--- a/net/minecraft/core/dispenser/DispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/DispenseItemBehavior.java
@@ -4,10 +_,8 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Consumer;
-import net.minecraft.Util;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
-import net.minecraft.core.Position;
 import net.minecraft.core.component.DataComponents;
 import net.minecraft.core.particles.ParticleTypes;
 import net.minecraft.server.level.ServerLevel;
@@ -15,7 +_,6 @@
 import net.minecraft.sounds.SoundSource;
 import net.minecraft.tags.BlockTags;
 import net.minecraft.tags.FluidTags;
-import net.minecraft.util.RandomSource;
 import net.minecraft.world.entity.EntitySelector;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.LivingEntity;
@@ -26,17 +_,6 @@
 import net.minecraft.world.entity.animal.horse.AbstractHorse;
 import net.minecraft.world.entity.decoration.ArmorStand;
 import net.minecraft.world.entity.item.PrimedTnt;
-import net.minecraft.world.entity.projectile.AbstractArrow;
-import net.minecraft.world.entity.projectile.Arrow;
-import net.minecraft.world.entity.projectile.FireworkRocketEntity;
-import net.minecraft.world.entity.projectile.Projectile;
-import net.minecraft.world.entity.projectile.SmallFireball;
-import net.minecraft.world.entity.projectile.Snowball;
-import net.minecraft.world.entity.projectile.SpectralArrow;
-import net.minecraft.world.entity.projectile.ThrownEgg;
-import net.minecraft.world.entity.projectile.ThrownExperienceBottle;
-import net.minecraft.world.entity.projectile.ThrownPotion;
-import net.minecraft.world.entity.projectile.windcharge.WindCharge;
 import net.minecraft.world.entity.vehicle.Boat;
 import net.minecraft.world.item.ArmorItem;
 import net.minecraft.world.item.BoneMealItem;
@@ -53,7 +_,6 @@
 import net.minecraft.world.level.LevelAccessor;
 import net.minecraft.world.level.block.BaseFireBlock;
 import net.minecraft.world.level.block.BeehiveBlock;
-import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.BucketPickup;
 import net.minecraft.world.level.block.CampfireBlock;
@@ -61,7 +_,6 @@
 import net.minecraft.world.level.block.CandleCakeBlock;
 import net.minecraft.world.level.block.CarvedPumpkinBlock;
 import net.minecraft.world.level.block.DispenserBlock;
-import net.minecraft.world.level.block.PotatoZombieHeadBlock;
 import net.minecraft.world.level.block.RespawnAnchorBlock;
 import net.minecraft.world.level.block.ShulkerBoxBlock;
 import net.minecraft.world.level.block.SkullBlock;
@@ -76,7 +_,6 @@
 import net.minecraft.world.level.block.state.properties.RotationSegment;
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.phys.AABB;
-import net.minecraft.world.phys.Vec3;
 import org.slf4j.Logger;
 
 public interface DispenseItemBehavior {
@@ -86,122 +_,39 @@
     ItemStack dispense(BlockSource p_302445_, ItemStack p_123404_);
 
     static void bootStrap() {
-        DispenserBlock.registerBehavior(Items.ARROW, new AbstractProjectileDispenseBehavior() {
-            @Override
-            protected Projectile getProjectile(Level p_123407_, Position p_123408_, ItemStack p_123409_) {
-                Arrow arrow = new Arrow(p_123407_, p_123408_.x(), p_123408_.y(), p_123408_.z(), p_123409_.copyWithCount(1));
-                arrow.pickup = AbstractArrow.Pickup.ALLOWED;
-                return arrow;
-            }
-        });
-        DispenserBlock.registerBehavior(Items.TIPPED_ARROW, new AbstractProjectileDispenseBehavior() {
-            @Override
-            protected Projectile getProjectile(Level p_123420_, Position p_123421_, ItemStack p_123422_) {
-                Arrow arrow = new Arrow(p_123420_, p_123421_.x(), p_123421_.y(), p_123421_.z(), p_123422_.copyWithCount(1));
-                arrow.pickup = AbstractArrow.Pickup.ALLOWED;
-                return arrow;
-            }
-        });
-        DispenserBlock.registerBehavior(Items.SPECTRAL_ARROW, new AbstractProjectileDispenseBehavior() {
-            @Override
-            protected Projectile getProjectile(Level p_123456_, Position p_123457_, ItemStack p_123458_) {
-                AbstractArrow abstractarrow = new SpectralArrow(p_123456_, p_123457_.x(), p_123457_.y(), p_123457_.z(), p_123458_.copyWithCount(1));
-                abstractarrow.pickup = AbstractArrow.Pickup.ALLOWED;
-                return abstractarrow;
-            }
-        });
-        DispenserBlock.registerBehavior(Items.EGG, new AbstractProjectileDispenseBehavior() {
-            @Override
-            protected Projectile getProjectile(Level p_123468_, Position p_123469_, ItemStack p_123470_) {
-                return Util.make(new ThrownEgg(p_123468_, p_123469_.x(), p_123469_.y(), p_123469_.z()), p_123466_ -> p_123466_.setItem(p_123470_));
-            }
-        });
-        DispenserBlock.registerBehavior(Items.SNOWBALL, new AbstractProjectileDispenseBehavior() {
-            @Override
-            protected Projectile getProjectile(Level p_123476_, Position p_123477_, ItemStack p_123478_) {
-                return Util.make(new Snowball(p_123476_, p_123477_.x(), p_123477_.y(), p_123477_.z()), p_123474_ -> p_123474_.setItem(p_123478_));
-            }
-        });
-        DispenserBlock.registerBehavior(Items.EXPERIENCE_BOTTLE, new AbstractProjectileDispenseBehavior() {
-            @Override
-            protected Projectile getProjectile(Level p_123485_, Position p_123486_, ItemStack p_123487_) {
-                return Util.make(new ThrownExperienceBottle(p_123485_, p_123486_.x(), p_123486_.y(), p_123486_.z()), p_123483_ -> p_123483_.setItem(p_123487_));
-            }
-
-            @Override
-            protected float getUncertainty() {
-                return super.getUncertainty() * 0.5F;
-            }
-
-            @Override
-            protected float getPower() {
-                return super.getPower() * 1.25F;
-            }
-        });
-        DispenserBlock.registerBehavior(Items.SPLASH_POTION, new DispenseItemBehavior() {
-            @Override
-            public ItemStack dispense(BlockSource p_302458_, ItemStack p_123492_) {
-                return (new AbstractProjectileDispenseBehavior() {
-                    @Override
-                    protected Projectile getProjectile(Level p_123501_, Position p_123502_, ItemStack p_123503_) {
-                        return Util.make(new ThrownPotion(p_123501_, p_123502_.x(), p_123502_.y(), p_123502_.z()), p_123499_ -> p_123499_.setItem(p_123503_));
-                    }
-
-                    @Override
-                    protected float getUncertainty() {
-                        return super.getUncertainty() * 0.5F;
-                    }
-
-                    @Override
-                    protected float getPower() {
-                        return super.getPower() * 1.25F;
-                    }
-                }).dispense(p_302458_, p_123492_);
-            }
-        });
-        DispenserBlock.registerBehavior(Items.LINGERING_POTION, new DispenseItemBehavior() {
-            @Override
-            public ItemStack dispense(BlockSource p_302461_, ItemStack p_123508_) {
-                return (new AbstractProjectileDispenseBehavior() {
-                    @Override
-                    protected Projectile getProjectile(Level p_123517_, Position p_123518_, ItemStack p_123519_) {
-                        return Util.make(new ThrownPotion(p_123517_, p_123518_.x(), p_123518_.y(), p_123518_.z()), p_123515_ -> p_123515_.setItem(p_123519_));
-                    }
-
-                    @Override
-                    protected float getUncertainty() {
-                        return super.getUncertainty() * 0.5F;
-                    }
-
-                    @Override
-                    protected float getPower() {
-                        return super.getPower() * 1.25F;
-                    }
-                }).dispense(p_302461_, p_123508_);
-            }
-        });
+        DispenserBlock.registerProjectileBehavior(Items.ARROW);
+        DispenserBlock.registerProjectileBehavior(Items.TIPPED_ARROW);
+        DispenserBlock.registerProjectileBehavior(Items.SPECTRAL_ARROW);
+        DispenserBlock.registerProjectileBehavior(Items.EGG);
+        DispenserBlock.registerProjectileBehavior(Items.SNOWBALL);
+        DispenserBlock.registerProjectileBehavior(Items.EXPERIENCE_BOTTLE);
+        DispenserBlock.registerProjectileBehavior(Items.SPLASH_POTION);
+        DispenserBlock.registerProjectileBehavior(Items.LINGERING_POTION);
+        DispenserBlock.registerProjectileBehavior(Items.FIREWORK_ROCKET);
+        DispenserBlock.registerProjectileBehavior(Items.FIRE_CHARGE);
+        DispenserBlock.registerProjectileBehavior(Items.WIND_CHARGE);
         DefaultDispenseItemBehavior defaultdispenseitembehavior = new DefaultDispenseItemBehavior() {
             @Override
-            public ItemStack execute(BlockSource p_302450_, ItemStack p_123524_) {
-                Direction direction = p_302450_.state().getValue(DispenserBlock.FACING);
-                EntityType<?> entitytype = ((SpawnEggItem)p_123524_.getItem()).getType(p_123524_);
+            public ItemStack execute(BlockSource p_338275_, ItemStack p_338216_) {
+                Direction direction = p_338275_.state().getValue(DispenserBlock.FACING);
+                EntityType<?> entitytype = ((SpawnEggItem)p_338216_.getItem()).getType(p_338216_);
 
                 try {
                     entitytype.spawn(
-                        p_302450_.level(), p_123524_, null, p_302450_.pos().relative(direction), MobSpawnType.DISPENSER, direction != Direction.UP, false
+                        p_338275_.level(), p_338216_, null, p_338275_.pos().relative(direction), MobSpawnType.DISPENSER, direction != Direction.UP, false
                     );
                 } catch (Exception exception) {
-                    LOGGER.error("Error while dispensing spawn egg from dispenser at {}", p_302450_.pos(), exception);
+                    LOGGER.error("Error while dispensing spawn egg from dispenser at {}", p_338275_.pos(), exception);
                     return ItemStack.EMPTY;
                 }
 
-                p_123524_.shrink(1);
-                p_302450_.level().gameEvent(null, GameEvent.ENTITY_PLACE, p_302450_.pos());
-                return p_123524_;
+                p_338216_.shrink(1);
+                p_338275_.level().gameEvent(null, GameEvent.ENTITY_PLACE, p_338275_.pos());
+                return p_338216_;
             }
         };
 
-        for(SpawnEggItem spawneggitem : SpawnEggItem.eggs()) {
+        for (SpawnEggItem spawneggitem : SpawnEggItem.eggs()) {
             DispenserBlock.registerBehavior(spawneggitem, defaultdispenseitembehavior);
         }
 
@@ -209,59 +_,60 @@
             Items.ARMOR_STAND,
             new DefaultDispenseItemBehavior() {
                 @Override
-                public ItemStack execute(BlockSource p_302430_, ItemStack p_123462_) {
-                    Direction direction = p_302430_.state().getValue(DispenserBlock.FACING);
-                    BlockPos blockpos = p_302430_.pos().relative(direction);
-                    ServerLevel serverlevel = p_302430_.level();
+                public ItemStack execute(BlockSource p_338813_, ItemStack p_338809_) {
+                    Direction direction = p_338813_.state().getValue(DispenserBlock.FACING);
+                    BlockPos blockpos = p_338813_.pos().relative(direction);
+                    ServerLevel serverlevel = p_338813_.level();
                     Consumer<ArmorStand> consumer = EntityType.appendDefaultStackConfig(
-                        p_293695_ -> p_293695_.setYRot(direction.toYRot()), serverlevel, p_123462_, null
+                        p_338707_ -> p_338707_.setYRot(direction.toYRot()), serverlevel, p_338809_, null
                     );
                     ArmorStand armorstand = EntityType.ARMOR_STAND.spawn(serverlevel, consumer, blockpos, MobSpawnType.DISPENSER, false, false);
                     if (armorstand != null) {
-                        p_123462_.shrink(1);
+                        p_338809_.shrink(1);
                     }
     
-                    return p_123462_;
+                    return p_338809_;
                 }
             }
         );
-        DispenserBlock.registerBehavior(Items.SADDLE, new OptionalDispenseItemBehavior() {
-            @Override
-            public ItemStack execute(BlockSource p_302463_, ItemStack p_123530_) {
-                BlockPos blockpos = p_302463_.pos().relative(p_302463_.state().getValue(DispenserBlock.FACING));
-                List<LivingEntity> list = p_302463_.level().getEntitiesOfClass(LivingEntity.class, new AABB(blockpos), p_123527_ -> {
-                    if (!(p_123527_ instanceof Saddleable)) {
-                        return false;
+        DispenserBlock.registerBehavior(
+            Items.SADDLE,
+            new OptionalDispenseItemBehavior() {
+                @Override
+                public ItemStack execute(BlockSource p_338360_, ItemStack p_338306_) {
+                    BlockPos blockpos = p_338360_.pos().relative(p_338360_.state().getValue(DispenserBlock.FACING));
+                    List<LivingEntity> list = p_338360_.level()
+                        .getEntitiesOfClass(
+                            LivingEntity.class,
+                            new AABB(blockpos),
+                            p_338882_ -> !(p_338882_ instanceof Saddleable saddleable) ? false : !saddleable.isSaddled() && saddleable.isSaddleable()
+                        );
+                    if (!list.isEmpty()) {
+                        ((Saddleable)list.get(0)).equipSaddle(SoundSource.BLOCKS);
+                        p_338306_.shrink(1);
+                        this.setSuccess(true);
+                        return p_338306_;
                     } else {
-                        Saddleable saddleable = (Saddleable)p_123527_;
-                        return !saddleable.isSaddled() && saddleable.isSaddleable();
+                        return super.execute(p_338360_, p_338306_);
                     }
-                });
-                if (!list.isEmpty()) {
-                    ((Saddleable)list.get(0)).equipSaddle(SoundSource.BLOCKS);
-                    p_123530_.shrink(1);
-                    this.setSuccess(true);
-                    return p_123530_;
-                } else {
-                    return super.execute(p_302463_, p_123530_);
                 }
             }
-        });
+        );
         DefaultDispenseItemBehavior defaultdispenseitembehavior1 = new OptionalDispenseItemBehavior() {
             @Override
-            protected ItemStack execute(BlockSource p_302425_, ItemStack p_123536_) {
-                BlockPos blockpos = p_302425_.pos().relative(p_302425_.state().getValue(DispenserBlock.FACING));
+            protected ItemStack execute(BlockSource p_338193_, ItemStack p_338600_) {
+                BlockPos blockpos = p_338193_.pos().relative(p_338193_.state().getValue(DispenserBlock.FACING));
 
-                for(AbstractHorse abstracthorse : p_302425_.level()
-                    .getEntitiesOfClass(AbstractHorse.class, new AABB(blockpos), p_339557_ -> p_339557_.isAlive() && p_339557_.canWearBodyArmor())) {
-                    if (abstracthorse.isBodyArmorItem(p_123536_) && !abstracthorse.isWearingBodyArmor() && abstracthorse.isTamed()) {
-                        abstracthorse.setBodyArmorItem(p_123536_.split(1));
+                for (AbstractHorse abstracthorse : p_338193_.level()
+                    .getEntitiesOfClass(AbstractHorse.class, new AABB(blockpos), p_337459_ -> p_337459_.isAlive() && p_337459_.canWearBodyArmor())) {
+                    if (abstracthorse.isBodyArmorItem(p_338600_) && !abstracthorse.isWearingBodyArmor() && abstracthorse.isTamed()) {
+                        abstracthorse.setBodyArmorItem(p_338600_.split(1));
                         this.setSuccess(true);
-                        return p_123536_;
+                        return p_338600_;
                     }
                 }
 
-                return super.execute(p_302425_, p_123536_);
+                return super.execute(p_338193_, p_338600_);
             }
         };
         DispenserBlock.registerBehavior(Items.LEATHER_HORSE_ARMOR, defaultdispenseitembehavior1);
@@ -288,87 +_,19 @@
             Items.CHEST,
             new OptionalDispenseItemBehavior() {
                 @Override
-                public ItemStack execute(BlockSource p_302452_, ItemStack p_123542_) {
-                    BlockPos blockpos = p_302452_.pos().relative(p_302452_.state().getValue(DispenserBlock.FACING));
+                public ItemStack execute(BlockSource p_338297_, ItemStack p_338735_) {
+                    BlockPos blockpos = p_338297_.pos().relative(p_338297_.state().getValue(DispenserBlock.FACING));
     
-                    for(AbstractChestedHorse abstractchestedhorse : p_302452_.level()
-                        .getEntitiesOfClass(AbstractChestedHorse.class, new AABB(blockpos), p_339558_ -> p_339558_.isAlive() && !p_339558_.hasChest())) {
-                        if (abstractchestedhorse.isTamed() && abstractchestedhorse.getSlot(499).set(p_123542_)) {
-                            p_123542_.shrink(1);
+                    for (AbstractChestedHorse abstractchestedhorse : p_338297_.level()
+                        .getEntitiesOfClass(AbstractChestedHorse.class, new AABB(blockpos), p_337460_ -> p_337460_.isAlive() && !p_337460_.hasChest())) {
+                        if (abstractchestedhorse.isTamed() && abstractchestedhorse.getSlot(499).set(p_338735_)) {
+                            p_338735_.shrink(1);
                             this.setSuccess(true);
-                            return p_123542_;
+                            return p_338735_;
                         }
                     }
     
-                    return super.execute(p_302452_, p_123542_);
-                }
-            }
-        );
-        DispenserBlock.registerBehavior(Items.FIREWORK_ROCKET, new DefaultDispenseItemBehavior() {
-            @Override
-            public ItemStack execute(BlockSource p_302433_, ItemStack p_123548_) {
-                Direction direction = p_302433_.state().getValue(DispenserBlock.FACING);
-                Vec3 vec3 = DispenseItemBehavior.getEntityPokingOutOfBlockPos(p_302433_, EntityType.FIREWORK_ROCKET, direction);
-                FireworkRocketEntity fireworkrocketentity = new FireworkRocketEntity(p_302433_.level(), p_123548_, vec3.x(), vec3.y(), vec3.z(), true);
-                fireworkrocketentity.shoot((double)direction.getStepX(), (double)direction.getStepY(), (double)direction.getStepZ(), 0.5F, 1.0F);
-                p_302433_.level().addFreshEntity(fireworkrocketentity);
-                p_123548_.shrink(1);
-                return p_123548_;
-            }
-
-            @Override
-            protected void playSound(BlockSource p_302457_) {
-                p_302457_.level().levelEvent(1004, p_302457_.pos(), 0);
-            }
-        });
-        DispenserBlock.registerBehavior(Items.FIRE_CHARGE, new DefaultDispenseItemBehavior() {
-            @Override
-            public ItemStack execute(BlockSource p_302423_, ItemStack p_123557_) {
-                Direction direction = p_302423_.state().getValue(DispenserBlock.FACING);
-                Position position = DispenserBlock.getDispensePosition(p_302423_);
-                double d0 = position.x() + (double)((float)direction.getStepX() * 0.3F);
-                double d1 = position.y() + (double)((float)direction.getStepY() * 0.3F);
-                double d2 = position.z() + (double)((float)direction.getStepZ() * 0.3F);
-                Level level = p_302423_.level();
-                RandomSource randomsource = level.random;
-                double d3 = randomsource.triangle((double)direction.getStepX(), 0.11485000000000001);
-                double d4 = randomsource.triangle((double)direction.getStepY(), 0.11485000000000001);
-                double d5 = randomsource.triangle((double)direction.getStepZ(), 0.11485000000000001);
-                SmallFireball smallfireball = new SmallFireball(level, d0, d1, d2, d3, d4, d5);
-                level.addFreshEntity(Util.make(smallfireball, p_123552_ -> p_123552_.setItem(p_123557_)));
-                p_123557_.shrink(1);
-                return p_123557_;
-            }
-
-            @Override
-            protected void playSound(BlockSource p_302436_) {
-                p_302436_.level().levelEvent(1018, p_302436_.pos(), 0);
-            }
-        });
-        DispenserBlock.registerBehavior(
-            Items.WIND_CHARGE,
-            new DefaultDispenseItemBehavior() {
-                @Override
-                public ItemStack execute(BlockSource p_302435_, ItemStack p_123562_) {
-                    Direction direction = p_302435_.state().getValue(DispenserBlock.FACING);
-                    Position position = DispenserBlock.getDispensePosition(p_302435_);
-                    Level level = p_302435_.level();
-                    RandomSource randomsource = level.random;
-                    double d0 = randomsource.triangle((double)direction.getStepX(), 0.11485000000000001);
-                    double d1 = randomsource.triangle((double)direction.getStepY(), 0.11485000000000001);
-                    double d2 = randomsource.triangle((double)direction.getStepZ(), 0.11485000000000001);
-                    WindCharge windcharge = new WindCharge(
-                        level,
-                        position.x() + (double)((float)direction.getStepX() * 0.3F),
-                        position.y() + (double)((float)direction.getStepY() * 0.3F),
-                        position.z() + (double)((float)direction.getStepZ() * 0.3F),
-                        d0,
-                        d1,
-                        d2
-                    );
-                    level.addFreshEntity(windcharge);
-                    p_123562_.shrink(1);
-                    return p_123562_;
+                    return super.execute(p_338297_, p_338735_);
                 }
             }
         );
@@ -394,15 +_,15 @@
             private final DefaultDispenseItemBehavior defaultDispenseItemBehavior = new DefaultDispenseItemBehavior();
 
             @Override
-            public ItemStack execute(BlockSource p_302464_, ItemStack p_123567_) {
-                DispensibleContainerItem dispensiblecontaineritem = (DispensibleContainerItem)p_123567_.getItem();
-                BlockPos blockpos = p_302464_.pos().relative(p_302464_.state().getValue(DispenserBlock.FACING));
-                Level level = p_302464_.level();
-                if (dispensiblecontaineritem.emptyContents(null, level, blockpos, null)) {
-                    dispensiblecontaineritem.checkExtraContent(null, level, p_123567_, blockpos);
+            public ItemStack execute(BlockSource p_338850_, ItemStack p_338251_) {
+                DispensibleContainerItem dispensiblecontaineritem = (DispensibleContainerItem)p_338251_.getItem();
+                BlockPos blockpos = p_338850_.pos().relative(p_338850_.state().getValue(DispenserBlock.FACING));
+                Level level = p_338850_.level();
+                if (dispensiblecontaineritem.emptyContents(null, level, blockpos, null, p_338251_)) {
+                    dispensiblecontaineritem.checkExtraContent(null, level, p_338251_, blockpos);
                     return new ItemStack(Items.BUCKET);
                 } else {
-                    return this.defaultDispenseItemBehavior.dispense(p_302464_, p_123567_);
+                    return this.defaultDispenseItemBehavior.dispense(p_338850_, p_338251_);
                 }
             }
         };
@@ -419,41 +_,40 @@
             private final DefaultDispenseItemBehavior defaultDispenseItemBehavior = new DefaultDispenseItemBehavior();
 
             @Override
-            public ItemStack execute(BlockSource p_302453_, ItemStack p_123413_) {
-                LevelAccessor levelaccessor = p_302453_.level();
-                BlockPos blockpos = p_302453_.pos().relative(p_302453_.state().getValue(DispenserBlock.FACING));
+            public ItemStack execute(BlockSource p_338386_, ItemStack p_338526_) {
+                LevelAccessor levelaccessor = p_338386_.level();
+                BlockPos blockpos = p_338386_.pos().relative(p_338386_.state().getValue(DispenserBlock.FACING));
                 BlockState blockstate = levelaccessor.getBlockState(blockpos);
-                Block block = blockstate.getBlock();
-                if (block instanceof BucketPickup bucketpickup) {
+                if (blockstate.getBlock() instanceof BucketPickup bucketpickup) {
                     ItemStack itemstack = bucketpickup.pickupBlock(null, levelaccessor, blockpos, blockstate);
                     if (itemstack.isEmpty()) {
-                        return super.execute(p_302453_, p_123413_);
+                        return super.execute(p_338386_, p_338526_);
                     } else {
                         levelaccessor.gameEvent(null, GameEvent.FLUID_PICKUP, blockpos);
                         Item item = itemstack.getItem();
-                        p_123413_.shrink(1);
-                        if (p_123413_.isEmpty()) {
+                        p_338526_.shrink(1);
+                        if (p_338526_.isEmpty()) {
                             return new ItemStack(item);
                         } else {
-                            if (p_302453_.blockEntity().addItem(new ItemStack(item)) < 0) {
-                                this.defaultDispenseItemBehavior.dispense(p_302453_, new ItemStack(item));
+                            if (p_338386_.blockEntity().addItem(new ItemStack(item)) < 0) {
+                                this.defaultDispenseItemBehavior.dispense(p_338386_, new ItemStack(item));
                             }
 
-                            return p_123413_;
+                            return p_338526_;
                         }
                     }
                 } else {
-                    return super.execute(p_302453_, p_123413_);
+                    return super.execute(p_338386_, p_338526_);
                 }
             }
         });
         DispenserBlock.registerBehavior(Items.FLINT_AND_STEEL, new OptionalDispenseItemBehavior() {
             @Override
-            protected ItemStack execute(BlockSource p_302429_, ItemStack p_123417_) {
-                Level level = p_302429_.level();
+            protected ItemStack execute(BlockSource p_338494_, ItemStack p_338444_) {
+                Level level = p_338494_.level();
                 this.setSuccess(true);
-                Direction direction = p_302429_.state().getValue(DispenserBlock.FACING);
-                BlockPos blockpos = p_302429_.pos().relative(direction);
+                Direction direction = p_338494_.state().getValue(DispenserBlock.FACING);
+                BlockPos blockpos = p_338494_.pos().relative(direction);
                 BlockState blockstate = level.getBlockState(blockpos);
                 if (BaseFireBlock.canBePlacedAt(level, blockpos, direction)) {
                     level.setBlockAndUpdate(blockpos, BaseFireBlock.getState(level, blockpos));
@@ -461,53 +_,54 @@
                 } else if (CampfireBlock.canLight(blockstate) || CandleBlock.canLight(blockstate) || CandleCakeBlock.canLight(blockstate)) {
                     level.setBlockAndUpdate(blockpos, blockstate.setValue(BlockStateProperties.LIT, Boolean.valueOf(true)));
                     level.gameEvent(null, GameEvent.BLOCK_CHANGE, blockpos);
-                } else if (blockstate.getBlock() instanceof TntBlock) {
-                    TntBlock.explode(level, blockpos);
-                    level.removeBlock(blockpos, false);
+                } else if (blockstate.isFlammable(level, blockpos, p_338494_.state().getValue(DispenserBlock.FACING).getOpposite())) {
+                    blockstate.onCaughtFire(level, blockpos, p_338494_.state().getValue(DispenserBlock.FACING).getOpposite(), null);
+                    if (blockstate.getBlock() instanceof TntBlock)
+                        level.removeBlock(blockpos, false);
                 } else {
                     this.setSuccess(false);
                 }
 
                 if (this.isSuccess()) {
-                    p_123417_.hurtAndBreak(1, level.getRandom(), null, () -> p_123417_.setCount(0));
+                    p_338444_.hurtAndBreak(1, level.getRandom(), null, () -> p_338444_.setCount(0));
                 }
 
-                return p_123417_;
+                return p_338444_;
             }
         });
         DispenserBlock.registerBehavior(Items.BONE_MEAL, new OptionalDispenseItemBehavior() {
             @Override
-            protected ItemStack execute(BlockSource p_302466_, ItemStack p_123426_) {
+            protected ItemStack execute(BlockSource p_302450_, ItemStack p_123524_) {
                 this.setSuccess(true);
-                Level level = p_302466_.level();
-                BlockPos blockpos = p_302466_.pos().relative(p_302466_.state().getValue(DispenserBlock.FACING));
-                if (!BoneMealItem.growCrop(p_123426_, level, blockpos) && !BoneMealItem.growWaterPlant(p_123426_, level, blockpos, null)) {
+                Level level = p_302450_.level();
+                BlockPos blockpos = p_302450_.pos().relative(p_302450_.state().getValue(DispenserBlock.FACING));
+                if (!BoneMealItem.growCrop(p_123524_, level, blockpos) && !BoneMealItem.growWaterPlant(p_123524_, level, blockpos, null)) {
                     this.setSuccess(false);
                 } else if (!level.isClientSide) {
                     level.levelEvent(1505, blockpos, 15);
                 }
 
-                return p_123426_;
+                return p_123524_;
             }
         });
         DispenserBlock.registerBehavior(Blocks.TNT, new DefaultDispenseItemBehavior() {
             @Override
-            protected ItemStack execute(BlockSource p_302449_, ItemStack p_123430_) {
-                Level level = p_302449_.level();
-                BlockPos blockpos = p_302449_.pos().relative(p_302449_.state().getValue(DispenserBlock.FACING));
+            protected ItemStack execute(BlockSource p_302430_, ItemStack p_123462_) {
+                Level level = p_302430_.level();
+                BlockPos blockpos = p_302430_.pos().relative(p_302430_.state().getValue(DispenserBlock.FACING));
                 PrimedTnt primedtnt = new PrimedTnt(level, (double)blockpos.getX() + 0.5, (double)blockpos.getY(), (double)blockpos.getZ() + 0.5, null);
                 level.addFreshEntity(primedtnt);
                 level.playSound(null, primedtnt.getX(), primedtnt.getY(), primedtnt.getZ(), SoundEvents.TNT_PRIMED, SoundSource.BLOCKS, 1.0F, 1.0F);
                 level.gameEvent(null, GameEvent.ENTITY_PLACE, blockpos);
-                p_123430_.shrink(1);
-                return p_123430_;
+                p_123462_.shrink(1);
+                return p_123462_;
             }
         });
         DispenseItemBehavior dispenseitembehavior = new OptionalDispenseItemBehavior() {
             @Override
-            protected ItemStack execute(BlockSource p_302428_, ItemStack p_123434_) {
-                this.setSuccess(ArmorItem.dispenseArmor(p_302428_, p_123434_));
-                return p_123434_;
+            protected ItemStack execute(BlockSource p_302463_, ItemStack p_123530_) {
+                this.setSuccess(ArmorItem.dispenseArmor(p_302463_, p_123530_));
+                return p_123530_;
             }
         };
         DispenserBlock.registerBehavior(Items.CREEPER_HEAD, dispenseitembehavior);
@@ -520,11 +_,11 @@
             Items.WITHER_SKELETON_SKULL,
             new OptionalDispenseItemBehavior() {
                 @Override
-                protected ItemStack execute(BlockSource p_302468_, ItemStack p_123438_) {
-                    Level level = p_302468_.level();
-                    Direction direction = p_302468_.state().getValue(DispenserBlock.FACING);
-                    BlockPos blockpos = p_302468_.pos().relative(direction);
-                    if (level.isEmptyBlock(blockpos) && WitherSkullBlock.canSpawnMob(level, blockpos, p_123438_)) {
+                protected ItemStack execute(BlockSource p_302425_, ItemStack p_123536_) {
+                    Level level = p_302425_.level();
+                    Direction direction = p_302425_.state().getValue(DispenserBlock.FACING);
+                    BlockPos blockpos = p_302425_.pos().relative(direction);
+                    if (level.isEmptyBlock(blockpos) && WitherSkullBlock.canSpawnMob(level, blockpos, p_123536_)) {
                         level.setBlock(
                             blockpos,
                             Blocks.WITHER_SKELETON_SKULL
@@ -538,21 +_,21 @@
                             WitherSkullBlock.checkSpawn(level, blockpos, (SkullBlockEntity)blockentity);
                         }
     
-                        p_123438_.shrink(1);
+                        p_123536_.shrink(1);
                         this.setSuccess(true);
                     } else {
-                        this.setSuccess(ArmorItem.dispenseArmor(p_302468_, p_123438_));
+                        this.setSuccess(ArmorItem.dispenseArmor(p_302425_, p_123536_));
                     }
     
-                    return p_123438_;
+                    return p_123536_;
                 }
             }
         );
         DispenserBlock.registerBehavior(Blocks.CARVED_PUMPKIN, new OptionalDispenseItemBehavior() {
             @Override
-            protected ItemStack execute(BlockSource p_302437_, ItemStack p_123445_) {
-                Level level = p_302437_.level();
-                BlockPos blockpos = p_302437_.pos().relative(p_302437_.state().getValue(DispenserBlock.FACING));
+            protected ItemStack execute(BlockSource p_302452_, ItemStack p_123542_) {
+                Level level = p_302452_.level();
+                BlockPos blockpos = p_302452_.pos().relative(p_302452_.state().getValue(DispenserBlock.FACING));
                 CarvedPumpkinBlock carvedpumpkinblock = (CarvedPumpkinBlock)Blocks.CARVED_PUMPKIN;
                 if (level.isEmptyBlock(blockpos) && carvedpumpkinblock.canSpawnGolem(level, blockpos)) {
                     if (!level.isClientSide) {
@@ -560,28 +_,18 @@
                         level.gameEvent(null, GameEvent.BLOCK_PLACE, blockpos);
                     }
 
-                    p_123445_.shrink(1);
+                    p_123542_.shrink(1);
                     this.setSuccess(true);
                 } else {
-                    this.setSuccess(ArmorItem.dispenseArmor(p_302437_, p_123445_));
+                    this.setSuccess(ArmorItem.dispenseArmor(p_302452_, p_123542_));
                 }
 
-                return p_123445_;
-            }
-        });
-        DispenserBlock.registerBehavior(Blocks.POTATO_ZOMBIE_HEAD_HAT, new OptionalDispenseItemBehavior() {
-            @Override
-            protected ItemStack execute(BlockSource p_302454_, ItemStack p_123453_) {
-                Level level = p_302454_.level();
-                BlockPos blockpos = p_302454_.pos().relative(p_302454_.state().getValue(DispenserBlock.FACING));
-                PotatoZombieHeadBlock potatozombieheadblock = (PotatoZombieHeadBlock)Blocks.POTATO_ZOMBIE_HEAD_HAT;
-                this.setSuccess(ArmorItem.dispenseArmor(p_302454_, p_123453_));
-                return p_123453_;
+                return p_123542_;
             }
         });
         DispenserBlock.registerBehavior(Blocks.SHULKER_BOX.asItem(), new ShulkerBoxDispenseBehavior());
 
-        for(DyeColor dyecolor : DyeColor.values()) {
+        for (DyeColor dyecolor : DyeColor.values()) {
             DispenserBlock.registerBehavior(ShulkerBoxBlock.getBlockByColor(dyecolor).asItem(), new ShulkerBoxDispenseBehavior());
         }
 
@@ -590,106 +_,106 @@
             new OptionalDispenseItemBehavior() {
                 private final DefaultDispenseItemBehavior defaultDispenseItemBehavior = new DefaultDispenseItemBehavior();
     
-                private ItemStack takeLiquid(BlockSource p_341049_, ItemStack p_340757_, ItemStack p_341735_) {
-                    p_340757_.shrink(1);
-                    if (p_340757_.isEmpty()) {
-                        p_341049_.level().gameEvent(null, GameEvent.FLUID_PICKUP, p_341049_.pos());
-                        return p_341735_.copy();
+                private ItemStack takeLiquid(BlockSource p_338528_, ItemStack p_338340_, ItemStack p_338879_) {
+                    p_338340_.shrink(1);
+                    if (p_338340_.isEmpty()) {
+                        p_338528_.level().gameEvent(null, GameEvent.FLUID_PICKUP, p_338528_.pos());
+                        return p_338879_.copy();
                     } else {
-                        if (p_341049_.blockEntity().addItem(p_341735_.copy()) < 0) {
-                            this.defaultDispenseItemBehavior.dispense(p_341049_, p_341735_.copy());
+                        if (p_338528_.blockEntity().addItem(p_338879_.copy()) < 0) {
+                            this.defaultDispenseItemBehavior.dispense(p_338528_, p_338879_.copy());
                         }
     
-                        return p_340757_;
+                        return p_338340_;
                     }
                 }
     
                 @Override
-                public ItemStack execute(BlockSource p_302447_, ItemStack p_175748_) {
+                public ItemStack execute(BlockSource p_302433_, ItemStack p_123548_) {
                     this.setSuccess(false);
-                    ServerLevel serverlevel = p_302447_.level();
-                    BlockPos blockpos = p_302447_.pos().relative(p_302447_.state().getValue(DispenserBlock.FACING));
+                    ServerLevel serverlevel = p_302433_.level();
+                    BlockPos blockpos = p_302433_.pos().relative(p_302433_.state().getValue(DispenserBlock.FACING));
                     BlockState blockstate = serverlevel.getBlockState(blockpos);
                     if (blockstate.is(
-                            BlockTags.BEEHIVES, p_341308_ -> p_341308_.hasProperty(BeehiveBlock.HONEY_LEVEL) && p_341308_.getBlock() instanceof BeehiveBlock
+                            BlockTags.BEEHIVES, p_338544_ -> p_338544_.hasProperty(BeehiveBlock.HONEY_LEVEL) && p_338544_.getBlock() instanceof BeehiveBlock
                         )
                         && blockstate.getValue(BeehiveBlock.HONEY_LEVEL) >= 5) {
                         ((BeehiveBlock)blockstate.getBlock())
                             .releaseBeesAndResetHoneyLevel(serverlevel, blockstate, blockpos, null, BeehiveBlockEntity.BeeReleaseStatus.BEE_RELEASED);
                         this.setSuccess(true);
-                        return this.takeLiquid(p_302447_, p_175748_, new ItemStack(Items.HONEY_BOTTLE));
+                        return this.takeLiquid(p_302433_, p_123548_, new ItemStack(Items.HONEY_BOTTLE));
                     } else if (serverlevel.getFluidState(blockpos).is(FluidTags.WATER)) {
                         this.setSuccess(true);
-                        return this.takeLiquid(p_302447_, p_175748_, PotionContents.createItemStack(Items.POTION, Potions.WATER));
+                        return this.takeLiquid(p_302433_, p_123548_, PotionContents.createItemStack(Items.POTION, Potions.WATER));
                     } else {
-                        return super.execute(p_302447_, p_175748_);
+                        return super.execute(p_302433_, p_123548_);
                     }
                 }
             }
         );
         DispenserBlock.registerBehavior(Items.GLOWSTONE, new OptionalDispenseItemBehavior() {
             @Override
-            public ItemStack execute(BlockSource p_302442_, ItemStack p_235897_) {
-                Direction direction = p_302442_.state().getValue(DispenserBlock.FACING);
-                BlockPos blockpos = p_302442_.pos().relative(direction);
-                Level level = p_302442_.level();
+            public ItemStack execute(BlockSource p_302423_, ItemStack p_123557_) {
+                Direction direction = p_302423_.state().getValue(DispenserBlock.FACING);
+                BlockPos blockpos = p_302423_.pos().relative(direction);
+                Level level = p_302423_.level();
                 BlockState blockstate = level.getBlockState(blockpos);
                 this.setSuccess(true);
                 if (blockstate.is(Blocks.RESPAWN_ANCHOR)) {
                     if (blockstate.getValue(RespawnAnchorBlock.CHARGE) != 4) {
                         RespawnAnchorBlock.charge(null, level, blockpos, blockstate);
-                        p_235897_.shrink(1);
+                        p_123557_.shrink(1);
                     } else {
                         this.setSuccess(false);
                     }
 
-                    return p_235897_;
+                    return p_123557_;
                 } else {
-                    return super.execute(p_302442_, p_235897_);
+                    return super.execute(p_302423_, p_123557_);
                 }
             }
         });
         DispenserBlock.registerBehavior(Items.SHEARS.asItem(), new ShearsDispenseItemBehavior());
         DispenserBlock.registerBehavior(Items.BRUSH.asItem(), new OptionalDispenseItemBehavior() {
             @Override
-            protected ItemStack execute(BlockSource p_316451_, ItemStack p_316386_) {
-                ServerLevel serverlevel = p_316451_.level();
-                BlockPos blockpos = p_316451_.pos().relative(p_316451_.state().getValue(DispenserBlock.FACING));
+            protected ItemStack execute(BlockSource p_302435_, ItemStack p_123562_) {
+                ServerLevel serverlevel = p_302435_.level();
+                BlockPos blockpos = p_302435_.pos().relative(p_302435_.state().getValue(DispenserBlock.FACING));
                 List<Armadillo> list = serverlevel.getEntitiesOfClass(Armadillo.class, new AABB(blockpos), EntitySelector.NO_SPECTATORS);
                 if (list.isEmpty()) {
                     this.setSuccess(false);
-                    return p_316386_;
+                    return p_123562_;
                 } else {
-                    for(Armadillo armadillo : list) {
+                    for (Armadillo armadillo : list) {
                         if (armadillo.brushOffScute()) {
-                            p_316386_.hurtAndBreak(16, serverlevel.getRandom(), null, () -> {
-                                p_316386_.shrink(1);
-                                p_316386_.setDamageValue(0);
+                            p_123562_.hurtAndBreak(16, serverlevel.getRandom(), null, () -> {
+                                p_123562_.shrink(1);
+                                p_123562_.setDamageValue(0);
                             });
-                            return p_316386_;
+                            return p_123562_;
                         }
                     }
 
                     this.setSuccess(false);
-                    return p_316386_;
+                    return p_123562_;
                 }
             }
         });
         DispenserBlock.registerBehavior(Items.HONEYCOMB, new OptionalDispenseItemBehavior() {
             @Override
-            public ItemStack execute(BlockSource p_326393_, ItemStack p_326473_) {
-                BlockPos blockpos = p_326393_.pos().relative(p_326393_.state().getValue(DispenserBlock.FACING));
-                Level level = p_326393_.level();
+            public ItemStack execute(BlockSource p_302464_, ItemStack p_123567_) {
+                BlockPos blockpos = p_302464_.pos().relative(p_302464_.state().getValue(DispenserBlock.FACING));
+                Level level = p_302464_.level();
                 BlockState blockstate = level.getBlockState(blockpos);
                 Optional<BlockState> optional = HoneycombItem.getWaxed(blockstate);
                 if (optional.isPresent()) {
                     level.setBlockAndUpdate(blockpos, optional.get());
                     level.levelEvent(3003, blockpos, 0);
-                    p_326473_.shrink(1);
+                    p_123567_.shrink(1);
                     this.setSuccess(true);
-                    return p_326473_;
+                    return p_123567_;
                 } else {
-                    return super.execute(p_326393_, p_326473_);
+                    return super.execute(p_302464_, p_123567_);
                 }
             }
         });
@@ -699,19 +_,19 @@
                 private final DefaultDispenseItemBehavior defaultDispenseItemBehavior = new DefaultDispenseItemBehavior();
     
                 @Override
-                public ItemStack execute(BlockSource p_341895_, ItemStack p_341771_) {
-                    PotionContents potioncontents = p_341771_.getOrDefault(DataComponents.POTION_CONTENTS, PotionContents.EMPTY);
+                public ItemStack execute(BlockSource p_302453_, ItemStack p_123413_) {
+                    PotionContents potioncontents = p_123413_.getOrDefault(DataComponents.POTION_CONTENTS, PotionContents.EMPTY);
                     if (!potioncontents.is(Potions.WATER)) {
-                        return this.defaultDispenseItemBehavior.dispense(p_341895_, p_341771_);
+                        return this.defaultDispenseItemBehavior.dispense(p_302453_, p_123413_);
                     } else {
-                        ServerLevel serverlevel = p_341895_.level();
-                        BlockPos blockpos = p_341895_.pos();
-                        BlockPos blockpos1 = p_341895_.pos().relative(p_341895_.state().getValue(DispenserBlock.FACING));
+                        ServerLevel serverlevel = p_302453_.level();
+                        BlockPos blockpos = p_302453_.pos();
+                        BlockPos blockpos1 = p_302453_.pos().relative(p_302453_.state().getValue(DispenserBlock.FACING));
                         if (!serverlevel.getBlockState(blockpos1).is(BlockTags.CONVERTABLE_TO_MUD)) {
-                            return this.defaultDispenseItemBehavior.dispense(p_341895_, p_341771_);
+                            return this.defaultDispenseItemBehavior.dispense(p_302453_, p_123413_);
                         } else {
                             if (!serverlevel.isClientSide) {
-                                for(int i = 0; i < 5; ++i) {
+                                for (int i = 0; i < 5; i++) {
                                     serverlevel.sendParticles(
                                         ParticleTypes.SPLASH,
                                         (double)blockpos.getX() + serverlevel.random.nextDouble(),
@@ -735,14 +_,5 @@
                 }
             }
         );
-    }
-
-    static Vec3 getEntityPokingOutOfBlockPos(BlockSource p_302467_, EntityType<?> p_302422_, Direction p_302439_) {
-        return p_302467_.center()
-            .add(
-                (double)p_302439_.getStepX() * (0.5000099999997474 - (double)p_302422_.getWidth() / 2.0),
-                (double)p_302439_.getStepY() * (0.5000099999997474 - (double)p_302422_.getHeight() / 2.0) - (double)p_302422_.getHeight() / 2.0,
-                (double)p_302439_.getStepZ() * (0.5000099999997474 - (double)p_302422_.getWidth() / 2.0)
-            );
     }
 }
