--- a/net/minecraft/core/dispenser/MinecartDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/MinecartDispenseItemBehavior.java
@@ -34,7 +_,7 @@
         BlockState blockstate = serverlevel.getBlockState(blockpos);
         double d3;
         if (blockstate.is(BlockTags.RAILS)) {
-            if (getRailShape(blockstate).isSlope()) {
+            if (getRailShape(blockstate, serverlevel, blockpos).isSlope()) {
                 d3 = 0.6;
             } else {
                 d3 = 0.1;
@@ -49,7 +_,7 @@
                 return this.defaultDispenseItemBehavior.dispense(source, dispensed);
             }
 
-            if (direction != Direction.DOWN && getRailShape(blockstate1).isSlope()) {
+            if (direction != Direction.DOWN && getRailShape(blockstate1, serverlevel, blockpos.below()).isSlope()) {
                 d3 = -0.4;
             } else {
                 d3 = -0.9;
@@ -68,8 +_,8 @@
         return dispensed;
     }
 
-    private static RailShape getRailShape(BlockState blockFront) {
-        return blockFront.getBlock() instanceof BaseRailBlock baserailblock ? blockFront.getValue(baserailblock.getShapeProperty()) : RailShape.NORTH_SOUTH;
+    private static RailShape getRailShape(BlockState blockFront, ServerLevel level, BlockPos pos) {
+        return blockFront.getBlock() instanceof BaseRailBlock baserailblock ? baserailblock.getRailDirection(blockFront, level, pos, null) : RailShape.NORTH_SOUTH;
     }
 
     @Override
