--- a/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
@@ -1,15 +_,13 @@
 package net.minecraft.world.level.levelgen.heightproviders;
 
-import com.mojang.serialization.Codec;
-import net.minecraft.util.ExtraCodecs;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.levelgen.VerticalAnchor;
 import net.minecraft.world.level.levelgen.WorldGenerationContext;
 
 public class ConstantHeight extends HeightProvider {
     public static final ConstantHeight ZERO = new ConstantHeight(VerticalAnchor.absolute(0));
-    public static final Codec<ConstantHeight> CODEC = ExtraCodecs.withAlternative(VerticalAnchor.CODEC, VerticalAnchor.CODEC.fieldOf("value").codec())
-        .xmap(ConstantHeight::new, ConstantHeight::getValue);
+    public static final MapCodec<ConstantHeight> CODEC = VerticalAnchor.CODEC.fieldOf("value").xmap(ConstantHeight::new, ConstantHeight::getValue);
     private final VerticalAnchor value;
 
     public static ConstantHeight of(VerticalAnchor p_161957_) {
