--- a/net/minecraft/world/level/levelgen/heightproviders/HeightProvider.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/HeightProvider.java
@@ -12,7 +_,7 @@
         VerticalAnchor.CODEC, BuiltInRegistries.HEIGHT_PROVIDER_TYPE.byNameCodec().dispatch(HeightProvider::getType, HeightProviderType::codec)
     );
     public static final Codec<HeightProvider> CODEC = CONSTANT_OR_DISPATCH_CODEC.xmap(
-        p_161974_ -> p_161974_.map(ConstantHeight::of, p_161980_ -> p_161980_),
+        p_161974_ -> p_161974_.map(ConstantHeight::of, p_161980_ -> (HeightProvider)p_161980_),
         p_161976_ -> p_161976_.getType() == HeightProviderType.CONSTANT ? Either.left(((ConstantHeight)p_161976_).getValue()) : Either.right(p_161976_)
     );
 
