--- a/net/minecraft/world/level/levelgen/heightproviders/TrapezoidHeight.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/TrapezoidHeight.java
@@ -2,6 +_,7 @@
 
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.util.Mth;
@@ -11,7 +_,7 @@
 import org.slf4j.Logger;
 
 public class TrapezoidHeight extends HeightProvider {
-    public static final Codec<TrapezoidHeight> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<TrapezoidHeight> CODEC = RecordCodecBuilder.mapCodec(
         p_162005_ -> p_162005_.group(
                     VerticalAnchor.CODEC.fieldOf("min_inclusive").forGetter(p_162021_ -> p_162021_.minInclusive),
                     VerticalAnchor.CODEC.fieldOf("max_inclusive").forGetter(p_162019_ -> p_162019_.maxInclusive),
