--- a/net/minecraft/world/level/levelgen/heightproviders/UniformHeight.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/UniformHeight.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.levelgen.heightproviders;
 
 import com.mojang.logging.LogUtils;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
@@ -13,7 +_,7 @@
 import org.slf4j.Logger;
 
 public class UniformHeight extends HeightProvider {
-    public static final Codec<UniformHeight> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<UniformHeight> CODEC = RecordCodecBuilder.mapCodec(
         p_162033_ -> p_162033_.group(
                     VerticalAnchor.CODEC.fieldOf("min_inclusive").forGetter(p_162043_ -> p_162043_.minInclusive),
                     VerticalAnchor.CODEC.fieldOf("max_inclusive").forGetter(p_162038_ -> p_162038_.maxInclusive)
