--- a/net/minecraft/world/level/levelgen/synth/SimplexNoise.java
+++ b/net/minecraft/world/level/levelgen/synth/SimplexNoise.java
@@ -36,11 +_,11 @@
         this.zo = p_230549_.nextDouble() * 256.0;
         int i = 0;
 
-        while(i < 256) {
+        while (i < 256) {
             this.p[i] = i++;
         }
 
-        for(int l = 0; l < 256; ++l) {
+        for (int l = 0; l < 256; l++) {
             int j = p_230549_.nextInt(256 - l);
             int k = this.p[l];
             this.p[l] = this.p[j + l];
