--- a/net/minecraft/server/packs/CompositePackResources.java
+++ b/net/minecraft/server/packs/CompositePackResources.java
@@ -35,7 +_,7 @@
     @Nullable
     @Override
     public IoSupplier<InputStream> getResource(PackType p_295406_, ResourceLocation p_296382_) {
-        for(PackResources packresources : this.packResourcesStack) {
+        for (PackResources packresources : this.packResourcesStack) {
             IoSupplier<InputStream> iosupplier = packresources.getResource(p_295406_, p_296382_);
             if (iosupplier != null) {
                 return iosupplier;
@@ -49,7 +_,7 @@
     public void listResources(PackType p_295490_, String p_296164_, String p_294691_, PackResources.ResourceOutput p_295313_) {
         Map<ResourceLocation, IoSupplier<InputStream>> map = new HashMap<>();
 
-        for(PackResources packresources : this.packResourcesStack) {
+        for (PackResources packresources : this.packResourcesStack) {
             packresources.listResources(p_295490_, p_296164_, p_294691_, map::putIfAbsent);
         }
 
@@ -60,7 +_,7 @@
     public Set<String> getNamespaces(PackType p_294708_) {
         Set<String> set = new HashSet<>();
 
-        for(PackResources packresources : this.packResourcesStack) {
+        for (PackResources packresources : this.packResourcesStack) {
             set.addAll(packresources.getNamespaces(p_294708_));
         }
 
