--- a/net/minecraft/server/packs/DownloadCacheCleaner.java
+++ b/net/minecraft/server/packs/DownloadCacheCleaner.java
@@ -37,7 +_,7 @@
             list1.sort(DownloadCacheCleaner.PathAndPriority.HIGHEST_PRIORITY_FIRST);
             Set<Path> set = new HashSet<>();
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 DownloadCacheCleaner.PathAndPriority downloadcachecleaner$pathandpriority = list1.get(j);
                 Path path = downloadcachecleaner$pathandpriority.path;
 
@@ -53,7 +_,7 @@
 
             set.remove(p_314919_);
 
-            for(Path path1 : set) {
+            for (Path path1 : set) {
                 try {
                     Files.delete(path1);
                 } catch (DirectoryNotEmptyException directorynotemptyexception) {
@@ -89,7 +_,7 @@
         List<DownloadCacheCleaner.PathAndPriority> list = new ArrayList<>();
         Object2IntOpenHashMap<Path> object2intopenhashmap = new Object2IntOpenHashMap<>();
 
-        for(DownloadCacheCleaner.PathAndTime downloadcachecleaner$pathandtime : p_314924_) {
+        for (DownloadCacheCleaner.PathAndTime downloadcachecleaner$pathandtime : p_314924_) {
             int i = object2intopenhashmap.addTo(downloadcachecleaner$pathandtime.path.getParent(), 1);
             list.add(new DownloadCacheCleaner.PathAndPriority(downloadcachecleaner$pathandtime.path, i));
         }
