--- a/net/minecraft/server/packs/PackLocationInfo.java
+++ b/net/minecraft/server/packs/PackLocationInfo.java
@@ -14,7 +_,7 @@
     public Component createChatLink(boolean p_326100_, Component p_325972_) {
         return ComponentUtils.wrapInSquareBrackets(this.source.decorate(Component.literal(this.id)))
             .withStyle(
-                p_339856_ -> p_339856_.withColor(p_326100_ ? ChatFormatting.GREEN : ChatFormatting.RED)
+                p_325937_ -> p_325937_.withColor(p_326100_ ? ChatFormatting.GREEN : ChatFormatting.RED)
                         .withInsertion(StringArgumentType.escapeIfRequired(this.id))
                         .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.empty().append(this.title).append("\n").append(p_325972_)))
             );
