--- a/net/minecraft/server/packs/VanillaPackResources.java
+++ b/net/minecraft/server/packs/VanillaPackResources.java
@@ -1,7 +_,7 @@
 package net.minecraft.server.packs;
 
 import com.mojang.logging.LogUtils;
-import com.mojang.serialization.DataResult.PartialResult;
+import com.mojang.serialization.DataResult.Error;
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.file.Files;
@@ -45,7 +_,7 @@
         FileUtil.validatePath(p_250530_);
         List<String> list = List.of(p_250530_);
 
-        for(Path path : this.rootPaths) {
+        for (Path path : this.rootPaths) {
             Path path1 = FileUtil.resolvePath(path, list);
             if (Files.exists(path1) && PathPackResources.validatePath(path1)) {
                 return IoSupplier.create(path1);
@@ -56,39 +_,39 @@
     }
 
     public void listRawPaths(PackType p_252103_, ResourceLocation p_250441_, Consumer<Path> p_251968_) {
-        FileUtil.decomposePath(p_250441_.getPath()).get().ifLeft(p_248238_ -> {
+        FileUtil.decomposePath(p_250441_.getPath()).ifSuccess(p_248238_ -> {
             String s = p_250441_.getNamespace();
 
-            for(Path path : this.pathsForType.get(p_252103_)) {
+            for (Path path : this.pathsForType.get(p_252103_)) {
                 Path path1 = path.resolve(s);
-                p_251968_.accept(FileUtil.resolvePath(path1, p_248238_));
+                p_251968_.accept(FileUtil.resolvePath(path1, (List<String>)p_248238_));
             }
-        }).ifRight(p_248232_ -> LOGGER.error("Invalid path {}: {}", p_250441_, p_248232_.message()));
+        }).ifError(p_337562_ -> LOGGER.error("Invalid path {}: {}", p_250441_, p_337562_.message()));
     }
 
     @Override
     public void listResources(PackType p_248974_, String p_248703_, String p_250848_, PackResources.ResourceOutput p_249668_) {
-        FileUtil.decomposePath(p_250848_).get().ifLeft(p_248228_ -> {
+        FileUtil.decomposePath(p_250848_).ifSuccess(p_248228_ -> {
             List<Path> list = this.pathsForType.get(p_248974_);
             int i = list.size();
             if (i == 1) {
-                getResources(p_249668_, p_248703_, list.get(0), p_248228_);
+                getResources(p_249668_, p_248703_, list.get(0), (List<String>)p_248228_);
             } else if (i > 1) {
                 Map<ResourceLocation, IoSupplier<InputStream>> map = new HashMap<>();
 
-                for(int j = 0; j < i - 1; ++j) {
-                    getResources(map::putIfAbsent, p_248703_, list.get(j), p_248228_);
+                for (int j = 0; j < i - 1; j++) {
+                    getResources(map::putIfAbsent, p_248703_, list.get(j), (List<String>)p_248228_);
                 }
 
                 Path path = list.get(i - 1);
                 if (map.isEmpty()) {
-                    getResources(p_249668_, p_248703_, path, p_248228_);
+                    getResources(p_249668_, p_248703_, path, (List<String>)p_248228_);
                 } else {
-                    getResources(map::putIfAbsent, p_248703_, path, p_248228_);
+                    getResources(map::putIfAbsent, p_248703_, path, (List<String>)p_248228_);
                     map.forEach(p_249668_);
                 }
             }
-        }).ifRight(p_248234_ -> LOGGER.error("Invalid path {}: {}", p_250848_, p_248234_.message()));
+        }).ifError(p_337564_ -> LOGGER.error("Invalid path {}: {}", p_250848_, p_337564_.message()));
     }
 
     private static void getResources(PackResources.ResourceOutput p_249662_, String p_251249_, Path p_251290_, List<String> p_250451_) {
@@ -99,19 +_,19 @@
     @Nullable
     @Override
     public IoSupplier<InputStream> getResource(PackType p_250512_, ResourceLocation p_251554_) {
-        return FileUtil.decomposePath(p_251554_.getPath()).get().map(p_248224_ -> {
+        return FileUtil.decomposePath(p_251554_.getPath()).mapOrElse(p_248224_ -> {
             String s = p_251554_.getNamespace();
 
-            for(Path path : this.pathsForType.get(p_250512_)) {
-                Path path1 = FileUtil.resolvePath(path.resolve(s), p_248224_);
+            for (Path path : this.pathsForType.get(p_250512_)) {
+                Path path1 = FileUtil.resolvePath(path.resolve(s), (List<String>)p_248224_);
                 if (Files.exists(path1) && PathPackResources.validatePath(path1)) {
                     return IoSupplier.create(path1);
                 }
             }
 
             return null;
-        }, p_248230_ -> {
-            LOGGER.error("Invalid path {}: {}", p_251554_, p_248230_.message());
+        }, p_337566_ -> {
+            LOGGER.error("Invalid path {}: {}", p_251554_, p_337566_.message());
             return null;
         });
     }
@@ -150,6 +_,7 @@
     }
 
     public ResourceProvider asProvider() {
-        return p_248239_ -> Optional.ofNullable(this.getResource(PackType.CLIENT_RESOURCES, p_248239_)).map(p_248221_ -> new Resource(this, p_248221_));
+        return p_248239_ -> Optional.ofNullable(this.getResource(PackType.CLIENT_RESOURCES, p_248239_))
+                .map(p_248221_ -> new Resource(this, (IoSupplier<InputStream>)p_248221_));
     }
 }
