--- a/net/minecraft/commands/execution/tasks/ContinuationTask.java
+++ b/net/minecraft/commands/execution/tasks/ContinuationTask.java
@@ -29,7 +_,7 @@
 
     public static <T, P> void schedule(ExecutionContext<T> p_306309_, Frame p_309712_, List<P> p_306081_, ContinuationTask.TaskProvider<T, P> p_305883_) {
         int i = p_306081_.size();
-        switch(i) {
+        switch (i) {
             case 0:
                 break;
             case 1:
