--- a/net/minecraft/commands/execution/tasks/FallthroughTask.java
+++ b/net/minecraft/commands/execution/tasks/FallthroughTask.java
@@ -6,7 +_,9 @@
 import net.minecraft.commands.execution.Frame;
 
 public class FallthroughTask<T extends ExecutionCommandSource<T>> implements EntryAction<T> {
-    private static final FallthroughTask<? extends ExecutionCommandSource<?>> INSTANCE = new FallthroughTask<>();
+    private static final FallthroughTask<? extends ExecutionCommandSource<?>> INSTANCE = (FallthroughTask<? extends ExecutionCommandSource<?>>)(new FallthroughTask<>(
+
+    ));
 
     public static <T extends ExecutionCommandSource<T>> EntryAction<T> instance() {
         return (EntryAction<T>)INSTANCE;
