--- a/com/mojang/math/SymmetricGroup3.java
+++ b/com/mojang/math/SymmetricGroup3.java
@@ -16,11 +_,11 @@
     private final Matrix3f transformation;
     private static final int ORDER = 3;
     private static final SymmetricGroup3[][] cayleyTable = Util.make(new SymmetricGroup3[values().length][values().length], p_109188_ -> {
-        for(SymmetricGroup3 symmetricgroup3 : values()) {
-            for(SymmetricGroup3 symmetricgroup31 : values()) {
+        for (SymmetricGroup3 symmetricgroup3 : values()) {
+            for (SymmetricGroup3 symmetricgroup31 : values()) {
                 int[] aint = new int[3];
 
-                for(int i = 0; i < 3; ++i) {
+                for (int i = 0; i < 3; i++) {
                     aint[i] = symmetricgroup3.permutation[symmetricgroup31.permutation[i]];
                 }
 
