--- a/net/minecraft/world/level/border/WorldBorder.java
+++ b/net/minecraft/world/level/border/WorldBorder.java
@@ -117,7 +_,7 @@
         this.centerZ = p_61951_;
         this.extent.onCenterChange();
 
-        for(BorderChangeListener borderchangelistener : this.getListeners()) {
+        for (BorderChangeListener borderchangelistener : this.getListeners()) {
             borderchangelistener.onBorderCenterSet(this, p_61950_, p_61951_);
         }
     }
@@ -137,7 +_,7 @@
     public void setSize(double p_61918_) {
         this.extent = new WorldBorder.StaticBorderExtent(p_61918_);
 
-        for(BorderChangeListener borderchangelistener : this.getListeners()) {
+        for (BorderChangeListener borderchangelistener : this.getListeners()) {
             borderchangelistener.onBorderSizeSet(this, p_61918_);
         }
     }
@@ -147,7 +_,7 @@
             ? new WorldBorder.StaticBorderExtent(p_61921_)
             : new WorldBorder.MovingBorderExtent(p_61920_, p_61921_, p_61922_));
 
-        for(BorderChangeListener borderchangelistener : this.getListeners()) {
+        for (BorderChangeListener borderchangelistener : this.getListeners()) {
             borderchangelistener.onBorderSizeLerping(this, p_61920_, p_61921_, p_61922_);
         }
     }
@@ -180,7 +_,7 @@
     public void setDamageSafeZone(double p_61940_) {
         this.damageSafeZone = p_61940_;
 
-        for(BorderChangeListener borderchangelistener : this.getListeners()) {
+        for (BorderChangeListener borderchangelistener : this.getListeners()) {
             borderchangelistener.onBorderSetDamageSafeZOne(this, p_61940_);
         }
     }
@@ -192,7 +_,7 @@
     public void setDamagePerBlock(double p_61948_) {
         this.damagePerBlock = p_61948_;
 
-        for(BorderChangeListener borderchangelistener : this.getListeners()) {
+        for (BorderChangeListener borderchangelistener : this.getListeners()) {
             borderchangelistener.onBorderSetDamagePerBlock(this, p_61948_);
         }
     }
@@ -208,7 +_,7 @@
     public void setWarningTime(int p_61945_) {
         this.warningTime = p_61945_;
 
-        for(BorderChangeListener borderchangelistener : this.getListeners()) {
+        for (BorderChangeListener borderchangelistener : this.getListeners()) {
             borderchangelistener.onBorderSetWarningTime(this, p_61945_);
         }
     }
@@ -220,7 +_,7 @@
     public void setWarningBlocks(int p_61953_) {
         this.warningBlocks = p_61953_;
 
-        for(BorderChangeListener borderchangelistener : this.getListeners()) {
+        for (BorderChangeListener borderchangelistener : this.getListeners()) {
             borderchangelistener.onBorderSetWarningBlocks(this, p_61953_);
         }
     }
