--- a/net/minecraft/server/PlayerAdvancements.java
+++ b/net/minecraft/server/PlayerAdvancements.java
@@ -164,6 +_,8 @@
     }
 
     public boolean award(AdvancementHolder holder, String criterion) {
+        // Forge: don't grant advancements for fake players
+        if (this.player instanceof net.neoforged.neoforge.common.util.FakePlayer) return false;
         boolean flag = false;
         AdvancementProgress advancementprogress = this.getOrStartProgress(holder);
         boolean flag1 = advancementprogress.isDone();
@@ -171,12 +_,14 @@
             this.unregisterListeners(holder);
             this.progressChanged.add(holder);
             flag = true;
+            net.neoforged.neoforge.event.EventHooks.onAdvancementProgressedEvent(this.player, holder, advancementprogress, criterion, net.neoforged.neoforge.event.entity.player.AdvancementEvent.AdvancementProgressEvent.ProgressType.GRANT);
             if (!flag1 && advancementprogress.isDone()) {
                 holder.value().rewards().grant(this.player);
                 holder.value().display().ifPresent(display -> {
                     if (display.shouldAnnounceChat() && this.player.level().getGameRules().get(GameRules.SHOW_ADVANCEMENT_MESSAGES)) {
                         this.playerList.broadcastSystemMessage(display.getType().createAnnouncement(holder, this.player), false);
                     }
+                    net.neoforged.neoforge.event.EventHooks.onAdvancementEarnedEvent(this.player, holder);
                 });
             }
         }
@@ -196,6 +_,7 @@
             this.registerListeners(advancement);
             this.progressChanged.add(advancement);
             flag = true;
+            net.neoforged.neoforge.event.EventHooks.onAdvancementProgressedEvent(this.player, advancement, advancementprogress, criterion, net.neoforged.neoforge.event.entity.player.AdvancementEvent.AdvancementProgressEvent.ProgressType.REVOKE);
         }
 
         if (flag1 && !advancementprogress.isDone()) {
