--- a/net/minecraft/server/ReloadableServerRegistries.java
+++ b/net/minecraft/server/ReloadableServerRegistries.java
@@ -49,10 +_,10 @@
         RegistryOps<JsonElement> registryops = new ReloadableServerRegistries.EmptyTagLookupWrapper(registryaccess$frozen)
             .createSerializationContext(JsonOps.INSTANCE);
         List<CompletableFuture<WritableRegistry<?>>> list = LootDataType.values()
-            .map(p_335899_ -> scheduleElementParse(p_335899_, registryops, p_335786_, p_335516_))
+            .map(p_335899_ -> scheduleElementParse((LootDataType<?>)p_335899_, registryops, p_335786_, p_335516_))
             .toList();
         CompletableFuture<List<WritableRegistry<?>>> completablefuture = Util.sequence(list);
-        return completablefuture.thenApplyAsync(p_335383_ -> apply(p_335950_, p_335383_), p_335516_);
+        return completablefuture.thenApplyAsync(p_335383_ -> apply(p_335950_, (List<WritableRegistry<?>>)p_335383_), p_335516_);
     }
 
     private static <T> CompletableFuture<WritableRegistry<?>> scheduleElementParse(
@@ -67,7 +_,7 @@
                     (p_335614_, p_335474_) -> p_335741_.deserialize(p_335614_, p_336173_, p_335474_)
                             .ifPresent(
                                 p_335683_ -> writableregistry.register(
-                                        ResourceKey.create(p_335741_.registryKey(), p_335614_), p_335683_, DEFAULT_REGISTRATION_INFO
+                                        ResourceKey.create(p_335741_.registryKey(), p_335614_), (T)p_335683_, DEFAULT_REGISTRATION_INFO
                                     )
                             )
                 );
@@ -84,7 +_,7 @@
         ValidationContext validationcontext = new ValidationContext(
             problemreporter$collector, LootContextParamSets.ALL_PARAMS, registryaccess$frozen.asGetterLookup()
         );
-        LootDataType.values().forEach(p_336006_ -> validateRegistry(validationcontext, p_336006_, registryaccess$frozen));
+        LootDataType.values().forEach(p_336006_ -> validateRegistry(validationcontext, (LootDataType<?>)p_336006_, registryaccess$frozen));
         problemreporter$collector.get()
             .forEach((p_336001_, p_335424_) -> LOGGER.warn("Found loot table element validation problem in {}: {}", p_336001_, p_335424_));
         return layeredregistryaccess;
