--- a/net/minecraft/server/ServerTickRateManager.java
+++ b/net/minecraft/server/ServerTickRateManager.java
@@ -94,7 +_,7 @@
             return false;
         } else if (this.remainingSprintTicks > 0L) {
             this.sprintTickStartTime = System.nanoTime();
-            --this.remainingSprintTicks;
+            this.remainingSprintTicks--;
             return true;
         } else {
             this.finishTickSprint();
@@ -103,7 +_,7 @@
     }
 
     public void endTickWork() {
-        this.sprintTimeSpend += System.nanoTime() - this.sprintTickStartTime;
+        this.sprintTimeSpend = this.sprintTimeSpend + (System.nanoTime() - this.sprintTickStartTime);
     }
 
     @Override
