--- a/net/minecraft/server/ServerFunctionManager.java
+++ b/net/minecraft/server/ServerFunctionManager.java
@@ -51,7 +_,7 @@
     private void executeTagFunctions(Collection<CommandFunction<CommandSourceStack>> p_136116_, ResourceLocation p_136117_) {
         this.server.getProfiler().push(p_136117_::toString);
 
-        for(CommandFunction<CommandSourceStack> commandfunction : p_136116_) {
+        for (CommandFunction<CommandSourceStack> commandfunction : p_136116_) {
             this.execute(commandfunction, this.getGameLoopSender());
         }
 
