--- a/net/minecraft/client/gui/screens/telemetry/TelemetryEventWidget.java
+++ b/net/minecraft/client/gui/screens/telemetry/TelemetryEventWidget.java
@@ -58,7 +_,7 @@
         List<TelemetryEventType> list = new ArrayList<>(TelemetryEventType.values());
         list.sort(Comparator.comparing(TelemetryEventType::isOptIn));
 
-        for(int i = 0; i < list.size(); ++i) {
+        for (int i = 0; i < list.size(); i++) {
             TelemetryEventType telemetryeventtype = list.get(i);
             boolean flag = telemetryeventtype.isOptIn() && !p_261628_;
             this.addEventType(telemetryeventwidget$contentbuilder, telemetryeventtype, flag);
@@ -121,7 +_,7 @@
     }
 
     private void addEventTypeProperties(TelemetryEventType p_262105_, TelemetryEventWidget.ContentBuilder p_261932_, boolean p_308932_) {
-        for(TelemetryProperty<?> telemetryproperty : p_262105_.properties()) {
+        for (TelemetryProperty<?> telemetryproperty : p_262105_.properties()) {
             p_261932_.addLine(this.font, this.grayOutIfDisabled(telemetryproperty.title(), p_308932_));
         }
     }
