--- a/net/minecraft/client/renderer/texture/atlas/sources/LazyLoadedImage.java
+++ b/net/minecraft/client/renderer/texture/atlas/sources/LazyLoadedImage.java
@@ -26,7 +_,7 @@
     public NativeImage get() throws IOException {
         NativeImage nativeimage = this.image.get();
         if (nativeimage == null) {
-            synchronized(this) {
+            synchronized (this) {
                 nativeimage = this.image.get();
                 if (nativeimage == null) {
                     try (InputStream inputstream = this.resource.open()) {
