--- a/net/minecraft/client/renderer/texture/atlas/sources/Unstitcher.java
+++ b/net/minecraft/client/renderer/texture/atlas/sources/Unstitcher.java
@@ -3,6 +_,7 @@
 import com.mojang.blaze3d.platform.NativeImage;
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -27,7 +_,7 @@
 @OnlyIn(Dist.CLIENT)
 public class Unstitcher implements SpriteSource {
     static final Logger LOGGER = LogUtils.getLogger();
-    public static final Codec<Unstitcher> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<Unstitcher> CODEC = RecordCodecBuilder.mapCodec(
         p_262047_ -> p_262047_.group(
                     ResourceLocation.CODEC.fieldOf("resource").forGetter(p_261910_ -> p_261910_.resource),
                     ExtraCodecs.nonEmptyList(Unstitcher.Region.CODEC.listOf()).fieldOf("regions").forGetter(p_261944_ -> p_261944_.regions),
@@ -55,7 +_,7 @@
         if (optional.isPresent()) {
             LazyLoadedImage lazyloadedimage = new LazyLoadedImage(resourcelocation, optional.get(), this.regions.size());
 
-            for(Unstitcher.Region unstitcher$region : this.regions) {
+            for (Unstitcher.Region unstitcher$region : this.regions) {
                 p_261828_.add(unstitcher$region.sprite, new Unstitcher.RegionInstance(lazyloadedimage, unstitcher$region, this.xDivisor, this.yDivisor));
             }
         } else {
