--- a/com/mojang/blaze3d/vertex/VertexConsumer.java
+++ b/com/mojang/blaze3d/vertex/VertexConsumer.java
@@ -101,10 +_,12 @@
             long k = quad.packedUV(j);
             float f = brightness[j];
             int l = ARGB.colorFromFloat(a, f * r, f * g, f * b);
+            l = ARGB.multiply(l, ARGB.toABGR(quad.bakedColors().color(j))); // Neo: apply baked color from the quad
             int i1 = LightTexture.lightCoordsWithEmission(lightmapCoord[j], i);
             Vector3f vector3f1 = matrix4f.transformPosition(vector3fc1, new Vector3f());
             float f1 = UVPair.unpackU(k);
             float f2 = UVPair.unpackV(k);
+            applyBakedNormals(vector3f, quad.bakedNormals(), j, pose.normal()); // Neo: apply baked normals from the quad
             this.addVertex(vector3f1.x(), vector3f1.y(), vector3f1.z(), l, f1, f2, overlayCoords, i1, vector3f.x(), vector3f.y(), vector3f.z());
         }
     }
