--- a/net/minecraft/world/entity/ai/behavior/AnimalMakeLove.java
+++ b/net/minecraft/world/entity/ai/behavior/AnimalMakeLove.java
@@ -101,8 +_,8 @@
     }
 
     private Optional<? extends Animal> findValidBreedPartner(Animal p_22432_) {
-        return p_22432_.getBrain().getMemory(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES).get().findClosest(p_339952_ -> {
-            if (p_339952_.getType() == this.partnerType && p_339952_ instanceof Animal animal && p_22432_.canMate(animal) && !animal.isPanicking()) {
+        return p_22432_.getBrain().getMemory(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES).get().findClosest(p_337706_ -> {
+            if (p_337706_.getType() == this.partnerType && p_337706_ instanceof Animal animal && p_22432_.canMate(animal) && !animal.isPanicking()) {
                 return true;
             }
 
