--- a/net/minecraft/world/entity/ai/behavior/Behavior.java
+++ b/net/minecraft/world/entity/ai/behavior/Behavior.java
@@ -89,7 +_,7 @@
     }
 
     protected boolean hasRequiredMemories(E p_22544_) {
-        for(Entry<MemoryModuleType<?>, MemoryStatus> entry : this.entryCondition.entrySet()) {
+        for (Entry<MemoryModuleType<?>, MemoryStatus> entry : this.entryCondition.entrySet()) {
             MemoryModuleType<?> memorymoduletype = entry.getKey();
             MemoryStatus memorystatus = entry.getValue();
             if (!p_22544_.getBrain().checkMemory(memorymoduletype, memorystatus)) {
