--- a/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
+++ b/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
@@ -2,7 +_,7 @@
 
 import java.util.Collections;
 import java.util.List;
-import java.util.stream.Collectors;
+import net.minecraft.Util;
 import net.minecraft.core.BlockPos;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.PathfinderMob;
@@ -24,7 +_,7 @@
                                     BlockPos blockpos = p_258394_.blockPosition();
                                     List<BlockPos> list = BlockPos.betweenClosedStream(blockpos.offset(-1, -1, -1), blockpos.offset(1, 1, 1))
                                         .map(BlockPos::immutable)
-                                        .collect(Collectors.toList());
+                                        .collect(Util.toMutableList());
                                     Collections.shuffle(list);
                                     list.stream()
                                         .filter(p_313604_ -> !p_258393_.canSeeSky(p_313604_))
