--- a/net/minecraft/world/entity/ai/behavior/InteractWith.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWith.java
@@ -28,7 +_,7 @@
         int p_259065_
     ) {
         int i = p_259564_ * p_259564_;
-        Predicate<LivingEntity> predicate = p_339967_ -> p_259366_.equals(p_339967_.getType()) && p_260254_.test((T)p_339967_);
+        Predicate<LivingEntity> predicate = p_337721_ -> p_259366_.equals(p_337721_.getType()) && p_260254_.test((T)p_337721_);
         return BehaviorBuilder.create(
             p_258426_ -> p_258426_.group(
                         p_258426_.registered(p_260229_),
