--- a/net/minecraft/world/entity/ai/behavior/JumpOnBed.java
+++ b/net/minecraft/world/entity/ai/behavior/JumpOnBed.java
@@ -66,13 +_,13 @@
 
     protected void tick(ServerLevel p_23394_, Mob p_23395_, long p_23396_) {
         if (!this.onOrOverBed(p_23394_, p_23395_)) {
-            --this.remainingTimeToReachBed;
+            this.remainingTimeToReachBed--;
         } else if (this.remainingCooldownUntilNextJump > 0) {
-            --this.remainingCooldownUntilNextJump;
+            this.remainingCooldownUntilNextJump--;
         } else {
             if (this.onBedSurface(p_23394_, p_23395_)) {
                 p_23395_.getJumpControl().jump();
-                --this.remainingJumps;
+                this.remainingJumps--;
                 this.remainingCooldownUntilNextJump = 5;
             }
         }
