--- a/net/minecraft/world/entity/ai/behavior/LongJumpUtil.java
+++ b/net/minecraft/world/entity/ai/behavior/LongJumpUtil.java
@@ -41,7 +_,7 @@
                     Vec3 vec34 = null;
                     EntityDimensions entitydimensions = p_311799_.getDimensions(Pose.LONG_JUMPING);
 
-                    for(int j = 0; j < i - 1; ++j) {
+                    for (int j = 0; j < i - 1; j++) {
                         d15 += d2 / (double)i;
                         double d16 = d7 / d8 * d15 - Math.pow(d15, 2.0) * d4 / (2.0 * d11 * Math.pow(d8, 2.0));
                         double d17 = d15 * d10;
@@ -67,7 +_,7 @@
         Vec3 vec31 = vec3.normalize();
         Vec3 vec32 = p_311995_;
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             vec32 = j == i - 1 ? p_312896_ : vec32.add(vec31.scale(d0 * 0.9F));
             if (!p_312910_.level().noCollision(p_312910_, p_312734_.makeBoundingBox(vec32))) {
                 return false;
