--- a/net/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
+++ b/net/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
@@ -139,13 +_,13 @@
                 p_147680_.playSound(null, p_147681_, this.getJumpSound.apply(p_147681_), SoundSource.NEUTRAL, 1.0F, 1.0F);
             }
         } else {
-            --this.findJumpTries;
+            this.findJumpTries--;
             this.pickCandidate(p_147680_, p_147681_, p_147682_);
         }
     }
 
     protected void pickCandidate(ServerLevel p_217319_, E p_217320_, long p_217321_) {
-        while(!this.jumpCandidates.isEmpty()) {
+        while (!this.jumpCandidates.isEmpty()) {
             Optional<LongJumpToRandomPos.PossibleJump> optional = this.getJumpCandidate(p_217319_);
             if (!optional.isEmpty()) {
                 LongJumpToRandomPos.PossibleJump longjumptorandompos$possiblejump = optional.get();
@@ -187,7 +_,7 @@
         Collections.shuffle(list);
         float f = (float)(p_217304_.getAttributeValue(Attributes.JUMP_STRENGTH) * (double)this.maxJumpVelocityMultiplier);
 
-        for(int i : list) {
+        for (int i : list) {
             Optional<Vec3> optional = LongJumpUtil.calculateJumpVectorForAngle(p_217304_, p_217305_, f, i, true);
             if (optional.isPresent()) {
                 return optional.get();
