--- a/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
+++ b/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
@@ -49,7 +_,7 @@
                                         return true;
                                     }
                                 } else {
-                                    for(int i = 0; i < 10; ++i) {
+                                    for (int i = 0; i < 10; i++) {
                                         Vec3 vec3 = LandRandomPos.getPos(p_275029_, 20, 8);
                                         if (vec3 != null && p_275028_.isVillage(BlockPos.containing(vec3))) {
                                             p_258560_.set(new WalkTarget(vec3, 0.6F, 0));
