--- a/net/minecraft/world/entity/ai/behavior/ReactToBell.java
+++ b/net/minecraft/world/entity/ai/behavior/ReactToBell.java
@@ -12,10 +_,10 @@
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
             p_259349_ -> p_259349_.group(p_259349_.present(MemoryModuleType.HEARD_BELL_TIME))
-                    .apply(p_259349_, p_259472_ -> (p_340011_, p_340012_, p_340013_) -> {
-                            Raid raid = p_340011_.getRaidAt(p_340012_.blockPosition());
+                    .apply(p_259349_, p_259472_ -> (p_337765_, p_337766_, p_337767_) -> {
+                            Raid raid = p_337765_.getRaidAt(p_337766_.blockPosition());
                             if (raid == null) {
-                                p_340012_.getBrain().setActiveActivityIfPossible(Activity.HIDE);
+                                p_337766_.getBrain().setActiveActivityIfPossible(Activity.HIDE);
                             }
         
                             return true;
