--- a/net/minecraft/world/entity/ai/behavior/ResetRaidStatus.java
+++ b/net/minecraft/world/entity/ai/behavior/ResetRaidStatus.java
@@ -9,15 +_,15 @@
 
 public class ResetRaidStatus {
     public static BehaviorControl<LivingEntity> create() {
-        return BehaviorBuilder.create(p_259870_ -> p_259870_.point((p_340014_, p_340015_, p_340016_) -> {
-                if (p_340014_.random.nextInt(20) != 0) {
+        return BehaviorBuilder.create(p_259870_ -> p_259870_.point((p_330025_, p_330026_, p_330027_) -> {
+                if (p_330025_.random.nextInt(20) != 0) {
                     return false;
                 } else {
-                    Brain<?> brain = p_340015_.getBrain();
-                    Raid raid = p_340014_.getRaidAt(p_340015_.blockPosition());
+                    Brain<?> brain = p_330026_.getBrain();
+                    Raid raid = p_330025_.getRaidAt(p_330026_.blockPosition());
                     if (raid == null || raid.isStopped() || raid.isLoss()) {
                         brain.setDefaultActivity(Activity.IDLE);
-                        brain.updateActivityFromSchedule(p_340014_.getDayTime(), p_340014_.getGameTime());
+                        brain.updateActivityFromSchedule(p_330025_.getDayTime(), p_330025_.getGameTime());
                     }
 
                     return true;
