--- a/net/minecraft/world/entity/ai/behavior/RingBell.java
+++ b/net/minecraft/world/entity/ai/behavior/RingBell.java
@@ -1,7 +_,6 @@
 package net.minecraft.world.entity.ai.behavior;
 
 import net.minecraft.core.BlockPos;
-import net.minecraft.core.GlobalPos;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.ai.behavior.declarative.BehaviorBuilder;
@@ -22,7 +_,7 @@
                             if (p_259026_.random.nextFloat() <= 0.95F) {
                                 return false;
                             } else {
-                                BlockPos blockpos = p_259094_.<GlobalPos>get(p_259028_).pos();
+                                BlockPos blockpos = p_259094_.get(p_259028_).pos();
                                 if (blockpos.closerThan(p_260317_.blockPosition(), 3.0)) {
                                     BlockState blockstate = p_259026_.getBlockState(blockpos);
                                     if (blockstate.is(Blocks.BELL)) {
