--- a/net/minecraft/world/entity/ai/behavior/SetEntityLookTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/SetEntityLookTarget.java
@@ -13,11 +_,11 @@
 
 public class SetEntityLookTarget {
     public static BehaviorControl<LivingEntity> create(MobCategory p_259154_, float p_260240_) {
-        return create(p_340020_ -> p_259154_.equals(p_340020_.getType().getCategory()), p_260240_);
+        return create(p_337769_ -> p_259154_.equals(p_337769_.getType().getCategory()), p_260240_);
     }
 
     public static OneShot<LivingEntity> create(EntityType<?> p_260318_, float p_259522_) {
-        return create(p_340018_ -> p_260318_.equals(p_340018_.getType()), p_259522_);
+        return create(p_337771_ -> p_260318_.equals(p_337771_.getType()), p_259522_);
     }
 
     public static OneShot<LivingEntity> create(float p_259830_) {
@@ -27,9 +_,7 @@
     public static OneShot<LivingEntity> create(Predicate<LivingEntity> p_260088_, float p_259747_) {
         float f = p_259747_ * p_259747_;
         return BehaviorBuilder.create(
-            p_258663_ -> p_258663_.group(
-                        p_258663_.absent(MemoryModuleType.LOOK_TARGET), p_258663_.present(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES)
-                    )
+            p_258663_ -> p_258663_.group(p_258663_.absent(MemoryModuleType.LOOK_TARGET), p_258663_.present(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES))
                     .apply(
                         p_258663_,
                         (p_258656_, p_258657_) -> (p_258650_, p_258651_, p_258652_) -> {
