--- a/net/minecraft/world/entity/ai/behavior/SetEntityLookTargetSometimes.java
+++ b/net/minecraft/world/entity/ai/behavior/SetEntityLookTargetSometimes.java
@@ -19,16 +_,14 @@
     }
 
     public static BehaviorControl<LivingEntity> create(EntityType<?> p_259588_, float p_259105_, UniformInt p_259422_) {
-        return create(p_259105_, p_259422_, p_340022_ -> p_259588_.equals(p_340022_.getType()));
+        return create(p_259105_, p_259422_, p_337773_ -> p_259588_.equals(p_337773_.getType()));
     }
 
     private static BehaviorControl<LivingEntity> create(float p_260336_, UniformInt p_259266_, Predicate<LivingEntity> p_260173_) {
         float f = p_260336_ * p_260336_;
         SetEntityLookTargetSometimes.Ticker setentitylooktargetsometimes$ticker = new SetEntityLookTargetSometimes.Ticker(p_259266_);
         return BehaviorBuilder.create(
-            p_259288_ -> p_259288_.group(
-                        p_259288_.absent(MemoryModuleType.LOOK_TARGET), p_259288_.present(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES)
-                    )
+            p_259288_ -> p_259288_.group(p_259288_.absent(MemoryModuleType.LOOK_TARGET), p_259288_.present(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES))
                     .apply(
                         p_259288_,
                         (p_259350_, p_260134_) -> (p_264952_, p_264953_, p_264954_) -> {
