--- a/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
+++ b/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
@@ -22,7 +_,7 @@
                         p_258685_,
                         (p_258680_, p_258681_, p_258682_) -> (p_258670_, p_258671_, p_258672_) -> {
                                 Optional<LivingEntity> optional = p_258685_.<NearestVisibleLivingEntities>get(p_258682_)
-                                    .findClosest(p_340035_ -> p_340035_.distanceToSqr(p_258671_) <= (double)i && p_259642_.equals(p_340035_.getType()));
+                                    .findClosest(p_337777_ -> p_337777_.distanceToSqr(p_258671_) <= (double)i && p_259642_.equals(p_337777_.getType()));
                                 if (optional.isEmpty()) {
                                     return false;
                                 } else {
