--- a/net/minecraft/world/entity/ai/behavior/ShowTradesToPlayer.java
+++ b/net/minecraft/world/entity/ai/behavior/ShowTradesToPlayer.java
@@ -67,7 +_,7 @@
             this.lookTime = Math.min(this.lookTime, 40);
         }
 
-        --this.lookTime;
+        this.lookTime--;
     }
 
     public void stop(ServerLevel p_24144_, Villager p_24145_, long p_24146_) {
@@ -100,7 +_,7 @@
     }
 
     private void updateDisplayItems(Villager p_24128_) {
-        for(MerchantOffer merchantoffer : p_24128_.getOffers()) {
+        for (MerchantOffer merchantoffer : p_24128_.getOffers()) {
             if (!merchantoffer.isOutOfStock() && this.playerItemStackMatchesCostOfOffer(merchantoffer)) {
                 this.displayItems.add(merchantoffer.assemble());
             }
@@ -130,7 +_,7 @@
 
     private void displayCyclingItems(Villager p_24148_) {
         if (this.displayItems.size() >= 2 && ++this.cycleCounter >= 40) {
-            ++this.displayIndex;
+            this.displayIndex++;
             this.cycleCounter = 0;
             if (this.displayIndex > this.displayItems.size() - 1) {
                 this.displayIndex = 0;
