--- a/net/minecraft/world/entity/ai/behavior/StartAttacking.java
+++ b/net/minecraft/world/entity/ai/behavior/StartAttacking.java
@@ -17,9 +_,7 @@
 
     public static <E extends Mob> BehaviorControl<E> create(Predicate<E> p_259618_, Function<E, Optional<? extends LivingEntity>> p_259435_) {
         return BehaviorBuilder.create(
-            p_258782_ -> p_258782_.group(
-                        p_258782_.absent(MemoryModuleType.ATTACK_TARGET), p_258782_.registered(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE)
-                    )
+            p_258782_ -> p_258782_.group(p_258782_.absent(MemoryModuleType.ATTACK_TARGET), p_258782_.registered(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE))
                     .apply(p_258782_, (p_258778_, p_258779_) -> (p_258773_, p_258774_, p_258775_) -> {
                             if (!p_259618_.test(p_258774_)) {
                                 return false;
@@ -32,7 +_,11 @@
                                     if (!p_258774_.canAttack(livingentity)) {
                                         return false;
                                     } else {
-                                        p_258778_.set(livingentity);
+                                        net.neoforged.neoforge.event.entity.living.LivingChangeTargetEvent changeTargetEvent = net.neoforged.neoforge.common.CommonHooks.onLivingChangeTarget(p_258774_, livingentity, net.neoforged.neoforge.event.entity.living.LivingChangeTargetEvent.LivingTargetType.BEHAVIOR_TARGET);
+                                        if (changeTargetEvent.isCanceled())
+                                            return false;
+
+                                        p_258778_.set(changeTargetEvent.getNewTarget());
                                         p_258779_.erase();
                                         return true;
                                     }
