--- a/net/minecraft/world/entity/ai/behavior/TradeWithVillager.java
+++ b/net/minecraft/world/entity/ai/behavior/TradeWithVillager.java
@@ -76,7 +_,7 @@
         ItemStack itemstack = ItemStack.EMPTY;
         int i = 0;
 
-        while(i < simplecontainer.getContainerSize()) {
+        while (i < simplecontainer.getContainerSize()) {
             ItemStack itemstack1;
             Item item;
             int j;
@@ -97,7 +_,7 @@
                     }
                 }
 
-                ++i;
+                i++;
                 continue;
             }
 
