--- a/net/minecraft/world/entity/ai/behavior/TriggerGate.java
+++ b/net/minecraft/world/entity/ai/behavior/TriggerGate.java
@@ -16,13 +_,13 @@
         List<Pair<? extends Trigger<? super E>, Integer>> p_259442_, GateBehavior.OrderPolicy p_259823_, GateBehavior.RunningPolicy p_259632_
     ) {
         ShufflingList<Trigger<? super E>> shufflinglist = new ShufflingList<>();
-        p_259442_.forEach(p_260333_ -> shufflinglist.add(p_260333_.getFirst(), p_260333_.getSecond()));
+        p_259442_.forEach(p_260333_ -> shufflinglist.add((Trigger<? super E>)p_260333_.getFirst(), p_260333_.getSecond()));
         return BehaviorBuilder.create(p_259457_ -> p_259457_.point((p_260107_, p_259505_, p_259999_) -> {
                 if (p_259823_ == GateBehavior.OrderPolicy.SHUFFLED) {
                     shufflinglist.shuffle();
                 }
 
-                for(Trigger<? super E> trigger : shufflinglist) {
+                for (Trigger<? super E> trigger : shufflinglist) {
                     if (trigger.trigger(p_260107_, p_259505_, p_259999_) && p_259632_ == GateBehavior.RunningPolicy.RUN_ONE) {
                         break;
                     }
