--- a/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
+++ b/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
@@ -36,13 +_,13 @@
                                     BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
             
                                     label45:
-                                    for(BlockPos blockpos1 : BlockPos.withinManhattan(blockpos, p_259739_, p_259739_, p_259739_)) {
+                                    for (BlockPos blockpos1 : BlockPos.withinManhattan(blockpos, p_259739_, p_259739_, p_259739_)) {
                                         if ((blockpos1.getX() != blockpos.getX() || blockpos1.getZ() != blockpos.getZ())
                                             && p_259876_.getBlockState(blockpos1).getCollisionShape(p_259876_, blockpos1, collisioncontext).isEmpty()
                                             && !p_259876_.getBlockState(blockpos$mutableblockpos.setWithOffset(blockpos1, Direction.DOWN))
                                                 .getCollisionShape(p_259876_, blockpos1, collisioncontext)
                                                 .isEmpty()) {
-                                            for(Direction direction : Direction.Plane.HORIZONTAL) {
+                                            for (Direction direction : Direction.Plane.HORIZONTAL) {
                                                 blockpos$mutableblockpos.setWithOffset(blockpos1, direction);
                                                 if (p_259876_.getBlockState(blockpos$mutableblockpos).isAir()
                                                     && p_259876_.getBlockState(blockpos$mutableblockpos.move(Direction.DOWN)).is(Blocks.WATER)) {
