--- a/net/minecraft/world/entity/ai/behavior/TryLaySpawnOnWaterNearLand.java
+++ b/net/minecraft/world/entity/ai/behavior/TryLaySpawnOnWaterNearLand.java
@@ -28,7 +_,7 @@
                                 if (!p_269882_.isInWater() && p_269882_.onGround()) {
                                     BlockPos blockpos = p_269882_.blockPosition().below();
             
-                                    for(Direction direction : Direction.Plane.HORIZONTAL) {
+                                    for (Direction direction : Direction.Plane.HORIZONTAL) {
                                         BlockPos blockpos1 = blockpos.relative(direction);
                                         if (p_269881_.getBlockState(blockpos1).getCollisionShape(p_269881_, blockpos1).getFaceShape(Direction.UP).isEmpty()
                                             && p_269881_.getFluidState(blockpos1).is(Fluids.WATER)) {
