--- a/net/minecraft/world/entity/ai/behavior/UseBonemeal.java
+++ b/net/minecraft/world/entity/ai/behavior/UseBonemeal.java
@@ -50,9 +_,9 @@
         Optional<BlockPos> optional = Optional.empty();
         int i = 0;
 
-        for(int j = -1; j <= 1; ++j) {
-            for(int k = -1; k <= 1; ++k) {
-                for(int l = -1; l <= 1; ++l) {
+        for (int j = -1; j <= 1; j++) {
+            for (int k = -1; k <= 1; k++) {
+                for (int l = -1; l <= 1; l++) {
                     blockpos$mutableblockpos.setWithOffset(p_24494_.blockPosition(), j, k, l);
                     if (this.validPos(blockpos$mutableblockpos, p_24493_)) {
                         if (p_24493_.random.nextInt(++i) == 0) {
@@ -99,7 +_,7 @@
             SimpleContainer simplecontainer = p_24513_.getInventory();
             int i = simplecontainer.getContainerSize();
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 ItemStack itemstack1 = simplecontainer.getItem(j);
                 if (itemstack1.is(Items.BONE_MEAL)) {
                     itemstack = itemstack1;
@@ -114,7 +_,7 @@
                 this.nextWorkCycleTime = p_24514_ + 40L;
             }
 
-            ++this.timeWorkedSoFar;
+            this.timeWorkedSoFar++;
         }
     }
 }
