--- a/net/minecraft/world/entity/ai/behavior/VillagerMakeLove.java
+++ b/net/minecraft/world/entity/ai/behavior/VillagerMakeLove.java
@@ -84,12 +_,10 @@
 
     private boolean isBreedingPossible(Villager p_24640_) {
         Brain<Villager> brain = p_24640_.getBrain();
-        Optional<AgeableMob> optional = brain.getMemory(MemoryModuleType.BREED_TARGET).filter(p_340051_ -> p_340051_.getType() == EntityType.VILLAGER);
-        if (optional.isEmpty()) {
-            return false;
-        } else {
-            return BehaviorUtils.targetIsValid(brain, MemoryModuleType.BREED_TARGET, EntityType.VILLAGER) && p_24640_.canBreed() && optional.get().canBreed();
-        }
+        Optional<AgeableMob> optional = brain.getMemory(MemoryModuleType.BREED_TARGET).filter(p_337783_ -> p_337783_.getType() == EntityType.VILLAGER);
+        return optional.isEmpty()
+            ? false
+            : BehaviorUtils.targetIsValid(brain, MemoryModuleType.BREED_TARGET, EntityType.VILLAGER) && p_24640_.canBreed() && optional.get().canBreed();
     }
 
     private Optional<BlockPos> takeVacantBed(ServerLevel p_24649_, Villager p_24650_) {
