--- a/net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
+++ b/net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
@@ -18,14 +_,14 @@
                     )
                     .apply(
                         p_258884_,
-                        (p_258886_, p_258887_, p_258888_) -> (p_340048_, p_340049_, p_340050_) -> {
+                        (p_258886_, p_258887_, p_258888_) -> (p_330046_, p_330047_, p_330048_) -> {
                                 boolean flag = p_258884_.tryGet(p_258886_).isPresent()
                                     || p_258884_.tryGet(p_258888_).isPresent()
-                                    || p_258884_.<LivingEntity>tryGet(p_258887_).filter(p_325748_ -> p_325748_.distanceToSqr(p_340049_) <= 36.0).isPresent();
+                                    || p_258884_.<LivingEntity>tryGet(p_258887_).filter(p_325748_ -> p_325748_.distanceToSqr(p_330047_) <= 36.0).isPresent();
                                 if (!flag) {
                                     p_258886_.erase();
                                     p_258887_.erase();
-                                    p_340049_.getBrain().updateActivityFromSchedule(p_340048_.getDayTime(), p_340048_.getGameTime());
+                                    p_330047_.getBrain().updateActivityFromSchedule(p_330046_.getDayTime(), p_330046_.getGameTime());
                                 }
             
                                 return true;
