--- a/net/minecraft/world/entity/ai/behavior/WorkAtComposter.java
+++ b/net/minecraft/world/entity/ai/behavior/WorkAtComposter.java
@@ -45,7 +_,7 @@
         int k = simplecontainer.getContainerSize();
         BlockState blockstate = p_24796_;
 
-        for(int l = k - 1; l >= 0 && i > 0; --l) {
+        for (int l = k - 1; l >= 0 && i > 0; l--) {
             ItemStack itemstack = simplecontainer.getItem(l);
             int i1 = COMPOSTABLE_ITEMS.indexOf(itemstack.getItem());
             if (i1 != -1) {
@@ -56,7 +_,7 @@
                 if (l1 > 0) {
                     i -= l1;
 
-                    for(int i2 = 0; i2 < l1; ++i2) {
+                    for (int i2 = 0; i2 < l1; i2++) {
                         blockstate = ComposterBlock.insertItem(p_24794_, blockstate, p_24793_, itemstack, blockpos);
                         if (blockstate.getValue(ComposterBlock.LEVEL) == 7) {
                             this.spawnComposterFillEffects(p_24793_, p_24796_, blockpos, blockstate);
