--- a/net/minecraft/stats/ServerRecipeBook.java
+++ b/net/minecraft/stats/ServerRecipeBook.java
@@ -27,14 +_,14 @@
         List<ResourceLocation> list = Lists.newArrayList();
         int i = 0;
 
-        for(RecipeHolder<?> recipeholder : p_12792_) {
+        for (RecipeHolder<?> recipeholder : p_12792_) {
             ResourceLocation resourcelocation = recipeholder.id();
             if (!this.known.contains(resourcelocation) && !recipeholder.value().isSpecial()) {
                 this.add(resourcelocation);
                 this.addHighlight(resourcelocation);
                 list.add(resourcelocation);
                 CriteriaTriggers.RECIPE_UNLOCKED.trigger(p_12793_, recipeholder);
-                ++i;
+                i++;
             }
         }
 
@@ -49,12 +_,12 @@
         List<ResourceLocation> list = Lists.newArrayList();
         int i = 0;
 
-        for(RecipeHolder<?> recipeholder : p_12807_) {
+        for (RecipeHolder<?> recipeholder : p_12807_) {
             ResourceLocation resourcelocation = recipeholder.id();
             if (this.known.contains(resourcelocation)) {
                 this.remove(resourcelocation);
                 list.add(resourcelocation);
-                ++i;
+                i++;
             }
         }
 
@@ -71,14 +_,14 @@
         this.getBookSettings().write(compoundtag);
         ListTag listtag = new ListTag();
 
-        for(ResourceLocation resourcelocation : this.known) {
+        for (ResourceLocation resourcelocation : this.known) {
             listtag.add(StringTag.valueOf(resourcelocation.toString()));
         }
 
         compoundtag.put("recipes", listtag);
         ListTag listtag1 = new ListTag();
 
-        for(ResourceLocation resourcelocation1 : this.highlight) {
+        for (ResourceLocation resourcelocation1 : this.highlight) {
             listtag1.add(StringTag.valueOf(resourcelocation1.toString()));
         }
 
@@ -95,7 +_,7 @@
     }
 
     private void loadRecipes(ListTag p_12798_, Consumer<RecipeHolder<?>> p_12799_, RecipeManager p_12800_) {
-        for(int i = 0; i < p_12798_.size(); ++i) {
+        for (int i = 0; i < p_12798_.size(); i++) {
             String s = p_12798_.getString(i);
 
             try {
