--- a/net/minecraft/world/entity/ai/targeting/LivingBlockTargetingConditions.java
+++ b/net/minecraft/world/entity/ai/targeting/LivingBlockTargetingConditions.java
@@ -2,6 +_,7 @@
 
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.Targetable;
 import org.jspecify.annotations.Nullable;
 
@@ -12,6 +_,6 @@
 
     @Override
     public <T extends Entity & Targetable> boolean test(ServerLevel level, @Nullable Entity targeter, Entity target) {
-        return this.selector == null || this.selector.test(target, level);
+        return this.selector == null || !(target instanceof LivingEntity) || this.selector.test((LivingEntity)target, level);
     }
 }
