--- a/net/minecraft/world/entity/ai/goal/BegGoal.java
+++ b/net/minecraft/world/entity/ai/goal/BegGoal.java
@@ -37,10 +_,10 @@
     public boolean canContinueToUse() {
         if (!this.player.isAlive()) {
             return false;
-        } else if (this.wolf.distanceToSqr(this.player) > (double)(this.lookDistance * this.lookDistance)) {
-            return false;
         } else {
-            return this.lookTime > 0 && this.playerHoldingInteresting(this.player);
+            return this.wolf.distanceToSqr(this.player) > (double)(this.lookDistance * this.lookDistance)
+                ? false
+                : this.lookTime > 0 && this.playerHoldingInteresting(this.player);
         }
     }
 
@@ -59,11 +_,11 @@
     @Override
     public void tick() {
         this.wolf.getLookControl().setLookAt(this.player.getX(), this.player.getEyeY(), this.player.getZ(), 10.0F, (float)this.wolf.getMaxHeadXRot());
-        --this.lookTime;
+        this.lookTime--;
     }
 
     private boolean playerHoldingInteresting(Player p_25067_) {
-        for(InteractionHand interactionhand : InteractionHand.values()) {
+        for (InteractionHand interactionhand : InteractionHand.values()) {
             ItemStack itemstack = p_25067_.getItemInHand(interactionhand);
             if (this.wolf.isTame() && itemstack.is(Items.BONE)) {
                 return true;
