--- a/net/minecraft/world/entity/ai/goal/BreedGoal.java
+++ b/net/minecraft/world/entity/ai/goal/BreedGoal.java
@@ -19,7 +_,7 @@
     private final double speedModifier;
 
     public BreedGoal(Animal p_25122_, double p_25123_) {
-        this(p_25122_, p_25123_, p_25122_.getClass());
+        this(p_25122_, p_25123_, (Class<? extends Animal>)p_25122_.getClass());
     }
 
     public BreedGoal(Animal p_25125_, double p_25126_, Class<? extends Animal> p_25127_) {
@@ -55,7 +_,7 @@
     public void tick() {
         this.animal.getLookControl().setLookAt(this.partner, 10.0F, (float)this.animal.getMaxHeadXRot());
         this.animal.getNavigation().moveTo(this.partner, this.speedModifier);
-        ++this.loveTime;
+        this.loveTime++;
         if (this.loveTime >= this.adjustedTickDelay(60) && this.animal.distanceToSqr(this.partner) < 9.0) {
             this.breed();
         }
@@ -68,7 +_,7 @@
         double d0 = Double.MAX_VALUE;
         Animal animal = null;
 
-        for(Animal animal1 : list) {
+        for (Animal animal1 : list) {
             if (this.animal.canMate(animal1) && !animal1.isPanicking() && this.animal.distanceToSqr(animal1) < d0) {
                 animal = animal1;
                 d0 = this.animal.distanceToSqr(animal1);
