--- a/net/minecraft/world/entity/ai/goal/CatSitOnBlockGoal.java
+++ b/net/minecraft/world/entity/ai/goal/CatSitOnBlockGoal.java
@@ -55,7 +_,7 @@
                 return blockstate.is(Blocks.FURNACE) && blockstate.getValue(FurnaceBlock.LIT)
                     ? true
                     : blockstate.is(
-                        BlockTags.BEDS, p_25156_ -> p_25156_.<BedPart>getOptionalValue(BedBlock.PART).map(p_148084_ -> p_148084_ != BedPart.HEAD).orElse(true)
+                        BlockTags.BEDS, p_25156_ -> p_25156_.getOptionalValue(BedBlock.PART).map(p_148084_ -> p_148084_ != BedPart.HEAD).orElse(true)
                     );
             }
         }
