--- a/net/minecraft/world/entity/ai/goal/FollowParentGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowParentGoal.java
@@ -24,11 +_,13 @@
         if (this.animal.getAge() >= 0) {
             return false;
         } else {
-            List<? extends Animal> list = this.animal.level().getEntitiesOfClass(this.animal.getClass(), this.animal.getBoundingBox().inflate(8.0, 4.0, 8.0));
+            List<? extends Animal> list = this.animal
+                .level()
+                .getEntitiesOfClass((Class<? extends Animal>)this.animal.getClass(), this.animal.getBoundingBox().inflate(8.0, 4.0, 8.0));
             Animal animal = null;
             double d0 = Double.MAX_VALUE;
 
-            for(Animal animal1 : list) {
+            for (Animal animal1 : list) {
                 if (animal1.getAge() >= 0) {
                     double d1 = this.animal.distanceToSqr(animal1);
                     if (!(d1 > d0)) {
