--- a/net/minecraft/world/entity/ai/goal/FollowFlockLeaderGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowFlockLeaderGoal.java
@@ -27,14 +_,14 @@
         } else if (this.mob.isFollower()) {
             return true;
         } else if (this.nextStartTick > 0) {
-            --this.nextStartTick;
+            this.nextStartTick--;
             return false;
         } else {
             this.nextStartTick = this.nextStartTick(this.mob);
             Predicate<AbstractSchoolingFish> predicate = p_25258_ -> p_25258_.canBeFollowed() || !p_25258_.isFollower();
             List<? extends AbstractSchoolingFish> list = this.mob
                 .level()
-                .getEntitiesOfClass(this.mob.getClass(), this.mob.getBoundingBox().inflate(8.0, 8.0, 8.0), predicate);
+                .getEntitiesOfClass((Class<? extends AbstractSchoolingFish>)this.mob.getClass(), this.mob.getBoundingBox().inflate(8.0, 8.0, 8.0), predicate);
             AbstractSchoolingFish abstractschoolingfish = DataFixUtils.orElse(list.stream().filter(AbstractSchoolingFish::canBeFollowed).findAny(), this.mob);
             abstractschoolingfish.addFollowers(list.stream().filter(p_25255_ -> !p_25255_.isFollower()));
             return this.mob.isFollower();
