--- a/net/minecraft/world/entity/ai/goal/FollowMobGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowMobGoal.java
@@ -40,7 +_,7 @@
     public boolean canUse() {
         List<Mob> list = this.mob.level().getEntitiesOfClass(Mob.class, this.mob.getBoundingBox().inflate((double)this.areaSize), this.followPredicate);
         if (!list.isEmpty()) {
-            for(Mob mob : list) {
+            for (Mob mob : list) {
                 if (!mob.isInvisible()) {
                     this.followingMob = mob;
                     return true;
