--- a/net/minecraft/world/entity/ai/goal/FollowBoatGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowBoatGoal.java
@@ -6,7 +_,6 @@
 import net.minecraft.core.Direction;
 import net.minecraft.util.Mth;
 import net.minecraft.world.entity.Entity;
-import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.MoverType;
 import net.minecraft.world.entity.PathfinderMob;
 import net.minecraft.world.entity.player.Player;
@@ -29,7 +_,7 @@
         List<Boat> list = this.mob.level().getEntitiesOfClass(Boat.class, this.mob.getBoundingBox().inflate(5.0));
         boolean flag = false;
 
-        for(Boat boat : list) {
+        for (Boat boat : list) {
             Entity entity = boat.getControllingPassenger();
             if (entity instanceof Player && (Mth.abs(((Player)entity).xxa) > 0.0F || Mth.abs(((Player)entity).zza) > 0.0F)) {
                 flag = true;
@@ -52,9 +_,8 @@
 
     @Override
     public void start() {
-        for(Boat boat : this.mob.level().getEntitiesOfClass(Boat.class, this.mob.getBoundingBox().inflate(5.0))) {
-            LivingEntity livingentity = boat.getControllingPassenger();
-            if (livingentity instanceof Player player) {
+        for (Boat boat : this.mob.level().getEntitiesOfClass(Boat.class, this.mob.getBoundingBox().inflate(5.0))) {
+            if (boat.getControllingPassenger() instanceof Player player) {
                 this.following = player;
                 break;
             }
