--- a/net/minecraft/world/entity/ai/goal/Goal.java
+++ b/net/minecraft/world/entity/ai/goal/Goal.java
@@ -54,11 +_,11 @@
         return Mth.positiveCeilDiv(ticks, 2);
     }
 
-    protected static ServerLevel getServerLevel(Entity entity) {
+    protected ServerLevel getServerLevel(Entity entity) {
         return (ServerLevel)entity.level();
     }
 
-    protected static ServerLevel getServerLevel(Level level) {
+    protected ServerLevel getServerLevel(Level level) {
         return (ServerLevel)level;
     }
 
