--- a/net/minecraft/world/entity/ai/goal/LeapAtTargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/LeapAtTargetGoal.java
@@ -28,10 +_,8 @@
                 double d0 = this.mob.distanceToSqr(this.target);
                 if (d0 < 4.0 || d0 > 16.0) {
                     return false;
-                } else if (!this.mob.onGround()) {
-                    return false;
                 } else {
-                    return this.mob.getRandom().nextInt(reducedTickDelay(5)) == 0;
+                    return !this.mob.onGround() ? false : this.mob.getRandom().nextInt(reducedTickDelay(5)) == 0;
                 }
             }
         }
