--- a/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
+++ b/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
@@ -30,7 +_,7 @@
             Llama llama = null;
             double d0 = Double.MAX_VALUE;
 
-            for(Entity entity : list) {
+            for (Entity entity : list) {
                 Llama llama1 = (Llama)entity;
                 if (llama1.inCaravan() && !llama1.hasCaravanTail()) {
                     double d1 = this.llama.distanceToSqr(llama1);
@@ -42,7 +_,7 @@
             }
 
             if (llama == null) {
-                for(Entity entity1 : list) {
+                for (Entity entity1 : list) {
                     Llama llama2 = (Llama)entity1;
                     if (llama2.isLeashed() && !llama2.hasCaravanTail()) {
                         double d2 = this.llama.distanceToSqr(llama2);
@@ -86,7 +_,7 @@
             }
 
             if (this.distCheckCounter > 0) {
-                --this.distCheckCounter;
+                this.distCheckCounter--;
             }
 
             return true;
