--- a/net/minecraft/world/entity/ai/goal/LookAtPlayerGoal.java
+++ b/net/minecraft/world/entity/ai/goal/LookAtPlayerGoal.java
@@ -15,7 +_,7 @@
     @Nullable
     protected Entity lookAt;
     protected final float lookDistance;
-    protected int lookTime;
+    private int lookTime;
     protected final float probability;
     private final boolean onlyHorizontal;
     protected final Class<? extends LivingEntity> lookAtType;
@@ -83,10 +_,8 @@
     public boolean canContinueToUse() {
         if (!this.lookAt.isAlive()) {
             return false;
-        } else if (this.mob.distanceToSqr(this.lookAt) > (double)(this.lookDistance * this.lookDistance)) {
-            return false;
         } else {
-            return this.lookTime > 0;
+            return this.mob.distanceToSqr(this.lookAt) > (double)(this.lookDistance * this.lookDistance) ? false : this.lookTime > 0;
         }
     }
 
@@ -105,7 +_,7 @@
         if (this.lookAt.isAlive()) {
             double d0 = this.onlyHorizontal ? this.mob.getEyeY() : this.lookAt.getEyeY();
             this.mob.getLookControl().setLookAt(this.lookAt.getX(), d0, this.lookAt.getZ());
-            --this.lookTime;
+            this.lookTime--;
         }
     }
 }
