--- a/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
+++ b/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
@@ -36,7 +_,7 @@
     @Override
     public boolean canUse() {
         if (this.nextStartTick > 0) {
-            --this.nextStartTick;
+            this.nextStartTick--;
             return false;
         } else {
             this.nextStartTick = this.nextStartTick(this.mob);
@@ -84,13 +_,13 @@
         BlockPos blockpos = this.getMoveToTarget();
         if (!blockpos.closerToCenterThan(this.mob.position(), this.acceptedDistance())) {
             this.reachedTarget = false;
-            ++this.tryTicks;
+            this.tryTicks++;
             if (this.shouldRecalculatePath()) {
                 this.mob.getNavigation().moveTo((double)blockpos.getX() + 0.5, (double)blockpos.getY(), (double)blockpos.getZ() + 0.5, this.speedModifier);
             }
         } else {
             this.reachedTarget = true;
-            --this.tryTicks;
+            this.tryTicks--;
         }
     }
 
@@ -108,10 +_,10 @@
         BlockPos blockpos = this.mob.blockPosition();
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int k = this.verticalSearchStart; k <= j; k = k > 0 ? -k : 1 - k) {
-            for(int l = 0; l < i; ++l) {
-                for(int i1 = 0; i1 <= l; i1 = i1 > 0 ? -i1 : 1 - i1) {
-                    for(int j1 = i1 < l && i1 > -l ? l : 0; j1 <= l; j1 = j1 > 0 ? -j1 : 1 - j1) {
+        for (int k = this.verticalSearchStart; k <= j; k = k > 0 ? -k : 1 - k) {
+            for (int l = 0; l < i; l++) {
+                for (int i1 = 0; i1 <= l; i1 = i1 > 0 ? -i1 : 1 - i1) {
+                    for (int j1 = i1 < l && i1 > -l ? l : 0; j1 <= l; j1 = j1 > 0 ? -j1 : 1 - j1) {
                         blockpos$mutableblockpos.setWithOffset(blockpos, i1, k - 1, j1);
                         if (this.mob.isWithinRestriction(blockpos$mutableblockpos) && this.isValidTarget(this.mob.level(), blockpos$mutableblockpos)) {
                             this.blockPos = blockpos$mutableblockpos;
