--- a/net/minecraft/world/entity/ai/goal/TradeWithPlayerGoal.java
+++ b/net/minecraft/world/entity/ai/goal/TradeWithPlayerGoal.java
@@ -26,10 +_,8 @@
             Player player = this.mob.getTradingPlayer();
             if (player == null) {
                 return false;
-            } else if (this.mob.distanceToSqr(player) > 16.0) {
-                return false;
             } else {
-                return player.containerMenu != null;
+                return this.mob.distanceToSqr(player) > 16.0 ? false : player.containerMenu != null;
             }
         }
     }
