--- a/net/minecraft/world/entity/ai/goal/TryFindWaterGoal.java
+++ b/net/minecraft/world/entity/ai/goal/TryFindWaterGoal.java
@@ -21,7 +_,7 @@
     public void start() {
         BlockPos blockpos = null;
 
-        for(BlockPos blockpos1 : BlockPos.betweenClosed(
+        for (BlockPos blockpos1 : BlockPos.betweenClosed(
             Mth.floor(this.mob.getX() - 2.0),
             Mth.floor(this.mob.getY() - 2.0),
             Mth.floor(this.mob.getZ() - 2.0),
