--- a/net/minecraft/world/entity/ai/goal/ZombieAttackGoal.java
+++ b/net/minecraft/world/entity/ai/goal/ZombieAttackGoal.java
@@ -1,14 +_,14 @@
 package net.minecraft.world.entity.ai.goal;
 
-import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.monster.Zombie;
 
-public class ZombieAttackGoal<T extends PathfinderMob> extends MeleeAttackGoal {
-    private final T zombie;
+public class ZombieAttackGoal extends MeleeAttackGoal {
+    private final Zombie zombie;
     private int raiseArmTicks;
 
-    public ZombieAttackGoal(T p_342302_, double p_26020_, boolean p_26021_) {
-        super(p_342302_, p_26020_, p_26021_);
-        this.zombie = p_342302_;
+    public ZombieAttackGoal(Zombie p_26019_, double p_26020_, boolean p_26021_) {
+        super(p_26019_, p_26020_, p_26021_);
+        this.zombie = p_26019_;
     }
 
     @Override
@@ -26,7 +_,7 @@
     @Override
     public void tick() {
         super.tick();
-        ++this.raiseArmTicks;
+        this.raiseArmTicks++;
         if (this.raiseArmTicks >= 5 && this.getTicksUntilNextAttack() < this.getAttackInterval() / 2) {
             this.zombie.setAggressive(true);
         } else {
