--- a/net/minecraft/util/thread/ProcessorHandle.java
+++ b/net/minecraft/util/thread/ProcessorHandle.java
@@ -16,14 +_,14 @@
 
     default <Source> CompletableFuture<Source> ask(Function<? super ProcessorHandle<Source>, ? extends Msg> p_18721_) {
         CompletableFuture<Source> completablefuture = new CompletableFuture<>();
-        Msg msg = p_18721_.apply(of("ask future procesor handle", completablefuture::complete));
+        Msg msg = (Msg)p_18721_.apply(of("ask future procesor handle", completablefuture::complete));
         this.tell(msg);
         return completablefuture;
     }
 
     default <Source> CompletableFuture<Source> askEither(Function<? super ProcessorHandle<Either<Source, Exception>>, ? extends Msg> p_18723_) {
         CompletableFuture<Source> completablefuture = new CompletableFuture<>();
-        Msg msg = p_18723_.apply(of("ask future procesor handle", p_18719_ -> {
+        Msg msg = (Msg)p_18723_.apply(of("ask future procesor handle", p_18719_ -> {
             p_18719_.ifLeft(completablefuture::complete);
             p_18719_.ifRight(completablefuture::completeExceptionally);
         }));
