--- a/net/minecraft/util/thread/ProcessorMailbox.java
+++ b/net/minecraft/util/thread/ProcessorMailbox.java
@@ -42,7 +_,7 @@
             if ((i & 3) != 0) {
                 return false;
             }
-        } while(!this.status.compareAndSet(i, i | 2));
+        } while (!this.status.compareAndSet(i, i | 2));
 
         return true;
     }
@@ -51,15 +_,11 @@
         int i;
         do {
             i = this.status.get();
-        } while(!this.status.compareAndSet(i, i & -3));
+        } while (!this.status.compareAndSet(i, i & -3));
     }
 
     private boolean canBeScheduled() {
-        if ((this.status.get() & 1) != 0) {
-            return false;
-        } else {
-            return !this.queue.isEmpty();
-        }
+        return (this.status.get() & 1) != 0 ? false : !this.queue.isEmpty();
     }
 
     @Override
@@ -67,7 +_,7 @@
         int i;
         do {
             i = this.status.get();
-        } while(!this.status.compareAndSet(i, i | 1));
+        } while (!this.status.compareAndSet(i, i | 1));
     }
 
     private boolean shouldProcess() {
@@ -130,8 +_,8 @@
     private int pollUntil(Int2BooleanFunction p_18748_) {
         int i = 0;
 
-        while(p_18748_.get(i) && this.pollTask()) {
-            ++i;
+        while (p_18748_.get(i) && this.pollTask()) {
+            i++;
         }
 
         return i;
