--- a/net/minecraft/network/syncher/EntityDataAccessor.java
+++ b/net/minecraft/network/syncher/EntityDataAccessor.java
@@ -6,7 +_,7 @@
         if (this == p_135018_) {
             return true;
         } else if (p_135018_ != null && this.getClass() == p_135018_.getClass()) {
-            EntityDataAccessor<?> entitydataaccessor = (EntityDataAccessor)p_135018_;
+            EntityDataAccessor<?> entitydataaccessor = (EntityDataAccessor<?>)p_135018_;
             return this.id == entitydataaccessor.id;
         } else {
             return false;
