--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -27,11 +_,13 @@
     }
 
     public static <T> EntityDataAccessor<T> defineId(Class<? extends SyncedDataHolder> p_135354_, EntityDataSerializer<T> p_135355_) {
-        if (LOGGER.isDebugEnabled()) {
+        if (true || LOGGER.isDebugEnabled()) { // Forge: This is very useful for mods that register keys on classes that are not their own
             try {
                 Class<?> oclass = Class.forName(Thread.currentThread().getStackTrace()[2].getClassName());
                 if (!oclass.equals(p_135354_)) {
-                    LOGGER.debug("defineId called for: {} from {}", p_135354_, oclass, new RuntimeException());
+                    // Forge: log at warn, mods should not add to classes that they don't own, and only add stacktrace when in debug is enabled as it is mostly not needed and consumes time
+                    if (LOGGER.isDebugEnabled()) LOGGER.warn("defineId called for: {} from {}", p_135354_, oclass, new RuntimeException());
+                    else LOGGER.warn("defineId called for: {} from {}", p_135354_, oclass);
                 }
             } catch (ClassNotFoundException classnotfoundexception) {
             }
@@ -79,7 +_,7 @@
             this.isDirty = false;
             List<SynchedEntityData.DataValue<?>> list = new ArrayList<>();
 
-            for(SynchedEntityData.DataItem<?> dataitem : this.itemsById) {
+            for (SynchedEntityData.DataItem<?> dataitem : this.itemsById) {
                 if (dataitem.isDirty()) {
                     dataitem.setDirty(false);
                     list.add(dataitem.value());
@@ -94,7 +_,7 @@
     public List<SynchedEntityData.DataValue<?>> getNonDefaultValues() {
         List<SynchedEntityData.DataValue<?>> list = null;
 
-        for(SynchedEntityData.DataItem<?> dataitem : this.itemsById) {
+        for (SynchedEntityData.DataItem<?> dataitem : this.itemsById) {
             if (!dataitem.isSetToDefault()) {
                 if (list == null) {
                     list = new ArrayList<>();
@@ -108,7 +_,7 @@
     }
 
     public void assignValues(List<SynchedEntityData.DataValue<?>> p_135357_) {
-        for(SynchedEntityData.DataValue<?> datavalue : p_135357_) {
+        for (SynchedEntityData.DataValue<?> datavalue : p_135357_) {
             SynchedEntityData.DataItem<?> dataitem = this.itemsById[datavalue.id];
             this.assignValue(dataitem, datavalue);
             this.entity.onSyncedDataUpdated(dataitem.getAccessor());
@@ -160,7 +_,7 @@
         }
 
         public SynchedEntityData build() {
-            for(int i = 0; i < this.itemsById.length; ++i) {
+            for (int i = 0; i < this.itemsById.length; i++) {
                 if (this.itemsById[i] == null) {
                     throw new IllegalStateException("Entity " + this.entity.getClass() + " has not defined synched data value " + i);
                 }
