--- a/net/minecraft/data/info/BiomeParametersDumpReport.java
+++ b/net/minecraft/data/info/BiomeParametersDumpReport.java
@@ -29,9 +_,7 @@
     private final Path topPath;
     private final CompletableFuture<HolderLookup.Provider> registries;
     private static final MapCodec<ResourceKey<Biome>> ENTRY_CODEC = ResourceKey.codec(Registries.BIOME).fieldOf("biome");
-    private static final Codec<Climate.ParameterList<ResourceKey<Biome>>> CODEC = Climate.ParameterList.<ResourceKey<Biome>>codec(ENTRY_CODEC)
-        .fieldOf("biomes")
-        .codec();
+    private static final Codec<Climate.ParameterList<ResourceKey<Biome>>> CODEC = Climate.ParameterList.codec(ENTRY_CODEC).fieldOf("biomes").codec();
 
     public BiomeParametersDumpReport(PackOutput p_256322_, CompletableFuture<HolderLookup.Provider> p_256222_) {
         this.topPath = p_256322_.getOutputFolder(PackOutput.Target.REPORTS).resolve("biome_parameters");
@@ -46,8 +_,14 @@
                     DynamicOps<JsonElement> dynamicops = p_326734_.createSerializationContext(JsonOps.INSTANCE);
                     List<CompletableFuture<?>> list = new ArrayList<>();
                     MultiNoiseBiomeSourceParameterList.knownPresets()
-                        .forEach((p_274759_, p_274760_) -> list.add(dumpValue(this.createPath(p_274759_.id()), p_254091_, dynamicops, CODEC, p_274760_)));
-                    return CompletableFuture.allOf(list.toArray(p_253398_ -> new CompletableFuture[p_253398_]));
+                        .forEach(
+                            (p_274759_, p_274760_) -> list.add(
+                                    dumpValue(
+                                        this.createPath(p_274759_.id()), p_254091_, dynamicops, CODEC, (Climate.ParameterList<ResourceKey<Biome>>)p_274760_
+                                    )
+                                )
+                        );
+                    return CompletableFuture.allOf(list.toArray(CompletableFuture[]::new));
                 }
             );
     }
