--- a/net/minecraft/data/info/BlockListReport.java
+++ b/net/minecraft/data/info/BlockListReport.java
@@ -42,14 +_,14 @@
                         .forEach(
                             p_330336_ -> {
                                 JsonObject jsonobject1 = new JsonObject();
-                                StateDefinition<Block, BlockState> statedefinition = ((Block)p_330336_.value()).getStateDefinition();
+                                StateDefinition<Block, BlockState> statedefinition = p_330336_.value().getStateDefinition();
                                 if (!statedefinition.getProperties().isEmpty()) {
                                     JsonObject jsonobject2 = new JsonObject();
                 
-                                    for(Property<?> property : statedefinition.getProperties()) {
+                                    for (Property<?> property : statedefinition.getProperties()) {
                                         JsonArray jsonarray = new JsonArray();
                 
-                                        for(Comparable<?> comparable : property.getPossibleValues()) {
+                                        for (Comparable<?> comparable : property.getPossibleValues()) {
                                             jsonarray.add(Util.getPropertyName(property, comparable));
                                         }
                 
@@ -61,11 +_,11 @@
                 
                                 JsonArray jsonarray1 = new JsonArray();
                 
-                                for(BlockState blockstate : statedefinition.getPossibleStates()) {
+                                for (BlockState blockstate : statedefinition.getPossibleStates()) {
                                     JsonObject jsonobject3 = new JsonObject();
                                     JsonObject jsonobject4 = new JsonObject();
                 
-                                    for(Property<?> property1 : statedefinition.getProperties()) {
+                                    for (Property<?> property1 : statedefinition.getProperties()) {
                                         jsonobject4.addProperty(property1.getName(), Util.getPropertyName(property1, blockstate.getValue(property1)));
                                     }
                 
@@ -74,7 +_,7 @@
                                     }
                 
                                     jsonobject3.addProperty("id", Block.getId(blockstate));
-                                    if (blockstate == ((Block)p_330336_.value()).defaultBlockState()) {
+                                    if (blockstate == p_330336_.value().defaultBlockState()) {
                                         jsonobject3.addProperty("default", true);
                                     }
                 
@@ -83,10 +_,12 @@
                 
                                 jsonobject1.add("states", jsonarray1);
                                 String s = p_330336_.getRegisteredName();
-                                JsonElement jsonelement = Util.getOrThrow(
-                                    BlockTypes.CODEC.codec().encodeStart(registryops, (Block)p_330336_.value()),
-                                    p_331670_ -> new AssertionError("Failed to serialize block " + s + " (is type registered in BlockTypes?): " + p_331670_)
-                                );
+                                JsonElement jsonelement = BlockTypes.CODEC
+                                    .codec()
+                                    .encodeStart(registryops, p_330336_.value())
+                                    .getOrThrow(
+                                        p_331670_ -> new AssertionError("Failed to serialize block " + s + " (is type registered in BlockTypes?): " + p_331670_)
+                                    );
                                 jsonobject1.add("definition", jsonelement);
                                 jsonobject.add(s, jsonobject1);
                             }
