--- a/net/minecraft/server/chase/ChaseClient.java
+++ b/net/minecraft/server/chase/ChaseClient.java
@@ -66,14 +_,14 @@
     public void run() {
         String s = this.serverHost + ":" + this.serverPort;
 
-        while(this.wantsToRun) {
+        while (this.wantsToRun) {
             try {
                 LOGGER.info("Connecting to remote control server {}", s);
                 this.socket = new Socket(this.serverHost, this.serverPort);
                 LOGGER.info("Connected to remote control server! Will continuously execute the command broadcasted by that server.");
 
                 try (BufferedReader bufferedreader = new BufferedReader(new InputStreamReader(this.socket.getInputStream(), Charsets.US_ASCII))) {
-                    while(this.wantsToRun) {
+                    while (this.wantsToRun) {
                         String s1 = bufferedreader.readLine();
                         if (s1 == null) {
                             LOGGER.warn("Lost connection to remote control server {}. Will retry in {}s.", s, 5);
