--- a/net/minecraft/server/chase/ChaseServer.java
+++ b/net/minecraft/server/chase/ChaseServer.java
@@ -56,14 +_,14 @@
     private void runSender() {
         ChaseServer.PlayerPosition chaseserver$playerposition = null;
 
-        while(this.wantsToRun) {
+        while (this.wantsToRun) {
             if (!this.clientSockets.isEmpty()) {
                 ChaseServer.PlayerPosition chaseserver$playerposition1 = this.getPlayerPosition();
                 if (chaseserver$playerposition1 != null && !chaseserver$playerposition1.equals(chaseserver$playerposition)) {
                     chaseserver$playerposition = chaseserver$playerposition1;
                     byte[] abyte = chaseserver$playerposition1.format().getBytes(StandardCharsets.US_ASCII);
 
-                    for(Socket socket : this.clientSockets) {
+                    for (Socket socket : this.clientSockets) {
                         if (!socket.isClosed()) {
                             Util.ioPool().submit(() -> {
                                 try {
@@ -100,7 +_,7 @@
 
     private void runAcceptor() {
         try {
-            while(this.wantsToRun) {
+            while (this.wantsToRun) {
                 if (this.serverSocket != null) {
                     LOGGER.info("Remote control server is listening for connections on port {}", this.serverPort);
                     Socket socket = this.serverSocket.accept();
