--- a/net/minecraft/advancements/AdvancementNode.java
+++ b/net/minecraft/advancements/AdvancementNode.java
@@ -37,7 +_,7 @@
     public static AdvancementNode getRoot(AdvancementNode p_300869_) {
         AdvancementNode advancementnode = p_300869_;
 
-        while(true) {
+        while (true) {
             AdvancementNode advancementnode1 = advancementnode.parent();
             if (advancementnode1 == null) {
                 return advancementnode;
