--- a/net/minecraft/advancements/AdvancementProgress.java
+++ b/net/minecraft/advancements/AdvancementProgress.java
@@ -28,20 +_,18 @@
         .xmap(Instant::from, p_300659_ -> p_300659_.atZone(ZoneId.systemDefault()));
     private static final Codec<Map<String, CriterionProgress>> CRITERIA_CODEC = Codec.unboundedMap(Codec.STRING, OBTAINED_TIME_CODEC)
         .xmap(
-            p_300661_ -> p_300661_.entrySet()
-                    .stream()
-                    .collect(Collectors.toMap(Entry::getKey, p_300660_ -> new CriterionProgress((Instant)p_300660_.getValue()))),
+            p_300661_ -> p_300661_.entrySet().stream().collect(Collectors.toMap(Entry::getKey, p_300660_ -> new CriterionProgress(p_300660_.getValue()))),
             p_300663_ -> p_300663_.entrySet()
                     .stream()
-                    .filter(p_300656_ -> ((CriterionProgress)p_300656_.getValue()).isDone())
-                    .collect(Collectors.toMap(Entry::getKey, p_300655_ -> Objects.requireNonNull(((CriterionProgress)p_300655_.getValue()).getObtained())))
+                    .filter(p_300656_ -> p_300656_.getValue().isDone())
+                    .collect(Collectors.toMap(Entry::getKey, p_300655_ -> Objects.requireNonNull(p_300655_.getValue().getObtained())))
         );
     public static final Codec<AdvancementProgress> CODEC = RecordCodecBuilder.create(
-        p_300664_ -> p_300664_.group(
-                    ExtraCodecs.strictOptionalField(CRITERIA_CODEC, "criteria", Map.of()).forGetter(p_300662_ -> p_300662_.criteria),
+        p_337335_ -> p_337335_.group(
+                    CRITERIA_CODEC.optionalFieldOf("criteria", Map.of()).forGetter(p_300662_ -> p_300662_.criteria),
                     Codec.BOOL.fieldOf("done").orElse(true).forGetter(AdvancementProgress::isDone)
                 )
-                .apply(p_300664_, (p_300657_, p_300658_) -> new AdvancementProgress(new HashMap<>(p_300657_)))
+                .apply(p_337335_, (p_300657_, p_300658_) -> new AdvancementProgress(new HashMap<>(p_300657_)))
     );
     private final Map<String, CriterionProgress> criteria;
     private AdvancementRequirements requirements = AdvancementRequirements.EMPTY;
@@ -58,7 +_,7 @@
         Set<String> set = p_301278_.names();
         this.criteria.entrySet().removeIf(p_8203_ -> !set.contains(p_8203_.getKey()));
 
-        for(String s : set) {
+        for (String s : set) {
             this.criteria.putIfAbsent(s, new CriterionProgress());
         }
 
@@ -70,7 +_,7 @@
     }
 
     public boolean hasProgress() {
-        for(CriterionProgress criterionprogress : this.criteria.values()) {
+        for (CriterionProgress criterionprogress : this.criteria.values()) {
             if (criterionprogress.isDone()) {
                 return true;
             }
@@ -155,7 +_,7 @@
     public Iterable<String> getRemainingCriteria() {
         List<String> list = Lists.newArrayList();
 
-        for(Entry<String, CriterionProgress> entry : this.criteria.entrySet()) {
+        for (Entry<String, CriterionProgress> entry : this.criteria.entrySet()) {
             if (!entry.getValue().isDone()) {
                 list.add(entry.getKey());
             }
@@ -167,7 +_,7 @@
     public Iterable<String> getCompletedCriteria() {
         List<String> list = Lists.newArrayList();
 
-        for(Entry<String, CriterionProgress> entry : this.criteria.entrySet()) {
+        for (Entry<String, CriterionProgress> entry : this.criteria.entrySet()) {
             if (entry.getValue().isDone()) {
                 list.add(entry.getKey());
             }
