--- a/net/minecraft/advancements/AdvancementTree.java
+++ b/net/minecraft/advancements/AdvancementTree.java
@@ -22,7 +_,7 @@
     private AdvancementTree.Listener listener;
 
     private void remove(AdvancementNode p_301231_) {
-        for(AdvancementNode advancementnode : p_301231_.children()) {
+        for (AdvancementNode advancementnode : p_301231_.children()) {
             this.remove(advancementnode);
         }
 
@@ -42,7 +_,7 @@
     }
 
     public void remove(Set<ResourceLocation> p_300917_) {
-        for(ResourceLocation resourcelocation : p_300917_) {
+        for (ResourceLocation resourcelocation : p_300917_) {
             AdvancementNode advancementnode = this.nodes.get(resourcelocation);
             if (advancementnode == null) {
                 LOGGER.warn("Told to remove advancement {} but I don't know what that is", resourcelocation);
@@ -55,7 +_,7 @@
     public void addAll(Collection<AdvancementHolder> p_301192_) {
         List<AdvancementHolder> list = new ArrayList<>(p_301192_);
 
-        while(!list.isEmpty()) {
+        while (!list.isEmpty()) {
             if (!list.removeIf(this::tryInsert)) {
                 LOGGER.error("Couldn't load advancements: {}", list);
                 break;
@@ -123,11 +_,11 @@
     public void setListener(@Nullable AdvancementTree.Listener p_301114_) {
         this.listener = p_301114_;
         if (p_301114_ != null) {
-            for(AdvancementNode advancementnode : this.roots) {
+            for (AdvancementNode advancementnode : this.roots) {
                 p_301114_.onAddAdvancementRoot(advancementnode);
             }
 
-            for(AdvancementNode advancementnode1 : this.tasks) {
+            for (AdvancementNode advancementnode1 : this.tasks) {
                 p_301114_.onAddAdvancementTask(advancementnode1);
             }
         }
