--- a/net/minecraft/realms/RepeatedNarrator.java
+++ b/net/minecraft/realms/RepeatedNarrator.java
@@ -20,8 +_,8 @@
     public void narrate(GameNarrator p_240528_, Component p_240604_) {
         RepeatedNarrator.Params repeatednarrator$params = this.params
             .updateAndGet(
-                p_175080_ -> p_175080_ != null && p_240604_.equals(p_175080_.narration)
-                        ? p_175080_
+                p_339420_ -> p_339420_ != null && p_240604_.equals(p_339420_.narration)
+                        ? p_339420_
                         : new RepeatedNarrator.Params(p_240604_, RateLimiter.create((double)this.permitsPerSecond))
             );
         if (repeatednarrator$params.rateLimiter.tryAcquire(1)) {
