--- a/net/minecraft/world/item/component/BlocksAttacks.java
+++ b/net/minecraft/world/item/component/BlocksAttacks.java
@@ -112,14 +_,22 @@
     }
 
     public void hurtBlockingItem(Level level, ItemStack item, LivingEntity user, InteractionHand hand, float damage) {
+        this.hurtBlockingItem(level, item, user, hand, damage, -1);
+    }
+
+    public void hurtBlockingItem(Level level, ItemStack item, LivingEntity user, InteractionHand hand, float damage, int fixedDamage) {
         if (user instanceof Player player) {
             if (!level.isClientSide()) {
                 player.awardStat(Stats.ITEM_USED.get(item.getItem()));
             }
 
-            int i = this.itemDamage.apply(damage);
-            if (i > 0) {
-                item.hurtAndBreak(i, user, hand.asEquipmentSlot());
+            int i = fixedDamage < 0 ? this.itemDamage.apply(damage) : fixedDamage;
+            if (i > 0 && level instanceof ServerLevel serverLevel) {
+                item.hurtAndBreak(i, serverLevel, user, item -> {
+                    user.onEquippedItemBroken(item, hand.asEquipmentSlot());
+                    net.neoforged.neoforge.event.EventHooks.onPlayerDestroyItem(player, player.getUseItem(), hand);
+                    player.stopUsingItem(); // Neo: Fix MC-168573 ("After breaking a shield, the player's off-hand can't finish using some items")
+                });
             }
         }
     }
