--- a/net/minecraft/world/item/component/BundleContents.java
+++ b/net/minecraft/world/item/component/BundleContents.java
@@ -9,52 +_,50 @@
 import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.inventory.Slot;
 import net.minecraft.world.inventory.tooltip.TooltipComponent;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.block.entity.BeehiveBlockEntity;
+import org.apache.commons.lang3.math.Fraction;
 
 public final class BundleContents implements TooltipComponent {
-    public static final int MAX_WEIGHT = 64;
     public static final BundleContents EMPTY = new BundleContents(List.of());
-    public static final Codec<BundleContents> CODEC = ExtraCodecs.sizeLimitedList(ItemStack.CODEC.listOf(), 64)
-        .xmap(BundleContents::new, p_331551_ -> p_331551_.items);
+    public static final Codec<BundleContents> CODEC = ItemStack.CODEC.listOf().xmap(BundleContents::new, p_331551_ -> p_331551_.items);
     public static final StreamCodec<RegistryFriendlyByteBuf, BundleContents> STREAM_CODEC = ItemStack.STREAM_CODEC
-        .apply(ByteBufCodecs.list(64))
+        .apply(ByteBufCodecs.list())
         .map(BundleContents::new, p_331649_ -> p_331649_.items);
-    private static final int BUNDLE_IN_BUNDLE_WEIGHT = 4;
+    private static final Fraction BUNDLE_IN_BUNDLE_WEIGHT = Fraction.getFraction(1, 16);
     private static final int NO_STACK_INDEX = -1;
     final List<ItemStack> items;
-    final int weight;
+    final Fraction weight;
 
-    BundleContents(List<ItemStack> p_331277_, int p_330585_) {
+    BundleContents(List<ItemStack> p_331277_, Fraction p_339622_) {
         this.items = p_331277_;
-        this.weight = p_330585_;
+        this.weight = p_339622_;
     }
 
     public BundleContents(List<ItemStack> p_331417_) {
         this(p_331417_, computeContentWeight(p_331417_));
     }
 
-    private static int computeContentWeight(List<ItemStack> p_331148_) {
-        int i = 0;
+    private static Fraction computeContentWeight(List<ItemStack> p_331148_) {
+        Fraction fraction = Fraction.ZERO;
 
-        for(ItemStack itemstack : p_331148_) {
-            i += getWeight(itemstack) * itemstack.getCount();
+        for (ItemStack itemstack : p_331148_) {
+            fraction = fraction.add(getWeight(itemstack).multiplyBy(Fraction.getFraction(itemstack.getCount(), 1)));
         }
 
-        return i;
+        return fraction;
     }
 
-    static int getWeight(ItemStack p_332084_) {
+    static Fraction getWeight(ItemStack p_332084_) {
         BundleContents bundlecontents = p_332084_.get(DataComponents.BUNDLE_CONTENTS);
         if (bundlecontents != null) {
-            return 4 + bundlecontents.weight();
+            return BUNDLE_IN_BUNDLE_WEIGHT.add(bundlecontents.weight());
         } else {
             List<BeehiveBlockEntity.Occupant> list = p_332084_.getOrDefault(DataComponents.BEES, List.of());
-            return !list.isEmpty() ? 64 : 64 / p_332084_.getMaxStackSize();
+            return !list.isEmpty() ? Fraction.ONE : Fraction.getFraction(1, p_332084_.getMaxStackSize());
         }
     }
 
@@ -70,7 +_,7 @@
         return this.items.size();
     }
 
-    public int weight() {
+    public Fraction weight() {
         return this.weight;
     }
 
@@ -82,11 +_,10 @@
     public boolean equals(Object p_331898_) {
         if (this == p_331898_) {
             return true;
-        } else if (!(p_331898_ instanceof BundleContents)) {
-            return false;
         } else {
-            BundleContents bundlecontents = (BundleContents)p_331898_;
-            return this.weight == bundlecontents.weight && ItemStack.listMatches(this.items, bundlecontents.items);
+            return !(p_331898_ instanceof BundleContents bundlecontents)
+                ? false
+                : this.weight.equals(bundlecontents.weight) && ItemStack.listMatches(this.items, bundlecontents.items);
         }
     }
 
@@ -102,7 +_,7 @@
 
     public static class Mutable {
         private final List<ItemStack> items;
-        private int weight;
+        private Fraction weight;
 
         public Mutable(BundleContents p_332039_) {
             this.items = new ArrayList<>(p_332039_.items);
@@ -113,7 +_,7 @@
             if (!p_331941_.isStackable()) {
                 return -1;
             } else {
-                for(int i = 0; i < this.items.size(); ++i) {
+                for (int i = 0; i < this.items.size(); i++) {
                     if (ItemStack.isSameItemSameComponents(this.items.get(i), p_331941_)) {
                         return i;
                     }
@@ -124,7 +_,8 @@
         }
 
         private int getMaxAmountToAdd(ItemStack p_330527_) {
-            return Math.max(64 - this.weight, 0) / BundleContents.getWeight(p_330527_);
+            Fraction fraction = Fraction.ONE.subtract(this.weight);
+            return Math.max(fraction.divideBy(BundleContents.getWeight(p_330527_)).intValue(), 0);
         }
 
         public int tryInsert(ItemStack p_331789_) {
@@ -133,7 +_,7 @@
                 if (i == 0) {
                     return 0;
                 } else {
-                    this.weight += BundleContents.getWeight(p_331789_) * i;
+                    this.weight = this.weight.add(BundleContents.getWeight(p_331789_).multiplyBy(Fraction.getFraction(i, 1)));
                     int j = this.findStackIndex(p_331789_);
                     if (j != -1) {
                         ItemStack itemstack = this.items.remove(j);
@@ -163,12 +_,12 @@
                 return null;
             } else {
                 ItemStack itemstack = this.items.remove(0).copy();
-                this.weight -= BundleContents.getWeight(itemstack) * itemstack.getCount();
+                this.weight = this.weight.subtract(BundleContents.getWeight(itemstack).multiplyBy(Fraction.getFraction(itemstack.getCount(), 1)));
                 return itemstack;
             }
         }
 
-        public int weight() {
+        public Fraction weight() {
             return this.weight;
         }
 
