--- a/net/minecraft/world/item/component/DyedItemColor.java
+++ b/net/minecraft/world/item/component/DyedItemColor.java
@@ -13,7 +_,6 @@
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.tags.ItemTags;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.FastColor;
 import net.minecraft.world.item.DyeItem;
 import net.minecraft.world.item.ItemStack;
@@ -21,13 +_,13 @@
 
 public record DyedItemColor(int rgb, boolean showInTooltip) implements TooltipProvider {
     private static final Codec<DyedItemColor> FULL_CODEC = RecordCodecBuilder.create(
-        p_331362_ -> p_331362_.group(
+        p_337944_ -> p_337944_.group(
                     Codec.INT.fieldOf("rgb").forGetter(DyedItemColor::rgb),
-                    ExtraCodecs.strictOptionalField(Codec.BOOL, "show_in_tooltip", true).forGetter(DyedItemColor::showInTooltip)
+                    Codec.BOOL.optionalFieldOf("show_in_tooltip", Boolean.valueOf(true)).forGetter(DyedItemColor::showInTooltip)
                 )
-                .apply(p_331362_, DyedItemColor::new)
+                .apply(p_337944_, DyedItemColor::new)
     );
-    public static final Codec<DyedItemColor> CODEC = ExtraCodecs.withAlternative(FULL_CODEC, Codec.INT, p_332619_ -> new DyedItemColor(p_332619_, true));
+    public static final Codec<DyedItemColor> CODEC = Codec.withAlternative(FULL_CODEC, Codec.INT, p_332619_ -> new DyedItemColor(p_332619_, true));
     public static final StreamCodec<ByteBuf, DyedItemColor> STREAM_CODEC = StreamCodec.composite(
         ByteBufCodecs.INT, DyedItemColor::rgb, ByteBufCodecs.BOOL, DyedItemColor::showInTooltip, DyedItemColor::new
     );
@@ -57,10 +_,10 @@
                 i += j1;
                 j += k1;
                 k += l1;
-                ++i1;
+                i1++;
             }
 
-            for(DyeItem dyeitem : p_330568_) {
+            for (DyeItem dyeitem : p_330568_) {
                 float[] afloat = dyeitem.getDyeColor().getTextureDiffuseColors();
                 int i2 = (int)(afloat[0] * 255.0F);
                 int j2 = (int)(afloat[1] * 255.0F);
@@ -69,7 +_,7 @@
                 i += i2;
                 j += j2;
                 k += k2;
-                ++i1;
+                i1++;
             }
 
             int l2 = i / i1;
