--- a/net/minecraft/world/item/component/ResolvableProfile.java
+++ b/net/minecraft/world/item/component/ResolvableProfile.java
@@ -18,14 +_,14 @@
 
 public record ResolvableProfile(Optional<String> name, Optional<UUID> id, PropertyMap properties, GameProfile gameProfile) {
     private static final Codec<ResolvableProfile> FULL_CODEC = RecordCodecBuilder.create(
-        p_332622_ -> p_332622_.group(
-                    ExtraCodecs.strictOptionalField(ExtraCodecs.PLAYER_NAME, "name").forGetter(ResolvableProfile::name),
-                    ExtraCodecs.strictOptionalField(UUIDUtil.CODEC, "id").forGetter(ResolvableProfile::id),
-                    ExtraCodecs.strictOptionalField(ExtraCodecs.PROPERTY_MAP, "properties", new PropertyMap()).forGetter(ResolvableProfile::properties)
+        p_337950_ -> p_337950_.group(
+                    ExtraCodecs.PLAYER_NAME.optionalFieldOf("name").forGetter(ResolvableProfile::name),
+                    UUIDUtil.CODEC.optionalFieldOf("id").forGetter(ResolvableProfile::id),
+                    ExtraCodecs.PROPERTY_MAP.optionalFieldOf("properties", new PropertyMap()).forGetter(ResolvableProfile::properties)
                 )
-                .apply(p_332622_, ResolvableProfile::new)
+                .apply(p_337950_, ResolvableProfile::new)
     );
-    public static final Codec<ResolvableProfile> CODEC = ExtraCodecs.withAlternative(
+    public static final Codec<ResolvableProfile> CODEC = Codec.withAlternative(
         FULL_CODEC, ExtraCodecs.PLAYER_NAME, p_332623_ -> new ResolvableProfile(Optional.of(p_332623_), Optional.empty(), new PropertyMap())
     );
     public static final StreamCodec<ByteBuf, ResolvableProfile> STREAM_CODEC = StreamCodec.composite(
@@ -47,12 +_,17 @@
     }
 
     public CompletableFuture<ResolvableProfile> resolve() {
-        return this.isResolved()
-            ? CompletableFuture.completedFuture(this)
-            : SkullBlockEntity.fetchGameProfile(this.name.orElseThrow()).thenApply(p_332081_ -> {
-                GameProfile gameprofile = p_332081_.orElseGet(() -> new GameProfile(Util.NIL_UUID, this.name.get()));
+        if (this.isResolved()) {
+            return CompletableFuture.completedFuture(this);
+        } else {
+            return this.id.isPresent() ? SkullBlockEntity.fetchGameProfile(this.id.get()).thenApply(p_332081_ -> {
+                GameProfile gameprofile = p_332081_.orElseGet(() -> new GameProfile(this.id.get(), this.name.orElse("")));
+                return new ResolvableProfile(gameprofile);
+            }) : SkullBlockEntity.fetchGameProfile(this.name.orElseThrow()).thenApply(p_339530_ -> {
+                GameProfile gameprofile = p_339530_.orElseGet(() -> new GameProfile(Util.NIL_UUID, this.name.get()));
                 return new ResolvableProfile(gameprofile);
             });
+        }
     }
 
     private static GameProfile createProfile(Optional<String> p_331092_, Optional<UUID> p_332678_, PropertyMap p_331544_) {
@@ -62,6 +_,6 @@
     }
 
     public boolean isResolved() {
-        return this.id.isPresent() || !this.properties.isEmpty() || this.name.isEmpty();
+        return !this.properties.isEmpty() ? true : this.id.isPresent() == this.name.isPresent();
     }
 }
