--- a/net/minecraft/world/item/component/SeededContainerLoot.java
+++ b/net/minecraft/world/item/component/SeededContainerLoot.java
@@ -5,15 +_,14 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.resources.ResourceKey;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.storage.loot.LootTable;
 
 public record SeededContainerLoot(ResourceKey<LootTable> lootTable, long seed) {
     public static final Codec<SeededContainerLoot> CODEC = RecordCodecBuilder.create(
-        p_335288_ -> p_335288_.group(
+        p_337951_ -> p_337951_.group(
                     ResourceKey.codec(Registries.LOOT_TABLE).fieldOf("loot_table").forGetter(SeededContainerLoot::lootTable),
-                    ExtraCodecs.strictOptionalField(Codec.LONG, "seed", 0L).forGetter(SeededContainerLoot::seed)
+                    Codec.LONG.optionalFieldOf("seed", Long.valueOf(0L)).forGetter(SeededContainerLoot::seed)
                 )
-                .apply(p_335288_, SeededContainerLoot::new)
+                .apply(p_337951_, SeededContainerLoot::new)
     );
 }
