--- a/net/minecraft/world/item/component/Tool.java
+++ b/net/minecraft/world/item/component/Tool.java
@@ -20,12 +_,12 @@
 
 public record Tool(List<Tool.Rule> rules, float defaultMiningSpeed, int damagePerBlock) {
     public static final Codec<Tool> CODEC = RecordCodecBuilder.create(
-        p_335820_ -> p_335820_.group(
+        p_337953_ -> p_337953_.group(
                     Tool.Rule.CODEC.listOf().fieldOf("rules").forGetter(Tool::rules),
-                    ExtraCodecs.strictOptionalField(Codec.FLOAT, "default_mining_speed", 1.0F).forGetter(Tool::defaultMiningSpeed),
-                    ExtraCodecs.strictOptionalField(ExtraCodecs.NON_NEGATIVE_INT, "damage_per_block", 1).forGetter(Tool::damagePerBlock)
+                    Codec.FLOAT.optionalFieldOf("default_mining_speed", Float.valueOf(1.0F)).forGetter(Tool::defaultMiningSpeed),
+                    ExtraCodecs.NON_NEGATIVE_INT.optionalFieldOf("damage_per_block", 1).forGetter(Tool::damagePerBlock)
                 )
-                .apply(p_335820_, Tool::new)
+                .apply(p_337953_, Tool::new)
     );
     public static final StreamCodec<RegistryFriendlyByteBuf, Tool> STREAM_CODEC = StreamCodec.composite(
         Tool.Rule.STREAM_CODEC.apply(ByteBufCodecs.list()),
@@ -38,7 +_,7 @@
     );
 
     public float getMiningSpeed(BlockState p_336131_) {
-        for(Tool.Rule tool$rule : this.rules) {
+        for (Tool.Rule tool$rule : this.rules) {
             if (tool$rule.speed.isPresent() && p_336131_.is(tool$rule.blocks)) {
                 return tool$rule.speed.get();
             }
@@ -48,7 +_,7 @@
     }
 
     public boolean isCorrectForDrops(BlockState p_336189_) {
-        for(Tool.Rule tool$rule : this.rules) {
+        for (Tool.Rule tool$rule : this.rules) {
             if (tool$rule.correctForDrops.isPresent() && p_336189_.is(tool$rule.blocks)) {
                 return tool$rule.correctForDrops.get();
             }
@@ -59,12 +_,12 @@
 
     public static record Rule(HolderSet<Block> blocks, Optional<Float> speed, Optional<Boolean> correctForDrops) {
         public static final Codec<Tool.Rule> CODEC = RecordCodecBuilder.create(
-            p_335834_ -> p_335834_.group(
+            p_337954_ -> p_337954_.group(
                         RegistryCodecs.homogeneousList(Registries.BLOCK).fieldOf("blocks").forGetter(Tool.Rule::blocks),
-                        ExtraCodecs.strictOptionalField(ExtraCodecs.POSITIVE_FLOAT, "speed").forGetter(Tool.Rule::speed),
-                        ExtraCodecs.strictOptionalField(Codec.BOOL, "correct_for_drops").forGetter(Tool.Rule::correctForDrops)
+                        ExtraCodecs.POSITIVE_FLOAT.optionalFieldOf("speed").forGetter(Tool.Rule::speed),
+                        Codec.BOOL.optionalFieldOf("correct_for_drops").forGetter(Tool.Rule::correctForDrops)
                     )
-                    .apply(p_335834_, Tool.Rule::new)
+                    .apply(p_337954_, Tool.Rule::new)
         );
         public static final StreamCodec<RegistryFriendlyByteBuf, Tool.Rule> STREAM_CODEC = StreamCodec.composite(
             ByteBufCodecs.holderSet(Registries.BLOCK),
