--- a/net/minecraft/world/item/component/Unbreakable.java
+++ b/net/minecraft/world/item/component/Unbreakable.java
@@ -9,13 +_,12 @@
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.item.TooltipFlag;
 
 public record Unbreakable(boolean showInTooltip) implements TooltipProvider {
     public static final Codec<Unbreakable> CODEC = RecordCodecBuilder.create(
-        p_330469_ -> p_330469_.group(ExtraCodecs.strictOptionalField(Codec.BOOL, "show_in_tooltip", true).forGetter(Unbreakable::showInTooltip))
-                .apply(p_330469_, Unbreakable::new)
+        p_337955_ -> p_337955_.group(Codec.BOOL.optionalFieldOf("show_in_tooltip", Boolean.valueOf(true)).forGetter(Unbreakable::showInTooltip))
+                .apply(p_337955_, Unbreakable::new)
     );
     public static final StreamCodec<ByteBuf, Unbreakable> STREAM_CODEC = ByteBufCodecs.BOOL.map(Unbreakable::new, Unbreakable::showInTooltip);
     private static final Component TOOLTIP = Component.translatable("item.unbreakable").withStyle(ChatFormatting.BLUE);
