--- a/net/minecraft/world/item/component/WritableBookContent.java
+++ b/net/minecraft/world/item/component/WritableBookContent.java
@@ -9,16 +_,15 @@
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.server.network.Filterable;
-import net.minecraft.util.ExtraCodecs;
 
 public record WritableBookContent(List<Filterable<String>> pages) implements BookContent<String, WritableBookContent> {
     public static final WritableBookContent EMPTY = new WritableBookContent(List.of());
     public static final int PAGE_EDIT_LENGTH = 1024;
-    private static final Codec<Filterable<String>> PAGE_CODEC = Filterable.codec(ExtraCodecs.sizeLimitedString(0, 1024));
-    public static final Codec<List<Filterable<String>>> PAGES_CODEC = ExtraCodecs.sizeLimitedList(PAGE_CODEC.listOf(), 100);
+    private static final Codec<Filterable<String>> PAGE_CODEC = Filterable.codec(Codec.string(0, 1024));
+    public static final Codec<List<Filterable<String>>> PAGES_CODEC = PAGE_CODEC.sizeLimitedListOf(100);
     public static final Codec<WritableBookContent> CODEC = RecordCodecBuilder.create(
-        p_330710_ -> p_330710_.group(ExtraCodecs.strictOptionalField(PAGES_CODEC, "pages", List.of()).forGetter(WritableBookContent::pages))
-                .apply(p_330710_, WritableBookContent::new)
+        p_337956_ -> p_337956_.group(PAGES_CODEC.optionalFieldOf("pages", List.of()).forGetter(WritableBookContent::pages))
+                .apply(p_337956_, WritableBookContent::new)
     );
     public static final StreamCodec<ByteBuf, WritableBookContent> STREAM_CODEC = Filterable.streamCodec(ByteBufCodecs.stringUtf8(1024))
         .apply(ByteBufCodecs.list(100))
