--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -2125,6 +2136,8 @@
         } else {
             ItemEntity entity = new ItemEntity(level, this.getX() + offset.x, this.getY() + offset.y, this.getZ() + offset.z, itemStack);
             entity.setDefaultPickUpDelay();
+            if (captureDrops() != null) captureDrops().add(entity);
+            else
             level.addFreshEntity(entity);
             return entity;
         }
