--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1176,11 +1214,18 @@
         } else if (source.is(DamageTypeTags.IS_FIRE) && this.hasEffect(MobEffects.FIRE_RESISTANCE)) {
             return false;
         } else {
+            this.damageContainers.push(new net.neoforged.neoforge.common.damagesource.DamageContainer(source, damage));
+            if (net.neoforged.neoforge.common.CommonHooks.onEntityIncomingDamage(this, this.damageContainers.peek())) {
+                this.damageContainers.pop();
+                return false;
+            }
+
             if (this.isSleeping()) {
                 this.stopSleeping();
             }
             this.noActionTime = 0;
+            damage = this.damageContainers.peek().getNewDamage(); // Neo: enforce damage container as source of truth for damage amount
             if (damage < 0.0F) {
                 damage = 0.0F;
             }
@@ -1201,24 +1246,38 @@
             if (Float.isNaN(damage) || Float.isInfinite(damage)) {
                 damage = Float.MAX_VALUE;
             }
+            this.damageContainers.peek().setNewDamage(damage); // Update container with vanilla changes
             boolean tookFullDamage = true;
             if (this.invulnerableTime > 10.0F && !source.is(DamageTypeTags.BYPASSES_COOLDOWN)) {
                 if (damage <= this.lastHurt) {
+                    // Neo: Pop the container when vanilla would abort the incoming damage due to the invulnerability ticks.
+                    this.damageContainers.pop();
                     return false;
                 }
+                // If we are going to do damage despite invulnerability (due to the damage being higher than prior), we need to setup the container differently.
+                this.damageContainers.peek().setPostAttackInvulnerabilityTicks(this.invulnerableTime);
+                this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.INVULNERABILITY, this.lastHurt);
                 this.actuallyHurt(level, source, damage - this.lastHurt);
-                this.lastHurt = damage;
+                // Vanilla updates lastHurt to the local var `damage`, but we throw an event in actuallyHurt that might mess with the damage.
+                // Also, since we're in the "damage while invuln is active" block, we need to make sure we don't reduce the lastHurt amount.
+                this.lastHurt = java.lang.Math.max(this.lastHurt, this.damageContainers.peek().getInflictedDamage());
+                this.invulnerableTime = this.damageContainers.peek().getPostAttackInvulnerabilityTicks();
                 tookFullDamage = false;
             } else {
-                this.lastHurt = damage;
-                this.invulnerableTime = 20;
                 this.actuallyHurt(level, source, damage);
+                this.lastHurt = this.damageContainers.peek().getInflictedDamage();
+                this.invulnerableTime = this.damageContainers.peek().getPostAttackInvulnerabilityTicks();
                 this.hurtDuration = 10;
                 this.hurtTime = this.hurtDuration;
             }
+            // Neo: Note that we don't update the local tookFullDamage.
+            //      Its name doesn't really mean that, and it is instead used to track if we're inside the invuln block above.
+            //      Also, we need to update the `damage` local with the post-Pre-event value.
+            damage = this.damageContainers.peek().getInflictedDamage();
+
             this.resolveMobResponsibleForDamage(source);
             this.resolvePlayerResponsibleForDamage(source);
             if (tookFullDamage) {
@@ -1287,6 +1346,10 @@
                 CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(sourcePlayer, this, source, damage, damage, blocked);
             }
+            // Neo: Fire our post-damage event hooks. As usual, fire the extension method first, then the event version.
+            this.onDamageTaken(this.damageContainers.peek());
+            net.neoforged.neoforge.common.CommonHooks.onLivingDamagePost(this, this.damageContainers.peek());
+            this.damageContainers.pop();
             return success;
         }
     }
@@ -1316,7 +1379,13 @@
                         }
                         float damageBlocked = blocksAttacks.resolveBlockedDamage(source, damage, angle);
-                        blocksAttacks.hurtBlockingItem(this.level(), blockingWith, this, this.getUsedItemHand(), damageBlocked);
+
+                        var ev = net.neoforged.neoforge.common.CommonHooks.onDamageBlock(this, this.damageContainers.peek(), damageBlocked, !blocksAttacks.bypassedBy().map(t -> t.contains(source.typeHolder())).orElse(false));
+                        if (!ev.getBlocked()) return 0.0F;
+
+                        damageBlocked = ev.getBlockedDamage();
+                        this.damageContainers.peek().setBlockedDamage(ev);
+                        blocksAttacks.hurtBlockingItem(this.level(), blockingWith, this, this.getUsedItemHand(), damageBlocked, ev.shieldDamage());
                         if (damageBlocked > 0.0F && !source.is(DamageTypeTags.IS_PROJECTILE) && source.getDirectEntity() instanceof LivingEntity livingEntity) {
                             this.blockUsingItem(level, livingEntity);
                         }
@@ -1437,6 +1506,8 @@
     public void die(DamageSource source) {
         if (!this.isRemoved() && !this.dead) {
+            // Neo: Fire the LivingDeathEvent after validating that the entity isn't already dead, but before trying to kill it.
+            if (net.neoforged.neoforge.common.CommonHooks.onLivingDeath(this, source)) return;
             Entity sourceEntity = source.getEntity();
             LivingEntity killer = this.getKillCredit();
@@ -1489,6 +1560,7 @@
     }
     protected void dropAllDeathLoot(ServerLevel level, DamageSource source) {
+        this.captureDrops(new java.util.ArrayList<>());
         boolean playerKilled = this.lastHurtByPlayerMemoryTime > 0;
         if (this.shouldDropLoot(level)) {
             this.dropFromLootTable(level, source, playerKilled);
@@ -1497,6 +1569,10 @@
         this.dropEquipment(level);
         this.dropExperience(level, source.getEntity());
+
+        Collection<LivingBlock> drops = captureDrops(null);
+        if (!net.neoforged.neoforge.common.CommonHooks.onLivingDrops(this, source, drops, lastHurtByPlayerMemoryTime > 0))
+            drops.forEach(e -> level().addFreshEntity(e));
     }
     protected void dropEquipment(ServerLevel level) {
@@ -1508,7 +1584,8 @@
                 this.isAlwaysExperienceDropper()
                     || this.lastHurtByPlayerMemoryTime > 0 && this.shouldDropExperience() && level.getGameRules().get(GameRules.MOB_DROPS)
             )) {
-            ExperienceOrb.award(level, this.position(), this.getExperienceReward(level, killer));
+            int reward = net.neoforged.neoforge.event.EventHooks.getExperienceDrop(this, this.getLastHurtByPlayer(), this.getExperienceReward(level, killer));
+            ExperienceOrb.award((ServerLevel) this.level(), this.position(), reward);
         }
     }
@@ -1609,6 +1686,11 @@
     @Override
     public void knockback(double power, double xd, double zd) {
+        net.neoforged.neoforge.event.entity.living.LivingKnockBackEvent event = net.neoforged.neoforge.common.CommonHooks.onLivingKnockBack(this, (float) power, xd, zd);
+        if(event.isCanceled()) return;
+        power = event.getStrength();
+        xd = event.getRatioX();
+        zd = event.getRatioZ();
         power *= 1.0 - this.getAttributeValue(Attributes.KNOCKBACK_RESISTANCE);
         if (!(power <= 0.0)) {
             this.needsSync = true;
@@ -1691,17 +1773,11 @@
         } else {
             BlockPos ladderCheckPos = this.blockPosition();
             BlockState state = this.getInBlockState();
-            if (this.isFallFlying() && state.is(BlockTags.CAN_GLIDE_THROUGH)) {
-                return false;
-            } else if (state.is(BlockTags.CLIMBABLE)) {
-                this.lastClimbablePos = Optional.of(ladderCheckPos);
-                return true;
-            } else if (state.getBlock() instanceof TrapDoorBlock && this.trapdoorUsableAsLadder(ladderCheckPos, state)) {
-                this.lastClimbablePos = Optional.of(ladderCheckPos);
-                return true;
-            } else {
-                return false;
-            }
+            if (this.isFallFlying() && state.is(BlockTags.CAN_GLIDE_THROUGH))
+                return false;
+            Optional<BlockPos> ladderPos = net.neoforged.neoforge.common.CommonHooks.isLivingOnLadder(state, level(), ladderCheckPos, this);
+            if (ladderPos.isPresent()) this.lastClimbablePos = ladderPos;
+            return ladderPos.isPresent();
         }
     }
@@ -1762,6 +1838,11 @@
             effectiveFallDistance = fallDistance;
         }
+        var event = net.neoforged.neoforge.common.CommonHooks.onLivingFall(this, effectiveFallDistance, damageModifier);
+        if (event.isCanceled()) return false;
+        effectiveFallDistance = event.getDistance();
+        damageModifier = event.getDamageMultiplier();
+
         boolean damaged = super.causeFallDamage(effectiveFallDistance, damageModifier, damageSource);
         int dmg = this.calculateFallDamage(effectiveFallDistance, damageModifier);
         if (dmg > 0) {
@@ -1825,15 +1906,15 @@
         playBlockFallSoundForEntity(this);
     }
-    public static void playBlockFallSoundForEntity(Entity entity) {
+    public static void playBlockFallSoundForEntity(LivingEntity entity) {
         if (!entity.isSilent()) {
             int xx = Mth.floor(entity.getX());
             int yy = Mth.floor(entity.getY() - 0.2F);
             int zz = Mth.floor(entity.getZ());
-            BlockState state = entity.level().getBlockState(new BlockPos(xx, yy, zz));
+            var pos = new BlockPos(xx, yy, zz);
+            BlockState state = entity.level().getBlockState(pos);
             if (!state.isAir()) {
-                SoundType soundType = state.getSoundType();
-                entity.playSound(soundType.getFallSound(), soundType.getVolume() * 0.5F, soundType.getPitch() * 0.75F);
+                state.playFallSound(entity.level(), pos, entity);
             }
         }
     }
@@ -1858,6 +1939,8 @@
         if (!(damage <= 0.0F)) {
             int durabilityDamage = (int)Math.max(1.0F, damage / 4.0F);
+            net.neoforged.neoforge.common.CommonHooks.onArmorHurt(damageSource, slots, durabilityDamage, this);
+            if (true) return; //Neo: Invalidates the loop. Armor damage happens in common hook.
             for (EquipmentSlot slot : slots) {
                 ItemStack itemStack = this.getItemBySlot(slot);
                 Equippable equippable = itemStack.get(DataComponents.EQUIPPABLE);
@@ -1913,6 +1997,7 @@
                 if (enchantmentArmor > 0.0F) {
                     damage = CombatRules.getDamageAfterMagicAbsorb(damage, enchantmentArmor);
+                    this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ENCHANTMENTS,this.damageContainers.peek().getNewDamage() - damage);
                 }
                 return damage;
@@ -1922,21 +2007,37 @@
     protected void actuallyHurt(ServerLevel level, DamageSource source, float dmg) {
         if (!this.isInvulnerableTo(level, source)) {
-            dmg = this.getDamageAfterArmorAbsorb(source, dmg);
-            dmg = this.getDamageAfterMagicAbsorb(source, dmg);
-            float var10 = Math.max(dmg - this.getAbsorptionAmount(), 0.0F);
-            this.setAbsorptionAmount(this.getAbsorptionAmount() - (dmg - var10));
-            float absorbedDamage = dmg - var10;
+            // Neo (ImplNote): Keep this in-sync with the equivalent patches in Player#actuallyHurt.
+            // Neo: Primary damage pipeline hooks. Compute the armor and enchantment reductions against the "original" damage, and then fire LivingDamageEvent.Pre
+            this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ARMOR, this.damageContainers.peek().getNewDamage() - this.getDamageAfterArmorAbsorb(source, this.damageContainers.peek().getNewDamage()));
+            this.getDamageAfterMagicAbsorb(source, this.damageContainers.peek().getNewDamage());
+
+            // LivingDamageEvent.Pre is then able to mutate the container to change the reductions and change the "real" damage that will be inflicted.
+            float damage = net.neoforged.neoforge.common.CommonHooks.onLivingDamagePre(this, this.damageContainers.peek());
+            // Throw an exception if someone kills the entity during Pre. Vanilla is going to kill any entity which isDeadOrDying() when it bubbles back up to hurtServer, so it dying early is invalid.
+            com.google.common.base.Preconditions.checkArgument(!this.dead, "It is invalid to have killed an entity during LivingDamageEvent.Pre. Use LivingIncomingDamageEvent or LivingDamageEvent.Post instead.");
+
+            // Store a copy of the "true" inflicted damage so we can propagate it back to hurtServer.
+            this.damageContainers.peek().captureInflictedDamage();
+
+            // Next we attribute damage to absorption hearts. We record the amount blocked by absorption in the container for later calculations.
+            // Mods reduction functions may adjust the split of damage between absorption and health, so we need to first apply the reductions and then compute the amounts that go to each pool.
+            this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ABSORPTION, Math.min(this.getAbsorptionAmount(), damage));
+            float absorbedDamage = Math.min(damage, this.damageContainers.peek().getReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ABSORPTION));
+            this.setAbsorptionAmount(Math.max(0, this.getAbsorptionAmount() - absorbedDamage));
+            float var10 = this.damageContainers.peek().getNewDamage();
             if (absorbedDamage > 0.0F && absorbedDamage < 3.4028235E37F && source.getEntity() instanceof ServerPlayer serverPlayer) {
                 serverPlayer.awardStat(Stats.DAMAGE_DEALT_ABSORBED, Math.round(absorbedDamage * 10.0F));
             }
+            // Last thing here, inflict damage to the entity's health, based on how much damage is remaining after absorption has been processes.
             if (var10 != 0.0F) {
                 this.getCombatTracker().recordDamage(source, var10);
                 this.setHealth(this.getHealth() - var10);
-                this.setAbsorptionAmount(this.getAbsorptionAmount() - var10);
                 this.gameEvent(GameEvent.ENTITY_DAMAGE);
             }
+
+            // Neo: Now we go back to hurtServer, where we'll update various metadata fields, process the side effects of damage (i.e. killing the entity), and firing LivingDamageEvent.Post
         }
     }
 
