--- a/net/minecraft/world/entity/monster/warden/AngerLevel.java
+++ b/net/minecraft/world/entity/monster/warden/AngerLevel.java
@@ -36,7 +_,7 @@
     }
 
     public static AngerLevel byAnger(int p_219228_) {
-        for(AngerLevel angerlevel : SORTED_LEVELS) {
+        for (AngerLevel angerlevel : SORTED_LEVELS) {
             if (p_219228_ >= angerlevel.minimumAnger) {
                 return angerlevel;
             }
