--- a/net/minecraft/world/entity/monster/warden/WardenAi.java
+++ b/net/minecraft/world/entity/monster/warden/WardenAi.java
@@ -84,14 +_,13 @@
         MemoryModuleType.SONIC_BOOM_SOUND_DELAY
     );
     private static final BehaviorControl<Warden> DIG_COOLDOWN_SETTER = BehaviorBuilder.create(
-        p_258953_ -> p_258953_.group(p_258953_.registered(MemoryModuleType.DIG_COOLDOWN))
-                .apply(p_258953_, p_258960_ -> (p_258956_, p_258957_, p_258958_) -> {
-                        if (p_258953_.tryGet(p_258960_).isPresent()) {
-                            p_258960_.setWithExpiry(Unit.INSTANCE, 1200L);
-                        }
-        
-                        return true;
-                    })
+        p_258953_ -> p_258953_.group(p_258953_.registered(MemoryModuleType.DIG_COOLDOWN)).apply(p_258953_, p_258960_ -> (p_258956_, p_258957_, p_258958_) -> {
+                    if (p_258953_.tryGet(p_258960_).isPresent()) {
+                        p_258960_.setWithExpiry(Unit.INSTANCE, 1200L);
+                    }
+
+                    return true;
+                })
     );
 
     public static void updateActivity(Warden p_219513_) {
