--- a/net/minecraft/world/entity/monster/warden/Warden.java
+++ b/net/minecraft/world/entity/monster/warden/Warden.java
@@ -237,8 +_,7 @@
 
     @Override
     public void tick() {
-        Level level = this.level();
-        if (level instanceof ServerLevel serverlevel) {
+        if (this.level() instanceof ServerLevel serverlevel) {
             VibrationSystem.Ticker.tick(serverlevel, this.vibrationData, this.vibrationUser);
             if (this.isPersistenceRequired() || this.requiresCustomPersistence()) {
                 WardenAi.setDigCooldown(this);
@@ -259,15 +_,15 @@
 
             this.tendrilAnimationO = this.tendrilAnimation;
             if (this.tendrilAnimation > 0) {
-                --this.tendrilAnimation;
+                this.tendrilAnimation--;
             }
 
             this.heartAnimationO = this.heartAnimation;
             if (this.heartAnimation > 0) {
-                --this.heartAnimation;
+                this.heartAnimation--;
             }
 
-            switch(this.getPose()) {
+            switch (this.getPose()) {
                 case EMERGING:
                     this.clientDiggingParticles(this.emergeAnimationState);
                     break;
@@ -328,7 +_,7 @@
             RandomSource randomsource = this.getRandom();
             BlockState blockstate = this.getBlockStateOn();
             if (blockstate.getRenderShape() != RenderShape.INVISIBLE) {
-                for(int i = 0; i < 30; ++i) {
+                for (int i = 0; i < 30; i++) {
                     double d0 = this.getX() + (double)Mth.randomBetween(randomsource, -0.7F, 0.7F);
                     double d1 = this.getY();
                     double d2 = this.getZ() + (double)Mth.randomBetween(randomsource, -0.7F, 0.7F);
@@ -341,7 +_,7 @@
     @Override
     public void onSyncedDataUpdated(EntityDataAccessor<?> p_219422_) {
         if (DATA_POSE.equals(p_219422_)) {
-            switch(this.getPose()) {
+            switch (this.getPose()) {
                 case EMERGING:
                     this.emergeAnimationState.start(this.tickCount);
                     break;
@@ -382,8 +_,7 @@
 
     @Override
     public void updateDynamicGameEventListener(BiConsumer<DynamicGameEventListener<?>, ServerLevel> p_219413_) {
-        Level level = this.level();
-        if (level instanceof ServerLevel serverlevel) {
+        if (this.level() instanceof ServerLevel serverlevel) {
             p_219413_.accept(this.dynamicGameEventListener, serverlevel);
         }
     }
@@ -616,8 +_,7 @@
                 && !Warden.this.getBrain().hasMemoryValue(MemoryModuleType.VIBRATION_COOLDOWN)
                 && !Warden.this.isDiggingOrEmerging()
                 && p_282574_.getWorldBorder().isWithinBounds(p_282323_)) {
-                Entity entity = p_282515_.sourceEntity();
-                if (entity instanceof LivingEntity livingentity && !Warden.this.canTargetEntity(livingentity)) {
+                if (p_282515_.sourceEntity() instanceof LivingEntity livingentity && !Warden.this.canTargetEntity(livingentity)) {
                     return false;
                 }
 
