--- a/net/minecraft/world/entity/monster/warden/WardenSpawnTracker.java
+++ b/net/minecraft/world/entity/monster/warden/WardenSpawnTracker.java
@@ -47,11 +_,11 @@
             this.decreaseWarningLevel();
             this.ticksSinceLastWarning = 0;
         } else {
-            ++this.ticksSinceLastWarning;
+            this.ticksSinceLastWarning++;
         }
 
         if (this.cooldownTicks > 0) {
-            --this.cooldownTicks;
+            this.cooldownTicks--;
         }
     }
 
@@ -99,7 +_,7 @@
 
     private static List<ServerPlayer> getNearbyPlayers(ServerLevel p_219595_, BlockPos p_219596_) {
         Vec3 vec3 = Vec3.atCenterOf(p_219596_);
-        Predicate<ServerPlayer> predicate = p_340155_ -> p_340155_.position().closerThan(vec3, 16.0);
+        Predicate<ServerPlayer> predicate = p_337873_ -> p_337873_.position().closerThan(vec3, 16.0);
         return p_219595_.getPlayers(predicate.and(LivingEntity::isAlive).and(EntitySelector.NO_SPECTATORS));
     }
 
