--- a/net/minecraft/world/level/levelgen/structure/templatesystem/AlwaysTrueTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/AlwaysTrueTest.java
@@ -1,11 +_,11 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class AlwaysTrueTest extends RuleTest {
-    public static final Codec<AlwaysTrueTest> CODEC = Codec.unit(() -> AlwaysTrueTest.INSTANCE);
+    public static final MapCodec<AlwaysTrueTest> CODEC = MapCodec.unit(() -> AlwaysTrueTest.INSTANCE);
     public static final AlwaysTrueTest INSTANCE = new AlwaysTrueTest();
 
     private AlwaysTrueTest() {
