--- a/net/minecraft/world/level/levelgen/structure/templatesystem/BlockIgnoreProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/BlockIgnoreProcessor.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
 import com.google.common.collect.ImmutableList;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
@@ -12,12 +_,11 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class BlockIgnoreProcessor extends StructureProcessor {
-    public static final Codec<BlockIgnoreProcessor> CODEC = BlockState.CODEC
+    public static final MapCodec<BlockIgnoreProcessor> CODEC = BlockState.CODEC
         .xmap(BlockBehaviour.BlockStateBase::getBlock, Block::defaultBlockState)
         .listOf()
         .fieldOf("blocks")
-        .xmap(BlockIgnoreProcessor::new, p_74062_ -> p_74062_.toIgnore)
-        .codec();
+        .xmap(BlockIgnoreProcessor::new, p_74062_ -> p_74062_.toIgnore);
     public static final BlockIgnoreProcessor STRUCTURE_BLOCK = new BlockIgnoreProcessor(ImmutableList.of(Blocks.STRUCTURE_BLOCK));
     public static final BlockIgnoreProcessor AIR = new BlockIgnoreProcessor(ImmutableList.of(Blocks.AIR));
     public static final BlockIgnoreProcessor STRUCTURE_AND_AIR = new BlockIgnoreProcessor(ImmutableList.of(Blocks.AIR, Blocks.STRUCTURE_BLOCK));
