--- a/net/minecraft/world/level/levelgen/structure/templatesystem/BlockStateMatchTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/BlockStateMatchTest.java
@@ -1,14 +_,13 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class BlockStateMatchTest extends RuleTest {
-    public static final Codec<BlockStateMatchTest> CODEC = BlockState.CODEC
+    public static final MapCodec<BlockStateMatchTest> CODEC = BlockState.CODEC
         .fieldOf("block_state")
-        .xmap(BlockStateMatchTest::new, p_74099_ -> p_74099_.blockState)
-        .codec();
+        .xmap(BlockStateMatchTest::new, p_74099_ -> p_74099_.blockState);
     private final BlockState blockState;
 
     public BlockStateMatchTest(BlockState p_74093_) {
