--- a/net/minecraft/world/level/levelgen/structure/templatesystem/BlockMatchTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/BlockMatchTest.java
@@ -1,17 +_,16 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class BlockMatchTest extends RuleTest {
-    public static final Codec<BlockMatchTest> CODEC = BuiltInRegistries.BLOCK
+    public static final MapCodec<BlockMatchTest> CODEC = BuiltInRegistries.BLOCK
         .byNameCodec()
         .fieldOf("block")
-        .xmap(BlockMatchTest::new, p_74073_ -> p_74073_.block)
-        .codec();
+        .xmap(BlockMatchTest::new, p_74073_ -> p_74073_.block);
     private final Block block;
 
     public BlockMatchTest(Block p_74067_) {
