--- a/net/minecraft/world/level/levelgen/structure/templatesystem/JigsawReplacementProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/JigsawReplacementProcessor.java
@@ -2,7 +_,7 @@
 
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
 import com.mojang.logging.LogUtils;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import javax.annotation.Nullable;
 import net.minecraft.commands.arguments.blocks.BlockStateParser;
 import net.minecraft.core.BlockPos;
@@ -14,7 +_,7 @@
 
 public class JigsawReplacementProcessor extends StructureProcessor {
     private static final Logger LOGGER = LogUtils.getLogger();
-    public static final Codec<JigsawReplacementProcessor> CODEC = Codec.unit(() -> JigsawReplacementProcessor.INSTANCE);
+    public static final MapCodec<JigsawReplacementProcessor> CODEC = MapCodec.unit(() -> JigsawReplacementProcessor.INSTANCE);
     public static final JigsawReplacementProcessor INSTANCE = new JigsawReplacementProcessor();
 
     private JigsawReplacementProcessor() {
