--- a/net/minecraft/world/level/levelgen/structure/templatesystem/LavaSubmergedBlockProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/LavaSubmergedBlockProcessor.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.LevelReader;
@@ -8,7 +_,7 @@
 import net.minecraft.world.level.block.Blocks;
 
 public class LavaSubmergedBlockProcessor extends StructureProcessor {
-    public static final Codec<LavaSubmergedBlockProcessor> CODEC = Codec.unit(() -> LavaSubmergedBlockProcessor.INSTANCE);
+    public static final MapCodec<LavaSubmergedBlockProcessor> CODEC = MapCodec.unit(() -> LavaSubmergedBlockProcessor.INSTANCE);
     public static final LavaSubmergedBlockProcessor INSTANCE = new LavaSubmergedBlockProcessor();
 
     @Nullable
