--- a/net/minecraft/world/level/levelgen/structure/templatesystem/LinearPosTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/LinearPosTest.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -8,7 +_,7 @@
 import net.minecraft.util.RandomSource;
 
 public class LinearPosTest extends PosRuleTest {
-    public static final Codec<LinearPosTest> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<LinearPosTest> CODEC = RecordCodecBuilder.mapCodec(
         p_74160_ -> p_74160_.group(
                     Codec.FLOAT.fieldOf("min_chance").orElse(0.0F).forGetter(p_163737_ -> p_163737_.minChance),
                     Codec.FLOAT.fieldOf("max_chance").orElse(0.0F).forGetter(p_163735_ -> p_163735_.maxChance),
