--- a/net/minecraft/world/level/levelgen/structure/templatesystem/NopProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/NopProcessor.java
@@ -1,9 +_,9 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 
 public class NopProcessor extends StructureProcessor {
-    public static final Codec<NopProcessor> CODEC = Codec.unit(() -> NopProcessor.INSTANCE);
+    public static final MapCodec<NopProcessor> CODEC = MapCodec.unit(() -> NopProcessor.INSTANCE);
     public static final NopProcessor INSTANCE = new NopProcessor();
 
     private NopProcessor() {
