--- a/net/minecraft/world/level/levelgen/structure/templatesystem/PosAlwaysTrueTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/PosAlwaysTrueTest.java
@@ -1,11 +_,11 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.BlockPos;
 import net.minecraft.util.RandomSource;
 
 public class PosAlwaysTrueTest extends PosRuleTest {
-    public static final Codec<PosAlwaysTrueTest> CODEC = Codec.unit(() -> PosAlwaysTrueTest.INSTANCE);
+    public static final MapCodec<PosAlwaysTrueTest> CODEC = MapCodec.unit(() -> PosAlwaysTrueTest.INSTANCE);
     public static final PosAlwaysTrueTest INSTANCE = new PosAlwaysTrueTest();
 
     private PosAlwaysTrueTest() {
