--- a/net/minecraft/world/level/levelgen/structure/templatesystem/ProcessorRule.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/ProcessorRule.java
@@ -14,16 +_,16 @@
 public class ProcessorRule {
     public static final Passthrough DEFAULT_BLOCK_ENTITY_MODIFIER = Passthrough.INSTANCE;
     public static final Codec<ProcessorRule> CODEC = RecordCodecBuilder.create(
-        p_277334_ -> p_277334_.group(
+        p_338110_ -> p_338110_.group(
                     RuleTest.CODEC.fieldOf("input_predicate").forGetter(p_163747_ -> p_163747_.inputPredicate),
                     RuleTest.CODEC.fieldOf("location_predicate").forGetter(p_163745_ -> p_163745_.locPredicate),
-                    PosRuleTest.CODEC.optionalFieldOf("position_predicate", PosAlwaysTrueTest.INSTANCE).forGetter(p_163743_ -> p_163743_.posPredicate),
+                    PosRuleTest.CODEC.lenientOptionalFieldOf("position_predicate", PosAlwaysTrueTest.INSTANCE).forGetter(p_163743_ -> p_163743_.posPredicate),
                     BlockState.CODEC.fieldOf("output_state").forGetter(p_163741_ -> p_163741_.outputState),
                     RuleBlockEntityModifier.CODEC
-                        .optionalFieldOf("block_entity_modifier", DEFAULT_BLOCK_ENTITY_MODIFIER)
+                        .lenientOptionalFieldOf("block_entity_modifier", DEFAULT_BLOCK_ENTITY_MODIFIER)
                         .forGetter(p_277333_ -> p_277333_.blockEntityModifier)
                 )
-                .apply(p_277334_, ProcessorRule::new)
+                .apply(p_338110_, ProcessorRule::new)
     );
     private final RuleTest inputPredicate;
     private final RuleTest locPredicate;
