--- a/net/minecraft/world/level/levelgen/structure/templatesystem/ProtectedBlockProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/ProtectedBlockProcessor.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.registries.Registries;
@@ -11,8 +_,9 @@
 
 public class ProtectedBlockProcessor extends StructureProcessor {
     public final TagKey<Block> cannotReplace;
-    public static final Codec<ProtectedBlockProcessor> CODEC = TagKey.hashedCodec(Registries.BLOCK)
-        .xmap(ProtectedBlockProcessor::new, p_205053_ -> p_205053_.cannotReplace);
+    public static final MapCodec<ProtectedBlockProcessor> CODEC = TagKey.hashedCodec(Registries.BLOCK)
+        .xmap(ProtectedBlockProcessor::new, p_205053_ -> p_205053_.cannotReplace)
+        .fieldOf("value");
 
     public ProtectedBlockProcessor(TagKey<Block> p_205051_) {
         this.cannotReplace = p_205051_;
