--- a/net/minecraft/world/level/levelgen/structure/templatesystem/RuleProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/RuleProcessor.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
 import com.google.common.collect.ImmutableList;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class RuleProcessor extends StructureProcessor {
-    public static final Codec<RuleProcessor> CODEC = ProcessorRule.CODEC.listOf().fieldOf("rules").xmap(RuleProcessor::new, p_74306_ -> p_74306_.rules).codec();
+    public static final MapCodec<RuleProcessor> CODEC = ProcessorRule.CODEC.listOf().fieldOf("rules").xmap(RuleProcessor::new, p_74306_ -> p_74306_.rules);
     private final ImmutableList<ProcessorRule> rules;
 
     public RuleProcessor(List<? extends ProcessorRule> p_74296_) {
@@ -31,7 +_,7 @@
         RandomSource randomsource = RandomSource.create(Mth.getSeed(p_74303_.pos()));
         BlockState blockstate = p_74299_.getBlockState(p_74303_.pos());
 
-        for(ProcessorRule processorrule : this.rules) {
+        for (ProcessorRule processorrule : this.rules) {
             if (processorrule.test(p_74303_.state(), blockstate, p_74302_.pos(), p_74303_.pos(), p_74301_, randomsource)) {
                 return new StructureTemplate.StructureBlockInfo(
                     p_74303_.pos(), processorrule.getOutputState(), processorrule.getOutputTag(randomsource, p_74303_.nbt())
