--- a/net/minecraft/world/level/levelgen/structure/templatesystem/RuleTestType.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/RuleTestType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 
@@ -12,9 +_,9 @@
     RuleTestType<RandomBlockMatchTest> RANDOM_BLOCK_TEST = register("random_block_match", RandomBlockMatchTest.CODEC);
     RuleTestType<RandomBlockStateMatchTest> RANDOM_BLOCKSTATE_TEST = register("random_blockstate_match", RandomBlockStateMatchTest.CODEC);
 
-    Codec<P> codec();
+    MapCodec<P> codec();
 
-    static <P extends RuleTest> RuleTestType<P> register(String p_74322_, Codec<P> p_74323_) {
-        return Registry.register(BuiltInRegistries.RULE_TEST, p_74322_, () -> p_74323_);
+    static <P extends RuleTest> RuleTestType<P> register(String p_74322_, MapCodec<P> p_338228_) {
+        return Registry.register(BuiltInRegistries.RULE_TEST, p_74322_, () -> p_338228_);
     }
 }
