--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
@@ -96,7 +_,7 @@
     }
 
     private Optional<StructureTemplate> tryLoad(ResourceLocation p_230426_) {
-        for(StructureTemplateManager.Source structuretemplatemanager$source : this.sources) {
+        for (StructureTemplateManager.Source structuretemplatemanager$source : this.sources) {
             try {
                 Optional<StructureTemplate> optional = structuretemplatemanager$source.loader().apply(p_230426_);
                 if (optional.isPresent()) {
