--- a/net/minecraft/world/level/levelgen/structure/templatesystem/TagMatchTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/TagMatchTest.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.tags.TagKey;
 import net.minecraft.util.RandomSource;
@@ -8,7 +_,7 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class TagMatchTest extends RuleTest {
-    public static final Codec<TagMatchTest> CODEC = TagKey.codec(Registries.BLOCK).fieldOf("tag").xmap(TagMatchTest::new, p_205065_ -> p_205065_.tag).codec();
+    public static final MapCodec<TagMatchTest> CODEC = TagKey.codec(Registries.BLOCK).fieldOf("tag").xmap(TagMatchTest::new, p_205065_ -> p_205065_.tag);
     private final TagKey<Block> tag;
 
     public TagMatchTest(TagKey<Block> p_205063_) {
