--- a/net/minecraft/world/level/entity/EntityTickList.java
+++ b/net/minecraft/world/level/entity/EntityTickList.java
@@ -18,7 +_,7 @@
         if (this.iterated == this.active) {
             this.passive.clear();
 
-            for(Entry<Entity> entry : Int2ObjectMaps.fastIterable(this.active)) {
+            for (Entry<Entity> entry : Int2ObjectMaps.fastIterable(this.active)) {
                 this.passive.put(entry.getIntKey(), entry.getValue());
             }
 
@@ -49,7 +_,7 @@
             this.iterated = this.active;
 
             try {
-                for(Entity entity : this.active.values()) {
+                for (Entity entity : this.active.values()) {
                     p_156911_.accept(entity);
                 }
             } finally {
