--- a/net/minecraft/world/level/entity/EntitySection.java
+++ b/net/minecraft/world/level/entity/EntitySection.java
@@ -28,7 +_,7 @@
     }
 
     public AbortableIterationConsumer.Continuation getEntities(AABB p_262016_, AbortableIterationConsumer<T> p_261863_) {
-        for(T t : this.storage) {
+        for (T t : this.storage) {
             if (t.getBoundingBox().intersects(p_262016_) && p_261863_.accept(t).shouldAbort()) {
                 return AbortableIterationConsumer.Continuation.ABORT;
             }
@@ -44,8 +_,8 @@
         if (collection.isEmpty()) {
             return AbortableIterationConsumer.Continuation.CONTINUE;
         } else {
-            for(T t : collection) {
-                U u = p_188349_.tryCast(t);
+            for (T t : collection) {
+                U u = (U)p_188349_.tryCast(t);
                 if (u != null && t.getBoundingBox().intersects(p_188350_) && p_261535_.accept(u).shouldAbort()) {
                     return AbortableIterationConsumer.Continuation.ABORT;
                 }
