--- a/net/minecraft/data/loot/packs/UpdateOneTwentyOneChestLoot.java
+++ b/net/minecraft/data/loot/packs/UpdateOneTwentyOneChestLoot.java
@@ -16,6 +_,7 @@
 import net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction;
 import net.minecraft.world.level.storage.loot.functions.SetItemCountFunction;
 import net.minecraft.world.level.storage.loot.functions.SetItemDamageFunction;
+import net.minecraft.world.level.storage.loot.functions.SetOminousBottleAmplifierFunction;
 import net.minecraft.world.level.storage.loot.functions.SetPotionFunction;
 import net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceCondition;
 import net.minecraft.world.level.storage.loot.providers.number.ConstantValue;
@@ -321,19 +_,9 @@
                 .withPool(
                     LootPool.lootPool()
                         .setRolls(ConstantValue.exactly(1.0F))
-                        .add(LootItem.lootTableItem(Items.EMERALD_BLOCK).setWeight(3))
+                        .add(LootItem.lootTableItem(Items.EMERALD).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 4.0F))))
                         .add(LootItem.lootTableItem(Items.SHIELD).setWeight(3).apply(SetItemDamageFunction.setDamage(UniformGenerator.between(0.5F, 1.0F))))
                         .add(
-                            LootItem.lootTableItem(Items.IRON_CHESTPLATE)
-                                .setWeight(3)
-                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 15.0F)).allowTreasure())
-                        )
-                        .add(
-                            LootItem.lootTableItem(Items.IRON_AXE)
-                                .setWeight(3)
-                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 15.0F)).allowTreasure())
-                        )
-                        .add(
                             LootItem.lootTableItem(Items.BOW)
                                 .setWeight(3)
                                 .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 15.0F)).allowTreasure())
@@ -344,16 +_,18 @@
                                 .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 20.0F)).allowTreasure())
                         )
                         .add(
-                            LootItem.lootTableItem(Items.DIAMOND_AXE)
-                                .setWeight(2)
-                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 15.0F)).allowTreasure())
-                        )
-                        .add(
-                            LootItem.lootTableItem(Items.DIAMOND_CHESTPLATE)
-                                .setWeight(2)
-                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 15.0F)).allowTreasure())
-                        )
-                        .add(LootItem.lootTableItem(Items.GOLDEN_APPLE).setWeight(2))
+                            LootItem.lootTableItem(Items.IRON_AXE)
+                                .setWeight(2)
+                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(0.0F, 10.0F)).allowTreasure())
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.IRON_CHESTPLATE)
+                                .setWeight(2)
+                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(0.0F, 10.0F)).allowTreasure())
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.GOLDEN_CARROT).setWeight(2).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F)))
+                        )
                         .add(
                             LootItem.lootTableItem(Items.BOOK)
                                 .setWeight(2)
@@ -379,6 +_,16 @@
                                         .withEnchantment(Enchantments.MENDING)
                                 )
                         )
+                        .add(
+                            LootItem.lootTableItem(Items.DIAMOND_CHESTPLATE)
+                                .setWeight(1)
+                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 15.0F)).allowTreasure())
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.DIAMOND_AXE)
+                                .setWeight(1)
+                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 15.0F)).allowTreasure())
+                        )
                 )
         );
         p_312616_.accept(
@@ -394,10 +_,16 @@
                                 .apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 8.0F)))
                                 .apply(SetPotionFunction.setPotion(Potions.POISON))
                         )
-                        .add(LootItem.lootTableItem(Items.EMERALD).setWeight(4).apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 8.0F))))
+                        .add(LootItem.lootTableItem(Items.EMERALD).setWeight(4).apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 4.0F))))
                         .add(LootItem.lootTableItem(Items.WIND_CHARGE).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F))))
                         .add(LootItem.lootTableItem(Items.IRON_INGOT).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 4.0F))))
                         .add(LootItem.lootTableItem(Items.HONEY_BOTTLE).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F))))
+                        .add(
+                            LootItem.lootTableItem(Items.OMINOUS_BOTTLE)
+                                .setWeight(2)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetOminousBottleAmplifierFunction.setAmplifier(UniformGenerator.between(0.0F, 1.0F)))
+                        )
                         .add(LootItem.lootTableItem(Items.WIND_CHARGE).setWeight(1).apply(SetItemCountFunction.setCount(UniformGenerator.between(4.0F, 12.0F))))
                         .add(LootItem.lootTableItem(Items.DIAMOND).setWeight(1).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F))))
                 )
@@ -408,12 +_,10 @@
                 .withPool(
                     LootPool.lootPool()
                         .setRolls(ConstantValue.exactly(1.0F))
-                        .add(LootItem.lootTableItem(Items.ENCHANTED_GOLDEN_APPLE).setWeight(3))
+                        .add(LootItem.lootTableItem(Items.GOLDEN_APPLE).setWeight(6))
                         .add(LootItem.lootTableItem(Items.BOLT_ARMOR_TRIM_SMITHING_TEMPLATE).setWeight(3))
-                        .add(LootItem.lootTableItem(Items.FLOW_ARMOR_TRIM_SMITHING_TEMPLATE).setWeight(3))
-                        .add(LootItem.lootTableItem(Items.FLOW_BANNER_PATTERN).setWeight(1))
-                        .add(LootItem.lootTableItem(Items.GUSTER_BANNER_PATTERN).setWeight(1))
-                        .add(LootItem.lootTableItem(Items.HEAVY_CORE).setWeight(1))
+                        .add(LootItem.lootTableItem(Items.GUSTER_BANNER_PATTERN).setWeight(2))
+                        .add(LootItem.lootTableItem(Items.TRIDENT).setWeight(1))
                 )
         );
         p_312616_.accept(
@@ -438,6 +_,111 @@
                 )
         );
         p_312616_.accept(
+            BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS_RARE,
+            LootTable.lootTable()
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .add(LootItem.lootTableItem(Items.EMERALD_BLOCK).setWeight(6))
+                        .add(LootItem.lootTableItem(Items.IRON_BLOCK).setWeight(4))
+                        .add(
+                            LootItem.lootTableItem(Items.CROSSBOW)
+                                .setWeight(4)
+                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(5.0F, 20.0F)).allowTreasure())
+                        )
+                        .add(LootItem.lootTableItem(Items.GOLDEN_APPLE).setWeight(4))
+                        .add(
+                            LootItem.lootTableItem(Items.DIAMOND_AXE)
+                                .setWeight(3)
+                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(10.0F, 20.0F)).allowTreasure())
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.DIAMOND_CHESTPLATE)
+                                .setWeight(3)
+                                .apply(EnchantWithLevelsFunction.enchantWithLevels(UniformGenerator.between(10.0F, 20.0F)).allowTreasure())
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.BOOK)
+                                .setWeight(2)
+                                .apply(
+                                    new EnchantRandomlyFunction.Builder()
+                                        .withEnchantment(Enchantments.KNOCKBACK)
+                                        .withEnchantment(Enchantments.PUNCH)
+                                        .withEnchantment(Enchantments.SMITE)
+                                        .withEnchantment(Enchantments.LOOTING)
+                                        .withEnchantment(Enchantments.MULTISHOT)
+                                )
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.BOOK)
+                                .setWeight(2)
+                                .apply(new EnchantRandomlyFunction.Builder().withEnchantment(Enchantments.BREACH).withEnchantment(Enchantments.DENSITY))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.BOOK)
+                                .setWeight(1)
+                                .apply(new EnchantRandomlyFunction.Builder().withEnchantment(Enchantments.WIND_BURST))
+                        )
+                        .add(LootItem.lootTableItem(Items.DIAMOND_BLOCK).setWeight(1))
+                )
+        );
+        p_312616_.accept(
+            BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS_COMMON,
+            LootTable.lootTable()
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .add(LootItem.lootTableItem(Items.EMERALD).setWeight(5).apply(SetItemCountFunction.setCount(UniformGenerator.between(4.0F, 10.0F))))
+                        .add(LootItem.lootTableItem(Items.WIND_CHARGE).setWeight(4).apply(SetItemCountFunction.setCount(UniformGenerator.between(8.0F, 12.0F))))
+                        .add(
+                            LootItem.lootTableItem(Items.TIPPED_ARROW)
+                                .setWeight(3)
+                                .apply(SetItemCountFunction.setCount(UniformGenerator.between(4.0F, 12.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.STRONG_SLOWNESS))
+                        )
+                        .add(LootItem.lootTableItem(Items.DIAMOND).setWeight(2).apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 3.0F))))
+                        .add(
+                            LootItem.lootTableItem(Items.OMINOUS_BOTTLE)
+                                .setWeight(1)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetOminousBottleAmplifierFunction.setAmplifier(UniformGenerator.between(2.0F, 4.0F)))
+                        )
+                )
+        );
+        p_312616_.accept(
+            BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS_UNIQUE,
+            LootTable.lootTable()
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .add(LootItem.lootTableItem(Items.ENCHANTED_GOLDEN_APPLE).setWeight(3))
+                        .add(LootItem.lootTableItem(Items.FLOW_ARMOR_TRIM_SMITHING_TEMPLATE).setWeight(3))
+                        .add(LootItem.lootTableItem(Items.FLOW_BANNER_PATTERN).setWeight(2))
+                        .add(LootItem.lootTableItem(Items.HEAVY_CORE).setWeight(1))
+                )
+        );
+        p_312616_.accept(
+            BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS,
+            LootTable.lootTable()
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .add(NestedLootTable.lootTableReference(BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS_RARE).setWeight(8))
+                        .add(NestedLootTable.lootTableReference(BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS_COMMON).setWeight(2))
+                )
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(UniformGenerator.between(1.0F, 3.0F))
+                        .add(NestedLootTable.lootTableReference(BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS_COMMON))
+                )
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .when(LootItemRandomChanceCondition.randomChance(0.75F))
+                        .add(NestedLootTable.lootTableReference(BuiltInLootTables.TRIAL_CHAMBERS_REWARD_OMINOUS_UNIQUE))
+                )
+        );
+        p_312616_.accept(
             BuiltInLootTables.SPAWNER_TRIAL_CHAMBER_KEY,
             LootTable.lootTable().withPool(LootPool.lootPool().setRolls(ConstantValue.exactly(1.0F)).add(LootItem.lootTableItem(Items.TRIAL_KEY)))
         );
@@ -447,25 +_,109 @@
                 .withPool(
                     LootPool.lootPool()
                         .setRolls(ConstantValue.exactly(1.0F))
-                        .add(LootItem.lootTableItem(Items.EMERALD).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 6.0F))))
-                        .add(
-                            LootItem.lootTableItem(Items.GOLDEN_CARROT).setWeight(1).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F)))
-                        )
-                        .add(LootItem.lootTableItem(Items.BAKED_POTATO).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F))))
-                        .add(
-                            LootItem.lootTableItem(Items.GLOW_BERRIES).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 10.0F)))
-                        )
-                        .add(LootItem.lootTableItem(Items.ENDER_PEARL).apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F))))
-                        .add(
-                            LootItem.lootTableItem(Items.POTION)
-                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
-                                .apply(SetPotionFunction.setPotion(Potions.REGENERATION))
-                        )
-                        .add(
-                            LootItem.lootTableItem(Items.POTION)
-                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
-                                .apply(SetPotionFunction.setPotion(Potions.STRENGTH))
-                        )
+                        .add(LootItem.lootTableItem(Items.COOKED_CHICKEN).setWeight(3).apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F))))
+                        .add(LootItem.lootTableItem(Items.BREAD).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F))))
+                        .add(LootItem.lootTableItem(Items.BAKED_POTATO).setWeight(2).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F))))
+                        .add(
+                            LootItem.lootTableItem(Items.POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.REGENERATION))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.SWIFTNESS))
+                        )
+                        .add(LootItem.lootTableItem(Items.ROTTEN_FLESH).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 4.0F))))
+                )
+        );
+        p_312616_.accept(
+            BuiltInLootTables.SPAWNER_OMINOUS_TRIAL_CHAMBER_KEY,
+            LootTable.lootTable().withPool(LootPool.lootPool().setRolls(ConstantValue.exactly(1.0F)).add(LootItem.lootTableItem(Items.OMINOUS_TRIAL_KEY)))
+        );
+        p_312616_.accept(
+            BuiltInLootTables.SPAWNER_OMINOUS_TRIAL_CHAMBER_CONSUMABLES,
+            LootTable.lootTable()
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .add(LootItem.lootTableItem(Items.COOKED_BEEF).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F))))
+                        .add(LootItem.lootTableItem(Items.BAKED_POTATO).setWeight(3).apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 4.0F))))
+                        .add(
+                            LootItem.lootTableItem(Items.GOLDEN_CARROT).setWeight(2).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F)))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.REGENERATION))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.STRENGTH))
+                        )
+                        .add(LootItem.lootTableItem(Items.ROTTEN_FLESH).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 4.0F))))
+                )
+        );
+        p_312616_.accept(
+            BuiltInLootTables.SPAWNER_TRIAL_ITEMS_TO_DROP_WHEN_OMINOUS,
+            LootTable.lootTable()
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .add(
+                            LootItem.lootTableItem(Items.LINGERING_POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.WIND_CHARGED))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.LINGERING_POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.OOZING))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.LINGERING_POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.WEAVING))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.LINGERING_POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.INFESTED))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.LINGERING_POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.STRENGTH))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.LINGERING_POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.SWIFTNESS))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.LINGERING_POTION)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.SLOW_FALLING))
+                        )
+                )
+                .withPool(
+                    LootPool.lootPool()
+                        .setRolls(ConstantValue.exactly(1.0F))
+                        .add(LootItem.lootTableItem(Items.ARROW).apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F))))
+                        .add(
+                            LootItem.lootTableItem(Items.ARROW)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.POISON))
+                        )
+                        .add(
+                            LootItem.lootTableItem(Items.ARROW)
+                                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
+                                .apply(SetPotionFunction.setPotion(Potions.STRONG_SLOWNESS))
+                        )
+                        .add(LootItem.lootTableItem(Items.FIRE_CHARGE).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F))))
+                        .add(LootItem.lootTableItem(Items.WIND_CHARGE).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F))))
+                        .add(LootItem.lootTableItem(Items.EXPERIENCE_BOTTLE).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 3.0F))))
                 )
         );
     }
