--- a/net/minecraft/data/loot/packs/VanillaFishingLoot.java
+++ b/net/minecraft/data/loot/packs/VanillaFishingLoot.java
@@ -19,7 +_,6 @@
 import net.minecraft.world.level.storage.loot.LootPool;
 import net.minecraft.world.level.storage.loot.LootTable;
 import net.minecraft.world.level.storage.loot.entries.LootItem;
-import net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer;
 import net.minecraft.world.level.storage.loot.entries.NestedLootTable;
 import net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction;
 import net.minecraft.world.level.storage.loot.functions.SetItemCountFunction;
@@ -79,19 +_,19 @@
                         .add(LootItem.lootTableItem(Blocks.TRIPWIRE_HOOK).setWeight(10))
                         .add(LootItem.lootTableItem(Items.ROTTEN_FLESH).setWeight(10))
                         .add(
-                            ((LootPoolSingletonContainer.Builder)LootItem.lootTableItem(Blocks.BAMBOO)
-                                    .when(
-                                        LocationCheck.checkLocation(
-                                            LocationPredicate.Builder.location()
-                                                .setBiomes(
-                                                    HolderSet.direct(
-                                                        registrylookup.getOrThrow(Biomes.JUNGLE),
-                                                        registrylookup.getOrThrow(Biomes.SPARSE_JUNGLE),
-                                                        registrylookup.getOrThrow(Biomes.BAMBOO_JUNGLE)
-                                                    )
+                            LootItem.lootTableItem(Blocks.BAMBOO)
+                                .when(
+                                    LocationCheck.checkLocation(
+                                        LocationPredicate.Builder.location()
+                                            .setBiomes(
+                                                HolderSet.direct(
+                                                    registrylookup.getOrThrow(Biomes.JUNGLE),
+                                                    registrylookup.getOrThrow(Biomes.SPARSE_JUNGLE),
+                                                    registrylookup.getOrThrow(Biomes.BAMBOO_JUNGLE)
                                                 )
-                                        )
-                                    ))
+                                            )
+                                    )
+                                )
                                 .setWeight(10)
                         )
                 )
