--- a/net/minecraft/data/models/blockstates/MultiVariantGenerator.java
+++ b/net/minecraft/data/models/blockstates/MultiVariantGenerator.java
@@ -29,11 +_,11 @@
     }
 
     public MultiVariantGenerator with(PropertyDispatch p_125272_) {
-        p_125272_.getDefinedProperties().forEach(p_125263_ -> {
-            if (this.block.getStateDefinition().getProperty(p_125263_.getName()) != p_125263_) {
-                throw new IllegalStateException("Property " + p_125263_ + " is not defined for block " + this.block);
-            } else if (!this.seenProperties.add(p_125263_)) {
-                throw new IllegalStateException("Values of property " + p_125263_ + " already defined for block " + this.block);
+        p_125272_.getDefinedProperties().forEach(p_339378_ -> {
+            if (this.block.getStateDefinition().getProperty(p_339378_.getName()) != p_339378_) {
+                throw new IllegalStateException("Property " + p_339378_ + " is not defined for block " + this.block);
+            } else if (!this.seenProperties.add((Property<?>)p_339378_)) {
+                throw new IllegalStateException("Values of property " + p_339378_ + " already defined for block " + this.block);
             }
         });
         this.declaredPropertySets.add(p_125272_);
@@ -43,7 +_,7 @@
     public JsonElement get() {
         Stream<Pair<Selector, List<Variant>>> stream = Stream.of(Pair.of(Selector.empty(), this.baseVariants));
 
-        for(PropertyDispatch propertydispatch : this.declaredPropertySets) {
+        for (PropertyDispatch propertydispatch : this.declaredPropertySets) {
             Map<Selector, List<Variant>> map = propertydispatch.getEntries();
             stream = stream.flatMap(p_125289_ -> map.entrySet().stream().map(p_176309_ -> {
                     Selector selector = ((Selector)p_125289_.getFirst()).extend(p_176309_.getKey());
