--- a/net/minecraft/data/models/blockstates/PropertyDispatch.java
+++ b/net/minecraft/data/models/blockstates/PropertyDispatch.java
@@ -31,7 +_,7 @@
         List<Property<?>> list = this.getDefinedProperties();
         Stream<Selector> stream = Stream.of(Selector.empty());
 
-        for(Property<?> property : list) {
+        for (Property<?> property : list) {
             stream = stream.flatMap(p_125316_ -> property.getAllValues().map(p_125316_::extend));
         }
 
@@ -92,12 +_,12 @@
         }
 
         public PropertyDispatch generate(Function<T1, Variant> p_125336_) {
-            this.property1.getPossibleValues().forEach(p_125340_ -> this.select(p_125340_, p_125336_.apply(p_125340_)));
+            this.property1.getPossibleValues().forEach(p_125340_ -> this.select((T1)p_125340_, p_125336_.apply((T1)p_125340_)));
             return this;
         }
 
         public PropertyDispatch generateList(Function<T1, List<Variant>> p_176314_) {
-            this.property1.getPossibleValues().forEach(p_176312_ -> this.select(p_176312_, p_176314_.apply(p_176312_)));
+            this.property1.getPossibleValues().forEach(p_176312_ -> this.select((T1)p_176312_, p_176314_.apply((T1)p_176312_)));
             return this;
         }
     }
@@ -132,7 +_,7 @@
                 .forEach(
                     p_125376_ -> this.property2
                             .getPossibleValues()
-                            .forEach(p_176322_ -> this.select((T1)p_125376_, p_176322_, p_125363_.apply((T1)p_125376_, p_176322_)))
+                            .forEach(p_176322_ -> this.select((T1)p_125376_, (T2)p_176322_, p_125363_.apply((T1)p_125376_, (T2)p_176322_)))
                 );
             return this;
         }
@@ -143,7 +_,7 @@
                 .forEach(
                     p_125366_ -> this.property2
                             .getPossibleValues()
-                            .forEach(p_176318_ -> this.select((T1)p_125366_, p_176318_, p_125373_.apply((T1)p_125366_, p_176318_)))
+                            .forEach(p_176318_ -> this.select((T1)p_125366_, (T2)p_176318_, p_125373_.apply((T1)p_125366_, (T2)p_176318_)))
                 );
             return this;
         }
@@ -186,7 +_,7 @@
                                         .getPossibleValues()
                                         .forEach(
                                             p_176339_ -> this.select(
-                                                    (T1)p_125404_, (T2)p_176343_, p_176339_, p_125390_.apply((T1)p_125404_, (T2)p_176343_, p_176339_)
+                                                    (T1)p_125404_, (T2)p_176343_, (T3)p_176339_, p_125390_.apply((T1)p_125404_, (T2)p_176343_, (T3)p_176339_)
                                                 )
                                         )
                             )
@@ -205,7 +_,7 @@
                                         .getPossibleValues()
                                         .forEach(
                                             p_176327_ -> this.select(
-                                                    (T1)p_176334_, (T2)p_176331_, p_176327_, p_176345_.apply((T1)p_176334_, (T2)p_176331_, p_176327_)
+                                                    (T1)p_176334_, (T2)p_176331_, (T3)p_176327_, p_176345_.apply((T1)p_176334_, (T2)p_176331_, (T3)p_176327_)
                                                 )
                                         )
                             )
@@ -261,8 +_,8 @@
                                                                 (T1)p_176385_,
                                                                 (T2)p_176380_,
                                                                 (T3)p_176376_,
-                                                                p_176371_,
-                                                                p_176362_.apply((T1)p_176385_, (T2)p_176380_, (T3)p_176376_, p_176371_)
+                                                                (T4)p_176371_,
+                                                                p_176362_.apply((T1)p_176385_, (T2)p_176380_, (T3)p_176376_, (T4)p_176371_)
                                                             )
                                                     )
                                         )
@@ -288,8 +_,8 @@
                                                                 (T1)p_176365_,
                                                                 (T2)p_176360_,
                                                                 (T3)p_176356_,
-                                                                p_176351_,
-                                                                p_176382_.apply((T1)p_176365_, (T2)p_176360_, (T3)p_176356_, p_176351_)
+                                                                (T4)p_176351_,
+                                                                p_176382_.apply((T1)p_176365_, (T2)p_176360_, (T3)p_176356_, (T4)p_176351_)
                                                             )
                                                     )
                                         )
@@ -357,9 +_,9 @@
                                                                             (T2)p_176434_,
                                                                             (T3)p_176430_,
                                                                             (T4)p_176425_,
-                                                                            p_176419_,
+                                                                            (T5)p_176419_,
                                                                             p_176409_.apply(
-                                                                                (T1)p_176439_, (T2)p_176434_, (T3)p_176430_, (T4)p_176425_, p_176419_
+                                                                                (T1)p_176439_, (T2)p_176434_, (T3)p_176430_, (T4)p_176425_, (T5)p_176419_
                                                                             )
                                                                         )
                                                                 )
@@ -391,9 +_,9 @@
                                                                             (T2)p_176407_,
                                                                             (T3)p_176403_,
                                                                             (T4)p_176398_,
-                                                                            p_176392_,
+                                                                            (T5)p_176392_,
                                                                             p_176436_.apply(
-                                                                                (T1)p_176412_, (T2)p_176407_, (T3)p_176403_, (T4)p_176398_, p_176392_
+                                                                                (T1)p_176412_, (T2)p_176407_, (T3)p_176403_, (T4)p_176398_, (T5)p_176392_
                                                                             )
                                                                         )
                                                                 )
