--- a/net/minecraft/client/gui/screens/packs/PackSelectionScreen.java
+++ b/net/minecraft/client/gui/screens/packs/PackSelectionScreen.java
@@ -28,6 +_,7 @@
 import net.minecraft.Util;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.ComponentPath;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.StringWidget;
 import net.minecraft.client.gui.components.Tooltip;
@@ -116,7 +_,9 @@
         );
         this.doneButton = linearlayout1.addChild(Button.builder(CommonComponents.GUI_DONE, p_100036_ -> this.onClose()).build());
         this.reload();
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_329736_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_329736_);
+        });
         this.repositionElements();
     }
 
@@ -159,12 +_,12 @@
         String s = transferableselectionlist$packentry == null ? "" : transferableselectionlist$packentry.getPackId();
         p_100014_.setSelected(null);
         p_100015_.forEach(
-            p_332533_ -> {
+            p_339286_ -> {
                 TransferableSelectionList.PackEntry transferableselectionlist$packentry1 = new TransferableSelectionList.PackEntry(
-                    this.minecraft, p_100014_, p_332533_
+                    this.minecraft, p_100014_, p_339286_
                 );
                 p_100014_.children().add(transferableselectionlist$packentry1);
-                if (p_332533_.getId().equals(s)) {
+                if (p_339286_.getId().equals(s)) {
                     p_100014_.setSelected(transferableselectionlist$packentry1);
                 }
             }
@@ -231,7 +_,7 @@
                             };
                             List<ForbiddenSymlinkInfo> list1 = new ArrayList<>();
             
-                            for(Path path : p_100029_) {
+                            for (Path path : p_100029_) {
                                 try {
                                     Path path1 = packdetector.detectPackResources(path, list1);
                                     if (path1 == null) {
@@ -326,7 +_,7 @@
                 this.watchDir(p_250327_);
 
                 try (DirectoryStream<Path> directorystream = Files.newDirectoryStream(p_250327_)) {
-                    for(Path path : directorystream) {
+                    for (Path path : directorystream) {
                         if (Files.isDirectory(path, LinkOption.NOFOLLOW_LINKS)) {
                             this.watchDir(path);
                         }
@@ -356,8 +_,8 @@
             boolean flag = false;
 
             WatchKey watchkey;
-            while((watchkey = this.watcher.poll()) != null) {
-                for(WatchEvent<?> watchevent : watchkey.pollEvents()) {
+            while ((watchkey = this.watcher.poll()) != null) {
+                for (WatchEvent<?> watchevent : watchkey.pollEvents()) {
                     flag = true;
                     if (watchkey.watchable() == this.packPath && watchevent.kind() == StandardWatchEventKinds.ENTRY_CREATE) {
                         Path path = this.packPath.resolve((Path)watchevent.context());
