--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -81,7 +_,7 @@
         "reducedDebugInfo", GameRules.Category.MISC, GameRules.BooleanValue.create(false, (p_294061_, p_294062_) -> {
             byte b0 = (byte)(p_294062_.get() ? 22 : 23);
     
-            for(ServerPlayer serverplayer : p_294061_.getPlayerList().getPlayers()) {
+            for (ServerPlayer serverplayer : p_294061_.getPlayerList().getPlayers()) {
                 serverplayer.connection.send(new ClientboundEntityEventPacket(serverplayer, b0));
             }
         })
@@ -103,7 +_,7 @@
     );
     public static final GameRules.Key<GameRules.BooleanValue> RULE_LIMITED_CRAFTING = register(
         "doLimitedCrafting", GameRules.Category.PLAYER, GameRules.BooleanValue.create(false, (p_301943_, p_301944_) -> {
-            for(ServerPlayer serverplayer : p_301943_.getPlayerList().getPlayers()) {
+            for (ServerPlayer serverplayer : p_301943_.getPlayerList().getPlayers()) {
                 serverplayer.connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.LIMITED_CRAFTING, p_301944_.get() ? 1.0F : 0.0F));
             }
         })
@@ -117,9 +_,6 @@
     public static final GameRules.Key<GameRules.IntegerValue> RULE_COMMAND_MODIFICATION_BLOCK_LIMIT = register(
         "commandModificationBlockLimit", GameRules.Category.MISC, GameRules.IntegerValue.create(32768)
     );
-    public static final GameRules.Key<GameRules.IntegerValue> RULE_FLOATATER_SIZE_LIMIT = register(
-        "floataterSizeLimit", GameRules.Category.MISC, GameRules.IntegerValue.create(32)
-    );
     public static final GameRules.Key<GameRules.BooleanValue> RULE_ANNOUNCE_ADVANCEMENTS = register(
         "announceAdvancements", GameRules.Category.CHAT, GameRules.BooleanValue.create(true)
     );
@@ -131,7 +_,7 @@
     );
     public static final GameRules.Key<GameRules.BooleanValue> RULE_DO_IMMEDIATE_RESPAWN = register(
         "doImmediateRespawn", GameRules.Category.PLAYER, GameRules.BooleanValue.create(false, (p_294059_, p_294060_) -> {
-            for(ServerPlayer serverplayer : p_294059_.getPlayerList().getPlayers()) {
+            for (ServerPlayer serverplayer : p_294059_.getPlayerList().getPlayers()) {
                 serverplayer.connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.IMMEDIATE_RESPAWN, p_294060_.get() ? 1.0F : 0.0F));
             }
         })
@@ -200,14 +_,11 @@
         "enderPearlsVanishOnDeath", GameRules.Category.PLAYER, GameRules.BooleanValue.create(true)
     );
     public static final GameRules.Key<GameRules.IntegerValue> RULE_SPAWN_CHUNK_RADIUS = register(
-        "spawnChunkRadius", GameRules.Category.MISC, GameRules.IntegerValue.create(2, 0, 32, (p_340208_, p_340209_) -> {
-            ServerLevel serverlevel = p_340208_.overworld();
+        "spawnChunkRadius", GameRules.Category.MISC, GameRules.IntegerValue.create(2, 0, 32, (p_330122_, p_330123_) -> {
+            ServerLevel serverlevel = p_330122_.overworld();
             serverlevel.setDefaultSpawnPos(serverlevel.getSharedSpawnPos(), serverlevel.getSharedSpawnAngle());
         })
     );
-    public static final GameRules.Key<GameRules.BooleanValue> RULE_NEVER_EAT_ARMOR = register(
-        "neverEatArmor", GameRules.Category.PLAYER, GameRules.BooleanValue.create(false)
-    );
     private final Map<GameRules.Key<?>, GameRules.Value<?>> rules;
 
     public static <T extends GameRules.Value<T>> GameRules.Key<T> register(String p_46190_, GameRules.Category p_46191_, GameRules.Type<T> p_46192_) {
@@ -226,9 +_,7 @@
     }
 
     public GameRules() {
-        this.rules = GAME_RULE_TYPES.entrySet()
-            .stream()
-            .collect(ImmutableMap.toImmutableMap(Entry::getKey, p_46210_ -> ((GameRules.Type)p_46210_.getValue()).createRule()));
+        this.rules = GAME_RULE_TYPES.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, p_46210_ -> p_46210_.getValue().createRule()));
     }
 
     private GameRules(Map<GameRules.Key<?>, GameRules.Value<?>> p_46162_) {
@@ -246,17 +_,15 @@
     }
 
     private void loadFromTag(DynamicLike<?> p_46184_) {
-        this.rules.forEach((p_46187_, p_46188_) -> p_46184_.get(p_46187_.id).asString().result().ifPresent(p_46188_::deserialize));
+        this.rules.forEach((p_337968_, p_337969_) -> p_46184_.get(p_337968_.id).asString().ifSuccess(p_337969_::deserialize));
     }
 
     public GameRules copy() {
-        return new GameRules(
-            this.rules.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, p_46194_ -> ((GameRules.Value)p_46194_.getValue()).copy()))
-        );
+        return new GameRules(this.rules.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, p_46194_ -> p_46194_.getValue().copy())));
     }
 
     public static void visitGameRuleTypes(GameRules.GameRuleTypeVisitor p_46165_) {
-        GAME_RULE_TYPES.forEach((p_46205_, p_46206_) -> callVisitorCap(p_46165_, p_46205_, p_46206_));
+        GAME_RULE_TYPES.forEach((p_46205_, p_46206_) -> callVisitorCap(p_46165_, (GameRules.Key<?>)p_46205_, (GameRules.Type<?>)p_46206_));
     }
 
     private static <T extends GameRules.Value<T>> void callVisitorCap(
@@ -267,7 +_,7 @@
     }
 
     public void assignFrom(GameRules p_46177_, @Nullable MinecraftServer p_46178_) {
-        p_46177_.rules.keySet().forEach(p_46182_ -> this.assignCap(p_46182_, p_46177_, p_46178_));
+        p_46177_.rules.keySet().forEach(p_46182_ -> this.assignCap((GameRules.Key<?>)p_46182_, p_46177_, p_46178_));
     }
 
     private <T extends GameRules.Value<T>> void assignCap(GameRules.Key<T> p_46173_, GameRules p_46174_, @Nullable MinecraftServer p_46175_) {
@@ -433,7 +_,7 @@
         public boolean tryDeserialize(String p_46315_) {
             try {
                 StringReader stringreader = new StringReader(p_46315_);
-                this.value = (int)this.type.argument.get().parse(stringreader);
+                this.value = (Integer)this.type.argument.get().parse(stringreader);
                 return !stringreader.canRead();
             } catch (CommandSyntaxException commandsyntaxexception) {
                 return false;
@@ -487,11 +_,7 @@
 
         @Override
         public boolean equals(Object p_46334_) {
-            if (this == p_46334_) {
-                return true;
-            } else {
-                return p_46334_ instanceof GameRules.Key && ((GameRules.Key)p_46334_).id.equals(this.id);
-            }
+            return this == p_46334_ ? true : p_46334_ instanceof GameRules.Key && ((GameRules.Key)p_46334_).id.equals(this.id);
         }
 
         @Override
@@ -531,7 +_,7 @@
         }
 
         public RequiredArgumentBuilder<CommandSourceStack, ?> createArgument(String p_46359_) {
-            return Commands.argument(p_46359_, (ArgumentType<T>)this.argument.get());
+            return Commands.argument(p_46359_, this.argument.get());
         }
 
         public T createRule() {
