--- a/net/minecraft/world/level/PathNavigationRegion.java
+++ b/net/minecraft/world/level/PathNavigationRegion.java
@@ -43,14 +_,14 @@
         ChunkSource chunksource = p_47164_.getChunkSource();
         this.allEmpty = true;
 
-        for(int k = this.centerX; k <= i; ++k) {
-            for(int l = this.centerZ; l <= j; ++l) {
+        for (int k = this.centerX; k <= i; k++) {
+            for (int l = this.centerZ; l <= j; l++) {
                 this.chunks[k - this.centerX][l - this.centerZ] = chunksource.getChunkNow(k, l);
             }
         }
 
-        for(int i1 = SectionPos.blockToSectionCoord(p_47165_.getX()); i1 <= SectionPos.blockToSectionCoord(p_47166_.getX()); ++i1) {
-            for(int j1 = SectionPos.blockToSectionCoord(p_47165_.getZ()); j1 <= SectionPos.blockToSectionCoord(p_47166_.getZ()); ++j1) {
+        for (int i1 = SectionPos.blockToSectionCoord(p_47165_.getX()); i1 <= SectionPos.blockToSectionCoord(p_47166_.getX()); i1++) {
+            for (int j1 = SectionPos.blockToSectionCoord(p_47165_.getZ()); j1 <= SectionPos.blockToSectionCoord(p_47166_.getZ()); j1++) {
                 ChunkAccess chunkaccess = this.chunks[i1 - this.centerX][j1 - this.centerZ];
                 if (chunkaccess != null && !chunkaccess.isYSpaceEmpty(p_47165_.getY(), p_47166_.getY())) {
                     this.allEmpty = false;
@@ -129,10 +_,5 @@
 
     public ProfilerFiller getProfiler() {
         return this.level.getProfiler();
-    }
-
-    @Override
-    public boolean isPotato() {
-        return this.level.isPotato();
     }
 }
