--- a/net/minecraft/world/level/SignalGetter.java
+++ b/net/minecraft/world/level/SignalGetter.java
@@ -65,7 +_,7 @@
     default int getSignal(BlockPos p_277961_, Direction p_277351_) {
         BlockState blockstate = this.getBlockState(p_277961_);
         int i = blockstate.getSignal(this, p_277961_, p_277351_);
-        return blockstate.isRedstoneConductor(this, p_277961_) ? Math.max(i, this.getDirectSignalTo(p_277961_)) : i;
+        return blockstate.shouldCheckWeakPower(this, p_277961_, p_277351_) ? Math.max(i, this.getDirectSignalTo(p_277961_)) : i;
     }
 
     default boolean hasNeighborSignal(BlockPos p_277626_) {
@@ -77,17 +_,15 @@
             return true;
         } else if (this.getSignal(p_277626_.south(), Direction.SOUTH) > 0) {
             return true;
-        } else if (this.getSignal(p_277626_.west(), Direction.WEST) > 0) {
-            return true;
         } else {
-            return this.getSignal(p_277626_.east(), Direction.EAST) > 0;
+            return this.getSignal(p_277626_.west(), Direction.WEST) > 0 ? true : this.getSignal(p_277626_.east(), Direction.EAST) > 0;
         }
     }
 
     default int getBestNeighborSignal(BlockPos p_277977_) {
         int i = 0;
 
-        for(Direction direction : DIRECTIONS) {
+        for (Direction direction : DIRECTIONS) {
             int j = this.getSignal(p_277977_.relative(direction), direction);
             if (j >= 15) {
                 return 15;
