--- a/net/minecraft/world/level/SpawnData.java
+++ b/net/minecraft/world/level/SpawnData.java
@@ -10,26 +_,26 @@
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.InclusiveRange;
 import net.minecraft.util.random.SimpleWeightedRandomList;
 
-public record SpawnData(CompoundTag entityToSpawn, Optional<SpawnData.CustomSpawnRules> customSpawnRules) {
+public record SpawnData(CompoundTag entityToSpawn, Optional<SpawnData.CustomSpawnRules> customSpawnRules, Optional<ResourceLocation> equipmentLootTable) {
     public static final String ENTITY_TAG = "entity";
     public static final Codec<SpawnData> CODEC = RecordCodecBuilder.create(
-        p_186571_ -> p_186571_.group(
+        p_337971_ -> p_337971_.group(
                     CompoundTag.CODEC.fieldOf("entity").forGetter(p_186576_ -> p_186576_.entityToSpawn),
-                    SpawnData.CustomSpawnRules.CODEC.optionalFieldOf("custom_spawn_rules").forGetter(p_186569_ -> p_186569_.customSpawnRules)
+                    SpawnData.CustomSpawnRules.CODEC.optionalFieldOf("custom_spawn_rules").forGetter(p_186569_ -> p_186569_.customSpawnRules),
+                    ResourceLocation.CODEC.optionalFieldOf("equipment_loot_table").forGetter(p_337970_ -> p_337970_.equipmentLootTable)
                 )
-                .apply(p_186571_, SpawnData::new)
+                .apply(p_337971_, SpawnData::new)
     );
     public static final Codec<SimpleWeightedRandomList<SpawnData>> LIST_CODEC = SimpleWeightedRandomList.wrappedCodecAllowingEmpty(CODEC);
 
     public SpawnData() {
-        this(new CompoundTag(), Optional.empty());
+        this(new CompoundTag(), Optional.empty(), Optional.empty());
     }
 
-    public SpawnData(CompoundTag entityToSpawn, Optional<SpawnData.CustomSpawnRules> customSpawnRules) {
+    public SpawnData(CompoundTag entityToSpawn, Optional<SpawnData.CustomSpawnRules> customSpawnRules, Optional<ResourceLocation> equipmentLootTable) {
         if (entityToSpawn.contains("id")) {
             ResourceLocation resourcelocation = ResourceLocation.tryParse(entityToSpawn.getString("id"));
             if (resourcelocation != null) {
@@ -41,6 +_,7 @@
 
         this.entityToSpawn = entityToSpawn;
         this.customSpawnRules = customSpawnRules;
+        this.equipmentLootTable = equipmentLootTable;
     }
 
     public CompoundTag getEntityToSpawn() {
@@ -51,6 +_,10 @@
         return this.customSpawnRules;
     }
 
+    public Optional<ResourceLocation> getEquipmentLootTable() {
+        return this.equipmentLootTable;
+    }
+
     public static record CustomSpawnRules(InclusiveRange<Integer> blockLightLimit, InclusiveRange<Integer> skyLightLimit) {
         private static final InclusiveRange<Integer> LIGHT_RANGE = new InclusiveRange<>(0, 15);
         public static final Codec<SpawnData.CustomSpawnRules> CODEC = RecordCodecBuilder.create(
@@ -68,7 +_,7 @@
         }
 
         private static MapCodec<InclusiveRange<Integer>> lightLimit(String p_286409_) {
-            return ExtraCodecs.validate(InclusiveRange.INT.optionalFieldOf(p_286409_, LIGHT_RANGE), SpawnData.CustomSpawnRules::checkLightBoundaries);
+            return InclusiveRange.INT.lenientOptionalFieldOf(p_286409_, LIGHT_RANGE).validate(SpawnData.CustomSpawnRules::checkLightBoundaries);
         }
 
         public boolean isValidPosition(BlockPos p_321770_, ServerLevel p_321818_) {
