--- a/net/minecraft/world/level/WorldDataConfiguration.java
+++ b/net/minecraft/world/level/WorldDataConfiguration.java
@@ -9,11 +_,13 @@
 public record WorldDataConfiguration(DataPackConfig dataPacks, FeatureFlagSet enabledFeatures) {
     public static final String ENABLED_FEATURES_ID = "enabled_features";
     public static final Codec<WorldDataConfiguration> CODEC = RecordCodecBuilder.create(
-        p_249474_ -> p_249474_.group(
-                    DataPackConfig.CODEC.optionalFieldOf("DataPacks", DataPackConfig.DEFAULT).forGetter(WorldDataConfiguration::dataPacks),
-                    FeatureFlags.CODEC.optionalFieldOf("enabled_features", FeatureFlags.DEFAULT_FLAGS).forGetter(WorldDataConfiguration::enabledFeatures)
+        p_337972_ -> p_337972_.group(
+                    DataPackConfig.CODEC.lenientOptionalFieldOf("DataPacks", DataPackConfig.DEFAULT).forGetter(WorldDataConfiguration::dataPacks),
+                    FeatureFlags.CODEC
+                        .lenientOptionalFieldOf("enabled_features", FeatureFlags.DEFAULT_FLAGS)
+                        .forGetter(WorldDataConfiguration::enabledFeatures)
                 )
-                .apply(p_249474_, WorldDataConfiguration::new)
+                .apply(p_337972_, WorldDataConfiguration::new)
     );
     public static final WorldDataConfiguration DEFAULT = new WorldDataConfiguration(DataPackConfig.DEFAULT, FeatureFlags.DEFAULT_FLAGS);
 
