--- a/net/minecraft/data/worldgen/placement/CavePlacements.java
+++ b/net/minecraft/data/worldgen/placement/CavePlacements.java
@@ -16,7 +_,6 @@
 import net.minecraft.world.level.levelgen.blockpredicates.BlockPredicate;
 import net.minecraft.world.level.levelgen.feature.ConfiguredFeature;
 import net.minecraft.world.level.levelgen.placement.BiomeFilter;
-import net.minecraft.world.level.levelgen.placement.CountOnEveryLayerPlacement;
 import net.minecraft.world.level.levelgen.placement.CountPlacement;
 import net.minecraft.world.level.levelgen.placement.EnvironmentScanPlacement;
 import net.minecraft.world.level.levelgen.placement.HeightRangePlacement;
@@ -33,7 +_,6 @@
     public static final ResourceKey<PlacedFeature> FOSSIL_LOWER = PlacementUtils.createKey("fossil_lower");
     public static final ResourceKey<PlacedFeature> DRIPSTONE_CLUSTER = PlacementUtils.createKey("dripstone_cluster");
     public static final ResourceKey<PlacedFeature> LARGE_DRIPSTONE = PlacementUtils.createKey("large_dripstone");
-    public static final ResourceKey<PlacedFeature> LARGE_POTATOSTONE = PlacementUtils.createKey("large_potatostone");
     public static final ResourceKey<PlacedFeature> POINTED_DRIPSTONE = PlacementUtils.createKey("pointed_dripstone");
     public static final ResourceKey<PlacedFeature> UNDERWATER_MAGMA = PlacementUtils.createKey("underwater_magma");
     public static final ResourceKey<PlacedFeature> GLOW_LICHEN = PlacementUtils.createKey("glow_lichen");
@@ -45,12 +_,9 @@
     public static final ResourceKey<PlacedFeature> SPORE_BLOSSOM = PlacementUtils.createKey("spore_blossom");
     public static final ResourceKey<PlacedFeature> CLASSIC_VINES = PlacementUtils.createKey("classic_vines_cave_feature");
     public static final ResourceKey<PlacedFeature> AMETHYST_GEODE = PlacementUtils.createKey("amethyst_geode");
-    public static final ResourceKey<PlacedFeature> POTATO_GEODE = PlacementUtils.createKey("potato_geode");
     public static final ResourceKey<PlacedFeature> SCULK_PATCH_DEEP_DARK = PlacementUtils.createKey("sculk_patch_deep_dark");
     public static final ResourceKey<PlacedFeature> SCULK_PATCH_ANCIENT_CITY = PlacementUtils.createKey("sculk_patch_ancient_city");
     public static final ResourceKey<PlacedFeature> SCULK_VEIN = PlacementUtils.createKey("sculk_vein");
-    public static final ResourceKey<PlacedFeature> POTATO_LEAF = PlacementUtils.createKey("potato_leaf");
-    public static final ResourceKey<PlacedFeature> TWISTED_POTATO = PlacementUtils.createKey("twisted_potato");
 
     public static void bootstrap(BootstrapContext<PlacedFeature> p_321491_) {
         HolderGetter<ConfiguredFeature<?, ?>> holdergetter = p_321491_.lookup(Registries.CONFIGURED_FEATURE);
@@ -59,24 +_,20 @@
         Holder<ConfiguredFeature<?, ?>> holder2 = holdergetter.getOrThrow(CaveFeatures.FOSSIL_DIAMONDS);
         Holder<ConfiguredFeature<?, ?>> holder3 = holdergetter.getOrThrow(CaveFeatures.DRIPSTONE_CLUSTER);
         Holder<ConfiguredFeature<?, ?>> holder4 = holdergetter.getOrThrow(CaveFeatures.LARGE_DRIPSTONE);
-        Holder<ConfiguredFeature<?, ?>> holder5 = holdergetter.getOrThrow(CaveFeatures.LARGE_POTATOSTONE);
-        Holder<ConfiguredFeature<?, ?>> holder6 = holdergetter.getOrThrow(CaveFeatures.POINTED_DRIPSTONE);
-        Holder<ConfiguredFeature<?, ?>> holder7 = holdergetter.getOrThrow(CaveFeatures.UNDERWATER_MAGMA);
-        Holder<ConfiguredFeature<?, ?>> holder8 = holdergetter.getOrThrow(CaveFeatures.GLOW_LICHEN);
-        Holder<ConfiguredFeature<?, ?>> holder9 = holdergetter.getOrThrow(CaveFeatures.ROOTED_AZALEA_TREE);
-        Holder<ConfiguredFeature<?, ?>> holder10 = holdergetter.getOrThrow(CaveFeatures.CAVE_VINE);
-        Holder<ConfiguredFeature<?, ?>> holder11 = holdergetter.getOrThrow(CaveFeatures.MOSS_PATCH);
-        Holder<ConfiguredFeature<?, ?>> holder12 = holdergetter.getOrThrow(CaveFeatures.LUSH_CAVES_CLAY);
-        Holder<ConfiguredFeature<?, ?>> holder13 = holdergetter.getOrThrow(CaveFeatures.MOSS_PATCH_CEILING);
-        Holder<ConfiguredFeature<?, ?>> holder14 = holdergetter.getOrThrow(CaveFeatures.SPORE_BLOSSOM);
-        Holder<ConfiguredFeature<?, ?>> holder15 = holdergetter.getOrThrow(VegetationFeatures.VINES);
-        Holder<ConfiguredFeature<?, ?>> holder16 = holdergetter.getOrThrow(CaveFeatures.AMETHYST_GEODE);
-        Holder<ConfiguredFeature<?, ?>> holder17 = holdergetter.getOrThrow(CaveFeatures.POTATO_GEODE);
-        Holder<ConfiguredFeature<?, ?>> holder18 = holdergetter.getOrThrow(CaveFeatures.SCULK_PATCH_DEEP_DARK);
-        Holder<ConfiguredFeature<?, ?>> holder19 = holdergetter.getOrThrow(CaveFeatures.SCULK_PATCH_ANCIENT_CITY);
-        Holder<ConfiguredFeature<?, ?>> holder20 = holdergetter.getOrThrow(CaveFeatures.SCULK_VEIN);
-        Holder<ConfiguredFeature<?, ?>> holder21 = holdergetter.getOrThrow(CaveFeatures.POTATO_LEAF);
-        Holder<ConfiguredFeature<?, ?>> holder22 = holdergetter.getOrThrow(CaveFeatures.TWISTED_POTATO);
+        Holder<ConfiguredFeature<?, ?>> holder5 = holdergetter.getOrThrow(CaveFeatures.POINTED_DRIPSTONE);
+        Holder<ConfiguredFeature<?, ?>> holder6 = holdergetter.getOrThrow(CaveFeatures.UNDERWATER_MAGMA);
+        Holder<ConfiguredFeature<?, ?>> holder7 = holdergetter.getOrThrow(CaveFeatures.GLOW_LICHEN);
+        Holder<ConfiguredFeature<?, ?>> holder8 = holdergetter.getOrThrow(CaveFeatures.ROOTED_AZALEA_TREE);
+        Holder<ConfiguredFeature<?, ?>> holder9 = holdergetter.getOrThrow(CaveFeatures.CAVE_VINE);
+        Holder<ConfiguredFeature<?, ?>> holder10 = holdergetter.getOrThrow(CaveFeatures.MOSS_PATCH);
+        Holder<ConfiguredFeature<?, ?>> holder11 = holdergetter.getOrThrow(CaveFeatures.LUSH_CAVES_CLAY);
+        Holder<ConfiguredFeature<?, ?>> holder12 = holdergetter.getOrThrow(CaveFeatures.MOSS_PATCH_CEILING);
+        Holder<ConfiguredFeature<?, ?>> holder13 = holdergetter.getOrThrow(CaveFeatures.SPORE_BLOSSOM);
+        Holder<ConfiguredFeature<?, ?>> holder14 = holdergetter.getOrThrow(VegetationFeatures.VINES);
+        Holder<ConfiguredFeature<?, ?>> holder15 = holdergetter.getOrThrow(CaveFeatures.AMETHYST_GEODE);
+        Holder<ConfiguredFeature<?, ?>> holder16 = holdergetter.getOrThrow(CaveFeatures.SCULK_PATCH_DEEP_DARK);
+        Holder<ConfiguredFeature<?, ?>> holder17 = holdergetter.getOrThrow(CaveFeatures.SCULK_PATCH_ANCIENT_CITY);
+        Holder<ConfiguredFeature<?, ?>> holder18 = holdergetter.getOrThrow(CaveFeatures.SCULK_VEIN);
         PlacementUtils.register(
             p_321491_,
             MONSTER_ROOM,
@@ -133,17 +_,8 @@
         );
         PlacementUtils.register(
             p_321491_,
-            LARGE_POTATOSTONE,
-            holder5,
-            CountPlacement.of(UniformInt.of(5, 10)),
-            InSquarePlacement.spread(),
-            PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
-            BiomeFilter.biome()
-        );
-        PlacementUtils.register(
-            p_321491_,
             POINTED_DRIPSTONE,
-            holder6,
+            holder5,
             CountPlacement.of(UniformInt.of(192, 256)),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -154,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             UNDERWATER_MAGMA,
-            holder7,
+            holder6,
             CountPlacement.of(UniformInt.of(44, 52)),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -164,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             GLOW_LICHEN,
-            holder8,
+            holder7,
             CountPlacement.of(UniformInt.of(104, 157)),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
             InSquarePlacement.spread(),
@@ -174,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             ROOTED_AZALEA_TREE,
-            holder9,
+            holder8,
             CountPlacement.of(UniformInt.of(1, 2)),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -185,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             CAVE_VINES,
-            holder10,
+            holder9,
             CountPlacement.of(188),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -196,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             LUSH_CAVES_VEGETATION,
-            holder11,
+            holder10,
             CountPlacement.of(125),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -207,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             LUSH_CAVES_CLAY,
-            holder12,
+            holder11,
             CountPlacement.of(62),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -218,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             LUSH_CAVES_CEILING_VEGETATION,
-            holder13,
+            holder12,
             CountPlacement.of(125),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -229,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             SPORE_BLOSSOM,
-            holder14,
+            holder13,
             CountPlacement.of(25),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -240,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             CLASSIC_VINES,
-            holder15,
+            holder14,
             CountPlacement.of(256),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
@@ -249,7 +_,7 @@
         PlacementUtils.register(
             p_321491_,
             AMETHYST_GEODE,
-            holder16,
+            holder15,
             RarityFilter.onAverageOnceEvery(24),
             InSquarePlacement.spread(),
             HeightRangePlacement.uniform(VerticalAnchor.aboveBottom(6), VerticalAnchor.absolute(30)),
@@ -257,33 +_,22 @@
         );
         PlacementUtils.register(
             p_321491_,
-            POTATO_GEODE,
-            holder17,
-            RarityFilter.onAverageOnceEvery(18),
-            InSquarePlacement.spread(),
-            HeightRangePlacement.uniform(VerticalAnchor.aboveBottom(6), VerticalAnchor.absolute(30)),
-            BiomeFilter.biome()
-        );
-        PlacementUtils.register(
-            p_321491_,
             SCULK_PATCH_DEEP_DARK,
-            holder18,
+            holder16,
             CountPlacement.of(ConstantInt.of(256)),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
             BiomeFilter.biome()
         );
-        PlacementUtils.register(p_321491_, SCULK_PATCH_ANCIENT_CITY, holder19);
+        PlacementUtils.register(p_321491_, SCULK_PATCH_ANCIENT_CITY, holder17);
         PlacementUtils.register(
             p_321491_,
             SCULK_VEIN,
-            holder20,
+            holder18,
             CountPlacement.of(UniformInt.of(204, 250)),
             InSquarePlacement.spread(),
             PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT,
             BiomeFilter.biome()
         );
-        PlacementUtils.register(p_321491_, POTATO_LEAF, holder21, CountOnEveryLayerPlacement.of(1), RarityFilter.onAverageOnceEvery(5), BiomeFilter.biome());
-        PlacementUtils.register(p_321491_, TWISTED_POTATO, holder22, CountOnEveryLayerPlacement.of(1), RarityFilter.onAverageOnceEvery(13), BiomeFilter.biome());
     }
 }
