--- a/net/minecraft/data/worldgen/placement/VillagePlacements.java
+++ b/net/minecraft/data/worldgen/placement/VillagePlacements.java
@@ -15,12 +_,10 @@
 public class VillagePlacements {
     public static final ResourceKey<PlacedFeature> PILE_HAY_VILLAGE = PlacementUtils.createKey("pile_hay");
     public static final ResourceKey<PlacedFeature> PILE_MELON_VILLAGE = PlacementUtils.createKey("pile_melon");
-    public static final ResourceKey<PlacedFeature> PILE_POTATO_FRUIT_VILLAGE = PlacementUtils.createKey("pile_potato_fruit");
     public static final ResourceKey<PlacedFeature> PILE_SNOW_VILLAGE = PlacementUtils.createKey("pile_snow");
     public static final ResourceKey<PlacedFeature> PILE_ICE_VILLAGE = PlacementUtils.createKey("pile_ice");
     public static final ResourceKey<PlacedFeature> PILE_PUMPKIN_VILLAGE = PlacementUtils.createKey("pile_pumpkin");
     public static final ResourceKey<PlacedFeature> OAK_VILLAGE = PlacementUtils.createKey("oak");
-    public static final ResourceKey<PlacedFeature> POTATO_VILLAGE = PlacementUtils.createKey("potato");
     public static final ResourceKey<PlacedFeature> ACACIA_VILLAGE = PlacementUtils.createKey("acacia");
     public static final ResourceKey<PlacedFeature> SPRUCE_VILLAGE = PlacementUtils.createKey("spruce");
     public static final ResourceKey<PlacedFeature> PINE_VILLAGE = PlacementUtils.createKey("pine");
@@ -32,34 +_,30 @@
     public static void bootstrap(BootstrapContext<PlacedFeature> p_321555_) {
         HolderGetter<ConfiguredFeature<?, ?>> holdergetter = p_321555_.lookup(Registries.CONFIGURED_FEATURE);
         Holder<ConfiguredFeature<?, ?>> holder = holdergetter.getOrThrow(PileFeatures.PILE_HAY);
-        Holder<ConfiguredFeature<?, ?>> holder1 = holdergetter.getOrThrow(PileFeatures.PILE_POTATO_FRUIT);
-        Holder<ConfiguredFeature<?, ?>> holder2 = holdergetter.getOrThrow(PileFeatures.PILE_MELON);
-        Holder<ConfiguredFeature<?, ?>> holder3 = holdergetter.getOrThrow(PileFeatures.PILE_SNOW);
-        Holder<ConfiguredFeature<?, ?>> holder4 = holdergetter.getOrThrow(PileFeatures.PILE_ICE);
-        Holder<ConfiguredFeature<?, ?>> holder5 = holdergetter.getOrThrow(PileFeatures.PILE_PUMPKIN);
-        Holder<ConfiguredFeature<?, ?>> holder6 = holdergetter.getOrThrow(TreeFeatures.OAK);
-        Holder<ConfiguredFeature<?, ?>> holder7 = holdergetter.getOrThrow(TreeFeatures.POTATO_TREE);
-        Holder<ConfiguredFeature<?, ?>> holder8 = holdergetter.getOrThrow(TreeFeatures.ACACIA);
-        Holder<ConfiguredFeature<?, ?>> holder9 = holdergetter.getOrThrow(TreeFeatures.SPRUCE);
-        Holder<ConfiguredFeature<?, ?>> holder10 = holdergetter.getOrThrow(TreeFeatures.PINE);
-        Holder<ConfiguredFeature<?, ?>> holder11 = holdergetter.getOrThrow(VegetationFeatures.PATCH_CACTUS);
-        Holder<ConfiguredFeature<?, ?>> holder12 = holdergetter.getOrThrow(VegetationFeatures.FLOWER_PLAIN);
-        Holder<ConfiguredFeature<?, ?>> holder13 = holdergetter.getOrThrow(VegetationFeatures.PATCH_TAIGA_GRASS);
-        Holder<ConfiguredFeature<?, ?>> holder14 = holdergetter.getOrThrow(VegetationFeatures.PATCH_BERRY_BUSH);
+        Holder<ConfiguredFeature<?, ?>> holder1 = holdergetter.getOrThrow(PileFeatures.PILE_MELON);
+        Holder<ConfiguredFeature<?, ?>> holder2 = holdergetter.getOrThrow(PileFeatures.PILE_SNOW);
+        Holder<ConfiguredFeature<?, ?>> holder3 = holdergetter.getOrThrow(PileFeatures.PILE_ICE);
+        Holder<ConfiguredFeature<?, ?>> holder4 = holdergetter.getOrThrow(PileFeatures.PILE_PUMPKIN);
+        Holder<ConfiguredFeature<?, ?>> holder5 = holdergetter.getOrThrow(TreeFeatures.OAK);
+        Holder<ConfiguredFeature<?, ?>> holder6 = holdergetter.getOrThrow(TreeFeatures.ACACIA);
+        Holder<ConfiguredFeature<?, ?>> holder7 = holdergetter.getOrThrow(TreeFeatures.SPRUCE);
+        Holder<ConfiguredFeature<?, ?>> holder8 = holdergetter.getOrThrow(TreeFeatures.PINE);
+        Holder<ConfiguredFeature<?, ?>> holder9 = holdergetter.getOrThrow(VegetationFeatures.PATCH_CACTUS);
+        Holder<ConfiguredFeature<?, ?>> holder10 = holdergetter.getOrThrow(VegetationFeatures.FLOWER_PLAIN);
+        Holder<ConfiguredFeature<?, ?>> holder11 = holdergetter.getOrThrow(VegetationFeatures.PATCH_TAIGA_GRASS);
+        Holder<ConfiguredFeature<?, ?>> holder12 = holdergetter.getOrThrow(VegetationFeatures.PATCH_BERRY_BUSH);
         PlacementUtils.register(p_321555_, PILE_HAY_VILLAGE, holder);
-        PlacementUtils.register(p_321555_, PILE_POTATO_FRUIT_VILLAGE, holder1);
-        PlacementUtils.register(p_321555_, PILE_MELON_VILLAGE, holder2);
-        PlacementUtils.register(p_321555_, PILE_SNOW_VILLAGE, holder3);
-        PlacementUtils.register(p_321555_, PILE_ICE_VILLAGE, holder4);
-        PlacementUtils.register(p_321555_, PILE_PUMPKIN_VILLAGE, holder5);
-        PlacementUtils.register(p_321555_, OAK_VILLAGE, holder6, PlacementUtils.filteredByBlockSurvival(Blocks.OAK_SAPLING));
-        PlacementUtils.register(p_321555_, POTATO_VILLAGE, holder7, PlacementUtils.filteredByBlockSurvival(Blocks.POTATO_SPROUTS));
-        PlacementUtils.register(p_321555_, ACACIA_VILLAGE, holder8, PlacementUtils.filteredByBlockSurvival(Blocks.ACACIA_SAPLING));
-        PlacementUtils.register(p_321555_, SPRUCE_VILLAGE, holder9, PlacementUtils.filteredByBlockSurvival(Blocks.SPRUCE_SAPLING));
-        PlacementUtils.register(p_321555_, PINE_VILLAGE, holder10, PlacementUtils.filteredByBlockSurvival(Blocks.SPRUCE_SAPLING));
-        PlacementUtils.register(p_321555_, PATCH_CACTUS_VILLAGE, holder11);
-        PlacementUtils.register(p_321555_, FLOWER_PLAIN_VILLAGE, holder12);
-        PlacementUtils.register(p_321555_, PATCH_TAIGA_GRASS_VILLAGE, holder13);
-        PlacementUtils.register(p_321555_, PATCH_BERRY_BUSH_VILLAGE, holder14);
+        PlacementUtils.register(p_321555_, PILE_MELON_VILLAGE, holder1);
+        PlacementUtils.register(p_321555_, PILE_SNOW_VILLAGE, holder2);
+        PlacementUtils.register(p_321555_, PILE_ICE_VILLAGE, holder3);
+        PlacementUtils.register(p_321555_, PILE_PUMPKIN_VILLAGE, holder4);
+        PlacementUtils.register(p_321555_, OAK_VILLAGE, holder5, PlacementUtils.filteredByBlockSurvival(Blocks.OAK_SAPLING));
+        PlacementUtils.register(p_321555_, ACACIA_VILLAGE, holder6, PlacementUtils.filteredByBlockSurvival(Blocks.ACACIA_SAPLING));
+        PlacementUtils.register(p_321555_, SPRUCE_VILLAGE, holder7, PlacementUtils.filteredByBlockSurvival(Blocks.SPRUCE_SAPLING));
+        PlacementUtils.register(p_321555_, PINE_VILLAGE, holder8, PlacementUtils.filteredByBlockSurvival(Blocks.SPRUCE_SAPLING));
+        PlacementUtils.register(p_321555_, PATCH_CACTUS_VILLAGE, holder9);
+        PlacementUtils.register(p_321555_, FLOWER_PLAIN_VILLAGE, holder10);
+        PlacementUtils.register(p_321555_, PATCH_TAIGA_GRASS_VILLAGE, holder11);
+        PlacementUtils.register(p_321555_, PATCH_BERRY_BUSH_VILLAGE, holder12);
     }
 }
