--- a/net/minecraft/server/packs/repository/FolderRepositorySource.java
+++ b/net/minecraft/server/packs/repository/FolderRepositorySource.java
@@ -70,7 +_,7 @@
         FolderRepositorySource.FolderPackDetector folderrepositorysource$folderpackdetector = new FolderRepositorySource.FolderPackDetector(p_294483_);
 
         try (DirectoryStream<Path> directorystream = Files.newDirectoryStream(p_248794_)) {
-            for(Path path : directorystream) {
+            for (Path path : directorystream) {
                 try {
                     List<ForbiddenSymlinkInfo> list = new ArrayList<>();
                     Pack.ResourcesSupplier pack$resourcessupplier = folderrepositorysource$folderpackdetector.detectPackResources(path, list);
