--- a/net/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings.java
+++ b/net/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings.java
@@ -37,12 +_,12 @@
     public static final Codec<FlatLevelGeneratorSettings> CODEC = RecordCodecBuilder.<FlatLevelGeneratorSettings>create(
             p_209800_ -> p_209800_.group(
                         RegistryCodecs.homogeneousList(Registries.STRUCTURE_SET)
-                            .optionalFieldOf("structure_overrides")
+                            .lenientOptionalFieldOf("structure_overrides")
                             .forGetter(p_209812_ -> p_209812_.structureOverrides),
                         FlatLayerInfo.CODEC.listOf().fieldOf("layers").forGetter(FlatLevelGeneratorSettings::getLayersInfo),
                         Codec.BOOL.fieldOf("lakes").orElse(false).forGetter(p_161912_ -> p_161912_.addLakes),
                         Codec.BOOL.fieldOf("features").orElse(false).forGetter(p_209809_ -> p_209809_.decoration),
-                        Biome.CODEC.optionalFieldOf("biome").orElseGet(Optional::empty).forGetter(p_209807_ -> Optional.of(p_209807_.biome)),
+                        Biome.CODEC.lenientOptionalFieldOf("biome").orElseGet(Optional::empty).forGetter(p_209807_ -> Optional.of(p_209807_.biome)),
                         RegistryOps.retrieveElement(Biomes.PLAINS),
                         RegistryOps.retrieveElement(MiscOverworldPlacements.LAKE_LAVA_UNDERGROUND),
                         RegistryOps.retrieveElement(MiscOverworldPlacements.LAKE_LAVA_SURFACE)
@@ -95,7 +_,7 @@
             LOGGER.error("Unknown biome, defaulting to plains");
             return p_256475_;
         } else {
-            return p_256142_.get();
+            return (Holder<Biome>)p_256142_.get();
         }
     }
 
@@ -109,7 +_,7 @@
     public FlatLevelGeneratorSettings withBiomeAndLayers(List<FlatLayerInfo> p_256587_, Optional<HolderSet<StructureSet>> p_256500_, Holder<Biome> p_256598_) {
         FlatLevelGeneratorSettings flatlevelgeneratorsettings = new FlatLevelGeneratorSettings(p_256500_, p_256598_, this.lakes);
 
-        for(FlatLayerInfo flatlayerinfo : p_256587_) {
+        for (FlatLayerInfo flatlayerinfo : p_256587_) {
             flatlevelgeneratorsettings.layersInfo.add(new FlatLayerInfo(flatlayerinfo.getHeight(), flatlayerinfo.getBlockState().getBlock()));
             flatlevelgeneratorsettings.updateLayers();
         }
@@ -140,7 +_,7 @@
             BiomeGenerationSettings biomegenerationsettings = this.getBiome().value().getGenerationSettings();
             BiomeGenerationSettings.PlainBuilder biomegenerationsettings$plainbuilder = new BiomeGenerationSettings.PlainBuilder();
             if (this.addLakes) {
-                for(Holder<PlacedFeature> holder : this.lakes) {
+                for (Holder<PlacedFeature> holder : this.lakes) {
                     biomegenerationsettings$plainbuilder.addFeature(GenerationStep.Decoration.LAKES, holder);
                 }
             }
@@ -149,11 +_,11 @@
             if (flag) {
                 List<HolderSet<PlacedFeature>> list = biomegenerationsettings.features();
 
-                for(int i = 0; i < list.size(); ++i) {
+                for (int i = 0; i < list.size(); i++) {
                     if (i != GenerationStep.Decoration.UNDERGROUND_STRUCTURES.ordinal()
                         && i != GenerationStep.Decoration.SURFACE_STRUCTURES.ordinal()
                         && (!this.addLakes || i != GenerationStep.Decoration.LAKES.ordinal())) {
-                        for(Holder<PlacedFeature> holder1 : list.get(i)) {
+                        for (Holder<PlacedFeature> holder1 : list.get(i)) {
                             biomegenerationsettings$plainbuilder.addFeature(i, holder1);
                         }
                     }
@@ -162,7 +_,7 @@
 
             List<BlockState> list1 = this.getLayers();
 
-            for(int j = 0; j < list1.size(); ++j) {
+            for (int j = 0; j < list1.size(); j++) {
                 BlockState blockstate = list1.get(j);
                 if (!Heightmap.Types.MOTION_BLOCKING.isOpaque().test(blockstate)) {
                     list1.set(j, null);
@@ -196,8 +_,8 @@
     public void updateLayers() {
         this.layers.clear();
 
-        for(FlatLayerInfo flatlayerinfo : this.layersInfo) {
-            for(int i = 0; i < flatlayerinfo.getHeight(); ++i) {
+        for (FlatLayerInfo flatlayerinfo : this.layersInfo) {
+            for (int i = 0; i < flatlayerinfo.getHeight(); i++) {
                 this.layers.add(flatlayerinfo.getBlockState());
             }
         }
