--- a/net/minecraft/world/flag/FeatureElement.java
+++ b/net/minecraft/world/flag/FeatureElement.java
@@ -7,7 +_,7 @@
 
 public interface FeatureElement {
     Set<ResourceKey<? extends Registry<? extends FeatureElement>>> FILTERED_REGISTRIES = Set.of(
-        Registries.ITEM, Registries.BLOCK, Registries.ENTITY_TYPE, Registries.MENU
+        Registries.ITEM, Registries.BLOCK, Registries.ENTITY_TYPE, Registries.MENU, Registries.ENCHANTMENT, Registries.POTION, Registries.MOB_EFFECT
     );
 
     FeatureFlagSet requiredFeatures();
