--- a/net/minecraft/world/flag/FeatureFlagSet.java
+++ b/net/minecraft/world/flag/FeatureFlagSet.java
@@ -40,7 +_,7 @@
     }
 
     private static long computeMask(FeatureFlagUniverse p_249684_, long p_250982_, Iterable<FeatureFlag> p_251734_) {
-        for(FeatureFlag featureflag : p_251734_) {
+        for (FeatureFlag featureflag : p_251734_) {
             if (p_249684_ != featureflag.universe) {
                 throw new IllegalStateException("Mismatched feature universe, expected '" + p_249684_ + "', but got '" + featureflag.universe + "'");
             }
@@ -52,11 +_,7 @@
     }
 
     public boolean contains(FeatureFlag p_249521_) {
-        if (this.universe != p_249521_.universe) {
-            return false;
-        } else {
-            return (this.mask & p_249521_.mask) != 0L;
-        }
+        return this.universe != p_249521_.universe ? false : (this.mask & p_249521_.mask) != 0L;
     }
 
     public boolean isEmpty() {
@@ -66,10 +_,8 @@
     public boolean isSubsetOf(FeatureFlagSet p_249164_) {
         if (this.universe == null) {
             return true;
-        } else if (this.universe != p_249164_.universe) {
-            return false;
         } else {
-            return (this.mask & ~p_249164_.mask) == 0L;
+            return this.universe != p_249164_.universe ? false : (this.mask & ~p_249164_.mask) == 0L;
         }
     }
 
