--- a/net/minecraft/client/multiplayer/chat/report/ReportEnvironment.java
+++ b/net/minecraft/client/multiplayer/chat/report/ReportEnvironment.java
@@ -34,23 +_,21 @@
 
     @Nullable
     public ThirdPartyServerInfo thirdPartyServerInfo() {
-        ReportEnvironment.Server reportenvironment$server = this.server;
-        return reportenvironment$server instanceof ReportEnvironment.Server.ThirdParty reportenvironment$server$thirdparty
+        return this.server instanceof ReportEnvironment.Server.ThirdParty reportenvironment$server$thirdparty
             ? new ThirdPartyServerInfo(reportenvironment$server$thirdparty.ip)
             : null;
     }
 
     @Nullable
     public RealmInfo realmInfo() {
-        ReportEnvironment.Server reportenvironment$server = this.server;
-        return reportenvironment$server instanceof ReportEnvironment.Server.Realm reportenvironment$server$realm
+        return this.server instanceof ReportEnvironment.Server.Realm reportenvironment$server$realm
             ? new RealmInfo(String.valueOf(reportenvironment$server$realm.realmId()), reportenvironment$server$realm.slotId())
             : null;
     }
 
     private static String getClientVersion() {
         StringBuilder stringbuilder = new StringBuilder();
-        stringbuilder.append("24w14potato");
+        stringbuilder.append("24w14a");
         if (Minecraft.checkModStatus().shouldReportAsModified()) {
             stringbuilder.append(" (modded)");
         }
