--- a/net/minecraft/world/entity/animal/Panda.java
+++ b/net/minecraft/world/entity/animal/Panda.java
@@ -100,9 +_,9 @@
     private float rollAmount;
     private float rollAmountO;
     Panda.PandaLookAtPlayerGoal lookAtPlayerGoal;
-    static final Predicate<ItemEntity> PANDA_ITEMS = p_340111_ -> {
-        ItemStack itemstack = p_340111_.getItem();
-        return (itemstack.is(Blocks.BAMBOO.asItem()) || itemstack.is(Blocks.CAKE.asItem())) && p_340111_.isAlive() && !p_340111_.hasPickUpDelay();
+    static final Predicate<ItemEntity> PANDA_ITEMS = p_337831_ -> {
+        ItemStack itemstack = p_337831_.getItem();
+        return (itemstack.is(Blocks.BAMBOO.asItem()) || itemstack.is(Blocks.CAKE.asItem())) && p_337831_.isAlive() && !p_337831_.hasPickUpDelay();
     };
 
     public Panda(EntityType<? extends Panda> p_29086_, Level p_29087_) {
@@ -116,11 +_,7 @@
     @Override
     public boolean canTakeItem(ItemStack p_29146_) {
         EquipmentSlot equipmentslot = Mob.getEquipmentSlotForItem(p_29146_);
-        if (!this.getItemBySlot(equipmentslot).isEmpty()) {
-            return false;
-        } else {
-            return equipmentslot == EquipmentSlot.MAINHAND && super.canTakeItem(p_29146_);
-        }
+        return !this.getItemBySlot(equipmentslot).isEmpty() ? false : equipmentslot == EquipmentSlot.MAINHAND && super.canTakeItem(p_29146_);
     }
 
     public int getUnhappyCounter() {
@@ -429,7 +_,7 @@
                 SoundEvents.PANDA_EAT, 0.5F + 0.5F * (float)this.random.nextInt(2), (this.random.nextFloat() - this.random.nextFloat()) * 0.2F + 1.0F
             );
 
-            for(int i = 0; i < 6; ++i) {
+            for (int i = 0; i < 6; i++) {
                 Vec3 vec3 = new Vec3(((double)this.random.nextFloat() - 0.5) * 0.1, Math.random() * 0.1 + 0.1, ((double)this.random.nextFloat() - 0.5) * 0.1);
                 vec3 = vec3.xRot(-this.getXRot() * (float) (Math.PI / 180.0));
                 vec3 = vec3.yRot(-this.getYRot() * (float) (Math.PI / 180.0));
@@ -491,7 +_,7 @@
     }
 
     private void handleRoll() {
-        ++this.rollCounter;
+        this.rollCounter++;
         if (this.rollCounter > 32) {
             this.roll(false);
         } else {
@@ -525,7 +_,7 @@
         );
         this.playSound(SoundEvents.PANDA_SNEEZE, 1.0F, 1.0F);
 
-        for(Panda panda : level.getEntitiesOfClass(Panda.class, this.getBoundingBox().inflate(10.0))) {
+        for (Panda panda : level.getEntitiesOfClass(Panda.class, this.getBoundingBox().inflate(10.0))) {
             if (!panda.isBaby() && panda.onGround() && !panda.isInWater() && panda.canPerformAction()) {
                 panda.jumpFromGround();
             }
@@ -539,7 +_,7 @@
                 .withParameter(LootContextParams.THIS_ENTITY, this)
                 .create(LootContextParamSets.GIFT);
 
-            for(ItemStack itemstack : loottable.getRandomItems(lootparams)) {
+            for (ItemStack itemstack : loottable.getRandomItems(lootparams)) {
                 this.spawnAtLocation(itemstack);
             }
         }
@@ -844,10 +_,10 @@
             BlockPos blockpos = this.panda.blockPosition();
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(int i = 0; i < 3; ++i) {
-                for(int j = 0; j < 8; ++j) {
-                    for(int k = 0; k <= j; k = k > 0 ? -k : 1 - k) {
-                        for(int l = k < j && k > -j ? j : 0; l <= j; l = l > 0 ? -l : 1 - l) {
+            for (int i = 0; i < 3; i++) {
+                for (int j = 0; j < 8; j++) {
+                    for (int k = 0; k <= j; k = k > 0 ? -k : 1 - k) {
+                        for (int l = k < j && k > -j ? j : 0; l <= j; l = l > 0 ? -l : 1 - l) {
                             blockpos$mutableblockpos.setWithOffset(blockpos, k, i, l);
                             if (this.level.getBlockState(blockpos$mutableblockpos).is(Blocks.BAMBOO)) {
                                 return true;
@@ -905,11 +_,9 @@
 
         @Override
         public boolean canContinueToUse() {
-            if (!this.panda.isInWater() && (this.panda.isLazy() || this.panda.random.nextInt(reducedTickDelay(600)) != 1)) {
-                return this.panda.random.nextInt(reducedTickDelay(2000)) != 1;
-            } else {
-                return false;
-            }
+            return !this.panda.isInWater() && (this.panda.isLazy() || this.panda.random.nextInt(reducedTickDelay(600)) != 1)
+                ? this.panda.random.nextInt(reducedTickDelay(2000)) != 1
+                : false;
         }
 
         @Override
@@ -1043,10 +_,10 @@
                     int j = (double)Math.abs(f2) > 0.5 ? Mth.sign((double)f2) : 0;
                     if (this.panda.level().getBlockState(this.panda.blockPosition().offset(i, -1, j)).isAir()) {
                         return true;
-                    } else if (this.panda.isPlayful() && this.panda.random.nextInt(reducedTickDelay(60)) == 1) {
-                        return true;
                     } else {
-                        return this.panda.random.nextInt(reducedTickDelay(500)) == 1;
+                        return this.panda.isPlayful() && this.panda.random.nextInt(reducedTickDelay(60)) == 1
+                            ? true
+                            : this.panda.random.nextInt(reducedTickDelay(500)) == 1;
                     }
                 }
             } else {
@@ -1094,11 +_,9 @@
 
         @Override
         public boolean canContinueToUse() {
-            if (!Panda.this.isInWater() && (Panda.this.isLazy() || Panda.this.random.nextInt(reducedTickDelay(600)) != 1)) {
-                return Panda.this.random.nextInt(reducedTickDelay(2000)) != 1;
-            } else {
-                return false;
-            }
+            return !Panda.this.isInWater() && (Panda.this.isLazy() || Panda.this.random.nextInt(reducedTickDelay(600)) != 1)
+                ? Panda.this.random.nextInt(reducedTickDelay(2000)) != 1
+                : false;
         }
 
         @Override
@@ -1145,11 +_,9 @@
         @Override
         public boolean canUse() {
             if (this.panda.isBaby() && this.panda.canPerformAction()) {
-                if (this.panda.isWeak() && this.panda.random.nextInt(reducedTickDelay(500)) == 1) {
-                    return true;
-                } else {
-                    return this.panda.random.nextInt(reducedTickDelay(6000)) == 1;
-                }
+                return this.panda.isWeak() && this.panda.random.nextInt(reducedTickDelay(500)) == 1
+                    ? true
+                    : this.panda.random.nextInt(reducedTickDelay(6000)) == 1;
             } else {
                 return false;
             }
