--- a/net/minecraft/world/entity/animal/Pufferfish.java
+++ b/net/minecraft/world/entity/animal/Pufferfish.java
@@ -30,12 +_,12 @@
     private static final EntityDataAccessor<Integer> PUFF_STATE = SynchedEntityData.defineId(Pufferfish.class, EntityDataSerializers.INT);
     int inflateCounter;
     int deflateTimer;
-    private static final Predicate<LivingEntity> SCARY_MOB = p_340113_ -> {
-        if (p_340113_ instanceof Player player && player.isCreative()) {
+    private static final Predicate<LivingEntity> SCARY_MOB = p_337832_ -> {
+        if (p_337832_ instanceof Player player && player.isCreative()) {
             return false;
         }
 
-        return !p_340113_.getType().is(EntityTypeTags.NOT_SCARY_FOR_PUFFERFISH);
+        return !p_337832_.getType().is(EntityTypeTags.NOT_SCARY_FOR_PUFFERFISH);
     };
     static final TargetingConditions targetingConditions = TargetingConditions.forNonCombat()
         .ignoreInvisibilityTesting()
@@ -108,7 +_,7 @@
                     this.setPuffState(2);
                 }
 
-                ++this.inflateCounter;
+                this.inflateCounter++;
             } else if (this.getPuffState() != 0) {
                 if (this.deflateTimer > 60 && this.getPuffState() == 2) {
                     this.makeSound(SoundEvents.PUFFER_FISH_BLOW_OUT);
@@ -118,7 +_,7 @@
                     this.setPuffState(0);
                 }
 
-                ++this.deflateTimer;
+                this.deflateTimer++;
             }
         }
 
@@ -129,7 +_,7 @@
     public void aiStep() {
         super.aiStep();
         if (this.isAlive() && this.getPuffState() > 0) {
-            for(Mob mob : this.level()
+            for (Mob mob : this.level()
                 .getEntitiesOfClass(Mob.class, this.getBoundingBox().inflate(0.3), p_149013_ -> targetingConditions.test(this, p_149013_))) {
                 if (mob.isAlive()) {
                     this.touch(mob);
@@ -184,7 +_,7 @@
     }
 
     private static float getScale(int p_29639_) {
-        switch(p_29639_) {
+        switch (p_29639_) {
             case 0:
                 return 0.5F;
             case 1:
