--- a/net/minecraft/world/entity/animal/Sheep.java
+++ b/net/minecraft/world/entity/animal/Sheep.java
@@ -7,7 +_,6 @@
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 import net.minecraft.Util;
-import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.core.BlockPos;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.network.syncher.EntityDataAccessor;
@@ -15,7 +_,6 @@
 import net.minecraft.network.syncher.SynchedEntityData;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.sounds.SoundEvents;
 import net.minecraft.sounds.SoundSource;
@@ -53,7 +_,6 @@
 import net.minecraft.world.item.DyeItem;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
-import net.minecraft.world.item.PotatoPeelerItem;
 import net.minecraft.world.item.crafting.RecipeHolder;
 import net.minecraft.world.item.crafting.RecipeType;
 import net.minecraft.world.level.ItemLike;
@@ -87,10 +_,7 @@
         p_29841_.put(DyeColor.BLACK, Blocks.BLACK_WOOL);
     });
     private static final Map<DyeColor, float[]> COLORARRAY_BY_COLOR = Maps.<DyeColor, float[]>newEnumMap(
-        Arrays.stream(DyeColor.values()).collect(Collectors.toMap(p_29868_ -> p_29868_, Sheep::createSheepColor))
-    );
-    private static final Map<DyeColor, float[]> COLORARRAY_BY_POTATO = Maps.<DyeColor, float[]>newEnumMap(
-        Arrays.stream(DyeColor.values()).collect(Collectors.toMap(p_340118_ -> p_340118_, Sheep::createSheepColorPotato))
+        Arrays.stream(DyeColor.values()).collect(Collectors.toMap(p_29868_ -> (DyeColor)p_29868_, Sheep::createSheepColor))
     );
     private int eatAnimationTick;
     private EatBlockGoal eatBlockGoal;
@@ -105,12 +_,8 @@
         }
     }
 
-    private static float[] createSheepColorPotato(DyeColor p_341364_) {
-        return p_341364_ == DyeColor.WHITE ? new float[]{1.0F, 1.0F, 1.0F} : p_341364_.getTextureDiffuseColors();
-    }
-
-    public float[] getColorArray(DyeColor p_29830_) {
-        return this.isPotato() ? COLORARRAY_BY_POTATO.get(p_29830_) : COLORARRAY_BY_COLOR.get(p_29830_);
+    public static float[] getColorArray(DyeColor p_29830_) {
+        return COLORARRAY_BY_COLOR.get(p_29830_);
     }
 
     public Sheep(EntityType<? extends Sheep> p_29806_, Level p_29807_) {
@@ -118,11 +_,6 @@
     }
 
     @Override
-    public boolean hasPotatoVariant() {
-        return true;
-    }
-
-    @Override
     protected void registerGoals() {
         this.eatBlockGoal = new EatBlockGoal(this);
         this.goalSelector.addGoal(0, new FloatGoal(this));
@@ -170,27 +_,8 @@
     public ResourceKey<LootTable> getDefaultLootTable() {
         if (this.isSheared()) {
             return this.getType().getDefaultLootTable();
-        } else if (this.isPotato()) {
-            return switch(this.getColor()) {
-                case WHITE -> BuiltInLootTables.SHEEP_WHITE_POTATO;
-                case ORANGE -> BuiltInLootTables.SHEEP_ORANGE_POTATO;
-                case MAGENTA -> BuiltInLootTables.SHEEP_MAGENTA_POTATO;
-                case LIGHT_BLUE -> BuiltInLootTables.SHEEP_LIGHT_BLUE_POTATO;
-                case YELLOW -> BuiltInLootTables.SHEEP_YELLOW_POTATO;
-                case LIME -> BuiltInLootTables.SHEEP_LIME_POTATO;
-                case PINK -> BuiltInLootTables.SHEEP_PINK_POTATO;
-                case GRAY -> BuiltInLootTables.SHEEP_GRAY_POTATO;
-                case LIGHT_GRAY -> BuiltInLootTables.SHEEP_LIGHT_GRAY_POTATO;
-                case CYAN -> BuiltInLootTables.SHEEP_CYAN_POTATO;
-                case PURPLE -> BuiltInLootTables.SHEEP_PURPLE_POTATO;
-                case BLUE -> BuiltInLootTables.SHEEP_BLUE_POTATO;
-                case BROWN -> BuiltInLootTables.SHEEP_BROWN_POTATO;
-                case GREEN -> BuiltInLootTables.SHEEP_GREEN_POTATO;
-                case RED -> BuiltInLootTables.SHEEP_RED_POTATO;
-                case BLACK -> BuiltInLootTables.SHEEP_BLACK_POTATO;
-            };
         } else {
-            return switch(this.getColor()) {
+            return switch (this.getColor()) {
                 case WHITE -> BuiltInLootTables.SHEEP_WHITE;
                 case ORANGE -> BuiltInLootTables.SHEEP_ORANGE;
                 case MAGENTA -> BuiltInLootTables.SHEEP_MAGENTA;
@@ -207,6 +_,7 @@
                 case GREEN -> BuiltInLootTables.SHEEP_GREEN;
                 case RED -> BuiltInLootTables.SHEEP_RED;
                 case BLACK -> BuiltInLootTables.SHEEP_BLACK;
+                default -> throw new MatchException(null, null);
             };
         }
     }
@@ -242,14 +_,10 @@
     @Override
     public InteractionResult mobInteract(Player p_29853_, InteractionHand p_29854_) {
         ItemStack itemstack = p_29853_.getItemInHand(p_29854_);
-        if (this.isCorrectShears(itemstack)) {
+        if (false && itemstack.getItem() == Items.SHEARS) { //Forge: Moved to onSheared
             if (!this.level().isClientSide && this.readyForShearing()) {
                 this.shear(SoundSource.PLAYERS);
                 this.gameEvent(GameEvent.SHEAR, p_29853_);
-                if (this.isPotato() && p_29853_ instanceof ServerPlayer serverplayer) {
-                    CriteriaTriggers.PEEL_POTATO_SHEEP.trigger(serverplayer);
-                }
-
                 itemstack.hurtAndBreak(1, p_29853_, getSlotForHand(p_29854_));
                 return InteractionResult.SUCCESS;
             } else {
@@ -260,23 +_,14 @@
         }
     }
 
-    private boolean isCorrectShears(ItemStack p_340710_) {
-        return this.isPotato() ? p_340710_.is(Items.POTATO_PEELER) : p_340710_.is(Items.SHEARS);
-    }
-
     @Override
     public void shear(SoundSource p_29819_) {
-        if (this.isPotato()) {
-            PotatoPeelerItem.playPeelSound(this.level(), this, p_29819_);
-        } else {
-            this.level().playSound(null, this, SoundEvents.SHEEP_SHEAR, p_29819_, 1.0F, 1.0F);
-        }
-
+        this.level().playSound(null, this, SoundEvents.SHEEP_SHEAR, p_29819_, 1.0F, 1.0F);
         this.setSheared(true);
         int i = 1 + this.random.nextInt(3);
 
-        for(int j = 0; j < i; ++j) {
-            ItemEntity itementity = this.spawnAtLocation(this.getShearDrop(), 1);
+        for (int j = 0; j < i; j++) {
+            ItemEntity itementity = this.spawnAtLocation(ITEM_BY_DYE.get(this.getColor()), 1);
             if (itementity != null) {
                 itementity.setDeltaMovement(
                     itementity.getDeltaMovement()
@@ -290,10 +_,6 @@
         }
     }
 
-    private ItemLike getShearDrop() {
-        return this.isPotato() ? Items.POTATO_PEELS_MAP.get(this.getColor()) : ITEM_BY_DYE.get(this.getColor());
-    }
-
     @Override
     public boolean readyForShearing() {
         return this.isAlive() && !this.isSheared() && !this.isBaby();
@@ -403,7 +_,7 @@
         return this.level()
             .getRecipeManager()
             .getRecipeFor(RecipeType.CRAFTING, craftingcontainer, this.level())
-            .map(p_340117_ -> p_340117_.value().assemble(craftingcontainer, this.level().registryAccess()))
+            .map(p_337836_ -> p_337836_.value().assemble(craftingcontainer, this.level().registryAccess()))
             .map(ItemStack::getItem)
             .filter(DyeItem.class::isInstance)
             .map(DyeItem.class::cast)
