--- a/net/minecraft/world/entity/animal/WolfVariant.java
+++ b/net/minecraft/world/entity/animal/WolfVariant.java
@@ -65,14 +_,13 @@
     public boolean equals(Object p_332811_) {
         if (p_332811_ == this) {
             return true;
-        } else if (!(p_332811_ instanceof WolfVariant)) {
-            return false;
         } else {
-            WolfVariant wolfvariant = (WolfVariant)p_332811_;
-            return Objects.equals(this.wildTexture, wolfvariant.wildTexture)
-                && Objects.equals(this.tameTexture, wolfvariant.tameTexture)
-                && Objects.equals(this.angryTexture, wolfvariant.angryTexture)
-                && Objects.equals(this.biomes, wolfvariant.biomes);
+            return !(p_332811_ instanceof WolfVariant wolfvariant)
+                ? false
+                : Objects.equals(this.wildTexture, wolfvariant.wildTexture)
+                    && Objects.equals(this.tameTexture, wolfvariant.tameTexture)
+                    && Objects.equals(this.angryTexture, wolfvariant.angryTexture)
+                    && Objects.equals(this.biomes, wolfvariant.biomes);
         }
     }
 
