--- a/net/minecraft/world/level/levelgen/feature/trunkplacers/CherryTrunkPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/trunkplacers/CherryTrunkPlacer.java
@@ -2,6 +_,7 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.ArrayList;
@@ -10,7 +_,6 @@
 import java.util.function.Function;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.RandomSource;
 import net.minecraft.util.valueproviders.IntProvider;
 import net.minecraft.util.valueproviders.UniformInt;
@@ -21,25 +_,26 @@
 import net.minecraft.world.level.levelgen.feature.foliageplacers.FoliagePlacer;
 
 public class CherryTrunkPlacer extends TrunkPlacer {
-    private static final Codec<UniformInt> BRANCH_START_CODEC = ExtraCodecs.validate(
-        UniformInt.CODEC,
-        p_275181_ -> p_275181_.getMaxValue() - p_275181_.getMinValue() < 1
-                ? DataResult.error(() -> "Need at least 2 blocks variation for the branch starts to fit both branches")
-                : DataResult.success(p_275181_)
-    );
-    public static final Codec<CherryTrunkPlacer> CODEC = RecordCodecBuilder.create(
-        p_273579_ -> trunkPlacerParts(p_273579_)
+    private static final Codec<UniformInt> BRANCH_START_CODEC = UniformInt.CODEC
+        .codec()
+        .validate(
+            p_275181_ -> p_275181_.getMaxValue() - p_275181_.getMinValue() < 1
+                    ? DataResult.error(() -> "Need at least 2 blocks variation for the branch starts to fit both branches")
+                    : DataResult.success(p_275181_)
+        );
+    public static final MapCodec<CherryTrunkPlacer> CODEC = RecordCodecBuilder.mapCodec(
+        p_338099_ -> trunkPlacerParts(p_338099_)
                 .and(
-                    p_273579_.group(
+                    p_338099_.group(
                         IntProvider.codec(1, 3).fieldOf("branch_count").forGetter(p_272644_ -> p_272644_.branchCount),
                         IntProvider.codec(2, 16).fieldOf("branch_horizontal_length").forGetter(p_273612_ -> p_273612_.branchHorizontalLength),
-                        IntProvider.codec(-16, 0, BRANCH_START_CODEC)
+                        IntProvider.validateCodec(-16, 0, BRANCH_START_CODEC)
                             .fieldOf("branch_start_offset_from_top")
                             .forGetter(p_272705_ -> p_272705_.branchStartOffsetFromTop),
                         IntProvider.codec(-16, 16).fieldOf("branch_end_offset_from_top").forGetter(p_273633_ -> p_273633_.branchEndOffsetFromTop)
                     )
                 )
-                .apply(p_273579_, CherryTrunkPlacer::new)
+                .apply(p_338099_, CherryTrunkPlacer::new)
     );
     private final IntProvider branchCount;
     private final IntProvider branchHorizontalLength;
@@ -76,7 +_,7 @@
         int i = Math.max(0, p_272990_ - 1 + this.branchStartOffsetFromTop.sample(p_272993_));
         int j = Math.max(0, p_272990_ - 1 + this.secondBranchStartOffsetFromTop.sample(p_272993_));
         if (j >= i) {
-            ++j;
+            j++;
         }
 
         int k = this.branchCount.sample(p_272993_);
@@ -91,7 +_,7 @@
             l = i + 1;
         }
 
-        for(int i1 = 0; i1 < l; ++i1) {
+        for (int i1 = 0; i1 < l; i1++) {
             this.placeLog(p_272827_, p_272650_, p_272993_, p_273471_.above(i1), p_273355_);
         }
 
@@ -137,13 +_,13 @@
         BlockPos blockpos = p_273743_.relative(p_273712_, j).above(i);
         int k = flag ? 2 : 1;
 
-        for(int l = 0; l < k; ++l) {
+        for (int l = 0; l < k; l++) {
             this.placeLog(p_272736_, p_273092_, p_273449_, p_273496_.move(p_273712_), p_273027_, p_273558_);
         }
 
         Direction direction = blockpos.getY() > p_273496_.getY() ? Direction.UP : Direction.DOWN;
 
-        while(true) {
+        while (true) {
             int i1 = p_273496_.distManhattan(blockpos);
             if (i1 == 0) {
                 return new FoliagePlacer.FoliageAttachment(blockpos.above(), 0, false);
