--- a/net/minecraft/client/gui/screens/recipebook/GhostRecipe.java
+++ b/net/minecraft/client/gui/screens/recipebook/GhostRecipe.java
@@ -53,7 +_,7 @@
             this.time += p_282256_;
         }
 
-        for(int i = 0; i < this.ingredients.size(); ++i) {
+        for (int i = 0; i < this.ingredients.size(); i++) {
             GhostRecipe.GhostIngredient ghostrecipe$ghostingredient = this.ingredients.get(i);
             int j = ghostrecipe$ghostingredient.getX() + p_283169_;
             int k = ghostrecipe$ghostingredient.getY() + p_282326_;
