--- a/net/minecraft/client/gui/screens/recipebook/RecipeBookPage.java
+++ b/net/minecraft/client/gui/screens/recipebook/RecipeBookPage.java
@@ -44,7 +_,7 @@
     private RecipeCollection lastClickedRecipeCollection;
 
     public RecipeBookPage() {
-        for(int i = 0; i < 20; ++i) {
+        for (int i = 0; i < 20; i++) {
             this.buttons.add(new RecipeButton());
         }
     }
@@ -53,7 +_,7 @@
         this.minecraft = p_100429_;
         this.recipeBook = p_100429_.player.getRecipeBook();
 
-        for(int i = 0; i < this.buttons.size(); ++i) {
+        for (int i = 0; i < this.buttons.size(); i++) {
             this.buttons.get(i).setPosition(p_100430_ + 11 + 25 * (i % 5), p_100431_ + 31 + 25 * (i / 5));
         }
 
@@ -81,7 +_,7 @@
     private void updateButtonsForPage() {
         int i = 20 * this.currentPage;
 
-        for(int j = 0; j < this.buttons.size(); ++j) {
+        for (int j = 0; j < this.buttons.size(); j++) {
             RecipeButton recipebutton = this.buttons.get(j);
             if (i + j < this.recipeCollections.size()) {
                 RecipeCollection recipecollection = this.recipeCollections.get(i + j);
@@ -109,7 +_,7 @@
 
         this.hoveredButton = null;
 
-        for(RecipeButton recipebutton : this.buttons) {
+        for (RecipeButton recipebutton : this.buttons) {
             recipebutton.render(p_281416_, p_282278_, p_282424_, p_281712_);
             if (recipebutton.visible && recipebutton.isHoveredOrFocused()) {
                 this.hoveredButton = recipebutton;
@@ -154,15 +_,15 @@
 
             return true;
         } else if (this.forwardButton.mouseClicked(p_100410_, p_100411_, p_100412_)) {
-            ++this.currentPage;
+            this.currentPage++;
             this.updateButtonsForPage();
             return true;
         } else if (this.backButton.mouseClicked(p_100410_, p_100411_, p_100412_)) {
-            --this.currentPage;
+            this.currentPage--;
             this.updateButtonsForPage();
             return true;
         } else {
-            for(RecipeButton recipebutton : this.buttons) {
+            for (RecipeButton recipebutton : this.buttons) {
                 if (recipebutton.mouseClicked(p_100410_, p_100411_, p_100412_)) {
                     if (p_100412_ == 0) {
                         this.lastClickedRecipe = recipebutton.getRecipe();
@@ -189,7 +_,7 @@
     }
 
     public void recipesShown(List<RecipeHolder<?>> p_100435_) {
-        for(RecipeShownListener recipeshownlistener : this.showListeners) {
+        for (RecipeShownListener recipeshownlistener : this.showListeners) {
             recipeshownlistener.recipesShown(p_100435_);
         }
     }
