--- a/net/minecraft/client/gui/screens/recipebook/RecipeCollection.java
+++ b/net/minecraft/client/gui/screens/recipebook/RecipeCollection.java
@@ -36,7 +_,7 @@
         int i = p_100509_.size();
         ItemStack itemstack = p_100509_.get(0).value().getResultItem(p_267210_);
 
-        for(int j = 1; j < i; ++j) {
+        for (int j = 1; j < i; j++) {
             ItemStack itemstack1 = p_100509_.get(j).value().getResultItem(p_267210_);
             if (!ItemStack.isSameItemSameComponents(itemstack, itemstack1)) {
                 return false;
@@ -55,7 +_,7 @@
     }
 
     public void updateKnownRecipes(RecipeBook p_100500_) {
-        for(RecipeHolder<?> recipeholder : this.recipes) {
+        for (RecipeHolder<?> recipeholder : this.recipes) {
             if (p_100500_.contains(recipeholder)) {
                 this.known.add(recipeholder);
             }
@@ -63,7 +_,7 @@
     }
 
     public void canCraft(StackedContents p_100502_, int p_100503_, int p_100504_, RecipeBook p_100505_) {
-        for(RecipeHolder<?> recipeholder : this.recipes) {
+        for (RecipeHolder<?> recipeholder : this.recipes) {
             boolean flag = recipeholder.value().canCraftInDimensions(p_100503_, p_100504_) && p_100505_.contains(recipeholder);
             if (flag) {
                 this.fitsDimensions.add(recipeholder);
@@ -99,7 +_,7 @@
         List<RecipeHolder<?>> list = Lists.newArrayList();
         Set<RecipeHolder<?>> set = p_100511_ ? this.craftable : this.fitsDimensions;
 
-        for(RecipeHolder<?> recipeholder : this.recipes) {
+        for (RecipeHolder<?> recipeholder : this.recipes) {
             if (set.contains(recipeholder)) {
                 list.add(recipeholder);
             }
@@ -111,7 +_,7 @@
     public List<RecipeHolder<?>> getDisplayRecipes(boolean p_100514_) {
         List<RecipeHolder<?>> list = Lists.newArrayList();
 
-        for(RecipeHolder<?> recipeholder : this.recipes) {
+        for (RecipeHolder<?> recipeholder : this.recipes) {
             if (this.fitsDimensions.contains(recipeholder) && this.craftable.contains(recipeholder) == p_100514_) {
                 list.add(recipeholder);
             }
