--- a/net/minecraft/client/gui/screens/recipebook/RecipeBookTabButton.java
+++ b/net/minecraft/client/gui/screens/recipebook/RecipeBookTabButton.java
@@ -33,8 +_,8 @@
         ClientRecipeBook clientrecipebook = p_100452_.player.getRecipeBook();
         List<RecipeCollection> list = clientrecipebook.getCollection(this.category);
         if (p_100452_.player.containerMenu instanceof RecipeBookMenu) {
-            for(RecipeCollection recipecollection : list) {
-                for(RecipeHolder<?> recipeholder : recipecollection.getRecipes(clientrecipebook.isFiltering((RecipeBookMenu<?>)p_100452_.player.containerMenu))) {
+            for (RecipeCollection recipecollection : list) {
+                for (RecipeHolder<?> recipeholder : recipecollection.getRecipes(clientrecipebook.isFiltering((RecipeBookMenu<?>)p_100452_.player.containerMenu))) {
                     if (clientrecipebook.willHighlight(recipeholder)) {
                         this.animationTime = 15.0F;
                         return;
@@ -92,7 +_,7 @@
         List<RecipeCollection> list = p_100450_.getCollection(this.category);
         this.visible = false;
         if (list != null) {
-            for(RecipeCollection recipecollection : list) {
+            for (RecipeCollection recipecollection : list) {
                 if (recipecollection.hasKnownRecipes() && recipecollection.hasFitting()) {
                     this.visible = true;
                     break;
