--- a/net/minecraft/world/level/levelgen/structure/templatesystem/rule/blockentity/AppendLoot.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/rule/blockentity/AppendLoot.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem.rule.blockentity;
 
 import com.mojang.logging.LogUtils;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import javax.annotation.Nullable;
@@ -16,7 +_,7 @@
 
 public class AppendLoot implements RuleBlockEntityModifier {
     private static final Logger LOGGER = LogUtils.getLogger();
-    public static final Codec<AppendLoot> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<AppendLoot> CODEC = RecordCodecBuilder.mapCodec(
         p_335310_ -> p_335310_.group(ResourceKey.codec(Registries.LOOT_TABLE).fieldOf("loot_table").forGetter(p_335311_ -> p_335311_.lootTable))
                 .apply(p_335310_, AppendLoot::new)
     );
