--- a/net/minecraft/nbt/visitors/CollectFields.java
+++ b/net/minecraft/nbt/visitors/CollectFields.java
@@ -19,7 +_,7 @@
         Builder<TagType<?>> builder = ImmutableSet.builder();
         FieldTree fieldtree = FieldTree.createRoot();
 
-        for(FieldSelector fieldselector : p_202496_) {
+        for (FieldSelector fieldselector : p_202496_) {
             fieldtree.addEntry(fieldselector);
             builder.add(fieldselector.type());
         }
@@ -52,7 +_,7 @@
         if (this.depth() > fieldtree.depth()) {
             return super.visitEntry(p_197610_, p_197611_);
         } else if (fieldtree.selectedFields().remove(p_197611_, p_197610_)) {
-            --this.fieldsToGetCount;
+            this.fieldsToGetCount--;
             return super.visitEntry(p_197610_, p_197611_);
         } else {
             if (p_197610_ == CompoundTag.TYPE) {
