--- a/net/minecraft/util/datafix/ComponentDataFixUtils.java
+++ b/net/minecraft/util/datafix/ComponentDataFixUtils.java
@@ -65,4 +_,20 @@
             }
         }
     }
+
+    public static Optional<String> extractTranslationString(String p_338737_) {
+        try {
+            JsonElement jsonelement = JsonParser.parseString(p_338737_);
+            if (jsonelement.isJsonObject()) {
+                JsonObject jsonobject = jsonelement.getAsJsonObject();
+                JsonElement jsonelement1 = jsonobject.get("translate");
+                if (jsonelement1 != null && jsonelement1.isJsonPrimitive()) {
+                    return Optional.of(jsonelement1.getAsString());
+                }
+            }
+        } catch (JsonParseException jsonparseexception) {
+        }
+
+        return Optional.empty();
+    }
 }
