--- a/net/minecraft/util/datafix/DataFixTypes.java
+++ b/net/minecraft/util/datafix/DataFixTypes.java
@@ -51,7 +_,9 @@
             public <T> DataResult<T> encode(A p_301090_, DynamicOps<T> p_300954_, T p_301331_) {
                 return p_301309_.encode(p_301090_, p_300954_, p_301331_)
                     .flatMap(
-                        p_300998_ -> p_300954_.mergeToMap(p_300998_, p_300954_.createString("DataVersion"), p_300954_.createInt(DataFixTypes.currentVersion()))
+                        p_300998_ -> p_300954_.mergeToMap(
+                                (T)p_300998_, p_300954_.createString("DataVersion"), p_300954_.createInt(DataFixTypes.currentVersion())
+                            )
                     );
             }
 
