--- a/net/minecraft/util/datafix/DataFixers.java
+++ b/net/minecraft/util/datafix/DataFixers.java
@@ -28,6 +_,7 @@
 import net.minecraft.util.datafix.fixes.AdvancementsRenameFix;
 import net.minecraft.util.datafix.fixes.AreaEffectCloudPotionFix;
 import net.minecraft.util.datafix.fixes.AttributesRename;
+import net.minecraft.util.datafix.fixes.BannerEntityCustomNameToOverrideComponentFix;
 import net.minecraft.util.datafix.fixes.BannerPatternFormatFix;
 import net.minecraft.util.datafix.fixes.BedItemColorFix;
 import net.minecraft.util.datafix.fixes.BeehiveFieldRenameFix;
@@ -130,6 +_,7 @@
 import net.minecraft.util.datafix.fixes.ItemShulkerBoxColorFix;
 import net.minecraft.util.datafix.fixes.ItemSpawnEggFix;
 import net.minecraft.util.datafix.fixes.ItemStackComponentizationFix;
+import net.minecraft.util.datafix.fixes.ItemStackCustomNameToOverrideComponentFix;
 import net.minecraft.util.datafix.fixes.ItemStackEnchantmentNamesFix;
 import net.minecraft.util.datafix.fixes.ItemStackMapIdFix;
 import net.minecraft.util.datafix.fixes.ItemStackSpawnEggFix;
@@ -198,6 +_,7 @@
 import net.minecraft.util.datafix.fixes.TeamDisplayNameFix;
 import net.minecraft.util.datafix.fixes.TippedArrowPotionToItemFix;
 import net.minecraft.util.datafix.fixes.TrappedChestBlockEntityFix;
+import net.minecraft.util.datafix.fixes.TrialSpawnerConfigFix;
 import net.minecraft.util.datafix.fixes.VariantRenameFix;
 import net.minecraft.util.datafix.fixes.VillagerDataFix;
 import net.minecraft.util.datafix.fixes.VillagerFollowRangeFix;
@@ -286,7 +_,7 @@
 import net.minecraft.util.datafix.schemas.V3818;
 import net.minecraft.util.datafix.schemas.V3818_3;
 import net.minecraft.util.datafix.schemas.V3818_4;
-import net.minecraft.util.datafix.schemas.V3819;
+import net.minecraft.util.datafix.schemas.V3825;
 import net.minecraft.util.datafix.schemas.V501;
 import net.minecraft.util.datafix.schemas.V700;
 import net.minecraft.util.datafix.schemas.V701;
@@ -1310,12 +_,14 @@
         p_14514_.addFixer(new ItemStackComponentizationFix(schema214));
         Schema schema215 = p_14514_.addSchema(3818, 5, SAME_NAMESPACED);
         p_14514_.addFixer(new AreaEffectCloudPotionFix(schema215));
-        Schema schema216 = p_14514_.addSchema(3819, V3819::new);
-        p_14514_.addFixer(new AddNewChoices(schema216, "Added April Fools Block Entity", References.BLOCK_ENTITY));
-        p_14514_.addFixer(new AddNewChoices(schema216, "Added April Fools Entity", References.ENTITY));
-        Schema schema217 = p_14514_.addSchema(3820, SAME_NAMESPACED);
-        p_14514_.addFixer(new PlayerHeadBlockProfileFix(schema217));
-        p_14514_.addFixer(new LodestoneCompassComponentFix(schema217));
+        Schema schema216 = p_14514_.addSchema(3820, SAME_NAMESPACED);
+        p_14514_.addFixer(new PlayerHeadBlockProfileFix(schema216));
+        p_14514_.addFixer(new LodestoneCompassComponentFix(schema216));
+        Schema schema217 = p_14514_.addSchema(3825, V3825::new);
+        p_14514_.addFixer(new ItemStackCustomNameToOverrideComponentFix(schema217));
+        p_14514_.addFixer(new BannerEntityCustomNameToOverrideComponentFix(schema217));
+        p_14514_.addFixer(new TrialSpawnerConfigFix(schema217));
+        p_14514_.addFixer(new AddNewChoices(schema217, "Added Ominous Item Spawner", References.ENTITY));
     }
 
     private static UnaryOperator<String> createRenamerNoNamespace(Map<String, String> p_326818_) {
