--- a/net/minecraft/client/gui/spectator/SpectatorMenu.java
+++ b/net/minecraft/client/gui/spectator/SpectatorMenu.java
@@ -69,7 +_,7 @@
     public List<SpectatorMenuItem> getItems() {
         List<SpectatorMenuItem> list = Lists.newArrayList();
 
-        for(int i = 0; i <= 8; ++i) {
+        for (int i = 0; i <= 8; i++) {
             list.add(this.getItem(i));
         }
 
@@ -148,7 +_,7 @@
 
         @Override
         public void selectItem(SpectatorMenu p_101836_) {
-            p_101836_.page += this.direction;
+            p_101836_.page = p_101836_.page + this.direction;
         }
 
         @Override
