--- a/net/minecraft/world/entity/projectile/windcharge/AbstractWindCharge.java
+++ b/net/minecraft/world/entity/projectile/windcharge/AbstractWindCharge.java
@@ -23,6 +_,8 @@
 import net.minecraft.world.phys.HitResult;
 
 public abstract class AbstractWindCharge extends AbstractHurtingProjectile implements ItemSupplier {
+    public static final AbstractWindCharge.WindChargeDamageCalculator EXPLOSION_DAMAGE_CALCULATOR = new AbstractWindCharge.WindChargeDamageCalculator();
+
     public AbstractWindCharge(EntityType<? extends AbstractWindCharge> p_325927_, Level p_326350_) {
         super(p_325927_, p_326350_);
     }
@@ -69,15 +_,18 @@
 
     @Override
     protected boolean canHitEntity(Entity p_326159_) {
-        return p_326159_ instanceof AbstractWindCharge ? false : super.canHitEntity(p_326159_);
+        if (p_326159_ instanceof AbstractWindCharge) {
+            return false;
+        } else {
+            return p_326159_.getType() == EntityType.END_CRYSTAL ? false : super.canHitEntity(p_326159_);
+        }
     }
 
     @Override
     protected void onHitEntity(EntityHitResult p_326121_) {
         super.onHitEntity(p_326121_);
         if (!this.level().isClientSide) {
-            Entity entity = this.getOwner();
-            LivingEntity livingentity = entity instanceof LivingEntity livingentity1 ? livingentity1 : null;
+            LivingEntity livingentity = this.getOwner() instanceof LivingEntity livingentity1 ? livingentity1 : null;
             if (livingentity != null) {
                 livingentity.setLastHurtMob(p_326121_.getEntity());
             }
@@ -105,11 +_,9 @@
     @Override
     protected void onHit(HitResult p_326337_) {
         super.onHit(p_326337_);
-        if (!this.level().isClientSide && !this.isDeflected) {
+        if (!this.level().isClientSide) {
             this.discard();
         }
-
-        this.isDeflected = false;
     }
 
     @Override
