--- a/net/minecraft/world/entity/projectile/windcharge/BreezeWindCharge.java
+++ b/net/minecraft/world/entity/projectile/windcharge/BreezeWindCharge.java
@@ -7,7 +_,6 @@
 import net.minecraft.world.level.Level;
 
 public class BreezeWindCharge extends AbstractWindCharge {
-    private static final AbstractWindCharge.WindChargeDamageCalculator EXPLOSION_DAMAGE_CALCULATOR = new AbstractWindCharge.WindChargeDamageCalculator();
     private static final float RADIUS = 3.0F;
 
     public BreezeWindCharge(EntityType<? extends AbstractWindCharge> p_326366_, Level p_325976_) {
