--- a/net/minecraft/world/entity/monster/zombie/Drowned.java
+++ b/net/minecraft/world/entity/monster/zombie/Drowned.java
@@ -18,6 +_,7 @@
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.EntityAttachment;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.EntityAttachments;
 import net.minecraft.world.entity.EntityDimensions;
 import net.minecraft.world.entity.EntitySpawnReason;
@@ -278,7 +_,7 @@
     }
 
     @Override
-    public void performRangedAttack(Entity target, float power) {
+    public void performRangedAttack(LivingEntity target, float power) {
         ItemStack mainHandItem = this.getMainHandItem();
         ItemStack tridentItemStack = mainHandItem.is(Items.TRIDENT) ? mainHandItem : new ItemStack(Items.TRIDENT);
         ThrownTrident trident = new ThrownTrident(this.level(), this, tridentItemStack);
