--- a/net/minecraft/world/entity/player/StackedContents.java
+++ b/net/minecraft/world/entity/player/StackedContents.java
@@ -105,10 +_,10 @@
             this.itemCount = this.items.length;
             this.data = new BitSet(this.ingredientCount + this.itemCount + this.ingredientCount + this.ingredientCount * this.itemCount);
 
-            for(int i = 0; i < this.ingredients.size(); ++i) {
+            for (int i = 0; i < this.ingredients.size(); i++) {
                 IntList intlist = this.ingredients.get(i).getStackingIds();
 
-                for(int j = 0; j < this.itemCount; ++j) {
+                for (int j = 0; j < this.itemCount; j++) {
                     if (intlist.contains(this.items[j])) {
                         this.data.set(this.getIndex(true, j, i));
                     }
@@ -121,12 +_,12 @@
                 return true;
             } else {
                 int i;
-                for(i = 0; this.dfs(p_36513_); ++i) {
+                for (i = 0; this.dfs(p_36513_); i++) {
                     StackedContents.this.take(this.items[this.path.getInt(0)], p_36513_);
                     int j = this.path.size() - 1;
                     this.setSatisfied(this.path.getInt(j));
 
-                    for(int k = 0; k < j; ++k) {
+                    for (int k = 0; k < j; k++) {
                         this.toggleResidual((k & 1) == 0, this.path.get(k), this.path.get(k + 1));
                     }
 
@@ -143,11 +_,11 @@
                 this.data.clear(0, this.ingredientCount + this.itemCount + this.ingredientCount);
                 int l = 0;
 
-                for(Ingredient ingredient : this.recipe.getIngredients()) {
+                for (Ingredient ingredient : this.recipe.getIngredients()) {
                     if (flag1 && ingredient.isEmpty()) {
                         p_36514_.add(0);
                     } else {
-                        for(int i1 = 0; i1 < this.itemCount; ++i1) {
+                        for (int i1 = 0; i1 < this.itemCount; i1++) {
                             if (this.hasResidual(false, l, i1)) {
                                 this.toggleResidual(true, i1, l);
                                 StackedContents.this.put(this.items[i1], p_36513_);
@@ -157,7 +_,7 @@
                             }
                         }
 
-                        ++l;
+                        l++;
                     }
                 }
 
@@ -168,13 +_,13 @@
         private int[] getUniqueAvailableIngredientItems() {
             IntCollection intcollection = new IntAVLTreeSet();
 
-            for(Ingredient ingredient : this.ingredients) {
+            for (Ingredient ingredient : this.ingredients) {
                 intcollection.addAll(ingredient.getStackingIds());
             }
 
             IntIterator intiterator = intcollection.iterator();
 
-            while(intiterator.hasNext()) {
+            while (intiterator.hasNext()) {
                 if (!StackedContents.this.has(intiterator.nextInt())) {
                     intiterator.remove();
                 }
@@ -186,11 +_,11 @@
         private boolean dfs(int p_36511_) {
             int i = this.itemCount;
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 if (StackedContents.this.contents.get(this.items[j]) >= p_36511_) {
                     this.visit(false, j);
 
-                    while(!this.path.isEmpty()) {
+                    while (!this.path.isEmpty()) {
                         int k = this.path.size();
                         boolean flag = (k & 1) == 1;
                         int l = this.path.getInt(k - 1);
@@ -201,10 +_,10 @@
                         int i1 = flag ? this.ingredientCount : i;
                         int j1 = 0;
 
-                        while(true) {
+                        while (true) {
                             if (j1 < i1) {
                                 if (this.hasVisited(flag, j1) || !this.hasConnection(flag, l, j1) || !this.hasResidual(flag, l, j1)) {
-                                    ++j1;
+                                    j1++;
                                     continue;
                                 }
 
@@ -274,7 +_,7 @@
             int i = 0;
             int j = Math.min(p_36526_, this.getMinIngredientCount()) + 1;
 
-            while(true) {
+            while (true) {
                 int k = (i + j) / 2;
                 if (this.tryPick(k, null)) {
                     if (j - i <= 1) {
@@ -295,10 +_,10 @@
         private int getMinIngredientCount() {
             int i = Integer.MAX_VALUE;
 
-            for(Ingredient ingredient : this.ingredients) {
+            for (Ingredient ingredient : this.ingredients) {
                 int j = 0;
 
-                for(int k : ingredient.getStackingIds()) {
+                for (int k : ingredient.getStackingIds()) {
                     j = Math.max(j, StackedContents.this.contents.get(k));
                 }
 
