--- a/net/minecraft/world/item/trading/Merchant.java
+++ b/net/minecraft/world/item/trading/Merchant.java
@@ -1,8 +_,11 @@
 package net.minecraft.world.item.trading;
 
+import java.util.OptionalInt;
 import net.minecraft.network.chat.Component;
 import net.minecraft.sounds.SoundEvent;
+import net.minecraft.world.SimpleMenuProvider;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.MerchantMenu;
 import net.minecraft.world.item.ItemStack;
 import org.jspecify.annotations.Nullable;
 
@@ -32,6 +_,13 @@
     }
 
     default void openTradingScreen(Player player, Component title, int level) {
+        OptionalInt containerId = player.openMenu(new SimpleMenuProvider((id, inventory, p) -> new MerchantMenu(id, inventory, this), title));
+        if (containerId.isPresent()) {
+            MerchantOffers offers = this.getOffers();
+            if (!offers.isEmpty()) {
+                player.sendMerchantOffers(containerId.getAsInt(), offers, level, this.getVillagerXp(), this.showProgressBar(), this.canRestock());
+            }
+        }
     }
 
     boolean isClientSide();
