--- a/net/minecraft/world/item/trading/MerchantOffers.java
+++ b/net/minecraft/world/item/trading/MerchantOffers.java
@@ -32,7 +_,7 @@
             MerchantOffer merchantoffer1 = this.get(p_45392_);
             return merchantoffer1.satisfiedBy(p_45390_, p_45391_) ? merchantoffer1 : null;
         } else {
-            for(int i = 0; i < this.size(); ++i) {
+            for (int i = 0; i < this.size(); i++) {
                 MerchantOffer merchantoffer = this.get(i);
                 if (merchantoffer.satisfiedBy(p_45390_, p_45391_)) {
                     return merchantoffer;
@@ -46,7 +_,7 @@
     public MerchantOffers copy() {
         MerchantOffers merchantoffers = new MerchantOffers(this.size());
 
-        for(MerchantOffer merchantoffer : this) {
+        for (MerchantOffer merchantoffer : this) {
             merchantoffers.add(merchantoffer.copy());
         }
 
