--- a/net/minecraft/client/gui/components/AbstractSelectionList.java
+++ b/net/minecraft/client/gui/components/AbstractSelectionList.java
@@ -376,10 +_,11 @@
 
     @Nullable
     protected E nextEntry(ScreenDirection p_265159_, Predicate<E> p_265109_, @Nullable E p_265379_) {
-        int i = switch(p_265159_) {
+        int i = switch (p_265159_) {
             case RIGHT, LEFT -> 0;
             case UP -> -1;
             case DOWN -> 1;
+            default -> throw new MatchException(null, null);
         };
         if (!this.children().isEmpty() && i != 0) {
             int j;
@@ -389,7 +_,7 @@
                 j = this.children().indexOf(p_265379_) + i;
             }
 
-            for(int k = j; k >= 0 && k < this.children.size(); k += i) {
+            for (int k = j; k >= 0 && k < this.children.size(); k += i) {
                 E e = this.children().get(k);
                 if (p_265109_.test(e)) {
                     return e;
@@ -414,7 +_,7 @@
         int k = this.itemHeight - 4;
         int l = this.getItemCount();
 
-        for(int i1 = 0; i1 < l; ++i1) {
+        for (int i1 = 0; i1 < l; i1++) {
             int j1 = this.getRowTop(i1);
             int k1 = this.getRowBottom(i1);
             if (k1 >= this.getY() && j1 <= this.getBottom()) {
@@ -485,7 +_,7 @@
     protected boolean removeEntry(E p_93503_) {
         boolean flag = this.children.remove(p_93503_);
         if (flag && p_93503_ == this.getSelected()) {
-            this.setSelected((E)null);
+            this.setSelected(null);
         }
 
         return flag;
