--- a/net/minecraft/client/gui/components/ChatComponent.java
+++ b/net/minecraft/client/gui/components/ChatComponent.java
@@ -75,7 +_,7 @@
                 int l1 = (int)Math.round(-8.0 * (d2 + 1.0) + 4.0 * d2);
                 int i2 = 0;
 
-                for(int j2 = 0; j2 + this.chatScrollbarPos < this.trimmedMessages.size() && j2 < i; ++j2) {
+                for (int j2 = 0; j2 + this.chatScrollbarPos < this.trimmedMessages.size() && j2 < i; j2++) {
                     int k2 = j2 + this.chatScrollbarPos;
                     GuiMessage.Line guimessage$line = this.trimmedMessages.get(k2);
                     if (guimessage$line != null) {
@@ -84,7 +_,7 @@
                             double d3 = p_316855_ ? 1.0 : getTimeFactor(l2);
                             int j3 = (int)(255.0 * d3 * d0);
                             int k3 = (int)(255.0 * d3 * d1);
-                            ++i2;
+                            i2++;
                             if (j3 > 3) {
                                 int l3 = 0;
                                 int i4 = i1 - j2 * k1;
@@ -180,13 +_,15 @@
     }
 
     public void addMessage(Component p_241484_, @Nullable MessageSignature p_241323_, @Nullable GuiMessageTag p_241297_) {
-        this.logChatMessage(p_241484_, p_241297_);
-        this.addMessage(p_241484_, p_241323_, this.minecraft.gui.getGuiTicks(), p_241297_, false);
+        GuiMessage guimessage = new GuiMessage(this.minecraft.gui.getGuiTicks(), p_241484_, p_241323_, p_241297_);
+        this.logChatMessage(guimessage);
+        this.addMessageToDisplayQueue(guimessage);
+        this.addMessageToQueue(guimessage);
     }
 
-    private void logChatMessage(Component p_242919_, @Nullable GuiMessageTag p_242840_) {
-        String s = p_242919_.getString().replaceAll("\r", "\\\\r").replaceAll("\n", "\\\\n");
-        String s1 = Optionull.map(p_242840_, GuiMessageTag::logTag);
+    private void logChatMessage(GuiMessage p_338237_) {
+        String s = p_338237_.content().getString().replaceAll("\r", "\\\\r").replaceAll("\n", "\\\\n");
+        String s1 = Optionull.map(p_338237_.tag(), GuiMessageTag::logTag);
         if (s1 != null) {
             LOGGER.info("[{}] [CHAT] {}", s1, s);
         } else {
@@ -194,16 +_,17 @@
         }
     }
 
-    private void addMessage(Component p_240562_, @Nullable MessageSignature p_241566_, int p_240583_, @Nullable GuiMessageTag p_240624_, boolean p_240558_) {
+    private void addMessageToDisplayQueue(GuiMessage p_338816_) {
         int i = Mth.floor((double)this.getWidth() / this.getScale());
-        if (p_240624_ != null && p_240624_.icon() != null) {
-            i -= p_240624_.icon().width + 4 + 2;
+        GuiMessageTag.Icon guimessagetag$icon = p_338816_.icon();
+        if (guimessagetag$icon != null) {
+            i -= guimessagetag$icon.width + 4 + 2;
         }
 
-        List<FormattedCharSequence> list = ComponentRenderUtils.wrapComponents(p_240562_, i, this.minecraft.font);
+        List<FormattedCharSequence> list = ComponentRenderUtils.wrapComponents(p_338816_.content(), i, this.minecraft.font);
         boolean flag = this.isChatFocused();
 
-        for(int j = 0; j < list.size(); ++j) {
+        for (int j = 0; j < list.size(); j++) {
             FormattedCharSequence formattedcharsequence = list.get(j);
             if (flag && this.chatScrollbarPos > 0) {
                 this.newMessageSinceScroll = true;
@@ -211,31 +_,25 @@
             }
 
             boolean flag1 = j == list.size() - 1;
-            this.trimmedMessages.add(0, new GuiMessage.Line(p_240583_, formattedcharsequence, p_240624_, flag1));
+            this.trimmedMessages.add(0, new GuiMessage.Line(p_338816_.addedTime(), formattedcharsequence, p_338816_.tag(), flag1));
         }
 
-        while(this.trimmedMessages.size() > 100) {
+        while (this.trimmedMessages.size() > 100) {
             this.trimmedMessages.remove(this.trimmedMessages.size() - 1);
         }
-
-        if (!p_240558_) {
-            this.allMessages.add(0, new GuiMessage(p_240583_, p_240562_, p_241566_, p_240624_));
-
-            while(this.allMessages.size() > 100) {
-                this.allMessages.remove(this.allMessages.size() - 1);
-            }
+    }
+
+    private void addMessageToQueue(GuiMessage p_338828_) {
+        this.allMessages.add(0, p_338828_);
+
+        while (this.allMessages.size() > 100) {
+            this.allMessages.remove(this.allMessages.size() - 1);
         }
     }
 
     private void processMessageDeletionQueue() {
         int i = this.minecraft.gui.getGuiTicks();
-        this.messageDeletionQueue.removeIf(p_250713_ -> {
-            if (i >= p_250713_.deletableAfter()) {
-                return this.deleteMessageOrDelay(p_250713_.signature()) == null;
-            } else {
-                return false;
-            }
-        });
+        this.messageDeletionQueue.removeIf(p_250713_ -> i >= p_250713_.deletableAfter() ? this.deleteMessageOrDelay(p_250713_.signature()) == null : false);
     }
 
     public void deleteMessage(MessageSignature p_241324_) {
@@ -250,13 +_,13 @@
         int i = this.minecraft.gui.getGuiTicks();
         ListIterator<GuiMessage> listiterator = this.allMessages.listIterator();
 
-        while(listiterator.hasNext()) {
+        while (listiterator.hasNext()) {
             GuiMessage guimessage = listiterator.next();
             if (p_251812_.equals(guimessage.signature())) {
                 int j = guimessage.addedTime() + 60;
                 if (i >= j) {
                     listiterator.set(this.createDeletedMarker(guimessage));
-                    this.refreshTrimmedMessage();
+                    this.refreshTrimmedMessages();
                     return null;
                 }
 
@@ -273,15 +_,14 @@
 
     public void rescaleChat() {
         this.resetChatScroll();
-        this.refreshTrimmedMessage();
+        this.refreshTrimmedMessages();
     }
 
-    private void refreshTrimmedMessage() {
+    private void refreshTrimmedMessages() {
         this.trimmedMessages.clear();
 
-        for(int i = this.allMessages.size() - 1; i >= 0; --i) {
-            GuiMessage guimessage = this.allMessages.get(i);
-            this.addMessage(guimessage.content(), guimessage.signature(), guimessage.addedTime(), guimessage.tag(), true);
+        for (GuiMessage guimessage : Lists.reverse(this.allMessages)) {
+            this.addMessageToDisplayQueue(guimessage);
         }
     }
 
@@ -399,12 +_,12 @@
         if (i == -1) {
             return -1;
         } else {
-            while(i >= 0) {
+            while (i >= 0) {
                 if (this.trimmedMessages.get(i).endOfEntry()) {
                     return i;
                 }
 
-                --i;
+                i--;
             }
 
             return i;
@@ -473,7 +_,34 @@
         return (int)(9.0 * (this.minecraft.options.chatLineSpacing().get() + 1.0));
     }
 
+    public ChatComponent.State storeState() {
+        return new ChatComponent.State(List.copyOf(this.allMessages), List.copyOf(this.recentChat), List.copyOf(this.messageDeletionQueue));
+    }
+
+    public void restoreState(ChatComponent.State p_338814_) {
+        this.recentChat.clear();
+        this.recentChat.addAll(p_338814_.history);
+        this.messageDeletionQueue.clear();
+        this.messageDeletionQueue.addAll(p_338814_.delayedMessageDeletions);
+        this.allMessages.clear();
+        this.allMessages.addAll(p_338814_.messages);
+        this.refreshTrimmedMessages();
+    }
+
     @OnlyIn(Dist.CLIENT)
     static record DelayedMessageDeletion(MessageSignature signature, int deletableAfter) {
+    }
+
+    @OnlyIn(Dist.CLIENT)
+    public static class State {
+        final List<GuiMessage> messages;
+        final List<String> history;
+        final List<ChatComponent.DelayedMessageDeletion> delayedMessageDeletions;
+
+        public State(List<GuiMessage> p_338549_, List<String> p_338655_, List<ChatComponent.DelayedMessageDeletion> p_338701_) {
+            this.messages = p_338549_;
+            this.history = p_338655_;
+            this.delayedMessageDeletions = p_338701_;
+        }
     }
 }
