--- a/net/minecraft/client/gui/components/CommandSuggestions.java
+++ b/net/minecraft/client/gui/components/CommandSuggestions.java
@@ -139,7 +_,7 @@
             if (!suggestions.isEmpty()) {
                 int i = 0;
 
-                for(Suggestion suggestion : suggestions.getList()) {
+                for (Suggestion suggestion : suggestions.getList()) {
                     i = Math.max(i, this.font.width(suggestion.getText()));
                 }
 
@@ -177,7 +_,7 @@
         List<Suggestion> list = Lists.newArrayList();
         List<Suggestion> list1 = Lists.newArrayList();
 
-        for(Suggestion suggestion : p_93899_.getList()) {
+        for (Suggestion suggestion : p_93899_.getList()) {
             if (!suggestion.getText().startsWith(s1) && !suggestion.getText().startsWith("minecraft:" + s1)) {
                 list1.add(suggestion);
             } else {
@@ -239,7 +_,7 @@
             int i = 0;
             Matcher matcher = WHITESPACE_PATTERN.matcher(p_93913_);
 
-            while(matcher.find()) {
+            while (matcher.find()) {
                 i = matcher.end();
             }
 
@@ -261,10 +_,10 @@
             if (this.pendingSuggestions.join().isEmpty() && !this.currentParse.getExceptions().isEmpty()) {
                 int i = 0;
 
-                for(Entry<CommandNode<SharedSuggestionProvider>, CommandSyntaxException> entry : this.currentParse.getExceptions().entrySet()) {
+                for (Entry<CommandNode<SharedSuggestionProvider>, CommandSyntaxException> entry : this.currentParse.getExceptions().entrySet()) {
                     CommandSyntaxException commandsyntaxexception = entry.getValue();
                     if (commandsyntaxexception.getType() == CommandSyntaxException.BUILT_IN_EXCEPTIONS.literalIncorrect()) {
-                        ++i;
+                        i++;
                     } else {
                         this.commandUsage.add(getExceptionMessage(commandsyntaxexception));
                     }
@@ -302,7 +_,7 @@
         int i = 0;
         Style style = Style.EMPTY.withColor(p_289002_);
 
-        for(Entry<CommandNode<SharedSuggestionProvider>, String> entry : map.entrySet()) {
+        for (Entry<CommandNode<SharedSuggestionProvider>, String> entry : map.entrySet()) {
             if (!(entry.getKey() instanceof LiteralCommandNode)) {
                 list.add(FormattedCharSequence.forward(entry.getValue(), style));
                 i = Math.max(i, this.font.width(entry.getValue()));
@@ -336,7 +_,7 @@
         int j = -1;
         CommandContextBuilder<SharedSuggestionProvider> commandcontextbuilder = p_93893_.getContext().getLastChild();
 
-        for(ParsedArgument<SharedSuggestionProvider, ?> parsedargument : commandcontextbuilder.getArguments().values()) {
+        for (ParsedArgument<SharedSuggestionProvider, ?> parsedargument : commandcontextbuilder.getArguments().values()) {
             if (++j >= ARGUMENT_STYLES.size()) {
                 j = 0;
             }
@@ -386,11 +_,11 @@
     public void renderUsage(GuiGraphics p_282763_) {
         int i = 0;
 
-        for(FormattedCharSequence formattedcharsequence : this.commandUsage) {
+        for (FormattedCharSequence formattedcharsequence : this.commandUsage) {
             int j = this.anchorToBottom ? this.screen.height - 14 - 13 - 12 * i : 72 + 12 * i;
             p_282763_.fill(this.commandUsagePosition - 1, j, this.commandUsagePosition + this.commandUsageWidth + 1, j + 12, this.fillColor);
             p_282763_.drawString(this.font, formattedcharsequence, this.commandUsagePosition, j + 2, -1);
-            ++i;
+            i++;
         }
     }
 
@@ -444,7 +_,7 @@
                     CommandSuggestions.this.fillColor
                 );
                 if (flag) {
-                    for(int k = 0; k < this.rect.getWidth(); ++k) {
+                    for (int k = 0; k < this.rect.getWidth(); k++) {
                         if (k % 2 == 0) {
                             p_282264_.fill(this.rect.getX() + k, this.rect.getY() - 1, this.rect.getX() + k + 1, this.rect.getY(), -1);
                         }
@@ -452,7 +_,7 @@
                 }
 
                 if (flag1) {
-                    for(int i1 = 0; i1 < this.rect.getWidth(); ++i1) {
+                    for (int i1 = 0; i1 < this.rect.getWidth(); i1++) {
                         if (i1 % 2 == 0) {
                             p_282264_.fill(
                                 this.rect.getX() + i1,
@@ -468,7 +_,7 @@
 
             boolean flag4 = false;
 
-            for(int l = 0; l < i; ++l) {
+            for (int l = 0; l < i; l++) {
                 Suggestion suggestion = this.suggestionList.get(l + this.offset);
                 p_282264_.fill(
                     this.rect.getX(),
@@ -583,11 +_,11 @@
         public void select(int p_93987_) {
             this.current = p_93987_;
             if (this.current < 0) {
-                this.current += this.suggestionList.size();
+                this.current = this.current + this.suggestionList.size();
             }
 
             if (this.current >= this.suggestionList.size()) {
-                this.current -= this.suggestionList.size();
+                this.current = this.current - this.suggestionList.size();
             }
 
             Suggestion suggestion = this.suggestionList.get(this.current);
