--- a/net/minecraft/client/gui/components/ContainerObjectSelectionList.java
+++ b/net/minecraft/client/gui/components/ContainerObjectSelectionList.java
@@ -29,10 +_,9 @@
     public ComponentPath nextFocusPath(FocusNavigationEvent p_265385_) {
         if (this.getItemCount() == 0) {
             return null;
-        } else if (!(p_265385_ instanceof FocusNavigationEvent.ArrowNavigation)) {
+        } else if (!(p_265385_ instanceof FocusNavigationEvent.ArrowNavigation focusnavigationevent$arrownavigation)) {
             return super.nextFocusPath(p_265385_);
         } else {
-            FocusNavigationEvent.ArrowNavigation focusnavigationevent$arrownavigation = (FocusNavigationEvent.ArrowNavigation)p_265385_;
             E e = this.getFocused();
             if (focusnavigationevent$arrownavigation.direction().getAxis() == ScreenAxis.HORIZONTAL && e != null) {
                 return ComponentPath.path(this, e.nextFocusPath(p_265385_));
@@ -44,7 +_,7 @@
                 }
 
                 if (i == -1) {
-                    switch(screendirection) {
+                    switch (screendirection) {
                         case LEFT:
                             i = Integer.MAX_VALUE;
                             screendirection = ScreenDirection.DOWN;
@@ -62,13 +_,13 @@
 
                 ComponentPath componentpath;
                 do {
-                    e1 = this.nextEntry(screendirection, p_332530_ -> !p_332530_.children().isEmpty(), e1);
+                    e1 = this.nextEntry(screendirection, p_339276_ -> !p_339276_.children().isEmpty(), e1);
                     if (e1 == null) {
                         return null;
                     }
 
                     componentpath = e1.focusPathAtIndex(focusnavigationevent$arrownavigation, i);
-                } while(componentpath == null);
+                } while (componentpath == null);
 
                 return ComponentPath.path(this, componentpath);
             }
@@ -79,7 +_,7 @@
     public void setFocused(@Nullable GuiEventListener p_265559_) {
         super.setFocused(p_265559_);
         if (p_265559_ == null) {
-            this.setSelected((E)null);
+            this.setSelected(null);
         }
     }
 
@@ -166,10 +_,11 @@
         @Override
         public ComponentPath nextFocusPath(FocusNavigationEvent p_265672_) {
             if (p_265672_ instanceof FocusNavigationEvent.ArrowNavigation focusnavigationevent$arrownavigation) {
-                int i = switch(focusnavigationevent$arrownavigation.direction()) {
+                int i = switch (focusnavigationevent$arrownavigation.direction()) {
                     case LEFT -> -1;
                     case RIGHT -> 1;
                     case UP, DOWN -> 0;
+                    default -> throw new MatchException(null, null);
                 };
                 if (i == 0) {
                     return null;
@@ -177,7 +_,7 @@
 
                 int j = Mth.clamp(i + this.children().indexOf(this.getFocused()), 0, this.children().size() - 1);
 
-                for(int k = j; k >= 0 && k < this.children().size(); k += i) {
+                for (int k = j; k >= 0 && k < this.children().size(); k += i) {
                     GuiEventListener guieventlistener = this.children().get(k);
                     ComponentPath componentpath = guieventlistener.nextFocusPath(p_265672_);
                     if (componentpath != null) {
