--- a/net/minecraft/client/gui/components/CycleButton.java
+++ b/net/minecraft/client/gui/components/CycleButton.java
@@ -241,7 +_,7 @@
             if (list.isEmpty()) {
                 throw new IllegalStateException("No values for cycle button");
             } else {
-                T t = (T)(this.initialValue != null ? this.initialValue : list.get(this.initialIndex));
+                T t = this.initialValue != null ? this.initialValue : list.get(this.initialIndex);
                 Component component = this.valueStringifier.apply(t);
                 Component component1 = (Component)(this.displayOnlyValue ? component : CommonComponents.optionNameValue(p_168941_, component));
                 return new CycleButton<>(
