--- a/net/minecraft/client/gui/components/EditBox.java
+++ b/net/minecraft/client/gui/components/EditBox.java
@@ -124,7 +_,7 @@
             int l = s.length();
             if (k < l) {
                 if (Character.isHighSurrogate(s.charAt(k - 1))) {
-                    --k;
+                    k--;
                 }
 
                 s = s.substring(0, k);
@@ -200,24 +_,24 @@
         boolean flag = p_94141_ < 0;
         int j = Math.abs(p_94141_);
 
-        for(int k = 0; k < j; ++k) {
+        for (int k = 0; k < j; k++) {
             if (!flag) {
                 int l = this.value.length();
                 i = this.value.indexOf(32, i);
                 if (i == -1) {
                     i = l;
                 } else {
-                    while(p_94143_ && i < l && this.value.charAt(i) == ' ') {
-                        ++i;
+                    while (p_94143_ && i < l && this.value.charAt(i) == ' ') {
+                        i++;
                     }
                 }
             } else {
-                while(p_94143_ && i > 0 && this.value.charAt(i - 1) == ' ') {
-                    --i;
+                while (p_94143_ && i > 0 && this.value.charAt(i - 1) == ' ') {
+                    i--;
                 }
 
-                while(i > 0 && this.value.charAt(i - 1) != ' ') {
-                    --i;
+                while (i > 0 && this.value.charAt(i - 1) != ' ') {
+                    i--;
                 }
             }
         }
@@ -258,7 +_,7 @@
     @Override
     public boolean keyPressed(int p_94132_, int p_94133_, int p_94134_) {
         if (this.isActive() && this.isFocused()) {
-            switch(p_94132_) {
+            switch (p_94132_) {
                 case 259:
                     if (this.isEditable) {
                         this.deleteText(-1);
@@ -392,7 +_,7 @@
                 j1 = i > 0 ? j + this.width : j;
             } else if (flag2) {
                 j1 = l - 1;
-                --l;
+                l--;
             }
 
             if (!s.isEmpty() && flag && i < s.length()) {
@@ -512,13 +_,13 @@
             String s = this.font.plainSubstrByWidth(this.value.substring(this.displayPos), i);
             int j = s.length() + this.displayPos;
             if (p_296429_ == this.displayPos) {
-                this.displayPos -= this.font.plainSubstrByWidth(this.value, i, true).length();
+                this.displayPos = this.displayPos - this.font.plainSubstrByWidth(this.value, i, true).length();
             }
 
             if (p_296429_ > j) {
                 this.displayPos += p_296429_ - j;
             } else if (p_296429_ <= this.displayPos) {
-                this.displayPos -= this.displayPos - p_296429_;
+                this.displayPos = this.displayPos - (this.displayPos - p_296429_);
             }
 
             this.displayPos = Mth.clamp(this.displayPos, 0, this.value.length());
