--- a/net/minecraft/client/gui/components/MultiLineEditBox.java
+++ b/net/minecraft/client/gui/components/MultiLineEditBox.java
@@ -115,7 +_,7 @@
             int k = 0;
             int l = this.getY() + this.innerPadding();
 
-            for(MultilineTextField.StringView multilinetextfield$stringview : this.textField.iterateLines()) {
+            for (MultilineTextField.StringView multilinetextfield$stringview : this.textField.iterateLines()) {
                 boolean flag2 = this.withinContentAreaTopBottom(l, l + 9);
                 if (flag && flag1 && i >= multilinetextfield$stringview.beginIndex() && i <= multilinetextfield$stringview.endIndex()) {
                     if (flag2) {
@@ -153,7 +_,7 @@
                 int k1 = this.getX() + this.innerPadding();
                 l = this.getY() + this.innerPadding();
 
-                for(MultilineTextField.StringView multilinetextfield$stringview1 : this.textField.iterateLines()) {
+                for (MultilineTextField.StringView multilinetextfield$stringview1 : this.textField.iterateLines()) {
                     if (multilinetextfield$stringview2.beginIndex() > multilinetextfield$stringview1.endIndex()) {
                         l += 9;
                     } else {
