--- a/net/minecraft/client/gui/components/PopupScreen.java
+++ b/net/minecraft/client/gui/components/PopupScreen.java
@@ -72,7 +_,9 @@
 
         this.layout.addChild(new MultiLineTextWidget(this.message, this.font).setMaxWidth(this.contentWidth).setCentered(true));
         this.layout.addChild(this.buildButtonRow());
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321353_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321353_);
+        });
         this.repositionElements();
     }
 
@@ -82,7 +_,7 @@
         LinearLayout linearlayout = LinearLayout.horizontal();
         linearlayout.spacing(6);
 
-        for(PopupScreen.ButtonOption popupscreen$buttonoption : this.buttons) {
+        for (PopupScreen.ButtonOption popupscreen$buttonoption : this.buttons) {
             linearlayout.addChild(
                 Button.builder(popupscreen$buttonoption.message(), p_304831_ -> popupscreen$buttonoption.action().accept(this)).width(j).build()
             );
