--- a/net/minecraft/client/gui/components/SubtitleOverlay.java
+++ b/net/minecraft/client/gui/components/SubtitleOverlay.java
@@ -47,7 +_,7 @@
             Vec3 vec32 = listenertransform.right();
             this.audibleSubtitles.clear();
 
-            for(SubtitleOverlay.Subtitle subtitleoverlay$subtitle : this.subtitles) {
+            for (SubtitleOverlay.Subtitle subtitleoverlay$subtitle : this.subtitles) {
                 if (subtitleoverlay$subtitle.isAudibleFrom(vec3)) {
                     this.audibleSubtitles.add(subtitleoverlay$subtitle);
                 }
@@ -59,7 +_,7 @@
                 double d0 = this.minecraft.options.notificationDisplayTime().get();
                 Iterator<SubtitleOverlay.Subtitle> iterator = this.audibleSubtitles.iterator();
 
-                while(iterator.hasNext()) {
+                while (iterator.hasNext()) {
                     SubtitleOverlay.Subtitle subtitleoverlay$subtitle1 = iterator.next();
                     if ((double)subtitleoverlay$subtitle1.getTime() + 3000.0 * d0 <= (double)Util.getMillis()) {
                         iterator.remove();
@@ -70,7 +_,7 @@
 
                 j2 += this.minecraft.font.width("<") + this.minecraft.font.width(" ") + this.minecraft.font.width(">") + this.minecraft.font.width(" ");
 
-                for(SubtitleOverlay.Subtitle subtitleoverlay$subtitle2 : this.audibleSubtitles) {
+                for (SubtitleOverlay.Subtitle subtitleoverlay$subtitle2 : this.audibleSubtitles) {
                     int i = 255;
                     Component component = subtitleoverlay$subtitle2.getText();
                     Vec3 vec33 = subtitleoverlay$subtitle2.getLocation().subtract(vec3).normalize();
@@ -102,7 +_,7 @@
 
                     p_282562_.drawString(this.minecraft.font, component, -i1 / 2, -l, l1);
                     p_282562_.pose().popPose();
-                    ++i2;
+                    i2++;
                 }
             }
         }
@@ -113,7 +_,7 @@
         if (p_94646_.getSubtitle() != null) {
             Component component = p_94646_.getSubtitle();
             if (!this.subtitles.isEmpty()) {
-                for(SubtitleOverlay.Subtitle subtitleoverlay$subtitle : this.subtitles) {
+                for (SubtitleOverlay.Subtitle subtitleoverlay$subtitle : this.subtitles) {
                     if (subtitleoverlay$subtitle.getText().equals(component)) {
                         subtitleoverlay$subtitle.refresh(new Vec3(p_94645_.getX(), p_94645_.getY(), p_94645_.getZ()));
                         return;
