--- a/net/minecraft/world/entity/ai/control/LookControl.java
+++ b/net/minecraft/world/entity/ai/control/LookControl.java
@@ -51,9 +_,9 @@
         }
 
         if (this.lookAtCooldown > 0) {
-            --this.lookAtCooldown;
+            this.lookAtCooldown--;
             this.getYRotD().ifPresent(p_287447_ -> this.mob.yHeadRot = this.rotateTowards(this.mob.yHeadRot, p_287447_, this.yMaxRotSpeed));
-            this.getXRotD().ifPresent(p_340069_ -> this.mob.setXRot(this.rotateTowards(this.mob.getXRot(), p_340069_, this.xMaxRotAngle)));
+            this.getXRotD().ifPresent(p_337788_ -> this.mob.setXRot(this.rotateTowards(this.mob.getXRot(), p_337788_, this.xMaxRotAngle)));
         } else {
             this.mob.yHeadRot = this.rotateTowards(this.mob.yHeadRot, this.mob.yBodyRot, 10.0F);
         }
