--- a/net/minecraft/world/entity/schedule/Timeline.java
+++ b/net/minecraft/world/entity/schedule/Timeline.java
@@ -45,7 +_,7 @@
             int i = flag ? 0 : this.previousIndex;
             float f = flag ? keyframe1.getValue() : keyframe.getValue();
 
-            for(int j = i; j < this.keyframes.size(); ++j) {
+            for (int j = i; j < this.keyframes.size(); j++) {
                 Keyframe keyframe2 = this.keyframes.get(j);
                 if (keyframe2.getTimeStamp() > p_38059_) {
                     break;
