--- a/net/minecraft/client/multiplayer/chat/ChatTrustLevel.java
+++ b/net/minecraft/client/multiplayer/chat/ChatTrustLevel.java
@@ -56,7 +_,7 @@
 
     @Nullable
     public GuiMessageTag createTag(PlayerChatMessage p_240632_) {
-        return switch(this) {
+        return switch (this) {
             case MODIFIED -> GuiMessageTag.chatModified(p_240632_.signedContent());
             case NOT_SECURE -> GuiMessageTag.chatNotSecure();
             default -> null;
