--- a/net/minecraft/client/multiplayer/chat/ChatListener.java
+++ b/net/minecraft/client/multiplayer/chat/ChatListener.java
@@ -133,7 +_,9 @@
         Instant instant = Instant.now();
         this.handleMessage(null, () -> {
             Component component = boundChatType.decorate(message);
-            this.minecraft.gui.getChat().addMessage(component);
+            Component forgeComponent = net.neoforged.neoforge.client.ClientHooks.onClientChat(boundChatType, component, Util.NIL_UUID);
+            if (forgeComponent == null) return false;
+            this.minecraft.gui.getChat().addMessage(forgeComponent);
             this.narrateChatMessage(boundChatType, message);
             this.logSystemMessage(component, instant);
             this.previousMessageTime = Util.getMillis();
@@ -152,12 +_,16 @@
             MessageSignature messagesignature = message.signature();
             FilterMask filtermask = message.filterMask();
             if (filtermask.isEmpty()) {
-                this.minecraft.gui.getChat().addMessage(decoratedMessage, messagesignature, guimessagetag);
+                Component forgeComponent = net.neoforged.neoforge.client.ClientHooks.onClientPlayerChat(boundChatType, decoratedMessage, message, message.sender());
+                if (forgeComponent == null) return false;
+                this.minecraft.gui.getChat().addMessage(forgeComponent, messagesignature, guimessagetag);
                 this.narrateChatMessage(boundChatType, message.decoratedContent());
             } else {
                 Component component = filtermask.applyWithFormatting(message.signedContent());
                 if (component != null) {
-                    this.minecraft.gui.getChat().addMessage(boundChatType.decorate(component), messagesignature, guimessagetag);
+                    Component forgeComponent = net.neoforged.neoforge.client.ClientHooks.onClientPlayerChat(boundChatType, boundChatType.decorate(component), message, message.sender());
+                    if (forgeComponent == null) return false;
+                    this.minecraft.gui.getChat().addMessage(forgeComponent, messagesignature, guimessagetag);
                     this.narrateChatMessage(boundChatType, component);
                 }
             }
@@ -190,6 +_,8 @@
 
     public void handleSystemMessage(Component message, boolean overlay) {
         if (!this.minecraft.options.hideMatchedNames().get() || !this.minecraft.isBlocked(this.guessChatUUID(message))) {
+            message = net.neoforged.neoforge.client.ClientHooks.onClientSystemChat(message, overlay);
+            if (message == null) return;
             if (overlay) {
                 this.minecraft.gui.setOverlayMessage(message, false);
                 this.minecraft.getNarrator().saySystemQueued(message);
