--- a/net/minecraft/client/multiplayer/chat/LoggedChatMessage.java
+++ b/net/minecraft/client/multiplayer/chat/LoggedChatMessage.java
@@ -1,7 +_,7 @@
 package net.minecraft.client.multiplayer.chat;
 
 import com.mojang.authlib.GameProfile;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.time.Instant;
@@ -38,7 +_,7 @@
 
     @OnlyIn(Dist.CLIENT)
     public static record Player(GameProfile profile, PlayerChatMessage message, ChatTrustLevel trustLevel) implements LoggedChatMessage {
-        public static final Codec<LoggedChatMessage.Player> CODEC = RecordCodecBuilder.create(
+        public static final MapCodec<LoggedChatMessage.Player> CODEC = RecordCodecBuilder.mapCodec(
             p_261382_ -> p_261382_.group(
                         ExtraCodecs.GAME_PROFILE.fieldOf("profile").forGetter(LoggedChatMessage.Player::profile),
                         PlayerChatMessage.MAP_CODEC.forGetter(LoggedChatMessage.Player::message),
@@ -92,7 +_,7 @@
 
     @OnlyIn(Dist.CLIENT)
     public static record System(Component message, Instant timeStamp) implements LoggedChatMessage {
-        public static final Codec<LoggedChatMessage.System> CODEC = RecordCodecBuilder.create(
+        public static final MapCodec<LoggedChatMessage.System> CODEC = RecordCodecBuilder.mapCodec(
             p_304056_ -> p_304056_.group(
                         ComponentSerialization.CODEC.fieldOf("message").forGetter(LoggedChatMessage.System::message),
                         ExtraCodecs.INSTANT_ISO8601.fieldOf("time_stamp").forGetter(LoggedChatMessage.System::timeStamp)
