--- a/net/minecraft/client/multiplayer/chat/LoggedChatEvent.java
+++ b/net/minecraft/client/multiplayer/chat/LoggedChatEvent.java
@@ -1,6 +_,7 @@
 package net.minecraft.client.multiplayer.chat;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.function.Supplier;
 import net.minecraft.util.StringRepresentable;
 import net.neoforged.api.distmarker.Dist;
@@ -18,14 +_,14 @@
         SYSTEM("system", () -> LoggedChatMessage.System.CODEC);
 
         private final String serializedName;
-        private final Supplier<Codec<? extends LoggedChatEvent>> codec;
+        private final Supplier<MapCodec<? extends LoggedChatEvent>> codec;
 
-        private Type(String p_254335_, Supplier<Codec<? extends LoggedChatEvent>> p_254115_) {
+        private Type(String p_254335_, Supplier<MapCodec<? extends LoggedChatEvent>> p_254115_) {
             this.serializedName = p_254335_;
             this.codec = p_254115_;
         }
 
-        private Codec<? extends LoggedChatEvent> codec() {
+        private MapCodec<? extends LoggedChatEvent> codec() {
             return this.codec.get();
         }
 
