--- a/net/minecraft/world/entity/ai/sensing/AxolotlAttackablesSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/AxolotlAttackablesSensor.java
@@ -12,7 +_,7 @@
     public static final float TARGET_DETECTION_DISTANCE = 8.0F;
 
     @Override
-    protected boolean isMatchingEntity(ServerLevel level, LivingEntity body, Entity mob) {
+    protected boolean isMatchingEntity(ServerLevel level, LivingEntity body, LivingEntity mob) {
         return this.isClose(body, mob)
             && mob.isInWater()
             && (this.isHostileTarget(mob) || this.isHuntTarget(body, mob))
