--- a/com/mojang/blaze3d/shaders/EffectProgram.java
+++ b/com/mojang/blaze3d/shaders/EffectProgram.java
@@ -23,14 +_,14 @@
 
     public void attachToEffect(Effect p_166587_) {
         RenderSystem.assertOnRenderThread();
-        ++this.references;
+        this.references++;
         this.attachToShader(p_166587_);
     }
 
     @Override
     public void close() {
         RenderSystem.assertOnRenderThread();
-        --this.references;
+        this.references--;
         if (this.references <= 0) {
             super.close();
         }
