--- a/net/minecraft/client/CommandHistory.java
+++ b/net/minecraft/client/CommandHistory.java
@@ -45,7 +_,7 @@
 
     private void save() {
         try (BufferedWriter bufferedwriter = Files.newBufferedWriter(this.commandsPath, Charsets.UTF_8)) {
-            for(String s : this.lastCommands) {
+            for (String s : this.lastCommands) {
                 bufferedwriter.write(s);
                 bufferedwriter.newLine();
             }
