--- a/net/minecraft/client/Options.java
+++ b/net/minecraft/client/Options.java
@@ -613,6 +_,7 @@
     public final KeyMapping keySprint = new ToggleKeyMapping("key.sprint", 341, KeyMapping.Category.MOVEMENT, this.toggleSprint::get, true);
     public final KeyMapping keyInventory = new KeyMapping("key.inventory", 69, KeyMapping.Category.INVENTORY);
     public final KeyMapping keyDrop = new KeyMapping("key.drop", 81, KeyMapping.Category.INVENTORY);
+    public final KeyMapping keySwapOffhand = new KeyMapping("key.swapOffhand", 70, KeyMapping.Category.INVENTORY);
     public final KeyMapping keyUse = new ToggleKeyMapping("key.use", InputConstants.Type.MOUSE, 1, KeyMapping.Category.GAMEPLAY, this.toggleUse::get, false);
     public final KeyMapping keyAttack = new ToggleKeyMapping(
         "key.attack", InputConstants.Type.MOUSE, 0, KeyMapping.Category.GAMEPLAY, this.toggleAttack::get, true
@@ -709,6 +_,7 @@
                 this.keyShift,
                 this.keySprint,
                 this.keyDrop,
+                this.keySwapOffhand,
                 this.keyInventory,
                 this.keyChat,
                 this.keyPlayerList,
@@ -1027,7 +_,7 @@
         this.incompatibleResourcePacks.clear();
 
         for (Pack entry : packRepository.getSelectedPacks()) {
-            if (!entry.isFixedPosition()) {
+            if (!entry.isFixedPosition() && !entry.isHidden()) {
                 this.resourcePacks.add(entry.getId());
                 if (!entry.getCompatibility().isCompatible()) {
                     this.incompatibleResourcePacks.add(entry.getId());
@@ -1402,6 +_,7 @@
     }
 
     public Options(Minecraft minecraft, File workingDirectory) {
+        setForgeKeybindProperties();
         this.minecraft = minecraft;
         this.optionsFile = new File(workingDirectory, "options.txt");
         boolean largeDistances = Runtime.getRuntime().maxMemory() >= 1000000000L;
@@ -1425,6 +_,7 @@
             value -> this.setGraphicsPresetToCustom()
         );
         this.syncWrites = Util.getPlatform() == Util.OS.WINDOWS;
+        net.neoforged.neoforge.client.ClientHooks.onRegisterKeyMappings(this, KeyMapping.Category.SORT_ORDER);
         this.load();
         this.initialExclusiveFullscreen = this.exclusiveFullscreen.get();
     }
@@ -1556,10 +_,15 @@
         access.process("musicFrequency", this.musicFrequency);
 
         for (KeyMapping keyMapping : this.keyMappings) {
-            String currentValue = keyMapping.saveString();
+            String currentValue = keyMapping.saveString() + (keyMapping.getKeyModifier() != net.neoforged.neoforge.client.settings.KeyModifier.NONE ? ":" + keyMapping.getKeyModifier() : "");
             String newValue = access.process("key_" + keyMapping.getName(), currentValue);
             if (!currentValue.equals(newValue)) {
-                keyMapping.setKey(InputConstants.getKey(newValue));
+                if (newValue.indexOf(':') != -1) {
+                    String[] pts = newValue.split(":");
+                    keyMapping.setKeyModifierAndCode(net.neoforged.neoforge.client.settings.KeyModifier.valueFromString(pts[1]), InputConstants.getKey(pts[0]));
+                } else {
+                    keyMapping.setKeyModifierAndCode(net.neoforged.neoforge.client.settings.KeyModifier.NONE, InputConstants.getKey(newValue));
+                }
             }
         }
 
@@ -1855,6 +_,23 @@
         }
 
         repository.setSelected(selected);
+    }
+
+    private void setForgeKeybindProperties() {
+        net.neoforged.neoforge.client.settings.KeyConflictContext inGame = net.neoforged.neoforge.client.settings.KeyConflictContext.IN_GAME;
+        keyUp.setKeyConflictContext(inGame);
+        keyLeft.setKeyConflictContext(inGame);
+        keyDown.setKeyConflictContext(inGame);
+        keyRight.setKeyConflictContext(inGame);
+        keyJump.setKeyConflictContext(inGame);
+        keyShift.setKeyConflictContext(inGame);
+        keySprint.setKeyConflictContext(inGame);
+        keyAttack.setKeyConflictContext(inGame);
+        keyChat.setKeyConflictContext(inGame);
+        keyPlayerList.setKeyConflictContext(inGame);
+        keyCommand.setKeyConflictContext(inGame);
+        keyTogglePerspective.setKeyConflictContext(inGame);
+        keySmoothCamera.setKeyConflictContext(inGame);
     }
 
     public CameraType getCameraType() {
