--- a/net/minecraft/client/Options.java
+++ b/net/minecraft/client/Options.java
@@ -21,7 +_,7 @@
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
 import com.mojang.serialization.JsonOps;
-import com.mojang.serialization.DataResult.PartialResult;
+import com.mojang.serialization.DataResult.Error;
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileOutputStream;
@@ -65,7 +_,6 @@
 import net.minecraft.server.packs.repository.PackRepository;
 import net.minecraft.sounds.SoundEvents;
 import net.minecraft.sounds.SoundSource;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.GsonHelper;
 import net.minecraft.util.Mth;
 import net.minecraft.util.datafix.DataFixTypes;
@@ -151,7 +_,7 @@
         OptionInstance.forOptionEnum(),
         new OptionInstance.Enum<>(
             Arrays.asList(CloudStatus.values()),
-            ExtraCodecs.withAlternative(CloudStatus.CODEC, Codec.BOOL, p_232082_ -> p_232082_ ? CloudStatus.FANCY : CloudStatus.OFF)
+            Codec.withAlternative(CloudStatus.CODEC, Codec.BOOL, p_232082_ -> p_232082_ ? CloudStatus.FANCY : CloudStatus.OFF)
         ),
         CloudStatus.FANCY,
         p_231854_ -> {
@@ -170,11 +_,12 @@
     private static final Component GRAPHICS_TOOLTIP_FANCY = Component.translatable("options.graphics.fancy.tooltip");
     private final OptionInstance<GraphicsStatus> graphicsMode = new OptionInstance<>(
         "options.graphics",
-        p_325463_ -> {
-            return switch(p_325463_) {
+        p_339270_ -> {
+            return switch (p_339270_) {
                 case FANCY -> Tooltip.create(GRAPHICS_TOOLTIP_FANCY);
                 case FAST -> Tooltip.create(GRAPHICS_TOOLTIP_FAST);
                 case FABULOUS -> Tooltip.create(GRAPHICS_TOOLTIP_FABULOUS);
+                default -> throw new MatchException(null, null);
             };
         },
         (p_231904_, p_231905_) -> {
@@ -209,11 +_,12 @@
     private static final Component PRIORITIZE_CHUNK_TOOLTIP_NEARBY = Component.translatable("options.prioritizeChunkUpdates.nearby.tooltip");
     private final OptionInstance<PrioritizeChunkUpdates> prioritizeChunkUpdates = new OptionInstance<>(
         "options.prioritizeChunkUpdates",
-        p_325462_ -> {
-            return switch(p_325462_) {
+        p_339269_ -> {
+            return switch (p_339269_) {
                 case NONE -> Tooltip.create(PRIORITIZE_CHUNK_TOOLTIP_NONE);
                 case PLAYER_AFFECTED -> Tooltip.create(PRIORITIZE_CHUNK_TOOLTIP_PLAYER_AFFECTED);
                 case NEARBY -> Tooltip.create(PRIORITIZE_CHUNK_TOOLTIP_NEARBY);
+                default -> throw new MatchException(null, null);
             };
         },
         OptionInstance.forOptionEnum(),
@@ -272,12 +_,12 @@
     private final OptionInstance<Boolean> highContrast = OptionInstance.createBoolean(
         "options.accessibility.high_contrast", OptionInstance.cachedConstantTooltip(ACCESSIBILITY_TOOLTIP_CONTRAST_MODE), false, p_275860_ -> {
             PackRepository packrepository = Minecraft.getInstance().getResourcePackRepository();
-            boolean flag2 = packrepository.getSelectedIds().contains("high_contrast");
-            if (!flag2 && p_275860_) {
+            boolean flag1 = packrepository.getSelectedIds().contains("high_contrast");
+            if (!flag1 && p_275860_) {
                 if (packrepository.addPack("high_contrast")) {
                     this.updateResourcePacks(packrepository);
                 }
-            } else if (flag2 && !p_275860_ && packrepository.removePack("high_contrast")) {
+            } else if (flag1 && !p_275860_ && packrepository.removePack("high_contrast")) {
                 this.updateResourcePacks(packrepository);
             }
         }
@@ -434,7 +_,7 @@
     );
     private final OptionInstance<Boolean> reducedDebugInfo = OptionInstance.createBoolean("options.reducedDebugInfo", false);
     private final Map<SoundSource, OptionInstance<Double>> soundSourceVolumes = Util.make(new EnumMap<>(SoundSource.class), p_247766_ -> {
-        for(SoundSource soundsource : SoundSource.values()) {
+        for (SoundSource soundsource : SoundSource.values()) {
             p_247766_.put(soundsource, this.createSoundSliderOptionInstance("soundCategory." + soundsource.getName(), soundsource));
         }
     });
@@ -492,7 +_,6 @@
         }
     );
     public boolean skipMultiplayerWarning;
-    public boolean skipRealms32bitWarning;
     private static final Component CHAT_TOOLTIP_HIDE_MATCHED_NAMES = Component.translatable("options.hideMatchedNames.tooltip");
     private final OptionInstance<Boolean> hideMatchedNames = OptionInstance.createBoolean(
         "options.hideMatchedNames", OptionInstance.cachedConstantTooltip(CHAT_TOOLTIP_HIDE_MATCHED_NAMES), true
@@ -519,7 +_,6 @@
     public final KeyMapping keyPlayerList = new KeyMapping("key.playerlist", 258, "key.categories.multiplayer");
     public final KeyMapping keyCommand = new KeyMapping("key.command", 47, "key.categories.multiplayer");
     public final KeyMapping keySocialInteractions = new KeyMapping("key.socialInteractions", 80, "key.categories.multiplayer");
-    public final KeyMapping keyPotato = new KeyMapping("key.potato", 39, "key.categories.multiplayer");
     public final KeyMapping keyScreenshot = new KeyMapping("key.screenshot", 291, "key.categories.misc");
     public final KeyMapping keyTogglePerspective = new KeyMapping("key.togglePerspective", 294, "key.categories.misc");
     public final KeyMapping keySmoothCamera = new KeyMapping("key.smoothCamera", InputConstants.UNKNOWN.getValue(), "key.categories.misc");
@@ -557,7 +_,6 @@
             this.keyPickItem,
             this.keyCommand,
             this.keySocialInteractions,
-            this.keyPotato,
             this.keyScreenshot,
             this.keyTogglePerspective,
             this.keySmoothCamera,
@@ -580,7 +_,7 @@
         "options.fov",
         OptionInstance.noTooltip(),
         (p_231999_, p_232000_) -> {
-            return switch(p_232000_) {
+            return switch (p_232000_) {
                 case 70 -> genericValueLabel(p_231999_, Component.translatable("options.fov.min"));
                 case 110 -> genericValueLabel(p_231999_, Component.translatable("options.fov.max"));
                 default -> genericValueLabel(p_231999_, p_232000_);
@@ -693,8 +_,7 @@
             return !minecraft.isRunning() ? 2147483646 : minecraft.getWindow().calculateScale(0, minecraft.isEnforceUnicode());
         }, 2147483646),
         0,
-        p_330905_ -> {
-        }
+        p_337405_ -> this.minecraft.resizeDisplay()
     );
     private final OptionInstance<ParticleStatus> particles = new OptionInstance<>(
         "options.particles",
@@ -702,7 +_,7 @@
         OptionInstance.forOptionEnum(),
         new OptionInstance.Enum<>(Arrays.asList(ParticleStatus.values()), Codec.INT.xmap(ParticleStatus::byId, ParticleStatus::getId)),
         ParticleStatus.ALL,
-        p_329693_ -> {
+        p_338218_ -> {
         }
     );
     private final OptionInstance<NarratorStatus> narrator = new OptionInstance<>(
@@ -729,9 +_,10 @@
             }
         },
         new OptionInstance.LazyEnum<>(
+            // FORGE: fix incorrect string comparison - PR #8767
             () -> Stream.concat(Stream.of(""), Minecraft.getInstance().getSoundManager().getAvailableSoundDevices().stream()).toList(),
             p_232011_ -> Minecraft.getInstance().isRunning()
-                        && p_232011_ != ""
+                        && (p_232011_ == null || !p_232011_.isEmpty())
                         && !Minecraft.getInstance().getSoundManager().getAvailableSoundDevices().contains(p_232011_)
                     ? Optional.empty()
                     : Optional.of(p_232011_),
@@ -800,8 +_,8 @@
         this.resourcePacks.clear();
         this.incompatibleResourcePacks.clear();
 
-        for(Pack pack : p_275268_.getSelectedPacks()) {
-            if (!pack.isFixedPosition()) {
+        for (Pack pack : p_275268_.getSelectedPacks()) {
+            if (!pack.isFixedPosition() && !pack.isHidden()) {
                 this.resourcePacks.add(pack.getId());
                 if (!pack.getCompatibility().isCompatible()) {
                     this.incompatibleResourcePacks.add(pack.getId());
@@ -1096,24 +_,24 @@
     }
 
     public Options(Minecraft p_92138_, File p_92139_) {
+        setForgeKeybindProperties();
         this.minecraft = p_92138_;
         this.optionsFile = new File(p_92139_, "options.txt");
-        boolean flag = p_92138_.is64Bit();
-        boolean flag1 = flag && Runtime.getRuntime().maxMemory() >= 1000000000L;
+        boolean flag = Runtime.getRuntime().maxMemory() >= 1000000000L;
         this.renderDistance = new OptionInstance<>(
             "options.renderDistance",
             OptionInstance.noTooltip(),
             (p_264664_, p_296150_) -> genericValueLabel(p_264664_, Component.translatable("options.chunks", p_296150_)),
-            new OptionInstance.IntRange(2, flag1 ? 32 : 16),
-            flag ? 12 : 8,
+            new OptionInstance.IntRange(2, flag ? 32 : 16),
+            12,
             p_231951_ -> Minecraft.getInstance().levelRenderer.needsUpdate()
         );
         this.simulationDistance = new OptionInstance<>(
             "options.simulationDistance",
             OptionInstance.noTooltip(),
             (p_231962_, p_331290_) -> genericValueLabel(p_231962_, Component.translatable("options.chunks", p_331290_)),
-            new OptionInstance.IntRange(5, flag1 ? 32 : 16),
-            flag ? 12 : 8,
+            new OptionInstance.IntRange(5, flag ? 32 : 16),
+            12,
             p_326374_ -> {
             }
         );
@@ -1222,7 +_,6 @@
         p_168428_.process("rawMouseInput", this.rawMouseInput);
         this.glDebugVerbosity = p_168428_.process("glDebugVerbosity", this.glDebugVerbosity);
         this.skipMultiplayerWarning = p_168428_.process("skipMultiplayerWarning", this.skipMultiplayerWarning);
-        this.skipRealms32bitWarning = p_168428_.process("skipRealms32bitWarning", this.skipRealms32bitWarning);
         p_168428_.process("hideMatchedNames", this.hideMatchedNames);
         this.joinedFirstServer = p_168428_.process("joinedFirstServer", this.joinedFirstServer);
         this.hideBundleTutorial = p_168428_.process("hideBundleTutorial", this.hideBundleTutorial);
@@ -1235,19 +_,29 @@
         this.onboardAccessibility = p_168428_.process("onboardAccessibility", this.onboardAccessibility);
         p_168428_.process("menuBackgroundBlurriness", this.menuBackgroundBlurriness);
 
-        for(KeyMapping keymapping : this.keyMappings) {
-            String s = keymapping.saveString();
+        net.neoforged.neoforge.forge.snapshots.ForgeSnapshotsMod.processOptions(p_168428_);
+        processOptionsForge(p_168428_);
+    }
+    // FORGE: split off to allow reloading options after mod loading is done
+    private void processOptionsForge(Options.FieldAccess p_168428_)
+    {
+        for (KeyMapping keymapping : this.keyMappings) {
+            String s = keymapping.saveString() + (keymapping.getKeyModifier() != net.neoforged.neoforge.client.settings.KeyModifier.NONE ? ":" + keymapping.getKeyModifier() : "");
             String s1 = p_168428_.process("key_" + keymapping.getName(), s);
             if (!s.equals(s1)) {
-                keymapping.setKey(InputConstants.getKey(s1));
+                if (s1.indexOf(':') != -1) {
+                    String[] pts = s1.split(":");
+                    keymapping.setKeyModifierAndCode(net.neoforged.neoforge.client.settings.KeyModifier.valueFromString(pts[1]), InputConstants.getKey(pts[0]));
+                } else
+                    keymapping.setKeyModifierAndCode(net.neoforged.neoforge.client.settings.KeyModifier.NONE, InputConstants.getKey(s1));
             }
         }
 
-        for(SoundSource soundsource : SoundSource.values()) {
+        for (SoundSource soundsource : SoundSource.values()) {
             p_168428_.process("soundCategory_" + soundsource.getName(), this.soundSourceVolumes.get(soundsource));
         }
 
-        for(PlayerModelPart playermodelpart : PlayerModelPart.values()) {
+        for (PlayerModelPart playermodelpart : PlayerModelPart.values()) {
             boolean flag = this.modelParts.contains(playermodelpart);
             boolean flag1 = p_168428_.process("modelPart_" + playermodelpart.getId(), flag);
             if (flag1 != flag) {
@@ -1257,6 +_,9 @@
     }
 
     public void load() {
+        this.load(false);
+    }
+    public void load(boolean limited) {
         try {
             if (!this.optionsFile.exists()) {
                 return;
@@ -1284,7 +_,8 @@
                 }
             }
 
-            this.processOptions(
+            java.util.function.Consumer<FieldAccess> processor = limited ? this::processOptionsForge : this::processOptions;
+            processor.accept(
                 new Options.FieldAccess() {
                     @Nullable
                     private String getValueOrNull(String p_168459_) {
@@ -1300,10 +_,10 @@
                             DataResult<T> dataresult = p_232126_.codec().parse(JsonOps.INSTANCE, jsonelement);
                             dataresult.error()
                                 .ifPresent(
-                                    p_232130_ -> Options.LOGGER
-                                            .error("Error parsing option value " + s + " for option " + p_232126_ + ": " + p_232130_.message())
+                                    p_339273_ -> Options.LOGGER
+                                            .error("Error parsing option value " + s + " for option " + p_232126_ + ": " + p_339273_.message())
                                 );
-                            dataresult.result().ifPresent(p_232126_::set);
+                            dataresult.ifSuccess(p_232126_::set);
                         }
                     }
     
@@ -1354,7 +_,7 @@
                     @Override
                     public <T> T process(String p_168470_, T p_168471_, Function<String, T> p_168472_, Function<T, String> p_168473_) {
                         String s = this.getValueOrNull(p_168470_);
-                        return (T)(s == null ? p_168471_ : p_168472_.apply(s));
+                        return s == null ? p_168471_ : p_168472_.apply(s);
                     }
                 }
             );
@@ -1394,57 +_,60 @@
     public void save() {
         try (final PrintWriter printwriter = new PrintWriter(new OutputStreamWriter(new FileOutputStream(this.optionsFile), StandardCharsets.UTF_8))) {
             printwriter.println("version:" + SharedConstants.getCurrentVersion().getDataVersion().getVersion());
-            this.processOptions(new Options.FieldAccess() {
-                public void writePrefix(String p_168491_) {
-                    printwriter.print(p_168491_);
-                    printwriter.print(':');
-                }
-
-                @Override
-                public <T> void process(String p_232135_, OptionInstance<T> p_232136_) {
-                    DataResult<JsonElement> dataresult = p_232136_.codec().encodeStart(JsonOps.INSTANCE, p_232136_.get());
-                    dataresult.error().ifPresent(p_232133_ -> Options.LOGGER.error("Error saving option " + p_232136_ + ": " + p_232133_));
-                    dataresult.result().ifPresent(p_232140_ -> {
-                        this.writePrefix(p_232135_);
-                        printwriter.println(Options.GSON.toJson(p_232140_));
-                    });
-                }
-
-                @Override
-                public int process(String p_168499_, int p_168500_) {
-                    this.writePrefix(p_168499_);
-                    printwriter.println(p_168500_);
-                    return p_168500_;
-                }
-
-                @Override
-                public boolean process(String p_168515_, boolean p_168516_) {
-                    this.writePrefix(p_168515_);
-                    printwriter.println(p_168516_);
-                    return p_168516_;
-                }
-
-                @Override
-                public String process(String p_168512_, String p_168513_) {
-                    this.writePrefix(p_168512_);
-                    printwriter.println(p_168513_);
-                    return p_168513_;
-                }
-
-                @Override
-                public float process(String p_168496_, float p_168497_) {
-                    this.writePrefix(p_168496_);
-                    printwriter.println(p_168497_);
-                    return p_168497_;
-                }
-
-                @Override
-                public <T> T process(String p_168502_, T p_168503_, Function<String, T> p_168504_, Function<T, String> p_168505_) {
-                    this.writePrefix(p_168502_);
-                    printwriter.println(p_168505_.apply(p_168503_));
-                    return p_168503_;
-                }
-            });
+            this.processOptions(
+                new Options.FieldAccess() {
+                    public void writePrefix(String p_168491_) {
+                        printwriter.print(p_168491_);
+                        printwriter.print(':');
+                    }
+
+                    @Override
+                    public <T> void process(String p_232135_, OptionInstance<T> p_232136_) {
+                        p_232136_.codec()
+                            .encodeStart(JsonOps.INSTANCE, p_232136_.get())
+                            .ifError(p_339275_ -> Options.LOGGER.error("Error saving option " + p_232136_ + ": " + p_339275_))
+                            .ifSuccess(p_232140_ -> {
+                                this.writePrefix(p_232135_);
+                                printwriter.println(Options.GSON.toJson(p_232140_));
+                            });
+                    }
+
+                    @Override
+                    public int process(String p_168499_, int p_168500_) {
+                        this.writePrefix(p_168499_);
+                        printwriter.println(p_168500_);
+                        return p_168500_;
+                    }
+
+                    @Override
+                    public boolean process(String p_168515_, boolean p_168516_) {
+                        this.writePrefix(p_168515_);
+                        printwriter.println(p_168516_);
+                        return p_168516_;
+                    }
+
+                    @Override
+                    public String process(String p_168512_, String p_168513_) {
+                        this.writePrefix(p_168512_);
+                        printwriter.println(p_168513_);
+                        return p_168513_;
+                    }
+
+                    @Override
+                    public float process(String p_168496_, float p_168497_) {
+                        this.writePrefix(p_168496_);
+                        printwriter.println(p_168497_);
+                        return p_168497_;
+                    }
+
+                    @Override
+                    public <T> T process(String p_168502_, T p_168503_, Function<String, T> p_168504_, Function<T, String> p_168505_) {
+                        this.writePrefix(p_168502_);
+                        printwriter.println(p_168505_.apply(p_168503_));
+                        return p_168503_;
+                    }
+                }
+            );
             if (this.minecraft.getWindow().getPreferredFullscreenVideoMode().isPresent()) {
                 printwriter.println("fullscreenResolution:" + this.minecraft.getWindow().getPreferredFullscreenVideoMode().get().write());
             }
@@ -1456,9 +_,10 @@
     }
 
     public ClientInformation buildPlayerInformation() {
+        if (net.neoforged.neoforge.client.loading.ClientModLoader.isLoading()) return ClientInformation.createDefault(); //Don't save settings before mods add keybindigns and the like to prevent them from being deleted.
         int i = 0;
 
-        for(PlayerModelPart playermodelpart : this.modelParts) {
+        for (PlayerModelPart playermodelpart : this.modelParts) {
             i |= playermodelpart.getMask();
         }
 
@@ -1509,7 +_,7 @@
         Set<String> set = Sets.newLinkedHashSet();
         Iterator<String> iterator = this.resourcePacks.iterator();
 
-        while(iterator.hasNext()) {
+        while (iterator.hasNext()) {
             String s = iterator.next();
             Pack pack = p_92146_.getPack(s);
             if (pack == null && !s.startsWith("file/")) {
@@ -1533,6 +_,23 @@
         p_92146_.setSelected(set);
     }
 
+    private void setForgeKeybindProperties() {
+        net.neoforged.neoforge.client.settings.KeyConflictContext inGame = net.neoforged.neoforge.client.settings.KeyConflictContext.IN_GAME;
+        keyUp.setKeyConflictContext(inGame);
+        keyLeft.setKeyConflictContext(inGame);
+        keyDown.setKeyConflictContext(inGame);
+        keyRight.setKeyConflictContext(inGame);
+        keyJump.setKeyConflictContext(inGame);
+        keyShift.setKeyConflictContext(inGame);
+        keySprint.setKeyConflictContext(inGame);
+        keyAttack.setKeyConflictContext(inGame);
+        keyChat.setKeyConflictContext(inGame);
+        keyPlayerList.setKeyConflictContext(inGame);
+        keyCommand.setKeyConflictContext(inGame);
+        keyTogglePerspective.setKeyConflictContext(inGame);
+        keySmoothCamera.setKeyConflictContext(inGame);
+    }
+
     public CameraType getCameraType() {
         return this.cameraType;
     }
@@ -1567,7 +_,7 @@
         list.add(Pair.of("resourcePacks", this.resourcePacks));
         return list.stream()
             .sorted(Comparator.comparing(Pair::getFirst))
-            .map(p_231848_ -> (String)p_231848_.getFirst() + ": " + p_231848_.getSecond())
+            .map(p_339268_ -> p_339268_.getFirst() + ": " + p_339268_.getSecond())
             .collect(Collectors.joining(System.lineSeparator()));
     }
 
