--- a/net/minecraft/client/OptionInstance.java
+++ b/net/minecraft/client/OptionInstance.java
@@ -24,7 +_,6 @@
 import net.minecraft.client.gui.components.Tooltip;
 import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.Mth;
 import net.minecraft.util.OptionEnum;
 import net.neoforged.api.distmarker.Dist;
@@ -199,15 +_,15 @@
 
         @Override
         public Codec<Integer> codec() {
-            return ExtraCodecs.validate(
-                Codec.INT,
-                p_276098_ -> {
-                    int i = this.encodableMaxInclusive + 1;
-                    return p_276098_.compareTo(this.minInclusive) >= 0 && p_276098_.compareTo(i) <= 0
-                        ? DataResult.success(p_276098_)
-                        : DataResult.error(() -> "Value " + p_276098_ + " outside of range [" + this.minInclusive + ":" + i + "]", p_276098_);
-                }
-            );
+            return Codec.INT
+                .validate(
+                    p_276098_ -> {
+                        int i = this.encodableMaxInclusive + 1;
+                        return p_276098_.compareTo(this.minInclusive) >= 0 && p_276098_.compareTo(i) <= 0
+                            ? DataResult.success(p_276098_)
+                            : DataResult.error(() -> "Value " + p_276098_ + " outside of range [" + this.minInclusive + ":" + i + "]", p_276098_);
+                    }
+                );
         }
 
         @Override
@@ -303,7 +_,7 @@
 
                 @Override
                 public R fromSliderValue(double p_231676_) {
-                    return p_231658_.apply(IntRangeBase.this.fromSliderValue(p_231676_));
+                    return (R)p_231658_.apply(IntRangeBase.this.fromSliderValue(p_231676_));
                 }
 
                 @Override
@@ -435,7 +_,7 @@
 
                 @Override
                 public R fromSliderValue(double p_231775_) {
-                    return p_231751_.apply(UnitDouble.this.fromSliderValue(p_231775_));
+                    return (R)p_231751_.apply(UnitDouble.this.fromSliderValue(p_231775_));
                 }
 
                 @Override
@@ -447,7 +_,7 @@
 
         @Override
         public Codec<Double> codec() {
-            return ExtraCodecs.withAlternative(Codec.doubleRange(0.0, 1.0), Codec.BOOL, p_231745_ -> p_231745_ ? 1.0 : 0.0);
+            return Codec.withAlternative(Codec.doubleRange(0.0, 1.0), Codec.BOOL, p_231745_ -> p_231745_ ? 1.0 : 0.0);
         }
     }
 
