--- a/net/minecraft/client/ResourceLoadStateTracker.java
+++ b/net/minecraft/client/ResourceLoadStateTracker.java
@@ -21,7 +_,7 @@
     private int reloadCount;
 
     public void startReload(ResourceLoadStateTracker.ReloadReason p_168558_, List<PackResources> p_168559_) {
-        ++this.reloadCount;
+        this.reloadCount++;
         if (this.reloadState != null && !this.reloadState.finished) {
             LOGGER.warn("Reload already ongoing, replacing");
         }
