--- a/net/minecraft/client/Screenshot.java
+++ b/net/minecraft/client/Screenshot.java
@@ -43,6 +_,13 @@
                     file2 = new File(file1, forceName);
                 }
 
+                net.neoforged.neoforge.client.event.ScreenshotEvent event = net.neoforged.neoforge.client.ClientHooks.onScreenshot(image, file2);
+                if (event.isCanceled()) {
+                    callback.accept(event.getCancelMessage());
+                    return;
+                }
+                final File target = event.getScreenshotFile();
+
                 Util.ioPool()
                     .execute(
                         () -> {
@@ -50,11 +_,15 @@
                                 NativeImage $$4x = image;
 
                                 try {
-                                    image.writeToFile(file2);
-                                    Component component = Component.literal(file2.getName())
+                                    $$4x.writeToFile(target);
+                                    if (event.getResultMessage() != null) {
+                                        callback.accept(event.getResultMessage());
+                                    } else {
+                                    Component component = Component.literal(target.getName())
                                         .withStyle(ChatFormatting.UNDERLINE)
-                                        .withStyle(s -> s.withClickEvent(new ClickEvent.OpenFile(file2.getAbsoluteFile())));
+                                        .withStyle(s -> s.withClickEvent(new ClickEvent.OpenFile(target.getAbsoluteFile())));
                                     callback.accept(Component.translatable("screenshot.success", component));
+                                    }
                                 } catch (Throwable throwable1) {
                                     if (image != null) {
                                         try {
