--- a/com/mojang/blaze3d/pipeline/MainTarget.java
+++ b/com/mojang/blaze3d/pipeline/MainTarget.java
@@ -58,7 +_,7 @@
         this.depthBufferId = TextureUtil.generateTextureId();
         MainTarget.AttachmentState maintarget$attachmentstate = MainTarget.AttachmentState.NONE;
 
-        for(MainTarget.Dimension maintarget$dimension : MainTarget.Dimension.listWithFallback(p_166147_, p_166148_)) {
+        for (MainTarget.Dimension maintarget$dimension : MainTarget.Dimension.listWithFallback(p_166147_, p_166148_)) {
             maintarget$attachmentstate = MainTarget.AttachmentState.NONE;
             if (this.allocateColorAttachment(maintarget$dimension)) {
                 maintarget$attachmentstate = maintarget$attachmentstate.with(MainTarget.AttachmentState.COLOR);
