--- a/net/minecraft/world/entity/AgeableMob.java
+++ b/net/minecraft/world/entity/AgeableMob.java
@@ -128,7 +_,7 @@
                     this.level().addParticle(ParticleTypes.HAPPY_VILLAGER, this.getRandomX(1.0), this.getRandomY() + 0.5, this.getRandomZ(1.0), 0.0, 0.0, 0.0);
                 }
 
-                --this.forcedAgeTimer;
+                this.forcedAgeTimer--;
             }
         } else if (this.isAlive()) {
             int i = this.getAge();
@@ -141,11 +_,8 @@
     }
 
     protected void ageBoundaryReached() {
-        if (!this.isBaby() && this.isPassenger()) {
-            Entity entity = this.getVehicle();
-            if (entity instanceof Boat boat && !boat.hasEnoughSpaceFor(this)) {
-                this.stopRiding();
-            }
+        if (!this.isBaby() && this.isPassenger() && this.getVehicle() instanceof Boat boat && !boat.hasEnoughSpaceFor(this)) {
+            this.stopRiding();
         }
     }
 
@@ -186,7 +_,7 @@
         }
 
         public void increaseGroupSizeByOne() {
-            ++this.groupSize;
+            this.groupSize++;
         }
 
         public boolean isShouldSpawnBaby() {
