--- a/net/minecraft/world/entity/AnimationState.java
+++ b/net/minecraft/world/entity/AnimationState.java
@@ -40,7 +_,7 @@
     public void updateTime(float p_216975_, float p_216976_) {
         if (this.isStarted()) {
             long i = Mth.lfloor((double)(p_216975_ * 1000.0F / 20.0F));
-            this.accumulatedTime += (long)((float)(i - this.lastTime) * p_216976_);
+            this.accumulatedTime = this.accumulatedTime + (long)((float)(i - this.lastTime) * p_216976_);
             this.lastTime = i;
         }
     }
