--- a/net/minecraft/world/entity/AreaEffectCloud.java
+++ b/net/minecraft/world/entity/AreaEffectCloud.java
@@ -10,7 +_,6 @@
 import java.util.UUID;
 import java.util.Map.Entry;
 import javax.annotation.Nullable;
-import net.minecraft.Util;
 import net.minecraft.commands.arguments.ParticleArgument;
 import net.minecraft.core.particles.ColorParticleOption;
 import net.minecraft.core.particles.ParticleOptions;
@@ -33,7 +_,6 @@
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final int TIME_BETWEEN_APPLICATIONS = 5;
     private static final EntityDataAccessor<Float> DATA_RADIUS = SynchedEntityData.defineId(AreaEffectCloud.class, EntityDataSerializers.FLOAT);
-    private static final EntityDataAccessor<Integer> DATA_COLOR = SynchedEntityData.defineId(AreaEffectCloud.class, EntityDataSerializers.INT);
     private static final EntityDataAccessor<Boolean> DATA_WAITING = SynchedEntityData.defineId(AreaEffectCloud.class, EntityDataSerializers.BOOLEAN);
     private static final EntityDataAccessor<ParticleOptions> DATA_PARTICLE = SynchedEntityData.defineId(AreaEffectCloud.class, EntityDataSerializers.PARTICLE);
     private static final float MAX_RADIUS = 32.0F;
@@ -66,10 +_,9 @@
 
     @Override
     protected void defineSynchedData(SynchedEntityData.Builder p_326332_) {
-        p_326332_.define(DATA_COLOR, 0);
         p_326332_.define(DATA_RADIUS, 3.0F);
         p_326332_.define(DATA_WAITING, false);
-        p_326332_.define(DATA_PARTICLE, ColorParticleOption.create(ParticleTypes.ENTITY_EFFECT, 0.0F, 0.0F, 0.0F));
+        p_326332_.define(DATA_PARTICLE, ColorParticleOption.create(ParticleTypes.ENTITY_EFFECT, -1));
     }
 
     public void setRadius(float p_19713_) {
@@ -97,17 +_,17 @@
     }
 
     private void updateColor() {
-        this.entityData.set(DATA_COLOR, this.potionContents.equals(PotionContents.EMPTY) ? 0 : this.potionContents.getColor());
+        ParticleOptions particleoptions = this.entityData.get(DATA_PARTICLE);
+        if (particleoptions instanceof ColorParticleOption colorparticleoption) {
+            int i = this.potionContents.equals(PotionContents.EMPTY) ? 0 : this.potionContents.getColor();
+            this.entityData.set(DATA_PARTICLE, ColorParticleOption.create(colorparticleoption.getType(), i));
+        }
     }
 
     public void addEffect(MobEffectInstance p_19717_) {
         this.setPotionContents(this.potionContents.withEffectAdded(p_19717_));
     }
 
-    public int getColor() {
-        return this.getEntityData().get(DATA_COLOR);
-    }
-
     public ParticleOptions getParticle() {
         return this.getEntityData().get(DATA_PARTICLE);
     }
@@ -153,31 +_,26 @@
                 f1 = f;
             }
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 float f2 = this.random.nextFloat() * (float) (Math.PI * 2);
                 float f3 = Mth.sqrt(this.random.nextFloat()) * f1;
                 double d0 = this.getX() + (double)(Mth.cos(f2) * f3);
                 double d2 = this.getY();
                 double d4 = this.getZ() + (double)(Mth.sin(f2) * f3);
-                double d5;
-                double d6;
-                double d7;
                 if (particleoptions.getType() == ParticleTypes.ENTITY_EFFECT) {
-                    int k = flag && this.random.nextBoolean() ? 16777215 : this.getColor();
-                    d5 = (double)((float)(k >> 16 & 0xFF) / 255.0F);
-                    d6 = (double)((float)(k >> 8 & 0xFF) / 255.0F);
-                    d7 = (double)((float)(k & 0xFF) / 255.0F);
+                    if (flag && this.random.nextBoolean()) {
+                        this.level().addAlwaysVisibleParticle(ColorParticleOption.create(ParticleTypes.ENTITY_EFFECT, -1), d0, d2, d4, 0.0, 0.0, 0.0);
+                    } else {
+                        this.level().addAlwaysVisibleParticle(particleoptions, d0, d2, d4, 0.0, 0.0, 0.0);
+                    }
                 } else if (flag) {
-                    d5 = 0.0;
-                    d6 = 0.0;
-                    d7 = 0.0;
+                    this.level().addAlwaysVisibleParticle(particleoptions, d0, d2, d4, 0.0, 0.0, 0.0);
                 } else {
-                    d5 = (0.5 - this.random.nextDouble()) * 0.15;
-                    d6 = 0.01F;
-                    d7 = (0.5 - this.random.nextDouble()) * 0.15;
+                    this.level()
+                        .addAlwaysVisibleParticle(
+                            particleoptions, d0, d2, d4, (0.5 - this.random.nextDouble()) * 0.15, 0.01F, (0.5 - this.random.nextDouble()) * 0.15
+                        );
                 }
-
-                this.level().addAlwaysVisibleParticle(particleoptions, d0, d2, d4, d5, d6, d7);
             }
         } else {
             if (this.tickCount >= this.waitTime + this.duration) {
@@ -211,14 +_,14 @@
                 } else {
                     List<MobEffectInstance> list = Lists.newArrayList();
                     if (this.potionContents.potion().isPresent()) {
-                        for(MobEffectInstance mobeffectinstance : this.potionContents.potion().get().value().getEffects()) {
+                        for (MobEffectInstance mobeffectinstance1 : this.potionContents.potion().get().value().getEffects()) {
                             list.add(
                                 new MobEffectInstance(
-                                    mobeffectinstance.getEffect(),
-                                    mobeffectinstance.mapDuration(p_267926_ -> p_267926_ / 4),
-                                    mobeffectinstance.getAmplifier(),
-                                    mobeffectinstance.isAmbient(),
-                                    mobeffectinstance.isVisible()
+                                    mobeffectinstance1.getEffect(),
+                                    mobeffectinstance1.mapDuration(p_267926_ -> p_267926_ / 4),
+                                    mobeffectinstance1.getAmplifier(),
+                                    mobeffectinstance1.isAmbient(),
+                                    mobeffectinstance1.isVisible()
                                 )
                             );
                         }
@@ -227,21 +_,21 @@
                     list.addAll(this.potionContents.customEffects());
                     List<LivingEntity> list1 = this.level().getEntitiesOfClass(LivingEntity.class, this.getBoundingBox());
                     if (!list1.isEmpty()) {
-                        for(LivingEntity livingentity : list1) {
+                        for (LivingEntity livingentity : list1) {
                             if (!this.victims.containsKey(livingentity) && livingentity.isAffectedByPotions()) {
-                                double d8 = livingentity.getX() - this.getX();
+                                double d5 = livingentity.getX() - this.getX();
                                 double d1 = livingentity.getZ() - this.getZ();
-                                double d3 = d8 * d8 + d1 * d1;
+                                double d3 = d5 * d5 + d1 * d1;
                                 if (d3 <= (double)(f * f)) {
                                     this.victims.put(livingentity, this.tickCount + this.reapplicationDelay);
 
-                                    for(MobEffectInstance mobeffectinstance1 : list) {
-                                        if (mobeffectinstance1.getEffect().value().isInstantenous()) {
-                                            mobeffectinstance1.getEffect()
+                                    for (MobEffectInstance mobeffectinstance : list) {
+                                        if (mobeffectinstance.getEffect().value().isInstantenous()) {
+                                            mobeffectinstance.getEffect()
                                                 .value()
-                                                .applyInstantenousEffect(this, this.getOwner(), livingentity, mobeffectinstance1.getAmplifier(), 0.5);
+                                                .applyInstantenousEffect(this, this.getOwner(), livingentity, mobeffectinstance.getAmplifier(), 0.5);
                                         } else {
-                                            livingentity.addEffect(new MobEffectInstance(mobeffectinstance1), this);
+                                            livingentity.addEffect(new MobEffectInstance(mobeffectinstance), this);
                                         }
                                     }
 
@@ -256,7 +_,7 @@
                                     }
 
                                     if (this.durationOnUse != 0) {
-                                        this.duration += this.durationOnUse;
+                                        this.duration = this.duration + this.durationOnUse;
                                         if (this.duration <= 0) {
                                             this.discard();
                                             return;
@@ -366,7 +_,7 @@
         }
 
         if (!this.potionContents.equals(PotionContents.EMPTY)) {
-            Tag tag = Util.getOrThrow(PotionContents.CODEC.encodeStart(NbtOps.INSTANCE, this.potionContents), IllegalStateException::new);
+            Tag tag = PotionContents.CODEC.encodeStart(NbtOps.INSTANCE, this.potionContents).getOrThrow();
             p_19737_.put("potion_contents", tag);
         }
     }
