--- a/net/minecraft/world/entity/EquipmentSlot.java
+++ b/net/minecraft/world/entity/EquipmentSlot.java
@@ -63,7 +_,7 @@
     }
 
     public static EquipmentSlot byTypeAndIndex(EquipmentSlot.Type p_20745_, int p_20746_) {
-        for(EquipmentSlot equipmentslot : values()) {
+        for (EquipmentSlot equipmentslot : values()) {
             if (equipmentslot.getType() == p_20745_ && equipmentslot.getIndex() == p_20746_) {
                 return equipmentslot;
             }
