--- /dev/null
+++ b/net/minecraft/world/entity/EquipmentUser.java
@@ -1,0 +_,61 @@
+package net.minecraft.world.entity;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Equipable;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.storage.loot.BuiltInLootTables;
+import net.minecraft.world.level.storage.loot.LootParams;
+import net.minecraft.world.level.storage.loot.LootTable;
+
+public interface EquipmentUser {
+    void setItemSlot(EquipmentSlot p_338576_, ItemStack p_338519_);
+
+    ItemStack getItemBySlot(EquipmentSlot p_338597_);
+
+    void setDropChance(EquipmentSlot p_338643_, float p_338569_);
+
+    default void equip(ResourceLocation p_338473_, LootParams p_338202_) {
+        this.equip(p_338473_, p_338202_, 0L);
+    }
+
+    default void equip(ResourceLocation p_338692_, LootParams p_338408_, long p_338485_) {
+        ResourceKey<LootTable> resourcekey = ResourceKey.create(Registries.LOOT_TABLE, p_338692_);
+        if (!resourcekey.equals(BuiltInLootTables.EMPTY)) {
+            LootTable loottable = p_338408_.getLevel().getServer().reloadableRegistries().getLootTable(resourcekey);
+            if (loottable != LootTable.EMPTY) {
+                List<ItemStack> list = loottable.getRandomItems(p_338408_, p_338485_);
+                List<EquipmentSlot> list1 = new ArrayList<>();
+
+                for (ItemStack itemstack : list) {
+                    EquipmentSlot equipmentslot = this.resolveSlot(itemstack, list1);
+                    if (equipmentslot != null) {
+                        ItemStack itemstack1 = equipmentslot.isArmor() ? itemstack.copyWithCount(1) : itemstack;
+                        this.setItemSlot(equipmentslot, itemstack1);
+                        this.setDropChance(equipmentslot, 0.085F);
+                        list1.add(equipmentslot);
+                    }
+                }
+            }
+        }
+    }
+
+    @Nullable
+    default EquipmentSlot resolveSlot(ItemStack p_338225_, List<EquipmentSlot> p_338547_) {
+        Equipable equipable = Equipable.get(p_338225_);
+        if (equipable != null) {
+            EquipmentSlot equipmentslot = equipable.getEquipmentSlot();
+            if (!p_338547_.contains(equipmentslot)) {
+                return equipmentslot;
+            }
+        } else if (!p_338547_.contains(EquipmentSlot.MAINHAND)) {
+            return EquipmentSlot.MAINHAND;
+        }
+
+        return null;
+    }
+}
