--- a/net/minecraft/world/entity/ExperienceOrb.java
+++ b/net/minecraft/world/entity/ExperienceOrb.java
@@ -136,7 +_,8 @@
             this.applyEffectsFromBlocks();
             float f = 0.98F;
             if (this.onGround()) {
-                f = this.level().getBlockState(this.getBlockPosBelowThatAffectsMyMovement()).getBlock().getFriction() * 0.98F;
+                BlockPos pos = this.getBlockPosBelowThatAffectsMyMovement();
+                f = this.level().getBlockState(pos).getFriction(this.level(), pos, this) * 0.98F;
             }
 
             this.setDeltaMovement(this.getDeltaMovement().scale(f));
@@ -153,7 +_,7 @@
 
     private void followNearbyPlayer() {
         if (this.followingPlayer == null || this.followingPlayer.isSpectator() || this.followingPlayer.distanceToSqr(this) > 64.0) {
-            Player player = this.level().getNearestPlayer(this, 8.0);
+            Player player = net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.entity.XpOrbTargetingEvent(this, 8.0)).getFollowingPlayer();
             if (player != null && !player.isSpectator() && !player.isDeadOrDying()) {
                 this.followingPlayer = player;
             } else {
@@ -278,6 +_,7 @@
     public void playerTouch(Player player) {
         if (player instanceof ServerPlayer serverplayer) {
             if (player.takeXpDelay == 0) {
+                if (net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.entity.player.PlayerXpEvent.PickupXp(player, this)).isCanceled()) return;
                 player.takeXpDelay = 2;
                 player.take(this, 1);
                 int i = this.repairPlayerItems(serverplayer, this.getValue());
@@ -297,7 +_,7 @@
         Optional<EnchantedItemInUse> optional = EnchantmentHelper.getRandomItemWith(EnchantmentEffectComponents.REPAIR_WITH_XP, player, ItemStack::isDamaged);
         if (optional.isPresent()) {
             ItemStack itemstack = optional.get().itemStack();
-            int i = EnchantmentHelper.modifyDurabilityToRepairFromXp(player.level(), itemstack, amount);
+            int i = EnchantmentHelper.modifyDurabilityToRepairFromXp(player.level(), itemstack, (int) (amount * itemstack.getXpRepairRatio()));
             int j = Math.min(i, itemstack.getDamageValue());
             itemstack.setDamageValue(itemstack.getDamageValue() - j);
             if (j > 0) {
