--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -195,13 +_,19 @@
     public static final float EXTRA_RENDER_CULLING_SIZE_WITH_BIG_HAT = 0.5F;
     public static final float DEFAULT_BABY_SCALE = 0.5F;
     private static final float WATER_FLOAT_IMPULSE = 0.04F;
-    public static final Predicate<LivingEntity> PLAYER_NOT_WEARING_DISGUISE_ITEM = livingEntity -> {
+    // Neo: Support IItemExtension#isGazeDisguise
+    public static final java.util.function.BiPredicate<LivingEntity, @org.jspecify.annotations.Nullable LivingEntity> PLAYER_NOT_WEARING_DISGUISE_ITEM_FOR_TARGET = (livingEntity, target) -> {
         if (livingEntity instanceof Player player) {
             ItemStack itemstack = player.getItemBySlot(EquipmentSlot.HEAD);
-            return !itemstack.is(ItemTags.GAZE_DISGUISE_EQUIPMENT);
+            return !itemstack.isGazeDisguise(player, target);
         } else {
             return true;
         }
+    };
+    /** @deprecated Neo: use {@link #PLAYER_NOT_WEARING_DISGUISE_ITEM_FOR_TARGET} with target info instead */
+    @Deprecated
+    public static final Predicate<LivingEntity> PLAYER_NOT_WEARING_DISGUISE_ITEM = livingEntity -> {
+        return PLAYER_NOT_WEARING_DISGUISE_ITEM_FOR_TARGET.test(livingEntity, null);
     };
     private static final Dynamic<?> EMPTY_BRAIN = new Dynamic<>(JavaOps.INSTANCE, Map.of("memories", Map.of()));
     private final AttributeMap attributes;
@@ -268,6 +_,13 @@
     );
     protected final EntityEquipment equipment;
     private Waypoint.Icon locatorBarIcon = new Waypoint.Icon();
+    /**
+     * This field stores information about damage dealt to this entity.
+     * a new {@link net.neoforged.neoforge.common.damagesource.DamageContainer} is instantiated
+     * via {@link #hurt(DamageSource, float)} after invulnerability checks, and is removed from
+     * the stack before the method's return.
+     **/
+    protected java.util.@Nullable Stack<net.neoforged.neoforge.common.damagesource.DamageContainer> damageContainers = new java.util.Stack<>();
 
     protected LivingEntity(EntityType<? extends LivingEntity> type, Level level) {
         super(type, level);
@@ -344,13 +_,15 @@
             .add(Attributes.MOVEMENT_EFFICIENCY)
             .add(Attributes.ATTACK_KNOCKBACK)
             .add(Attributes.CAMERA_DISTANCE)
-            .add(Attributes.WAYPOINT_TRANSMIT_RANGE);
+            .add(Attributes.WAYPOINT_TRANSMIT_RANGE)
+            .add(net.neoforged.neoforge.common.NeoForgeMod.SWIM_SPEED)
+            .add(net.neoforged.neoforge.common.NeoForgeMod.NAMETAG_DISTANCE);
     }
 
     @Override
     protected void checkFallDamage(double ya, boolean onGround, BlockState onState, BlockPos pos) {
         if (!this.isInWater()) {
-            this.updateInWaterStateAndDoWaterCurrentPushing();
+            this.updateInWaterStateAndDoWaterCurrentPushing(false);
         }
 
         if (this.level() instanceof ServerLevel serverlevel && onGround && this.fallDistance > 0.0) {
@@ -371,7 +_,8 @@
 
                 double d7 = Math.min(0.2F + d6 / 15.0, 2.5);
                 int i = (int)(150.0 * d7);
-                serverlevel.sendParticles(new BlockParticleOption(ParticleTypes.BLOCK, onState), d0, d1, d2, i, 0.0, 0.0, 0.0, 0.15F);
+                if (!onState.addLandingEffects(serverlevel, pos, onState, this, i))
+                serverlevel.sendParticles(new BlockParticleOption(ParticleTypes.BLOCK, onState, pos), d0, d1, d2, i, 0.0, 0.0, 0.0, 0.15F);
             }
         }
 
@@ -381,6 +_,7 @@
         }
     }
 
+    @Deprecated //FORGE: Use canDrownInFluidType instead
     public boolean canBreatheUnderwater() {
         return this.getType().is(EntityTypeTags.CAN_BREATHE_UNDER_WATER);
     }
@@ -421,6 +_,9 @@
                 }
             }
 
+            int airSupply = this.getAirSupply();
+            net.neoforged.neoforge.common.CommonHooks.onLivingBreathe(this, airSupply - decreaseAirSupply(airSupply), increaseAirSupply(airSupply) - airSupply);
+            if (false) // Forge: Handled in ForgeHooks#onLivingBreathe(LivingEntity, int, int)
             if (this.isEyeInFluid(FluidTags.WATER)
                 && !serverlevel1.getBlockState(BlockPos.containing(this.getX(), this.getEyeY(), this.getZ())).is(Blocks.BUBBLE_COLUMN)) {
                 boolean flag1 = !this.canBreatheUnderwater()
@@ -769,7 +_,11 @@
         } else {
             ItemEntity itementity = this.createItemStackToDrop(itemStack, randomly, thrownFromHand);
             if (itementity != null) {
-                this.level().addFreshEntity(itementity);
+                if (captureDrops() != null) {
+                    captureDrops().add(itementity);
+                } else {
+                    this.level().addFreshEntity(itementity);
+                }
             }
 
             return itementity;
@@ -835,8 +_,10 @@
                     Holder<MobEffect> holder = iterator.next();
                     MobEffectInstance mobeffectinstance = this.activeEffects.get(holder);
                     if (!mobeffectinstance.tickServer(serverlevel, this, () -> this.onEffectUpdated(mobeffectinstance, true, null))) {
+                        if (!net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.entity.living.MobEffectEvent.Expired(this, mobeffectinstance)).isCanceled()) {
                         iterator.remove();
                         this.onEffectsRemoved(List.of(mobeffectinstance));
+                        }
                     } else if (mobeffectinstance.getDuration() % 600 == 0) {
                         this.onEffectUpdated(mobeffectinstance, false, null);
                     }
@@ -882,8 +_,9 @@
         List<ParticleOptions> list = this.activeEffects
             .values()
             .stream()
-            .filter(MobEffectInstance::isVisible)
-            .map(MobEffectInstance::getParticleOptions)
+            .map(effect -> net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.entity.living.EffectParticleModificationEvent(this, effect)))
+            .filter(net.neoforged.neoforge.event.entity.living.EffectParticleModificationEvent::isVisible)
+            .map(net.neoforged.neoforge.event.entity.living.EffectParticleModificationEvent::getParticleOptions)
             .toList();
         this.entityData.set(DATA_EFFECT_PARTICLES, list);
         this.entityData.set(DATA_EFFECT_AMBIENCE_ID, areAllEffectsAmbient(this.activeEffects.values()));
@@ -923,6 +_,7 @@
             }
         }
 
+        d0 = net.neoforged.neoforge.common.CommonHooks.getEntityVisibilityMultiplier(this, targetingEntity, d0);
         return d0;
     }
 
@@ -958,8 +_,13 @@
         } else if (this.activeEffects.isEmpty()) {
             return false;
         } else {
-            Map<Holder<MobEffect>, MobEffectInstance> map = Maps.newHashMap(this.activeEffects);
-            this.activeEffects.clear();
+            Map<Holder<MobEffect>, MobEffectInstance> map = new java.util.HashMap<>(this.activeEffects.size());
+            for (Map.Entry<Holder<MobEffect>, MobEffectInstance> entry : this.activeEffects.entrySet()) {
+                if (!net.neoforged.neoforge.event.EventHooks.onEffectRemoved(this, entry.getValue())) {
+                    map.put(entry.getKey(), entry.getValue());
+                }
+            }
+            map.keySet().forEach(this.activeEffects::remove);
             this.onEffectsRemoved(map.values());
             return true;
         }
@@ -991,11 +_,12 @@
     }
 
     public boolean addEffect(MobEffectInstance newEffect, @Nullable Entity source) {
-        if (!this.canBeAffected(newEffect)) {
+        if (!net.neoforged.neoforge.common.CommonHooks.canMobEffectBeApplied(this, newEffect, source)) {
             return false;
         } else {
             MobEffectInstance mobeffectinstance = this.activeEffects.get(newEffect.getEffect());
             boolean flag = false;
+            net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.entity.living.MobEffectEvent.Added(this, mobeffectinstance, newEffect, source));
             if (mobeffectinstance == null) {
                 this.activeEffects.put(newEffect.getEffect(), newEffect);
                 this.onEffectAdded(newEffect, source);
@@ -1011,6 +_,14 @@
         }
     }
 
+    /**
+     * Neo: Override-Only. Call via {@link net.neoforged.neoforge.common.CommonHooks#canMobEffectBeApplied(LivingEntity, MobEffectInstance, Entity)}
+     *
+     * @param newEffect A mob effect instance
+     * @return If the mob effect instance can be applied to this entity
+     */
+    @Deprecated
+    @org.jetbrains.annotations.ApiStatus.OverrideOnly
     public boolean canBeAffected(MobEffectInstance newEffect) {
         if (this.getType().is(EntityTypeTags.IMMUNE_TO_INFESTED)) {
             return !newEffect.is(MobEffects.INFESTED);
@@ -1024,7 +_,7 @@
     }
 
     public void forceAddEffect(MobEffectInstance newEffect, @Nullable Entity source) {
-        if (this.canBeAffected(newEffect)) {
+        if (net.neoforged.neoforge.common.CommonHooks.canMobEffectBeApplied(this, newEffect, source)) {
             MobEffectInstance mobeffectinstance = this.activeEffects.put(newEffect.getEffect(), newEffect);
             if (mobeffectinstance == null) {
                 this.onEffectAdded(newEffect, source);
@@ -1044,6 +_,7 @@
     }
 
     public boolean removeEffect(Holder<MobEffect> effect) {
+        if (net.neoforged.neoforge.event.EventHooks.onEffectRemoved(this, effect)) return false;
         MobEffectInstance mobeffectinstance = this.removeEffectNoUpdate(effect);
         if (mobeffectinstance != null) {
             this.onEffectsRemoved(List.of(mobeffectinstance));
@@ -1135,6 +_,8 @@
     }
 
     public void heal(float heal) {
+        heal = net.neoforged.neoforge.event.EventHooks.onLivingHeal(this, heal);
+        if (heal <= 0) return;
         float f = this.getHealth();
         if (f > 0.0F) {
             this.setHealth(f + heal);
@@ -1162,11 +_,14 @@
         } else if (source.is(DamageTypeTags.IS_FIRE) && this.hasEffect(MobEffects.FIRE_RESISTANCE)) {
             return false;
         } else {
+            this.damageContainers.push(new net.neoforged.neoforge.common.damagesource.DamageContainer(source, damage));
+            if (net.neoforged.neoforge.common.CommonHooks.onEntityIncomingDamage(this, this.damageContainers.peek())) return false;
             if (this.isSleeping()) {
                 this.stopSleeping();
             }
 
             this.noActionTime = 0;
+            damage = this.damageContainers.peek().getNewDamage(); //Neo: enforce damage container as source of truth for damage amount
             if (damage < 0.0F) {
                 damage = 0.0F;
             }
@@ -1187,24 +_,28 @@
             if (Float.isNaN(damage) || Float.isInfinite(damage)) {
                 damage = Float.MAX_VALUE;
             }
+            this.damageContainers.peek().setNewDamage(damage); //update container with vanilla changes
 
             boolean flag1 = true;
             if (this.invulnerableTime > 10.0F && !source.is(DamageTypeTags.BYPASSES_COOLDOWN)) {
                 if (damage <= this.lastHurt) {
+                    this.damageContainers.pop();
                     return false;
                 }
 
+                this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.INVULNERABILITY, this.lastHurt);
                 this.actuallyHurt(level, source, damage - this.lastHurt);
                 this.lastHurt = damage;
                 flag1 = false;
             } else {
                 this.lastHurt = damage;
-                this.invulnerableTime = 20;
+                this.invulnerableTime = this.damageContainers.peek().getPostAttackInvulnerabilityTicks();
                 this.actuallyHurt(level, source, damage);
                 this.hurtDuration = 10;
                 this.hurtTime = this.hurtDuration;
             }
 
+            damage = this.damageContainers.peek().getNewDamage(); //update local with container value
             this.resolveMobResponsibleForDamage(source);
             this.resolvePlayerResponsibleForDamage(source);
             if (flag1) {
@@ -1273,6 +_,7 @@
                 CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(serverplayer1, this, source, damage, damage, flag);
             }
 
+            this.damageContainers.pop();
             return flag2;
         }
     }
@@ -1286,7 +_,7 @@
                 return 0.0F;
             } else {
                 BlocksAttacks blocksattacks = itemstack.get(DataComponents.BLOCKS_ATTACKS);
-                if (blocksattacks != null && !blocksattacks.bypassedBy().map(source::is).orElse(false)) {
+                if (blocksattacks != null) {
                     if (source.getDirectEntity() instanceof AbstractArrow abstractarrow && abstractarrow.getPierceLevel() > 0) {
                         return 0.0F;
                     } else {
@@ -1302,7 +_,14 @@
                         }
 
                         float f = blocksattacks.resolveBlockedDamage(source, damage, d0);
-                        blocksattacks.hurtBlockingItem(this.level(), itemstack, this, this.getUsedItemHand(), f);
+
+                        var ev = net.neoforged.neoforge.common.CommonHooks.onDamageBlock(this, this.damageContainers.peek(), f, !blocksattacks.bypassedBy().map(source::is).orElse(false));
+                        if (!ev.getBlocked()) return 0.0F;
+
+                        f = ev.getBlockedDamage();
+                        this.damageContainers.peek().setBlockedDamage(ev);
+
+                        blocksattacks.hurtBlockingItem(this.level(), itemstack, this, this.getUsedItemHand(), f, ev.shieldDamage());
                         if (f > 0.0F && !source.is(DamageTypeTags.IS_PROJECTILE) && source.getDirectEntity() instanceof LivingEntity livingentity) {
                             this.blockUsingItem(level, livingentity);
                         }
@@ -1335,9 +_,9 @@
         Entity entity = source.getEntity();
         if (entity instanceof Player player) {
             this.setLastHurtByPlayer(player, 100);
-        } else if (entity instanceof Wolf wolf && wolf.isTame()) {
-            if (wolf.getOwnerReference() != null) {
-                this.setLastHurtByPlayer(wolf.getOwnerReference().getUUID(), 100);
+        } else if (entity instanceof TamableAnimal tamableAnimal && tamableAnimal.isTame()) {
+            if (tamableAnimal.getOwnerReference() != null) {
+                this.setLastHurtByPlayer(tamableAnimal.getOwnerReference().getUUID(), 100);
             } else {
                 this.lastHurtByPlayer = null;
                 this.lastHurtByPlayerMemoryTime = 0;
@@ -1365,7 +_,7 @@
             for (InteractionHand interactionhand : InteractionHand.values()) {
                 ItemStack itemstack1 = this.getItemInHand(interactionhand);
                 deathprotection = itemstack1.get(DataComponents.DEATH_PROTECTION);
-                if (deathprotection != null) {
+                if (deathprotection != null && net.neoforged.neoforge.common.CommonHooks.onLivingUseTotem(this, killingDamage, itemstack1, interactionhand)) {
                     itemstack = itemstack1.copy();
                     itemstack1.shrink(1);
                     break;
@@ -1428,6 +_,7 @@
     }
 
     public void die(DamageSource source) {
+        if (net.neoforged.neoforge.common.CommonHooks.onLivingDeath(this, source)) return;
         if (!this.isRemoved() && !this.dead) {
             Entity entity = source.getEntity();
             LivingEntity livingentity = this.getKillCredit();
@@ -1464,7 +_,7 @@
         if (this.level() instanceof ServerLevel serverlevel) {
             boolean flag = false;
             if (killer instanceof WitherBoss) {
-                if (serverlevel.getGameRules().get(GameRules.MOB_GRIEFING)) {
+                if (net.neoforged.neoforge.event.EventHooks.canEntityGrief(serverlevel, killer)) {
                     BlockPos blockpos = this.blockPosition();
                     BlockState blockstate = Blocks.WITHER_ROSE.defaultBlockState();
                     if (this.level().getBlockState(blockpos).isAir() && blockstate.canSurvive(this.level(), blockpos)) {
@@ -1482,6 +_,7 @@
     }
 
     protected void dropAllDeathLoot(ServerLevel level, DamageSource source) {
+        this.captureDrops(new java.util.ArrayList<>());
         boolean flag = this.lastHurtByPlayerMemoryTime > 0;
         if (this.shouldDropLoot(level)) {
             this.dropFromLootTable(level, source, flag);
@@ -1490,6 +_,10 @@
 
         this.dropEquipment(level);
         this.dropExperience(level, source.getEntity());
+
+        Collection<ItemEntity> drops = captureDrops(null);
+        if (!net.neoforged.neoforge.common.CommonHooks.onLivingDrops(this, source, drops, lastHurtByPlayerMemoryTime > 0))
+            drops.forEach(e -> level().addFreshEntity(e));
     }
 
     protected void dropEquipment(ServerLevel level) {
@@ -1501,7 +_,8 @@
                 this.isAlwaysExperienceDropper()
                     || this.lastHurtByPlayerMemoryTime > 0 && this.shouldDropExperience() && level.getGameRules().get(GameRules.MOB_DROPS)
             )) {
-            ExperienceOrb.award(level, this.position(), this.getExperienceReward(level, killer));
+            int reward = net.neoforged.neoforge.event.EventHooks.getExperienceDrop(this, this.getLastHurtByPlayer(), this.getExperienceReward(level, killer));
+            ExperienceOrb.award((ServerLevel) this.level(), this.position(), reward);
         }
     }
 
@@ -1606,6 +_,11 @@
     }
 
     public void knockback(double power, double xd, double zd) {
+        net.neoforged.neoforge.event.entity.living.LivingKnockBackEvent event = net.neoforged.neoforge.common.CommonHooks.onLivingKnockBack(this, (float) power, xd, zd);
+        if(event.isCanceled()) return;
+        power = event.getStrength();
+        xd = event.getRatioX();
+        zd = event.getRatioZ();
         power *= 1.0 - this.getAttributeValue(Attributes.KNOCKBACK_RESISTANCE);
         if (!(power <= 0.0)) {
             this.needsSync = true;
@@ -1683,17 +_,11 @@
         } else {
             BlockPos blockpos = this.blockPosition();
             BlockState blockstate = this.getInBlockState();
-            if (this.isFallFlying() && blockstate.is(BlockTags.CAN_GLIDE_THROUGH)) {
-                return false;
-            } else if (blockstate.is(BlockTags.CLIMBABLE)) {
-                this.lastClimbablePos = Optional.of(blockpos);
-                return true;
-            } else if (blockstate.getBlock() instanceof TrapDoorBlock && this.trapdoorUsableAsLadder(blockpos, blockstate)) {
-                this.lastClimbablePos = Optional.of(blockpos);
-                return true;
-            } else {
-                return false;
-            }
+            if (this.isFallFlying() && blockstate.is(BlockTags.CAN_GLIDE_THROUGH))
+                return false;
+            Optional<BlockPos> ladderPos = net.neoforged.neoforge.common.CommonHooks.isLivingOnLadder(blockstate, level(), blockpos, this);
+            if (ladderPos.isPresent()) this.lastClimbablePos = ladderPos;
+            return ladderPos.isPresent();
         }
     }
 
@@ -1739,6 +_,11 @@
 
     @Override
     public boolean causeFallDamage(double fallDistance, float damageModifier, DamageSource damageSource) {
+        var event = net.neoforged.neoforge.common.CommonHooks.onLivingFall(this, fallDistance, damageModifier);
+        if (event.isCanceled()) return false;
+        fallDistance = event.getDistance();
+        damageModifier = event.getDamageMultiplier();
+
         boolean flag = super.causeFallDamage(fallDistance, damageModifier, damageSource);
         int i = this.calculateFallDamage(fallDistance, damageModifier);
         if (i > 0) {
@@ -1769,10 +_,10 @@
             int i = Mth.floor(this.getX());
             int j = Mth.floor(this.getY() - 0.2F);
             int k = Mth.floor(this.getZ());
-            BlockState blockstate = this.level().getBlockState(new BlockPos(i, j, k));
+            BlockPos pos = new BlockPos(i, j, k);
+            BlockState blockstate = this.level().getBlockState(pos);
             if (!blockstate.isAir()) {
-                SoundType soundtype = blockstate.getSoundType();
-                this.playSound(soundtype.getFallSound(), soundtype.getVolume() * 0.5F, soundtype.getPitch() * 0.75F);
+                blockstate.playFallSound(this.level(), pos, this);
             }
         }
     }
@@ -1797,6 +_,8 @@
         if (!(damage <= 0.0F)) {
             int i = (int)Math.max(1.0F, damage / 4.0F);
 
+            net.neoforged.neoforge.common.CommonHooks.onArmorHurt(damageSource, slots, i, this);
+            if (true) return; //Neo: Invalidates the loop. Armor damage happens in common hook.
             for (EquipmentSlot equipmentslot : slots) {
                 ItemStack itemstack = this.getItemBySlot(equipmentslot);
                 Equippable equippable = itemstack.get(DataComponents.EQUIPPABLE);
@@ -1830,6 +_,7 @@
                 damage = Math.max(f / 25.0F, 0.0F);
                 float f2 = f1 - damage;
                 if (f2 > 0.0F && f2 < 3.4028235E37F) {
+                    this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.MOB_EFFECTS, f2);
                     if (this instanceof ServerPlayer) {
                         ((ServerPlayer)this).awardStat(Stats.DAMAGE_RESISTED, Math.round(f2 * 10.0F));
                     } else if (damageSource.getEntity() instanceof ServerPlayer) {
@@ -1852,6 +_,7 @@
 
                 if (f3 > 0.0F) {
                     damage = CombatRules.getDamageAfterMagicAbsorb(damage, f3);
+                    this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ENCHANTMENTS,this.damageContainers.peek().getNewDamage() - damage);
                 }
 
                 return damage;
@@ -1861,11 +_,14 @@
 
     protected void actuallyHurt(ServerLevel level, DamageSource source, float dmg) {
         if (!this.isInvulnerableTo(level, source)) {
-            dmg = this.getDamageAfterArmorAbsorb(source, dmg);
-            dmg = this.getDamageAfterMagicAbsorb(source, dmg);
-            float f1 = Math.max(dmg - this.getAbsorptionAmount(), 0.0F);
-            this.setAbsorptionAmount(this.getAbsorptionAmount() - (dmg - f1));
-            float f = dmg - f1;
+            this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ARMOR, this.damageContainers.peek().getNewDamage() - this.getDamageAfterArmorAbsorb(source, this.damageContainers.peek().getNewDamage()));
+            this.getDamageAfterMagicAbsorb(source, this.damageContainers.peek().getNewDamage());
+            float damage = net.neoforged.neoforge.common.CommonHooks.onLivingDamagePre(this, this.damageContainers.peek());
+            this.damageContainers.peek().setReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ABSORPTION, Math.min(this.getAbsorptionAmount(), damage));
+            float absorbed = Math.min(damage, this.damageContainers.peek().getReduction(net.neoforged.neoforge.common.damagesource.DamageContainer.Reduction.ABSORPTION));
+            this.setAbsorptionAmount(Math.max(0, this.getAbsorptionAmount() - absorbed));
+            float f1 = this.damageContainers.peek().getNewDamage();
+            float f = absorbed;
             if (f > 0.0F && f < 3.4028235E37F && source.getEntity() instanceof ServerPlayer serverplayer) {
                 serverplayer.awardStat(Stats.DAMAGE_DEALT_ABSORBED, Math.round(f * 10.0F));
             }
@@ -1873,9 +_,10 @@
             if (f1 != 0.0F) {
                 this.getCombatTracker().recordDamage(source, f1);
                 this.setHealth(this.getHealth() - f1);
-                this.setAbsorptionAmount(this.getAbsorptionAmount() - f1);
                 this.gameEvent(GameEvent.ENTITY_DAMAGE);
+                this.onDamageTaken(this.damageContainers.peek());
             }
+            net.neoforged.neoforge.common.CommonHooks.onLivingDamagePost(this, this.damageContainers.peek());
         }
     }
 
@@ -1930,6 +_,8 @@
     }
 
     public void swing(InteractionHand hand, boolean sendToSwingingEntity) {
+        ItemStack stack = this.getItemInHand(hand);
+        if (!stack.isEmpty() && stack.onEntitySwing(this, hand)) return;
         if (!this.swinging || this.swingTime >= this.getCurrentSwingDuration() / 2 || this.swingTime < 0) {
             this.swingTime = -1;
             this.swinging = true;
@@ -2071,8 +_,10 @@
 
     private void swapHandItems() {
         ItemStack itemstack = this.getItemBySlot(EquipmentSlot.OFFHAND);
-        this.setItemSlot(EquipmentSlot.OFFHAND, this.getItemBySlot(EquipmentSlot.MAINHAND));
-        this.setItemSlot(EquipmentSlot.MAINHAND, itemstack);
+        var event = net.neoforged.neoforge.common.CommonHooks.onLivingSwapHandItems(this);
+        if (event.isCanceled()) return;
+        this.setItemSlot(EquipmentSlot.OFFHAND, event.getItemSwappedToOffHand());
+        this.setItemSlot(EquipmentSlot.MAINHAND, event.getItemSwappedToMainHand());
     }
 
     @Override
@@ -2180,7 +_,13 @@
     }
 
     public void setItemSlot(EquipmentSlot slot, ItemStack itemStack) {
-        this.onEquipItem(slot, this.equipment.set(slot, itemStack), itemStack);
+        setItemSlot(slot, itemStack, false);
+    }
+    /** Neo: Passing {@code true} for {@code insideTransaction} sets the item without side-effects (calling {@link #onEquipItem}).
+     * Callers are responsible for performing the deferred side-effects, when the active transaction commits. */
+    public void setItemSlot(EquipmentSlot slot, ItemStack itemStack, boolean insideTransaction) {
+        var oldStack = this.equipment.set(slot, itemStack);
+        if (!insideTransaction) this.onEquipItem(slot, oldStack, itemStack);
     }
 
     public float getArmorCoverPercentage() {
@@ -2285,15 +_,18 @@
             }
 
             this.needsSync = true;
+            net.neoforged.neoforge.common.CommonHooks.onLivingJump(this);
         }
     }
 
+    @Deprecated // FORGE: use sinkInFluid instead
     protected void goDownInWater() {
-        this.setDeltaMovement(this.getDeltaMovement().add(0.0, -0.04F, 0.0));
+        this.sinkInFluid(net.neoforged.neoforge.common.NeoForgeMod.WATER_TYPE.value());
     }
 
+    @Deprecated // FORGE: use jumpInFluid instead
     protected void jumpInLiquid(TagKey<Fluid> type) {
-        this.setDeltaMovement(this.getDeltaMovement().add(0.0, 0.04F, 0.0));
+        this.setDeltaMovement(this.getDeltaMovement().add(0.0D, (double)0.04F * this.getAttributeValue(net.neoforged.neoforge.common.NeoForgeMod.SWIM_SPEED), 0.0D));
     }
 
     protected float getWaterSlowDown() {
@@ -2325,7 +_,8 @@
     }
 
     protected boolean shouldTravelInFluid(FluidState fluidState) {
-        return (this.isInWater() || this.isInLava()) && this.isAffectedByFluids() && !this.canStandOnFluid(fluidState);
+        // Neo: Call (patched-in overload of) #travelInFluid for custom fluid types
+        return (this.isInWater() || this.isInLava() || this.isInFluidType(fluidState)) && this.isAffectedByFluids() && !this.canStandOnFluid(fluidState);
     }
 
     protected void travelFlying(Vec3 input, float speed) {
@@ -2350,7 +_,7 @@
 
     private void travelInAir(Vec3 input) {
         BlockPos blockpos = this.getBlockPosBelowThatAffectsMyMovement();
-        float f = this.onGround() ? this.level().getBlockState(blockpos).getBlock().getFriction() : 1.0F;
+        float f = this.onGround() ? this.level().getBlockState(blockpos).getFriction(this.level(), blockpos, this) : 1.0F;
         float f1 = f * 0.91F;
         Vec3 vec3 = this.handleRelativeFrictionAndCalculateMovement(input, f);
         double d0 = vec3.y;
@@ -2373,11 +_,20 @@
         }
     }
 
+    /**
+     * @deprecated Neo: use {@link #travelInFluid(Vec3, FluidState)} instead
+     */
+    @Deprecated
     private void travelInFluid(Vec3 input) {
+        travelInFluid(input, net.minecraft.world.level.material.Fluids.EMPTY.defaultFluidState());
+    }
+
+    private void travelInFluid(Vec3 input, FluidState fluidState) {
         boolean flag = this.getDeltaMovement().y <= 0.0;
         double d0 = this.getY();
         double d1 = this.getEffectiveGravity();
-        if (this.isInWater()) {
+        // Neo: patched in to use the fluid the eye is in because fluidState is always EMPTY fluid.
+        if (this.isInWater() || (this.isInFluidType(this.getEyeInFluidType()) && !this.moveInFluid(this.level().getFluidState(BlockPos.containing(this.getEyePosition())), input, d1))) {
             this.travelInWater(input, d1, flag, d0);
             this.floatInWaterWhileRidden();
         } else {
@@ -2402,6 +_,7 @@
             f = 0.96F;
         }
 
+        f1 *= (float)this.getAttributeValue(net.neoforged.neoforge.common.NeoForgeMod.SWIM_SPEED);
         this.moveRelative(f1, input);
         this.move(MoverType.SELF, this.getDeltaMovement());
         Vec3 vec3 = this.getDeltaMovement();
@@ -2573,7 +_,7 @@
             double d0 = Mth.clamp(delta.x, -0.15F, 0.15F);
             double d1 = Mth.clamp(delta.z, -0.15F, 0.15F);
             double d2 = Math.max(delta.y, -0.15F);
-            if (d2 < 0.0 && !this.getInBlockState().is(Blocks.SCAFFOLDING) && this.isSuppressingSlidingDownLadder() && this instanceof Player) {
+            if (d2 < 0.0D && !this.getInBlockState().isScaffolding(this) && this.isSuppressingSlidingDownLadder() && this instanceof Player) {
                 d2 = 0.0;
             }
 
@@ -2806,6 +_,7 @@
             ItemStack itemstack = this.lastEquipmentItems.get(equipmentslot);
             ItemStack itemstack1 = this.getItemBySlot(equipmentslot);
             if (this.equipmentHasChanged(itemstack, itemstack1)) {
+                net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.entity.living.LivingEquipmentChangeEvent(this, equipmentslot, itemstack, itemstack1));
                 if (map == null) {
                     map = Maps.newEnumMap(EquipmentSlot.class);
                 }
@@ -2941,6 +_,9 @@
         profilerfiller.push("jump");
         if (this.jumping && this.isAffectedByFluids()) {
             double d3;
+            net.neoforged.neoforge.fluids.FluidType fluidType = this.getMaxHeightFluidType();
+            if (!fluidType.isAir()) d3 = this.getFluidTypeHeight(fluidType);
+            else
             if (this.isInLava()) {
                 d3 = this.getFluidHeight(FluidTags.LAVA);
             } else {
@@ -2951,15 +_,17 @@
             double d4 = this.getFluidJumpThreshold();
             if (!flag || this.onGround() && !(d3 > d4)) {
                 if (!this.isInLava() || this.onGround() && !(d3 > d4)) {
+                    if (fluidType.isAir() || this.onGround() && !(d3 > d4)) {
                     if ((this.onGround() || flag && d3 <= d4) && this.noJumpDelay == 0) {
                         this.jumpFromGround();
                         this.noJumpDelay = 10;
                     }
+                    } else this.jumpInFluid(fluidType);
                 } else {
-                    this.jumpInLiquid(FluidTags.LAVA);
+                    this.jumpInFluid(net.neoforged.neoforge.common.NeoForgeMod.LAVA_TYPE.value());
                 }
             } else {
-                this.jumpInLiquid(FluidTags.WATER);
+                this.jumpInFluid(net.neoforged.neoforge.common.NeoForgeMod.WATER_TYPE.value());
             }
         } else {
             this.noJumpDelay = 0;
@@ -3280,8 +_,11 @@
 
     private void updatingUsingItem() {
         if (this.isUsingItem()) {
-            if (ItemStack.isSameItem(this.getItemInHand(this.getUsedItemHand()), this.useItem)) {
-                this.useItem = this.getItemInHand(this.getUsedItemHand());
+            ItemStack itemStack = this.getItemInHand(this.getUsedItemHand());
+            if (net.neoforged.neoforge.common.CommonHooks.canContinueUsing(this.useItem, itemStack)) {
+                this.useItem = itemStack;
+            }
+            if (itemStack == this.useItem) {
                 this.updateUsingItem(this.useItem);
             } else {
                 this.stopUsingItem();
@@ -3324,8 +_,11 @@
     }
 
     protected void updateUsingItem(ItemStack useItem) {
+        if (!useItem.isEmpty())
+            this.useItemRemaining = net.neoforged.neoforge.event.EventHooks.onItemUseTick(this, useItem, this.getUseItemRemainingTicks());
+        if (this.getUseItemRemainingTicks() > 0)
         useItem.onUseTick(this.level(), this, this.getUseItemRemainingTicks());
-        if (--this.useItemRemaining == 0 && !this.level().isClientSide() && !useItem.useOnRelease()) {
+        if (--this.useItemRemaining <= 0 && !this.level().isClientSide() && !useItem.useOnRelease()) {
             this.completeUsingItem();
         }
     }
@@ -3353,8 +_,10 @@
     public void startUsingItem(InteractionHand hand) {
         ItemStack itemstack = this.getItemInHand(hand);
         if (!itemstack.isEmpty() && !this.isUsingItem()) {
+            int duration = net.neoforged.neoforge.event.EventHooks.onItemUseStart(this, itemstack, hand, itemstack.getUseDuration(this));
+            if (duration < 0) return; // Neo: Early return for negative values, as that indicates event cancellation.
             this.useItem = itemstack;
-            this.useItemRemaining = itemstack.getUseDuration(this);
+            this.useItemRemaining = duration;
             if (!this.level().isClientSide()) {
                 this.setLivingEntityFlag(1, true);
                 this.setLivingEntityFlag(2, hand == InteractionHand.OFF_HAND);
@@ -3420,7 +_,8 @@
                 this.releaseUsingItem();
             } else {
                 if (!this.useItem.isEmpty() && this.isUsingItem()) {
-                    ItemStack itemstack = this.useItem.finishUsingItem(this.level(), this);
+                    ItemStack copy = this.useItem.copy();
+                    ItemStack itemstack = net.neoforged.neoforge.event.EventHooks.onItemUseFinish(this, copy, getUseItemRemainingTicks(), this.useItem.finishUsingItem(this.level(), this));
                     if (itemstack != this.useItem) {
                         this.setItemInHand(interactionhand, itemstack);
                     }
@@ -3454,7 +_,11 @@
         ItemStack itemstack = this.getItemInHand(this.getUsedItemHand());
         if (!this.useItem.isEmpty() && ItemStack.isSameItem(itemstack, this.useItem)) {
             this.useItem = itemstack;
+            if (!net.neoforged.neoforge.event.EventHooks.onUseItemStop(this, useItem, this.getUseItemRemainingTicks())) {
+            ItemStack copy = this instanceof Player ? useItem.copy() : null;
             this.useItem.releaseUsing(this.level(), this, this.getUseItemRemainingTicks());
+            if (copy != null && useItem.isEmpty()) net.neoforged.neoforge.event.EventHooks.onPlayerDestroyItem((Player)this, copy, getUsedItemHand());
+            }
             if (this.useItem.useOnRelease()) {
                 this.updatingUsingItem();
             }
@@ -3464,6 +_,7 @@
     }
 
     public void stopUsingItem() {
+        if (this.isUsingItem() && !this.useItem.isEmpty()) this.useItem.onStopUsing(this, useItemRemaining);
         if (!this.level().isClientSide()) {
             boolean flag = this.isUsingItem();
             this.recentKineticEnemies = null;
@@ -3632,8 +_,8 @@
         }
 
         BlockState blockstate = this.level().getBlockState(bedPosition);
-        if (blockstate.getBlock() instanceof BedBlock) {
-            this.level().setBlock(bedPosition, blockstate.setValue(BedBlock.OCCUPIED, true), 3);
+        if (blockstate.isBed(level(), bedPosition, this)) {
+            blockstate.setBedOccupied(level(), bedPosition, this, true);
         }
 
         this.setPose(Pose.SLEEPING);
@@ -3648,15 +_,17 @@
     }
 
     private boolean checkBedExists() {
-        return this.getSleepingPos().map(bedPosition -> this.level().getBlockState(bedPosition).getBlock() instanceof BedBlock).orElse(false);
+        // Neo: Overwrite the vanilla instanceof BedBlock check with isBed and fire the CanContinueSleepingEvent.
+        boolean hasBed = this.getSleepingPos().map(pos -> this.level().getBlockState(pos).isBed(this.level(), pos, this)).orElse(false);
+        return net.neoforged.neoforge.event.EventHooks.canEntityContinueSleeping(this, hasBed ? null : Player.BedSleepingProblem.OTHER_PROBLEM);
     }
 
     public void stopSleeping() {
         this.getSleepingPos().filter(this.level()::hasChunkAt).ifPresent(bedPosition -> {
             BlockState blockstate = this.level().getBlockState(bedPosition);
-            if (blockstate.getBlock() instanceof BedBlock) {
+            if (blockstate.isBed(level(), bedPosition, this)) {
                 Direction direction = blockstate.getValue(BedBlock.FACING);
-                this.level().setBlock(bedPosition, blockstate.setValue(BedBlock.OCCUPIED, false), 3);
+                blockstate.setBedOccupied(level(), bedPosition, this, false);
                 Vec3 vec31 = BedBlock.findStandUpPosition(this.getType(), this.level(), bedPosition, direction, this.getYRot()).orElseGet(() -> {
                     BlockPos blockpos = bedPosition.above();
                     return new Vec3(blockpos.getX() + 0.5, blockpos.getY() + 0.1, blockpos.getZ() + 0.5);
@@ -3676,7 +_,9 @@
 
     public @Nullable Direction getBedOrientation() {
         BlockPos blockpos = this.getSleepingPos().orElse(null);
-        return blockpos != null ? BedBlock.getBedOrientation(this.level(), blockpos) : null;
+        if (blockpos == null) return null;
+        BlockState state = this.level().getBlockState(blockpos);
+        return !state.isBed(level(), blockpos, this) ? null : state.getBedDirection(level(), blockpos);
     }
 
     @Override
@@ -3685,7 +_,7 @@
     }
 
     public ItemStack getProjectile(ItemStack heldWeapon) {
-        return ItemStack.EMPTY;
+        return net.neoforged.neoforge.common.CommonHooks.getProjectile(this, heldWeapon, ItemStack.EMPTY);
     }
 
     private static byte entityEventForEquipmentBreak(EquipmentSlot equipmentSlot) {
@@ -3737,6 +_,8 @@
     }
 
     public final EquipmentSlot getEquipmentSlotForItem(ItemStack itemStack) {
+        final EquipmentSlot slot = itemStack.getEquipmentSlot();
+        if (slot != null) return slot; // FORGE: Allow modders to set a non-default equipment slot for a stack; e.g. a non-armor chestplate-slot item
         Equippable equippable = itemStack.get(DataComponents.EQUIPPABLE);
         return equippable != null && this.canUseSlot(equippable.slot()) ? equippable.slot() : EquipmentSlot.MAINHAND;
     }
