--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -79,10 +_,13 @@
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.pathfinder.PathType;
+import net.minecraft.world.level.storage.loot.LootParams;
 import net.minecraft.world.level.storage.loot.LootTable;
+import net.minecraft.world.level.storage.loot.parameters.LootContextParamSets;
+import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
 import net.minecraft.world.phys.AABB;
 
-public abstract class Mob extends LivingEntity implements Targeting {
+public abstract class Mob extends LivingEntity implements EquipmentUser, Targeting {
     private static final EntityDataAccessor<Byte> DATA_MOB_FLAGS_ID = SynchedEntityData.defineId(Mob.class, EntityDataSerializers.BYTE);
     private static final int MOB_FLAG_NO_AI = 1;
     private static final int MOB_FLAG_LEFTHANDED = 2;
@@ -129,6 +_,9 @@
     private Either<UUID, BlockPos> delayedLeashInfo;
     private BlockPos restrictCenter = BlockPos.ZERO;
     private float restrictRadius = -1.0F;
+    @Nullable
+    private MobSpawnType spawnType;
+    private boolean spawnCancelled = false;
 
     protected Mob(EntityType<? extends Mob> p_21368_, Level p_21369_) {
         super(p_21368_, p_21369_);
@@ -166,8 +_,7 @@
     public float getPathfindingMalus(PathType p_326934_) {
         Mob mob;
         label17: {
-            Entity entity = this.getControlledVehicle();
-            if (entity instanceof Mob mob1 && mob1.shouldPassengersInheritMalus()) {
+            if (this.getControlledVehicle() instanceof Mob mob1 && mob1.shouldPassengersInheritMalus()) {
                 mob = mob1;
                 break label17;
             }
@@ -198,8 +_,7 @@
     }
 
     public MoveControl getMoveControl() {
-        Entity entity = this.getControlledVehicle();
-        return entity instanceof Mob mob ? mob.getMoveControl() : this.moveControl;
+        return this.getControlledVehicle() instanceof Mob mob ? mob.getMoveControl() : this.moveControl;
     }
 
     public JumpControl getJumpControl() {
@@ -207,8 +_,7 @@
     }
 
     public PathNavigation getNavigation() {
-        Entity entity = this.getControlledVehicle();
-        return entity instanceof Mob mob ? mob.getNavigation() : this.navigation;
+        return this.getControlledVehicle() instanceof Mob mob ? mob.getNavigation() : this.navigation;
     }
 
     @Nullable
@@ -233,7 +_,10 @@
     }
 
     public void setTarget(@Nullable LivingEntity p_21544_) {
-        this.target = p_21544_;
+        net.neoforged.neoforge.event.entity.living.LivingChangeTargetEvent changeTargetEvent = net.neoforged.neoforge.common.CommonHooks.onLivingChangeTarget(this, p_21544_, net.neoforged.neoforge.event.entity.living.LivingChangeTargetEvent.LivingTargetType.MOB_TARGET);
+        if(!changeTargetEvent.isCanceled()) {
+             this.target = changeTargetEvent.getNewTarget();
+        }
     }
 
     @Override
@@ -290,13 +_,13 @@
         if (this.xpReward > 0) {
             int i = this.xpReward;
 
-            for(int j = 0; j < this.armorItems.size(); ++j) {
+            for (int j = 0; j < this.armorItems.size(); j++) {
                 if (!this.armorItems.get(j).isEmpty() && this.armorDropChances[j] <= 1.0F) {
                     i += 1 + this.random.nextInt(3);
                 }
             }
 
-            for(int k = 0; k < this.handItems.size(); ++k) {
+            for (int k = 0; k < this.handItems.size(); k++) {
                 if (!this.handItems.get(k).isEmpty() && this.handDropChances[k] <= 1.0F) {
                     i += 1 + this.random.nextInt(3);
                 }
@@ -314,7 +_,7 @@
 
     public void spawnAnim() {
         if (this.level().isClientSide) {
-            for(int i = 0; i < 20; ++i) {
+            for (int i = 0; i < 20; i++) {
                 double d0 = this.random.nextGaussian() * 0.02;
                 double d1 = this.random.nextGaussian() * 0.02;
                 double d2 = this.random.nextGaussian() * 0.02;
@@ -345,6 +_,12 @@
                 this.updateControlFlags();
             }
         }
+
+        // Neo: Animal armor tick patch
+        if (this.canWearBodyArmor()) {
+            ItemStack stack = this.getBodyArmorItem();
+            if (isBodyArmorItem(stack)) stack.onAnimalArmorTick(level(), this);
+        }
     }
 
     protected void updateControlFlags() {
@@ -373,7 +_,7 @@
         p_21484_.putBoolean("PersistenceRequired", this.persistenceRequired);
         ListTag listtag = new ListTag();
 
-        for(ItemStack itemstack : this.armorItems) {
+        for (ItemStack itemstack : this.armorItems) {
             if (!itemstack.isEmpty()) {
                 listtag.add(itemstack.save(this.registryAccess()));
             } else {
@@ -384,14 +_,14 @@
         p_21484_.put("ArmorItems", listtag);
         ListTag listtag1 = new ListTag();
 
-        for(float f : this.armorDropChances) {
+        for (float f : this.armorDropChances) {
             listtag1.add(FloatTag.valueOf(f));
         }
 
         p_21484_.put("ArmorDropChances", listtag1);
         ListTag listtag2 = new ListTag();
 
-        for(ItemStack itemstack1 : this.handItems) {
+        for (ItemStack itemstack1 : this.handItems) {
             if (!itemstack1.isEmpty()) {
                 listtag2.add(itemstack1.save(this.registryAccess()));
             } else {
@@ -402,7 +_,7 @@
         p_21484_.put("HandItems", listtag2);
         ListTag listtag3 = new ListTag();
 
-        for(float f1 : this.handDropChances) {
+        for (float f1 : this.handDropChances) {
             listtag3.add(FloatTag.valueOf(f1));
         }
 
@@ -415,11 +_,8 @@
         Either<UUID, BlockPos> either = this.delayedLeashInfo;
         if (this.leashHolder instanceof LivingEntity) {
             either = Either.left(this.leashHolder.getUUID());
-        } else {
-            Entity entity = this.leashHolder;
-            if (entity instanceof HangingEntity hangingentity) {
-                either = Either.right(hangingentity.getPos());
-            }
+        } else if (this.leashHolder instanceof HangingEntity hangingentity) {
+            either = Either.right(hangingentity.getPos());
         }
 
         if (either != null) {
@@ -441,6 +_,9 @@
         if (this.isNoAi()) {
             p_21484_.putBoolean("NoAI", this.isNoAi());
         }
+        if (this.spawnType != null) {
+            p_21484_.putString("neoforge:spawn_type", this.spawnType.name());
+        }
     }
 
     @Override
@@ -454,7 +_,7 @@
         if (p_21450_.contains("ArmorItems", 9)) {
             ListTag listtag = p_21450_.getList("ArmorItems", 10);
 
-            for(int i = 0; i < this.armorItems.size(); ++i) {
+            for (int i = 0; i < this.armorItems.size(); i++) {
                 CompoundTag compoundtag = listtag.getCompound(i);
                 this.armorItems.set(i, ItemStack.parseOptional(this.registryAccess(), compoundtag));
             }
@@ -463,7 +_,7 @@
         if (p_21450_.contains("ArmorDropChances", 9)) {
             ListTag listtag1 = p_21450_.getList("ArmorDropChances", 5);
 
-            for(int j = 0; j < listtag1.size(); ++j) {
+            for (int j = 0; j < listtag1.size(); j++) {
                 this.armorDropChances[j] = listtag1.getFloat(j);
             }
         }
@@ -471,7 +_,7 @@
         if (p_21450_.contains("HandItems", 9)) {
             ListTag listtag2 = p_21450_.getList("HandItems", 10);
 
-            for(int k = 0; k < this.handItems.size(); ++k) {
+            for (int k = 0; k < this.handItems.size(); k++) {
                 CompoundTag compoundtag1 = listtag2.getCompound(k);
                 this.handItems.set(k, ItemStack.parseOptional(this.registryAccess(), compoundtag1));
             }
@@ -480,7 +_,7 @@
         if (p_21450_.contains("HandDropChances", 9)) {
             ListTag listtag3 = p_21450_.getList("HandDropChances", 5);
 
-            for(int l = 0; l < listtag3.size(); ++l) {
+            for (int l = 0; l < listtag3.size(); l++) {
                 this.handDropChances[l] = listtag3.getFloat(l);
             }
         }
@@ -495,7 +_,7 @@
         if (p_21450_.contains("leash", 10)) {
             this.delayedLeashInfo = Either.left(p_21450_.getCompound("leash").getUUID("UUID"));
         } else if (p_21450_.contains("leash", 11)) {
-            this.delayedLeashInfo = (Either)NbtUtils.readBlockPos(p_21450_, "leash").map(Either::right).orElse(null);
+            this.delayedLeashInfo = NbtUtils.readBlockPos(p_21450_, "leash").<Either<UUID, BlockPos>>map(Either::right).orElse(null);
         } else {
             this.delayedLeashInfo = null;
         }
@@ -507,6 +_,14 @@
         }
 
         this.setNoAi(p_21450_.getBoolean("NoAI"));
+
+        if (p_21450_.contains("neoforge:spawn_type")) {
+            try {
+                this.spawnType = MobSpawnType.valueOf(p_21450_.getString("neoforge:spawn_type"));
+            } catch (Exception ex) {
+                p_21450_.remove("neoforge:spawn_type");
+            }
+        }
     }
 
     @Override
@@ -562,10 +_,10 @@
             && this.canPickUpLoot()
             && this.isAlive()
             && !this.dead
-            && this.level().getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING)) {
+            && net.neoforged.neoforge.event.EventHooks.getMobGriefingEvent(this.level(), this)) {
             Vec3i vec3i = this.getPickupReach();
 
-            for(ItemEntity itementity : this.level()
+            for (ItemEntity itementity : this.level()
                 .getEntitiesOfClass(ItemEntity.class, this.getBoundingBox().inflate((double)vec3i.getX(), (double)vec3i.getY(), (double)vec3i.getZ()))) {
                 if (!itementity.isRemoved() && !itementity.getItem().isEmpty() && !itementity.hasPickUpDelay() && this.wantsToPickUp(itementity.getItem())) {
                     this.pickUpItem(itementity);
@@ -629,7 +_,7 @@
     }
 
     public void setGuaranteedDrop(EquipmentSlot p_21509_) {
-        switch(p_21509_.getType()) {
+        switch (p_21509_.getType()) {
             case HAND:
                 this.handDropChances[p_21509_.getIndex()] = 2.0F;
                 break;
@@ -650,52 +_,45 @@
             } else {
                 double d2 = this.getApproximateAttackDamageWithItem(p_21428_);
                 double d3 = this.getApproximateAttackDamageWithItem(p_21429_);
-                if (d2 != d3) {
-                    return d2 > d3;
-                } else {
-                    return this.canReplaceEqualItem(p_21428_, p_21429_);
-                }
+                return d2 != d3 ? d2 > d3 : this.canReplaceEqualItem(p_21428_, p_21429_);
             }
         } else if (p_21428_.getItem() instanceof BowItem && p_21429_.getItem() instanceof BowItem) {
             return this.canReplaceEqualItem(p_21428_, p_21429_);
         } else if (p_21428_.getItem() instanceof CrossbowItem && p_21429_.getItem() instanceof CrossbowItem) {
             return this.canReplaceEqualItem(p_21428_, p_21429_);
-        } else {
-            Item $$6 = p_21428_.getItem();
-            if ($$6 instanceof ArmorItem armoritem) {
-                if (EnchantmentHelper.hasBindingCurse(p_21429_)) {
-                    return false;
-                } else if (!(p_21429_.getItem() instanceof ArmorItem)) {
-                    return true;
-                } else {
-                    ArmorItem armoritem1 = (ArmorItem)p_21429_.getItem();
-                    if (armoritem.getDefense() != armoritem1.getDefense()) {
-                        return armoritem.getDefense() > armoritem1.getDefense();
-                    } else if (armoritem.getToughness() != armoritem1.getToughness()) {
-                        return armoritem.getToughness() > armoritem1.getToughness();
-                    } else {
-                        return this.canReplaceEqualItem(p_21428_, p_21429_);
-                    }
-                }
-            } else {
-                if (p_21428_.getItem() instanceof DiggerItem) {
-                    if (p_21429_.getItem() instanceof BlockItem) {
-                        return true;
-                    }
-
-                    if (p_21429_.getItem() instanceof DiggerItem) {
-                        double d1 = this.getApproximateAttackDamageWithItem(p_21428_);
-                        double d0 = this.getApproximateAttackDamageWithItem(p_21429_);
-                        if (d1 != d0) {
-                            return d1 > d0;
-                        }
-
-                        return this.canReplaceEqualItem(p_21428_, p_21429_);
-                    }
-                }
-
+        } else if (p_21428_.getItem() instanceof ArmorItem armoritem) {
+            if (EnchantmentHelper.hasBindingCurse(p_21429_)) {
                 return false;
-            }
+            } else if (!(p_21429_.getItem() instanceof ArmorItem)) {
+                return true;
+            } else {
+                ArmorItem armoritem1 = (ArmorItem)p_21429_.getItem();
+                if (armoritem.getDefense() != armoritem1.getDefense()) {
+                    return armoritem.getDefense() > armoritem1.getDefense();
+                } else {
+                    return armoritem.getToughness() != armoritem1.getToughness()
+                        ? armoritem.getToughness() > armoritem1.getToughness()
+                        : this.canReplaceEqualItem(p_21428_, p_21429_);
+                }
+            }
+        } else {
+            if (p_21428_.getItem() instanceof DiggerItem) {
+                if (p_21429_.getItem() instanceof BlockItem) {
+                    return true;
+                }
+
+                if (p_21429_.getItem() instanceof DiggerItem) {
+                    double d1 = this.getApproximateAttackDamageWithItem(p_21428_);
+                    double d0 = this.getApproximateAttackDamageWithItem(p_21429_);
+                    if (d1 != d0) {
+                        return d1 > d0;
+                    }
+
+                    return this.canReplaceEqualItem(p_21428_, p_21429_);
+                }
+            }
+
+            return false;
         }
     }
 
@@ -705,11 +_,7 @@
     }
 
     public boolean canReplaceEqualItem(ItemStack p_21478_, ItemStack p_21479_) {
-        if (p_21478_.getDamageValue() < p_21479_.getDamageValue()) {
-            return true;
-        } else {
-            return hasAnyComponentExceptDamage(p_21478_) && !hasAnyComponentExceptDamage(p_21479_);
-        }
+        return p_21478_.getDamageValue() < p_21479_.getDamageValue() ? true : hasAnyComponentExceptDamage(p_21478_) && !hasAnyComponentExceptDamage(p_21479_);
     }
 
     private static boolean hasAnyComponentExceptDamage(ItemStack p_332136_) {
@@ -744,6 +_,14 @@
             this.discard();
         } else if (!this.isPersistenceRequired() && !this.requiresCustomPersistence()) {
             Entity entity = this.level().getNearestPlayer(this, -1.0);
+            net.neoforged.bus.api.Event.Result result = net.neoforged.neoforge.event.EventHooks.canEntityDespawn(this, (ServerLevel) this.level());
+            if (result == net.neoforged.bus.api.Event.Result.DENY) {
+                noActionTime = 0;
+                entity = null;
+            } else if (result == net.neoforged.bus.api.Event.Result.ALLOW) {
+                this.discard();
+                entity = null;
+            }
             if (entity != null) {
                 double d0 = entity.distanceToSqr(this);
                 int i = this.getType().getCategory().getDespawnDistance();
@@ -767,7 +_,7 @@
 
     @Override
     protected final void serverAiStep() {
-        ++this.noActionTime;
+        this.noActionTime++;
         ProfilerFiller profilerfiller = this.level().getProfiler();
         profilerfiller.push("sensing");
         this.sensing.tick();
@@ -945,17 +_,18 @@
 
     @Override
     public ItemStack getItemBySlot(EquipmentSlot p_21467_) {
-        return switch(p_21467_.getType()) {
+        return switch (p_21467_.getType()) {
             case HAND -> (ItemStack)this.handItems.get(p_21467_.getIndex());
             case ARMOR -> (ItemStack)this.armorItems.get(p_21467_.getIndex());
             case BODY -> this.bodyArmorItem;
+            default -> throw new MatchException(null, null);
         };
     }
 
     @Override
     public void setItemSlot(EquipmentSlot p_21416_, ItemStack p_21417_) {
         this.verifyEquippedItem(p_21417_);
-        switch(p_21416_.getType()) {
+        switch (p_21416_.getType()) {
             case HAND:
                 this.onEquipItem(p_21416_, this.handItems.set(p_21416_.getIndex(), p_21417_), p_21417_);
                 break;
@@ -973,7 +_,7 @@
     protected void dropCustomDeathLoot(DamageSource p_21385_, int p_21386_, boolean p_21387_) {
         super.dropCustomDeathLoot(p_21385_, p_21386_, p_21387_);
 
-        for(EquipmentSlot equipmentslot : EquipmentSlot.values()) {
+        for (EquipmentSlot equipmentslot : EquipmentSlot.values()) {
             ItemStack itemstack = this.getItemBySlot(equipmentslot);
             float f = this.getEquipmentDropChance(equipmentslot);
             boolean flag = f > 1.0F;
@@ -992,11 +_,25 @@
     }
 
     protected float getEquipmentDropChance(EquipmentSlot p_21520_) {
-        return switch(p_21520_.getType()) {
+        return switch (p_21520_.getType()) {
             case HAND -> this.handDropChances[p_21520_.getIndex()];
             case ARMOR -> this.armorDropChances[p_21520_.getIndex()];
             case BODY -> this.bodyArmorDropChance;
+            default -> throw new MatchException(null, null);
         };
+    }
+
+    private LootParams createEquipmentParams(ServerLevel p_338758_) {
+        return new LootParams.Builder(p_338758_)
+            .withParameter(LootContextParams.ORIGIN, this.position())
+            .withParameter(LootContextParams.THIS_ENTITY, this)
+            .create(LootContextParamSets.EQUIPMENT);
+    }
+
+    public void equip(ResourceLocation p_338889_) {
+        if (this.level() instanceof ServerLevel serverlevel) {
+            this.equip(p_338889_, this.createEquipmentParams(serverlevel));
+        }
     }
 
     protected void populateDefaultEquipmentSlots(RandomSource p_217055_, DifficultyInstance p_217056_) {
@@ -1004,20 +_,20 @@
             int i = p_217055_.nextInt(2);
             float f = this.level().getDifficulty() == Difficulty.HARD ? 0.1F : 0.25F;
             if (p_217055_.nextFloat() < 0.095F) {
-                ++i;
-            }
-
-            if (p_217055_.nextFloat() < 0.095F) {
-                ++i;
-            }
-
-            if (p_217055_.nextFloat() < 0.095F) {
-                ++i;
+                i++;
+            }
+
+            if (p_217055_.nextFloat() < 0.095F) {
+                i++;
+            }
+
+            if (p_217055_.nextFloat() < 0.095F) {
+                i++;
             }
 
             boolean flag = true;
 
-            for(EquipmentSlot equipmentslot : EquipmentSlot.values()) {
+            for (EquipmentSlot equipmentslot : EquipmentSlot.values()) {
                 if (equipmentslot.getType() == EquipmentSlot.Type.ARMOR) {
                     ItemStack itemstack = this.getItemBySlot(equipmentslot);
                     if (!flag && p_217055_.nextFloat() < f) {
@@ -1038,7 +_,7 @@
 
     @Nullable
     public static Item getEquipmentForSlot(EquipmentSlot p_21413_, int p_21414_) {
-        switch(p_21413_) {
+        switch (p_21413_) {
             case HEAD:
                 if (p_21414_ == 0) {
                     return Items.LEATHER_HELMET;
@@ -1096,7 +_,7 @@
         float f = p_217064_.getSpecialMultiplier();
         this.enchantSpawnedWeapon(p_217063_, f);
 
-        for(EquipmentSlot equipmentslot : EquipmentSlot.values()) {
+        for (EquipmentSlot equipmentslot : EquipmentSlot.values()) {
             if (equipmentslot.getType() == EquipmentSlot.Type.ARMOR) {
                 this.enchantSpawnedArmor(p_217063_, f, equipmentslot);
             }
@@ -1107,7 +_,9 @@
         if (!this.getMainHandItem().isEmpty() && p_217049_.nextFloat() < 0.25F * p_217050_) {
             this.setItemSlot(
                 EquipmentSlot.MAINHAND,
-                EnchantmentHelper.enchantItem(p_217049_, this.getMainHandItem(), (int)(5.0F + p_217050_ * (float)p_217049_.nextInt(18)), false)
+                EnchantmentHelper.enchantItem(
+                    this.level().enabledFeatures(), p_217049_, this.getMainHandItem(), (int)(5.0F + p_217050_ * (float)p_217049_.nextInt(18)), false
+                )
             );
         }
     }
@@ -1115,10 +_,23 @@
     protected void enchantSpawnedArmor(RandomSource p_217052_, float p_217053_, EquipmentSlot p_217054_) {
         ItemStack itemstack = this.getItemBySlot(p_217054_);
         if (!itemstack.isEmpty() && p_217052_.nextFloat() < 0.5F * p_217053_) {
-            this.setItemSlot(p_217054_, EnchantmentHelper.enchantItem(p_217052_, itemstack, (int)(5.0F + p_217053_ * (float)p_217052_.nextInt(18)), false));
+            this.setItemSlot(
+                p_217054_,
+                EnchantmentHelper.enchantItem(
+                    this.level().enabledFeatures(), p_217052_, itemstack, (int)(5.0F + p_217053_ * (float)p_217052_.nextInt(18)), false
+                )
+            );
         }
     }
 
+    /**
+     * Forge: Override-Only, call via EventHooks.onFinalizeSpawn.<br>
+     * Overrides are allowed. Do not wrap super calls within override (as that will cause stack overflows).<br>
+     * Vanilla calls are replaced with a transformer, and are not visible in source.<br>
+     * <p>
+     */
+    @Deprecated
+    @org.jetbrains.annotations.ApiStatus.OverrideOnly
     @Nullable
     public SpawnGroupData finalizeSpawn(ServerLevelAccessor p_21434_, DifficultyInstance p_21435_, MobSpawnType p_21436_, @Nullable SpawnGroupData p_21437_) {
         RandomSource randomsource = p_21434_.getRandom();
@@ -1127,6 +_,7 @@
                 new AttributeModifier("Random spawn bonus", randomsource.triangle(0.0, 0.11485000000000001), AttributeModifier.Operation.ADD_MULTIPLIED_BASE)
             );
         this.setLeftHanded(randomsource.nextFloat() < 0.05F);
+        this.spawnType = p_21436_;
         return p_21437_;
     }
 
@@ -1134,8 +_,9 @@
         this.persistenceRequired = true;
     }
 
+    @Override
     public void setDropChance(EquipmentSlot p_21410_, float p_21411_) {
-        switch(p_21410_.getType()) {
+        switch (p_21410_.getType()) {
             case HAND:
                 this.handDropChances[p_21410_.getIndex()] = p_21411_;
                 break;
@@ -1233,11 +_,7 @@
     }
 
     public boolean isWithinRestriction(BlockPos p_21445_) {
-        if (this.restrictRadius == -1.0F) {
-            return true;
-        } else {
-            return this.restrictCenter.distSqr(p_21445_) < (double)(this.restrictRadius * this.restrictRadius);
-        }
+        return this.restrictRadius == -1.0F ? true : this.restrictCenter.distSqr(p_21445_) < (double)(this.restrictRadius * this.restrictRadius);
     }
 
     public void restrictTo(BlockPos p_21447_, int p_21448_) {
@@ -1266,7 +_,7 @@
         if (this.isRemoved()) {
             return null;
         } else {
-            T t = p_21407_.create(this.level());
+            T t = (T)p_21407_.create(this.level());
             if (t == null) {
                 return null;
             } else {
@@ -1286,7 +_,7 @@
                 if (p_21408_) {
                     t.setCanPickUpLoot(this.canPickUpLoot());
 
-                    for(EquipmentSlot equipmentslot : EquipmentSlot.values()) {
+                    for (EquipmentSlot equipmentslot : EquipmentSlot.values()) {
                         ItemStack itemstack = this.getItemBySlot(equipmentslot);
                         if (!itemstack.isEmpty()) {
                             t.setItemSlot(equipmentslot, itemstack.copyAndClear());
@@ -1384,26 +_,23 @@
     }
 
     private void restoreLeashFromSave() {
-        if (this.delayedLeashInfo != null) {
-            Level $$1 = this.level();
-            if ($$1 instanceof ServerLevel serverlevel) {
-                Optional<UUID> optional1 = this.delayedLeashInfo.left();
-                Optional<BlockPos> optional = this.delayedLeashInfo.right();
-                if (optional1.isPresent()) {
-                    Entity entity = serverlevel.getEntity(optional1.get());
-                    if (entity != null) {
-                        this.setLeashedTo(entity, true);
-                        return;
-                    }
-                } else if (optional.isPresent()) {
-                    this.setLeashedTo(LeashFenceKnotEntity.getOrCreateKnot(this.level(), optional.get()), true);
+        if (this.delayedLeashInfo != null && this.level() instanceof ServerLevel serverlevel) {
+            Optional<UUID> optional1 = this.delayedLeashInfo.left();
+            Optional<BlockPos> optional = this.delayedLeashInfo.right();
+            if (optional1.isPresent()) {
+                Entity entity = serverlevel.getEntity(optional1.get());
+                if (entity != null) {
+                    this.setLeashedTo(entity, true);
                     return;
                 }
+            } else if (optional.isPresent()) {
+                this.setLeashedTo(LeashFenceKnotEntity.getOrCreateKnot(this.level(), optional.get()), true);
+                return;
+            }
 
-                if (this.tickCount > 100) {
-                    this.spawnAtLocation(Items.LEAD);
-                    this.delayedLeashInfo = null;
-                }
+            if (this.tickCount > 100) {
+                this.spawnAtLocation(Items.LEAD);
+                this.delayedLeashInfo = null;
             }
         }
     }
@@ -1520,14 +_,24 @@
     }
 
     @Override
+    @Deprecated // FORGE: use jumpInFluid instead
     protected void jumpInLiquid(TagKey<Fluid> p_204045_) {
+        this.jumpInLiquidInternal(() -> super.jumpInLiquid(p_204045_));
+    }
+
+    private void jumpInLiquidInternal(Runnable onSuper) {
         if (this.getNavigation().canFloat()) {
-            super.jumpInLiquid(p_204045_);
+            onSuper.run();
         } else {
             this.setDeltaMovement(this.getDeltaMovement().add(0.0, 0.3, 0.0));
         }
     }
 
+    @Override
+    public void jumpInFluid(net.neoforged.neoforge.fluids.FluidType type) {
+        this.jumpInLiquidInternal(() -> super.jumpInFluid(type));
+    }
+
     @VisibleForTesting
     public void removeFreeWill() {
         this.removeAllGoals(p_262562_ -> true);
@@ -1554,5 +_,40 @@
     public ItemStack getPickResult() {
         SpawnEggItem spawneggitem = SpawnEggItem.byId(this.getType());
         return spawneggitem == null ? null : new ItemStack(spawneggitem);
+    }
+
+    /**
+    * Returns the type of spawn that created this mob, if applicable.
+    * If it could not be determined, this will return null.
+    * <p>
+    * This is set via {@link Mob#finalizeSpawn}, so you should not call this from within that method, instead using the parameter.
+    */
+    @Nullable
+    public final MobSpawnType getSpawnType() {
+        return this.spawnType;
+    }
+
+    /**
+     * This method exists so that spawns can be cancelled from the {@link net.neoforged.neoforge.event.entity.living.MobSpawnEvent.FinalizeSpawn FinalizeSpawnEvent}
+     * without needing to hook up an additional handler for the {@link net.neoforged.neoforge.event.entity.EntityJoinLevelEvent EntityJoinLevelEvent}.
+     * @return if this mob will be blocked from spawning during {@link Level#addFreshEntity(Entity)}
+     * @apiNote Not public-facing API.
+     */
+    @org.jetbrains.annotations.ApiStatus.Internal
+    public final boolean isSpawnCancelled() {
+        return this.spawnCancelled;
+    }
+
+    /**
+     * Marks this mob as being disallowed to spawn during {@link Level#addFreshEntity(Entity)}.<p>
+     * @throws UnsupportedOperationException if this entity has already been {@link Entity#isAddedToWorld() added to the world}.
+     * @apiNote Not public-facing API.
+     */
+    @org.jetbrains.annotations.ApiStatus.Internal
+    public final void setSpawnCancelled(boolean cancel) {
+        if (this.isAddedToWorld()) {
+            throw new UnsupportedOperationException("Late invocations of Mob#setSpawnCancelled are not permitted.");
+        }
+        this.spawnCancelled = cancel;
     }
 }
