--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -128,7 +_,9 @@
     protected PathNavigation navigation;
     public final GoalSelector goalSelector;
     public final GoalSelector targetSelector;
-    private @Nullable Entity target;
+    public static boolean HOSTILE_MOBS_ATTACK_LIVINGBLOCKS = true;
+    public static boolean LIVINGBLOCK_ITEMS_PICKABLE = false;
+    private @Nullable LivingEntity target;
     private final Sensing sensing;
     private DropChances dropChances = DropChances.DEFAULT;
     private boolean canPickUpLoot = false;
@@ -139,6 +_,11 @@
     private Leashable.@Nullable LeashData leashData;
     private BlockPos homePosition = BlockPos.ZERO;
     private int homeRadius = -1;
+    @Nullable
+    private EntitySpawnReason spawnType;
+    private boolean spawnCancelled = false;
+    /** Neo: Prevent immediate spawning from conversions to capture conversion results for events */
+    protected boolean preventConversionSpawns = false;
 
     protected Mob(EntityType<? extends Mob> type, Level level) {
         super(type, level);
@@ -223,15 +_,16 @@
     }
 
     @Override
-    public @Nullable Entity getTarget() {
-        return this.asValidTarget(this.target);
-    }
-
-    public @Nullable Entity getTargetUnchecked() {
-        return this.target;
-    }
-
-    protected @Nullable Entity asValidTarget(@Nullable Entity target) {
+    public @Nullable LivingEntity getTarget() {
+        Entity valid = this.asValidTarget(this.target);
+        return valid instanceof LivingEntity livingTarget ? livingTarget : null;
+    }
+
+    public @Nullable LivingEntity getTargetUnchecked() {
+        return this.target instanceof LivingEntity livingTarget ? livingTarget : null;
+    }
+
+    protected @Nullable LivingEntity asValidTarget(@Nullable LivingEntity target) {
         if (target instanceof Player player && (player.isCreative() || player.isSpectator())) {
             return null;
         } else {
@@ -239,12 +_,25 @@
         }
     }
 
-    protected final @Nullable Entity getTargetFromBrain() {
-        return this.asValidTarget(this.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null));
-    }
-
-    public void setTarget(@Nullable Entity target) {
-        this.target = this.asValidTarget(target);
+    protected final @Nullable LivingEntity getTargetFromBrain() {
+        Entity valid = this.asValidTarget(this.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null));
+        return valid instanceof LivingEntity livingTarget ? livingTarget : null;
+    }
+
+    public void setTarget(@Nullable LivingEntity target) {
+        target = this.asValidTarget(target);
+        if (target != null) {
+            net.neoforged.neoforge.event.entity.living.LivingChangeTargetEvent changeTargetEvent = net.neoforged.neoforge.common.CommonHooks.onLivingChangeTarget(this, target, net.neoforged.neoforge.event.entity.living.LivingChangeTargetEvent.LivingTargetType.MOB_TARGET);
+            if (!changeTargetEvent.isCanceled()) {
+                this.target = changeTargetEvent.getNewAboutToBeSetTarget();
+            }
+        } else {
+            this.target = target;
+        }
+    }
+
+    public void setAttackTarget(@Nullable LivingEntity target) {
+        this.target = target;
     }
 
     @Override
@@ -340,6 +_,12 @@
         if (!this.level().isClientSide() && this.tickCount % 5 == 0) {
             this.updateControlFlags();
         }
+
+        // Neo: Animal armor tick patch
+        if (this.canUseSlot(EquipmentSlot.BODY)) {
+            ItemStack stack = this.getBodyArmorItem();
+            if (stack.has(DataComponents.EQUIPPABLE)) stack.onAnimalArmorTick(level(), this);
+        }
     }
 
     protected void updateControlFlags() {
@@ -383,6 +_,9 @@
         if (this.isNoAi()) {
             output.putBoolean("NoAI", this.isNoAi());
         }
+        if (this.spawnType != null) {
+            output.putString("neoforge:spawn_type", this.spawnType.name());
+        }
     }
 
     @Override
@@ -401,6 +_,14 @@
         this.lootTable = input.read("DeathLootTable", LootTable.KEY_CODEC);
         this.lootTableSeed = input.getLongOr("DeathLootTableSeed", 0L);
         this.setNoAi(input.getBooleanOr("NoAI", false));
+
+        input.getString("neoforge:spawn_type")
+                .ifPresent(spawnType -> {
+                    try {
+                        this.spawnType = EntitySpawnReason.valueOf(spawnType);
+                    } catch (Exception ignored) {
+                    }
+                });
     }
 
     @Override
@@ -459,7 +_,7 @@
             && this.canPickUpLoot()
             && this.isAlive()
             && !this.dead
-            && serverLevel.getGameRules().get(GameRules.MOB_GRIEFING)) {
+            && net.neoforged.neoforge.event.EventHooks.canEntityGrief(serverLevel, this)) {
             Vec3i pickupReach = this.getPickupReach();
 
             for (LivingBlock entity : this.level()
@@ -617,6 +_,8 @@
     private double getApproximateAttributeWith(ItemStack itemStack, Holder<Attribute> attribute, EquipmentSlot slot) {
         double baseValue = this.getAttributes().hasAttribute(attribute) ? this.getAttributeBaseValue(attribute) : 0.0;
         ItemAttributeModifiers attributeModifiers = itemStack.getOrDefault(DataComponents.ATTRIBUTE_MODIFIERS, ItemAttributeModifiers.EMPTY);
+        // Neo: Respect gameplay modifiers
+        attributeModifiers = itemStack.getAttributeModifiers();
         return attributeModifiers.compute(attribute, baseValue, slot);
     }
 
@@ -656,6 +_,7 @@
 
     @Override
     public void checkDespawn() {
+        if (net.neoforged.neoforge.event.EventHooks.checkMobDespawn(this)) return;
         if (this.level().getDifficulty() == Difficulty.PEACEFUL && !this.getType().isAllowedInPeaceful()) {
             this.discard();
         } else if (!this.isPersistenceRequired() && !this.requiresCustomPersistence()) {
@@ -1063,6 +_,11 @@
         }
     }
 
+    /**
+     * @deprecated Override-Only. External callers should call via {@link net.neoforged.neoforge.event.EventHooks#finalizeMobSpawn}.
+     */
+    @Deprecated
+    @org.jetbrains.annotations.ApiStatus.OverrideOnly
     public @Nullable SpawnGroupData finalizeSpawn(
         ServerLevelAccessor level, DifficultyInstance difficulty, EntitySpawnReason spawnReason, @Nullable SpawnGroupData groupData
     ) {
@@ -1075,6 +_,7 @@
         }
 
         this.setLeftHanded(random.nextFloat() < 0.05F);
+        this.spawnType = spawnReason;
         return groupData;
     }
 
@@ -1221,6 +_,7 @@
             } else {
                 params.type().convert(this, newMob, params);
                 afterConversion.finalizeConversion(newMob);
+                if (!preventConversionSpawns)
                 if (this.level() instanceof ServerLevel serverLevel) {
                     serverLevel.addFreshEntity(newMob);
                 }
@@ -1388,14 +_,24 @@
     }
 
     @Override
+    @Deprecated // FORGE: use jumpInFluid instead
     protected void jumpInLiquid(TagKey<Fluid> type) {
+        this.jumpInLiquidInternal(() -> super.jumpInLiquid(type));
+    }
+
+    private void jumpInLiquidInternal(Runnable onSuper) {
         if (this.getNavigation().canFloat()) {
-            super.jumpInLiquid(type);
+            onSuper.run();
         } else {
             this.setDeltaMovement(this.getDeltaMovement().add(0.0, 0.3, 0.0));
         }
     }
 
+    @Override
+    public void jumpInFluid(net.neoforged.neoforge.fluids.FluidType type) {
+        this.jumpInLiquidInternal(() -> super.jumpInFluid(type));
+    }
+
     @VisibleForTesting
     public void removeFreeWill() {
         this.removeAllGoals(goal -> true);
@@ -1455,5 +_,40 @@
 
     public float chargeSpeedModifier() {
         return 1.0F;
+    }
+
+    /**
+     * Returns the type of spawn that created this mob, if applicable.
+     * If it could not be determined, this will return null.
+     * <p>
+     * This is set via {@link Mob#finalizeSpawn}, so you should not call this from within that method, instead using the parameter.
+     */
+    @Nullable
+    public final EntitySpawnReason getSpawnType() {
+        return this.spawnType;
+    }
+
+    /**
+     * This method exists so that spawns can be cancelled from the {@link net.neoforged.neoforge.event.entity.living.FinalizeSpawnEvent FinalizeSpawnEvent}
+     * without needing to hook up an additional handler for the {@link net.neoforged.neoforge.event.entity.EntityJoinLevelEvent EntityJoinLevelEvent}.
+     * @return if this mob will be blocked from spawning during {@link Level#addFreshEntity(Entity)}
+     * @apiNote Not public-facing API.
+     */
+    @org.jetbrains.annotations.ApiStatus.Internal
+    public final boolean isSpawnCancelled() {
+        return this.spawnCancelled;
+    }
+
+    /**
+     * Marks this mob as being disallowed to spawn during {@link Level#addFreshEntity(Entity)}.<p>
+     * @throws UnsupportedOperationException if this entity has already been {@link Entity#isAddedToLevel()} added to the level.
+     * @apiNote Not public-facing API.
+     */
+    @org.jetbrains.annotations.ApiStatus.Internal
+    public final void setSpawnCancelled(boolean cancel) {
+        if (this.isAddedToLevel()) {
+            throw new UnsupportedOperationException("Late invocations of Mob#setSpawnCancelled are not permitted.");
+        }
+        this.spawnCancelled = cancel;
     }
 }
