--- a/net/minecraft/world/entity/SpawnPlacementTypes.java
+++ b/net/minecraft/world/entity/SpawnPlacementTypes.java
@@ -10,16 +_,16 @@
 
 public interface SpawnPlacementTypes {
     SpawnPlacementType NO_RESTRICTIONS = (p_321554_, p_321832_, p_321540_) -> true;
-    SpawnPlacementType IN_WATER = (p_339936_, p_339937_, p_339938_) -> {
-        if (p_339938_ != null && p_339936_.getWorldBorder().isWithinBounds(p_339937_)) {
-            BlockPos blockpos = p_339937_.above();
-            return p_339936_.getFluidState(p_339937_).is(FluidTags.WATER) && !p_339936_.getBlockState(blockpos).isRedstoneConductor(p_339936_, blockpos);
+    SpawnPlacementType IN_WATER = (p_325672_, p_325673_, p_325674_) -> {
+        if (p_325674_ != null && p_325672_.getWorldBorder().isWithinBounds(p_325673_)) {
+            BlockPos blockpos = p_325673_.above();
+            return p_325672_.getFluidState(p_325673_).is(FluidTags.WATER) && !p_325672_.getBlockState(blockpos).isRedstoneConductor(p_325672_, blockpos);
         } else {
             return false;
         }
     };
-    SpawnPlacementType IN_LAVA = (p_339939_, p_339940_, p_339941_) -> p_339941_ != null && p_339939_.getWorldBorder().isWithinBounds(p_339940_)
-            ? p_339939_.getFluidState(p_339940_).is(FluidTags.LAVA)
+    SpawnPlacementType IN_LAVA = (p_325669_, p_325670_, p_325671_) -> p_325671_ != null && p_325669_.getWorldBorder().isWithinBounds(p_325670_)
+            ? p_325669_.getFluidState(p_325670_).is(FluidTags.LAVA)
             : false;
     SpawnPlacementType ON_GROUND = new SpawnPlacementType() {
         @Override
@@ -28,11 +_,9 @@
                 BlockPos blockpos = p_321783_.above();
                 BlockPos blockpos1 = p_321783_.below();
                 BlockState blockstate = p_321666_.getBlockState(blockpos1);
-                if (!blockstate.isValidSpawn(p_321666_, blockpos1, p_321839_)) {
-                    return false;
-                } else {
-                    return this.isValidEmptySpawnBlock(p_321666_, p_321783_, p_321839_) && this.isValidEmptySpawnBlock(p_321666_, blockpos, p_321839_);
-                }
+                return !blockstate.isValidSpawn(p_321666_, blockpos1, p_321839_)
+                    ? false
+                    : this.isValidEmptySpawnBlock(p_321666_, p_321783_, p_321839_) && this.isValidEmptySpawnBlock(p_321666_, blockpos, p_321839_);
             } else {
                 return false;
             }
