--- a/net/minecraft/data/worldgen/biome/BiomeData.java
+++ b/net/minecraft/data/worldgen/biome/BiomeData.java
@@ -15,11 +_,6 @@
         p_321862_.register(Biomes.THE_VOID, OverworldBiomes.theVoid(holdergetter, holdergetter1));
         p_321862_.register(Biomes.PLAINS, OverworldBiomes.plains(holdergetter, holdergetter1, false, false, false));
         p_321862_.register(Biomes.SUNFLOWER_PLAINS, OverworldBiomes.plains(holdergetter, holdergetter1, true, false, false));
-        p_321862_.register(Biomes.FIELDS, PotatoBiomes.fields(holdergetter, holdergetter1));
-        p_321862_.register(Biomes.ARBORETUM, PotatoBiomes.arboretum(holdergetter, holdergetter1));
-        p_321862_.register(Biomes.HASH, PotatoBiomes.hash(holdergetter, holdergetter1));
-        p_321862_.register(Biomes.WASTELAND, PotatoBiomes.wasteland(holdergetter, holdergetter1));
-        p_321862_.register(Biomes.CORRUPTION, PotatoBiomes.corruption(holdergetter, holdergetter1));
         p_321862_.register(Biomes.SNOWY_PLAINS, OverworldBiomes.plains(holdergetter, holdergetter1, false, true, false));
         p_321862_.register(Biomes.ICE_SPIKES, OverworldBiomes.plains(holdergetter, holdergetter1, false, true, true));
         p_321862_.register(Biomes.DESERT, OverworldBiomes.desert(holdergetter, holdergetter1));
