--- a/net/minecraft/world/entity/ai/memory/ExpirableValue.java
+++ b/net/minecraft/world/entity/ai/memory/ExpirableValue.java
@@ -17,7 +_,7 @@
 
     public void tick() {
         if (this.canExpire()) {
-            --this.timeToLive;
+            this.timeToLive--;
         }
     }
 
@@ -53,11 +_,13 @@
 
     public static <T> Codec<ExpirableValue<T>> codec(Codec<T> p_26305_) {
         return RecordCodecBuilder.create(
-            p_26308_ -> p_26308_.group(
+            p_337791_ -> p_337791_.group(
                         p_26305_.fieldOf("value").forGetter(p_148193_ -> p_148193_.value),
-                        Codec.LONG.optionalFieldOf("ttl").forGetter(p_148187_ -> p_148187_.canExpire() ? Optional.of(p_148187_.timeToLive) : Optional.empty())
+                        Codec.LONG
+                            .lenientOptionalFieldOf("ttl")
+                            .forGetter(p_148187_ -> p_148187_.canExpire() ? Optional.of(p_148187_.timeToLive) : Optional.empty())
                     )
-                    .apply(p_26308_, (p_148189_, p_148190_) -> new ExpirableValue<>(p_148189_, p_148190_.orElse(Long.MAX_VALUE)))
+                    .apply(p_337791_, (p_148189_, p_148190_) -> new ExpirableValue<>(p_148189_, p_148190_.orElse(Long.MAX_VALUE)))
         );
     }
 }
