--- a/net/minecraft/gametest/framework/GameTestServer.java
+++ b/net/minecraft/gametest/framework/GameTestServer.java
@@ -145,6 +_,7 @@
         ServerLevel serverlevel = thegame.overworld();
         this.testBatches = this.evaluateTestsToRun(serverlevel);
         LOGGER.info("Started game test server");
+        net.neoforged.neoforge.server.ServerLifecycleHooks.handleServerStarting(thegame);
         return thegame;
     }
 
