--- a/net/minecraft/gametest/framework/ReportGameListener.java
+++ b/net/minecraft/gametest/framework/ReportGameListener.java
@@ -33,7 +_,7 @@
     @Override
     public void testStructureLoaded(GameTestInfo p_177718_) {
         spawnBeacon(p_177718_, Blocks.LIGHT_GRAY_STAINED_GLASS);
-        ++this.attempts;
+        this.attempts++;
     }
 
     private void handleRetry(GameTestInfo p_319864_, GameTestRunner p_319782_, boolean p_320742_) {
@@ -59,7 +_,7 @@
 
     @Override
     public void testPassed(GameTestInfo p_177729_, GameTestRunner p_320359_) {
-        ++this.successes;
+        this.successes++;
         if (p_177729_.retryOptions().hasRetries()) {
             this.handleRetry(p_177729_, p_320359_, true);
         } else if (!p_177729_.isFlaky()) {
@@ -143,8 +_,8 @@
         BlockPos blockpos3 = blockpos2.offset(0, 1, 0);
         serverlevel.setBlockAndUpdate(blockpos3, p_177721_.defaultBlockState());
 
-        for(int i = -1; i <= 1; ++i) {
-            for(int j = -1; j <= 1; ++j) {
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
                 BlockPos blockpos4 = blockpos2.offset(i, -1, j);
                 serverlevel.setBlockAndUpdate(blockpos4, Blocks.IRON_BLOCK.defaultBlockState());
             }
