--- a/net/minecraft/client/gui/font/providers/BitmapProvider.java
+++ b/net/minecraft/client/gui/font/providers/BitmapProvider.java
@@ -23,7 +_,6 @@
 import net.minecraft.client.gui.font.glyphs.BakedGlyph;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.packs.resources.ResourceManager;
-import net.minecraft.util.ExtraCodecs;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 import org.slf4j.Logger;
@@ -57,11 +_,11 @@
 
     @OnlyIn(Dist.CLIENT)
     public static record Definition(ResourceLocation file, int height, int ascent, int[][] codepointGrid) implements GlyphProviderDefinition {
-        private static final Codec<int[][]> CODEPOINT_GRID_CODEC = ExtraCodecs.validate(Codec.STRING.listOf().xmap(p_286900_ -> {
+        private static final Codec<int[][]> CODEPOINT_GRID_CODEC = Codec.STRING.listOf().xmap(p_286900_ -> {
             int i = p_286900_.size();
             int[][] aint = new int[i][];
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 aint[j] = p_286900_.get(j).codePoints().toArray();
             }
 
@@ -69,14 +_,13 @@
         }, p_286828_ -> {
             List<String> list = new ArrayList<>(p_286828_.length);
 
-            for(int[] aint : p_286828_) {
+            for (int[] aint : p_286828_) {
                 list.add(new String(aint, 0, aint.length));
             }
 
             return list;
-        }), BitmapProvider.Definition::validateDimensions);
-        public static final MapCodec<BitmapProvider.Definition> CODEC = ExtraCodecs.validate(
-            RecordCodecBuilder.mapCodec(
+        }).validate(BitmapProvider.Definition::validateDimensions);
+        public static final MapCodec<BitmapProvider.Definition> CODEC = RecordCodecBuilder.<BitmapProvider.Definition>mapCodec(
                 p_286905_ -> p_286905_.group(
                             ResourceLocation.CODEC.fieldOf("file").forGetter(BitmapProvider.Definition::file),
                             Codec.INT.optionalFieldOf("height", Integer.valueOf(8)).forGetter(BitmapProvider.Definition::height),
@@ -84,9 +_,8 @@
                             CODEPOINT_GRID_CODEC.fieldOf("chars").forGetter(BitmapProvider.Definition::codepointGrid)
                         )
                         .apply(p_286905_, BitmapProvider.Definition::new)
-            ),
-            BitmapProvider.Definition::validate
-        );
+            )
+            .validate(BitmapProvider.Definition::validate);
 
         private static DataResult<int[][]> validateDimensions(int[][] p_286348_) {
             int i = p_286348_.length;
@@ -98,7 +_,7 @@
                 if (j == 0) {
                     return DataResult.error(() -> "Expected to find data in codepoint grid");
                 } else {
-                    for(int k = 1; k < i; ++k) {
+                    for (int k = 1; k < i; k++) {
                         int[] aint1 = p_286348_[k];
                         if (aint1.length != j) {
                             return DataResult.error(
@@ -143,14 +_,12 @@
                 int k = i / this.codepointGrid[0].length;
                 int l = j / this.codepointGrid.length;
                 float f = (float)this.height / (float)l;
-                CodepointMap<BitmapProvider.Glyph> codepointmap = new CodepointMap<>(
-                    p_286343_ -> new BitmapProvider.Glyph[p_286343_], p_286759_ -> new BitmapProvider.Glyph[p_286759_][]
-                );
+                CodepointMap<BitmapProvider.Glyph> codepointmap = new CodepointMap<>(BitmapProvider.Glyph[]::new, BitmapProvider.Glyph[][]::new);
 
-                for(int i1 = 0; i1 < this.codepointGrid.length; ++i1) {
+                for (int i1 = 0; i1 < this.codepointGrid.length; i1++) {
                     int j1 = 0;
 
-                    for(int k1 : this.codepointGrid[i1]) {
+                    for (int k1 : this.codepointGrid[i1]) {
                         int l1 = j1++;
                         if (k1 != 0) {
                             int i2 = this.getActualGlyphWidth(nativeimage, k, l, l1, i1);
@@ -172,10 +_,10 @@
 
         private int getActualGlyphWidth(NativeImage p_286449_, int p_286656_, int p_286554_, int p_286657_, int p_286307_) {
             int i;
-            for(i = p_286656_ - 1; i >= 0; --i) {
+            for (i = p_286656_ - 1; i >= 0; i--) {
                 int j = p_286657_ * p_286656_ + i;
 
-                for(int k = 0; k < p_286554_; ++k) {
+                for (int k = 0; k < p_286554_; k++) {
                     int l = p_286307_ * p_286554_ + k;
                     if (p_286449_.getLuminanceOrAlpha(j, l) != 0) {
                         return i + 1;
