--- a/net/minecraft/client/gui/font/providers/UnihexProvider.java
+++ b/net/minecraft/client/gui/font/providers/UnihexProvider.java
@@ -66,7 +_,7 @@
         int i = 32 - p_285312_ - 1;
         int j = 32 - p_285412_ - 1;
 
-        for(int k = i; k >= j; --k) {
+        for (int k = i; k >= j; k--) {
             if (k < 32 && k >= 0) {
                 boolean flag = (p_285508_ >> k & 1) != 0;
                 p_285211_.put(flag ? -1 : 0);
@@ -77,7 +_,7 @@
     }
 
     static void unpackBitsToBytes(IntBuffer p_285283_, UnihexProvider.LineData p_285485_, int p_284940_, int p_284950_) {
-        for(int i = 0; i < 16; ++i) {
+        for (int i = 0; i < 16; i++) {
             int j = p_285485_.line(i);
             unpackBitsToBytes(p_285283_, j, p_284940_, p_284950_);
         }
@@ -88,7 +_,7 @@
         int i = 0;
         ByteList bytelist = new ByteArrayList(128);
 
-        while(true) {
+        while (true) {
             boolean flag = copyUntil(p_285315_, bytelist, 58);
             int j = bytelist.size();
             if (j == 0 && !flag) {
@@ -101,7 +_,7 @@
 
             int k = 0;
 
-            for(int l = 0; l < j; ++l) {
+            for (int l = 0; l < j; l++) {
                 k = k << 4 | decodeHex(i, bytelist.getByte(l));
             }
 
@@ -109,7 +_,7 @@
             copyUntil(p_285315_, bytelist, 10);
             int i1 = bytelist.size();
 
-            UnihexProvider.LineData unihexprovider$linedata = switch(i1) {
+            UnihexProvider.LineData unihexprovider$linedata = switch (i1) {
                 case 32 -> UnihexProvider.ByteContents.read(i, bytelist);
                 case 64 -> UnihexProvider.ShortContents.read(i, bytelist);
                 case 96 -> UnihexProvider.IntContents.read24(i, bytelist);
@@ -119,7 +_,7 @@
             );
             };
             p_285353_.accept(k, unihexprovider$linedata);
-            ++i;
+            i++;
             bytelist.clear();
         }
     }
@@ -129,7 +_,7 @@
     }
 
     private static int decodeHex(int p_284952_, byte p_285036_) {
-        return switch(p_285036_) {
+        return switch (p_285036_) {
             case 48 -> 0;
             case 49 -> 1;
             case 50 -> 2;
@@ -151,7 +_,7 @@
     }
 
     private static boolean copyUntil(InputStream p_284994_, ByteList p_285351_, int p_285177_) throws IOException {
-        while(true) {
+        while (true) {
             int i = p_284994_.read();
             if (i == -1) {
                 return false;
@@ -176,7 +_,7 @@
             byte[] abyte = new byte[16];
             int i = 0;
 
-            for(int j = 0; j < 16; ++j) {
+            for (int j = 0; j < 16; j++) {
                 int k = UnihexProvider.decodeHex(p_285080_, p_285481_, i++);
                 int l = UnihexProvider.decodeHex(p_285080_, p_285481_, i++);
                 byte b0 = (byte)(k << 4 | l);
@@ -229,15 +_,13 @@
         }
 
         private UnihexProvider loadData(InputStream p_286795_) throws IOException {
-            CodepointMap<UnihexProvider.LineData> codepointmap = new CodepointMap<>(
-                p_286908_ -> new UnihexProvider.LineData[p_286908_], p_286615_ -> new UnihexProvider.LineData[p_286615_][]
-            );
+            CodepointMap<UnihexProvider.LineData> codepointmap = new CodepointMap<>(UnihexProvider.LineData[]::new, UnihexProvider.LineData[][]::new);
             UnihexProvider.ReaderOutput unihexprovider$readeroutput = codepointmap::put;
 
             UnihexProvider unihexprovider;
             try (ZipInputStream zipinputstream = new ZipInputStream(p_286795_)) {
                 ZipEntry zipentry;
-                while((zipentry = zipinputstream.getNextEntry()) != null) {
+                while ((zipentry = zipinputstream.getNextEntry()) != null) {
                     String s = zipentry.getName();
                     if (s.endsWith(".hex")) {
                         UnihexProvider.LOGGER.info("Found {}, loading", s);
@@ -245,16 +_,14 @@
                     }
                 }
 
-                CodepointMap<UnihexProvider.Glyph> codepointmap1 = new CodepointMap<>(
-                    p_286831_ -> new UnihexProvider.Glyph[p_286831_], p_286340_ -> new UnihexProvider.Glyph[p_286340_][]
-                );
+                CodepointMap<UnihexProvider.Glyph> codepointmap1 = new CodepointMap<>(UnihexProvider.Glyph[]::new, UnihexProvider.Glyph[][]::new);
 
-                for(UnihexProvider.OverrideRange unihexprovider$overriderange : this.sizeOverrides) {
+                for (UnihexProvider.OverrideRange unihexprovider$overriderange : this.sizeOverrides) {
                     int i = unihexprovider$overriderange.from;
                     int j = unihexprovider$overriderange.to;
                     UnihexProvider.Dimensions unihexprovider$dimensions = unihexprovider$overriderange.dimensions;
 
-                    for(int k = i; k <= j; ++k) {
+                    for (int k = i; k <= j; k++) {
                         UnihexProvider.LineData unihexprovider$linedata = codepointmap.remove(k);
                         if (unihexprovider$linedata != null) {
                             codepointmap1.put(
@@ -374,7 +_,7 @@
             int i = 0;
             int j = 0;
 
-            for(int k = 0; k < 16; ++k) {
+            for (int k = 0; k < 16; k++) {
                 int l = UnihexProvider.decodeHex(p_285362_, p_285123_, j++);
                 int i1 = UnihexProvider.decodeHex(p_285362_, p_285123_, j++);
                 int j1 = UnihexProvider.decodeHex(p_285362_, p_285123_, j++);
@@ -394,7 +_,7 @@
             int i = 0;
             int j = 0;
 
-            for(int k = 0; k < 16; ++k) {
+            for (int k = 0; k < 16; k++) {
                 int l = UnihexProvider.decodeHex(p_285222_, p_285346_, j++);
                 int i1 = UnihexProvider.decodeHex(p_285222_, p_285346_, j++);
                 int j1 = UnihexProvider.decodeHex(p_285222_, p_285346_, j++);
@@ -421,7 +_,7 @@
         default int mask() {
             int i = 0;
 
-            for(int j = 0; j < 16; ++j) {
+            for (int j = 0; j < 16; j++) {
                 i |= this.line(j);
             }
 
@@ -455,8 +_,7 @@
                     )
                     .apply(p_285088_, UnihexProvider.OverrideRange::new)
         );
-        public static final Codec<UnihexProvider.OverrideRange> CODEC = ExtraCodecs.validate(
-            RAW_CODEC,
+        public static final Codec<UnihexProvider.OverrideRange> CODEC = RAW_CODEC.validate(
             p_285215_ -> p_285215_.from >= p_285215_.to
                     ? DataResult.error(() -> "Invalid range: [" + p_285215_.from + ";" + p_285215_.to + "]")
                     : DataResult.success(p_285215_)
@@ -480,7 +_,7 @@
             short[] ashort = new short[16];
             int i = 0;
 
-            for(int j = 0; j < 16; ++j) {
+            for (int j = 0; j < 16; j++) {
                 int k = UnihexProvider.decodeHex(p_285528_, p_284958_, i++);
                 int l = UnihexProvider.decodeHex(p_285528_, p_284958_, i++);
                 int i1 = UnihexProvider.decodeHex(p_285528_, p_284958_, i++);
