--- a/net/minecraft/world/level/levelgen/structure/structures/BuriedTreasureStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/BuriedTreasureStructure.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.structures;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.Optional;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.levelgen.Heightmap;
@@ -9,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.pieces.StructurePiecesBuilder;
 
 public class BuriedTreasureStructure extends Structure {
-    public static final Codec<BuriedTreasureStructure> CODEC = simpleCodec(BuriedTreasureStructure::new);
+    public static final MapCodec<BuriedTreasureStructure> CODEC = simpleCodec(BuriedTreasureStructure::new);
 
     public BuriedTreasureStructure(Structure.StructureSettings p_227385_) {
         super(p_227385_);
