--- a/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidPiece.java
@@ -73,7 +_,7 @@
                 false
             );
 
-            for(int i = 1; i <= 9; ++i) {
+            for (int i = 1; i <= 9; i++) {
                 this.generateBox(
                     p_227405_,
                     p_227409_,
@@ -102,8 +_,8 @@
                 );
             }
 
-            for(int k1 = 0; k1 < this.width; ++k1) {
-                for(int j = 0; j < this.depth; ++j) {
+            for (int k1 = 0; k1 < this.width; k1++) {
+                for (int j = 0; j < this.depth; j++) {
                     int k = -5;
                     this.fillColumnDown(p_227405_, Blocks.SANDSTONE.defaultBlockState(), k1, -5, j, p_227409_);
                 }
@@ -271,7 +_,7 @@
                 p_227405_, p_227409_, this.width - 6, 1, 5, this.width - 5, 2, 16, Blocks.AIR.defaultBlockState(), Blocks.AIR.defaultBlockState(), false
             );
 
-            for(int l = 5; l <= 17; l += 2) {
+            for (int l = 5; l <= 17; l += 2) {
                 this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), 4, 1, l, p_227409_);
                 this.placeBlock(p_227405_, Blocks.CHISELED_SANDSTONE.defaultBlockState(), 4, 2, l, p_227409_);
                 this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), this.width - 5, 1, l, p_227409_);
@@ -292,7 +_,7 @@
             this.placeBlock(p_227405_, Blocks.ORANGE_TERRACOTTA.defaultBlockState(), 10, 0, 13, p_227409_);
             this.placeBlock(p_227405_, Blocks.BLUE_TERRACOTTA.defaultBlockState(), 10, 0, 10, p_227409_);
 
-            for(int l1 = 0; l1 <= this.width - 1; l1 += this.width - 1) {
+            for (int l1 = 0; l1 <= this.width - 1; l1 += this.width - 1) {
                 this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), l1, 2, 1, p_227409_);
                 this.placeBlock(p_227405_, Blocks.ORANGE_TERRACOTTA.defaultBlockState(), l1, 2, 2, p_227409_);
                 this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), l1, 2, 3, p_227409_);
@@ -316,7 +_,7 @@
                 this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), l1, 8, 3, p_227409_);
             }
 
-            for(int i2 = 2; i2 <= this.width - 3; i2 += this.width - 3 - 2) {
+            for (int i2 = 2; i2 <= this.width - 3; i2 += this.width - 3 - 2) {
                 this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), i2 - 1, 2, 0, p_227409_);
                 this.placeBlock(p_227405_, Blocks.ORANGE_TERRACOTTA.defaultBlockState(), i2, 2, 0, p_227409_);
                 this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), i2 + 1, 2, 0, p_227409_);
@@ -386,7 +_,7 @@
             this.placeBlock(p_227405_, Blocks.CHISELED_SANDSTONE.defaultBlockState(), 10, -10, 13, p_227409_);
             this.placeBlock(p_227405_, Blocks.CUT_SANDSTONE.defaultBlockState(), 10, -11, 13, p_227409_);
 
-            for(Direction direction : Direction.Plane.HORIZONTAL) {
+            for (Direction direction : Direction.Plane.HORIZONTAL) {
                 if (!this.hasPlacedChest[direction.get2DDataValue()]) {
                     int i1 = direction.getStepX() * 2;
                     int j1 = direction.getStepZ() * 2;
@@ -487,9 +_,9 @@
     }
 
     private void placeSandBox(int p_279483_, int p_279321_, int p_279271_, int p_279471_, int p_279229_, int p_279111_) {
-        for(int i = p_279321_; i <= p_279229_; ++i) {
-            for(int j = p_279483_; j <= p_279471_; ++j) {
-                for(int k = p_279271_; k <= p_279111_; ++k) {
+        for (int i = p_279321_; i <= p_279229_; i++) {
+            for (int j = p_279483_; j <= p_279471_; j++) {
+                for (int k = p_279271_; k <= p_279111_; k++) {
                     this.placeSand(j, i, k);
                 }
             }
@@ -507,8 +_,8 @@
     }
 
     private void placeCollapsedRoof(WorldGenLevel p_273438_, BoundingBox p_273058_, int p_272638_, int p_272826_, int p_273026_, int p_272750_, int p_272639_) {
-        for(int i = p_272638_; i <= p_272750_; ++i) {
-            for(int j = p_273026_; j <= p_272639_; ++j) {
+        for (int i = p_272638_; i <= p_272750_; i++) {
+            for (int j = p_273026_; j <= p_272639_; j++) {
                 this.placeCollapsedRoofPiece(p_273438_, i, p_272826_, j, p_273058_);
             }
         }
