--- a/net/minecraft/world/level/levelgen/structure/structures/EndCityStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/EndCityStructure.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.structures;
 
 import com.google.common.collect.Lists;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import java.util.Optional;
 import net.minecraft.core.BlockPos;
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.pieces.StructurePiecesBuilder;
 
 public class EndCityStructure extends Structure {
-    public static final Codec<EndCityStructure> CODEC = simpleCodec(EndCityStructure::new);
+    public static final MapCodec<EndCityStructure> CODEC = simpleCodec(EndCityStructure::new);
 
     public EndCityStructure(Structure.StructureSettings p_227526_) {
         super(p_227526_);
