--- a/net/minecraft/world/level/levelgen/structure/structures/JungleTemplePiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/JungleTemplePiece.java
@@ -95,7 +_,7 @@
             this.placeBlock(p_227674_, Blocks.AIR.defaultBlockState(), 1, 5, 9, p_227678_);
             this.placeBlock(p_227674_, Blocks.AIR.defaultBlockState(), 10, 5, 9, p_227678_);
 
-            for(int i = 0; i <= 14; i += 14) {
+            for (int i = 0; i <= 14; i += 14) {
                 this.generateBox(p_227674_, p_227678_, 2, 4, i, 2, 5, i, false, p_227677_, STONE_SELECTOR);
                 this.generateBox(p_227674_, p_227678_, 4, 4, i, 4, 5, i, false, p_227677_, STONE_SELECTOR);
                 this.generateBox(p_227674_, p_227678_, 7, 4, i, 7, 5, i, false, p_227677_, STONE_SELECTOR);
@@ -104,8 +_,8 @@
 
             this.generateBox(p_227674_, p_227678_, 5, 6, 0, 6, 6, 0, false, p_227677_, STONE_SELECTOR);
 
-            for(int l = 0; l <= 11; l += 11) {
-                for(int j = 2; j <= 12; j += 2) {
+            for (int l = 0; l <= 11; l += 11) {
+                for (int j = 2; j <= 12; j += 2) {
                     this.generateBox(p_227674_, p_227678_, l, 4, j, l, 5, j, false, p_227677_, STONE_SELECTOR);
                 }
 
@@ -147,7 +_,7 @@
             this.placeBlock(p_227674_, blockstate3, 4, 4, 5, p_227678_);
             this.placeBlock(p_227674_, blockstate4, 7, 4, 5, p_227678_);
 
-            for(int k = 0; k < 4; ++k) {
+            for (int k = 0; k < 4; k++) {
                 this.placeBlock(p_227674_, blockstate, 5, 0 - k, 6 + k, p_227678_);
                 this.placeBlock(p_227674_, blockstate, 6, 0 - k, 6 + k, p_227678_);
                 this.generateAirBox(p_227674_, p_227678_, 5, 0 - k, 7 + k, 6, 0 - k, 9 + k);
@@ -157,11 +_,11 @@
             this.generateAirBox(p_227674_, p_227678_, 1, -3, 1, 3, -1, 13);
             this.generateAirBox(p_227674_, p_227678_, 1, -3, 1, 9, -1, 5);
 
-            for(int i1 = 1; i1 <= 13; i1 += 2) {
+            for (int i1 = 1; i1 <= 13; i1 += 2) {
                 this.generateBox(p_227674_, p_227678_, 1, -3, i1, 1, -2, i1, false, p_227677_, STONE_SELECTOR);
             }
 
-            for(int j1 = 2; j1 <= 12; j1 += 2) {
+            for (int j1 = 2; j1 <= 12; j1 += 2) {
                 this.generateBox(p_227674_, p_227678_, 1, -1, j1, 3, -1, j1, false, p_227677_, STONE_SELECTOR);
             }
 
