--- a/net/minecraft/world/level/levelgen/structure/structures/NetherFortressStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/NetherFortressStructure.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.structures;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import java.util.Optional;
 import net.minecraft.core.BlockPos;
@@ -21,7 +_,7 @@
         new MobSpawnSettings.SpawnerData(EntityType.SKELETON, 2, 5, 5),
         new MobSpawnSettings.SpawnerData(EntityType.MAGMA_CUBE, 3, 4, 4)
     );
-    public static final Codec<NetherFortressStructure> CODEC = simpleCodec(NetherFortressStructure::new);
+    public static final MapCodec<NetherFortressStructure> CODEC = simpleCodec(NetherFortressStructure::new);
 
     public NetherFortressStructure(Structure.StructureSettings p_228521_) {
         super(p_228521_);
@@ -42,7 +_,7 @@
         netherfortresspieces$startpiece.addChildren(netherfortresspieces$startpiece, p_228528_, p_228529_.random());
         List<StructurePiece> list = netherfortresspieces$startpiece.pendingChildren;
 
-        while(!list.isEmpty()) {
+        while (!list.isEmpty()) {
             int i = p_228529_.random().nextInt(list.size());
             StructurePiece structurepiece = list.remove(i);
             structurepiece.addChildren(netherfortresspieces$startpiece, p_228528_, p_228529_.random());
