--- a/net/minecraft/world/level/levelgen/structure/structures/NetherFortressPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/NetherFortressPieces.java
@@ -14,7 +_,6 @@
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.FenceBlock;
 import net.minecraft.world.level.block.StairBlock;
-import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.entity.SpawnerBlockEntity;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.chunk.ChunkGenerator;
@@ -203,8 +_,8 @@
                 p_228035_, p_228039_, 7, 0, 15, 11, 1, 18, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false
             );
 
-            for(int i = 7; i <= 11; ++i) {
-                for(int j = 0; j <= 2; ++j) {
+            for (int i = 7; i <= 11; i++) {
+                for (int j = 0; j <= 2; j++) {
                     this.fillColumnDown(p_228035_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228039_);
                     this.fillColumnDown(p_228035_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, 18 - j, p_228039_);
                 }
@@ -219,8 +_,8 @@
                 p_228035_, p_228039_, 15, 0, 7, 18, 1, 11, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false
             );
 
-            for(int k = 0; k <= 2; ++k) {
-                for(int l = 7; l <= 11; ++l) {
+            for (int k = 0; k <= 2; k++) {
+                for (int l = 7; l <= 11; l++) {
                     this.fillColumnDown(p_228035_, Blocks.NETHER_BRICKS.defaultBlockState(), k, -1, l, p_228039_);
                     this.fillColumnDown(p_228035_, Blocks.NETHER_BRICKS.defaultBlockState(), 18 - k, -1, l, p_228039_);
                 }
@@ -272,8 +_,8 @@
         ) {
             RandomSource randomsource = RandomSource.create((long)this.selfSeed);
 
-            for(int i = 0; i <= 4; ++i) {
-                for(int j = 3; j <= 4; ++j) {
+            for (int i = 0; i <= 4; i++) {
+                for (int j = 3; j <= 4; j++) {
                     int k = randomsource.nextInt(8);
                     this.generateBox(
                         p_228065_, p_228069_, i, j, 0, i, j, k, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false
@@ -286,15 +_,15 @@
             l = randomsource.nextInt(8);
             this.generateBox(p_228065_, p_228069_, 4, 5, 0, 4, 5, l, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
 
-            for(int i1 = 0; i1 <= 4; ++i1) {
+            for (int i1 = 0; i1 <= 4; i1++) {
                 int k1 = randomsource.nextInt(5);
                 this.generateBox(
                     p_228065_, p_228069_, i1, 2, 0, i1, 2, k1, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false
                 );
             }
 
-            for(int j1 = 0; j1 <= 4; ++j1) {
-                for(int l1 = 0; l1 <= 1; ++l1) {
+            for (int j1 = 0; j1 <= 4; j1++) {
+                for (int l1 = 0; l1 <= 1; l1++) {
                     int i2 = randomsource.nextInt(3);
                     this.generateBox(
                         p_228065_, p_228069_, j1, l1, 0, j1, l1, i2, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false
@@ -355,8 +_,8 @@
                 p_228094_, p_228098_, 0, 0, 15, 4, 1, 18, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false
             );
 
-            for(int i = 0; i <= 4; ++i) {
-                for(int j = 0; j <= 2; ++j) {
+            for (int i = 0; i <= 4; i++) {
+                for (int j = 0; j <= 2; j++) {
                     this.fillColumnDown(p_228094_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228098_);
                     this.fillColumnDown(p_228094_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, 18 - j, p_228098_);
                 }
@@ -423,7 +_,7 @@
                 .setValue(FenceBlock.NORTH, Boolean.valueOf(true))
                 .setValue(FenceBlock.SOUTH, Boolean.valueOf(true));
 
-            for(int i = 0; i <= 9; ++i) {
+            for (int i = 0; i <= 9; i++) {
                 int j = Math.max(1, 7 - i);
                 int k = Math.min(Math.max(j + 5, 14 - i), 13);
                 int l = i;
@@ -451,7 +_,7 @@
                     this.generateBox(p_228123_, p_228127_, 4, j + 2, i, 4, j + 3, i, blockstate1, blockstate1, false);
                 }
 
-                for(int i1 = 0; i1 <= 4; ++i1) {
+                for (int i1 = 0; i1 <= 4; i1++) {
                     this.fillColumnDown(p_228123_, Blocks.NETHER_BRICKS.defaultBlockState(), i1, -1, l, p_228127_);
                 }
             }
@@ -553,8 +_,8 @@
             this.generateBox(p_228151_, p_228155_, 1, 4, 5, 1, 5, 5, blockstate1, blockstate1, false);
             this.generateBox(p_228151_, p_228155_, 7, 4, 5, 7, 5, 5, blockstate1, blockstate1, false);
 
-            for(int i = 0; i <= 5; ++i) {
-                for(int j = 0; j <= 8; ++j) {
+            for (int i = 0; i <= 5; i++) {
+                for (int j = 0; j <= 8; j++) {
                     this.fillColumnDown(p_228151_, Blocks.NETHER_BRICKS.defaultBlockState(), j, -1, i, p_228155_);
                 }
             }
@@ -638,7 +_,7 @@
                 .setValue(FenceBlock.NORTH, Boolean.valueOf(true))
                 .setValue(FenceBlock.SOUTH, Boolean.valueOf(true));
 
-            for(int i = 1; i <= 11; i += 2) {
+            for (int i = 1; i <= 11; i += 2) {
                 this.generateBox(p_228180_, p_228184_, i, 10, 0, i, 11, 0, blockstate, blockstate, false);
                 this.generateBox(p_228180_, p_228184_, i, 10, 12, i, 11, 12, blockstate, blockstate, false);
                 this.generateBox(p_228180_, p_228184_, 0, 10, i, 0, 11, i, blockstate1, blockstate1, false);
@@ -700,7 +_,7 @@
                 p_228184_
             );
 
-            for(int k = 3; k <= 9; k += 2) {
+            for (int k = 3; k <= 9; k += 2) {
                 this.generateBox(
                     p_228180_,
                     p_228184_,
@@ -736,15 +_,15 @@
             this.generateBox(p_228180_, p_228184_, 0, 0, 4, 3, 1, 8, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
             this.generateBox(p_228180_, p_228184_, 9, 0, 4, 12, 1, 8, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
 
-            for(int l = 4; l <= 8; ++l) {
-                for(int j = 0; j <= 2; ++j) {
+            for (int l = 4; l <= 8; l++) {
+                for (int j = 0; j <= 2; j++) {
                     this.fillColumnDown(p_228180_, Blocks.NETHER_BRICKS.defaultBlockState(), l, -1, j, p_228184_);
                     this.fillColumnDown(p_228180_, Blocks.NETHER_BRICKS.defaultBlockState(), l, -1, 12 - j, p_228184_);
                 }
             }
 
-            for(int i1 = 0; i1 <= 2; ++i1) {
-                for(int j1 = 4; j1 <= 8; ++j1) {
+            for (int i1 = 0; i1 <= 2; i1++) {
+                for (int j1 = 4; j1 <= 8; j1++) {
                     this.fillColumnDown(p_228180_, Blocks.NETHER_BRICKS.defaultBlockState(), i1, -1, j1, p_228184_);
                     this.fillColumnDown(p_228180_, Blocks.NETHER_BRICKS.defaultBlockState(), 12 - i1, -1, j1, p_228184_);
                 }
@@ -809,8 +_,8 @@
             this.generateBox(p_228209_, p_228213_, 4, 2, 4, 4, 5, 4, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
             this.generateBox(p_228209_, p_228213_, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
 
-            for(int i = 0; i <= 4; ++i) {
-                for(int j = 0; j <= 4; ++j) {
+            for (int i = 0; i <= 4; i++) {
+                for (int j = 0; j <= 4; j++) {
                     this.fillColumnDown(p_228209_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228213_);
                 }
             }
@@ -888,8 +_,8 @@
 
             this.generateBox(p_228239_, p_228243_, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
 
-            for(int i = 0; i <= 4; ++i) {
-                for(int j = 0; j <= 4; ++j) {
+            for (int i = 0; i <= 4; i++) {
+                for (int j = 0; j <= 4; j++) {
                     this.fillColumnDown(p_228239_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228243_);
                 }
             }
@@ -948,8 +_,8 @@
             this.generateBox(p_228271_, p_228275_, 4, 3, 3, 4, 4, 3, blockstate, blockstate, false);
             this.generateBox(p_228271_, p_228275_, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
 
-            for(int i = 0; i <= 4; ++i) {
-                for(int j = 0; j <= 4; ++j) {
+            for (int i = 0; i <= 4; i++) {
+                for (int j = 0; j <= 4; j++) {
                     this.fillColumnDown(p_228271_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228275_);
                 }
             }
@@ -1027,8 +_,8 @@
 
             this.generateBox(p_228301_, p_228305_, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
 
-            for(int i = 0; i <= 4; ++i) {
-                for(int j = 0; j <= 4; ++j) {
+            for (int i = 0; i <= 4; i++) {
+                for (int j = 0; j <= 4; j++) {
                     this.fillColumnDown(p_228301_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228305_);
                 }
             }
@@ -1112,7 +_,7 @@
             BlockState blockstate2 = blockstate1.setValue(FenceBlock.WEST, Boolean.valueOf(true));
             BlockState blockstate3 = blockstate1.setValue(FenceBlock.EAST, Boolean.valueOf(true));
 
-            for(int i = 1; i <= 11; i += 2) {
+            for (int i = 1; i <= 11; i += 2) {
                 this.generateBox(p_228333_, p_228337_, i, 10, 0, i, 11, 0, blockstate, blockstate, false);
                 this.generateBox(p_228333_, p_228337_, i, 10, 12, i, 11, 12, blockstate, blockstate, false);
                 this.generateBox(p_228333_, p_228337_, 0, 10, i, 0, 11, i, blockstate1, blockstate1, false);
@@ -1174,17 +_,17 @@
                 p_228337_
             );
 
-            for(int j1 = 3; j1 <= 9; j1 += 2) {
+            for (int j1 = 3; j1 <= 9; j1 += 2) {
                 this.generateBox(p_228333_, p_228337_, 1, 7, j1, 1, 8, j1, blockstate2, blockstate2, false);
                 this.generateBox(p_228333_, p_228337_, 11, 7, j1, 11, 8, j1, blockstate3, blockstate3, false);
             }
 
             BlockState blockstate4 = Blocks.NETHER_BRICK_STAIRS.defaultBlockState().setValue(StairBlock.FACING, Direction.NORTH);
 
-            for(int j = 0; j <= 6; ++j) {
+            for (int j = 0; j <= 6; j++) {
                 int k = j + 4;
 
-                for(int l = 5; l <= 7; ++l) {
+                for (int l = 5; l <= 7; l++) {
                     this.placeBlock(p_228333_, blockstate4, l, 5 + j, k, p_228337_);
                 }
 
@@ -1203,7 +_,7 @@
                 }
             }
 
-            for(int k1 = 5; k1 <= 7; ++k1) {
+            for (int k1 = 5; k1 <= 7; k1++) {
                 this.placeBlock(p_228333_, blockstate4, k1, 12, 11, p_228337_);
             }
 
@@ -1241,15 +_,15 @@
             this.generateBox(p_228333_, p_228337_, 0, 0, 4, 3, 1, 8, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
             this.generateBox(p_228333_, p_228337_, 9, 0, 4, 12, 1, 8, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
 
-            for(int l1 = 4; l1 <= 8; ++l1) {
-                for(int i1 = 0; i1 <= 2; ++i1) {
+            for (int l1 = 4; l1 <= 8; l1++) {
+                for (int i1 = 0; i1 <= 2; i1++) {
                     this.fillColumnDown(p_228333_, Blocks.NETHER_BRICKS.defaultBlockState(), l1, -1, i1, p_228337_);
                     this.fillColumnDown(p_228333_, Blocks.NETHER_BRICKS.defaultBlockState(), l1, -1, 12 - i1, p_228337_);
                 }
             }
 
-            for(int i2 = 0; i2 <= 2; ++i2) {
-                for(int j2 = 4; j2 <= 8; ++j2) {
+            for (int i2 = 0; i2 <= 2; i2++) {
+                for (int j2 = 4; j2 <= 8; j2++) {
                     this.fillColumnDown(p_228333_, Blocks.NETHER_BRICKS.defaultBlockState(), i2, -1, j2, p_228337_);
                     this.fillColumnDown(p_228333_, Blocks.NETHER_BRICKS.defaultBlockState(), 12 - i2, -1, j2, p_228337_);
                 }
@@ -1378,15 +_,14 @@
                 if (p_228366_.isInside(blockpos)) {
                     this.hasPlacedSpawner = true;
                     p_228362_.setBlock(blockpos, Blocks.SPAWNER.defaultBlockState(), 2);
-                    BlockEntity blockentity = p_228362_.getBlockEntity(blockpos);
-                    if (blockentity instanceof SpawnerBlockEntity spawnerblockentity) {
+                    if (p_228362_.getBlockEntity(blockpos) instanceof SpawnerBlockEntity spawnerblockentity) {
                         spawnerblockentity.setEntityId(EntityType.BLAZE, p_228365_);
                     }
                 }
             }
 
-            for(int i = 0; i <= 6; ++i) {
-                for(int j = 0; j <= 6; ++j) {
+            for (int i = 0; i <= 6; i++) {
+                for (int j = 0; j <= 6; j++) {
                     this.fillColumnDown(p_228362_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228366_);
                 }
             }
@@ -1410,7 +_,7 @@
             boolean flag = false;
             int i = 0;
 
-            for(NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight : p_228419_) {
+            for (NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight : p_228419_) {
                 if (netherfortresspieces$pieceweight.maxPlaceCount > 0
                     && netherfortresspieces$pieceweight.placeCount < netherfortresspieces$pieceweight.maxPlaceCount) {
                     flag = true;
@@ -1437,11 +_,11 @@
             boolean flag = i > 0 && p_228417_ <= 30;
             int j = 0;
 
-            while(j < 5 && flag) {
-                ++j;
+            while (j < 5 && flag) {
+                j++;
                 int k = p_228412_.nextInt(i);
 
-                for(NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight : p_228410_) {
+                for (NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight : p_228410_) {
                     k -= netherfortresspieces$pieceweight.weight;
                     if (k < 0) {
                         if (!netherfortresspieces$pieceweight.doPlace(p_228417_)
@@ -1453,7 +_,7 @@
                             netherfortresspieces$pieceweight, p_228411_, p_228412_, p_228413_, p_228414_, p_228415_, p_228416_, p_228417_
                         );
                         if (netherfortresspieces$netherbridgepiece != null) {
-                            ++netherfortresspieces$pieceweight.placeCount;
+                            netherfortresspieces$pieceweight.placeCount++;
                             p_228409_.previousPiece = netherfortresspieces$pieceweight;
                             if (!netherfortresspieces$pieceweight.isValid()) {
                                 p_228410_.remove(netherfortresspieces$pieceweight);
@@ -1510,7 +_,7 @@
         ) {
             Direction direction = this.getOrientation();
             if (direction != null) {
-                switch(direction) {
+                switch (direction) {
                     case NORTH:
                         return this.generateAndAddPiece(
                             p_228402_,
@@ -1576,7 +_,7 @@
         ) {
             Direction direction = this.getOrientation();
             if (direction != null) {
-                switch(direction) {
+                switch (direction) {
                     case NORTH:
                         return this.generateAndAddPiece(
                             p_228421_,
@@ -1642,7 +_,7 @@
         ) {
             Direction direction = this.getOrientation();
             if (direction != null) {
-                switch(direction) {
+                switch (direction) {
                     case NORTH:
                         return this.generateAndAddPiece(
                             p_228428_,
@@ -1796,8 +_,8 @@
             this.generateBox(p_228461_, p_228465_, 6, 6, 2, 6, 6, 4, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
             this.generateBox(p_228461_, p_228465_, 6, 5, 2, 6, 5, 4, blockstate1, blockstate1, false);
 
-            for(int i = 0; i <= 6; ++i) {
-                for(int j = 0; j <= 6; ++j) {
+            for (int i = 0; i <= 6; i++) {
+                for (int j = 0; j <= 6; j++) {
                     this.fillColumnDown(p_228461_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228465_);
                 }
             }
@@ -1870,8 +_,8 @@
             this.generateBox(p_228489_, p_228493_, 2, 6, 0, 4, 8, 0, Blocks.NETHER_BRICKS.defaultBlockState(), Blocks.NETHER_BRICKS.defaultBlockState(), false);
             this.generateBox(p_228489_, p_228493_, 2, 5, 0, 4, 5, 0, blockstate, blockstate, false);
 
-            for(int i = 0; i <= 6; ++i) {
-                for(int j = 0; j <= 6; ++j) {
+            for (int i = 0; i <= 6; i++) {
+                for (int j = 0; j <= 6; j++) {
                     this.fillColumnDown(p_228489_, Blocks.NETHER_BRICKS.defaultBlockState(), i, -1, j, p_228493_);
                 }
             }
@@ -1888,14 +_,14 @@
             super(p_228513_, p_228514_, getRandomHorizontalDirection(p_228512_));
             this.availableBridgePieces = Lists.newArrayList();
 
-            for(NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight : NetherFortressPieces.BRIDGE_PIECE_WEIGHTS) {
+            for (NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight : NetherFortressPieces.BRIDGE_PIECE_WEIGHTS) {
                 netherfortresspieces$pieceweight.placeCount = 0;
                 this.availableBridgePieces.add(netherfortresspieces$pieceweight);
             }
 
             this.availableCastlePieces = Lists.newArrayList();
 
-            for(NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight1 : NetherFortressPieces.CASTLE_PIECE_WEIGHTS) {
+            for (NetherFortressPieces.PieceWeight netherfortresspieces$pieceweight1 : NetherFortressPieces.CASTLE_PIECE_WEIGHTS) {
                 netherfortresspieces$pieceweight1.placeCount = 0;
                 this.availableCastlePieces.add(netherfortresspieces$pieceweight1);
             }
