--- a/net/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces.java
@@ -182,9 +_,9 @@
             list1.add(new OceanMonumentPieces.FitSimpleTopRoom());
             list1.add(new OceanMonumentPieces.FitSimpleRoom());
 
-            for(OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition : list) {
+            for (OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition : list) {
                 if (!oceanmonumentpieces$roomdefinition.claimed && !oceanmonumentpieces$roomdefinition.isSpecial()) {
-                    for(OceanMonumentPieces.MonumentRoomFitter oceanmonumentpieces$monumentroomfitter : list1) {
+                    for (OceanMonumentPieces.MonumentRoomFitter oceanmonumentpieces$monumentroomfitter : list1) {
                         if (oceanmonumentpieces$monumentroomfitter.fits(oceanmonumentpieces$roomdefinition)) {
                             this.childPieces.add(oceanmonumentpieces$monumentroomfitter.create(p_228651_, oceanmonumentpieces$roomdefinition, p_228648_));
                             break;
@@ -195,7 +_,7 @@
 
             BlockPos blockpos = this.getWorldPos(9, 0, 22);
 
-            for(OceanMonumentPieces.OceanMonumentPiece oceanmonumentpieces$oceanmonumentpiece : this.childPieces) {
+            for (OceanMonumentPieces.OceanMonumentPiece oceanmonumentpieces$oceanmonumentpiece : this.childPieces) {
                 oceanmonumentpieces$oceanmonumentpiece.getBoundingBox().move(blockpos);
             }
 
@@ -215,24 +_,24 @@
         private List<OceanMonumentPieces.RoomDefinition> generateRoomGraph(RandomSource p_228673_) {
             OceanMonumentPieces.RoomDefinition[] aoceanmonumentpieces$roomdefinition = new OceanMonumentPieces.RoomDefinition[75];
 
-            for(int i = 0; i < 5; ++i) {
-                for(int j = 0; j < 4; ++j) {
+            for (int i = 0; i < 5; i++) {
+                for (int j = 0; j < 4; j++) {
                     int k = 0;
                     int l = getRoomIndex(i, 0, j);
                     aoceanmonumentpieces$roomdefinition[l] = new OceanMonumentPieces.RoomDefinition(l);
                 }
             }
 
-            for(int i2 = 0; i2 < 5; ++i2) {
-                for(int l2 = 0; l2 < 4; ++l2) {
+            for (int i2 = 0; i2 < 5; i2++) {
+                for (int l2 = 0; l2 < 4; l2++) {
                     int k3 = 1;
                     int j4 = getRoomIndex(i2, 1, l2);
                     aoceanmonumentpieces$roomdefinition[j4] = new OceanMonumentPieces.RoomDefinition(j4);
                 }
             }
 
-            for(int j2 = 1; j2 < 4; ++j2) {
-                for(int i3 = 0; i3 < 2; ++i3) {
+            for (int j2 = 1; j2 < 4; j2++) {
+                for (int i3 = 0; i3 < 2; i3++) {
                     int l3 = 2;
                     int k4 = getRoomIndex(j2, 2, i3);
                     aoceanmonumentpieces$roomdefinition[k4] = new OceanMonumentPieces.RoomDefinition(k4);
@@ -241,12 +_,12 @@
 
             this.sourceRoom = aoceanmonumentpieces$roomdefinition[GRIDROOM_SOURCE_INDEX];
 
-            for(int k2 = 0; k2 < 5; ++k2) {
-                for(int j3 = 0; j3 < 5; ++j3) {
-                    for(int i4 = 0; i4 < 3; ++i4) {
+            for (int k2 = 0; k2 < 5; k2++) {
+                for (int j3 = 0; j3 < 5; j3++) {
+                    for (int i4 = 0; i4 < 3; i4++) {
                         int l4 = getRoomIndex(k2, i4, j3);
                         if (aoceanmonumentpieces$roomdefinition[l4] != null) {
-                            for(Direction direction : Direction.values()) {
+                            for (Direction direction : Direction.values()) {
                                 int i1 = k2 + direction.getStepX();
                                 int j1 = i4 + direction.getStepY();
                                 int k1 = j3 + direction.getStepZ();
@@ -288,7 +_,7 @@
             this.coreRoom.connections[Direction.EAST.get3DDataValue()].connections[Direction.NORTH.get3DDataValue()].connections[Direction.UP.get3DDataValue()].claimed = true;
             ObjectArrayList<OceanMonumentPieces.RoomDefinition> objectarraylist = new ObjectArrayList<>();
 
-            for(OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition4 : aoceanmonumentpieces$roomdefinition) {
+            for (OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition4 : aoceanmonumentpieces$roomdefinition) {
                 if (oceanmonumentpieces$roomdefinition4 != null) {
                     oceanmonumentpieces$roomdefinition4.updateOpenings();
                     objectarraylist.add(oceanmonumentpieces$roomdefinition4);
@@ -299,19 +_,19 @@
             Util.shuffle(objectarraylist, p_228673_);
             int i5 = 1;
 
-            for(OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition3 : objectarraylist) {
+            for (OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition3 : objectarraylist) {
                 int j5 = 0;
                 int k5 = 0;
 
-                while(j5 < 2 && k5 < 5) {
-                    ++k5;
+                while (j5 < 2 && k5 < 5) {
+                    k5++;
                     int l5 = p_228673_.nextInt(6);
                     if (oceanmonumentpieces$roomdefinition3.hasOpening[l5]) {
                         int i6 = Direction.from3DDataValue(l5).getOpposite().get3DDataValue();
                         oceanmonumentpieces$roomdefinition3.hasOpening[l5] = false;
                         oceanmonumentpieces$roomdefinition3.connections[l5].hasOpening[i6] = false;
                         if (oceanmonumentpieces$roomdefinition3.findSource(i5++) && oceanmonumentpieces$roomdefinition3.connections[l5].findSource(i5++)) {
-                            ++j5;
+                            j5++;
                         } else {
                             oceanmonumentpieces$roomdefinition3.hasOpening[l5] = true;
                             oceanmonumentpieces$roomdefinition3.connections[l5].hasOpening[i6] = true;
@@ -347,10 +_,10 @@
             this.generateMiddleWall(p_228659_, p_228662_, p_228663_);
             this.generateUpperWall(p_228659_, p_228662_, p_228663_);
 
-            for(int j = 0; j < 7; ++j) {
+            for (int j = 0; j < 7; j++) {
                 int k = 0;
 
-                while(k < 7) {
+                while (k < 7) {
                     if (k == 0 && j == 3) {
                         k = 6;
                     }
@@ -358,8 +_,8 @@
                     int l = j * 9;
                     int i1 = k * 9;
 
-                    for(int j1 = 0; j1 < 4; ++j1) {
-                        for(int k1 = 0; k1 < 4; ++k1) {
+                    for (int j1 = 0; j1 < 4; j1++) {
+                        for (int k1 = 0; k1 < 4; k1++) {
                             this.placeBlock(p_228659_, BASE_LIGHT, l + j1, 0, i1 + k1, p_228663_);
                             this.fillColumnDown(p_228659_, BASE_LIGHT, l + j1, -1, i1 + k1, p_228663_);
                         }
@@ -368,19 +_,19 @@
                     if (j != 0 && j != 6) {
                         k += 6;
                     } else {
-                        ++k;
+                        k++;
                     }
                 }
             }
 
-            for(int l1 = 0; l1 < 5; ++l1) {
+            for (int l1 = 0; l1 < 5; l1++) {
                 this.generateWaterBox(p_228659_, p_228663_, -1 - l1, 0 + l1 * 2, -1 - l1, -1 - l1, 23, 58 + l1);
                 this.generateWaterBox(p_228659_, p_228663_, 58 + l1, 0 + l1 * 2, -1 - l1, 58 + l1, 23, 58 + l1);
                 this.generateWaterBox(p_228659_, p_228663_, 0 - l1, 0 + l1 * 2, -1 - l1, 57 + l1, 23, -1 - l1);
                 this.generateWaterBox(p_228659_, p_228663_, 0 - l1, 0 + l1 * 2, 58 + l1, 57 + l1, 23, 58 + l1);
             }
 
-            for(OceanMonumentPieces.OceanMonumentPiece oceanmonumentpieces$oceanmonumentpiece : this.childPieces) {
+            for (OceanMonumentPieces.OceanMonumentPiece oceanmonumentpieces$oceanmonumentpiece : this.childPieces) {
                 if (oceanmonumentpieces$oceanmonumentpiece.getBoundingBox().intersects(p_228663_)) {
                     oceanmonumentpieces$oceanmonumentpiece.postProcess(p_228659_, p_228660_, p_228661_, p_228662_, p_228663_, p_228664_, p_228665_);
                 }
@@ -393,7 +_,7 @@
                 this.generateBox(p_228669_, p_228671_, p_228668_ + 0, 0, 0, p_228668_ + 24, 0, 20, BASE_GRAY, BASE_GRAY, false);
                 this.generateWaterBox(p_228669_, p_228671_, p_228668_ + 0, 1, 0, p_228668_ + 24, 10, 20);
 
-                for(int j = 0; j < 4; ++j) {
+                for (int j = 0; j < 4; j++) {
                     this.generateBox(p_228669_, p_228671_, p_228668_ + j, j + 1, j, p_228668_ + j, j + 1, 20, BASE_LIGHT, BASE_LIGHT, false);
                     this.generateBox(p_228669_, p_228671_, p_228668_ + j + 7, j + 5, j + 7, p_228668_ + j + 7, j + 5, 20, BASE_LIGHT, BASE_LIGHT, false);
                     this.generateBox(p_228669_, p_228671_, p_228668_ + 17 - j, j + 5, j + 7, p_228668_ + 17 - j, j + 5, 20, BASE_LIGHT, BASE_LIGHT, false);
@@ -412,15 +_,15 @@
                 int j1 = p_228668_ + (p_228667_ ? 19 : 5);
                 int k = p_228668_ + (p_228667_ ? 5 : 19);
 
-                for(int l = 20; l >= 5; l -= 3) {
+                for (int l = 20; l >= 5; l -= 3) {
                     this.placeBlock(p_228669_, DOT_DECO_DATA, j1, 5, l, p_228671_);
                 }
 
-                for(int k1 = 19; k1 >= 7; k1 -= 3) {
+                for (int k1 = 19; k1 >= 7; k1 -= 3) {
                     this.placeBlock(p_228669_, DOT_DECO_DATA, k, 5, k1, p_228671_);
                 }
 
-                for(int l1 = 0; l1 < 4; ++l1) {
+                for (int l1 = 0; l1 < 4; l1++) {
                     int i1 = p_228667_ ? p_228668_ + 24 - (17 - l1 * 3) : p_228668_ + 17 - l1 * 3;
                     this.placeBlock(p_228669_, DOT_DECO_DATA, i1, 5, 5, p_228671_);
                 }
@@ -435,7 +_,7 @@
             if (this.chunkIntersects(p_228657_, 22, 5, 35, 17)) {
                 this.generateWaterBox(p_228655_, p_228657_, 25, 0, 0, 32, 8, 20);
 
-                for(int i = 0; i < 4; ++i) {
+                for (int i = 0; i < 4; i++) {
                     this.generateBox(p_228655_, p_228657_, 24, 2, 5 + i * 4, 24, 4, 5 + i * 4, BASE_LIGHT, BASE_LIGHT, false);
                     this.generateBox(p_228655_, p_228657_, 22, 4, 5 + i * 4, 23, 4, 5 + i * 4, BASE_LIGHT, BASE_LIGHT, false);
                     this.placeBlock(p_228655_, BASE_LIGHT, 25, 5, 5 + i * 4, p_228657_);
@@ -474,12 +_,12 @@
                 this.placeBlock(p_228675_, BASE_LIGHT, 25, 1, 21, p_228677_);
                 this.placeBlock(p_228675_, BASE_LIGHT, 32, 1, 21, p_228677_);
 
-                for(int i = 0; i < 7; ++i) {
+                for (int i = 0; i < 7; i++) {
                     this.placeBlock(p_228675_, BASE_BLACK, 28 - i, 6 + i, 21, p_228677_);
                     this.placeBlock(p_228675_, BASE_BLACK, 29 + i, 6 + i, 21, p_228677_);
                 }
 
-                for(int j = 0; j < 4; ++j) {
+                for (int j = 0; j < 4; j++) {
                     this.placeBlock(p_228675_, BASE_BLACK, 28 - j, 9 + j, 21, p_228677_);
                     this.placeBlock(p_228675_, BASE_BLACK, 29 + j, 9 + j, 21, p_228677_);
                 }
@@ -487,7 +_,7 @@
                 this.placeBlock(p_228675_, BASE_BLACK, 28, 12, 21, p_228677_);
                 this.placeBlock(p_228675_, BASE_BLACK, 29, 12, 21, p_228677_);
 
-                for(int k = 0; k < 3; ++k) {
+                for (int k = 0; k < 3; k++) {
                     this.placeBlock(p_228675_, BASE_BLACK, 22 - k * 2, 8, 21, p_228677_);
                     this.placeBlock(p_228675_, BASE_BLACK, 22 - k * 2, 9, 21, p_228677_);
                     this.placeBlock(p_228675_, BASE_BLACK, 35 + k * 2, 8, 21, p_228677_);
@@ -515,7 +_,7 @@
                 this.generateBox(p_228679_, p_228681_, 21, 0, 22, 36, 0, 36, BASE_GRAY, BASE_GRAY, false);
                 this.generateWaterBox(p_228679_, p_228681_, 21, 1, 22, 36, 23, 36);
 
-                for(int i = 0; i < 4; ++i) {
+                for (int i = 0; i < 4; i++) {
                     this.generateBox(p_228679_, p_228681_, 21 + i, 13 + i, 21 + i, 36 - i, 13 + i, 21 + i, BASE_LIGHT, BASE_LIGHT, false);
                     this.generateBox(p_228679_, p_228681_, 21 + i, 13 + i, 36 - i, 36 - i, 13 + i, 36 - i, BASE_LIGHT, BASE_LIGHT, false);
                     this.generateBox(p_228679_, p_228681_, 21 + i, 13 + i, 22 + i, 21 + i, 13 + i, 35 - i, BASE_LIGHT, BASE_LIGHT, false);
@@ -552,17 +_,17 @@
                 this.generateWaterBox(p_228683_, p_228685_, 0, 1, 21, 6, 7, 57);
                 this.generateBox(p_228683_, p_228685_, 4, 4, 21, 6, 4, 53, BASE_GRAY, BASE_GRAY, false);
 
-                for(int i = 0; i < 4; ++i) {
+                for (int i = 0; i < 4; i++) {
                     this.generateBox(p_228683_, p_228685_, i, i + 1, 21, i, i + 1, 57 - i, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
-                for(int j = 23; j < 53; j += 3) {
+                for (int j = 23; j < 53; j += 3) {
                     this.placeBlock(p_228683_, DOT_DECO_DATA, 5, 5, j, p_228685_);
                 }
 
                 this.placeBlock(p_228683_, DOT_DECO_DATA, 5, 5, 52, p_228685_);
 
-                for(int k = 0; k < 4; ++k) {
+                for (int k = 0; k < 4; k++) {
                     this.generateBox(p_228683_, p_228685_, k, k + 1, 21, k, k + 1, 57 - k, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
@@ -575,11 +_,11 @@
                 this.generateWaterBox(p_228683_, p_228685_, 51, 1, 21, 57, 7, 57);
                 this.generateBox(p_228683_, p_228685_, 51, 4, 21, 53, 4, 53, BASE_GRAY, BASE_GRAY, false);
 
-                for(int l = 0; l < 4; ++l) {
+                for (int l = 0; l < 4; l++) {
                     this.generateBox(p_228683_, p_228685_, 57 - l, l + 1, 21, 57 - l, l + 1, 57 - l, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
-                for(int i1 = 23; i1 < 53; i1 += 3) {
+                for (int i1 = 23; i1 < 53; i1 += 3) {
                     this.placeBlock(p_228683_, DOT_DECO_DATA, 52, 5, i1, p_228685_);
                 }
 
@@ -592,7 +_,7 @@
                 this.generateBox(p_228683_, p_228685_, 7, 0, 51, 50, 0, 57, BASE_GRAY, BASE_GRAY, false);
                 this.generateWaterBox(p_228683_, p_228685_, 7, 1, 51, 50, 10, 57);
 
-                for(int j1 = 0; j1 < 4; ++j1) {
+                for (int j1 = 0; j1 < 4; j1++) {
                     this.generateBox(p_228683_, p_228685_, j1 + 1, j1 + 1, 57 - j1, 56 - j1, j1 + 1, 57 - j1, BASE_LIGHT, BASE_LIGHT, false);
                 }
             }
@@ -604,11 +_,11 @@
                 this.generateWaterBox(p_228687_, p_228689_, 7, 1, 21, 13, 10, 50);
                 this.generateBox(p_228687_, p_228689_, 11, 8, 21, 13, 8, 53, BASE_GRAY, BASE_GRAY, false);
 
-                for(int i = 0; i < 4; ++i) {
+                for (int i = 0; i < 4; i++) {
                     this.generateBox(p_228687_, p_228689_, i + 7, i + 5, 21, i + 7, i + 5, 54, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
-                for(int j = 21; j <= 45; j += 3) {
+                for (int j = 21; j <= 45; j += 3) {
                     this.placeBlock(p_228687_, DOT_DECO_DATA, 12, 9, j, p_228689_);
                 }
             }
@@ -618,11 +_,11 @@
                 this.generateWaterBox(p_228687_, p_228689_, 44, 1, 21, 50, 10, 50);
                 this.generateBox(p_228687_, p_228689_, 44, 8, 21, 46, 8, 53, BASE_GRAY, BASE_GRAY, false);
 
-                for(int k = 0; k < 4; ++k) {
+                for (int k = 0; k < 4; k++) {
                     this.generateBox(p_228687_, p_228689_, 50 - k, k + 5, 21, 50 - k, k + 5, 54, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
-                for(int l = 21; l <= 45; l += 3) {
+                for (int l = 21; l <= 45; l += 3) {
                     this.placeBlock(p_228687_, DOT_DECO_DATA, 45, 9, l, p_228689_);
                 }
             }
@@ -631,7 +_,7 @@
                 this.generateBox(p_228687_, p_228689_, 14, 0, 44, 43, 0, 50, BASE_GRAY, BASE_GRAY, false);
                 this.generateWaterBox(p_228687_, p_228689_, 14, 1, 44, 43, 10, 50);
 
-                for(int i1 = 12; i1 <= 45; i1 += 3) {
+                for (int i1 = 12; i1 <= 45; i1 += 3) {
                     this.placeBlock(p_228687_, DOT_DECO_DATA, i1, 9, 45, p_228689_);
                     this.placeBlock(p_228687_, DOT_DECO_DATA, i1, 9, 52, p_228689_);
                     if (i1 == 12 || i1 == 18 || i1 == 24 || i1 == 33 || i1 == 39 || i1 == 45) {
@@ -648,7 +_,7 @@
                     }
                 }
 
-                for(int j1 = 0; j1 < 3; ++j1) {
+                for (int j1 = 0; j1 < 3; j1++) {
                     this.generateBox(p_228687_, p_228689_, 8 + j1, 5 + j1, 54, 49 - j1, 5 + j1, 54, BASE_GRAY, BASE_GRAY, false);
                 }
 
@@ -664,11 +_,11 @@
                 this.generateBox(p_228691_, p_228693_, 18, 12, 22, 20, 12, 39, BASE_GRAY, BASE_GRAY, false);
                 this.generateBox(p_228691_, p_228693_, 18, 12, 21, 20, 12, 21, BASE_LIGHT, BASE_LIGHT, false);
 
-                for(int i = 0; i < 4; ++i) {
+                for (int i = 0; i < 4; i++) {
                     this.generateBox(p_228691_, p_228693_, i + 14, i + 9, 21, i + 14, i + 9, 43 - i, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
-                for(int j = 23; j <= 39; j += 3) {
+                for (int j = 23; j <= 39; j += 3) {
                     this.placeBlock(p_228691_, DOT_DECO_DATA, 19, 13, j, p_228693_);
                 }
             }
@@ -679,11 +_,11 @@
                 this.generateBox(p_228691_, p_228693_, 37, 12, 22, 39, 12, 39, BASE_GRAY, BASE_GRAY, false);
                 this.generateBox(p_228691_, p_228693_, 37, 12, 21, 39, 12, 21, BASE_LIGHT, BASE_LIGHT, false);
 
-                for(int k = 0; k < 4; ++k) {
+                for (int k = 0; k < 4; k++) {
                     this.generateBox(p_228691_, p_228693_, 43 - k, k + 9, 21, 43 - k, k + 9, 43 - k, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
-                for(int l = 23; l <= 39; l += 3) {
+                for (int l = 23; l <= 39; l += 3) {
                     this.placeBlock(p_228691_, DOT_DECO_DATA, 38, 13, l, p_228693_);
                 }
             }
@@ -693,11 +_,11 @@
                 this.generateWaterBox(p_228691_, p_228693_, 21, 1, 37, 36, 14, 43);
                 this.generateBox(p_228691_, p_228693_, 21, 12, 37, 36, 12, 39, BASE_GRAY, BASE_GRAY, false);
 
-                for(int i1 = 0; i1 < 4; ++i1) {
+                for (int i1 = 0; i1 < 4; i1++) {
                     this.generateBox(p_228691_, p_228693_, 15 + i1, i1 + 9, 43 - i1, 42 - i1, i1 + 9, 43 - i1, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
-                for(int j1 = 21; j1 <= 36; j1 += 3) {
+                for (int j1 = 21; j1 <= 36; j1 += 3) {
                     this.placeBlock(p_228691_, DOT_DECO_DATA, j1, 13, 38, p_228693_);
                 }
             }
@@ -737,13 +_,13 @@
             this.generateBox(p_228704_, p_228708_, 1, 7, 0, 15, 7, 0, blockstate, blockstate, false);
             this.generateBox(p_228704_, p_228708_, 1, 7, 15, 14, 7, 15, blockstate, blockstate, false);
 
-            for(int k = 1; k <= 6; ++k) {
+            for (int k = 1; k <= 6; k++) {
                 blockstate = BASE_LIGHT;
                 if (k == 2 || k == 6) {
                     blockstate = BASE_GRAY;
                 }
 
-                for(int j = 0; j <= 15; j += 15) {
+                for (int j = 0; j <= 15; j += 15) {
                     this.generateBox(p_228704_, p_228708_, j, k, 0, j, k, 1, blockstate, blockstate, false);
                     this.generateBox(p_228704_, p_228708_, j, k, 6, j, k, 9, blockstate, blockstate, false);
                     this.generateBox(p_228704_, p_228708_, j, k, 14, j, k, 15, blockstate, blockstate, false);
@@ -758,8 +_,8 @@
             this.generateBox(p_228704_, p_228708_, 6, 3, 6, 9, 6, 9, BASE_BLACK, BASE_BLACK, false);
             this.generateBox(p_228704_, p_228708_, 7, 4, 7, 8, 5, 8, Blocks.GOLD_BLOCK.defaultBlockState(), Blocks.GOLD_BLOCK.defaultBlockState(), false);
 
-            for(int l = 3; l <= 6; l += 3) {
-                for(int i1 = 6; i1 <= 9; i1 += 3) {
+            for (int l = 3; l <= 6; l += 3) {
+                for (int i1 = 6; i1 <= 9; i1 += 3) {
                     this.placeBlock(p_228704_, LAMP_BLOCK, i1, l, 6, p_228708_);
                     this.placeBlock(p_228704_, LAMP_BLOCK, i1, l, 9, p_228708_);
                 }
@@ -911,7 +_,7 @@
                 this.generateBoxOnFillOnly(p_228730_, p_228734_, 8, 8, 1, 14, 8, 6, BASE_GRAY);
             }
 
-            for(int i = 1; i <= 7; ++i) {
+            for (int i = 1; i <= 7; i++) {
                 BlockState blockstate = BASE_LIGHT;
                 if (i == 2 || i == 6) {
                     blockstate = BASE_GRAY;
@@ -1038,7 +_,7 @@
             this.generateBox(p_228743_, p_228747_, 6, 4, 5, 6, 4, 5, BASE_LIGHT, BASE_LIGHT, false);
             OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.roomDefinition;
 
-            for(int i = 1; i <= 5; i += 4) {
+            for (int i = 1; i <= 5; i += 4) {
                 int j = 0;
                 if (oceanmonumentpieces$roomdefinition1.hasOpening[Direction.SOUTH.get3DDataValue()]) {
                     this.generateBox(p_228743_, p_228747_, 2, i, j, 2, i + 2, j, BASE_LIGHT, BASE_LIGHT, false);
@@ -1122,7 +_,7 @@
                 this.generateBoxOnFillOnly(p_228756_, p_228760_, 1, 8, 8, 6, 8, 14, BASE_GRAY);
             }
 
-            for(int i = 1; i <= 7; ++i) {
+            for (int i = 1; i <= 7; i++) {
                 BlockState blockstate = BASE_LIGHT;
                 if (i == 2 || i == 6) {
                     blockstate = BASE_GRAY;
@@ -1134,7 +_,7 @@
                 this.generateBox(p_228756_, p_228760_, 1, i, 15, 6, i, 15, blockstate, blockstate, false);
             }
 
-            for(int j = 1; j <= 7; ++j) {
+            for (int j = 1; j <= 7; j++) {
                 BlockState blockstate1 = BASE_BLACK;
                 if (j == 2 || j == 6) {
                     blockstate1 = LAMP_BLOCK;
@@ -1372,7 +_,7 @@
             this.generateBox(p_228795_, p_228799_, 1, 0, 0, 12, 0, 0, BASE_LIGHT, BASE_LIGHT, false);
             this.generateBox(p_228795_, p_228799_, 1, 0, 13, 12, 0, 13, BASE_LIGHT, BASE_LIGHT, false);
 
-            for(int i = 2; i <= 11; i += 3) {
+            for (int i = 2; i <= 11; i += 3) {
                 this.placeBlock(p_228795_, LAMP_BLOCK, 0, 0, i, p_228799_);
                 this.placeBlock(p_228795_, LAMP_BLOCK, 13, 0, i, p_228799_);
                 this.placeBlock(p_228795_, LAMP_BLOCK, i, 0, 0, p_228799_);
@@ -1390,8 +_,8 @@
             this.generateBox(p_228795_, p_228799_, 6, 0, 10, 7, 0, 10, BASE_BLACK, BASE_BLACK, false);
             int l = 3;
 
-            for(int j = 0; j < 2; ++j) {
-                for(int k = 2; k <= 8; k += 3) {
+            for (int j = 0; j < 2; j++) {
+                for (int k = 2; k <= 8; k += 3) {
                     this.generateBox(p_228795_, p_228799_, l, 0, k, l, 2, k, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
@@ -1468,7 +_,7 @@
             int k = i / 5 % 5;
             int l = i / 25;
             BoundingBox boundingbox = makeBoundingBox(0, 0, 0, p_228875_, p_228877_ * 8, p_228878_ * 4, p_228879_ * 8);
-            switch(p_228875_) {
+            switch (p_228875_) {
                 case NORTH:
                     boundingbox.move(j * 8, l * 4, -(k + p_228879_) * 8 + 1);
                     break;
@@ -1497,9 +_,9 @@
         protected void generateWaterBox(
             WorldGenLevel p_228881_, BoundingBox p_228882_, int p_228883_, int p_228884_, int p_228885_, int p_228886_, int p_228887_, int p_228888_
         ) {
-            for(int i = p_228884_; i <= p_228887_; ++i) {
-                for(int j = p_228883_; j <= p_228886_; ++j) {
-                    for(int k = p_228885_; k <= p_228888_; ++k) {
+            for (int i = p_228884_; i <= p_228887_; i++) {
+                for (int j = p_228883_; j <= p_228886_; j++) {
+                    for (int k = p_228885_; k <= p_228888_; k++) {
                         BlockState blockstate = this.getBlock(p_228881_, j, i, k, p_228882_);
                         if (!FILL_KEEP.contains(blockstate.getBlock())) {
                             if (this.getWorldY(i) >= p_228881_.getSeaLevel() && blockstate != FILL_BLOCK) {
@@ -1539,9 +_,9 @@
             int p_228857_,
             BlockState p_228858_
         ) {
-            for(int i = p_228853_; i <= p_228856_; ++i) {
-                for(int j = p_228852_; j <= p_228855_; ++j) {
-                    for(int k = p_228854_; k <= p_228857_; ++k) {
+            for (int i = p_228853_; i <= p_228856_; i++) {
+                for (int j = p_228852_; j <= p_228855_; j++) {
+                    for (int k = p_228854_; k <= p_228857_; k++) {
                         if (this.getBlock(p_228850_, j, i, k, p_228851_) == FILL_BLOCK) {
                             this.placeBlock(p_228850_, p_228858_, j, i, k, p_228851_);
                         }
@@ -1776,8 +_,8 @@
                 this.generateBoxOnFillOnly(p_228914_, p_228918_, 1, 4, 1, 6, 4, 6, BASE_GRAY);
             }
 
-            for(int i = 1; i <= 6; ++i) {
-                for(int j = 1; j <= 6; ++j) {
+            for (int i = 1; i <= 6; i++) {
+                for (int j = 1; j <= 6; j++) {
                     if (p_228917_.nextInt(3) != 0) {
                         int k = 2 + (p_228917_.nextInt(4) == 0 ? 0 : 1);
                         BlockState blockstate = Blocks.WET_SPONGE.defaultBlockState();
@@ -1831,7 +_,7 @@
             BlockPos p_228935_
         ) {
             if (this.mainDesign == 0) {
-                for(int i = 0; i < 4; ++i) {
+                for (int i = 0; i < 4; i++) {
                     this.generateBox(p_228929_, p_228933_, 10 - i, 3 - i, 20 - i, 12 + i, 3 - i, 20, BASE_LIGHT, BASE_LIGHT, false);
                 }
 
@@ -1849,7 +_,7 @@
                 this.generateBox(p_228929_, p_228933_, 8, 0, 10, 8, 0, 12, BASE_BLACK, BASE_BLACK, false);
                 this.generateBox(p_228929_, p_228933_, 14, 0, 10, 14, 0, 12, BASE_BLACK, BASE_BLACK, false);
 
-                for(int i1 = 18; i1 >= 7; i1 -= 3) {
+                for (int i1 = 18; i1 >= 7; i1 -= 3) {
                     this.placeBlock(p_228929_, LAMP_BLOCK, 6, 3, i1, p_228933_);
                     this.placeBlock(p_228929_, LAMP_BLOCK, 16, 3, i1, p_228933_);
                 }
@@ -1885,7 +_,7 @@
                 int j = 20;
                 int k = 5;
 
-                for(int l = 0; l < 2; ++l) {
+                for (int l = 0; l < 2; l++) {
                     this.placeBlock(p_228929_, BASE_LIGHT, j1, 6, 20, p_228933_);
                     this.placeBlock(p_228929_, LAMP_BLOCK, j1, 5, 20, p_228933_);
                     this.placeBlock(p_228929_, BASE_LIGHT, j1, 4, 20, p_228933_);
@@ -1895,7 +_,7 @@
                 this.generateBox(p_228929_, p_228933_, 7, 3, 7, 15, 3, 14, BASE_LIGHT, BASE_LIGHT, false);
                 int b0 = 10;
 
-                for(int k1 = 0; k1 < 2; ++k1) {
+                for (int k1 = 0; k1 < 2; k1++) {
                     this.generateBox(p_228929_, p_228933_, b0, 0, 10, b0, 6, 10, BASE_LIGHT, BASE_LIGHT, false);
                     this.generateBox(p_228929_, p_228933_, b0, 0, 12, b0, 6, 12, BASE_LIGHT, BASE_LIGHT, false);
                     this.placeBlock(p_228929_, LAMP_BLOCK, b0, 0, 10, p_228933_);
@@ -1907,7 +_,7 @@
 
                 b0 = 8;
 
-                for(int l1 = 0; l1 < 2; ++l1) {
+                for (int l1 = 0; l1 < 2; l1++) {
                     this.generateBox(p_228929_, p_228933_, b0, 0, 7, b0, 2, 7, BASE_LIGHT, BASE_LIGHT, false);
                     this.generateBox(p_228929_, p_228933_, b0, 0, 14, b0, 2, 14, BASE_LIGHT, BASE_LIGHT, false);
                     b0 = 14;
@@ -1938,7 +_,7 @@
         }
 
         public void updateOpenings() {
-            for(int i = 0; i < 6; ++i) {
+            for (int i = 0; i < 6; i++) {
                 this.hasOpening[i] = this.connections[i] != null;
             }
         }
@@ -1949,7 +_,7 @@
             } else {
                 this.scanIndex = p_228946_;
 
-                for(int i = 0; i < 6; ++i) {
+                for (int i = 0; i < 6; i++) {
                     if (this.connections[i] != null
                         && this.hasOpening[i]
                         && this.connections[i].scanIndex != p_228946_
@@ -1969,9 +_,9 @@
         public int countOpenings() {
             int i = 0;
 
-            for(int j = 0; j < 6; ++j) {
+            for (int j = 0; j < 6; j++) {
                 if (this.hasOpening[j]) {
-                    ++i;
+                    i++;
                 }
             }
 
