--- a/net/minecraft/world/level/levelgen/structure/structures/OceanMonumentStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/OceanMonumentStructure.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.structures;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.Objects;
 import java.util.Optional;
 import net.minecraft.core.Direction;
@@ -20,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.pieces.StructurePiecesBuilder;
 
 public class OceanMonumentStructure extends Structure {
-    public static final Codec<OceanMonumentStructure> CODEC = simpleCodec(OceanMonumentStructure::new);
+    public static final MapCodec<OceanMonumentStructure> CODEC = simpleCodec(OceanMonumentStructure::new);
 
     public OceanMonumentStructure(Structure.StructureSettings p_228955_) {
         super(p_228955_);
@@ -31,7 +_,7 @@
         int i = p_228964_.chunkPos().getBlockX(9);
         int j = p_228964_.chunkPos().getBlockZ(9);
 
-        for(Holder<Biome> holder : p_228964_.biomeSource()
+        for (Holder<Biome> holder : p_228964_.biomeSource()
             .getBiomesWithin(i, p_228964_.chunkGenerator().getSeaLevel(), j, 29, p_228964_.randomState().sampler())) {
             if (!holder.is(BiomeTags.REQUIRED_OCEAN_MONUMENT_SURROUNDING)) {
                 return Optional.empty();
