--- a/net/minecraft/world/level/levelgen/structure/structures/ShipwreckStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/ShipwreckStructure.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.structures;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Optional;
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.pieces.StructurePiecesBuilder;
 
 public class ShipwreckStructure extends Structure {
-    public static final Codec<ShipwreckStructure> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<ShipwreckStructure> CODEC = RecordCodecBuilder.mapCodec(
         p_229401_ -> p_229401_.group(settingsCodec(p_229401_), Codec.BOOL.fieldOf("is_beached").forGetter(p_229399_ -> p_229399_.isBeached))
                 .apply(p_229401_, ShipwreckStructure::new)
     );
