--- a/net/minecraft/world/level/levelgen/structure/structures/ShipwreckPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/ShipwreckPieces.java
@@ -137,7 +_,7 @@
             } else {
                 BlockPos blockpos = this.templatePosition.offset(vec3i.getX() - 1, 0, vec3i.getZ() - 1);
 
-                for(BlockPos blockpos1 : BlockPos.betweenClosed(this.templatePosition, blockpos)) {
+                for (BlockPos blockpos1 : BlockPos.betweenClosed(this.templatePosition, blockpos)) {
                     int l = p_229363_.getHeight(heightmap$types, blockpos1.getX(), blockpos1.getZ());
                     j += l;
                     i = Math.min(i, l);
