--- a/net/minecraft/world/level/levelgen/structure/structures/SwampHutPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/SwampHutPiece.java
@@ -88,8 +_,8 @@
             this.placeBlock(p_229961_, blockstate3.setValue(StairBlock.SHAPE, StairsShape.OUTER_LEFT), 0, 4, 8, p_229965_);
             this.placeBlock(p_229961_, blockstate3.setValue(StairBlock.SHAPE, StairsShape.OUTER_RIGHT), 6, 4, 8, p_229965_);
 
-            for(int i = 2; i <= 7; i += 5) {
-                for(int j = 1; j <= 5; j += 4) {
+            for (int i = 2; i <= 7; i += 5) {
+                for (int j = 1; j <= 5; j += 4) {
                     this.fillColumnDown(p_229961_, Blocks.OAK_LOG.defaultBlockState(), j, -1, i, p_229965_);
                 }
             }
