--- a/net/minecraft/world/level/levelgen/structure/structures/WoodlandMansionStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/WoodlandMansionStructure.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.structures;
 
 import com.google.common.collect.Lists;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import java.util.Optional;
 import net.minecraft.core.BlockPos;
@@ -19,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.pieces.StructurePiecesBuilder;
 
 public class WoodlandMansionStructure extends Structure {
-    public static final Codec<WoodlandMansionStructure> CODEC = simpleCodec(WoodlandMansionStructure::new);
+    public static final MapCodec<WoodlandMansionStructure> CODEC = simpleCodec(WoodlandMansionStructure::new);
 
     public WoodlandMansionStructure(Structure.StructureSettings p_230225_) {
         super(p_230225_);
@@ -55,13 +_,13 @@
         BoundingBox boundingbox = p_230233_.calculateBoundingBox();
         int j = boundingbox.minY();
 
-        for(int k = p_230231_.minX(); k <= p_230231_.maxX(); ++k) {
-            for(int l = p_230231_.minZ(); l <= p_230231_.maxZ(); ++l) {
+        for (int k = p_230231_.minX(); k <= p_230231_.maxX(); k++) {
+            for (int l = p_230231_.minZ(); l <= p_230231_.maxZ(); l++) {
                 blockpos$mutableblockpos.set(k, j, l);
                 if (!p_230227_.isEmptyBlock(blockpos$mutableblockpos)
                     && boundingbox.isInside(blockpos$mutableblockpos)
                     && p_230233_.isInsidePiece(blockpos$mutableblockpos)) {
-                    for(int i1 = j - 1; i1 > i; --i1) {
+                    for (int i1 = j - 1; i1 > i; i1--) {
                         blockpos$mutableblockpos.setY(i1);
                         if (!p_230227_.isEmptyBlock(blockpos$mutableblockpos) && !p_230227_.getBlockState(blockpos$mutableblockpos).liquid()) {
                             break;
