--- a/net/minecraft/util/profiling/jfr/parse/JfrStatsParser.java
+++ b/net/minecraft/util/profiling/jfr/parse/JfrStatsParser.java
@@ -116,7 +_,7 @@
             }
 
             String s = p_325662_.getEventType().getName();
-            switch(s) {
+            switch (s) {
                 case "minecraft.ChunkGeneration":
                     this.chunkGenStats.add(ChunkGenStat.from(p_325662_));
                     break;
@@ -154,7 +_,7 @@
                     this.appendFileIO(p_325662_, this.fileReads, "bytesRead");
                     break;
                 case "jdk.GarbageCollection":
-                    ++this.garbageCollections;
+                    this.garbageCollections++;
                     this.gcTotalDuration = this.gcTotalDuration.plus(p_325662_.getDuration());
             }
         });
@@ -186,7 +_,7 @@
 
         public void increment(int p_185477_) {
             this.totalSize += (long)p_185477_;
-            ++this.count;
+            this.count++;
         }
 
         public IoSummary.CountAndSize toCountAndSize() {
