--- a/net/minecraft/nbt/ByteArrayTag.java
+++ b/net/minecraft/nbt/ByteArrayTag.java
@@ -56,7 +_,7 @@
     private static byte[] toArray(List<Byte> p_128207_) {
         byte[] abyte = new byte[p_128207_.size()];
 
-        for(int i = 0; i < p_128207_.size(); ++i) {
+        for (int i = 0; i < p_128207_.size(); i++) {
             Byte obyte = p_128207_.get(i);
             abyte[i] = obyte == null ? 0 : obyte;
         }
@@ -99,11 +_,7 @@
 
     @Override
     public boolean equals(Object p_128233_) {
-        if (this == p_128233_) {
-            return true;
-        } else {
-            return p_128233_ instanceof ByteArrayTag && Arrays.equals(this.data, ((ByteArrayTag)p_128233_).data);
-        }
+        return this == p_128233_ ? true : p_128233_ instanceof ByteArrayTag && Arrays.equals(this.data, ((ByteArrayTag)p_128233_).data);
     }
 
     @Override
