--- a/net/minecraft/nbt/ListTag.java
+++ b/net/minecraft/nbt/ListTag.java
@@ -36,7 +_,7 @@
                 TagType<?> tagtype = TagTypes.getType(b0);
                 List<Tag> list = Lists.newArrayListWithCapacity(i);
 
-                for(int j = 0; j < i; ++j) {
+                for (int j = 0; j < i; j++) {
                     list.add(tagtype.load(p_302382_, p_302349_));
                 }
 
@@ -62,7 +_,7 @@
             p_302361_.accountBytes(37L);
             TagType<?> tagtype = TagTypes.getType(p_302369_.readByte());
             int i = p_302369_.readInt();
-            switch(p_302372_.visitList(tagtype, i)) {
+            switch (p_302372_.visitList(tagtype, i)) {
                 case HALT:
                     return StreamTagVisitor.ValueResult.HALT;
                 case BREAK:
@@ -72,10 +_,10 @@
                     p_302361_.accountBytes(4L, (long)i);
                     int j = 0;
 
-                    while(true) {
+                    while (true) {
                         label41: {
                             if (j < i) {
-                                switch(p_302372_.visitElement(tagtype, j)) {
+                                switch (p_302372_.visitElement(tagtype, j)) {
                                     case HALT:
                                         return StreamTagVisitor.ValueResult.HALT;
                                     case BREAK:
@@ -85,7 +_,7 @@
                                         tagtype.skip(p_302369_, p_302361_);
                                         break label41;
                                     default:
-                                        switch(tagtype.parse(p_302369_, p_302372_, p_302361_)) {
+                                        switch (tagtype.parse(p_302369_, p_302372_, p_302361_)) {
                                             case HALT:
                                                 return StreamTagVisitor.ValueResult.HALT;
                                             case BREAK:
@@ -104,7 +_,7 @@
                             return p_302372_.visitContainerEnd();
                         }
 
-                        ++j;
+                        j++;
                     }
             }
         }
@@ -155,7 +_,7 @@
         p_128734_.writeByte(this.type);
         p_128734_.writeInt(this.list.size());
 
-        for(Tag tag : this.list) {
+        for (Tag tag : this.list) {
             tag.write(p_128734_);
         }
     }
@@ -165,7 +_,7 @@
         int i = 37;
         i += 4 * this.list.size();
 
-        for(Tag tag : this.list) {
+        for (Tag tag : this.list) {
             i += tag.sizeInBytes();
         }
 
@@ -367,11 +_,7 @@
 
     @Override
     public boolean equals(Object p_128766_) {
-        if (this == p_128766_) {
-            return true;
-        } else {
-            return p_128766_ instanceof ListTag && Objects.equals(this.list, ((ListTag)p_128766_).list);
-        }
+        return this == p_128766_ ? true : p_128766_ instanceof ListTag && Objects.equals(this.list, ((ListTag)p_128766_).list);
     }
 
     @Override
@@ -397,7 +_,7 @@
 
     @Override
     public StreamTagVisitor.ValueResult accept(StreamTagVisitor p_197487_) {
-        switch(p_197487_.visitList(TagTypes.getType(this.type), this.list.size())) {
+        switch (p_197487_.visitList(TagTypes.getType(this.type), this.list.size())) {
             case HALT:
                 return StreamTagVisitor.ValueResult.HALT;
             case BREAK:
@@ -405,22 +_,22 @@
             default:
                 int i = 0;
 
-                while(i < this.list.size()) {
+                while (i < this.list.size()) {
                     Tag tag = this.list.get(i);
-                    switch(p_197487_.visitElement(tag.getType(), i)) {
+                    switch (p_197487_.visitElement(tag.getType(), i)) {
                         case HALT:
                             return StreamTagVisitor.ValueResult.HALT;
                         case BREAK:
                             return p_197487_.visitContainerEnd();
                         default:
-                            switch(tag.accept(p_197487_)) {
+                            switch (tag.accept(p_197487_)) {
                                 case HALT:
                                     return StreamTagVisitor.ValueResult.HALT;
                                 case BREAK:
                                     return p_197487_.visitContainerEnd();
                             }
                         case SKIP:
-                            ++i;
+                            i++;
                     }
                 }
 
