--- a/net/minecraft/nbt/LongTag.java
+++ b/net/minecraft/nbt/LongTag.java
@@ -77,11 +_,7 @@
 
     @Override
     public boolean equals(Object p_128894_) {
-        if (this == p_128894_) {
-            return true;
-        } else {
-            return p_128894_ instanceof LongTag && this.data == ((LongTag)p_128894_).data;
-        }
+        return this == p_128894_ ? true : p_128894_ instanceof LongTag && this.data == ((LongTag)p_128894_).data;
     }
 
     @Override
@@ -143,7 +_,7 @@
         }
 
         static {
-            for(int i = 0; i < cache.length; ++i) {
+            for (int i = 0; i < cache.length; i++) {
                 cache[i] = new LongTag((long)(-128 + i));
             }
         }
