--- a/net/minecraft/nbt/LongArrayTag.java
+++ b/net/minecraft/nbt/LongArrayTag.java
@@ -26,7 +_,7 @@
             p_302364_.accountBytes(8L, (long)i);
             long[] along = new long[i];
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 along[j] = p_302379_.readLong();
             }
 
@@ -65,7 +_,7 @@
     private static long[] toArray(List<Long> p_128824_) {
         long[] along = new long[p_128824_.size()];
 
-        for(int i = 0; i < p_128824_.size(); ++i) {
+        for (int i = 0; i < p_128824_.size(); i++) {
             Long olong = p_128824_.get(i);
             along[i] = olong == null ? 0L : olong;
         }
@@ -77,7 +_,7 @@
     public void write(DataOutput p_128819_) throws IOException {
         p_128819_.writeInt(this.data.length);
 
-        for(long i : this.data) {
+        for (long i : this.data) {
             p_128819_.writeLong(i);
         }
     }
@@ -110,11 +_,7 @@
 
     @Override
     public boolean equals(Object p_128850_) {
-        if (this == p_128850_) {
-            return true;
-        } else {
-            return p_128850_ instanceof LongArrayTag && Arrays.equals(this.data, ((LongArrayTag)p_128850_).data);
-        }
+        return this == p_128850_ ? true : p_128850_ instanceof LongArrayTag && Arrays.equals(this.data, ((LongArrayTag)p_128850_).data);
     }
 
     @Override
