--- a/net/minecraft/nbt/NbtUtils.java
+++ b/net/minecraft/nbt/NbtUtils.java
@@ -75,7 +_,7 @@
             if (compoundtag1.size() < compoundtag.size()) {
                 return false;
             } else {
-                for(String s : compoundtag.getAllKeys()) {
+                for (String s : compoundtag.getAllKeys()) {
                     Tag tag2 = compoundtag.get(s);
                     if (!compareNbt(tag2, compoundtag1.get(s), p_129238_)) {
                         return false;
@@ -95,10 +_,10 @@
                     return false;
                 }
 
-                for(Tag tag : listtag) {
+                for (Tag tag : listtag) {
                     boolean flag = false;
 
-                    for(Tag tag1 : listtag1) {
+                    for (Tag tag1 : listtag1) {
                         if (compareNbt(tag, tag1, p_129238_)) {
                             flag = true;
                             break;
@@ -160,7 +_,7 @@
                     CompoundTag compoundtag = p_250775_.getCompound("Properties");
                     StateDefinition<Block, BlockState> statedefinition = block.getStateDefinition();
 
-                    for(String s : compoundtag.getAllKeys()) {
+                    for (String s : compoundtag.getAllKeys()) {
                         Property<?> property = statedefinition.getProperty(s);
                         if (property != null) {
                             blockstate = setValueHelper(blockstate, property, s, compoundtag, p_250775_);
@@ -192,7 +_,7 @@
         if (!map.isEmpty()) {
             CompoundTag compoundtag1 = new CompoundTag();
 
-            for(Entry<Property<?>, Comparable<?>> entry : map.entrySet()) {
+            for (Entry<Property<?>, Comparable<?>> entry : map.entrySet()) {
                 Property<?> property = entry.getKey();
                 compoundtag1.putString(property.getName(), getName(property, entry.getValue()));
             }
@@ -210,7 +_,7 @@
         if (!map.isEmpty()) {
             CompoundTag compoundtag1 = new CompoundTag();
 
-            for(Entry<Property<?>, Comparable<?>> entry : map.entrySet()) {
+            for (Entry<Property<?>, Comparable<?>> entry : map.entrySet()) {
                 Property<?> property = entry.getKey();
                 compoundtag1.putString(property.getName(), getName(property, entry.getValue()));
             }
@@ -234,7 +_,7 @@
     }
 
     public static StringBuilder prettyPrint(StringBuilder p_178027_, Tag p_178028_, int p_178029_, boolean p_178030_) {
-        switch(p_178028_.getId()) {
+        switch (p_178028_.getId()) {
             case 0:
                 break;
             case 1:
@@ -254,7 +_,7 @@
                 if (p_178030_) {
                     indent(p_178029_ + 1, p_178027_);
 
-                    for(int i2 = 0; i2 < abyte.length; ++i2) {
+                    for (int i2 = 0; i2 < abyte.length; i2++) {
                         if (i2 != 0) {
                             p_178027_.append(',');
                         }
@@ -287,7 +_,7 @@
                     p_178027_.append('\n');
                 }
 
-                for(int i3 = 0; i3 < k; ++i3) {
+                for (int i3 = 0; i3 < k; i3++) {
                     if (i3 != 0) {
                         p_178027_.append(",\n");
                     }
@@ -315,7 +_,7 @@
                 int i1 = list.stream().mapToInt(String::length).max().orElse(0);
                 String s = Strings.repeat(" ", i1);
 
-                for(int l2 = 0; l2 < list.size(); ++l2) {
+                for (int l2 = 0; l2 < list.size(); l2++) {
                     if (l2 != 0) {
                         p_178027_.append(",\n");
                     }
@@ -336,7 +_,7 @@
                 int[] aint = intarraytag.getAsIntArray();
                 int l = 0;
 
-                for(int k3 : aint) {
+                for (int k3 : aint) {
                     l = Math.max(l, String.format(Locale.ROOT, "%X", k3).length());
                 }
 
@@ -345,7 +_,7 @@
                 if (p_178030_) {
                     indent(p_178029_ + 1, p_178027_);
 
-                    for(int k2 = 0; k2 < aint.length; ++k2) {
+                    for (int k2 = 0; k2 < aint.length; k2++) {
                         if (k2 != 0) {
                             p_178027_.append(',');
                         }
@@ -373,7 +_,7 @@
                 long[] along = longarraytag.getAsLongArray();
                 long i = 0L;
 
-                for(long j : along) {
+                for (long j : along) {
                     i = Math.max(i, (long)String.format(Locale.ROOT, "%X", j).length());
                 }
 
@@ -382,7 +_,7 @@
                 if (p_178030_) {
                     indent(p_178029_ + 1, p_178027_);
 
-                    for(int j3 = 0; j3 < along.length; ++j3) {
+                    for (int j3 = 0; j3 < along.length; j3++) {
                         if (j3 != 0) {
                             p_178027_.append(',');
                         }
@@ -416,7 +_,7 @@
         int i = p_178021_.lastIndexOf("\n") + 1;
         int j = p_178021_.length() - i;
 
-        for(int k = 0; k < 2 * p_178020_ - j; ++k) {
+        for (int k = 0; k < 2 * p_178020_ - j; k++) {
             p_178021_.append(' ');
         }
 
@@ -457,7 +_,7 @@
             listtag3.stream().map(ListTag.class::cast).forEach(p_178049_ -> {
                 CompoundTag compoundtag = new CompoundTag();
 
-                for(int i = 0; i < p_178049_.size(); ++i) {
+                for (int i = 0; i < p_178049_.size(); i++) {
                     compoundtag.putString(listtag1.getString(i), packBlockState(p_178049_.getCompound(i)));
                 }
 
@@ -517,13 +_,13 @@
             Object2IntMap<String> object2intmap = new Object2IntOpenHashMap<>();
             object2intmap.defaultReturnValue(-1);
 
-            for(int i = 0; i < listtag.size(); ++i) {
+            for (int i = 0; i < listtag.size(); i++) {
                 object2intmap.put(listtag.getString(i), i);
             }
 
             ListTag listtag1 = p_178072_.getList("data", 10);
 
-            for(int j = 0; j < listtag1.size(); ++j) {
+            for (int j = 0; j < listtag1.size(); j++) {
                 CompoundTag compoundtag = listtag1.getCompound(j);
                 String s = compoundtag.getString("state");
                 int k = object2intmap.getInt(s);
