--- a/net/minecraft/nbt/SnbtPrinterTagVisitor.java
+++ b/net/minecraft/nbt/SnbtPrinterTagVisitor.java
@@ -91,7 +_,7 @@
         StringBuilder stringbuilder = new StringBuilder("[").append("B").append(";");
         byte[] abyte = p_178116_.getAsByteArray();
 
-        for(int i = 0; i < abyte.length; ++i) {
+        for (int i = 0; i < abyte.length; i++) {
             stringbuilder.append(" ").append(abyte[i]).append("B");
             if (i != abyte.length - 1) {
                 stringbuilder.append(ELEMENT_SEPARATOR);
@@ -107,7 +_,7 @@
         StringBuilder stringbuilder = new StringBuilder("[").append("I").append(";");
         int[] aint = p_178128_.getAsIntArray();
 
-        for(int i = 0; i < aint.length; ++i) {
+        for (int i = 0; i < aint.length; i++) {
             stringbuilder.append(" ").append(aint[i]);
             if (i != aint.length - 1) {
                 stringbuilder.append(ELEMENT_SEPARATOR);
@@ -124,7 +_,7 @@
         StringBuilder stringbuilder = new StringBuilder("[").append("L").append(";");
         long[] along = p_178134_.getAsLongArray();
 
-        for(int i = 0; i < along.length; ++i) {
+        for (int i = 0; i < along.length; i++) {
             stringbuilder.append(" ").append(along[i]).append("L");
             if (i != along.length - 1) {
                 stringbuilder.append(ELEMENT_SEPARATOR);
@@ -147,7 +_,7 @@
                 stringbuilder.append("\n");
             }
 
-            for(int i = 0; i < p_178132_.size(); ++i) {
+            for (int i = 0; i < p_178132_.size(); i++) {
                 stringbuilder.append(Strings.repeat(s, this.depth + 1));
                 stringbuilder.append(new SnbtPrinterTagVisitor(s, this.depth + 1, this.path).visit(p_178132_.get(i)));
                 if (i != p_178132_.size() - 1) {
@@ -180,7 +_,7 @@
             Collection<String> collection = this.getKeys(p_178120_);
             Iterator<String> iterator = collection.iterator();
 
-            while(iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 String s1 = iterator.next();
                 Tag tag = p_178120_.get(s1);
                 this.pushPath(s1);
@@ -218,7 +_,7 @@
         List<String> list = Lists.newArrayList();
         List<String> list1 = KEY_ORDER.get(this.pathString());
         if (list1 != null) {
-            for(String s : list1) {
+            for (String s : list1) {
                 if (set.remove(s)) {
                     list.add(s);
                 }
