--- a/net/minecraft/nbt/StringTagVisitor.java
+++ b/net/minecraft/nbt/StringTagVisitor.java
@@ -54,7 +_,7 @@
         this.builder.append("[B;");
         byte[] abyte = p_178162_.getAsByteArray();
 
-        for(int i = 0; i < abyte.length; ++i) {
+        for (int i = 0; i < abyte.length; i++) {
             if (i != 0) {
                 this.builder.append(',');
             }
@@ -70,7 +_,7 @@
         this.builder.append("[I;");
         int[] aint = p_178174_.getAsIntArray();
 
-        for(int i = 0; i < aint.length; ++i) {
+        for (int i = 0; i < aint.length; i++) {
             if (i != 0) {
                 this.builder.append(',');
             }
@@ -86,7 +_,7 @@
         this.builder.append("[L;");
         long[] along = p_178180_.getAsLongArray();
 
-        for(int i = 0; i < along.length; ++i) {
+        for (int i = 0; i < along.length; i++) {
             if (i != 0) {
                 this.builder.append(',');
             }
@@ -101,7 +_,7 @@
     public void visitList(ListTag p_178178_) {
         this.builder.append('[');
 
-        for(int i = 0; i < p_178178_.size(); ++i) {
+        for (int i = 0; i < p_178178_.size(); i++) {
             if (i != 0) {
                 this.builder.append(',');
             }
@@ -118,7 +_,7 @@
         List<String> list = Lists.newArrayList(p_178166_.getAllKeys());
         Collections.sort(list);
 
-        for(String s : list) {
+        for (String s : list) {
             if (this.builder.length() != 1) {
                 this.builder.append(',');
             }
