--- a/net/minecraft/server/rcon/thread/QueryThreadGs4.java
+++ b/net/minecraft/server/rcon/thread/QueryThreadGs4.java
@@ -94,7 +_,7 @@
         LOGGER.debug("Packet len {} [{}]", i, socketaddress);
         if (3 <= i && -2 == abyte[0] && -3 == abyte[1]) {
             LOGGER.debug("Packet '{}' [{}]", PktUtils.toHexString(abyte[2]), socketaddress);
-            switch(abyte[2]) {
+            switch (abyte[2]) {
                 case 0:
                     if (!this.validChallenge(p_11550_)) {
                         LOGGER.debug("Invalid challenge [{}]", socketaddress);
@@ -173,7 +_,7 @@
             this.rulesResponse.write(0);
             String[] astring = this.serverInterface.getPlayerNames();
 
-            for(String s : astring) {
+            for (String s : astring) {
                 this.rulesResponse.writeString(s);
             }
 
@@ -219,7 +_,7 @@
         DatagramPacket datagrampacket = new DatagramPacket(this.buffer, this.buffer.length);
 
         try {
-            while(this.running) {
+            while (this.running) {
                 try {
                     this.socket.receive(datagrampacket);
                     this.pruneChallenges();
