--- a/net/minecraft/server/rcon/thread/RconThread.java
+++ b/net/minecraft/server/rcon/thread/RconThread.java
@@ -34,7 +_,7 @@
     @Override
     public void run() {
         try {
-            while(this.running) {
+            while (this.running) {
                 try {
                     Socket socket = this.socket.accept();
                     RconClient rconclient = new RconClient(this.serverInterface, this.rconPassword, socket);
@@ -96,7 +_,7 @@
         this.closeSocket(this.socket);
         super.stop();
 
-        for(RconClient rconclient : this.clients) {
+        for (RconClient rconclient : this.clients) {
             if (rconclient.isRunning()) {
                 rconclient.stop();
             }
