--- a/net/minecraft/server/rcon/thread/RconClient.java
+++ b/net/minecraft/server/rcon/thread/RconClient.java
@@ -43,7 +_,7 @@
     public void run() {
         try {
             try {
-                while(this.running) {
+                while (this.running) {
                     BufferedInputStream bufferedinputstream = new BufferedInputStream(this.client.getInputStream());
                     int i = bufferedinputstream.read(this.buf, 0, 1460);
                     if (10 > i) {
@@ -61,7 +_,7 @@
                     j += 4;
                     int i1 = PktUtils.intFromByteArray(this.buf, j);
                     j += 4;
-                    switch(i1) {
+                    switch (i1) {
                         case 2:
                             if (this.authed) {
                                 String s1 = PktUtils.stringFromByteArray(this.buf, j, i);
@@ -123,14 +_,15 @@
     }
 
     private void sendCmdResponse(int p_11595_, String p_11596_) throws IOException {
-        int i = p_11596_.length();
-
+        byte[] whole = p_11596_.getBytes(StandardCharsets.UTF_8);
+        int i = whole.length;
+        int start = 0;
         do {
             int j = 4096 <= i ? 4096 : i;
-            this.send(p_11595_, 0, p_11596_.substring(0, j));
-            p_11596_ = p_11596_.substring(j);
-            i = p_11596_.length();
-        } while(0 != i);
+            this.send(p_11595_, 0, new String(java.util.Arrays.copyOfRange(whole, start, j+start), StandardCharsets.UTF_8));
+            i -= j;
+            start += j;
+        } while (0 != i);
     }
 
     @Override
