--- a/net/minecraft/world/entity/animal/armadillo/ArmadilloAi.java
+++ b/net/minecraft/world/entity/animal/armadillo/ArmadilloAi.java
@@ -182,7 +_,7 @@
         protected void tick(ServerLevel p_316315_, Armadillo p_316310_, long p_316415_) {
             super.tick(p_316315_, p_316310_, p_316415_);
             if (this.nextPeekTimer > 0) {
-                --this.nextPeekTimer;
+                this.nextPeekTimer--;
             }
 
             if (p_316310_.shouldSwitchToScaredState()) {
