--- a/net/minecraft/world/entity/projectile/Arrow.java
+++ b/net/minecraft/world/entity/projectile/Arrow.java
@@ -85,7 +_,7 @@
     private void makeParticle(int p_36877_) {
         int i = this.getColor();
         if (i != -1 && p_36877_ > 0) {
-            for(int j = 0; j < p_36877_; ++j) {
+            for (int j = 0; j < p_36877_; j++) {
                 this.level()
                     .addParticle(
                         ColorParticleOption.create(ParticleTypes.ENTITY_EFFECT, i),
@@ -110,7 +_,7 @@
         Entity entity = this.getEffectSource();
         PotionContents potioncontents = this.getPotionContents();
         if (potioncontents.potion().isPresent()) {
-            for(MobEffectInstance mobeffectinstance : potioncontents.potion().get().value().getEffects()) {
+            for (MobEffectInstance mobeffectinstance : potioncontents.potion().get().value().getEffects()) {
                 p_36873_.addEffect(
                     new MobEffectInstance(
                         mobeffectinstance.getEffect(),
@@ -124,7 +_,7 @@
             }
         }
 
-        for(MobEffectInstance mobeffectinstance1 : potioncontents.customEffects()) {
+        for (MobEffectInstance mobeffectinstance1 : potioncontents.customEffects()) {
             p_36873_.addEffect(mobeffectinstance1, entity);
         }
     }
@@ -143,7 +_,7 @@
                 float f1 = (float)(i >> 8 & 0xFF) / 255.0F;
                 float f2 = (float)(i >> 0 & 0xFF) / 255.0F;
 
-                for(int j = 0; j < 20; ++j) {
+                for (int j = 0; j < 20; j++) {
                     this.level()
                         .addParticle(
                             ColorParticleOption.create(ParticleTypes.ENTITY_EFFECT, f, f1, f2),
