--- a/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
+++ b/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
@@ -103,8 +_,8 @@
         super.tick();
         if (this.isAttachedToEntity()) {
             if (this.attachedToEntity == null) {
-                this.entityData.get(DATA_ATTACHED_TO_TARGET).ifPresent(p_340164_ -> {
-                    Entity entity = this.level().getEntity(p_340164_);
+                this.entityData.get(DATA_ATTACHED_TO_TARGET).ifPresent(p_337881_ -> {
+                    Entity entity = this.level().getEntity(p_337881_);
                     if (entity instanceof LivingEntity) {
                         this.attachedToEntity = (LivingEntity)entity;
                     }
@@ -147,7 +_,7 @@
 
         HitResult hitresult = ProjectileUtil.getHitResultOnMoveVector(this, this::canHitEntity);
         if (!this.noPhysics) {
-            this.onHit(hitresult);
+            this.hitOrDeflect(hitresult);
             this.hasImpulse = true;
         }
 
@@ -156,7 +_,7 @@
             this.level().playSound(null, this.getX(), this.getY(), this.getZ(), SoundEvents.FIREWORK_ROCKET_LAUNCH, SoundSource.AMBIENT, 3.0F, 1.0F);
         }
 
-        ++this.life;
+        this.life++;
         if (this.level().isClientSide && this.life % 2 < 2) {
             this.level()
                 .addParticle(
@@ -175,6 +_,13 @@
         }
     }
 
+    @Override
+    protected void onHit(HitResult result) {
+        if (result.getType() == HitResult.Type.MISS || !net.neoforged.neoforge.event.EventHooks.onProjectileImpact(this, result)) {
+            super.onHit(result);
+        }
+    }
+
     private void explode() {
         this.level().broadcastEntityEvent(this, (byte)17);
         this.gameEvent(GameEvent.EXPLODE, this.getOwner());
@@ -220,11 +_,11 @@
             double d0 = 5.0;
             Vec3 vec3 = this.position();
 
-            for(LivingEntity livingentity : this.level().getEntitiesOfClass(LivingEntity.class, this.getBoundingBox().inflate(5.0))) {
+            for (LivingEntity livingentity : this.level().getEntitiesOfClass(LivingEntity.class, this.getBoundingBox().inflate(5.0))) {
                 if (livingentity != this.attachedToEntity && !(this.distanceToSqr(livingentity) > 25.0)) {
                     boolean flag = false;
 
-                    for(int i = 0; i < 2; ++i) {
+                    for (int i = 0; i < 2; i++) {
                         Vec3 vec31 = new Vec3(livingentity.getX(), livingentity.getY(0.5 * (double)i), livingentity.getZ());
                         HitResult hitresult = this.level().clip(new ClipContext(vec3, vec31, ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, this));
                         if (hitresult.getType() == HitResult.Type.MISS) {
