--- a/net/minecraft/world/entity/projectile/LargeFireball.java
+++ b/net/minecraft/world/entity/projectile/LargeFireball.java
@@ -10,30 +_,24 @@
 import net.minecraft.world.phys.HitResult;
 
 public class LargeFireball extends Fireball {
-    private final boolean fire;
     private int explosionPower = 1;
 
     public LargeFireball(EntityType<? extends LargeFireball> p_37199_, Level p_37200_) {
         super(p_37199_, p_37200_);
-        this.fire = true;
     }
 
     public LargeFireball(Level p_181151_, LivingEntity p_181152_, double p_181153_, double p_181154_, double p_181155_, int p_181156_) {
-        this(p_181151_, p_181152_, p_181153_, p_181154_, p_181155_, p_181156_, true);
-    }
-
-    public LargeFireball(Level p_340660_, LivingEntity p_341305_, double p_341941_, double p_340784_, double p_342349_, int p_341818_, boolean p_341232_) {
-        super(EntityType.FIREBALL, p_341305_, p_341941_, p_340784_, p_342349_, p_340660_);
-        this.explosionPower = p_341818_;
-        this.fire = p_341232_;
+        super(EntityType.FIREBALL, p_181152_, p_181153_, p_181154_, p_181155_, p_181151_);
+        this.explosionPower = p_181156_;
     }
 
     @Override
     protected void onHit(HitResult p_37218_) {
         super.onHit(p_37218_);
         if (!this.level().isClientSide) {
-            boolean flag = this.level().getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING);
-            this.level().explode(this, this.getX(), this.getY(), this.getZ(), (float)this.explosionPower, this.fire && flag, Level.ExplosionInteraction.MOB);
+            // TODO 1.19.3: The creation of Level.ExplosionInteraction means this code path will fire EntityMobGriefingEvent twice. Should we try and fix it? -SS
+            boolean flag = net.neoforged.neoforge.event.EventHooks.getMobGriefingEvent(this.level(), this.getOwner());
+            this.level().explode(this, this.getX(), this.getY(), this.getZ(), (float)this.explosionPower, flag, Level.ExplosionInteraction.MOB);
             this.discard();
         }
     }
