--- a/net/minecraft/world/entity/projectile/LargeFireball.java
+++ b/net/minecraft/world/entity/projectile/LargeFireball.java
@@ -37,7 +_,7 @@
                 flag = true;
                 level$explosioninteraction = Level.ExplosionInteraction.TNT;
             } else {
-                flag = serverlevel.getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING);
+                flag = net.neoforged.neoforge.event.EventHooks.canEntityGrief(serverlevel, this.getOwner());
                 level$explosioninteraction = Level.ExplosionInteraction.MOB;
             }
 
