--- a/net/minecraft/world/entity/projectile/ProjectileDeflection.java
+++ b/net/minecraft/world/entity/projectile/ProjectileDeflection.java
@@ -7,11 +_,13 @@
 public interface ProjectileDeflection {
     ProjectileDeflection NONE = (p_320379_, p_320626_, p_320122_) -> {
     };
-    ProjectileDeflection REVERSE = (p_340165_, p_340166_, p_340167_) -> {
-        float f = 180.0F + p_340167_.nextFloat() * 20.0F;
-        p_340165_.setDeltaMovement(p_340165_.getDeltaMovement().scale(-0.25));
-        p_340165_.setYRot(p_340165_.getYRot() + f);
-        p_340165_.yRotO += f;
+    ProjectileDeflection REVERSE = (p_339516_, p_339517_, p_339518_) -> {
+        float f = 180.0F + p_339518_.nextFloat() * 20.0F;
+        p_339516_.setDeltaMovement(p_339516_.getDeltaMovement().scale(-0.5));
+        p_339516_.setYRot(p_339516_.getYRot() + f);
+        p_339516_.yRotO += f;
+        p_339516_.hurtMarked = true;
+        p_339516_.onDeflection();
     };
 
     void deflect(Projectile p_320311_, Entity p_320130_, RandomSource p_320125_);
