--- a/net/minecraft/world/entity/projectile/Projectile.java
+++ b/net/minecraft/world/entity/projectile/Projectile.java
@@ -31,7 +_,6 @@
     private Entity cachedOwner;
     private boolean leftOwner;
     private boolean hasBeenShot;
-    protected boolean isDeflected;
 
     protected Projectile(EntityType<? extends Projectile> p_37248_, Level p_37249_) {
         super(p_37248_, p_37249_);
@@ -49,15 +_,10 @@
     public Entity getOwner() {
         if (this.cachedOwner != null && !this.cachedOwner.isRemoved()) {
             return this.cachedOwner;
+        } else if (this.ownerUUID != null && this.level() instanceof ServerLevel serverlevel) {
+            this.cachedOwner = serverlevel.getEntity(this.ownerUUID);
+            return this.cachedOwner;
         } else {
-            if (this.ownerUUID != null) {
-                Level level = this.level();
-                if (level instanceof ServerLevel serverlevel) {
-                    this.cachedOwner = serverlevel.getEntity(this.ownerUUID);
-                    return this.cachedOwner;
-                }
-            }
-
             return null;
         }
     }
@@ -119,7 +_,7 @@
     private boolean checkLeftOwner() {
         Entity entity = this.getOwner();
         if (entity != null) {
-            for(Entity entity1 : this.level()
+            for (Entity entity1 : this.level()
                 .getEntities(
                     this,
                     this.getBoundingBox().expandTowards(this.getDeltaMovement()).inflate(1.0),
@@ -134,15 +_,19 @@
         return true;
     }
 
-    public void shoot(double p_37266_, double p_37267_, double p_37268_, float p_37269_, float p_37270_) {
-        Vec3 vec3 = new Vec3(p_37266_, p_37267_, p_37268_)
+    public Vec3 getMovementToShoot(double p_338345_, double p_338731_, double p_338427_, float p_338430_, float p_338697_) {
+        return new Vec3(p_338345_, p_338731_, p_338427_)
             .normalize()
             .add(
-                this.random.triangle(0.0, 0.0172275 * (double)p_37270_),
-                this.random.triangle(0.0, 0.0172275 * (double)p_37270_),
-                this.random.triangle(0.0, 0.0172275 * (double)p_37270_)
+                this.random.triangle(0.0, 0.0172275 * (double)p_338697_),
+                this.random.triangle(0.0, 0.0172275 * (double)p_338697_),
+                this.random.triangle(0.0, 0.0172275 * (double)p_338697_)
             )
-            .scale((double)p_37269_);
+            .scale((double)p_338430_);
+    }
+
+    public void shoot(double p_37266_, double p_37267_, double p_37268_, float p_37269_, float p_37270_) {
+        Vec3 vec3 = this.getMovementToShoot(p_37266_, p_37267_, p_37268_, p_37269_, p_37270_);
         this.setDeltaMovement(vec3);
         double d0 = vec3.horizontalDistance();
         this.setYRot((float)(Mth.atan2(vec3.x, vec3.z) * 180.0F / (float)Math.PI));
@@ -160,19 +_,25 @@
         this.setDeltaMovement(this.getDeltaMovement().add(vec3.x, p_37252_.onGround() ? 0.0 : vec3.y, vec3.z));
     }
 
+    protected ProjectileDeflection hitOrDeflect(HitResult p_339680_) {
+        if (p_339680_.getType() == HitResult.Type.ENTITY) {
+            EntityHitResult entityhitresult = (EntityHitResult)p_339680_;
+            ProjectileDeflection projectiledeflection = entityhitresult.getEntity().deflection(this);
+            if (projectiledeflection != ProjectileDeflection.NONE) {
+                projectiledeflection.deflect(this, entityhitresult.getEntity(), this.random);
+                this.markHurt();
+                return projectiledeflection;
+            }
+        }
+
+        this.onHit(p_339680_);
+        return ProjectileDeflection.NONE;
+    }
+
     protected void onHit(HitResult p_37260_) {
         HitResult.Type hitresult$type = p_37260_.getType();
         if (hitresult$type == HitResult.Type.ENTITY) {
             EntityHitResult entityhitresult = (EntityHitResult)p_37260_;
-            if (!this.isDeflected) {
-                ProjectileDeflection projectiledeflection = entityhitresult.getEntity().deflection(this);
-                if (projectiledeflection != ProjectileDeflection.NONE) {
-                    projectiledeflection.deflect(this, entityhitresult.getEntity(), this.random);
-                    this.isDeflected = true;
-                    return;
-                }
-            }
-
             this.onHitEntity(entityhitresult);
             this.level().gameEvent(GameEvent.PROJECTILE_LAND, p_37260_.getLocation(), GameEvent.Context.of(this, null));
         } else if (hitresult$type == HitResult.Type.BLOCK) {
@@ -221,11 +_,11 @@
     }
 
     protected static float lerpRotation(float p_37274_, float p_37275_) {
-        while(p_37275_ - p_37274_ < -180.0F) {
+        while (p_37275_ - p_37274_ < -180.0F) {
             p_37274_ -= 360.0F;
         }
 
-        while(p_37275_ - p_37274_ >= 180.0F) {
+        while (p_37275_ - p_37274_ >= 180.0F) {
             p_37274_ += 360.0F;
         }
 
@@ -250,14 +_,15 @@
     @Override
     public boolean mayInteract(Level p_150167_, BlockPos p_150168_) {
         Entity entity = this.getOwner();
-        if (entity instanceof Player) {
-            return entity.mayInteract(p_150167_, p_150168_);
-        } else {
-            return entity == null || p_150167_.getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING);
-        }
+        return entity instanceof Player
+            ? entity.mayInteract(p_150167_, p_150168_)
+            : entity == null || net.neoforged.neoforge.event.EventHooks.getMobGriefingEvent(p_150167_, entity);
     }
 
     public boolean mayBreak(Level p_307481_) {
         return this.getType().is(EntityTypeTags.IMPACT_PROJECTILES) && p_307481_.getGameRules().getBoolean(GameRules.RULE_PROJECTILESCANBREAKBLOCKS);
+    }
+
+    public void onDeflection() {
     }
 }
