--- a/net/minecraft/world/entity/projectile/ThrownTrident.java
+++ b/net/minecraft/world/entity/projectile/ThrownTrident.java
@@ -39,6 +_,12 @@
         this.entityData.set(ID_FOIL, p_37571_.hasFoil());
     }
 
+    public ThrownTrident(Level p_338686_, double p_338771_, double p_338674_, double p_338477_, ItemStack p_338255_) {
+        super(EntityType.TRIDENT, p_338771_, p_338674_, p_338477_, p_338686_, p_338255_);
+        this.entityData.set(ID_LOYALTY, (byte)EnchantmentHelper.getLoyalty(p_338255_));
+        this.entityData.set(ID_FOIL, p_338255_.hasFoil());
+    }
+
     @Override
     protected void defineSynchedData(SynchedEntityData.Builder p_326249_) {
         super.defineSynchedData(p_326249_);
@@ -75,7 +_,7 @@
                     this.playSound(SoundEvents.TRIDENT_RETURN, 10.0F, 1.0F);
                 }
 
-                ++this.clientSideReturnTridentTickCount;
+                this.clientSideReturnTridentTickCount++;
             }
         }
 
@@ -84,11 +_,7 @@
 
     private boolean isAcceptibleReturnOwner() {
         Entity entity = this.getOwner();
-        if (entity == null || !entity.isAlive()) {
-            return false;
-        } else {
-            return !(entity instanceof ServerPlayer) || !entity.isSpectator();
-        }
+        return entity == null || !entity.isAlive() ? false : !(entity instanceof ServerPlayer) || !entity.isSpectator();
     }
 
     public boolean isFoil() {
