--- a/net/minecraft/world/entity/projectile/ThrownEgg.java
+++ b/net/minecraft/world/entity/projectile/ThrownEgg.java
@@ -29,7 +_,7 @@
         if (p_37484_ == 3) {
             double d0 = 0.08;
 
-            for(int i = 0; i < 8; ++i) {
+            for (int i = 0; i < 8; i++) {
                 this.level()
                     .addParticle(
                         new ItemParticleOption(ParticleTypes.ITEM, this.getItem()),
@@ -60,7 +_,7 @@
                     i = 4;
                 }
 
-                for(int j = 0; j < i; ++j) {
+                for (int j = 0; j < i; j++) {
                     Chicken chicken = EntityType.CHICKEN.create(this.level());
                     if (chicken != null) {
                         chicken.setAge(-24000);
