--- a/com/mojang/realmsclient/gui/screens/RealmsBackupInfoScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsBackupInfoScreen.java
@@ -4,6 +_,7 @@
 import java.util.Locale;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.ObjectSelectionList;
 import net.minecraft.client.gui.components.events.GuiEventListener;
@@ -36,7 +_,9 @@
         this.backupInfoList = this.layout.addToContents(new RealmsBackupInfoScreen.BackupInfoList(this.minecraft));
         this.layout.addToFooter(Button.builder(CommonComponents.GUI_BACK, p_299754_ -> this.onClose()).build());
         this.repositionElements();
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321336_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321336_);
+        });
     }
 
     @Override
@@ -134,7 +_,7 @@
         }
 
         private Component translateKey(String p_287652_) {
-            return switch(p_287652_) {
+            return switch (p_287652_) {
                 case "template_name" -> TEMPLATE_NAME;
                 case "game_difficulty" -> GAME_DIFFICULTY;
                 case "name" -> NAME;
