--- a/com/mojang/realmsclient/gui/screens/RealmsLongRunningMcoTaskScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsLongRunningMcoTaskScreen.java
@@ -7,6 +_,7 @@
 import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.client.GameNarrator;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.LoadingDotsWidget;
 import net.minecraft.client.gui.components.events.GuiEventListener;
@@ -42,7 +_,7 @@
         } else {
             this.title = this.queuedTasks.get(0).getTitle();
             Runnable runnable = () -> {
-                for(LongRunningTask longrunningtask : p_305868_) {
+                for (LongRunningTask longrunningtask : p_305868_) {
                     this.setTitle(longrunningtask.getTitle());
                     if (longrunningtask.aborted()) {
                         break;
@@ -84,7 +_,9 @@
         this.loadingDotsWidget = new LoadingDotsWidget(this.font, this.title);
         this.layout.addChild(this.loadingDotsWidget, p_300631_ -> p_300631_.paddingBottom(30));
         this.layout.addChild(Button.builder(CommonComponents.GUI_CANCEL, p_300632_ -> this.cancel()).build());
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321340_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321340_);
+        });
         this.repositionElements();
     }
 
@@ -95,7 +_,7 @@
     }
 
     protected void cancel() {
-        for(LongRunningTask longrunningtask : this.queuedTasks) {
+        for (LongRunningTask longrunningtask : this.queuedTasks) {
             longrunningtask.abortTask();
         }
 
