--- a/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
@@ -18,6 +_,7 @@
 import javax.annotation.Nullable;
 import net.minecraft.Util;
 import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.ImageButton;
 import net.minecraft.client.gui.components.ObjectSelectionList;
@@ -96,7 +_,9 @@
         linearlayout.addChild(Button.builder(CommonComponents.GUI_CANCEL, p_89691_ -> this.onClose()).width(100).build());
         this.publisherButton = linearlayout.addChild(Button.builder(PUBLISHER_BUTTON_NAME, p_89679_ -> this.onPublish()).width(100).build());
         this.updateButtonStates();
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321348_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321348_);
+        });
         this.repositionElements();
     }
 
@@ -151,10 +_,12 @@
                 @Override
                 public void run() {
                     WorldTemplatePaginatedList worldtemplatepaginatedlist = p_89654_;
-    
                     RealmsClient realmsclient = RealmsClient.create();
-                    while(worldtemplatepaginatedlist != null) {
-                        Either<WorldTemplatePaginatedList, Exception> either = RealmsSelectWorldTemplateScreen.this.fetchTemplates(worldtemplatepaginatedlist, realmsclient);
+
+                    while (worldtemplatepaginatedlist != null) {
+                        Either<WorldTemplatePaginatedList, Exception> either = RealmsSelectWorldTemplateScreen.this.fetchTemplates(
+                            worldtemplatepaginatedlist, realmsclient
+                        );
                         worldtemplatepaginatedlist = RealmsSelectWorldTemplateScreen.this.minecraft
                             .submit(
                                 () -> {
@@ -165,15 +_,15 @@
                                                 I18n.get("mco.template.select.failure")
                                             );
                                         }
-                
+
                                         return null;
                                     } else {
                                         WorldTemplatePaginatedList worldtemplatepaginatedlist1 = either.left().get();
-                
-                                        for(WorldTemplate worldtemplate : worldtemplatepaginatedlist1.templates) {
+
+                                        for (WorldTemplate worldtemplate : worldtemplatepaginatedlist1.templates) {
                                             RealmsSelectWorldTemplateScreen.this.worldTemplateList.addEntry(worldtemplate);
                                         }
-                
+
                                         if (worldtemplatepaginatedlist1.templates.isEmpty()) {
                                             if (RealmsSelectWorldTemplateScreen.this.worldTemplateList.isEmpty()) {
                                                 String s = I18n.get("mco.template.select.none", "%link");
@@ -184,7 +_,7 @@
                                                     s, textrenderingutils$linesegment
                                                 );
                                             }
-                
+
                                             return null;
                                         } else {
                                             return worldtemplatepaginatedlist1;
@@ -216,7 +_,7 @@
         }
 
         if (this.warning != null) {
-            for(int i = 0; i < this.warning.length; ++i) {
+            for (int i = 0; i < this.warning.length; i++) {
                 Component component = this.warning[i];
                 p_282162_.drawCenteredString(this.font, component, this.width / 2, row(-1 + i), -6250336);
             }
@@ -224,13 +_,13 @@
     }
 
     private void renderMultilineMessage(GuiGraphics p_282398_, int p_282163_, int p_282021_, List<TextRenderingUtils.Line> p_282203_) {
-        for(int i = 0; i < p_282203_.size(); ++i) {
+        for (int i = 0; i < p_282203_.size(); i++) {
             TextRenderingUtils.Line textrenderingutils$line = p_282203_.get(i);
             int j = row(4 + i);
             int k = textrenderingutils$line.segments.stream().mapToInt(p_280748_ -> this.font.width(p_280748_.renderedText())).sum();
             int l = this.width / 2 - k / 2;
 
-            for(TextRenderingUtils.LineSegment textrenderingutils$linesegment : textrenderingutils$line.segments) {
+            for (TextRenderingUtils.LineSegment textrenderingutils$linesegment : textrenderingutils$line.segments) {
                 int i1 = textrenderingutils$linesegment.isLink() ? 3368635 : -1;
                 int j1 = p_282398_.drawString(this.font, textrenderingutils$linesegment.renderedText(), l, j, i1);
                 if (textrenderingutils$linesegment.isLink() && p_282163_ > l && p_282163_ < j1 && p_282021_ > j - 3 && p_282021_ < j + 8) {
