--- a/com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
@@ -7,6 +_,7 @@
 import javax.annotation.Nullable;
 import net.minecraft.Util;
 import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.EditBox;
 import net.minecraft.client.gui.components.events.GuiEventListener;
@@ -53,7 +_,9 @@
         linearlayout.addChild(CommonLayouts.labeledElement(this.font, this.profileName, NAME_LABEL));
         this.inviteButton = linearlayout.addChild(Button.builder(TITLE, p_88721_ -> this.onInvite()).width(200).build());
         this.layout.addToFooter(Button.builder(CommonComponents.GUI_BACK, p_329651_ -> this.onClose()).width(200).build());
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_329653_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_329653_);
+        });
         this.repositionElements();
     }
 
