--- a/com/mojang/realmsclient/gui/screens/RealmsDownloadLatestWorldScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsDownloadLatestWorldScreen.java
@@ -103,7 +_,7 @@
     @Override
     public void tick() {
         super.tick();
-        ++this.animTick;
+        this.animTick++;
         if (this.status != null && this.narrationRateLimiter.tryAcquire(1)) {
             Component component = this.createProgressNarrationMessage();
             this.minecraft.getNarrator().sayNow(component);
@@ -167,7 +_,7 @@
     private void drawDots(GuiGraphics p_281948_) {
         int i = this.font.width(this.status);
         if (this.animTick % 10 == 0) {
-            ++this.dotIndex;
+            this.dotIndex++;
         }
 
         p_281948_.drawString(this.font, DOTS[this.dotIndex % DOTS.length], this.width / 2 + i / 2 + 5, 50, -1, false);
@@ -230,7 +_,7 @@
                     filedownload.contentLength(this.worldDownload.downloadLink);
                     filedownload.download(this.worldDownload, this.worldName, this.downloadStatus, this.minecraft.getLevelSource());
 
-                    while(!filedownload.isFinished()) {
+                    while (!filedownload.isFinished()) {
                         if (filedownload.isError()) {
                             filedownload.cancel();
                             this.errorMessage = Component.translatable("mco.download.failed");
