--- a/com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
@@ -5,6 +_,7 @@
 import com.mojang.realmsclient.util.WorldGenerationInfo;
 import com.mojang.realmsclient.util.task.CreateSnapshotRealmTask;
 import com.mojang.realmsclient.util.task.RealmCreationTask;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.EditBox;
 import net.minecraft.client.gui.components.events.GuiEventListener;
@@ -58,7 +_,9 @@
         LinearLayout linearlayout1 = this.layout.addToFooter(LinearLayout.horizontal().spacing(10));
         linearlayout1.addChild(button);
         linearlayout1.addChild(Button.builder(CommonComponents.GUI_BACK, p_293570_ -> this.onClose()).build());
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321338_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321338_);
+        });
         this.repositionElements();
     }
 
