--- a/com/mojang/realmsclient/gui/screens/RealmsConfigureWorldScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsConfigureWorldScreen.java
@@ -102,7 +_,7 @@
         );
         this.slotButtonList.clear();
 
-        for(int i = 1; i < 5; ++i) {
+        for (int i = 1; i < 5; i++) {
             this.slotButtonList.add(this.addSlotButton(i));
         }
 
@@ -176,10 +_,10 @@
     private RealmsWorldSlotButton addSlotButton(int p_167386_) {
         int i = this.frame(p_167386_);
         int j = row(5) + 5;
-        RealmsWorldSlotButton realmsworldslotbutton = new RealmsWorldSlotButton(i, j, 80, 80, p_167386_, p_167389_ -> {
-            RealmsWorldSlotButton.State realmsworldslotbutton$state = ((RealmsWorldSlotButton)p_167389_).getState();
+        RealmsWorldSlotButton realmsworldslotbutton = new RealmsWorldSlotButton(i, j, 80, 80, p_167386_, p_339255_ -> {
+            RealmsWorldSlotButton.State realmsworldslotbutton$state = ((RealmsWorldSlotButton)p_339255_).getState();
             if (realmsworldslotbutton$state != null) {
-                switch(realmsworldslotbutton$state.action) {
+                switch (realmsworldslotbutton$state.action) {
                     case NOTHING:
                         break;
                     case JOIN:
@@ -268,7 +_,7 @@
                         this.show(this.resetWorldButton);
                     }
 
-                    for(RealmsWorldSlotButton realmsworldslotbutton : this.slotButtonList) {
+                    for (RealmsWorldSlotButton realmsworldslotbutton : this.slotButtonList) {
                         realmsworldslotbutton.setServerData(realmsserver);
                     }
                 });
