--- a/net/minecraft/network/codec/ByteBufCodecs.java
+++ b/net/minecraft/network/codec/ByteBufCodecs.java
@@ -18,7 +_,6 @@
 import java.util.function.IntFunction;
 import java.util.function.Supplier;
 import java.util.function.ToIntFunction;
-import net.minecraft.Util;
 import net.minecraft.core.Holder;
 import net.minecraft.core.HolderSet;
 import net.minecraft.core.IdMap;
@@ -70,67 +_,58 @@
             p_320028_.writeShort(p_320388_);
         }
     };
-    StreamCodec<ByteBuf, Character> CHAR = new StreamCodec<ByteBuf, Character>() {
-        public Character decode(ByteBuf p_320319_) {
-            return p_320319_.readChar();
-        }
-
-        public void encode(ByteBuf p_320669_, Character p_340746_) {
-            p_320669_.writeChar(p_340746_);
-        }
-    };
     StreamCodec<ByteBuf, Integer> INT = new StreamCodec<ByteBuf, Integer>() {
+        public Integer decode(ByteBuf p_320319_) {
+            return p_320319_.readInt();
+        }
+
+        public void encode(ByteBuf p_320669_, Integer p_320205_) {
+            p_320669_.writeInt(p_320205_);
+        }
+    };
+    StreamCodec<ByteBuf, Integer> VAR_INT = new StreamCodec<ByteBuf, Integer>() {
         public Integer decode(ByteBuf p_320253_) {
-            return p_320253_.readInt();
+            return VarInt.read(p_320253_);
         }
 
         public void encode(ByteBuf p_320753_, Integer p_330380_) {
-            p_320753_.writeInt(p_330380_);
-        }
-    };
-    StreamCodec<ByteBuf, Integer> VAR_INT = new StreamCodec<ByteBuf, Integer>() {
-        public Integer decode(ByteBuf p_320759_) {
-            return VarInt.read(p_320759_);
-        }
-
-        public void encode(ByteBuf p_320314_, Integer p_341582_) {
-            VarInt.write(p_320314_, p_341582_);
+            VarInt.write(p_320753_, p_330380_);
         }
     };
     StreamCodec<ByteBuf, Long> VAR_LONG = new StreamCodec<ByteBuf, Long>() {
-        public Long decode(ByteBuf p_320635_) {
-            return VarLong.read(p_320635_);
+        public Long decode(ByteBuf p_320759_) {
+            return VarLong.read(p_320759_);
         }
 
-        public void encode(ByteBuf p_320545_, Long p_341467_) {
-            VarLong.write(p_320545_, p_341467_);
+        public void encode(ByteBuf p_320314_, Long p_330753_) {
+            VarLong.write(p_320314_, p_330753_);
         }
     };
     StreamCodec<ByteBuf, Float> FLOAT = new StreamCodec<ByteBuf, Float>() {
-        public Float decode(ByteBuf p_320259_) {
-            return p_320259_.readFloat();
+        public Float decode(ByteBuf p_320635_) {
+            return p_320635_.readFloat();
         }
 
-        public void encode(ByteBuf p_320199_, Float p_341938_) {
-            p_320199_.writeFloat(p_341938_);
+        public void encode(ByteBuf p_320545_, Float p_332060_) {
+            p_320545_.writeFloat(p_332060_);
         }
     };
     StreamCodec<ByteBuf, Double> DOUBLE = new StreamCodec<ByteBuf, Double>() {
-        public Double decode(ByteBuf p_320599_) {
-            return p_320599_.readDouble();
+        public Double decode(ByteBuf p_320259_) {
+            return p_320259_.readDouble();
         }
 
-        public void encode(ByteBuf p_320880_, Double p_341144_) {
-            p_320880_.writeDouble(p_341144_);
+        public void encode(ByteBuf p_320199_, Double p_330368_) {
+            p_320199_.writeDouble(p_330368_);
         }
     };
     StreamCodec<ByteBuf, byte[]> BYTE_ARRAY = new StreamCodec<ByteBuf, byte[]>() {
-        public byte[] decode(ByteBuf p_320167_) {
-            return FriendlyByteBuf.readByteArray(p_320167_);
+        public byte[] decode(ByteBuf p_319947_) {
+            return FriendlyByteBuf.readByteArray(p_319947_);
         }
 
-        public void encode(ByteBuf p_320240_, byte[] p_341768_) {
-            FriendlyByteBuf.writeByteArray(p_320240_, p_341768_);
+        public void encode(ByteBuf p_320370_, byte[] p_331189_) {
+            FriendlyByteBuf.writeByteArray(p_320370_, p_331189_);
         }
     };
     StreamCodec<ByteBuf, String> STRING_UTF8 = stringUtf8(32767);
@@ -139,30 +_,30 @@
     StreamCodec<ByteBuf, CompoundTag> COMPOUND_TAG = compoundTagCodec(() -> NbtAccounter.create(2097152L));
     StreamCodec<ByteBuf, CompoundTag> TRUSTED_COMPOUND_TAG = compoundTagCodec(NbtAccounter::unlimitedHeap);
     StreamCodec<ByteBuf, Optional<CompoundTag>> OPTIONAL_COMPOUND_TAG = new StreamCodec<ByteBuf, Optional<CompoundTag>>() {
-        public Optional<CompoundTag> decode(ByteBuf p_320103_) {
-            return Optional.ofNullable(FriendlyByteBuf.readNbt(p_320103_));
+        public Optional<CompoundTag> decode(ByteBuf p_319992_) {
+            return Optional.ofNullable(FriendlyByteBuf.readNbt(p_319992_));
         }
 
-        public void encode(ByteBuf p_320012_, Optional<CompoundTag> p_340971_) {
-            FriendlyByteBuf.writeNbt(p_320012_, p_340971_.orElse(null));
+        public void encode(ByteBuf p_320613_, Optional<CompoundTag> p_330370_) {
+            FriendlyByteBuf.writeNbt(p_320613_, p_330370_.orElse(null));
         }
     };
     StreamCodec<ByteBuf, Vector3f> VECTOR3F = new StreamCodec<ByteBuf, Vector3f>() {
-        public Vector3f decode(ByteBuf p_319897_) {
-            return FriendlyByteBuf.readVector3f(p_319897_);
+        public Vector3f decode(ByteBuf p_320103_) {
+            return FriendlyByteBuf.readVector3f(p_320103_);
         }
 
-        public void encode(ByteBuf p_320441_, Vector3f p_341536_) {
-            FriendlyByteBuf.writeVector3f(p_320441_, p_341536_);
+        public void encode(ByteBuf p_320012_, Vector3f p_330529_) {
+            FriendlyByteBuf.writeVector3f(p_320012_, p_330529_);
         }
     };
     StreamCodec<ByteBuf, Quaternionf> QUATERNIONF = new StreamCodec<ByteBuf, Quaternionf>() {
-        public Quaternionf decode(ByteBuf p_324083_) {
-            return FriendlyByteBuf.readQuaternion(p_324083_);
+        public Quaternionf decode(ByteBuf p_319897_) {
+            return FriendlyByteBuf.readQuaternion(p_319897_);
         }
 
-        public void encode(ByteBuf p_324192_, Quaternionf p_340541_) {
-            FriendlyByteBuf.writeQuaternion(p_324192_, p_340541_);
+        public void encode(ByteBuf p_320441_, Quaternionf p_331734_) {
+            FriendlyByteBuf.writeQuaternion(p_320441_, p_331734_);
         }
     };
     StreamCodec<ByteBuf, PropertyMap> GAME_PROFILE_PROPERTIES = new StreamCodec<ByteBuf, PropertyMap>() {
@@ -171,14 +_,14 @@
         private static final int MAX_PROPERTY_SIGNATURE_LENGTH = 1024;
         private static final int MAX_PROPERTIES = 16;
 
-        public PropertyMap decode(ByteBuf p_331129_) {
-            int i = ByteBufCodecs.readCount(p_331129_, 16);
+        public PropertyMap decode(ByteBuf p_332022_) {
+            int i = ByteBufCodecs.readCount(p_332022_, 16);
             PropertyMap propertymap = new PropertyMap();
 
-            for(int j = 0; j < i; ++j) {
-                String s = Utf8String.read(p_331129_, 64);
-                String s1 = Utf8String.read(p_331129_, 32767);
-                String s2 = FriendlyByteBuf.readNullable(p_331129_, p_340942_ -> Utf8String.read(p_340942_, 1024));
+            for (int j = 0; j < i; j++) {
+                String s = Utf8String.read(p_332022_, 64);
+                String s1 = Utf8String.read(p_332022_, 32767);
+                String s2 = FriendlyByteBuf.readNullable(p_332022_, p_330807_ -> Utf8String.read(p_330807_, 1024));
                 Property property = new Property(s, s1, s2);
                 propertymap.put(property.name(), property);
             }
@@ -186,43 +_,43 @@
             return propertymap;
         }
 
-        public void encode(ByteBuf p_331394_, PropertyMap p_340592_) {
-            ByteBufCodecs.writeCount(p_331394_, p_340592_.size(), 16);
+        public void encode(ByteBuf p_330916_, PropertyMap p_331841_) {
+            ByteBufCodecs.writeCount(p_330916_, p_331841_.size(), 16);
 
-            for(Property property : p_340592_.values()) {
-                Utf8String.write(p_331394_, property.name(), 64);
-                Utf8String.write(p_331394_, property.value(), 32767);
-                FriendlyByteBuf.writeNullable(p_331394_, property.signature(), (p_341490_, p_341759_) -> Utf8String.write(p_341490_, p_341759_, 1024));
+            for (Property property : p_331841_.values()) {
+                Utf8String.write(p_330916_, property.name(), 64);
+                Utf8String.write(p_330916_, property.value(), 32767);
+                FriendlyByteBuf.writeNullable(p_330916_, property.signature(), (p_330760_, p_331768_) -> Utf8String.write(p_330760_, p_331768_, 1024));
             }
         }
     };
     StreamCodec<ByteBuf, GameProfile> GAME_PROFILE = new StreamCodec<ByteBuf, GameProfile>() {
-        public GameProfile decode(ByteBuf p_341368_) {
-            UUID uuid = UUIDUtil.STREAM_CODEC.decode(p_341368_);
-            String s = Utf8String.read(p_341368_, 16);
+        public GameProfile decode(ByteBuf p_331129_) {
+            UUID uuid = UUIDUtil.STREAM_CODEC.decode(p_331129_);
+            String s = Utf8String.read(p_331129_, 16);
             GameProfile gameprofile = new GameProfile(uuid, s);
-            gameprofile.getProperties().putAll(ByteBufCodecs.GAME_PROFILE_PROPERTIES.decode(p_341368_));
+            gameprofile.getProperties().putAll(ByteBufCodecs.GAME_PROFILE_PROPERTIES.decode(p_331129_));
             return gameprofile;
         }
 
-        public void encode(ByteBuf p_342502_, GameProfile p_341105_) {
-            UUIDUtil.STREAM_CODEC.encode(p_342502_, p_341105_.getId());
-            Utf8String.write(p_342502_, p_341105_.getName(), 16);
-            ByteBufCodecs.GAME_PROFILE_PROPERTIES.encode(p_342502_, p_341105_.getProperties());
+        public void encode(ByteBuf p_331394_, GameProfile p_330726_) {
+            UUIDUtil.STREAM_CODEC.encode(p_331394_, p_330726_.getId());
+            Utf8String.write(p_331394_, p_330726_.getName(), 16);
+            ByteBufCodecs.GAME_PROFILE_PROPERTIES.encode(p_331394_, p_330726_.getProperties());
         }
     };
 
     static StreamCodec<ByteBuf, byte[]> byteArray(final int p_324182_) {
         return new StreamCodec<ByteBuf, byte[]>() {
-            public byte[] decode(ByteBuf p_319947_) {
-                return FriendlyByteBuf.readByteArray(p_319947_, p_324182_);
+            public byte[] decode(ByteBuf p_320599_) {
+                return FriendlyByteBuf.readByteArray(p_320599_, p_324182_);
             }
 
-            public void encode(ByteBuf p_320370_, byte[] p_331189_) {
-                if (p_331189_.length > p_324182_) {
-                    throw new EncoderException("ByteArray with size " + p_331189_.length + " is bigger than allowed " + p_324182_);
+            public void encode(ByteBuf p_320880_, byte[] p_324207_) {
+                if (p_324207_.length > p_324182_) {
+                    throw new EncoderException("ByteArray with size " + p_324207_.length + " is bigger than allowed " + p_324182_);
                 } else {
-                    FriendlyByteBuf.writeByteArray(p_320370_, p_331189_);
+                    FriendlyByteBuf.writeByteArray(p_320880_, p_324207_);
                 }
             }
         };
@@ -230,20 +_,20 @@
 
     static StreamCodec<ByteBuf, String> stringUtf8(final int p_320693_) {
         return new StreamCodec<ByteBuf, String>() {
-            public String decode(ByteBuf p_332176_) {
-                return Utf8String.read(p_332176_, p_320693_);
+            public String decode(ByteBuf p_320167_) {
+                return Utf8String.read(p_320167_, p_320693_);
             }
 
-            public void encode(ByteBuf p_331068_, String p_342128_) {
-                Utf8String.write(p_331068_, p_342128_, p_320693_);
+            public void encode(ByteBuf p_320240_, String p_330603_) {
+                Utf8String.write(p_320240_, p_330603_, p_320693_);
             }
         };
     }
 
     static StreamCodec<ByteBuf, Tag> tagCodec(final Supplier<NbtAccounter> p_320506_) {
         return new StreamCodec<ByteBuf, Tag>() {
-            public Tag decode(ByteBuf p_342226_) {
-                Tag tag = FriendlyByteBuf.readNbt(p_342226_, p_320506_.get());
+            public Tag decode(ByteBuf p_332176_) {
+                Tag tag = FriendlyByteBuf.readNbt(p_332176_, p_320506_.get());
                 if (tag == null) {
                     throw new DecoderException("Expected non-null compound tag");
                 } else {
@@ -251,24 +_,24 @@
                 }
             }
 
-            public void encode(ByteBuf p_341169_, Tag p_341010_) {
-                if (p_341010_ == EndTag.INSTANCE) {
+            public void encode(ByteBuf p_331068_, Tag p_330299_) {
+                if (p_330299_ == EndTag.INSTANCE) {
                     throw new EncoderException("Expected non-null compound tag");
                 } else {
-                    FriendlyByteBuf.writeNbt(p_341169_, p_341010_);
+                    FriendlyByteBuf.writeNbt(p_331068_, p_330299_);
                 }
             }
         };
     }
 
     static StreamCodec<ByteBuf, CompoundTag> compoundTagCodec(Supplier<NbtAccounter> p_331128_) {
-        return tagCodec(p_331128_).map(p_320934_ -> {
-            if (p_320934_ instanceof CompoundTag) {
-                return (CompoundTag)p_320934_;
+        return tagCodec(p_331128_).map(p_339405_ -> {
+            if (p_339405_ instanceof CompoundTag) {
+                return (CompoundTag)p_339405_;
             } else {
-                throw new DecoderException("Not a compound tag: " + p_320934_);
+                throw new DecoderException("Not a compound tag: " + p_339405_);
             }
-        }, p_330975_ -> p_330975_);
+        }, p_330975_ -> (Tag)p_330975_);
     }
 
     static <T> StreamCodec<ByteBuf, T> fromCodecTrusted(Codec<T> p_331105_) {
@@ -282,13 +_,10 @@
     static <T> StreamCodec<ByteBuf, T> fromCodec(Codec<T> p_330943_, Supplier<NbtAccounter> p_330382_) {
         return tagCodec(p_330382_)
             .map(
-                p_320291_ -> Util.getOrThrow(
-                        p_330943_.parse(NbtOps.INSTANCE, p_320291_), p_320724_ -> new DecoderException("Failed to decode: " + p_320724_ + " " + p_320291_)
-                    ),
-                p_320777_ -> Util.getOrThrow(
-                        p_330943_.encodeStart(NbtOps.INSTANCE, p_320777_),
-                        p_320664_ -> new EncoderException("Failed to encode: " + p_320664_ + " " + p_320777_)
-                    )
+                p_337514_ -> p_330943_.parse(NbtOps.INSTANCE, p_337514_)
+                        .getOrThrow(p_339407_ -> new DecoderException("Failed to decode: " + p_339407_ + " " + p_337514_)),
+                p_337516_ -> p_330943_.encodeStart(NbtOps.INSTANCE, (T)p_337516_)
+                        .getOrThrow(p_339409_ -> new EncoderException("Failed to encode: " + p_339409_ + " " + p_337516_))
             );
     }
 
@@ -303,34 +_,33 @@
     static <T> StreamCodec<RegistryFriendlyByteBuf, T> fromCodecWithRegistries(final Codec<T> p_331571_, Supplier<NbtAccounter> p_331922_) {
         final StreamCodec<ByteBuf, Tag> streamcodec = tagCodec(p_331922_);
         return new StreamCodec<RegistryFriendlyByteBuf, T>() {
-            public T decode(RegistryFriendlyByteBuf p_342562_) {
-                Tag tag = streamcodec.decode(p_342562_);
-                RegistryOps<Tag> registryops = p_342562_.registryAccess().createSerializationContext(NbtOps.INSTANCE);
-                return Util.getOrThrow(p_331571_.parse(registryops, tag), p_341631_ -> new DecoderException("Failed to decode: " + p_341631_ + " " + tag));
+            public T decode(RegistryFriendlyByteBuf p_331380_) {
+                Tag tag = streamcodec.decode(p_331380_);
+                RegistryOps<Tag> registryops = p_331380_.registryAccess().createSerializationContext(NbtOps.INSTANCE);
+                return p_331571_.parse(registryops, tag).getOrThrow(p_339413_ -> new DecoderException("Failed to decode: " + p_339413_ + " " + tag));
             }
 
-            public void encode(RegistryFriendlyByteBuf p_340877_, T p_341761_) {
-                RegistryOps<Tag> registryops = p_340877_.registryAccess().createSerializationContext(NbtOps.INSTANCE);
-                Tag tag = Util.getOrThrow(
-                    p_331571_.encodeStart(registryops, p_341761_), p_341698_ -> new EncoderException("Failed to encode: " + p_341698_ + " " + p_341761_)
-                );
-                streamcodec.encode(p_340877_, tag);
+            public void encode(RegistryFriendlyByteBuf p_330803_, T p_332115_) {
+                RegistryOps<Tag> registryops = p_330803_.registryAccess().createSerializationContext(NbtOps.INSTANCE);
+                Tag tag = p_331571_.encodeStart(registryops, p_332115_)
+                    .getOrThrow(p_339411_ -> new EncoderException("Failed to encode: " + p_339411_ + " " + p_332115_));
+                streamcodec.encode(p_330803_, tag);
             }
         };
     }
 
     static <B extends ByteBuf, V> StreamCodec<B, Optional<V>> optional(final StreamCodec<B, V> p_320522_) {
         return new StreamCodec<B, Optional<V>>() {
-            public Optional<V> decode(B p_324595_) {
-                return p_324595_.readBoolean() ? Optional.of(p_320522_.decode(p_324595_)) : Optional.empty();
+            public Optional<V> decode(B p_324083_) {
+                return p_324083_.readBoolean() ? Optional.of(p_320522_.decode(p_324083_)) : Optional.empty();
             }
 
-            public void encode(B p_324147_, Optional<V> p_342591_) {
-                if (p_342591_.isPresent()) {
-                    p_324147_.writeBoolean(true);
-                    p_320522_.encode(p_324147_, p_342591_.get());
+            public void encode(B p_324192_, Optional<V> p_331101_) {
+                if (p_331101_.isPresent()) {
+                    p_324192_.writeBoolean(true);
+                    p_320522_.encode(p_324192_, p_331101_.get());
                 } else {
-                    p_324147_.writeBoolean(false);
+                    p_324192_.writeBoolean(false);
                 }
             }
         };
@@ -361,22 +_,22 @@
         final IntFunction<C> p_332198_, final StreamCodec<? super B, V> p_332183_, final int p_332173_
     ) {
         return new StreamCodec<B, C>() {
-            public C decode(B p_324220_) {
-                int i = ByteBufCodecs.readCount(p_324220_, p_332173_);
+            public C decode(B p_324595_) {
+                int i = ByteBufCodecs.readCount(p_324595_, p_332173_);
                 C c = p_332198_.apply(Math.min(i, 65536));
 
-                for(int j = 0; j < i; ++j) {
-                    c.add(p_332183_.decode(p_324220_));
+                for (int j = 0; j < i; j++) {
+                    c.add(p_332183_.decode(p_324595_));
                 }
 
                 return c;
             }
 
-            public void encode(B p_323874_, C p_341479_) {
-                ByteBufCodecs.writeCount(p_323874_, p_341479_.size(), p_332173_);
+            public void encode(B p_324147_, C p_330874_) {
+                ByteBufCodecs.writeCount(p_324147_, p_330874_.size(), p_332173_);
 
-                for(V v : p_341479_) {
-                    p_332183_.encode(p_323874_, v);
+                for (V v : p_330874_) {
+                    p_332183_.encode(p_324147_, v);
                 }
             }
         };
@@ -404,21 +_,21 @@
         final IntFunction<? extends M> p_331325_, final StreamCodec<? super B, K> p_331975_, final StreamCodec<? super B, V> p_331254_, final int p_330938_
     ) {
         return new StreamCodec<B, M>() {
-            public void encode(B p_331539_, M p_342481_) {
-                ByteBufCodecs.writeCount(p_331539_, p_342481_.size(), p_330938_);
-                p_342481_.forEach((p_339758_, p_339759_) -> {
-                    p_331975_.encode(p_331539_, p_339758_);
-                    p_331254_.encode(p_331539_, p_339759_);
+            public void encode(B p_323874_, M p_331240_) {
+                ByteBufCodecs.writeCount(p_323874_, p_331240_.size(), p_330938_);
+                p_331240_.forEach((p_331386_, p_331781_) -> {
+                    p_331975_.encode(p_323874_, (K)p_331386_);
+                    p_331254_.encode(p_323874_, (V)p_331781_);
                 });
             }
 
-            public M decode(B p_331901_) {
-                int i = ByteBufCodecs.readCount(p_331901_, p_330938_);
-                M m = p_331325_.apply(Math.min(i, 65536));
+            public M decode(B p_324220_) {
+                int i = ByteBufCodecs.readCount(p_324220_, p_330938_);
+                M m = (M)p_331325_.apply(Math.min(i, 65536));
 
-                for(int j = 0; j < i; ++j) {
-                    K k = p_331975_.decode(p_331901_);
-                    V v = p_331254_.decode(p_331901_);
+                for (int j = 0; j < i; j++) {
+                    K k = p_331975_.decode(p_324220_);
+                    V v = p_331254_.decode(p_324220_);
                     m.put(k, v);
                 }
 
@@ -429,17 +_,17 @@
 
     static <B extends ByteBuf, L, R> StreamCodec<B, Either<L, R>> either(final StreamCodec<? super B, L> p_331983_, final StreamCodec<? super B, R> p_332156_) {
         return new StreamCodec<B, Either<L, R>>() {
-            public Either<L, R> decode(B p_332082_) {
-                return p_332082_.readBoolean() ? Either.left(p_331983_.decode(p_332082_)) : Either.right(p_332156_.decode(p_332082_));
+            public Either<L, R> decode(B p_331901_) {
+                return p_331901_.readBoolean() ? Either.left(p_331983_.decode(p_331901_)) : Either.right(p_332156_.decode(p_331901_));
             }
 
-            public void encode(B p_331172_, Either<L, R> p_340726_) {
-                p_340726_.ifLeft(p_340712_ -> {
-                    p_331172_.writeBoolean(true);
-                    p_331983_.encode(p_331172_, p_340712_);
-                }).ifRight(p_340869_ -> {
-                    p_331172_.writeBoolean(false);
-                    p_332156_.encode(p_331172_, p_340869_);
+            public void encode(B p_331539_, Either<L, R> p_330504_) {
+                p_330504_.ifLeft(p_330214_ -> {
+                    p_331539_.writeBoolean(true);
+                    p_331983_.encode(p_331539_, (L)p_330214_);
+                }).ifRight(p_330242_ -> {
+                    p_331539_.writeBoolean(false);
+                    p_332156_.encode(p_331539_, (R)p_330242_);
                 });
             }
         };
@@ -447,14 +_,14 @@
 
     static <T> StreamCodec<ByteBuf, T> idMapper(final IntFunction<T> p_320877_, final ToIntFunction<T> p_319985_) {
         return new StreamCodec<ByteBuf, T>() {
-            public T decode(ByteBuf p_341020_) {
-                int i = VarInt.read(p_341020_);
+            public T decode(ByteBuf p_332082_) {
+                int i = VarInt.read(p_332082_);
                 return p_320877_.apply(i);
             }
 
-            public void encode(ByteBuf p_342341_, T p_330257_) {
-                int i = p_319985_.applyAsInt(p_330257_);
-                VarInt.write(p_342341_, i);
+            public void encode(ByteBuf p_331172_, T p_331760_) {
+                int i = p_319985_.applyAsInt(p_331760_);
+                VarInt.write(p_331172_, i);
             }
         };
     }
@@ -467,18 +_,22 @@
         final ResourceKey<? extends Registry<T>> p_319942_, final Function<Registry<T>, IdMap<R>> p_320353_
     ) {
         return new StreamCodec<RegistryFriendlyByteBuf, R>() {
-            private IdMap<R> getRegistryOrThrow(RegistryFriendlyByteBuf p_330361_) {
-                return p_320353_.apply(p_330361_.registryAccess().registryOrThrow(p_319942_));
-            }
-
-            public R decode(RegistryFriendlyByteBuf p_331253_) {
-                int i = VarInt.read(p_331253_);
-                return (R)this.getRegistryOrThrow(p_331253_).byIdOrThrow(i);
-            }
-
-            public void encode(RegistryFriendlyByteBuf p_331775_, R p_342413_) {
-                int i = this.getRegistryOrThrow(p_331775_).getIdOrThrow(p_342413_);
-                VarInt.write(p_331775_, i);
+            private IdMap<R> getRegistryOrThrow(RegistryFriendlyByteBuf p_331260_) {
+                var registry = p_331260_.registryAccess().registryOrThrow(p_319942_);
+                if (net.neoforged.neoforge.registries.RegistryManager.isNonSyncedBuiltInRegistry(registry)) {
+                    throw new IllegalStateException("Cannot use ID syncing for non-synced built-in registry: " + registry.key());
+                }
+                return p_320353_.apply(registry);
+            }
+
+            public R decode(RegistryFriendlyByteBuf p_331078_) {
+                int i = VarInt.read(p_331078_);
+                return (R)this.getRegistryOrThrow(p_331078_).byIdOrThrow(i);
+            }
+
+            public void encode(RegistryFriendlyByteBuf p_331573_, R p_330257_) {
+                int i = this.getRegistryOrThrow(p_331573_).getIdOrThrow(p_330257_);
+                VarInt.write(p_331573_, i);
             }
         };
     }
@@ -497,24 +_,24 @@
         return new StreamCodec<RegistryFriendlyByteBuf, Holder<T>>() {
             private static final int DIRECT_HOLDER_ID = 0;
 
-            private IdMap<Holder<T>> getRegistryOrThrow(RegistryFriendlyByteBuf p_341649_) {
-                return p_341649_.registryAccess().registryOrThrow(p_320391_).asHolderIdMap();
-            }
-
-            public Holder<T> decode(RegistryFriendlyByteBuf p_330998_) {
-                int i = VarInt.read(p_330998_);
-                return i == 0 ? Holder.direct(p_320595_.decode(p_330998_)) : (Holder)this.getRegistryOrThrow(p_330998_).byIdOrThrow(i - 1);
-            }
-
-            public void encode(RegistryFriendlyByteBuf p_330557_, Holder<T> p_341448_) {
-                switch(p_341448_.kind()) {
+            private IdMap<Holder<T>> getRegistryOrThrow(RegistryFriendlyByteBuf p_330361_) {
+                return p_330361_.registryAccess().registryOrThrow(p_320391_).asHolderIdMap();
+            }
+
+            public Holder<T> decode(RegistryFriendlyByteBuf p_331253_) {
+                int i = VarInt.read(p_331253_);
+                return i == 0 ? Holder.direct(p_320595_.decode(p_331253_)) : (Holder)this.getRegistryOrThrow(p_331253_).byIdOrThrow(i - 1);
+            }
+
+            public void encode(RegistryFriendlyByteBuf p_331775_, Holder<T> p_330732_) {
+                switch (p_330732_.kind()) {
                     case REFERENCE:
-                        int i = this.getRegistryOrThrow(p_330557_).getIdOrThrow(p_341448_);
-                        VarInt.write(p_330557_, i + 1);
+                        int i = this.getRegistryOrThrow(p_331775_).getIdOrThrow(p_330732_);
+                        VarInt.write(p_331775_, i + 1);
                         break;
                     case DIRECT:
-                        VarInt.write(p_330557_, 0);
-                        p_320595_.encode(p_330557_, p_341448_.value());
+                        VarInt.write(p_331775_, 0);
+                        p_320595_.encode(p_331775_, p_330732_.value());
                 }
             }
         };
@@ -525,32 +_,32 @@
             private static final int NAMED_SET = -1;
             private final StreamCodec<RegistryFriendlyByteBuf, Holder<T>> holderCodec = ByteBufCodecs.holderRegistry(p_332137_);
 
-            public HolderSet<T> decode(RegistryFriendlyByteBuf p_342350_) {
-                int i = VarInt.read(p_342350_) - 1;
+            public HolderSet<T> decode(RegistryFriendlyByteBuf p_330998_) {
+                int i = VarInt.read(p_330998_) - 1;
                 if (i == -1) {
-                    Registry<T> registry = p_342350_.registryAccess().registryOrThrow(p_332137_);
-                    return registry.getTag(TagKey.create(p_332137_, ResourceLocation.STREAM_CODEC.decode(p_342350_))).orElseThrow();
+                    Registry<T> registry = p_330998_.registryAccess().registryOrThrow(p_332137_);
+                    return registry.getTag(TagKey.create(p_332137_, ResourceLocation.STREAM_CODEC.decode(p_330998_))).orElseThrow();
                 } else {
                     List<Holder<T>> list = new ArrayList<>(Math.min(i, 65536));
 
-                    for(int j = 0; j < i; ++j) {
-                        list.add(this.holderCodec.decode(p_342350_));
+                    for (int j = 0; j < i; j++) {
+                        list.add(this.holderCodec.decode(p_330998_));
                     }
 
                     return HolderSet.direct(list);
                 }
             }
 
-            public void encode(RegistryFriendlyByteBuf p_340559_, HolderSet<T> p_341929_) {
-                Optional<TagKey<T>> optional = p_341929_.unwrapKey();
+            public void encode(RegistryFriendlyByteBuf p_330557_, HolderSet<T> p_331716_) {
+                Optional<TagKey<T>> optional = p_331716_.unwrapKey();
                 if (optional.isPresent()) {
-                    VarInt.write(p_340559_, 0);
-                    ResourceLocation.STREAM_CODEC.encode(p_340559_, optional.get().location());
+                    VarInt.write(p_330557_, 0);
+                    ResourceLocation.STREAM_CODEC.encode(p_330557_, optional.get().location());
                 } else {
-                    VarInt.write(p_340559_, p_341929_.size() + 1);
+                    VarInt.write(p_330557_, p_331716_.size() + 1);
 
-                    for(Holder<T> holder : p_341929_) {
-                        this.holderCodec.encode(p_340559_, holder);
+                    for (Holder<T> holder : p_331716_) {
+                        this.holderCodec.encode(p_330557_, holder);
                     }
                 }
             }
