--- a/net/minecraft/network/codec/IdDispatchCodec.java
+++ b/net/minecraft/network/codec/IdDispatchCodec.java
@@ -28,7 +_,7 @@
             IdDispatchCodec.Entry<B, V, T> entry = this.byId.get(i);
 
             try {
-                return entry.serializer.decode(p_320374_);
+                return (V)entry.serializer.decode(p_320374_);
             } catch (Exception exception) {
                 throw new DecoderException("Failed to decode packet '" + entry.type + "'", exception);
             }
@@ -38,7 +_,7 @@
     }
 
     public void encode(B p_319955_, V p_320138_) {
-        T t = this.typeGetter.apply(p_320138_);
+        T t = (T)this.typeGetter.apply(p_320138_);
         int i = this.toId.getOrDefault(t, -1);
         if (i == -1) {
             throw new EncoderException("Sending unknown packet '" + t + "'");
@@ -76,7 +_,7 @@
             Object2IntOpenHashMap<T> object2intopenhashmap = new Object2IntOpenHashMap<>();
             object2intopenhashmap.defaultReturnValue(-2);
 
-            for(IdDispatchCodec.Entry<B, V, T> entry : this.entries) {
+            for (IdDispatchCodec.Entry<B, V, T> entry : this.entries) {
                 int i = object2intopenhashmap.size();
                 int j = object2intopenhashmap.putIfAbsent(entry.type, i);
                 if (j != -2) {
