--- a/net/minecraft/client/tutorial/FindTreeTutorialStepInstance.java
+++ b/net/minecraft/client/tutorial/FindTreeTutorialStepInstance.java
@@ -32,7 +_,7 @@
 
     @Override
     public void tick() {
-        ++this.timeWaiting;
+        this.timeWaiting++;
         if (!this.tutorial.isSurvival()) {
             this.tutorial.setStep(TutorialSteps.NONE);
         } else {
@@ -81,7 +_,7 @@
     }
 
     public static boolean hasPunchedTreesPreviously(LocalPlayer p_120504_) {
-        for(Holder<Block> holder : BuiltInRegistries.BLOCK.getTagOrEmpty(BlockTags.COMPLETES_FIND_TREE_TUTORIAL)) {
+        for (Holder<Block> holder : BuiltInRegistries.BLOCK.getTagOrEmpty(BlockTags.COMPLETES_FIND_TREE_TUTORIAL)) {
             Block block = holder.value();
             if (p_120504_.getStats().getValue(Stats.BLOCK_MINED.get(block)) > 0) {
                 return true;
