--- a/net/minecraft/client/tutorial/MovementTutorialStepInstance.java
+++ b/net/minecraft/client/tutorial/MovementTutorialStepInstance.java
@@ -36,14 +_,14 @@
 
     @Override
     public void tick() {
-        ++this.timeWaiting;
+        this.timeWaiting++;
         if (this.moved) {
-            ++this.timeMoved;
+            this.timeMoved++;
             this.moved = false;
         }
 
         if (this.turned) {
-            ++this.timeLooked;
+            this.timeLooked++;
             this.turned = false;
         }
 
