--- a/net/minecraft/client/tutorial/TutorialSteps.java
+++ b/net/minecraft/client/tutorial/TutorialSteps.java
@@ -30,7 +_,7 @@
     }
 
     public static TutorialSteps getByName(String p_120643_) {
-        for(TutorialSteps tutorialsteps : values()) {
+        for (TutorialSteps tutorialsteps : values()) {
             if (tutorialsteps.name.equals(p_120643_)) {
                 return tutorialsteps;
             }
