--- a/net/minecraft/client/tutorial/Tutorial.java
+++ b/net/minecraft/client/tutorial/Tutorial.java
@@ -120,11 +_,7 @@
     }
 
     public boolean isSurvival() {
-        if (this.minecraft.gameMode == null) {
-            return false;
-        } else {
-            return this.minecraft.gameMode.getPlayerMode() == GameType.SURVIVAL;
-        }
+        return this.minecraft.gameMode == null ? false : this.minecraft.gameMode.getPlayerMode() == GameType.SURVIVAL;
     }
 
     public static Component key(String p_120593_) {
