--- a/net/minecraft/client/gui/screens/achievement/StatsScreen.java
+++ b/net/minecraft/client/gui/screens/achievement/StatsScreen.java
@@ -9,6 +_,7 @@
 import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.LoadingDotsWidget;
 import net.minecraft.client.gui.components.ObjectSelectionList;
@@ -101,7 +_,9 @@
         }
 
         this.layout = headerandfooterlayout;
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_329726_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_329726_);
+        });
         this.repositionElements();
     }
 
@@ -154,9 +_,9 @@
         public GeneralStatisticsList(Minecraft p_96995_) {
             super(p_96995_, StatsScreen.this.width, StatsScreen.this.height - 33 - 58, 33, 14);
             ObjectArrayList<Stat<ResourceLocation>> objectarraylist = new ObjectArrayList<>(Stats.CUSTOM.iterator());
-            objectarraylist.sort(Comparator.comparing(p_96997_ -> I18n.get(StatsScreen.getTranslationKey(p_96997_))));
+            objectarraylist.sort(Comparator.comparing(p_96997_ -> I18n.get(StatsScreen.getTranslationKey((Stat<ResourceLocation>)p_96997_))));
 
-            for(Stat<ResourceLocation> stat : objectarraylist) {
+            for (Stat<ResourceLocation> stat : objectarraylist) {
                 this.addEntry(new StatsScreen.GeneralStatisticsList.Entry(stat));
             }
         }
@@ -241,10 +_,10 @@
             this.setRenderHeader(true, 22);
             Set<Item> set = Sets.newIdentityHashSet();
 
-            for(Item item : BuiltInRegistries.ITEM) {
+            for (Item item : BuiltInRegistries.ITEM) {
                 boolean flag = false;
 
-                for(StatType<Item> stattype : this.itemColumns) {
+                for (StatType<Item> stattype : this.itemColumns) {
                     if (stattype.contains(item) && StatsScreen.this.stats.getValue(stattype.get(item)) > 0) {
                         flag = true;
                     }
@@ -255,10 +_,10 @@
                 }
             }
 
-            for(Block block : BuiltInRegistries.BLOCK) {
+            for (Block block : BuiltInRegistries.BLOCK) {
                 boolean flag1 = false;
 
-                for(StatType<Block> stattype1 : this.blockColumns) {
+                for (StatType<Block> stattype1 : this.blockColumns) {
                     if (stattype1.contains(block) && StatsScreen.this.stats.getValue(stattype1.get(block)) > 0) {
                         flag1 = true;
                     }
@@ -271,7 +_,7 @@
 
             set.remove(Items.AIR);
 
-            for(Item item1 : set) {
+            for (Item item1 : set) {
                 this.addEntry(new StatsScreen.ItemStatisticsList.ItemRow(item1));
             }
         }
@@ -286,7 +_,7 @@
                 this.headerPressed = -1;
             }
 
-            for(int i = 0; i < this.iconSprites.length; ++i) {
+            for (int i = 0; i < this.iconSprites.length; i++) {
                 ResourceLocation resourcelocation = this.headerPressed == i ? StatsScreen.SLOT_SPRITE : StatsScreen.HEADER_SPRITE;
                 p_282214_.blitSprite(resourcelocation, p_97050_ + this.getColumnX(i) - 18, p_97051_ + 1, 0, 18, 18);
             }
@@ -297,7 +_,7 @@
                 p_282214_.blitSprite(resourcelocation1, p_97050_ + j, p_97051_ + 1, 0, 18, 18);
             }
 
-            for(int k = 0; k < this.iconSprites.length; ++k) {
+            for (int k = 0; k < this.iconSprites.length; k++) {
                 int l = this.headerPressed == k ? 1 : 0;
                 p_282214_.blitSprite(this.iconSprites[k], p_97050_ + this.getColumnX(k) - 18 + l, p_97051_ + 1 + l, 0, 18, 18);
             }
@@ -312,7 +_,7 @@
         protected boolean clickedHeader(int p_97036_, int p_97037_) {
             this.headerPressed = -1;
 
-            for(int i = 0; i < this.iconSprites.length; ++i) {
+            for (int i = 0; i < this.iconSprites.length; i++) {
                 int j = p_97036_ - this.getColumnX(i);
                 if (j >= -36 && j <= 0) {
                     this.headerPressed = i;
@@ -359,7 +_,7 @@
                     Component component = null;
                     int j = p_97046_ - i;
 
-                    for(int k = 0; k < this.iconSprites.length; ++k) {
+                    for (int k = 0; k < this.iconSprites.length; k++) {
                         int l = this.getColumnX(k);
                         if (j >= l - 18 && j <= l) {
                             component = this.getColumn(k).getDisplayName();
@@ -416,10 +_,9 @@
                 p_283614_.blitSprite(StatsScreen.SLOT_SPRITE, p_97084_, p_97083_, 0, 18, 18);
                 p_283614_.renderFakeItem(this.item.getDefaultInstance(), p_97084_ + 1, p_97083_ + 1);
                 if (StatsScreen.this.itemStatsList != null) {
-                    for(int i = 0; i < StatsScreen.this.itemStatsList.blockColumns.size(); ++i) {
-                        Item itemx = this.item;
+                    for (int i = 0; i < StatsScreen.this.itemStatsList.blockColumns.size(); i++) {
                         Stat<Block> stat;
-                        if (itemx instanceof BlockItem blockitem) {
+                        if (this.item instanceof BlockItem blockitem) {
                             stat = StatsScreen.this.itemStatsList.blockColumns.get(i).get(blockitem.getBlock());
                         } else {
                             stat = null;
@@ -428,7 +_,7 @@
                         this.renderStat(p_283614_, stat, p_97084_ + ItemStatisticsList.this.getColumnX(i), p_97083_ + p_97086_ / 2 - 9 / 2, p_97082_ % 2 == 0);
                     }
 
-                    for(int j = 0; j < StatsScreen.this.itemStatsList.itemColumns.size(); ++j) {
+                    for (int j = 0; j < StatsScreen.this.itemStatsList.itemColumns.size(); j++) {
                         this.renderStat(
                             p_283614_,
                             StatsScreen.this.itemStatsList.itemColumns.get(j).get(this.item),
@@ -485,7 +_,7 @@
         public MobsStatisticsList(Minecraft p_97100_) {
             super(p_97100_, StatsScreen.this.width, StatsScreen.this.height - 33 - 58, 33, 9 * 4);
 
-            for(EntityType<?> entitytype : BuiltInRegistries.ENTITY_TYPE) {
+            for (EntityType<?> entitytype : BuiltInRegistries.ENTITY_TYPE) {
                 if (StatsScreen.this.stats.getValue(Stats.ENTITY_KILLED.get(entitytype)) > 0
                     || StatsScreen.this.stats.getValue(Stats.ENTITY_KILLED_BY.get(entitytype)) > 0) {
                     this.addEntry(new StatsScreen.MobsStatisticsList.MobRow(entitytype));
