--- a/net/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
+++ b/net/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
@@ -82,7 +_,7 @@
             if (!blockstate.is(this.state.getBlock())) {
                 return false;
             } else {
-                for(Property<?> property : this.properties) {
+                for (Property<?> property : this.properties) {
                     if (blockstate.getValue(property) != this.state.getValue(property)) {
                         return false;
                     }
@@ -124,13 +_,13 @@
             if (!blockstate.is(this.tag)) {
                 return false;
             } else {
-                for(Entry<String, String> entry : this.vagueProperties.entrySet()) {
+                for (Entry<String, String> entry : this.vagueProperties.entrySet()) {
                     Property<?> property = blockstate.getBlock().getStateDefinition().getProperty(entry.getKey());
                     if (property == null) {
                         return false;
                     }
 
-                    Comparable<?> comparable = (Comparable)property.getValue(entry.getValue()).orElse(null);
+                    Comparable<?> comparable = (Comparable<?>)property.getValue(entry.getValue()).orElse(null);
                     if (comparable == null) {
                         return false;
                     }
