--- a/net/minecraft/commands/arguments/blocks/BlockStateParser.java
+++ b/net/minecraft/commands/arguments/blocks/BlockStateParser.java
@@ -192,7 +_,7 @@
     private CompletableFuture<Suggestions> suggestPropertyName(SuggestionsBuilder p_234729_) {
         String s = p_234729_.getRemaining().toLowerCase(Locale.ROOT);
 
-        for(Property<?> property : this.state.getProperties()) {
+        for (Property<?> property : this.state.getProperties()) {
             if (!this.properties.containsKey(property) && property.getName().startsWith(s)) {
                 p_234729_.suggest(property.getName() + "=");
             }
@@ -204,8 +_,8 @@
     private CompletableFuture<Suggestions> suggestVaguePropertyName(SuggestionsBuilder p_234731_) {
         String s = p_234731_.getRemaining().toLowerCase(Locale.ROOT);
         if (this.tag != null) {
-            for(Holder<Block> holder : this.tag) {
-                for(Property<?> property : holder.value().getStateDefinition().getProperties()) {
+            for (Holder<Block> holder : this.tag) {
+                for (Property<?> property : holder.value().getStateDefinition().getProperties()) {
                     if (!this.vagueProperties.containsKey(property.getName()) && property.getName().startsWith(s)) {
                         p_234731_.suggest(property.getName() + "=");
                     }
@@ -229,7 +_,7 @@
             return this.state.hasBlockEntity();
         } else {
             if (this.tag != null) {
-                for(Holder<Block> holder : this.tag) {
+                for (Holder<Block> holder : this.tag) {
                     if (holder.value().defaultBlockState().hasBlockEntity()) {
                         return true;
                     }
@@ -261,7 +_,7 @@
     }
 
     private static <T extends Comparable<T>> SuggestionsBuilder addSuggestions(SuggestionsBuilder p_116787_, Property<T> p_116788_) {
-        for(T t : p_116788_.getPossibleValues()) {
+        for (T t : p_116788_.getPossibleValues()) {
             if (t instanceof Integer integer) {
                 p_116787_.suggest(integer);
             } else {
@@ -275,7 +_,7 @@
     private CompletableFuture<Suggestions> suggestVaguePropertyValue(SuggestionsBuilder p_234686_, String p_234687_) {
         boolean flag = false;
         if (this.tag != null) {
-            for(Holder<Block> holder : this.tag) {
+            for (Holder<Block> holder : this.tag) {
                 Block block = holder.value();
                 Property<?> property = block.getStateDefinition().getProperty(p_234687_);
                 if (property != null) {
@@ -283,7 +_,7 @@
                 }
 
                 if (!flag) {
-                    for(Property<?> property1 : block.getStateDefinition().getProperties()) {
+                    for (Property<?> property1 : block.getStateDefinition().getProperties()) {
                         if (!this.vagueProperties.containsKey(property1.getName())) {
                             flag = true;
                             break;
@@ -306,7 +_,7 @@
             boolean flag = false;
             boolean flag1 = false;
 
-            for(Holder<Block> holder : this.tag) {
+            for (Holder<Block> holder : this.tag) {
                 Block block = holder.value();
                 flag |= !block.getStateDefinition().getProperties().isEmpty();
                 flag1 |= block.defaultBlockState().hasBlockEntity();
@@ -386,7 +_,7 @@
         this.suggestions = this::suggestPropertyNameOrEnd;
         this.reader.skipWhitespace();
 
-        while(this.reader.canRead() && this.reader.peek() != ']') {
+        while (this.reader.canRead() && this.reader.peek() != ']') {
             this.reader.skipWhitespace();
             int i = this.reader.getCursor();
             String s = this.reader.readString();
@@ -440,7 +_,7 @@
         int i = -1;
         this.reader.skipWhitespace();
 
-        while(this.reader.canRead() && this.reader.peek() != ']') {
+        while (this.reader.canRead() && this.reader.peek() != ']') {
             this.reader.skipWhitespace();
             int j = this.reader.getCursor();
             String s = this.reader.readString();
@@ -508,7 +_,7 @@
             stringbuilder.append('[');
             boolean flag = false;
 
-            for(Entry<Property<?>, Comparable<?>> entry : p_116770_.getValues().entrySet()) {
+            for (Entry<Property<?>, Comparable<?>> entry : p_116770_.getValues().entrySet()) {
                 if (flag) {
                     stringbuilder.append(',');
                 }
