--- a/net/minecraft/data/advancements/packs/VanillaAdvancementProvider.java
+++ b/net/minecraft/data/advancements/packs/VanillaAdvancementProvider.java
@@ -16,8 +_,7 @@
                 new VanillaHusbandryAdvancements(),
                 new VanillaAdventureAdvancements(),
                 new VanillaNetherAdvancements(),
-                new VanillaStoryAdvancements(),
-                new PoisonousPotatoAdvancements()
+                new VanillaStoryAdvancements()
             )
         );
     }
