--- a/net/minecraft/data/advancements/packs/VanillaHusbandryAdvancements.java
+++ b/net/minecraft/data/advancements/packs/VanillaHusbandryAdvancements.java
@@ -101,10 +_,6 @@
         Items.POTATO,
         Items.BAKED_POTATO,
         Items.POISONOUS_POTATO,
-        Items.POISONOUS_POTATO_STICKS,
-        Items.POISONOUS_POTATO_SLICES,
-        Items.POISONOUS_POTATO_FRIES,
-        Items.POISONOUS_POTATO_CHIPS,
         Items.GOLDEN_CARROT,
         Items.PUMPKIN_PIE,
         Items.RABBIT,
@@ -599,7 +_,7 @@
     }
 
     private static Advancement.Builder addFood(Advancement.Builder p_248532_) {
-        for(Item item : EDIBLE_ITEMS) {
+        for (Item item : EDIBLE_ITEMS) {
             p_248532_.addCriterion(BuiltInRegistries.ITEM.getKey(item).getPath(), ConsumeItemTrigger.TriggerInstance.usedItem(item));
         }
 
@@ -609,15 +_,16 @@
     private static Advancement.Builder addBreedable(Advancement.Builder p_266978_, Stream<EntityType<?>> p_267147_, Stream<EntityType<?>> p_267091_) {
         p_267147_.forEach(
             p_300705_ -> p_266978_.addCriterion(
-                    EntityType.getKey(p_300705_).toString(), BredAnimalsTrigger.TriggerInstance.bredAnimals(EntityPredicate.Builder.entity().of(p_300705_))
+                    EntityType.getKey((EntityType<?>)p_300705_).toString(),
+                    BredAnimalsTrigger.TriggerInstance.bredAnimals(EntityPredicate.Builder.entity().of((EntityType<?>)p_300705_))
                 )
         );
         p_267091_.forEach(
             p_300701_ -> p_266978_.addCriterion(
-                    EntityType.getKey(p_300701_).toString(),
+                    EntityType.getKey((EntityType<?>)p_300701_).toString(),
                     BredAnimalsTrigger.TriggerInstance.bredAnimals(
-                        Optional.of(EntityPredicate.Builder.entity().of(p_300701_).build()),
-                        Optional.of(EntityPredicate.Builder.entity().of(p_300701_).build()),
+                        Optional.of(EntityPredicate.Builder.entity().of((EntityType<?>)p_300701_).build()),
+                        Optional.of(EntityPredicate.Builder.entity().of((EntityType<?>)p_300701_).build()),
                         Optional.empty()
                     )
                 )
@@ -626,7 +_,7 @@
     }
 
     private static Advancement.Builder addFishBuckets(Advancement.Builder p_249285_) {
-        for(Item item : FISH_BUCKETS) {
+        for (Item item : FISH_BUCKETS) {
             p_249285_.addCriterion(
                 BuiltInRegistries.ITEM.getKey(item).getPath(), FilledBucketTrigger.TriggerInstance.filledBucket(ItemPredicate.Builder.item().of(item))
             );
@@ -636,7 +_,7 @@
     }
 
     private static Advancement.Builder addFish(Advancement.Builder p_248725_) {
-        for(Item item : FISH) {
+        for (Item item : FISH) {
             p_248725_.addCriterion(
                 BuiltInRegistries.ITEM.getKey(item).getPath(),
                 FishingRodHookedTrigger.TriggerInstance.fishedItem(
