--- a/net/minecraft/data/advancements/packs/VanillaNetherAdvancements.java
+++ b/net/minecraft/data/advancements/packs/VanillaNetherAdvancements.java
@@ -50,7 +_,6 @@
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.RespawnAnchorBlock;
 import net.minecraft.world.level.levelgen.structure.BuiltinStructures;
-import net.minecraft.world.level.levelgen.structure.Structure;
 import net.minecraft.world.level.storage.loot.BuiltInLootTables;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition;
@@ -140,7 +_,7 @@
             .addCriterion(
                 "fortress",
                 PlayerTrigger.TriggerInstance.located(
-                    LocationPredicate.Builder.inStructure(p_256338_.<Structure>lookupOrThrow(Registries.STRUCTURE).getOrThrow(BuiltinStructures.FORTRESS))
+                    LocationPredicate.Builder.inStructure(p_256338_.lookupOrThrow(Registries.STRUCTURE).getOrThrow(BuiltinStructures.FORTRESS))
                 )
             )
             .save(p_249760_, "nether/find_fortress");
@@ -512,9 +_,7 @@
             .addCriterion(
                 "bastion",
                 PlayerTrigger.TriggerInstance.located(
-                    LocationPredicate.Builder.inStructure(
-                        p_256338_.<Structure>lookupOrThrow(Registries.STRUCTURE).getOrThrow(BuiltinStructures.BASTION_REMNANT)
-                    )
+                    LocationPredicate.Builder.inStructure(p_256338_.lookupOrThrow(Registries.STRUCTURE).getOrThrow(BuiltinStructures.BASTION_REMNANT))
                 )
             )
             .save(p_249760_, "nether/find_bastion");
