--- a/net/minecraft/world/level/redstone/CollectingNeighborUpdater.java
+++ b/net/minecraft/world/level/redstone/CollectingNeighborUpdater.java
@@ -52,7 +_,7 @@
     private void addAndRun(BlockPos p_230661_, CollectingNeighborUpdater.NeighborUpdates p_230662_) {
         boolean flag = this.count > 0;
         boolean flag1 = this.maxChainedNeighborUpdates >= 0 && this.count >= this.maxChainedNeighborUpdates;
-        ++this.count;
+        this.count++;
         if (!flag1) {
             if (flag) {
                 this.addedThisLayer.add(p_230662_);
@@ -70,15 +_,15 @@
 
     private void runUpdates() {
         try {
-            while(!this.stack.isEmpty() || !this.addedThisLayer.isEmpty()) {
-                for(int i = this.addedThisLayer.size() - 1; i >= 0; --i) {
+            while (!this.stack.isEmpty() || !this.addedThisLayer.isEmpty()) {
+                for (int i = this.addedThisLayer.size() - 1; i >= 0; i--) {
                     this.stack.push(this.addedThisLayer.get(i));
                 }
 
                 this.addedThisLayer.clear();
                 CollectingNeighborUpdater.NeighborUpdates collectingneighborupdater$neighborupdates = this.stack.peek();
 
-                while(this.addedThisLayer.isEmpty()) {
+                while (this.addedThisLayer.isEmpty()) {
                     if (!collectingneighborupdater$neighborupdates.runNext(this.level)) {
                         this.stack.pop();
                         break;
@@ -113,7 +_,7 @@
             this.sourceBlock = p_230698_;
             this.skipDirection = p_230699_;
             if (NeighborUpdater.UPDATE_ORDER[this.idx] == p_230699_) {
-                ++this.idx;
+                this.idx++;
             }
         }
 
@@ -123,7 +_,7 @@
             BlockState blockstate = p_230701_.getBlockState(blockpos);
             NeighborUpdater.executeUpdate(p_230701_, blockstate, blockpos, this.sourceBlock, this.sourcePos, false);
             if (this.idx < NeighborUpdater.UPDATE_ORDER.length && NeighborUpdater.UPDATE_ORDER[this.idx] == this.skipDirection) {
-                ++this.idx;
+                this.idx++;
             }
 
             return this.idx < NeighborUpdater.UPDATE_ORDER.length;
