--- a/net/minecraft/world/inventory/CrafterMenu.java
+++ b/net/minecraft/world/inventory/CrafterMenu.java
@@ -40,20 +_,20 @@
     }
 
     private void addSlots(Inventory p_307214_) {
-        for(int i = 0; i < 3; ++i) {
-            for(int j = 0; j < 3; ++j) {
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
                 int k = j + i * 3;
                 this.addSlot(new CrafterSlot(this.container, k, 26 + j * 18, 17 + i * 18, this));
             }
         }
 
-        for(int l = 0; l < 3; ++l) {
-            for(int j1 = 0; j1 < 9; ++j1) {
+        for (int l = 0; l < 3; l++) {
+            for (int j1 = 0; j1 < 9; j1++) {
                 this.addSlot(new Slot(p_307214_, j1 + l * 9 + 9, 8 + j1 * 18, 84 + l * 18));
             }
         }
 
-        for(int i1 = 0; i1 < 9; ++i1) {
+        for (int i1 = 0; i1 < 9; i1++) {
             this.addSlot(new Slot(p_307214_, i1, 8 + i1 * 18, 142));
         }
 
@@ -69,11 +_,7 @@
     }
 
     public boolean isSlotDisabled(int p_307609_) {
-        if (p_307609_ > -1 && p_307609_ < 9) {
-            return this.containerData.get(p_307609_) == 1;
-        } else {
-            return false;
-        }
+        return p_307609_ > -1 && p_307609_ < 9 ? this.containerData.get(p_307609_) == 1 : false;
     }
 
     public boolean isPowered() {
@@ -117,8 +_,7 @@
     }
 
     private void refreshRecipeResult() {
-        Player $$1 = this.player;
-        if ($$1 instanceof ServerPlayer serverplayer) {
+        if (this.player instanceof ServerPlayer serverplayer) {
             Level level = serverplayer.level();
             ItemStack itemstack = CrafterBlock.getPotentialResults(level, this.container)
                 .map(p_335274_ -> p_335274_.value().assemble(this.container, level.registryAccess()))
