--- a/net/minecraft/world/inventory/DispenserMenu.java
+++ b/net/minecraft/world/inventory/DispenserMenu.java
@@ -24,19 +_,19 @@
         this.dispenser = p_39438_;
         p_39438_.startOpen(p_39437_.player);
 
-        for(int i = 0; i < 3; ++i) {
-            for(int j = 0; j < 3; ++j) {
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
                 this.addSlot(new Slot(p_39438_, j + i * 3, 62 + j * 18, 17 + i * 18));
             }
         }
 
-        for(int k = 0; k < 3; ++k) {
-            for(int i1 = 0; i1 < 9; ++i1) {
+        for (int k = 0; k < 3; k++) {
+            for (int i1 = 0; i1 < 9; i1++) {
                 this.addSlot(new Slot(p_39437_, i1 + k * 9 + 9, 8 + i1 * 18, 84 + k * 18));
             }
         }
 
-        for(int l = 0; l < 9; ++l) {
+        for (int l = 0; l < 9; l++) {
             this.addSlot(new Slot(p_39437_, l, 8 + l * 18, 142));
         }
     }
