--- a/net/minecraft/world/inventory/FurnaceResultSlot.java
+++ b/net/minecraft/world/inventory/FurnaceResultSlot.java
@@ -23,7 +_,7 @@
     @Override
     public ItemStack remove(int p_39548_) {
         if (this.hasItem()) {
-            this.removeCount += Math.min(p_39548_, this.getItem().getCount());
+            this.removeCount = this.removeCount + Math.min(p_39548_, this.getItem().getCount());
         }
 
         return super.remove(p_39548_);
@@ -41,16 +_,14 @@
         this.checkTakeAchievements(p_39555_);
     }
 
+    @Override
     protected void checkTakeAchievements(ItemStack p_39558_) {
         p_39558_.onCraftedBy(this.player.level(), this.player, this.removeCount);
-        Player playerx = this.player;
-        if (playerx instanceof ServerPlayer serverplayer) {
-            Container container = this.container;
-            if (container instanceof AbstractFurnaceBlockEntity abstractfurnaceblockentity) {
-                abstractfurnaceblockentity.awardUsedRecipesAndPopExperience(serverplayer);
-            }
+        if (this.player instanceof ServerPlayer serverplayer && this.container instanceof AbstractFurnaceBlockEntity abstractfurnaceblockentity) {
+            abstractfurnaceblockentity.awardUsedRecipesAndPopExperience(serverplayer);
         }
 
         this.removeCount = 0;
+        net.neoforged.neoforge.event.EventHooks.firePlayerSmeltedEvent(this.player, p_39558_);
     }
 }
