--- a/net/minecraft/world/inventory/ItemCombinerMenu.java
+++ b/net/minecraft/world/inventory/ItemCombinerMenu.java
@@ -41,7 +_,7 @@
     }
 
     private void createInputSlots(ItemCombinerMenuSlotDefinition p_267172_) {
-        for(final ItemCombinerMenuSlotDefinition.SlotDefinition itemcombinermenuslotdefinition$slotdefinition : p_267172_.getSlots()) {
+        for (final ItemCombinerMenuSlotDefinition.SlotDefinition itemcombinermenuslotdefinition$slotdefinition : p_267172_.getSlots()) {
             this.addSlot(
                 new Slot(
                     this.inputSlots,
@@ -78,13 +_,13 @@
     }
 
     private void createInventorySlots(Inventory p_267325_) {
-        for(int i = 0; i < 3; ++i) {
-            for(int j = 0; j < 9; ++j) {
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 9; j++) {
                 this.addSlot(new Slot(p_267325_, j + i * 9 + 9, 8 + j * 18, 84 + i * 18));
             }
         }
 
-        for(int k = 0; k < 9; ++k) {
+        for (int k = 0; k < 9; k++) {
             this.addSlot(new Slot(p_267325_, k, 8 + k * 18, 142));
         }
     }
@@ -121,10 +_,7 @@
     public boolean stillValid(Player p_39780_) {
         return this.access
             .evaluate(
-                (p_39785_, p_39786_) -> !this.isValidBlock(p_39785_.getBlockState(p_39786_))
-                        ? false
-                        : p_39780_.distanceToSqr((double)p_39786_.getX() + 0.5, (double)p_39786_.getY() + 0.5, (double)p_39786_.getZ() + 0.5) <= 64.0,
-                true
+                (p_339525_, p_339526_) -> !this.isValidBlock(p_339525_.getBlockState(p_339526_)) ? false : p_39780_.canInteractWithBlock(p_339526_, 4.0), true
             );
     }
 
