--- a/net/minecraft/world/item/armortrim/TrimMaterials.java
+++ b/net/minecraft/world/item/armortrim/TrimMaterials.java
@@ -28,27 +_,22 @@
     public static final ResourceKey<TrimMaterial> DIAMOND = registryKey("diamond");
     public static final ResourceKey<TrimMaterial> LAPIS = registryKey("lapis");
     public static final ResourceKey<TrimMaterial> AMETHYST = registryKey("amethyst");
-    public static final ResourceKey<TrimMaterial> AMBER = registryKey("amber");
 
     public static void bootstrap(BootstrapContext<TrimMaterial> p_321806_) {
-        register(p_321806_, QUARTZ, Items.QUARTZ, Style.EMPTY.withColor(14931140), 0.01F);
-        register(p_321806_, IRON, Items.IRON_INGOT, Style.EMPTY.withColor(15527148), 0.02F, Map.of(ArmorMaterials.IRON, "iron_darker"));
-        register(p_321806_, NETHERITE, Items.NETHERITE_INGOT, Style.EMPTY.withColor(6445145), 0.03F, Map.of(ArmorMaterials.NETHERITE, "netherite_darker"));
-        register(p_321806_, REDSTONE, Items.REDSTONE, Style.EMPTY.withColor(9901575), 0.04F);
-        register(p_321806_, COPPER, Items.COPPER_INGOT, Style.EMPTY.withColor(11823181), 0.05F);
-        register(p_321806_, GOLD, Items.GOLD_INGOT, Style.EMPTY.withColor(14594349), 0.06F, Map.of(ArmorMaterials.GOLD, "gold_darker"));
-        register(p_321806_, EMERALD, Items.EMERALD, Style.EMPTY.withColor(1155126), 0.07F);
-        register(p_321806_, DIAMOND, Items.DIAMOND, Style.EMPTY.withColor(7269586), 0.08F, Map.of(ArmorMaterials.DIAMOND, "diamond_darker"));
-        register(p_321806_, LAPIS, Items.LAPIS_LAZULI, Style.EMPTY.withColor(4288151), 0.09F);
-        register(p_321806_, AMETHYST, Items.AMETHYST_SHARD, Style.EMPTY.withColor(10116294), 0.1F);
-        register(p_321806_, AMBER, Items.AMBER_GEM, Style.EMPTY.withColor(15042820), 1.0F);
+        register(p_321806_, QUARTZ, Items.QUARTZ, Style.EMPTY.withColor(14931140), 0.1F);
+        register(p_321806_, IRON, Items.IRON_INGOT, Style.EMPTY.withColor(15527148), 0.2F, Map.of(ArmorMaterials.IRON, "iron_darker"));
+        register(p_321806_, NETHERITE, Items.NETHERITE_INGOT, Style.EMPTY.withColor(6445145), 0.3F, Map.of(ArmorMaterials.NETHERITE, "netherite_darker"));
+        register(p_321806_, REDSTONE, Items.REDSTONE, Style.EMPTY.withColor(9901575), 0.4F);
+        register(p_321806_, COPPER, Items.COPPER_INGOT, Style.EMPTY.withColor(11823181), 0.5F);
+        register(p_321806_, GOLD, Items.GOLD_INGOT, Style.EMPTY.withColor(14594349), 0.6F, Map.of(ArmorMaterials.GOLD, "gold_darker"));
+        register(p_321806_, EMERALD, Items.EMERALD, Style.EMPTY.withColor(1155126), 0.7F);
+        register(p_321806_, DIAMOND, Items.DIAMOND, Style.EMPTY.withColor(7269586), 0.8F, Map.of(ArmorMaterials.DIAMOND, "diamond_darker"));
+        register(p_321806_, LAPIS, Items.LAPIS_LAZULI, Style.EMPTY.withColor(4288151), 0.9F);
+        register(p_321806_, AMETHYST, Items.AMETHYST_SHARD, Style.EMPTY.withColor(10116294), 1.0F);
     }
 
     public static Optional<Holder.Reference<TrimMaterial>> getFromIngredient(HolderLookup.Provider p_335701_, ItemStack p_267327_) {
-        return p_335701_.<TrimMaterial>lookupOrThrow(Registries.TRIM_MATERIAL)
-            .listElements()
-            .filter(p_266876_ -> p_267327_.is(p_266876_.value().ingredient()))
-            .findFirst();
+        return p_335701_.lookupOrThrow(Registries.TRIM_MATERIAL).listElements().filter(p_266876_ -> p_267327_.is(p_266876_.value().ingredient())).findFirst();
     }
 
     private static void register(
