--- a/net/minecraft/world/level/pathfinder/BinaryHeap.java
+++ b/net/minecraft/world/level/pathfinder/BinaryHeap.java
@@ -73,10 +_,10 @@
 
     private void upHeap(int p_77083_) {
         Node node = this.heap[p_77083_];
+        float f = node.f;
 
-        int i;
-        for(float f = node.f; p_77083_ > 0; p_77083_ = i) {
-            i = p_77083_ - 1 >> 1;
+        while (p_77083_ > 0) {
+            int i = p_77083_ - 1 >> 1;
             Node node1 = this.heap[i];
             if (!(f < node1.f)) {
                 break;
@@ -84,6 +_,7 @@
 
             this.heap[p_77083_] = node1;
             node1.heapIdx = p_77083_;
+            p_77083_ = i;
         }
 
         this.heap[p_77083_] = node;
@@ -94,7 +_,7 @@
         Node node = this.heap[p_77090_];
         float f = node.f;
 
-        while(true) {
+        while (true) {
             int i = 1 + (p_77090_ << 1);
             int j = i + 1;
             if (i >= this.size) {
