--- a/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java
+++ b/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java
@@ -39,7 +_,7 @@
             i = this.mob.getBlockY();
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(this.mob.getX(), (double)i, this.mob.getZ());
 
-            for(BlockState blockstate = this.currentContext.getBlockState(blockpos$mutableblockpos);
+            for (BlockState blockstate = this.currentContext.getBlockState(blockpos$mutableblockpos);
                 blockstate.is(Blocks.WATER);
                 blockstate = this.currentContext.getBlockState(blockpos$mutableblockpos)
             ) {
@@ -51,7 +_,7 @@
 
         BlockPos blockpos1 = BlockPos.containing(this.mob.getX(), (double)i, this.mob.getZ());
         if (!this.canStartAt(blockpos1)) {
-            for(BlockPos blockpos : this.iteratePathfindingStartNodeCandidatePositions(this.mob)) {
+            for (BlockPos blockpos : this.iteratePathfindingStartNodeCandidatePositions(this.mob)) {
                 if (this.canStartAt(blockpos)) {
                     return super.getStartNode(blockpos);
                 }
@@ -274,7 +_,7 @@
             node.type = pathtype;
             node.costMalus = Math.max(node.costMalus, f);
             if (pathtype == PathType.WALKABLE) {
-                ++node.costMalus;
+                node.costMalus++;
             }
         }
 
