--- a/net/minecraft/world/level/pathfinder/PathFinder.java
+++ b/net/minecraft/world/level/pathfinder/PathFinder.java
@@ -66,7 +_,7 @@
         Set<Target> set2 = Sets.newHashSetWithExpectedSize(set.size());
         int j = (int)((float)this.maxVisitedNodes * p_164722_);
 
-        while(!this.openSet.isEmpty()) {
+        while (!this.openSet.isEmpty()) {
             if (++i >= j) {
                 break;
             }
@@ -74,7 +_,7 @@
             Node node = this.openSet.pop();
             node.closed = true;
 
-            for(Target target : set) {
+            for (Target target : set) {
                 if (node.distanceManhattan(target) <= (float)p_164721_) {
                     target.setReached();
                     set2.add(target);
@@ -88,7 +_,7 @@
             if (!(node.distanceTo(p_164718_) >= p_164720_)) {
                 int k = this.nodeEvaluator.getNeighbors(this.neighbors, node);
 
-                for(int l = 0; l < k; ++l) {
+                for (int l = 0; l < k; l++) {
                     Node node1 = this.neighbors[l];
                     float f = this.distance(node, node1);
                     node1.walkedDistance = node.walkedDistance + f;
@@ -126,7 +_,7 @@
     private float getBestH(Node p_77445_, Set<Target> p_77446_) {
         float f = Float.MAX_VALUE;
 
-        for(Target target : p_77446_) {
+        for (Target target : p_77446_) {
             float f1 = p_77445_.distanceTo(target);
             target.updateBest(f1, p_77445_);
             f = Math.min(f1, f);
@@ -140,7 +_,7 @@
         Node node = p_77435_;
         list.add(0, p_77435_);
 
-        while(node.cameFrom != null) {
+        while (node.cameFrom != null) {
             node = node.cameFrom;
             list.add(0, node);
         }
