--- a/net/minecraft/world/level/pathfinder/Path.java
+++ b/net/minecraft/world/level/pathfinder/Path.java
@@ -27,7 +_,7 @@
     }
 
     public void advance() {
-        ++this.nextNodeIndex;
+        this.nextNodeIndex++;
     }
 
     public boolean notStarted() {
@@ -104,7 +_,7 @@
         } else if (p_77386_.nodes.size() != this.nodes.size()) {
             return false;
         } else {
-            for(int i = 0; i < this.nodes.size(); ++i) {
+            for (int i = 0; i < this.nodes.size(); i++) {
                 Node node = this.nodes.get(i);
                 Node node1 = p_77386_.nodes.get(i);
                 if (node.x != node1.x || node.y != node1.y || node.z != node1.z) {
@@ -168,7 +_,7 @@
     static Node[] readNodeArray(FriendlyByteBuf p_294398_) {
         Node[] anode = new Node[p_294398_.readVarInt()];
 
-        for(int i = 0; i < anode.length; ++i) {
+        for (int i = 0; i < anode.length; i++) {
             anode[i] = Node.createFromStream(p_294398_);
         }
 
@@ -178,7 +_,7 @@
     static void writeNodeArray(FriendlyByteBuf p_296066_, Node[] p_294231_) {
         p_296066_.writeVarInt(p_294231_.length);
 
-        for(Node node : p_294231_) {
+        for (Node node : p_294231_) {
             node.writeToStream(p_296066_);
         }
     }
