--- a/net/minecraft/client/gui/font/CodepointMap.java
+++ b/net/minecraft/client/gui/font/CodepointMap.java
@@ -88,10 +_,10 @@
     }
 
     public void forEach(CodepointMap.Output<T> p_285048_) {
-        for(int i = 0; i < this.blockMap.length; ++i) {
+        for (int i = 0; i < this.blockMap.length; i++) {
             T[] at = this.blockMap[i];
             if (at != this.empty) {
-                for(int j = 0; j < at.length; ++j) {
+                for (int j = 0; j < at.length; j++) {
                     T t = at[j];
                     if (t != null) {
                         int k = i << 8 | j;
