--- a/net/minecraft/client/gui/font/FontOption.java
+++ b/net/minecraft/client/gui/font/FontOption.java
@@ -38,7 +_,7 @@
         }
 
         public boolean apply(Set<FontOption> p_326085_) {
-            for(Entry<FontOption, Boolean> entry : this.values.entrySet()) {
+            for (Entry<FontOption, Boolean> entry : this.values.entrySet()) {
                 if (p_326085_.contains(entry.getKey()) != entry.getValue()) {
                     return false;
                 }
