--- a/net/minecraft/client/gui/font/FontSet.java
+++ b/net/minecraft/client/gui/font/FontSet.java
@@ -33,10 +_,8 @@
     private BakedGlyph whiteGlyph;
     private List<GlyphProvider.Conditional> allProviders = List.of();
     private List<GlyphProvider> activeProviders = List.of();
-    private final CodepointMap<BakedGlyph> glyphs = new CodepointMap<>(p_284630_ -> new BakedGlyph[p_284630_], p_284629_ -> new BakedGlyph[p_284629_][]);
-    private final CodepointMap<FontSet.GlyphInfoFilter> glyphInfos = new CodepointMap<>(
-        p_284631_ -> new FontSet.GlyphInfoFilter[p_284631_], p_284632_ -> new FontSet.GlyphInfoFilter[p_284632_][]
-    );
+    private final CodepointMap<BakedGlyph> glyphs = new CodepointMap<>(BakedGlyph[]::new, BakedGlyph[][]::new);
+    private final CodepointMap<FontSet.GlyphInfoFilter> glyphInfos = new CodepointMap<>(FontSet.GlyphInfoFilter[]::new, FontSet.GlyphInfoFilter[][]::new);
     private final Int2ObjectMap<IntList> glyphsByWidth = new Int2ObjectOpenHashMap<>();
     private final List<FontTexture> textures = Lists.newArrayList();
 
@@ -69,7 +_,7 @@
         IntSet intset = new IntOpenHashSet();
         List<GlyphProvider> list = new ArrayList<>();
 
-        for(GlyphProvider.Conditional glyphprovider$conditional : p_326151_) {
+        for (GlyphProvider.Conditional glyphprovider$conditional : p_326151_) {
             if (glyphprovider$conditional.filter().apply(p_326291_)) {
                 list.add(glyphprovider$conditional.provider());
                 intset.addAll(glyphprovider$conditional.provider().getSupportedGlyphs());
@@ -78,7 +_,7 @@
 
         Set<GlyphProvider> set = Sets.newHashSet();
         intset.forEach(p_232561_ -> {
-            for(GlyphProvider glyphprovider : list) {
+            for (GlyphProvider glyphprovider : list) {
                 GlyphInfo glyphinfo = glyphprovider.getGlyph(p_232561_);
                 if (glyphinfo != null) {
                     set.add(glyphprovider);
@@ -98,7 +_,7 @@
     }
 
     private void closeTextures() {
-        for(FontTexture fonttexture : this.textures) {
+        for (FontTexture fonttexture : this.textures) {
             fonttexture.close();
         }
 
@@ -118,7 +_,7 @@
     private FontSet.GlyphInfoFilter computeGlyphInfo(int p_243321_) {
         GlyphInfo glyphinfo = null;
 
-        for(GlyphProvider glyphprovider : this.activeProviders) {
+        for (GlyphProvider glyphprovider : this.activeProviders) {
             GlyphInfo glyphinfo1 = glyphprovider.getGlyph(p_243321_);
             if (glyphinfo1 != null) {
                 if (glyphinfo == null) {
@@ -139,7 +_,7 @@
     }
 
     private BakedGlyph computeBakedGlyph(int p_232565_) {
-        for(GlyphProvider glyphprovider : this.activeProviders) {
+        for (GlyphProvider glyphprovider : this.activeProviders) {
             GlyphInfo glyphinfo = glyphprovider.getGlyph(p_232565_);
             if (glyphinfo != null) {
                 return glyphinfo.bake(this::stitch);
@@ -154,7 +_,7 @@
     }
 
     private BakedGlyph stitch(SheetGlyphInfo p_232557_) {
-        for(FontTexture fonttexture : this.textures) {
+        for (FontTexture fonttexture : this.textures) {
             BakedGlyph bakedglyph = fonttexture.add(p_232557_);
             if (bakedglyph != null) {
                 return bakedglyph;
