--- a/net/minecraft/client/gui/font/TextFieldHelper.java
+++ b/net/minecraft/client/gui/font/TextFieldHelper.java
@@ -136,7 +_,7 @@
     }
 
     public void moveBy(int p_232576_, boolean p_232577_, TextFieldHelper.CursorStep p_232578_) {
-        switch(p_232578_) {
+        switch (p_232578_) {
             case CHARACTER:
                 this.moveByChars(p_232576_, p_232577_);
                 break;
@@ -164,7 +_,7 @@
     }
 
     public void removeFromCursor(int p_232573_, TextFieldHelper.CursorStep p_232574_) {
-        switch(p_232574_) {
+        switch (p_232574_) {
             case CHARACTER:
                 this.removeCharsFromCursor(p_232573_);
                 break;
