--- a/net/minecraft/world/level/levelgen/carver/WorldCarver.java
+++ b/net/minecraft/world/level/levelgen/carver/WorldCarver.java
@@ -2,6 +_,7 @@
 
 import com.google.common.collect.ImmutableSet;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.Set;
 import java.util.function.Function;
 import javax.annotation.Nullable;
@@ -10,7 +_,6 @@
 import net.minecraft.core.Holder;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
-import net.minecraft.tags.BlockTags;
 import net.minecraft.util.Mth;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.ChunkPos;
@@ -36,21 +_,21 @@
     protected static final FluidState WATER = Fluids.WATER.defaultFluidState();
     protected static final FluidState LAVA = Fluids.LAVA.defaultFluidState();
     protected Set<Fluid> liquids = ImmutableSet.of(Fluids.WATER);
-    private final Codec<ConfiguredWorldCarver<C>> configuredCodec;
+    private final MapCodec<ConfiguredWorldCarver<C>> configuredCodec;
 
     private static <C extends CarverConfiguration, F extends WorldCarver<C>> F register(String p_65066_, F p_65067_) {
         return Registry.register(BuiltInRegistries.CARVER, p_65066_, p_65067_);
     }
 
     public WorldCarver(Codec<C> p_159366_) {
-        this.configuredCodec = p_159366_.fieldOf("config").xmap(this::configured, ConfiguredWorldCarver::config).codec();
+        this.configuredCodec = p_159366_.fieldOf("config").xmap(this::configured, ConfiguredWorldCarver::config);
     }
 
     public ConfiguredWorldCarver<C> configured(C p_65064_) {
         return new ConfiguredWorldCarver<>(this, p_65064_);
     }
 
-    public Codec<ConfiguredWorldCarver<C>> configuredCodec() {
+    public MapCodec<ConfiguredWorldCarver<C>> configuredCodec() {
         return this.configuredCodec;
     }
 
@@ -90,17 +_,17 @@
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
             BlockPos.MutableBlockPos blockpos$mutableblockpos1 = new BlockPos.MutableBlockPos();
 
-            for(int j2 = k; j2 <= l; ++j2) {
+            for (int j2 = k; j2 <= l; j2++) {
                 int k2 = chunkpos.getBlockX(j2);
                 double d3 = ((double)k2 + 0.5 - p_190759_) / p_190762_;
 
-                for(int l2 = l1; l2 <= i2; ++l2) {
+                for (int l2 = l1; l2 <= i2; l2++) {
                     int i3 = chunkpos.getBlockZ(l2);
                     double d4 = ((double)i3 + 0.5 - p_190761_) / p_190762_;
                     if (!(d3 * d3 + d4 * d4 >= 1.0)) {
                         MutableBoolean mutableboolean = new MutableBoolean(false);
 
-                        for(int j3 = k1; j3 > i1; --j3) {
+                        for (int j3 = k1; j3 > i1; j3--) {
                             double d5 = ((double)j3 - 0.5 - p_190760_) / p_190763_;
                             if (!p_190765_.shouldSkip(p_190754_, d3, d5, d4, j3) && (!p_190764_.get(j2, j3, l2) || isDebugEnabled(p_190755_))) {
                                 p_190764_.set(j2, j3, l2);
@@ -140,16 +_,14 @@
         MutableBoolean p_190752_
     ) {
         BlockState blockstate = p_190746_.getBlockState(p_190749_);
-        if (blockstate.is(BlockTags.ANIMALS_SPAWNABLE_ON) || blockstate.is(Blocks.MYCELIUM)) {
+        if (blockstate.is(Blocks.GRASS_BLOCK) || blockstate.is(Blocks.MYCELIUM)) {
             p_190752_.setTrue();
         }
 
         if (!this.canReplaceBlock(p_190745_, blockstate) && !isDebugEnabled(p_190745_)) {
             return false;
         } else {
-            BlockState blockstate1 = p_190746_.isPotato()
-                ? Blocks.CAVE_AIR.defaultBlockState()
-                : this.getCarveState(p_190744_, p_190745_, p_190749_, p_190751_);
+            BlockState blockstate1 = this.getCarveState(p_190744_, p_190745_, p_190749_, p_190751_);
             if (blockstate1 == null) {
                 return false;
             } else {
@@ -160,8 +_,7 @@
 
                 if (p_190752_.isTrue()) {
                     p_190750_.setWithOffset(p_190749_, Direction.DOWN);
-                    BlockState blockstate2 = p_190746_.getBlockState(p_190750_);
-                    if (blockstate2.is(Blocks.DIRT) || blockstate2.is(Blocks.TERREDEPOMME)) {
+                    if (p_190746_.getBlockState(p_190750_).is(Blocks.DIRT)) {
                         p_190744_.topMaterial(p_190747_, p_190746_, p_190750_, !blockstate1.getFluidState().isEmpty()).ifPresent(p_284918_ -> {
                             p_190746_.setBlockState(p_190750_, p_284918_, false);
                             if (!p_284918_.getFluidState().isEmpty()) {
