--- a/net/minecraft/world/level/levelgen/structure/pools/alias/Direct.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/alias/Direct.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.pools.alias;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.function.BiConsumer;
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.pools.StructureTemplatePool;
 
 record Direct(ResourceKey<StructureTemplatePool> alias, ResourceKey<StructureTemplatePool> target) implements PoolAliasBinding {
-    static Codec<Direct> CODEC = RecordCodecBuilder.create(
+    static MapCodec<Direct> CODEC = RecordCodecBuilder.mapCodec(
         p_307312_ -> p_307312_.group(
                     ResourceKey.codec(Registries.TEMPLATE_POOL).fieldOf("alias").forGetter(Direct::alias),
                     ResourceKey.codec(Registries.TEMPLATE_POOL).fieldOf("target").forGetter(Direct::target)
@@ -30,7 +_,7 @@
     }
 
     @Override
-    public Codec<Direct> codec() {
+    public MapCodec<Direct> codec() {
         return CODEC;
     }
 }
