--- a/net/minecraft/world/level/levelgen/structure/pools/alias/PoolAliasBindings.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/alias/PoolAliasBindings.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.pools.alias;
 
 import com.mojang.datafixers.util.Pair;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import net.minecraft.core.Holder;
 import net.minecraft.core.Registry;
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.pools.StructureTemplatePool;
 
 public class PoolAliasBindings {
-    public static Codec<? extends PoolAliasBinding> bootstrap(Registry<Codec<? extends PoolAliasBinding>> p_307584_) {
+    public static MapCodec<? extends PoolAliasBinding> bootstrap(Registry<MapCodec<? extends PoolAliasBinding>> p_307584_) {
         Registry.register(p_307584_, "random", Random.CODEC);
         Registry.register(p_307584_, "random_group", RandomGroup.CODEC);
         return Registry.register(p_307584_, "direct", Direct.CODEC);
