--- a/net/minecraft/world/level/levelgen/structure/pools/alias/PoolAliasBinding.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/alias/PoolAliasBinding.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.levelgen.structure.pools.alias;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import java.util.function.BiConsumer;
 import java.util.function.Function;
@@ -30,7 +_,7 @@
 
     static Random random(String p_307435_, SimpleWeightedRandomList<String> p_307653_) {
         SimpleWeightedRandomList.Builder<ResourceKey<StructureTemplatePool>> builder = SimpleWeightedRandomList.builder();
-        p_307653_.unwrap().forEach(p_307355_ -> builder.add(Pools.createKey(p_307355_.getData()), p_307355_.getWeight().asInt()));
+        p_307653_.unwrap().forEach(p_338103_ -> builder.add(Pools.createKey(p_338103_.data()), p_338103_.getWeight().asInt()));
         return random(Pools.createKey(p_307435_), builder.build());
     }
 
@@ -42,5 +_,5 @@
         return new RandomGroup(p_307498_);
     }
 
-    Codec<? extends PoolAliasBinding> codec();
+    MapCodec<? extends PoolAliasBinding> codec();
 }
