--- a/net/minecraft/client/gui/components/debugchart/AbstractDebugChart.java
+++ b/net/minecraft/client/gui/components/debugchart/AbstractDebugChart.java
@@ -35,7 +_,7 @@
         int i1 = Math.max(0, this.sampleStorage.capacity() - (p_298301_ - 2));
         int j1 = this.sampleStorage.size() - i1;
 
-        for(int k1 = 0; k1 < j1; ++k1) {
+        for (int k1 = 0; k1 < j1; k1++) {
             int l1 = p_298986_ + k1 + 1;
             int i2 = i1 + k1;
             long j2 = this.getValueForAggregation(i2);
