--- a/net/minecraft/client/gui/screens/multiplayer/ServerReconfigScreen.java
+++ b/net/minecraft/client/gui/screens/multiplayer/ServerReconfigScreen.java
@@ -1,5 +_,6 @@
 package net.minecraft.client.gui.screens.multiplayer;
 
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.StringWidget;
 import net.minecraft.client.gui.components.events.GuiEventListener;
@@ -39,7 +_,9 @@
             .addChild(Button.builder(CommonComponents.GUI_DISCONNECT, p_295623_ -> this.connection.disconnect(ConnectScreen.ABORT_CONNECTION)).build());
         this.disconnectButton.active = false;
         this.layout.arrangeElements();
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321356_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321356_);
+        });
         this.repositionElements();
     }
 
@@ -51,7 +_,7 @@
     @Override
     public void tick() {
         super.tick();
-        ++this.delayTicker;
+        this.delayTicker++;
         if (this.delayTicker == 600) {
             this.disconnectButton.active = true;
         }
