--- a/net/minecraft/client/gui/screens/multiplayer/ServerSelectionList.java
+++ b/net/minecraft/client/gui/screens/multiplayer/ServerSelectionList.java
@@ -103,7 +_,7 @@
     public void updateOnlineServers(ServerList p_99798_) {
         this.onlineServers.clear();
 
-        for(int i = 0; i < p_99798_.size(); ++i) {
+        for (int i = 0; i < p_99798_.size(); i++) {
             this.onlineServers.add(new ServerSelectionList.OnlineServerEntry(this.screen, p_99798_.get(i)));
         }
 
@@ -114,13 +_,13 @@
         int i = p_99800_.size() - this.networkServers.size();
         this.networkServers.clear();
 
-        for(LanServer lanserver : p_99800_) {
+        for (LanServer lanserver : p_99800_) {
             this.networkServers.add(new ServerSelectionList.NetworkServerEntry(this.screen, lanserver));
         }
 
         this.refreshEntries();
 
-        for(int i1 = this.networkServers.size() - i; i1 < this.networkServers.size(); ++i1) {
+        for (int i1 = this.networkServers.size() - i; i1 < this.networkServers.size(); i1++) {
             ServerSelectionList.NetworkServerEntry serverselectionlist$networkserverentry = this.networkServers.get(i1);
             int j = i1 - this.networkServers.size() + this.children().size();
             int k = this.getRowTop(j);
@@ -328,7 +_,7 @@
             p_281406_.drawString(this.minecraft.font, this.serverData.name, p_281363_ + 32 + 3, p_282921_ + 1, 16777215, false);
             List<FormattedCharSequence> list = this.minecraft.font.split(this.serverData.motd, p_283596_ - 32 - 2);
 
-            for(int i = 0; i < Math.min(list.size(), 2); ++i) {
+            for (int i = 0; i < Math.min(list.size(), 2); i++) {
                 p_281406_.drawString(this.minecraft.font, list.get(i), p_281363_ + 32 + 3, p_282921_ + 12 + 9 * i, -8355712, false);
             }
 
@@ -338,7 +_,7 @@
                 if (j1 > 4) {
                     j1 = 8 - j1;
                 }
-                this.statusIcon = switch(j1) {
+                this.statusIcon = switch (j1) {
                     case 1 -> ServerSelectionList.PINGING_2_SPRITE;
                     case 2 -> ServerSelectionList.PINGING_3_SPRITE;
                     case 3 -> ServerSelectionList.PINGING_4_SPRITE;
@@ -406,7 +_,7 @@
 
         private void refreshStatus() {
             this.onlinePlayersTooltip = null;
-            switch(this.serverData.state()) {
+            switch (this.serverData.state()) {
                 case INITIAL:
                 case PINGING:
                     this.statusIcon = ServerSelectionList.PING_1_SPRITE;
@@ -535,7 +_,7 @@
             MutableComponent mutablecomponent = Component.empty();
             mutablecomponent.append(Component.translatable("narrator.select", this.serverData.name));
             mutablecomponent.append(CommonComponents.NARRATION_SEPARATOR);
-            switch(this.serverData.state()) {
+            switch (this.serverData.state()) {
                 case PINGING:
                     mutablecomponent.append(ServerSelectionList.PINGING_STATUS);
                     break;
