--- a/net/minecraft/world/entity/npc/VillagerType.java
+++ b/net/minecraft/world/entity/npc/VillagerType.java
@@ -20,43 +_,37 @@
     public static final VillagerType SNOW = register("snow");
     public static final VillagerType SWAMP = register("swamp");
     public static final VillagerType TAIGA = register("taiga");
-    public static final VillagerType POTATO = register("potato");
     private final String name;
-    private static final Map<ResourceKey<Biome>, VillagerType> BY_BIOME = Util.make(Maps.newHashMap(), p_340157_ -> {
-        p_340157_.put(Biomes.BADLANDS, DESERT);
-        p_340157_.put(Biomes.DESERT, DESERT);
-        p_340157_.put(Biomes.ERODED_BADLANDS, DESERT);
-        p_340157_.put(Biomes.WOODED_BADLANDS, DESERT);
-        p_340157_.put(Biomes.BAMBOO_JUNGLE, JUNGLE);
-        p_340157_.put(Biomes.JUNGLE, JUNGLE);
-        p_340157_.put(Biomes.SPARSE_JUNGLE, JUNGLE);
-        p_340157_.put(Biomes.SAVANNA_PLATEAU, SAVANNA);
-        p_340157_.put(Biomes.SAVANNA, SAVANNA);
-        p_340157_.put(Biomes.WINDSWEPT_SAVANNA, SAVANNA);
-        p_340157_.put(Biomes.DEEP_FROZEN_OCEAN, SNOW);
-        p_340157_.put(Biomes.FROZEN_OCEAN, SNOW);
-        p_340157_.put(Biomes.FROZEN_RIVER, SNOW);
-        p_340157_.put(Biomes.ICE_SPIKES, SNOW);
-        p_340157_.put(Biomes.SNOWY_BEACH, SNOW);
-        p_340157_.put(Biomes.SNOWY_TAIGA, SNOW);
-        p_340157_.put(Biomes.SNOWY_PLAINS, SNOW);
-        p_340157_.put(Biomes.GROVE, SNOW);
-        p_340157_.put(Biomes.SNOWY_SLOPES, SNOW);
-        p_340157_.put(Biomes.FROZEN_PEAKS, SNOW);
-        p_340157_.put(Biomes.JAGGED_PEAKS, SNOW);
-        p_340157_.put(Biomes.SWAMP, SWAMP);
-        p_340157_.put(Biomes.MANGROVE_SWAMP, SWAMP);
-        p_340157_.put(Biomes.OLD_GROWTH_SPRUCE_TAIGA, TAIGA);
-        p_340157_.put(Biomes.OLD_GROWTH_PINE_TAIGA, TAIGA);
-        p_340157_.put(Biomes.WINDSWEPT_GRAVELLY_HILLS, TAIGA);
-        p_340157_.put(Biomes.WINDSWEPT_HILLS, TAIGA);
-        p_340157_.put(Biomes.TAIGA, TAIGA);
-        p_340157_.put(Biomes.WINDSWEPT_FOREST, TAIGA);
-        p_340157_.put(Biomes.ARBORETUM, POTATO);
-        p_340157_.put(Biomes.FIELDS, POTATO);
-        p_340157_.put(Biomes.HASH, POTATO);
-        p_340157_.put(Biomes.CORRUPTION, POTATO);
-        p_340157_.put(Biomes.WASTELAND, POTATO);
+    private static final Map<ResourceKey<Biome>, VillagerType> BY_BIOME = Util.make(Maps.newHashMap(), p_35834_ -> {
+        p_35834_.put(Biomes.BADLANDS, DESERT);
+        p_35834_.put(Biomes.DESERT, DESERT);
+        p_35834_.put(Biomes.ERODED_BADLANDS, DESERT);
+        p_35834_.put(Biomes.WOODED_BADLANDS, DESERT);
+        p_35834_.put(Biomes.BAMBOO_JUNGLE, JUNGLE);
+        p_35834_.put(Biomes.JUNGLE, JUNGLE);
+        p_35834_.put(Biomes.SPARSE_JUNGLE, JUNGLE);
+        p_35834_.put(Biomes.SAVANNA_PLATEAU, SAVANNA);
+        p_35834_.put(Biomes.SAVANNA, SAVANNA);
+        p_35834_.put(Biomes.WINDSWEPT_SAVANNA, SAVANNA);
+        p_35834_.put(Biomes.DEEP_FROZEN_OCEAN, SNOW);
+        p_35834_.put(Biomes.FROZEN_OCEAN, SNOW);
+        p_35834_.put(Biomes.FROZEN_RIVER, SNOW);
+        p_35834_.put(Biomes.ICE_SPIKES, SNOW);
+        p_35834_.put(Biomes.SNOWY_BEACH, SNOW);
+        p_35834_.put(Biomes.SNOWY_TAIGA, SNOW);
+        p_35834_.put(Biomes.SNOWY_PLAINS, SNOW);
+        p_35834_.put(Biomes.GROVE, SNOW);
+        p_35834_.put(Biomes.SNOWY_SLOPES, SNOW);
+        p_35834_.put(Biomes.FROZEN_PEAKS, SNOW);
+        p_35834_.put(Biomes.JAGGED_PEAKS, SNOW);
+        p_35834_.put(Biomes.SWAMP, SWAMP);
+        p_35834_.put(Biomes.MANGROVE_SWAMP, SWAMP);
+        p_35834_.put(Biomes.OLD_GROWTH_SPRUCE_TAIGA, TAIGA);
+        p_35834_.put(Biomes.OLD_GROWTH_PINE_TAIGA, TAIGA);
+        p_35834_.put(Biomes.WINDSWEPT_GRAVELLY_HILLS, TAIGA);
+        p_35834_.put(Biomes.WINDSWEPT_HILLS, TAIGA);
+        p_35834_.put(Biomes.TAIGA, TAIGA);
+        p_35834_.put(Biomes.WINDSWEPT_FOREST, TAIGA);
     });
 
     public VillagerType(String p_35830_) {
