--- a/net/minecraft/world/entity/npc/VillagerTrades.java
+++ b/net/minecraft/world/entity/npc/VillagerTrades.java
@@ -74,8 +_,8 @@
     private static final float HIGH_TIER_PRICE_MULTIPLIER = 0.2F;
     public static final Map<VillagerProfession, Int2ObjectMap<VillagerTrades.ItemListing[]>> TRADES = Util.make(
         Maps.newHashMap(),
-        p_340156_ -> {
-            p_340156_.put(
+        p_335271_ -> {
+            p_335271_.put(
                 VillagerProfession.FARMER,
                 toIntMap(
                     ImmutableMap.of(
@@ -115,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.FISHERMAN,
                 toIntMap(
                     ImmutableMap.of(
@@ -154,14 +_,13 @@
                                     .put(VillagerType.JUNGLE, Items.JUNGLE_BOAT)
                                     .put(VillagerType.SAVANNA, Items.ACACIA_BOAT)
                                     .put(VillagerType.SWAMP, Items.DARK_OAK_BOAT)
-                                    .put(VillagerType.POTATO, Items.BAMBOO_RAFT)
                                     .build()
                             )
                         }
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.SHEPHERD,
                 toIntMap(
                     ImmutableMap.of(
@@ -267,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.FLETCHER,
                 toIntMap(
                     ImmutableMap.of(
@@ -298,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.LIBRARIAN,
                 toIntMap(
                     ImmutableMap.<Integer, VillagerTrades.ItemListing[]>builder()
@@ -339,7 +_,7 @@
                         .build()
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.CARTOGRAPHER,
                 toIntMap(
                     ImmutableMap.of(
@@ -389,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.CLERIC,
                 toIntMap(
                     ImmutableMap.of(
@@ -420,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.ARMORER,
                 toIntMap(
                     ImmutableMap.of(
@@ -460,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.WEAPONSMITH,
                 toIntMap(
                     ImmutableMap.of(
@@ -487,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.TOOLSMITH,
                 toIntMap(
                     ImmutableMap.of(
@@ -523,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.BUTCHER,
                 toIntMap(
                     ImmutableMap.of(
@@ -551,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.LEATHERWORKER,
                 toIntMap(
                     ImmutableMap.of(
@@ -585,7 +_,7 @@
                     )
                 )
             );
-            p_340156_.put(
+            p_335271_.put(
                 VillagerProfession.MASON,
                 toIntMap(
                     ImmutableMap.of(
@@ -717,7 +_,7 @@
                 new VillagerTrades.ItemsForEmeralds(Items.RED_MUSHROOM, 1, 1, 12, 1),
                 new VillagerTrades.ItemsForEmeralds(Items.LILY_PAD, 1, 2, 5, 1),
                 new VillagerTrades.ItemsForEmeralds(Items.SMALL_DRIPLEAF, 1, 2, 5, 1),
-                new VillagerTrades.ItemsForEmeralds(Items.GRAVTATER, 1, 8, 8, 1),
+                new VillagerTrades.ItemsForEmeralds(Items.SAND, 1, 8, 8, 1),
                 new VillagerTrades.ItemsForEmeralds(Items.RED_SAND, 1, 4, 6, 1),
                 new VillagerTrades.ItemsForEmeralds(Items.POINTED_DRIPSTONE, 1, 2, 5, 1),
                 new VillagerTrades.ItemsForEmeralds(Items.ROOTED_DIRT, 1, 2, 5, 1),
@@ -810,8 +_,7 @@
                             VillagerType.PLAINS,
                             VillagerType.SAVANNA,
                             VillagerType.SNOW,
-                            VillagerType.TAIGA,
-                            VillagerType.POTATO
+                            VillagerType.TAIGA
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(Items.CHAINMAIL_BOOTS, 4, 1, 12, 5, 0.05F), VillagerType.JUNGLE, VillagerType.SWAMP
@@ -822,8 +_,7 @@
                             VillagerType.PLAINS,
                             VillagerType.SAVANNA,
                             VillagerType.SNOW,
-                            VillagerType.TAIGA,
-                            VillagerType.POTATO
+                            VillagerType.TAIGA
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(Items.CHAINMAIL_HELMET, 5, 1, 12, 5, 0.05F), VillagerType.JUNGLE, VillagerType.SWAMP
@@ -834,8 +_,7 @@
                             VillagerType.PLAINS,
                             VillagerType.SAVANNA,
                             VillagerType.SNOW,
-                            VillagerType.TAIGA,
-                            VillagerType.POTATO
+                            VillagerType.TAIGA
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(Items.CHAINMAIL_LEGGINGS, 7, 1, 12, 5, 0.05F), VillagerType.JUNGLE, VillagerType.SWAMP
@@ -846,8 +_,7 @@
                             VillagerType.PLAINS,
                             VillagerType.SAVANNA,
                             VillagerType.SNOW,
-                            VillagerType.TAIGA,
-                            VillagerType.POTATO
+                            VillagerType.TAIGA
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(Items.CHAINMAIL_CHESTPLATE, 9, 1, 12, 5, 0.05F), VillagerType.JUNGLE, VillagerType.SWAMP
@@ -879,24 +_,19 @@
                             VillagerType.DESERT
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
-                            new VillagerTrades.ItemsForEmeralds(enchant(Items.IRON_BOOTS, Enchantments.PROTECTION, 1), 8, 1, 3, 15, 0.05F),
-                            VillagerType.PLAINS,
-                            VillagerType.POTATO
+                            new VillagerTrades.ItemsForEmeralds(enchant(Items.IRON_BOOTS, Enchantments.PROTECTION, 1), 8, 1, 3, 15, 0.05F), VillagerType.PLAINS
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(enchant(Items.IRON_HELMET, Enchantments.PROTECTION, 1), 9, 1, 3, 15, 0.05F),
-                            VillagerType.PLAINS,
-                            VillagerType.POTATO
+                            VillagerType.PLAINS
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(enchant(Items.IRON_LEGGINGS, Enchantments.PROTECTION, 1), 11, 1, 3, 15, 0.05F),
-                            VillagerType.PLAINS,
-                            VillagerType.POTATO
+                            VillagerType.PLAINS
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(enchant(Items.IRON_CHESTPLATE, Enchantments.PROTECTION, 1), 13, 1, 3, 15, 0.05F),
-                            VillagerType.PLAINS,
-                            VillagerType.POTATO
+                            VillagerType.PLAINS
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsForEmeralds(enchant(Items.IRON_BOOTS, Enchantments.BINDING_CURSE, 1), 2, 1, 3, 15, 0.05F),
@@ -988,15 +_,13 @@
                             new VillagerTrades.ItemsAndEmeraldsToItems(
                                 Items.DIAMOND, 3, 16, enchant(Items.DIAMOND_LEGGINGS, Enchantments.PROTECTION, 1), 1, 3, 30, 0.05F
                             ),
-                            VillagerType.PLAINS,
-                            VillagerType.POTATO
+                            VillagerType.PLAINS
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsAndEmeraldsToItems(
                                 Items.DIAMOND, 2, 12, enchant(Items.DIAMOND_BOOTS, Enchantments.PROTECTION, 1), 1, 3, 30, 0.05F
                             ),
-                            VillagerType.PLAINS,
-                            VillagerType.POTATO
+                            VillagerType.PLAINS
                         ),
                         VillagerTrades.TypeSpecificTrade.oneTradeInBiomes(
                             new VillagerTrades.ItemsAndEmeraldsToItems(
@@ -1060,8 +_,7 @@
                             VillagerType.PLAINS,
                             VillagerType.SAVANNA,
                             VillagerType.SNOW,
-                            VillagerType.SWAMP,
-                            VillagerType.POTATO
+                            VillagerType.SWAMP
                         )
                     }
                 )
@@ -1086,7 +_,6 @@
                             .put(VillagerType.SNOW, PLAINS_MAP)
                             .put(VillagerType.JUNGLE, SAVANNA_MAP)
                             .put(VillagerType.SWAMP, SNOWY_MAP)
-                            .put(VillagerType.POTATO, PLAINS_MAP)
                             .build()
                     ),
                     new VillagerTrades.TypeSpecificTrade(
@@ -1098,7 +_,6 @@
                             .put(VillagerType.SNOW, TAIGA_MAP)
                             .put(VillagerType.JUNGLE, DESERT_MAP)
                             .put(VillagerType.SWAMP, TAIGA_MAP)
-                            .put(VillagerType.POTATO, PLAINS_MAP)
                             .build()
                     ),
                     new VillagerTrades.TypeSpecificTrade(
@@ -1110,7 +_,6 @@
                             .put(VillagerType.SNOW, SWAMP_MAP)
                             .put(VillagerType.JUNGLE, SWAMP_MAP)
                             .put(VillagerType.SWAMP, JUNGLE_MAP)
-                            .put(VillagerType.POTATO, PLAINS_MAP)
                             .build()
                     )
                 },
@@ -1254,7 +_,7 @@
                     new VillagerTrades.ItemsForEmeralds(Items.RED_MUSHROOM, 1, 3, 4, 1),
                     new VillagerTrades.ItemsForEmeralds(Items.LILY_PAD, 1, 5, 2, 1),
                     new VillagerTrades.ItemsForEmeralds(Items.SMALL_DRIPLEAF, 1, 2, 5, 1),
-                    new VillagerTrades.ItemsForEmeralds(Items.GRAVTATER, 1, 8, 8, 1),
+                    new VillagerTrades.ItemsForEmeralds(Items.SAND, 1, 8, 8, 1),
                     new VillagerTrades.ItemsForEmeralds(Items.RED_SAND, 1, 4, 6, 1),
                     new VillagerTrades.ItemsForEmeralds(Items.POINTED_DRIPSTONE, 1, 2, 5, 1),
                     new VillagerTrades.ItemsForEmeralds(Items.ROOTED_DIRT, 1, 2, 5, 1),
@@ -1296,10 +_,6 @@
                     VillagerType.TAIGA,
                     new VillagerTrades.EnchantBookForEmeralds(p_294584_, Enchantments.BLAST_PROTECTION, Enchantments.FIRE_ASPECT, Enchantments.FLAME)
                 )
-                .put(
-                    VillagerType.POTATO,
-                    new VillagerTrades.EnchantBookForEmeralds(p_294584_, Enchantments.LOYALTY, Enchantments.IMPALING, Enchantments.MULTISHOT)
-                )
                 .build()
         );
     }
@@ -1314,7 +_,6 @@
                 .put(VillagerType.SNOW, new VillagerTrades.EnchantBookForEmeralds(30, Enchantments.SILK_TOUCH))
                 .put(VillagerType.SWAMP, new VillagerTrades.EnchantBookForEmeralds(30, Enchantments.MENDING))
                 .put(VillagerType.TAIGA, new VillagerTrades.EnchantBookForEmeralds(30, 2, 2, Enchantments.FORTUNE))
-                .put(VillagerType.POTATO, new VillagerTrades.EnchantBookForEmeralds(30, 2, 2, Enchantments.LUCK_OF_THE_SEA))
                 .build()
         );
     }
@@ -1356,7 +_,7 @@
 
         @Override
         public MerchantOffer getOffer(Entity p_219679_, RandomSource p_219680_) {
-            ItemCost itemcost = new ItemCost(Items.AMBER_GEM, this.value);
+            ItemCost itemcost = new ItemCost(Items.EMERALD, this.value);
             ItemStack itemstack = new ItemStack(this.item);
             if (itemstack.is(ItemTags.DYEABLE)) {
                 List<DyeItem> list = Lists.newArrayList();
@@ -1405,9 +_,7 @@
 
         @Override
         public MerchantOffer getOffer(Entity p_219682_, RandomSource p_219683_) {
-            return new MerchantOffer(
-                this.itemStack, new ItemStack(Items.POISONOUS_POTATO, this.emeraldAmount), this.maxUses, this.villagerXp, this.priceMultiplier
-            );
+            return new MerchantOffer(this.itemStack, new ItemStack(Items.EMERALD, this.emeraldAmount), this.maxUses, this.villagerXp, this.priceMultiplier);
         }
     }
 
@@ -1418,8 +_,8 @@
         private final int villagerXp;
 
         public EmeraldsForVillagerTypeItem(int p_35669_, int p_35670_, int p_35671_, Map<VillagerType, Item> p_35672_) {
-            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)).findAny().ifPresent(p_258962_ -> {
-                throw new IllegalStateException("Missing trade for villager type: " + BuiltInRegistries.VILLAGER_TYPE.getKey(p_258962_));
+            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)).findAny().ifPresent(p_339515_ -> {
+                throw new IllegalStateException("Missing trade for villager type: " + BuiltInRegistries.VILLAGER_TYPE.getKey(p_339515_));
             });
             this.trades = p_35672_;
             this.cost = p_35669_;
@@ -1432,7 +_,7 @@
         public MerchantOffer getOffer(Entity p_219685_, RandomSource p_219686_) {
             if (p_219685_ instanceof VillagerDataHolder villagerdataholder) {
                 ItemCost itemcost = new ItemCost(this.trades.get(villagerdataholder.getVillagerData().getType()), this.cost);
-                return new MerchantOffer(itemcost, new ItemStack(Items.POISONOUS_POTATO), this.maxUses, this.villagerXp, 0.05F);
+                return new MerchantOffer(itemcost, new ItemStack(Items.EMERALD), this.maxUses, this.villagerXp, 0.05F);
             } else {
                 return null;
             }
@@ -1446,7 +_,7 @@
         private final int maxLevel;
 
         public EnchantBookForEmeralds(int p_35683_) {
-            this(p_35683_, BuiltInRegistries.ENCHANTMENT.stream().filter(Enchantment::isTradeable).toArray(p_294101_ -> new Enchantment[p_294101_]));
+            this(p_35683_, BuiltInRegistries.ENCHANTMENT.stream().filter(Enchantment::isTradeable).toArray(Enchantment[]::new));
         }
 
         public EnchantBookForEmeralds(int p_296323_, Enchantment... p_294878_) {
@@ -1476,7 +_,7 @@
                 l = 64;
             }
 
-            return new MerchantOffer(new ItemCost(Items.AMBER_GEM, l), Optional.of(new ItemCost(Items.BOOK)), itemstack, 12, this.villagerXp, 0.2F);
+            return new MerchantOffer(new ItemCost(Items.EMERALD, l), Optional.of(new ItemCost(Items.BOOK)), itemstack, 12, this.villagerXp, 0.2F);
         }
     }
 
@@ -1502,9 +_,11 @@
         @Override
         public MerchantOffer getOffer(Entity p_219691_, RandomSource p_219692_) {
             int i = 5 + p_219692_.nextInt(15);
-            ItemStack itemstack = EnchantmentHelper.enchantItem(p_219692_, new ItemStack(this.itemStack.getItem()), i, false);
+            ItemStack itemstack = EnchantmentHelper.enchantItem(
+                p_219691_.level().enabledFeatures(), p_219692_, new ItemStack(this.itemStack.getItem()), i, false
+            );
             int j = Math.min(this.baseEmeraldCost + i, 64);
-            ItemCost itemcost = new ItemCost(Items.AMBER_GEM, j);
+            ItemCost itemcost = new ItemCost(Items.EMERALD, j);
             return new MerchantOffer(itemcost, itemstack, this.maxUses, this.villagerXp, this.priceMultiplier);
         }
     }
@@ -1550,7 +_,7 @@
         @Override
         public MerchantOffer getOffer(Entity p_219696_, RandomSource p_219697_) {
             return new MerchantOffer(
-                new ItemCost(Items.AMBER_GEM, this.emeraldCost),
+                new ItemCost(Items.EMERALD, this.emeraldCost),
                 Optional.of(this.fromItem),
                 this.toItem.copy(),
                 0,
@@ -1600,9 +_,7 @@
 
         @Override
         public MerchantOffer getOffer(Entity p_219699_, RandomSource p_219700_) {
-            return new MerchantOffer(
-                new ItemCost(Items.AMBER_GEM, this.emeraldCost), this.itemStack.copy(), this.maxUses, this.villagerXp, this.priceMultiplier
-            );
+            return new MerchantOffer(new ItemCost(Items.EMERALD, this.emeraldCost), this.itemStack.copy(), this.maxUses, this.villagerXp, this.priceMultiplier);
         }
     }
 
@@ -1626,7 +_,7 @@
         public MerchantOffer getOffer(Entity p_219702_, RandomSource p_219703_) {
             ItemStack itemstack = new ItemStack(Items.SUSPICIOUS_STEW, 1);
             itemstack.set(DataComponents.SUSPICIOUS_STEW_EFFECTS, this.effects);
-            return new MerchantOffer(new ItemCost(Items.AMBER_GEM), itemstack, 12, this.xp, this.priceMultiplier);
+            return new MerchantOffer(new ItemCost(Items.EMERALD), itemstack, 12, this.xp, this.priceMultiplier);
         }
     }
 
@@ -1653,7 +_,7 @@
 
         @Override
         public MerchantOffer getOffer(Entity p_219705_, RandomSource p_219706_) {
-            ItemCost itemcost = new ItemCost(Items.AMBER_GEM, this.emeraldCost);
+            ItemCost itemcost = new ItemCost(Items.EMERALD, this.emeraldCost);
             List<Holder<Potion>> list = BuiltInRegistries.POTION
                 .holders()
                 .filter(p_316045_ -> !p_316045_.value().getEffects().isEmpty() && PotionBrewing.isBrewablePotion(p_316045_))
@@ -1698,14 +_,9 @@
                     ItemStack itemstack = MapItem.create(serverlevel, blockpos.getX(), blockpos.getZ(), (byte)2, true, true);
                     MapItem.renderBiomePreviewMap(serverlevel, itemstack);
                     MapItemSavedData.addTargetDecoration(itemstack, blockpos, "+", this.destinationType);
-                    itemstack.set(DataComponents.CUSTOM_NAME, Component.translatable(this.displayName));
+                    itemstack.set(DataComponents.ITEM_NAME, Component.translatable(this.displayName));
                     return new MerchantOffer(
-                        new ItemCost(Items.AMBER_GEM, this.emeraldCost),
-                        Optional.of(new ItemCost(Items.COMPASS)),
-                        itemstack,
-                        this.maxUses,
-                        this.villagerXp,
-                        0.2F
+                        new ItemCost(Items.EMERALD, this.emeraldCost), Optional.of(new ItemCost(Items.COMPASS)), itemstack, this.maxUses, this.villagerXp, 0.2F
                     );
                 } else {
                     return null;
@@ -1716,7 +_,9 @@
 
     static record TypeSpecificTrade(Map<VillagerType, VillagerTrades.ItemListing> trades) implements VillagerTrades.ItemListing {
         public static VillagerTrades.TypeSpecificTrade oneTradeInBiomes(VillagerTrades.ItemListing p_302030_, VillagerType... p_301996_) {
-            return new VillagerTrades.TypeSpecificTrade(Arrays.stream(p_301996_).collect(Collectors.toMap(p_301920_ -> p_301920_, p_301922_ -> p_302030_)));
+            return new VillagerTrades.TypeSpecificTrade(
+                Arrays.stream(p_301996_).collect(Collectors.toMap(p_301920_ -> (VillagerType)p_301920_, p_301922_ -> p_302030_))
+            );
         }
 
         @Nullable
