--- a/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
+++ b/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
@@ -105,7 +_,7 @@
 
                 WanderingTrader wanderingtrader = EntityType.WANDERING_TRADER.spawn(p_35916_, blockpos2, MobSpawnType.EVENT);
                 if (wanderingtrader != null) {
-                    for(int j = 0; j < 2; ++j) {
+                    for (int j = 0; j < 2; j++) {
                         this.tryToSpawnLlamaFor(p_35916_, wanderingtrader, 4);
                     }
 
@@ -136,7 +_,7 @@
         BlockPos blockpos = null;
         SpawnPlacementType spawnplacementtype = SpawnPlacements.getPlacementType(EntityType.WANDERING_TRADER);
 
-        for(int i = 0; i < 10; ++i) {
+        for (int i = 0; i < 10; i++) {
             int j = p_35930_.getX() + this.random.nextInt(p_35931_ * 2) - p_35931_;
             int k = p_35930_.getZ() + this.random.nextInt(p_35931_ * 2) - p_35931_;
             int l = p_35929_.getHeight(Heightmap.Types.WORLD_SURFACE, j, k);
@@ -151,7 +_,7 @@
     }
 
     private boolean hasEnoughSpace(BlockGetter p_35926_, BlockPos p_35927_) {
-        for(BlockPos blockpos : BlockPos.betweenClosed(p_35927_, p_35927_.offset(1, 2, 1))) {
+        for (BlockPos blockpos : BlockPos.betweenClosed(p_35927_, p_35927_.offset(1, 2, 1))) {
             if (!p_35926_.getBlockState(blockpos).getCollisionShape(p_35926_, blockpos).isEmpty()) {
                 return false;
             }
