--- a/net/minecraft/client/model/geom/builders/PartDefinition.java
+++ b/net/minecraft/client/model/geom/builders/PartDefinition.java
@@ -40,7 +_,7 @@
             .collect(
                 Collectors.toMap(
                     Entry::getKey,
-                    p_171593_ -> ((PartDefinition)p_171593_.getValue()).bake(p_171584_, p_171585_),
+                    p_171593_ -> p_171593_.getValue().bake(p_171584_, p_171585_),
                     (p_171595_, p_171596_) -> p_171595_,
                     Object2ObjectArrayMap::new
                 )
