--- a/net/minecraft/client/renderer/item/BlockModelWrapper.java
+++ b/net/minecraft/client/renderer/item/BlockModelWrapper.java
@@ -56,7 +_,7 @@
     private final boolean animated;
     private final Function<ItemStack, RenderType> renderType;
 
-    BlockModelWrapper(List<ItemTintSource> tints, List<BakedQuad> quads, ModelRenderProperties properties, Function<ItemStack, RenderType> renderType) {
+    public BlockModelWrapper(List<ItemTintSource> tints, List<BakedQuad> quads, ModelRenderProperties properties, Function<ItemStack, RenderType> renderType) {
         this.tints = tints;
         this.quads = quads;
         this.properties = properties;
@@ -176,7 +_,9 @@
             TextureSlots textureslots = resolvedmodel.getTopTextureSlots();
             List<BakedQuad> list = resolvedmodel.bakeTopGeometry(textureslots, modelbaker, BlockModelRotation.IDENTITY).getAll();
             ModelRenderProperties modelrenderproperties = ModelRenderProperties.fromResolvedModel(modelbaker, resolvedmodel, textureslots);
-            Function<ItemStack, RenderType> function = BlockModelWrapper.detectRenderType(list);
+            var renderTypeGroup = resolvedmodel.getTopAdditionalProperties().getOptional(
+                    net.neoforged.neoforge.client.model.NeoForgeModelProperties.RENDER_TYPE);
+            Function<ItemStack, RenderType> function = renderTypeGroup == null ? BlockModelWrapper.detectRenderType(list) : net.neoforged.neoforge.client.RenderTypeHelper.detectItemModelRenderType(list, renderTypeGroup);
             return new BlockModelWrapper(this.tints, list, modelrenderproperties, function);
         }
 
