--- a/net/minecraft/client/animation/AnimationDefinition.java
+++ b/net/minecraft/client/animation/AnimationDefinition.java
@@ -29,7 +_,7 @@
         }
 
         public AnimationDefinition.Builder addAnimation(String p_232280_, AnimationChannel p_232281_) {
-            this.animationByBone.computeIfAbsent(p_232280_, p_329694_ -> new ArrayList()).add(p_232281_);
+            this.animationByBone.computeIfAbsent(p_232280_, p_329694_ -> new ArrayList<>()).add(p_232281_);
             return this;
         }
 
