--- a/net/minecraft/world/entity/vehicle/Boat.java
+++ b/net/minecraft/world/entity/vehicle/Boat.java
@@ -51,7 +_,7 @@
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
 
-public class Boat extends VehicleEntity implements VariantHolder<Boat.Type> {
+public class Boat extends VehicleEntity implements VariantHolder<Boat.Type>, net.neoforged.neoforge.common.extensions.IBoatExtension {
     private static final EntityDataAccessor<Integer> DATA_ID_TYPE = SynchedEntityData.defineId(Boat.class, EntityDataSerializers.INT);
     private static final EntityDataAccessor<Boolean> DATA_ID_PADDLE_LEFT = SynchedEntityData.defineId(Boat.class, EntityDataSerializers.BOOLEAN);
     private static final EntityDataAccessor<Boolean> DATA_ID_PADDLE_RIGHT = SynchedEntityData.defineId(Boat.class, EntityDataSerializers.BOOLEAN);
@@ -200,7 +_,7 @@
 
     @Override
     public Item getDropItem() {
-        return switch(this.getVariant()) {
+        return switch (this.getVariant()) {
             case SPRUCE -> Items.SPRUCE_BOAT;
             case BIRCH -> Items.BIRCH_BOAT;
             case JUNGLE -> Items.JUNGLE_BOAT;
@@ -272,7 +_,7 @@
         if (this.status != Boat.Status.UNDER_WATER && this.status != Boat.Status.UNDER_FLOWING_WATER) {
             this.outOfControlTicks = 0.0F;
         } else {
-            ++this.outOfControlTicks;
+            this.outOfControlTicks++;
         }
 
         if (!this.level().isClientSide && this.outOfControlTicks >= 60.0F) {
@@ -307,7 +_,7 @@
 
         this.tickBubbleColumn();
 
-        for(int i = 0; i <= 1; ++i) {
+        for (int i = 0; i <= 1; i++) {
             if (this.getPaddleState(i)) {
                 if (!this.isSilent()
                     && (double)(this.paddlePositions[i] % (float) (Math.PI * 2)) <= (float) (Math.PI / 4)
@@ -331,7 +_,7 @@
                     }
                 }
 
-                this.paddlePositions[i] += (float) (Math.PI / 8);
+                this.paddlePositions[i] = this.paddlePositions[i] + (float) (Math.PI / 8);
             } else {
                 this.paddlePositions[i] = 0.0F;
             }
@@ -342,7 +_,7 @@
         if (!list.isEmpty()) {
             boolean flag = !this.level().isClientSide && !(this.getControllingPassenger() instanceof Player);
 
-            for(Entity entity : list) {
+            for (Entity entity : list) {
                 if (!entity.hasPassenger(this)) {
                     if (flag
                         && this.getPassengers().size() < this.getMaxPassengers()
@@ -399,7 +_,7 @@
 
     @Nullable
     protected SoundEvent getPaddleSound() {
-        switch(this.getStatus()) {
+        switch (this.getStatus()) {
             case IN_WATER:
             case UNDER_WATER:
             case UNDER_FLOWING_WATER:
@@ -420,7 +_,7 @@
 
         if (this.lerpSteps > 0) {
             this.lerpPositionAndRotationStep(this.lerpSteps, this.lerpX, this.lerpY, this.lerpZ, this.lerpYRot, this.lerpXRot);
-            --this.lerpSteps;
+            this.lerpSteps--;
         }
     }
 
@@ -464,14 +_,14 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
         label39:
-        for(int k1 = k; k1 < l; ++k1) {
+        for (int k1 = k; k1 < l; k1++) {
             float f = 0.0F;
 
-            for(int l1 = i; l1 < j; ++l1) {
-                for(int i2 = i1; i2 < j1; ++i2) {
+            for (int l1 = i; l1 < j; l1++) {
+                for (int i2 = i1; i2 < j1; i2++) {
                     blockpos$mutableblockpos.set(l1, k1, i2);
                     FluidState fluidstate = this.level().getFluidState(blockpos$mutableblockpos);
-                    if (fluidstate.is(FluidTags.WATER)) {
+                    if (this.canBoatInFluid(fluidstate)) {
                         f = Math.max(f, fluidstate.getHeight(this.level(), blockpos$mutableblockpos));
                     }
 
@@ -503,11 +_,11 @@
         int k1 = 0;
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int l1 = i; l1 < j; ++l1) {
-            for(int i2 = i1; i2 < j1; ++i2) {
+        for (int l1 = i; l1 < j; l1++) {
+            for (int i2 = i1; i2 < j1; i2++) {
                 int j2 = (l1 != i && l1 != j - 1 ? 0 : 1) + (i2 != i1 && i2 != j1 - 1 ? 0 : 1);
                 if (j2 != 2) {
-                    for(int k2 = k; k2 < l; ++k2) {
+                    for (int k2 = k; k2 < l; k2++) {
                         if (j2 <= 0 || k2 != k && k2 != l - 1) {
                             blockpos$mutableblockpos.set(l1, k2, i2);
                             BlockState blockstate = this.level().getBlockState(blockpos$mutableblockpos);
@@ -517,8 +_,8 @@
                                     voxelshape,
                                     BooleanOp.AND
                                 )) {
-                                f += blockstate.getBlock().getFriction();
-                                ++k1;
+                                f += blockstate.getFriction(this.level(), blockpos$mutableblockpos, this);
+                                k1++;
                             }
                         }
                     }
@@ -541,12 +_,12 @@
         this.waterLevel = -Double.MAX_VALUE;
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int k1 = i; k1 < j; ++k1) {
-            for(int l1 = k; l1 < l; ++l1) {
-                for(int i2 = i1; i2 < j1; ++i2) {
+        for (int k1 = i; k1 < j; k1++) {
+            for (int l1 = k; l1 < l; l1++) {
+                for (int i2 = i1; i2 < j1; i2++) {
                     blockpos$mutableblockpos.set(k1, l1, i2);
                     FluidState fluidstate = this.level().getFluidState(blockpos$mutableblockpos);
-                    if (fluidstate.is(FluidTags.WATER)) {
+                    if (this.canBoatInFluid(fluidstate)) {
                         float f = (float)l1 + fluidstate.getHeight(this.level(), blockpos$mutableblockpos);
                         this.waterLevel = Math.max((double)f, this.waterLevel);
                         flag |= aabb.minY < (double)f;
@@ -571,12 +_,12 @@
         boolean flag = false;
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int k1 = i; k1 < j; ++k1) {
-            for(int l1 = k; l1 < l; ++l1) {
-                for(int i2 = i1; i2 < j1; ++i2) {
+        for (int k1 = i; k1 < j; k1++) {
+            for (int l1 = k; l1 < l; l1++) {
+                for (int i2 = i1; i2 < j1; i2++) {
                     blockpos$mutableblockpos.set(k1, l1, i2);
                     FluidState fluidstate = this.level().getFluidState(blockpos$mutableblockpos);
-                    if (fluidstate.is(FluidTags.WATER)
+                    if (this.canBoatInFluid(fluidstate)
                         && d0 < (double)((float)blockpos$mutableblockpos.getY() + fluidstate.getHeight(this.level(), blockpos$mutableblockpos))) {
                         if (!fluidstate.isSource()) {
                             return Boat.Status.UNDER_FLOWING_WATER;
@@ -627,7 +_,7 @@
 
             Vec3 vec3 = this.getDeltaMovement();
             this.setDeltaMovement(vec3.x * (double)this.invFriction, vec3.y + d0, vec3.z * (double)this.invFriction);
-            this.deltaRotation *= this.invFriction;
+            this.deltaRotation = this.deltaRotation * this.invFriction;
             if (d1 > 0.0) {
                 Vec3 vec31 = this.getDeltaMovement();
                 this.setDeltaMovement(vec31.x, (vec31.y + d1 * (this.getDefaultGravity() / 0.65)) * 0.75, vec31.z);
@@ -639,11 +_,11 @@
         if (this.isVehicle()) {
             float f = 0.0F;
             if (this.inputLeft) {
-                --this.deltaRotation;
+                this.deltaRotation--;
             }
 
             if (this.inputRight) {
-                ++this.deltaRotation;
+                this.deltaRotation++;
             }
 
             if (this.inputRight != this.inputLeft && !this.inputUp && !this.inputDown) {
@@ -713,8 +_,8 @@
                 list.add(new Vec3(d0, (double)blockpos1.getY() + d3, d1));
             }
 
-            for(Pose pose : p_38357_.getDismountPoses()) {
-                for(Vec3 vec31 : list) {
+            for (Pose pose : p_38357_.getDismountPoses()) {
+                for (Vec3 vec31 : list) {
                     if (DismountHelper.canDismountTo(this.level(), vec31, p_38357_, pose)) {
                         p_38357_.setPose(pose);
                         return vec31;
@@ -782,11 +_,11 @@
                     if (!this.level().isClientSide && !this.isRemoved()) {
                         this.kill();
                         if (this.level().getGameRules().getBoolean(GameRules.RULE_DOENTITYDROPS)) {
-                            for(int i = 0; i < 3; ++i) {
+                            for (int i = 0; i < 3; i++) {
                                 this.spawnAtLocation(this.getVariant().getPlanks());
                             }
 
-                            for(int j = 0; j < 2; ++j) {
+                            for (int j = 0; j < 2; j++) {
                                 this.spawnAtLocation(Items.STICK);
                             }
                         }
@@ -794,14 +_,14 @@
                 }
 
                 this.resetFallDistance();
-            } else if (!this.level().getFluidState(this.blockPosition().below()).is(FluidTags.WATER) && p_38307_ < 0.0) {
+            } else if (!this.canBoatInFluid(this.level().getFluidState(this.blockPosition().below())) && p_38307_ < 0.0D) {
                 this.fallDistance -= (float)p_38307_;
             }
         }
     }
 
     public boolean getPaddleState(int p_38314_) {
-        return this.entityData.<Boolean>get(p_38314_ == 0 ? DATA_ID_PADDLE_LEFT : DATA_ID_PADDLE_RIGHT) && this.getControllingPassenger() != null;
+        return this.entityData.get(p_38314_ == 0 ? DATA_ID_PADDLE_LEFT : DATA_ID_PADDLE_RIGHT) && this.getControllingPassenger() != null;
     }
 
     private void setBubbleTime(int p_38367_) {
@@ -826,7 +_,7 @@
 
     @Override
     protected boolean canAddPassenger(Entity p_38390_) {
-        return this.getPassengers().size() < this.getMaxPassengers() && !this.isEyeInFluid(FluidTags.WATER);
+        return this.getPassengers().size() < this.getMaxPassengers() && !this.canBoatInFluid(this.getEyeInFluidType());
     }
 
     protected int getMaxPassengers() {
@@ -836,8 +_,7 @@
     @Nullable
     @Override
     public LivingEntity getControllingPassenger() {
-        Entity entity = this.getFirstPassenger();
-        return entity instanceof LivingEntity livingentity ? livingentity : super.getControllingPassenger();
+        return this.getFirstPassenger() instanceof LivingEntity livingentity ? livingentity : super.getControllingPassenger();
     }
 
     public void setInput(boolean p_38343_, boolean p_38344_, boolean p_38345_, boolean p_38346_) {
