--- a/net/minecraft/world/entity/vehicle/DismountHelper.java
+++ b/net/minecraft/world/entity/vehicle/DismountHelper.java
@@ -39,7 +_,7 @@
     }
 
     public static boolean canDismountTo(CollisionGetter p_38457_, LivingEntity p_38458_, AABB p_38459_) {
-        for(VoxelShape voxelshape : p_38457_.getBlockCollisions(p_38458_, p_38459_)) {
+        for (VoxelShape voxelshape : p_38457_.getBlockCollisions(p_38458_, p_38459_)) {
             if (!voxelshape.isEmpty()) {
                 return false;
             }
@@ -63,13 +_,13 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_38464_.mutable();
         int i = 0;
 
-        while(i < p_38465_) {
+        while (i < p_38465_) {
             VoxelShape voxelshape = p_38466_.apply(blockpos$mutableblockpos);
             if (!voxelshape.isEmpty()) {
                 return (double)(p_38464_.getY() + i) + voxelshape.min(Direction.Axis.Y);
             }
 
-            ++i;
+            i++;
             blockpos$mutableblockpos.move(Direction.UP);
         }
 
@@ -90,7 +_,7 @@
                 Vec3 vec3 = Vec3.upFromBottomCenterOf(p_38444_, d0);
                 AABB aabb = p_38442_.getDimensions().makeBoundingBox(vec3);
 
-                for(VoxelShape voxelshape : p_38443_.getBlockCollisions(null, aabb)) {
+                for (VoxelShape voxelshape : p_38443_.getBlockCollisions(null, aabb)) {
                     if (!voxelshape.isEmpty()) {
                         return null;
                     }
