--- a/net/minecraft/client/resources/IndexedAssetSource.java
+++ b/net/minecraft/client/resources/IndexedAssetSource.java
@@ -32,7 +_,7 @@
             JsonObject jsonobject = GsonHelper.parse(bufferedreader);
             JsonObject jsonobject1 = GsonHelper.getAsJsonObject(jsonobject, "objects", null);
             if (jsonobject1 != null) {
-                for(Entry<String, JsonElement> entry : jsonobject1.entrySet()) {
+                for (Entry<String, JsonElement> entry : jsonobject1.entrySet()) {
                     JsonObject jsonobject2 = (JsonObject)entry.getValue();
                     String s = entry.getKey();
                     List<String> list = PATH_SPLITTER.splitToList(s);
