--- a/net/minecraft/client/resources/MobEffectTextureManager.java
+++ b/net/minecraft/client/resources/MobEffectTextureManager.java
@@ -1,5 +_,7 @@
 package net.minecraft.client.resources;
 
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.client.renderer.texture.MissingTextureAtlasSprite;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.client.renderer.texture.TextureManager;
@@ -7,6 +_,8 @@
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.flag.FeatureFlags;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
@@ -17,6 +_,13 @@
     }
 
     public TextureAtlasSprite get(Holder<MobEffect> p_316705_) {
+        if (p_316705_ == MobEffects.BAD_OMEN) {
+            ClientLevel clientlevel = Minecraft.getInstance().level;
+            if (clientlevel != null && clientlevel.enabledFeatures().contains(FeatureFlags.UPDATE_1_21)) {
+                return this.getSprite(new ResourceLocation("bad_omen_121"));
+            }
+        }
+
         return this.getSprite(p_316705_.unwrapKey().map(ResourceKey::location).orElseGet(MissingTextureAtlasSprite::getLocation));
     }
 }
