--- a/net/minecraft/client/resources/SkinManager.java
+++ b/net/minecraft/client/resources/SkinManager.java
@@ -161,10 +_,11 @@
         }
 
         private ResourceLocation getTextureLocation(String p_295006_) {
-            String s = switch(this.type) {
+            String s = switch (this.type) {
                 case SKIN -> "skins";
                 case CAPE -> "capes";
                 case ELYTRA -> "elytra";
+                default -> throw new MatchException(null, null);
             };
             return new ResourceLocation(s + "/" + p_295006_);
         }
