--- a/net/minecraft/core/BlockPos.java
+++ b/net/minecraft/core/BlockPos.java
@@ -33,7 +_,7 @@
 public class BlockPos extends Vec3i {
     public static final Codec<BlockPos> CODEC = Codec.INT_STREAM
         .<BlockPos>comapFlatMap(
-            p_121967_ -> Util.fixedSize(p_121967_, 3).map(p_175270_ -> new BlockPos(p_175270_[0], p_175270_[1], p_175270_[2])),
+            p_337445_ -> Util.fixedSize(p_337445_, 3).map(p_175270_ -> new BlockPos(p_175270_[0], p_175270_[1], p_175270_[2])),
             p_121924_ -> IntStream.of(p_121924_.getX(), p_121924_.getY(), p_121924_.getZ())
         )
         .stable();
@@ -223,7 +_,7 @@
     }
 
     public BlockPos rotate(Rotation p_121918_) {
-        switch(p_121918_) {
+        switch (p_121918_) {
             case NONE:
             default:
                 return this;
@@ -256,10 +_,6 @@
         return new BlockPos.MutableBlockPos(this.getX(), this.getY(), this.getZ());
     }
 
-    public BlockPos step(Direction p_340635_) {
-        return this.offset(p_340635_.getNormal());
-    }
-
     public static Iterable<BlockPos> randomInCube(RandomSource p_235651_, int p_235652_, BlockPos p_235653_, int p_235654_) {
         return randomBetweenClosed(
             p_235651_,
@@ -294,7 +_,7 @@
                     } else {
                         BlockPos blockpos = this.nextPos
                             .set(p_235644_ + p_235642_.nextInt(i), p_235645_ + p_235642_.nextInt(j), p_235646_ + p_235642_.nextInt(k));
-                        --this.counter;
+                        this.counter--;
                         return blockpos;
                     }
                 }
@@ -322,11 +_,11 @@
                         return this.cursor;
                     } else {
                         BlockPos blockpos;
-                        for(blockpos = null; blockpos == null; ++this.y) {
+                        for (blockpos = null; blockpos == null; this.y++) {
                             if (this.y > this.maxY) {
-                                ++this.x;
+                                this.x++;
                                 if (this.x > this.maxX) {
-                                    ++this.currentDepth;
+                                    this.currentDepth++;
                                     if (this.currentDepth > i) {
                                         return this.endOfData();
                                     }
@@ -355,7 +_,7 @@
     }
 
     public static Optional<BlockPos> findClosestMatch(BlockPos p_121931_, int p_121932_, int p_121933_, Predicate<BlockPos> p_121934_) {
-        for(BlockPos blockpos : withinManhattan(p_121931_, p_121932_, p_121933_, p_121932_)) {
+        for (BlockPos blockpos : withinManhattan(p_121931_, p_121932_, p_121933_, p_121932_)) {
             if (p_121934_.test(blockpos)) {
                 return Optional.of(blockpos);
             }
@@ -426,7 +_,7 @@
                         int j1 = this.index / i;
                         int k1 = j1 % j;
                         int l1 = j1 / j;
-                        ++this.index;
+                        this.index++;
                         return this.cursor.set(p_121977_ + i1, p_121978_ + k1, p_121979_ + l1);
                     }
                 }
@@ -456,12 +_,12 @@
                             return this.endOfData();
                         }
 
-                        ++this.leg;
+                        this.leg++;
                         this.legIndex = 0;
                         this.legSize = this.leg / 2 + 1;
                     }
 
-                    ++this.legIndex;
+                    this.legIndex++;
                     return this.cursor;
                 }
             };
@@ -475,7 +_,7 @@
         queue.add(Pair.of(p_278078_, 0));
         int i = 0;
 
-        while(!queue.isEmpty()) {
+        while (!queue.isEmpty()) {
             Pair<BlockPos, Integer> pair = queue.poll();
             BlockPos blockpos = pair.getLeft();
             int j = pair.getRight();
@@ -507,10 +_,6 @@
             this(Mth.floor(p_122126_), Mth.floor(p_122127_), Mth.floor(p_122128_));
         }
 
-        public MutableBlockPos(BlockPos p_341378_) {
-            super(p_341378_);
-        }
-
         @Override
         public BlockPos offset(int p_122163_, int p_122164_, int p_122165_) {
             return super.offset(p_122163_, p_122164_, p_122165_).immutable();
@@ -594,7 +_,7 @@
         }
 
         public BlockPos.MutableBlockPos clamp(Direction.Axis p_122148_, int p_122149_, int p_122150_) {
-            switch(p_122148_) {
+            switch (p_122148_) {
                 case X:
                     return this.set(Mth.clamp(this.getX(), p_122149_, p_122150_), this.getY(), this.getZ());
                 case Y:
