--- a/net/minecraft/core/BlockMath.java
+++ b/net/minecraft/core/BlockMath.java
@@ -23,7 +_,7 @@
         p_121851_.put(Direction.DOWN, new Transformation(null, new Quaternionf().rotateX((float) (Math.PI / 2)), null, null));
     });
     public static final Map<Direction, Transformation> VANILLA_UV_TRANSFORM_GLOBAL_TO_LOCAL = Util.make(Maps.newEnumMap(Direction.class), p_121849_ -> {
-        for(Direction direction : Direction.values()) {
+        for (Direction direction : Direction.values()) {
             p_121849_.put(direction, VANILLA_UV_TRANSFORM_LOCAL_TO_GLOBAL.get(direction).inverse());
         }
     });
