--- a/net/minecraft/core/Cursor3D.java
+++ b/net/minecraft/core/Cursor3D.java
@@ -35,7 +_,7 @@
             int i = this.index / this.width;
             this.y = i % this.height;
             this.z = i / this.height;
-            ++this.index;
+            this.index++;
             return true;
         }
     }
@@ -55,15 +_,15 @@
     public int getNextType() {
         int i = 0;
         if (this.x == 0 || this.x == this.width - 1) {
-            ++i;
+            i++;
         }
 
         if (this.y == 0 || this.y == this.height - 1) {
-            ++i;
+            i++;
         }
 
         if (this.z == 0 || this.z == this.depth - 1) {
-            ++i;
+            i++;
         }
 
         return i;
