--- a/net/minecraft/core/Direction.java
+++ b/net/minecraft/core/Direction.java
@@ -324,6 +_,11 @@
     }
 
     @Contract("_,_,_,!null->!null;_,_,_,_->_")
+    public static @Nullable Direction getNearest(int x, int y, int z, @Nullable Direction orElse) {
+        return getNearest((float)x, (float)y, (float)z, orElse);
+    }
+
+    @Contract("_,_,_,!null->!null;_,_,_,_->_")
     public static @Nullable Direction getNearest(float x, float y, float z, @Nullable Direction orElse) {
         float absX = Math.abs(x);
         float absY = Math.abs(y);
