--- a/net/minecraft/core/FrontAndTop.java
+++ b/net/minecraft/core/FrontAndTop.java
@@ -20,7 +_,7 @@
     SOUTH_UP("south_up", Direction.SOUTH, Direction.UP);
 
     private static final Int2ObjectMap<FrontAndTop> LOOKUP_TOP_FRONT = Util.make(new Int2ObjectOpenHashMap<>(values().length), p_175377_ -> {
-        for(FrontAndTop frontandtop : values()) {
+        for (FrontAndTop frontandtop : values()) {
             p_175377_.put(lookupKey(frontandtop.front, frontandtop.top), frontandtop);
         }
     });
