--- a/net/minecraft/core/Holder.java
+++ b/net/minecraft/core/Holder.java
@@ -1,6 +_,7 @@
 package net.minecraft.core;
 
 import com.mojang.datafixers.util.Either;
+import com.mojang.logging.LogUtils;
 import java.util.Collection;
 import java.util.Objects;
 import java.util.Optional;
@@ -12,6 +_,7 @@
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.tags.TagKey;
 import org.jspecify.annotations.Nullable;
+import org.slf4j.Logger;
 
 public interface Holder<T> extends net.neoforged.neoforge.common.extensions.IHolderExtension<T> {
     T value();
@@ -128,6 +_,7 @@
     }
 
     public static class Reference<T> implements Holder<T> {
+        private static final Logger LOGGER = LogUtils.getLogger();
         private final HolderOwner<T> owner;
         private @Nullable Set<TagKey<T>> tags;
         private @Nullable DataComponentMap components;
@@ -160,9 +_,10 @@
         }
 
         @Override
-        public T value() {
+        public @Nullable T value() {
             if (this.value == null) {
-                throw new IllegalStateException("Trying to access unbound value '" + this.key + "' from registry " + this.owner);
+                LOGGER.warn("Trying to access unbound value '{}' from registry {}. Returning null.", this.key, this.owner);
+                return null;
             } else {
                 return this.value;
             }
@@ -247,6 +_,14 @@
             }
         }
 
+        @org.jspecify.annotations.Nullable
+        public <A> A getData(net.neoforged.neoforge.registries.datamaps.DataMapType<T, A> type) {
+            if (owner instanceof HolderLookup.RegistryLookup<T> lookup) {
+                return lookup.getData(type, key());
+            }
+            return null;
+        }
+
         void bindTags(Collection<TagKey<T>> tags) {
             this.tags = Set.copyOf(tags);
         }
@@ -268,6 +_,32 @@
         @Override
         public String toString() {
             return "Reference{" + this.key + "=" + this.value + "}";
+        }
+
+        // Neo: Add key getter that doesn't allocate
+        @Override
+        @org.jspecify.annotations.Nullable
+        public ResourceKey<T> getKey() {
+            return this.key;
+        }
+
+        // Neo: Add DeferredHolder-compatible hashCode() overrides
+        @Override
+        public int hashCode() {
+            return key().hashCode();
+        }
+
+        // Neo: Add DeferredHolder-compatible equals() overrides
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            return obj instanceof Holder<?> h && h.kind() == Kind.REFERENCE && h.getKey() == this.key();
+        }
+
+        // Neo: Helper method to get the registry lookup from a holder
+        @Override
+        public HolderLookup.@Nullable RegistryLookup<T> unwrapLookup() {
+            return this.owner instanceof HolderLookup.RegistryLookup<T> rl ? rl : null;
         }
 
         protected static enum Type {
