--- a/net/minecraft/core/IdMapper.java
+++ b/net/minecraft/core/IdMapper.java
@@ -27,8 +_,8 @@
     public void addMapping(T p_122665_, int p_122666_) {
         this.tToId.put(p_122665_, p_122666_);
 
-        while(this.idToT.size() <= p_122666_) {
-            this.idToT.add((T)null);
+        while (this.idToT.size() <= p_122666_) {
+            this.idToT.add(null);
         }
 
         this.idToT.set(p_122666_, p_122665_);
