--- a/net/minecraft/core/LayeredRegistryAccess.java
+++ b/net/minecraft/core/LayeredRegistryAccess.java
@@ -67,13 +_,13 @@
         } else {
             List<RegistryAccess.Frozen> list = new ArrayList<>();
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 list.add(this.values.get(j));
             }
 
             list.addAll(p_250124_);
 
-            while(list.size() < this.values.size()) {
+            while (list.size() < this.values.size()) {
                 list.add(RegistryAccess.EMPTY);
             }
 
@@ -87,9 +_,9 @@
 
     private static Map<ResourceKey<? extends Registry<?>>, Registry<?>> collectRegistries(Stream<? extends RegistryAccess> p_248595_) {
         Map<ResourceKey<? extends Registry<?>>, Registry<?>> map = new HashMap<>();
-        p_248595_.forEach(p_252003_ -> p_252003_.registries().forEach(p_250413_ -> {
-                if (map.put(p_250413_.key(), p_250413_.value()) != null) {
-                    throw new IllegalStateException("Duplicated registry " + p_250413_.key());
+        p_248595_.forEach(p_252003_ -> p_252003_.registries().forEach(p_339330_ -> {
+                if (map.put(p_339330_.key(), p_339330_.value()) != null) {
+                    throw new IllegalStateException("Duplicated registry " + p_339330_.key());
                 }
             }));
         return map;
