--- a/net/minecraft/core/MappedRegistry.java
+++ b/net/minecraft/core/MappedRegistry.java
@@ -31,7 +_,7 @@
 import net.minecraft.util.RandomSource;
 import org.slf4j.Logger;
 
-public class MappedRegistry<T> implements WritableRegistry<T> {
+public class MappedRegistry<T> extends net.neoforged.neoforge.registries.BaseMappedRegistry<T> implements WritableRegistry<T> {
     private static final Logger LOGGER = LogUtils.getLogger();
     final ResourceKey<? extends Registry<T>> key;
     private final ObjectList<Holder.Reference<T>> byId = new ObjectArrayList<>(256);
@@ -75,6 +_,12 @@
         public Stream<HolderSet.Named<T>> listTags() {
             return MappedRegistry.this.getTags().map(Pair::getSecond);
         }
+
+        @Override
+        @org.jetbrains.annotations.Nullable
+        public <A> A getData(net.neoforged.neoforge.registries.datamaps.DataMapType<T, A> type, ResourceKey<T> key) {
+            return MappedRegistry.this.getData(type, key);
+        }
     };
 
     public MappedRegistry(ResourceKey<? extends Registry<T>> p_249899_, Lifecycle p_252249_) {
@@ -113,9 +_,17 @@
 
     @Override
     public Holder.Reference<T> register(ResourceKey<T> p_256252_, T p_256591_, RegistrationInfo p_326235_) {
+        return register(this.byId.size(), p_256252_, p_256591_, p_326235_);
+    }
+
+    public Holder.Reference<T> register(int id, ResourceKey<T> p_256252_, T p_256591_, RegistrationInfo p_326235_) {
         this.validateWrite(p_256252_);
         Objects.requireNonNull(p_256252_);
         Objects.requireNonNull(p_256591_);
+        int i = id;
+        if (i > this.getMaxId())
+            throw new IllegalStateException(String.format(java.util.Locale.ENGLISH, "Invalid id %d - maximum id range of %d exceeded.", i, this.getMaxId()));
+
         if (this.byLocation.containsKey(p_256252_.location())) {
             Util.pauseInIde(new IllegalStateException("Adding duplicate key '" + p_256252_ + "' to registry"));
         }
@@ -133,17 +_,19 @@
 
             reference.bindKey(p_256252_);
         } else {
-            reference = this.byKey.computeIfAbsent(p_256252_, p_258168_ -> Holder.Reference.createStandAlone(this.holderOwner(), p_258168_));
+            reference = this.byKey.computeIfAbsent(p_256252_, p_258168_ -> Holder.Reference.createStandAlone(this.holderOwner(), (ResourceKey<T>)p_258168_));
+            // Forge: Bind the value immediately so it can be queried while the registry is not frozen
+            reference.bindValue(p_256591_);
         }
 
         this.byKey.put(p_256252_, reference);
         this.byLocation.put(p_256252_.location(), reference);
         this.byValue.put(p_256591_, reference);
-        int i = this.byId.size();
         this.byId.add(reference);
         this.toId.put(p_256591_, i);
         this.registrationInfos.put(p_256252_, p_326235_);
         this.registryLifecycle = this.registryLifecycle.add(p_326235_.lifecycle());
+        this.addCallbacks.forEach(addCallback -> addCallback.onAdd(this, i, p_256252_, p_256591_));
         return reference;
     }
 
@@ -167,7 +_,7 @@
     @Nullable
     @Override
     public T get(@Nullable ResourceKey<T> p_122714_) {
-        return getValueFromNullable(this.byKey.get(p_122714_));
+        return getValueFromNullable(this.byKey.get(resolve(p_122714_)));
     }
 
     @Nullable
@@ -188,7 +_,7 @@
 
     @Override
     public Optional<Holder.Reference<T>> getHolder(ResourceKey<T> p_205905_) {
-        return Optional.ofNullable(this.byKey.get(p_205905_));
+        return Optional.ofNullable(this.byKey.get(resolve(p_205905_)));
     }
 
     @Override
@@ -198,12 +_,12 @@
     }
 
     Holder.Reference<T> getOrCreateHolderOrThrow(ResourceKey<T> p_248831_) {
-        return this.byKey.computeIfAbsent(p_248831_, p_258169_ -> {
+        return this.byKey.computeIfAbsent(resolve(p_248831_), p_258169_ -> {
             if (this.unregisteredIntrusiveHolders != null) {
                 throw new IllegalStateException("This registry can't create new holders without value");
             } else {
-                this.validateWrite(p_258169_);
-                return Holder.Reference.createStandAlone(this.holderOwner(), p_258169_);
+                this.validateWrite((ResourceKey<T>)p_258169_);
+                return Holder.Reference.createStandAlone(this.holderOwner(), (ResourceKey<T>)p_258169_);
             }
         });
     }
@@ -231,7 +_,7 @@
     @Nullable
     @Override
     public T get(@Nullable ResourceLocation p_122739_) {
-        Holder.Reference<T> reference = this.byLocation.get(p_122739_);
+        Holder.Reference<T> reference = this.byLocation.get(resolve(p_122739_));
         return getValueFromNullable(reference);
     }
 
@@ -307,13 +_,18 @@
         return this.byKey.containsKey(p_175392_);
     }
 
+    /** @deprecated Forge: For internal use only. Use the Register events when registering values. */
+    @Deprecated
+    public void unfreeze() {
+        this.frozen = false;
+    }
+
     @Override
     public Registry<T> freeze() {
         if (this.frozen) {
             return this;
         } else {
             this.frozen = true;
-            this.byValue.forEach((p_247989_, p_247990_) -> p_247990_.bindValue(p_247989_));
             List<ResourceLocation> list = this.byKey
                 .entrySet()
                 .stream()
@@ -329,8 +_,10 @@
                         throw new IllegalStateException("Some intrusive holders were not registered: " + this.unregisteredIntrusiveHolders.values());
                     }
 
-                    this.unregisteredIntrusiveHolders = null;
+                    // Neo: We freeze/unfreeze vanilla registries more than once, so we need to keep the unregistered intrusive holders map around.
+                    // this.unregisteredIntrusiveHolders = null;
                 }
+                this.bakeCallbacks.forEach(bakeCallback -> bakeCallback.onBake(this));
 
                 return this;
             }
@@ -343,7 +_,7 @@
             throw new IllegalStateException("This registry can't create intrusive holders");
         } else {
             this.validateWrite();
-            return this.unregisteredIntrusiveHolders.computeIfAbsent(p_205915_, p_258166_ -> Holder.Reference.createIntrusive(this.asLookup(), p_258166_));
+            return this.unregisteredIntrusiveHolders.computeIfAbsent(p_205915_, p_258166_ -> Holder.Reference.createIntrusive(this.asLookup(), (T)p_258166_));
         }
     }
 
@@ -355,19 +_,18 @@
     @Override
     public void bindTags(Map<TagKey<T>, List<Holder<T>>> p_205875_) {
         Map<Holder.Reference<T>, List<TagKey<T>>> map = new IdentityHashMap<>();
-        this.byKey.values().forEach(p_211801_ -> map.put(p_211801_, new ArrayList<>()));
-        p_205875_.forEach((p_211806_, p_211807_) -> {
-            for(Holder<T> holder : p_211807_) {
+        this.byKey.values().forEach(p_211801_ -> map.put((Holder.Reference<T>)p_211801_, new ArrayList<>()));
+        p_205875_.forEach((p_339332_, p_339333_) -> {
+            for (Holder<T> holder : p_339333_) {
                 if (!holder.canSerializeIn(this.asLookup())) {
-                    throw new IllegalStateException("Can't create named set " + p_211806_ + " containing value " + holder + " from outside registry " + this);
-                }
-
-                if (!(holder instanceof Holder.Reference)) {
-                    throw new IllegalStateException("Found direct holder " + holder + " value in tag " + p_211806_);
-                }
-
-                Holder.Reference<T> reference = (Holder.Reference)holder;
-                map.get(reference).add(p_211806_);
+                    throw new IllegalStateException("Can't create named set " + p_339332_ + " containing value " + holder + " from outside registry " + this);
+                }
+
+                if (!(holder instanceof Holder.Reference<T> reference)) {
+                    throw new IllegalStateException("Found direct holder " + holder + " value in tag " + p_339332_);
+                }
+
+                map.get(reference).add((TagKey<T>)p_339332_);
             }
         });
         Set<TagKey<T>> set = Sets.difference(this.tags.keySet(), p_205875_.keySet());
@@ -380,7 +_,7 @@
         }
 
         Map<TagKey<T>, HolderSet.Named<T>> map1 = new IdentityHashMap<>(this.tags);
-        p_205875_.forEach((p_211797_, p_211798_) -> map1.computeIfAbsent(p_211797_, this::createTag).bind(p_211798_));
+        p_205875_.forEach((p_211797_, p_211798_) -> map1.computeIfAbsent((TagKey<T>)p_211797_, this::createTag).bind((List<Holder<T>>)p_211798_));
         map.forEach(Holder.Reference::bindTags);
         this.tags = map1;
     }
@@ -425,5 +_,53 @@
     @Override
     public HolderLookup.RegistryLookup<T> asLookup() {
         return this.lookup;
+    }
+
+    @Override
+    protected void clear(boolean full) {
+        this.validateWrite();
+        this.clearCallbacks.forEach(clearCallback -> clearCallback.onClear(this, full));
+        super.clear(full);
+        this.byId.clear();
+        this.toId.clear();
+        if (full) {
+            this.byLocation.clear();
+            this.byKey.clear();
+            this.byValue.clear();
+            this.tags.clear();
+            if (unregisteredIntrusiveHolders != null) {
+                unregisteredIntrusiveHolders.clear();
+                unregisteredIntrusiveHolders = null;
+            }
+        }
+    }
+
+    @Override
+    protected void registerIdMapping(ResourceKey<T> key, int id) {
+        this.validateWrite(key);
+        if (id > this.getMaxId())
+            throw new IllegalStateException(String.format(java.util.Locale.ENGLISH, "Invalid id %d - maximum id range of %d exceeded.", id, this.getMaxId()));
+        if (0 <= id && id < this.byId.size() && this.byId.get(id) != null) { // Don't use byId() method, it will return the default value if the entry is absent
+            throw new IllegalStateException("Duplicate id " + id + " for " + key + " and " + this.getKey(this.byId.get(id).value()));
+        }
+        var holder = byKey.get(key);
+        while (this.byId.size() < (id + 1)) this.byId.add(null);
+        this.byId.set(id, holder);
+        this.toId.put(holder.value(), id);
+    }
+
+    @Override
+    public int getId(ResourceLocation name) {
+        return getId(get(name));
+    }
+
+    @Override
+    public int getId(ResourceKey<T> key) {
+        return getId(get(key));
+    }
+
+    @Override
+    public boolean containsValue(T value) {
+        return byValue.containsKey(value);
     }
 }
