--- a/net/minecraft/core/RegistryAccess.java
+++ b/net/minecraft/core/RegistryAccess.java
@@ -7,7 +_,6 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.Map.Entry;
-import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import net.minecraft.resources.ResourceKey;
@@ -76,8 +_,7 @@
         private final Map<? extends ResourceKey<? extends Registry<?>>, ? extends Registry<?>> registries;
 
         public ImmutableRegistryAccess(List<? extends Registry<?>> p_248540_) {
-            this.registries = p_248540_.stream()
-                .collect(Collectors.toUnmodifiableMap(Registry::key, (p_206232_ -> p_206232_)));
+            this.registries = p_248540_.stream().collect(Collectors.toUnmodifiableMap(Registry::key, p_206232_ -> p_206232_));
         }
 
         public ImmutableRegistryAccess(Map<? extends ResourceKey<? extends Registry<?>>, ? extends Registry<?>> p_206225_) {
@@ -90,7 +_,7 @@
 
         @Override
         public <E> Optional<Registry<E>> registry(ResourceKey<? extends Registry<? extends E>> p_206229_) {
-            return Optional.ofNullable(this.registries.get(p_206229_)).map((Function<? super Registry<?>, ? extends Registry<E>>)(p_247993_ -> (Registry<E>)p_247993_));
+            return Optional.ofNullable(this.registries.get(p_206229_)).map(p_247993_ -> (Registry<E>)p_247993_);
         }
 
         @Override
@@ -103,11 +_,11 @@
         private static <T, R extends Registry<? extends T>> RegistryAccess.RegistryEntry<T> fromMapEntry(
             Entry<? extends ResourceKey<? extends Registry<?>>, R> p_206242_
         ) {
-            return fromUntyped(p_206242_.getKey(), p_206242_.getValue());
+            return fromUntyped((ResourceKey<? extends Registry<?>>)p_206242_.getKey(), p_206242_.getValue());
         }
 
         private static <T> RegistryAccess.RegistryEntry<T> fromUntyped(ResourceKey<? extends Registry<?>> p_206244_, Registry<?> p_206245_) {
-            return new RegistryAccess.RegistryEntry<T>((ResourceKey<? extends Registry<T>>)p_206244_, (Registry<T>)p_206245_);
+            return new RegistryAccess.RegistryEntry<>((ResourceKey<? extends Registry<T>>)p_206244_, (Registry<T>)p_206245_);
         }
 
         private RegistryAccess.RegistryEntry<T> freeze() {
