--- a/net/minecraft/core/RegistrySynchronization.java
+++ b/net/minecraft/core/RegistrySynchronization.java
@@ -9,7 +_,6 @@
 import java.util.function.BiConsumer;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import net.minecraft.Util;
 import net.minecraft.nbt.Tag;
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
@@ -31,7 +_,8 @@
         Set<KnownPack> p_326143_,
         BiConsumer<ResourceKey<? extends Registry<?>>, List<RegistrySynchronization.PackedRegistryEntry>> p_321539_
     ) {
-        RegistryDataLoader.SYNCHRONIZED_REGISTRIES.forEach(p_325532_ -> packRegistry(p_321796_, p_325532_, p_321733_, p_326143_, p_321539_));
+        RegistryDataLoader.SYNCHRONIZED_REGISTRIES
+            .forEach(p_325532_ -> packRegistry(p_321796_, (RegistryDataLoader.RegistryData<?>)p_325532_, p_321733_, p_326143_, p_321539_));
     }
 
     private static <T> void packRegistry(
@@ -56,10 +_,9 @@
                                 if (flag) {
                                     optional = Optional.empty();
                                 } else {
-                                    Tag tag = Util.getOrThrow(
-                                        p_321701_.elementCodec().encodeStart(p_321608_, p_325522_.value()),
-                                        p_321385_ -> new IllegalArgumentException("Failed to serialize " + p_325522_.key() + ": " + p_321385_)
-                                    );
+                                    Tag tag = p_321701_.elementCodec()
+                                        .encodeStart(p_321608_, p_325522_.value())
+                                        .getOrThrow(p_339341_ -> new IllegalArgumentException("Failed to serialize " + p_325522_.key() + ": " + p_339341_));
                                     optional = Optional.of(tag);
                                 }
                 
