--- a/net/minecraft/core/UUIDUtil.java
+++ b/net/minecraft/core/UUIDUtil.java
@@ -3,7 +_,6 @@
 import com.google.common.collect.Lists;
 import com.google.common.collect.Sets;
 import com.mojang.authlib.GameProfile;
-import com.mojang.datafixers.util.Either;
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
 import com.mojang.serialization.Dynamic;
@@ -23,7 +_,7 @@
 
 public final class UUIDUtil {
     public static final Codec<UUID> CODEC = Codec.INT_STREAM
-        .comapFlatMap(p_235884_ -> Util.fixedSize(p_235884_, 4).map(UUIDUtil::uuidFromIntArray), p_235888_ -> Arrays.stream(uuidToIntArray(p_235888_)));
+        .comapFlatMap(p_337446_ -> Util.fixedSize(p_337446_, 4).map(UUIDUtil::uuidFromIntArray), p_235888_ -> Arrays.stream(uuidToIntArray(p_235888_)));
     public static final Codec<Set<UUID>> CODEC_SET = Codec.list(CODEC).xmap(Sets::newHashSet, Lists::newArrayList);
     public static final Codec<Set<UUID>> CODEC_LINKED_SET = Codec.list(CODEC).xmap(Sets::newLinkedHashSet, Lists::newArrayList);
     public static final Codec<UUID> STRING_CODEC = Codec.STRING.comapFlatMap(p_274732_ -> {
@@ -33,15 +_,14 @@
             return DataResult.error(() -> "Invalid UUID " + p_274732_ + ": " + illegalargumentexception.getMessage());
         }
     }, UUID::toString);
-    public static Codec<UUID> AUTHLIB_CODEC = Codec.either(CODEC, Codec.STRING.comapFlatMap(p_293693_ -> {
+    public static Codec<UUID> AUTHLIB_CODEC = Codec.withAlternative(Codec.STRING.comapFlatMap(p_293693_ -> {
         try {
             return DataResult.success(UndashedUuid.fromStringLenient(p_293693_), Lifecycle.stable());
         } catch (IllegalArgumentException illegalargumentexception) {
             return DataResult.error(() -> "Invalid UUID " + p_293693_ + ": " + illegalargumentexception.getMessage());
         }
-    }, UndashedUuid::toString)).xmap(p_304145_ -> p_304145_.map(p_253361_ -> p_253361_, p_304144_ -> p_304144_), Either::right);
-    public static Codec<UUID> LENIENT_CODEC = Codec.either(CODEC, STRING_CODEC)
-        .xmap(p_253364_ -> p_253364_.map(p_304143_ -> p_304143_, p_253362_ -> p_253362_), Either::left);
+    }, UndashedUuid::toString), CODEC);
+    public static Codec<UUID> LENIENT_CODEC = Codec.withAlternative(CODEC, STRING_CODEC);
     public static StreamCodec<ByteBuf, UUID> STREAM_CODEC = new StreamCodec<ByteBuf, UUID>() {
         public UUID decode(ByteBuf p_320929_) {
             return FriendlyByteBuf.readUUID(p_320929_);
