--- a/net/minecraft/commands/arguments/coordinates/SwizzleArgument.java
+++ b/net/minecraft/commands/arguments/coordinates/SwizzleArgument.java
@@ -27,10 +_,10 @@
     public EnumSet<Direction.Axis> parse(StringReader p_120809_) throws CommandSyntaxException {
         EnumSet<Direction.Axis> enumset = EnumSet.noneOf(Direction.Axis.class);
 
-        while(p_120809_.canRead() && p_120809_.peek() != ' ') {
+        while (p_120809_.canRead() && p_120809_.peek() != ' ') {
             char c0 = p_120809_.read();
 
-            Direction.Axis $$6 = switch(c0) {
+            Direction.Axis $$6 = switch (c0) {
                 case 'x' -> Direction.Axis.X;
                 case 'y' -> Direction.Axis.Y;
                 case 'z' -> Direction.Axis.Z;
