--- a/net/minecraft/commands/arguments/coordinates/WorldCoordinate.java
+++ b/net/minecraft/commands/arguments/coordinates/WorldCoordinate.java
@@ -77,15 +_,10 @@
     public boolean equals(Object p_120877_) {
         if (this == p_120877_) {
             return true;
-        } else if (!(p_120877_ instanceof WorldCoordinate)) {
+        } else if (!(p_120877_ instanceof WorldCoordinate worldcoordinate)) {
             return false;
         } else {
-            WorldCoordinate worldcoordinate = (WorldCoordinate)p_120877_;
-            if (this.relative != worldcoordinate.relative) {
-                return false;
-            } else {
-                return Double.compare(worldcoordinate.value, this.value) == 0;
-            }
+            return this.relative != worldcoordinate.relative ? false : Double.compare(worldcoordinate.value, this.value) == 0;
         }
     }
 
