--- a/net/minecraft/world/item/alchemy/PotionContents.java
+++ b/net/minecraft/world/item/alchemy/PotionContents.java
@@ -8,6 +_,7 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
 import java.util.Optional;
+import java.util.OptionalInt;
 import java.util.function.Consumer;
 import net.minecraft.ChatFormatting;
 import net.minecraft.Util;
@@ -21,7 +_,6 @@
 import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.FastColor;
 import net.minecraft.world.effect.MobEffect;
 import net.minecraft.world.effect.MobEffectInstance;
@@ -35,18 +_,17 @@
 public record PotionContents(Optional<Holder<Potion>> potion, Optional<Integer> customColor, List<MobEffectInstance> customEffects) {
     public static final PotionContents EMPTY = new PotionContents(Optional.empty(), Optional.empty(), List.of());
     private static final Component NO_EFFECT = Component.translatable("effect.none").withStyle(ChatFormatting.GRAY);
-    private static final int EMPTY_COLOR = 16253176;
-    private static final int BASE_POTION_COLOR = 3694022;
-    public static final int NO_POTION_COLOR = -1;
+    private static final int EMPTY_COLOR = -524040;
+    private static final int BASE_POTION_COLOR = -13083194;
     private static final Codec<PotionContents> FULL_CODEC = RecordCodecBuilder.create(
-        p_332008_ -> p_332008_.group(
-                    ExtraCodecs.strictOptionalField(BuiltInRegistries.POTION.holderByNameCodec(), "potion").forGetter(PotionContents::potion),
-                    ExtraCodecs.strictOptionalField(Codec.INT, "custom_color").forGetter(PotionContents::customColor),
-                    ExtraCodecs.strictOptionalField(MobEffectInstance.CODEC.listOf(), "custom_effects", List.of()).forGetter(PotionContents::customEffects)
+        p_337942_ -> p_337942_.group(
+                    BuiltInRegistries.POTION.holderByNameCodec().optionalFieldOf("potion").forGetter(PotionContents::potion),
+                    Codec.INT.optionalFieldOf("custom_color").forGetter(PotionContents::customColor),
+                    MobEffectInstance.CODEC.listOf().optionalFieldOf("custom_effects", List.of()).forGetter(PotionContents::customEffects)
                 )
-                .apply(p_332008_, PotionContents::new)
+                .apply(p_337942_, PotionContents::new)
     );
-    public static final Codec<PotionContents> CODEC = ExtraCodecs.withAlternative(FULL_CODEC, BuiltInRegistries.POTION.holderByNameCodec(), PotionContents::new);
+    public static final Codec<PotionContents> CODEC = Codec.withAlternative(FULL_CODEC, BuiltInRegistries.POTION.holderByNameCodec(), PotionContents::new);
     public static final StreamCodec<RegistryFriendlyByteBuf, PotionContents> STREAM_CODEC = StreamCodec.composite(
         ByteBufCodecs.holderRegistry(Registries.POTION).apply(ByteBufCodecs::optional),
         PotionContents::potion,
@@ -83,12 +_,12 @@
 
     public void forEachEffect(Consumer<MobEffectInstance> p_331190_) {
         if (this.potion.isPresent()) {
-            for(MobEffectInstance mobeffectinstance : this.potion.get().value().getEffects()) {
+            for (MobEffectInstance mobeffectinstance : this.potion.get().value().getEffects()) {
                 p_331190_.accept(new MobEffectInstance(mobeffectinstance));
             }
         }
 
-        for(MobEffectInstance mobeffectinstance1 : this.customEffects) {
+        for (MobEffectInstance mobeffectinstance1 : this.customEffects) {
             p_331190_.accept(new MobEffectInstance(mobeffectinstance1));
         }
     }
@@ -105,7 +_,7 @@
         if (this.customColor.isPresent()) {
             return this.customColor.get();
         } else {
-            return this.potion.isEmpty() ? 16253176 : getColor(this.getAllEffects());
+            return this.potion.isEmpty() ? -524040 : getColor(this.getAllEffects());
         }
     }
 
@@ -118,17 +_,16 @@
     }
 
     public static int getColor(Iterable<MobEffectInstance> p_331934_) {
-        int i = getColorOptional(p_331934_);
-        return i == -1 ? 3694022 : i;
+        return getColorOptional(p_331934_).orElse(-13083194);
     }
 
-    public static int getColorOptional(Iterable<MobEffectInstance> p_332699_) {
+    public static OptionalInt getColorOptional(Iterable<MobEffectInstance> p_332699_) {
         int i = 0;
         int j = 0;
         int k = 0;
         int l = 0;
 
-        for(MobEffectInstance mobeffectinstance : p_332699_) {
+        for (MobEffectInstance mobeffectinstance : p_332699_) {
             if (mobeffectinstance.isVisible()) {
                 int i1 = mobeffectinstance.getEffect().value().getColor();
                 int j1 = mobeffectinstance.getAmplifier() + 1;
@@ -139,15 +_,11 @@
             }
         }
 
-        return l == 0 ? -1 : FastColor.ARGB32.color(0, i / l, j / l, k / l);
+        return l == 0 ? OptionalInt.empty() : OptionalInt.of(FastColor.ARGB32.color(i / l, j / l, k / l));
     }
 
     public boolean hasEffects() {
-        if (!this.customEffects.isEmpty()) {
-            return true;
-        } else {
-            return this.potion.isPresent() && !this.potion.get().value().getEffects().isEmpty();
-        }
+        return !this.customEffects.isEmpty() ? true : this.potion.isPresent() && !this.potion.get().value().getEffects().isEmpty();
     }
 
     public List<MobEffectInstance> customEffects() {
@@ -162,7 +_,7 @@
         List<Pair<Holder<Attribute>, AttributeModifier>> list = Lists.newArrayList();
         boolean flag = true;
 
-        for(MobEffectInstance mobeffectinstance : p_331825_) {
+        for (MobEffectInstance mobeffectinstance : p_331825_) {
             flag = false;
             MutableComponent mutablecomponent = Component.translatable(mobeffectinstance.getDescriptionId());
             Holder<MobEffect> holder = mobeffectinstance.getEffect();
@@ -190,7 +_,7 @@
             p_331296_.accept(CommonComponents.EMPTY);
             p_331296_.accept(Component.translatable("potion.whenDrank").withStyle(ChatFormatting.DARK_PURPLE));
 
-            for(Pair<Holder<Attribute>, AttributeModifier> pair : list) {
+            for (Pair<Holder<Attribute>, AttributeModifier> pair : list) {
                 AttributeModifier attributemodifier = pair.getSecond();
                 double d1 = attributemodifier.amount();
                 double d0;
