--- a/net/minecraft/world/item/alchemy/Potions.java
+++ b/net/minecraft/world/item/alchemy/Potions.java
@@ -6,16 +_,13 @@
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.effect.MobEffectInstance;
 import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.flag.FeatureFlags;
 
 public class Potions {
     public static final Holder<Potion> WATER = register("water", new Potion());
     public static final Holder<Potion> MUNDANE = register("mundane", new Potion());
     public static final Holder<Potion> THICK = register("thick", new Potion());
     public static final Holder<Potion> AWKWARD = register("awkward", new Potion());
-    public static final Holder<Potion> POTATO_OIL = register("potato_oil", new Potion(new MobEffectInstance(MobEffects.POTATO_OIL, 1)));
-    public static final Holder<Potion> POISONOUS_POTATO_OIL = register(
-        "poisonous_potato_oil", new Potion(new MobEffectInstance(MobEffects.POTATO_OIL, 1), new MobEffectInstance(MobEffects.POISON, 200))
-    );
     public static final Holder<Potion> NIGHT_VISION = register("night_vision", new Potion(new MobEffectInstance(MobEffects.NIGHT_VISION, 3600)));
     public static final Holder<Potion> LONG_NIGHT_VISION = register(
         "long_night_vision", new Potion("night_vision", new MobEffectInstance(MobEffects.NIGHT_VISION, 9600))
@@ -87,7 +_,18 @@
     public static final Holder<Potion> LONG_SLOW_FALLING = register(
         "long_slow_falling", new Potion("slow_falling", new MobEffectInstance(MobEffects.SLOW_FALLING, 4800))
     );
-    public static final Holder<Potion> STICKY = register("sticky", new Potion("sticky", new MobEffectInstance(MobEffects.STICKY, 1800)));
+    public static final Holder<Potion> WIND_CHARGED = register(
+        "wind_charged", new Potion("wind_charged", new MobEffectInstance(MobEffects.WIND_CHARGED, 3600)).requiredFeatures(FeatureFlags.UPDATE_1_21)
+    );
+    public static final Holder<Potion> WEAVING = register(
+        "weaving", new Potion("weaving", new MobEffectInstance(MobEffects.WEAVING, 3600)).requiredFeatures(FeatureFlags.UPDATE_1_21)
+    );
+    public static final Holder<Potion> OOZING = register(
+        "oozing", new Potion("oozing", new MobEffectInstance(MobEffects.OOZING, 3600)).requiredFeatures(FeatureFlags.UPDATE_1_21)
+    );
+    public static final Holder<Potion> INFESTED = register(
+        "infested", new Potion("infested", new MobEffectInstance(MobEffects.INFESTED, 3600)).requiredFeatures(FeatureFlags.UPDATE_1_21)
+    );
 
     private static Holder<Potion> register(String p_43626_, Potion p_43627_) {
         return Registry.registerForHolder(BuiltInRegistries.POTION, new ResourceLocation(p_43626_), p_43627_);
