--- a/net/minecraft/world/item/alchemy/Potion.java
+++ b/net/minecraft/world/item/alchemy/Potion.java
@@ -6,11 +_,16 @@
 import net.minecraft.core.Holder;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.flag.FeatureElement;
+import net.minecraft.world.flag.FeatureFlag;
+import net.minecraft.world.flag.FeatureFlagSet;
+import net.minecraft.world.flag.FeatureFlags;
 
-public class Potion {
+public class Potion implements FeatureElement {
     @Nullable
     private final String name;
     private final List<MobEffectInstance> effects;
+    private FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET;
 
     public Potion(MobEffectInstance... p_43487_) {
         this(null, p_43487_);
@@ -21,6 +_,16 @@
         this.effects = List.of(p_43485_);
     }
 
+    public Potion requiredFeatures(FeatureFlag... p_338520_) {
+        this.requiredFeatures = FeatureFlags.REGISTRY.subset(p_338520_);
+        return this;
+    }
+
+    @Override
+    public FeatureFlagSet requiredFeatures() {
+        return this.requiredFeatures;
+    }
+
     public static String getName(Optional<Holder<Potion>> p_330503_, String p_43493_) {
         if (p_330503_.isPresent()) {
             String s = p_330503_.get().value().name;
@@ -39,7 +_,7 @@
 
     public boolean hasInstantEffects() {
         if (!this.effects.isEmpty()) {
-            for(MobEffectInstance mobeffectinstance : this.effects) {
+            for (MobEffectInstance mobeffectinstance : this.effects) {
                 if (mobeffectinstance.getEffect().value().isInstantenous()) {
                     return true;
                 }
