--- a/net/minecraft/world/entity/monster/piglin/Piglin.java
+++ b/net/minecraft/world/entity/monster/piglin/Piglin.java
@@ -320,7 +_,7 @@
         } else if (this.isChargingCrossbow()) {
             return PiglinArmPose.CROSSBOW_CHARGE;
         } else {
-            return this.isHolding(Items.CROSSBOW) && CrossbowItem.isCharged(this.getWeaponItem()) ? PiglinArmPose.CROSSBOW_HOLD : PiglinArmPose.DEFAULT;
+            return this.isHolding(is -> is.getItem() instanceof net.minecraft.world.item.CrossbowItem) && CrossbowItem.isCharged(this.getWeaponItem()) ? PiglinArmPose.CROSSBOW_HOLD : PiglinArmPose.DEFAULT;
         }
     }
 
@@ -335,9 +_,8 @@
     @Override
     public boolean hurtServer(ServerLevel level, DamageSource source, float damage) {
         boolean wasHurt = super.hurtServer(level, source, damage);
-        Entity attacker = source.getEntity();
-        if (wasHurt && (attacker instanceof LivingEntity || attacker instanceof LivingBlock)) {
-            PiglinAi.wasHurtBy(level, this, attacker);
+        if (wasHurt && source.getEntity() instanceof LivingEntity livingEntity) {
+            PiglinAi.wasHurtBy(level, this, livingEntity);
         }
 
         return wasHurt;
@@ -348,9 +_,13 @@
         this.performCrossbowAttack(this, 1.6F);
     }
 
+    public void performRangedAttack(LivingEntity target, float power) {
+        this.performRangedAttack((net.minecraft.world.entity.Entity) target, power);
+    }
+
     @Override
     public boolean canUseNonMeleeWeapon(ItemStack item) {
-        return item.getItem() == Items.CROSSBOW || item.has(DataComponents.KINETIC_WEAPON);
+        return item.is(net.neoforged.neoforge.common.Tags.Items.PIGLIN_USABLE_CROSSBOWS) || item.has(DataComponents.KINETIC_WEAPON);
     }
 
     protected void holdInMainHand(ItemStack itemStack) {
@@ -358,7 +_,7 @@
     }
 
     protected void holdInOffHand(ItemStack itemStack) {
-        if (itemStack.is(PiglinAi.BARTERING_ITEM)) {
+        if (itemStack.isPiglinCurrency()) {
             this.setItemSlot(EquipmentSlot.OFFHAND, itemStack);
             this.setGuaranteedDrop(EquipmentSlot.OFFHAND);
         } else {
@@ -368,7 +_,7 @@
 
     @Override
     public boolean wantsToPickUp(ServerLevel level, ItemStack itemStack) {
-        return level.getGameRules().get(GameRules.MOB_GRIEFING) && this.canPickUpLoot() && PiglinAi.wantsToPickup(this, itemStack);
+        return net.neoforged.neoforge.event.EventHooks.canEntityGrief(level, this) && this.canPickUpLoot() && PiglinAi.wantsToPickup(this, itemStack);
     }
 
     protected boolean canReplaceCurrentItem(ItemStack newItemStack) {
