--- a/net/minecraft/world/entity/monster/piglin/PiglinAi.java
+++ b/net/minecraft/world/entity/monster/piglin/PiglinAi.java
@@ -385,7 +_,7 @@
         ItemStack itemStack = body.getItemInHand(InteractionHand.OFF_HAND);
         body.setItemInHand(InteractionHand.OFF_HAND, ItemStack.EMPTY);
         if (body.isAdult()) {
-            boolean barterCurrency = isBarterCurrency(itemStack);
+            boolean barterCurrency = itemStack.isPiglinCurrency();
             if (barteringEnabled && barterCurrency) {
                 throwItems(body, getBarterResponseItems(body));
             } else if (!barterCurrency) {
@@ -467,7 +_,7 @@
             return false;
         } else if (isAdmiringDisabled(body) && body.getBrain().hasMemoryValue(MemoryModuleType.ATTACK_TARGET)) {
             return false;
-        } else if (isBarterCurrency(itemStack)) {
+        } else if (itemStack.isPiglinCurrency()) {
             return isNotHoldingLovedItemInOffHand(body);
         } else {
             boolean hasSpace = body.canAddToInventory(itemStack);
@@ -514,7 +_,7 @@
         if (isNearZombified(body)) {
             return Optional.empty();
         } else {
-            Optional<Entity> angryAt = BehaviorUtils.getLivingEntityFromUUIDMemory(body, MemoryModuleType.ANGRY_AT);
+            Optional<LivingEntity> angryAt = BehaviorUtils.getLivingEntityFromUUIDMemory(body, MemoryModuleType.ANGRY_AT);
             if (angryAt.isPresent() && Sensor.isEntityAttackableIgnoringLineOfSight(level, body, angryAt.get())) {
                 return angryAt;
             } else {
@@ -563,10 +_,10 @@
     }
 
     protected static boolean canAdmire(Piglin body, ItemStack playerHeldItemStack) {
-        return !isAdmiringDisabled(body) && !isAdmiringItem(body) && body.isAdult() && isBarterCurrency(playerHeldItemStack);
+        return !isAdmiringDisabled(body) && !isAdmiringItem(body) && body.isAdult() && playerHeldItemStack.isPiglinCurrency();
     }
 
-    protected static void wasHurtBy(ServerLevel level, Piglin body, Entity attacker) {
+    protected static void wasHurtBy(ServerLevel level, Piglin body, LivingEntity attacker) {
         if (!(attacker instanceof Piglin)) {
             if (isHoldingItemInOffHand(body)) {
                 stopHoldingOffHandItem(level, body, false);
@@ -599,7 +_,7 @@
         }
     }
 
-    protected static void maybeRetaliate(ServerLevel level, AbstractPiglin body, Entity attacker) {
+    protected static void maybeRetaliate(ServerLevel level, AbstractPiglin body, LivingEntity attacker) {
         if (!body.getBrain().isActive(Activity.AVOID)) {
             if (Sensor.isEntityAttackableIgnoringLineOfSight(level, body, attacker)) {
                 if (!BehaviorUtils.isOtherTargetMuchFurtherAwayThanCurrentAttackTarget(body, attacker, 4.0)) {
@@ -652,7 +_,7 @@
 
     public static boolean isWearingSafeArmor(LivingEntity livingEntity) {
         for (EquipmentSlot slot : EquipmentSlotGroup.ARMOR) {
-            if (livingEntity.getItemBySlot(slot).is(ItemTags.PIGLIN_SAFE_ARMOR)) {
+            if (livingEntity.getItemBySlot(slot).makesPiglinsNeutral(livingEntity)) {
                 return true;
             }
         }
@@ -675,7 +_,7 @@
         );
     }
 
-    protected static void broadcastAngerTarget(ServerLevel level, AbstractPiglin body, Entity target) {
+    protected static void broadcastAngerTarget(ServerLevel level, AbstractPiglin body, LivingEntity target) {
         getAdultPiglins(body).forEach(piglin -> {
             if (!(target instanceof Hoglin hoglin && (!piglin.canHunt() || !hoglin.canBeHunted()))) {
                 setAngerTargetIfCloserThanCurrent(level, piglin, target);
@@ -687,7 +_,7 @@
         getAdultPiglins(body).forEach(piglin -> getNearestVisibleTargetablePlayer(piglin).ifPresent(player -> setAngerTarget(level, piglin, player)));
     }
 
-    protected static void setAngerTarget(ServerLevel level, AbstractPiglin body, Entity target) {
+    protected static void setAngerTarget(ServerLevel level, AbstractPiglin body, LivingEntity target) {
         if (Sensor.isEntityAttackableIgnoringLineOfSight(level, body, target)) {
             body.getBrain().eraseMemory(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE);
             body.getBrain().setMemoryWithExpiry(MemoryModuleType.ANGRY_AT, target.getUUID(), 600L);
@@ -701,7 +_,7 @@
         }
     }
 
-    private static void setAngerTargetToNearestTargetablePlayerIfFound(ServerLevel level, AbstractPiglin body, Entity targetIfNoPlayerFound) {
+    private static void setAngerTargetToNearestTargetablePlayerIfFound(ServerLevel level, AbstractPiglin body, LivingEntity targetIfNoPlayerFound) {
         Optional<Player> nearestPlayer = getNearestVisibleTargetablePlayer(body);
         if (nearestPlayer.isPresent()) {
             setAngerTarget(level, body, nearestPlayer.get());
@@ -710,20 +_,20 @@
         }
     }
 
-    private static void setAngerTargetIfCloserThanCurrent(ServerLevel level, AbstractPiglin body, Entity newTarget) {
-        Optional<Entity> currentTarget = getAngerTarget(body);
-        Entity nearest = BehaviorUtils.getNearestTarget(body, currentTarget, newTarget);
+    private static void setAngerTargetIfCloserThanCurrent(ServerLevel level, AbstractPiglin body, LivingEntity newTarget) {
+        Optional<LivingEntity> currentTarget = getAngerTarget(body);
+        LivingEntity nearest = BehaviorUtils.getNearestTarget(body, currentTarget, newTarget);
         if (!currentTarget.isPresent() || currentTarget.get() != nearest) {
             setAngerTarget(level, body, nearest);
         }
     }
 
-    private static Optional<Entity> getAngerTarget(AbstractPiglin body) {
+    private static Optional<LivingEntity> getAngerTarget(AbstractPiglin body) {
         return BehaviorUtils.getLivingEntityFromUUIDMemory(body, MemoryModuleType.ANGRY_AT);
     }
 
-    public static Optional<Entity> getAvoidTarget(Piglin body) {
-        return body.getBrain().hasMemoryValue(MemoryModuleType.AVOID_TARGET) ? body.getBrain().getMemory(MemoryModuleType.AVOID_TARGET) : Optional.empty();
+    public static Optional<LivingEntity> getAvoidTarget(Piglin body) {
+        return body.getBrain().getMemory(MemoryModuleType.AVOID_TARGET).filter(LivingEntity.class::isInstance).map(LivingEntity.class::cast);
     }
 
     public static Optional<Player> getNearestVisibleTargetablePlayer(AbstractPiglin body) {
@@ -732,7 +_,7 @@
             : Optional.empty();
     }
 
-    private static void broadcastRetreat(Piglin body, Entity target) {
+    private static void broadcastRetreat(Piglin body, LivingEntity target) {
         getVisibleAdultPiglins(body).forEach(abstractPiglin -> {
             if (abstractPiglin instanceof Piglin piglin) {
                 retreatFromNearestTarget(piglin, target);
@@ -740,9 +_,10 @@
         });
     }
 
-    private static void retreatFromNearestTarget(Piglin body, Entity newAvoidTarget) {
+    private static void retreatFromNearestTarget(Piglin body, LivingEntity newAvoidTarget) {
         Brain<Piglin> brain = body.getBrain();
-        Entity nearest = BehaviorUtils.getNearestTarget(body, brain.getMemory(MemoryModuleType.AVOID_TARGET), newAvoidTarget);
+        Optional<LivingEntity> avoidTarget = brain.getMemory(MemoryModuleType.AVOID_TARGET).filter(LivingEntity.class::isInstance).map(LivingEntity.class::cast);
+        LivingEntity nearest = BehaviorUtils.getNearestTarget(body, avoidTarget, newAvoidTarget);
         nearest = BehaviorUtils.getNearestTarget(body, brain.getMemory(MemoryModuleType.ATTACK_TARGET), nearest);
         setAvoidTargetAndDontHuntForAWhile(body, nearest);
     }
@@ -752,7 +_,7 @@
         if (!brain.hasMemoryValue(MemoryModuleType.AVOID_TARGET)) {
             return true;
         } else {
-            Entity avoidedEntity = brain.getMemory(MemoryModuleType.AVOID_TARGET).get();
+            LivingEntity avoidedEntity = (LivingEntity)brain.getMemory(MemoryModuleType.AVOID_TARGET).get();
             if (avoidedEntity.is(EntityType.HOGLIN)) {
                 return piglinsEqualOrOutnumberHoglins(body);
             } else {
@@ -771,7 +_,7 @@
         return hoglinCount > piglinCount;
     }
 
-    private static void setAvoidTargetAndDontHuntForAWhile(Piglin body, Entity target) {
+    private static void setAvoidTargetAndDontHuntForAWhile(Piglin body, LivingEntity target) {
         body.getBrain().eraseMemory(MemoryModuleType.ANGRY_AT);
         body.getBrain().eraseMemory(MemoryModuleType.ATTACK_TARGET);
         body.getBrain().eraseMemory(MemoryModuleType.WALK_TARGET);
@@ -801,7 +_,7 @@
     }
 
     private static boolean hasCrossbow(LivingEntity body) {
-        return body.isHolding(Items.CROSSBOW);
+        return body.isHolding(is -> is.getItem() instanceof net.minecraft.world.item.CrossbowItem);
     }
 
     private static void admireGoldItem(LivingEntity body) {
