--- a/net/minecraft/client/renderer/debug/BeeDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/BeeDebugRenderer.java
@@ -113,7 +_,7 @@
         });
         this.renderFlowerInfos(p_270886_, p_270808_);
 
-        for(BlockPos blockpos1 : this.hives.keySet()) {
+        for (BlockPos blockpos1 : this.hives.keySet()) {
             if (blockpos.closerThan(blockpos1, 30.0)) {
                 highlightHive(p_270886_, p_270808_, blockpos1);
             }
@@ -128,7 +_,7 @@
         });
         this.getGhostHives().forEach((p_269699_, p_269700_) -> {
             if (blockpos.closerThan(p_269699_, 30.0)) {
-                this.renderGhostHive(p_270886_, p_270808_, p_269699_, p_269700_);
+                this.renderGhostHive(p_270886_, p_270808_, p_269699_, (List<String>)p_269700_);
             }
         });
     }
@@ -148,16 +_,16 @@
         Map<BlockPos, Set<UUID>> map = Maps.newHashMap();
         this.beeInfosPerEntity.values().forEach(p_293651_ -> {
             if (p_293651_.flowerPos() != null) {
-                map.computeIfAbsent(p_293651_.flowerPos(), p_293649_ -> new HashSet()).add(p_293651_.uuid());
+                map.computeIfAbsent(p_293651_.flowerPos(), p_293649_ -> new HashSet<>()).add(p_293651_.uuid());
             }
         });
-        map.forEach((p_293631_, p_293632_) -> {
-            Set<String> set = p_293632_.stream().map(DebugEntityNameGenerator::getEntityName).collect(Collectors.toSet());
+        map.forEach((p_339302_, p_339303_) -> {
+            Set<String> set = p_339303_.stream().map(DebugEntityNameGenerator::getEntityName).collect(Collectors.toSet());
             int i = 1;
-            renderTextOverPos(p_270578_, p_270098_, set.toString(), p_293631_, i++, -256);
-            renderTextOverPos(p_270578_, p_270098_, "Flower", p_293631_, i++, -1);
+            renderTextOverPos(p_270578_, p_270098_, set.toString(), p_339302_, i++, -256);
+            renderTextOverPos(p_270578_, p_270098_, "Flower", p_339302_, i++, -1);
             float f = 0.05F;
-            DebugRenderer.renderFilledBox(p_270578_, p_270098_, p_293631_, 0.05F, 0.8F, 0.8F, 0.0F, 0.3F);
+            DebugRenderer.renderFilledBox(p_270578_, p_270098_, p_339302_, 0.05F, 0.8F, 0.8F, 0.0F, 0.3F);
         });
     }
 
@@ -234,7 +_,7 @@
             renderTextOverMob(p_270154_, p_270397_, p_294247_.pos(), i++, "Flower: " + this.getPosDescription(p_294247_, p_294247_.flowerPos()), -256, 0.02F);
         }
 
-        for(String s : p_294247_.goals()) {
+        for (String s : p_294247_.goals()) {
             renderTextOverMob(p_270154_, p_270397_, p_294247_.pos(), i++, s, -16711936, 0.02F);
         }
 
@@ -313,7 +_,7 @@
     private Map<BlockPos, List<String>> getGhostHives() {
         Map<BlockPos, List<String>> map = Maps.newHashMap();
 
-        for(BeeDebugPayload.BeeInfo beedebugpayload$beeinfo : this.beeInfosPerEntity.values()) {
+        for (BeeDebugPayload.BeeInfo beedebugpayload$beeinfo : this.beeInfosPerEntity.values()) {
             if (beedebugpayload$beeinfo.hivePos() != null && !this.hives.containsKey(beedebugpayload$beeinfo.hivePos())) {
                 map.computeIfAbsent(beedebugpayload$beeinfo.hivePos(), p_113140_ -> Lists.newArrayList()).add(beedebugpayload$beeinfo.generateName());
             }
