--- a/net/minecraft/client/renderer/debug/BreezeDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/BreezeDebugRenderer.java
@@ -42,14 +_,14 @@
         localplayer.level()
             .getEntities(EntityType.BREEZE, localplayer.getBoundingBox().inflate(100.0), p_312383_ -> true)
             .forEach(
-                p_339521_ -> {
-                    Optional<BreezeDebugPayload.BreezeInfo> optional = Optional.ofNullable(this.perEntity.get(p_339521_.getId()));
+                p_337428_ -> {
+                    Optional<BreezeDebugPayload.BreezeInfo> optional = Optional.ofNullable(this.perEntity.get(p_337428_.getId()));
                     optional.map(BreezeDebugPayload.BreezeInfo::attackTarget)
-                        .map(p_339514_ -> localplayer.level().getEntity(p_339514_))
+                        .map(p_337430_ -> localplayer.level().getEntity(p_337430_))
                         .map(p_312810_ -> p_312810_.getPosition(this.minecraft.getFrameTime()))
                         .ifPresent(
                             p_312926_ -> {
-                                drawLine(p_312334_, p_312422_, p_312916_, p_312212_, p_312705_, p_339521_.position(), p_312926_, TARGET_LINE_COLOR);
+                                drawLine(p_312334_, p_312422_, p_312916_, p_312212_, p_312705_, p_337428_.position(), p_312926_, TARGET_LINE_COLOR);
                                 Vec3 vec3 = p_312926_.add(0.0, 0.01F, 0.0);
                                 drawCircle(
                                     p_312334_.last().pose(),
@@ -85,14 +_,14 @@
                         );
                     optional.map(BreezeDebugPayload.BreezeInfo::jumpTarget)
                         .ifPresent(
-                            p_339512_ -> {
+                            p_337437_ -> {
                                 drawLine(
-                                    p_312334_, p_312422_, p_312916_, p_312212_, p_312705_, p_339521_.position(), p_339512_.getCenter(), JUMP_TARGET_LINE_COLOR
+                                    p_312334_, p_312422_, p_312916_, p_312212_, p_312705_, p_337428_.position(), p_337437_.getCenter(), JUMP_TARGET_LINE_COLOR
                                 );
                                 DebugRenderer.renderFilledBox(
                                     p_312334_,
                                     p_312422_,
-                                    AABB.unitCubeFromLowerCorner(Vec3.atLowerCornerOf(p_339512_)).move(-p_312916_, -p_312212_, -p_312705_),
+                                    AABB.unitCubeFromLowerCorner(Vec3.atLowerCornerOf(p_337437_)).move(-p_312916_, -p_312212_, -p_312705_),
                                     1.0F,
                                     0.0F,
                                     0.0F,
@@ -119,7 +_,7 @@
     private static void drawCircle(
         Matrix4f p_312289_, double p_312777_, double p_312720_, double p_312378_, VertexConsumer p_312840_, Vec3 p_312033_, float p_312193_, int p_312116_
     ) {
-        for(int i = 0; i < 20; ++i) {
+        for (int i = 0; i < 20; i++) {
             drawCircleVertex(i, p_312289_, p_312777_, p_312720_, p_312378_, p_312840_, p_312033_, p_312193_, p_312116_);
         }
 
