--- a/net/minecraft/client/renderer/debug/ChunkBorderRenderer.java
+++ b/net/minecraft/client/renderer/debug/ChunkBorderRenderer.java
@@ -33,8 +_,8 @@
         VertexConsumer vertexconsumer = p_113359_.getBuffer(RenderType.debugLineStrip(1.0));
         Matrix4f matrix4f = p_113358_.last().pose();
 
-        for(int i = -16; i <= 32; i += 16) {
-            for(int j = -16; j <= 32; j += 16) {
+        for (int i = -16; i <= 32; i += 16) {
+            for (int j = -16; j <= 32; j += 16) {
                 vertexconsumer.vertex(matrix4f, f2 + (float)i, f, f3 + (float)j).color(1.0F, 0.0F, 0.0F, 0.0F).endVertex();
                 vertexconsumer.vertex(matrix4f, f2 + (float)i, f, f3 + (float)j).color(1.0F, 0.0F, 0.0F, 0.5F).endVertex();
                 vertexconsumer.vertex(matrix4f, f2 + (float)i, f1, f3 + (float)j).color(1.0F, 0.0F, 0.0F, 0.5F).endVertex();
@@ -42,7 +_,7 @@
             }
         }
 
-        for(int l = 2; l < 16; l += 2) {
+        for (int l = 2; l < 16; l += 2) {
             int i2 = l % 4 == 0 ? CELL_BORDER : YELLOW;
             vertexconsumer.vertex(matrix4f, f2 + (float)l, f, f3).color(1.0F, 1.0F, 0.0F, 0.0F).endVertex();
             vertexconsumer.vertex(matrix4f, f2 + (float)l, f, f3).color(i2).endVertex();
@@ -54,7 +_,7 @@
             vertexconsumer.vertex(matrix4f, f2 + (float)l, f1, f3 + 16.0F).color(1.0F, 1.0F, 0.0F, 0.0F).endVertex();
         }
 
-        for(int i1 = 2; i1 < 16; i1 += 2) {
+        for (int i1 = 2; i1 < 16; i1 += 2) {
             int j2 = i1 % 4 == 0 ? CELL_BORDER : YELLOW;
             vertexconsumer.vertex(matrix4f, f2, f, f3 + (float)i1).color(1.0F, 1.0F, 0.0F, 0.0F).endVertex();
             vertexconsumer.vertex(matrix4f, f2, f, f3 + (float)i1).color(j2).endVertex();
@@ -66,7 +_,7 @@
             vertexconsumer.vertex(matrix4f, f2 + 16.0F, f1, f3 + (float)i1).color(1.0F, 1.0F, 0.0F, 0.0F).endVertex();
         }
 
-        for(int j1 = this.minecraft.level.getMinBuildHeight(); j1 <= this.minecraft.level.getMaxBuildHeight(); j1 += 2) {
+        for (int j1 = this.minecraft.level.getMinBuildHeight(); j1 <= this.minecraft.level.getMaxBuildHeight(); j1 += 2) {
             float f4 = (float)((double)j1 - p_113361_);
             int k = j1 % 8 == 0 ? CELL_BORDER : YELLOW;
             vertexconsumer.vertex(matrix4f, f2, f4, f3).color(1.0F, 1.0F, 0.0F, 0.0F).endVertex();
@@ -80,8 +_,8 @@
 
         vertexconsumer = p_113359_.getBuffer(RenderType.debugLineStrip(2.0));
 
-        for(int k1 = 0; k1 <= 16; k1 += 16) {
-            for(int k2 = 0; k2 <= 16; k2 += 16) {
+        for (int k1 = 0; k1 <= 16; k1 += 16) {
+            for (int k2 = 0; k2 <= 16; k2 += 16) {
                 vertexconsumer.vertex(matrix4f, f2 + (float)k1, f, f3 + (float)k2).color(0.25F, 0.25F, 1.0F, 0.0F).endVertex();
                 vertexconsumer.vertex(matrix4f, f2 + (float)k1, f, f3 + (float)k2).color(0.25F, 0.25F, 1.0F, 1.0F).endVertex();
                 vertexconsumer.vertex(matrix4f, f2 + (float)k1, f1, f3 + (float)k2).color(0.25F, 0.25F, 1.0F, 1.0F).endVertex();
@@ -89,7 +_,7 @@
             }
         }
 
-        for(int l1 = this.minecraft.level.getMinBuildHeight(); l1 <= this.minecraft.level.getMaxBuildHeight(); l1 += 16) {
+        for (int l1 = this.minecraft.level.getMinBuildHeight(); l1 <= this.minecraft.level.getMaxBuildHeight(); l1 += 16) {
             float f5 = (float)((double)l1 - p_113361_);
             vertexconsumer.vertex(matrix4f, f2, f5, f3).color(0.25F, 0.25F, 1.0F, 0.0F).endVertex();
             vertexconsumer.vertex(matrix4f, f2, f5, f3).color(0.25F, 0.25F, 1.0F, 1.0F).endVertex();
