--- a/net/minecraft/client/renderer/debug/LightSectionDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/LightSectionDebugRenderer.java
@@ -117,7 +_,7 @@
         float f8 = p_283316_.z();
         float f9 = p_283316_.w();
         Matrix4f matrix4f = p_283612_.last().pose();
-        switch(p_282340_) {
+        switch (p_282340_) {
             case DOWN:
                 p_281996_.vertex(matrix4f, f, f1, f2).color(f6, f7, f8, f9).endVertex();
                 p_281996_.vertex(matrix4f, f3, f1, f2).color(f6, f7, f8, f9).endVertex();
@@ -192,9 +_,9 @@
             this.lightAndBlocksShape = new BitSetDiscreteVoxelShape(i, i, i);
             this.lightShape = new BitSetDiscreteVoxelShape(i, i, i);
 
-            for(int j = 0; j < i; ++j) {
-                for(int k = 0; k < i; ++k) {
-                    for(int l = 0; l < i; ++l) {
+            for (int j = 0; j < i; j++) {
+                for (int k = 0; k < i; k++) {
+                    for (int l = 0; l < i; l++) {
                         SectionPos sectionpos = SectionPos.of(p_282370_.x() + l - p_282804_, p_282370_.y() + k - p_282804_, p_282370_.z() + j - p_282804_);
                         LayerLightSectionStorage.SectionType layerlightsectionstorage$sectiontype = p_283220_.getDebugSectionType(p_283151_, sectionpos);
                         if (layerlightsectionstorage$sectiontype == LayerLightSectionStorage.SectionType.LIGHT_AND_DATA) {
