--- a/net/minecraft/client/renderer/debug/NeighborsUpdateRenderer.java
+++ b/net/minecraft/client/renderer/debug/NeighborsUpdateRenderer.java
@@ -43,7 +_,7 @@
         VertexConsumer vertexconsumer = p_113601_.getBuffer(RenderType.lines());
         Iterator<Entry<Long, Map<BlockPos, Integer>>> iterator = this.lastUpdate.entrySet().iterator();
 
-        while(iterator.hasNext()) {
+        while (iterator.hasNext()) {
             Entry<Long, Map<BlockPos, Integer>> entry = iterator.next();
             Long olong = entry.getKey();
             Map<BlockPos, Integer> map1 = entry.getValue();
@@ -51,7 +_,7 @@
             if (k > 200L) {
                 iterator.remove();
             } else {
-                for(Entry<BlockPos, Integer> entry1 : map1.entrySet()) {
+                for (Entry<BlockPos, Integer> entry1 : map1.entrySet()) {
                     BlockPos blockpos = entry1.getKey();
                     Integer integer = entry1.getValue();
                     if (set.add(blockpos)) {
@@ -69,7 +_,7 @@
             }
         }
 
-        for(Entry<BlockPos, Integer> entry2 : map.entrySet()) {
+        for (Entry<BlockPos, Integer> entry2 : map.entrySet()) {
             BlockPos blockpos1 = entry2.getKey();
             Integer integer1 = entry2.getValue();
             DebugRenderer.renderFloatingText(p_113600_, p_113601_, String.valueOf(integer1), blockpos1.getX(), blockpos1.getY(), blockpos1.getZ(), -1);
