--- a/net/minecraft/client/renderer/debug/PathfindingRenderer.java
+++ b/net/minecraft/client/renderer/debug/PathfindingRenderer.java
@@ -41,13 +_,13 @@
         if (!this.pathMap.isEmpty()) {
             long i = Util.getMillis();
 
-            for(Integer integer : this.pathMap.keySet()) {
+            for (Integer integer : this.pathMap.keySet()) {
                 Path path = this.pathMap.get(integer);
                 float f = this.pathMaxDist.get(integer);
                 renderPath(p_113629_, p_113630_, path, f, true, true, p_113631_, p_113632_, p_113633_);
             }
 
-            for(Integer integer1 : this.creationMap.keySet().toArray(new Integer[0])) {
+            for (Integer integer1 : this.creationMap.keySet().toArray(new Integer[0])) {
                 if (i - this.creationMap.get(integer1) > 5000L) {
                     this.pathMap.remove(integer1);
                     this.creationMap.remove(integer1);
@@ -88,7 +_,7 @@
                 0.5F
             );
 
-            for(int i = 0; i < p_270189_.getNodeCount(); ++i) {
+            for (int i = 0; i < p_270189_.getNodeCount(); i++) {
                 Node node = p_270189_.getNode(i);
                 if (distanceToCamera(node.asBlockPos(), p_270187_, p_270252_, p_270371_) <= 80.0F) {
                     float f = i == p_270189_.getNextNodeIndex() ? 1.0F : 0.0F;
@@ -116,7 +_,7 @@
 
         Path.DebugData path$debugdata = p_270189_.debugData();
         if (p_270481_ && path$debugdata != null) {
-            for(Node node1 : path$debugdata.closedSet()) {
+            for (Node node1 : path$debugdata.closedSet()) {
                 if (distanceToCamera(node1.asBlockPos(), p_270187_, p_270252_, p_270371_) <= 80.0F) {
                     DebugRenderer.renderFilledBox(
                         p_270399_,
@@ -138,7 +_,7 @@
                 }
             }
 
-            for(Node node3 : path$debugdata.openSet()) {
+            for (Node node3 : path$debugdata.openSet()) {
                 if (distanceToCamera(node3.asBlockPos(), p_270187_, p_270252_, p_270371_) <= 80.0F) {
                     DebugRenderer.renderFilledBox(
                         p_270399_,
@@ -162,7 +_,7 @@
         }
 
         if (p_270748_) {
-            for(int j = 0; j < p_270189_.getNodeCount(); ++j) {
+            for (int j = 0; j < p_270189_.getNodeCount(); j++) {
                 Node node2 = p_270189_.getNode(j);
                 if (distanceToCamera(node2.asBlockPos(), p_270187_, p_270252_, p_270371_) <= 80.0F) {
                     DebugRenderer.renderFloatingText(
@@ -197,7 +_,7 @@
     }
 
     public static void renderPathLine(PoseStack p_270666_, VertexConsumer p_270602_, Path p_270511_, double p_270524_, double p_270163_, double p_270176_) {
-        for(int i = 0; i < p_270511_.getNodeCount(); ++i) {
+        for (int i = 0; i < p_270511_.getNodeCount(); i++) {
             Node node = p_270511_.getNode(i);
             if (!(distanceToCamera(node.asBlockPos(), p_270524_, p_270163_, p_270176_) > 80.0F)) {
                 float f = (float)i / (float)p_270511_.getNodeCount() * 0.33F;
