--- a/net/minecraft/client/renderer/debug/SupportBlockRenderer.java
+++ b/net/minecraft/client/renderer/debug/SupportBlockRenderer.java
@@ -41,7 +_,7 @@
             this.drawHighlights(p_286297_, p_286436_, p_286291_, p_286388_, p_286330_, player, () -> 0.0, 1.0F, 0.0F, 0.0F);
         }
 
-        for(Entity entity1 : this.surroundEntities) {
+        for (Entity entity1 : this.surroundEntities) {
             if (entity1 != player) {
                 this.drawHighlights(p_286297_, p_286436_, p_286291_, p_286388_, p_286330_, entity1, () -> this.getBias(entity1), 0.0F, 1.0F, 0.0F);
             }
