--- a/net/minecraft/client/renderer/debug/WaterDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/WaterDebugRenderer.java
@@ -24,7 +_,7 @@
         BlockPos blockpos = this.minecraft.player.blockPosition();
         LevelReader levelreader = this.minecraft.player.level();
 
-        for(BlockPos blockpos1 : BlockPos.betweenClosed(blockpos.offset(-10, -10, -10), blockpos.offset(10, 10, 10))) {
+        for (BlockPos blockpos1 : BlockPos.betweenClosed(blockpos.offset(-10, -10, -10), blockpos.offset(10, 10, 10))) {
             FluidState fluidstate = levelreader.getFluidState(blockpos1);
             if (fluidstate.is(FluidTags.WATER)) {
                 double d0 = (double)((float)blockpos1.getY() + fluidstate.getHeight(levelreader, blockpos1));
@@ -48,7 +_,7 @@
             }
         }
 
-        for(BlockPos blockpos2 : BlockPos.betweenClosed(blockpos.offset(-10, -10, -10), blockpos.offset(10, 10, 10))) {
+        for (BlockPos blockpos2 : BlockPos.betweenClosed(blockpos.offset(-10, -10, -10), blockpos.offset(10, 10, 10))) {
             FluidState fluidstate1 = levelreader.getFluidState(blockpos2);
             if (fluidstate1.is(FluidTags.WATER)) {
                 DebugRenderer.renderFloatingText(
