--- a/net/minecraft/world/level/dimension/end/EndDragonFight.java
+++ b/net/minecraft/world/level/dimension/end/EndDragonFight.java
@@ -245,7 +_,7 @@
                 this.dragonKilled = false;
                 EnderDragon enderdragon = this.createNewDragon();
                 if (enderdragon != null) {
-                    for(ServerPlayer serverplayer : this.dragonEvent.getPlayers()) {
+                    for (ServerPlayer serverplayer : this.dragonEvent.getPlayers()) {
                         CriteriaTriggers.SUMMONED_ENTITY.trigger(serverplayer, enderdragon);
                     }
                 }
@@ -256,11 +_,11 @@
     }
 
     private boolean hasActiveExitPortal() {
-        for(int i = -8; i <= 8; ++i) {
-            for(int j = -8; j <= 8; ++j) {
+        for (int i = -8; i <= 8; i++) {
+            for (int j = -8; j <= 8; j++) {
                 LevelChunk levelchunk = this.level.getChunk(i, j);
 
-                for(BlockEntity blockentity : levelchunk.getBlockEntities().values()) {
+                for (BlockEntity blockentity : levelchunk.getBlockEntities().values()) {
                     if (blockentity instanceof TheEndPortalBlockEntity) {
                         return true;
                     }
@@ -275,11 +_,11 @@
     private BlockPattern.BlockPatternMatch findExitPortal() {
         ChunkPos chunkpos = new ChunkPos(this.origin);
 
-        for(int i = -8 + chunkpos.x; i <= 8 + chunkpos.x; ++i) {
-            for(int j = -8 + chunkpos.z; j <= 8 + chunkpos.z; ++j) {
+        for (int i = -8 + chunkpos.x; i <= 8 + chunkpos.x; i++) {
+            for (int j = -8 + chunkpos.z; j <= 8 + chunkpos.z; j++) {
                 LevelChunk levelchunk = this.level.getChunk(i, j);
 
-                for(BlockEntity blockentity : levelchunk.getBlockEntities().values()) {
+                for (BlockEntity blockentity : levelchunk.getBlockEntities().values()) {
                     if (blockentity instanceof TheEndPortalBlockEntity) {
                         BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch = this.exitPortalPattern.find(this.level, blockentity.getBlockPos());
                         if (blockpattern$blockpatternmatch != null) {
@@ -298,7 +_,7 @@
         BlockPos blockpos1 = EndPodiumFeature.getLocation(this.origin);
         int k = this.level.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING, blockpos1).getY();
 
-        for(int l = k; l >= this.level.getMinBuildHeight(); --l) {
+        for (int l = k; l >= this.level.getMinBuildHeight(); l--) {
             BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch1 = this.exitPortalPattern
                 .find(this.level, new BlockPos(blockpos1.getX(), l, blockpos1.getZ()));
             if (blockpattern$blockpatternmatch1 != null) {
@@ -319,8 +_,8 @@
         } else {
             ChunkPos chunkpos = new ChunkPos(this.origin);
 
-            for(int i = -8 + chunkpos.x; i <= 8 + chunkpos.x; ++i) {
-                for(int j = 8 + chunkpos.z; j <= 8 + chunkpos.z; ++j) {
+            for (int i = -8 + chunkpos.x; i <= 8 + chunkpos.x; i++) {
+                for (int j = 8 + chunkpos.z; j <= 8 + chunkpos.z; j++) {
                     ChunkAccess chunkaccess = this.level.getChunk(i, j, ChunkStatus.FULL, false);
                     if (!(chunkaccess instanceof LevelChunk)) {
                         return false;
@@ -340,7 +_,7 @@
     private void updatePlayers() {
         Set<ServerPlayer> set = Sets.newHashSet();
 
-        for(ServerPlayer serverplayer : this.level.getPlayers(this.validPlayer)) {
+        for (ServerPlayer serverplayer : this.level.getPlayers(this.validPlayer)) {
             this.dragonEvent.addPlayer(serverplayer);
             set.add(serverplayer);
         }
@@ -348,7 +_,7 @@
         Set<ServerPlayer> set1 = Sets.newHashSet(this.dragonEvent.getPlayers());
         set1.removeAll(set);
 
-        for(ServerPlayer serverplayer1 : set1) {
+        for (ServerPlayer serverplayer1 : set1) {
             this.dragonEvent.removePlayer(serverplayer1);
         }
     }
@@ -357,8 +_,8 @@
         this.ticksSinceCrystalsScanned = 0;
         this.crystalsAlive = 0;
 
-        for(SpikeFeature.EndSpike spikefeature$endspike : SpikeFeature.getSpikesForLevel(this.level)) {
-            this.crystalsAlive += this.level.getEntitiesOfClass(EndCrystal.class, spikefeature$endspike.getTopBoundingBox()).size();
+        for (SpikeFeature.EndSpike spikefeature$endspike : SpikeFeature.getSpikesForLevel(this.level)) {
+            this.crystalsAlive = this.crystalsAlive + this.level.getEntitiesOfClass(EndCrystal.class, spikefeature$endspike.getTopBoundingBox()).size();
         }
 
         LOGGER.debug("Found {} end crystals still alive", this.crystalsAlive);
@@ -412,7 +_,7 @@
         if (this.portalLocation == null) {
             this.portalLocation = this.level.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, EndPodiumFeature.getLocation(this.origin)).below();
 
-            while(this.level.getBlockState(this.portalLocation).is(Blocks.BEDROCK) && this.portalLocation.getY() > this.level.getSeaLevel()) {
+            while (this.level.getBlockState(this.portalLocation).is(Blocks.BEDROCK) && this.portalLocation.getY() > this.level.getSeaLevel()) {
                 this.portalLocation = this.portalLocation.below();
             }
         }
@@ -496,7 +_,7 @@
             List<EndCrystal> list1 = Lists.newArrayList();
             BlockPos blockpos1 = blockpos.above(1);
 
-            for(Direction direction : Direction.Plane.HORIZONTAL) {
+            for (Direction direction : Direction.Plane.HORIZONTAL) {
                 List<EndCrystal> list = this.level.getEntitiesOfClass(EndCrystal.class, new AABB(blockpos1.relative(direction, 2)));
                 if (list.isEmpty()) {
                     return;
@@ -512,13 +_,13 @@
 
     private void respawnDragon(List<EndCrystal> p_64092_) {
         if (this.dragonKilled && this.respawnStage == null) {
-            for(BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch = this.findExitPortal();
+            for (BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch = this.findExitPortal();
                 blockpattern$blockpatternmatch != null;
                 blockpattern$blockpatternmatch = this.findExitPortal()
             ) {
-                for(int i = 0; i < this.exitPortalPattern.getWidth(); ++i) {
-                    for(int j = 0; j < this.exitPortalPattern.getHeight(); ++j) {
-                        for(int k = 0; k < this.exitPortalPattern.getDepth(); ++k) {
+                for (int i = 0; i < this.exitPortalPattern.getWidth(); i++) {
+                    for (int j = 0; j < this.exitPortalPattern.getHeight(); j++) {
+                        for (int k = 0; k < this.exitPortalPattern.getDepth(); k++) {
                             BlockInWorld blockinworld = blockpattern$blockpatternmatch.getBlock(i, j, k);
                             if (blockinworld.getState().is(Blocks.BEDROCK) || blockinworld.getState().is(Blocks.END_PORTAL)) {
                                 this.level.setBlockAndUpdate(blockinworld.getPos(), Blocks.END_STONE.defaultBlockState());
@@ -536,14 +_,22 @@
     }
 
     public void resetSpikeCrystals() {
-        for(SpikeFeature.EndSpike spikefeature$endspike : SpikeFeature.getSpikesForLevel(this.level)) {
-            for(EndCrystal endcrystal : this.level.getEntitiesOfClass(EndCrystal.class, spikefeature$endspike.getTopBoundingBox())) {
+        for (SpikeFeature.EndSpike spikefeature$endspike : SpikeFeature.getSpikesForLevel(this.level)) {
+            for (EndCrystal endcrystal : this.level.getEntitiesOfClass(EndCrystal.class, spikefeature$endspike.getTopBoundingBox())) {
                 endcrystal.setInvulnerable(false);
                 endcrystal.setBeamTarget(null);
             }
         }
     }
 
+    public void addPlayer(ServerPlayer player) {
+        this.dragonEvent.addPlayer(player);
+    }
+
+    public void removePlayer(ServerPlayer player) {
+        this.dragonEvent.removePlayer(player);
+    }
+
     @Nullable
     public UUID getDragonUUID() {
         return this.dragonUUID;
@@ -563,10 +_,10 @@
                         Codec.BOOL.fieldOf("NeedsStateScanning").orElse(true).forGetter(EndDragonFight.Data::needsStateScanning),
                         Codec.BOOL.fieldOf("DragonKilled").orElse(false).forGetter(EndDragonFight.Data::dragonKilled),
                         Codec.BOOL.fieldOf("PreviouslyKilled").orElse(false).forGetter(EndDragonFight.Data::previouslyKilled),
-                        Codec.BOOL.optionalFieldOf("IsRespawning", Boolean.valueOf(false)).forGetter(EndDragonFight.Data::isRespawning),
-                        UUIDUtil.CODEC.optionalFieldOf("Dragon").forGetter(EndDragonFight.Data::dragonUUID),
-                        BlockPos.CODEC.optionalFieldOf("ExitPortalLocation").forGetter(EndDragonFight.Data::exitPortalLocation),
-                        Codec.list(Codec.INT).optionalFieldOf("Gateways").forGetter(EndDragonFight.Data::gateways)
+                        Codec.BOOL.lenientOptionalFieldOf("IsRespawning", Boolean.valueOf(false)).forGetter(EndDragonFight.Data::isRespawning),
+                        UUIDUtil.CODEC.lenientOptionalFieldOf("Dragon").forGetter(EndDragonFight.Data::dragonUUID),
+                        BlockPos.CODEC.lenientOptionalFieldOf("ExitPortalLocation").forGetter(EndDragonFight.Data::exitPortalLocation),
+                        Codec.list(Codec.INT).lenientOptionalFieldOf("Gateways").forGetter(EndDragonFight.Data::gateways)
                     )
                     .apply(p_289803_, EndDragonFight.Data::new)
         );
