--- a/net/minecraft/client/gui/components/tabs/TabNavigationBar.java
+++ b/net/minecraft/client/gui/components/tabs/TabNavigationBar.java
@@ -47,7 +_,7 @@
         this.layout.defaultCellSetting().alignHorizontallyCenter();
         ImmutableList.Builder<TabButton> builder = ImmutableList.builder();
 
-        for(Tab tab : p_275279_) {
+        for (Tab tab : p_275279_) {
             builder.add(this.layout.addChild(new TabButton(p_275624_, tab, 0, 24)));
         }
 
@@ -134,7 +_,7 @@
         p_281720_.blit(Screen.HEADER_SEPARATOR, i, this.layout.getY() + this.layout.getHeight() - 2, 0.0F, 0.0F, this.width, 2, 32, 2);
         RenderSystem.disableBlend();
 
-        for(TabButton tabbutton : this.tabButtons) {
+        for (TabButton tabbutton : this.tabButtons) {
             tabbutton.render(p_281720_, p_282085_, p_281687_, p_283048_);
         }
     }
@@ -148,7 +_,7 @@
         int i = Math.min(400, this.width) - 28;
         int j = Mth.roundToward(i / this.tabs.size(), 2);
 
-        for(TabButton tabbutton : this.tabButtons) {
+        for (TabButton tabbutton : this.tabButtons) {
             tabbutton.setWidth(j);
         }
 
