--- a/net/minecraft/server/gui/PlayerListComponent.java
+++ b/net/minecraft/server/gui/PlayerListComponent.java
@@ -17,7 +_,7 @@
         if (this.tickCount++ % 20 == 0) {
             Vector<String> vector = new Vector<>();
 
-            for(int i = 0; i < this.server.getPlayerList().getPlayers().size(); ++i) {
+            for (int i = 0; i < this.server.getPlayerList().getPlayers().size(); i++) {
                 vector.add(this.server.getPlayerList().getPlayers().get(i).getGameProfile().getName());
             }
 
