--- a/net/minecraft/client/gui/navigation/ScreenRectangle.java
+++ b/net/minecraft/client/gui/navigation/ScreenRectangle.java
@@ -17,9 +_,10 @@
     }
 
     public static ScreenRectangle of(ScreenAxis p_265648_, int p_265317_, int p_265685_, int p_265218_, int p_265226_) {
-        return switch(p_265648_) {
+        return switch (p_265648_) {
             case HORIZONTAL -> new ScreenRectangle(p_265317_, p_265685_, p_265218_, p_265226_);
             case VERTICAL -> new ScreenRectangle(p_265685_, p_265317_, p_265226_, p_265218_);
+            default -> throw new MatchException(null, null);
         };
     }
 
@@ -28,9 +_,10 @@
     }
 
     public int getLength(ScreenAxis p_265463_) {
-        return switch(p_265463_) {
+        return switch (p_265463_) {
             case HORIZONTAL -> this.width;
             case VERTICAL -> this.height;
+            default -> throw new MatchException(null, null);
         };
     }
 
