--- a/net/minecraft/client/gui/navigation/ScreenDirection.java
+++ b/net/minecraft/client/gui/navigation/ScreenDirection.java
@@ -16,14 +_,14 @@
             : (this.isBefore(p_265081_, p_265641_) ? -1 : 1);
 
     public ScreenAxis getAxis() {
-        return switch(this) {
+        return switch (this) {
             case UP, DOWN -> ScreenAxis.VERTICAL;
             case LEFT, RIGHT -> ScreenAxis.HORIZONTAL;
         };
     }
 
     public ScreenDirection getOpposite() {
-        return switch(this) {
+        return switch (this) {
             case UP -> DOWN;
             case DOWN -> UP;
             case LEFT -> RIGHT;
@@ -32,26 +_,18 @@
     }
 
     public boolean isPositive() {
-        return switch(this) {
+        return switch (this) {
             case UP, LEFT -> false;
             case DOWN, RIGHT -> true;
         };
     }
 
     public boolean isAfter(int p_265461_, int p_265553_) {
-        if (this.isPositive()) {
-            return p_265461_ > p_265553_;
-        } else {
-            return p_265553_ > p_265461_;
-        }
+        return this.isPositive() ? p_265461_ > p_265553_ : p_265553_ > p_265461_;
     }
 
     public boolean isBefore(int p_265215_, int p_265040_) {
-        if (this.isPositive()) {
-            return p_265215_ < p_265040_;
-        } else {
-            return p_265040_ < p_265215_;
-        }
+        return this.isPositive() ? p_265215_ < p_265040_ : p_265040_ < p_265215_;
     }
 
     public IntComparator coordinateValueComparator() {
