--- a/net/minecraft/world/scores/Score.java
+++ b/net/minecraft/world/scores/Score.java
@@ -69,8 +_,7 @@
         if (this.numberFormat != null) {
             NumberFormatTypes.CODEC
                 .encodeStart(p_330377_.createSerializationContext(NbtOps.INSTANCE), this.numberFormat)
-                .result()
-                .ifPresent(p_313666_ -> compoundtag.put("format", p_313666_));
+                .ifSuccess(p_313666_ -> compoundtag.put("format", p_313666_));
         }
 
         return compoundtag;
@@ -87,8 +_,7 @@
         if (p_313855_.contains("format", 10)) {
             NumberFormatTypes.CODEC
                 .parse(p_331997_.createSerializationContext(NbtOps.INSTANCE), p_313855_.get("format"))
-                .result()
-                .ifPresent(p_313664_ -> score.numberFormat = p_313664_);
+                .ifSuccess(p_313664_ -> score.numberFormat = p_313664_);
         }
 
         return score;
