--- a/net/minecraft/world/scores/Scoreboard.java
+++ b/net/minecraft/world/scores/Scoreboard.java
@@ -222,7 +_,7 @@
     public void removeObjective(Objective p_83503_) {
         this.objectivesByName.remove(p_83503_.getName());
 
-        for(DisplaySlot displayslot : DisplaySlot.values()) {
+        for (DisplaySlot displayslot : DisplaySlot.values()) {
             if (this.getDisplayObjective(displayslot) == p_83503_) {
                 this.setDisplayObjective(displayslot, null);
             }
@@ -233,7 +_,7 @@
             list.remove(p_83503_);
         }
 
-        for(PlayerScores playerscores : this.playerScores.values()) {
+        for (PlayerScores playerscores : this.playerScores.values()) {
             playerscores.remove(p_83503_);
         }
 
@@ -270,7 +_,7 @@
     public void removePlayerTeam(PlayerTeam p_83476_) {
         this.teamsByName.remove(p_83476_.getName());
 
-        for(String s : p_83476_.getPlayers()) {
+        for (String s : p_83476_.getPlayers()) {
             this.teamsByPlayer.remove(s);
         }
 
@@ -367,7 +_,7 @@
     }
 
     protected void loadPlayerScores(ListTag p_83446_, HolderLookup.Provider p_330625_) {
-        for(int i = 0; i < p_83446_.size(); ++i) {
+        for (int i = 0; i < p_83446_.size(); i++) {
             CompoundTag compoundtag = p_83446_.getCompound(i);
             Score score = Score.read(compoundtag, p_330625_);
             String s = compoundtag.getString("Name");
