--- a/net/minecraft/world/scores/Team.java
+++ b/net/minecraft/world/scores/Team.java
@@ -11,11 +_,7 @@
 
 public abstract class Team {
     public boolean isAlliedTo(@Nullable Team p_83537_) {
-        if (p_83537_ == null) {
-            return false;
-        } else {
-            return this == p_83537_;
-        }
+        return p_83537_ == null ? false : this == p_83537_;
     }
 
     public abstract String getName();
@@ -43,7 +_,7 @@
         PUSH_OWN_TEAM("pushOwnTeam", 3);
 
         private static final Map<String, Team.CollisionRule> BY_NAME = Arrays.stream(values())
-            .collect(Collectors.toMap(p_83559_ -> p_83559_.name, p_83554_ -> p_83554_));
+            .collect(Collectors.toMap(p_83559_ -> p_83559_.name, p_83554_ -> (Team.CollisionRule)p_83554_));
         public final String name;
         public final int id;
 
@@ -69,7 +_,7 @@
         HIDE_FOR_OWN_TEAM("hideForOwnTeam", 3);
 
         private static final Map<String, Team.Visibility> BY_NAME = Arrays.stream(values())
-            .collect(Collectors.toMap(p_83583_ -> p_83583_.name, p_83578_ -> p_83578_));
+            .collect(Collectors.toMap(p_83583_ -> p_83583_.name, p_83578_ -> (Team.Visibility)p_83578_));
         public final String name;
         public final int id;
 
