--- a/net/minecraft/world/entity/ai/goal/target/HurtByTargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/target/HurtByTargetGoal.java
@@ -38,7 +_,7 @@
             if (livingentity.getType() == EntityType.PLAYER && this.mob.level().getGameRules().getBoolean(GameRules.RULE_UNIVERSAL_ANGER)) {
                 return false;
             } else {
-                for(Class<?> oclass : this.toIgnoreDamage) {
+                for (Class<?> oclass : this.toIgnoreDamage) {
                     if (oclass.isAssignableFrom(livingentity.getClass())) {
                         return false;
                     }
@@ -73,12 +_,12 @@
     protected void alertOthers() {
         double d0 = this.getFollowDistance();
         AABB aabb = AABB.unitCubeFromLowerCorner(this.mob.position()).inflate(d0, 10.0, d0);
-        List<? extends Mob> list = this.mob.level().getEntitiesOfClass(this.mob.getClass(), aabb, EntitySelector.NO_SPECTATORS);
+        List<? extends Mob> list = this.mob.level().getEntitiesOfClass((Class<? extends Mob>)this.mob.getClass(), aabb, EntitySelector.NO_SPECTATORS);
         Iterator iterator = list.iterator();
 
-        while(true) {
+        while (true) {
             Mob mob;
-            while(true) {
+            while (true) {
                 if (!iterator.hasNext()) {
                     return;
                 }
@@ -94,7 +_,7 @@
 
                     boolean flag = false;
 
-                    for(Class<?> oclass : this.toIgnoreAlert) {
+                    for (Class<?> oclass : this.toIgnoreAlert) {
                         if (mob.getClass() == oclass) {
                             flag = true;
                             break;
