--- a/net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
@@ -4,6 +_,7 @@
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.Mob;
 import net.minecraft.world.entity.Targetable;
 import net.minecraft.world.entity.ai.goal.Goal;
@@ -73,7 +_,9 @@
 
     @Override
     public void start() {
-        this.mob.setTarget(this.target);
+        if (this.target instanceof LivingEntity livingTarget) {
+            this.mob.setTarget(livingTarget);
+        }
         super.start();
     }
 
