--- a/net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
@@ -4,6 +_,7 @@
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.Mob;
 import net.minecraft.world.entity.Targetable;
 import net.minecraft.world.entity.ai.goal.Goal;
@@ -16,7 +_,7 @@
     private static final int DEFAULT_RANDOM_INTERVAL = 10;
     protected final Class<T> targetType;
     protected final int randomInterval;
-    protected @Nullable T target;
+    protected @Nullable LivingEntity target;
     protected final TargetingConditions targetConditions;
 
     public NearestAttackableTargetGoal(Mob mob, Class<T> targetType, boolean mustSee) {
@@ -58,7 +_,7 @@
     protected void findTarget() {
         ServerLevel level = getServerLevel(this.mob);
         if (this.targetType != Player.class && this.targetType != ServerPlayer.class) {
-            this.target = level.getNearestEntity(
+            this.target = (LivingEntity) level.getNearestEntity(
                 this.mob.level().getEntitiesOfClass(this.targetType, this.getTargetSearchArea(this.getFollowDistance()), entity -> true),
                 this.getTargetConditions(),
                 this.mob,
@@ -67,18 +_,20 @@
                 this.mob.getZ()
             );
         } else {
-            this.target = (T)level.getNearestPlayer(this.getTargetConditions(), this.mob, this.mob.getX(), this.mob.getEyeY(), this.mob.getZ());
+            this.target = level.getNearestPlayer(this.getTargetConditions(), this.mob, this.mob.getX(), this.mob.getEyeY(), this.mob.getZ());
         }
     }
 
     @Override
     public void start() {
-        this.mob.setTarget(this.target);
+        if (this.target instanceof LivingEntity livingTarget) {
+            this.mob.setTarget(livingTarget);
+        }
         super.start();
     }
 
     public void setTarget(@Nullable T target) {
-        this.target = target;
+        this.target = (LivingEntity) target;
     }
 
     private TargetingConditions getTargetConditions() {
