--- a/net/minecraft/world/entity/ai/goal/target/TargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/target/TargetGoal.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.entity.ai.goal.target;
 
 import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.Mob;
 import net.minecraft.world.entity.ai.attributes.Attributes;
 import net.minecraft.world.entity.ai.goal.Goal;
@@ -62,7 +_,9 @@
                         }
                     }
 
-                    this.mob.setTarget(target);
+                    if (target instanceof LivingEntity livingTarget) {
+                        this.mob.setTarget(livingTarget);
+                    }
                     return true;
                 }
             }
@@ -86,7 +_,7 @@
         this.targetMob = null;
     }
 
-    protected boolean canAttack(@Nullable Entity target, TargetingConditions targetConditions) {
+    protected boolean canAttack(@Nullable LivingEntity target, TargetingConditions targetConditions) {
         if (target == null) {
             return false;
         } else if (!targetConditions.test(getServerLevel(this.mob), this.mob, target)) {
