--- a/com/mojang/blaze3d/systems/RenderSystem.java
+++ b/com/mojang/blaze3d/systems/RenderSystem.java
@@ -45,6 +_,8 @@
 @DontObfuscate
 public class RenderSystem {
     public static final ScissorState SCISSOR_STATE = new ScissorState();
+    @Nullable
+    public static net.neoforged.neoforge.client.stencil.StencilTest STENCIL_TEST = null;
     static final Logger LOGGER = LogUtils.getLogger();
     public static final int MINIMUM_ATLAS_TEXTURE_SIZE = 1024;
     @Nullable
@@ -152,6 +_,15 @@
 
     public static void disableScissor() {
         SCISSOR_STATE.disable();
+    }
+
+
+    public static void enableStencil(net.neoforged.neoforge.client.stencil.StencilTest stencilTest) {
+        STENCIL_TEST = stencilTest;
+    }
+
+    public static void disableStencil() {
+        STENCIL_TEST = null;
     }
 
     public static void setShaderFog(FogParameters p_360454_) {
