--- a/net/minecraft/world/entity/monster/illager/Pillager.java
+++ b/net/minecraft/world/entity/monster/illager/Pillager.java
@@ -17,6 +_,7 @@
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.EntitySpawnReason;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.EquipmentSlot;
 import net.minecraft.world.entity.Mob;
@@ -98,7 +_,7 @@
 
     @Override
     public boolean canUseNonMeleeWeapon(ItemStack item) {
-        return item.getItem() == Items.CROSSBOW;
+        return item.is(net.neoforged.neoforge.common.Tags.Items.PILLAGER_USABLE_CROSSBOWS);
     }
 
     public boolean isChargingCrossbow() {
@@ -130,7 +_,7 @@
     public AbstractIllager.IllagerArmPose getArmPose() {
         if (this.isChargingCrossbow()) {
             return AbstractIllager.IllagerArmPose.CROSSBOW_CHARGE;
-        } else if (this.isHolding(Items.CROSSBOW)) {
+        } else if (this.isHolding(is -> is.getItem() instanceof net.minecraft.world.item.CrossbowItem)) {
             return AbstractIllager.IllagerArmPose.CROSSBOW_HOLD;
         } else {
             return this.isAggressive() ? AbstractIllager.IllagerArmPose.ATTACKING : AbstractIllager.IllagerArmPose.NEUTRAL;
@@ -200,6 +_,10 @@
     @Override
     public void performRangedAttack(Entity target, float power) {
         this.performCrossbowAttack(this, 1.6F);
+    }
+
+    public void performRangedAttack(LivingEntity target, float power) {
+        this.performRangedAttack((net.minecraft.world.entity.Entity) target, power);
     }
 
     @Override
