--- a/net/minecraft/world/level/levelgen/structure/BoundingBox.java
+++ b/net/minecraft/world/level/levelgen/structure/BoundingBox.java
@@ -22,7 +_,7 @@
     private static final Logger LOGGER = LogUtils.getLogger();
     public static final Codec<BoundingBox> CODEC = Codec.INT_STREAM
         .<BoundingBox>comapFlatMap(
-            p_162383_ -> Util.fixedSize(p_162383_, 6)
+            p_338100_ -> Util.fixedSize(p_338100_, 6)
                     .map(p_162385_ -> new BoundingBox(p_162385_[0], p_162385_[1], p_162385_[2], p_162385_[3], p_162385_[4], p_162385_[5])),
             p_162391_ -> IntStream.of(p_162391_.minX, p_162391_.minY, p_162391_.minZ, p_162391_.maxX, p_162391_.maxY, p_162391_.maxZ)
         )
@@ -79,7 +_,7 @@
     public static BoundingBox orientBox(
         int p_71032_, int p_71033_, int p_71034_, int p_71035_, int p_71036_, int p_71037_, int p_71038_, int p_71039_, int p_71040_, Direction p_71041_
     ) {
-        switch(p_71041_) {
+        switch (p_71041_) {
             case SOUTH:
             default:
                 return new BoundingBox(
@@ -221,17 +_,6 @@
         );
     }
 
-    public BoundingBox inflatedBy(int p_341109_, int p_341453_, int p_342241_) {
-        return new BoundingBox(
-            this.minX() - p_341109_,
-            this.minY() - p_341453_,
-            this.minZ() - p_342241_,
-            this.maxX() + p_341109_,
-            this.maxY() + p_341453_,
-            this.maxZ() + p_342241_
-        );
-    }
-
     public boolean isInside(Vec3i p_71052_) {
         return this.isInside(p_71052_.getX(), p_71052_.getY(), p_71052_.getZ());
     }
@@ -295,16 +_,15 @@
     public boolean equals(Object p_162393_) {
         if (this == p_162393_) {
             return true;
-        } else if (!(p_162393_ instanceof BoundingBox)) {
-            return false;
         } else {
-            BoundingBox boundingbox = (BoundingBox)p_162393_;
-            return this.minX == boundingbox.minX
-                && this.minY == boundingbox.minY
-                && this.minZ == boundingbox.minZ
-                && this.maxX == boundingbox.maxX
-                && this.maxY == boundingbox.maxY
-                && this.maxZ == boundingbox.maxZ;
+            return !(p_162393_ instanceof BoundingBox boundingbox)
+                ? false
+                : this.minX == boundingbox.minX
+                    && this.minY == boundingbox.minY
+                    && this.minZ == boundingbox.minZ
+                    && this.maxX == boundingbox.maxX
+                    && this.maxY == boundingbox.maxY
+                    && this.maxZ == boundingbox.maxZ;
         }
     }
 
