--- a/net/minecraft/world/level/levelgen/structure/BuiltinStructureSets.java
+++ b/net/minecraft/world/level/levelgen/structure/BuiltinStructureSets.java
@@ -16,12 +_,10 @@
     ResourceKey<StructureSet> BURIED_TREASURES = register("buried_treasures");
     ResourceKey<StructureSet> MINESHAFTS = register("mineshafts");
     ResourceKey<StructureSet> RUINED_PORTALS = register("ruined_portals");
-    ResourceKey<StructureSet> RUINED_PORTATOLS = register("ruined_portatols");
     ResourceKey<StructureSet> SHIPWRECKS = register("shipwrecks");
     ResourceKey<StructureSet> OCEAN_RUINS = register("ocean_ruins");
     ResourceKey<StructureSet> NETHER_COMPLEXES = register("nether_complexes");
     ResourceKey<StructureSet> NETHER_FOSSILS = register("nether_fossils");
-    ResourceKey<StructureSet> COLOSSEA = register("colossea");
     ResourceKey<StructureSet> END_CITIES = register("end_cities");
     ResourceKey<StructureSet> ANCIENT_CITIES = register("ancient_cities");
     ResourceKey<StructureSet> STRONGHOLDS = register("strongholds");
