--- a/net/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java
+++ b/net/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java
@@ -75,7 +_,7 @@
         p_192426_.putString("rotation", this.rotation.name());
         ListTag listtag = new ListTag();
 
-        for(JigsawJunction jigsawjunction : this.junctions) {
+        for (JigsawJunction jigsawjunction : this.junctions) {
             listtag.add(jigsawjunction.serialize(dynamicops).getValue());
         }
 
