--- a/net/minecraft/world/level/levelgen/structure/ScatteredFeaturePiece.java
+++ b/net/minecraft/world/level/levelgen/structure/ScatteredFeaturePiece.java
@@ -48,12 +_,12 @@
             int j = 0;
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(int k = this.boundingBox.minZ(); k <= this.boundingBox.maxZ(); ++k) {
-                for(int l = this.boundingBox.minX(); l <= this.boundingBox.maxX(); ++l) {
+            for (int k = this.boundingBox.minZ(); k <= this.boundingBox.maxZ(); k++) {
+                for (int l = this.boundingBox.minX(); l <= this.boundingBox.maxX(); l++) {
                     blockpos$mutableblockpos.set(l, 64, k);
                     if (p_72805_.isInside(blockpos$mutableblockpos)) {
                         i += p_72804_.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, blockpos$mutableblockpos).getY();
-                        ++j;
+                        j++;
                     }
                 }
             }
@@ -76,8 +_,8 @@
             boolean flag = false;
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(int j = this.boundingBox.minZ(); j <= this.boundingBox.maxZ(); ++j) {
-                for(int k = this.boundingBox.minX(); k <= this.boundingBox.maxX(); ++k) {
+            for (int j = this.boundingBox.minZ(); j <= this.boundingBox.maxZ(); j++) {
+                for (int k = this.boundingBox.minX(); k <= this.boundingBox.maxX(); k++) {
                     blockpos$mutableblockpos.set(k, 0, j);
                     i = Math.min(i, p_192468_.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, blockpos$mutableblockpos).getY());
                     flag = true;
