--- a/net/minecraft/world/level/levelgen/structure/StructureStart.java
+++ b/net/minecraft/world/level/levelgen/structure/StructureStart.java
@@ -87,7 +_,7 @@
             BlockPos blockpos = boundingbox.getCenter();
             BlockPos blockpos1 = new BlockPos(blockpos.getX(), boundingbox.minY(), blockpos.getZ());
 
-            for(StructurePiece structurepiece : list) {
+            for (StructurePiece structurepiece : list) {
                 if (structurepiece.getBoundingBox().intersects(p_226855_)) {
                     structurepiece.postProcess(p_226851_, p_226852_, p_226853_, p_226854_, p_226855_, p_226856_, blockpos1);
                 }
@@ -100,6 +_,9 @@
     public CompoundTag createTag(StructurePieceSerializationContext p_192661_, ChunkPos p_192662_) {
         CompoundTag compoundtag = new CompoundTag();
         if (this.isValid()) {
+            if (p_192661_.registryAccess().registryOrThrow(Registries.STRUCTURE).getKey(this.getStructure()) == null) { // FORGE: This is just a more friendly error instead of the 'Null String' below
+                throw new RuntimeException("StructureStart \"" + this.getClass().getName() + "\": \"" + this.getStructure() + "\" unregistered, serializing impossible.");
+            }
             compoundtag.putString("id", p_192661_.registryAccess().registryOrThrow(Registries.STRUCTURE).getKey(this.structure).toString());
             compoundtag.putInt("ChunkX", p_192662_.x);
             compoundtag.putInt("ChunkZ", p_192662_.z);
@@ -125,7 +_,7 @@
     }
 
     public void addReference() {
-        ++this.references;
+        this.references++;
     }
 
     public int getReferences() {
