--- a/net/minecraft/world/level/levelgen/structure/StructureCheck.java
+++ b/net/minecraft/world/level/levelgen/structure/StructureCheck.java
@@ -11,14 +_,12 @@
 import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.function.BiFunction;
 import javax.annotation.Nullable;
 import net.minecraft.core.Registry;
 import net.minecraft.core.RegistryAccess;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.IntTag;
-import net.minecraft.nbt.Tag;
 import net.minecraft.nbt.visitors.CollectFields;
 import net.minecraft.nbt.visitors.FieldSelector;
 import net.minecraft.resources.ResourceKey;
@@ -107,8 +_,7 @@
                     this.seed,
                     p_226756_,
                     this.heightAccessor,
-                    p_226757_.biomes()::contains,
-                    p_226757_.densityChecks()
+                    p_226757_.biomes()::contains
                 )
             )
             .isPresent();
@@ -129,11 +_,9 @@
             return StructureCheckResult.CHUNK_LOAD_NEEDED;
         }
 
-        Tag tag = collectfields.getResult();
-        if (!(tag instanceof CompoundTag)) {
+        if (!(collectfields.getResult() instanceof CompoundTag compoundtag)) {
             return null;
         } else {
-            CompoundTag compoundtag = (CompoundTag)tag;
             int i = ChunkStorage.getVersion(compoundtag);
             if (i <= 1493) {
                 return StructureCheckResult.CHUNK_LOAD_NEEDED;
@@ -175,7 +_,7 @@
                     Object2IntMap<Structure> object2intmap = new Object2IntOpenHashMap<>();
                     Registry<Structure> registry = this.registryAccess.registryOrThrow(Registries.STRUCTURE);
 
-                    for(String s : compoundtag1.getAllKeys()) {
+                    for (String s : compoundtag1.getAllKeys()) {
                         ResourceLocation resourcelocation = ResourceLocation.tryParse(s);
                         if (resourcelocation != null) {
                             Structure structure = registry.get(resourcelocation);
@@ -224,17 +_,13 @@
     }
 
     public void incrementReference(ChunkPos p_226723_, Structure p_226724_) {
-        this.loadedChunks
-            .compute(
-                p_226723_.toLong(),
-                (BiFunction<? super Long, ? super Object2IntMap<Structure>, ? extends Object2IntMap<Structure>>)((p_226745_, p_226746_) -> {
-                    if (p_226746_ == null || p_226746_.isEmpty()) {
-                        p_226746_ = new Object2IntOpenHashMap<>();
-                    }
-        
-                    p_226746_.computeInt(p_226724_, (p_226741_, p_226742_) -> p_226742_ == null ? 1 : p_226742_ + 1);
-                    return p_226746_;
-                })
-            );
+        this.loadedChunks.compute(p_226723_.toLong(), (p_226745_, p_226746_) -> {
+            if (p_226746_ == null || p_226746_.isEmpty()) {
+                p_226746_ = new Object2IntOpenHashMap<>();
+            }
+
+            p_226746_.computeInt(p_226724_, (p_226741_, p_226742_) -> p_226742_ == null ? 1 : p_226742_ + 1);
+            return p_226746_;
+        });
     }
 }
