--- a/net/minecraft/world/level/levelgen/structure/TemplateStructurePiece.java
+++ b/net/minecraft/world/level/levelgen/structure/TemplateStructurePiece.java
@@ -88,7 +_,7 @@
         this.placeSettings.setBoundingBox(p_226903_);
         this.boundingBox = this.template.getBoundingBox(this.placeSettings, this.templatePosition);
         if (this.template.placeInWorld(p_226899_, this.templatePosition, p_226905_, this.placeSettings, p_226902_, 2)) {
-            for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : this.template
+            for (StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : this.template
                 .filterBlocks(this.templatePosition, this.placeSettings, Blocks.STRUCTURE_BLOCK)) {
                 if (structuretemplate$structureblockinfo.nbt() != null) {
                     StructureMode structuremode = StructureMode.valueOf(structuretemplate$structureblockinfo.nbt().getString("mode"));
@@ -104,7 +_,7 @@
                 }
             }
 
-            for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo1 : this.template
+            for (StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo1 : this.template
                 .filterBlocks(this.templatePosition, this.placeSettings, Blocks.JIGSAW)) {
                 if (structuretemplate$structureblockinfo1.nbt() != null) {
                     String s = structuretemplate$structureblockinfo1.nbt().getString("final_state");
