--- a/net/minecraft/world/entity/monster/breeze/BreezeUtil.java
+++ b/net/minecraft/world/entity/monster/breeze/BreezeUtil.java
@@ -20,11 +_,9 @@
 
     public static boolean hasLineOfSight(Breeze p_316785_, Vec3 p_316249_) {
         Vec3 vec3 = new Vec3(p_316785_.getX(), p_316785_.getY(), p_316785_.getZ());
-        if (p_316249_.distanceTo(vec3) > 50.0) {
-            return false;
-        } else {
-            return p_316785_.level().clip(new ClipContext(vec3, p_316249_, ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, p_316785_)).getType()
+        return p_316249_.distanceTo(vec3) > 50.0
+            ? false
+            : p_316785_.level().clip(new ClipContext(vec3, p_316249_, ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, p_316785_)).getType()
                 == HitResult.Type.MISS;
-        }
     }
 }
