--- a/net/minecraft/world/entity/monster/breeze/LongJump.java
+++ b/net/minecraft/world/entity/monster/breeze/LongJump.java
@@ -24,6 +_,7 @@
 import net.minecraft.world.entity.ai.memory.MemoryModuleType;
 import net.minecraft.world.entity.ai.memory.MemoryStatus;
 import net.minecraft.world.level.ClipContext;
+import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
 
@@ -58,38 +_,48 @@
         );
     }
 
-    protected boolean checkExtraStartConditions(ServerLevel p_312131_, Breeze p_312686_) {
-        if (!p_312686_.onGround() && !p_312686_.isInWater()) {
-            return false;
-        } else if (Swim.shouldSwim(p_312686_)) {
-            return false;
-        } else if (p_312686_.getBrain().checkMemory(MemoryModuleType.BREEZE_JUMP_TARGET, MemoryStatus.VALUE_PRESENT)) {
+    public static boolean canRun(ServerLevel p_338176_, Breeze p_338459_) {
+        if (!p_338459_.onGround() && !p_338459_.isInWater()) {
+            return false;
+        } else if (Swim.shouldSwim(p_338459_)) {
+            return false;
+        } else if (p_338459_.getBrain().checkMemory(MemoryModuleType.BREEZE_JUMP_TARGET, MemoryStatus.VALUE_PRESENT)) {
             return true;
         } else {
-            LivingEntity livingentity = p_312686_.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null);
+            LivingEntity livingentity = p_338459_.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null);
             if (livingentity == null) {
                 return false;
-            } else if (outOfAggroRange(p_312686_, livingentity)) {
-                p_312686_.getBrain().eraseMemory(MemoryModuleType.ATTACK_TARGET);
-                return false;
-            } else if (tooCloseForJump(p_312686_, livingentity)) {
-                return false;
-            } else if (!canJumpFromCurrentPosition(p_312131_, p_312686_)) {
+            } else if (outOfAggroRange(p_338459_, livingentity)) {
+                p_338459_.getBrain().eraseMemory(MemoryModuleType.ATTACK_TARGET);
+                return false;
+            } else if (tooCloseForJump(p_338459_, livingentity)) {
+                return false;
+            } else if (!canJumpFromCurrentPosition(p_338176_, p_338459_)) {
                 return false;
             } else {
-                BlockPos blockpos = snapToSurface(p_312686_, BreezeUtil.randomPointBehindTarget(livingentity, p_312686_.getRandom()));
+                BlockPos blockpos = snapToSurface(p_338459_, BreezeUtil.randomPointBehindTarget(livingentity, p_338459_.getRandom()));
                 if (blockpos == null) {
                     return false;
-                } else if (!BreezeUtil.hasLineOfSight(p_312686_, blockpos.getCenter()) && !BreezeUtil.hasLineOfSight(p_312686_, blockpos.above(4).getCenter())) {
-                    return false;
                 } else {
-                    p_312686_.getBrain().setMemory(MemoryModuleType.BREEZE_JUMP_TARGET, blockpos);
-                    return true;
+                    BlockState blockstate = p_338176_.getBlockState(blockpos.below());
+                    if (p_338459_.getType().isBlockDangerous(blockstate)) {
+                        return false;
+                    } else if (!BreezeUtil.hasLineOfSight(p_338459_, blockpos.getCenter())
+                        && !BreezeUtil.hasLineOfSight(p_338459_, blockpos.above(4).getCenter())) {
+                        return false;
+                    } else {
+                        p_338459_.getBrain().setMemory(MemoryModuleType.BREEZE_JUMP_TARGET, blockpos);
+                        return true;
+                    }
                 }
             }
         }
     }
 
+    protected boolean checkExtraStartConditions(ServerLevel p_312131_, Breeze p_312686_) {
+        return canRun(p_312131_, p_312686_);
+    }
+
     protected boolean canStillUse(ServerLevel p_312482_, Breeze p_312019_, long p_312448_) {
         return p_312019_.getPose() != Pose.STANDING && !p_312019_.getBrain().hasMemoryValue(MemoryModuleType.BREEZE_JUMP_COOLDOWN);
     }
@@ -115,7 +_,7 @@
         if (isFinishedInhaling(p_312923_)) {
             Vec3 vec3 = p_312923_.getBrain()
                 .getMemory(MemoryModuleType.BREEZE_JUMP_TARGET)
-                .flatMap(p_340135_ -> calculateOptimalJumpVector(p_312923_, p_312923_.getRandom(), Vec3.atBottomCenterOf(p_340135_)))
+                .flatMap(p_337853_ -> calculateOptimalJumpVector(p_312923_, p_312923_.getRandom(), Vec3.atBottomCenterOf(p_337853_)))
                 .orElse(null);
             if (vec3 == null) {
                 p_312923_.setPose(Pose.STANDING);
@@ -175,7 +_,7 @@
                 p_312597_, p_312597_.relative(Direction.UP, 10.0), ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, p_311909_
             );
             HitResult hitresult1 = p_311909_.level().clip(clipcontext1);
-            return hitresult1.getType() == HitResult.Type.BLOCK ? BlockPos.containing(hitresult.getLocation()).above() : null;
+            return hitresult1.getType() == HitResult.Type.BLOCK ? BlockPos.containing(hitresult1.getLocation()).above() : null;
         }
     }
 
@@ -190,7 +_,7 @@
     private static boolean canJumpFromCurrentPosition(ServerLevel p_312261_, Breeze p_312824_) {
         BlockPos blockpos = p_312824_.blockPosition();
 
-        for(int i = 1; i <= 4; ++i) {
+        for (int i = 1; i <= 4; i++) {
             BlockPos blockpos1 = blockpos.relative(Direction.UP, i);
             if (!p_312261_.getBlockState(blockpos1).isAir() && !p_312261_.getFluidState(blockpos1).is(FluidTags.WATER)) {
                 return false;
@@ -201,7 +_,7 @@
     }
 
     private static Optional<Vec3> calculateOptimalJumpVector(Breeze p_312651_, RandomSource p_312364_, Vec3 p_312758_) {
-        for(int i : Util.shuffledCopy(ALLOWED_ANGLES, p_312364_)) {
+        for (int i : Util.shuffledCopy(ALLOWED_ANGLES, p_312364_)) {
             Optional<Vec3> optional = LongJumpUtil.calculateJumpVectorForAngle(p_312651_, p_312758_, 1.4F, i, false);
             if (optional.isPresent()) {
                 return optional;
