--- a/net/minecraft/world/level/portal/PortalShape.java
+++ b/net/minecraft/world/level/portal/PortalShape.java
@@ -28,7 +_,7 @@
     public static final int MAX_WIDTH = 21;
     private static final int MIN_HEIGHT = 3;
     public static final int MAX_HEIGHT = 21;
-    private static final BlockBehaviour.StatePredicate FRAME = (p_77720_, p_77721_, p_77722_) -> p_77720_.is(Blocks.OBSIDIAN);
+    private static final BlockBehaviour.StatePredicate FRAME = net.neoforged.neoforge.common.extensions.IBlockStateExtension::isPortalFrame;
     private static final float SAFE_TRAVEL_MAX_ENTITY_XY = 4.0F;
     private static final double SAFE_TRAVEL_MAX_VERTICAL_DELTA = 1.0;
     private final LevelAccessor level;
@@ -75,7 +_,7 @@
     private BlockPos calculateBottomLeft(BlockPos p_77734_) {
         int i = Math.max(this.level.getMinBuildHeight(), p_77734_.getY() - 21);
 
-        while(p_77734_.getY() > i && isEmpty(this.level.getBlockState(p_77734_.below()))) {
+        while (p_77734_.getY() > i && isEmpty(this.level.getBlockState(p_77734_.below()))) {
             p_77734_ = p_77734_.below();
         }
 
@@ -92,7 +_,7 @@
     private int getDistanceUntilEdgeAboveFrame(BlockPos p_77736_, Direction p_77737_) {
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int i = 0; i <= 21; ++i) {
+        for (int i = 0; i <= 21; i++) {
             blockpos$mutableblockpos.set(p_77736_).move(p_77737_, i);
             BlockState blockstate = this.level.getBlockState(blockpos$mutableblockpos);
             if (!isEmpty(blockstate)) {
@@ -118,7 +_,7 @@
     }
 
     private boolean hasTopFrame(BlockPos.MutableBlockPos p_77731_, int p_77732_) {
-        for(int i = 0; i < this.width; ++i) {
+        for (int i = 0; i < this.width; i++) {
             BlockPos.MutableBlockPos blockpos$mutableblockpos = p_77731_.set(this.bottomLeft).move(Direction.UP, p_77732_).move(this.rightDir, i);
             if (!FRAME.test(this.level.getBlockState(blockpos$mutableblockpos), this.level, blockpos$mutableblockpos)) {
                 return false;
@@ -129,7 +_,7 @@
     }
 
     private int getDistanceUntilTop(BlockPos.MutableBlockPos p_77729_) {
-        for(int i = 0; i < 21; ++i) {
+        for (int i = 0; i < 21; i++) {
             p_77729_.set(this.bottomLeft).move(Direction.UP, i).move(this.rightDir, -1);
             if (!FRAME.test(this.level.getBlockState(p_77729_), this.level, p_77729_)) {
                 return i;
@@ -140,7 +_,7 @@
                 return i;
             }
 
-            for(int j = 0; j < this.width; ++j) {
+            for (int j = 0; j < this.width; j++) {
                 p_77729_.set(this.bottomLeft).move(Direction.UP, i).move(this.rightDir, j);
                 BlockState blockstate = this.level.getBlockState(p_77729_);
                 if (!isEmpty(blockstate)) {
@@ -148,7 +_,7 @@
                 }
 
                 if (blockstate.is(Blocks.NETHER_PORTAL)) {
-                    ++this.numPortalBlocks;
+                    this.numPortalBlocks++;
                 }
             }
         }
