--- a/config/inject/mcp/client/Start.java
+++ b/config/inject/mcp/client/Start.java
@@ -14,7 +_,7 @@
          * Working directory is used as gameDir if not provided
          */
         String assets = System.getenv().containsKey("assetDirectory") ? System.getenv("assetDirectory") : "assets";
-        Main.main(concat(new String[]{"--version", "mcp", "--accessToken", "0", "--assetsDir", assets, "--assetIndex", "15", "--userProperties", "{}"}, args));
+        Main.main(concat(new String[] { "--version", "mcp", "--accessToken", "0", "--assetsDir", assets, "--assetIndex", "16", "--userProperties", "{}" }, args));
     }
 
     public static <T> T[] concat(T[] first, T[] second)
