--- a/net/minecraft/world/level/chunk/BulkSectionAccess.java
+++ b/net/minecraft/world/level/chunk/BulkSectionAccess.java
@@ -29,7 +_,7 @@
                 this.lastSection = this.acquiredSections
                     .computeIfAbsent(
                         j,
-                        p_340277_ -> {
+                        p_156109_ -> {
                             ChunkAccess chunkaccess = this.level
                                 .getChunk(SectionPos.blockToSectionCoord(p_156105_.getX()), SectionPos.blockToSectionCoord(p_156105_.getZ()));
                             LevelChunkSection levelchunksection = chunkaccess.getSection(i);
@@ -60,7 +_,7 @@
 
     @Override
     public void close() {
-        for(LevelChunkSection levelchunksection : this.acquiredSections.values()) {
+        for (LevelChunkSection levelchunksection : this.acquiredSections.values()) {
             levelchunksection.release();
         }
     }
