--- a/net/minecraft/world/level/chunk/ChunkGenerator.java
+++ b/net/minecraft/world/level/chunk/ChunkGenerator.java
@@ -3,6 +_,7 @@
 import com.google.common.base.Suppliers;
 import com.mojang.datafixers.util.Pair;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import it.unimi.dsi.fastutil.ints.IntArraySet;
 import it.unimi.dsi.fastutil.ints.IntSet;
 import it.unimi.dsi.fastutil.longs.LongSet;
@@ -97,13 +_,13 @@
         );
     }
 
-    protected abstract Codec<? extends ChunkGenerator> codec();
+    protected abstract MapCodec<? extends ChunkGenerator> codec();
 
     public ChunkGeneratorStructureState createState(HolderLookup<StructureSet> p_256405_, RandomState p_256101_, long p_256018_) {
         return ChunkGeneratorStructureState.createForNormal(p_256101_, p_256018_, this.biomeSource, p_256405_);
     }
 
-    public Optional<ResourceKey<Codec<? extends ChunkGenerator>>> getTypeNameForDataFixer() {
+    public Optional<ResourceKey<MapCodec<? extends ChunkGenerator>>> getTypeNameForDataFixer() {
         return BuiltInRegistries.CHUNK_GENERATOR.getResourceKey(this.codec());
     }
 
@@ -133,9 +_,9 @@
         ChunkGeneratorStructureState chunkgeneratorstructurestate = p_223038_.getChunkSource().getGeneratorState();
         Map<StructurePlacement, Set<Holder<Structure>>> map = new Object2ObjectArrayMap<>();
 
-        for(Holder<Structure> holder : p_223039_) {
-            for(StructurePlacement structureplacement : chunkgeneratorstructurestate.getPlacementsForStructure(holder)) {
-                map.computeIfAbsent(structureplacement, p_223127_ -> new ObjectArraySet()).add(holder);
+        for (Holder<Structure> holder : p_223039_) {
+            for (StructurePlacement structureplacement : chunkgeneratorstructurestate.getPlacementsForStructure(holder)) {
+                map.computeIfAbsent(structureplacement, p_223127_ -> new ObjectArraySet<>()).add(holder);
             }
         }
 
@@ -147,9 +_,10 @@
             StructureManager structuremanager = p_223038_.structureManager();
             List<Entry<StructurePlacement, Set<Holder<Structure>>>> list = new ArrayList<>(map.size());
 
-            for(Entry<StructurePlacement, Set<Holder<Structure>>> entry : map.entrySet()) {
+            for (Entry<StructurePlacement, Set<Holder<Structure>>> entry : map.entrySet()) {
                 StructurePlacement structureplacement1 = entry.getKey();
-                if (structureplacement1 instanceof ConcentricRingsStructurePlacement concentricringsstructureplacement) {
+                if (structureplacement1 instanceof ConcentricRingsStructurePlacement) {
+                    ConcentricRingsStructurePlacement concentricringsstructureplacement = (ConcentricRingsStructurePlacement)structureplacement1;
                     Pair<BlockPos, Holder<Structure>> pair = this.getNearestGeneratedStructure(
                         entry.getValue(), p_223038_, structuremanager, p_223040_, p_223042_, concentricringsstructureplacement
                     );
@@ -170,10 +_,10 @@
                 int i = SectionPos.blockToSectionCoord(p_223040_.getX());
                 int j = SectionPos.blockToSectionCoord(p_223040_.getZ());
 
-                for(int k = 0; k <= p_223041_; ++k) {
+                for (int k = 0; k <= p_223041_; k++) {
                     boolean flag = false;
 
-                    for(Entry<StructurePlacement, Set<Holder<Structure>>> entry1 : list) {
+                    for (Entry<StructurePlacement, Set<Holder<Structure>>> entry1 : list) {
                         RandomSpreadStructurePlacement randomspreadstructureplacement = (RandomSpreadStructurePlacement)entry1.getKey();
                         Pair<BlockPos, Holder<Structure>> pair1 = getNearestGeneratedStructure(
                             entry1.getValue(),
@@ -223,7 +_,7 @@
             double d0 = Double.MAX_VALUE;
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(ChunkPos chunkpos : list) {
+            for (ChunkPos chunkpos : list) {
                 blockpos$mutableblockpos.set(SectionPos.sectionToBlockCoord(chunkpos.x, 8), 32, SectionPos.sectionToBlockCoord(chunkpos.z, 8));
                 double d1 = blockpos$mutableblockpos.distSqr(p_223185_);
                 boolean flag = pair == null || d1 < d0;
@@ -254,10 +_,10 @@
     ) {
         int i = p_223197_.spacing();
 
-        for(int j = -p_223194_; j <= p_223194_; ++j) {
+        for (int j = -p_223194_; j <= p_223194_; j++) {
             boolean flag = j == -p_223194_ || j == p_223194_;
 
-            for(int k = -p_223194_; k <= p_223194_; ++k) {
+            for (int k = -p_223194_; k <= p_223194_; k++) {
                 boolean flag1 = k == -p_223194_ || k == p_223194_;
                 if (flag || flag1) {
                     int l = p_223192_ + i * j;
@@ -283,7 +_,7 @@
         StructurePlacement p_223203_,
         ChunkPos p_223204_
     ) {
-        for(Holder<Structure> holder : p_223199_) {
+        for (Holder<Structure> holder : p_223199_) {
             StructureCheckResult structurecheckresult = p_223201_.checkStructurePresence(p_223204_, holder.value(), p_223203_, p_223202_);
             if (structurecheckresult != StructureCheckResult.START_NOT_PRESENT) {
                 if (!p_223202_ && structurecheckresult == StructureCheckResult.START_PRESENT) {
@@ -321,10 +_,10 @@
             WorldgenRandom worldgenrandom = new WorldgenRandom(new XoroshiroRandomSource(RandomSupport.generateUniqueSeed()));
             long i = worldgenrandom.setDecorationSeed(p_223087_.getSeed(), blockpos.getX(), blockpos.getZ());
             Set<Holder<Biome>> set = new ObjectArraySet<>();
-            ChunkPos.rangeClosed(sectionpos.chunk(), 1).forEach(p_340280_ -> {
-                ChunkAccess chunkaccess = p_223087_.getChunk(p_340280_.x, p_340280_.z);
+            ChunkPos.rangeClosed(sectionpos.chunk(), 1).forEach(p_223093_ -> {
+                ChunkAccess chunkaccess = p_223087_.getChunk(p_223093_.x, p_223093_.z);
 
-                for(LevelChunkSection levelchunksection : chunkaccess.getSections()) {
+                for (LevelChunkSection levelchunksection : chunkaccess.getSections()) {
                     levelchunksection.getBiomes().getAll(set::add);
                 }
             });
@@ -335,10 +_,10 @@
                 Registry<PlacedFeature> registry1 = p_223087_.registryAccess().registryOrThrow(Registries.PLACED_FEATURE);
                 int i1 = Math.max(GenerationStep.Decoration.values().length, j);
 
-                for(int k = 0; k < i1; ++k) {
+                for (int k = 0; k < i1; k++) {
                     int l = 0;
                     if (p_223089_.shouldGenerateStructures()) {
-                        for(Structure structure : map.getOrDefault(k, Collections.emptyList())) {
+                        for (Structure structure : map.getOrDefault(k, Collections.emptyList())) {
                             worldgenrandom.setFeatureSeed(i, l, k);
                             Supplier<String> supplier = () -> registry.getResourceKey(structure).map(Object::toString).orElseGet(structure::toString);
 
@@ -354,14 +_,14 @@
                                 throw new ReportedException(crashreport1);
                             }
 
-                            ++l;
+                            l++;
                         }
                     }
 
                     if (k < j) {
                         IntSet intset = new IntArraySet();
 
-                        for(Holder<Biome> holder : set) {
+                        for (Holder<Biome> holder : set) {
                             List<HolderSet<PlacedFeature>> list1 = this.generationSettingsGetter.apply(holder).features();
                             if (k < list1.size()) {
                                 HolderSet<PlacedFeature> holderset = list1.get(k);
@@ -377,7 +_,7 @@
                         Arrays.sort(aint);
                         FeatureSorter.StepFeatureData featuresorter$stepfeaturedata = list.get(k);
 
-                        for(int k1 = 0; k1 < j1; ++k1) {
+                        for (int k1 = 0; k1 < j1; k1++) {
                             int l1 = aint[k1];
                             PlacedFeature placedfeature = featuresorter$stepfeaturedata.features().get(l1);
                             Supplier<String> supplier1 = () -> registry1.getResourceKey(placedfeature).map(Object::toString).orElseGet(placedfeature::toString);
@@ -433,7 +_,7 @@
     ) {
         Map<Structure, LongSet> map = p_223135_.getAllStructuresAt(p_223137_);
 
-        for(Entry<Structure, LongSet> entry : map.entrySet()) {
+        for (Entry<Structure, LongSet> entry : map.entrySet()) {
             Structure structure = entry.getKey();
             StructureSpawnOverride structurespawnoverride = structure.spawnOverrides().get(p_223136_);
             if (structurespawnoverride != null) {
@@ -467,7 +_,7 @@
                     StructurePlacement structureplacement = p_255564_.value().placement();
                     List<StructureSet.StructureSelectionEntry> list = p_255564_.value().structures();
         
-                    for(StructureSet.StructureSelectionEntry structureset$structureselectionentry : list) {
+                    for (StructureSet.StructureSelectionEntry structureset$structureselectionentry : list) {
                         StructureStart structurestart = p_255934_.getStartForStructure(
                             sectionpos, structureset$structureselectionentry.structure().value(), p_255767_
                         );
@@ -488,21 +_,21 @@
                             worldgenrandom.setLargeFeatureSeed(p_256505_.getLevelSeed(), chunkpos.x, chunkpos.z);
                             int i = 0;
         
-                            for(StructureSet.StructureSelectionEntry structureset$structureselectionentry1 : arraylist) {
+                            for (StructureSet.StructureSelectionEntry structureset$structureselectionentry1 : arraylist) {
                                 i += structureset$structureselectionentry1.weight();
                             }
         
-                            while(!arraylist.isEmpty()) {
+                            while (!arraylist.isEmpty()) {
                                 int j = worldgenrandom.nextInt(i);
                                 int k = 0;
         
-                                for(StructureSet.StructureSelectionEntry structureset$structureselectionentry2 : arraylist) {
+                                for (StructureSet.StructureSelectionEntry structureset$structureselectionentry2 : arraylist) {
                                     j -= structureset$structureselectionentry2.weight();
                                     if (j < 0) {
                                         break;
                                     }
         
-                                    ++k;
+                                    k++;
                                 }
         
                                 StructureSet.StructureSelectionEntry structureset$structureselectionentry3 = arraylist.get(k);
@@ -545,7 +_,7 @@
         HolderSet<Biome> holderset = structure.biomes();
         Predicate<Holder<Biome>> predicate = holderset::contains;
         StructureStart structurestart = structure.generate(
-            p_223107_, this, this.biomeSource, p_223108_, p_223109_, p_223110_, p_223112_, i, p_223111_, predicate, structure.densityChecks()
+            p_223107_, this, this.biomeSource, p_223108_, p_223109_, p_223110_, p_223112_, i, p_223111_, predicate
         );
         if (structurestart.isValid()) {
             p_223106_.setStartForStructure(p_223113_, structure, structurestart, p_223111_);
@@ -569,11 +_,11 @@
         int i1 = chunkpos.getMinBlockZ();
         SectionPos sectionpos = SectionPos.bottomOf(p_223079_);
 
-        for(int j1 = j - 8; j1 <= j + 8; ++j1) {
-            for(int k1 = k - 8; k1 <= k + 8; ++k1) {
+        for (int j1 = j - 8; j1 <= j + 8; j1++) {
+            for (int k1 = k - 8; k1 <= k + 8; k1++) {
                 long l1 = ChunkPos.asLong(j1, k1);
 
-                for(StructureStart structurestart : p_223077_.getChunk(j1, k1).getAllStarts().values()) {
+                for (StructureStart structurestart : p_223077_.getChunk(j1, k1).getAllStarts().values()) {
                     try {
                         if (structurestart.isValid() && structurestart.getBoundingBox().intersects(l, i1, l + 15, i1 + 15)) {
                             p_223078_.addReferenceForStructure(sectionpos, structurestart.getStructure(), l1, p_223079_);
