--- a/net/minecraft/world/level/chunk/LinearPalette.java
+++ b/net/minecraft/world/level/chunk/LinearPalette.java
@@ -23,7 +_,7 @@
             p_188018_.size() <= this.values.length, "Can't initialize LinearPalette of size %d with %d entries", this.values.length, p_188018_.size()
         );
 
-        for(int i = 0; i < p_188018_.size(); ++i) {
+        for (int i = 0; i < p_188018_.size(); i++) {
             this.values[i] = p_188018_.get(i);
         }
 
@@ -44,7 +_,7 @@
 
     @Override
     public int idFor(T p_63040_) {
-        for(int i = 0; i < this.size; ++i) {
+        for (int i = 0; i < this.size; i++) {
             if (this.values[i] == p_63040_) {
                 return i;
             }
@@ -53,7 +_,7 @@
         int j = this.size;
         if (j < this.values.length) {
             this.values[j] = p_63040_;
-            ++this.size;
+            this.size++;
             return j;
         } else {
             return this.resizeHandler.onResize(this.bits + 1, p_63040_);
@@ -62,7 +_,7 @@
 
     @Override
     public boolean maybeHas(Predicate<T> p_63042_) {
-        for(int i = 0; i < this.size; ++i) {
+        for (int i = 0; i < this.size; i++) {
             if (p_63042_.test(this.values[i])) {
                 return true;
             }
@@ -84,7 +_,7 @@
     public void read(FriendlyByteBuf p_63046_) {
         this.size = p_63046_.readVarInt();
 
-        for(int i = 0; i < this.size; ++i) {
+        for (int i = 0; i < this.size; i++) {
             this.values[i] = this.registry.byIdOrThrow(p_63046_.readVarInt());
         }
     }
@@ -93,7 +_,7 @@
     public void write(FriendlyByteBuf p_63049_) {
         p_63049_.writeVarInt(this.size);
 
-        for(int i = 0; i < this.size; ++i) {
+        for (int i = 0; i < this.size; i++) {
             p_63049_.writeVarInt(this.registry.getId(this.values[i]));
         }
     }
@@ -102,7 +_,7 @@
     public int getSerializedSize() {
         int i = VarInt.getByteSize(this.getSize());
 
-        for(int j = 0; j < this.getSize(); ++j) {
+        for (int j = 0; j < this.getSize(); j++) {
             i += VarInt.getByteSize(this.registry.getId(this.values[j]));
         }
 
