--- a/net/minecraft/world/level/chunk/ProtoChunk.java
+++ b/net/minecraft/world/level/chunk/ProtoChunk.java
@@ -136,7 +_,7 @@
                         this.lightEngine.updateSectionStatus(pos, flag1);
                     }
 
-                    if (LightEngine.hasDifferentLightProperties(blockstate, state)) {
+                    if (LightEngine.hasDifferentLightProperties(this, pos, blockstate, state)) {
                         this.skyLightSources.update(this, i1, j, k1);
                         this.lightEngine.checkBlock(pos);
                     }
