--- a/net/minecraft/world/level/chunk/UpgradeData.java
+++ b/net/minecraft/world/level/chunk/UpgradeData.java
@@ -62,7 +_,7 @@
         if (p_156508_.contains("Indices", 10)) {
             CompoundTag compoundtag = p_156508_.getCompound("Indices");
 
-            for(int i = 0; i < this.index.length; ++i) {
+            for (int i = 0; i < this.index.length; i++) {
                 String s = String.valueOf(i);
                 if (compoundtag.contains(s, 11)) {
                     this.index[i] = compoundtag.getIntArray(s);
@@ -72,7 +_,7 @@
 
         int j = p_156508_.getInt("Sides");
 
-        for(Direction8 direction8 : Direction8.values()) {
+        for (Direction8 direction8 : Direction8.values()) {
             if ((j & 1 << direction8.ordinal()) != 0) {
                 this.sides.add(direction8);
             }
@@ -94,7 +_,7 @@
 
     private static <T> void loadTicks(CompoundTag p_208133_, String p_208134_, Function<String, Optional<T>> p_208135_, List<SavedTick<T>> p_208136_) {
         if (p_208133_.contains(p_208134_, 9)) {
-            for(Tag tag : p_208133_.getList(p_208134_, 10)) {
+            for (Tag tag : p_208133_.getList(p_208134_, 10)) {
                 SavedTick.loadTick((CompoundTag)tag, p_208135_).ifPresent(p_208136_::add);
             }
         }
@@ -103,7 +_,7 @@
     public void upgrade(LevelChunk p_63342_) {
         this.upgradeInside(p_63342_);
 
-        for(Direction8 direction8 : DIRECTIONS) {
+        for (Direction8 direction8 : DIRECTIONS) {
             upgradeSides(p_63342_, direction8);
         }
 
@@ -138,11 +_,11 @@
             Direction[] adirection = Direction.values();
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(BlockPos blockpos : BlockPos.betweenClosed(k, level.getMinBuildHeight(), i1, l, level.getMaxBuildHeight() - 1, j1)) {
+            for (BlockPos blockpos : BlockPos.betweenClosed(k, level.getMinBuildHeight(), i1, l, level.getMaxBuildHeight() - 1, j1)) {
                 BlockState blockstate = level.getBlockState(blockpos);
                 BlockState blockstate1 = blockstate;
 
-                for(Direction direction : adirection) {
+                for (Direction direction : adirection) {
                     blockpos$mutableblockpos.setWithOffset(blockpos, direction);
                     blockstate1 = updateState(blockstate1, direction, level, blockpos, blockpos$mutableblockpos);
                 }
@@ -163,7 +_,7 @@
         ChunkPos chunkpos = p_63348_.getPos();
         LevelAccessor levelaccessor = p_63348_.getLevel();
 
-        for(int i = 0; i < this.index.length; ++i) {
+        for (int i = 0; i < this.index.length; i++) {
             LevelChunkSection levelchunksection = p_63348_.getSection(i);
             int[] aint = this.index[i];
             this.index[i] = null;
@@ -173,7 +_,7 @@
                 int j = p_63348_.getSectionYFromSectionIndex(i);
                 int k = SectionPos.sectionToBlockCoord(j);
 
-                for(int l : aint) {
+                for (int l : aint) {
                     int i1 = l & 15;
                     int j1 = l >> 8 & 15;
                     int k1 = l >> 4 & 15;
@@ -181,7 +_,7 @@
                     BlockState blockstate = palettedcontainer.get(l);
                     BlockState blockstate1 = blockstate;
 
-                    for(Direction direction : adirection) {
+                    for (Direction direction : adirection) {
                         blockpos$mutableblockpos1.setWithOffset(blockpos$mutableblockpos, direction);
                         if (SectionPos.blockToSectionCoord(blockpos$mutableblockpos.getX()) == chunkpos.x
                             && SectionPos.blockToSectionCoord(blockpos$mutableblockpos.getZ()) == chunkpos.z) {
@@ -194,7 +_,7 @@
             }
         }
 
-        for(int l1 = 0; l1 < this.index.length; ++l1) {
+        for (int l1 = 0; l1 < this.index.length; l1++) {
             if (this.index[l1] != null) {
                 LOGGER.warn("Discarding update data for section {} for chunk ({} {})", levelaccessor.getSectionYFromSectionIndex(l1), chunkpos.x, chunkpos.z);
             }
@@ -204,7 +_,7 @@
     }
 
     public boolean isEmpty() {
-        for(int[] aint : this.index) {
+        for (int[] aint : this.index) {
             if (aint != null) {
                 return false;
             }
@@ -217,7 +_,7 @@
         CompoundTag compoundtag = new CompoundTag();
         CompoundTag compoundtag1 = new CompoundTag();
 
-        for(int i = 0; i < this.index.length; ++i) {
+        for (int i = 0; i < this.index.length; i++) {
             String s = String.valueOf(i);
             if (this.index[i] != null && this.index[i].length != 0) {
                 compoundtag1.putIntArray(s, this.index[i]);
@@ -230,7 +_,7 @@
 
         int j = 0;
 
-        for(Direction8 direction8 : this.sides) {
+        for (Direction8 direction8 : this.sides) {
             j |= 1 << direction8.ordinal();
         }
 
@@ -283,7 +_,6 @@
             Blocks.DRAGON_EGG,
             Blocks.GRAVEL,
             Blocks.SAND,
-            Blocks.GRAVTATER,
             Blocks.RED_SAND,
             Blocks.OAK_SIGN,
             Blocks.SPRUCE_SIGN,
@@ -356,7 +_,6 @@
         },
         LEAVES(
             true,
-            Blocks.POTATO_LEAVES,
             Blocks.ACACIA_LEAVES,
             Blocks.CHERRY_LEAVES,
             Blocks.BIRCH_LEAVES,
@@ -376,7 +_,7 @@
                     int i = blockstate.getValue(BlockStateProperties.DISTANCE);
                     List<ObjectSet<BlockPos>> list = this.queue.get();
                     if (list.isEmpty()) {
-                        for(int j = 0; j < 7; ++j) {
+                        for (int j = 0; j < 7; j++) {
                             list.add(new ObjectOpenHashSet<>());
                         }
                     }
@@ -392,17 +_,17 @@
                 BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
                 List<ObjectSet<BlockPos>> list = this.queue.get();
 
-                for(int i = 2; i < list.size(); ++i) {
+                for (int i = 2; i < list.size(); i++) {
                     int j = i - 1;
                     ObjectSet<BlockPos> objectset = list.get(j);
                     ObjectSet<BlockPos> objectset1 = list.get(i);
 
-                    for(BlockPos blockpos : objectset) {
+                    for (BlockPos blockpos : objectset) {
                         BlockState blockstate = p_63430_.getBlockState(blockpos);
                         if (blockstate.getValue(BlockStateProperties.DISTANCE) >= j) {
                             p_63430_.setBlock(blockpos, blockstate.setValue(BlockStateProperties.DISTANCE, Integer.valueOf(j)), 18);
                             if (i != 7) {
-                                for(Direction direction : DIRECTIONS) {
+                                for (Direction direction : DIRECTIONS) {
                                     blockpos$mutableblockpos.setWithOffset(blockpos, direction);
                                     BlockState blockstate1 = p_63430_.getBlockState(blockpos$mutableblockpos);
                                     if (blockstate1.hasProperty(BlockStateProperties.DISTANCE) && blockstate.getValue(BlockStateProperties.DISTANCE) > i) {
@@ -442,7 +_,7 @@
         }
 
         BlockFixers(boolean p_63369_, Block... p_63370_) {
-            for(Block block : p_63370_) {
+            for (Block block : p_63370_) {
                 UpgradeData.MAP.put(block, this);
             }
 
