--- a/net/minecraft/client/sounds/SoundBufferLibrary.java
+++ b/net/minecraft/client/sounds/SoundBufferLibrary.java
@@ -62,8 +_,6 @@
     }
 
     public CompletableFuture<?> preload(Collection<Sound> p_120199_) {
-        return CompletableFuture.allOf(
-            p_120199_.stream().map(p_120197_ -> this.getCompleteBuffer(p_120197_.getPath())).toArray(p_120195_ -> new CompletableFuture[p_120195_])
-        );
+        return CompletableFuture.allOf(p_120199_.stream().map(p_120197_ -> this.getCompleteBuffer(p_120197_.getPath())).toArray(CompletableFuture[]::new));
     }
 }
