--- a/net/minecraft/world/level/storage/loot/providers/number/BinomialDistributionGenerator.java
+++ b/net/minecraft/world/level/storage/loot/providers/number/BinomialDistributionGenerator.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.storage.loot.providers.number;
 
 import com.google.common.collect.Sets;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Set;
@@ -10,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 
 public record BinomialDistributionGenerator(NumberProvider n, NumberProvider p) implements NumberProvider {
-    public static final Codec<BinomialDistributionGenerator> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<BinomialDistributionGenerator> CODEC = RecordCodecBuilder.mapCodec(
         p_299136_ -> p_299136_.group(
                     NumberProviders.CODEC.fieldOf("n").forGetter(BinomialDistributionGenerator::n),
                     NumberProviders.CODEC.fieldOf("p").forGetter(BinomialDistributionGenerator::p)
@@ -30,9 +_,9 @@
         RandomSource randomsource = p_165663_.getRandom();
         int j = 0;
 
-        for(int k = 0; k < i; ++k) {
+        for (int k = 0; k < i; k++) {
             if (randomsource.nextFloat() < f) {
-                ++j;
+                j++;
             }
         }
 
