--- a/net/minecraft/world/level/storage/loot/providers/number/ConstantValue.java
+++ b/net/minecraft/world/level/storage/loot/providers/number/ConstantValue.java
@@ -1,12 +_,13 @@
 package net.minecraft.world.level.storage.loot.providers.number;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.world.level.storage.loot.LootContext;
 
 public record ConstantValue(float value) implements NumberProvider {
-    public static final Codec<ConstantValue> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<ConstantValue> CODEC = RecordCodecBuilder.mapCodec(
         p_299242_ -> p_299242_.group(Codec.FLOAT.fieldOf("value").forGetter(ConstantValue::value)).apply(p_299242_, ConstantValue::new)
     );
     public static final Codec<ConstantValue> INLINE_CODEC = Codec.FLOAT.xmap(ConstantValue::new, ConstantValue::value);
@@ -29,10 +_,8 @@
     public boolean equals(Object p_165697_) {
         if (this == p_165697_) {
             return true;
-        } else if (p_165697_ != null && this.getClass() == p_165697_.getClass()) {
-            return Float.compare(((ConstantValue)p_165697_).value, this.value) == 0;
         } else {
-            return false;
+            return p_165697_ != null && this.getClass() == p_165697_.getClass() ? Float.compare(((ConstantValue)p_165697_).value, this.value) == 0 : false;
         }
     }
 
