--- a/net/minecraft/world/level/storage/loot/providers/number/ScoreboardValue.java
+++ b/net/minecraft/world/level/storage/loot/providers/number/ScoreboardValue.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.storage.loot.providers.number;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Set;
@@ -15,7 +_,7 @@
 import net.minecraft.world.scores.Scoreboard;
 
 public record ScoreboardValue(ScoreboardNameProvider target, String score, float scale) implements NumberProvider {
-    public static final Codec<ScoreboardValue> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<ScoreboardValue> CODEC = RecordCodecBuilder.mapCodec(
         p_298582_ -> p_298582_.group(
                     ScoreboardNameProviders.CODEC.fieldOf("target").forGetter(ScoreboardValue::target),
                     Codec.STRING.fieldOf("score").forGetter(ScoreboardValue::score),
