--- a/net/minecraft/world/entity/ambient/Bat.java
+++ b/net/minecraft/world/entity/ambient/Bat.java
@@ -218,7 +_,7 @@
     }
 
     public static boolean checkBatSpawnRules(
-        EntityType<? extends Bat> p_218099_, LevelAccessor p_218100_, MobSpawnType p_218101_, BlockPos p_218102_, RandomSource p_218103_
+        EntityType<Bat> p_218099_, LevelAccessor p_218100_, MobSpawnType p_218101_, BlockPos p_218102_, RandomSource p_218103_
     ) {
         if (p_218102_.getY() >= p_218100_.getSeaLevel()) {
             return false;
