--- a/net/minecraft/core/component/DataComponentType.java
+++ b/net/minecraft/core/component/DataComponentType.java
@@ -1,21 +_,29 @@
 package net.minecraft.core.component;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import java.util.Map;
 import java.util.Objects;
 import javax.annotation.Nullable;
 import net.minecraft.Util;
+import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
-import net.minecraft.util.ExtraCodecs;
 
 public interface DataComponentType<T> {
-    Codec<DataComponentType<?>> CODEC = ExtraCodecs.lazyInitializedCodec(() -> BuiltInRegistries.DATA_COMPONENT_TYPE.byNameCodec());
+    Codec<DataComponentType<?>> CODEC = Codec.lazyInitialized(() -> BuiltInRegistries.DATA_COMPONENT_TYPE.byNameCodec());
     StreamCodec<RegistryFriendlyByteBuf, DataComponentType<?>> STREAM_CODEC = StreamCodec.recursive(
         p_330812_ -> ByteBufCodecs.registry(Registries.DATA_COMPONENT_TYPE)
     );
+    Codec<DataComponentType<?>> PERSISTENT_CODEC = CODEC.validate(
+        p_337456_ -> p_337456_.isTransient()
+                ? DataResult.error(() -> "Encountered transient component " + BuiltInRegistries.DATA_COMPONENT_TYPE.getKey(p_337456_))
+                : DataResult.success(p_337456_)
+    );
+    Codec<Map<DataComponentType<?>, Object>> VALUE_MAP_CODEC = Codec.dispatchedMap(PERSISTENT_CODEC, DataComponentType::codecOrThrow);
 
     static <T> DataComponentType.Builder<T> builder() {
         return new DataComponentType.Builder<>();
