--- a/net/minecraft/core/component/DataComponents.java
+++ b/net/minecraft/core/component/DataComponents.java
@@ -1,8 +_,6 @@
 package net.minecraft.core.component;
 
 import com.mojang.serialization.Codec;
-import it.unimi.dsi.fastutil.ints.Int2IntMap;
-import it.unimi.dsi.fastutil.ints.IntIntPair;
 import java.util.List;
 import java.util.function.UnaryOperator;
 import net.minecraft.core.Holder;
@@ -41,21 +_,15 @@
 import net.minecraft.world.item.component.MapPostProcessing;
 import net.minecraft.world.item.component.ResolvableProfile;
 import net.minecraft.world.item.component.SeededContainerLoot;
-import net.minecraft.world.item.component.SnekComponent;
 import net.minecraft.world.item.component.SuspiciousStewEffects;
 import net.minecraft.world.item.component.Tool;
 import net.minecraft.world.item.component.Unbreakable;
 import net.minecraft.world.item.component.WritableBookContent;
 import net.minecraft.world.item.component.WrittenBookContent;
-import net.minecraft.world.item.component.XpComponent;
 import net.minecraft.world.item.enchantment.ItemEnchantments;
 import net.minecraft.world.level.block.entity.BannerPatternLayers;
 import net.minecraft.world.level.block.entity.BeehiveBlockEntity;
-import net.minecraft.world.level.block.entity.FletchingBlockEntity;
-import net.minecraft.world.level.block.entity.InventoryHeatComponent;
-import net.minecraft.world.level.block.entity.LubricationComponent;
 import net.minecraft.world.level.block.entity.PotDecorations;
-import net.minecraft.world.level.block.entity.PotatoBaneComponent;
 import net.minecraft.world.level.saveddata.maps.MapId;
 
 public class DataComponents {
@@ -75,6 +_,9 @@
     public static final DataComponentType<Component> CUSTOM_NAME = register(
         "custom_name", p_331838_ -> p_331838_.persistent(ComponentSerialization.FLAT_CODEC).networkSynchronized(ComponentSerialization.STREAM_CODEC)
     );
+    public static final DataComponentType<Component> ITEM_NAME = register(
+        "item_name", p_337457_ -> p_337457_.persistent(ComponentSerialization.FLAT_CODEC).networkSynchronized(ComponentSerialization.STREAM_CODEC)
+    );
     public static final DataComponentType<ItemLore> LORE = register(
         "lore", p_330480_ -> p_330480_.persistent(ItemLore.CODEC).networkSynchronized(ItemLore.STREAM_CODEC)
     );
@@ -85,10 +_,10 @@
         "enchantments", p_330612_ -> p_330612_.persistent(ItemEnchantments.CODEC).networkSynchronized(ItemEnchantments.STREAM_CODEC)
     );
     public static final DataComponentType<AdventureModePredicate> CAN_PLACE_ON = register(
-        "can_place_on", p_330922_ -> p_330922_.persistent(AdventureModePredicate.CODEC).networkSynchronized(AdventureModePredicate.STREAM_CODEC)
+        "can_place_on", p_330714_ -> p_330714_.persistent(AdventureModePredicate.CODEC).networkSynchronized(AdventureModePredicate.STREAM_CODEC)
     );
     public static final DataComponentType<AdventureModePredicate> CAN_BREAK = register(
-        "can_break", p_330714_ -> p_330714_.persistent(AdventureModePredicate.CODEC).networkSynchronized(AdventureModePredicate.STREAM_CODEC)
+        "can_break", p_330922_ -> p_330922_.persistent(AdventureModePredicate.CODEC).networkSynchronized(AdventureModePredicate.STREAM_CODEC)
     );
     public static final DataComponentType<ItemAttributeModifiers> ATTRIBUTE_MODIFIERS = register(
         "attribute_modifiers", p_332098_ -> p_332098_.persistent(ItemAttributeModifiers.CODEC).networkSynchronized(ItemAttributeModifiers.STREAM_CODEC)
@@ -164,12 +_,6 @@
     public static final DataComponentType<CustomData> ENTITY_DATA = register(
         "entity_data", p_332024_ -> p_332024_.persistent(CustomData.CODEC_WITH_ID).networkSynchronized(CustomData.STREAM_CODEC)
     );
-    public static final DataComponentType<FletchingBlockEntity.Fletching> FLETCHING = register(
-        "fletching", p_339555_ -> p_339555_.persistent(FletchingBlockEntity.Fletching.CODEC).networkSynchronized(FletchingBlockEntity.Fletching.STREAM_CODEC)
-    );
-    public static final DataComponentType<FletchingBlockEntity.Resin> RESIN = register(
-        "resin", p_339554_ -> p_339554_.persistent(FletchingBlockEntity.Resin.CODEC).networkSynchronized(FletchingBlockEntity.Resin.STREAM_CODEC)
-    );
     public static final DataComponentType<CustomData> BUCKET_ENTITY_DATA = register(
         "bucket_entity_data", p_331109_ -> p_331109_.persistent(CustomData.CODEC).networkSynchronized(CustomData.STREAM_CODEC)
     );
@@ -179,6 +_,9 @@
     public static final DataComponentType<Holder<Instrument>> INSTRUMENT = register(
         "instrument", p_331450_ -> p_331450_.persistent(Instrument.CODEC).networkSynchronized(Instrument.STREAM_CODEC)
     );
+    public static final DataComponentType<Integer> OMINOUS_BOTTLE_AMPLIFIER = register(
+        "ominous_bottle_amplifier", p_337458_ -> p_337458_.persistent(ExtraCodecs.intRange(0, 4)).networkSynchronized(ByteBufCodecs.VAR_INT)
+    );
     public static final DataComponentType<List<ResourceLocation>> RECIPES = register(
         "recipes", p_330709_ -> p_330709_.persistent(ResourceLocation.CODEC.listOf())
     );
@@ -220,42 +_,6 @@
     public static final DataComponentType<LockCode> LOCK = register("lock", p_330909_ -> p_330909_.persistent(LockCode.CODEC));
     public static final DataComponentType<SeededContainerLoot> CONTAINER_LOOT = register(
         "container_loot", p_331929_ -> p_331929_.persistent(SeededContainerLoot.CODEC)
-    );
-    public static final DataComponentType<XpComponent> XP = register(
-        "xp", p_339553_ -> p_339553_.persistent(XpComponent.CODEC).networkSynchronized(XpComponent.STREAM_CODEC)
-    );
-    public static final DataComponentType<SnekComponent> SNEK = register(
-        "snek", p_339548_ -> p_339548_.persistent(SnekComponent.CODEC).networkSynchronized(SnekComponent.STREAM_CODEC)
-    );
-    public static final DataComponentType<Boolean> HOVERED = register(
-        "hovered", p_339551_ -> p_339551_.persistent(Codec.BOOL).networkSynchronized(ByteBufCodecs.BOOL)
-    );
-    public static final DataComponentType<Integer> CLICKS = register(
-        "clicks", p_339552_ -> p_339552_.persistent(Codec.INT).networkSynchronized(ByteBufCodecs.INT)
-    );
-    public static final DataComponentType<Integer> VIEWS = register(
-        "views", p_339556_ -> p_339556_.persistent(Codec.INT).networkSynchronized(ByteBufCodecs.INT)
-    );
-    public static final DataComponentType<Integer> UNDERCOVER_ID = register(
-        "undercover_id", p_339546_ -> p_339546_.persistent(Codec.INT).networkSynchronized(ByteBufCodecs.INT)
-    );
-    public static final DataComponentType<Int2IntMap> CONTACTS_MESSAGES = register(
-        "contacts_messages", p_339549_ -> p_339549_.persistent(ExtraCodecs.int2IntMap()).networkSynchronized(ByteBufCodecs.fromCodec(ExtraCodecs.int2IntMap()))
-    );
-    public static final DataComponentType<IntIntPair> SECRET_MESSAGE = register(
-        "secret_message", p_339545_ -> p_339545_.persistent(ExtraCodecs.intIntPair()).networkSynchronized(ByteBufCodecs.fromCodec(ExtraCodecs.intIntPair()))
-    );
-    public static final DataComponentType<LubricationComponent> LUBRICATION = register(
-        "lubrication", p_339547_ -> p_339547_.persistent(LubricationComponent.CODEC).networkSynchronized(LubricationComponent.STREAM_CODEC)
-    );
-    public static final DataComponentType<Boolean> EXPLICIT_FOIL = register(
-        "explicit_foil", p_339544_ -> p_339544_.persistent(Codec.BOOL).networkSynchronized(ByteBufCodecs.BOOL)
-    );
-    public static final DataComponentType<InventoryHeatComponent> INVENTORY_HEAT = register(
-        "heat", p_339543_ -> p_339543_.persistent(InventoryHeatComponent.CODEC).networkSynchronized(InventoryHeatComponent.STREAM_CODEC)
-    );
-    public static final DataComponentType<PotatoBaneComponent> POTATO_BANE = register(
-        "potato_bane", p_339550_ -> p_339550_.persistent(PotatoBaneComponent.CODEC).networkSynchronized(PotatoBaneComponent.STREAM_CODEC)
     );
     public static final DataComponentMap COMMON_ITEM_COMPONENTS = DataComponentMap.builder()
         .set(MAX_STACK_SIZE, 64)
