--- a/net/minecraft/world/level/block/state/predicate/BlockStatePredicate.java
+++ b/net/minecraft/world/level/block/state/predicate/BlockStatePredicate.java
@@ -28,7 +_,7 @@
             if (this.properties.isEmpty()) {
                 return true;
             } else {
-                for(Entry<Property<?>, Predicate<Object>> entry : this.properties.entrySet()) {
+                for (Entry<Property<?>, Predicate<Object>> entry : this.properties.entrySet()) {
                     if (!this.applies(p_61290_, entry.getKey(), entry.getValue())) {
                         return false;
                     }
