--- a/net/minecraft/client/gui/screens/LevelLoadingScreen.java
+++ b/net/minecraft/client/gui/screens/LevelLoadingScreen.java
@@ -104,8 +_,8 @@
                 p_283467_.fill(p_96152_ - l1, p_96153_ + l1 - 1, p_96152_ + l1, p_96153_ + l1, -16772609);
             }
 
-            for(int j2 = 0; j2 < l; ++j2) {
-                for(int k2 = 0; k2 < l; ++k2) {
+            for (int j2 = 0; j2 < l; j2++) {
+                for (int k2 = 0; k2 < l; k2++) {
                     ChunkStatus chunkstatus = p_96151_.getStatus(j2, k2);
                     int l2 = j1 + j2 * i;
                     int i3 = k1 + k2 * i;
