--- a/net/minecraft/client/gui/screens/OptionsScreen.java
+++ b/net/minecraft/client/gui/screens/OptionsScreen.java
@@ -8,6 +_,7 @@
 import net.minecraft.client.gui.components.CycleButton;
 import net.minecraft.client.gui.components.LockIconButton;
 import net.minecraft.client.gui.components.StringWidget;
+import net.minecraft.client.gui.components.Tooltip;
 import net.minecraft.client.gui.layouts.EqualSpacingLayout;
 import net.minecraft.client.gui.layouts.GridLayout;
 import net.minecraft.client.gui.layouts.HeaderAndFooterLayout;
@@ -38,6 +_,7 @@
     private static final Component RESOURCEPACK = Component.translatable("options.resourcepack");
     private static final Component ACCESSIBILITY = Component.translatable("options.accessibility");
     private static final Component TELEMETRY = Component.translatable("options.telemetry");
+    private static final Tooltip TELEMETRY_DISABLED_TOOLTIP = Tooltip.create(Component.translatable("options.telemetry.disabled"));
     private static final Component CREDITS_AND_ATTRIBUTION = Component.translatable("options.credits_and_attribution");
     private static final int COLUMNS = 2;
     private final HeaderAndFooterLayout layout = new HeaderAndFooterLayout(this, 61, 33);
@@ -82,7 +_,12 @@
             )
         );
         gridlayout$rowhelper.addChild(this.openScreenButton(ACCESSIBILITY, () -> new AccessibilityOptionsScreen(this, this.options)));
-        gridlayout$rowhelper.addChild(this.openScreenButton(TELEMETRY, () -> new TelemetryInfoScreen(this, this.options)));
+        Button button = gridlayout$rowhelper.addChild(this.openScreenButton(TELEMETRY, () -> new TelemetryInfoScreen(this, this.options)));
+        if (!this.minecraft.allowsTelemetry()) {
+            button.active = false;
+            button.setTooltip(TELEMETRY_DISABLED_TOOLTIP);
+        }
+
         gridlayout$rowhelper.addChild(this.openScreenButton(CREDITS_AND_ATTRIBUTION, () -> new CreditsAndAttributionScreen(this)));
         this.layout.addToContents(gridlayout);
         this.layout.addToFooter(Button.builder(CommonComponents.GUI_DONE, p_315820_ -> this.onClose()).width(200).build());
@@ -97,7 +_,10 @@
 
     @Override
     public void onClose() {
-        this.minecraft.setScreen(this.lastScreen);
+        // We need to consider 2 potential parent screens here:
+        // 1. From the main menu, in which case display the main menu
+        // 2. From the pause menu, in which case exit back to game
+        this.minecraft.setScreen(this.lastScreen instanceof PauseScreen ? null : this.lastScreen);
     }
 
     private void applyPacks(PackRepository p_275714_) {
