--- a/net/minecraft/client/gui/screens/Screen.java
+++ b/net/minecraft/client/gui/screens/Screen.java
@@ -10,14 +_,12 @@
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.nio.file.Path;
-import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Locale;
 import java.util.Set;
 import java.util.concurrent.Executor;
 import java.util.concurrent.TimeUnit;
-import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 import net.minecraft.CrashReport;
 import net.minecraft.CrashReportCategory;
@@ -50,6 +_,7 @@
 import net.minecraft.sounds.Music;
 import net.minecraft.util.FormattedCharSequence;
 import net.minecraft.util.StringUtil;
+import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.TooltipFlag;
 import net.neoforged.api.distmarker.Dist;
@@ -61,7 +_,7 @@
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final Set<String> ALLOWED_PROTOCOLS = Sets.newHashSet("http", "https");
     private static final Component USAGE_NARRATION = Component.translatable("narrator.screen.usage");
-    protected static final CubeMap CUBE_MAP = new CubeMap(new ResourceLocation("nothingtoseeheremovealong", "textures/gui/title/background/panorama"));
+    protected static final CubeMap CUBE_MAP = new CubeMap(new ResourceLocation("textures/gui/title/background/panorama"));
     protected static final PanoramaRenderer PANORAMA = new PanoramaRenderer(CUBE_MAP);
     public static final ResourceLocation MENU_BACKGROUND = new ResourceLocation("textures/gui/menu_background.png");
     public static final ResourceLocation HEADER_SEPARATOR = new ResourceLocation("textures/gui/header_separator.png");
@@ -119,11 +_,12 @@
         }
     }
 
+    /** Neo: mixins targeting this method won't fire for container screens as {@link net.minecraft.client.gui.screens.inventory.AbstractContainerScreen#render} replicates this method in place of a super call to insert an event */
     @Override
     public void render(GuiGraphics p_281549_, int p_281550_, int p_282878_, float p_282465_) {
         this.renderBackground(p_281549_, p_281550_, p_282878_, p_282465_);
 
-        for(Renderable renderable : this.renderables) {
+        for (Renderable renderable : this.renderables) {
             renderable.render(p_281549_, p_281550_, p_282878_, p_282465_);
         }
     }
@@ -136,7 +_,7 @@
         } else if (super.keyPressed(p_96552_, p_96553_, p_96554_)) {
             return true;
         } else {
-            FocusNavigationEvent focusnavigationevent = (FocusNavigationEvent)(switch(p_96552_) {
+            FocusNavigationEvent focusnavigationevent = (FocusNavigationEvent)(switch (p_96552_) {
                 case 258 -> this.createTabEvent();
                 default -> null;
                 case 262 -> this.createArrowEvent(ScreenDirection.RIGHT);
@@ -204,7 +_,7 @@
     }
 
     public void onClose() {
-        this.minecraft.setScreen(null);
+        this.minecraft.popGuiLayer();
     }
 
     protected <T extends GuiEventListener & Renderable & NarratableEntry> T addRenderableWidget(T p_169406_) {
@@ -242,7 +_,11 @@
     }
 
     public static List<Component> getTooltipFromItem(Minecraft p_281881_, ItemStack p_282833_) {
-        return p_282833_.getTooltipLines(p_281881_.player, p_281881_.options.advancedItemTooltips ? TooltipFlag.Default.ADVANCED : TooltipFlag.Default.NORMAL);
+        return p_282833_.getTooltipLines(
+            Item.TooltipContext.of(p_281881_.level),
+            p_281881_.player,
+            p_281881_.options.advancedItemTooltips ? TooltipFlag.Default.ADVANCED : TooltipFlag.Default.NORMAL
+        );
     }
 
     protected void insertText(String p_96587_, boolean p_96588_) {
@@ -316,8 +_,11 @@
         this.width = p_96608_;
         this.height = p_96609_;
         if (!this.initialized) {
+            if (!net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.ScreenEvent.Init.Pre(this, this.children, this::addEventWidget, this::removeWidget)).isCanceled()) {
             this.init();
             this.setInitialFocus();
+            }
+            net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.ScreenEvent.Init.Post(this, this.children, this::addEventWidget, this::removeWidget));
         } else {
             this.repositionElements();
         }
@@ -330,8 +_,11 @@
     protected void rebuildWidgets() {
         this.clearWidgets();
         this.clearFocus();
+        if (!net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.ScreenEvent.Init.Pre(this, this.children, this::addEventWidget, this::removeWidget)).isCanceled()) {
         this.init();
         this.setInitialFocus();
+        }
+        net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.ScreenEvent.Init.Post(this, this.children, this::addEventWidget, this::removeWidget));
     }
 
     @Override
@@ -358,6 +_,7 @@
 
         this.renderBlurredBackground(p_294317_);
         this.renderMenuBackground(p_283688_);
+        net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.ScreenEvent.BackgroundRendered(this, p_283688_));
     }
 
     protected void renderBlurredBackground(float p_330683_) {
@@ -410,13 +_,11 @@
     }
 
     public static boolean hasControlDown() {
-        if (Minecraft.ON_OSX) {
-            return InputConstants.isKeyDown(Minecraft.getInstance().getWindow().getWindow(), 343)
-                || InputConstants.isKeyDown(Minecraft.getInstance().getWindow().getWindow(), 347);
-        } else {
-            return InputConstants.isKeyDown(Minecraft.getInstance().getWindow().getWindow(), 341)
+        return Minecraft.ON_OSX
+            ? InputConstants.isKeyDown(Minecraft.getInstance().getWindow().getWindow(), 343)
+                || InputConstants.isKeyDown(Minecraft.getInstance().getWindow().getWindow(), 347)
+            : InputConstants.isKeyDown(Minecraft.getInstance().getWindow().getWindow(), 341)
                 || InputConstants.isKeyDown(Minecraft.getInstance().getWindow().getWindow(), 345);
-        }
     }
 
     public static boolean hasShiftDown() {
@@ -471,10 +_,10 @@
         int j = p_96584_.indexOf(47);
         if (p_96585_ == ':') {
             return (j == -1 || p_96586_ <= j) && i == -1;
-        } else if (p_96585_ == '/') {
-            return p_96586_ > i;
         } else {
-            return p_96585_ == '_' || p_96585_ == '-' || p_96585_ >= 'a' && p_96585_ <= 'z' || p_96585_ >= '0' && p_96585_ <= '9' || p_96585_ == '.';
+            return p_96585_ == '/'
+                ? p_96586_ > i
+                : p_96585_ == '_' || p_96585_ == '-' || p_96585_ >= 'a' && p_96585_ <= 'z' || p_96585_ >= '0' && p_96585_ <= '9' || p_96585_ == '.';
         }
     }
 
@@ -486,6 +_,10 @@
     public void onFilesDrop(List<Path> p_96591_) {
     }
 
+    public Minecraft getMinecraft() {
+        return this.minecraft;
+    }
+
     private void scheduleNarration(long p_169381_, boolean p_169382_) {
         this.nextNarrationTime = Util.getMillis() + p_169381_;
         if (p_169382_) {
@@ -551,8 +_,11 @@
     }
 
     protected void updateNarratedWidget(NarrationElementOutput p_169403_) {
-        List<NarratableEntry> list = this.narratables.stream().filter(NarratableEntry::isActive).collect(Collectors.toList());
-        Collections.sort(list, Comparator.comparingInt(TabOrderedElement::getTabOrderGroup));
+        List<NarratableEntry> list = this.narratables
+            .stream()
+            .filter(NarratableEntry::isActive)
+            .sorted(Comparator.comparingInt(TabOrderedElement::getTabOrderGroup))
+            .toList();
         Screen.NarratableSearchResult screen$narratablesearchresult = findNarratableWidget(list, this.lastNarratable);
         if (screen$narratablesearchresult != null) {
             if (screen$narratablesearchresult.priority.isTerminal()) {
@@ -582,7 +_,7 @@
         Screen.NarratableSearchResult screen$narratablesearchresult1 = null;
         int i = 0;
 
-        for(int j = p_169401_.size(); i < j; ++i) {
+        for (int j = p_169401_.size(); i < j; i++) {
             NarratableEntry narratableentry = p_169401_.get(i);
             NarratableEntry.NarrationPriority narratableentry$narrationpriority = narratableentry.narrationPriority();
             if (narratableentry$narrationpriority.isTerminal()) {
@@ -649,5 +_,13 @@
             this.index = p_169425_;
             this.priority = p_169426_;
         }
+    }
+
+    private void addEventWidget(GuiEventListener b) {
+        if (b instanceof Renderable r)
+            this.renderables.add(r);
+        if (b instanceof NarratableEntry ne)
+            this.narratables.add(ne);
+        children.add(b);
     }
 }
