--- a/net/minecraft/client/gui/screens/SkinCustomizationScreen.java
+++ b/net/minecraft/client/gui/screens/SkinCustomizationScreen.java
@@ -27,7 +_,7 @@
         this.list = this.addRenderableWidget(new OptionsList(this.minecraft, this.width, this.height, this));
         List<AbstractWidget> list = new ArrayList<>();
 
-        for(PlayerModelPart playermodelpart : PlayerModelPart.values()) {
+        for (PlayerModelPart playermodelpart : PlayerModelPart.values()) {
             list.add(
                 CycleButton.onOffBuilder(this.options.isModelPartEnabled(playermodelpart))
                     .create(playermodelpart.getName(), (p_169436_, p_169437_) -> this.options.toggleModelPart(playermodelpart, p_169437_))
