--- a/net/minecraft/client/gui/screens/WinScreen.java
+++ b/net/minecraft/client/gui/screens/WinScreen.java
@@ -152,29 +_,27 @@
         RandomSource randomsource = RandomSource.create(8124371L);
 
         String s;
-        while((s = bufferedreader.readLine()) != null) {
+        while ((s = bufferedreader.readLine()) != null) {
+            s = s.replaceAll("PLAYERNAME", this.minecraft.getUser().getName());
+
             int i;
-            String s1;
-            String s2;
-            for(s = s.replaceAll("PLAYERNAME", this.minecraft.getUser().getName());
-                (i = s.indexOf(OBFUSCATE_TOKEN)) != -1;
-                s = s1 + ChatFormatting.WHITE + ChatFormatting.OBFUSCATED + "XXXXXXXX".substring(0, randomsource.nextInt(4) + 3) + s2
-            ) {
-                s1 = s.substring(0, i);
-                s2 = s.substring(i + OBFUSCATE_TOKEN.length());
+            while ((i = s.indexOf(OBFUSCATE_TOKEN)) != -1) {
+                String s1 = s.substring(0, i);
+                String s2 = s.substring(i + OBFUSCATE_TOKEN.length());
+                s = s1 + ChatFormatting.WHITE + ChatFormatting.OBFUSCATED + "XXXXXXXX".substring(0, randomsource.nextInt(4) + 3) + s2;
             }
 
             this.addPoemLines(s);
             this.addEmptyLine();
         }
 
-        for(int j = 0; j < 8; ++j) {
+        for (int j = 0; j < 8; j++) {
             this.addEmptyLine();
         }
     }
 
     private void addCreditsFile(Reader p_232820_) {
-        for(JsonElement jsonelement : GsonHelper.parseArray(p_232820_)) {
+        for (JsonElement jsonelement : GsonHelper.parseArray(p_232820_)) {
             JsonObject jsonobject = jsonelement.getAsJsonObject();
             String s = jsonobject.get("section").getAsString();
             this.addCreditsLine(SECTION_HEADING, true);
@@ -183,7 +_,7 @@
             this.addEmptyLine();
             this.addEmptyLine();
 
-            for(JsonElement jsonelement1 : jsonobject.getAsJsonArray("disciplines")) {
+            for (JsonElement jsonelement1 : jsonobject.getAsJsonArray("disciplines")) {
                 JsonObject jsonobject1 = jsonelement1.getAsJsonObject();
                 String s1 = jsonobject1.get("discipline").getAsString();
                 if (StringUtils.isNotEmpty(s1)) {
@@ -192,13 +_,13 @@
                     this.addEmptyLine();
                 }
 
-                for(JsonElement jsonelement2 : jsonobject1.getAsJsonArray("titles")) {
+                for (JsonElement jsonelement2 : jsonobject1.getAsJsonArray("titles")) {
                     JsonObject jsonobject2 = jsonelement2.getAsJsonObject();
                     String s2 = jsonobject2.get("title").getAsString();
                     JsonArray jsonarray = jsonobject2.getAsJsonArray("names");
                     this.addCreditsLine(Component.literal(s2).withStyle(ChatFormatting.GRAY), false);
 
-                    for(JsonElement jsonelement3 : jsonarray) {
+                    for (JsonElement jsonelement3 : jsonarray) {
                         String s3 = jsonelement3.getAsString();
                         this.addCreditsLine(Component.literal("           ").append(s3).withStyle(ChatFormatting.WHITE), false);
                     }
@@ -239,7 +_,7 @@
         this.logoRenderer.renderLogo(p_281907_, this.width, 1.0F, j);
         int k = j + 100;
 
-        for(int l = 0; l < this.lines.size(); ++l) {
+        for (int l = 0; l < this.lines.size(); l++) {
             if (l == this.lines.size() - 1) {
                 float f1 = (float)k + f - (float)(this.height / 2 - 6);
                 if (f1 < 0.0F) {
