--- a/net/minecraft/data/BlockFamilies.java
+++ b/net/minecraft/data/BlockFamilies.java
@@ -128,19 +_,6 @@
         .recipeGroupPrefix("wooden")
         .recipeUnlockedBy("has_planks")
         .getFamily();
-    public static final BlockFamily POTATO_PLANKS = familyBuilder(Blocks.POTATO_PLANKS)
-        .button(Blocks.POTATO_BUTTON)
-        .fence(Blocks.POTATO_FENCE)
-        .fenceGate(Blocks.POTATO_FENCE_GATE)
-        .pressurePlate(Blocks.POTATO_PRESSURE_PLATE)
-        .sign(Blocks.POTATO_SIGN, Blocks.POTATO_WALL_SIGN)
-        .slab(Blocks.POTATO_SLAB)
-        .stairs(Blocks.POTATO_STAIRS)
-        .door(Blocks.POTATO_DOOR)
-        .trapdoor(Blocks.POTATO_TRAPDOOR)
-        .recipeGroupPrefix("wooden")
-        .recipeUnlockedBy("has_planks")
-        .getFamily();
     public static final BlockFamily MANGROVE_PLANKS = familyBuilder(Blocks.MANGROVE_PLANKS)
         .button(Blocks.MANGROVE_BUTTON)
         .slab(Blocks.MANGROVE_SLAB)
@@ -332,26 +_,6 @@
         .stairs(Blocks.COBBLESTONE_STAIRS)
         .slab(Blocks.COBBLESTONE_SLAB)
         .getFamily();
-    public static final BlockFamily TATERSTONE = familyBuilder(Blocks.TATERSTONE)
-        .slab(Blocks.TATERSTONE_SLAB)
-        .stairs(Blocks.TATERSTONE_STAIRS)
-        .wall(Blocks.TATERSTONE_WALL)
-        .getFamily();
-    public static final BlockFamily BAKED_POTATO_BRICKS = familyBuilder(Blocks.BAKED_POTATO_BRICKS)
-        .wall(Blocks.BAKED_POTATO_BRICK_WALL)
-        .stairs(Blocks.BAKED_POTATO_BRICK_STAIRS)
-        .slab(Blocks.BAKED_POTATO_BRICK_SLAB)
-        .getFamily();
-    public static final BlockFamily EXPIRED_BAKED_POTATO_BRICKS = familyBuilder(Blocks.EXPIRED_BAKED_POTATO_BRICKS)
-        .wall(Blocks.EXPIRED_BAKED_POTATO_BRICK_WALL)
-        .stairs(Blocks.EXPIRED_BAKED_POTATO_BRICK_STAIRS)
-        .slab(Blocks.EXPIRED_BAKED_POTATO_BRICK_SLAB)
-        .getFamily();
-    public static final BlockFamily CHARRED_BAKED_POTATO_BRICKS = familyBuilder(Blocks.CHARRED_BAKED_POTATO_BRICKS)
-        .wall(Blocks.CHARRED_BAKED_POTATO_BRICK_WALL)
-        .stairs(Blocks.CHARRED_BAKED_POTATO_BRICK_STAIRS)
-        .slab(Blocks.CHARRED_BAKED_POTATO_BRICK_SLAB)
-        .getFamily();
     public static final BlockFamily MOSSY_COBBLESTONE = familyBuilder(Blocks.MOSSY_COBBLESTONE)
         .wall(Blocks.MOSSY_COBBLESTONE_WALL)
         .stairs(Blocks.MOSSY_COBBLESTONE_STAIRS)
@@ -468,11 +_,6 @@
         .pressurePlate(Blocks.STONE_PRESSURE_PLATE)
         .button(Blocks.STONE_BUTTON)
         .stairs(Blocks.STONE_STAIRS)
-        .getFamily();
-    public static final BlockFamily POTONE = familyBuilder(Blocks.POTONE)
-        .slab(Blocks.POTONE_SLAB)
-        .stairs(Blocks.POTONE_STAIRS)
-        .wall(Blocks.POTONE_WALL)
         .getFamily();
     public static final BlockFamily STONE_BRICK = familyBuilder(Blocks.STONE_BRICKS)
         .wall(Blocks.STONE_BRICK_WALL)
