--- a/net/minecraft/world/level/block/AbstractBannerBlock.java
+++ b/net/minecraft/world/level/block/AbstractBannerBlock.java
@@ -33,8 +_,7 @@
 
     @Override
     public ItemStack getCloneItemStack(LevelReader p_304796_, BlockPos p_48665_, BlockState p_48666_) {
-        BlockEntity blockentity = p_304796_.getBlockEntity(p_48665_);
-        return blockentity instanceof BannerBlockEntity bannerblockentity
+        return p_304796_.getBlockEntity(p_48665_) instanceof BannerBlockEntity bannerblockentity
             ? bannerblockentity.getItem()
             : super.getCloneItemStack(p_304796_, p_48665_, p_48666_);
     }
