--- a/net/minecraft/world/level/block/AmethystClusterBlock.java
+++ b/net/minecraft/world/level/block/AmethystClusterBlock.java
@@ -69,7 +_,7 @@
     @Override
     protected VoxelShape getShape(BlockState p_152021_, BlockGetter p_152022_, BlockPos p_152023_, CollisionContext p_152024_) {
         Direction direction = p_152021_.getValue(FACING);
-        switch(direction) {
+        switch (direction) {
             case NORTH:
                 return this.northAabb;
             case SOUTH:
