--- a/net/minecraft/world/level/block/BaseCoralPlantTypeBlock.java
+++ b/net/minecraft/world/level/block/BaseCoralPlantTypeBlock.java
@@ -41,7 +_,7 @@
         if (p_49187_.getValue(WATERLOGGED)) {
             return true;
         } else {
-            for(Direction direction : Direction.values()) {
+            for (Direction direction : Direction.values()) {
                 if (p_49188_.getFluidState(p_49189_.relative(direction)).is(FluidTags.WATER)) {
                     return true;
                 }
