--- a/net/minecraft/world/level/block/BedBlock.java
+++ b/net/minecraft/world/level/block/BedBlock.java
@@ -201,7 +_,7 @@
     @Override
     protected VoxelShape getShape(BlockState p_49547_, BlockGetter p_49548_, BlockPos p_49549_, CollisionContext p_49550_) {
         Direction direction = getConnectedDirection(p_49547_).getOpposite();
-        switch(direction) {
+        switch (direction) {
             case NORTH:
                 return NORTH_SHAPE;
             case SOUTH:
@@ -276,7 +_,7 @@
     ) {
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int[] aint : p_49473_) {
+        for (int[] aint : p_49473_) {
             blockpos$mutableblockpos.set(p_49472_.getX() + aint[0], p_49472_.getY(), p_49472_.getZ() + aint[1]);
             Vec3 vec3 = DismountHelper.findSafeDismountLocation(p_49470_, p_49471_, blockpos$mutableblockpos, p_49474_);
             if (vec3 != null) {
