--- a/net/minecraft/world/level/block/BeehiveBlock.java
+++ b/net/minecraft/world/level/block/BeehiveBlock.java
@@ -105,7 +_,7 @@
                 return;
             }
 
-            for(Bee bee : list) {
+            for (Bee bee : list) {
                 if (bee.getTarget() == null) {
                     Player player = Util.getRandom(list1, p_49650_.random);
                     bee.setTarget(player);
@@ -126,7 +_,7 @@
         boolean flag = false;
         if (i >= 5) {
             Item item = p_316844_.getItem();
-            if (p_316844_.is(Items.SHEARS)) {
+            if (p_316844_.canPerformAction(net.neoforged.neoforge.common.ToolActions.SHEARS_HARVEST)) {
                 p_316306_.playSound(p_316824_, p_316824_.getX(), p_316824_.getY(), p_316824_.getZ(), SoundEvents.BEEHIVE_SHEAR, SoundSource.BLOCKS, 1.0F, 1.0F);
                 dropHoneycomb(p_316306_, p_316497_);
                 p_316844_.hurtAndBreak(1, p_316824_, LivingEntity.getSlotForHand(p_316436_));
@@ -168,20 +_,14 @@
     }
 
     private boolean hiveContainsBees(Level p_49655_, BlockPos p_49656_) {
-        BlockEntity blockentity = p_49655_.getBlockEntity(p_49656_);
-        if (blockentity instanceof BeehiveBlockEntity beehiveblockentity) {
-            return !beehiveblockentity.isEmpty();
-        } else {
-            return false;
-        }
+        return p_49655_.getBlockEntity(p_49656_) instanceof BeehiveBlockEntity beehiveblockentity ? !beehiveblockentity.isEmpty() : false;
     }
 
     public void releaseBeesAndResetHoneyLevel(
         Level p_49595_, BlockState p_49596_, BlockPos p_49597_, @Nullable Player p_49598_, BeehiveBlockEntity.BeeReleaseStatus p_49599_
     ) {
         this.resetHoneyLevel(p_49595_, p_49596_, p_49597_);
-        BlockEntity blockentity = p_49595_.getBlockEntity(p_49597_);
-        if (blockentity instanceof BeehiveBlockEntity beehiveblockentity) {
+        if (p_49595_.getBlockEntity(p_49597_) instanceof BeehiveBlockEntity beehiveblockentity) {
             beehiveblockentity.emptyAllLivingFromHive(p_49598_, p_49596_, p_49599_);
         }
     }
@@ -193,7 +_,7 @@
     @Override
     public void animateTick(BlockState p_220773_, Level p_220774_, BlockPos p_220775_, RandomSource p_220776_) {
         if (p_220773_.getValue(HONEY_LEVEL) >= 5) {
-            for(int i = 0; i < p_220776_.nextInt(1) + 1; ++i) {
+            for (int i = 0; i < p_220776_.nextInt(1) + 1; i++) {
                 this.trySpawnDripParticles(p_220774_, p_220775_, p_220773_);
             }
         }
@@ -272,19 +_,19 @@
 
     @Override
     public BlockState playerWillDestroy(Level p_49608_, BlockPos p_49609_, BlockState p_49610_, Player p_49611_) {
-        if (!p_49608_.isClientSide && p_49611_.isCreative() && p_49608_.getGameRules().getBoolean(GameRules.RULE_DOBLOCKDROPS)) {
-            BlockEntity blockentity = p_49608_.getBlockEntity(p_49609_);
-            if (blockentity instanceof BeehiveBlockEntity beehiveblockentity) {
-                int i = p_49610_.getValue(HONEY_LEVEL);
-                boolean flag = !beehiveblockentity.isEmpty();
-                if (flag || i > 0) {
-                    ItemStack itemstack = new ItemStack(this);
-                    itemstack.applyComponents(beehiveblockentity.collectComponents());
-                    itemstack.set(DataComponents.BLOCK_STATE, BlockItemStateProperties.EMPTY.with(HONEY_LEVEL, i));
-                    ItemEntity itementity = new ItemEntity(p_49608_, (double)p_49609_.getX(), (double)p_49609_.getY(), (double)p_49609_.getZ(), itemstack);
-                    itementity.setDefaultPickUpDelay();
-                    p_49608_.addFreshEntity(itementity);
-                }
+        if (!p_49608_.isClientSide
+            && p_49611_.isCreative()
+            && p_49608_.getGameRules().getBoolean(GameRules.RULE_DOBLOCKDROPS)
+            && p_49608_.getBlockEntity(p_49609_) instanceof BeehiveBlockEntity beehiveblockentity) {
+            int i = p_49610_.getValue(HONEY_LEVEL);
+            boolean flag = !beehiveblockentity.isEmpty();
+            if (flag || i > 0) {
+                ItemStack itemstack = new ItemStack(this);
+                itemstack.applyComponents(beehiveblockentity.collectComponents());
+                itemstack.set(DataComponents.BLOCK_STATE, BlockItemStateProperties.EMPTY.with(HONEY_LEVEL, i));
+                ItemEntity itementity = new ItemEntity(p_49608_, (double)p_49609_.getX(), (double)p_49609_.getY(), (double)p_49609_.getZ(), itemstack);
+                itementity.setDefaultPickUpDelay();
+                p_49608_.addFreshEntity(itementity);
             }
         }
 
@@ -310,11 +_,9 @@
 
     @Override
     protected BlockState updateShape(BlockState p_49639_, Direction p_49640_, BlockState p_49641_, LevelAccessor p_49642_, BlockPos p_49643_, BlockPos p_49644_) {
-        if (p_49642_.getBlockState(p_49644_).getBlock() instanceof FireBlock) {
-            BlockEntity blockentity = p_49642_.getBlockEntity(p_49643_);
-            if (blockentity instanceof BeehiveBlockEntity beehiveblockentity) {
-                beehiveblockentity.emptyAllLivingFromHive(null, p_49639_, BeehiveBlockEntity.BeeReleaseStatus.EMERGENCY);
-            }
+        if (p_49642_.getBlockState(p_49644_).getBlock() instanceof FireBlock
+            && p_49642_.getBlockEntity(p_49643_) instanceof BeehiveBlockEntity beehiveblockentity) {
+            beehiveblockentity.emptyAllLivingFromHive(null, p_49639_, BeehiveBlockEntity.BeeReleaseStatus.EMERGENCY);
         }
 
         return super.updateShape(p_49639_, p_49640_, p_49641_, p_49642_, p_49643_, p_49644_);
