--- a/net/minecraft/world/level/block/Blocks.java
+++ b/net/minecraft/world/level/block/Blocks.java
@@ -1,8 +_,5 @@
 package net.minecraft.world.level.block;
 
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.ImmutableMap.Builder;
-import java.util.Map;
 import java.util.function.ToIntFunction;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
@@ -24,7 +_,6 @@
 import net.minecraft.world.item.DyeColor;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.biome.Biome;
-import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.entity.BlockEntityType;
 import net.minecraft.world.level.block.entity.ShulkerBoxBlockEntity;
 import net.minecraft.world.level.block.grower.TreeGrower;
@@ -44,10 +_,9 @@
 import net.minecraft.world.level.material.PushReaction;
 
 public class Blocks {
-    private static final BlockBehaviour.StatePredicate NOT_CLOSED_SHULKER = (p_304352_, p_304353_, p_304354_) -> {
-        BlockEntity blockentity = p_304353_.getBlockEntity(p_304354_);
-        return blockentity instanceof ShulkerBoxBlockEntity shulkerboxblockentity ? shulkerboxblockentity.isClosed() : true;
-    };
+    private static final BlockBehaviour.StatePredicate NOT_CLOSED_SHULKER = (p_304352_, p_304353_, p_304354_) -> p_304353_.getBlockEntity(p_304354_) instanceof ShulkerBoxBlockEntity shulkerboxblockentity
+            ? shulkerboxblockentity.isClosed()
+            : true;
     public static final Block AIR = register("air", new AirBlock(BlockBehaviour.Properties.of().replaceable().noCollission().noLootTable().air()));
     public static final Block STONE = register(
         "stone",
@@ -55,17 +_,6 @@
             BlockBehaviour.Properties.of().mapColor(MapColor.STONE).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(1.5F, 6.0F)
         )
     );
-    public static final Block POTONE = register(
-        "potone",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(1.5F, 6.0F)
-        )
-    );
     public static final Block GRANITE = register(
         "granite",
         new Block(
@@ -113,18 +_,7 @@
     public static final Block GRASS_BLOCK = register(
         "grass_block", new GrassBlock(BlockBehaviour.Properties.of().mapColor(MapColor.GRASS).randomTicks().strength(0.6F).sound(SoundType.GRASS))
     );
-    public static final Block PEELGRASS_BLOCK = register(
-        "peelgrass_block",
-        new GrassBlock(BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_LIGHT_GREEN).randomTicks().strength(0.6F).sound(SoundType.PEELGRASS))
-    );
-    public static final Block CORRUPTED_PEELGRASS_BLOCK = register(
-        "corrupted_peelgrass_block",
-        new GrassBlock(BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_GREEN).randomTicks().strength(0.6F).sound(SoundType.CORRUPTED_PEELGRASS))
-    );
     public static final Block DIRT = register("dirt", new Block(BlockBehaviour.Properties.of().mapColor(MapColor.DIRT).strength(0.5F).sound(SoundType.GRAVEL)));
-    public static final Block TERREDEPOMME = register(
-        "terredepomme", new Block(BlockBehaviour.Properties.of().mapColor(MapColor.TERRACOTTA_YELLOW).strength(0.5F).sound(SoundType.TERREDEPOMME))
-    );
     public static final Block COARSE_DIRT = register(
         "coarse_dirt", new Block(BlockBehaviour.Properties.of().mapColor(MapColor.DIRT).strength(0.5F).sound(SoundType.GRAVEL))
     );
@@ -137,281 +_,6 @@
             BlockBehaviour.Properties.of().mapColor(MapColor.STONE).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(2.0F, 6.0F)
         )
     );
-    public static final Block TATERSTONE = register(
-        "taterstone",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_YELLOW)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 6.0F)
-        )
-    );
-    public static final Block BAKED_POTATO_BRICKS = register(
-        "baked_potato_bricks",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(4.0F, 6.0F)
-        )
-    );
-    public static final Block EXPIRED_BAKED_POTATO_BRICKS = register(
-        "expired_baked_potato_bricks",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_LIGHT_GREEN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 1.0F)
-        )
-    );
-    public static final Block CHARRED_BAKED_POTATO_BRICKS = register(
-        "charred_baked_potato_bricks",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_BROWN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(1000.0F, 100.0F)
-        )
-    );
-    public static final Block TATERSTONE_SLAB = register(
-        "taterstone_slab",
-        new SlabBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_YELLOW)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 6.0F)
-        )
-    );
-    public static final Block POTONE_SLAB = register(
-        "potone_slab",
-        new SlabBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 6.0F)
-        )
-    );
-    public static final Block BAKED_POTATO_BRICK_SLAB = register(
-        "baked_potato_brick_slab",
-        new SlabBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(4.0F, 6.0F)
-        )
-    );
-    public static final Block EXPIRED_BAKED_POTATO_BRICK_SLAB = register(
-        "expired_baked_potato_brick_slab",
-        new SlabBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_LIGHT_GREEN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 1.0F)
-        )
-    );
-    public static final Block CHARRED_BAKED_POTATO_BRICK_SLAB = register(
-        "charred_baked_potato_brick_slab",
-        new SlabBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_BROWN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(1000.0F, 100.0F)
-        )
-    );
-    public static final Block TATERSTONE_STAIRS = register(
-        "taterstone_stairs",
-        new StairBlock(
-            TATERSTONE.defaultBlockState(),
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_YELLOW)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 6.0F)
-        )
-    );
-    public static final Block POTONE_STAIRS = register(
-        "potone_stairs",
-        new StairBlock(
-            POTONE.defaultBlockState(),
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 6.0F)
-        )
-    );
-    public static final Block BAKED_POTATO_BRICK_STAIRS = register(
-        "baked_potato_brick_stairs",
-        new StairBlock(
-            POTONE.defaultBlockState(),
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(4.0F, 6.0F)
-        )
-    );
-    public static final Block EXPIRED_BAKED_POTATO_BRICK_STAIRS = register(
-        "expired_baked_potato_brick_stairs",
-        new StairBlock(
-            POTONE.defaultBlockState(),
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_LIGHT_GREEN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 1.0F)
-        )
-    );
-    public static final Block CHARRED_BAKED_POTATO_BRICK_STAIRS = register(
-        "charred_baked_potato_brick_stairs",
-        new StairBlock(
-            POTONE.defaultBlockState(),
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_BROWN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(1000.0F, 100.0F)
-        )
-    );
-    public static final Block TATERSTONE_WALL = register(
-        "taterstone_wall",
-        new WallBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_YELLOW)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 6.0F)
-                .forceSolidOn()
-        )
-    );
-    public static final Block POTONE_WALL = register(
-        "potone_wall",
-        new WallBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 6.0F)
-                .forceSolidOn()
-        )
-    );
-    public static final Block BAKED_POTATO_BRICK_WALL = register(
-        "baked_potato_brick_wall",
-        new WallBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(4.0F, 6.0F)
-                .forceSolidOn()
-        )
-    );
-    public static final Block EXPIRED_BAKED_POTATO_BRICK_WALL = register(
-        "expired_baked_potato_brick_wall",
-        new WallBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_LIGHT_GREEN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(2.0F, 1.0F)
-                .forceSolidOn()
-        )
-    );
-    public static final Block CHARRED_BAKED_POTATO_BRICK_WALL = register(
-        "charred_baked_potato_brick_wall",
-        new WallBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_BROWN)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .requiresCorrectToolForDrops()
-                .strength(1000.0F, 100.0F)
-                .forceSolidOn()
-        )
-    );
-    public static final Block POISONOUS_MASHED_POTATO = register(
-        "poisonous_mashed_potato",
-        new MashedPotatoBlock(
-            BlockBehaviour.Properties.of()
-                .replaceable()
-                .forceSolidOff()
-                .mapColor(MapColor.GOLD)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.SLIME_BLOCK)
-                .strength(0.1F)
-                .isViewBlocking((p_340225_, p_340226_, p_340227_) -> p_340225_.getValue(MashedPotatoBlock.LAYERS) >= 8)
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
-    public static final Block POISONOUS_POTATO_BLOCK = register(
-        "poisonous_potato_block",
-        new Block(BlockBehaviour.Properties.of().mapColor(MapColor.GRASS).instrument(NoteBlockInstrument.BASEDRUM).sound(SoundType.POTONE).strength(0.2F))
-    );
-    public static final Block COMPRESSED_POISONOUS_POTATO_BLOCK = register(
-        "compressed_poisonous_potato_block",
-        new Block(
-            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_ORANGE).instrument(NoteBlockInstrument.BASEDRUM).sound(SoundType.POTONE).strength(1.8F)
-        )
-    );
-    public static final Block DOUBLE_COMPRESSED_POISONOUS_POTATO_BLOCK = register(
-        "double_compressed_poisonous_potato_block",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_ORANGE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .strength(16.2F)
-        )
-    );
-    public static final Block TRIPLE_COMPRESSED_POISONOUS_POTATO_BLOCK = register(
-        "triple_compressed_poisonous_potato_block",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_BLACK)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.POTONE)
-                .strength(145.8F)
-        )
-    );
-    public static final Block QUADRUPLE_COMPRESSED_POISONOUS_POTATO_BLOCK = register(
-        "quadruple_compressed_poisonous_potato_block",
-        new Block(
-            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_BLACK).instrument(NoteBlockInstrument.BASEDRUM).sound(SoundType.POTONE).strength(1312.2F)
-        )
-    );
-    public static final Block FLOATATO = register(
-        "floatato",
-        new Block(BlockBehaviour.Properties.of().mapColor(MapColor.QUARTZ).instrument(NoteBlockInstrument.BASEDRUM).sound(SoundType.WOOL).strength(0.3F))
-    );
-    public static final Block FLOATATER = register(
-        "floatater",
-        new FloataterBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.QUARTZ)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .sound(SoundType.STONE)
-                .strength(0.5F)
-                .requiresCorrectToolForDrops()
-        )
-    );
     public static final Block OAK_PLANKS = register(
         "oak_planks",
         new Block(
@@ -663,7 +_,7 @@
                 .noCollission()
                 .randomTicks()
                 .strength(100.0F)
-                .lightLevel(p_340230_ -> 15)
+                .lightLevel(p_220867_ -> 15)
                 .pushReaction(PushReaction.DESTROY)
                 .noLootTable()
                 .liquid()
@@ -677,13 +_,6 @@
             BlockBehaviour.Properties.of().mapColor(MapColor.SAND).instrument(NoteBlockInstrument.SNARE).strength(0.5F).sound(SoundType.SAND)
         )
     );
-    public static final Block GRAVTATER = register(
-        "gravtater",
-        new ColoredFallingBlock(
-            new ColorRGBA(14188339),
-            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_ORANGE).instrument(NoteBlockInstrument.SNARE).strength(0.6F).sound(SoundType.GRAVTATER)
-        )
-    );
     public static final Block SUSPICIOUS_SAND = register(
         "suspicious_sand",
         new BrushableBlock(
@@ -752,19 +_,6 @@
             ConstantInt.of(0), BlockBehaviour.Properties.ofLegacyCopy(IRON_ORE).mapColor(MapColor.DEEPSLATE).strength(4.5F, 3.0F).sound(SoundType.DEEPSLATE)
         )
     );
-    public static final Block POISONOUS_POTATO_ORE = register(
-        "poisonous_potato_ore",
-        new DropExperienceBlock(
-            ConstantInt.of(0),
-            BlockBehaviour.Properties.of().mapColor(MapColor.STONE).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(3.0F, 3.0F)
-        )
-    );
-    public static final Block DEEPSLATE_POISONOUS_POTATO_ORE = register(
-        "deepslate_poisonous_potato_ore",
-        new DropExperienceBlock(
-            ConstantInt.of(0), BlockBehaviour.Properties.ofLegacyCopy(IRON_ORE).mapColor(MapColor.DEEPSLATE).strength(4.5F, 3.0F).sound(SoundType.DEEPSLATE)
-        )
-    );
     public static final Block COAL_ORE = register(
         "coal_ore",
         new DropExperienceBlock(
@@ -1026,19 +_,6 @@
             BlockBehaviour.Properties.of().mapColor(MapColor.STONE).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(3.0F, 3.0F)
         )
     );
-    public static final Block RESIN_ORE = register(
-        "resin_ore",
-        new DropExperienceTransparentBlock(
-            UniformInt.of(0, 1),
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_ORANGE)
-                .sound(SoundType.POTONE)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(3.0F, 1.0F)
-                .lightLevel(p_340232_ -> 3)
-        )
-    );
     public static final Block DEEPSLATE_LAPIS_ORE = register(
         "deepslate_lapis_ore",
         new DropExperienceBlock(
@@ -1054,17 +_,6 @@
             BlockBehaviour.Properties.of().mapColor(MapColor.STONE).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(3.5F)
         )
     );
-    public static final Block VICIOUS_POTATO = register(
-        "vicious_potato",
-        new ViciousPotatoBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_BROWN)
-                .sound(SoundType.GRAVTATER)
-                .instrument(NoteBlockInstrument.BANJO)
-                .randomTicks()
-                .strength(3.5F)
-        )
-    );
     public static final Block SANDSTONE = register(
         "sandstone",
         new Block(BlockBehaviour.Properties.of().mapColor(MapColor.SAND).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(0.8F))
@@ -1100,7 +_,7 @@
     public static final Block RED_BED = register("red_bed", bed(DyeColor.RED));
     public static final Block BLACK_BED = register("black_bed", bed(DyeColor.BLACK));
     public static final Block POWERED_RAIL = register(
-        "powered_rail", new PoweredRailBlock(BlockBehaviour.Properties.of().noCollission().strength(0.7F).sound(SoundType.METAL))
+        "powered_rail", new PoweredRailBlock(BlockBehaviour.Properties.of().noCollission().strength(0.7F).sound(SoundType.METAL), true)
     );
     public static final Block DETECTOR_RAIL = register(
         "detector_rail", new DetectorRailBlock(BlockBehaviour.Properties.of().noCollission().strength(0.7F).sound(SoundType.METAL))
@@ -1111,6 +_,7 @@
         new WebBlock(
             BlockBehaviour.Properties.of()
                 .mapColor(MapColor.WOOL)
+                .sound(SoundType.COBWEB)
                 .forceSolidOn()
                 .noCollission()
                 .requiresCorrectToolForDrops()
@@ -1572,20 +_,6 @@
                 .pushReaction(PushReaction.DESTROY)
         )
     );
-    public static final Block POTATO_FLOWER = register(
-        "potato_flower",
-        new FlowerBlock(
-            MobEffects.POISON,
-            12.0F,
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.PLANT)
-                .noCollission()
-                .instabreak()
-                .sound(SoundType.GRASS)
-                .offsetType(BlockBehaviour.OffsetType.XZ)
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
     public static final Block BROWN_MUSHROOM = register(
         "brown_mushroom",
         new MushroomBlock(
@@ -1596,7 +_,7 @@
                 .randomTicks()
                 .instabreak()
                 .sound(SoundType.GRASS)
-                .lightLevel(p_50856_ -> 1)
+                .lightLevel(p_50892_ -> 1)
                 .hasPostProcess(Blocks::always)
                 .pushReaction(PushReaction.DESTROY)
         )
@@ -1690,7 +_,7 @@
         "torch",
         new TorchBlock(
             ParticleTypes.FLAME,
-            BlockBehaviour.Properties.of().noCollission().instabreak().lightLevel(p_50886_ -> 14).sound(SoundType.WOOD).pushReaction(PushReaction.DESTROY)
+            BlockBehaviour.Properties.of().noCollission().instabreak().lightLevel(p_50755_ -> 14).sound(SoundType.WOOD).pushReaction(PushReaction.DESTROY)
         )
     );
     public static final Block WALL_TORCH = register(
@@ -1700,7 +_,7 @@
             BlockBehaviour.Properties.of()
                 .noCollission()
                 .instabreak()
-                .lightLevel(p_152607_ -> 14)
+                .lightLevel(p_50886_ -> 14)
                 .sound(SoundType.WOOD)
                 .dropsLike(TORCH)
                 .pushReaction(PushReaction.DESTROY)
@@ -1714,7 +_,7 @@
                 .replaceable()
                 .noCollission()
                 .instabreak()
-                .lightLevel(p_152605_ -> 15)
+                .lightLevel(p_152607_ -> 15)
                 .sound(SoundType.WOOL)
                 .pushReaction(PushReaction.DESTROY)
         )
@@ -1727,7 +_,7 @@
                 .replaceable()
                 .noCollission()
                 .instabreak()
-                .lightLevel(p_50884_ -> 10)
+                .lightLevel(p_152605_ -> 10)
                 .sound(SoundType.WOOL)
                 .pushReaction(PushReaction.DESTROY)
         )
@@ -1803,18 +_,6 @@
                 .isSuffocating(Blocks::always)
         )
     );
-    public static final Block POISON_FARMLAND = register(
-        "poison_farmland",
-        new FarmBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_YELLOW)
-                .randomTicks()
-                .strength(0.6F)
-                .sound(SoundType.GRAVTATER)
-                .isViewBlocking(Blocks::always)
-                .isSuffocating(Blocks::always)
-        )
-    );
     public static final Block FURNACE = register(
         "furnace",
         new FurnaceBlock(
@@ -2193,13 +_,6 @@
             BlockBehaviour.Properties.of().mapColor(MapColor.WARPED_STEM).forceSolidOn().instrument(NoteBlockInstrument.BASS).noCollission().strength(1.0F)
         )
     );
-    public static final Block POTATO_HANGING_SIGN = register(
-        "potato_hanging_sign",
-        new CeilingHangingSignBlock(
-            WoodType.POTATO,
-            BlockBehaviour.Properties.of().mapColor(MapColor.EMERALD).forceSolidOn().instrument(NoteBlockInstrument.BASS).noCollission().strength(1.0F)
-        )
-    );
     public static final Block MANGROVE_HANGING_SIGN = register(
         "mangrove_hanging_sign",
         new CeilingHangingSignBlock(
@@ -2364,19 +_,6 @@
                 .dropsLike(WARPED_HANGING_SIGN)
         )
     );
-    public static final Block POTATO_WALL_HANGING_SIGN = register(
-        "potato_wall_hanging_sign",
-        new WallHangingSignBlock(
-            WoodType.POTATO,
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.WOOD)
-                .forceSolidOn()
-                .instrument(NoteBlockInstrument.BASS)
-                .noCollission()
-                .strength(1.0F)
-                .dropsLike(POTATO_HANGING_SIGN)
-        )
-    );
     public static final Block BAMBOO_WALL_HANGING_SIGN = register(
         "bamboo_wall_hanging_sign",
         new WallHangingSignBlock(
@@ -2712,7 +_,7 @@
         "soul_torch",
         new TorchBlock(
             ParticleTypes.SOUL_FIRE_FLAME,
-            BlockBehaviour.Properties.of().noCollission().instabreak().lightLevel(p_50876_ -> 10).sound(SoundType.WOOD).pushReaction(PushReaction.DESTROY)
+            BlockBehaviour.Properties.of().noCollission().instabreak().lightLevel(p_50884_ -> 10).sound(SoundType.WOOD).pushReaction(PushReaction.DESTROY)
         )
     );
     public static final Block SOUL_WALL_TORCH = register(
@@ -2722,7 +_,7 @@
             BlockBehaviour.Properties.of()
                 .noCollission()
                 .instabreak()
-                .lightLevel(p_50874_ -> 10)
+                .lightLevel(p_50876_ -> 10)
                 .sound(SoundType.WOOD)
                 .dropsLike(SOUL_TORCH)
                 .pushReaction(PushReaction.DESTROY)
@@ -2736,7 +_,7 @@
                 .instrument(NoteBlockInstrument.PLING)
                 .strength(0.3F)
                 .sound(SoundType.GLASS)
-                .lightLevel(p_50872_ -> 15)
+                .lightLevel(p_50874_ -> 15)
                 .isRedstoneConductor(Blocks::never)
         )
     );
@@ -2748,26 +_,8 @@
                 .randomTicks()
                 .strength(-1.0F)
                 .sound(SoundType.GLASS)
-                .lightLevel(p_50870_ -> 11)
-                .pushReaction(PushReaction.BLOCK)
-        )
-    );
-    public static final Block POTATO_PORTAL = register(
-        "potato_portal",
-        new PotatoPortalBlock(
-            BlockBehaviour.Properties.of()
-                .noCollission()
-                .randomTicks()
-                .strength(-1.0F)
-                .sound(SoundType.GLASS)
-                .lightLevel(p_187437_ -> 15)
-                .pushReaction(PushReaction.BLOCK)
-        )
-    );
-    public static final Block PEDESTAL = register(
-        "pedestal",
-        new PedestalBlock(
-            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_ORANGE).strength(55.0F, 1200.0F).sound(SoundType.STONE).pushReaction(PushReaction.BLOCK)
+                .lightLevel(p_50872_ -> 11)
+                .pushReaction(PushReaction.BLOCK)
         )
     );
     public static final Block CARVED_PUMPKIN = register(
@@ -2781,28 +_,6 @@
                 .pushReaction(PushReaction.DESTROY)
         )
     );
-    public static final Block POTATO_ZOMBIE_HEAD_HAT = register(
-        "poisonous_potato_zombie_head_hat",
-        new EquipablePotatoZombieHeadBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_YELLOW)
-                .strength(1.0F)
-                .sound(SoundType.CROP)
-                .isValidSpawn(Blocks::always)
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
-    public static final Block POTATO_ZOMBIE_HEAD_BLOCK = register(
-        "poisonous_potato_zombie_head_block",
-        new PotatoZombieHeadBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_YELLOW)
-                .strength(1.0F)
-                .sound(SoundType.CROP)
-                .isValidSpawn(Blocks::always)
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
     public static final Block JACK_O_LANTERN = register(
         "jack_o_lantern",
         new CarvedPumpkinBlock(
@@ -2810,7 +_,7 @@
                 .mapColor(MapColor.COLOR_ORANGE)
                 .strength(1.0F)
                 .sound(SoundType.WOOD)
-                .lightLevel(p_152692_ -> 15)
+                .lightLevel(p_50870_ -> 15)
                 .isValidSpawn(Blocks::always)
                 .pushReaction(PushReaction.DESTROY)
         )
@@ -3197,14 +_,14 @@
                 .mapColor(MapColor.COLOR_RED)
                 .instrument(NoteBlockInstrument.BASEDRUM)
                 .requiresCorrectToolForDrops()
-                .lightLevel(p_152690_ -> 7)
+                .lightLevel(p_187437_ -> 7)
                 .strength(5.0F, 1200.0F)
         )
     );
     public static final Block BREWING_STAND = register(
         "brewing_stand",
         new BrewingStandBlock(
-            BlockBehaviour.Properties.of().mapColor(MapColor.METAL).requiresCorrectToolForDrops().strength(0.5F).lightLevel(p_50847_ -> 1).noOcclusion()
+            BlockBehaviour.Properties.of().mapColor(MapColor.METAL).requiresCorrectToolForDrops().strength(0.5F).lightLevel(p_50856_ -> 1).noOcclusion()
         )
     );
     public static final Block CAULDRON = register(
@@ -3214,7 +_,7 @@
         "water_cauldron", new LayeredCauldronBlock(Biome.Precipitation.RAIN, CauldronInteraction.WATER, BlockBehaviour.Properties.ofLegacyCopy(CAULDRON))
     );
     public static final Block LAVA_CAULDRON = register(
-        "lava_cauldron", new LavaCauldronBlock(BlockBehaviour.Properties.ofLegacyCopy(CAULDRON).lightLevel(p_50854_ -> 15))
+        "lava_cauldron", new LavaCauldronBlock(BlockBehaviour.Properties.ofLegacyCopy(CAULDRON).lightLevel(p_152692_ -> 15))
     );
     public static final Block POWDER_SNOW_CAULDRON = register(
         "powder_snow_cauldron",
@@ -3226,7 +_,7 @@
             BlockBehaviour.Properties.of()
                 .mapColor(MapColor.COLOR_BLACK)
                 .noCollission()
-                .lightLevel(p_50828_ -> 15)
+                .lightLevel(p_152690_ -> 15)
                 .strength(-1.0F, 3600000.0F)
                 .noLootTable()
                 .pushReaction(PushReaction.BLOCK)
@@ -3239,7 +_,7 @@
                 .mapColor(MapColor.COLOR_GREEN)
                 .instrument(NoteBlockInstrument.BASEDRUM)
                 .sound(SoundType.GLASS)
-                .lightLevel(p_50840_ -> 1)
+                .lightLevel(p_50847_ -> 1)
                 .strength(-1.0F, 3600000.0F)
                 .noLootTable()
         )
@@ -3256,7 +_,7 @@
             BlockBehaviour.Properties.of()
                 .mapColor(MapColor.COLOR_BLACK)
                 .strength(3.0F, 9.0F)
-                .lightLevel(p_187409_ -> 1)
+                .lightLevel(p_50840_ -> 1)
                 .noOcclusion()
                 .pushReaction(PushReaction.DESTROY)
         )
@@ -3302,7 +_,7 @@
                 .instrument(NoteBlockInstrument.BASEDRUM)
                 .requiresCorrectToolForDrops()
                 .strength(22.5F, 600.0F)
-                .lightLevel(p_152688_ -> 7)
+                .lightLevel(p_50854_ -> 7)
         )
     );
     public static final Block TRIPWIRE_HOOK = register(
@@ -3322,17 +_,6 @@
                 .sound(SoundType.METAL)
         )
     );
-    public static final Block AMBER_BLOCK = register(
-        "amber_block",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_ORANGE)
-                .instrument(NoteBlockInstrument.BIT)
-                .requiresCorrectToolForDrops()
-                .strength(5.0F, 6.0F)
-                .sound(SoundType.METAL)
-        )
-    );
     public static final Block SPRUCE_STAIRS = register("spruce_stairs", legacyStair(SPRUCE_PLANKS));
     public static final Block BIRCH_STAIRS = register("birch_stairs", legacyStair(BIRCH_PLANKS));
     public static final Block JUNGLE_STAIRS = register("jungle_stairs", legacyStair(JUNGLE_PLANKS));
@@ -3349,7 +_,7 @@
                 .mapColor(MapColor.DIAMOND)
                 .instrument(NoteBlockInstrument.HAT)
                 .strength(3.0F)
-                .lightLevel(p_152686_ -> 15)
+                .lightLevel(p_50828_ -> 15)
                 .noOcclusion()
                 .isRedstoneConductor(Blocks::never)
         )
@@ -3381,7 +_,6 @@
     public static final Block POTTED_OXEYE_DAISY = register("potted_oxeye_daisy", flowerPot(OXEYE_DAISY));
     public static final Block POTTED_CORNFLOWER = register("potted_cornflower", flowerPot(CORNFLOWER));
     public static final Block POTTED_LILY_OF_THE_VALLEY = register("potted_lily_of_the_valley", flowerPot(LILY_OF_THE_VALLEY));
-    public static final Block POTTED_POTATO_FLOWER = register("potted_potato_flower", flowerPot(POTATO_FLOWER));
     public static final Block POTTED_WITHER_ROSE = register("potted_wither_rose", flowerPot(WITHER_ROSE));
     public static final Block POTTED_RED_MUSHROOM = register("potted_red_mushroom", flowerPot(RED_MUSHROOM));
     public static final Block POTTED_BROWN_MUSHROOM = register("potted_brown_mushroom", flowerPot(BROWN_MUSHROOM));
@@ -3409,7 +_,6 @@
                 .instabreak()
                 .sound(SoundType.CROP)
                 .pushReaction(PushReaction.DESTROY)
-                .offsetType(BlockBehaviour.OffsetType.POTATO)
         )
     );
     public static final Block OAK_BUTTON = register("oak_button", woodenButton(BlockSetType.OAK));
@@ -3982,7 +_,7 @@
                 .instrument(NoteBlockInstrument.HAT)
                 .strength(0.3F)
                 .sound(SoundType.GLASS)
-                .lightLevel(p_187435_ -> 15)
+                .lightLevel(p_152688_ -> 15)
                 .isRedstoneConductor(Blocks::never)
         )
     );
@@ -5214,7 +_,7 @@
         )
     );
     public static final Block END_ROD = register(
-        "end_rod", new EndRodBlock(BlockBehaviour.Properties.of().forceSolidOff().instabreak().lightLevel(p_50804_ -> 14).sound(SoundType.WOOD).noOcclusion())
+        "end_rod", new EndRodBlock(BlockBehaviour.Properties.of().forceSolidOff().instabreak().lightLevel(p_152686_ -> 14).sound(SoundType.WOOD).noOcclusion())
     );
     public static final Block CHORUS_PLANT = register(
         "chorus_plant",
@@ -5244,47 +_,6 @@
                 .isRedstoneConductor(Blocks::never)
         )
     );
-    public static final Block STRONG_ROOTS = register(
-        "strong_roots",
-        new StrongRootsBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_GREEN)
-                .forceSolidOff()
-                .strength(0.4F)
-                .sound(SoundType.WOOD)
-                .randomTicks()
-                .noOcclusion()
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
-    public static final Block POWERFUL_POTATO = register(
-        "powerful_potato",
-        new PowerfulPotatoBlock(
-            STRONG_ROOTS,
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_GREEN)
-                .forceSolidOff()
-                .randomTicks()
-                .strength(0.4F)
-                .sound(SoundType.STEM)
-                .noOcclusion()
-                .isValidSpawn(Blocks::never)
-                .pushReaction(PushReaction.DESTROY)
-                .isRedstoneConductor(Blocks::never)
-        )
-    );
-    public static final Block WEAK_ROOTS = register(
-        "weak_roots",
-        new StrongRootsBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.COLOR_GREEN)
-                .forceSolidOff()
-                .strength(0.4F)
-                .sound(SoundType.WOOD)
-                .noOcclusion()
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
     public static final Block PURPUR_BLOCK = register(
         "purpur_block",
         new Block(
@@ -5372,24 +_,13 @@
                 .isSuffocating(Blocks::always)
         )
     );
-    public static final Block POISON_PATH = register(
-        "poison_path",
-        new DirtPathBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_LIGHT_GREEN)
-                .strength(0.65F)
-                .sound(SoundType.MOSS)
-                .isViewBlocking(Blocks::always)
-                .isSuffocating(Blocks::always)
-        )
-    );
     public static final Block END_GATEWAY = register(
         "end_gateway",
         new EndGatewayBlock(
             BlockBehaviour.Properties.of()
                 .mapColor(MapColor.COLOR_BLACK)
                 .noCollission()
-                .lightLevel(p_152677_ -> 15)
+                .lightLevel(p_187435_ -> 15)
                 .strength(-1.0F, 3600000.0F)
                 .noLootTable()
                 .pushReaction(PushReaction.BLOCK)
@@ -6392,7 +_,7 @@
                 .forceSolidOn()
                 .instrument(NoteBlockInstrument.HAT)
                 .strength(3.0F)
-                .lightLevel(p_187433_ -> 15)
+                .lightLevel(p_50804_ -> 15)
                 .noOcclusion()
         )
     );
@@ -6596,12 +_,6 @@
             BlockBehaviour.Properties.of().mapColor(MapColor.STONE).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(3.5F)
         )
     );
-    public static final Block POISONOUS_POTATO_CUTTER = register(
-        "poisonous_potato_cutter",
-        new PoisonousPotatoCutterBlock(
-            BlockBehaviour.Properties.of().mapColor(MapColor.STONE).instrument(NoteBlockInstrument.BASEDRUM).requiresCorrectToolForDrops().strength(3.5F)
-        )
-    );
     public static final Block BELL = register(
         "bell",
         new BellBlock(
@@ -6623,7 +_,7 @@
                 .requiresCorrectToolForDrops()
                 .strength(3.5F)
                 .sound(SoundType.LANTERN)
-                .lightLevel(p_187431_ -> 15)
+                .lightLevel(p_152677_ -> 15)
                 .noOcclusion()
                 .pushReaction(PushReaction.DESTROY)
         )
@@ -6637,7 +_,7 @@
                 .requiresCorrectToolForDrops()
                 .strength(3.5F)
                 .sound(SoundType.LANTERN)
-                .lightLevel(p_152663_ -> 10)
+                .lightLevel(p_187433_ -> 10)
                 .noOcclusion()
                 .pushReaction(PushReaction.DESTROY)
         )
@@ -6672,21 +_,6 @@
                 .ignitedByLava()
         )
     );
-    public static final Block FRYING_TABLE = register(
-        "frying_table",
-        new CampfireBlock(
-            true,
-            0,
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.PODZOL)
-                .instrument(NoteBlockInstrument.BASS)
-                .strength(2.0F)
-                .sound(SoundType.WOOD)
-                .lightLevel(litBlockEmission(0))
-                .noOcclusion()
-                .ignitedByLava()
-        )
-    );
     public static final Block SWEET_BERRY_BUSH = register(
         "sweet_berry_bush",
         new SweetBerryBushBlock(
@@ -6797,7 +_,7 @@
     );
     public static final Block SHROOMLIGHT = register(
         "shroomlight",
-        new Block(BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_RED).strength(1.0F).sound(SoundType.SHROOMLIGHT).lightLevel(p_152651_ -> 15))
+        new Block(BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_RED).strength(1.0F).sound(SoundType.SHROOMLIGHT).lightLevel(p_187431_ -> 15))
     );
     public static final Block WEEPING_VINES = register(
         "weeping_vines",
@@ -6878,12 +_,6 @@
                 .sound(SoundType.NETHER_WOOD)
         )
     );
-    public static final Block POTATO_PLANKS = register(
-        "potato_planks",
-        new Block(
-            BlockBehaviour.Properties.of().mapColor(MapColor.EMERALD).instrument(NoteBlockInstrument.BASS).strength(2.0F, 3.0F).sound(SoundType.NETHER_WOOD)
-        )
-    );
     public static final Block CRIMSON_SLAB = register(
         "crimson_slab",
         new SlabBlock(
@@ -6904,16 +_,6 @@
                 .sound(SoundType.NETHER_WOOD)
         )
     );
-    public static final Block POTATO_SLAB = register(
-        "potato_slab",
-        new SlabBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(POTATO_PLANKS.defaultMapColor())
-                .instrument(NoteBlockInstrument.BASS)
-                .strength(2.0F, 3.0F)
-                .sound(SoundType.NETHER_WOOD)
-        )
-    );
     public static final Block CRIMSON_PRESSURE_PLATE = register(
         "crimson_pressure_plate",
         new PressurePlateBlock(
@@ -6940,19 +_,6 @@
                 .pushReaction(PushReaction.DESTROY)
         )
     );
-    public static final Block POTATO_PRESSURE_PLATE = register(
-        "potato_pressure_plate",
-        new PressurePlateBlock(
-            BlockSetType.POTATO,
-            BlockBehaviour.Properties.of()
-                .mapColor(POTATO_PLANKS.defaultMapColor())
-                .forceSolidOn()
-                .instrument(NoteBlockInstrument.BASS)
-                .noCollission()
-                .strength(0.5F)
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
     public static final Block CRIMSON_FENCE = register(
         "crimson_fence",
         new FenceBlock(
@@ -6973,16 +_,6 @@
                 .sound(SoundType.NETHER_WOOD)
         )
     );
-    public static final Block POTATO_FENCE = register(
-        "potato_fence",
-        new FenceBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(POTATO_PLANKS.defaultMapColor())
-                .instrument(NoteBlockInstrument.BASS)
-                .strength(2.0F, 3.0F)
-                .sound(SoundType.NETHER_WOOD)
-        )
-    );
     public static final Block CRIMSON_TRAPDOOR = register(
         "crimson_trapdoor",
         new TrapDoorBlock(
@@ -7007,18 +_,6 @@
                 .isValidSpawn(Blocks::never)
         )
     );
-    public static final Block POTATO_TRAPDOOR = register(
-        "potato_trapdoor",
-        new TrapDoorBlock(
-            BlockSetType.POTATO,
-            BlockBehaviour.Properties.of()
-                .mapColor(POTATO_PLANKS.defaultMapColor())
-                .instrument(NoteBlockInstrument.BASS)
-                .strength(3.0F)
-                .noOcclusion()
-                .isValidSpawn(Blocks::never)
-        )
-    );
     public static final Block CRIMSON_FENCE_GATE = register(
         "crimson_fence_gate",
         new FenceGateBlock(
@@ -7033,19 +_,10 @@
             BlockBehaviour.Properties.of().mapColor(WARPED_PLANKS.defaultMapColor()).forceSolidOn().instrument(NoteBlockInstrument.BASS).strength(2.0F, 3.0F)
         )
     );
-    public static final Block POTATO_FENCE_GATE = register(
-        "potato_fence_gate",
-        new FenceGateBlock(
-            WoodType.POTATO,
-            BlockBehaviour.Properties.of().mapColor(POTATO_PLANKS.defaultMapColor()).forceSolidOn().instrument(NoteBlockInstrument.BASS).strength(2.0F, 3.0F)
-        )
-    );
     public static final Block CRIMSON_STAIRS = register("crimson_stairs", legacyStair(CRIMSON_PLANKS));
     public static final Block WARPED_STAIRS = register("warped_stairs", legacyStair(WARPED_PLANKS));
-    public static final Block POTATO_STAIRS = register("potato_stairs", legacyStair(POTATO_PLANKS));
     public static final Block CRIMSON_BUTTON = register("crimson_button", woodenButton(BlockSetType.CRIMSON));
     public static final Block WARPED_BUTTON = register("warped_button", woodenButton(BlockSetType.WARPED));
-    public static final Block POTATO_BUTTON = register("potato_button", woodenButton(BlockSetType.POTATO));
     public static final Block CRIMSON_DOOR = register(
         "crimson_door",
         new DoorBlock(
@@ -7070,18 +_,6 @@
                 .pushReaction(PushReaction.DESTROY)
         )
     );
-    public static final Block POTATO_DOOR = register(
-        "potato_door",
-        new DoorBlock(
-            BlockSetType.POTATO,
-            BlockBehaviour.Properties.of()
-                .mapColor(POTATO_PLANKS.defaultMapColor())
-                .instrument(NoteBlockInstrument.BASS)
-                .strength(3.0F)
-                .noOcclusion()
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
     public static final Block CRIMSON_SIGN = register(
         "crimson_sign",
         new StandingSignBlock(
@@ -7106,18 +_,6 @@
                 .strength(1.0F)
         )
     );
-    public static final Block POTATO_SIGN = register(
-        "potato_sign",
-        new StandingSignBlock(
-            WoodType.POTATO,
-            BlockBehaviour.Properties.of()
-                .mapColor(POTATO_PLANKS.defaultMapColor())
-                .instrument(NoteBlockInstrument.BASS)
-                .forceSolidOn()
-                .noCollission()
-                .strength(1.0F)
-        )
-    );
     public static final Block CRIMSON_WALL_SIGN = register(
         "crimson_wall_sign",
         new WallSignBlock(
@@ -7144,19 +_,6 @@
                 .dropsLike(WARPED_SIGN)
         )
     );
-    public static final Block POTATO_WALL_SIGN = register(
-        "potato_wall_sign",
-        new WallSignBlock(
-            WoodType.POTATO,
-            BlockBehaviour.Properties.of()
-                .mapColor(POTATO_PLANKS.defaultMapColor())
-                .instrument(NoteBlockInstrument.BASS)
-                .forceSolidOn()
-                .noCollission()
-                .strength(1.0F)
-                .dropsLike(POTATO_SIGN)
-        )
-    );
     public static final Block STRUCTURE_BLOCK = register(
         "structure_block",
         new StructureBlock(
@@ -7232,7 +_,7 @@
                 .instrument(NoteBlockInstrument.BASEDRUM)
                 .requiresCorrectToolForDrops()
                 .strength(50.0F, 1200.0F)
-                .lightLevel(p_220873_ -> 10)
+                .lightLevel(p_152663_ -> 10)
         )
     );
     public static final Block RESPAWN_ANCHOR = register(
@@ -7457,7 +_,7 @@
     public static final Block SMALL_AMETHYST_BUD = register(
         "small_amethyst_bud",
         new AmethystClusterBlock(
-            3.0F, 4.0F, BlockBehaviour.Properties.ofLegacyCopy(AMETHYST_CLUSTER).sound(SoundType.SMALL_AMETHYST_BUD).lightLevel(p_187406_ -> 1)
+            3.0F, 4.0F, BlockBehaviour.Properties.ofLegacyCopy(AMETHYST_CLUSTER).sound(SoundType.SMALL_AMETHYST_BUD).lightLevel(p_187409_ -> 1)
         )
     );
     public static final Block TUFF = register(
@@ -7543,7 +_,7 @@
                 .mapColor(MapColor.COLOR_CYAN)
                 .strength(1.5F)
                 .sound(SoundType.SCULK_SENSOR)
-                .lightLevel(p_50892_ -> 1)
+                .lightLevel(p_187406_ -> 1)
                 .emissiveRendering((p_187412_, p_187413_, p_187414_) -> SculkSensorBlock.getPhase(p_187412_) == SculkSensorPhase.ACTIVE)
         )
     );
@@ -7568,7 +_,7 @@
     public static final Block SCULK_CATALYST = register(
         "sculk_catalyst",
         new SculkCatalystBlock(
-            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_BLACK).strength(3.0F, 3.0F).sound(SoundType.SCULK_CATALYST).lightLevel(p_220871_ -> 6)
+            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_BLACK).strength(3.0F, 3.0F).sound(SoundType.SCULK_CATALYST).lightLevel(p_152651_ -> 6)
         )
     );
     public static final Block SCULK_SHRIEKER = register(
@@ -7927,7 +_,6 @@
     public static final Block POINTED_DRIPSTONE = register(
         "pointed_dripstone",
         new PointedDripstoneBlock(
-            false,
             BlockBehaviour.Properties.of()
                 .mapColor(MapColor.TERRACOTTA_BROWN)
                 .forceSolidOn()
@@ -7942,113 +_,6 @@
                 .isRedstoneConductor(Blocks::never)
         )
     );
-    public static final Block POTATO_BUD = register(
-        "potato_bud",
-        new PointedDripstoneBlock(
-            true,
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.CRIMSON_STEM)
-                .forceSolidOn()
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .noOcclusion()
-                .sound(SoundType.NETHER_WOOD)
-                .randomTicks()
-                .strength(1.5F, 3.0F)
-                .dynamicShape()
-                .offsetType(BlockBehaviour.OffsetType.XZ)
-                .pushReaction(PushReaction.DESTROY)
-                .isRedstoneConductor(Blocks::never)
-        )
-    );
-    public static final Block POTATO_STEM = register(
-        "potato_stem",
-        new RotatedPillarBlock(
-            BlockBehaviour.Properties.of().mapColor(p_340228_ -> MapColor.EMERALD).instrument(NoteBlockInstrument.BASS).strength(2.0F).sound(SoundType.STEM)
-        )
-    );
-    public static final Block POTATO_LEAVES = register(
-        "potato_leaves",
-        new LeavesBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.PLANT)
-                .strength(0.2F)
-                .randomTicks()
-                .sound(SoundType.GRASS)
-                .noOcclusion()
-                .isValidSpawn(Blocks::ocelotOrParrot)
-                .isSuffocating(Blocks::never)
-                .isViewBlocking(Blocks::never)
-                .ignitedByLava()
-                .pushReaction(PushReaction.DESTROY)
-                .isRedstoneConductor(Blocks::never)
-        )
-    );
-    public static final Block POTATO_FRUIT = register(
-        "potato_fruit",
-        new Block(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.PLANT)
-                .strength(1.0F)
-                .sound(SoundType.NETHER_WOOD)
-                .pushReaction(PushReaction.DESTROY)
-                .lightLevel(p_220869_ -> 15)
-        )
-    );
-    public static final Block POTATO_PEDICULE = register(
-        "potato_pedicule",
-        new ChainBlock(BlockBehaviour.Properties.of().mapColor(MapColor.PLANT).strength(0.2F).sound(SoundType.NETHER_WOOD).pushReaction(PushReaction.DESTROY))
-    );
-    public static final Block POTATO_SPROUTS = register(
-        "potato_sprouts",
-        new PotatoSproutsBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.TERRACOTTA_WHITE)
-                .sound(SoundType.NETHER_SPROUTS)
-                .noOcclusion()
-                .dynamicShape()
-                .offsetType(BlockBehaviour.OffsetType.XZ)
-                .pushReaction(PushReaction.DESTROY)
-        )
-    );
-    public static final Block POTONE_REDSTONE_ORE = register(
-        "potone_redstone_ore",
-        new RedStoneOreBlock(
-            BlockBehaviour.Properties.ofLegacyCopy(REDSTONE_ORE).mapColor(MapColor.TERRACOTTA_YELLOW).sound(SoundType.POTONE).strength(4.5F, 3.0F)
-        )
-    );
-    public static final Block POTONE_LAPIS_ORE = register(
-        "potone_lapis_ore",
-        new DropExperienceBlock(
-            UniformInt.of(2, 5), BlockBehaviour.Properties.ofLegacyCopy(LAPIS_ORE).mapColor(MapColor.COLOR_BLUE).sound(SoundType.POTONE).strength(4.5F, 3.0F)
-        )
-    );
-    public static final Block POTONE_DIAMOND_ORE = register(
-        "potone_diamond_ore",
-        new DropExperienceBlock(
-            UniformInt.of(3, 7),
-            BlockBehaviour.Properties.ofLegacyCopy(DIAMOND_ORE).mapColor(MapColor.COLOR_LIGHT_BLUE).sound(SoundType.POTONE).strength(4.5F, 3.0F)
-        )
-    );
-    public static final Block POTONE_GOLD_ORE = register(
-        "potone_gold_ore",
-        new DropExperienceBlock(
-            UniformInt.of(2, 5), BlockBehaviour.Properties.ofLegacyCopy(GOLD_ORE).mapColor(MapColor.COLOR_YELLOW).sound(SoundType.POTONE).strength(4.5F, 3.0F)
-        )
-    );
-    public static final Block POTONE_IRON_ORE = register(
-        "potone_iron_ore",
-        new DropExperienceBlock(
-            UniformInt.of(1, 2),
-            BlockBehaviour.Properties.ofLegacyCopy(IRON_ORE).mapColor(MapColor.TERRACOTTA_WHITE).sound(SoundType.POTONE).strength(4.5F, 3.0F)
-        )
-    );
-    public static final Block POTONE_COPPER_ORE = register(
-        "potone_copper_ore",
-        new DropExperienceBlock(
-            UniformInt.of(1, 3),
-            BlockBehaviour.Properties.ofLegacyCopy(COPPER_ORE).mapColor(MapColor.COLOR_ORANGE).sound(SoundType.POTONE).strength(4.5F, 3.0F)
-        )
-    );
     public static final Block DRIPSTONE_BLOCK = register(
         "dripstone_block",
         new Block(
@@ -8230,17 +_,6 @@
     public static final Block POLISHED_DEEPSLATE_WALL = register(
         "polished_deepslate_wall", new WallBlock(BlockBehaviour.Properties.ofLegacyCopy(POLISHED_DEEPSLATE).forceSolidOn())
     );
-    public static final Block POTATO_REFINERY = register(
-        "potato_refinery",
-        new PotatoRefineryBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.POTATO)
-                .instrument(NoteBlockInstrument.BASEDRUM)
-                .requiresCorrectToolForDrops()
-                .strength(3.5F)
-                .lightLevel(litBlockEmission(13))
-        )
-    );
     public static final Block DEEPSLATE_TILES = register(
         "deepslate_tiles", new Block(BlockBehaviour.Properties.ofLegacyCopy(COBBLED_DEEPSLATE).sound(SoundType.DEEPSLATE_TILES))
     );
@@ -8298,18 +_,18 @@
     public static final Block POTTED_FLOWERING_AZALEA = register("potted_flowering_azalea_bush", flowerPot(FLOWERING_AZALEA));
     public static final Block OCHRE_FROGLIGHT = register(
         "ochre_froglight",
-        new RotatedPillarBlock(BlockBehaviour.Properties.of().mapColor(MapColor.SAND).strength(0.3F).lightLevel(p_220867_ -> 15).sound(SoundType.FROGLIGHT))
+        new RotatedPillarBlock(BlockBehaviour.Properties.of().mapColor(MapColor.SAND).strength(0.3F).lightLevel(p_220873_ -> 15).sound(SoundType.FROGLIGHT))
     );
     public static final Block VERDANT_FROGLIGHT = register(
         "verdant_froglight",
         new RotatedPillarBlock(
-            BlockBehaviour.Properties.of().mapColor(MapColor.GLOW_LICHEN).strength(0.3F).lightLevel(p_50755_ -> 15).sound(SoundType.FROGLIGHT)
+            BlockBehaviour.Properties.of().mapColor(MapColor.GLOW_LICHEN).strength(0.3F).lightLevel(p_220871_ -> 15).sound(SoundType.FROGLIGHT)
         )
     );
     public static final Block PEARLESCENT_FROGLIGHT = register(
         "pearlescent_froglight",
         new RotatedPillarBlock(
-            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_PINK).strength(0.3F).lightLevel(p_340231_ -> 15).sound(SoundType.FROGLIGHT)
+            BlockBehaviour.Properties.of().mapColor(MapColor.COLOR_PINK).strength(0.3F).lightLevel(p_220869_ -> 15).sound(SoundType.FROGLIGHT)
         )
     );
     public static final Block FROGSPAWN = register(
@@ -8386,20 +_,6 @@
                 .requiredFeatures(FeatureFlags.UPDATE_1_21)
         )
     );
-    public static final Block BIG_BRAIN = register(
-        "big_brain",
-        new BigBrainBlock(
-            BlockBehaviour.Properties.of()
-                .mapColor(MapColor.PLANT)
-                .instrument(NoteBlockInstrument.BASS)
-                .strength(2.0F, 3.0F)
-                .sound(SoundType.SLIME_BLOCK)
-                .ignitedByLava()
-        )
-    );
-    public static final Map<DyeColor, Block> POTATO_PEELS_BLOCK_MAP;
-    public static final Block CORRUPTED_POTATO_PEELS_BLOCK;
-    public static final Block POTATO_BATTERY;
 
     private static ToIntFunction<BlockState> litBlockEmission(int p_50760_) {
         return p_50763_ -> p_50763_.getValue(BlockStateProperties.LIT) ? p_50760_ : 0;
@@ -8574,49 +_,10 @@
     }
 
     static {
-        Builder<DyeColor, Block> builder = ImmutableMap.builder();
-
-        for(DyeColor dyecolor : DyeColor.values()) {
-            String s = dyecolor == DyeColor.WHITE ? "potato_peels_block" : dyecolor.getName() + "_potato_peels_block";
-            builder.put(
-                dyecolor,
-                register(
-                    s,
-                    new PotatoPeelsBlock(
-                        BlockBehaviour.Properties.of()
-                            .mapColor(dyecolor)
-                            .instrument(NoteBlockInstrument.BANJO)
-                            .strength(2.0F)
-                            .sound(SoundType.POTONE)
-                            .ignitedByLava(),
-                        dyecolor
-                    )
-                )
-            );
-        }
-
-        POTATO_PEELS_BLOCK_MAP = builder.build();
-        CORRUPTED_POTATO_PEELS_BLOCK = register(
-            "corrupted_potato_peels_block",
-            new CorruptedPotatoPeelsBlock(
-                BlockBehaviour.Properties.of().mapColor(MapColor.WARPED_NYLIUM).sound(SoundType.CORRUPTED_PEELGRASS).randomTicks().strength(20.0F, 500.0F)
-            )
-        );
-        POTATO_BATTERY = register(
-            "potato_battery",
-            new PotatoBatteryBlock(
-                BlockBehaviour.Properties.of()
-                    .strength(3.0F)
-                    .mapColor(MapColor.TERRACOTTA_ORANGE)
-                    .sound(SoundType.TERREDEPOMME)
-                    .isRedstoneConductor(Blocks::never)
-                    .lightLevel(p_340229_ -> p_340229_.getValue(PotatoBatteryBlock.INVERTED) ? 8 : 0)
-            )
-        );
-
-        for(Block block : BuiltInRegistries.BLOCK) {
-            for(BlockState blockstate : block.getStateDefinition().getPossibleStates()) {
-                Block.BLOCK_STATE_REGISTRY.add(blockstate);
+        for (Block block : BuiltInRegistries.BLOCK) {
+            for (BlockState blockstate : block.getStateDefinition().getPossibleStates()) {
+                // Neo: comment out, it's done in NeoForgeRegistryCallbacks
+                //Block.BLOCK_STATE_REGISTRY.add(blockstate);
                 blockstate.initCache();
             }
 
