--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -83,7 +_,7 @@
     public static final MapCodec<Block> CODEC = simpleCodec(Block::new);
     private static final Logger LOGGER = LogUtils.getLogger();
     private final Holder.Reference<Block> builtInRegistryHolder = BuiltInRegistries.BLOCK.createIntrusiveHolder(this);
-    public static final IdMapper<BlockState> BLOCK_STATE_REGISTRY = new IdMapper<>();
+    public static final IdMapper<BlockState> BLOCK_STATE_REGISTRY = net.neoforged.neoforge.registries.GameData.getBlockStateIDMap();
     private static final LoadingCache<VoxelShape, Boolean> SHAPE_FULL_BLOCK_CACHE = CacheBuilder.newBuilder()
         .maximumSize(512L)
         .weakKeys()
@@ -242,7 +_,7 @@
         this.createBlockStateDefinition(builder);
         this.stateDefinition = builder.create(Block::defaultBlockState, BlockState::new);
         this.registerDefaultState(this.stateDefinition.any());
-        if (SharedConstants.IS_RUNNING_IN_IDE) {
+        if (SharedConstants.IS_RUNNING_IN_IDE && false) {
             String className = this.getClass().getSimpleName();
             if (!className.endsWith("Block")) {
                 LOGGER.error("Block classes should end with Block and {} doesn't.", className);
@@ -295,12 +_,22 @@
         }
     }
 
+    /**
+     * @deprecated Neo: use overload with level context instead
+     */
+    @Deprecated
     public static boolean shouldRenderFace(BlockState state, BlockState neighborState, Direction direction) {
+        return shouldRenderFace(net.minecraft.world.level.EmptyBlockGetter.INSTANCE, BlockPos.ZERO, state, neighborState, direction);
+    }
+
+    public static boolean shouldRenderFace(BlockGetter level, BlockPos pos, BlockState state, BlockState neighborState, Direction direction) {
         VoxelShape occluder = neighborState.getFaceOcclusionShape(direction.getOpposite());
         if (occluder == Shapes.block()) {
             return false;
         } else if (state.skipRendering(neighborState, direction)) {
             return false;
+        } else if (neighborState.hidesNeighborFace(level, pos.relative(direction), state, direction.getOpposite()) && state.supportsExternalFaceHiding()) {
+            return false;
         } else if (occluder == Shapes.empty()) {
             return true;
         } else {
@@ -522,6 +_,7 @@
     public void handlePrecipitation(BlockState state, Level level, BlockPos pos, Biome.Precipitation precipitation) {
     }
 
+    @Deprecated //Forge: Use more sensitive version
     public boolean dropFromExplosion(Explosion explosion) {
         return true;
     }
@@ -604,6 +_,35 @@
         return state.setValue(property, (T)value);
     }
 
+    /**
+     * Neo: Short-lived holder of dropped item entities. Used mainly for Neo hooks and event logic.
+     * <p>
+     * When not null, records all item entities from {@link #popResource(Level, Supplier, ItemStack)} instead of adding them to the world.
+     */
+    @Nullable
+    private static List<ItemEntity> capturedDrops = null;
+
+    /**
+     * Initializes {@link #capturedDrops}, starting the drop capture process.
+     * <p>
+     * Must only be called on the server thread.
+     */
+    private static void beginCapturingDrops() {
+        capturedDrops = new java.util.ArrayList<>();
+    }
+
+    /**
+     * Ends the drop capture process by setting {@link #capturedDrops} to null and returning the old list.
+     * <p>
+     * Must only be called on the server thread.
+     */
+    private static List<ItemEntity> stopCapturingDrops() {
+        List<ItemEntity> drops = capturedDrops;
+        capturedDrops = null;
+        return drops;
+    }
+
+    /** @deprecated */
     @Deprecated
     public Holder.Reference<Block> builtInRegistryHolder() {
         return this.builtInRegistryHolder;
