--- a/net/minecraft/world/level/block/BrushableBlock.java
+++ b/net/minecraft/world/level/block/BrushableBlock.java
@@ -79,8 +_,7 @@
 
     @Override
     public void tick(BlockState p_277544_, ServerLevel p_277779_, BlockPos p_278019_, RandomSource p_277471_) {
-        BlockEntity blockentity = p_277779_.getBlockEntity(p_278019_);
-        if (blockentity instanceof BrushableBlockEntity brushableblockentity) {
+        if (p_277779_.getBlockEntity(p_278019_) instanceof BrushableBlockEntity brushableblockentity) {
             brushableblockentity.checkReset();
         }
 
