--- a/net/minecraft/world/level/block/CalibratedSculkSensorBlock.java
+++ b/net/minecraft/world/level/block/CalibratedSculkSensorBlock.java
@@ -45,8 +_,8 @@
             ? createTickerHelper(
                 p_277641_,
                 BlockEntityType.CALIBRATED_SCULK_SENSOR,
-                (p_340233_, p_340234_, p_340235_, p_340236_) -> VibrationSystem.Ticker.tick(
-                        p_340233_, p_340236_.getVibrationData(), p_340236_.getVibrationUser()
+                (p_333671_, p_333672_, p_333673_, p_333674_) -> VibrationSystem.Ticker.tick(
+                        p_333671_, p_333674_.getVibrationData(), p_333674_.getVibrationUser()
                     )
             )
             : null;
