--- a/net/minecraft/world/level/block/CampfireBlock.java
+++ b/net/minecraft/world/level/block/CampfireBlock.java
@@ -60,7 +_,6 @@
                 .apply(p_308808_, CampfireBlock::new)
     );
     protected static final VoxelShape SHAPE = Block.box(0.0, 0.0, 0.0, 16.0, 7.0, 16.0);
-    protected static final VoxelShape SHAPE_FRYING = Block.box(0.0, 0.0, 0.0, 16.0, 12.0, 16.0);
     public static final BooleanProperty LIT = BlockStateProperties.LIT;
     public static final BooleanProperty SIGNAL_FIRE = BlockStateProperties.SIGNAL_FIRE;
     public static final BooleanProperty WATERLOGGED = BlockStateProperties.WATERLOGGED;
@@ -93,8 +_,7 @@
     protected ItemInteractionResult useItemOn(
         ItemStack p_316347_, BlockState p_51274_, Level p_51275_, BlockPos p_51276_, Player p_51277_, InteractionHand p_51278_, BlockHitResult p_51279_
     ) {
-        BlockEntity blockentity = p_51275_.getBlockEntity(p_51276_);
-        if (blockentity instanceof CampfireBlockEntity campfireblockentity) {
+        if (p_51275_.getBlockEntity(p_51276_) instanceof CampfireBlockEntity campfireblockentity) {
             ItemStack itemstack = p_51277_.getItemInHand(p_51278_);
             Optional<RecipeHolder<CampfireCookingRecipe>> optional = campfireblockentity.getCookableRecipe(itemstack);
             if (optional.isPresent()) {
@@ -164,7 +_,7 @@
 
     @Override
     protected VoxelShape getShape(BlockState p_51309_, BlockGetter p_51310_, BlockPos p_51311_, CollisionContext p_51312_) {
-        return p_51309_.is(Blocks.FRYING_TABLE) ? SHAPE_FRYING : SHAPE;
+        return SHAPE;
     }
 
     @Override
@@ -189,7 +_,7 @@
             }
 
             if (this.spawnParticles && p_220921_.nextInt(5) == 0) {
-                for(int i = 0; i < p_220921_.nextInt(1) + 1; ++i) {
+                for (int i = 0; i < p_220921_.nextInt(1) + 1; i++) {
                     p_220919_.addParticle(
                         ParticleTypes.LAVA,
                         (double)p_220920_.getX() + 0.5,
@@ -206,7 +_,7 @@
 
     public static void dowse(@Nullable Entity p_152750_, LevelAccessor p_152751_, BlockPos p_152752_, BlockState p_152753_) {
         if (p_152751_.isClientSide()) {
-            for(int i = 0; i < 20; ++i) {
+            for (int i = 0; i < 20; i++) {
                 makeParticles((Level)p_152751_, p_152752_, p_152753_.getValue(SIGNAL_FIRE), true);
             }
         }
@@ -278,14 +_,14 @@
     }
 
     public static boolean isSmokeyPos(Level p_51249_, BlockPos p_51250_) {
-        for(int i = 1; i <= 5; ++i) {
+        for (int i = 1; i <= 5; i++) {
             BlockPos blockpos = p_51250_.below(i);
             BlockState blockstate = p_51249_.getBlockState(blockpos);
             if (isLitCampfire(blockstate)) {
                 return true;
             }
 
-            boolean flag = Shapes.joinIsNotEmpty(VIRTUAL_FENCE_POST, blockstate.getCollisionShape(p_51249_, p_51250_, CollisionContext.empty()), BooleanOp.AND);
+            boolean flag = Shapes.joinIsNotEmpty(VIRTUAL_FENCE_POST, blockstate.getCollisionShape(p_51249_, blockpos, CollisionContext.empty()), BooleanOp.AND); // FORGE: Fix MC-201374
             if (flag) {
                 BlockState blockstate1 = p_51249_.getBlockState(blockpos.below());
                 return isLitCampfire(blockstate1);
