--- a/net/minecraft/world/level/block/CandleBlock.java
+++ b/net/minecraft/world/level/block/CandleBlock.java
@@ -124,7 +_,7 @@
 
     @Override
     protected VoxelShape getShape(BlockState p_152817_, BlockGetter p_152818_, BlockPos p_152819_, CollisionContext p_152820_) {
-        switch(p_152817_.getValue(CANDLES)) {
+        switch (p_152817_.getValue(CANDLES)) {
             case 1:
             default:
                 return ONE_AABB;
