--- a/net/minecraft/world/level/block/CeilingHangingSignBlock.java
+++ b/net/minecraft/world/level/block/CeilingHangingSignBlock.java
@@ -82,10 +_,8 @@
     protected ItemInteractionResult useItemOn(
         ItemStack p_316602_, BlockState p_316839_, Level p_316805_, BlockPos p_316894_, Player p_316202_, InteractionHand p_316538_, BlockHitResult p_316895_
     ) {
-        BlockEntity blockentity = p_316805_.getBlockEntity(p_316894_);
-        if (blockentity instanceof SignBlockEntity signblockentity && this.shouldTryToChainAnotherHangingSign(p_316202_, p_316895_, signblockentity, p_316602_)
-            )
-         {
+        if (p_316805_.getBlockEntity(p_316894_) instanceof SignBlockEntity signblockentity
+            && this.shouldTryToChainAnotherHangingSign(p_316202_, p_316895_, signblockentity, p_316602_)) {
             return ItemInteractionResult.SKIP_DEFAULT_BLOCK_INTERACTION;
         }
 
