--- a/net/minecraft/world/level/block/ChestBlock.java
+++ b/net/minecraft/world/level/block/ChestBlock.java
@@ -9,7 +_,6 @@
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
-import net.minecraft.core.component.DataComponents;
 import net.minecraft.network.chat.Component;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
@@ -27,7 +_,6 @@
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.inventory.ChestMenu;
-import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.Level;
@@ -171,7 +_,7 @@
         if (p_51569_.getValue(TYPE) == ChestType.SINGLE) {
             return AABB;
         } else {
-            switch(getConnectedDirection(p_51569_)) {
+            switch (getConnectedDirection(p_51569_)) {
                 case NORTH:
                 default:
                     return NORTH_AABB;
@@ -248,16 +_,6 @@
                 PiglinAi.angerNearbyPiglins(p_51534_, true);
             }
 
-            Container container = getContainer(this, p_51531_, p_51532_, p_51533_, false);
-            if (container != null) {
-                container.forEach(p_340240_ -> {
-                    if (p_340240_.has(DataComponents.VIEWS)) {
-                        p_340240_.set(DataComponents.VIEWS, p_340240_.getOrDefault(DataComponents.VIEWS, Integer.valueOf(0)) + 1);
-                        p_340240_.getItem().onViewedInContainer(p_340240_, p_51532_, p_51533_, container);
-                    }
-                });
-            }
-
             return InteractionResult.CONSUME;
         }
     }
@@ -272,7 +_,7 @@
 
     @Nullable
     public static Container getContainer(ChestBlock p_51512_, BlockState p_51513_, Level p_51514_, BlockPos p_51515_, boolean p_51516_) {
-        return p_51512_.combine(p_51513_, p_51514_, p_51515_, p_51516_).<Optional<Container>>apply(CHEST_COMBINER).orElse(null);
+        return p_51512_.combine(p_51513_, p_51514_, p_51515_, p_51516_).apply(CHEST_COMBINER).orElse(null);
     }
 
     @Override
@@ -294,7 +_,7 @@
     @Nullable
     @Override
     protected MenuProvider getMenuProvider(BlockState p_51574_, Level p_51575_, BlockPos p_51576_) {
-        return this.combine(p_51574_, p_51575_, p_51576_, false).<Optional<MenuProvider>>apply(MENU_PROVIDER_COMBINER).orElse(null);
+        return this.combine(p_51574_, p_51575_, p_51576_, false).apply(MENU_PROVIDER_COMBINER).orElse(null);
     }
 
     public static DoubleBlockCombiner.Combiner<ChestBlockEntity, Float2FloatFunction> opennessCombiner(final LidBlockEntity p_51518_) {
@@ -346,7 +_,7 @@
             )
         );
         if (!list.isEmpty()) {
-            for(Cat cat : list) {
+            for (Cat cat : list) {
                 if (cat.isInSittingPose()) {
                     return true;
                 }
@@ -373,7 +_,8 @@
 
     @Override
     protected BlockState mirror(BlockState p_51549_, Mirror p_51550_) {
-        return p_51549_.rotate(p_51550_.getRotation(p_51549_.getValue(FACING)));
+        BlockState rotated = p_51549_.rotate(p_51550_.getRotation(p_51549_.getValue(FACING)));
+        return p_51550_ == Mirror.NONE ? rotated : rotated.setValue(TYPE, rotated.getValue(TYPE).getOpposite());  // Forge: Fixed MC-134110 Structure mirroring breaking apart double chests
     }
 
     @Override
