--- a/net/minecraft/world/level/block/ChorusPlantBlock.java
+++ b/net/minecraft/world/level/block/ChorusPlantBlock.java
@@ -80,7 +_,7 @@
         BlockState blockstate = p_51725_.getBlockState(p_51726_.below());
         boolean flag = !p_51725_.getBlockState(p_51726_.above()).isAir() && !blockstate.isAir();
 
-        for(Direction direction : Direction.Plane.HORIZONTAL) {
+        for (Direction direction : Direction.Plane.HORIZONTAL) {
             BlockPos blockpos = p_51726_.relative(direction);
             BlockState blockstate1 = p_51725_.getBlockState(blockpos);
             if (blockstate1.is(this)) {
