--- a/net/minecraft/world/level/block/ChorusPlantBlock.java
+++ b/net/minecraft/world/level/block/ChorusPlantBlock.java
@@ -49,7 +_,8 @@
         BlockState blockstate4 = level.getBlockState(pos.south());
         BlockState blockstate5 = level.getBlockState(pos.west());
         Block block = defaultState.getBlock();
-        return defaultState.trySetValue(DOWN, blockstate.is(block) || blockstate.is(Blocks.CHORUS_FLOWER) || blockstate.is(Blocks.END_STONE))
+        var soilDecision = blockstate.canSustainPlant(level, pos.below(), Direction.UP, defaultState);
+        return defaultState.trySetValue(DOWN, blockstate.is(block) || blockstate.is(Blocks.CHORUS_FLOWER) || blockstate.is(Blocks.END_STONE) || soilDecision.isTrue())
             .trySetValue(UP, blockstate1.is(block) || blockstate1.is(Blocks.CHORUS_FLOWER))
             .trySetValue(NORTH, blockstate2.is(block) || blockstate2.is(Blocks.CHORUS_FLOWER))
             .trySetValue(EAST, blockstate3.is(block) || blockstate3.is(Blocks.CHORUS_FLOWER))
@@ -73,6 +_,12 @@
             return super.updateShape(state, level, ticks, pos, directionToNeighbour, neighbourPos, neighbourState, random);
         } else {
             boolean flag = neighbourState.is(this) || neighbourState.is(Blocks.CHORUS_FLOWER) || directionToNeighbour == Direction.DOWN && neighbourState.is(Blocks.END_STONE);
+            if (directionToNeighbour == Direction.DOWN) {
+                var soilDecision = neighbourState.canSustainPlant(level, neighbourPos.relative(directionToNeighbour), directionToNeighbour.getOpposite(), state);
+                if (!soilDecision.isDefault()) {
+                    flag = soilDecision.isTrue();
+                }
+            }
             return state.setValue(PROPERTY_BY_DIRECTION.get(directionToNeighbour), flag);
         }
     }
@@ -104,6 +_,8 @@
             }
         }
 
+        var soilDecision = blockstate.canSustainPlant(level, pos.below(), Direction.UP, state);
+        if (!soilDecision.isDefault()) return soilDecision.isTrue();
         return blockstate.is(this) || blockstate.is(Blocks.END_STONE);
     }
 
