--- a/net/minecraft/world/level/block/CocoaBlock.java
+++ b/net/minecraft/world/level/block/CocoaBlock.java
@@ -64,10 +_,11 @@
 
     @Override
     protected void randomTick(BlockState p_221000_, ServerLevel p_221001_, BlockPos p_221002_, RandomSource p_221003_) {
-        if (p_221001_.random.nextInt(5) == 0) {
+        if (true) {
             int i = p_221000_.getValue(AGE);
-            if (i < 2) {
+            if (i < 2 && net.neoforged.neoforge.common.CommonHooks.onCropsGrowPre(p_221001_, p_221002_, p_221000_, p_221001_.random.nextInt(5) == 0)) {
                 p_221001_.setBlock(p_221002_, p_221000_.setValue(AGE, Integer.valueOf(i + 1)), 2);
+                net.neoforged.neoforge.common.CommonHooks.onCropsGrowPost(p_221001_, p_221002_, p_221000_);
             }
         }
     }
@@ -81,7 +_,7 @@
     @Override
     protected VoxelShape getShape(BlockState p_51787_, BlockGetter p_51788_, BlockPos p_51789_, CollisionContext p_51790_) {
         int i = p_51787_.getValue(AGE);
-        switch((Direction)p_51787_.getValue(FACING)) {
+        switch ((Direction)p_51787_.getValue(FACING)) {
             case SOUTH:
                 return SOUTH_AABB[i];
             case NORTH:
@@ -101,7 +_,7 @@
         LevelReader levelreader = p_51750_.getLevel();
         BlockPos blockpos = p_51750_.getClickedPos();
 
-        for(Direction direction : p_51750_.getNearestLookingDirections()) {
+        for (Direction direction : p_51750_.getNearestLookingDirections()) {
             if (direction.getAxis().isHorizontal()) {
                 blockstate = blockstate.setValue(FACING, direction);
                 if (blockstate.canSurvive(levelreader, blockpos)) {
