--- a/net/minecraft/world/level/block/CropBlock.java
+++ b/net/minecraft/world/level/block/CropBlock.java
@@ -54,7 +_,7 @@
 
     @Override
     protected boolean mayPlaceOn(BlockState p_52302_, BlockGetter p_52303_, BlockPos p_52304_) {
-        return p_52302_.is(Blocks.FARMLAND) || p_52302_.is(Blocks.POISON_FARMLAND);
+        return p_52302_.is(Blocks.FARMLAND);
     }
 
     protected IntegerProperty getAgeProperty() {
@@ -69,7 +_,7 @@
         return p_52306_.getValue(this.getAgeProperty());
     }
 
-    public BlockState getStateForAge(int p_52290_, BlockState p_341287_) {
+    public BlockState getStateForAge(int p_52290_) {
         return this.defaultBlockState().setValue(this.getAgeProperty(), Integer.valueOf(p_52290_));
     }
 
@@ -84,16 +_,14 @@
 
     @Override
     protected void randomTick(BlockState p_221050_, ServerLevel p_221051_, BlockPos p_221052_, RandomSource p_221053_) {
+        if (!p_221051_.isAreaLoaded(p_221052_, 1)) return; // Forge: prevent loading unloaded chunks when checking neighbor's light
         if (p_221051_.getRawBrightness(p_221052_, 0) >= 9) {
             int i = this.getAge(p_221050_);
             if (i < this.getMaxAge()) {
                 float f = getGrowthSpeed(this, p_221051_, p_221052_);
-                if (p_221050_.is(Blocks.POTATOES)) {
-                    if (p_221053_.nextInt((int)(6.25F / f) + 1) == 0) {
-                        p_221051_.setBlock(p_221052_, this.getStateForAge(i + 1, p_221050_), 2);
-                    }
-                } else if (p_221053_.nextInt((int)(25.0F / f) + 1) == 0) {
-                    p_221051_.setBlock(p_221052_, this.getStateForAge(i + 1, p_221050_), 2);
+                if (net.neoforged.neoforge.common.CommonHooks.onCropsGrowPre(p_221051_, p_221052_, p_221050_, p_221053_.nextInt((int)(25.0F / f) + 1) == 0)) {
+                    p_221051_.setBlock(p_221052_, this.getStateForAge(i + 1), 2);
+                    net.neoforged.neoforge.common.CommonHooks.onCropsGrowPost(p_221051_, p_221052_, p_221050_);
                 }
             }
         }
@@ -106,7 +_,7 @@
             i = j;
         }
 
-        p_52264_.setBlock(p_52265_, this.getStateForAge(i, p_52266_), 2);
+        p_52264_.setBlock(p_52265_, this.getStateForAge(i), 2);
     }
 
     protected int getBonemealAgeIncrease(Level p_52262_) {
@@ -117,18 +_,13 @@
         float f = 1.0F;
         BlockPos blockpos = p_52275_.below();
 
-        for(int i = -1; i <= 1; ++i) {
-            for(int j = -1; j <= 1; ++j) {
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
                 float f1 = 0.0F;
-                BlockPos blockpos1 = blockpos.offset(i, 0, j);
-                BlockState blockstate = p_52274_.getBlockState(blockpos1);
-                if (p_52273_ instanceof BushBlock bushblock && bushblock.mayPlaceOn(blockstate, p_52274_, blockpos1)) {
+                BlockState blockstate = p_52274_.getBlockState(blockpos.offset(i, 0, j));
+                if (blockstate.canSustainPlant(p_52274_, blockpos.offset(i, 0, j), net.minecraft.core.Direction.UP, (net.neoforged.neoforge.common.IPlantable) p_52273_)) {
                     f1 = 1.0F;
-                    if ((blockstate.is(Blocks.FARMLAND) || blockstate.is(Blocks.POISON_FARMLAND)) && blockstate.getValue(FarmBlock.MOISTURE) > 0) {
-                        f1 = 3.0F;
-                    }
-
-                    if (blockstate.is(Blocks.CORRUPTED_PEELGRASS_BLOCK)) {
+                    if (blockstate.isFertile(p_52274_, p_52275_.offset(i, 0, j))) {
                         f1 = 3.0F;
                     }
                 }
@@ -141,20 +_,20 @@
             }
         }
 
-        BlockPos blockpos2 = p_52275_.north();
-        BlockPos blockpos3 = p_52275_.south();
-        BlockPos blockpos4 = p_52275_.west();
-        BlockPos blockpos5 = p_52275_.east();
-        boolean flag1 = p_52274_.getBlockState(blockpos4).is(p_52273_) || p_52274_.getBlockState(blockpos5).is(p_52273_);
-        boolean flag2 = p_52274_.getBlockState(blockpos2).is(p_52273_) || p_52274_.getBlockState(blockpos3).is(p_52273_);
-        if (flag1 && flag2) {
+        BlockPos blockpos1 = p_52275_.north();
+        BlockPos blockpos2 = p_52275_.south();
+        BlockPos blockpos3 = p_52275_.west();
+        BlockPos blockpos4 = p_52275_.east();
+        boolean flag = p_52274_.getBlockState(blockpos3).is(p_52273_) || p_52274_.getBlockState(blockpos4).is(p_52273_);
+        boolean flag1 = p_52274_.getBlockState(blockpos1).is(p_52273_) || p_52274_.getBlockState(blockpos2).is(p_52273_);
+        if (flag && flag1) {
             f /= 2.0F;
         } else {
-            boolean flag = p_52274_.getBlockState(blockpos4.north()).is(p_52273_)
-                || p_52274_.getBlockState(blockpos5.north()).is(p_52273_)
-                || p_52274_.getBlockState(blockpos5.south()).is(p_52273_)
-                || p_52274_.getBlockState(blockpos4.south()).is(p_52273_);
-            if (flag) {
+            boolean flag2 = p_52274_.getBlockState(blockpos3.north()).is(p_52273_)
+                || p_52274_.getBlockState(blockpos4.north()).is(p_52273_)
+                || p_52274_.getBlockState(blockpos4.south()).is(p_52273_)
+                || p_52274_.getBlockState(blockpos3.south()).is(p_52273_);
+            if (flag2) {
                 f /= 2.0F;
             }
         }
@@ -173,7 +_,7 @@
 
     @Override
     protected void entityInside(BlockState p_52277_, Level p_52278_, BlockPos p_52279_, Entity p_52280_) {
-        if (p_52280_ instanceof Ravager && p_52278_.getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING)) {
+        if (p_52280_ instanceof Ravager && net.neoforged.neoforge.event.EventHooks.getMobGriefingEvent(p_52278_, p_52280_)) {
             p_52278_.destroyBlock(p_52279_, true, p_52280_);
         }
 
