--- a/net/minecraft/world/level/block/DispenserBlock.java
+++ b/net/minecraft/world/level/block/DispenserBlock.java
@@ -11,6 +_,7 @@
 import net.minecraft.core.dispenser.BlockSource;
 import net.minecraft.core.dispenser.DefaultDispenseItemBehavior;
 import net.minecraft.core.dispenser.DispenseItemBehavior;
+import net.minecraft.core.dispenser.ProjectileDispenseBehavior;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.stats.Stats;
 import net.minecraft.util.RandomSource;
@@ -35,6 +_,7 @@
 import net.minecraft.world.level.block.state.properties.DirectionProperty;
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.Vec3;
 import org.slf4j.Logger;
 
 public class DispenserBlock extends BaseEntityBlock {
@@ -42,7 +_,7 @@
     public static final MapCodec<DispenserBlock> CODEC = simpleCodec(DispenserBlock::new);
     public static final DirectionProperty FACING = DirectionalBlock.FACING;
     public static final BooleanProperty TRIGGERED = BlockStateProperties.TRIGGERED;
-    private static final Map<Item, DispenseItemBehavior> DISPENSER_REGISTRY = Util.make(
+    public static final Map<Item, DispenseItemBehavior> DISPENSER_REGISTRY = Util.make(
         new Object2ObjectOpenHashMap<>(), p_52723_ -> p_52723_.defaultReturnValue(new DefaultDispenseItemBehavior())
     );
     private static final int TRIGGER_DURATION = 4;
@@ -56,6 +_,10 @@
         DISPENSER_REGISTRY.put(p_52673_.asItem(), p_52674_);
     }
 
+    public static void registerProjectileBehavior(ItemLike p_338817_) {
+        DISPENSER_REGISTRY.put(p_338817_.asItem(), new ProjectileDispenseBehavior(p_338817_.asItem()));
+    }
+
     public DispenserBlock(BlockBehaviour.Properties p_52664_) {
         super(p_52664_);
         this.registerDefaultState(this.stateDefinition.any().setValue(FACING, Direction.NORTH).setValue(TRIGGERED, Boolean.valueOf(false)));
@@ -138,8 +_,17 @@
     }
 
     public static Position getDispensePosition(BlockSource p_302440_) {
-        Direction direction = p_302440_.state().getValue(FACING);
-        return p_302440_.center().add(0.7 * (double)direction.getStepX(), 0.7 * (double)direction.getStepY(), 0.7 * (double)direction.getStepZ());
+        return getDispensePosition(p_302440_, 0.7, Vec3.ZERO);
+    }
+
+    public static Position getDispensePosition(BlockSource p_338419_, double p_338633_, Vec3 p_338798_) {
+        Direction direction = p_338419_.state().getValue(FACING);
+        return p_338419_.center()
+            .add(
+                p_338633_ * (double)direction.getStepX() + p_338798_.x(),
+                p_338633_ * (double)direction.getStepY() + p_338798_.y(),
+                p_338633_ * (double)direction.getStepZ() + p_338798_.z()
+            );
     }
 
     @Override
