--- a/net/minecraft/world/level/block/DoorBlock.java
+++ b/net/minecraft/world/level/block/DoorBlock.java
@@ -54,10 +_,6 @@
     protected static final VoxelShape EAST_AABB = Block.box(0.0, 0.0, 0.0, 3.0, 16.0, 16.0);
     private final BlockSetType type;
 
-    public static boolean shouldTrigger(Level p_341025_, BlockPos p_342123_) {
-        return p_341025_.hasNeighborSignal(p_342123_) || p_341025_.hasNeighborSignal(p_342123_.above());
-    }
-
     @Override
     public MapCodec<? extends DoorBlock> codec() {
         return CODEC;
@@ -87,7 +_,7 @@
         boolean flag = !p_52807_.getValue(OPEN);
         boolean flag1 = p_52807_.getValue(HINGE) == DoorHingeSide.RIGHT;
 
-        return switch(direction) {
+        return switch (direction) {
             case SOUTH -> flag ? SOUTH_AABB : (flag1 ? EAST_AABB : WEST_AABB);
             case WEST -> flag ? WEST_AABB : (flag1 ? SOUTH_AABB : NORTH_AABB);
             case NORTH -> flag ? NORTH_AABB : (flag1 ? WEST_AABB : EAST_AABB);
@@ -133,9 +_,10 @@
 
     @Override
     protected boolean isPathfindable(BlockState p_52764_, PathComputationType p_52767_) {
-        return switch(p_52767_) {
+        return switch (p_52767_) {
             case LAND, AIR -> p_52764_.getValue(OPEN);
             case WATER -> false;
+            default -> throw new MatchException(null, null);
         };
     }
 
@@ -145,7 +_,7 @@
         BlockPos blockpos = p_52739_.getClickedPos();
         Level level = p_52739_.getLevel();
         if (blockpos.getY() < level.getMaxBuildHeight() - 1 && level.getBlockState(blockpos.above()).canBeReplaced(p_52739_)) {
-            boolean flag = shouldTrigger(level, blockpos);
+            boolean flag = level.hasNeighborSignal(blockpos) || level.hasNeighborSignal(blockpos.above());
             return this.defaultBlockState()
                 .setValue(FACING, p_52739_.getHorizontalDirection())
                 .setValue(HINGE, this.getHinge(p_52739_))
@@ -283,8 +_,7 @@
     }
 
     public static boolean isWoodenDoor(BlockState p_52818_) {
-        Block block = p_52818_.getBlock();
-        if (block instanceof DoorBlock doorblock && doorblock.type().canOpenByHand()) {
+        if (p_52818_.getBlock() instanceof DoorBlock doorblock && doorblock.type().canOpenByHand()) {
             return true;
         }
 
