--- a/net/minecraft/world/level/block/EndGatewayBlock.java
+++ b/net/minecraft/world/level/block/EndGatewayBlock.java
@@ -49,7 +_,7 @@
         if (blockentity instanceof TheEndGatewayBlockEntity) {
             int i = ((TheEndGatewayBlockEntity)blockentity).getParticleAmount();
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 double d0 = (double)p_221099_.getX() + p_221100_.nextDouble();
                 double d1 = (double)p_221099_.getY() + p_221100_.nextDouble();
                 double d2 = (double)p_221099_.getZ() + p_221100_.nextDouble();
