--- a/net/minecraft/world/level/block/FaceAttachedHorizontalDirectionalBlock.java
+++ b/net/minecraft/world/level/block/FaceAttachedHorizontalDirectionalBlock.java
@@ -36,7 +_,7 @@
     @Nullable
     @Override
     public BlockState getStateForPlacement(BlockPlaceContext p_53184_) {
-        for(Direction direction : p_53184_.getNearestLookingDirections()) {
+        for (Direction direction : p_53184_.getNearestLookingDirections()) {
             BlockState blockstate;
             if (direction.getAxis() == Direction.Axis.Y) {
                 blockstate = this.defaultBlockState()
@@ -62,7 +_,7 @@
     }
 
     protected static Direction getConnectedDirection(BlockState p_53201_) {
-        switch((AttachFace)p_53201_.getValue(FACE)) {
+        switch ((AttachFace)p_53201_.getValue(FACE)) {
             case CEILING:
                 return Direction.DOWN;
             case FLOOR:
