--- a/net/minecraft/world/level/block/FletchingTableBlock.java
+++ b/net/minecraft/world/level/block/FletchingTableBlock.java
@@ -1,23 +_,15 @@
 package net.minecraft.world.level.block;
 
 import com.mojang.serialization.MapCodec;
-import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
-import net.minecraft.stats.Stats;
-import net.minecraft.world.Containers;
 import net.minecraft.world.InteractionResult;
 import net.minecraft.world.entity.player.Player;
-import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.level.Level;
-import net.minecraft.world.level.block.entity.BlockEntity;
-import net.minecraft.world.level.block.entity.BlockEntityTicker;
-import net.minecraft.world.level.block.entity.BlockEntityType;
-import net.minecraft.world.level.block.entity.FletchingBlockEntity;
 import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.BlockHitResult;
 
-public class FletchingTableBlock extends BaseEntityBlock {
+public class FletchingTableBlock extends CraftingTableBlock {
     public static final MapCodec<FletchingTableBlock> CODEC = simpleCodec(FletchingTableBlock::new);
 
     @Override
@@ -31,48 +_,6 @@
 
     @Override
     protected InteractionResult useWithoutItem(BlockState p_53501_, Level p_53502_, BlockPos p_53503_, Player p_53504_, BlockHitResult p_53506_) {
-        if (p_53502_.isClientSide) {
-            return InteractionResult.SUCCESS;
-        } else {
-            BlockEntity blockentity = p_53502_.getBlockEntity(p_53503_);
-            if (blockentity instanceof FletchingBlockEntity fletchingblockentity) {
-                p_53504_.openMenu(fletchingblockentity);
-                p_53504_.awardStat(Stats.INTERACT_WITH_FLETCHING);
-            }
-
-            return InteractionResult.CONSUME;
-        }
-    }
-
-    @Override
-    public BlockEntity newBlockEntity(BlockPos p_342184_, BlockState p_341053_) {
-        return new FletchingBlockEntity(p_342184_, p_341053_);
-    }
-
-    @Nullable
-    @Override
-    public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level p_342465_, BlockState p_341371_, BlockEntityType<T> p_342062_) {
-        return p_342465_.isClientSide ? null : createTickerHelper(p_342062_, BlockEntityType.FLETCHING, FletchingBlockEntity::serverTick);
-    }
-
-    @Override
-    protected void onRemove(BlockState p_341080_, Level p_341728_, BlockPos p_341553_, BlockState p_340975_, boolean p_341585_) {
-        Containers.dropContentsOnDestroy(p_341080_, p_340975_, p_341728_, p_341553_);
-        super.onRemove(p_341080_, p_341728_, p_341553_, p_340975_, p_341585_);
-    }
-
-    @Override
-    protected boolean hasAnalogOutputSignal(BlockState p_341189_) {
-        return true;
-    }
-
-    @Override
-    protected int getAnalogOutputSignal(BlockState p_340513_, Level p_341995_, BlockPos p_340311_) {
-        return AbstractContainerMenu.getRedstoneSignalFromBlockEntity(p_341995_.getBlockEntity(p_340311_));
-    }
-
-    @Override
-    protected RenderShape getRenderShape(BlockState p_341548_) {
-        return RenderShape.MODEL;
+        return InteractionResult.PASS;
     }
 }
