--- a/net/minecraft/world/level/block/FrostedIceBlock.java
+++ b/net/minecraft/world/level/block/FrostedIceBlock.java
@@ -45,7 +_,7 @@
             && this.slightlyMelt(p_221233_, p_221234_, p_221235_)) {
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(Direction direction : Direction.values()) {
+            for (Direction direction : Direction.values()) {
                 blockpos$mutableblockpos.setWithOffset(p_221235_, direction);
                 BlockState blockstate = p_221234_.getBlockState(blockpos$mutableblockpos);
                 if (blockstate.is(this) && !this.slightlyMelt(blockstate, p_221234_, blockpos$mutableblockpos)) {
@@ -81,7 +_,7 @@
         int i = 0;
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(Direction direction : Direction.values()) {
+        for (Direction direction : Direction.values()) {
             blockpos$mutableblockpos.setWithOffset(p_53567_, direction);
             if (p_53566_.getBlockState(blockpos$mutableblockpos).is(this)) {
                 if (++i >= p_53568_) {
