--- a/net/minecraft/world/level/block/GrassBlock.java
+++ b/net/minecraft/world/level/block/GrassBlock.java
@@ -48,10 +_,10 @@
             .getHolder(VegetationPlacements.GRASS_BONEMEAL);
 
         label49:
-        for(int i = 0; i < 128; ++i) {
+        for (int i = 0; i < 128; i++) {
             BlockPos blockpos1 = blockpos;
 
-            for(int j = 0; j < i / 16; ++j) {
+            for (int j = 0; j < i / 16; j++) {
                 blockpos1 = blockpos1.offset(p_221271_.nextInt(3) - 1, (p_221271_.nextInt(3) - 1) * p_221271_.nextInt(3) / 2, p_221271_.nextInt(3) - 1);
                 if (!p_221270_.getBlockState(blockpos1.below()).is(this) || p_221270_.getBlockState(blockpos1).isCollisionShapeFullBlock(p_221270_, blockpos1)) {
                     continue label49;
