--- a/net/minecraft/world/level/block/HoneyBlock.java
+++ b/net/minecraft/world/level/block/HoneyBlock.java
@@ -126,7 +_,7 @@
         if (p_53989_.level().isClientSide) {
             BlockState blockstate = Blocks.HONEY_BLOCK.defaultBlockState();
 
-            for(int i = 0; i < p_53990_; ++i) {
+            for (int i = 0; i < p_53990_; i++) {
                 p_53989_.level()
                     .addParticle(new BlockParticleOption(ParticleTypes.BLOCK, blockstate), p_53989_.getX(), p_53989_.getY(), p_53989_.getZ(), 0.0, 0.0, 0.0);
             }
