--- a/net/minecraft/world/level/block/HopperBlock.java
+++ b/net/minecraft/world/level/block/HopperBlock.java
@@ -62,7 +_,7 @@
 
     @Override
     protected VoxelShape getShape(BlockState p_54105_, BlockGetter p_54106_, BlockPos p_54107_, CollisionContext p_54108_) {
-        switch((Direction)p_54105_.getValue(FACING)) {
+        switch ((Direction)p_54105_.getValue(FACING)) {
             case DOWN:
                 return DOWN_SHAPE;
             case NORTH:
@@ -80,7 +_,7 @@
 
     @Override
     protected VoxelShape getInteractionShape(BlockState p_54099_, BlockGetter p_54100_, BlockPos p_54101_) {
-        switch((Direction)p_54099_.getValue(FACING)) {
+        switch ((Direction)p_54099_.getValue(FACING)) {
             case DOWN:
                 return DOWN_INTERACTION_SHAPE;
             case NORTH:
