--- a/net/minecraft/world/level/block/LecternBlock.java
+++ b/net/minecraft/world/level/block/LecternBlock.java
@@ -124,7 +_,7 @@
 
     @Override
     protected VoxelShape getShape(BlockState p_54561_, BlockGetter p_54562_, BlockPos p_54563_, CollisionContext p_54564_) {
-        switch((Direction)p_54561_.getValue(FACING)) {
+        switch ((Direction)p_54561_.getValue(FACING)) {
             case NORTH:
                 return SHAPE_NORTH;
             case SOUTH:
@@ -171,8 +_,7 @@
     }
 
     private static void placeBook(@Nullable Entity p_270891_, Level p_270065_, BlockPos p_270155_, BlockState p_270753_, ItemStack p_270173_) {
-        BlockEntity blockentity = p_270065_.getBlockEntity(p_270155_);
-        if (blockentity instanceof LecternBlockEntity lecternblockentity) {
+        if (p_270065_.getBlockEntity(p_270155_) instanceof LecternBlockEntity lecternblockentity) {
             lecternblockentity.setBook(p_270173_.split(1));
             resetBookState(p_270891_, p_270065_, p_270155_, p_270753_, true);
             p_270065_.playSound(null, p_270155_, SoundEvents.BOOK_PUT, SoundSource.BLOCKS, 1.0F, 1.0F);
@@ -222,8 +_,7 @@
     }
 
     private void popBook(BlockState p_54588_, Level p_54589_, BlockPos p_54590_) {
-        BlockEntity blockentity = p_54589_.getBlockEntity(p_54590_);
-        if (blockentity instanceof LecternBlockEntity lecternblockentity) {
+        if (p_54589_.getBlockEntity(p_54590_) instanceof LecternBlockEntity lecternblockentity) {
             Direction direction = p_54588_.getValue(FACING);
             ItemStack itemstack = lecternblockentity.getBook().copy();
             float f = 0.25F * (float)direction.getStepX();
