--- a/net/minecraft/world/level/block/MultifaceSpreader.java
+++ b/net/minecraft/world/level/block/MultifaceSpreader.java
@@ -83,7 +_,7 @@
         if (p_221617_.getAxis() == p_221616_.getAxis()) {
             return Optional.empty();
         } else if (this.config.isOtherBlockValidAsSource(p_221613_) || this.config.hasFace(p_221613_, p_221616_) && !this.config.hasFace(p_221613_, p_221617_)) {
-            for(MultifaceSpreader.SpreadType multifacespreader$spreadtype : this.config.getSpreadTypes()) {
+            for (MultifaceSpreader.SpreadType multifacespreader$spreadtype : this.config.getSpreadTypes()) {
                 MultifaceSpreader.SpreadPos multifacespreader$spreadpos = multifacespreader$spreadtype.getSpreadPos(p_221615_, p_221617_, p_221616_);
                 if (p_221618_.test(p_221614_, p_221615_, multifacespreader$spreadpos)) {
                     return Optional.of(multifacespreader$spreadpos);
