--- a/net/minecraft/world/level/block/MultifaceBlock.java
+++ b/net/minecraft/world/level/block/MultifaceBlock.java
@@ -71,7 +_,7 @@
         } else {
             Set<Direction> set = EnumSet.noneOf(Direction.class);
 
-            for(Direction direction : Direction.values()) {
+            for (Direction direction : Direction.values()) {
                 if (hasFace(p_221585_, direction)) {
                     set.add(direction);
                 }
@@ -84,7 +_,7 @@
     public static Set<Direction> unpack(byte p_221570_) {
         Set<Direction> set = EnumSet.noneOf(Direction.class);
 
-        for(Direction direction : Direction.values()) {
+        for (Direction direction : Direction.values()) {
             if ((p_221570_ & (byte)(1 << direction.ordinal())) > 0) {
                 set.add(direction);
             }
@@ -96,7 +_,7 @@
     public static byte pack(Collection<Direction> p_221577_) {
         byte b0 = 0;
 
-        for(Direction direction : p_221577_) {
+        for (Direction direction : p_221577_) {
             b0 = (byte)(b0 | 1 << direction.ordinal());
         }
 
@@ -109,7 +_,7 @@
 
     @Override
     protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> p_153917_) {
-        for(Direction direction : DIRECTIONS) {
+        for (Direction direction : DIRECTIONS) {
             if (this.isFaceSupported(direction)) {
                 p_153917_.add(getFaceProperty(direction));
             }
@@ -138,7 +_,7 @@
     protected boolean canSurvive(BlockState p_153888_, LevelReader p_153889_, BlockPos p_153890_) {
         boolean flag = false;
 
-        for(Direction direction : DIRECTIONS) {
+        for (Direction direction : DIRECTIONS) {
             if (hasFace(p_153888_, direction)) {
                 BlockPos blockpos = p_153890_.relative(direction);
                 if (!canAttachTo(p_153889_, direction, blockpos, p_153889_.getBlockState(blockpos))) {
@@ -214,7 +_,7 @@
     private BlockState mapDirections(BlockState p_153911_, Function<Direction, Direction> p_153912_) {
         BlockState blockstate = p_153911_;
 
-        for(Direction direction : DIRECTIONS) {
+        for (Direction direction : DIRECTIONS) {
             if (this.isFaceSupported(direction)) {
                 blockstate = blockstate.setValue(getFaceProperty(p_153912_.apply(direction)), p_153911_.getValue(getFaceProperty(direction)));
             }
@@ -249,7 +_,7 @@
     private static BlockState getDefaultMultifaceState(StateDefinition<Block, BlockState> p_153919_) {
         BlockState blockstate = p_153919_.any();
 
-        for(BooleanProperty booleanproperty : PROPERTY_BY_DIRECTION.values()) {
+        for (BooleanProperty booleanproperty : PROPERTY_BY_DIRECTION.values()) {
             if (blockstate.hasProperty(booleanproperty)) {
                 blockstate = blockstate.setValue(booleanproperty, Boolean.valueOf(false));
             }
@@ -261,7 +_,7 @@
     private static VoxelShape calculateMultifaceShape(BlockState p_153959_) {
         VoxelShape voxelshape = Shapes.empty();
 
-        for(Direction direction : DIRECTIONS) {
+        for (Direction direction : DIRECTIONS) {
             if (hasFace(p_153959_, direction)) {
                 voxelshape = Shapes.or(voxelshape, SHAPE_BY_DIRECTION.get(direction));
             }
