--- a/net/minecraft/world/level/block/NetherrackBlock.java
+++ b/net/minecraft/world/level/block/NetherrackBlock.java
@@ -27,7 +_,7 @@
         if (!p_256620_.getBlockState(p_55003_.above()).propagatesSkylightDown(p_256620_, p_55003_)) {
             return false;
         } else {
-            for(BlockPos blockpos : BlockPos.betweenClosed(p_55003_.offset(-1, -1, -1), p_55003_.offset(1, 1, 1))) {
+            for (BlockPos blockpos : BlockPos.betweenClosed(p_55003_.offset(-1, -1, -1), p_55003_.offset(1, 1, 1))) {
                 if (p_256620_.getBlockState(blockpos).is(BlockTags.NYLIUM)) {
                     return true;
                 }
@@ -47,7 +_,7 @@
         boolean flag = false;
         boolean flag1 = false;
 
-        for(BlockPos blockpos : BlockPos.betweenClosed(p_221813_.offset(-1, -1, -1), p_221813_.offset(1, 1, 1))) {
+        for (BlockPos blockpos : BlockPos.betweenClosed(p_221813_.offset(-1, -1, -1), p_221813_.offset(1, 1, 1))) {
             BlockState blockstate = p_221811_.getBlockState(blockpos);
             if (blockstate.is(Blocks.WARPED_NYLIUM)) {
                 flag1 = true;
