--- a/net/minecraft/world/level/block/NetherPortalBlock.java
+++ b/net/minecraft/world/level/block/NetherPortalBlock.java
@@ -45,7 +_,7 @@
 
     @Override
     protected VoxelShape getShape(BlockState p_54942_, BlockGetter p_54943_, BlockPos p_54944_, CollisionContext p_54945_) {
-        switch((Direction.Axis)p_54942_.getValue(AXIS)) {
+        switch ((Direction.Axis)p_54942_.getValue(AXIS)) {
             case Z:
                 return Z_AXIS_AABB;
             case X:
@@ -59,7 +_,7 @@
         if (p_221800_.dimensionType().natural()
             && p_221800_.getGameRules().getBoolean(GameRules.RULE_DOMOBSPAWNING)
             && p_221802_.nextInt(2000) < p_221800_.getDifficulty().getId()) {
-            while(p_221800_.getBlockState(p_221801_).is(this)) {
+            while (p_221800_.getBlockState(p_221801_).is(this)) {
                 p_221801_ = p_221801_.below();
             }
 
@@ -104,7 +_,7 @@
             );
         }
 
-        for(int i = 0; i < 4; ++i) {
+        for (int i = 0; i < 4; i++) {
             double d0 = (double)p_221796_.getX() + p_221797_.nextDouble();
             double d1 = (double)p_221796_.getY() + p_221797_.nextDouble();
             double d2 = (double)p_221796_.getZ() + p_221797_.nextDouble();
@@ -131,10 +_,10 @@
 
     @Override
     protected BlockState rotate(BlockState p_54925_, Rotation p_54926_) {
-        switch(p_54926_) {
+        switch (p_54926_) {
             case COUNTERCLOCKWISE_90:
             case CLOCKWISE_90:
-                switch((Direction.Axis)p_54925_.getValue(AXIS)) {
+                switch ((Direction.Axis)p_54925_.getValue(AXIS)) {
                     case Z:
                         return p_54925_.setValue(AXIS, Direction.Axis.X);
                     case X:
