--- a/net/minecraft/world/level/block/PinkPetalsBlock.java
+++ b/net/minecraft/world/level/block/PinkPetalsBlock.java
@@ -38,7 +_,7 @@
             };
             VoxelShape voxelshape = Shapes.empty();
     
-            for(int i = 0; i < p_294775_; ++i) {
+            for (int i = 0; i < p_294775_; i++) {
                 int j = Math.floorMod(i - p_296142_.get2DDataValue(), 4);
                 voxelshape = Shapes.or(voxelshape, avoxelshape[j]);
             }
