--- a/net/minecraft/world/level/block/PressurePlateBlock.java
+++ b/net/minecraft/world/level/block/PressurePlateBlock.java
@@ -43,9 +_,10 @@
 
     @Override
     protected int getSignalStrength(Level p_55264_, BlockPos p_55265_) {
-        Class<? extends Entity> oclass = switch(this.type.pressurePlateSensitivity()) {
+        Class<? extends Entity> oclass = switch (this.type.pressurePlateSensitivity()) {
             case EVERYTHING -> Entity.class;
             case MOBS -> LivingEntity.class;
+            default -> throw new MatchException(null, null);
         };
         return getEntityCount(p_55264_, TOUCH_AABB.move(p_55265_), oclass) > 0 ? 15 : 0;
     }
