--- a/net/minecraft/world/level/block/RailState.java
+++ b/net/minecraft/world/level/block/RailState.java
@@ -16,14 +_,16 @@
     private BlockState state;
     private final boolean isStraight;
     private final List<BlockPos> connections = Lists.newArrayList();
+    private final boolean canMakeSlopes;
 
     public RailState(Level p_55421_, BlockPos p_55422_, BlockState p_55423_) {
         this.level = p_55421_;
         this.pos = p_55422_;
         this.state = p_55423_;
         this.block = (BaseRailBlock)p_55423_.getBlock();
-        RailShape railshape = p_55423_.getValue(this.block.getShapeProperty());
-        this.isStraight = this.block.isStraight();
+        RailShape railshape = this.block.getRailDirection(state, p_55421_, p_55422_, null);
+        this.isStraight = !this.block.isFlexibleRail(state, p_55421_, p_55422_);
+        this.canMakeSlopes = this.block.canMakeSlopes(state, p_55421_, p_55422_);
         this.updateConnections(railshape);
     }
 
@@ -33,7 +_,7 @@
 
     private void updateConnections(RailShape p_55428_) {
         this.connections.clear();
-        switch(p_55428_) {
+        switch (p_55428_) {
             case NORTH_SOUTH:
                 this.connections.add(this.pos.north());
                 this.connections.add(this.pos.south());
@@ -77,7 +_,7 @@
     }
 
     private void removeSoftConnections() {
-        for(int i = 0; i < this.connections.size(); ++i) {
+        for (int i = 0; i < this.connections.size(); i++) {
             RailState railstate = this.getRail(this.connections.get(i));
             if (railstate != null && railstate.connectsTo(this)) {
                 this.connections.set(i, railstate.pos);
@@ -116,7 +_,7 @@
     }
 
     private boolean hasConnection(BlockPos p_55444_) {
-        for(int i = 0; i < this.connections.size(); ++i) {
+        for (int i = 0; i < this.connections.size(); i++) {
             BlockPos blockpos = this.connections.get(i);
             if (blockpos.getX() == p_55444_.getX() && blockpos.getZ() == p_55444_.getZ()) {
                 return true;
@@ -129,9 +_,9 @@
     protected int countPotentialConnections() {
         int i = 0;
 
-        for(Direction direction : Direction.Plane.HORIZONTAL) {
+        for (Direction direction : Direction.Plane.HORIZONTAL) {
             if (this.hasRail(this.pos.relative(direction))) {
-                ++i;
+                i++;
             }
         }
 
@@ -179,7 +_,7 @@
             }
         }
 
-        if (railshape == RailShape.NORTH_SOUTH) {
+        if (railshape == RailShape.NORTH_SOUTH && canMakeSlopes) {
             if (BaseRailBlock.isRail(this.level, blockpos.above())) {
                 railshape = RailShape.ASCENDING_NORTH;
             }
@@ -189,7 +_,7 @@
             }
         }
 
-        if (railshape == RailShape.EAST_WEST) {
+        if (railshape == RailShape.EAST_WEST && canMakeSlopes) {
             if (BaseRailBlock.isRail(this.level, blockpos3.above())) {
                 railshape = RailShape.ASCENDING_EAST;
             }
@@ -203,6 +_,10 @@
             railshape = RailShape.NORTH_SOUTH;
         }
 
+        if (!this.block.isValidRailShape(railshape)) { // Forge: allow rail block to decide if the new shape is valid
+            this.connections.remove(p_55442_.pos);
+            return;
+        }
         this.state = this.state.setValue(this.block.getShapeProperty(), railshape);
         this.level.setBlock(this.pos, this.state, 3);
     }
@@ -305,7 +_,7 @@
             }
         }
 
-        if (railshape == RailShape.NORTH_SOUTH) {
+        if (railshape == RailShape.NORTH_SOUTH && canMakeSlopes) {
             if (BaseRailBlock.isRail(this.level, blockpos.above())) {
                 railshape = RailShape.ASCENDING_NORTH;
             }
@@ -315,7 +_,7 @@
             }
         }
 
-        if (railshape == RailShape.EAST_WEST) {
+        if (railshape == RailShape.EAST_WEST && canMakeSlopes) {
             if (BaseRailBlock.isRail(this.level, blockpos3.above())) {
                 railshape = RailShape.ASCENDING_EAST;
             }
@@ -325,7 +_,7 @@
             }
         }
 
-        if (railshape == null) {
+        if (railshape == null || !this.block.isValidRailShape(railshape)) { // Forge: allow rail block to decide if the new shape is valid
             railshape = p_55434_;
         }
 
@@ -334,7 +_,7 @@
         if (p_55433_ || this.level.getBlockState(this.pos) != this.state) {
             this.level.setBlock(this.pos, this.state, 3);
 
-            for(int i = 0; i < this.connections.size(); ++i) {
+            for (int i = 0; i < this.connections.size(); i++) {
                 RailState railstate = this.getRail(this.connections.get(i));
                 if (railstate != null) {
                     railstate.removeSoftConnections();
