--- a/net/minecraft/world/level/block/RailBlock.java
+++ b/net/minecraft/world/level/block/RailBlock.java
@@ -41,9 +_,9 @@
     protected BlockState rotate(BlockState p_55405_, Rotation p_55406_) {
         RailShape railshape = p_55405_.getValue(SHAPE);
 
-        return p_55405_.setValue(SHAPE, switch(p_55406_) {
+        return p_55405_.setValue(SHAPE, switch (p_55406_) {
             case CLOCKWISE_180 -> {
-                switch(railshape) {
+                switch (railshape) {
                     case NORTH_SOUTH:
                         yield RailShape.NORTH_SOUTH;
                     case EAST_WEST:
@@ -65,11 +_,11 @@
                     case NORTH_EAST:
                         yield RailShape.SOUTH_WEST;
                     default:
-                        throw new IncompatibleClassChangeError();
+                        throw new MatchException(null, null);
                 }
             }
             case COUNTERCLOCKWISE_90 -> {
-                switch(railshape) {
+                switch (railshape) {
                     case NORTH_SOUTH:
                         yield RailShape.EAST_WEST;
                     case EAST_WEST:
@@ -91,11 +_,11 @@
                     case NORTH_EAST:
                         yield RailShape.NORTH_WEST;
                     default:
-                        throw new IncompatibleClassChangeError();
+                        throw new MatchException(null, null);
                 }
             }
             case CLOCKWISE_90 -> {
-                switch(railshape) {
+                switch (railshape) {
                     case NORTH_SOUTH:
                         yield RailShape.EAST_WEST;
                     case EAST_WEST:
@@ -117,7 +_,7 @@
                     case NORTH_EAST:
                         yield RailShape.SOUTH_EAST;
                     default:
-                        throw new IncompatibleClassChangeError();
+                        throw new MatchException(null, null);
                 }
             }
             default -> railshape;
@@ -127,9 +_,9 @@
     @Override
     protected BlockState mirror(BlockState p_55402_, Mirror p_55403_) {
         RailShape railshape = p_55402_.getValue(SHAPE);
-        switch(p_55403_) {
+        switch (p_55403_) {
             case LEFT_RIGHT:
-                switch(railshape) {
+                switch (railshape) {
                     case ASCENDING_NORTH:
                         return p_55402_.setValue(SHAPE, RailShape.ASCENDING_SOUTH);
                     case ASCENDING_SOUTH:
@@ -146,7 +_,7 @@
                         return super.mirror(p_55402_, p_55403_);
                 }
             case FRONT_BACK:
-                switch(railshape) {
+                switch (railshape) {
                     case ASCENDING_EAST:
                         return p_55402_.setValue(SHAPE, RailShape.ASCENDING_WEST);
                     case ASCENDING_WEST:
@@ -163,6 +_,9 @@
                         return p_55402_.setValue(SHAPE, RailShape.NORTH_EAST);
                     case NORTH_EAST:
                         return p_55402_.setValue(SHAPE, RailShape.NORTH_WEST);
+                    case NORTH_SOUTH: //Forge fix: MC-196102
+                    case EAST_WEST:
+                        return p_55402_; // TODO 1.20.3 PORTING: reevaluate
                 }
         }
 
