--- a/net/minecraft/world/level/block/RedstoneTorchBlock.java
+++ b/net/minecraft/world/level/block/RedstoneTorchBlock.java
@@ -39,7 +_,7 @@
 
     @Override
     protected void onPlace(BlockState p_55724_, Level p_55725_, BlockPos p_55726_, BlockState p_55727_, boolean p_55728_) {
-        for(Direction direction : Direction.values()) {
+        for (Direction direction : Direction.values()) {
             p_55725_.updateNeighborsAt(p_55726_.relative(direction), this);
         }
     }
@@ -47,7 +_,7 @@
     @Override
     protected void onRemove(BlockState p_55706_, Level p_55707_, BlockPos p_55708_, BlockState p_55709_, boolean p_55710_) {
         if (!p_55710_) {
-            for(Direction direction : Direction.values()) {
+            for (Direction direction : Direction.values()) {
                 p_55707_.updateNeighborsAt(p_55708_.relative(direction), this);
             }
         }
@@ -67,7 +_,7 @@
         boolean flag = this.hasNeighborSignal(p_221950_, p_221951_, p_221949_);
         List<RedstoneTorchBlock.Toggle> list = RECENT_TOGGLES.get(p_221950_);
 
-        while(list != null && !list.isEmpty() && p_221950_.getGameTime() - list.get(0).when > 60L) {
+        while (list != null && !list.isEmpty() && p_221950_.getGameTime() - list.get(0).when > 60L) {
             list.remove(0);
         }
 
@@ -124,7 +_,7 @@
 
         int i = 0;
 
-        for(RedstoneTorchBlock.Toggle redstonetorchblock$toggle : list) {
+        for (RedstoneTorchBlock.Toggle redstonetorchblock$toggle : list) {
             if (redstonetorchblock$toggle.pos.equals(p_55686_)) {
                 if (++i >= 8) {
                     return true;
