--- a/net/minecraft/world/level/block/RodBlock.java
+++ b/net/minecraft/world/level/block/RodBlock.java
@@ -25,7 +_,7 @@
 
     @Override
     protected VoxelShape getShape(BlockState p_154346_, BlockGetter p_154347_, BlockPos p_154348_, CollisionContext p_154349_) {
-        switch(p_154346_.getValue(FACING).getAxis()) {
+        switch (p_154346_.getValue(FACING).getAxis()) {
             case X:
             default:
                 return X_AXIS_AABB;
