--- a/net/minecraft/world/level/block/ScaffoldingBlock.java
+++ b/net/minecraft/world/level/block/ScaffoldingBlock.java
@@ -156,7 +_,7 @@
             return 0;
         }
 
-        for(Direction direction : Direction.Plane.HORIZONTAL) {
+        for (Direction direction : Direction.Plane.HORIZONTAL) {
             BlockState blockstate1 = p_56025_.getBlockState(blockpos$mutableblockpos.setWithOffset(p_56026_, direction));
             if (blockstate1.is(Blocks.SCAFFOLDING)) {
                 i = Math.min(i, blockstate1.getValue(DISTANCE) + 1);
