--- a/net/minecraft/world/level/block/SculkVeinBlock.java
+++ b/net/minecraft/world/level/block/SculkVeinBlock.java
@@ -53,7 +_,7 @@
         boolean flag = false;
         BlockState blockstate = Blocks.SCULK_VEIN.defaultBlockState();
 
-        for(Direction direction : p_222367_) {
+        for (Direction direction : p_222367_) {
             BlockPos blockpos = p_222365_.relative(direction);
             if (canAttachTo(p_222364_, direction, blockpos, p_222364_.getBlockState(blockpos))) {
                 blockstate = blockstate.setValue(getFaceProperty(direction), Boolean.valueOf(true));
@@ -76,7 +_,7 @@
     @Override
     public void onDischarged(LevelAccessor p_222359_, BlockState p_222360_, BlockPos p_222361_, RandomSource p_222362_) {
         if (p_222360_.is(this)) {
-            for(Direction direction : DIRECTIONS) {
+            for (Direction direction : DIRECTIONS) {
                 BooleanProperty booleanproperty = getFaceProperty(direction);
                 if (p_222360_.getValue(booleanproperty) && p_222359_.getBlockState(p_222361_.relative(direction)).is(Blocks.SCULK)) {
                     p_222360_ = p_222360_.setValue(booleanproperty, Boolean.valueOf(false));
@@ -108,7 +_,7 @@
         BlockState blockstate = p_222377_.getBlockState(p_222378_);
         TagKey<Block> tagkey = p_222376_.replaceableBlocks();
 
-        for(Direction direction : Direction.allShuffled(p_222379_)) {
+        for (Direction direction : Direction.allShuffled(p_222379_)) {
             if (hasFace(blockstate, direction)) {
                 BlockPos blockpos = p_222378_.relative(direction);
                 BlockState blockstate1 = p_222377_.getBlockState(blockpos);
@@ -120,7 +_,7 @@
                     this.veinSpreader.spreadAll(blockstate2, p_222377_, blockpos, p_222376_.isWorldGeneration());
                     Direction direction1 = direction.getOpposite();
 
-                    for(Direction direction2 : DIRECTIONS) {
+                    for (Direction direction2 : DIRECTIONS) {
                         if (direction2 != direction1) {
                             BlockPos blockpos1 = blockpos.relative(direction2);
                             BlockState blockstate3 = p_222377_.getBlockState(blockpos1);
@@ -142,7 +_,7 @@
         if (!p_222356_.is(Blocks.SCULK_VEIN)) {
             return false;
         } else {
-            for(Direction direction : DIRECTIONS) {
+            for (Direction direction : DIRECTIONS) {
                 if (hasFace(p_222356_, direction) && p_222355_.getBlockState(p_222357_.relative(direction)).is(BlockTags.SCULK_REPLACEABLE)) {
                     return true;
                 }
@@ -201,10 +_,10 @@
                 FluidState fluidstate = p_222409_.getFluidState();
                 if (!fluidstate.isEmpty() && !fluidstate.is(Fluids.WATER)) {
                     return false;
-                } else if (p_222409_.is(BlockTags.FIRE)) {
-                    return false;
                 } else {
-                    return p_222409_.canBeReplaced() || super.stateCanBeReplaced(p_222405_, p_222406_, p_222407_, p_222408_, p_222409_);
+                    return p_222409_.is(BlockTags.FIRE)
+                        ? false
+                        : p_222409_.canBeReplaced() || super.stateCanBeReplaced(p_222405_, p_222406_, p_222407_, p_222408_, p_222409_);
                 }
             } else {
                 return false;
