--- a/net/minecraft/world/level/block/SlabBlock.java
+++ b/net/minecraft/world/level/block/SlabBlock.java
@@ -55,7 +_,7 @@
     @Override
     protected VoxelShape getShape(BlockState p_56390_, BlockGetter p_56391_, BlockPos p_56392_, CollisionContext p_56393_) {
         SlabType slabtype = p_56390_.getValue(TYPE);
-        switch(slabtype) {
+        switch (slabtype) {
             case DOUBLE:
                 return Shapes.block();
             case TOP:
@@ -93,11 +_,9 @@
         } else if (p_56374_.replacingClickedOnBlock()) {
             boolean flag = p_56374_.getClickLocation().y - (double)p_56374_.getClickedPos().getY() > 0.5;
             Direction direction = p_56374_.getClickedFace();
-            if (slabtype == SlabType.BOTTOM) {
-                return direction == Direction.UP || flag && direction.getAxis().isHorizontal();
-            } else {
-                return direction == Direction.DOWN || !flag && direction.getAxis().isHorizontal();
-            }
+            return slabtype == SlabType.BOTTOM
+                ? direction == Direction.UP || flag && direction.getAxis().isHorizontal()
+                : direction == Direction.DOWN || !flag && direction.getAxis().isHorizontal();
         } else {
             return true;
         }
@@ -131,7 +_,7 @@
 
     @Override
     protected boolean isPathfindable(BlockState p_56376_, PathComputationType p_56379_) {
-        switch(p_56379_) {
+        switch (p_56379_) {
             case LAND:
                 return false;
             case WATER:
