--- a/net/minecraft/world/level/block/SmallDripleafBlock.java
+++ b/net/minecraft/world/level/block/SmallDripleafBlock.java
@@ -88,6 +_,8 @@
         } else {
             BlockPos blockpos = pos.below();
             BlockState blockstate = level.getBlockState(blockpos);
+            var soilDecision = blockstate.canSustainPlant(level, blockpos, Direction.UP, state);
+            if (!soilDecision.isDefault()) return soilDecision.isTrue();
             return this.mayPlaceOn(blockstate, level, blockpos);
         }
     }
