--- a/net/minecraft/world/level/block/SoundType.java
+++ b/net/minecraft/world/level/block/SoundType.java
@@ -652,39 +_,6 @@
         SoundEvents.NETHER_WOOD_HIT,
         SoundEvents.NETHER_WOOD_FALL
     );
-    public static final SoundType CORRUPTED_PEELGRASS = new SoundType(
-        1.0F,
-        1.0F,
-        SoundEvents.CORRUPTED_PEELGRASS_BLOCK_BREAK,
-        SoundEvents.CORRUPTED_PEELGRASS_BLOCK_STEP,
-        SoundEvents.CORRUPTED_PEELGRASS_BLOCK_PLACE,
-        SoundEvents.CORRUPTED_PEELGRASS_BLOCK_HIT,
-        SoundEvents.CORRUPTED_PEELGRASS_BLOCK_FALL
-    );
-    public static final SoundType PEELGRASS = new SoundType(
-        1.0F,
-        1.0F,
-        SoundEvents.PEELGRASS_BLOCK_BREAK,
-        SoundEvents.PEELGRASS_BLOCK_STEP,
-        SoundEvents.PEELGRASS_BLOCK_PLACE,
-        SoundEvents.PEELGRASS_BLOCK_HIT,
-        SoundEvents.PEELGRASS_BLOCK_FALL
-    );
-    public static final SoundType GRAVTATER = new SoundType(
-        1.0F, 1.0F, SoundEvents.GRAVTATER_BREAK, SoundEvents.GRAVTATER_STEP, SoundEvents.GRAVTATER_PLACE, SoundEvents.GRAVTATER_HIT, SoundEvents.GRAVTATER_FALL
-    );
-    public static final SoundType POTONE = new SoundType(
-        1.0F, 1.0F, SoundEvents.POTONE_BREAK, SoundEvents.POTONE_STEP, SoundEvents.POTONE_PLACE, SoundEvents.POTONE_HIT, SoundEvents.POTONE_FALL
-    );
-    public static final SoundType TERREDEPOMME = new SoundType(
-        1.0F,
-        1.0F,
-        SoundEvents.TERREDEPOMME_BREAK,
-        SoundEvents.TERREDEPOMME_STEP,
-        SoundEvents.TERREDEPOMME_PLACE,
-        SoundEvents.TERREDEPOMME_HIT,
-        SoundEvents.TERREDEPOMME_FALL
-    );
     public static final SoundType CHERRY_WOOD = new SoundType(
         1.0F,
         1.0F,
@@ -799,6 +_,9 @@
         SoundEvents.HEAVY_CORE_HIT,
         SoundEvents.HEAVY_CORE_FALL
     );
+    public static final SoundType COBWEB = new SoundType(
+        1.0F, 1.0F, SoundEvents.COBWEB_BREAK, SoundEvents.COBWEB_STEP, SoundEvents.COBWEB_PLACE, SoundEvents.COBWEB_HIT, SoundEvents.COBWEB_FALL
+    );
     public final float volume;
     public final float pitch;
     private final SoundEvent breakSound;
@@ -807,6 +_,7 @@
     private final SoundEvent hitSound;
     private final SoundEvent fallSound;
 
+    @Deprecated // Forge: Use {@link net.neoforged.neoforge.common.util.DeferredSoundType} instead for suppliers
     public SoundType(float p_56766_, float p_56767_, SoundEvent p_56768_, SoundEvent p_56769_, SoundEvent p_56770_, SoundEvent p_56771_, SoundEvent p_56772_) {
         this.volume = p_56766_;
         this.pitch = p_56767_;
