--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -49,12 +_,13 @@
     }
 
     private boolean removeWaterBreadthFirstSearch(Level p_56808_, BlockPos p_56809_) {
+        BlockState spongeState = p_56808_.getBlockState(p_56809_);
         return BlockPos.breadthFirstTraversal(
                 p_56809_,
                 6,
                 65,
                 (p_277519_, p_277492_) -> {
-                    for(Direction direction : ALL_DIRECTIONS) {
+                    for (Direction direction : ALL_DIRECTIONS) {
                         p_277492_.accept(p_277519_.relative(direction));
                     }
                 },
@@ -64,11 +_,11 @@
                     } else {
                         BlockState blockstate = p_56808_.getBlockState(p_294069_);
                         FluidState fluidstate = p_56808_.getFluidState(p_294069_);
-                        if (!fluidstate.is(FluidTags.WATER)) {
+                        if (!spongeState.canBeHydrated(p_56808_, p_56809_, fluidstate, p_294069_)) {
                             return false;
                         } else {
-                            Block block = blockstate.getBlock();
-                            if (block instanceof BucketPickup bucketpickup && !bucketpickup.pickupBlock(null, p_56808_, p_294069_, blockstate).isEmpty()) {
+                            if (blockstate.getBlock() instanceof BucketPickup bucketpickup
+                                && !bucketpickup.pickupBlock(null, p_56808_, p_294069_, blockstate).isEmpty()) {
                                 return true;
                             }
         
