--- a/net/minecraft/world/level/block/SporeBlossomBlock.java
+++ b/net/minecraft/world/level/block/SporeBlossomBlock.java
@@ -56,7 +_,7 @@
         p_222504_.addParticle(ParticleTypes.FALLING_SPORE_BLOSSOM, d0, d1, d2, 0.0, 0.0, 0.0);
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int l = 0; l < 14; ++l) {
+        for (int l = 0; l < 14; l++) {
             blockpos$mutableblockpos.set(i + Mth.nextInt(p_222506_, -10, 10), j - p_222506_.nextInt(10), k + Mth.nextInt(p_222506_, -10, 10));
             BlockState blockstate = p_222504_.getBlockState(blockpos$mutableblockpos);
             if (!blockstate.isCollisionShapeFullBlock(p_222504_, blockpos$mutableblockpos)) {
