--- a/net/minecraft/world/level/block/StairBlock.java
+++ b/net/minecraft/world/level/block/StairBlock.java
@@ -27,7 +_,7 @@
 
 public class StairBlock extends Block implements SimpleWaterloggedBlock {
     public static final MapCodec<StairBlock> CODEC = RecordCodecBuilder.mapCodec(
-        p_308839_ -> p_308839_.group(BlockState.CODEC.fieldOf("base_state").forGetter(p_304378_ -> p_304378_.baseState), propertiesCodec())
+        p_308839_ -> p_308839_.group(BlockState.CODEC.fieldOf("base_state").forGetter(StairBlock::getModelState), propertiesCodec())
                 .apply(p_308839_, StairBlock::new)
     );
     public static final DirectionProperty FACING = HorizontalDirectionalBlock.FACING;
@@ -47,6 +_,7 @@
     protected static final VoxelShape[] TOP_SHAPES = makeShapes(TOP_AABB, OCTET_NNN, OCTET_PNN, OCTET_NNP, OCTET_PNP);
     protected static final VoxelShape[] BOTTOM_SHAPES = makeShapes(BOTTOM_AABB, OCTET_NPN, OCTET_PPN, OCTET_NPP, OCTET_PPP);
     private static final int[] SHAPE_BY_STATE = new int[]{12, 5, 3, 10, 14, 13, 7, 11, 13, 7, 11, 14, 8, 4, 1, 2, 4, 1, 2, 8};
+    /** Neo: Field accesses are redirected to {@link #getModelBlock()} with a coremod. */
     private final Block base;
     protected final BlockState baseState;
 
@@ -58,7 +_,7 @@
     private static VoxelShape[] makeShapes(VoxelShape p_56934_, VoxelShape p_56935_, VoxelShape p_56936_, VoxelShape p_56937_, VoxelShape p_56938_) {
         return IntStream.range(0, 16)
             .mapToObj(p_56945_ -> makeStairShape(p_56945_, p_56934_, p_56935_, p_56936_, p_56937_, p_56938_))
-            .toArray(p_56949_ -> new VoxelShape[p_56949_]);
+            .toArray(VoxelShape[]::new);
     }
 
     private static VoxelShape makeStairShape(
@@ -84,7 +_,12 @@
         return voxelshape;
     }
 
+    @Deprecated // Forge: Use the other constructor that takes a Supplier
     public StairBlock(BlockState p_56862_, BlockBehaviour.Properties p_56863_) {
+        this(() -> p_56862_, p_56863_);
+    }
+
+    public StairBlock(java.util.function.Supplier<net.minecraft.world.level.block.state.BlockState> p_56862_, BlockBehaviour.Properties p_56863_) {
         super(p_56863_);
         this.registerDefaultState(
             this.stateDefinition
@@ -94,8 +_,9 @@
                 .setValue(SHAPE, StairsShape.STRAIGHT)
                 .setValue(WATERLOGGED, Boolean.valueOf(false))
         );
-        this.base = p_56862_.getBlock();
-        this.baseState = p_56862_;
+        this.base = Blocks.AIR;
+        this.baseState = Blocks.AIR.defaultBlockState();
+        this.stateSupplier = p_56862_;
     }
 
     @Override
@@ -192,10 +_,10 @@
     protected BlockState mirror(BlockState p_56919_, Mirror p_56920_) {
         Direction direction = p_56919_.getValue(FACING);
         StairsShape stairsshape = p_56919_.getValue(SHAPE);
-        switch(p_56920_) {
+        switch (p_56920_) {
             case LEFT_RIGHT:
                 if (direction.getAxis() == Direction.Axis.Z) {
-                    switch(stairsshape) {
+                    switch (stairsshape) {
                         case INNER_LEFT:
                             return p_56919_.rotate(Rotation.CLOCKWISE_180).setValue(SHAPE, StairsShape.INNER_RIGHT);
                         case INNER_RIGHT:
@@ -211,7 +_,7 @@
                 break;
             case FRONT_BACK:
                 if (direction.getAxis() == Direction.Axis.X) {
-                    switch(stairsshape) {
+                    switch (stairsshape) {
                         case INNER_LEFT:
                             return p_56919_.rotate(Rotation.CLOCKWISE_180).setValue(SHAPE, StairsShape.INNER_LEFT);
                         case INNER_RIGHT:
@@ -243,4 +_,18 @@
     protected boolean isPathfindable(BlockState p_56891_, PathComputationType p_56894_) {
         return false;
     }
+
+    // NEO START
+
+    private final java.util.function.Supplier<BlockState> stateSupplier;
+
+    private Block getModelBlock() {
+        return getModelState().getBlock();
+    }
+
+    protected BlockState getModelState() {
+        return stateSupplier.get();
+    }
+
+    // NEO END
 }
