--- a/net/minecraft/world/level/block/TrialSpawnerBlock.java
+++ b/net/minecraft/world/level/block/TrialSpawnerBlock.java
@@ -4,12 +_,11 @@
 import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
-import net.minecraft.core.RegistryAccess;
 import net.minecraft.network.chat.Component;
 import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.TooltipFlag;
-import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.Spawner;
 import net.minecraft.world.level.block.entity.BlockEntity;
@@ -21,11 +_,13 @@
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.StateDefinition;
 import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
 import net.minecraft.world.level.block.state.properties.EnumProperty;
 
 public class TrialSpawnerBlock extends BaseEntityBlock {
     public static final MapCodec<TrialSpawnerBlock> CODEC = simpleCodec(TrialSpawnerBlock::new);
     public static final EnumProperty<TrialSpawnerState> STATE = BlockStateProperties.TRIAL_SPAWNER_STATE;
+    public static final BooleanProperty OMINOUS = BlockStateProperties.OMINOUS;
 
     @Override
     public MapCodec<TrialSpawnerBlock> codec() {
@@ -34,12 +_,12 @@
 
     public TrialSpawnerBlock(BlockBehaviour.Properties p_312795_) {
         super(p_312795_);
-        this.registerDefaultState(this.stateDefinition.any().setValue(STATE, TrialSpawnerState.INACTIVE));
+        this.registerDefaultState(this.stateDefinition.any().setValue(STATE, TrialSpawnerState.INACTIVE).setValue(OMINOUS, Boolean.valueOf(false)));
     }
 
     @Override
     protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> p_312785_) {
-        p_312785_.add(STATE);
+        p_312785_.add(STATE, OMINOUS);
     }
 
     @Override
@@ -60,20 +_,20 @@
             ? createTickerHelper(
                 p_312122_,
                 BlockEntityType.TRIAL_SPAWNER,
-                (p_312628_, p_312290_, p_312639_, p_311782_) -> p_311782_.getTrialSpawner().tickServer(serverlevel, p_312290_)
+                (p_337976_, p_337977_, p_337978_, p_337979_) -> p_337979_.getTrialSpawner()
+                        .tickServer(serverlevel, p_337977_, p_337978_.getOptionalValue(BlockStateProperties.OMINOUS).orElse(false))
             )
             : createTickerHelper(
                 p_312122_,
                 BlockEntityType.TRIAL_SPAWNER,
-                (p_312408_, p_312921_, p_312435_, p_312269_) -> p_312269_.getTrialSpawner().tickClient(p_312408_, p_312921_)
+                (p_337980_, p_337981_, p_337982_, p_337983_) -> p_337983_.getTrialSpawner()
+                        .tickClient(p_337980_, p_337981_, p_337982_.getOptionalValue(BlockStateProperties.OMINOUS).orElse(false))
             );
     }
 
     @Override
-    public void appendHoverText(
-        ItemStack p_312446_, @Nullable BlockGetter p_312129_, List<Component> p_312088_, TooltipFlag p_311895_, @Nullable RegistryAccess p_324047_
-    ) {
-        super.appendHoverText(p_312446_, p_312129_, p_312088_, p_311895_, p_324047_);
+    public void appendHoverText(ItemStack p_312446_, Item.TooltipContext p_339621_, List<Component> p_312088_, TooltipFlag p_311895_) {
+        super.appendHoverText(p_312446_, p_339621_, p_312088_, p_311895_);
         Spawner.appendHoverText(p_312446_, p_312088_, "spawn_data");
     }
 }
