--- a/net/minecraft/world/level/block/VaultBlock.java
+++ b/net/minecraft/world/level/block/VaultBlock.java
@@ -20,6 +_,7 @@
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.StateDefinition;
 import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
 import net.minecraft.world.level.block.state.properties.DirectionProperty;
 import net.minecraft.world.level.block.state.properties.Property;
 import net.minecraft.world.phys.BlockHitResult;
@@ -28,6 +_,7 @@
     public static final MapCodec<VaultBlock> CODEC = simpleCodec(VaultBlock::new);
     public static final Property<VaultState> STATE = BlockStateProperties.VAULT_STATE;
     public static final DirectionProperty FACING = HorizontalDirectionalBlock.FACING;
+    public static final BooleanProperty OMINOUS = BlockStateProperties.OMINOUS;
 
     @Override
     public MapCodec<VaultBlock> codec() {
@@ -36,7 +_,9 @@
 
     public VaultBlock(BlockBehaviour.Properties p_324605_) {
         super(p_324605_);
-        this.registerDefaultState(this.stateDefinition.any().setValue(FACING, Direction.NORTH).setValue(STATE, VaultState.INACTIVE));
+        this.registerDefaultState(
+            this.stateDefinition.any().setValue(FACING, Direction.NORTH).setValue(STATE, VaultState.INACTIVE).setValue(OMINOUS, Boolean.valueOf(false))
+        );
     }
 
     @Override
@@ -46,8 +_,7 @@
         if (p_324161_.isEmpty() || p_323816_.getValue(STATE) != VaultState.ACTIVE) {
             return ItemInteractionResult.PASS_TO_DEFAULT_BLOCK_INTERACTION;
         } else if (p_324403_ instanceof ServerLevel serverlevel) {
-            BlockEntity blockentity = serverlevel.getBlockEntity(p_324623_);
-            if (blockentity instanceof VaultBlockEntity vaultblockentity) {
+            if (serverlevel.getBlockEntity(p_324623_) instanceof VaultBlockEntity vaultblockentity) {
                 VaultBlockEntity.Server.tryInsertKey(
                     serverlevel,
                     p_324623_,
@@ -75,7 +_,7 @@
 
     @Override
     protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> p_323673_) {
-        p_323673_.add(FACING, STATE);
+        p_323673_.add(FACING, STATE, OMINOUS);
     }
 
     @Nullable
