--- a/net/minecraft/world/level/block/WallSkullBlock.java
+++ b/net/minecraft/world/level/block/WallSkullBlock.java
@@ -63,7 +_,7 @@
         BlockPos blockpos = p_58104_.getClickedPos();
         Direction[] adirection = p_58104_.getNearestLookingDirections();
 
-        for(Direction direction : adirection) {
+        for (Direction direction : adirection) {
             if (direction.getAxis().isHorizontal()) {
                 Direction direction1 = direction.getOpposite();
                 blockstate = blockstate.setValue(FACING, direction1);
