--- a/net/minecraft/world/level/block/WallHangingSignBlock.java
+++ b/net/minecraft/world/level/block/WallHangingSignBlock.java
@@ -66,8 +_,7 @@
     protected ItemInteractionResult useItemOn(
         ItemStack p_316420_, BlockState p_316885_, Level p_316114_, BlockPos p_316297_, Player p_316609_, InteractionHand p_316187_, BlockHitResult p_316223_
     ) {
-        BlockEntity blockentity = p_316114_.getBlockEntity(p_316297_);
-        if (blockentity instanceof SignBlockEntity signblockentity
+        if (p_316114_.getBlockEntity(p_316297_) instanceof SignBlockEntity signblockentity
             && this.shouldTryToChainAnotherHangingSign(p_316885_, p_316609_, p_316223_, signblockentity, p_316420_)) {
             return ItemInteractionResult.SKIP_DEFAULT_BLOCK_INTERACTION;
         }
@@ -104,7 +_,7 @@
 
     @Override
     protected VoxelShape getCollisionShape(BlockState p_249963_, BlockGetter p_248542_, BlockPos p_252224_, CollisionContext p_251891_) {
-        switch((Direction)p_249963_.getValue(FACING)) {
+        switch ((Direction)p_249963_.getValue(FACING)) {
             case EAST:
             case WEST:
                 return PLANK_EASTWEST;
@@ -135,7 +_,7 @@
         LevelReader levelreader = p_251399_.getLevel();
         BlockPos blockpos = p_251399_.getClickedPos();
 
-        for(Direction direction : p_251399_.getNearestLookingDirections()) {
+        for (Direction direction : p_251399_.getNearestLookingDirections()) {
             if (direction.getAxis().isHorizontal() && !direction.getAxis().test(p_251399_.getClickedFace())) {
                 Direction direction1 = direction.getOpposite();
                 blockstate = blockstate.setValue(FACING, direction1);
