--- a/net/minecraft/world/level/block/WetSpongeBlock.java
+++ b/net/minecraft/world/level/block/WetSpongeBlock.java
@@ -51,14 +_,14 @@
                     if (direction.getAxis() == Direction.Axis.X) {
                         d2 += p_222685_.nextDouble();
                         if (direction == Direction.EAST) {
-                            ++d0;
+                            d0++;
                         } else {
                             d0 += 0.05;
                         }
                     } else {
                         d0 += p_222685_.nextDouble();
                         if (direction == Direction.SOUTH) {
-                            ++d2;
+                            d2++;
                         } else {
                             d2 += 0.05;
                         }
