--- a/net/minecraft/client/gui/screens/inventory/tooltip/ClientBundleTooltip.java
+++ b/net/minecraft/client/gui/screens/inventory/tooltip/ClientBundleTooltip.java
@@ -8,6 +_,7 @@
 import net.minecraft.world.item.component.BundleContents;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
+import org.apache.commons.lang3.math.Fraction;
 
 @OnlyIn(Dist.CLIENT)
 public class ClientBundleTooltip implements ClientTooltipComponent {
@@ -45,11 +_,11 @@
         int i = this.gridSizeX();
         int j = this.gridSizeY();
         p_282522_.blitSprite(BACKGROUND_SPRITE, p_194043_, p_194044_, this.backgroundWidth(), this.backgroundHeight());
-        boolean flag = this.contents.weight() >= 64;
+        boolean flag = this.contents.weight().compareTo(Fraction.ONE) >= 0;
         int k = 0;
 
-        for(int l = 0; l < j; ++l) {
-            for(int i1 = 0; i1 < i; ++i1) {
+        for (int l = 0; l < j; l++) {
+            for (int i1 = 0; i1 < i; i1++) {
                 int j1 = p_194043_ + i1 * 18 + 1;
                 int k1 = p_194044_ + l * 20 + 1;
                 this.renderSlot(j1, k1, k++, flag, p_282522_, p_194042_);
