--- a/net/minecraft/client/gui/screens/inventory/tooltip/MenuTooltipPositioner.java
+++ b/net/minecraft/client/gui/screens/inventory/tooltip/MenuTooltipPositioner.java
@@ -31,9 +_,9 @@
         int j = this.screenRectangle.bottom() + 3 + getOffset(0, 0, this.screenRectangle.height());
         int k = p_282509_ - 5;
         if (j + i <= k) {
-            vector2i.y += getOffset(vector2i.y, this.screenRectangle.top(), this.screenRectangle.height());
+            vector2i.y = vector2i.y + getOffset(vector2i.y, this.screenRectangle.top(), this.screenRectangle.height());
         } else {
-            vector2i.y -= i + getOffset(vector2i.y, this.screenRectangle.bottom(), this.screenRectangle.height());
+            vector2i.y = vector2i.y - (i + getOffset(vector2i.y, this.screenRectangle.bottom(), this.screenRectangle.height()));
         }
 
         return vector2i;
