--- a/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
+++ b/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
@@ -13,7 +_,6 @@
 import java.util.Optional;
 import java.util.function.Predicate;
 import javax.annotation.Nullable;
-import net.minecraft.Util;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Holder;
 import net.minecraft.core.HolderLookup;
@@ -117,7 +_,7 @@
 
         RegistryOps<Tag> registryops = p_324560_.createSerializationContext(NbtOps.INSTANCE);
 
-        for(MapBanner mapbanner : MapBanner.LIST_CODEC
+        for (MapBanner mapbanner : MapBanner.LIST_CODEC
             .parse(registryops, p_164808_.get("banners"))
             .resultOrPartial(p_323448_ -> LOGGER.warn("Failed to parse map banner: '{}'", p_323448_))
             .orElse(List.of())) {
@@ -135,7 +_,7 @@
 
         ListTag listtag = p_164808_.getList("frames", 10);
 
-        for(int k = 0; k < listtag.size(); ++k) {
+        for (int k = 0; k < listtag.size(); k++) {
             MapFrame mapframe = MapFrame.load(listtag.getCompound(k));
             if (mapframe != null) {
                 mapitemsaveddata.frameMarkers.put(mapframe.getId(), mapframe);
@@ -168,12 +_,10 @@
         p_77956_.putBoolean("unlimitedTracking", this.unlimitedTracking);
         p_77956_.putBoolean("locked", this.locked);
         RegistryOps<Tag> registryops = p_323858_.createSerializationContext(NbtOps.INSTANCE);
-        p_77956_.put(
-            "banners", Util.getOrThrow(MapBanner.LIST_CODEC.encodeStart(registryops, List.copyOf(this.bannerMarkers.values())), IllegalStateException::new)
-        );
+        p_77956_.put("banners", MapBanner.LIST_CODEC.encodeStart(registryops, List.copyOf(this.bannerMarkers.values())).getOrThrow());
         ListTag listtag = new ListTag();
 
-        for(MapFrame mapframe : this.frameMarkers.values()) {
+        for (MapFrame mapframe : this.frameMarkers.values()) {
             listtag.add(mapframe.save());
         }
 
@@ -201,13 +_,7 @@
 
     private static Predicate<ItemStack> mapMatcher(ItemStack p_316807_) {
         MapId mapid = p_316807_.get(DataComponents.MAP_ID);
-        return p_330169_ -> {
-            if (p_330169_ == p_316807_) {
-                return true;
-            } else {
-                return p_330169_.is(p_316807_.getItem()) && Objects.equals(mapid, p_330169_.get(DataComponents.MAP_ID));
-            }
-        };
+        return p_330169_ -> p_330169_ == p_316807_ ? true : p_330169_.is(p_316807_.getItem()) && Objects.equals(mapid, p_330169_.get(DataComponents.MAP_ID));
     }
 
     public void tickCarriedBy(Player p_77919_, ItemStack p_77920_) {
@@ -222,7 +_,7 @@
             this.removeDecoration(p_77919_.getName().getString());
         }
 
-        for(int i = 0; i < this.carriedBy.size(); ++i) {
+        for (int i = 0; i < this.carriedBy.size(); i++) {
             MapItemSavedData.HoldingPlayer mapitemsaveddata$holdingplayer1 = this.carriedBy.get(i);
             String s = mapitemsaveddata$holdingplayer1.player.getName().getString();
             if (!mapitemsaveddata$holdingplayer1.player.isRemoved()
@@ -268,9 +_,9 @@
 
         MapDecorations mapdecorations = p_77920_.getOrDefault(DataComponents.MAP_DECORATIONS, MapDecorations.EMPTY);
         if (!this.decorations.keySet().containsAll(mapdecorations.decorations().keySet())) {
-            mapdecorations.decorations().forEach((p_340308_, p_340309_) -> {
-                if (!this.decorations.containsKey(p_340308_)) {
-                    this.addDecoration(p_340309_.type(), p_77919_.level(), p_340308_, p_340309_.x(), p_340309_.z(), (double)p_340309_.rotation(), null);
+            mapdecorations.decorations().forEach((p_338113_, p_338114_) -> {
+                if (!this.decorations.containsKey(p_338113_)) {
+                    this.addDecoration(p_338114_.type(), p_77919_.level(), p_338113_, p_338114_.x(), p_338114_.z(), (double)p_338114_.rotation(), null);
                 }
             });
         }
@@ -279,7 +_,7 @@
     private void removeDecoration(String p_164800_) {
         MapDecoration mapdecoration = this.decorations.remove(p_164800_);
         if (mapdecoration != null && mapdecoration.type().value().trackCount()) {
-            --this.trackedDecorationCount;
+            this.trackedDecorationCount--;
         }
 
         this.setDecorationsDirty();
@@ -356,11 +_,11 @@
         MapDecoration mapdecoration = this.decorations.put(p_77940_, mapdecoration1);
         if (!mapdecoration1.equals(mapdecoration)) {
             if (mapdecoration != null && mapdecoration.type().value().trackCount()) {
-                --this.trackedDecorationCount;
+                this.trackedDecorationCount--;
             }
 
             if (p_335830_.value().trackCount()) {
-                ++this.trackedDecorationCount;
+                this.trackedDecorationCount++;
             }
 
             this.setDecorationsDirty();
@@ -376,7 +_,7 @@
     private void setColorsDirty(int p_164790_, int p_164791_) {
         this.setDirty();
 
-        for(MapItemSavedData.HoldingPlayer mapitemsaveddata$holdingplayer : this.carriedBy) {
+        for (MapItemSavedData.HoldingPlayer mapitemsaveddata$holdingplayer : this.carriedBy) {
             mapitemsaveddata$holdingplayer.markColorsDirty(p_164790_, p_164791_);
         }
     }
@@ -428,7 +_,7 @@
     public void checkBanners(BlockGetter p_77931_, int p_77932_, int p_77933_) {
         Iterator<MapBanner> iterator = this.bannerMarkers.values().iterator();
 
-        while(iterator.hasNext()) {
+        while (iterator.hasNext()) {
             MapBanner mapbanner = iterator.next();
             if (mapbanner.pos().getX() == p_77932_ && mapbanner.pos().getZ() == p_77933_) {
                 MapBanner mapbanner1 = MapBanner.fromWorld(p_77931_, mapbanner.pos());
@@ -465,7 +_,7 @@
     }
 
     public boolean isExplorationMap() {
-        for(MapDecoration mapdecoration : this.decorations.values()) {
+        for (MapDecoration mapdecoration : this.decorations.values()) {
             if (mapdecoration.type().value().explorationMapElement()) {
                 return true;
             }
@@ -478,11 +_,11 @@
         this.decorations.clear();
         this.trackedDecorationCount = 0;
 
-        for(int i = 0; i < p_164802_.size(); ++i) {
+        for (int i = 0; i < p_164802_.size(); i++) {
             MapDecoration mapdecoration = p_164802_.get(i);
             this.decorations.put("icon-" + i, mapdecoration);
             if (mapdecoration.type().value().trackCount()) {
-                ++this.trackedDecorationCount;
+                this.trackedDecorationCount++;
             }
         }
     }
@@ -517,8 +_,8 @@
             int l = this.maxDirtyY + 1 - this.minDirtyY;
             byte[] abyte = new byte[k * l];
 
-            for(int i1 = 0; i1 < k; ++i1) {
-                for(int j1 = 0; j1 < l; ++j1) {
+            for (int i1 = 0; i1 < k; i1++) {
+                for (int j1 = 0; j1 < l; j1++) {
                     abyte[i1 + j1 * k] = MapItemSavedData.this.colors[i + i1 + (j + j1) * 128];
                 }
             }
@@ -601,8 +_,8 @@
         }
 
         public void applyToMap(MapItemSavedData p_164833_) {
-            for(int i = 0; i < this.width; ++i) {
-                for(int j = 0; j < this.height; ++j) {
+            for (int i = 0; i < this.width; i++) {
+                for (int j = 0; j < this.height; j++) {
                     p_164833_.setColor(this.startX + i, this.startY + j, this.mapColors[i + j * this.width]);
                 }
             }
