--- a/net/minecraft/world/level/saveddata/maps/MapIndex.java
+++ b/net/minecraft/world/level/saveddata/maps/MapIndex.java
@@ -23,7 +_,7 @@
     public static MapIndex load(CompoundTag p_164763_, HolderLookup.Provider p_323806_) {
         MapIndex mapindex = new MapIndex();
 
-        for(String s : p_164763_.getAllKeys()) {
+        for (String s : p_164763_.getAllKeys()) {
             if (p_164763_.contains(s, 99)) {
                 mapindex.usedAuxIds.put(s, p_164763_.getInt(s));
             }
@@ -34,7 +_,7 @@
 
     @Override
     public CompoundTag save(CompoundTag p_77884_, HolderLookup.Provider p_324049_) {
-        for(Entry<String> entry : this.usedAuxIds.object2IntEntrySet()) {
+        for (Entry<String> entry : this.usedAuxIds.object2IntEntrySet()) {
             p_77884_.putInt(entry.getKey(), entry.getIntValue());
         }
 
