--- a/net/minecraft/server/network/Filterable.java
+++ b/net/minecraft/server/network/Filterable.java
@@ -8,19 +_,17 @@
 import java.util.function.Function;
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
-import net.minecraft.util.ExtraCodecs;
 
 public record Filterable<T>(T raw, Optional<T> filtered) {
     public static <T> Codec<Filterable<T>> codec(Codec<T> p_331745_) {
         Codec<Filterable<T>> codec = RecordCodecBuilder.create(
-            p_331913_ -> p_331913_.group(
-                        p_331745_.fieldOf("text").forGetter(Filterable::raw),
-                        ExtraCodecs.strictOptionalField(p_331745_, "filtered").forGetter(Filterable::filtered)
+            p_337552_ -> p_337552_.group(
+                        p_331745_.fieldOf("raw").forGetter(Filterable::raw), p_331745_.optionalFieldOf("filtered").forGetter(Filterable::filtered)
                     )
-                    .apply(p_331913_, Filterable::new)
+                    .apply(p_337552_, Filterable::new)
         );
         Codec<Filterable<T>> codec1 = p_331745_.xmap(Filterable::passThrough, Filterable::raw);
-        return ExtraCodecs.withAlternative(codec, codec1);
+        return Codec.withAlternative(codec, codec1);
     }
 
     public static <B extends ByteBuf, T> StreamCodec<B, Filterable<T>> streamCodec(StreamCodec<B, T> p_330521_) {
@@ -36,7 +_,7 @@
     }
 
     public T get(boolean p_331777_) {
-        return (T)(p_331777_ ? this.filtered.orElse(this.raw) : this.raw);
+        return p_331777_ ? this.filtered.orElse(this.raw) : this.raw;
     }
 
     public <U> Filterable<U> map(Function<T, U> p_331765_) {
