--- a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
@@ -28,6 +_,7 @@
 public abstract class ServerCommonPacketListenerImpl implements ServerCommonPacketListener {
     private static final Logger LOGGER = LogUtils.getLogger();
     public static final int LATENCY_CHECK_INTERVAL = 15000;
+    private static final int CLOSED_LISTENER_TIMEOUT = 15000;
     private static final Component TIMEOUT_DISCONNECTION_MESSAGE = Component.translatable("disconnect.timeout");
     static final Component DISCONNECT_UNEXPECTED_QUERY = Component.translatable("multiplayer.disconnect.unexpected_query_response");
     protected final MinecraftServer server;
@@ -36,6 +_,8 @@
     private long keepAliveTime;
     private boolean keepAlivePending;
     private long keepAliveChallenge;
+    private long closedListenerTime;
+    private boolean closed = false;
     private int latency;
     private volatile boolean suspendFlushingOnServerThread = false;
 
@@ -47,6 +_,13 @@
         this.transferred = p_301980_.transferred();
     }
 
+    private void close() {
+        if (!this.closed) {
+            this.closedListenerTime = Util.getMillis();
+            this.closed = true;
+        }
+    }
+
     @Override
     public void onDisconnect(Component p_295648_) {
         if (this.isSingleplayerOwner()) {
@@ -91,10 +_,10 @@
     protected void keepConnectionAlive() {
         this.server.getProfiler().push("keepAlive");
         long i = Util.getMillis();
-        if (i - this.keepAliveTime >= 15000L) {
+        if (!this.isSingleplayerOwner() && i - this.keepAliveTime >= 15000L) {
             if (this.keepAlivePending) {
                 this.disconnect(TIMEOUT_DISCONNECTION_MESSAGE);
-            } else {
+            } else if (this.checkIfClosed(i)) {
                 this.keepAlivePending = true;
                 this.keepAliveTime = i;
                 this.keepAliveChallenge = i;
@@ -105,6 +_,18 @@
         this.server.getProfiler().pop();
     }
 
+    private boolean checkIfClosed(long p_339648_) {
+        if (this.closed) {
+            if (p_339648_ - this.closedListenerTime >= 15000L) {
+                this.disconnect(TIMEOUT_DISCONNECTION_MESSAGE);
+            }
+
+            return false;
+        } else {
+            return true;
+        }
+    }
+
     public void suspendFlushing() {
         this.suspendFlushingOnServerThread = true;
     }
@@ -118,7 +_,16 @@
         this.send(p_294278_, null);
     }
 
+    @Override
     public void send(Packet<?> p_295099_, @Nullable PacketSendListener p_296321_) {
+        if (!net.neoforged.neoforge.network.registration.NetworkRegistry.getInstance().canSendPacket(p_295099_, this)) {
+            return;
+        }
+
+        if (p_295099_.isTerminal()) {
+            this.close();
+        }
+
         boolean flag = !this.suspendFlushingOnServerThread || !this.server.isSameThread();
 
         try {
@@ -152,7 +_,36 @@
         return this.latency;
     }
 
+    /**
+     * Creates a new cookie for this connection.
+     *
+     * @param p_301973_ The client information.
+     * @return The cookie.
+     * @deprecated Use {@link #createCookie(ClientInformation, net.neoforged.neoforge.network.connection.ConnectionType)} instead, keeping the connection type information available.
+     */
+    @Deprecated
     protected CommonListenerCookie createCookie(ClientInformation p_301973_) {
         return new CommonListenerCookie(this.playerProfile(), this.latency, p_301973_, this.transferred);
+    }
+
+    /**
+     * Creates a new cookie for this connection.
+     *
+     * @param p_301973_ The client information.
+     * @param connectionType Whether the connection is modded.
+     * @return The cookie.
+     */
+    protected CommonListenerCookie createCookie(ClientInformation p_301973_, net.neoforged.neoforge.network.connection.ConnectionType connectionType) {
+        return new CommonListenerCookie(this.playerProfile(), this.latency, p_301973_, this.transferred, connectionType);
+    }
+
+    @Override
+    public Connection getConnection() {
+        return connection;
+    }
+
+    @Override
+    public net.minecraft.util.thread.ReentrantBlockableEventLoop<?> getMainThreadEventLoop() {
+        return server;
     }
 }
