--- a/net/minecraft/server/network/ServerHandshakePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerHandshakePacketListenerImpl.java
@@ -23,7 +_,7 @@
 
     @Override
     public void handleIntention(ClientIntentionPacket p_9975_) {
-        switch(p_9975_.intention()) {
+        switch (p_9975_.intention()) {
             case LOGIN:
                 this.beginLogin(p_9975_, false);
                 break;
