--- a/net/minecraft/client/player/inventory/Hotbar.java
+++ b/net/minecraft/client/player/inventory/Hotbar.java
@@ -15,7 +_,6 @@
 import net.minecraft.nbt.NbtOps;
 import net.minecraft.nbt.Tag;
 import net.minecraft.resources.RegistryOps;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.item.ItemStack;
 import net.neoforged.api.distmarker.Dist;
@@ -26,12 +_,12 @@
 public class Hotbar {
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final int SIZE = Inventory.getSelectionSize();
-    public static final Codec<Hotbar> CODEC = ExtraCodecs.<List<Dynamic<?>>>validate(Codec.PASSTHROUGH.listOf(), p_324442_ -> Util.fixedSize(p_324442_, SIZE))
+    public static final Codec<Hotbar> CODEC = Codec.PASSTHROUGH
+        .listOf()
+        .validate(p_324442_ -> Util.fixedSize(p_324442_, SIZE))
         .xmap(Hotbar::new, p_323563_ -> p_323563_.items);
     private static final DynamicOps<Tag> DEFAULT_OPS = NbtOps.INSTANCE;
-    private static final Dynamic<?> EMPTY_STACK = new Dynamic<>(
-        DEFAULT_OPS, Util.getOrThrow(ItemStack.OPTIONAL_CODEC.encodeStart(DEFAULT_OPS, ItemStack.EMPTY), IllegalStateException::new)
-    );
+    private static final Dynamic<?> EMPTY_STACK = new Dynamic<>(DEFAULT_OPS, ItemStack.OPTIONAL_CODEC.encodeStart(DEFAULT_OPS, ItemStack.EMPTY).getOrThrow());
     private List<Dynamic<?>> items;
 
     private Hotbar(List<Dynamic<?>> p_324385_) {
@@ -46,8 +_,8 @@
         return this.items
             .stream()
             .map(
-                p_329785_ -> ItemStack.OPTIONAL_CODEC
-                        .parse(RegistryOps.injectRegistryContext(p_329785_, p_323741_))
+                p_337421_ -> ItemStack.OPTIONAL_CODEC
+                        .parse(RegistryOps.injectRegistryContext((Dynamic<?>)p_337421_, p_323741_))
                         .resultOrPartial(p_323502_ -> LOGGER.warn("Could not parse hotbar item: {}", p_323502_))
                         .orElse(ItemStack.EMPTY)
             )
@@ -58,7 +_,7 @@
         RegistryOps<Tag> registryops = p_324514_.createSerializationContext(DEFAULT_OPS);
         Builder<Dynamic<?>> builder = ImmutableList.builderWithExpectedSize(SIZE);
 
-        for(int i = 0; i < SIZE; ++i) {
+        for (int i = 0; i < SIZE; i++) {
             ItemStack itemstack = p_324559_.getItem(i);
             Optional<Dynamic<?>> optional = ItemStack.OPTIONAL_CODEC
                 .encodeStart(registryops, itemstack)
@@ -71,7 +_,7 @@
     }
 
     public boolean isEmpty() {
-        for(Dynamic<?> dynamic : this.items) {
+        for (Dynamic<?> dynamic : this.items) {
             if (!isEmpty(dynamic)) {
                 return false;
             }
