--- a/com/mojang/realmsclient/dto/RealmsServerPlayerLists.java
+++ b/com/mojang/realmsclient/dto/RealmsServerPlayerLists.java
@@ -28,7 +_,7 @@
                 JsonArray jsonarray = jsonobject.get("lists").getAsJsonArray();
                 Iterator<JsonElement> iterator = jsonarray.iterator();
 
-                while(iterator.hasNext()) {
+                while (iterator.hasNext()) {
                     realmsserverplayerlists.servers.add(RealmsServerPlayerList.parse(iterator.next().getAsJsonObject()));
                 }
             }
