--- a/com/mojang/realmsclient/dto/ServerActivityList.java
+++ b/com/mojang/realmsclient/dto/ServerActivityList.java
@@ -24,7 +_,7 @@
             serveractivitylist.periodInMillis = JsonUtils.getLongOr("periodInMillis", jsonobject, -1L);
             JsonElement jsonelement1 = jsonobject.get("playerActivityDto");
             if (jsonelement1 != null && jsonelement1.isJsonArray()) {
-                for(JsonElement jsonelement2 : jsonelement1.getAsJsonArray()) {
+                for (JsonElement jsonelement2 : jsonelement1.getAsJsonArray()) {
                     ServerActivity serveractivity = ServerActivity.parse(jsonelement2.getAsJsonObject());
                     serveractivitylist.serverActivities.add(serveractivity);
                 }
