--- a/com/mojang/realmsclient/dto/WorldTemplatePaginatedList.java
+++ b/com/mojang/realmsclient/dto/WorldTemplatePaginatedList.java
@@ -45,7 +_,7 @@
             if (jsonobject.get("templates").isJsonArray()) {
                 Iterator<JsonElement> iterator = jsonobject.get("templates").getAsJsonArray().iterator();
 
-                while(iterator.hasNext()) {
+                while (iterator.hasNext()) {
                     worldtemplatepaginatedlist.templates.add(WorldTemplate.parse(iterator.next().getAsJsonObject()));
                 }
             }
