--- a/net/minecraft/world/damagesource/CombatRules.java
+++ b/net/minecraft/world/damagesource/CombatRules.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.damagesource;
 
 import net.minecraft.util.Mth;
+import net.minecraft.world.item.enchantment.EnchantmentHelper;
 
 public class CombatRules {
     public static final float MAX_ARMOR = 20.0F;
@@ -9,10 +_,13 @@
     public static final float MIN_ARMOR_RATIO = 0.2F;
     private static final int NUM_ARMOR_ITEMS = 4;
 
-    public static float getDamageAfterAbsorb(float p_19273_, float p_19274_, float p_19275_) {
+    public static float getDamageAfterAbsorb(float p_19273_, DamageSource p_338892_, float p_19274_, float p_19275_) {
         float f = 2.0F + p_19275_ / 4.0F;
         float f1 = Mth.clamp(p_19274_ - p_19273_ / f, p_19274_ * 0.2F, 20.0F);
-        return p_19273_ * (1.0F - f1 / 25.0F);
+        float f2 = f1 / 25.0F;
+        float f3 = EnchantmentHelper.calculateArmorBreach(p_338892_.getEntity(), f2);
+        float f4 = 1.0F - f3;
+        return p_19273_ * f4;
     }
 
     public static float getDamageAfterMagicAbsorb(float p_19270_, float p_19271_) {
