--- a/net/minecraft/world/damagesource/DamageSources.java
+++ b/net/minecraft/world/damagesource/DamageSources.java
@@ -21,7 +_,6 @@
     private final DamageSource lightningBolt;
     private final DamageSource onFire;
     private final DamageSource lava;
-    private final DamageSource potatoHeat;
     private final DamageSource hotFloor;
     private final DamageSource inWall;
     private final DamageSource cramming;
@@ -41,7 +_,6 @@
     private final DamageSource stalagmite;
     private final DamageSource outsideBorder;
     private final DamageSource genericKill;
-    private final DamageSource potatoMagic;
 
     public DamageSources(RegistryAccess p_270740_) {
         this.damageTypes = p_270740_.registryOrThrow(Registries.DAMAGE_TYPE);
@@ -49,7 +_,6 @@
         this.lightningBolt = this.source(DamageTypes.LIGHTNING_BOLT);
         this.onFire = this.source(DamageTypes.ON_FIRE);
         this.lava = this.source(DamageTypes.LAVA);
-        this.potatoHeat = this.source(DamageTypes.POTATO_HEAT);
         this.hotFloor = this.source(DamageTypes.HOT_FLOOR);
         this.inWall = this.source(DamageTypes.IN_WALL);
         this.cramming = this.source(DamageTypes.CRAMMING);
@@ -69,7 +_,6 @@
         this.stalagmite = this.source(DamageTypes.STALAGMITE);
         this.outsideBorder = this.source(DamageTypes.OUTSIDE_BORDER);
         this.genericKill = this.source(DamageTypes.GENERIC_KILL);
-        this.potatoMagic = this.source(DamageTypes.POTATO_MAGIC);
     }
 
     public DamageSource source(ResourceKey<DamageType> p_270957_) {
@@ -100,10 +_,6 @@
         return this.lava;
     }
 
-    public DamageSource potatoHeat() {
-        return this.potatoHeat;
-    }
-
     public DamageSource hotFloor() {
         return this.hotFloor;
     }
@@ -146,10 +_,6 @@
 
     public DamageSource magic() {
         return this.magic;
-    }
-
-    public DamageSource potatoMagic() {
-        return this.potatoMagic;
     }
 
     public DamageSource wither() {
