--- a/net/minecraft/world/damagesource/DamageTypes.java
+++ b/net/minecraft/world/damagesource/DamageTypes.java
@@ -10,7 +_,6 @@
     ResourceKey<DamageType> LIGHTNING_BOLT = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("lightning_bolt"));
     ResourceKey<DamageType> ON_FIRE = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("on_fire"));
     ResourceKey<DamageType> LAVA = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("lava"));
-    ResourceKey<DamageType> POTATO_HEAT = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("potato_heat"));
     ResourceKey<DamageType> HOT_FLOOR = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("hot_floor"));
     ResourceKey<DamageType> IN_WALL = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("in_wall"));
     ResourceKey<DamageType> CRAMMING = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("cramming"));
@@ -53,14 +_,12 @@
     ResourceKey<DamageType> BAD_RESPAWN_POINT = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("bad_respawn_point"));
     ResourceKey<DamageType> OUTSIDE_BORDER = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("outside_border"));
     ResourceKey<DamageType> GENERIC_KILL = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("generic_kill"));
-    ResourceKey<DamageType> POTATO_MAGIC = ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation("potato_magic"));
 
     static void bootstrap(BootstrapContext<DamageType> p_321708_) {
         p_321708_.register(IN_FIRE, new DamageType("inFire", 0.1F, DamageEffects.BURNING));
         p_321708_.register(LIGHTNING_BOLT, new DamageType("lightningBolt", 0.1F));
         p_321708_.register(ON_FIRE, new DamageType("onFire", 0.0F, DamageEffects.BURNING));
         p_321708_.register(LAVA, new DamageType("lava", 0.1F, DamageEffects.BURNING));
-        p_321708_.register(POTATO_HEAT, new DamageType("potato_heat", 0.1F, DamageEffects.BURNING));
         p_321708_.register(HOT_FLOOR, new DamageType("hotFloor", 0.1F, DamageEffects.BURNING));
         p_321708_.register(IN_WALL, new DamageType("inWall", 0.0F));
         p_321708_.register(CRAMMING, new DamageType("cramming", 0.0F));
@@ -106,6 +_,5 @@
         );
         p_321708_.register(OUTSIDE_BORDER, new DamageType("outsideBorder", 0.0F));
         p_321708_.register(GENERIC_KILL, new DamageType("genericKill", 0.0F));
-        p_321708_.register(POTATO_MAGIC, new DamageType("potato_magic", DamageScaling.ALWAYS, 0.2F));
     }
 }
