--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonChargePlayerPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonChargePlayerPhase.java
@@ -27,7 +_,7 @@
         } else {
             double d0 = this.targetLocation.distanceToSqr(this.dragon.getX(), this.dragon.getY(), this.dragon.getZ());
             if (d0 < 100.0 || d0 > 22500.0 || this.dragon.horizontalCollision || this.dragon.verticalCollision) {
-                ++this.timeSinceCharge;
+                this.timeSinceCharge++;
             }
         }
     }
