--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingApproachPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingApproachPhase.java
@@ -88,7 +_,7 @@
             double d2;
             do {
                 d2 = (double)((float)vec3i.getY() + this.dragon.getRandom().nextFloat() * 20.0F);
-            } while(d2 < (double)vec3i.getY());
+            } while (d2 < (double)vec3i.getY());
 
             this.targetLocation = new Vec3(d0, d2, d1);
         }
