--- a/net/minecraft/data/recipes/RecipeBuilder.java
+++ b/net/minecraft/data/recipes/RecipeBuilder.java
@@ -38,7 +_,7 @@
     }
 
     static CraftingBookCategory determineBookCategory(RecipeCategory p_311837_) {
-        return switch(p_311837_) {
+        return switch (p_311837_) {
             case BUILDING_BLOCKS -> CraftingBookCategory.BUILDING;
             case TOOLS, COMBAT -> CraftingBookCategory.EQUIPMENT;
             case REDSTONE -> CraftingBookCategory.REDSTONE;
