--- a/net/minecraft/data/recipes/SingleItemRecipeBuilder.java
+++ b/net/minecraft/data/recipes/SingleItemRecipeBuilder.java
@@ -13,7 +_,6 @@
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.crafting.Ingredient;
-import net.minecraft.world.item.crafting.PoisonousPotatoCutterRecipe;
 import net.minecraft.world.item.crafting.SingleItemRecipe;
 import net.minecraft.world.item.crafting.StonecutterRecipe;
 import net.minecraft.world.level.ItemLike;
@@ -42,10 +_,6 @@
 
     public static SingleItemRecipeBuilder stonecutting(Ingredient p_251375_, RecipeCategory p_248984_, ItemLike p_250105_, int p_249506_) {
         return new SingleItemRecipeBuilder(p_248984_, StonecutterRecipe::new, p_251375_, p_250105_, p_249506_);
-    }
-
-    public static SingleItemRecipeBuilder poisonous_potato_cutting(Ingredient p_340468_, RecipeCategory p_341244_, ItemLike p_342157_, int p_340454_) {
-        return new SingleItemRecipeBuilder(p_341244_, PoisonousPotatoCutterRecipe::new, p_340468_, p_342157_, p_340454_);
     }
 
     public SingleItemRecipeBuilder unlockedBy(String p_176810_, Criterion<?> p_301267_) {
