--- a/net/minecraft/util/random/WeightedRandomList.java
+++ b/net/minecraft/util/random/WeightedRandomList.java
@@ -3,7 +_,9 @@
 import com.google.common.collect.ImmutableList;
 import com.mojang.serialization.Codec;
 import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
+import javax.annotation.Nullable;
 import net.minecraft.util.RandomSource;
 
 public class WeightedRandomList<E extends WeightedEntry> {
@@ -47,5 +_,22 @@
 
     public static <E extends WeightedEntry> Codec<WeightedRandomList<E>> codec(Codec<E> p_146334_) {
         return p_146334_.listOf().xmap(WeightedRandomList::create, WeightedRandomList::unwrap);
+    }
+
+    @Override
+    public boolean equals(@Nullable Object p_338205_) {
+        if (this == p_338205_) {
+            return true;
+        } else if (p_338205_ != null && this.getClass() == p_338205_.getClass()) {
+            WeightedRandomList<?> weightedrandomlist = (WeightedRandomList<?>)p_338205_;
+            return this.totalWeight == weightedrandomlist.totalWeight && Objects.equals(this.items, weightedrandomlist.items);
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.totalWeight, this.items);
     }
 }
