--- a/net/minecraft/world/level/levelgen/feature/stateproviders/BlockStateProviderType.java
+++ b/net/minecraft/world/level/levelgen/feature/stateproviders/BlockStateProviderType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.stateproviders;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 
@@ -16,17 +_,17 @@
     public static final BlockStateProviderType<RandomizedIntStateProvider> RANDOMIZED_INT_STATE_PROVIDER = register(
         "randomized_int_state_provider", RandomizedIntStateProvider.CODEC
     );
-    private final Codec<P> codec;
-
-    private static <P extends BlockStateProvider> BlockStateProviderType<P> register(String p_68763_, Codec<P> p_68764_) {
-        return Registry.register(BuiltInRegistries.BLOCKSTATE_PROVIDER_TYPE, p_68763_, new BlockStateProviderType<>(p_68764_));
-    }
-
-    public BlockStateProviderType(Codec<P> p_68760_) {
-        this.codec = p_68760_;
-    }
-
-    public Codec<P> codec() {
+    private final MapCodec<P> codec;
+
+    private static <P extends BlockStateProvider> BlockStateProviderType<P> register(String p_68763_, MapCodec<P> p_338587_) {
+        return Registry.register(BuiltInRegistries.BLOCKSTATE_PROVIDER_TYPE, p_68763_, new BlockStateProviderType<>(p_338587_));
+    }
+
+    private BlockStateProviderType(MapCodec<P> p_338337_) {
+        this.codec = p_338337_;
+    }
+
+    public MapCodec<P> codec() {
         return this.codec;
     }
 }
