--- a/net/minecraft/world/level/levelgen/feature/stateproviders/RotatedBlockProvider.java
+++ b/net/minecraft/world/level/levelgen/feature/stateproviders/RotatedBlockProvider.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.stateproviders;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.util.RandomSource;
@@ -10,11 +_,10 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class RotatedBlockProvider extends BlockStateProvider {
-    public static final Codec<RotatedBlockProvider> CODEC = BlockState.CODEC
+    public static final MapCodec<RotatedBlockProvider> CODEC = BlockState.CODEC
         .fieldOf("state")
         .xmap(BlockBehaviour.BlockStateBase::getBlock, Block::defaultBlockState)
-        .xmap(RotatedBlockProvider::new, p_68793_ -> p_68793_.block)
-        .codec();
+        .xmap(RotatedBlockProvider::new, p_68793_ -> p_68793_.block);
     private final Block block;
 
     public RotatedBlockProvider(Block p_68790_) {
