--- a/net/minecraft/world/level/levelgen/feature/stateproviders/WeightedStateProvider.java
+++ b/net/minecraft/world/level/levelgen/feature/stateproviders/WeightedStateProvider.java
@@ -1,17 +_,16 @@
 package net.minecraft.world.level.levelgen.feature.stateproviders;
 
-import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.BlockPos;
 import net.minecraft.util.RandomSource;
 import net.minecraft.util.random.SimpleWeightedRandomList;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class WeightedStateProvider extends BlockStateProvider {
-    public static final Codec<WeightedStateProvider> CODEC = SimpleWeightedRandomList.wrappedCodec(BlockState.CODEC)
+    public static final MapCodec<WeightedStateProvider> CODEC = SimpleWeightedRandomList.wrappedCodec(BlockState.CODEC)
         .comapFlatMap(WeightedStateProvider::create, p_161600_ -> p_161600_.weightedList)
-        .fieldOf("entries")
-        .codec();
+        .fieldOf("entries");
     private final SimpleWeightedRandomList<BlockState> weightedList;
 
     private static DataResult<WeightedStateProvider> create(SimpleWeightedRandomList<BlockState> p_161598_) {
