--- a/net/minecraft/world/entity/ai/gossip/GossipContainer.java
+++ b/net/minecraft/world/entity/ai/gossip/GossipContainer.java
@@ -48,7 +_,7 @@
     public void decay() {
         Iterator<GossipContainer.EntityGossips> iterator = this.gossips.values().iterator();
 
-        while(iterator.hasNext()) {
+        while (iterator.hasNext()) {
             GossipContainer.EntityGossips gossipcontainer$entitygossips = iterator.next();
             gossipcontainer$entitygossips.decay();
             if (gossipcontainer$entitygossips.isEmpty()) {
@@ -69,7 +_,7 @@
             int[] aint = new int[list.size()];
             int i = 0;
 
-            for(int j = 0; j < list.size(); ++j) {
+            for (int j = 0; j < list.size(); j++) {
                 GossipContainer.GossipEntry gossipcontainer$gossipentry = list.get(j);
                 i += Math.abs(gossipcontainer$gossipentry.weightedValue());
                 aint[j] = i - 1;
@@ -77,7 +_,7 @@
 
             Set<GossipContainer.GossipEntry> set = Sets.newIdentityHashSet();
 
-            for(int i1 = 0; i1 < p_217761_; ++i1) {
+            for (int i1 = 0; i1 < p_217761_; i1++) {
                 int k = p_217760_.nextInt(i);
                 int l = Arrays.binarySearch(aint, k);
                 set.add(list.get(l < 0 ? -l - 1 : l));
@@ -141,7 +_,7 @@
     public void remove(GossipType p_148161_) {
         Iterator<GossipContainer.EntityGossips> iterator = this.gossips.values().iterator();
 
-        while(iterator.hasNext()) {
+        while (iterator.hasNext()) {
             GossipContainer.EntityGossips gossipcontainer$entitygossips = iterator.next();
             gossipcontainer$entitygossips.remove(p_148161_);
             if (gossipcontainer$entitygossips.isEmpty()) {
@@ -162,7 +_,7 @@
             .decode(p_26178_)
             .resultOrPartial(p_262901_ -> LOGGER.warn("Failed to deserialize gossips: {}", p_262901_))
             .stream()
-            .flatMap(p_262899_ -> (p_262899_.getFirst()).stream())
+            .flatMap(p_262899_ -> p_262899_.getFirst().stream())
             .forEach(p_26162_ -> this.getOrCreate(p_26162_.target).entries.put(p_26162_.type, p_26162_.value));
     }
 
@@ -197,7 +_,7 @@
         public void decay() {
             ObjectIterator<Entry<GossipType>> objectiterator = this.entries.object2IntEntrySet().iterator();
 
-            while(objectiterator.hasNext()) {
+            while (objectiterator.hasNext()) {
                 Entry<GossipType> entry = objectiterator.next();
                 int i = entry.getIntValue() - entry.getKey().decayPerDay;
                 if (i < 2) {
