--- a/net/minecraft/client/gui/screens/debug/GameModeSwitcherScreen.java
+++ b/net/minecraft/client/gui/screens/debug/GameModeSwitcherScreen.java
@@ -64,7 +_,7 @@
         super.init();
         this.currentlyHovered = this.previousHovered;
 
-        for(int i = 0; i < GameModeSwitcherScreen.GameModeIcon.VALUES.length; ++i) {
+        for (int i = 0; i < GameModeSwitcherScreen.GameModeIcon.VALUES.length; i++) {
             GameModeSwitcherScreen.GameModeIcon gamemodeswitcherscreen$gamemodeicon = GameModeSwitcherScreen.GameModeIcon.VALUES[i];
             this.slots
                 .add(
@@ -95,7 +_,7 @@
 
             boolean flag = this.firstMouseX == p_283223_ && this.firstMouseY == p_282178_;
 
-            for(GameModeSwitcherScreen.GameModeSlot gamemodeswitcherscreen$gamemodeslot : this.slots) {
+            for (GameModeSwitcherScreen.GameModeSlot gamemodeswitcherscreen$gamemodeslot : this.slots) {
                 gamemodeswitcherscreen$gamemodeslot.render(p_281834_, p_283223_, p_282178_, p_281339_);
                 gamemodeswitcherscreen$gamemodeslot.setSelected(this.currentlyHovered == gamemodeswitcherscreen$gamemodeslot.icon);
                 if (!flag && gamemodeswitcherscreen$gamemodeslot.isHoveredOrFocused()) {
@@ -152,7 +_,7 @@
 
     @OnlyIn(Dist.CLIENT)
     static enum GameModeIcon {
-        CREATIVE(Component.translatable("gameMode.creative"), "gamemode creative", new ItemStack(Blocks.PEELGRASS_BLOCK)),
+        CREATIVE(Component.translatable("gameMode.creative"), "gamemode creative", new ItemStack(Blocks.GRASS_BLOCK)),
         SURVIVAL(Component.translatable("gameMode.survival"), "gamemode survival", new ItemStack(Items.IRON_SWORD)),
         ADVENTURE(Component.translatable("gameMode.adventure"), "gamemode adventure", new ItemStack(Items.MAP)),
         SPECTATOR(Component.translatable("gameMode.spectator"), "gamemode spectator", new ItemStack(Items.ENDER_EYE));
@@ -183,7 +_,7 @@
         }
 
         GameModeSwitcherScreen.GameModeIcon getNext() {
-            return switch(this) {
+            return switch (this) {
                 case CREATIVE -> SURVIVAL;
                 case SURVIVAL -> ADVENTURE;
                 case ADVENTURE -> SPECTATOR;
@@ -192,11 +_,12 @@
         }
 
         static GameModeSwitcherScreen.GameModeIcon getFromGameType(GameType p_283307_) {
-            return switch(p_283307_) {
+            return switch (p_283307_) {
                 case SPECTATOR -> SPECTATOR;
                 case SURVIVAL -> SURVIVAL;
                 case CREATIVE -> CREATIVE;
                 case ADVENTURE -> ADVENTURE;
+                default -> throw new MatchException(null, null);
             };
         }
     }
