--- a/net/minecraft/client/resources/metadata/animation/AnimationMetadataSectionSerializer.java
+++ b/net/minecraft/client/resources/metadata/animation/AnimationMetadataSectionSerializer.java
@@ -26,7 +_,7 @@
             try {
                 JsonArray jsonarray = GsonHelper.getAsJsonArray(p_119064_, "frames");
 
-                for(int j = 0; j < jsonarray.size(); ++j) {
+                for (int j = 0; j < jsonarray.size(); j++) {
                     JsonElement jsonelement = jsonarray.get(j);
                     AnimationFrame animationframe = this.getFrame(j, jsonelement);
                     if (animationframe != null) {
