--- a/net/minecraft/client/resources/metadata/animation/VillagerMetaDataSection.java
+++ b/net/minecraft/client/resources/metadata/animation/VillagerMetaDataSection.java
@@ -27,7 +_,7 @@
         FULL("full");
 
         private static final Map<String, VillagerMetaDataSection.Hat> BY_NAME = Arrays.stream(values())
-            .collect(Collectors.toMap(VillagerMetaDataSection.Hat::getName, p_119084_ -> p_119084_));
+            .collect(Collectors.toMap(VillagerMetaDataSection.Hat::getName, p_119084_ -> (VillagerMetaDataSection.Hat)p_119084_));
         private final String name;
 
         private Hat(String p_119081_) {
