--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1312,7 +1333,7 @@
                 case ABORT_DESTROY_BLOCK:
                 case STOP_DESTROY_BLOCK:
                     this.player.gameMode.handleBlockBreakAction(pos, action, packet.getDirection(), this.player.level().getMaxY(), packet.getSequence());
-                    this.ackBlockChangesUpTo(packet.getSequence());
+                    this.player.connection.ackBlockChangesUpTo = packet.getSequence();
                     return;
                 default:
                     throw new IllegalArgumentException("Invalid player action");
