--- a/net/minecraft/world/level/levelgen/structure/pools/EmptyPoolElement.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/EmptyPoolElement.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.structure.pools;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.Collections;
 import java.util.List;
 import net.minecraft.core.BlockPos;
@@ -15,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager;
 
 public class EmptyPoolElement extends StructurePoolElement {
-    public static final Codec<EmptyPoolElement> CODEC = Codec.unit(() -> EmptyPoolElement.INSTANCE);
+    public static final MapCodec<EmptyPoolElement> CODEC = MapCodec.unit(() -> EmptyPoolElement.INSTANCE);
     public static final EmptyPoolElement INSTANCE = new EmptyPoolElement();
 
     private EmptyPoolElement() {
