--- a/net/minecraft/world/level/levelgen/structure/pools/JigsawPlacement.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/JigsawPlacement.java
@@ -65,7 +_,7 @@
         Registry<StructureTemplatePool> registry = registryaccess.registryOrThrow(Registries.TEMPLATE_POOL);
         Rotation rotation = Rotation.getRandom(worldgenrandom);
         StructureTemplatePool structuretemplatepool = p_227240_.unwrapKey()
-            .flatMap(p_314915_ -> registry.getOptional(p_307522_.lookup(p_314915_)))
+            .flatMap(p_314915_ -> registry.getOptional(p_307522_.lookup((ResourceKey<StructureTemplatePool>)p_314915_)))
             .orElse(p_227240_.value());
         StructurePoolElement structurepoolelement = structuretemplatepool.getRandomTemplate(worldgenrandom);
         if (structurepoolelement == EmptyPoolElement.INSTANCE) {
@@ -163,7 +_,7 @@
         List<StructureTemplate.StructureBlockInfo> list = p_227248_.getShuffledJigsawBlocks(p_227252_, p_227250_, p_227251_, p_227253_);
         Optional<BlockPos> optional = Optional.empty();
 
-        for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : list) {
+        for (StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : list) {
             ResourceLocation resourcelocation = ResourceLocation.tryParse(
                 Objects.requireNonNull(structuretemplate$structureblockinfo.nbt(), () -> structuretemplate$structureblockinfo + " nbt was null")
                     .getString("name")
@@ -194,7 +_,7 @@
         JigsawPlacement.Placer jigsawplacement$placer = new JigsawPlacement.Placer(p_227218_, p_227212_, p_227214_, p_227215_, p_227220_, p_227217_);
         jigsawplacement$placer.tryPlacingChildren(p_227219_, new MutableObject<>(p_227221_), 0, p_227213_, p_227216_, p_227211_, p_307210_);
 
-        while(jigsawplacement$placer.placing.hasNext()) {
+        while (jigsawplacement$placer.placing.hasNext()) {
             JigsawPlacement.PieceState jigsawplacement$piecestate = jigsawplacement$placer.placing.next();
             jigsawplacement$placer.tryPlacingChildren(
                 jigsawplacement$piecestate.piece, jigsawplacement$piecestate.free, jigsawplacement$piecestate.depth, p_227213_, p_227216_, p_227211_, p_307210_
@@ -218,8 +_,7 @@
             p_227204_.getSeed(),
             new ChunkPos(p_227208_),
             p_227204_,
-            p_227255_ -> true,
-            List.of()
+            p_227255_ -> true
         );
         Optional<Structure.GenerationStub> optional = addPieces(
             structure$generationcontext, p_227205_, Optional.of(p_227206_), p_227207_, p_227208_, false, Optional.empty(), 128, PoolAliasLookup.EMPTY
@@ -227,7 +_,7 @@
         if (optional.isPresent()) {
             StructurePiecesBuilder structurepiecesbuilder = optional.get().getPiecesBuilder();
 
-            for(StructurePiece structurepiece : structurepiecesbuilder.build().pieces()) {
+            for (StructurePiece structurepiece : structurepiecesbuilder.build().pieces()) {
                 if (structurepiece instanceof PoolElementStructurePiece poolelementstructurepiece) {
                     poolelementstructurepiece.place(p_227204_, structuremanager, chunkgenerator, randomsource, BoundingBox.infinite(), p_227208_, p_227209_);
                 }
@@ -286,7 +_,7 @@
             int i = boundingbox.minY();
 
             label134:
-            for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : structurepoolelement.getShuffledJigsawBlocks(
+            for (StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : structurepoolelement.getShuffledJigsawBlocks(
                 this.structureTemplateManager, blockpos, rotation, this.random
             )) {
                 Direction direction = JigsawBlock.getFrontFacing(structuretemplate$structureblockinfo.state());
@@ -299,7 +_,7 @@
                 if (optional.isEmpty()) {
                     JigsawPlacement.LOGGER.warn("Empty or non-existent pool: {}", resourcekey.location());
                 } else {
-                    Holder<StructureTemplatePool> holder = optional.get();
+                    Holder<StructureTemplatePool> holder = (Holder<StructureTemplatePool>)optional.get();
                     if (holder.value().size() == 0 && !holder.is(Pools.EMPTY)) {
                         JigsawPlacement.LOGGER.warn("Empty or non-existent pool: {}", resourcekey.location());
                     } else {
@@ -332,12 +_,12 @@
                                 ? structuretemplate$structureblockinfo.nbt().getInt("placement_priority")
                                 : 0;
 
-                            for(StructurePoolElement structurepoolelement1 : list) {
+                            for (StructurePoolElement structurepoolelement1 : list) {
                                 if (structurepoolelement1 == EmptyPoolElement.INSTANCE) {
                                     break;
                                 }
 
-                                for(Rotation rotation1 : Rotation.getShuffled(this.random)) {
+                                for (Rotation rotation1 : Rotation.getShuffled(this.random)) {
                                     List<StructureTemplate.StructureBlockInfo> list1 = structurepoolelement1.getShuffledJigsawBlocks(
                                         this.structureTemplateManager, BlockPos.ZERO, rotation1, this.random
                                     );
@@ -373,7 +_,7 @@
                                         i1 = 0;
                                     }
 
-                                    for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo1 : list1) {
+                                    for (StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo1 : list1) {
                                         if (JigsawBlock.canAttach(structuretemplate$structureblockinfo, structuretemplate$structureblockinfo1)) {
                                             BlockPos blockpos3 = structuretemplate$structureblockinfo1.pos();
                                             BlockPos blockpos4 = blockpos2.subtract(blockpos3);
