--- a/net/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
@@ -6,6 +_,7 @@
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
 import com.mojang.serialization.DynamicOps;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import it.unimi.dsi.fastutil.objects.ObjectArrayList;
@@ -40,7 +_,7 @@
     private static final Codec<Either<ResourceLocation, StructureTemplate>> TEMPLATE_CODEC = Codec.of(
         SinglePoolElement::encodeTemplate, ResourceLocation.CODEC.map(Either::left)
     );
-    public static final Codec<SinglePoolElement> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SinglePoolElement> CODEC = RecordCodecBuilder.mapCodec(
         p_210429_ -> p_210429_.group(templateCodec(), processorsCodec(), projectionCodec()).apply(p_210429_, SinglePoolElement::new)
     );
     protected final Either<ResourceLocation, StructureTemplate> template;
@@ -88,7 +_,7 @@
         );
         List<StructureTemplate.StructureBlockInfo> list1 = Lists.newArrayList();
 
-        for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : list) {
+        for (StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : list) {
             CompoundTag compoundtag = structuretemplate$structureblockinfo.nbt();
             if (compoundtag != null) {
                 StructureMode structuremode = StructureMode.valueOf(compoundtag.getString("mode"));
@@ -148,7 +_,7 @@
         if (!structuretemplate.placeInWorld(p_227303_, p_227306_, p_227307_, structureplacesettings, p_227310_, 18)) {
             return false;
         } else {
-            for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : StructureTemplate.processBlockInfos(
+            for (StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : StructureTemplate.processBlockInfos(
                 p_227303_, p_227306_, p_227307_, structureplacesettings, this.getDataMarkers(p_227302_, p_227306_, p_227308_, false)
             )) {
                 this.handleDataMarker(p_227303_, structuretemplate$structureblockinfo, p_227306_, p_227308_, p_227310_, p_227309_);
