--- a/net/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java
@@ -14,7 +_,6 @@
 import net.minecraft.core.Holder;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.resources.RegistryFileCodec;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.RandomSource;
 import net.minecraft.util.StringRepresentable;
 import net.minecraft.world.level.block.Rotation;
@@ -23,21 +_,20 @@
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor;
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager;
 import org.apache.commons.lang3.mutable.MutableObject;
-import org.apache.commons.lang3.tuple.Triple;
 
 public class StructureTemplatePool {
     private static final int SIZE_UNSET = Integer.MIN_VALUE;
     private static final MutableObject<Codec<Holder<StructureTemplatePool>>> CODEC_REFERENCE = new MutableObject<>();
     public static final Codec<StructureTemplatePool> DIRECT_CODEC = RecordCodecBuilder.create(
-        p_255606_ -> p_255606_.group(
-                    ExtraCodecs.lazyInitializedCodec(CODEC_REFERENCE::getValue).fieldOf("fallback").forGetter(StructureTemplatePool::getFallback),
+        p_338101_ -> p_338101_.group(
+                    Codec.lazyInitialized(CODEC_REFERENCE::getValue).fieldOf("fallback").forGetter(StructureTemplatePool::getFallback),
                     Codec.mapPair(StructurePoolElement.CODEC.fieldOf("element"), Codec.intRange(1, 150).fieldOf("weight"))
                         .codec()
                         .listOf()
                         .fieldOf("elements")
                         .forGetter(p_210579_ -> p_210579_.rawTemplates)
                 )
-                .apply(p_255606_, StructureTemplatePool::new)
+                .apply(p_338101_, StructureTemplatePool::new)
     );
     public static final Codec<Holder<StructureTemplatePool>> CODEC = Util.make(
         RegistryFileCodec.create(Registries.TEMPLATE_POOL, DIRECT_CODEC), CODEC_REFERENCE::setValue
@@ -51,10 +_,10 @@
         this.rawTemplates = p_255919_;
         this.templates = new ObjectArrayList<>();
 
-        for(Pair<StructurePoolElement, Integer> pair : p_255919_) {
+        for (Pair<StructurePoolElement, Integer> pair : p_255919_) {
             StructurePoolElement structurepoolelement = pair.getFirst();
 
-            for(int i = 0; i < pair.getSecond(); ++i) {
+            for (int i = 0; i < pair.getSecond(); i++) {
                 this.templates.add(structurepoolelement);
             }
         }
@@ -70,35 +_,16 @@
         this.rawTemplates = Lists.newArrayList();
         this.templates = new ObjectArrayList<>();
 
-        for(Pair<Function<StructureTemplatePool.Projection, ? extends StructurePoolElement>, Integer> pair : p_256083_) {
+        for (Pair<Function<StructureTemplatePool.Projection, ? extends StructurePoolElement>, Integer> pair : p_256083_) {
             StructurePoolElement structurepoolelement = pair.getFirst().apply(p_255642_);
             this.rawTemplates.add(Pair.of(structurepoolelement, pair.getSecond()));
 
-            for(int i = 0; i < pair.getSecond(); ++i) {
+            for (int i = 0; i < pair.getSecond(); i++) {
                 this.templates.add(structurepoolelement);
             }
         }
 
         this.fallback = p_255795_;
-    }
-
-    public StructureTemplatePool(
-        List<Triple<Function<StructureTemplatePool.Projection, ? extends StructurePoolElement>, Integer, StructureTemplatePool.Projection>> p_342205_,
-        Holder<StructureTemplatePool> p_341142_
-    ) {
-        this.rawTemplates = Lists.newArrayList();
-        this.templates = new ObjectArrayList<>();
-
-        for(Triple<Function<StructureTemplatePool.Projection, ? extends StructurePoolElement>, Integer, StructureTemplatePool.Projection> triple : p_342205_) {
-            StructurePoolElement structurepoolelement = triple.getLeft().apply(triple.getRight());
-            this.rawTemplates.add(Pair.of(structurepoolelement, triple.getMiddle()));
-
-            for(int i = 0; i < triple.getMiddle(); ++i) {
-                this.templates.add(structurepoolelement);
-            }
-        }
-
-        this.fallback = p_341142_;
     }
 
     public int getMaxSize(StructureTemplateManager p_227358_) {
