--- a/net/minecraft/world/item/crafting/FireworkRocketRecipe.java
+++ b/net/minecraft/world/item/crafting/FireworkRocketRecipe.java
@@ -24,7 +_,7 @@
         boolean flag = false;
         int i = 0;
 
-        for(int j = 0; j < p_43854_.getContainerSize(); ++j) {
+        for (int j = 0; j < p_43854_.getContainerSize(); j++) {
             ItemStack itemstack = p_43854_.getItem(j);
             if (!itemstack.isEmpty()) {
                 if (PAPER_INGREDIENT.test(itemstack)) {
@@ -50,11 +_,11 @@
         List<FireworkExplosion> list = new ArrayList<>();
         int i = 0;
 
-        for(int j = 0; j < p_43852_.getContainerSize(); ++j) {
+        for (int j = 0; j < p_43852_.getContainerSize(); j++) {
             ItemStack itemstack = p_43852_.getItem(j);
             if (!itemstack.isEmpty()) {
                 if (GUNPOWDER_INGREDIENT.test(itemstack)) {
-                    ++i;
+                    i++;
                 } else if (STAR_INGREDIENT.test(itemstack)) {
                     FireworkExplosion fireworkexplosion = itemstack.get(DataComponents.FIREWORK_EXPLOSION);
                     if (fireworkexplosion != null) {
