--- a/net/minecraft/world/item/crafting/FireworkStarRecipe.java
+++ b/net/minecraft/world/item/crafting/FireworkStarRecipe.java
@@ -56,7 +_,7 @@
         boolean flag3 = false;
         boolean flag4 = false;
 
-        for(int i = 0; i < p_43895_.getContainerSize(); ++i) {
+        for (int i = 0; i < p_43895_.getContainerSize(); i++) {
             ItemStack itemstack = p_43895_.getItem(i);
             if (!itemstack.isEmpty()) {
                 if (SHAPE_INGREDIENT.test(itemstack)) {
@@ -102,7 +_,7 @@
         boolean flag1 = false;
         IntList intlist = new IntArrayList();
 
-        for(int i = 0; i < p_43893_.getContainerSize(); ++i) {
+        for (int i = 0; i < p_43893_.getContainerSize(); i++) {
             ItemStack itemstack = p_43893_.getItem(i);
             if (!itemstack.isEmpty()) {
                 if (SHAPE_INGREDIENT.test(itemstack)) {
