--- a/net/minecraft/world/item/crafting/FireworkStarFadeRecipe.java
+++ b/net/minecraft/world/item/crafting/FireworkStarFadeRecipe.java
@@ -23,7 +_,7 @@
         boolean flag = false;
         boolean flag1 = false;
 
-        for(int i = 0; i < p_43873_.getContainerSize(); ++i) {
+        for (int i = 0; i < p_43873_.getContainerSize(); i++) {
             ItemStack itemstack = p_43873_.getItem(i);
             if (!itemstack.isEmpty()) {
                 if (itemstack.getItem() instanceof DyeItem) {
@@ -49,7 +_,7 @@
         IntList intlist = new IntArrayList();
         ItemStack itemstack = null;
 
-        for(int i = 0; i < p_43871_.getContainerSize(); ++i) {
+        for (int i = 0; i < p_43871_.getContainerSize(); i++) {
             ItemStack itemstack1 = p_43871_.getItem(i);
             Item item = itemstack1.getItem();
             if (item instanceof DyeItem) {
