--- a/net/minecraft/world/item/crafting/MapCloningRecipe.java
+++ b/net/minecraft/world/item/crafting/MapCloningRecipe.java
@@ -15,7 +_,7 @@
         int i = 0;
         ItemStack itemstack = ItemStack.EMPTY;
 
-        for(int j = 0; j < p_43980_.getContainerSize(); ++j) {
+        for (int j = 0; j < p_43980_.getContainerSize(); j++) {
             ItemStack itemstack1 = p_43980_.getItem(j);
             if (!itemstack1.isEmpty()) {
                 if (itemstack1.is(Items.FILLED_MAP)) {
@@ -29,7 +_,7 @@
                         return false;
                     }
 
-                    ++i;
+                    i++;
                 }
             }
         }
@@ -41,7 +_,7 @@
         int i = 0;
         ItemStack itemstack = ItemStack.EMPTY;
 
-        for(int j = 0; j < p_43978_.getContainerSize(); ++j) {
+        for (int j = 0; j < p_43978_.getContainerSize(); j++) {
             ItemStack itemstack1 = p_43978_.getItem(j);
             if (!itemstack1.isEmpty()) {
                 if (itemstack1.is(Items.FILLED_MAP)) {
@@ -55,7 +_,7 @@
                         return ItemStack.EMPTY;
                     }
 
-                    ++i;
+                    i++;
                 }
             }
         }
