--- a/net/minecraft/world/item/crafting/RecipeCache.java
+++ b/net/minecraft/world/item/crafting/RecipeCache.java
@@ -24,7 +_,7 @@
         } else {
             this.validateRecipeManager(p_307315_);
 
-            for(int i = 0; i < this.entries.length; ++i) {
+            for (int i = 0; i < this.entries.length; i++) {
                 RecipeCache.Entry recipecache$entry = this.entries[i];
                 if (recipecache$entry != null && recipecache$entry.matches(p_307684_.getItems())) {
                     this.moveEntryToFront(i);
@@ -61,7 +_,7 @@
     private void insert(List<ItemStack> p_307307_, @Nullable RecipeHolder<CraftingRecipe> p_336146_) {
         NonNullList<ItemStack> nonnulllist = NonNullList.withSize(p_307307_.size(), ItemStack.EMPTY);
 
-        for(int i = 0; i < p_307307_.size(); ++i) {
+        for (int i = 0; i < p_307307_.size(); i++) {
             nonnulllist.set(i, p_307307_.get(i).copyWithCount(1));
         }
 
@@ -74,7 +_,7 @@
             if (this.key.size() != p_307411_.size()) {
                 return false;
             } else {
-                for(int i = 0; i < this.key.size(); ++i) {
+                for (int i = 0; i < this.key.size(); i++) {
                     if (!ItemStack.isSameItemSameComponents(this.key.get(i), p_307411_.get(i))) {
                         return false;
                     }
