--- a/net/minecraft/world/item/crafting/RecipeSerializer.java
+++ b/net/minecraft/world/item/crafting/RecipeSerializer.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.item.crafting;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.network.RegistryFriendlyByteBuf;
@@ -45,17 +_,13 @@
         "campfire_cooking", new SimpleCookingSerializer<>(CampfireCookingRecipe::new, 100)
     );
     RecipeSerializer<StonecutterRecipe> STONECUTTER = register("stonecutting", new SingleItemRecipe.Serializer<>(StonecutterRecipe::new));
-    RecipeSerializer<PoisonousPotatoCutterRecipe> POISONOUS_POTATO_CUTTER_RECIPE = register(
-        "poisonous_potato_cutting", new SingleItemRecipe.Serializer<>(PoisonousPotatoCutterRecipe::new)
-    );
     RecipeSerializer<SmithingTransformRecipe> SMITHING_TRANSFORM = register("smithing_transform", new SmithingTransformRecipe.Serializer());
     RecipeSerializer<SmithingTrimRecipe> SMITHING_TRIM = register("smithing_trim", new SmithingTrimRecipe.Serializer());
     RecipeSerializer<DecoratedPotRecipe> DECORATED_POT_RECIPE = register(
         "crafting_decorated_pot", new SimpleCraftingRecipeSerializer<>(DecoratedPotRecipe::new)
     );
-    RecipeSerializer<PotatoRefinementRecipe> POTATO_REFINEMENT_RECIPE = register("potato_refinement", new PotatoRefinementRecipe.Serializer());
 
-    Codec<T> codec();
+    MapCodec<T> codec();
 
     StreamCodec<RegistryFriendlyByteBuf, T> streamCodec();
 
