--- a/net/minecraft/world/item/crafting/ShapelessRecipe.java
+++ b/net/minecraft/world/item/crafting/ShapelessRecipe.java
@@ -2,6 +_,7 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -9,7 +_,6 @@
 import net.minecraft.core.NonNullList;
 import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.codec.StreamCodec;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.player.StackedContents;
 import net.minecraft.world.inventory.CraftingContainer;
 import net.minecraft.world.item.ItemStack;
@@ -20,12 +_,14 @@
     final CraftingBookCategory category;
     final ItemStack result;
     final NonNullList<Ingredient> ingredients;
+    private final boolean isSimple;
 
     public ShapelessRecipe(String p_249640_, CraftingBookCategory p_249390_, ItemStack p_252071_, NonNullList<Ingredient> p_250689_) {
         this.group = p_249640_;
         this.category = p_249390_;
         this.result = p_252071_;
         this.ingredients = p_250689_;
+        this.isSimple = p_250689_.stream().allMatch(Ingredient::isSimple);
     }
 
     @Override
@@ -55,17 +_,20 @@
 
     public boolean matches(CraftingContainer p_44262_, Level p_44263_) {
         StackedContents stackedcontents = new StackedContents();
+        java.util.List<ItemStack> inputs = new java.util.ArrayList<>();
         int i = 0;
 
-        for(int j = 0; j < p_44262_.getContainerSize(); ++j) {
+        for (int j = 0; j < p_44262_.getContainerSize(); j++) {
             ItemStack itemstack = p_44262_.getItem(j);
             if (!itemstack.isEmpty()) {
-                ++i;
+                i++;
+                if (isSimple)
                 stackedcontents.accountStack(itemstack, 1);
+                else inputs.add(itemstack);
             }
         }
 
-        return i == this.ingredients.size() && stackedcontents.canCraft(this, null);
+        return i == this.ingredients.size() && (isSimple ? stackedcontents.canCraft(this, null) : net.neoforged.neoforge.common.util.RecipeMatcher.findMatches(inputs,  this.ingredients) != null);
     }
 
     public ItemStack assemble(CraftingContainer p_44260_, HolderLookup.Provider p_335725_) {
@@ -78,9 +_,9 @@
     }
 
     public static class Serializer implements RecipeSerializer<ShapelessRecipe> {
-        private static final Codec<ShapelessRecipe> CODEC = RecordCodecBuilder.create(
-            p_332625_ -> p_332625_.group(
-                        ExtraCodecs.strictOptionalField(Codec.STRING, "group", "").forGetter(p_301127_ -> p_301127_.group),
+        private static final MapCodec<ShapelessRecipe> CODEC = RecordCodecBuilder.mapCodec(
+            p_337958_ -> p_337958_.group(
+                        Codec.STRING.optionalFieldOf("group", "").forGetter(p_301127_ -> p_301127_.group),
                         CraftingBookCategory.CODEC.fieldOf("category").orElse(CraftingBookCategory.MISC).forGetter(p_301133_ -> p_301133_.category),
                         ItemStack.CODEC.fieldOf("result").forGetter(p_301142_ -> p_301142_.result),
                         Ingredient.CODEC_NONEMPTY
@@ -88,14 +_,12 @@
                             .fieldOf("ingredients")
                             .flatXmap(
                                 p_301021_ -> {
-                                    Ingredient[] aingredient = p_301021_.stream()
-                                        .filter(p_300883_ -> !p_300883_.isEmpty())
-                                        .toArray(p_300865_ -> new Ingredient[p_300865_]);
+                                    Ingredient[] aingredient = p_301021_.toArray(Ingredient[]::new); // Neo skip the empty check and immediately create the array.
                                     if (aingredient.length == 0) {
                                         return DataResult.error(() -> "No ingredients for shapeless recipe");
                                     } else {
-                                        return aingredient.length > 9
-                                            ? DataResult.error(() -> "Too many ingredients for shapeless recipe")
+                                        return aingredient.length > ShapedRecipePattern.maxHeight * ShapedRecipePattern.maxWidth
+                                            ? DataResult.error(() -> "Too many ingredients for shapeless recipe. The maximum is: %s".formatted(ShapedRecipePattern.maxHeight * ShapedRecipePattern.maxWidth))
                                             : DataResult.success(NonNullList.of(Ingredient.EMPTY, aingredient));
                                     }
                                 },
@@ -103,14 +_,14 @@
                             )
                             .forGetter(p_300975_ -> p_300975_.ingredients)
                     )
-                    .apply(p_332625_, ShapelessRecipe::new)
+                    .apply(p_337958_, ShapelessRecipe::new)
         );
         public static final StreamCodec<RegistryFriendlyByteBuf, ShapelessRecipe> STREAM_CODEC = StreamCodec.of(
             ShapelessRecipe.Serializer::toNetwork, ShapelessRecipe.Serializer::fromNetwork
         );
 
         @Override
-        public Codec<ShapelessRecipe> codec() {
+        public MapCodec<ShapelessRecipe> codec() {
             return CODEC;
         }
 
@@ -134,7 +_,7 @@
             p_320371_.writeEnum(p_320323_.category);
             p_320371_.writeVarInt(p_320323_.ingredients.size());
 
-            for(Ingredient ingredient : p_320323_.ingredients) {
+            for (Ingredient ingredient : p_320323_.ingredients) {
                 Ingredient.CONTENTS_STREAM_CODEC.encode(p_320371_, ingredient);
             }
 
