--- a/net/minecraft/world/item/crafting/ShieldDecorationRecipe.java
+++ b/net/minecraft/world/item/crafting/ShieldDecorationRecipe.java
@@ -18,7 +_,7 @@
         ItemStack itemstack = ItemStack.EMPTY;
         ItemStack itemstack1 = ItemStack.EMPTY;
 
-        for(int i = 0; i < p_44308_.getContainerSize(); ++i) {
+        for (int i = 0; i < p_44308_.getContainerSize(); i++) {
             ItemStack itemstack2 = p_44308_.getItem(i);
             if (!itemstack2.isEmpty()) {
                 if (itemstack2.getItem() instanceof BannerItem) {
@@ -53,7 +_,7 @@
         ItemStack itemstack = ItemStack.EMPTY;
         ItemStack itemstack1 = ItemStack.EMPTY;
 
-        for(int i = 0; i < p_44306_.getContainerSize(); ++i) {
+        for (int i = 0; i < p_44306_.getContainerSize(); i++) {
             ItemStack itemstack2 = p_44306_.getItem(i);
             if (!itemstack2.isEmpty()) {
                 if (itemstack2.getItem() instanceof BannerItem) {
