--- a/net/minecraft/world/item/crafting/SingleItemRecipe.java
+++ b/net/minecraft/world/item/crafting/SingleItemRecipe.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.item.crafting;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.HolderLookup;
@@ -8,7 +_,6 @@
 import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.Container;
 import net.minecraft.world.item.ItemStack;
 
@@ -70,18 +_,18 @@
 
     public static class Serializer<T extends SingleItemRecipe> implements RecipeSerializer<T> {
         final SingleItemRecipe.Factory<T> factory;
-        private final Codec<T> codec;
+        private final MapCodec<T> codec;
         private final StreamCodec<RegistryFriendlyByteBuf, T> streamCodec;
 
         protected Serializer(SingleItemRecipe.Factory<T> p_312589_) {
             this.factory = p_312589_;
-            this.codec = RecordCodecBuilder.create(
-                p_332627_ -> p_332627_.group(
-                            ExtraCodecs.strictOptionalField(Codec.STRING, "group", "").forGetter(p_300947_ -> p_300947_.group),
+            this.codec = RecordCodecBuilder.mapCodec(
+                p_337960_ -> p_337960_.group(
+                            Codec.STRING.optionalFieldOf("group", "").forGetter(p_300947_ -> p_300947_.group),
                             Ingredient.CODEC_NONEMPTY.fieldOf("ingredient").forGetter(p_301068_ -> p_301068_.ingredient),
                             ItemStack.CODEC.fieldOf("result").forGetter(p_302316_ -> p_302316_.result)
                         )
-                        .apply(p_332627_, p_312589_::create)
+                        .apply(p_337960_, p_312589_::create)
             );
             this.streamCodec = StreamCodec.composite(
                 ByteBufCodecs.STRING_UTF8,
@@ -95,7 +_,7 @@
         }
 
         @Override
-        public Codec<T> codec() {
+        public MapCodec<T> codec() {
             return this.codec;
         }
 
