--- a/net/minecraft/client/particle/BreakingItemParticle.java
+++ b/net/minecraft/client/particle/BreakingItemParticle.java
@@ -33,7 +_,8 @@
 
     protected BreakingItemParticle(ClientLevel p_105665_, double p_105666_, double p_105667_, double p_105668_, ItemStack p_105669_) {
         super(p_105665_, p_105666_, p_105667_, p_105668_, 0.0, 0.0, 0.0);
-        this.setSprite(Minecraft.getInstance().getItemRenderer().getModel(p_105669_, p_105665_, null, 0).getParticleIcon());
+        var model = Minecraft.getInstance().getItemRenderer().getModel(p_105669_, p_105665_, null, 0);
+        this.setSprite(model.getOverrides().resolve(model, p_105669_, p_105665_, null, 0).getParticleIcon(net.neoforged.neoforge.client.model.data.ModelData.EMPTY));
         this.gravity = 1.0F;
         this.quadSize /= 2.0F;
         this.uo = this.random.nextFloat() * 3.0F;
@@ -58,6 +_,22 @@
     @Override
     protected float getV1() {
         return this.sprite.getV((this.vo + 1.0F) / 4.0F);
+    }
+
+    @OnlyIn(Dist.CLIENT)
+    public static class CobwebProvider implements ParticleProvider<SimpleParticleType> {
+        public Particle createParticle(
+            SimpleParticleType p_338579_,
+            ClientLevel p_338749_,
+            double p_338877_,
+            double p_338362_,
+            double p_338343_,
+            double p_338303_,
+            double p_338217_,
+            double p_338683_
+        ) {
+            return new BreakingItemParticle(p_338749_, p_338877_, p_338362_, p_338343_, new ItemStack(Items.COBWEB));
+        }
     }
 
     @OnlyIn(Dist.CLIENT)
