--- a/net/minecraft/client/particle/BubbleParticle.java
+++ b/net/minecraft/client/particle/BubbleParticle.java
@@ -12,7 +_,7 @@
     BubbleParticle(ClientLevel p_105773_, double p_105774_, double p_105775_, double p_105776_, double p_105777_, double p_105778_, double p_105779_) {
         super(p_105773_, p_105774_, p_105775_, p_105776_);
         this.setSize(0.02F, 0.02F);
-        this.quadSize *= this.random.nextFloat() * 0.6F + 0.2F;
+        this.quadSize = this.quadSize * (this.random.nextFloat() * 0.6F + 0.2F);
         this.xd = p_105777_ * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
         this.yd = p_105778_ * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
         this.zd = p_105779_ * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
