--- a/net/minecraft/client/particle/BubbleColumnUpParticle.java
+++ b/net/minecraft/client/particle/BubbleColumnUpParticle.java
@@ -14,7 +_,7 @@
         this.gravity = -0.125F;
         this.friction = 0.85F;
         this.setSize(0.02F, 0.02F);
-        this.quadSize *= this.random.nextFloat() * 0.6F + 0.2F;
+        this.quadSize = this.quadSize * (this.random.nextFloat() * 0.6F + 0.2F);
         this.xd = p_105737_ * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
         this.yd = p_105738_ * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
         this.zd = p_105739_ * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
