--- a/net/minecraft/client/particle/BubblePopParticle.java
+++ b/net/minecraft/client/particle/BubblePopParticle.java
@@ -30,7 +_,7 @@
         if (this.age++ >= this.lifetime) {
             this.remove();
         } else {
-            this.yd -= (double)this.gravity;
+            this.yd = this.yd - (double)this.gravity;
             this.move(this.xd, this.yd, this.zd);
             this.setSpriteFromAge(this.sprites);
         }
