--- a/net/minecraft/client/particle/CampfireSmokeParticle.java
+++ b/net/minecraft/client/particle/CampfireSmokeParticle.java
@@ -31,9 +_,9 @@
         this.yo = this.y;
         this.zo = this.z;
         if (this.age++ < this.lifetime && !(this.alpha <= 0.0F)) {
-            this.xd += (double)(this.random.nextFloat() / 5000.0F * (float)(this.random.nextBoolean() ? 1 : -1));
-            this.zd += (double)(this.random.nextFloat() / 5000.0F * (float)(this.random.nextBoolean() ? 1 : -1));
-            this.yd -= (double)this.gravity;
+            this.xd = this.xd + (double)(this.random.nextFloat() / 5000.0F * (float)(this.random.nextBoolean() ? 1 : -1));
+            this.zd = this.zd + (double)(this.random.nextFloat() / 5000.0F * (float)(this.random.nextBoolean() ? 1 : -1));
+            this.yd = this.yd - (double)this.gravity;
             this.move(this.xd, this.yd, this.zd);
             if (this.age >= this.lifetime - 60 && this.alpha > 0.01F) {
                 this.alpha -= 0.015F;
