--- a/net/minecraft/client/particle/GustParticle.java
+++ b/net/minecraft/client/particle/GustParticle.java
@@ -58,4 +_,28 @@
             return new GustParticle(p_312008_, p_312498_, p_312520_, p_311822_, this.sprites);
         }
     }
+
+    @OnlyIn(Dist.CLIENT)
+    public static class SmallProvider implements ParticleProvider<SimpleParticleType> {
+        private final SpriteSet sprites;
+
+        public SmallProvider(SpriteSet p_338457_) {
+            this.sprites = p_338457_;
+        }
+
+        public Particle createParticle(
+            SimpleParticleType p_338464_,
+            ClientLevel p_338628_,
+            double p_338495_,
+            double p_338874_,
+            double p_338214_,
+            double p_338609_,
+            double p_338492_,
+            double p_338605_
+        ) {
+            Particle particle = new GustParticle(p_338628_, p_338495_, p_338874_, p_338214_, this.sprites);
+            particle.scale(0.15F);
+            return particle;
+        }
+    }
 }
