--- a/net/minecraft/client/particle/ParticleEngine.java
+++ b/net/minecraft/client/particle/ParticleEngine.java
@@ -79,11 +_,11 @@
         ParticleRenderType.CUSTOM
     );
     protected ClientLevel level;
-    private final Map<ParticleRenderType, Queue<Particle>> particles = Maps.newIdentityHashMap();
+    private final Map<ParticleRenderType, Queue<Particle>> particles = Maps.newTreeMap(net.neoforged.neoforge.client.ClientHooks.makeParticleRenderTypeComparator(RENDER_ORDER));
     private final Queue<TrackingEmitter> trackingEmitters = Queues.newArrayDeque();
     private final TextureManager textureManager;
     private final RandomSource random = RandomSource.create();
-    private final Int2ObjectMap<ParticleProvider<?>> providers = new Int2ObjectOpenHashMap<>();
+    private final Map<ResourceLocation, ParticleProvider<?>> providers = new java.util.HashMap<>();
     private final Queue<Particle> particlesToAdd = Queues.newArrayDeque();
     private final Map<ResourceLocation, ParticleEngine.MutableSpriteSet> spriteSets = Maps.newHashMap();
     private final TextureAtlas textureAtlas;
@@ -131,11 +_,13 @@
         this.register(ParticleTypes.SONIC_BOOM, SonicBoomParticle.Provider::new);
         this.register(ParticleTypes.FALLING_DUST, FallingDustParticle.Provider::new);
         this.register(ParticleTypes.GUST, GustParticle.Provider::new);
+        this.register(ParticleTypes.SMALL_GUST, GustParticle.SmallProvider::new);
         this.register(ParticleTypes.GUST_EMITTER_LARGE, new GustSeedParticle.Provider(3.0, 7, 0));
         this.register(ParticleTypes.GUST_EMITTER_SMALL, new GustSeedParticle.Provider(1.0, 3, 2));
         this.register(ParticleTypes.FIREWORK, FireworkParticles.SparkProvider::new);
         this.register(ParticleTypes.FISHING, WakeParticle.Provider::new);
         this.register(ParticleTypes.FLAME, FlameParticle.Provider::new);
+        this.register(ParticleTypes.INFESTED, SpellParticle.Provider::new);
         this.register(ParticleTypes.SCULK_SOUL, SoulParticle.EmissiveProvider::new);
         this.register(ParticleTypes.SCULK_CHARGE, SculkChargeParticle.Provider::new);
         this.register(ParticleTypes.SCULK_CHARGE_POP, SculkChargePopParticle.Provider::new);
@@ -147,6 +_,7 @@
         this.register(ParticleTypes.INSTANT_EFFECT, SpellParticle.InstantProvider::new);
         this.register(ParticleTypes.ITEM, new BreakingItemParticle.Provider());
         this.register(ParticleTypes.ITEM_SLIME, new BreakingItemParticle.SlimeProvider());
+        this.register(ParticleTypes.ITEM_COBWEB, new BreakingItemParticle.CobwebProvider());
         this.register(ParticleTypes.ITEM_SNOWBALL, new BreakingItemParticle.SnowballProvider());
         this.register(ParticleTypes.LARGE_SMOKE, LargeSmokeParticle.Provider::new);
         this.register(ParticleTypes.LAVA, LavaParticle.Provider::new);
@@ -154,7 +_,7 @@
         this.register(ParticleTypes.NAUTILUS, FlyTowardsPositionParticle.NautilusProvider::new);
         this.register(ParticleTypes.NOTE, NoteParticle.Provider::new);
         this.register(ParticleTypes.POOF, ExplodeParticle.Provider::new);
-        this.register(ParticleTypes.PORTAL, p_339499_ -> new PortalParticle.Provider(p_339499_, 0.9F, 0.3F, 1.0F));
+        this.register(ParticleTypes.PORTAL, PortalParticle.Provider::new);
         this.register(ParticleTypes.RAIN, WaterDropParticle.Provider::new);
         this.register(ParticleTypes.SMOKE, SmokeParticle.Provider::new);
         this.register(ParticleTypes.WHITE_SMOKE, WhiteSmokeParticle.Provider::new);
@@ -171,7 +_,6 @@
         this.register(ParticleTypes.FALLING_HONEY, DripParticle::createHoneyFallParticle);
         this.register(ParticleTypes.LANDING_HONEY, DripParticle::createHoneyLandParticle);
         this.register(ParticleTypes.FALLING_NECTAR, DripParticle::createNectarFallParticle);
-        this.register(ParticleTypes.FALLING_POISON, DripParticle::createPoisonFallParticle);
         this.register(ParticleTypes.FALLING_SPORE_BLOSSOM, DripParticle::createSporeBlossomFallParticle);
         this.register(ParticleTypes.SPORE_BLOSSOM_AIR, SuspendedParticle.SporeBlossomAirProvider::new);
         this.register(ParticleTypes.ASH, AshParticle.Provider::new);
@@ -180,9 +_,7 @@
         this.register(ParticleTypes.DRIPPING_OBSIDIAN_TEAR, DripParticle::createObsidianTearHangParticle);
         this.register(ParticleTypes.FALLING_OBSIDIAN_TEAR, DripParticle::createObsidianTearFallParticle);
         this.register(ParticleTypes.LANDING_OBSIDIAN_TEAR, DripParticle::createObsidianTearLandParticle);
-        this.register(ParticleTypes.REVERSE_PORTAL, p_339501_ -> new ReversePortalParticle.ReversePortalProvider(p_339501_, 0.9F, 0.3F, 1.0F));
-        this.register(ParticleTypes.REVERSE_POTATO_LIGHTNING, p_339500_ -> new ReversePortalParticle.ReversePortalProvider(p_339500_, 0.6F, 1.0F, 0.2F));
-        this.register(ParticleTypes.POTATO_LIGHTNING, p_339502_ -> new PortalParticle.Provider(p_339502_, 0.6F, 1.0F, 0.2F));
+        this.register(ParticleTypes.REVERSE_PORTAL, ReversePortalParticle.ReversePortalProvider::new);
         this.register(ParticleTypes.WHITE_ASH, WhiteAshParticle.Provider::new);
         this.register(ParticleTypes.SMALL_FLAME, FlameParticle.SmallFlameProvider::new);
         this.register(ParticleTypes.DRIPPING_DRIPSTONE_WATER, DripParticle::createDripstoneWaterHangParticle);
@@ -203,17 +_,25 @@
         this.register(ParticleTypes.ELECTRIC_SPARK, GlowParticle.ElectricSparkProvider::new);
         this.register(ParticleTypes.SCRAPE, GlowParticle.ScrapeProvider::new);
         this.register(ParticleTypes.SHRIEK, ShriekParticle.Provider::new);
-        this.register(ParticleTypes.FOOTSTEP, FootprintParticle.Provider::new);
         this.register(ParticleTypes.EGG_CRACK, SuspendedTownParticle.EggCrackProvider::new);
         this.register(ParticleTypes.DUST_PLUME, DustPlumeParticle.Provider::new);
-        this.register(ParticleTypes.TRIAL_SPAWNER_DETECTION, TrialSpawnerDetectionParticle.Provider::new);
+        this.register(ParticleTypes.TRIAL_SPAWNER_DETECTED_PLAYER, TrialSpawnerDetectionParticle.Provider::new);
+        this.register(ParticleTypes.TRIAL_SPAWNER_DETECTED_PLAYER_OMINOUS, TrialSpawnerDetectionParticle.Provider::new);
         this.register(ParticleTypes.VAULT_CONNECTION, FlyTowardsPositionParticle.VaultConnectionProvider::new);
+        this.register(ParticleTypes.DUST_PILLAR, new TerrainParticle.DustPillarProvider());
+        this.register(ParticleTypes.RAID_OMEN, SpellParticle.Provider::new);
+        this.register(ParticleTypes.TRIAL_OMEN, SpellParticle.Provider::new);
+        this.register(ParticleTypes.OMINOUS_SPAWNING, FlyStraightTowardsParticle.OminousSpawnProvider::new);
     }
 
+    /** @deprecated Register via {@link net.neoforged.neoforge.client.event.RegisterParticleProvidersEvent} */
+    @Deprecated
     public <T extends ParticleOptions> void register(ParticleType<T> p_107382_, ParticleProvider<T> p_107383_) {
-        this.providers.put(BuiltInRegistries.PARTICLE_TYPE.getId(p_107382_), p_107383_);
+        this.providers.put(BuiltInRegistries.PARTICLE_TYPE.getKey(p_107382_), p_107383_);
     }
 
+    /** @deprecated Register via {@link net.neoforged.neoforge.client.event.RegisterParticleProvidersEvent} */
+    @Deprecated
     public <T extends ParticleOptions> void register(ParticleType<T> p_273423_, ParticleProvider.Sprite<T> p_273134_) {
         this.register(
             p_273423_,
@@ -230,10 +_,12 @@
         );
     }
 
+    /** @deprecated Register via {@link net.neoforged.neoforge.client.event.RegisterParticleProvidersEvent} */
+    @Deprecated
     public <T extends ParticleOptions> void register(ParticleType<T> p_107379_, ParticleEngine.SpriteParticleRegistration<T> p_107380_) {
         ParticleEngine.MutableSpriteSet particleengine$mutablespriteset = new ParticleEngine.MutableSpriteSet();
         this.spriteSets.put(BuiltInRegistries.PARTICLE_TYPE.getKey(p_107379_), particleengine$mutablespriteset);
-        this.providers.put(BuiltInRegistries.PARTICLE_TYPE.getId(p_107379_), p_107380_.create(particleengine$mutablespriteset));
+        this.providers.put(BuiltInRegistries.PARTICLE_TYPE.getKey(p_107379_), p_107380_.create(particleengine$mutablespriteset));
     }
 
     @Override
@@ -285,7 +_,7 @@
                 if (!optional.isEmpty()) {
                     List<TextureAtlasSprite> list = new ArrayList<>();
 
-                    for(ResourceLocation resourcelocation : optional.get()) {
+                    for (ResourceLocation resourcelocation : optional.get()) {
                         TextureAtlasSprite textureatlassprite1 = spriteloader$preparations.regions().get(resourcelocation);
                         if (textureatlassprite1 == null) {
                             set.add(resourcelocation);
@@ -359,7 +_,7 @@
     private <T extends ParticleOptions> Particle makeParticle(
         T p_107396_, double p_107397_, double p_107398_, double p_107399_, double p_107400_, double p_107401_, double p_107402_
     ) {
-        ParticleProvider<T> particleprovider = (ParticleProvider)this.providers.get(BuiltInRegistries.PARTICLE_TYPE.getId(p_107396_.getType()));
+        ParticleProvider<T> particleprovider = (ParticleProvider<T>) this.providers.get(BuiltInRegistries.PARTICLE_TYPE.getKey(p_107396_.getType()));
         return particleprovider == null
             ? null
             : particleprovider.createParticle(p_107396_, this.level, p_107397_, p_107398_, p_107399_, p_107400_, p_107401_, p_107402_);
@@ -378,15 +_,15 @@
     }
 
     public void tick() {
-        this.particles.forEach((p_339497_, p_339498_) -> {
-            this.level.getProfiler().push(p_339497_.toString());
-            this.tickParticleList(p_339498_);
+        this.particles.forEach((p_339289_, p_339290_) -> {
+            this.level.getProfiler().push(p_339289_.toString());
+            this.tickParticleList(p_339290_);
             this.level.getProfiler().pop();
         });
         if (!this.trackingEmitters.isEmpty()) {
             List<TrackingEmitter> list = Lists.newArrayList();
 
-            for(TrackingEmitter trackingemitter : this.trackingEmitters) {
+            for (TrackingEmitter trackingemitter : this.trackingEmitters) {
                 trackingemitter.tick();
                 if (!trackingemitter.isAlive()) {
                     list.add(trackingemitter);
@@ -398,7 +_,7 @@
 
         Particle particle;
         if (!this.particlesToAdd.isEmpty()) {
-            while((particle = this.particlesToAdd.poll()) != null) {
+            while ((particle = this.particlesToAdd.poll()) != null) {
                 this.particles.computeIfAbsent(particle.getRenderType(), p_107347_ -> EvictingQueue.create(16384)).add(particle);
             }
         }
@@ -408,7 +_,7 @@
         if (!p_107385_.isEmpty()) {
             Iterator<Particle> iterator = p_107385_.iterator();
 
-            while(iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 Particle particle = iterator.next();
                 this.tickParticle(particle);
                 if (!particle.isAlive()) {
@@ -435,11 +_,21 @@
         }
     }
 
+    /** @deprecated Neo: use {@link #render(LightTexture, Camera, float, net.minecraft.client.renderer.culling.Frustum)}  with Frustum as additional parameter */
+    @Deprecated
     public void render(LightTexture p_107339_, Camera p_107340_, float p_107341_) {
+        render(p_107339_, p_107340_, p_107341_, null);
+    }
+
+    public void render(LightTexture p_107339_, Camera p_107340_, float p_107341_, @Nullable net.minecraft.client.renderer.culling.Frustum frustum) {
         p_107339_.turnOnLightLayer();
         RenderSystem.enableDepthTest();
+        //TODO porting: is this even needed with the particle render order fix???
+        RenderSystem.activeTexture(org.lwjgl.opengl.GL13.GL_TEXTURE2);
+        RenderSystem.activeTexture(org.lwjgl.opengl.GL13.GL_TEXTURE0);
 
-        for(ParticleRenderType particlerendertype : RENDER_ORDER) {
+        for(ParticleRenderType particlerendertype : this.particles.keySet()) { // Forge: allow custom IParticleRenderType's
+            if (particlerendertype == ParticleRenderType.NO_RENDER) continue;
             Iterable<Particle> iterable = this.particles.get(particlerendertype);
             if (iterable != null) {
                 RenderSystem.setShader(GameRenderer::getParticleShader);
@@ -447,7 +_,8 @@
                 BufferBuilder bufferbuilder = tesselator.getBuilder();
                 particlerendertype.begin(bufferbuilder, this.textureManager);
 
-                for(Particle particle : iterable) {
+                for (Particle particle : iterable) {
+                    if (frustum != null && particle.shouldCull() && !frustum.isVisible(particle.getBoundingBox().inflate(1.0))) continue;
                     try {
                         particle.render(bufferbuilder, p_107340_, p_107341_);
                     } catch (Throwable throwable) {
@@ -475,7 +_,7 @@
     }
 
     public void destroy(BlockPos p_107356_, BlockState p_107357_) {
-        if (!p_107357_.isAir() && p_107357_.shouldSpawnTerrainParticles()) {
+        if (!p_107357_.isAir() && !net.neoforged.neoforge.client.extensions.common.IClientBlockExtensions.of(p_107357_).addDestroyEffects(p_107357_, this.level, p_107356_, this)) {
             VoxelShape voxelshape = p_107357_.getShape(this.level, p_107356_);
             double d0 = 0.25;
             voxelshape.forAllBoxes(
@@ -487,9 +_,9 @@
                     int j = Math.max(2, Mth.ceil(d2 / 0.25));
                     int k = Math.max(2, Mth.ceil(d3 / 0.25));
     
-                    for(int l = 0; l < i; ++l) {
-                        for(int i1 = 0; i1 < j; ++i1) {
-                            for(int j1 = 0; j1 < k; ++j1) {
+                    for (int l = 0; l < i; l++) {
+                        for (int i1 = 0; i1 < j; i1++) {
+                            for (int j1 = 0; j1 < k; j1++) {
                                 double d4 = ((double)l + 0.5) / (double)i;
                                 double d5 = ((double)i1 + 0.5) / (double)j;
                                 double d6 = ((double)j1 + 0.5) / (double)k;
@@ -507,7 +_,7 @@
                                         d6 - 0.5,
                                         p_107357_,
                                         p_107356_
-                                    )
+                                    ).updateSprite(p_107357_, p_107356_)
                                 );
                             }
                         }
@@ -552,12 +_,18 @@
                 d0 = (double)i + aabb.maxX + 0.1F;
             }
 
-            this.add(new TerrainParticle(this.level, d0, d1, d2, 0.0, 0.0, 0.0, blockstate, p_107368_).setPower(0.2F).scale(0.6F));
+            this.add((new TerrainParticle(this.level, d0, d1, d2, 0.0D, 0.0D, 0.0D, blockstate, p_107368_).updateSprite(blockstate, p_107368_)).setPower(0.2F).scale(0.6F));
         }
     }
 
     public String countParticles() {
         return String.valueOf(this.particles.values().stream().mapToInt(Collection::size).sum());
+    }
+
+    public void addBlockHitEffects(BlockPos pos, net.minecraft.world.phys.BlockHitResult target) {
+        BlockState state = level.getBlockState(pos);
+        if (!net.neoforged.neoforge.client.extensions.common.IClientBlockExtensions.of(state).addHitEffects(state, level, target, this))
+            crack(pos, target.getDirection());
     }
 
     private boolean hasSpaceInParticleLimit(ParticleGroup p_172280_) {
