--- a/net/minecraft/client/particle/PlayerCloudParticle.java
+++ b/net/minecraft/client/particle/PlayerCloudParticle.java
@@ -54,8 +_,8 @@
             if (player != null) {
                 double d0 = player.getY();
                 if (this.y > d0) {
-                    this.y += (d0 - this.y) * 0.2;
-                    this.yd += (player.getDeltaMovement().y - this.yd) * 0.2;
+                    this.y = this.y + (d0 - this.y) * 0.2;
+                    this.yd = this.yd + (player.getDeltaMovement().y - this.yd) * 0.2;
                     this.setPos(this.x, this.y, this.z);
                 }
             }
