--- a/net/minecraft/client/particle/RisingParticle.java
+++ b/net/minecraft/client/particle/RisingParticle.java
@@ -12,9 +_,9 @@
         this.xd = this.xd * 0.01F + p_107635_;
         this.yd = this.yd * 0.01F + p_107636_;
         this.zd = this.zd * 0.01F + p_107637_;
-        this.x += (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.05F);
-        this.y += (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.05F);
-        this.z += (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.05F);
+        this.x = this.x + (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.05F);
+        this.y = this.y + (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.05F);
+        this.z = this.z + (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.05F);
         this.lifetime = (int)(8.0 / (Math.random() * 0.8 + 0.2)) + 4;
     }
 }
