--- a/net/minecraft/client/particle/SingleQuadParticle.java
+++ b/net/minecraft/client/particle/SingleQuadParticle.java
@@ -47,7 +_,7 @@
         };
         float f3 = this.getQuadSize(p_107680_);
 
-        for(int i = 0; i < 4; ++i) {
+        for (int i = 0; i < 4; i++) {
             Vector3f vector3f = avector3f[i];
             vector3f.rotate(this.rotation);
             vector3f.mul(f3);
