--- a/net/minecraft/client/particle/SuspendedTownParticle.java
+++ b/net/minecraft/client/particle/SuspendedTownParticle.java
@@ -14,7 +_,7 @@
         this.gCol = f;
         this.bCol = f;
         this.setSize(0.02F, 0.02F);
-        this.quadSize *= this.random.nextFloat() * 0.6F + 0.5F;
+        this.quadSize = this.quadSize * (this.random.nextFloat() * 0.6F + 0.5F);
         this.xd *= 0.02F;
         this.yd *= 0.02F;
         this.zd *= 0.02F;
