--- a/net/minecraft/client/particle/SuspendedParticle.java
+++ b/net/minecraft/client/particle/SuspendedParticle.java
@@ -15,7 +_,7 @@
         super(p_172403_, p_172405_, p_172406_ - 0.125, p_172407_);
         this.setSize(0.01F, 0.01F);
         this.pickSprite(p_172404_);
-        this.quadSize *= this.random.nextFloat() * 0.6F + 0.2F;
+        this.quadSize = this.quadSize * (this.random.nextFloat() * 0.6F + 0.2F);
         this.lifetime = (int)(16.0 / (Math.random() * 0.8 + 0.2));
         this.hasPhysics = false;
         this.friction = 1.0F;
@@ -28,7 +_,7 @@
         super(p_172409_, p_172411_, p_172412_ - 0.125, p_172413_, p_172414_, p_172415_, p_172416_);
         this.setSize(0.01F, 0.01F);
         this.pickSprite(p_172410_);
-        this.quadSize *= this.random.nextFloat() * 0.6F + 0.6F;
+        this.quadSize = this.quadSize * (this.random.nextFloat() * 0.6F + 0.6F);
         this.lifetime = (int)(16.0 / (Math.random() * 0.8 + 0.2));
         this.hasPhysics = false;
         this.friction = 1.0F;
