--- a/net/minecraft/client/particle/WakeParticle.java
+++ b/net/minecraft/client/particle/WakeParticle.java
@@ -40,7 +_,7 @@
         if (this.lifetime-- <= 0) {
             this.remove();
         } else {
-            this.yd -= (double)this.gravity;
+            this.yd = this.yd - (double)this.gravity;
             this.move(this.xd, this.yd, this.zd);
             this.xd *= 0.98F;
             this.yd *= 0.98F;
