--- a/net/minecraft/client/particle/WaterDropParticle.java
+++ b/net/minecraft/client/particle/WaterDropParticle.java
@@ -32,7 +_,7 @@
         if (this.lifetime-- <= 0) {
             this.remove();
         } else {
-            this.yd -= (double)this.gravity;
+            this.yd = this.yd - (double)this.gravity;
             this.move(this.xd, this.yd, this.zd);
             this.xd *= 0.98F;
             this.yd *= 0.98F;
@@ -80,10 +_,6 @@
         ) {
             WaterDropParticle waterdropparticle = new WaterDropParticle(p_108504_, p_108505_, p_108506_, p_108507_);
             waterdropparticle.pickSprite(this.sprite);
-            if (p_108504_.isPotato()) {
-                waterdropparticle.setColor(1.0F, 1.0F, 0.0F);
-            }
-
             return waterdropparticle;
         }
     }
