--- a/net/minecraft/world/level/dimension/DimensionType.java
+++ b/net/minecraft/world/level/dimension/DimensionType.java
@@ -49,7 +_,7 @@
     public static final Codec<DimensionType> DIRECT_CODEC = ExtraCodecs.catchDecoderException(
         RecordCodecBuilder.create(
             p_223568_ -> p_223568_.group(
-                        ExtraCodecs.asOptionalLong(Codec.LONG.optionalFieldOf("fixed_time")).forGetter(DimensionType::fixedTime),
+                        ExtraCodecs.asOptionalLong(Codec.LONG.lenientOptionalFieldOf("fixed_time")).forGetter(DimensionType::fixedTime),
                         Codec.BOOL.fieldOf("has_skylight").forGetter(DimensionType::hasSkyLight),
                         Codec.BOOL.fieldOf("has_ceiling").forGetter(DimensionType::hasCeiling),
                         Codec.BOOL.fieldOf("ultrawarm").forGetter(DimensionType::ultraWarm),
