--- a/net/minecraft/client/multiplayer/ClientAdvancements.java
+++ b/net/minecraft/client/multiplayer/ClientAdvancements.java
@@ -47,7 +_,7 @@
         this.tree.remove(p_104400_.getRemoved());
         this.tree.addAll(p_104400_.getAdded());
 
-        for(Entry<ResourceLocation, AdvancementProgress> entry : p_104400_.getProgress().entrySet()) {
+        for (Entry<ResourceLocation, AdvancementProgress> entry : p_104400_.getProgress().entrySet()) {
             AdvancementNode advancementnode = this.tree.get(entry.getKey());
             if (advancementnode != null) {
                 AdvancementProgress advancementprogress = entry.getValue();
