--- a/net/minecraft/client/multiplayer/ClientHandshakePacketListenerImpl.java
+++ b/net/minecraft/client/multiplayer/ClientHandshakePacketListenerImpl.java
@@ -94,9 +_,9 @@
     }
 
     private void switchState(ClientHandshakePacketListenerImpl.State p_302230_) {
-        ClientHandshakePacketListenerImpl.State clienthandshakepacketlistenerimpl$state = this.state.updateAndGet(p_302151_ -> {
-            if (!p_302230_.fromStates.contains(p_302151_)) {
-                throw new IllegalStateException("Tried to switch to " + p_302230_ + " from " + p_302151_ + ", but expected one of " + p_302230_.fromStates);
+        ClientHandshakePacketListenerImpl.State clienthandshakepacketlistenerimpl$state = this.state.updateAndGet(p_339288_ -> {
+            if (!p_302230_.fromStates.contains(p_339288_)) {
+                throw new IllegalStateException("Tried to switch to " + p_302230_ + " from " + p_339288_ + ", but expected one of " + p_302230_.fromStates);
             } else {
                 return p_302230_;
             }
@@ -188,7 +_,8 @@
                         null,
                         this.serverData,
                         this.parent,
-                        this.cookies
+                        this.cookies,
+                        null
                     )
                 )
             );
