--- a/net/minecraft/client/multiplayer/ClientChunkCache.java
+++ b/net/minecraft/client/multiplayer/ClientChunkCache.java
@@ -63,6 +_,7 @@
             int i = this.storage.getIndex(p_295783_.x, p_295783_.z);
             LevelChunk levelchunk = this.storage.getChunk(i);
             if (isValidChunk(levelchunk, p_295783_.x, p_295783_.z)) {
+                net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.level.ChunkEvent.Unload(levelchunk));
                 this.storage.replace(i, levelchunk, null);
             }
         }
@@ -123,6 +_,7 @@
             }
 
             this.level.onChunkLoaded(chunkpos);
+            net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.level.ChunkEvent.Load(levelchunk, false));
             return levelchunk;
         }
     }
@@ -144,7 +_,7 @@
             clientchunkcache$storage.viewCenterX = this.storage.viewCenterX;
             clientchunkcache$storage.viewCenterZ = this.storage.viewCenterZ;
 
-            for(int k = 0; k < this.storage.chunks.length(); ++k) {
+            for (int k = 0; k < this.storage.chunks.length(); k++) {
                 LevelChunk levelchunk = this.storage.chunks.get(k);
                 if (levelchunk != null) {
                     ChunkPos chunkpos = levelchunk.getPos();
@@ -199,18 +_,18 @@
         protected void replace(int p_104485_, @Nullable LevelChunk p_104486_) {
             LevelChunk levelchunk = this.chunks.getAndSet(p_104485_, p_104486_);
             if (levelchunk != null) {
-                --this.chunkCount;
+                this.chunkCount--;
                 ClientChunkCache.this.level.unload(levelchunk);
             }
 
             if (p_104486_ != null) {
-                ++this.chunkCount;
+                this.chunkCount++;
             }
         }
 
         protected LevelChunk replace(int p_104488_, LevelChunk p_104489_, @Nullable LevelChunk p_104490_) {
             if (this.chunks.compareAndSet(p_104488_, p_104489_, p_104490_) && p_104490_ == null) {
-                --this.chunkCount;
+                this.chunkCount--;
             }
 
             ClientChunkCache.this.level.unload(p_104489_);
@@ -230,8 +_,8 @@
             try (FileOutputStream fileoutputstream = new FileOutputStream(p_171623_)) {
                 int i = ClientChunkCache.this.storage.chunkRadius;
 
-                for(int j = this.viewCenterZ - i; j <= this.viewCenterZ + i; ++j) {
-                    for(int k = this.viewCenterX - i; k <= this.viewCenterX + i; ++k) {
+                for (int j = this.viewCenterZ - i; j <= this.viewCenterZ + i; j++) {
+                    for (int k = this.viewCenterX - i; k <= this.viewCenterX + i; k++) {
                         LevelChunk levelchunk = ClientChunkCache.this.storage.chunks.get(ClientChunkCache.this.storage.getIndex(k, j));
                         if (levelchunk != null) {
                             ChunkPos chunkpos = levelchunk.getPos();
