--- a/net/minecraft/client/multiplayer/ClientSuggestionProvider.java
+++ b/net/minecraft/client/multiplayer/ClientSuggestionProvider.java
@@ -50,7 +_,7 @@
     public Collection<String> getOnlinePlayerNames() {
         List<String> list = Lists.newArrayList();
 
-        for(PlayerInfo playerinfo : this.connection.getOnlinePlayers()) {
+        for (PlayerInfo playerinfo : this.connection.getOnlinePlayers()) {
             list.add(playerinfo.getProfile().getName());
         }
 
@@ -104,7 +_,7 @@
         CommandContext<?> p_212432_
     ) {
         return this.registryAccess().registry(p_212429_).map(p_212427_ -> {
-            this.suggestRegistryElements(p_212427_, p_212430_, p_212431_);
+            this.suggestRegistryElements((Registry<?>)p_212427_, p_212430_, p_212431_);
             return p_212431_.buildFuture();
         }).orElseGet(() -> this.customSuggestion(p_212432_));
     }
@@ -177,7 +_,7 @@
     }
 
     public void modifyCustomCompletions(ClientboundCustomChatCompletionsPacket.Action p_240810_, List<String> p_240765_) {
-        switch(p_240810_) {
+        switch (p_240810_) {
             case ADD:
                 this.customCompletionSuggestions.addAll(p_240765_);
                 break;
