--- a/net/minecraft/client/multiplayer/RegistryDataCollector.java
+++ b/net/minecraft/client/multiplayer/RegistryDataCollector.java
@@ -62,7 +_,7 @@
         private final Map<ResourceKey<? extends Registry<?>>, List<RegistrySynchronization.PackedRegistryEntry>> elements = new HashMap<>();
 
         public void append(ResourceKey<? extends Registry<?>> p_321577_, List<RegistrySynchronization.PackedRegistryEntry> p_321551_) {
-            this.elements.computeIfAbsent(p_321577_, p_321745_ -> new ArrayList()).addAll(p_321551_);
+            this.elements.computeIfAbsent(p_321577_, p_321745_ -> new ArrayList<>()).addAll(p_321551_);
         }
 
         public RegistryAccess loadRegistries(ResourceProvider p_326039_, RegistryAccess p_321627_) {
