--- a/net/minecraft/client/multiplayer/ServerStatusPinger.java
+++ b/net/minecraft/client/multiplayer/ServerStatusPinger.java
@@ -79,7 +_,7 @@
                             if (!p_273230_.sample().isEmpty()) {
                                 List<Component> list = new ArrayList<>(p_273230_.sample().size());
 
-                                for(GameProfile gameprofile : p_273230_.sample()) {
+                                for (GameProfile gameprofile : p_273230_.sample()) {
                                     list.add(Component.literal(gameprofile.getName()));
                                 }
 
@@ -169,10 +_,10 @@
     }
 
     public void tick() {
-        synchronized(this.connections) {
+        synchronized (this.connections) {
             Iterator<Connection> iterator = this.connections.iterator();
 
-            while(iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 Connection connection = iterator.next();
                 if (connection.isConnected()) {
                     connection.tick();
@@ -185,10 +_,10 @@
     }
 
     public void removeAll() {
-        synchronized(this.connections) {
+        synchronized (this.connections) {
             Iterator<Connection> iterator = this.connections.iterator();
 
-            while(iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 Connection connection = iterator.next();
                 if (connection.isConnected()) {
                     iterator.remove();
