--- a/net/minecraft/world/entity/ai/Brain.java
+++ b/net/minecraft/world/entity/ai/Brain.java
@@ -72,7 +_,11 @@
                     @Override
                     public <T> Stream<T> keys(DynamicOps<T> p_22029_) {
                         return p_21947_.stream()
-                            .flatMap(p_22020_ -> p_22020_.getCodec().map(p_258254_ -> BuiltInRegistries.MEMORY_MODULE_TYPE.getKey(p_22020_)).stream())
+                            .flatMap(
+                                p_22020_ -> p_22020_.getCodec()
+                                        .map(p_258254_ -> BuiltInRegistries.MEMORY_MODULE_TYPE.getKey((MemoryModuleType<?>)p_22020_))
+                                        .stream()
+                            )
                             .map(p_22018_ -> p_22029_.createString(p_22018_.toString()));
                     }
         
@@ -83,12 +_,12 @@
                         );
                         p_22023_.entries()
                             .forEach(
-                                p_258252_ -> {
+                                p_337704_ -> {
                                     DataResult<MemoryModuleType<?>> dataresult = BuiltInRegistries.MEMORY_MODULE_TYPE
                                         .byNameCodec()
-                                        .parse(p_22022_, p_258252_.getFirst());
+                                        .parse(p_22022_, p_337704_.getFirst());
                                     DataResult<? extends Brain.MemoryValue<?>> dataresult1 = dataresult.flatMap(
-                                        p_147350_ -> this.captureRead(p_147350_, p_22022_, (T)p_258252_.getSecond())
+                                        p_147350_ -> this.captureRead((MemoryModuleType<T>)p_147350_, p_22022_, (T)p_337704_.getSecond())
                                     );
                                     mutableobject1.setValue(mutableobject1.getValue().apply2(Builder::add, dataresult1));
                                 }
@@ -127,21 +_,21 @@
     ) {
         this.codec = p_21858_;
 
-        for(MemoryModuleType<?> memorymoduletype : p_21855_) {
+        for (MemoryModuleType<?> memorymoduletype : p_21855_) {
             this.memories.put(memorymoduletype, Optional.empty());
         }
 
-        for(SensorType<? extends Sensor<? super E>> sensortype : p_21856_) {
-            this.sensors.put(sensortype, sensortype.create());
+        for (SensorType<? extends Sensor<? super E>> sensortype : p_21856_) {
+            this.sensors.put(sensortype, (Sensor<? super E>)sensortype.create());
         }
 
-        for(Sensor<? super E> sensor : this.sensors.values()) {
-            for(MemoryModuleType<?> memorymoduletype1 : sensor.requires()) {
+        for (Sensor<? super E> sensor : this.sensors.values()) {
+            for (MemoryModuleType<?> memorymoduletype1 : sensor.requires()) {
                 this.memories.put(memorymoduletype1, Optional.empty());
             }
         }
 
-        for(Brain.MemoryValue<?> memoryvalue : p_21857_) {
+        for (Brain.MemoryValue<?> memoryvalue : p_21857_) {
             memoryvalue.setMemoryInternal(this);
         }
     }
@@ -151,10 +_,7 @@
     }
 
     Stream<Brain.MemoryValue<?>> memories() {
-        return this.memories
-            .entrySet()
-            .stream()
-            .map(p_21929_ -> Brain.MemoryValue.createUnchecked((MemoryModuleType<? extends ExpirableValue<?>>)p_21929_.getKey(), p_21929_.getValue()));
+        return this.memories.entrySet().stream().map(p_21929_ -> Brain.MemoryValue.createUnchecked(p_21929_.getKey(), p_21929_.getValue()));
     }
 
     public boolean hasMemoryValue(MemoryModuleType<?> p_21875_) {
@@ -162,7 +_,7 @@
     }
 
     public void clearMemories() {
-        this.memories.keySet().forEach(p_276103_ -> this.memories.put(p_276103_, Optional.empty()));
+        this.memories.keySet().forEach(p_276103_ -> this.memories.put((MemoryModuleType<?>)p_276103_, Optional.empty()));
     }
 
     public <U> void eraseMemory(MemoryModuleType<U> p_21937_) {
@@ -223,13 +_,11 @@
 
     public boolean checkMemory(MemoryModuleType<?> p_21877_, MemoryStatus p_21878_) {
         Optional<? extends ExpirableValue<?>> optional = this.memories.get(p_21877_);
-        if (optional == null) {
-            return false;
-        } else {
-            return p_21878_ == MemoryStatus.REGISTERED
+        return optional == null
+            ? false
+            : p_21878_ == MemoryStatus.REGISTERED
                 || p_21878_ == MemoryStatus.VALUE_PRESENT && optional.isPresent()
                 || p_21878_ == MemoryStatus.VALUE_ABSENT && optional.isEmpty();
-        }
     }
 
     public Schedule getSchedule() {
@@ -255,9 +_,9 @@
     public List<BehaviorControl<? super E>> getRunningBehaviors() {
         List<BehaviorControl<? super E>> list = new ObjectArrayList<>();
 
-        for(Map<Activity, Set<BehaviorControl<? super E>>> map : this.availableBehaviorsByPriority.values()) {
-            for(Set<BehaviorControl<? super E>> set : map.values()) {
-                for(BehaviorControl<? super E> behaviorcontrol : set) {
+        for (Map<Activity, Set<BehaviorControl<? super E>>> map : this.availableBehaviorsByPriority.values()) {
+            for (Set<BehaviorControl<? super E>> set : map.values()) {
+                for (BehaviorControl<? super E> behaviorcontrol : set) {
                     if (behaviorcontrol.getStatus() == Behavior.Status.RUNNING) {
                         list.add(behaviorcontrol);
                     }
@@ -273,7 +_,7 @@
     }
 
     public Optional<Activity> getActiveNonCoreActivity() {
-        for(Activity activity : this.activeActivities) {
+        for (Activity activity : this.activeActivities) {
             if (!this.coreActivities.contains(activity)) {
                 return Optional.of(activity);
             }
@@ -300,11 +_,11 @@
     }
 
     private void eraseMemoriesForOtherActivitesThan(Activity p_21967_) {
-        for(Activity activity : this.activeActivities) {
+        for (Activity activity : this.activeActivities) {
             if (activity != p_21967_) {
                 Set<MemoryModuleType<?>> set = this.activityMemoriesToEraseWhenStopped.get(activity);
                 if (set != null) {
-                    for(MemoryModuleType<?> memorymoduletype : set) {
+                    for (MemoryModuleType<?> memorymoduletype : set) {
                         this.eraseMemory(memorymoduletype);
                     }
                 }
@@ -323,7 +_,7 @@
     }
 
     public void setActiveActivityToFirstValid(List<Activity> p_21927_) {
-        for(Activity activity : p_21927_) {
+        for (Activity activity : p_21927_) {
             if (this.activityRequirementsAreMet(activity)) {
                 this.setActiveActivity(activity);
                 break;
@@ -370,11 +_,11 @@
             this.activityMemoriesToEraseWhenStopped.put(p_21908_, p_21911_);
         }
 
-        for(Pair<Integer, ? extends BehaviorControl<? super E>> pair : p_21909_) {
+        for (Pair<Integer, ? extends BehaviorControl<? super E>> pair : p_21909_) {
             this.availableBehaviorsByPriority
                 .computeIfAbsent(pair.getFirst(), p_21917_ -> Maps.newHashMap())
                 .computeIfAbsent(p_21908_, p_21972_ -> Sets.newLinkedHashSet())
-                .add(pair.getSecond());
+                .add((BehaviorControl<? super E>)pair.getSecond());
         }
     }
 
@@ -390,7 +_,7 @@
     public Brain<E> copyWithoutBehaviors() {
         Brain<E> brain = new Brain<>(this.memories.keySet(), this.sensors.keySet(), ImmutableList.of(), this.codec);
 
-        for(Entry<MemoryModuleType<?>, Optional<? extends ExpirableValue<?>>> entry : this.memories.entrySet()) {
+        for (Entry<MemoryModuleType<?>, Optional<? extends ExpirableValue<?>>> entry : this.memories.entrySet()) {
             MemoryModuleType<?> memorymoduletype = entry.getKey();
             if (entry.getValue().isPresent()) {
                 brain.memories.put(memorymoduletype, entry.getValue());
@@ -408,15 +_,15 @@
     }
 
     private void tickSensors(ServerLevel p_21950_, E p_21951_) {
-        for(Sensor<? super E> sensor : this.sensors.values()) {
+        for (Sensor<? super E> sensor : this.sensors.values()) {
             sensor.tick(p_21950_, p_21951_);
         }
     }
 
     private void forgetOutdatedMemories() {
-        for(Entry<MemoryModuleType<?>, Optional<? extends ExpirableValue<?>>> entry : this.memories.entrySet()) {
+        for (Entry<MemoryModuleType<?>, Optional<? extends ExpirableValue<?>>> entry : this.memories.entrySet()) {
             if (entry.getValue().isPresent()) {
-                ExpirableValue<?> expirablevalue = entry.getValue().get();
+                ExpirableValue<?> expirablevalue = (ExpirableValue<?>)entry.getValue().get();
                 if (expirablevalue.hasExpired()) {
                     this.eraseMemory(entry.getKey());
                 }
@@ -429,7 +_,7 @@
     public void stopAll(ServerLevel p_21934_, E p_21935_) {
         long i = p_21935_.level().getGameTime();
 
-        for(BehaviorControl<? super E> behaviorcontrol : this.getRunningBehaviors()) {
+        for (BehaviorControl<? super E> behaviorcontrol : this.getRunningBehaviors()) {
             behaviorcontrol.doStop(p_21934_, p_21935_, i);
         }
     }
@@ -437,11 +_,11 @@
     private void startEachNonRunningBehavior(ServerLevel p_21958_, E p_21959_) {
         long i = p_21958_.getGameTime();
 
-        for(Map<Activity, Set<BehaviorControl<? super E>>> map : this.availableBehaviorsByPriority.values()) {
-            for(Entry<Activity, Set<BehaviorControl<? super E>>> entry : map.entrySet()) {
+        for (Map<Activity, Set<BehaviorControl<? super E>>> map : this.availableBehaviorsByPriority.values()) {
+            for (Entry<Activity, Set<BehaviorControl<? super E>>> entry : map.entrySet()) {
                 Activity activity = entry.getKey();
                 if (this.activeActivities.contains(activity)) {
-                    for(BehaviorControl<? super E> behaviorcontrol : entry.getValue()) {
+                    for (BehaviorControl<? super E> behaviorcontrol : entry.getValue()) {
                         if (behaviorcontrol.getStatus() == Behavior.Status.STOPPED) {
                             behaviorcontrol.tryStart(p_21958_, p_21959_, i);
                         }
@@ -454,7 +_,7 @@
     private void tickEachRunningBehavior(ServerLevel p_21964_, E p_21965_) {
         long i = p_21964_.getGameTime();
 
-        for(BehaviorControl<? super E> behaviorcontrol : this.getRunningBehaviors()) {
+        for (BehaviorControl<? super E> behaviorcontrol : this.getRunningBehaviors()) {
             behaviorcontrol.tickOrStop(p_21964_, p_21965_, i);
         }
     }
@@ -463,7 +_,7 @@
         if (!this.activityRequirements.containsKey(p_21970_)) {
             return false;
         } else {
-            for(Pair<MemoryModuleType<?>, MemoryStatus> pair : this.activityRequirements.get(p_21970_)) {
+            for (Pair<MemoryModuleType<?>, MemoryStatus> pair : this.activityRequirements.get(p_21970_)) {
                 MemoryModuleType<?> memorymoduletype = pair.getFirst();
                 MemoryStatus memorystatus = pair.getSecond();
                 if (!this.checkMemory(memorymoduletype, memorystatus)) {
@@ -485,7 +_,7 @@
         int i = p_21860_;
         Builder<Pair<Integer, ? extends BehaviorControl<? super E>>> builder = ImmutableList.builder();
 
-        for(BehaviorControl<? super E> behaviorcontrol : p_21861_) {
+        for (BehaviorControl<? super E> behaviorcontrol : p_21861_) {
             builder.add(Pair.of(i++, behaviorcontrol));
         }
 
@@ -497,7 +_,7 @@
         private final Optional<? extends ExpirableValue<U>> value;
 
         static <U> Brain.MemoryValue<U> createUnchecked(MemoryModuleType<U> p_22060_, Optional<? extends ExpirableValue<?>> p_22061_) {
-            return new Brain.MemoryValue<U>(p_22060_, (Optional<? extends ExpirableValue<U>>)p_22061_);
+            return new Brain.MemoryValue<>(p_22060_, (Optional<? extends ExpirableValue<U>>)p_22061_);
         }
 
         MemoryValue(MemoryModuleType<U> p_22033_, Optional<? extends ExpirableValue<U>> p_22034_) {
@@ -517,7 +_,7 @@
                             .ifPresent(
                                 p_258258_ -> p_22049_.add(
                                         BuiltInRegistries.MEMORY_MODULE_TYPE.byNameCodec().encodeStart(p_22048_, this.type),
-                                        p_22053_.encodeStart(p_22048_, p_258258_)
+                                        p_22053_.encodeStart(p_22048_, (ExpirableValue<U>)p_258258_)
                                     )
                             )
                 );
