--- a/net/minecraft/world/level/gameevent/EuclideanGameEventListenerRegistry.java
+++ b/net/minecraft/world/level/gameevent/EuclideanGameEventListenerRegistry.java
@@ -66,7 +_,7 @@
         try {
             Iterator<GameEventListener> iterator = this.listeners.iterator();
 
-            while(iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 GameEventListener gameeventlistener = iterator.next();
                 if (this.listenersToRemove.remove(gameeventlistener)) {
                     iterator.remove();
