--- a/net/minecraft/world/level/gameevent/PositionSourceType.java
+++ b/net/minecraft/world/level/gameevent/PositionSourceType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.gameevent;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.network.RegistryFriendlyByteBuf;
@@ -10,7 +_,7 @@
     PositionSourceType<BlockPositionSource> BLOCK = register("block", new BlockPositionSource.Type());
     PositionSourceType<EntityPositionSource> ENTITY = register("entity", new EntityPositionSource.Type());
 
-    Codec<T> codec();
+    MapCodec<T> codec();
 
     StreamCodec<RegistryFriendlyByteBuf, T> streamCodec();
 
