--- a/net/minecraft/world/entity/animal/goat/GoatAi.java
+++ b/net/minecraft/world/entity/animal/goat/GoatAi.java
@@ -50,7 +_,7 @@
     private static final UniformInt TIME_BETWEEN_RAMS = UniformInt.of(600, 6000);
     private static final UniformInt TIME_BETWEEN_RAMS_SCREAMER = UniformInt.of(100, 300);
     private static final TargetingConditions RAM_TARGET_CONDITIONS = TargetingConditions.forCombat()
-        .selector(p_340123_ -> !p_340123_.getType().equals(EntityType.GOAT) && p_340123_.level().getWorldBorder().isWithinBounds(p_340123_.getBoundingBox()));
+        .selector(p_337840_ -> !p_337840_.getType().equals(EntityType.GOAT) && p_337840_.level().getWorldBorder().isWithinBounds(p_337840_.getBoundingBox()));
     private static final float SPEED_MULTIPLIER_WHEN_RAMMING = 3.0F;
     public static final int RAM_MIN_DISTANCE = 4;
     public static final float ADULT_RAM_KNOCKBACK_FORCE = 2.5F;
@@ -146,7 +_,7 @@
                         p_149474_ -> p_149474_.isScreamingGoat() ? TIME_BETWEEN_RAMS_SCREAMER : TIME_BETWEEN_RAMS,
                         RAM_TARGET_CONDITIONS,
                         3.0F,
-                        p_340122_ -> p_340122_.isBaby() ? 1.0 : 2.5,
+                        p_330059_ -> p_330059_.isBaby() ? 1.0 : 2.5,
                         p_149468_ -> p_149468_.isScreamingGoat() ? SoundEvents.GOAT_SCREAMING_RAM_IMPACT : SoundEvents.GOAT_RAM_IMPACT,
                         p_218772_ -> p_218772_.isScreamingGoat() ? SoundEvents.GOAT_SCREAMING_HORN_BREAK : SoundEvents.GOAT_HORN_BREAK
                     )
