--- a/com/mojang/realmsclient/gui/task/DataFetcher.java
+++ b/com/mojang/realmsclient/gui/task/DataFetcher.java
@@ -95,19 +_,19 @@
         }
 
         public void forceUpdate() {
-            for(DataFetcher.SubscribedTask<?> subscribedtask : this.subscriptions) {
+            for (DataFetcher.SubscribedTask<?> subscribedtask : this.subscriptions) {
                 subscribedtask.runCallback();
             }
         }
 
         public void tick() {
-            for(DataFetcher.SubscribedTask<?> subscribedtask : this.subscriptions) {
+            for (DataFetcher.SubscribedTask<?> subscribedtask : this.subscriptions) {
                 subscribedtask.update(DataFetcher.this.timeSource.get(DataFetcher.this.resolution));
             }
         }
 
         public void reset() {
-            for(DataFetcher.SubscribedTask<?> subscribedtask : this.subscriptions) {
+            for (DataFetcher.SubscribedTask<?> subscribedtask : this.subscriptions) {
                 subscribedtask.reset();
             }
         }
@@ -146,7 +_,7 @@
                 this.pendingTask = null;
                 long i = computationresult.time;
                 computationresult.value().ifLeft(p_239691_ -> {
-                    this.lastResult = new DataFetcher.SuccessfulComputationResult<>(p_239691_, i);
+                    this.lastResult = new DataFetcher.SuccessfulComputationResult<>((T)p_239691_, i);
                     this.nextUpdate = i + this.period * this.repeatStrategy.delayCyclesAfterSuccess();
                 }).ifRight(p_239281_ -> {
                     long j = this.repeatStrategy.delayCyclesAfterFailure();
