--- a/com/mojang/realmsclient/gui/task/RepeatedDelayStrategy.java
+++ b/com/mojang/realmsclient/gui/task/RepeatedDelayStrategy.java
@@ -36,7 +_,7 @@
 
             @Override
             public long delayCyclesAfterFailure() {
-                ++this.failureCount;
+                this.failureCount++;
                 long i = Math.min(1L << this.failureCount, (long)p_239256_);
                 LOGGER.debug("Skipping for {} extra cycles", i);
                 return i;
