--- a/net/minecraft/util/profiling/metrics/profiling/ActiveMetricsRecorder.java
+++ b/net/minecraft/util/profiling/metrics/profiling/ActiveMetricsRecorder.java
@@ -97,18 +_,18 @@
         this.verifyStarted();
         this.thisTickSamplers = this.metricsSamplerProvider.samplers(() -> this.singleTickProfiler);
 
-        for(MetricSampler metricsampler : this.thisTickSamplers) {
+        for (MetricSampler metricsampler : this.thisTickSamplers) {
             metricsampler.onStartTick();
         }
 
-        ++this.currentTick;
+        this.currentTick++;
     }
 
     @Override
     public void endTick() {
         this.verifyStarted();
         if (this.currentTick != 0) {
-            for(MetricSampler metricsampler : this.thisTickSamplers) {
+            for (MetricSampler metricsampler : this.thisTickSamplers) {
                 metricsampler.onEndTick(this.currentTick);
                 if (metricsampler.triggersThreshold()) {
                     RecordedDeviation recordeddeviation = new RecordedDeviation(Instant.now(), this.currentTick, this.singleTickProfiler.getResults());
@@ -154,7 +_,7 @@
     }
 
     private void cleanup(Collection<MetricSampler> p_216817_) {
-        for(MetricSampler metricsampler : p_216817_) {
+        for (MetricSampler metricsampler : p_216817_) {
             metricsampler.onFinished();
         }
 
