--- a/net/minecraft/client/renderer/entity/layers/CapeLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/CapeLayer.java
@@ -40,7 +_,7 @@
             PlayerSkin playerskin = p_116618_.getSkin();
             if (playerskin.capeTexture() != null) {
                 ItemStack itemstack = p_116618_.getItemBySlot(EquipmentSlot.CHEST);
-                if (!itemstack.is(Items.ELYTRA) && !itemstack.is(Items.POISONOUS_POLYTRA)) {
+                if (!itemstack.is(Items.ELYTRA)) {
                     p_116615_.pushPose();
                     p_116615_.translate(0.0F, 0.0F, 0.125F);
                     double d0 = Mth.lerp((double)p_116621_, p_116618_.xCloakO, p_116618_.xCloak) - Mth.lerp((double)p_116621_, p_116618_.xo, p_116618_.getX());
