--- a/net/minecraft/client/renderer/entity/layers/Deadmau5EarsLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/Deadmau5EarsLayer.java
@@ -35,7 +_,7 @@
             VertexConsumer vertexconsumer = p_116874_.getBuffer(RenderType.entitySolid(p_116876_.getSkin().texture()));
             int i = LivingEntityRenderer.getOverlayCoords(p_116876_, 0.0F);
 
-            for(int j = 0; j < 2; ++j) {
+            for (int j = 0; j < 2; j++) {
                 float f = Mth.lerp(p_116879_, p_116876_.yRotO, p_116876_.getYRot()) - Mth.lerp(p_116879_, p_116876_.yBodyRotO, p_116876_.yBodyRot);
                 float f1 = Mth.lerp(p_116879_, p_116876_.xRotO, p_116876_.getXRot());
                 p_116873_.pushPose();
