--- a/net/minecraft/client/renderer/entity/layers/EquipmentLayerRenderer.java
+++ b/net/minecraft/client/renderer/entity/layers/EquipmentLayerRenderer.java
@@ -67,18 +_,22 @@
         int outlineColor,
         int order
     ) {
+        net.neoforged.neoforge.client.extensions.common.IClientItemExtensions extensions = net.neoforged.neoforge.client.extensions.common.IClientItemExtensions.of(itemStack);
+        model = extensions.getGenericArmorModel(itemStack, layerType, model);
         List<EquipmentClientInfo.Layer> list = this.equipmentAssets.get(equipmentAssetId).getLayers(layerType);
         if (!list.isEmpty()) {
-            int i = DyedItemColor.getOrDefault(itemStack, 0);
+            int i = extensions.getDefaultDyeColor(itemStack);
             boolean flag = itemStack.hasFoil();
             int j = order;
 
+            int idx = 0;
             for (EquipmentClientInfo.Layer equipmentclientinfo$layer : list) {
-                int k = getColorForLayer(equipmentclientinfo$layer, i);
+                int k = extensions.getArmorLayerTintColor(itemStack, equipmentclientinfo$layer, idx, i);
                 if (k != 0) {
                     Identifier identifier = equipmentclientinfo$layer.usePlayerTexture() && playerTextureOverride != null
                         ? playerTextureOverride
                         : this.layerTextureLookup.apply(new EquipmentLayerRenderer.LayerTextureKey(layerType, equipmentclientinfo$layer));
+                    identifier = net.neoforged.neoforge.client.ClientHooks.getArmorTexture(itemStack, layerType, equipmentclientinfo$layer, identifier);
                     submitNodeCollector.order(j++)
                         .submitModel(
                             model,
@@ -101,6 +_,7 @@
 
                     flag = false;
                 }
+                idx++;
             }
 
             ArmorTrim armortrim = itemStack.get(DataComponents.TRIM);
