--- a/net/minecraft/client/renderer/entity/layers/VillagerProfessionLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/VillagerProfessionLayer.java
@@ -11,7 +_,6 @@
 import net.minecraft.client.model.EntityModel;
 import net.minecraft.client.model.VillagerHeadModel;
 import net.minecraft.client.renderer.MultiBufferSource;
-import net.minecraft.client.renderer.entity.LivingEntityRenderer;
 import net.minecraft.client.renderer.entity.RenderLayerParent;
 import net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection;
 import net.minecraft.core.DefaultedRegistry;
@@ -64,7 +_,6 @@
             VillagerData villagerdata = p_117649_.getVillagerData();
             VillagerType villagertype = villagerdata.getType();
             VillagerProfession villagerprofession = villagerdata.getProfession();
-            boolean flag = p_117649_.isPotato();
             VillagerMetaDataSection.Hat villagermetadatasection$hat = this.getHatData(this.typeHatCache, "type", BuiltInRegistries.VILLAGER_TYPE, villagertype);
             VillagerMetaDataSection.Hat villagermetadatasection$hat1 = this.getHatData(
                 this.professionHatCache, "profession", BuiltInRegistries.VILLAGER_PROFESSION, villagerprofession
@@ -79,10 +_,6 @@
             m.hatVisible(true);
             if (villagerprofession != VillagerProfession.NONE && !p_117649_.isBaby()) {
                 ResourceLocation resourcelocation1 = this.getResourceLocation("profession", BuiltInRegistries.VILLAGER_PROFESSION.getKey(villagerprofession));
-                if (flag) {
-                    resourcelocation1 = LivingEntityRenderer.potatoify(resourcelocation1);
-                }
-
                 renderColoredCutoutModel(m, resourcelocation1, p_117646_, p_117647_, p_117648_, p_117649_, 1.0F, 1.0F, 1.0F);
                 if (villagerprofession != VillagerProfession.NITWIT) {
                     ResourceLocation resourcelocation2 = this.getResourceLocation(
