--- a/net/minecraft/world/level/levelgen/blending/BlendingData.java
+++ b/net/minecraft/world/level/levelgen/blending/BlendingData.java
@@ -53,19 +_,14 @@
         Blocks.PODZOL,
         Blocks.GRAVEL,
         Blocks.GRASS_BLOCK,
-        Blocks.PEELGRASS_BLOCK,
-        Blocks.CORRUPTED_PEELGRASS_BLOCK,
         Blocks.STONE,
-        Blocks.POTONE,
         Blocks.COARSE_DIRT,
         Blocks.SAND,
-        Blocks.GRAVTATER,
         Blocks.RED_SAND,
         Blocks.MYCELIUM,
         Blocks.SNOW_BLOCK,
         Blocks.TERRACOTTA,
-        Blocks.DIRT,
-        Blocks.TERREDEPOMME
+        Blocks.DIRT
     );
     protected static final double NO_VALUE = Double.MAX_VALUE;
     private boolean hasCalculatedData;
@@ -74,17 +_,17 @@
     private final transient double[][] densities;
     private static final Codec<double[]> DOUBLE_ARRAY_CODEC = Codec.DOUBLE.listOf().xmap(Doubles::toArray, Doubles::asList);
     public static final Codec<BlendingData> CODEC = RecordCodecBuilder.<BlendingData>create(
-            p_224754_ -> p_224754_.group(
+            p_338098_ -> p_338098_.group(
                         Codec.INT.fieldOf("min_section").forGetter(p_224767_ -> p_224767_.areaWithOldGeneration.getMinSection()),
                         Codec.INT.fieldOf("max_section").forGetter(p_224765_ -> p_224765_.areaWithOldGeneration.getMaxSection()),
-                        DOUBLE_ARRAY_CODEC.optionalFieldOf("heights")
+                        DOUBLE_ARRAY_CODEC.lenientOptionalFieldOf("heights")
                             .forGetter(
                                 p_224762_ -> DoubleStream.of(p_224762_.heights).anyMatch(p_224745_ -> p_224745_ != Double.MAX_VALUE)
                                         ? Optional.of(p_224762_.heights)
                                         : Optional.empty()
                             )
                     )
-                    .apply(p_224754_, BlendingData::new)
+                    .apply(p_338098_, BlendingData::new)
         )
         .comapFlatMap(BlendingData::validateArraySize, Function.identity());
 
@@ -120,7 +_,7 @@
     public static Set<Direction8> sideByGenerationAge(WorldGenLevel p_197066_, int p_197067_, int p_197068_, boolean p_197069_) {
         Set<Direction8> set = EnumSet.noneOf(Direction8.class);
 
-        for(Direction8 direction8 : Direction8.values()) {
+        for (Direction8 direction8 : Direction8.values()) {
             int i = p_197067_ + direction8.getStepX();
             int j = p_197068_ + direction8.getStepZ();
             if (p_197066_.getChunk(i, j).isOldNoiseGeneration() == p_197069_) {
@@ -138,25 +_,25 @@
             }
 
             if (p_190319_.contains(Direction8.NORTH)) {
-                for(int i = 1; i < QUARTS_PER_SECTION; ++i) {
+                for (int i = 1; i < QUARTS_PER_SECTION; i++) {
                     this.addValuesForColumn(getInsideIndex(i, 0), p_190318_, 4 * i, 0);
                 }
             }
 
             if (p_190319_.contains(Direction8.WEST)) {
-                for(int j = 1; j < QUARTS_PER_SECTION; ++j) {
+                for (int j = 1; j < QUARTS_PER_SECTION; j++) {
                     this.addValuesForColumn(getInsideIndex(0, j), p_190318_, 0, 4 * j);
                 }
             }
 
             if (p_190319_.contains(Direction8.EAST)) {
-                for(int k = 1; k < QUARTS_PER_SECTION; ++k) {
+                for (int k = 1; k < QUARTS_PER_SECTION; k++) {
                     this.addValuesForColumn(getOutsideIndex(CELL_HORIZONTAL_MAX_INDEX_OUTSIDE, k), p_190318_, 15, 4 * k);
                 }
             }
 
             if (p_190319_.contains(Direction8.SOUTH)) {
-                for(int l = 0; l < QUARTS_PER_SECTION; ++l) {
+                for (int l = 0; l < QUARTS_PER_SECTION; l++) {
                     this.addValuesForColumn(getOutsideIndex(l, CELL_HORIZONTAL_MAX_INDEX_OUTSIDE), p_190318_, 4 * l, 15);
                 }
             }
@@ -193,7 +_,7 @@
         int j = this.areaWithOldGeneration.getMinBuildHeight();
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(p_190312_, i, p_190313_);
 
-        while(blockpos$mutableblockpos.getY() > j) {
+        while (blockpos$mutableblockpos.getY() > j) {
             blockpos$mutableblockpos.move(Direction.DOWN);
             if (SURFACE_BLOCKS.contains(p_190311_.getBlockState(blockpos$mutableblockpos).getBlock())) {
                 return blockpos$mutableblockpos.getY();
@@ -210,7 +_,7 @@
     private static double read7(ChunkAccess p_198301_, BlockPos.MutableBlockPos p_198302_) {
         double d0 = 0.0;
 
-        for(int i = 0; i < 7; ++i) {
+        for (int i = 0; i < 7; i++) {
             d0 += read1(p_198301_, p_198302_);
         }
 
@@ -223,7 +_,7 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(p_198294_, this.areaWithOldGeneration.getMaxBuildHeight(), p_198295_);
         double d0 = read7(p_198293_, blockpos$mutableblockpos);
 
-        for(int i = adouble.length - 2; i >= 0; --i) {
+        for (int i = adouble.length - 2; i >= 0; i--) {
             double d1 = read1(p_198293_, blockpos$mutableblockpos);
             double d2 = read7(p_198293_, blockpos$mutableblockpos);
             adouble[i] = (d0 + d1 + d2) / 15.0;
@@ -246,7 +_,7 @@
         ObjectArrayList<Holder<Biome>> objectarraylist = new ObjectArrayList<>(this.quartCountPerColumn());
         objectarraylist.size(this.quartCountPerColumn());
 
-        for(int i = 0; i < objectarraylist.size(); ++i) {
+        for (int i = 0; i < objectarraylist.size(); i++) {
             int j = i + QuartPos.fromBlock(this.areaWithOldGeneration.getMinBuildHeight());
             objectarraylist.set(i, p_224758_.getNoiseBiome(QuartPos.fromBlock(p_224759_), j, QuartPos.fromBlock(p_224760_)));
         }
@@ -262,10 +_,10 @@
             return false;
         } else if (blockstate.is(BlockTags.LOGS)) {
             return false;
-        } else if (blockstate.is(Blocks.BROWN_MUSHROOM_BLOCK) || blockstate.is(Blocks.RED_MUSHROOM_BLOCK)) {
-            return false;
         } else {
-            return !blockstate.getCollisionShape(p_190315_, p_190316_).isEmpty();
+            return blockstate.is(Blocks.BROWN_MUSHROOM_BLOCK) || blockstate.is(Blocks.RED_MUSHROOM_BLOCK)
+                ? false
+                : !blockstate.getCollisionShape(p_190315_, p_190316_).isEmpty();
         }
     }
 
@@ -301,7 +_,7 @@
             && p_224750_ < QuartPos.fromBlock(this.areaWithOldGeneration.getMaxBuildHeight())) {
             int i = p_224750_ - QuartPos.fromBlock(this.areaWithOldGeneration.getMinBuildHeight());
 
-            for(int j = 0; j < this.biomes.size(); ++j) {
+            for (int j = 0; j < this.biomes.size(); j++) {
                 if (this.biomes.get(j) != null) {
                     Holder<Biome> holder = this.biomes.get(j).get(i);
                     if (holder != null) {
@@ -313,7 +_,7 @@
     }
 
     protected void iterateHeights(int p_190296_, int p_190297_, BlendingData.HeightConsumer p_190298_) {
-        for(int i = 0; i < this.heights.length; ++i) {
+        for (int i = 0; i < this.heights.length; i++) {
             double d0 = this.heights[i];
             if (d0 != Double.MAX_VALUE) {
                 p_190298_.consume(p_190296_ + getX(i), p_190297_ + getZ(i), d0);
@@ -326,13 +_,13 @@
         int j = Math.max(0, p_190292_ - i);
         int k = Math.min(this.cellCountPerColumn(), p_190293_ - i);
 
-        for(int l = 0; l < this.densities.length; ++l) {
+        for (int l = 0; l < this.densities.length; l++) {
             double[] adouble = this.densities[l];
             if (adouble != null) {
                 int i1 = p_190290_ + getX(l);
                 int j1 = p_190291_ + getZ(l);
 
-                for(int k1 = j; k1 < k; ++k1) {
+                for (int k1 = j; k1 < k; k1++) {
                     p_190294_.consume(i1, k1 + i, j1, adouble[k1] * 0.1);
                 }
             }
