--- a/net/minecraft/world/level/levelgen/blending/Blender.java
+++ b/net/minecraft/world/level/levelgen/blending/Blender.java
@@ -77,8 +_,8 @@
                 Long2ObjectOpenHashMap<BlendingData> long2objectopenhashmap1 = new Long2ObjectOpenHashMap<>();
                 int i = Mth.square(HEIGHT_BLENDING_RANGE_CHUNKS + 1);
 
-                for(int j = -HEIGHT_BLENDING_RANGE_CHUNKS; j <= HEIGHT_BLENDING_RANGE_CHUNKS; ++j) {
-                    for(int k = -HEIGHT_BLENDING_RANGE_CHUNKS; k <= HEIGHT_BLENDING_RANGE_CHUNKS; ++k) {
+                for (int j = -HEIGHT_BLENDING_RANGE_CHUNKS; j <= HEIGHT_BLENDING_RANGE_CHUNKS; j++) {
+                    for (int k = -HEIGHT_BLENDING_RANGE_CHUNKS; k <= HEIGHT_BLENDING_RANGE_CHUNKS; k++) {
                         if (j * j + k * k <= i) {
                             int l = chunkpos.x + j;
                             int i1 = chunkpos.z + k;
@@ -276,8 +_,8 @@
             int i = blendingdata.getAreaWithOldGeneration().getMinBuildHeight();
             int j = blendingdata.getAreaWithOldGeneration().getMaxBuildHeight() - 1;
             if (flag) {
-                for(int k = 0; k < 16; ++k) {
-                    for(int l = 0; l < 16; ++l) {
+                for (int k = 0; k < 16; k++) {
+                    for (int l = 0; l < 16; l++) {
                         generateBorderTick(p_197033_, blockpos$mutableblockpos.setWithOffset(blockpos, k, i - 1, l));
                         generateBorderTick(p_197033_, blockpos$mutableblockpos.setWithOffset(blockpos, k, i, l));
                         generateBorderTick(p_197033_, blockpos$mutableblockpos.setWithOffset(blockpos, k, j, l));
@@ -286,18 +_,18 @@
                 }
             }
 
-            for(Direction direction : Direction.Plane.HORIZONTAL) {
+            for (Direction direction : Direction.Plane.HORIZONTAL) {
                 if (p_197032_.getChunk(chunkpos.x + direction.getStepX(), chunkpos.z + direction.getStepZ()).isOldNoiseGeneration() != flag) {
                     int i1 = direction == Direction.EAST ? 15 : 0;
                     int j1 = direction == Direction.WEST ? 0 : 15;
                     int k1 = direction == Direction.SOUTH ? 15 : 0;
                     int l1 = direction == Direction.NORTH ? 0 : 15;
 
-                    for(int i2 = i1; i2 <= j1; ++i2) {
-                        for(int j2 = k1; j2 <= l1; ++j2) {
+                    for (int i2 = i1; i2 <= j1; i2++) {
+                        for (int j2 = k1; j2 <= l1; j2++) {
                             int k2 = Math.min(j, p_197033_.getHeight(Heightmap.Types.MOTION_BLOCKING, i2, j2)) + 1;
 
-                            for(int l2 = i; l2 < k2; ++l2) {
+                            for (int l2 = i; l2 < k2; l2++) {
                                 generateBorderTick(p_197033_, blockpos$mutableblockpos.setWithOffset(blockpos, i2, l2, j2));
                             }
                         }
@@ -323,7 +_,7 @@
         ChunkPos chunkpos = p_197036_.getPos();
         Builder<Direction8, BlendingData> builder = ImmutableMap.builder();
 
-        for(Direction8 direction8 : Direction8.values()) {
+        for (Direction8 direction8 : Direction8.values()) {
             int i = chunkpos.x + direction8.getStepX();
             int j = chunkpos.z + direction8.getStepZ();
             BlendingData blendingdata = p_197035_.getChunk(i, j).getBlendingData();
@@ -357,7 +_,7 @@
         return (p_202267_, p_202268_, p_202269_) -> {
             double d0 = Double.POSITIVE_INFINITY;
 
-            for(Blender.DistanceGetter blender$distancegetter : list) {
+            for (Blender.DistanceGetter blender$distancegetter : list) {
                 double d1 = blender$distancegetter.getDistance(p_202267_, p_202268_, p_202269_);
                 if (d1 < d0) {
                     d0 = d1;
@@ -372,7 +_,7 @@
         double d0 = 0.0;
         double d1 = 0.0;
         if (p_224730_ != null) {
-            for(Direction direction : p_224730_.getDirections()) {
+            for (Direction direction : p_224730_.getDirections()) {
                 d0 += (double)(direction.getStepX() * 16);
                 d1 += (double)(direction.getStepZ() * 16);
             }
