--- a/net/minecraft/client/gui/screens/reporting/AbstractReportScreen.java
+++ b/net/minecraft/client/gui/screens/reporting/AbstractReportScreen.java
@@ -92,9 +_,8 @@
 
     private void onReportSendError(Throwable p_299933_) {
         LOGGER.error("Encountered error while sending abuse report", p_299933_);
-        Throwable throwable = p_299933_.getCause();
         Component component;
-        if (throwable instanceof ThrowingComponent throwingcomponent) {
+        if (p_299933_.getCause() instanceof ThrowingComponent throwingcomponent) {
             component = throwingcomponent.getComponent();
         } else {
             component = REPORT_SEND_GENERIC_ERROR;
