--- a/net/minecraft/client/gui/screens/reporting/ChatSelectionScreen.java
+++ b/net/minecraft/client/gui/screens/reporting/ChatSelectionScreen.java
@@ -163,9 +_,9 @@
 
         @Override
         public void acceptDivider(Component p_239876_) {
-            this.addEntryToTop(new ChatSelectionScreen.ChatSelectionList.PaddingEntry());
+            this.addEntryToTop(new ChatSelectionScreen.ChatSelectionList.PaddingEntry(this));
             this.addEntryToTop(new ChatSelectionScreen.ChatSelectionList.DividerEntry(p_239876_));
-            this.addEntryToTop(new ChatSelectionScreen.ChatSelectionList.PaddingEntry());
+            this.addEntryToTop(new ChatSelectionScreen.ChatSelectionList.PaddingEntry(this));
             this.previousHeading = null;
         }
 
@@ -246,6 +_,7 @@
             private final Component text;
 
             public DividerEntry(Component p_239672_) {
+                super(ChatSelectionList.this);
                 this.text = p_239672_;
             }
 
@@ -278,6 +_,9 @@
 
         @OnlyIn(Dist.CLIENT)
         public abstract class Entry extends ObjectSelectionList.Entry<ChatSelectionScreen.ChatSelectionList.Entry> {
+            public Entry(ChatSelectionScreen.ChatSelectionList p_239445_) {
+            }
+
             @Override
             public Component getNarration() {
                 return CommonComponents.EMPTY;
@@ -324,6 +_,7 @@
             public MessageEntry(
                 int p_240650_, Component p_240525_, Component p_240539_, @Nullable GuiMessageTag p_240551_, boolean p_240596_, boolean p_240615_
             ) {
+                super(ChatSelectionList.this);
                 this.chatId = p_240650_;
                 this.tagIcon = Optionull.map(p_240551_, GuiMessageTag::icon);
                 this.tagHoverText = p_240551_ != null && p_240551_.text() != null
@@ -453,6 +_,7 @@
             private final boolean canReport;
 
             public MessageHeadingEntry(GameProfile p_240080_, Component p_240081_, boolean p_240082_) {
+                super(ChatSelectionList.this);
                 this.heading = p_240081_;
                 this.canReport = p_240082_;
                 this.skin = ChatSelectionList.this.minecraft.getSkinManager().lookupInsecure(p_240080_);
@@ -481,6 +_,10 @@
 
         @OnlyIn(Dist.CLIENT)
         public class PaddingEntry extends ChatSelectionScreen.ChatSelectionList.Entry {
+            public PaddingEntry(ChatSelectionScreen.ChatSelectionList p_240051_) {
+                super(p_240051_);
+            }
+
             @Override
             public void render(
                 GuiGraphics p_282007_,
