--- a/net/minecraft/client/gui/screens/reporting/ChatSelectionLogFiller.java
+++ b/net/minecraft/client/gui/screens/reporting/ChatSelectionLogFiller.java
@@ -36,7 +_,7 @@
     public void fillNextPage(int p_239016_, ChatSelectionLogFiller.Output p_239017_) {
         int i = 0;
 
-        while(i < p_239016_) {
+        while (i < p_239016_) {
             LoggedChatEvent loggedchatevent = this.log.lookup(this.eventId);
             if (loggedchatevent == null) {
                 break;
@@ -51,9 +_,9 @@
                     }
 
                     p_239017_.acceptMessage(j, loggedchatmessage$player);
-                    ++i;
+                    i++;
                 } else {
-                    ++this.missedCount;
+                    this.missedCount++;
                 }
 
                 this.lastMessage = loggedchatmessage$player.message();
