--- a/net/minecraft/client/gui/screens/reporting/ReportPlayerScreen.java
+++ b/net/minecraft/client/gui/screens/reporting/ReportPlayerScreen.java
@@ -1,5 +_,6 @@
 package net.minecraft.client.gui.screens.reporting;
 
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.MultiLineTextWidget;
 import net.minecraft.client.gui.components.StringWidget;
@@ -81,7 +_,9 @@
             );
         this.layout.addChild(SpacerElement.height(20));
         this.layout.addChild(Button.builder(CommonComponents.GUI_CANCEL, p_299992_ -> this.onClose()).build());
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321362_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321362_);
+        });
         this.repositionElements();
     }
 
