--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/BlobFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/BlobFoliagePlacer.java
@@ -2,6 +_,7 @@
 
 import com.mojang.datafixers.Products.P3;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Mu;
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class BlobFoliagePlacer extends FoliagePlacer {
-    public static final Codec<BlobFoliagePlacer> CODEC = RecordCodecBuilder.create(p_68427_ -> blobParts(p_68427_).apply(p_68427_, BlobFoliagePlacer::new));
+    public static final MapCodec<BlobFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(p_68427_ -> blobParts(p_68427_).apply(p_68427_, BlobFoliagePlacer::new));
     protected final int height;
 
     protected static <P extends BlobFoliagePlacer> P3<Mu<P>, IntProvider, IntProvider, Integer> blobParts(Instance<P> p_68414_) {
@@ -40,7 +_,7 @@
         int p_273511_,
         int p_273685_
     ) {
-        for(int i = p_273685_; i >= p_273685_ - p_273580_; --i) {
+        for (int i = p_273685_; i >= p_273685_ - p_273580_; i--) {
             int j = Math.max(p_273511_ + p_273711_.radiusOffset() - 1 - i / 2, 0);
             this.placeLeavesRow(p_273066_, p_272716_, p_273178_, p_272850_, p_273711_.pos(), j, i, p_273711_.doubleTrunk());
         }
