--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/CherryFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/CherryFoliagePlacer.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -10,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class CherryFoliagePlacer extends FoliagePlacer {
-    public static final Codec<CherryFoliagePlacer> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<CherryFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(
         p_273246_ -> foliagePlacerParts(p_273246_)
                 .and(
                     p_273246_.group(
@@ -63,7 +_,7 @@
         this.placeLeavesRow(p_272723_, p_273410_, p_273057_, p_273220_, blockpos, i - 2, p_273647_ - 3, flag);
         this.placeLeavesRow(p_272723_, p_273410_, p_273057_, p_273220_, blockpos, i - 1, p_273647_ - 4, flag);
 
-        for(int j = p_273647_ - 5; j >= 0; --j) {
+        for (int j = p_273647_ - 5; j >= 0; j--) {
             this.placeLeavesRow(p_272723_, p_273410_, p_273057_, p_273220_, blockpos, i, j, flag);
         }
 
@@ -87,11 +_,9 @@
         } else {
             boolean flag = p_273380_ == p_272631_ && p_272853_ == p_272631_;
             boolean flag1 = p_272631_ > 2;
-            if (flag1) {
-                return flag || p_273380_ + p_272853_ > p_272631_ * 2 - 2 && p_273294_.nextFloat() < this.cornerHoleChance;
-            } else {
-                return flag && p_273294_.nextFloat() < this.cornerHoleChance;
-            }
+            return flag1
+                ? flag || p_273380_ + p_272853_ > p_272631_ * 2 - 2 && p_273294_.nextFloat() < this.cornerHoleChance
+                : flag && p_273294_.nextFloat() < this.cornerHoleChance;
         }
     }
 }
