--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/DarkOakFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/DarkOakFoliagePlacer.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -10,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class DarkOakFoliagePlacer extends FoliagePlacer {
-    public static final Codec<DarkOakFoliagePlacer> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<DarkOakFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(
         p_68473_ -> foliagePlacerParts(p_68473_).apply(p_68473_, DarkOakFoliagePlacer::new)
     );
 
@@ -66,10 +_,8 @@
     protected boolean shouldSkipLocation(RandomSource p_225547_, int p_225548_, int p_225549_, int p_225550_, int p_225551_, boolean p_225552_) {
         if (p_225549_ == -1 && !p_225552_) {
             return p_225548_ == p_225551_ && p_225550_ == p_225551_;
-        } else if (p_225549_ == 1) {
-            return p_225548_ + p_225550_ > p_225551_ * 2 - 2;
         } else {
-            return false;
+            return p_225549_ == 1 ? p_225548_ + p_225550_ > p_225551_ * 2 - 2 : false;
         }
     }
 }
