--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/FancyFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/FancyFoliagePlacer.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.util.Mth;
@@ -10,7 +_,9 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class FancyFoliagePlacer extends BlobFoliagePlacer {
-    public static final Codec<FancyFoliagePlacer> CODEC = RecordCodecBuilder.create(p_68518_ -> blobParts(p_68518_).apply(p_68518_, FancyFoliagePlacer::new));
+    public static final MapCodec<FancyFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(
+        p_68518_ -> blobParts(p_68518_).apply(p_68518_, FancyFoliagePlacer::new)
+    );
 
     public FancyFoliagePlacer(IntProvider p_161397_, IntProvider p_161398_, int p_161399_) {
         super(p_161397_, p_161398_, p_161399_);
@@ -33,7 +_,7 @@
         int p_225589_,
         int p_225590_
     ) {
-        for(int i = p_225590_; i >= p_225590_ - p_225588_; --i) {
+        for (int i = p_225590_; i >= p_225590_ - p_225588_; i--) {
             int j = p_225589_ + (i != p_225590_ && i != p_225590_ - p_225588_ ? 1 : 0);
             this.placeLeavesRow(p_225582_, p_273184_, p_225584_, p_225585_, p_225587_.pos(), j, i, p_225587_.doubleTrunk());
         }
