--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/FoliagePlacerType.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/FoliagePlacerType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 
@@ -18,17 +_,17 @@
         "random_spread_foliage_placer", RandomSpreadFoliagePlacer.CODEC
     );
     public static final FoliagePlacerType<CherryFoliagePlacer> CHERRY_FOLIAGE_PLACER = register("cherry_foliage_placer", CherryFoliagePlacer.CODEC);
-    private final Codec<P> codec;
-
-    private static <P extends FoliagePlacer> FoliagePlacerType<P> register(String p_68606_, Codec<P> p_68607_) {
-        return Registry.register(BuiltInRegistries.FOLIAGE_PLACER_TYPE, p_68606_, new FoliagePlacerType<>(p_68607_));
-    }
-
-    public FoliagePlacerType(Codec<P> p_68603_) {
-        this.codec = p_68603_;
-    }
-
-    public Codec<P> codec() {
+    private final MapCodec<P> codec;
+
+    private static <P extends FoliagePlacer> FoliagePlacerType<P> register(String p_68606_, MapCodec<P> p_338511_) {
+        return Registry.register(BuiltInRegistries.FOLIAGE_PLACER_TYPE, p_68606_, new FoliagePlacerType<>(p_338511_));
+    }
+
+    private FoliagePlacerType(MapCodec<P> p_338854_) {
+        this.codec = p_338854_;
+    }
+
+    public MapCodec<P> codec() {
         return this.codec;
     }
 }
