--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaPineFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaPineFoliagePlacer.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class MegaPineFoliagePlacer extends FoliagePlacer {
-    public static final Codec<MegaPineFoliagePlacer> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<MegaPineFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(
         p_68664_ -> foliagePlacerParts(p_68664_)
                 .and(IntProvider.codec(0, 24).fieldOf("crown_height").forGetter(p_161484_ -> p_161484_.crownHeight))
                 .apply(p_68664_, MegaPineFoliagePlacer::new)
@@ -43,7 +_,7 @@
         BlockPos blockpos = p_225683_.pos();
         int i = 0;
 
-        for(int j = blockpos.getY() - p_225684_ + p_225686_; j <= blockpos.getY() + p_225686_; ++j) {
+        for (int j = blockpos.getY() - p_225684_ + p_225686_; j <= blockpos.getY() + p_225686_; j++) {
             int k = blockpos.getY() - j;
             int l = p_225685_ + p_225683_.radiusOffset() + Mth.floor((float)k / (float)p_225684_ * 3.5F);
             int i1;
@@ -65,10 +_,6 @@
 
     @Override
     protected boolean shouldSkipLocation(RandomSource p_225667_, int p_225668_, int p_225669_, int p_225670_, int p_225671_, boolean p_225672_) {
-        if (p_225668_ + p_225670_ >= 7) {
-            return true;
-        } else {
-            return p_225668_ * p_225668_ + p_225670_ * p_225670_ > p_225671_ * p_225671_;
-        }
+        return p_225668_ + p_225670_ >= 7 ? true : p_225668_ * p_225668_ + p_225670_ * p_225670_ > p_225671_ * p_225671_;
     }
 }
