--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/PineFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/PineFoliagePlacer.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.util.RandomSource;
@@ -9,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class PineFoliagePlacer extends FoliagePlacer {
-    public static final Codec<PineFoliagePlacer> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<PineFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(
         p_68698_ -> foliagePlacerParts(p_68698_)
                 .and(IntProvider.codec(0, 24).fieldOf("height").forGetter(p_161500_ -> p_161500_.height))
                 .apply(p_68698_, PineFoliagePlacer::new)
@@ -40,12 +_,12 @@
     ) {
         int i = 0;
 
-        for(int j = p_225710_; j >= p_225710_ - p_225708_; --j) {
+        for (int j = p_225710_; j >= p_225710_ - p_225708_; j--) {
             this.placeLeavesRow(p_225702_, p_272791_, p_225704_, p_225705_, p_225707_.pos(), i, j, p_225707_.doubleTrunk());
             if (i >= 1 && j == p_225710_ - p_225708_ + 1) {
-                --i;
+                i--;
             } else if (i < p_225709_ + p_225707_.radiusOffset()) {
-                ++i;
+                i++;
             }
         }
     }
