--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/RandomSpreadFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/RandomSpreadFoliagePlacer.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -10,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class RandomSpreadFoliagePlacer extends FoliagePlacer {
-    public static final Codec<RandomSpreadFoliagePlacer> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<RandomSpreadFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(
         p_161522_ -> foliagePlacerParts(p_161522_)
                 .and(
                     p_161522_.group(
@@ -49,7 +_,7 @@
         BlockPos blockpos = p_225728_.pos();
         BlockPos.MutableBlockPos blockpos$mutableblockpos = blockpos.mutable();
 
-        for(int i = 0; i < this.leafPlacementAttempts; ++i) {
+        for (int i = 0; i < this.leafPlacementAttempts; i++) {
             blockpos$mutableblockpos.setWithOffset(
                 blockpos,
                 p_225725_.nextInt(p_225730_) - p_225725_.nextInt(p_225730_),
