--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/SpruceFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/SpruceFoliagePlacer.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.foliageplacers;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -10,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class SpruceFoliagePlacer extends FoliagePlacer {
-    public static final Codec<SpruceFoliagePlacer> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SpruceFoliagePlacer> CODEC = RecordCodecBuilder.mapCodec(
         p_68735_ -> foliagePlacerParts(p_68735_)
                 .and(IntProvider.codec(0, 24).fieldOf("trunk_height").forGetter(p_161553_ -> p_161553_.trunkHeight))
                 .apply(p_68735_, SpruceFoliagePlacer::new)
@@ -44,14 +_,14 @@
         int j = 1;
         int k = 0;
 
-        for(int l = p_225752_; l >= -p_225750_; --l) {
+        for (int l = p_225752_; l >= -p_225750_; l--) {
             this.placeLeavesRow(p_225744_, p_273256_, p_225746_, p_225747_, blockpos, i, l, p_225749_.doubleTrunk());
             if (i >= j) {
                 i = k;
                 k = 1;
                 j = Math.min(j + 1, p_225751_ + p_225749_.radiusOffset());
             } else {
-                ++i;
+                i++;
             }
         }
     }
