--- a/net/minecraft/world/level/storage/loot/IntRange.java
+++ b/net/minecraft/world/level/storage/loot/IntRange.java
@@ -12,7 +_,6 @@
 import java.util.Set;
 import java.util.function.Function;
 import javax.annotation.Nullable;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.Mth;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.providers.number.ConstantValue;
@@ -21,11 +_,11 @@
 
 public class IntRange {
     private static final Codec<IntRange> RECORD_CODEC = RecordCodecBuilder.create(
-        p_297986_ -> p_297986_.group(
-                    ExtraCodecs.strictOptionalField(NumberProviders.CODEC, "min").forGetter(p_297985_ -> Optional.ofNullable(p_297985_.min)),
-                    ExtraCodecs.strictOptionalField(NumberProviders.CODEC, "max").forGetter(p_297984_ -> Optional.ofNullable(p_297984_.max))
+        p_338119_ -> p_338119_.group(
+                    NumberProviders.CODEC.optionalFieldOf("min").forGetter(p_297985_ -> Optional.ofNullable(p_297985_.min)),
+                    NumberProviders.CODEC.optionalFieldOf("max").forGetter(p_297984_ -> Optional.ofNullable(p_297984_.max))
                 )
-                .apply(p_297986_, IntRange::new)
+                .apply(p_338119_, IntRange::new)
     );
     public static final Codec<IntRange> CODEC = Codec.either(Codec.INT, RECORD_CODEC)
         .xmap(p_297983_ -> p_297983_.map(IntRange::exact, Function.identity()), p_297982_ -> {
@@ -102,14 +_,11 @@
     }
 
     private OptionalInt unpackExact() {
-        if (Objects.equals(this.min, this.max)) {
-            NumberProvider numberprovider = this.min;
-            if (numberprovider instanceof ConstantValue constantvalue && Math.floor((double)constantvalue.value()) == (double)constantvalue.value()) {
-                return OptionalInt.of((int)constantvalue.value());
-            }
-        }
-
-        return OptionalInt.empty();
+        return Objects.equals(this.min, this.max)
+                && this.min instanceof ConstantValue constantvalue
+                && Math.floor((double)constantvalue.value()) == (double)constantvalue.value()
+            ? OptionalInt.of((int)constantvalue.value())
+            : OptionalInt.empty();
     }
 
     @FunctionalInterface
