--- a/net/minecraft/commands/arguments/item/ComponentPredicateParser.java
+++ b/net/minecraft/commands/arguments/item/ComponentPredicateParser.java
@@ -70,13 +_,13 @@
         dictionary.put(
             atom5, Term.sequence(Term.named(atom6), Term.optional(Term.sequence(StringReaderTerms.character(','), Term.named(atom5)))), p_336093_ -> {
                 T t = p_335727_.anyOf(p_336093_.getOrThrow(atom6));
-                return Optional.ofNullable(p_336093_.get(atom5)).map(p_335806_ -> Util.copyAndAdd(t, p_335806_)).orElse(List.of(t));
+                return Optional.ofNullable(p_336093_.get(atom5)).map(p_335806_ -> Util.copyAndAdd(t, (List<T>)p_335806_)).orElse(List.of(t));
             }
         );
         dictionary.put(
             atom6, Term.sequence(Term.named(atom7), Term.optional(Term.sequence(StringReaderTerms.character('|'), Term.named(atom6)))), p_335542_ -> {
                 T t = p_335542_.getOrThrow(atom7);
-                return Optional.ofNullable(p_335542_.get(atom6)).map(p_335939_ -> Util.copyAndAdd(t, p_335939_)).orElse(List.of(t));
+                return Optional.ofNullable(p_335542_.get(atom6)).map(p_335939_ -> Util.copyAndAdd(t, (List<T>)p_335939_)).orElse(List.of(t));
             }
         );
         dictionary.put(
@@ -103,7 +_,7 @@
                         C c = p_336062_.getOrThrow(atom10);
                         Tag tag = p_336062_.get(atom13);
                         return Optional.of(
-                            (T)(tag != null ? p_335727_.createComponentTest(p_335570_.input(), c, tag) : p_335727_.createComponentTest(p_335570_.input(), c))
+                            tag != null ? p_335727_.createComponentTest(p_335570_.input(), c, tag) : p_335727_.createComponentTest(p_335570_.input(), c)
                         );
                     }
                 } catch (CommandSyntaxException commandsyntaxexception) {
