--- a/net/minecraft/commands/arguments/item/ItemPredicateArgument.java
+++ b/net/minecraft/commands/arguments/item/ItemPredicateArgument.java
@@ -68,11 +_,11 @@
                 COUNT_ID, p_335429_ -> true, MinMaxBounds.Ints.CODEC.map(p_335378_ -> p_336161_ -> p_335378_.matches(p_336161_.getCount()))
             )
         )
-        .collect(Collectors.toUnmodifiableMap(ItemPredicateArgument.ComponentWrapper::id, p_335476_ -> p_335476_));
+        .collect(Collectors.toUnmodifiableMap(ItemPredicateArgument.ComponentWrapper::id, p_335476_ -> (ItemPredicateArgument.ComponentWrapper)p_335476_));
     static final Map<ResourceLocation, ItemPredicateArgument.PredicateWrapper> PSEUDO_PREDICATES = Stream.of(
             new ItemPredicateArgument.PredicateWrapper(COUNT_ID, MinMaxBounds.Ints.CODEC.map(p_335489_ -> p_335603_ -> p_335489_.matches(p_335603_.getCount())))
         )
-        .collect(Collectors.toUnmodifiableMap(ItemPredicateArgument.PredicateWrapper::id, p_335496_ -> p_335496_));
+        .collect(Collectors.toUnmodifiableMap(ItemPredicateArgument.PredicateWrapper::id, p_335496_ -> (ItemPredicateArgument.PredicateWrapper)p_335496_));
     private final Grammar<List<Predicate<ItemStack>>> grammarWithContext;
 
     public ItemPredicateArgument(CommandBuildContext p_235352_) {
@@ -119,8 +_,8 @@
 
         public Predicate<ItemStack> decode(ImmutableStringReader p_335970_, RegistryOps<Tag> p_335855_, Tag p_335892_) throws CommandSyntaxException {
             DataResult<? extends Predicate<ItemStack>> dataresult = this.valueChecker.parse(p_335855_, p_335892_);
-            return Util.getOrThrow(
-                dataresult, p_335410_ -> ItemPredicateArgument.ERROR_MALFORMED_COMPONENT.createWithContext(p_335970_, this.id.toString(), p_335410_)
+            return (Predicate<ItemStack>)dataresult.getOrThrow(
+                p_335410_ -> ItemPredicateArgument.ERROR_MALFORMED_COMPONENT.createWithContext(p_335970_, this.id.toString(), p_335410_)
             );
         }
     }
@@ -227,8 +_,8 @@
 
         public Predicate<ItemStack> decode(ImmutableStringReader p_335497_, RegistryOps<Tag> p_335845_, Tag p_335487_) throws CommandSyntaxException {
             DataResult<? extends Predicate<ItemStack>> dataresult = this.type.parse(p_335845_, p_335487_);
-            return Util.getOrThrow(
-                dataresult, p_336129_ -> ItemPredicateArgument.ERROR_MALFORMED_PREDICATE.createWithContext(p_335497_, this.id.toString(), p_336129_)
+            return (Predicate<ItemStack>)dataresult.getOrThrow(
+                p_336129_ -> ItemPredicateArgument.ERROR_MALFORMED_PREDICATE.createWithContext(p_335497_, this.id.toString(), p_336129_)
             );
         }
     }
