--- a/net/minecraft/commands/arguments/item/ItemInput.java
+++ b/net/minecraft/commands/arguments/item/ItemInput.java
@@ -38,10 +_,10 @@
 
     public ItemStack createItemStack(int p_120981_, boolean p_120982_) throws CommandSyntaxException {
         ItemStack itemstack = new ItemStack(this.item, p_120981_);
+        itemstack.applyComponents(this.components);
         if (p_120982_ && p_120981_ > itemstack.getMaxStackSize()) {
             throw ERROR_STACK_TOO_BIG.create(this.getItemName(), itemstack.getMaxStackSize());
         } else {
-            itemstack.applyComponents(this.components);
             return itemstack;
         }
     }
@@ -60,10 +_,10 @@
 
     private String serializeComponents(HolderLookup.Provider p_331015_) {
         DynamicOps<Tag> dynamicops = p_331015_.createSerializationContext(NbtOps.INSTANCE);
-        return this.components.stream().flatMap(p_329808_ -> {
-            DataComponentType<?> datacomponenttype = p_329808_.type();
+        return this.components.stream().flatMap(p_339321_ -> {
+            DataComponentType<?> datacomponenttype = p_339321_.type();
             ResourceLocation resourcelocation = BuiltInRegistries.DATA_COMPONENT_TYPE.getKey(datacomponenttype);
-            Optional<Tag> optional = p_329808_.encodeValue(dynamicops).result();
+            Optional<Tag> optional = p_339321_.encodeValue(dynamicops).result();
             return resourcelocation != null && !optional.isEmpty() ? Stream.of(resourcelocation.toString() + "=" + optional.get()) : Stream.empty();
         }).collect(Collectors.joining(String.valueOf(',')));
     }
