--- a/net/minecraft/advancements/critereon/BlockPredicate.java
+++ b/net/minecraft/advancements/critereon/BlockPredicate.java
@@ -17,7 +_,6 @@
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.tags.TagKey;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.LevelReader;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.entity.BlockEntity;
@@ -26,12 +_,12 @@
 
 public record BlockPredicate(Optional<HolderSet<Block>> blocks, Optional<StatePropertiesPredicate> properties, Optional<NbtPredicate> nbt) {
     public static final Codec<BlockPredicate> CODEC = RecordCodecBuilder.create(
-        p_329679_ -> p_329679_.group(
-                    ExtraCodecs.strictOptionalField(RegistryCodecs.homogeneousList(Registries.BLOCK), "blocks").forGetter(BlockPredicate::blocks),
-                    ExtraCodecs.strictOptionalField(StatePropertiesPredicate.CODEC, "state").forGetter(BlockPredicate::properties),
-                    ExtraCodecs.strictOptionalField(NbtPredicate.CODEC, "nbt").forGetter(BlockPredicate::nbt)
+        p_337342_ -> p_337342_.group(
+                    RegistryCodecs.homogeneousList(Registries.BLOCK).optionalFieldOf("blocks").forGetter(BlockPredicate::blocks),
+                    StatePropertiesPredicate.CODEC.optionalFieldOf("state").forGetter(BlockPredicate::properties),
+                    NbtPredicate.CODEC.optionalFieldOf("nbt").forGetter(BlockPredicate::nbt)
                 )
-                .apply(p_329679_, BlockPredicate::new)
+                .apply(p_337342_, BlockPredicate::new)
     );
     public static final StreamCodec<RegistryFriendlyByteBuf, BlockPredicate> STREAM_CODEC = StreamCodec.composite(
         ByteBufCodecs.optional(ByteBufCodecs.holderSet(Registries.BLOCK)),
@@ -46,27 +_,21 @@
     public boolean matches(ServerLevel p_17915_, BlockPos p_17916_) {
         if (!p_17915_.isLoaded(p_17916_)) {
             return false;
-        } else if (!this.matchesState(p_17915_.getBlockState(p_17916_))) {
-            return false;
         } else {
-            return !this.nbt.isPresent() || matchesBlockEntity(p_17915_, p_17915_.getBlockEntity(p_17916_), this.nbt.get());
+            return !this.matchesState(p_17915_.getBlockState(p_17916_))
+                ? false
+                : !this.nbt.isPresent() || matchesBlockEntity(p_17915_, p_17915_.getBlockEntity(p_17916_), this.nbt.get());
         }
     }
 
     public boolean matches(BlockInWorld p_330448_) {
-        if (!this.matchesState(p_330448_.getState())) {
-            return false;
-        } else {
-            return !this.nbt.isPresent() || matchesBlockEntity(p_330448_.getLevel(), p_330448_.getEntity(), this.nbt.get());
-        }
+        return !this.matchesState(p_330448_.getState())
+            ? false
+            : !this.nbt.isPresent() || matchesBlockEntity(p_330448_.getLevel(), p_330448_.getEntity(), this.nbt.get());
     }
 
     private boolean matchesState(BlockState p_330349_) {
-        if (this.blocks.isPresent() && !p_330349_.is(this.blocks.get())) {
-            return false;
-        } else {
-            return !this.properties.isPresent() || this.properties.get().matches(p_330349_);
-        }
+        return this.blocks.isPresent() && !p_330349_.is(this.blocks.get()) ? false : !this.properties.isPresent() || this.properties.get().matches(p_330349_);
     }
 
     private static boolean matchesBlockEntity(LevelReader p_330457_, @Nullable BlockEntity p_332206_, NbtPredicate p_330422_) {
