--- a/net/minecraft/advancements/critereon/BredAnimalsTrigger.java
+++ b/net/minecraft/advancements/critereon/BredAnimalsTrigger.java
@@ -8,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.AgeableMob;
 import net.minecraft.world.entity.animal.Animal;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -33,13 +_,13 @@
         Optional<ContextAwarePredicate> child
     ) implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<BredAnimalsTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311402_ -> p_311402_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(BredAnimalsTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "parent").forGetter(BredAnimalsTrigger.TriggerInstance::parent),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "partner").forGetter(BredAnimalsTrigger.TriggerInstance::partner),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "child").forGetter(BredAnimalsTrigger.TriggerInstance::child)
+            p_337343_ -> p_337343_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(BredAnimalsTrigger.TriggerInstance::player),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("parent").forGetter(BredAnimalsTrigger.TriggerInstance::parent),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("partner").forGetter(BredAnimalsTrigger.TriggerInstance::partner),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("child").forGetter(BredAnimalsTrigger.TriggerInstance::child)
                     )
-                    .apply(p_311402_, BredAnimalsTrigger.TriggerInstance::new)
+                    .apply(p_337343_, BredAnimalsTrigger.TriggerInstance::new)
         );
 
         public static Criterion<BredAnimalsTrigger.TriggerInstance> bredAnimals() {
@@ -66,11 +_,9 @@
         }
 
         public boolean matches(LootContext p_18676_, LootContext p_18677_, @Nullable LootContext p_18678_) {
-            if (!this.child.isPresent() || p_18678_ != null && this.child.get().matches(p_18678_)) {
-                return matches(this.parent, p_18676_) && matches(this.partner, p_18677_) || matches(this.parent, p_18677_) && matches(this.partner, p_18676_);
-            } else {
-                return false;
-            }
+            return !this.child.isPresent() || p_18678_ != null && this.child.get().matches(p_18678_)
+                ? matches(this.parent, p_18676_) && matches(this.partner, p_18677_) || matches(this.parent, p_18677_) && matches(this.partner, p_18676_)
+                : false;
         }
 
         private static boolean matches(Optional<ContextAwarePredicate> p_298925_, LootContext p_298317_) {
