--- a/net/minecraft/advancements/critereon/BrewedPotionTrigger.java
+++ b/net/minecraft/advancements/critereon/BrewedPotionTrigger.java
@@ -9,7 +_,6 @@
 import net.minecraft.core.Holder;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.item.alchemy.Potion;
 
 public class BrewedPotionTrigger extends SimpleCriterionTrigger<BrewedPotionTrigger.TriggerInstance> {
@@ -25,12 +_,11 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, Optional<Holder<Potion>> potion)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<BrewedPotionTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_329680_ -> p_329680_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(BrewedPotionTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(BuiltInRegistries.POTION.holderByNameCodec(), "potion")
-                            .forGetter(BrewedPotionTrigger.TriggerInstance::potion)
+            p_337344_ -> p_337344_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(BrewedPotionTrigger.TriggerInstance::player),
+                        BuiltInRegistries.POTION.holderByNameCodec().optionalFieldOf("potion").forGetter(BrewedPotionTrigger.TriggerInstance::potion)
                     )
-                    .apply(p_329680_, BrewedPotionTrigger.TriggerInstance::new)
+                    .apply(p_337344_, BrewedPotionTrigger.TriggerInstance::new)
         );
 
         public static Criterion<BrewedPotionTrigger.TriggerInstance> brewedPotion() {
