--- a/net/minecraft/advancements/critereon/ChanneledLightningTrigger.java
+++ b/net/minecraft/advancements/critereon/ChanneledLightningTrigger.java
@@ -10,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.storage.loot.LootContext;
 
@@ -28,13 +_,14 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, List<ContextAwarePredicate> victims)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<ChanneledLightningTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_312247_ -> p_312247_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player")
-                            .forGetter(ChanneledLightningTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC.listOf(), "victims", List.of())
+            p_337346_ -> p_337346_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(ChanneledLightningTrigger.TriggerInstance::player),
+                        EntityPredicate.ADVANCEMENT_CODEC
+                            .listOf()
+                            .optionalFieldOf("victims", List.of())
                             .forGetter(ChanneledLightningTrigger.TriggerInstance::victims)
                     )
-                    .apply(p_312247_, ChanneledLightningTrigger.TriggerInstance::new)
+                    .apply(p_337346_, ChanneledLightningTrigger.TriggerInstance::new)
         );
 
         public static Criterion<ChanneledLightningTrigger.TriggerInstance> channeledLightning(EntityPredicate.Builder... p_298972_) {
@@ -43,10 +_,10 @@
         }
 
         public boolean matches(Collection<? extends LootContext> p_21745_) {
-            for(ContextAwarePredicate contextawarepredicate : this.victims) {
+            for (ContextAwarePredicate contextawarepredicate : this.victims) {
                 boolean flag = false;
 
-                for(LootContext lootcontext : p_21745_) {
+                for (LootContext lootcontext : p_21745_) {
                     if (contextawarepredicate.matches(lootcontext)) {
                         flag = true;
                         break;
