--- a/net/minecraft/advancements/critereon/ConsumeItemTrigger.java
+++ b/net/minecraft/advancements/critereon/ConsumeItemTrigger.java
@@ -7,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.ItemLike;
 
@@ -23,11 +_,11 @@
 
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, Optional<ItemPredicate> item) implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<ConsumeItemTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311405_ -> p_311405_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(ConsumeItemTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "item").forGetter(ConsumeItemTrigger.TriggerInstance::item)
+            p_337348_ -> p_337348_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(ConsumeItemTrigger.TriggerInstance::player),
+                        ItemPredicate.CODEC.optionalFieldOf("item").forGetter(ConsumeItemTrigger.TriggerInstance::item)
                     )
-                    .apply(p_311405_, ConsumeItemTrigger.TriggerInstance::new)
+                    .apply(p_337348_, ConsumeItemTrigger.TriggerInstance::new)
         );
 
         public static Criterion<ConsumeItemTrigger.TriggerInstance> usedItem() {
