--- a/net/minecraft/advancements/critereon/ContextAwarePredicate.java
+++ b/net/minecraft/advancements/critereon/ContextAwarePredicate.java
@@ -30,7 +_,7 @@
     }
 
     public void validate(ValidationContext p_312768_) {
-        for(int i = 0; i < this.conditions.size(); ++i) {
+        for (int i = 0; i < this.conditions.size(); i++) {
             LootItemCondition lootitemcondition = this.conditions.get(i);
             lootitemcondition.validate(p_312768_.forChild("[" + i + "]"));
         }
