--- a/net/minecraft/advancements/critereon/CuredZombieVillagerTrigger.java
+++ b/net/minecraft/advancements/critereon/CuredZombieVillagerTrigger.java
@@ -7,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.monster.Zombie;
 import net.minecraft.world.entity.npc.Villager;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -28,15 +_,12 @@
         Optional<ContextAwarePredicate> player, Optional<ContextAwarePredicate> zombie, Optional<ContextAwarePredicate> villager
     ) implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<CuredZombieVillagerTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311406_ -> p_311406_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player")
-                            .forGetter(CuredZombieVillagerTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "zombie")
-                            .forGetter(CuredZombieVillagerTrigger.TriggerInstance::zombie),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "villager")
-                            .forGetter(CuredZombieVillagerTrigger.TriggerInstance::villager)
+            p_337349_ -> p_337349_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(CuredZombieVillagerTrigger.TriggerInstance::player),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("zombie").forGetter(CuredZombieVillagerTrigger.TriggerInstance::zombie),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("villager").forGetter(CuredZombieVillagerTrigger.TriggerInstance::villager)
                     )
-                    .apply(p_311406_, CuredZombieVillagerTrigger.TriggerInstance::new)
+                    .apply(p_337349_, CuredZombieVillagerTrigger.TriggerInstance::new)
         );
 
         public static Criterion<CuredZombieVillagerTrigger.TriggerInstance> curedZombieVillager() {
@@ -45,11 +_,9 @@
         }
 
         public boolean matches(LootContext p_24300_, LootContext p_24301_) {
-            if (this.zombie.isPresent() && !this.zombie.get().matches(p_24300_)) {
-                return false;
-            } else {
-                return !this.villager.isPresent() || this.villager.get().matches(p_24301_);
-            }
+            return this.zombie.isPresent() && !this.zombie.get().matches(p_24300_)
+                ? false
+                : !this.villager.isPresent() || this.villager.get().matches(p_24301_);
         }
 
         @Override
