--- a/net/minecraft/advancements/critereon/DamageSourcePredicate.java
+++ b/net/minecraft/advancements/critereon/DamageSourcePredicate.java
@@ -9,20 +_,18 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.damagesource.DamageType;
 import net.minecraft.world.phys.Vec3;
 
 public record DamageSourcePredicate(List<TagPredicate<DamageType>> tags, Optional<EntityPredicate> directEntity, Optional<EntityPredicate> sourceEntity) {
     public static final Codec<DamageSourcePredicate> CODEC = RecordCodecBuilder.create(
-        p_298656_ -> p_298656_.group(
-                    ExtraCodecs.strictOptionalField(TagPredicate.codec(Registries.DAMAGE_TYPE).listOf(), "tags", List.of())
-                        .forGetter(DamageSourcePredicate::tags),
-                    ExtraCodecs.strictOptionalField(EntityPredicate.CODEC, "direct_entity").forGetter(DamageSourcePredicate::directEntity),
-                    ExtraCodecs.strictOptionalField(EntityPredicate.CODEC, "source_entity").forGetter(DamageSourcePredicate::sourceEntity)
+        p_337351_ -> p_337351_.group(
+                    TagPredicate.codec(Registries.DAMAGE_TYPE).listOf().optionalFieldOf("tags", List.of()).forGetter(DamageSourcePredicate::tags),
+                    EntityPredicate.CODEC.optionalFieldOf("direct_entity").forGetter(DamageSourcePredicate::directEntity),
+                    EntityPredicate.CODEC.optionalFieldOf("source_entity").forGetter(DamageSourcePredicate::sourceEntity)
                 )
-                .apply(p_298656_, DamageSourcePredicate::new)
+                .apply(p_337351_, DamageSourcePredicate::new)
     );
 
     public boolean matches(ServerPlayer p_25449_, DamageSource p_25450_) {
@@ -30,17 +_,15 @@
     }
 
     public boolean matches(ServerLevel p_25445_, Vec3 p_25446_, DamageSource p_25447_) {
-        for(TagPredicate<DamageType> tagpredicate : this.tags) {
+        for (TagPredicate<DamageType> tagpredicate : this.tags) {
             if (!tagpredicate.matches(p_25447_.typeHolder())) {
                 return false;
             }
         }
 
-        if (this.directEntity.isPresent() && !this.directEntity.get().matches(p_25445_, p_25446_, p_25447_.getDirectEntity())) {
-            return false;
-        } else {
-            return !this.sourceEntity.isPresent() || this.sourceEntity.get().matches(p_25445_, p_25446_, p_25447_.getEntity());
-        }
+        return this.directEntity.isPresent() && !this.directEntity.get().matches(p_25445_, p_25446_, p_25447_.getDirectEntity())
+            ? false
+            : !this.sourceEntity.isPresent() || this.sourceEntity.get().matches(p_25445_, p_25446_, p_25447_.getEntity());
     }
 
     public static class Builder {
