--- a/net/minecraft/advancements/critereon/EntityEquipmentPredicate.java
+++ b/net/minecraft/advancements/critereon/EntityEquipmentPredicate.java
@@ -7,7 +_,6 @@
 import javax.annotation.Nullable;
 import net.minecraft.core.HolderGetter;
 import net.minecraft.core.component.DataComponentPredicate;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.EquipmentSlot;
 import net.minecraft.world.entity.LivingEntity;
@@ -24,15 +_,15 @@
     Optional<ItemPredicate> offhand
 ) {
     public static final Codec<EntityEquipmentPredicate> CODEC = RecordCodecBuilder.create(
-        p_298479_ -> p_298479_.group(
-                    ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "head").forGetter(EntityEquipmentPredicate::head),
-                    ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "chest").forGetter(EntityEquipmentPredicate::chest),
-                    ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "legs").forGetter(EntityEquipmentPredicate::legs),
-                    ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "feet").forGetter(EntityEquipmentPredicate::feet),
-                    ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "mainhand").forGetter(EntityEquipmentPredicate::mainhand),
-                    ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "offhand").forGetter(EntityEquipmentPredicate::offhand)
+        p_337359_ -> p_337359_.group(
+                    ItemPredicate.CODEC.optionalFieldOf("head").forGetter(EntityEquipmentPredicate::head),
+                    ItemPredicate.CODEC.optionalFieldOf("chest").forGetter(EntityEquipmentPredicate::chest),
+                    ItemPredicate.CODEC.optionalFieldOf("legs").forGetter(EntityEquipmentPredicate::legs),
+                    ItemPredicate.CODEC.optionalFieldOf("feet").forGetter(EntityEquipmentPredicate::feet),
+                    ItemPredicate.CODEC.optionalFieldOf("mainhand").forGetter(EntityEquipmentPredicate::mainhand),
+                    ItemPredicate.CODEC.optionalFieldOf("offhand").forGetter(EntityEquipmentPredicate::offhand)
                 )
-                .apply(p_298479_, EntityEquipmentPredicate::new)
+                .apply(p_337359_, EntityEquipmentPredicate::new)
     );
 
     public static EntityEquipmentPredicate captainPredicate(HolderGetter<BannerPattern> p_332749_) {
@@ -55,10 +_,10 @@
                 return false;
             } else if (this.feet.isPresent() && !this.feet.get().matches(livingentity.getItemBySlot(EquipmentSlot.FEET))) {
                 return false;
-            } else if (this.mainhand.isPresent() && !this.mainhand.get().matches(livingentity.getItemBySlot(EquipmentSlot.MAINHAND))) {
-                return false;
             } else {
-                return !this.offhand.isPresent() || this.offhand.get().matches(livingentity.getItemBySlot(EquipmentSlot.OFFHAND));
+                return this.mainhand.isPresent() && !this.mainhand.get().matches(livingentity.getItemBySlot(EquipmentSlot.MAINHAND))
+                    ? false
+                    : !this.offhand.isPresent() || this.offhand.get().matches(livingentity.getItemBySlot(EquipmentSlot.OFFHAND));
             }
         } else {
             return false;
