--- a/net/minecraft/advancements/critereon/FluidPredicate.java
+++ b/net/minecraft/advancements/critereon/FluidPredicate.java
@@ -9,17 +_,16 @@
 import net.minecraft.core.RegistryCodecs;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.material.FluidState;
 
 public record FluidPredicate(Optional<HolderSet<Fluid>> fluids, Optional<StatePropertiesPredicate> properties) {
     public static final Codec<FluidPredicate> CODEC = RecordCodecBuilder.create(
-        p_329681_ -> p_329681_.group(
-                    ExtraCodecs.strictOptionalField(RegistryCodecs.homogeneousList(Registries.FLUID), "fluids").forGetter(FluidPredicate::fluids),
-                    ExtraCodecs.strictOptionalField(StatePropertiesPredicate.CODEC, "state").forGetter(FluidPredicate::properties)
+        p_337366_ -> p_337366_.group(
+                    RegistryCodecs.homogeneousList(Registries.FLUID).optionalFieldOf("fluids").forGetter(FluidPredicate::fluids),
+                    StatePropertiesPredicate.CODEC.optionalFieldOf("state").forGetter(FluidPredicate::properties)
                 )
-                .apply(p_329681_, FluidPredicate::new)
+                .apply(p_337366_, FluidPredicate::new)
     );
 
     public boolean matches(ServerLevel p_41105_, BlockPos p_41106_) {
@@ -27,11 +_,9 @@
             return false;
         } else {
             FluidState fluidstate = p_41105_.getFluidState(p_41106_);
-            if (this.fluids.isPresent() && !fluidstate.is(this.fluids.get())) {
-                return false;
-            } else {
-                return !this.properties.isPresent() || this.properties.get().matches(fluidstate);
-            }
+            return this.fluids.isPresent() && !fluidstate.is(this.fluids.get())
+                ? false
+                : !this.properties.isPresent() || this.properties.get().matches(fluidstate);
         }
     }
 
