--- a/net/minecraft/advancements/critereon/ItemUsedOnLocationTrigger.java
+++ b/net/minecraft/advancements/critereon/ItemUsedOnLocationTrigger.java
@@ -10,7 +_,6 @@
 import net.minecraft.core.BlockPos;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
@@ -45,12 +_,11 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, Optional<ContextAwarePredicate> location)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<ItemUsedOnLocationTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311428_ -> p_311428_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player")
-                            .forGetter(ItemUsedOnLocationTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(ContextAwarePredicate.CODEC, "location").forGetter(ItemUsedOnLocationTrigger.TriggerInstance::location)
+            p_337372_ -> p_337372_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(ItemUsedOnLocationTrigger.TriggerInstance::player),
+                        ContextAwarePredicate.CODEC.optionalFieldOf("location").forGetter(ItemUsedOnLocationTrigger.TriggerInstance::location)
                     )
-                    .apply(p_311428_, ItemUsedOnLocationTrigger.TriggerInstance::new)
+                    .apply(p_337372_, ItemUsedOnLocationTrigger.TriggerInstance::new)
         );
 
         public static Criterion<ItemUsedOnLocationTrigger.TriggerInstance> placedBlock(Block p_301158_) {
@@ -63,7 +_,7 @@
 
         public static Criterion<ItemUsedOnLocationTrigger.TriggerInstance> placedBlock(LootItemCondition.Builder... p_301013_) {
             ContextAwarePredicate contextawarepredicate = ContextAwarePredicate.create(
-                Arrays.stream(p_301013_).map(LootItemCondition.Builder::build).toArray(p_286827_ -> new LootItemCondition[p_286827_])
+                Arrays.stream(p_301013_).map(LootItemCondition.Builder::build).toArray(LootItemCondition[]::new)
             );
             return CriteriaTriggers.PLACED_BLOCK
                 .createCriterion(new ItemUsedOnLocationTrigger.TriggerInstance(Optional.empty(), Optional.of(contextawarepredicate)));
