--- a/net/minecraft/advancements/critereon/KilledTrigger.java
+++ b/net/minecraft/advancements/critereon/KilledTrigger.java
@@ -7,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -27,12 +_,12 @@
         Optional<ContextAwarePredicate> player, Optional<ContextAwarePredicate> entityPredicate, Optional<DamageSourcePredicate> killingBlow
     ) implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<KilledTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311429_ -> p_311429_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(KilledTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "entity").forGetter(KilledTrigger.TriggerInstance::entityPredicate),
-                        ExtraCodecs.strictOptionalField(DamageSourcePredicate.CODEC, "killing_blow").forGetter(KilledTrigger.TriggerInstance::killingBlow)
+            p_337374_ -> p_337374_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(KilledTrigger.TriggerInstance::player),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("entity").forGetter(KilledTrigger.TriggerInstance::entityPredicate),
+                        DamageSourcePredicate.CODEC.optionalFieldOf("killing_blow").forGetter(KilledTrigger.TriggerInstance::killingBlow)
                     )
-                    .apply(p_311429_, KilledTrigger.TriggerInstance::new)
+                    .apply(p_337374_, KilledTrigger.TriggerInstance::new)
         );
 
         public static Criterion<KilledTrigger.TriggerInstance> playerKilledEntity(Optional<EntityPredicate> p_298999_) {
@@ -119,11 +_,9 @@
         }
 
         public boolean matches(ServerPlayer p_48131_, LootContext p_48132_, DamageSource p_48133_) {
-            if (this.killingBlow.isPresent() && !this.killingBlow.get().matches(p_48131_, p_48133_)) {
-                return false;
-            } else {
-                return this.entityPredicate.isEmpty() || this.entityPredicate.get().matches(p_48132_);
-            }
+            return this.killingBlow.isPresent() && !this.killingBlow.get().matches(p_48131_, p_48133_)
+                ? false
+                : this.entityPredicate.isEmpty() || this.entityPredicate.get().matches(p_48132_);
         }
 
         @Override
