--- a/net/minecraft/advancements/critereon/KilledByCrossbowTrigger.java
+++ b/net/minecraft/advancements/critereon/KilledByCrossbowTrigger.java
@@ -13,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -28,7 +_,7 @@
         List<LootContext> list = Lists.newArrayList();
         Set<EntityType<?>> set = Sets.newHashSet();
 
-        for(Entity entity : p_46873_) {
+        for (Entity entity : p_46873_) {
             set.add(entity.getType());
             list.add(EntityPredicate.createContext(p_46872_, entity));
         }
@@ -39,14 +_,17 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, List<ContextAwarePredicate> victims, MinMaxBounds.Ints uniqueEntityTypes)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<KilledByCrossbowTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_312078_ -> p_312078_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(KilledByCrossbowTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC.listOf(), "victims", List.of())
+            p_337373_ -> p_337373_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(KilledByCrossbowTrigger.TriggerInstance::player),
+                        EntityPredicate.ADVANCEMENT_CODEC
+                            .listOf()
+                            .optionalFieldOf("victims", List.of())
                             .forGetter(KilledByCrossbowTrigger.TriggerInstance::victims),
-                        ExtraCodecs.strictOptionalField(MinMaxBounds.Ints.CODEC, "unique_entity_types", MinMaxBounds.Ints.ANY)
+                        MinMaxBounds.Ints.CODEC
+                            .optionalFieldOf("unique_entity_types", MinMaxBounds.Ints.ANY)
                             .forGetter(KilledByCrossbowTrigger.TriggerInstance::uniqueEntityTypes)
                     )
-                    .apply(p_312078_, KilledByCrossbowTrigger.TriggerInstance::new)
+                    .apply(p_337373_, KilledByCrossbowTrigger.TriggerInstance::new)
         );
 
         public static Criterion<KilledByCrossbowTrigger.TriggerInstance> crossbowKilled(EntityPredicate.Builder... p_301077_) {
@@ -62,11 +_,11 @@
             if (!this.victims.isEmpty()) {
                 List<LootContext> list = Lists.newArrayList(p_46898_);
 
-                for(ContextAwarePredicate contextawarepredicate : this.victims) {
+                for (ContextAwarePredicate contextawarepredicate : this.victims) {
                     boolean flag = false;
                     Iterator<LootContext> iterator = list.iterator();
 
-                    while(iterator.hasNext()) {
+                    while (iterator.hasNext()) {
                         LootContext lootcontext = iterator.next();
                         if (contextawarepredicate.matches(lootcontext)) {
                             iterator.remove();
