--- a/net/minecraft/advancements/critereon/LevitationTrigger.java
+++ b/net/minecraft/advancements/critereon/LevitationTrigger.java
@@ -7,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.phys.Vec3;
 
 public class LevitationTrigger extends SimpleCriterionTrigger<LevitationTrigger.TriggerInstance> {
@@ -23,13 +_,12 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, Optional<DistancePredicate> distance, MinMaxBounds.Ints duration)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<LevitationTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311430_ -> p_311430_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(LevitationTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(DistancePredicate.CODEC, "distance").forGetter(LevitationTrigger.TriggerInstance::distance),
-                        ExtraCodecs.strictOptionalField(MinMaxBounds.Ints.CODEC, "duration", MinMaxBounds.Ints.ANY)
-                            .forGetter(LevitationTrigger.TriggerInstance::duration)
+            p_337375_ -> p_337375_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(LevitationTrigger.TriggerInstance::player),
+                        DistancePredicate.CODEC.optionalFieldOf("distance").forGetter(LevitationTrigger.TriggerInstance::distance),
+                        MinMaxBounds.Ints.CODEC.optionalFieldOf("duration", MinMaxBounds.Ints.ANY).forGetter(LevitationTrigger.TriggerInstance::duration)
                     )
-                    .apply(p_311430_, LevitationTrigger.TriggerInstance::new)
+                    .apply(p_337375_, LevitationTrigger.TriggerInstance::new)
         );
 
         public static Criterion<LevitationTrigger.TriggerInstance> levitated(DistancePredicate p_49145_) {
@@ -38,12 +_,10 @@
         }
 
         public boolean matches(ServerPlayer p_49141_, Vec3 p_49142_, int p_49143_) {
-            if (this.distance.isPresent()
-                && !this.distance.get().matches(p_49142_.x, p_49142_.y, p_49142_.z, p_49141_.getX(), p_49141_.getY(), p_49141_.getZ())) {
-                return false;
-            } else {
-                return this.duration.matches(p_49143_);
-            }
+            return this.distance.isPresent()
+                    && !this.distance.get().matches(p_49142_.x, p_49142_.y, p_49142_.z, p_49141_.getX(), p_49141_.getY(), p_49141_.getZ())
+                ? false
+                : this.duration.matches(p_49143_);
         }
     }
 }
