--- a/net/minecraft/advancements/critereon/LightningBoltPredicate.java
+++ b/net/minecraft/advancements/critereon/LightningBoltPredicate.java
@@ -6,19 +_,17 @@
 import java.util.Optional;
 import javax.annotation.Nullable;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.LightningBolt;
 import net.minecraft.world.phys.Vec3;
 
 public record LightningBoltPredicate(MinMaxBounds.Ints blocksSetOnFire, Optional<EntityPredicate> entityStruck) implements EntitySubPredicate {
     public static final MapCodec<LightningBoltPredicate> CODEC = RecordCodecBuilder.mapCodec(
-        p_298977_ -> p_298977_.group(
-                    ExtraCodecs.strictOptionalField(MinMaxBounds.Ints.CODEC, "blocks_set_on_fire", MinMaxBounds.Ints.ANY)
-                        .forGetter(LightningBoltPredicate::blocksSetOnFire),
-                    ExtraCodecs.strictOptionalField(EntityPredicate.CODEC, "entity_struck").forGetter(LightningBoltPredicate::entityStruck)
+        p_337377_ -> p_337377_.group(
+                    MinMaxBounds.Ints.CODEC.optionalFieldOf("blocks_set_on_fire", MinMaxBounds.Ints.ANY).forGetter(LightningBoltPredicate::blocksSetOnFire),
+                    EntityPredicate.CODEC.optionalFieldOf("entity_struck").forGetter(LightningBoltPredicate::entityStruck)
                 )
-                .apply(p_298977_, LightningBoltPredicate::new)
+                .apply(p_337377_, LightningBoltPredicate::new)
     );
 
     public static LightningBoltPredicate blockSetOnFire(MinMaxBounds.Ints p_299013_) {
@@ -32,15 +_,12 @@
 
     @Override
     public boolean matches(Entity p_299034_, ServerLevel p_299101_, @Nullable Vec3 p_298929_) {
-        if (!(p_299034_ instanceof LightningBolt)) {
-            return false;
-        } else {
-            LightningBolt lightningbolt = (LightningBolt)p_299034_;
-            return this.blocksSetOnFire.matches(lightningbolt.getBlocksSetOnFire())
+        return !(p_299034_ instanceof LightningBolt lightningbolt)
+            ? false
+            : this.blocksSetOnFire.matches(lightningbolt.getBlocksSetOnFire())
                 && (
                     this.entityStruck.isEmpty()
                         || lightningbolt.getHitEntities().anyMatch(p_298360_ -> this.entityStruck.get().matches(p_299101_, p_298929_, p_298360_))
                 );
-        }
     }
 }
