--- a/net/minecraft/advancements/critereon/LightningStrikeTrigger.java
+++ b/net/minecraft/advancements/critereon/LightningStrikeTrigger.java
@@ -9,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.LightningBolt;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -30,14 +_,12 @@
         Optional<ContextAwarePredicate> player, Optional<ContextAwarePredicate> lightning, Optional<ContextAwarePredicate> bystander
     ) implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<LightningStrikeTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311431_ -> p_311431_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(LightningStrikeTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "lightning")
-                            .forGetter(LightningStrikeTrigger.TriggerInstance::lightning),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "bystander")
-                            .forGetter(LightningStrikeTrigger.TriggerInstance::bystander)
+            p_337378_ -> p_337378_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(LightningStrikeTrigger.TriggerInstance::player),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("lightning").forGetter(LightningStrikeTrigger.TriggerInstance::lightning),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("bystander").forGetter(LightningStrikeTrigger.TriggerInstance::bystander)
                     )
-                    .apply(p_311431_, LightningStrikeTrigger.TriggerInstance::new)
+                    .apply(p_337378_, LightningStrikeTrigger.TriggerInstance::new)
         );
 
         public static Criterion<LightningStrikeTrigger.TriggerInstance> lightningStrike(
@@ -48,11 +_,9 @@
         }
 
         public boolean matches(LootContext p_153419_, List<LootContext> p_153420_) {
-            if (this.lightning.isPresent() && !this.lightning.get().matches(p_153419_)) {
-                return false;
-            } else {
-                return !this.bystander.isPresent() || !p_153420_.stream().noneMatch(this.bystander.get()::matches);
-            }
+            return this.lightning.isPresent() && !this.lightning.get().matches(p_153419_)
+                ? false
+                : !this.bystander.isPresent() || !p_153420_.stream().noneMatch(this.bystander.get()::matches);
         }
 
         @Override
