--- a/net/minecraft/advancements/critereon/LocationPredicate.java
+++ b/net/minecraft/advancements/critereon/LocationPredicate.java
@@ -11,7 +_,6 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.biome.Biome;
 import net.minecraft.world.level.block.CampfireBlock;
@@ -29,15 +_,14 @@
 ) {
     public static final Codec<LocationPredicate> CODEC = RecordCodecBuilder.create(
         p_297907_ -> p_297907_.group(
-                    ExtraCodecs.strictOptionalField(LocationPredicate.PositionPredicate.CODEC, "position").forGetter(LocationPredicate::position),
-                    ExtraCodecs.strictOptionalField(RegistryCodecs.homogeneousList(Registries.BIOME), "biomes").forGetter(LocationPredicate::biomes),
-                    ExtraCodecs.strictOptionalField(RegistryCodecs.homogeneousList(Registries.STRUCTURE), "structures")
-                        .forGetter(LocationPredicate::structures),
-                    ExtraCodecs.strictOptionalField(ResourceKey.codec(Registries.DIMENSION), "dimension").forGetter(LocationPredicate::dimension),
-                    ExtraCodecs.strictOptionalField(Codec.BOOL, "smokey").forGetter(LocationPredicate::smokey),
-                    ExtraCodecs.strictOptionalField(LightPredicate.CODEC, "light").forGetter(LocationPredicate::light),
-                    ExtraCodecs.strictOptionalField(BlockPredicate.CODEC, "block").forGetter(LocationPredicate::block),
-                    ExtraCodecs.strictOptionalField(FluidPredicate.CODEC, "fluid").forGetter(LocationPredicate::fluid)
+                    LocationPredicate.PositionPredicate.CODEC.optionalFieldOf("position").forGetter(LocationPredicate::position),
+                    RegistryCodecs.homogeneousList(Registries.BIOME).optionalFieldOf("biomes").forGetter(LocationPredicate::biomes),
+                    RegistryCodecs.homogeneousList(Registries.STRUCTURE).optionalFieldOf("structures").forGetter(LocationPredicate::structures),
+                    ResourceKey.codec(Registries.DIMENSION).optionalFieldOf("dimension").forGetter(LocationPredicate::dimension),
+                    Codec.BOOL.optionalFieldOf("smokey").forGetter(LocationPredicate::smokey),
+                    LightPredicate.CODEC.optionalFieldOf("light").forGetter(LocationPredicate::light),
+                    BlockPredicate.CODEC.optionalFieldOf("block").forGetter(LocationPredicate::block),
+                    FluidPredicate.CODEC.optionalFieldOf("fluid").forGetter(LocationPredicate::fluid)
                 )
                 .apply(p_297907_, LocationPredicate::new)
     );
@@ -55,10 +_,10 @@
                     if (!this.smokey.isPresent() || flag && this.smokey.get() == CampfireBlock.isSmokeyPos(p_52618_, blockpos)) {
                         if (this.light.isPresent() && !this.light.get().matches(p_52618_, blockpos)) {
                             return false;
-                        } else if (this.block.isPresent() && !this.block.get().matches(p_52618_, blockpos)) {
-                            return false;
                         } else {
-                            return !this.fluid.isPresent() || this.fluid.get().matches(p_52618_, blockpos);
+                            return this.block.isPresent() && !this.block.get().matches(p_52618_, blockpos)
+                                ? false
+                                : !this.fluid.isPresent() || this.fluid.get().matches(p_52618_, blockpos);
                         }
                     } else {
                         return false;
@@ -162,15 +_,12 @@
 
     static record PositionPredicate(MinMaxBounds.Doubles x, MinMaxBounds.Doubles y, MinMaxBounds.Doubles z) {
         public static final Codec<LocationPredicate.PositionPredicate> CODEC = RecordCodecBuilder.create(
-            p_299107_ -> p_299107_.group(
-                        ExtraCodecs.strictOptionalField(MinMaxBounds.Doubles.CODEC, "x", MinMaxBounds.Doubles.ANY)
-                            .forGetter(LocationPredicate.PositionPredicate::x),
-                        ExtraCodecs.strictOptionalField(MinMaxBounds.Doubles.CODEC, "y", MinMaxBounds.Doubles.ANY)
-                            .forGetter(LocationPredicate.PositionPredicate::y),
-                        ExtraCodecs.strictOptionalField(MinMaxBounds.Doubles.CODEC, "z", MinMaxBounds.Doubles.ANY)
-                            .forGetter(LocationPredicate.PositionPredicate::z)
+            p_337379_ -> p_337379_.group(
+                        MinMaxBounds.Doubles.CODEC.optionalFieldOf("x", MinMaxBounds.Doubles.ANY).forGetter(LocationPredicate.PositionPredicate::x),
+                        MinMaxBounds.Doubles.CODEC.optionalFieldOf("y", MinMaxBounds.Doubles.ANY).forGetter(LocationPredicate.PositionPredicate::y),
+                        MinMaxBounds.Doubles.CODEC.optionalFieldOf("z", MinMaxBounds.Doubles.ANY).forGetter(LocationPredicate.PositionPredicate::z)
                     )
-                    .apply(p_299107_, LocationPredicate.PositionPredicate::new)
+                    .apply(p_337379_, LocationPredicate.PositionPredicate::new)
         );
 
         static Optional<LocationPredicate.PositionPredicate> of(MinMaxBounds.Doubles p_298771_, MinMaxBounds.Doubles p_298418_, MinMaxBounds.Doubles p_299133_) {
