--- a/net/minecraft/advancements/critereon/LootTableTrigger.java
+++ b/net/minecraft/advancements/critereon/LootTableTrigger.java
@@ -9,7 +_,6 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.storage.loot.LootTable;
 
 public class LootTableTrigger extends SimpleCriterionTrigger<LootTableTrigger.TriggerInstance> {
@@ -25,11 +_,11 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, ResourceKey<LootTable> lootTable)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<LootTableTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_335161_ -> p_335161_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(LootTableTrigger.TriggerInstance::player),
+            p_337380_ -> p_337380_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(LootTableTrigger.TriggerInstance::player),
                         ResourceKey.codec(Registries.LOOT_TABLE).fieldOf("loot_table").forGetter(LootTableTrigger.TriggerInstance::lootTable)
                     )
-                    .apply(p_335161_, LootTableTrigger.TriggerInstance::new)
+                    .apply(p_337380_, LootTableTrigger.TriggerInstance::new)
         );
 
         public static Criterion<LootTableTrigger.TriggerInstance> lootTableUsed(ResourceKey<LootTable> p_335395_) {
