--- a/net/minecraft/advancements/critereon/PickedUpItemTrigger.java
+++ b/net/minecraft/advancements/critereon/PickedUpItemTrigger.java
@@ -8,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -27,12 +_,12 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, Optional<ItemPredicate> item, Optional<ContextAwarePredicate> entity)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<PickedUpItemTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311432_ -> p_311432_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(PickedUpItemTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(ItemPredicate.CODEC, "item").forGetter(PickedUpItemTrigger.TriggerInstance::item),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "entity").forGetter(PickedUpItemTrigger.TriggerInstance::entity)
+            p_337385_ -> p_337385_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(PickedUpItemTrigger.TriggerInstance::player),
+                        ItemPredicate.CODEC.optionalFieldOf("item").forGetter(PickedUpItemTrigger.TriggerInstance::item),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("entity").forGetter(PickedUpItemTrigger.TriggerInstance::entity)
                     )
-                    .apply(p_311432_, PickedUpItemTrigger.TriggerInstance::new)
+                    .apply(p_337385_, PickedUpItemTrigger.TriggerInstance::new)
         );
 
         public static Criterion<PickedUpItemTrigger.TriggerInstance> thrownItemPickedUpByEntity(
@@ -49,11 +_,7 @@
         }
 
         public boolean matches(ServerPlayer p_221323_, ItemStack p_221324_, LootContext p_221325_) {
-            if (this.item.isPresent() && !this.item.get().matches(p_221324_)) {
-                return false;
-            } else {
-                return !this.entity.isPresent() || this.entity.get().matches(p_221325_);
-            }
+            return this.item.isPresent() && !this.item.get().matches(p_221324_) ? false : !this.entity.isPresent() || this.entity.get().matches(p_221325_);
         }
 
         @Override
