--- a/net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
+++ b/net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
@@ -7,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -26,12 +_,12 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, Optional<DamagePredicate> damage, Optional<ContextAwarePredicate> entity)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<PlayerHurtEntityTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311433_ -> p_311433_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(PlayerHurtEntityTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(DamagePredicate.CODEC, "damage").forGetter(PlayerHurtEntityTrigger.TriggerInstance::damage),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "entity").forGetter(PlayerHurtEntityTrigger.TriggerInstance::entity)
+            p_337386_ -> p_337386_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(PlayerHurtEntityTrigger.TriggerInstance::player),
+                        DamagePredicate.CODEC.optionalFieldOf("damage").forGetter(PlayerHurtEntityTrigger.TriggerInstance::damage),
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("entity").forGetter(PlayerHurtEntityTrigger.TriggerInstance::entity)
                     )
-                    .apply(p_311433_, PlayerHurtEntityTrigger.TriggerInstance::new)
+                    .apply(p_337386_, PlayerHurtEntityTrigger.TriggerInstance::new)
         );
 
         public static Criterion<PlayerHurtEntityTrigger.TriggerInstance> playerHurtEntity() {
@@ -69,11 +_,9 @@
         }
 
         public boolean matches(ServerPlayer p_60143_, LootContext p_60144_, DamageSource p_60145_, float p_60146_, float p_60147_, boolean p_60148_) {
-            if (this.damage.isPresent() && !this.damage.get().matches(p_60143_, p_60145_, p_60146_, p_60147_, p_60148_)) {
-                return false;
-            } else {
-                return !this.entity.isPresent() || this.entity.get().matches(p_60144_);
-            }
+            return this.damage.isPresent() && !this.damage.get().matches(p_60143_, p_60145_, p_60146_, p_60147_, p_60148_)
+                ? false
+                : !this.entity.isPresent() || this.entity.get().matches(p_60144_);
         }
 
         @Override
