--- a/net/minecraft/advancements/critereon/RecipeUnlockedTrigger.java
+++ b/net/minecraft/advancements/critereon/RecipeUnlockedTrigger.java
@@ -8,7 +_,6 @@
 import net.minecraft.advancements.Criterion;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.item.crafting.RecipeHolder;
 
 public class RecipeUnlockedTrigger extends SimpleCriterionTrigger<RecipeUnlockedTrigger.TriggerInstance> {
@@ -27,11 +_,11 @@
 
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, ResourceLocation recipe) implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<RecipeUnlockedTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_312775_ -> p_312775_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(RecipeUnlockedTrigger.TriggerInstance::player),
+            p_337392_ -> p_337392_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(RecipeUnlockedTrigger.TriggerInstance::player),
                         ResourceLocation.CODEC.fieldOf("recipe").forGetter(RecipeUnlockedTrigger.TriggerInstance::recipe)
                     )
-                    .apply(p_312775_, RecipeUnlockedTrigger.TriggerInstance::new)
+                    .apply(p_337392_, RecipeUnlockedTrigger.TriggerInstance::new)
         );
 
         public boolean matches(RecipeHolder<?> p_301295_) {
