--- a/net/minecraft/advancements/critereon/SlimePredicate.java
+++ b/net/minecraft/advancements/critereon/SlimePredicate.java
@@ -5,15 +_,14 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import javax.annotation.Nullable;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.monster.Slime;
 import net.minecraft.world.phys.Vec3;
 
 public record SlimePredicate(MinMaxBounds.Ints size) implements EntitySubPredicate {
     public static final MapCodec<SlimePredicate> CODEC = RecordCodecBuilder.mapCodec(
-        p_298436_ -> p_298436_.group(ExtraCodecs.strictOptionalField(MinMaxBounds.Ints.CODEC, "size", MinMaxBounds.Ints.ANY).forGetter(SlimePredicate::size))
-                .apply(p_298436_, SlimePredicate::new)
+        p_337395_ -> p_337395_.group(MinMaxBounds.Ints.CODEC.optionalFieldOf("size", MinMaxBounds.Ints.ANY).forGetter(SlimePredicate::size))
+                .apply(p_337395_, SlimePredicate::new)
     );
 
     public static SlimePredicate sized(MinMaxBounds.Ints p_223427_) {
