--- a/net/minecraft/advancements/critereon/SlotsPredicate.java
+++ b/net/minecraft/advancements/critereon/SlotsPredicate.java
@@ -14,7 +_,7 @@
         .xmap(SlotsPredicate::new, SlotsPredicate::slots);
 
     public boolean matches(Entity p_332796_) {
-        for(Entry<SlotRange, ItemPredicate> entry : this.slots.entrySet()) {
+        for (Entry<SlotRange, ItemPredicate> entry : this.slots.entrySet()) {
             if (!matchSlots(p_332796_, entry.getValue(), entry.getKey().slots())) {
                 return false;
             }
@@ -24,7 +_,7 @@
     }
 
     private static boolean matchSlots(Entity p_332687_, ItemPredicate p_332814_, IntList p_332676_) {
-        for(int i = 0; i < p_332676_.size(); ++i) {
+        for (int i = 0; i < p_332676_.size(); i++) {
             int j = p_332676_.getInt(i);
             SlotAccess slotaccess = p_332687_.getSlot(j);
             if (p_332814_.matches(slotaccess.get())) {
