--- a/net/minecraft/advancements/critereon/StartRidingTrigger.java
+++ b/net/minecraft/advancements/critereon/StartRidingTrigger.java
@@ -7,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 
 public class StartRidingTrigger extends SimpleCriterionTrigger<StartRidingTrigger.TriggerInstance> {
     @Override
@@ -21,10 +_,8 @@
 
     public static record TriggerInstance(Optional<ContextAwarePredicate> player) implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<StartRidingTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_312222_ -> p_312222_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(StartRidingTrigger.TriggerInstance::player)
-                    )
-                    .apply(p_312222_, StartRidingTrigger.TriggerInstance::new)
+            p_337396_ -> p_337396_.group(EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(StartRidingTrigger.TriggerInstance::player))
+                    .apply(p_337396_, StartRidingTrigger.TriggerInstance::new)
         );
 
         public static Criterion<StartRidingTrigger.TriggerInstance> playerStartsRiding(EntityPredicate.Builder p_160402_) {
