--- a/net/minecraft/advancements/critereon/TargetBlockTrigger.java
+++ b/net/minecraft/advancements/critereon/TargetBlockTrigger.java
@@ -7,7 +_,6 @@
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.advancements.Criterion;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.phys.Vec3;
@@ -26,14 +_,14 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, MinMaxBounds.Ints signalStrength, Optional<ContextAwarePredicate> projectile)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<TargetBlockTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_311448_ -> p_311448_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(TargetBlockTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(MinMaxBounds.Ints.CODEC, "signal_strength", MinMaxBounds.Ints.ANY)
+            p_337400_ -> p_337400_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(TargetBlockTrigger.TriggerInstance::player),
+                        MinMaxBounds.Ints.CODEC
+                            .optionalFieldOf("signal_strength", MinMaxBounds.Ints.ANY)
                             .forGetter(TargetBlockTrigger.TriggerInstance::signalStrength),
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "projectile")
-                            .forGetter(TargetBlockTrigger.TriggerInstance::projectile)
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("projectile").forGetter(TargetBlockTrigger.TriggerInstance::projectile)
                     )
-                    .apply(p_311448_, TargetBlockTrigger.TriggerInstance::new)
+                    .apply(p_337400_, TargetBlockTrigger.TriggerInstance::new)
         );
 
         public static Criterion<TargetBlockTrigger.TriggerInstance> targetHit(MinMaxBounds.Ints p_286700_, Optional<ContextAwarePredicate> p_298563_) {
@@ -41,11 +_,7 @@
         }
 
         public boolean matches(LootContext p_70242_, Vec3 p_70243_, int p_70244_) {
-            if (!this.signalStrength.matches(p_70244_)) {
-                return false;
-            } else {
-                return !this.projectile.isPresent() || this.projectile.get().matches(p_70242_);
-            }
+            return !this.signalStrength.matches(p_70244_) ? false : !this.projectile.isPresent() || this.projectile.get().matches(p_70242_);
         }
 
         @Override
