--- a/net/minecraft/advancements/critereon/UsedEnderEyeTrigger.java
+++ b/net/minecraft/advancements/critereon/UsedEnderEyeTrigger.java
@@ -6,7 +_,6 @@
 import java.util.Optional;
 import net.minecraft.core.BlockPos;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.ExtraCodecs;
 
 public class UsedEnderEyeTrigger extends SimpleCriterionTrigger<UsedEnderEyeTrigger.TriggerInstance> {
     @Override
@@ -24,12 +_,13 @@
     public static record TriggerInstance(Optional<ContextAwarePredicate> player, MinMaxBounds.Doubles distance)
         implements SimpleCriterionTrigger.SimpleInstance {
         public static final Codec<UsedEnderEyeTrigger.TriggerInstance> CODEC = RecordCodecBuilder.create(
-            p_312304_ -> p_312304_.group(
-                        ExtraCodecs.strictOptionalField(EntityPredicate.ADVANCEMENT_CODEC, "player").forGetter(UsedEnderEyeTrigger.TriggerInstance::player),
-                        ExtraCodecs.strictOptionalField(MinMaxBounds.Doubles.CODEC, "distance", MinMaxBounds.Doubles.ANY)
+            p_337402_ -> p_337402_.group(
+                        EntityPredicate.ADVANCEMENT_CODEC.optionalFieldOf("player").forGetter(UsedEnderEyeTrigger.TriggerInstance::player),
+                        MinMaxBounds.Doubles.CODEC
+                            .optionalFieldOf("distance", MinMaxBounds.Doubles.ANY)
                             .forGetter(UsedEnderEyeTrigger.TriggerInstance::distance)
                     )
-                    .apply(p_312304_, UsedEnderEyeTrigger.TriggerInstance::new)
+                    .apply(p_337402_, UsedEnderEyeTrigger.TriggerInstance::new)
         );
 
         public boolean matches(double p_73952_) {
