--- a/net/minecraft/advancements/critereon/WrappedMinMaxBounds.java
+++ b/net/minecraft/advancements/critereon/WrappedMinMaxBounds.java
@@ -35,10 +_,8 @@
     public boolean matches(float p_164420_) {
         if (this.min != null && this.max != null && this.min > this.max && this.min > p_164420_ && this.max < p_164420_) {
             return false;
-        } else if (this.min != null && this.min > p_164420_) {
-            return false;
         } else {
-            return this.max == null || !(this.max < p_164420_);
+            return this.min != null && this.min > p_164420_ ? false : this.max == null || !(this.max < p_164420_);
         }
     }
 
@@ -49,10 +_,8 @@
             && (double)(this.min * this.min) > p_164401_
             && (double)(this.max * this.max) < p_164401_) {
             return false;
-        } else if (this.min != null && (double)(this.min * this.min) > p_164401_) {
-            return false;
         } else {
-            return this.max == null || !((double)(this.max * this.max) < p_164401_);
+            return this.min != null && (double)(this.min * this.min) > p_164401_ ? false : this.max == null || !((double)(this.max * this.max) < p_164401_);
         }
     }
 
@@ -130,7 +_,7 @@
     private static Float readNumber(StringReader p_75368_, boolean p_75369_) throws CommandSyntaxException {
         int i = p_75368_.getCursor();
 
-        while(p_75368_.canRead() && isAllowedNumber(p_75368_, p_75369_)) {
+        while (p_75368_.canRead() && isAllowedNumber(p_75368_, p_75369_)) {
             p_75368_.skip();
         }
 
@@ -153,11 +_,7 @@
     private static boolean isAllowedNumber(StringReader p_75371_, boolean p_75372_) {
         char c0 = p_75371_.peek();
         if ((c0 < '0' || c0 > '9') && c0 != '-') {
-            if (p_75372_ && c0 == '.') {
-                return !p_75371_.canRead(2) || p_75371_.peek(1) != '.';
-            } else {
-                return false;
-            }
+            return p_75372_ && c0 == '.' ? !p_75371_.canRead(2) || p_75371_.peek(1) != '.' : false;
         } else {
             return true;
         }
