--- a/net/minecraft/resources/RegistryFixedCodec.java
+++ b/net/minecraft/resources/RegistryFixedCodec.java
@@ -23,7 +_,7 @@
     }
 
     public <T> DataResult<T> encode(Holder<E> p_206729_, DynamicOps<T> p_206730_, T p_206731_) {
-        if (p_206730_ instanceof RegistryOps registryops) {
+        if (p_206730_ instanceof RegistryOps<?> registryops) {
             Optional<HolderOwner<E>> optional = registryops.owner(this.registryKey);
             if (optional.isPresent()) {
                 if (!p_206729_.canSerializeIn(optional.get())) {
@@ -43,19 +_,19 @@
 
     @Override
     public <T> DataResult<Pair<Holder<E>, T>> decode(DynamicOps<T> p_206743_, T p_206744_) {
-        if (p_206743_ instanceof RegistryOps registryops) {
+        if (p_206743_ instanceof RegistryOps<?> registryops) {
             Optional<HolderGetter<E>> optional = registryops.getter(this.registryKey);
             if (optional.isPresent()) {
                 return ResourceLocation.CODEC
                     .decode(p_206743_, p_206744_)
                     .flatMap(
-                        p_255515_ -> {
-                            ResourceLocation resourcelocation = p_255515_.getFirst();
+                        p_337524_ -> {
+                            ResourceLocation resourcelocation = p_337524_.getFirst();
                             return optional.get()
                                 .get(ResourceKey.create(this.registryKey, resourcelocation))
                                 .map(DataResult::success)
                                 .orElseGet(() -> DataResult.error(() -> "Failed to get element " + resourcelocation))
-                                .<Pair<Holder<E>, T>>map(p_256041_ -> Pair.of(p_256041_, (T)p_255515_.getSecond()))
+                                .<Pair<Holder<E>, T>>map(p_256041_ -> Pair.of(p_256041_, (T)p_337524_.getSecond()))
                                 .setLifecycle(Lifecycle.stable());
                         }
                     );
