--- a/net/minecraft/client/telemetry/TelemetryEventType.java
+++ b/net/minecraft/client/telemetry/TelemetryEventType.java
@@ -4,6 +_,7 @@
 import com.mojang.authlib.minecraft.TelemetrySession;
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap;
 import java.util.ArrayList;
 import java.util.List;
@@ -87,7 +_,7 @@
     private final String exportKey;
     private final List<TelemetryProperty<?>> properties;
     private final boolean isOptIn;
-    private final Codec<TelemetryEventInstance> codec;
+    private final MapCodec<TelemetryEventInstance> codec;
 
     TelemetryEventType(String p_261787_, String p_262121_, List<TelemetryProperty<?>> p_261987_, boolean p_261511_) {
         this.id = p_261787_;
@@ -110,7 +_,7 @@
         return this.properties;
     }
 
-    public Codec<TelemetryEventInstance> codec() {
+    public MapCodec<TelemetryEventInstance> codec() {
         return this.codec;
     }
 
@@ -121,7 +_,7 @@
     public TelemetryEvent export(TelemetrySession p_262179_, TelemetryPropertyMap p_262018_) {
         TelemetryEvent telemetryevent = p_262179_.createNewEvent(this.exportKey);
 
-        for(TelemetryProperty<?> telemetryproperty : this.properties) {
+        for (TelemetryProperty<?> telemetryproperty : this.properties) {
             telemetryproperty.export(p_262018_, telemetryevent);
         }
 
