--- a/net/minecraft/client/telemetry/TelemetryPropertyMap.java
+++ b/net/minecraft/client/telemetry/TelemetryPropertyMap.java
@@ -1,6 +_,5 @@
 package net.minecraft.client.telemetry;
 
-import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
 import com.mojang.serialization.DynamicOps;
 import com.mojang.serialization.MapCodec;
@@ -27,52 +_,51 @@
         return new TelemetryPropertyMap.Builder();
     }
 
-    public static Codec<TelemetryPropertyMap> createCodec(final List<TelemetryProperty<?>> p_262139_) {
-        return (new MapCodec<TelemetryPropertyMap>() {
-                public <T> RecordBuilder<T> encode(TelemetryPropertyMap p_261525_, DynamicOps<T> p_262068_, RecordBuilder<T> p_261850_) {
-                    RecordBuilder<T> recordbuilder = p_261850_;
-    
-                    for(TelemetryProperty<?> telemetryproperty : p_262139_) {
-                        recordbuilder = this.encodeProperty(p_261525_, recordbuilder, telemetryproperty);
-                    }
-    
-                    return recordbuilder;
-                }
-    
-                private <T, V> RecordBuilder<T> encodeProperty(TelemetryPropertyMap p_262128_, RecordBuilder<T> p_261947_, TelemetryProperty<V> p_261911_) {
-                    V v = p_262128_.get(p_261911_);
-                    return v != null ? p_261947_.add(p_261911_.id(), v, p_261911_.codec()) : p_261947_;
-                }
-    
-                @Override
-                public <T> DataResult<TelemetryPropertyMap> decode(DynamicOps<T> p_261767_, MapLike<T> p_262176_) {
-                    DataResult<TelemetryPropertyMap.Builder> dataresult = DataResult.success(new TelemetryPropertyMap.Builder());
-    
-                    for(TelemetryProperty<?> telemetryproperty : p_262139_) {
-                        dataresult = this.decodeProperty(dataresult, p_261767_, p_262176_, telemetryproperty);
-                    }
-    
-                    return dataresult.map(TelemetryPropertyMap.Builder::build);
-                }
-    
-                private <T, V> DataResult<TelemetryPropertyMap.Builder> decodeProperty(
-                    DataResult<TelemetryPropertyMap.Builder> p_261892_, DynamicOps<T> p_261859_, MapLike<T> p_261668_, TelemetryProperty<V> p_261627_
-                ) {
-                    T t = p_261668_.get(p_261627_.id());
-                    if (t != null) {
-                        DataResult<V> dataresult = p_261627_.codec().parse(p_261859_, t);
-                        return p_261892_.apply2stable((p_262028_, p_261796_) -> p_262028_.put((TelemetryProperty<T>)p_261627_, (T)p_261796_), dataresult);
-                    } else {
-                        return p_261892_;
-                    }
-                }
-    
-                @Override
-                public <T> Stream<T> keys(DynamicOps<T> p_261746_) {
-                    return p_262139_.stream().map(TelemetryProperty::id).map(p_261746_::createString);
-                }
-            })
-            .codec();
+    public static MapCodec<TelemetryPropertyMap> createCodec(final List<TelemetryProperty<?>> p_262139_) {
+        return new MapCodec<TelemetryPropertyMap>() {
+            public <T> RecordBuilder<T> encode(TelemetryPropertyMap p_261525_, DynamicOps<T> p_262068_, RecordBuilder<T> p_261850_) {
+                RecordBuilder<T> recordbuilder = p_261850_;
+
+                for (TelemetryProperty<?> telemetryproperty : p_262139_) {
+                    recordbuilder = this.encodeProperty(p_261525_, recordbuilder, telemetryproperty);
+                }
+
+                return recordbuilder;
+            }
+
+            private <T, V> RecordBuilder<T> encodeProperty(TelemetryPropertyMap p_262128_, RecordBuilder<T> p_261947_, TelemetryProperty<V> p_261911_) {
+                V v = p_262128_.get(p_261911_);
+                return v != null ? p_261947_.add(p_261911_.id(), v, p_261911_.codec()) : p_261947_;
+            }
+
+            @Override
+            public <T> DataResult<TelemetryPropertyMap> decode(DynamicOps<T> p_261767_, MapLike<T> p_262176_) {
+                DataResult<TelemetryPropertyMap.Builder> dataresult = DataResult.success(new TelemetryPropertyMap.Builder());
+
+                for (TelemetryProperty<?> telemetryproperty : p_262139_) {
+                    dataresult = this.decodeProperty(dataresult, p_261767_, p_262176_, telemetryproperty);
+                }
+
+                return dataresult.map(TelemetryPropertyMap.Builder::build);
+            }
+
+            private <T, V> DataResult<TelemetryPropertyMap.Builder> decodeProperty(
+                DataResult<TelemetryPropertyMap.Builder> p_261892_, DynamicOps<T> p_261859_, MapLike<T> p_261668_, TelemetryProperty<V> p_261627_
+            ) {
+                T t = p_261668_.get(p_261627_.id());
+                if (t != null) {
+                    DataResult<V> dataresult = p_261627_.codec().parse(p_261859_, t);
+                    return p_261892_.apply2stable((p_262028_, p_261796_) -> p_262028_.put(p_261627_, (V)p_261796_), dataresult);
+                } else {
+                    return p_261892_;
+                }
+            }
+
+            @Override
+            public <T> Stream<T> keys(DynamicOps<T> p_261746_) {
+                return p_262139_.stream().map(TelemetryProperty::id).map(p_261746_::createString);
+            }
+        };
     }
 
     @Nullable
