--- a/net/minecraft/network/chat/ComponentUtils.java
+++ b/net/minecraft/network/chat/ComponentUtils.java
@@ -45,7 +_,7 @@
         } else {
             MutableComponent mutablecomponent = p_130733_.getContents().resolve(p_130732_, p_130734_, p_130735_ + 1);
 
-            for(Component component : p_130733_.getSiblings()) {
+            for (Component component : p_130733_.getSiblings()) {
                 mutablecomponent.append(updateForEntity(p_130732_, component, p_130734_, p_130735_ + 1));
             }
 
@@ -98,12 +_,12 @@
         if (p_178437_.isEmpty()) {
             return Component.empty();
         } else if (p_178437_.size() == 1) {
-            return p_178439_.apply(p_178437_.iterator().next()).copy();
+            return p_178439_.apply((T)p_178437_.iterator().next()).copy();
         } else {
             MutableComponent mutablecomponent = Component.empty();
             boolean flag = true;
 
-            for(T t : p_178437_) {
+            for (T t : p_178437_) {
                 if (!flag) {
                     mutablecomponent.append(p_178438_);
                 }
@@ -125,16 +_,13 @@
     }
 
     public static boolean isTranslationResolvable(@Nullable Component p_237135_) {
-        if (p_237135_ != null) {
-            ComponentContents $$2 = p_237135_.getContents();
-            if ($$2 instanceof TranslatableContents translatablecontents) {
-                String s1 = translatablecontents.getKey();
-                String s = translatablecontents.getFallback();
-                return s != null || Language.getInstance().has(s1);
-            }
+        if (p_237135_ != null && p_237135_.getContents() instanceof TranslatableContents translatablecontents) {
+            String s1 = translatablecontents.getKey();
+            String s = translatablecontents.getFallback();
+            return s != null || Language.getInstance().has(s1);
+        } else {
+            return true;
         }
-
-        return true;
     }
 
     public static MutableComponent copyOnClickText(String p_260039_) {
