--- a/net/minecraft/network/chat/Component.java
+++ b/net/minecraft/network/chat/Component.java
@@ -22,7 +_,6 @@
 import java.util.Optional;
 import java.util.UUID;
 import javax.annotation.Nullable;
-import net.minecraft.Util;
 import net.minecraft.core.HolderLookup;
 import net.minecraft.network.chat.contents.DataSource;
 import net.minecraft.network.chat.contents.KeybindContents;
@@ -63,8 +_,7 @@
 
     @Nullable
     default String tryCollapseToString() {
-        ComponentContents componentcontents = this.getContents();
-        if (componentcontents instanceof PlainTextContents plaintextcontents && this.getSiblings().isEmpty() && this.getStyle().isEmpty()) {
+        if (this.getContents() instanceof PlainTextContents plaintextcontents && this.getSiblings().isEmpty() && this.getStyle().isEmpty()) {
             return plaintextcontents.text();
         }
 
@@ -88,7 +_,7 @@
         if (optional.isPresent()) {
             return optional;
         } else {
-            for(Component component : this.getSiblings()) {
+            for (Component component : this.getSiblings()) {
                 Optional<T> optional1 = component.visit(p_130679_, style);
                 if (optional1.isPresent()) {
                     return optional1;
@@ -105,7 +_,7 @@
         if (optional.isPresent()) {
             return optional;
         } else {
-            for(Component component : this.getSiblings()) {
+            for (Component component : this.getSiblings()) {
                 Optional<T> optional1 = component.visit(p_130677_);
                 if (optional1.isPresent()) {
                     return optional1;
@@ -159,7 +_,7 @@
     }
 
     static MutableComponent translatableEscape(String p_304683_, Object... p_304399_) {
-        for(int i = 0; i < p_304399_.length; ++i) {
+        for (int i = 0; i < p_304399_.length; i++) {
             Object object = p_304399_[i];
             if (!TranslatableContents.isAllowedPrimitiveArgument(object) && !(object instanceof Component)) {
                 p_304399_[i] = String.valueOf(object);
@@ -224,15 +_,15 @@
         }
 
         static MutableComponent deserialize(JsonElement p_130720_, HolderLookup.Provider p_330752_) {
-            return (MutableComponent)Util.getOrThrow(
-                ComponentSerialization.CODEC.parse(p_330752_.createSerializationContext(JsonOps.INSTANCE), p_130720_), JsonParseException::new
-            );
+            return (MutableComponent)ComponentSerialization.CODEC
+                .parse(p_330752_.createSerializationContext(JsonOps.INSTANCE), p_130720_)
+                .getOrThrow(JsonParseException::new);
         }
 
         static JsonElement serialize(Component p_304983_, HolderLookup.Provider p_331501_) {
-            return Util.getOrThrow(
-                ComponentSerialization.CODEC.encodeStart(p_331501_.createSerializationContext(JsonOps.INSTANCE), p_304983_), JsonParseException::new
-            );
+            return ComponentSerialization.CODEC
+                .encodeStart(p_331501_.createSerializationContext(JsonOps.INSTANCE), p_304983_)
+                .getOrThrow(JsonParseException::new);
         }
 
         public static String toJson(Component p_130704_, HolderLookup.Provider p_330782_) {
