--- a/net/minecraft/network/chat/Style.java
+++ b/net/minecraft/network/chat/Style.java
@@ -12,7 +_,6 @@
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.resources.ResourceLocation;
-import net.minecraft.util.ExtraCodecs;
 
 public class Style {
     public static final Style EMPTY = new Style(null, null, null, null, null, null, null, null, null, null);
@@ -366,7 +_,7 @@
         Boolean obool2 = this.strikethrough;
         Boolean obool3 = this.underlined;
         Boolean obool4 = this.obfuscated;
-        switch(p_131158_) {
+        switch (p_131158_) {
             case OBFUSCATED:
                 obool4 = true;
                 break;
@@ -398,7 +_,7 @@
         Boolean obool2 = this.strikethrough;
         Boolean obool3 = this.underlined;
         Boolean obool4 = this.obfuscated;
-        switch(p_131165_) {
+        switch (p_131165_) {
             case OBFUSCATED:
                 obool4 = true;
                 break;
@@ -436,8 +_,8 @@
         Boolean obool3 = this.underlined;
         Boolean obool4 = this.obfuscated;
 
-        for(ChatFormatting chatformatting : p_131153_) {
-            switch(chatformatting) {
+        for (ChatFormatting chatformatting : p_131153_) {
+            switch (chatformatting) {
                 case OBFUSCATED:
                     obool4 = true;
                     break;
@@ -491,6 +_,9 @@
         class Collector {
             private boolean isNotFirst;
 
+            Collector(Style p_237286_) {
+            }
+
             private void prependSeparator() {
                 if (this.isNotFirst) {
                     stringbuilder.append(',');
@@ -520,7 +_,7 @@
             }
         }
 
-        Collector style$1collector = new Collector();
+        Collector style$1collector = new Collector(this);
         style$1collector.addValueString("color", this.color);
         style$1collector.addFlagString("bold", this.bold);
         style$1collector.addFlagString("italic", this.italic);
@@ -539,20 +_,19 @@
     public boolean equals(Object p_131175_) {
         if (this == p_131175_) {
             return true;
-        } else if (!(p_131175_ instanceof Style)) {
-            return false;
         } else {
-            Style style = (Style)p_131175_;
-            return this.bold == style.bold
-                && Objects.equals(this.getColor(), style.getColor())
-                && this.italic == style.italic
-                && this.obfuscated == style.obfuscated
-                && this.strikethrough == style.strikethrough
-                && this.underlined == style.underlined
-                && Objects.equals(this.clickEvent, style.clickEvent)
-                && Objects.equals(this.hoverEvent, style.hoverEvent)
-                && Objects.equals(this.insertion, style.insertion)
-                && Objects.equals(this.font, style.font);
+            return !(p_131175_ instanceof Style style)
+                ? false
+                : this.bold == style.bold
+                    && Objects.equals(this.getColor(), style.getColor())
+                    && this.italic == style.italic
+                    && this.obfuscated == style.obfuscated
+                    && this.strikethrough == style.strikethrough
+                    && this.underlined == style.underlined
+                    && Objects.equals(this.clickEvent, style.clickEvent)
+                    && Objects.equals(this.hoverEvent, style.hoverEvent)
+                    && Objects.equals(this.insertion, style.insertion)
+                    && Objects.equals(this.font, style.font);
         }
     }
 
@@ -566,16 +_,16 @@
     public static class Serializer {
         public static final MapCodec<Style> MAP_CODEC = RecordCodecBuilder.mapCodec(
             p_304583_ -> p_304583_.group(
-                        ExtraCodecs.strictOptionalField(TextColor.CODEC, "color").forGetter(p_304458_ -> Optional.ofNullable(p_304458_.color)),
-                        ExtraCodecs.strictOptionalField(Codec.BOOL, "bold").forGetter(p_304491_ -> Optional.ofNullable(p_304491_.bold)),
-                        ExtraCodecs.strictOptionalField(Codec.BOOL, "italic").forGetter(p_304980_ -> Optional.ofNullable(p_304980_.italic)),
-                        ExtraCodecs.strictOptionalField(Codec.BOOL, "underlined").forGetter(p_304946_ -> Optional.ofNullable(p_304946_.underlined)),
-                        ExtraCodecs.strictOptionalField(Codec.BOOL, "strikethrough").forGetter(p_304494_ -> Optional.ofNullable(p_304494_.strikethrough)),
-                        ExtraCodecs.strictOptionalField(Codec.BOOL, "obfuscated").forGetter(p_304916_ -> Optional.ofNullable(p_304916_.obfuscated)),
-                        ExtraCodecs.strictOptionalField(ClickEvent.CODEC, "clickEvent").forGetter(p_304578_ -> Optional.ofNullable(p_304578_.clickEvent)),
-                        ExtraCodecs.strictOptionalField(HoverEvent.CODEC, "hoverEvent").forGetter(p_304424_ -> Optional.ofNullable(p_304424_.hoverEvent)),
-                        ExtraCodecs.strictOptionalField(Codec.STRING, "insertion").forGetter(p_304670_ -> Optional.ofNullable(p_304670_.insertion)),
-                        ExtraCodecs.strictOptionalField(ResourceLocation.CODEC, "font").forGetter(p_304448_ -> Optional.ofNullable(p_304448_.font))
+                        TextColor.CODEC.optionalFieldOf("color").forGetter(p_304458_ -> Optional.ofNullable(p_304458_.color)),
+                        Codec.BOOL.optionalFieldOf("bold").forGetter(p_304491_ -> Optional.ofNullable(p_304491_.bold)),
+                        Codec.BOOL.optionalFieldOf("italic").forGetter(p_304980_ -> Optional.ofNullable(p_304980_.italic)),
+                        Codec.BOOL.optionalFieldOf("underlined").forGetter(p_304946_ -> Optional.ofNullable(p_304946_.underlined)),
+                        Codec.BOOL.optionalFieldOf("strikethrough").forGetter(p_304494_ -> Optional.ofNullable(p_304494_.strikethrough)),
+                        Codec.BOOL.optionalFieldOf("obfuscated").forGetter(p_304916_ -> Optional.ofNullable(p_304916_.obfuscated)),
+                        ClickEvent.CODEC.optionalFieldOf("clickEvent").forGetter(p_304578_ -> Optional.ofNullable(p_304578_.clickEvent)),
+                        HoverEvent.CODEC.optionalFieldOf("hoverEvent").forGetter(p_304424_ -> Optional.ofNullable(p_304424_.hoverEvent)),
+                        Codec.STRING.optionalFieldOf("insertion").forGetter(p_304670_ -> Optional.ofNullable(p_304670_.insertion)),
+                        ResourceLocation.CODEC.optionalFieldOf("font").forGetter(p_304448_ -> Optional.ofNullable(p_304448_.font))
                     )
                     .apply(p_304583_, Style::create)
         );
