--- a/net/minecraft/network/chat/SubStringSource.java
+++ b/net/minecraft/network/chat/SubStringSource.java
@@ -32,7 +_,7 @@
             Style style = this.charStyles.get(p_131237_);
             int i = p_131237_;
 
-            for(int j = 1; j < p_131238_; ++j) {
+            for (int j = 1; j < p_131238_; j++) {
                 int k = p_131237_ + j;
                 Style style1 = this.charStyles.get(k);
                 if (!style1.equals(style)) {
@@ -64,7 +_,7 @@
                 stringbuilder.appendCodePoint(p_178535_);
                 int i = Character.charCount(p_178535_);
 
-                for(int j = 0; j < i; ++j) {
+                for (int j = 0; j < i; j++) {
                     list.add(p_178534_);
                 }
 
