--- a/net/minecraft/world/level/block/entity/vault/VaultSharedData.java
+++ b/net/minecraft/world/level/block/entity/vault/VaultSharedData.java
@@ -15,14 +_,14 @@
 public class VaultSharedData {
     static final String TAG_NAME = "shared_data";
     static Codec<VaultSharedData> CODEC = RecordCodecBuilder.create(
-        p_324328_ -> p_324328_.group(
-                    ItemStack.optionalFieldOf("display_item").forGetter(p_324217_ -> p_324217_.displayItem),
-                    UUIDUtil.CODEC_LINKED_SET.optionalFieldOf("connected_players", Set.of()).forGetter(p_324110_ -> p_324110_.connectedPlayers),
+        p_338074_ -> p_338074_.group(
+                    ItemStack.lenientOptionalFieldOf("display_item").forGetter(p_324217_ -> p_324217_.displayItem),
+                    UUIDUtil.CODEC_LINKED_SET.lenientOptionalFieldOf("connected_players", Set.of()).forGetter(p_324110_ -> p_324110_.connectedPlayers),
                     Codec.DOUBLE
-                        .optionalFieldOf("connected_particles_range", Double.valueOf(VaultConfig.DEFAULT.deactivationRange()))
+                        .lenientOptionalFieldOf("connected_particles_range", Double.valueOf(VaultConfig.DEFAULT.deactivationRange()))
                         .forGetter(p_323486_ -> p_323486_.connectedParticlesRange)
                 )
-                .apply(p_324328_, VaultSharedData::new)
+                .apply(p_338074_, VaultSharedData::new)
     );
     private ItemStack displayItem = ItemStack.EMPTY;
     private Set<UUID> connectedPlayers = new ObjectLinkedOpenHashSet<>();
@@ -67,7 +_,7 @@
 
     void updateConnectedPlayersWithinRange(ServerLevel p_324193_, BlockPos p_324358_, VaultServerData p_324348_, VaultConfig p_324058_, double p_324450_) {
         Set<UUID> set = p_324058_.playerDetector()
-            .detect(p_324193_, p_324058_.entitySelector(), p_324358_, p_324450_)
+            .detect(p_324193_, p_324058_.entitySelector(), p_324358_, p_324450_, false)
             .stream()
             .filter(p_324308_ -> !p_324348_.getRewardedPlayers().contains(p_324308_))
             .collect(Collectors.toSet());
