--- a/net/minecraft/world/level/block/entity/vault/VaultBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/vault/VaultBlockEntity.java
@@ -12,6 +_,7 @@
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.core.HolderLookup;
+import net.minecraft.core.particles.ParticleOptions;
 import net.minecraft.core.particles.ParticleTypes;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.NbtOps;
@@ -72,12 +_,12 @@
     }
 
     private static <T> Tag encode(Codec<T> p_324465_, T p_324338_, HolderLookup.Provider p_331118_) {
-        return Util.getOrThrow(p_324465_.encodeStart(p_331118_.createSerializationContext(NbtOps.INSTANCE), p_324338_), IllegalStateException::new);
+        return p_324465_.encodeStart(p_331118_.createSerializationContext(NbtOps.INSTANCE), p_324338_).getOrThrow();
     }
 
     @Override
-    public void load(CompoundTag p_324011_, HolderLookup.Provider p_324430_) {
-        super.load(p_324011_, p_324430_);
+    protected void loadAdditional(CompoundTag p_324011_, HolderLookup.Provider p_324430_) {
+        super.loadAdditional(p_324011_, p_324430_);
         DynamicOps<Tag> dynamicops = p_324430_.createSerializationContext(NbtOps.INSTANCE);
         if (p_324011_.contains("server_data")) {
             VaultServerData.CODEC.parse(dynamicops, p_324011_.get("server_data")).resultOrPartial(LOGGER::error).ifPresent(this.serverData::set);
@@ -127,38 +_,42 @@
                 emitConnectionParticlesForNearbyPlayers(p_324312_, p_323810_, p_323862_, p_323914_);
             }
 
-            emitIdleParticles(p_324312_, p_323810_, p_323914_);
+            emitIdleParticles(
+                p_324312_, p_323810_, p_323914_, p_323862_.getValue(VaultBlock.OMINOUS) ? ParticleTypes.SOUL_FIRE_FLAME : ParticleTypes.SMALL_FLAME
+            );
             playIdleSounds(p_324312_, p_323810_, p_323914_);
         }
 
-        public static void emitActivationParticles(Level p_323761_, BlockPos p_324150_, BlockState p_324527_, VaultSharedData p_324466_) {
+        public static void emitActivationParticles(
+            Level p_323761_, BlockPos p_324150_, BlockState p_324527_, VaultSharedData p_324466_, ParticleOptions p_338551_
+        ) {
             emitConnectionParticlesForNearbyPlayers(p_323761_, p_324150_, p_324527_, p_324466_);
             RandomSource randomsource = p_323761_.random;
 
-            for(int i = 0; i < 20; ++i) {
+            for (int i = 0; i < 20; i++) {
                 Vec3 vec3 = randomPosInsideCage(p_324150_, randomsource);
                 p_323761_.addParticle(ParticleTypes.SMOKE, vec3.x(), vec3.y(), vec3.z(), 0.0, 0.0, 0.0);
-                p_323761_.addParticle(ParticleTypes.SMALL_FLAME, vec3.x(), vec3.y(), vec3.z(), 0.0, 0.0, 0.0);
+                p_323761_.addParticle(p_338551_, vec3.x(), vec3.y(), vec3.z(), 0.0, 0.0, 0.0);
             }
         }
 
-        public static void emitDeactivationParticles(Level p_324419_, BlockPos p_324587_) {
+        public static void emitDeactivationParticles(Level p_324419_, BlockPos p_324587_, ParticleOptions p_338582_) {
             RandomSource randomsource = p_324419_.random;
 
-            for(int i = 0; i < 20; ++i) {
+            for (int i = 0; i < 20; i++) {
                 Vec3 vec3 = randomPosCenterOfCage(p_324587_, randomsource);
                 Vec3 vec31 = new Vec3(randomsource.nextGaussian() * 0.02, randomsource.nextGaussian() * 0.02, randomsource.nextGaussian() * 0.02);
-                p_324419_.addParticle(ParticleTypes.SMALL_FLAME, vec3.x(), vec3.y(), vec3.z(), vec31.x(), vec31.y(), vec31.z());
+                p_324419_.addParticle(p_338582_, vec3.x(), vec3.y(), vec3.z(), vec31.x(), vec31.y(), vec31.z());
             }
         }
 
-        private static void emitIdleParticles(Level p_324004_, BlockPos p_324516_, VaultSharedData p_324173_) {
+        private static void emitIdleParticles(Level p_324004_, BlockPos p_324516_, VaultSharedData p_324173_, ParticleOptions p_338823_) {
             RandomSource randomsource = p_324004_.getRandom();
             if (randomsource.nextFloat() <= 0.5F) {
                 Vec3 vec3 = randomPosInsideCage(p_324516_, randomsource);
                 p_324004_.addParticle(ParticleTypes.SMOKE, vec3.x(), vec3.y(), vec3.z(), 0.0, 0.0, 0.0);
                 if (shouldDisplayActiveEffects(p_324173_)) {
-                    p_324004_.addParticle(ParticleTypes.SMALL_FLAME, vec3.x(), vec3.y(), vec3.z(), 0.0, 0.0, 0.0);
+                    p_324004_.addParticle(p_338823_, vec3.x(), vec3.y(), vec3.z(), 0.0, 0.0, 0.0);
                 }
             }
         }
@@ -168,7 +_,7 @@
             Vec3 vec3 = p_324592_.vectorTo(p_324428_.position().add(0.0, (double)(p_324428_.getBbHeight() / 2.0F), 0.0));
             int i = Mth.nextInt(randomsource, 2, 5);
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 Vec3 vec31 = vec3.offsetRandom(randomsource, 1.0F);
                 p_323629_.addParticle(ParticleTypes.VAULT_CONNECTION, p_324592_.x(), p_324592_.y(), p_324592_.z(), vec31.x(), vec31.y(), vec31.z());
             }
@@ -179,7 +_,7 @@
             if (!set.isEmpty()) {
                 Vec3 vec3 = keyholePos(p_324162_, p_324246_.getValue(VaultBlock.FACING));
 
-                for(UUID uuid : set) {
+                for (UUID uuid : set) {
                     Player player = p_324593_.getPlayerByUUID(uuid);
                     if (player != null && isWithinConnectionRange(p_324162_, p_323584_, player)) {
                         emitConnectionParticlesForPlayer(p_324593_, vec3, player);
@@ -297,7 +_,7 @@
             VaultState vaultstate = p_323759_.getValue(VaultBlock.STATE);
             VaultState vaultstate1 = p_324027_.getValue(VaultBlock.STATE);
             p_324091_.setBlock(p_324620_, p_324027_, 3);
-            vaultstate.onTransition(p_324091_, p_324620_, vaultstate1, p_324140_, p_323624_);
+            vaultstate.onTransition(p_324091_, p_324620_, vaultstate1, p_324140_, p_323624_, p_324027_.getValue(VaultBlock.OMINOUS));
         }
 
         static void cycleDisplayItemFromLootTable(
