--- a/net/minecraft/world/level/block/entity/vault/VaultServerData.java
+++ b/net/minecraft/world/level/block/entity/vault/VaultServerData.java
@@ -19,13 +_,13 @@
 public class VaultServerData {
     static final String TAG_NAME = "server_data";
     static Codec<VaultServerData> CODEC = RecordCodecBuilder.create(
-        p_324238_ -> p_324238_.group(
-                    UUIDUtil.CODEC_LINKED_SET.optionalFieldOf("rewarded_players", Set.of()).forGetter(p_323523_ -> p_323523_.rewardedPlayers),
-                    Codec.LONG.optionalFieldOf("state_updating_resumes_at", Long.valueOf(0L)).forGetter(p_323634_ -> p_323634_.stateUpdatingResumesAt),
-                    ItemStack.CODEC.listOf().optionalFieldOf("items_to_eject", List.of()).forGetter(p_323976_ -> p_323976_.itemsToEject),
-                    Codec.INT.optionalFieldOf("total_ejections_needed", Integer.valueOf(0)).forGetter(p_323753_ -> p_323753_.totalEjectionsNeeded)
+        p_338073_ -> p_338073_.group(
+                    UUIDUtil.CODEC_LINKED_SET.lenientOptionalFieldOf("rewarded_players", Set.of()).forGetter(p_323523_ -> p_323523_.rewardedPlayers),
+                    Codec.LONG.lenientOptionalFieldOf("state_updating_resumes_at", Long.valueOf(0L)).forGetter(p_323634_ -> p_323634_.stateUpdatingResumesAt),
+                    ItemStack.CODEC.listOf().lenientOptionalFieldOf("items_to_eject", List.of()).forGetter(p_323976_ -> p_323976_.itemsToEject),
+                    Codec.INT.lenientOptionalFieldOf("total_ejections_needed", Integer.valueOf(0)).forGetter(p_323753_ -> p_323753_.totalEjectionsNeeded)
                 )
-                .apply(p_324238_, VaultServerData::new)
+                .apply(p_338073_, VaultServerData::new)
     );
     private static final int MAX_REWARD_PLAYERS = 128;
     private final Set<UUID> rewardedPlayers = new ObjectLinkedOpenHashSet<>();
