--- a/net/minecraft/world/level/levelgen/blockpredicates/AllOfPredicate.java
+++ b/net/minecraft/world/level/levelgen/blockpredicates/AllOfPredicate.java
@@ -1,19 +_,19 @@
 package net.minecraft.world.level.levelgen.blockpredicates;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.WorldGenLevel;
 
 class AllOfPredicate extends CombiningPredicate {
-    public static final Codec<AllOfPredicate> CODEC = codec(AllOfPredicate::new);
+    public static final MapCodec<AllOfPredicate> CODEC = codec(AllOfPredicate::new);
 
     public AllOfPredicate(List<BlockPredicate> p_190373_) {
         super(p_190373_);
     }
 
     public boolean test(WorldGenLevel p_190376_, BlockPos p_190377_) {
-        for(BlockPredicate blockpredicate : this.predicates) {
+        for (BlockPredicate blockpredicate : this.predicates) {
             if (!blockpredicate.test(p_190376_, p_190377_)) {
                 return false;
             }
