--- a/net/minecraft/world/level/levelgen/blockpredicates/InsideWorldBoundsPredicate.java
+++ b/net/minecraft/world/level/levelgen/blockpredicates/InsideWorldBoundsPredicate.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.blockpredicates;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -8,7 +_,7 @@
 import net.minecraft.world.level.WorldGenLevel;
 
 public class InsideWorldBoundsPredicate implements BlockPredicate {
-    public static final Codec<InsideWorldBoundsPredicate> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<InsideWorldBoundsPredicate> CODEC = RecordCodecBuilder.mapCodec(
         p_190473_ -> p_190473_.group(Vec3i.offsetCodec(16).optionalFieldOf("offset", BlockPos.ZERO).forGetter(p_190475_ -> p_190475_.offset))
                 .apply(p_190473_, InsideWorldBoundsPredicate::new)
     );
