--- a/net/minecraft/client/renderer/BlockEntityWithoutLevelRenderer.java
+++ b/net/minecraft/client/renderer/BlockEntityWithoutLevelRenderer.java
@@ -55,7 +_,7 @@
     private static final ShulkerBoxBlockEntity[] SHULKER_BOXES = Arrays.stream(DyeColor.values())
         .sorted(Comparator.comparingInt(DyeColor::getId))
         .map(p_172557_ -> new ShulkerBoxBlockEntity(p_172557_, BlockPos.ZERO, Blocks.SHULKER_BOX.defaultBlockState()))
-        .toArray(p_172553_ -> new ShulkerBoxBlockEntity[p_172553_]);
+        .toArray(ShulkerBoxBlockEntity[]::new);
     private static final ShulkerBoxBlockEntity DEFAULT_SHULKER_BOX = new ShulkerBoxBlockEntity(BlockPos.ZERO, Blocks.SHULKER_BOX.defaultBlockState());
     private final ChestBlockEntity chest = new ChestBlockEntity(BlockPos.ZERO, Blocks.CHEST.defaultBlockState());
     private final ChestBlockEntity trappedChest = new TrappedChestBlockEntity(BlockPos.ZERO, Blocks.TRAPPED_CHEST.defaultBlockState());
