--- a/net/minecraft/client/renderer/CubeMap.java
+++ b/net/minecraft/client/renderer/CubeMap.java
@@ -22,7 +_,7 @@
     private final ResourceLocation[] images = new ResourceLocation[6];
 
     public CubeMap(ResourceLocation p_108848_) {
-        for(int i = 0; i < 6; ++i) {
+        for (int i = 0; i < 6; i++) {
             this.images[i] = p_108848_.withPath(p_108848_.getPath() + "_" + i + ".png");
         }
     }
@@ -43,7 +_,7 @@
         RenderSystem.depthMask(false);
         int i = 2;
 
-        for(int j = 0; j < 4; ++j) {
+        for (int j = 0; j < 4; j++) {
             matrix4fstack.pushMatrix();
             float f = ((float)(j % 2) / 2.0F - 0.5F) / 256.0F;
             float f1 = ((float)(j / 2) / 2.0F - 0.5F) / 256.0F;
@@ -53,7 +_,7 @@
             matrix4fstack.rotateY(p_108852_ * (float) (Math.PI / 180.0));
             RenderSystem.applyModelViewMatrix();
 
-            for(int k = 0; k < 6; ++k) {
+            for (int k = 0; k < 6; k++) {
                 RenderSystem.setShaderTexture(0, this.images[k]);
                 bufferbuilder.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_TEX_COLOR);
                 int l = Math.round(255.0F * p_108853_) / (j + 1);
@@ -118,7 +_,7 @@
     public CompletableFuture<Void> preload(TextureManager p_108855_, Executor p_108856_) {
         CompletableFuture<?>[] completablefuture = new CompletableFuture[6];
 
-        for(int i = 0; i < completablefuture.length; ++i) {
+        for (int i = 0; i < completablefuture.length; i++) {
             completablefuture[i] = p_108855_.preload(this.images[i], p_108856_);
         }
 
