--- a/net/minecraft/client/renderer/EffectInstance.java
+++ b/net/minecraft/client/renderer/EffectInstance.java
@@ -58,7 +_,8 @@
     private final EffectProgram fragmentProgram;
 
     public EffectInstance(ResourceProvider p_331011_, String p_108942_) throws IOException {
-        ResourceLocation resourcelocation = new ResourceLocation("shaders/program/" + p_108942_ + ".json");
+        ResourceLocation rl = ResourceLocation.tryParse(p_108942_);
+        ResourceLocation resourcelocation = new ResourceLocation(rl.getNamespace(), "shaders/program/" + rl.getPath() + ".json");
         this.name = p_108942_;
         Resource resource = p_331011_.getResourceOrThrow(resourcelocation);
 
@@ -70,7 +_,7 @@
             if (jsonarray != null) {
                 int i = 0;
 
-                for(JsonElement jsonelement : jsonarray) {
+                for (JsonElement jsonelement : jsonarray) {
                     try {
                         this.parseSamplerNode(jsonelement);
                     } catch (Exception exception2) {
@@ -79,7 +_,7 @@
                         throw chainedjsonexception1;
                     }
 
-                    ++i;
+                    i++;
                 }
             }
 
@@ -89,7 +_,7 @@
                 this.attributes = Lists.newArrayListWithCapacity(jsonarray1.size());
                 this.attributeNames = Lists.newArrayListWithCapacity(jsonarray1.size());
 
-                for(JsonElement jsonelement1 : jsonarray1) {
+                for (JsonElement jsonelement1 : jsonarray1) {
                     try {
                         this.attributeNames.add(GsonHelper.convertToString(jsonelement1, "attribute"));
                     } catch (Exception exception1) {
@@ -98,7 +_,7 @@
                         throw chainedjsonexception2;
                     }
 
-                    ++j;
+                    j++;
                 }
             } else {
                 this.attributes = null;
@@ -109,7 +_,7 @@
             if (jsonarray2 != null) {
                 int k = 0;
 
-                for(JsonElement jsonelement2 : jsonarray2) {
+                for (JsonElement jsonelement2 : jsonarray2) {
                     try {
                         this.parseUniformNode(jsonelement2);
                     } catch (Exception exception) {
@@ -118,7 +_,7 @@
                         throw chainedjsonexception3;
                     }
 
-                    ++k;
+                    k++;
                 }
             }
 
@@ -129,7 +_,7 @@
             ProgramManager.linkShader(this);
             this.updateLocations();
             if (this.attributeNames != null) {
-                for(String s2 : this.attributeNames) {
+                for (String s2 : this.attributeNames) {
                     int l = Uniform.glGetAttribLocation(this.programId, s2);
                     this.attributes.add(l);
                 }
@@ -150,7 +_,8 @@
         } else {
             EffectProgram effectprogram;
             if (program == null) {
-                ResourceLocation resourcelocation = new ResourceLocation("shaders/program/" + p_172569_ + p_172568_.getExtension());
+                ResourceLocation rl = ResourceLocation.tryParse(p_172569_);
+                ResourceLocation resourcelocation = new ResourceLocation(rl.getNamespace(), "shaders/program/" + rl.getPath() + p_172568_.getExtension());
                 Resource resource = p_330579_.getResourceOrThrow(resourcelocation);
 
                 try (InputStream inputstream = resource.open()) {
@@ -224,7 +_,7 @@
 
     @Override
     public void close() {
-        for(Uniform uniform : this.uniforms) {
+        for (Uniform uniform : this.uniforms) {
             uniform.close();
         }
 
@@ -237,7 +_,7 @@
         lastProgramId = -1;
         lastAppliedEffect = null;
 
-        for(int i = 0; i < this.samplerLocations.size(); ++i) {
+        for (int i = 0; i < this.samplerLocations.size(); i++) {
             if (this.samplerMap.get(this.samplerNames.get(i)) != null) {
                 GlStateManager._activeTexture(33984 + i);
                 GlStateManager._bindTexture(0);
@@ -255,7 +_,7 @@
             lastProgramId = this.programId;
         }
 
-        for(int i = 0; i < this.samplerLocations.size(); ++i) {
+        for (int i = 0; i < this.samplerLocations.size(); i++) {
             String s = this.samplerNames.get(i);
             IntSupplier intsupplier = this.samplerMap.get(s);
             if (intsupplier != null) {
@@ -268,7 +_,7 @@
             }
         }
 
-        for(Uniform uniform : this.uniforms) {
+        for (Uniform uniform : this.uniforms) {
             uniform.upload();
         }
     }
@@ -294,7 +_,7 @@
         RenderSystem.assertOnRenderThread();
         IntList intlist = new IntArrayList();
 
-        for(int i = 0; i < this.samplerNames.size(); ++i) {
+        for (int i = 0; i < this.samplerNames.size(); i++) {
             String s = this.samplerNames.get(i);
             int j = Uniform.glGetUniformLocation(this.programId, s);
             if (j == -1) {
@@ -306,11 +_,11 @@
             }
         }
 
-        for(int l = intlist.size() - 1; l >= 0; --l) {
+        for (int l = intlist.size() - 1; l >= 0; l--) {
             this.samplerNames.remove(intlist.getInt(l));
         }
 
-        for(Uniform uniform : this.uniforms) {
+        for (Uniform uniform : this.uniforms) {
             String s1 = uniform.getName();
             int k = Uniform.glGetUniformLocation(this.programId, s1);
             if (k == -1) {
@@ -355,7 +_,7 @@
         } else {
             int k = 0;
 
-            for(JsonElement jsonelement : jsonarray) {
+            for (JsonElement jsonelement : jsonarray) {
                 try {
                     afloat[k] = GsonHelper.convertToFloat(jsonelement, "value");
                 } catch (Exception exception) {
@@ -364,13 +_,13 @@
                     throw chainedjsonexception;
                 }
 
-                ++k;
+                k++;
             }
 
             if (j > 1 && jsonarray.size() == 1) {
-                while(k < j) {
+                while (k < j) {
                     afloat[k] = afloat[0];
-                    ++k;
+                    k++;
                 }
             }
 
