--- a/net/minecraft/client/renderer/ItemModelShaper.java
+++ b/net/minecraft/client/renderer/ItemModelShaper.java
@@ -24,6 +_,7 @@
 
     public BakedModel getItemModel(ItemStack p_109407_) {
         BakedModel bakedmodel = this.getItemModel(p_109407_.getItem());
+        // FORGE: Make sure to call the item overrides
         return bakedmodel == null ? this.modelManager.getMissingModel() : bakedmodel;
     }
 
@@ -47,7 +_,7 @@
     public void rebuildCache() {
         this.shapesCache.clear();
 
-        for(Entry<Integer, ModelResourceLocation> entry : this.shapes.entrySet()) {
+        for (Entry<Integer, ModelResourceLocation> entry : this.shapes.entrySet()) {
             this.shapesCache.put(entry.getKey(), this.modelManager.getModel(entry.getValue()));
         }
     }
