--- a/net/minecraft/client/renderer/LevelRenderer.java
+++ b/net/minecraft/client/renderer/LevelRenderer.java
@@ -490,7 +_,7 @@
         profilerfiller.push("entities");
         this.extractVisibleEntities(camera, frustum, deltaTracker, this.levelRenderState);
         profilerfiller.popPush("blockEntities");
-        this.extractVisibleBlockEntities(camera, f, this.levelRenderState);
+        this.extractVisibleBlockEntities(camera, f, this.levelRenderState, frustum);
         profilerfiller.popPush("blockOutline");
         this.extractBlockOutline(camera, this.levelRenderState);
         profilerfiller.popPush("blockBreaking");
@@ -504,6 +_,13 @@
             .extract(
                 this.level.getWorldBorder(), f, vec3, this.minecraft.options.getEffectiveRenderDistance() * 16, this.levelRenderState.worldBorderRenderState
             );
+        profilerfiller.popPush("neoforge_custom");
+        this.levelRenderState.customWeatherEffectRenderer = net.neoforged.neoforge.client.CustomEnvironmentEffectsRendererManager.getCustomWeatherEffectRenderer(this.level, camera.position());
+        this.levelRenderState.customSkyboxRenderer = net.neoforged.neoforge.client.CustomEnvironmentEffectsRendererManager.getCustomSkyboxRenderer(this.level, camera.position());
+        this.levelRenderState.customCloudsRenderer = net.neoforged.neoforge.client.CustomEnvironmentEffectsRendererManager.getCustomCloudsRenderer(this.level, camera.position());
+        net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.ExtractLevelRenderStateEvent(
+                this, this.levelRenderState, this.level, camera, frustum, deltaTracker, this.ticks
+        ));
         profilerfiller.pop();
         profilerfiller.popPush("debug");
         this.debugRenderer.emitGizmos(frustum, vec3.x, vec3.y, vec3.z, deltaTracker.getGameTimeDeltaPartialTick(false));
@@ -516,7 +_,7 @@
         this.targets.main = framegraphbuilder.importExternal("main", this.minecraft.getMainRenderTarget());
         int i = this.minecraft.getMainRenderTarget().width;
         int j = this.minecraft.getMainRenderTarget().height;
-        RenderTargetDescriptor rendertargetdescriptor = new RenderTargetDescriptor(i, j, true, 0);
+        RenderTargetDescriptor rendertargetdescriptor = new RenderTargetDescriptor(i, j, true, 0, this.minecraft.getMainRenderTarget().useStencil);
         PostChain postchain = this.getTransparencyChain();
         if (postchain != null) {
             this.targets.translucent = framegraphbuilder.createInternal("translucent", rendertargetdescriptor);
@@ -530,6 +_,9 @@
             this.targets.entityOutline = framegraphbuilder.importExternal("entity_outline", this.entityOutlineTarget);
         }
 
+        var setupEvent = net.neoforged.neoforge.client.ClientHooks.fireFrameGraphSetup(framegraphbuilder, this.targets, rendertargetdescriptor, frustum, camera, modelViewMatrix, projectionMatrix, deltaTracker, profilerfiller);
+        this.levelRenderState.haveGlowingEntities |= setupEvent.isOutlineProcessingEnabled();
+
         FramePass framepass = framegraphbuilder.addPass("clear");
         this.targets.main = framepass.readsAndWrites(this.targets.main);
         framepass.executes(
@@ -543,7 +_,7 @@
             }
         );
         if (shouldRenderSky) {
-            this.addSkyPass(framegraphbuilder, camera, terrainFog);
+            this.addSkyPass(framegraphbuilder, camera, terrainFog, modelViewMatrix);
         }
 
         this.addMainPass(framegraphbuilder, frustum, modelViewMatrix, terrainFog, renderOutline, this.levelRenderState, deltaTracker, profilerfiller);
@@ -553,17 +_,17 @@
         }
 
         this.minecraft.particleEngine.extract(this.particlesRenderState, new Frustum(frustum).offset(-3.0F), camera, f);
-        this.addParticlesPass(framegraphbuilder, terrainFog);
+        this.addParticlesPass(framegraphbuilder, terrainFog, modelViewMatrix);
         CloudStatus cloudstatus = this.minecraft.options.getCloudsType();
         if (cloudstatus != CloudStatus.OFF) {
             int k = camera.attributeProbe().getValue(EnvironmentAttributes.CLOUD_COLOR, f);
             if (ARGB.alpha(k) > 0) {
                 float f1 = camera.attributeProbe().getValue(EnvironmentAttributes.CLOUD_HEIGHT, f);
-                this.addCloudsPass(framegraphbuilder, cloudstatus, this.levelRenderState.cameraRenderState.pos, this.levelRenderState.gameTime, f, k, f1);
+                this.addCloudsPass(framegraphbuilder, cloudstatus, this.levelRenderState.cameraRenderState.pos, this.levelRenderState.gameTime, f, k, f1, modelViewMatrix);
             }
         }
 
-        this.addWeatherPass(framegraphbuilder, terrainFog);
+        this.addWeatherPass(framegraphbuilder, terrainFog, modelViewMatrix);
         if (postchain != null) {
             postchain.addToFrame(framegraphbuilder, i, j, this.targets);
         }
@@ -637,6 +_,9 @@
 
                 ChunkSectionsToRender chunksectionstorender = this.prepareChunkRenders(modelViewMatrix, d0, d1, d2);
                 chunksectionstorender.renderGroup(ChunkSectionLayerGroup.OPAQUE, this.chunkLayerSampler);
+                profiler.push("neoforge_render_after_opaque_blocks");
+                net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLevelStageEvent.AfterOpaqueBlocks(this, this.levelRenderState, null, modelViewMatrix, this.getRenderableSections()));
+                profiler.pop();
                 this.minecraft.gameRenderer.getLighting().setupFor(Lighting.Entry.LEVEL);
                 if (resourcehandle2 != null) {
                     resourcehandle2.get().copyDepthFrom(this.minecraft.getMainRenderTarget());
@@ -671,6 +_,8 @@
                 multibuffersource$buffersource.endBatch(Sheets.hangingSignSheet());
                 multibuffersource$buffersource.endBatch(Sheets.chestSheet());
                 this.renderBuffers.outlineBufferSource().endOutlineBatch();
+                profiler.popPush("neoforge_render_after_entities");
+                net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLevelStageEvent.AfterEntities(this, this.levelRenderState, posestack, modelViewMatrix, this.getRenderableSections()));
                 if (renderOutline) {
                     this.renderBlockOutline(multibuffersource$buffersource, posestack, false, levelRenderState);
                 }
@@ -700,8 +_,12 @@
 
                 profiler.push("translucent");
                 chunksectionstorender.renderGroup(ChunkSectionLayerGroup.TRANSLUCENT, this.chunkLayerSampler);
+                profiler.popPush("neoforge_render_after_translucent_blocks");
+                net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLevelStageEvent.AfterTranslucentBlocks(this, this.levelRenderState, null, modelViewMatrix, this.getRenderableSections()));
                 profiler.popPush("string");
                 chunksectionstorender.renderGroup(ChunkSectionLayerGroup.TRIPWIRE, this.chunkLayerSampler);
+                profiler.popPush("neoforge_render_after_tripwire_blocks");
+                net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLevelStageEvent.AfterTripwireBlocks(this, this.levelRenderState, null, modelViewMatrix, this.getRenderableSections()));
                 if (renderOutline) {
                     this.renderBlockOutline(multibuffersource$buffersource, posestack, true, levelRenderState);
                 }
@@ -712,7 +_,15 @@
         );
     }
 
+    /**
+     * @deprecated Neo: use {@link #addParticlesPass(FrameGraphBuilder, GpuBufferSlice, Matrix4f)} instead
+     */
+    @Deprecated
     private void addParticlesPass(FrameGraphBuilder frame, GpuBufferSlice fog) {
+        addParticlesPass(frame, fog, RenderSystem.getModelViewMatrix());
+    }
+
+    private void addParticlesPass(FrameGraphBuilder frame, GpuBufferSlice fog, Matrix4f modelViewMatrix) {
         FramePass framepass = frame.addPass("particles");
         if (this.targets.particles != null) {
             this.targets.particles = framepass.readsAndWrites(this.targets.particles);
@@ -732,12 +_,24 @@
             this.particlesRenderState.submit(this.submitNodeStorage, this.levelRenderState.cameraRenderState);
             this.featureRenderDispatcher.renderAllFeatures();
             this.particlesRenderState.reset();
+
+            Profiler.get().push("neoforge_render_after_particles");
+            net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLevelStageEvent.AfterParticles(this, this.levelRenderState, null, modelViewMatrix, this.getRenderableSections()));
+            Profiler.get().pop();
         });
     }
 
+    /**
+     * @deprecated Neo: use {@link #addCloudsPass(FrameGraphBuilder, CloudStatus, Vec3, long, float, int, float, Matrix4f)} instead
+     */
+    @Deprecated
     private void addCloudsPass(
         FrameGraphBuilder frame, CloudStatus cloudsType, Vec3 cameraPosition, long gameTime, float partialTicks, int cloudColor, float cloudHeight
     ) {
+        addCloudsPass(frame, cloudsType, cameraPosition, gameTime, partialTicks, cloudColor, cloudHeight, RenderSystem.getModelViewMatrix());
+    }
+
+    private void addCloudsPass(FrameGraphBuilder frame, CloudStatus cloudsType, Vec3 cameraPosition, long gameTime, float partialTicks, int cloudColor, float cloudHeight, Matrix4f modelViewMatrix) {
         FramePass framepass = frame.addPass("clouds");
         if (this.targets.clouds != null) {
             this.targets.clouds = framepass.readsAndWrites(this.targets.clouds);
@@ -745,10 +_,22 @@
             this.targets.main = framepass.readsAndWrites(this.targets.main);
         }
 
-        framepass.executes(() -> this.cloudRenderer.render(cloudColor, cloudsType, cloudHeight, cameraPosition, gameTime, partialTicks));
+        framepass.executes(() -> {
+            if (this.levelRenderState.customCloudsRenderer == null || !this.levelRenderState.customCloudsRenderer.renderClouds(this.levelRenderState, cameraPosition, cloudsType, cloudColor, cloudHeight, modelViewMatrix)) {
+                this.cloudRenderer.render(cloudColor, cloudsType, cloudHeight, cameraPosition, gameTime, partialTicks);
+            }
+        });
     }
 
+    /**
+     * @deprecated Neo: use {@link #addWeatherPass(FrameGraphBuilder, GpuBufferSlice, Matrix4f)} instead
+     */
+    @Deprecated
     private void addWeatherPass(FrameGraphBuilder frame, GpuBufferSlice fog) {
+        addWeatherPass(frame, fog, RenderSystem.getModelViewMatrix());
+    }
+
+    private void addWeatherPass(FrameGraphBuilder frame, GpuBufferSlice fog, Matrix4f modelViewMatrix) {
         int i = this.minecraft.options.getEffectiveRenderDistance() * 16;
         float f = this.minecraft.gameRenderer.getDepthFar();
         FramePass framepass = frame.addPass("weather");
@@ -763,6 +_,9 @@
             MultiBufferSource.BufferSource multibuffersource$buffersource = this.renderBuffers.bufferSource();
             CameraRenderState camerarenderstate = this.levelRenderState.cameraRenderState;
             this.weatherEffectRenderer.render(multibuffersource$buffersource, camerarenderstate.pos, this.levelRenderState.weatherRenderState);
+            Profiler.get().push("neoforge_render_after_weather");
+            net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLevelStageEvent.AfterWeather(this, this.levelRenderState, null, modelViewMatrix, this.getRenderableSections()));
+            Profiler.get().pop();
             this.worldBorderRenderer.render(this.levelRenderState.worldBorderRenderState, camerarenderstate.pos, i, f);
             multibuffersource$buffersource.endBatch();
         });
@@ -815,7 +_,7 @@
                             || camera.isDetached()
                             || camera.entity() instanceof LivingEntity && ((LivingEntity)camera.entity()).isSleeping()
                     )
-                    && (!(entity instanceof LocalPlayer) || camera.entity() == entity)) {
+                    && (!(entity instanceof LocalPlayer) || camera.entity() == entity || (entity == minecraft.player && !minecraft.player.isSpectator()))) { // Neo: render local player entity when it is not the camera entity
                     if (entity.tickCount == 0) {
                         entity.xOld = entity.getX();
                         entity.yOld = entity.getY();
@@ -828,6 +_,9 @@
                     if (entityrenderstate.appearsGlowing() && flag) {
                         output.haveGlowingEntities = true;
                     }
+                    else if (flag && entity.hasCustomOutlineRendering(this.minecraft.player)) { // FORGE: allow custom outline rendering
+                        output.haveGlowingEntities = true;
+                    }
                 }
             }
         }
@@ -857,12 +_,21 @@
         }
     }
 
+    /**
+     * @deprecated Neo: use {@link #extractVisibleBlockEntities(Camera, float, LevelRenderState, Frustum)} instead
+     */
+    @Deprecated
     private void extractVisibleBlockEntities(Camera camera, float deltaPartialTick, LevelRenderState levelRenderState) {
+        this.extractVisibleBlockEntities(camera, deltaPartialTick, levelRenderState, null);
+    }
+
+    private void extractVisibleBlockEntities(Camera camera, float deltaPartialTick, LevelRenderState levelRenderState, @Nullable Frustum frustum) {
         Vec3 vec3 = camera.position();
         double d0 = vec3.x();
         double d1 = vec3.y();
         double d2 = vec3.z();
         PoseStack posestack = new PoseStack();
+        boolean shouldShowEntityOutlines = this.shouldShowEntityOutlines();
 
         for (SectionRenderDispatcher.RenderSection sectionrenderdispatcher$rendersection : this.visibleSections) {
             List<BlockEntity> list = sectionrenderdispatcher$rendersection.getSectionMesh().getRenderableBlockEntities();
@@ -881,9 +_,12 @@
                     }
 
                     BlockEntityRenderState blockentityrenderstate = this.blockEntityRenderDispatcher
-                        .tryExtractRenderState(blockentity, deltaPartialTick, modelfeaturerenderer$crumblingoverlay);
+                        .tryExtractRenderState(blockentity, deltaPartialTick, modelfeaturerenderer$crumblingoverlay, frustum);
                     if (blockentityrenderstate != null) {
                         levelRenderState.blockEntityRenderStates.add(blockentityrenderstate);
+                        if (shouldShowEntityOutlines && blockentity.hasCustomOutlineRendering(this.minecraft.player)) { // Neo: allow custom outline rendering
+                            levelRenderState.haveGlowingEntities = true;
+                        }
                     }
                 }
             }
@@ -896,9 +_,12 @@
             if (blockentity1.isRemoved()) {
                 iterator.remove();
             } else {
-                BlockEntityRenderState blockentityrenderstate1 = this.blockEntityRenderDispatcher.tryExtractRenderState(blockentity1, deltaPartialTick, null);
+                BlockEntityRenderState blockentityrenderstate1 = this.blockEntityRenderDispatcher.tryExtractRenderState(blockentity1, deltaPartialTick, null, frustum);
                 if (blockentityrenderstate1 != null) {
                     levelRenderState.blockEntityRenderStates.add(blockentityrenderstate1);
+                    if (shouldShowEntityOutlines && blockentity1.hasCustomOutlineRendering(this.minecraft.player)) { // Neo: allow custom outline rendering
+                        levelRenderState.haveGlowingEntities = true;
+                    }
                 }
             }
         }
@@ -966,19 +_,23 @@
                 BlockPos blockpos = blockhitresult.getBlockPos();
                 BlockState blockstate = this.level.getBlockState(blockpos);
                 if (!blockstate.isAir() && this.level.getWorldBorder().isWithinBounds(blockpos)) {
-                    boolean flag = ItemBlockRenderTypes.getChunkRenderType(blockstate).sortOnUpload();
+                    boolean flag = net.neoforged.neoforge.client.ClientHooks.isInTranslucentBlockOutlinePass(this.level, blockpos, blockstate);
                     boolean flag1 = this.minecraft.options.highContrastBlockOutline().get();
                     CollisionContext collisioncontext = CollisionContext.of(camera.entity());
+                    var event = new net.neoforged.neoforge.client.event.ExtractBlockOutlineRenderStateEvent(this, this.level, blockpos, blockstate, blockhitresult, collisioncontext, flag, flag1, camera, levelRenderState);
+                    if (net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(event).isCanceled()) {
+                        return;
+                    }
                     VoxelShape voxelshape = blockstate.getShape(this.level, blockpos, collisioncontext);
                     if (SharedConstants.DEBUG_SHAPES) {
                         VoxelShape voxelshape1 = blockstate.getCollisionShape(this.level, blockpos, collisioncontext);
                         VoxelShape voxelshape2 = blockstate.getOcclusionShape();
                         VoxelShape voxelshape3 = blockstate.getInteractionShape(this.level, blockpos);
                         levelRenderState.blockOutlineRenderState = new BlockOutlineRenderState(
-                            blockpos, flag, flag1, voxelshape, voxelshape1, voxelshape2, voxelshape3
+                            blockpos, flag, flag1, voxelshape, voxelshape1, voxelshape2, voxelshape3, event.getCustomRenderers()
                         );
                     } else {
-                        levelRenderState.blockOutlineRenderState = new BlockOutlineRenderState(blockpos, flag, flag1, voxelshape);
+                        levelRenderState.blockOutlineRenderState = new BlockOutlineRenderState(blockpos, flag, flag1, voxelshape, event.getCustomRenderers());
                     }
                 }
             }
@@ -988,6 +_,14 @@
     private void renderBlockOutline(MultiBufferSource.BufferSource bufferSource, PoseStack poseStack, boolean onlyTranslucentBlocks, LevelRenderState levelRenderState) {
         BlockOutlineRenderState blockoutlinerenderstate = levelRenderState.blockOutlineRenderState;
         if (blockoutlinerenderstate != null) {
+            boolean cancel = false;
+            for (net.neoforged.neoforge.client.CustomBlockOutlineRenderer customRenderer : blockoutlinerenderstate.customRenderers()) {
+                cancel |= customRenderer.render(blockoutlinerenderstate, bufferSource, poseStack, onlyTranslucentBlocks, levelRenderState);
+            }
+            if (cancel) {
+                return;
+            }
+
             if (blockoutlinerenderstate.isTranslucent() == onlyTranslucentBlocks) {
                 Vec3 vec3 = levelRenderState.cameraRenderState.pos;
                 if (blockoutlinerenderstate.highContrast()) {
@@ -1168,7 +_,15 @@
         }
     }
 
+    /**
+     * @deprecated Neo: use {@link #addSkyPass(FrameGraphBuilder, Camera, GpuBufferSlice, Matrix4f)} instead
+     */
+    @Deprecated
     private void addSkyPass(FrameGraphBuilder frame, Camera camera, GpuBufferSlice skyFog) {
+        this.addSkyPass(frame, camera, skyFog, RenderSystem.getModelViewMatrix());
+    }
+
+    private void addSkyPass(FrameGraphBuilder frame, Camera camera, GpuBufferSlice skyFog, Matrix4f modelViewMatrix) {
         FogType fogtype = camera.getFluidInCamera();
         if (fogtype != FogType.POWDER_SNOW && fogtype != FogType.LAVA && !this.doesMobEffectBlockSky(camera)) {
             SkyRenderState skyrenderstate = this.levelRenderState.skyRenderState;
@@ -1179,6 +_,7 @@
                     this.targets.main = framepass.readsAndWrites(this.targets.main);
                     framepass.executes(
                         () -> {
+                            if (this.levelRenderState.customSkyboxRenderer == null || !this.levelRenderState.customSkyboxRenderer.renderSky(levelRenderState, skyrenderstate, modelViewMatrix, () -> RenderSystem.setShaderFog(skyFog))) {
                             RenderSystem.setShaderFog(skyFog);
                             if (skyrenderstate.skybox == DimensionType.Skybox.END) {
                                 skyrenderer.renderEndSky();
@@ -1205,6 +_,10 @@
                                     skyrenderer.renderDarkDisc();
                                 }
                             }
+                            } // Neo: End of skip if custom sky is in effect
+                            Profiler.get().push("neoforge_render_after_sky");
+                            net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLevelStageEvent.AfterSky(this, this.levelRenderState, null, modelViewMatrix, this.getRenderableSections()));
+                            Profiler.get().pop();
                         }
                     );
                 }
@@ -1459,7 +_,7 @@
         } else {
             int i = brightnessGetter.packedBrightness(level, pos);
             int j = LightTexture.block(i);
-            int k = state.getLightEmission();
+            int k = state.getLightEmission(level, pos);
             if (j < k) {
                 int l = LightTexture.sky(i);
                 return LightTexture.pack(k, l);
@@ -1516,6 +_,22 @@
 
     public CloudRenderer getCloudRenderer() {
         return this.cloudRenderer;
+    }
+
+    public int getTicks() {
+        return this.ticks;
+    }
+
+    public void iterateVisibleBlockEntities(java.util.function.Consumer<BlockEntity> blockEntityConsumer) {
+        for (var chunkInfo : this.visibleSections) {
+            chunkInfo.getSectionMesh().getRenderableBlockEntities().forEach(blockEntityConsumer);
+        }
+        this.level.getGloballyRenderedBlockEntities().forEach(blockEntityConsumer);
+    }
+
+    @org.jetbrains.annotations.ApiStatus.Internal
+    public Iterable<? extends net.neoforged.neoforge.client.IRenderableSection> getRenderableSections() {
+        return this.visibleSections;
     }
 
     public Gizmos.TemporaryCollection collectPerFrameGizmos() {
