--- a/net/minecraft/client/renderer/MapRenderer.java
+++ b/net/minecraft/client/renderer/MapRenderer.java
@@ -44,6 +_,10 @@
 
         for (MapRenderState.MapDecorationRenderState maprenderstate$mapdecorationrenderstate : mapRenderState.decorations) {
             if (!showOnlyFrame || maprenderstate$mapdecorationrenderstate.renderOnFrame) {
+                if (net.neoforged.neoforge.client.gui.map.MapDecorationRendererManager.render(maprenderstate$mapdecorationrenderstate, poseStack, submitNodeCollector, mapRenderState, MapRenderer.this.decorationSprites, showOnlyFrame, lightCoords, i)) {
+                    i++;
+                    continue;
+                }
                 poseStack.pushPose();
                 poseStack.translate(maprenderstate$mapdecorationrenderstate.x / 2.0F + 64.0F, maprenderstate$mapdecorationrenderstate.y / 2.0F + 64.0F, -0.02F);
                 poseStack.mulPose(Axis.ZP.rotationDegrees(maprenderstate$mapdecorationrenderstate.rot * 360 / 16.0F));
@@ -114,13 +_,15 @@
         mapRenderState.texture = this.mapTextureManager.prepareMapTexture(mapId, mapData);
         mapRenderState.decorations.clear();
 
+        net.neoforged.neoforge.client.renderstate.RenderStateExtensions.onUpdateMapRenderState(mapData, mapRenderState);
         for (MapDecoration mapdecoration : mapData.getDecorations()) {
-            mapRenderState.decorations.add(this.extractDecorationRenderState(mapdecoration));
+            mapRenderState.decorations.add(net.neoforged.neoforge.client.renderstate.RenderStateExtensions.onUpdateMapDecorationRenderState(mapdecoration.type(), mapData, mapRenderState, this.extractDecorationRenderState(mapdecoration)));
         }
     }
 
     private MapRenderState.MapDecorationRenderState extractDecorationRenderState(MapDecoration decoration) {
         MapRenderState.MapDecorationRenderState maprenderstate$mapdecorationrenderstate = new MapRenderState.MapDecorationRenderState();
+        maprenderstate$mapdecorationrenderstate.type = decoration.type();
         maprenderstate$mapdecorationrenderstate.atlasSprite = this.decorationSprites.getSprite(decoration.getSpriteLocation());
         maprenderstate$mapdecorationrenderstate.x = decoration.x();
         maprenderstate$mapdecorationrenderstate.y = decoration.y();
