--- a/net/minecraft/client/renderer/PostPass.java
+++ b/net/minecraft/client/renderer/PostPass.java
@@ -63,7 +_,7 @@
         RenderSystem.viewport(0, 0, (int)f, (int)f1);
         this.effect.setSampler("DiffuseSampler", this.inTarget::getColorTextureId);
 
-        for(int i = 0; i < this.auxAssets.size(); ++i) {
+        for (int i = 0; i < this.auxAssets.size(); i++) {
             this.effect.setSampler(this.auxNames.get(i), this.auxAssets.get(i));
             this.effect.safeGetUniform("AuxSize" + i).set((float)this.auxWidths.get(i).intValue(), (float)this.auxHeights.get(i).intValue());
         }
@@ -90,7 +_,7 @@
         this.outTarget.unbindWrite();
         this.inTarget.unbindRead();
 
-        for(Object object : this.auxAssets) {
+        for (Object object : this.auxAssets) {
             if (object instanceof RenderTarget) {
                 ((RenderTarget)object).unbindRead();
             }
