--- a/net/minecraft/client/renderer/RenderType.java
+++ b/net/minecraft/client/renderer/RenderType.java
@@ -587,23 +_,6 @@
             )
             .createCompositeState(false)
     );
-    private static final RenderType POISONOUS_POTATO = create(
-        "poisonous_potato",
-        DefaultVertexFormat.POSITION,
-        VertexFormat.Mode.QUADS,
-        1536,
-        false,
-        false,
-        RenderType.CompositeState.builder()
-            .setShaderState(RENDERTYPE_END_PORTAL_SHADER)
-            .setTextureState(
-                RenderStateShard.MultiTextureStateShard.builder()
-                    .add(new ResourceLocation("textures/item/poisonous_potato.png"), false, false)
-                    .add(new ResourceLocation("nothingtoseeheremovealong", "gui/menu_background.png"), false, false)
-                    .build()
-            )
-            .createCompositeState(false)
-    );
     private static final RenderType END_GATEWAY = create(
         "end_gateway",
         DefaultVertexFormat.POSITION,
@@ -806,7 +_,7 @@
         RenderType.CompositeState rendertype$compositestate = RenderType.CompositeState.builder()
             .setShaderState(RENDERTYPE_ARMOR_CUTOUT_NO_CULL_SHADER)
             .setTextureState(new RenderStateShard.TextureStateShard(p_299169_, false, false))
-            .setTransparencyState(RenderStateShard.TRANSLUCENT_TRANSPARENCY)
+            .setTransparencyState(NO_TRANSPARENCY)
             .setCullState(NO_CULL)
             .setLightmapState(LIGHTMAP)
             .setOverlayState(OVERLAY)
@@ -989,7 +_,7 @@
     }
 
     public static RenderType text(ResourceLocation p_110498_) {
-        return TEXT.apply(p_110498_);
+        return net.neoforged.neoforge.client.NeoForgeRenderTypes.getText(p_110498_);
     }
 
     public static RenderType textBackground() {
@@ -997,19 +_,19 @@
     }
 
     public static RenderType textIntensity(ResourceLocation p_173238_) {
-        return TEXT_INTENSITY.apply(p_173238_);
+        return net.neoforged.neoforge.client.NeoForgeRenderTypes.getTextIntensity(p_173238_);
     }
 
     public static RenderType textPolygonOffset(ResourceLocation p_181445_) {
-        return TEXT_POLYGON_OFFSET.apply(p_181445_);
+        return net.neoforged.neoforge.client.NeoForgeRenderTypes.getTextPolygonOffset(p_181445_);
     }
 
     public static RenderType textIntensityPolygonOffset(ResourceLocation p_181447_) {
-        return TEXT_INTENSITY_POLYGON_OFFSET.apply(p_181447_);
+        return net.neoforged.neoforge.client.NeoForgeRenderTypes.getTextIntensityPolygonOffset(p_181447_);
     }
 
     public static RenderType textSeeThrough(ResourceLocation p_110501_) {
-        return TEXT_SEE_THROUGH.apply(p_110501_);
+        return net.neoforged.neoforge.client.NeoForgeRenderTypes.getTextSeeThrough(p_110501_);
     }
 
     public static RenderType textBackgroundSeeThrough() {
@@ -1017,7 +_,7 @@
     }
 
     public static RenderType textIntensitySeeThrough(ResourceLocation p_173241_) {
-        return TEXT_INTENSITY_SEE_THROUGH.apply(p_173241_);
+        return net.neoforged.neoforge.client.NeoForgeRenderTypes.getTextIntensitySeeThrough(p_173241_);
     }
 
     public static RenderType lightning() {
@@ -1042,10 +_,6 @@
         return END_PORTAL;
     }
 
-    public static RenderType poisonousPotato() {
-        return POISONOUS_POTATO;
-    }
-
     public static RenderType endGateway() {
         return END_GATEWAY;
     }
@@ -1475,4 +_,17 @@
             return this.name;
         }
     }
+
+    // FORGE START
+    private int chunkLayerId = -1;
+    /** {@return the unique ID of this {@link RenderType} for chunk rendering purposes, or {@literal -1} if this is not a chunk {@link RenderType}} */
+    public final int getChunkLayerId() {
+        return chunkLayerId;
+    }
+    static {
+        int i = 0;
+        for (var layer : chunkBufferLayers())
+            layer.chunkLayerId = i++;
+    }
+    // FORGE END
 }
