--- a/net/minecraft/client/renderer/RenderStateShard.java
+++ b/net/minecraft/client/renderer/RenderStateShard.java
@@ -528,7 +_,7 @@
             super(() -> {
                 int i = 0;
 
-                for(Triple<ResourceLocation, Boolean, Boolean> triple : p_173123_) {
+                for (Triple<ResourceLocation, Boolean, Boolean> triple : p_173123_) {
                     TextureManager texturemanager = Minecraft.getInstance().getTextureManager();
                     texturemanager.getTexture(triple.getLeft()).setFilter(triple.getMiddle(), triple.getRight());
                     RenderSystem.setShaderTexture(i++, triple.getLeft());
