--- a/net/minecraft/client/renderer/RenderBuffers.java
+++ b/net/minecraft/client/renderer/RenderBuffers.java
@@ -43,6 +_,7 @@
             put((Object2ObjectLinkedOpenHashMap<RenderType, ByteBufferBuilder>)p_418569_, RenderType.entityGlint(ItemStackRenderState.FoilType.STANDARD));
             put((Object2ObjectLinkedOpenHashMap<RenderType, ByteBufferBuilder>)p_418569_, RenderType.waterMask());
         });
+        net.neoforged.fml.ModLoader.postEvent(new net.neoforged.neoforge.client.event.RegisterRenderBuffersEvent(sequencedmap));
         this.bufferSource = MultiBufferSource.immediateWithBuffers(sequencedmap, new ByteBufferBuilder(786432));
         this.outlineBufferSource = new OutlineBufferSource(this.bufferSource);
         SequencedMap<RenderType, ByteBufferBuilder> sequencedmap1 = Util.make(
