--- a/net/minecraft/client/renderer/ScreenEffectRenderer.java
+++ b/net/minecraft/client/renderer/ScreenEffectRenderer.java
@@ -30,18 +_,22 @@
     public static void renderScreenEffect(Minecraft p_110719_, PoseStack p_110720_) {
         Player player = p_110719_.player;
         if (!player.noPhysics) {
-            BlockState blockstate = getViewBlockingState(player);
-            if (blockstate != null) {
-                renderTex(p_110719_.getBlockRenderer().getBlockModelShaper().getParticleIcon(blockstate), p_110720_);
+            org.apache.commons.lang3.tuple.Pair<BlockState, BlockPos> overlay = getOverlayBlock(player);
+            if (overlay != null) {
+                if (!net.neoforged.neoforge.client.ClientHooks.renderBlockOverlay(player, p_110720_, net.neoforged.neoforge.client.event.RenderBlockScreenEffectEvent.OverlayType.BLOCK, overlay.getLeft(), overlay.getRight()))
+                    renderTex(p_110719_.getBlockRenderer().getBlockModelShaper().getTexture(overlay.getLeft(), p_110719_.level, overlay.getRight()), p_110720_);
             }
         }
 
         if (!p_110719_.player.isSpectator()) {
             if (p_110719_.player.isEyeInFluid(FluidTags.WATER)) {
+                if (!net.neoforged.neoforge.client.ClientHooks.renderWaterOverlay(player, p_110720_))
                 renderWater(p_110719_, p_110720_);
             }
+            else if (!player.getEyeInFluidType().isAir()) net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions.of(player.getEyeInFluidType()).renderOverlay(p_110719_, p_110720_);
 
             if (p_110719_.player.isOnFire()) {
+                if (!net.neoforged.neoforge.client.ClientHooks.renderFireOverlay(player, p_110720_))
                 renderFire(p_110719_, p_110720_);
             }
         }
@@ -49,16 +_,21 @@
 
     @Nullable
     private static BlockState getViewBlockingState(Player p_110717_) {
+        return getOverlayBlock(p_110717_).getLeft();
+    }
+
+    @Nullable
+    private static org.apache.commons.lang3.tuple.Pair<BlockState, BlockPos> getOverlayBlock(Player p_110717_) {
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int i = 0; i < 8; ++i) {
+        for (int i = 0; i < 8; i++) {
             double d0 = p_110717_.getX() + (double)(((float)((i >> 0) % 2) - 0.5F) * p_110717_.getBbWidth() * 0.8F);
             double d1 = p_110717_.getEyeY() + (double)(((float)((i >> 1) % 2) - 0.5F) * 0.1F * p_110717_.getScale());
             double d2 = p_110717_.getZ() + (double)(((float)((i >> 2) % 2) - 0.5F) * p_110717_.getBbWidth() * 0.8F);
             blockpos$mutableblockpos.set(d0, d1, d2);
             BlockState blockstate = p_110717_.level().getBlockState(blockpos$mutableblockpos);
             if (blockstate.getRenderShape() != RenderShape.INVISIBLE && blockstate.isViewBlocking(p_110717_.level(), blockpos$mutableblockpos)) {
-                return blockstate;
+                return org.apache.commons.lang3.tuple.Pair.of(blockstate, blockpos$mutableblockpos.immutable());
             }
         }
 
@@ -89,8 +_,12 @@
     }
 
     private static void renderWater(Minecraft p_110726_, PoseStack p_110727_) {
+        renderFluid(p_110726_, p_110727_, UNDERWATER_LOCATION);
+    }
+
+    public static void renderFluid(Minecraft p_110726_, PoseStack p_110727_, ResourceLocation texture) {
         RenderSystem.setShader(GameRenderer::getPositionTexShader);
-        RenderSystem.setShaderTexture(0, UNDERWATER_LOCATION);
+        RenderSystem.setShaderTexture(0, texture);
         BufferBuilder bufferbuilder = Tesselator.getInstance().getBuilder();
         BlockPos blockpos = BlockPos.containing(p_110726_.player.getX(), p_110726_.player.getEyeY(), p_110726_.player.getZ());
         float f = LightTexture.getBrightness(p_110726_.player.level().dimensionType(), p_110726_.player.level().getMaxLocalRawBrightness(blockpos));
@@ -136,7 +_,7 @@
         float f10 = Mth.lerp(f6, f4, f5);
         float f11 = 1.0F;
 
-        for(int i = 0; i < 2; ++i) {
+        for (int i = 0; i < 2; i++) {
             p_110730_.pushPose();
             float f12 = -0.5F;
             float f13 = 0.5F;
