--- a/net/minecraft/client/renderer/SectionBufferBuilderPack.java
+++ b/net/minecraft/client/renderer/SectionBufferBuilderPack.java
@@ -11,7 +_,7 @@
     public static final int TOTAL_BUFFERS_SIZE = RenderType.chunkBufferLayers().stream().mapToInt(RenderType::bufferSize).sum();
     private final Map<RenderType, BufferBuilder> builders = RenderType.chunkBufferLayers()
         .stream()
-        .collect(Collectors.toMap(p_294295_ -> p_294295_, p_295012_ -> new BufferBuilder(p_295012_.bufferSize())));
+        .collect(Collectors.toMap(p_294295_ -> (RenderType)p_294295_, p_295012_ -> new BufferBuilder(p_295012_.bufferSize())));
 
     public BufferBuilder builder(RenderType p_295847_) {
         return this.builders.get(p_295847_);
