--- a/net/minecraft/commands/arguments/selector/EntitySelectorParser.java
+++ b/net/minecraft/commands/arguments/selector/EntitySelectorParser.java
@@ -196,11 +_,7 @@
         double d1 = (double)Mth.wrapDegrees(p_121255_.max() == null ? 359.0F : p_121255_.max());
         return p_175137_ -> {
             double d2 = Mth.wrapDegrees(p_121256_.applyAsDouble(p_175137_));
-            if (d0 > d1) {
-                return d2 >= d0 || d2 <= d1;
-            } else {
-                return d2 >= d0 && d2 <= d1;
-            }
+            return d0 > d1 ? d2 >= d0 || d2 <= d1 : d2 >= d0 && d2 <= d1;
         };
     }
 
@@ -280,7 +_,7 @@
         this.suggestions = this::suggestOptionsKey;
         this.reader.skipWhitespace();
 
-        while(this.reader.canRead() && this.reader.peek() != ']') {
+        while (this.reader.canRead() && this.reader.peek() != ']') {
             this.reader.skipWhitespace();
             int i = this.reader.getCursor();
             String s = this.reader.readString();
@@ -463,6 +_,9 @@
             }
 
             this.reader.skip();
+            EntitySelector forgeSelector = net.neoforged.neoforge.common.command.EntitySelectorManager.parseSelector(this);
+            if (forgeSelector != null)
+                return forgeSelector;
             this.parseSelector();
         } else {
             this.parseNameOrUUID();
@@ -478,6 +_,7 @@
         p_121248_.suggest("@r", Component.translatable("argument.entity.selector.randomPlayer"));
         p_121248_.suggest("@s", Component.translatable("argument.entity.selector.self"));
         p_121248_.suggest("@e", Component.translatable("argument.entity.selector.allEntities"));
+        net.neoforged.neoforge.common.command.EntitySelectorManager.fillSelectorSuggestions(p_121248_);
     }
 
     private CompletableFuture<Suggestions> suggestNameOrSelector(SuggestionsBuilder p_121287_, Consumer<SuggestionsBuilder> p_121288_) {
