--- a/net/minecraft/commands/arguments/selector/EntitySelector.java
+++ b/net/minecraft/commands/arguments/selector/EntitySelector.java
@@ -104,7 +_,7 @@
     }
 
     private void checkPermissions(CommandSourceStack p_121169_) throws CommandSyntaxException {
-        if (this.usesSelector && !p_121169_.hasPermission(2)) {
+        if (this.usesSelector && !net.neoforged.neoforge.common.CommonHooks.canUseEntitySelectors(p_121169_)) {
             throw EntityArgument.ERROR_SELECTORS_NOT_ALLOWED.create();
         }
     }
@@ -133,7 +_,7 @@
             ServerPlayer serverplayer = p_251934_.getServer().getPlayerList().getPlayerByName(this.playerName);
             return (List<? extends Entity>)(serverplayer == null ? Collections.emptyList() : Lists.newArrayList(serverplayer));
         } else if (this.entityUUID != null) {
-            for(ServerLevel serverlevel1 : p_251934_.getServer().getAllLevels()) {
+            for (ServerLevel serverlevel1 : p_251934_.getServer().getAllLevels()) {
                 Entity entity = serverlevel1.getEntity(this.entityUUID);
                 if (entity != null) {
                     return Lists.newArrayList(entity);
@@ -153,7 +_,7 @@
                 if (this.isWorldLimited()) {
                     this.addEntities(list, p_251934_.getLevel(), vec3, predicate);
                 } else {
-                    for(ServerLevel serverlevel : p_251934_.getServer().getAllLevels()) {
+                    for (ServerLevel serverlevel : p_251934_.getServer().getAllLevels()) {
                         this.addEntities(list, serverlevel, vec3, predicate);
                     }
                 }
@@ -200,8 +_,7 @@
             Vec3 vec3 = this.position.apply(p_121167_.getPosition());
             Predicate<Entity> predicate = this.getPredicate(vec3);
             if (this.currentEntity) {
-                Entity entity = p_121167_.getEntity();
-                if (entity instanceof ServerPlayer serverplayer3 && predicate.test(serverplayer3)) {
+                if (p_121167_.getEntity() instanceof ServerPlayer serverplayer3 && predicate.test(serverplayer3)) {
                     return Lists.newArrayList(serverplayer3);
                 }
 
@@ -214,7 +_,7 @@
                 } else {
                     list = Lists.newArrayList();
 
-                    for(ServerPlayer serverplayer : p_121167_.getServer().getPlayerList().getPlayers()) {
+                    for (ServerPlayer serverplayer : p_121167_.getServer().getPlayerList().getPlayers()) {
                         if (predicate.test(serverplayer)) {
                             list.add(serverplayer);
                             if (list.size() >= i) {
