--- a/net/minecraft/world/level/storage/loot/functions/ApplyExplosionDecay.java
+++ b/net/minecraft/world/level/storage/loot/functions/ApplyExplosionDecay.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.functions;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class ApplyExplosionDecay extends LootItemConditionalFunction {
-    public static final Codec<ApplyExplosionDecay> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<ApplyExplosionDecay> CODEC = RecordCodecBuilder.mapCodec(
         p_298525_ -> commonFields(p_298525_).apply(p_298525_, ApplyExplosionDecay::new)
     );
 
@@ -33,9 +_,9 @@
             int i = p_80034_.getCount();
             int j = 0;
 
-            for(int k = 0; k < i; ++k) {
+            for (int k = 0; k < i; k++) {
                 if (randomsource.nextFloat() <= f1) {
-                    ++j;
+                    j++;
                 }
             }
 
