--- a/net/minecraft/world/level/storage/loot/functions/ApplyBonusCount.java
+++ b/net/minecraft/world/level/storage/loot/functions/ApplyBonusCount.java
@@ -43,7 +_,7 @@
     private static final MapCodec<ApplyBonusCount.Formula> FORMULA_CODEC = ExtraCodecs.dispatchOptionalValue(
         "formula", "parameters", FORMULA_TYPE_CODEC, ApplyBonusCount.Formula::getType, ApplyBonusCount.FormulaType::codec
     );
-    public static final Codec<ApplyBonusCount> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<ApplyBonusCount> CODEC = RecordCodecBuilder.mapCodec(
         p_298054_ -> commonFields(p_298054_)
                 .and(
                     p_298054_.group(
@@ -114,9 +_,9 @@
 
         @Override
         public int calculateNewCount(RandomSource p_230965_, int p_230966_, int p_230967_) {
-            for(int i = 0; i < p_230967_ + this.extraRounds; ++i) {
+            for (int i = 0; i < p_230967_ + this.extraRounds; i++) {
                 if (p_230965_.nextFloat() < this.probability) {
-                    ++p_230966_;
+                    p_230966_++;
                 }
             }
 
