--- a/net/minecraft/world/level/storage/loot/functions/CopyComponentsFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/CopyComponentsFunction.java
@@ -2,10 +_,15 @@
 
 import com.google.common.collect.ImmutableList;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 import java.util.Set;
+import java.util.function.Predicate;
+import net.minecraft.Util;
 import net.minecraft.core.component.DataComponentMap;
 import net.minecraft.core.component.DataComponentType;
 import net.minecraft.util.StringRepresentable;
@@ -17,23 +_,36 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class CopyComponentsFunction extends LootItemConditionalFunction {
-    public static final Codec<CopyComponentsFunction> CODEC = RecordCodecBuilder.create(
-        p_331155_ -> commonFields(p_331155_)
+    public static final MapCodec<CopyComponentsFunction> CODEC = RecordCodecBuilder.mapCodec(
+        p_338127_ -> commonFields(p_338127_)
                 .and(
-                    p_331155_.group(
+                    p_338127_.group(
                         CopyComponentsFunction.Source.CODEC.fieldOf("source").forGetter(p_331312_ -> p_331312_.source),
-                        DataComponentType.CODEC.listOf().fieldOf("components").forGetter(p_330277_ -> p_330277_.components)
+                        DataComponentType.CODEC.listOf().optionalFieldOf("include").forGetter(p_338132_ -> p_338132_.include),
+                        DataComponentType.CODEC.listOf().optionalFieldOf("exclude").forGetter(p_338126_ -> p_338126_.exclude)
                     )
                 )
-                .apply(p_331155_, CopyComponentsFunction::new)
+                .apply(p_338127_, CopyComponentsFunction::new)
     );
     private final CopyComponentsFunction.Source source;
-    private final List<DataComponentType<?>> components;
+    private final Optional<List<DataComponentType<?>>> include;
+    private final Optional<List<DataComponentType<?>>> exclude;
+    private final Predicate<DataComponentType<?>> bakedPredicate;
 
-    CopyComponentsFunction(List<LootItemCondition> p_330806_, CopyComponentsFunction.Source p_330881_, List<DataComponentType<?>> p_331546_) {
+    CopyComponentsFunction(
+        List<LootItemCondition> p_330806_,
+        CopyComponentsFunction.Source p_330881_,
+        Optional<List<DataComponentType<?>>> p_338636_,
+        Optional<List<DataComponentType<?>>> p_338680_
+    ) {
         super(p_330806_);
         this.source = p_330881_;
-        this.components = List.copyOf(p_331546_);
+        this.include = p_338636_.map(List::copyOf);
+        this.exclude = p_338680_.map(List::copyOf);
+        List<Predicate<DataComponentType<?>>> list = new ArrayList<>(2);
+        p_338680_.ifPresent(p_338129_ -> list.add(p_338134_ -> !p_338129_.contains(p_338134_)));
+        p_338636_.ifPresent(p_338131_ -> list.add(p_338131_::contains));
+        this.bakedPredicate = Util.allOf(list);
     }
 
     @Override
@@ -49,7 +_,7 @@
     @Override
     public ItemStack run(ItemStack p_330563_, LootContext p_331220_) {
         DataComponentMap datacomponentmap = this.source.get(p_331220_);
-        p_330563_.applyComponents(datacomponentmap.filter(this.components::contains));
+        p_330563_.applyComponents(datacomponentmap.filter(this.bakedPredicate));
         return p_330563_;
     }
 
@@ -59,14 +_,28 @@
 
     public static class Builder extends LootItemConditionalFunction.Builder<CopyComponentsFunction.Builder> {
         private final CopyComponentsFunction.Source source;
-        private final ImmutableList.Builder<DataComponentType<?>> components = ImmutableList.builder();
+        private Optional<ImmutableList.Builder<DataComponentType<?>>> include = Optional.empty();
+        private Optional<ImmutableList.Builder<DataComponentType<?>>> exclude = Optional.empty();
 
         Builder(CopyComponentsFunction.Source p_331304_) {
             this.source = p_331304_;
         }
 
-        public CopyComponentsFunction.Builder copy(DataComponentType<?> p_331969_) {
-            this.components.add(p_331969_);
+        public CopyComponentsFunction.Builder include(DataComponentType<?> p_338267_) {
+            if (this.include.isEmpty()) {
+                this.include = Optional.of(ImmutableList.builder());
+            }
+
+            this.include.get().add(p_338267_);
+            return this;
+        }
+
+        public CopyComponentsFunction.Builder exclude(DataComponentType<?> p_338302_) {
+            if (this.exclude.isEmpty()) {
+                this.exclude = Optional.of(ImmutableList.builder());
+            }
+
+            this.exclude.get().add(p_338302_);
             return this;
         }
 
@@ -76,7 +_,9 @@
 
         @Override
         public LootItemFunction build() {
-            return new CopyComponentsFunction(this.getConditions(), this.source, this.components.build());
+            return new CopyComponentsFunction(
+                this.getConditions(), this.source, this.include.map(ImmutableList.Builder::build), this.exclude.map(ImmutableList.Builder::build)
+            );
         }
     }
 
@@ -91,21 +_,21 @@
         }
 
         public DataComponentMap get(LootContext p_331014_) {
-            switch(this) {
+            switch (this) {
                 case BLOCK_ENTITY:
                     BlockEntity blockentity = p_331014_.getParamOrNull(LootContextParams.BLOCK_ENTITY);
                     return blockentity != null ? blockentity.collectComponents() : DataComponentMap.EMPTY;
                 default:
-                    throw new IncompatibleClassChangeError();
+                    throw new MatchException(null, null);
             }
         }
 
         public Set<LootContextParam<?>> getReferencedContextParams() {
-            switch(this) {
+            switch (this) {
                 case BLOCK_ENTITY:
                     return Set.of(LootContextParams.BLOCK_ENTITY);
                 default:
-                    throw new IncompatibleClassChangeError();
+                    throw new MatchException(null, null);
             }
         }
 
