--- a/net/minecraft/world/level/storage/loot/functions/CopyNameFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/CopyNameFunction.java
@@ -2,6 +_,7 @@
 
 import com.google.common.collect.ImmutableSet;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -16,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class CopyNameFunction extends LootItemConditionalFunction {
-    public static final Codec<CopyNameFunction> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<CopyNameFunction> CODEC = RecordCodecBuilder.mapCodec(
         p_298065_ -> commonFields(p_298065_)
                 .and(CopyNameFunction.NameSource.CODEC.fieldOf("source").forGetter(p_298068_ -> p_298068_.source))
                 .apply(p_298065_, CopyNameFunction::new)
@@ -40,8 +_,7 @@
 
     @Override
     public ItemStack run(ItemStack p_80185_, LootContext p_80186_) {
-        Object object = p_80186_.getParamOrNull(this.source.param);
-        if (object instanceof Nameable nameable) {
+        if (p_80186_.getParamOrNull(this.source.param) instanceof Nameable nameable) {
             p_80185_.set(DataComponents.CUSTOM_NAME, nameable.getCustomName());
         }
 
