--- a/net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction.java
@@ -2,6 +_,7 @@
 
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.ArrayList;
@@ -11,7 +_,6 @@
 import net.minecraft.core.Holder;
 import net.minecraft.core.HolderSet;
 import net.minecraft.core.registries.BuiltInRegistries;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.Mth;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.item.ItemStack;
@@ -27,10 +_,10 @@
         .holderByNameCodec()
         .listOf()
         .xmap(HolderSet::direct, p_298082_ -> p_298082_.stream().toList());
-    public static final Codec<EnchantRandomlyFunction> CODEC = RecordCodecBuilder.create(
-        p_298072_ -> commonFields(p_298072_)
-                .and(ExtraCodecs.strictOptionalField(ENCHANTMENT_SET_CODEC, "enchantments").forGetter(p_298083_ -> p_298083_.enchantments))
-                .apply(p_298072_, EnchantRandomlyFunction::new)
+    public static final MapCodec<EnchantRandomlyFunction> CODEC = RecordCodecBuilder.mapCodec(
+        p_338140_ -> commonFields(p_338140_)
+                .and(ENCHANTMENT_SET_CODEC.optionalFieldOf("enchantments").forGetter(p_298083_ -> p_298083_.enchantments))
+                .apply(p_338140_, EnchantRandomlyFunction::new)
     );
     private final Optional<HolderSet<Enchantment>> enchantments;
 
@@ -48,12 +_,13 @@
     public ItemStack run(ItemStack p_80429_, LootContext p_80430_) {
         RandomSource randomsource = p_80430_.getRandom();
         Optional<Holder<Enchantment>> optional = this.enchantments
-            .flatMap(p_298077_ -> p_298077_.getRandomElement(randomsource))
+            .<Holder<Enchantment>>flatMap(p_298077_ -> p_298077_.getRandomElement(randomsource))
             .or(
                 () -> {
                     boolean flag = p_80429_.is(Items.BOOK);
                     List<Holder.Reference<Enchantment>> list = BuiltInRegistries.ENCHANTMENT
                         .holders()
+                        .filter(p_338136_ -> p_338136_.value().isEnabled(p_80430_.getLevel().enabledFeatures()))
                         .filter(p_298078_ -> p_298078_.value().isDiscoverable())
                         .filter(p_298075_ -> flag || p_298075_.value().canEnchant(p_80429_))
                         .toList();
