--- a/net/minecraft/world/level/storage/loot/functions/FillPlayerHead.java
+++ b/net/minecraft/world/level/storage/loot/functions/FillPlayerHead.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.storage.loot.functions;
 
 import com.google.common.collect.ImmutableSet;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -16,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class FillPlayerHead extends LootItemConditionalFunction {
-    public static final Codec<FillPlayerHead> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<FillPlayerHead> CODEC = RecordCodecBuilder.mapCodec(
         p_298087_ -> commonFields(p_298087_)
                 .and(LootContext.EntityTarget.CODEC.fieldOf("entity").forGetter(p_298086_ -> p_298086_.entityTarget))
                 .apply(p_298087_, FillPlayerHead::new)
@@ -40,11 +_,8 @@
 
     @Override
     public ItemStack run(ItemStack p_80608_, LootContext p_80609_) {
-        if (p_80608_.is(Items.PLAYER_HEAD)) {
-            Object object = p_80609_.getParamOrNull(this.entityTarget.getParam());
-            if (object instanceof Player player) {
-                p_80608_.set(DataComponents.PROFILE, new ResolvableProfile(player.getGameProfile()));
-            }
+        if (p_80608_.is(Items.PLAYER_HEAD) && p_80609_.getParamOrNull(this.entityTarget.getParam()) instanceof Player player) {
+            p_80608_.set(DataComponents.PROFILE, new ResolvableProfile(player.getGameProfile()));
         }
 
         return p_80608_;
