--- a/net/minecraft/world/level/storage/loot/functions/FunctionReference.java
+++ b/net/minecraft/world/level/storage/loot/functions/FunctionReference.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.storage.loot.functions;
 
 import com.mojang.logging.LogUtils;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -16,7 +_,7 @@
 
 public class FunctionReference extends LootItemConditionalFunction {
     private static final Logger LOGGER = LogUtils.getLogger();
-    public static final Codec<FunctionReference> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<FunctionReference> CODEC = RecordCodecBuilder.mapCodec(
         p_335341_ -> commonFields(p_335341_)
                 .and(ResourceKey.codec(Registries.ITEM_MODIFIER).fieldOf("name").forGetter(p_335337_ -> p_335337_.name))
                 .apply(p_335341_, FunctionReference::new)
@@ -42,7 +_,7 @@
             p_279281_.resolver()
                 .get(Registries.ITEM_MODIFIER, this.name)
                 .ifPresentOrElse(
-                    p_335339_ -> ((LootItemFunction)p_335339_.value()).validate(p_279281_.enterElement(".{" + this.name.location() + "}", this.name)),
+                    p_339571_ -> p_339571_.value().validate(p_279281_.enterElement(".{" + this.name.location() + "}", this.name)),
                     () -> p_279281_.reportProblem("Unknown function table called " + this.name.location())
                 );
         }
