--- a/net/minecraft/world/level/storage/loot/functions/LootItemFunctionType.java
+++ b/net/minecraft/world/level/storage/loot/functions/LootItemFunctionType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.functions;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 
-public record LootItemFunctionType(Codec<? extends LootItemFunction> codec) {
+public record LootItemFunctionType(MapCodec<? extends LootItemFunction> codec) {
 }
