--- a/net/minecraft/world/level/storage/loot/functions/LootItemFunctions.java
+++ b/net/minecraft/world/level/storage/loot/functions/LootItemFunctions.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.storage.loot.functions;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.List;
 import java.util.function.BiFunction;
 import net.minecraft.core.Holder;
@@ -9,7 +_,6 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.resources.RegistryFileCodec;
 import net.minecraft.resources.ResourceLocation;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.storage.loot.LootContext;
 
@@ -18,9 +_,7 @@
     public static final Codec<LootItemFunction> TYPED_CODEC = BuiltInRegistries.LOOT_FUNCTION_TYPE
         .byNameCodec()
         .dispatch("function", LootItemFunction::getType, LootItemFunctionType::codec);
-    public static final Codec<LootItemFunction> ROOT_CODEC = ExtraCodecs.lazyInitializedCodec(
-        () -> ExtraCodecs.withAlternative(TYPED_CODEC, SequenceFunction.INLINE_CODEC)
-    );
+    public static final Codec<LootItemFunction> ROOT_CODEC = Codec.lazyInitialized(() -> Codec.withAlternative(TYPED_CODEC, SequenceFunction.INLINE_CODEC));
     public static final Codec<Holder<LootItemFunction>> CODEC = RegistryFileCodec.create(Registries.ITEM_MODIFIER, ROOT_CODEC);
     public static final LootItemFunctionType SET_COUNT = register("set_count", SetItemCountFunction.CODEC);
     public static final LootItemFunctionType ENCHANT_WITH_LEVELS = register("enchant_with_levels", EnchantWithLevelsFunction.CODEC);
@@ -57,15 +_,16 @@
     public static final LootItemFunctionType SET_WRITTEN_BOOK_PAGES = register("set_written_book_pages", SetWrittenBookPagesFunction.CODEC);
     public static final LootItemFunctionType SET_WRITABLE_BOOK_PAGES = register("set_writable_book_pages", SetWritableBookPagesFunction.CODEC);
     public static final LootItemFunctionType TOGGLE_TOOLTIPS = register("toggle_tooltips", ToggleTooltips.CODEC);
+    public static final LootItemFunctionType SET_OMINOUS_BOTTLE_AMPLIFIER = register("set_ominous_bottle_amplifier", SetOminousBottleAmplifierFunction.CODEC);
 
-    private static LootItemFunctionType register(String p_80763_, Codec<? extends LootItemFunction> p_298583_) {
-        return Registry.register(BuiltInRegistries.LOOT_FUNCTION_TYPE, new ResourceLocation(p_80763_), new LootItemFunctionType(p_298583_));
+    private static LootItemFunctionType register(String p_80763_, MapCodec<? extends LootItemFunction> p_338516_) {
+        return Registry.register(BuiltInRegistries.LOOT_FUNCTION_TYPE, new ResourceLocation(p_80763_), new LootItemFunctionType(p_338516_));
     }
 
     public static BiFunction<ItemStack, LootContext, ItemStack> compose(List<? extends BiFunction<ItemStack, LootContext, ItemStack>> p_298851_) {
         List<BiFunction<ItemStack, LootContext, ItemStack>> list = List.copyOf(p_298851_);
 
-        return switch(list.size()) {
+        return switch (list.size()) {
             case 0 -> IDENTITY;
             case 1 -> (BiFunction)list.get(0);
             case 2 -> {
@@ -74,7 +_,7 @@
                 yield (p_80768_, p_80769_) -> bifunction1.apply(bifunction.apply(p_80768_, p_80769_), p_80769_);
             }
             default -> (p_298097_, p_298098_) -> {
-            for(BiFunction<ItemStack, LootContext, ItemStack> bifunction2 : list) {
+            for (BiFunction<ItemStack, LootContext, ItemStack> bifunction2 : list) {
                 p_298097_ = bifunction2.apply(p_298097_, p_298098_);
             }
 
