--- a/net/minecraft/world/level/storage/loot/functions/SetContainerContents.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetContainerContents.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.storage.loot.functions;
 
 import com.google.common.collect.ImmutableList;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -21,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class SetContainerContents extends LootItemConditionalFunction {
-    public static final Codec<SetContainerContents> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SetContainerContents> CODEC = RecordCodecBuilder.mapCodec(
         p_298101_ -> commonFields(p_298101_)
                 .and(
                     p_298101_.group(
@@ -66,7 +_,7 @@
     public void validate(ValidationContext p_80918_) {
         super.validate(p_80918_);
 
-        for(int i = 0; i < this.entries.size(); ++i) {
+        for (int i = 0; i < this.entries.size(); i++) {
             this.entries.get(i).validate(p_80918_.forChild(".entry[" + i + "]"));
         }
     }
