--- a/net/minecraft/world/level/storage/loot/functions/SetComponentsFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetComponentsFunction.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.functions;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class SetComponentsFunction extends LootItemConditionalFunction {
-    public static final Codec<SetComponentsFunction> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SetComponentsFunction> CODEC = RecordCodecBuilder.mapCodec(
         p_331915_ -> commonFields(p_331915_)
                 .and(DataComponentPatch.CODEC.fieldOf("components").forGetter(p_331262_ -> p_331262_.components))
                 .apply(p_331915_, SetComponentsFunction::new)
