--- a/net/minecraft/world/level/storage/loot/functions/SetCustomDataFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetCustomDataFunction.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.functions;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -13,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class SetCustomDataFunction extends LootItemConditionalFunction {
-    public static final Codec<SetCustomDataFunction> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SetCustomDataFunction> CODEC = RecordCodecBuilder.mapCodec(
         p_330351_ -> commonFields(p_330351_)
                 .and(TagParser.AS_CODEC.fieldOf("tag").forGetter(p_330759_ -> p_330759_.tag))
                 .apply(p_330351_, SetCustomDataFunction::new)
