--- a/net/minecraft/world/level/storage/loot/functions/SetItemCountFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetItemCountFunction.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.storage.loot.functions;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -14,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.providers.number.NumberProviders;
 
 public class SetItemCountFunction extends LootItemConditionalFunction {
-    public static final Codec<SetItemCountFunction> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SetItemCountFunction> CODEC = RecordCodecBuilder.mapCodec(
         p_298131_ -> commonFields(p_298131_)
                 .and(
                     p_298131_.group(
