--- a/net/minecraft/world/level/storage/loot/functions/SetPotionFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetPotionFunction.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.functions;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -14,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class SetPotionFunction extends LootItemConditionalFunction {
-    public static final Codec<SetPotionFunction> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SetPotionFunction> CODEC = RecordCodecBuilder.mapCodec(
         p_330194_ -> commonFields(p_330194_)
                 .and(BuiltInRegistries.POTION.holderByNameCodec().fieldOf("id").forGetter(p_298158_ -> p_298158_.potion))
                 .apply(p_330194_, SetPotionFunction::new)
