--- a/net/minecraft/world/Containers.java
+++ b/net/minecraft/world/Containers.java
@@ -7,7 +_,6 @@
 import net.minecraft.world.entity.item.ItemEntity;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
-import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class Containers {
@@ -20,7 +_,7 @@
     }
 
     private static void dropContents(Level p_18987_, double p_18988_, double p_18989_, double p_18990_, Container p_18991_) {
-        for(int i = 0; i < p_18991_.getContainerSize(); ++i) {
+        for (int i = 0; i < p_18991_.getContainerSize(); i++) {
             dropItemStack(p_18987_, p_18988_, p_18989_, p_18990_, p_18991_.getItem(i));
         }
     }
@@ -37,7 +_,7 @@
         double d4 = Math.floor(p_18995_) + p_18993_.random.nextDouble() * d1;
         double d5 = Math.floor(p_18996_) + p_18993_.random.nextDouble() * d1 + d2;
 
-        while(!p_18997_.isEmpty()) {
+        while (!p_18997_.isEmpty()) {
             ItemEntity itementity = new ItemEntity(p_18993_, d3, d4, d5, p_18997_.split(p_18993_.random.nextInt(21) + 10));
             float f = 0.05F;
             itementity.setDeltaMovement(
@@ -51,8 +_,7 @@
 
     public static void dropContentsOnDestroy(BlockState p_306110_, BlockState p_306286_, Level p_305891_, BlockPos p_306156_) {
         if (!p_306110_.is(p_306286_.getBlock())) {
-            BlockEntity blockentity = p_305891_.getBlockEntity(p_306156_);
-            if (blockentity instanceof Container container) {
+            if (p_305891_.getBlockEntity(p_306156_) instanceof Container container) {
                 dropContents(p_305891_, p_306156_, container);
                 p_305891_.updateNeighbourForOutputSignal(p_306156_, p_306110_.getBlock());
             }
