--- a/com/mojang/realmsclient/client/FileDownload.java
+++ b/com/mojang/realmsclient/client/FileDownload.java
@@ -228,7 +_,7 @@
     public static String findAvailableFolderName(String p_87002_) {
         p_87002_ = p_87002_.replaceAll("[\\./\"]", "_");
 
-        for(String s : INVALID_FILE_NAMES) {
+        for (String s : INVALID_FILE_NAMES) {
             if (p_87002_.equalsIgnoreCase(s)) {
                 p_87002_ = "_" + p_87002_ + "_";
             }
@@ -241,7 +_,7 @@
         Pattern pattern = Pattern.compile(".*-([0-9]+)$");
         int i = 1;
 
-        for(char c0 : SharedConstants.ILLEGAL_FILE_CHARACTERS) {
+        for (char c0 : SharedConstants.ILLEGAL_FILE_CHARACTERS) {
             p_86992_ = p_86992_.replace(c0, '_');
         }
 
@@ -252,7 +_,7 @@
         p_86992_ = findAvailableFolderName(p_86992_);
 
         try {
-            for(LevelStorageSource.LevelDirectory levelstoragesource$leveldirectory : p_86994_.findLevelCandidates()) {
+            for (LevelStorageSource.LevelDirectory levelstoragesource$leveldirectory : p_86994_.findLevelCandidates()) {
                 String s1 = levelstoragesource$leveldirectory.directoryName();
                 if (s1.toLowerCase(Locale.ROOT).startsWith(p_86992_.toLowerCase(Locale.ROOT))) {
                     Matcher matcher = pattern.matcher(s1);
@@ -262,7 +_,7 @@
                             i = j;
                         }
                     } else {
-                        ++i;
+                        i++;
                     }
                 }
             }
@@ -280,8 +_,8 @@
             if (!p_86994_.isNewLevelIdAcceptable(s)) {
                 boolean flag = false;
 
-                while(!flag) {
-                    ++i;
+                while (!flag) {
+                    i++;
                     s = p_86992_ + (i == 1 ? "" : "-" + i);
                     if (p_86994_.isNewLevelIdAcceptable(s)) {
                         flag = true;
@@ -297,7 +_,7 @@
             file1.mkdir();
             tararchiveinputstream = new TarArchiveInputStream(new GzipCompressorInputStream(new BufferedInputStream(new FileInputStream(p_86993_))));
 
-            for(TarArchiveEntry tararchiveentry = tararchiveinputstream.getNextTarEntry();
+            for (TarArchiveEntry tararchiveentry = tararchiveinputstream.getNextTarEntry();
                 tararchiveentry != null;
                 tararchiveentry = tararchiveinputstream.getNextTarEntry()
             ) {
