--- a/com/mojang/realmsclient/client/FileUpload.java
+++ b/com/mojang/realmsclient/client/FileUpload.java
@@ -207,14 +_,14 @@
                 byte[] abyte = new byte[4096];
                 int j;
                 if (this.length < 0L) {
-                    while((j = inputstream.read(abyte)) != -1) {
+                    while ((j = inputstream.read(abyte)) != -1) {
                         p_87109_.write(abyte, 0, j);
                         this.uploadStatus.bytesWritten += (long)j;
                     }
                 } else {
                     long i = this.length;
 
-                    while(i > 0L) {
+                    while (i > 0L) {
                         j = inputstream.read(abyte, 0, (int)Math.min(4096L, i));
                         if (j == -1) {
                             break;
