--- a/com/mojang/realmsclient/client/Ping.java
+++ b/com/mojang/realmsclient/client/Ping.java
@@ -15,13 +_,13 @@
 @OnlyIn(Dist.CLIENT)
 public class Ping {
     public static List<RegionPingResult> ping(Ping.Region... p_87131_) {
-        for(Ping.Region ping$region : p_87131_) {
+        for (Ping.Region ping$region : p_87131_) {
             ping(ping$region.endpoint);
         }
 
         List<RegionPingResult> list = Lists.newArrayList();
 
-        for(Ping.Region ping$region1 : p_87131_) {
+        for (Ping.Region ping$region1 : p_87131_) {
             list.add(new RegionPingResult(ping$region1.name, ping(ping$region1.endpoint)));
         }
 
@@ -34,7 +_,7 @@
         long j = 0L;
         Socket socket = null;
 
-        for(int k = 0; k < 5; ++k) {
+        for (int k = 0; k < 5; k++) {
             try {
                 SocketAddress socketaddress = new InetSocketAddress(p_87127_, 80);
                 socket = new Socket();
