--- a/com/mojang/realmsclient/client/Request.java
+++ b/com/mojang/realmsclient/client/Request.java
@@ -108,7 +_,7 @@
             InputStreamReader inputstreamreader = new InputStreamReader(p_87315_, StandardCharsets.UTF_8);
             StringBuilder stringbuilder = new StringBuilder();
 
-            for(int i = inputstreamreader.read(); i != -1; i = inputstreamreader.read()) {
+            for (int i = inputstreamreader.read(); i != -1; i = inputstreamreader.read()) {
                 stringbuilder.append((char)i);
             }
 
@@ -122,7 +_,7 @@
         try {
             InputStream inputstream = this.connection.getInputStream();
 
-            while(inputstream.read(abyte) > 0) {
+            while (inputstream.read(abyte) > 0) {
             }
 
             inputstream.close();
@@ -131,7 +_,7 @@
             try {
                 InputStream inputstream1 = this.connection.getErrorStream();
                 if (inputstream1 != null) {
-                    while(inputstream1.read(abyte) > 0) {
+                    while (inputstream1.read(abyte) > 0) {
                     }
 
                     inputstream1.close();
