--- a/com/mojang/blaze3d/audio/OpenAlUtil.java
+++ b/com/mojang/blaze3d/audio/OpenAlUtil.java
@@ -14,7 +_,7 @@
     private static final Logger LOGGER = LogUtils.getLogger();
 
     private static String alErrorToString(int p_83783_) {
-        switch(p_83783_) {
+        switch (p_83783_) {
             case 40961:
                 return "Invalid name parameter.";
             case 40962:
@@ -41,7 +_,7 @@
     }
 
     private static String alcErrorToString(int p_83792_) {
-        switch(p_83792_) {
+        switch (p_83792_) {
             case 40961:
                 return "Invalid device.";
             case 40962:
