--- a/net/minecraft/world/level/block/state/pattern/BlockPattern.java
+++ b/net/minecraft/world/level/block/state/pattern/BlockPattern.java
@@ -60,9 +_,9 @@
 
     @Nullable
     private BlockPattern.BlockPatternMatch matches(BlockPos p_61198_, Direction p_61199_, Direction p_61200_, LoadingCache<BlockPos, BlockInWorld> p_61201_) {
-        for(int i = 0; i < this.width; ++i) {
-            for(int j = 0; j < this.height; ++j) {
-                for(int k = 0; k < this.depth; ++k) {
+        for (int i = 0; i < this.width; i++) {
+            for (int j = 0; j < this.height; j++) {
+                for (int k = 0; k < this.depth; k++) {
                     if (!this.pattern[k][j][i].test(p_61201_.getUnchecked(translateAndRotate(p_61198_, p_61199_, p_61200_, i, j, k)))) {
                         return null;
                     }
@@ -78,9 +_,9 @@
         LoadingCache<BlockPos, BlockInWorld> loadingcache = createLevelCache(p_61185_, false);
         int i = Math.max(Math.max(this.width, this.height), this.depth);
 
-        for(BlockPos blockpos : BlockPos.betweenClosed(p_61186_, p_61186_.offset(i - 1, i - 1, i - 1))) {
-            for(Direction direction : Direction.values()) {
-                for(Direction direction1 : Direction.values()) {
+        for (BlockPos blockpos : BlockPos.betweenClosed(p_61186_, p_61186_.offset(i - 1, i - 1, i - 1))) {
+            for (Direction direction : Direction.values()) {
+                for (Direction direction1 : Direction.values()) {
                     if (direction1 != direction && direction1 != direction.getOpposite()) {
                         BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch = this.matches(blockpos, direction, direction1, loadingcache);
                         if (blockpattern$blockpatternmatch != null) {
