--- a/net/minecraft/world/level/block/state/pattern/BlockPatternBuilder.java
+++ b/net/minecraft/world/level/block/state/pattern/BlockPatternBuilder.java
@@ -34,14 +_,14 @@
                     "Expected aisle with height of " + this.height + ", but was given one with a height of " + p_61248_.length + ")"
                 );
             } else {
-                for(String s : p_61248_) {
+                for (String s : p_61248_) {
                     if (s.length() != this.width) {
                         throw new IllegalArgumentException(
                             "Not all rows in the given aisle are the correct width (expected " + this.width + ", found one with " + s.length() + ")"
                         );
                     }
 
-                    for(char c0 : s.toCharArray()) {
+                    for (char c0 : s.toCharArray()) {
                         if (!this.lookup.containsKey(c0)) {
                             this.lookup.put(c0, null);
                         }
@@ -71,11 +_,13 @@
 
     private Predicate<BlockInWorld>[][][] createPattern() {
         this.ensureAllCharactersMatched();
-        Predicate<BlockInWorld>[][][] predicate = (Predicate[][][])Array.newInstance(Predicate.class, this.pattern.size(), this.height, this.width);
+        Predicate<BlockInWorld>[][][] predicate = (Predicate<BlockInWorld>[][][])Array.newInstance(
+            Predicate.class, this.pattern.size(), this.height, this.width
+        );
 
-        for(int i = 0; i < this.pattern.size(); ++i) {
-            for(int j = 0; j < this.height; ++j) {
-                for(int k = 0; k < this.width; ++k) {
+        for (int i = 0; i < this.pattern.size(); i++) {
+            for (int j = 0; j < this.height; j++) {
+                for (int k = 0; k < this.width; k++) {
                     predicate[i][j][k] = this.lookup.get(this.pattern.get(i)[j].charAt(k));
                 }
             }
@@ -87,7 +_,7 @@
     private void ensureAllCharactersMatched() {
         List<Character> list = Lists.newArrayList();
 
-        for(Entry<Character, Predicate<BlockInWorld>> entry : this.lookup.entrySet()) {
+        for (Entry<Character, Predicate<BlockInWorld>> entry : this.lookup.entrySet()) {
             if (entry.getValue() == null) {
                 list.add(entry.getKey());
             }
