--- a/net/minecraft/tags/DamageTypeTags.java
+++ b/net/minecraft/tags/DamageTypeTags.java
@@ -35,6 +_,7 @@
     TagKey<DamageType> ALWAYS_KILLS_ARMOR_STANDS = create("always_kills_armor_stands");
     TagKey<DamageType> CAN_BREAK_ARMOR_STAND = create("can_break_armor_stand");
     TagKey<DamageType> BYPASSES_WOLF_ARMOR = create("bypasses_wolf_armor");
+    TagKey<DamageType> IS_PLAYER_ATTACK = create("is_player_attack");
 
     private static TagKey<DamageType> create(String p_270635_) {
         return TagKey.create(Registries.DAMAGE_TYPE, new ResourceLocation(p_270635_));
