--- a/net/minecraft/tags/StructureTags.java
+++ b/net/minecraft/tags/StructureTags.java
@@ -25,8 +_,6 @@
     TagKey<Structure> SHIPWRECK = create("shipwreck");
     TagKey<Structure> RUINED_PORTAL = create("ruined_portal");
     TagKey<Structure> OCEAN_RUIN = create("ocean_ruin");
-    TagKey<Structure> RUINED_PORTATOL = create("ruined_portatol");
-    TagKey<Structure> COLOSSEUM = create("colosseum");
 
     private static TagKey<Structure> create(String p_215896_) {
         return TagKey.create(Registries.STRUCTURE, new ResourceLocation(p_215896_));
