--- a/net/minecraft/tags/TagKey.java
+++ b/net/minecraft/tags/TagKey.java
@@ -25,10 +_,10 @@
     public static <T> Codec<TagKey<T>> hashedCodec(ResourceKey<? extends Registry<T>> p_203887_) {
         return Codec.STRING
             .comapFlatMap(
-                p_274844_ -> p_274844_.startsWith("#")
-                        ? ResourceLocation.read(p_274844_.substring(1)).map(p_203890_ -> create(p_203887_, p_203890_))
+                p_337571_ -> p_337571_.startsWith("#")
+                        ? ResourceLocation.read(p_337571_.substring(1)).map(p_203890_ -> create(p_203887_, p_203890_))
                         : DataResult.error(() -> "Not a tag id"),
-                p_203876_ -> "#" + p_203876_.location
+                p_339468_ -> "#" + p_339468_.location
             );
     }
 
