--- a/net/minecraft/data/advancements/AdvancementProvider.java
+++ b/net/minecraft/data/advancements/AdvancementProvider.java
@@ -37,7 +_,7 @@
                     throw new IllegalStateException("Duplicate advancement " + holder.id());
                 } else {
                     Path path = this.pathProvider.json(holder.id());
-                    list.add(DataProvider.saveStable(cache, lookup, Advancement.CODEC, holder.value(), path));
+                    list.add(DataProvider.saveStable(cache, lookup, Advancement.CODEC, holder.value(), path));// TODO: make conditional
                 }
             };
 
