--- a/net/minecraft/world/level/levelgen/placement/BiomeFilter.java
+++ b/net/minecraft/world/level/levelgen/placement/BiomeFilter.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.placement;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Holder;
 import net.minecraft.util.RandomSource;
@@ -8,7 +_,7 @@
 
 public class BiomeFilter extends PlacementFilter {
     private static final BiomeFilter INSTANCE = new BiomeFilter();
-    public static Codec<BiomeFilter> CODEC = Codec.unit(() -> INSTANCE);
+    public static MapCodec<BiomeFilter> CODEC = MapCodec.unit(() -> INSTANCE);
 
     private BiomeFilter() {
     }
