--- a/net/minecraft/world/level/levelgen/placement/BlockPredicateFilter.java
+++ b/net/minecraft/world/level/levelgen/placement/BlockPredicateFilter.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.placement;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -8,7 +_,7 @@
 import net.minecraft.world.level.levelgen.blockpredicates.BlockPredicate;
 
 public class BlockPredicateFilter extends PlacementFilter {
-    public static final Codec<BlockPredicateFilter> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<BlockPredicateFilter> CODEC = RecordCodecBuilder.mapCodec(
         p_191575_ -> p_191575_.group(BlockPredicate.CODEC.fieldOf("predicate").forGetter(p_191579_ -> p_191579_.predicate))
                 .apply(p_191575_, BlockPredicateFilter::new)
     );
