--- a/net/minecraft/world/level/levelgen/placement/HeightRangePlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/HeightRangePlacement.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.placement;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.stream.Stream;
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.levelgen.heightproviders.UniformHeight;
 
 public class HeightRangePlacement extends PlacementModifier {
-    public static final Codec<HeightRangePlacement> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<HeightRangePlacement> CODEC = RecordCodecBuilder.mapCodec(
         p_191679_ -> p_191679_.group(HeightProvider.CODEC.fieldOf("height").forGetter(p_191686_ -> p_191686_.height))
                 .apply(p_191679_, HeightRangePlacement::new)
     );
