--- a/net/minecraft/world/level/levelgen/placement/NoiseBasedCountPlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/NoiseBasedCountPlacement.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.levelgen.placement;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.BlockPos;
@@ -8,7 +_,7 @@
 import net.minecraft.world.level.biome.Biome;
 
 public class NoiseBasedCountPlacement extends RepeatingPlacement {
-    public static final Codec<NoiseBasedCountPlacement> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<NoiseBasedCountPlacement> CODEC = RecordCodecBuilder.mapCodec(
         p_191736_ -> p_191736_.group(
                     Codec.INT.fieldOf("noise_to_count_ratio").forGetter(p_191746_ -> p_191746_.noiseToCountRatio),
                     Codec.DOUBLE.fieldOf("noise_factor").forGetter(p_191744_ -> p_191744_.noiseFactor),
