--- a/net/minecraft/world/level/levelgen/placement/PlacementModifierType.java
+++ b/net/minecraft/world/level/levelgen/placement/PlacementModifierType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.placement;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 
@@ -19,14 +_,13 @@
     PlacementModifierType<EnvironmentScanPlacement> ENVIRONMENT_SCAN = register("environment_scan", EnvironmentScanPlacement.CODEC);
     PlacementModifierType<HeightmapPlacement> HEIGHTMAP = register("heightmap", HeightmapPlacement.CODEC);
     PlacementModifierType<HeightRangePlacement> HEIGHT_RANGE = register("height_range", HeightRangePlacement.CODEC);
-    PlacementModifierType<CloudPlacement> CLOUD = register("cloud", CloudPlacement.CODEC);
     PlacementModifierType<InSquarePlacement> IN_SQUARE = register("in_square", InSquarePlacement.CODEC);
     PlacementModifierType<RandomOffsetPlacement> RANDOM_OFFSET = register("random_offset", RandomOffsetPlacement.CODEC);
     PlacementModifierType<CarvingMaskPlacement> CARVING_MASK_PLACEMENT = register("carving_mask", CarvingMaskPlacement.CODEC);
 
-    Codec<P> codec();
+    MapCodec<P> codec();
 
-    private static <P extends PlacementModifier> PlacementModifierType<P> register(String p_191867_, Codec<P> p_191868_) {
-        return Registry.register(BuiltInRegistries.PLACEMENT_MODIFIER_TYPE, p_191867_, () -> p_191868_);
+    private static <P extends PlacementModifier> PlacementModifierType<P> register(String p_191867_, MapCodec<P> p_338679_) {
+        return Registry.register(BuiltInRegistries.PLACEMENT_MODIFIER_TYPE, p_191867_, () -> p_338679_);
     }
 }
