--- a/net/minecraft/world/level/levelgen/placement/RarityFilter.java
+++ b/net/minecraft/world/level/levelgen/placement/RarityFilter.java
@@ -1,12 +_,12 @@
 package net.minecraft.world.level.levelgen.placement;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.BlockPos;
 import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.RandomSource;
 
 public class RarityFilter extends PlacementFilter {
-    public static final Codec<RarityFilter> CODEC = ExtraCodecs.POSITIVE_INT.fieldOf("chance").xmap(RarityFilter::new, p_191907_ -> p_191907_.chance).codec();
+    public static final MapCodec<RarityFilter> CODEC = ExtraCodecs.POSITIVE_INT.fieldOf("chance").xmap(RarityFilter::new, p_191907_ -> p_191907_.chance);
     private final int chance;
 
     private RarityFilter(int p_191899_) {
