--- a/net/minecraft/client/gui/components/events/ContainerEventHandler.java
+++ b/net/minecraft/client/gui/components/events/ContainerEventHandler.java
@@ -25,7 +_,7 @@
     List<? extends GuiEventListener> children();
 
     default Optional<GuiEventListener> getChildAt(double p_94730_, double p_94731_) {
-        for(GuiEventListener guieventlistener : this.children()) {
+        for (GuiEventListener guieventlistener : this.children()) {
             if (guieventlistener.isMouseOver(p_94730_, p_94731_)) {
                 return Optional.of(guieventlistener);
             }
@@ -36,7 +_,7 @@
 
     @Override
     default boolean mouseClicked(double p_94695_, double p_94696_, int p_94697_) {
-        for(GuiEventListener guieventlistener : this.children()) {
+        for (GuiEventListener guieventlistener : this.children()) {
             if (guieventlistener.mouseClicked(p_94695_, p_94696_, p_94697_)) {
                 this.setFocused(guieventlistener);
                 if (p_94697_ == 0) {
@@ -148,7 +_,7 @@
         BooleanSupplier booleansupplier = flag ? listiterator::hasNext : listiterator::hasPrevious;
         Supplier<? extends GuiEventListener> supplier = flag ? listiterator::next : listiterator::previous;
 
-        while(booleansupplier.getAsBoolean()) {
+        while (booleansupplier.getAsBoolean()) {
             GuiEventListener guieventlistener1 = supplier.get();
             ComponentPath componentpath = guieventlistener1.nextFocusPath(p_265354_);
             if (componentpath != null) {
@@ -182,7 +_,7 @@
         int i = p_265054_.getBoundInDirection(p_265167_.getOpposite());
         List<GuiEventListener> list = new ArrayList<>();
 
-        for(GuiEventListener guieventlistener : this.children()) {
+        for (GuiEventListener guieventlistener : this.children()) {
             if (guieventlistener != p_265476_) {
                 ScreenRectangle screenrectangle = guieventlistener.getRectangle();
                 if (screenrectangle.overlapsInAxis(p_265054_, screenaxis1)) {
@@ -204,7 +_,7 @@
         );
         list.sort(comparator.thenComparing(comparator1));
 
-        for(GuiEventListener guieventlistener1 : list) {
+        for (GuiEventListener guieventlistener1 : list) {
             ComponentPath componentpath = guieventlistener1.nextFocusPath(p_265762_);
             if (componentpath != null) {
                 return componentpath;
@@ -223,7 +_,7 @@
         List<Pair<GuiEventListener, Long>> list = new ArrayList<>();
         ScreenPosition screenposition = ScreenPosition.of(screenaxis, p_265390_.getBoundInDirection(p_265687_), p_265390_.getCenterInAxis(screenaxis1));
 
-        for(GuiEventListener guieventlistener : this.children()) {
+        for (GuiEventListener guieventlistener : this.children()) {
             if (guieventlistener != p_265498_) {
                 ScreenRectangle screenrectangle = guieventlistener.getRectangle();
                 ScreenPosition screenposition1 = ScreenPosition.of(
@@ -238,7 +_,7 @@
 
         list.sort(Comparator.comparingDouble(Pair::getSecond));
 
-        for(Pair<GuiEventListener, Long> pair : list) {
+        for (Pair<GuiEventListener, Long> pair : list) {
             ComponentPath componentpath = pair.getFirst().nextFocusPath(p_265048_);
             if (componentpath != null) {
                 return componentpath;
