--- a/net/minecraft/client/color/item/ItemTintSources.java
+++ b/net/minecraft/client/color/item/ItemTintSources.java
@@ -22,5 +_,6 @@
         ID_MAPPER.put(ResourceLocation.withDefaultNamespace("map_color"), MapColor.MAP_CODEC);
         ID_MAPPER.put(ResourceLocation.withDefaultNamespace("team"), TeamColor.MAP_CODEC);
         ID_MAPPER.put(ResourceLocation.withDefaultNamespace("grade"), GradeColor.MAP_CODEC);
+        net.neoforged.fml.ModLoader.postEvent(new net.neoforged.neoforge.client.event.RegisterColorHandlersEvent.ItemTintSources(ID_MAPPER));
     }
 }
