--- a/net/minecraft/world/entity/animal/horse/AbstractChestedHorse.java
+++ b/net/minecraft/world/entity/animal/horse/AbstractChestedHorse.java
@@ -88,7 +_,7 @@
         if (this.hasChest()) {
             ListTag listtag = new ListTag();
 
-            for(int i = 1; i < this.inventory.getContainerSize(); ++i) {
+            for (int i = 1; i < this.inventory.getContainerSize(); i++) {
                 ItemStack itemstack = this.inventory.getItem(i);
                 if (!itemstack.isEmpty()) {
                     CompoundTag compoundtag = new CompoundTag();
@@ -109,7 +_,7 @@
         if (this.hasChest()) {
             ListTag listtag = p_30488_.getList("Items", 10);
 
-            for(int i = 0; i < listtag.size(); ++i) {
+            for (int i = 0; i < listtag.size(); i++) {
                 CompoundTag compoundtag = listtag.getCompound(i);
                 int j = compoundtag.getByte("Slot") & 255;
                 if (j < this.inventory.getContainerSize() - 1) {
