--- a/net/minecraft/world/entity/animal/horse/Horse.java
+++ b/net/minecraft/world/entity/animal/horse/Horse.java
@@ -28,7 +_,6 @@
 import net.minecraft.world.entity.animal.Animal;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.AnimalArmorItem;
-import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.ServerLevelAccessor;
@@ -164,10 +_,8 @@
     public boolean canMate(Animal p_30698_) {
         if (p_30698_ == this) {
             return false;
-        } else if (!(p_30698_ instanceof Donkey) && !(p_30698_ instanceof Horse)) {
-            return false;
         } else {
-            return this.canParent() && ((AbstractHorse)p_30698_).canParent();
+            return !(p_30698_ instanceof Donkey) && !(p_30698_ instanceof Horse) ? false : this.canParent() && ((AbstractHorse)p_30698_).canParent();
         }
     }
 
@@ -220,8 +_,7 @@
 
     @Override
     public boolean isBodyArmorItem(ItemStack p_324434_) {
-        Item item = p_324434_.getItem();
-        if (item instanceof AnimalArmorItem animalarmoritem && animalarmoritem.getBodyType() == AnimalArmorItem.BodyType.EQUESTRIAN) {
+        if (p_324434_.getItem() instanceof AnimalArmorItem animalarmoritem && animalarmoritem.getBodyType() == AnimalArmorItem.BodyType.EQUESTRIAN) {
             return true;
         }
 
