--- a/net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
+++ b/net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
@@ -10,6 +_,7 @@
 import net.minecraft.world.entity.MobSpawnType;
 import net.minecraft.world.entity.ai.goal.Goal;
 import net.minecraft.world.entity.monster.Skeleton;
+import net.minecraft.world.flag.FeatureFlagSet;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.item.enchantment.EnchantmentHelper;
@@ -30,6 +_,10 @@
     @Override
     public void tick() {
         ServerLevel serverlevel = (ServerLevel)this.horse.level();
+        // Forge: Trigger the trap in a tick task to avoid crashes when mods add goals to skeleton horses
+        // (MC-206338/Forge PR #7509)
+        serverlevel.getServer().tell(new net.minecraft.server.TickTask(serverlevel.getServer().getTickCount(), () -> {
+        if (!this.horse.isAlive()) return;
         DifficultyInstance difficultyinstance = serverlevel.getCurrentDifficultyAt(this.horse.blockPosition());
         this.horse.setTrap(false);
         this.horse.setTamed(true);
@@ -44,7 +_,7 @@
                 skeleton.startRiding(this.horse);
                 serverlevel.addFreshEntityWithPassengers(skeleton);
 
-                for(int i = 0; i < 3; ++i) {
+                for (int i = 0; i < 3; i++) {
                     AbstractHorse abstracthorse = this.createHorse(difficultyinstance);
                     if (abstracthorse != null) {
                         Skeleton skeleton1 = this.createSkeleton(difficultyinstance, abstracthorse);
@@ -57,6 +_,7 @@
                 }
             }
         }
+        }));
     }
 
     @Nullable
@@ -86,9 +_,11 @@
                 skeleton.setItemSlot(EquipmentSlot.HEAD, new ItemStack(Items.IRON_HELMET));
             }
 
+            FeatureFlagSet featureflagset = p_30933_.level().enabledFeatures();
             skeleton.setItemSlot(
                 EquipmentSlot.MAINHAND,
                 EnchantmentHelper.enchantItem(
+                    featureflagset,
                     skeleton.getRandom(),
                     this.disenchant(skeleton.getMainHandItem()),
                     (int)(5.0F + p_30932_.getSpecialMultiplier() * (float)skeleton.getRandom().nextInt(18)),
@@ -98,6 +_,7 @@
             skeleton.setItemSlot(
                 EquipmentSlot.HEAD,
                 EnchantmentHelper.enchantItem(
+                    featureflagset,
                     skeleton.getRandom(),
                     this.disenchant(skeleton.getItemBySlot(EquipmentSlot.HEAD)),
                     (int)(5.0F + p_30932_.getSpecialMultiplier() * (float)skeleton.getRandom().nextInt(18)),
