--- a/net/minecraft/world/entity/animal/horse/TraderLlama.java
+++ b/net/minecraft/world/entity/animal/horse/TraderLlama.java
@@ -129,16 +_,12 @@
         public boolean canUse() {
             if (!this.llama.isLeashed()) {
                 return false;
+            } else if (!(this.llama.getLeashHolder() instanceof WanderingTrader wanderingtrader)) {
+                return false;
             } else {
-                Entity entity = this.llama.getLeashHolder();
-                if (!(entity instanceof WanderingTrader)) {
-                    return false;
-                } else {
-                    WanderingTrader wanderingtrader = (WanderingTrader)entity;
-                    this.ownerLastHurtBy = wanderingtrader.getLastHurtByMob();
-                    int i = wanderingtrader.getLastHurtByMobTimestamp();
-                    return i != this.timestamp && this.canAttack(this.ownerLastHurtBy, TargetingConditions.DEFAULT);
-                }
+                this.ownerLastHurtBy = wanderingtrader.getLastHurtByMob();
+                int i = wanderingtrader.getLastHurtByMobTimestamp();
+                return i != this.timestamp && this.canAttack(this.ownerLastHurtBy, TargetingConditions.DEFAULT);
             }
         }
 
