--- a/net/minecraft/world/entity/animal/horse/ZombieHorse.java
+++ b/net/minecraft/world/entity/animal/horse/ZombieHorse.java
@@ -40,11 +_,9 @@
     public static boolean checkZombieHorseSpawnRules(
         EntityType<? extends Animal> p_312450_, LevelAccessor p_312340_, MobSpawnType p_312031_, BlockPos p_311838_, RandomSource p_312514_
     ) {
-        if (!MobSpawnType.isSpawner(p_312031_)) {
-            return Animal.checkAnimalSpawnRules(p_312450_, p_312340_, p_312031_, p_311838_, p_312514_);
-        } else {
-            return MobSpawnType.ignoresLightRequirements(p_312031_) || isBrightEnoughToSpawn(p_312340_, p_311838_);
-        }
+        return !MobSpawnType.isSpawner(p_312031_)
+            ? Animal.checkAnimalSpawnRules(p_312450_, p_312340_, p_312031_, p_311838_, p_312514_)
+            : MobSpawnType.ignoresLightRequirements(p_312031_) || isBrightEnoughToSpawn(p_312340_, p_311838_);
     }
 
     @Override
