--- a/net/minecraft/util/eventlog/EventLogDirectory.java
+++ b/net/minecraft/util/eventlog/EventLogDirectory.java
@@ -103,7 +_,7 @@
             byte[] abyte = new byte[4096];
             ByteBuffer bytebuffer = ByteBuffer.wrap(abyte);
 
-            while(p_262066_.read(bytebuffer) >= 0) {
+            while (p_262066_.read(bytebuffer) >= 0) {
                 bytebuffer.flip();
                 outputstream.write(abyte, 0, bytebuffer.limit());
                 bytebuffer.clear();
@@ -118,7 +_,7 @@
         EventLogDirectory.FileId eventlogdirectory$fileid;
         do {
             eventlogdirectory$fileid = new EventLogDirectory.FileId(p_261865_, i++);
-        } while(set.contains(eventlogdirectory$fileid));
+        } while (set.contains(eventlogdirectory$fileid));
 
         EventLogDirectory.RawFile eventlogdirectory$rawfile = new EventLogDirectory.RawFile(
             this.root.resolve(eventlogdirectory$fileid.toFileName(this.extension)), eventlogdirectory$fileid
@@ -209,7 +_,7 @@
         public EventLogDirectory.FileList compressAll() {
             ListIterator<EventLogDirectory.File> listiterator = this.files.listIterator();
 
-            while(listiterator.hasNext()) {
+            while (listiterator.hasNext()) {
                 EventLogDirectory.File eventlogdirectory$file = listiterator.next();
 
                 try {
