--- a/net/minecraft/util/eventlog/JsonEventLog.java
+++ b/net/minecraft/util/eventlog/JsonEventLog.java
@@ -15,7 +_,6 @@
 import java.nio.file.StandardOpenOption;
 import java.util.concurrent.atomic.AtomicInteger;
 import javax.annotation.Nullable;
-import net.minecraft.Util;
 
 public class JsonEventLog<T> implements Closeable {
     private static final Gson GSON = new Gson();
@@ -34,7 +_,7 @@
     }
 
     public void write(T p_261929_) throws IOException, JsonIOException {
-        JsonElement jsonelement = Util.getOrThrow(this.codec.encodeStart(JsonOps.INSTANCE, p_261929_), IOException::new);
+        JsonElement jsonelement = this.codec.encodeStart(JsonOps.INSTANCE, p_261929_).getOrThrow(IOException::new);
         this.channel.position(this.channel.size());
         Writer writer = Channels.newWriter(this.channel, StandardCharsets.UTF_8);
         GSON.toJson(jsonelement, writer);
