--- a/net/minecraft/util/eventlog/JsonEventLogReader.java
+++ b/net/minecraft/util/eventlog/JsonEventLogReader.java
@@ -11,7 +_,6 @@
 import java.io.IOException;
 import java.io.Reader;
 import javax.annotation.Nullable;
-import net.minecraft.Util;
 
 public interface JsonEventLogReader<T> extends Closeable {
     static <T> JsonEventLogReader<T> create(final Codec<T> p_261600_, Reader p_261836_) {
@@ -26,7 +_,7 @@
                         return null;
                     } else {
                         JsonElement jsonelement = JsonParser.parseReader(jsonreader);
-                        return Util.getOrThrow(p_261600_.parse(JsonOps.INSTANCE, jsonelement), IOException::new);
+                        return p_261600_.parse(JsonOps.INSTANCE, jsonelement).getOrThrow(IOException::new);
                     }
                 } catch (JsonParseException jsonparseexception) {
                     throw new IOException(jsonparseexception);
