--- a/net/minecraft/network/protocol/game/ClientboundBossEventPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundBossEventPacket.java
@@ -9,7 +_,6 @@
 import net.minecraft.network.protocol.Packet;
 import net.minecraft.network.protocol.PacketType;
 import net.minecraft.world.BossEvent;
-import net.minecraft.world.phys.Vec3;
 
 public class ClientboundBossEventPacket implements Packet<ClientGamePacketListener> {
     public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundBossEventPacket> STREAM_CODEC = Packet.codec(
@@ -63,12 +_,6 @@
         return new ClientboundBossEventPacket(p_178652_.getId(), new ClientboundBossEventPacket.UpdateNameOperation(p_178652_.getName()));
     }
 
-    public static ClientboundBossEventPacket createUpdatePositionPacket(BossEvent p_341561_) {
-        return new ClientboundBossEventPacket(
-            p_341561_.getId(), new ClientboundBossEventPacket.UpdateCenterAndRadiusOperation(p_341561_.getCenter(), p_341561_.getRadius())
-        );
-    }
-
     public static ClientboundBossEventPacket createUpdateStylePacket(BossEvent p_178654_) {
         return new ClientboundBossEventPacket(
             p_178654_.getId(), new ClientboundBossEventPacket.UpdateStyleOperation(p_178654_.getColor(), p_178654_.getOverlay())
@@ -128,8 +_,6 @@
         private final boolean darkenScreen;
         private final boolean playMusic;
         private final boolean createWorldFog;
-        private final Vec3 center;
-        private final int radius;
 
         AddOperation(BossEvent p_178672_) {
             this.name = p_178672_.getName();
@@ -139,8 +_,6 @@
             this.darkenScreen = p_178672_.shouldDarkenScreen();
             this.playMusic = p_178672_.shouldPlayBossMusic();
             this.createWorldFog = p_178672_.shouldCreateWorldFog();
-            this.center = p_178672_.getCenter();
-            this.radius = p_178672_.getRadius();
         }
 
         private AddOperation(RegistryFriendlyByteBuf p_324153_) {
@@ -152,8 +_,6 @@
             this.darkenScreen = (i & 1) > 0;
             this.playMusic = (i & 2) > 0;
             this.createWorldFog = (i & 4) > 0;
-            this.center = p_324153_.readVec3();
-            this.radius = p_324153_.readVarInt();
         }
 
         @Override
@@ -163,9 +_,7 @@
 
         @Override
         public void dispatch(UUID p_178677_, ClientboundBossEventPacket.Handler p_178678_) {
-            p_178678_.add(
-                p_178677_, this.name, this.progress, this.color, this.overlay, this.darkenScreen, this.playMusic, this.createWorldFog, this.center, this.radius
-            );
+            p_178678_.add(p_178677_, this.name, this.progress, this.color, this.overlay, this.darkenScreen, this.playMusic, this.createWorldFog);
         }
 
         @Override
@@ -175,23 +_,19 @@
             p_323847_.writeEnum(this.color);
             p_323847_.writeEnum(this.overlay);
             p_323847_.writeByte(ClientboundBossEventPacket.encodeProperties(this.darkenScreen, this.playMusic, this.createWorldFog));
-            p_323847_.writeVec3(this.center);
-            p_323847_.writeVarInt(this.radius);
         }
     }
 
     public interface Handler {
         default void add(
-            UUID p_342544_,
-            Component p_341904_,
-            float p_340745_,
-            BossEvent.BossBarColor p_340897_,
-            BossEvent.BossBarOverlay p_341403_,
-            boolean p_340392_,
-            boolean p_341256_,
-            boolean p_341443_,
-            Vec3 p_341147_,
-            int p_341006_
+            UUID p_178689_,
+            Component p_178690_,
+            float p_178691_,
+            BossEvent.BossBarColor p_178692_,
+            BossEvent.BossBarOverlay p_178693_,
+            boolean p_178694_,
+            boolean p_178695_,
+            boolean p_178696_
         ) {
         }
 
@@ -204,9 +_,6 @@
         default void updateName(UUID p_178687_, Component p_178688_) {
         }
 
-        default void updateLocation(UUID p_178689_, Vec3 p_341960_, int p_340547_) {
-        }
-
         default void updateStyle(UUID p_178684_, BossEvent.BossBarColor p_178685_, BossEvent.BossBarOverlay p_178686_) {
         }
 
@@ -228,35 +_,12 @@
         UPDATE_PROGRESS(ClientboundBossEventPacket.UpdateProgressOperation::new),
         UPDATE_NAME(ClientboundBossEventPacket.UpdateNameOperation::new),
         UPDATE_STYLE(ClientboundBossEventPacket.UpdateStyleOperation::new),
-        UPDATE_PROPERTIES(ClientboundBossEventPacket.UpdatePropertiesOperation::new),
-        UPDATE_CENTER_AND_RADIUS(ClientboundBossEventPacket.UpdateCenterAndRadiusOperation::new);
+        UPDATE_PROPERTIES(ClientboundBossEventPacket.UpdatePropertiesOperation::new);
 
         final StreamDecoder<RegistryFriendlyByteBuf, ClientboundBossEventPacket.Operation> reader;
 
         private OperationType(StreamDecoder<RegistryFriendlyByteBuf, ClientboundBossEventPacket.Operation> p_324602_) {
             this.reader = p_324602_;
-        }
-    }
-
-    static record UpdateCenterAndRadiusOperation(Vec3 position, int radius) implements ClientboundBossEventPacket.Operation {
-        private UpdateCenterAndRadiusOperation(RegistryFriendlyByteBuf p_342384_) {
-            this(p_342384_.readVec3(), p_342384_.readVarInt());
-        }
-
-        @Override
-        public ClientboundBossEventPacket.OperationType getType() {
-            return ClientboundBossEventPacket.OperationType.UPDATE_CENTER_AND_RADIUS;
-        }
-
-        @Override
-        public void dispatch(UUID p_341847_, ClientboundBossEventPacket.Handler p_340463_) {
-            p_340463_.updateLocation(p_341847_, this.position, this.radius);
-        }
-
-        @Override
-        public void write(RegistryFriendlyByteBuf p_341463_) {
-            p_341463_.writeVec3(this.position);
-            p_341463_.writeVarInt(this.radius);
         }
     }
 
