--- a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
@@ -34,14 +_,14 @@
             EquipmentSlot equipmentslot = aequipmentslot[i & 127];
             ItemStack itemstack = ItemStack.OPTIONAL_STREAM_CODEC.decode(p_320957_);
             this.slots.add(Pair.of(equipmentslot, itemstack));
-        } while((i & -128) != 0);
+        } while ((i & -128) != 0);
     }
 
     private void write(RegistryFriendlyByteBuf p_320609_) {
         p_320609_.writeVarInt(this.entity);
         int i = this.slots.size();
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             Pair<EquipmentSlot, ItemStack> pair = this.slots.get(j);
             EquipmentSlot equipmentslot = pair.getFirst();
             boolean flag = j != i - 1;
