--- a/net/minecraft/network/protocol/game/ClientboundGameEventPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundGameEventPacket.java
@@ -25,7 +_,6 @@
     public static final ClientboundGameEventPacket.Type IMMEDIATE_RESPAWN = new ClientboundGameEventPacket.Type(11);
     public static final ClientboundGameEventPacket.Type LIMITED_CRAFTING = new ClientboundGameEventPacket.Type(12);
     public static final ClientboundGameEventPacket.Type LEVEL_CHUNKS_LOAD_START = new ClientboundGameEventPacket.Type(13);
-    public static final ClientboundGameEventPacket.Type POTATO_POEM = new ClientboundGameEventPacket.Type(15);
     public static final int DEMO_PARAM_INTRO = 0;
     public static final int DEMO_PARAM_HINT_1 = 101;
     public static final int DEMO_PARAM_HINT_2 = 102;
