--- a/net/minecraft/network/protocol/game/ClientGamePacketListener.java
+++ b/net/minecraft/network/protocol/game/ClientGamePacketListener.java
@@ -24,8 +_,6 @@
 
     void handleAddOrRemoveRecipes(ClientboundRecipePacket p_131417_);
 
-    void handleAddSubGrid(ClientboundAddSubGridPacket p_341457_);
-
     void handleBlockDestruction(ClientboundBlockDestructionPacket p_131375_);
 
     void handleOpenSignEditor(ClientboundOpenSignEditorPacket p_131410_);
@@ -124,8 +_,6 @@
 
     void handleSoundEvent(ClientboundSoundPacket p_131445_);
 
-    void handleSoundSequenceEvent(ClientboundSoundSequencePacket p_342563_);
-
     void handleSoundEntityEvent(ClientboundSoundEntityPacket p_131444_);
 
     void handleTakeItemEntity(ClientboundTakeItemEntityPacket p_131449_);
@@ -229,4 +_,6 @@
     void handleChunkBatchFinished(ClientboundChunkBatchFinishedPacket p_295302_);
 
     void handleDebugSample(ClientboundDebugSamplePacket p_323534_);
+
+    void handleProjectilePowerPacket(ClientboundProjectilePowerPacket p_339666_);
 }
