--- a/net/minecraft/network/protocol/game/ClientboundCommandsPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundCommandsPacket.java
@@ -68,7 +_,7 @@
     private static void validateEntries(List<ClientboundCommandsPacket.Entry> p_237631_, BiPredicate<ClientboundCommandsPacket.Entry, IntSet> p_237632_) {
         IntSet intset = new IntOpenHashSet(IntSets.fromTo(0, p_237631_.size()));
 
-        while(!intset.isEmpty()) {
+        while (!intset.isEmpty()) {
             boolean flag = intset.removeIf(p_237637_ -> p_237632_.test(p_237631_.get(p_237637_), intset));
             if (!flag) {
                 throw new IllegalStateException("Server sent an impossible command tree");
@@ -87,7 +_,7 @@
         queue.add(p_131863_);
 
         CommandNode<SharedSuggestionProvider> commandnode;
-        while((commandnode = queue.poll()) != null) {
+        while ((commandnode = queue.poll()) != null) {
             if (!object2intmap.containsKey(commandnode)) {
                 int i = object2intmap.size();
                 object2intmap.put(commandnode, i);
@@ -105,7 +_,7 @@
         ObjectArrayList<ClientboundCommandsPacket.Entry> objectarraylist = new ObjectArrayList<>(p_237627_.size());
         objectarraylist.size(p_237627_.size());
 
-        for(Object2IntMap.Entry<CommandNode<SharedSuggestionProvider>> entry : Object2IntMaps.fastIterable(p_237627_)) {
+        for (Object2IntMap.Entry<CommandNode<SharedSuggestionProvider>> entry : Object2IntMaps.fastIterable(p_237627_)) {
             objectarraylist.set(entry.getIntValue(), createEntry(entry.getKey(), p_237627_));
         }
 
@@ -162,18 +_,17 @@
         if (p_237622_ instanceof RootCommandNode) {
             i |= 0;
             clientboundcommandspacket$nodestub = null;
-        } else if (p_237622_ instanceof ArgumentCommandNode argumentcommandnode) {
+        } else if (p_237622_ instanceof ArgumentCommandNode<SharedSuggestionProvider, ?> argumentcommandnode) {
             clientboundcommandspacket$nodestub = new ClientboundCommandsPacket.ArgumentNodeStub(argumentcommandnode);
             i |= 2;
             if (argumentcommandnode.getCustomSuggestions() != null) {
                 i |= 16;
             }
         } else {
-            if (!(p_237622_ instanceof LiteralCommandNode)) {
+            if (!(p_237622_ instanceof LiteralCommandNode literalcommandnode)) {
                 throw new UnsupportedOperationException("Unknown node type " + p_237622_);
             }
 
-            LiteralCommandNode literalcommandnode = (LiteralCommandNode)p_237622_;
             clientboundcommandspacket$nodestub = new ClientboundCommandsPacket.LiteralNodeStub(literalcommandnode.getLiteral());
             i |= 1;
         }
@@ -275,15 +_,11 @@
         }
 
         public boolean canBuild(IntSet p_237673_) {
-            if ((this.flags & 8) != 0) {
-                return !p_237673_.contains(this.redirect);
-            } else {
-                return true;
-            }
+            return (this.flags & 8) != 0 ? !p_237673_.contains(this.redirect) : true;
         }
 
         public boolean canResolve(IntSet p_237677_) {
-            for(int i : this.children) {
+            for (int i : this.children) {
                 if (p_237677_.contains(i)) {
                     return false;
                 }
@@ -348,7 +_,7 @@
 
                 this.nodes.set(p_237692_, commandnode1);
 
-                for(int i : clientboundcommandspacket$entry.children) {
+                for (int i : clientboundcommandspacket$entry.children) {
                     CommandNode<SharedSuggestionProvider> commandnode2 = this.resolve(i);
                     if (!(commandnode2 instanceof RootCommandNode)) {
                         commandnode1.addChild(commandnode2);
