--- a/net/minecraft/network/protocol/game/ServerboundSignUpdatePacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundSignUpdatePacket.java
@@ -26,7 +_,7 @@
         this.isFrontText = p_179790_.readBoolean();
         this.lines = new String[4];
 
-        for(int i = 0; i < 4; ++i) {
+        for (int i = 0; i < 4; i++) {
             this.lines[i] = p_179790_.readUtf(384);
         }
     }
@@ -35,7 +_,7 @@
         p_134662_.writeBlockPos(this.pos);
         p_134662_.writeBoolean(this.isFrontText);
 
-        for(int i = 0; i < 4; ++i) {
+        for (int i = 0; i < 4; i++) {
             p_134662_.writeUtf(this.lines[i]);
         }
     }
