--- a/net/minecraft/commands/ParserUtils.java
+++ b/net/minecraft/commands/ParserUtils.java
@@ -47,9 +_,7 @@
         Object object;
         try {
             JsonElement jsonelement = Streams.parse(jsonreader);
-            object = Util.<T, JsonParseException>getOrThrow(
-                p_313876_.parse(p_323581_.createSerializationContext(JsonOps.INSTANCE), jsonelement), JsonParseException::new
-            );
+            object = p_313876_.parse(p_323581_.createSerializationContext(JsonOps.INSTANCE), jsonelement).getOrThrow(JsonParseException::new);
         } catch (StackOverflowError stackoverflowerror) {
             throw new JsonParseException(stackoverflowerror);
         } finally {
@@ -62,7 +_,7 @@
     public static String readWhile(StringReader p_332772_, CharPredicate p_332664_) {
         int i = p_332772_.getCursor();
 
-        while(p_332772_.canRead() && p_332664_.test(p_332772_.peek())) {
+        while (p_332772_.canRead() && p_332664_.test(p_332772_.peek())) {
             p_332772_.skip();
         }
 
