--- a/net/minecraft/client/model/geom/ModelLayerLocation.java
+++ b/net/minecraft/client/model/geom/ModelLayerLocation.java
@@ -26,11 +_,10 @@
     public boolean equals(Object p_171126_) {
         if (this == p_171126_) {
             return true;
-        } else if (!(p_171126_ instanceof ModelLayerLocation)) {
-            return false;
         } else {
-            ModelLayerLocation modellayerlocation = (ModelLayerLocation)p_171126_;
-            return this.model.equals(modellayerlocation.model) && this.layer.equals(modellayerlocation.layer);
+            return !(p_171126_ instanceof ModelLayerLocation modellayerlocation)
+                ? false
+                : this.model.equals(modellayerlocation.model) && this.layer.equals(modellayerlocation.layer);
         }
     }
 
