--- a/net/minecraft/client/model/geom/ModelLayers.java
+++ b/net/minecraft/client/model/geom/ModelLayers.java
@@ -20,7 +_,6 @@
     public static final ModelLayerLocation ARMOR_STAND_OUTER_ARMOR = registerOuterArmor("armor_stand");
     public static final ModelLayerLocation AXOLOTL = register("axolotl");
     public static final ModelLayerLocation BANNER = register("banner");
-    public static final ModelLayerLocation BATATO = register("batato");
     public static final ModelLayerLocation BAT = register("bat");
     public static final ModelLayerLocation BED_FOOT = register("bed_foot");
     public static final ModelLayerLocation BED_HEAD = register("bed_head");
@@ -62,9 +_,7 @@
     public static final ModelLayerLocation DROWNED_OUTER_ARMOR = registerOuterArmor("drowned");
     public static final ModelLayerLocation DROWNED_OUTER_LAYER = register("drowned", "outer");
     public static final ModelLayerLocation ELDER_GUARDIAN = register("elder_guardian");
-    public static final ModelLayerLocation ELDER_GUARDIAN_SLAB = register("elder_guardian_slab");
     public static final ModelLayerLocation ELYTRA = register("elytra");
-    public static final ModelLayerLocation POISONOUS_POLYTRA = register("poisonous_polytra");
     public static final ModelLayerLocation ENDERMAN = register("enderman");
     public static final ModelLayerLocation ENDERMITE = register("endermite");
     public static final ModelLayerLocation ENDER_DRAGON = register("ender_dragon");
@@ -81,7 +_,6 @@
     public static final ModelLayerLocation GLOW_SQUID = register("glow_squid");
     public static final ModelLayerLocation GOAT = register("goat");
     public static final ModelLayerLocation GUARDIAN = register("guardian");
-    public static final ModelLayerLocation GUARDIAN_SLAB = register("guardian_slab");
     public static final ModelLayerLocation HOGLIN = register("hoglin");
     public static final ModelLayerLocation HOPPER_MINECART = register("hopper_minecart");
     public static final ModelLayerLocation HORSE = register("horse");
@@ -141,8 +_,6 @@
     public static final ModelLayerLocation SKELETON_SKULL = register("skeleton_skull");
     public static final ModelLayerLocation SLIME = register("slime");
     public static final ModelLayerLocation SLIME_OUTER = register("slime", "outer");
-    public static final ModelLayerLocation MEGA_SPUD = register("mega_spud");
-    public static final ModelLayerLocation MEGA_SPUD_OUTER = register("mega_spud", "outer");
     public static final ModelLayerLocation SNIFFER = register("sniffer");
     public static final ModelLayerLocation SNOW_GOLEM = register("snow_golem");
     public static final ModelLayerLocation SPAWNER_MINECART = register("spawner_minecart");
@@ -234,11 +_,13 @@
     }
 
     public static ModelLayerLocation createSignModelName(WoodType p_171292_) {
-        return createLocation("sign/" + p_171292_.name(), "main");
+        ResourceLocation location = new ResourceLocation(p_171292_.name());
+        return new ModelLayerLocation(location.withPrefix("sign/"), "main");
     }
 
     public static ModelLayerLocation createHangingSignModelName(WoodType p_252225_) {
-        return createLocation("hanging_sign/" + p_252225_.name(), "main");
+        ResourceLocation location = new ResourceLocation(p_252225_.name());
+        return new ModelLayerLocation(location.withPrefix("hanging_sign/"), "main");
     }
 
     public static Stream<ModelLayerLocation> getKnownLocations() {
