--- a/net/minecraft/client/model/geom/ModelPart.java
+++ b/net/minecraft/client/model/geom/ModelPart.java
@@ -118,7 +_,7 @@
                     this.compile(p_104307_.last(), p_104308_, p_104309_, p_104310_, p_104311_, p_104312_, p_104313_, p_104314_);
                 }
 
-                for(ModelPart modelpart : this.children.values()) {
+                for (ModelPart modelpart : this.children.values()) {
                     modelpart.render(p_104307_, p_104308_, p_104309_, p_104310_, p_104311_, p_104312_, p_104313_, p_104314_);
                 }
 
@@ -137,7 +_,7 @@
             this.translateAndRotate(p_171313_);
             PoseStack.Pose posestack$pose = p_171313_.last();
 
-            for(int i = 0; i < this.cubes.size(); ++i) {
+            for (int i = 0; i < this.cubes.size(); i++) {
                 p_171314_.visit(posestack$pose, p_171315_, i, this.cubes.get(i));
             }
 
@@ -161,7 +_,7 @@
     private void compile(
         PoseStack.Pose p_104291_, VertexConsumer p_104292_, int p_104293_, int p_104294_, float p_104295_, float p_104296_, float p_104297_, float p_104298_
     ) {
-        for(ModelPart.Cube modelpart$cube : this.cubes) {
+        for (ModelPart.Cube modelpart$cube : this.cubes) {
             modelpart$cube.compile(p_104291_, p_104292_, p_104293_, p_104294_, p_104295_, p_104296_, p_104297_, p_104298_);
         }
     }
@@ -175,21 +_,21 @@
     }
 
     public void offsetPos(Vector3f p_253873_) {
-        this.x += p_253873_.x();
-        this.y += p_253873_.y();
-        this.z += p_253873_.z();
+        this.x = this.x + p_253873_.x();
+        this.y = this.y + p_253873_.y();
+        this.z = this.z + p_253873_.z();
     }
 
     public void offsetRotation(Vector3f p_253983_) {
-        this.xRot += p_253983_.x();
-        this.yRot += p_253983_.y();
-        this.zRot += p_253983_.z();
+        this.xRot = this.xRot + p_253983_.x();
+        this.yRot = this.yRot + p_253983_.y();
+        this.zRot = this.zRot + p_253983_.z();
     }
 
     public void offsetScale(Vector3f p_253957_) {
-        this.xScale += p_253957_.x();
-        this.yScale += p_253957_.y();
-        this.zScale += p_253957_.z();
+        this.xScale = this.xScale + p_253957_.x();
+        this.yScale = this.yScale + p_253957_.y();
+        this.zScale = this.zScale + p_253957_.z();
     }
 
     public Stream<ModelPart> getAllParts() {
@@ -361,13 +_,13 @@
             Matrix4f matrix4f = p_171333_.pose();
             Vector3f vector3f = new Vector3f();
 
-            for(ModelPart.Polygon modelpart$polygon : this.polygons) {
+            for (ModelPart.Polygon modelpart$polygon : this.polygons) {
                 Vector3f vector3f1 = p_171333_.transformNormal(modelpart$polygon.normal, vector3f);
                 float f = vector3f1.x();
                 float f1 = vector3f1.y();
                 float f2 = vector3f1.z();
 
-                for(ModelPart.Vertex modelpart$vertex : modelpart$polygon.vertices) {
+                for (ModelPart.Vertex modelpart$vertex : modelpart$polygon.vertices) {
                     float f3 = modelpart$vertex.pos.x() / 16.0F;
                     float f4 = modelpart$vertex.pos.y() / 16.0F;
                     float f5 = modelpart$vertex.pos.z() / 16.0F;
@@ -419,7 +_,7 @@
             if (p_104369_) {
                 int i = p_104362_.length;
 
-                for(int j = 0; j < i / 2; ++j) {
+                for (int j = 0; j < i / 2; j++) {
                     ModelPart.Vertex modelpart$vertex = p_104362_[j];
                     p_104362_[j] = p_104362_[i - 1 - j];
                     p_104362_[i - 1 - j] = modelpart$vertex;
