--- a/net/minecraft/client/Minecraft.java
+++ b/net/minecraft/client/Minecraft.java
@@ -1699,8 +1750,10 @@
                             }
 
                             this.player.resetAttackStrengthTicker();
+                            net.neoforged.neoforge.common.CommonHooks.onEmptyLeftClick(this.player);
                     }
 
+                    if (inputEvent.shouldSwingHand())
                     this.player.swing(InteractionHand.MAIN_HAND);
                     return endAttack;
                 }
