--- a/net/minecraft/world/entity/raid/Raider.java
+++ b/net/minecraft/world/entity/raid/Raider.java
@@ -38,6 +_,7 @@
 import net.minecraft.world.entity.monster.AbstractIllager;
 import net.minecraft.world.entity.monster.PatrollingMonster;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.flag.FeatureFlags;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.Level;
@@ -46,9 +_,9 @@
 
 public abstract class Raider extends PatrollingMonster {
     protected static final EntityDataAccessor<Boolean> IS_CELEBRATING = SynchedEntityData.defineId(Raider.class, EntityDataSerializers.BOOLEAN);
-    static final Predicate<ItemEntity> ALLOWED_ITEMS = p_340170_ -> !p_340170_.hasPickUpDelay()
-            && p_340170_.isAlive()
-            && ItemStack.matches(p_340170_.getItem(), Raid.getLeaderBannerInstance(p_340170_.registryAccess().lookupOrThrow(Registries.BANNER_PATTERN)));
+    static final Predicate<ItemEntity> ALLOWED_ITEMS = p_337887_ -> !p_337887_.hasPickUpDelay()
+            && p_337887_.isAlive()
+            && ItemStack.matches(p_337887_.getItem(), Raid.getLeaderBannerInstance(p_337887_.registryAccess().lookupOrThrow(Registries.BANNER_PATTERN)));
     @Nullable
     protected Raid raid;
     private int wave;
@@ -130,7 +_,10 @@
                 raid.removeFromRaid(this, false);
             }
 
-            if (this.isPatrolLeader() && raid == null && ((ServerLevel)this.level()).getRaidAt(this.blockPosition()) == null) {
+            if (!this.level().enabledFeatures().contains(FeatureFlags.UPDATE_1_21)
+                && this.isPatrolLeader()
+                && raid == null
+                && ((ServerLevel)this.level()).getRaidAt(this.blockPosition()) == null) {
                 ItemStack itemstack = this.getItemBySlot(EquipmentSlot.HEAD);
                 Player player = null;
                 if (entity instanceof Player) {
@@ -151,7 +_,7 @@
                         i += mobeffectinstance1.getAmplifier();
                         player.removeEffectNoUpdate(MobEffects.BAD_OMEN);
                     } else {
-                        --i;
+                        i--;
                     }
 
                     i = Mth.clamp(i, 0, 4);
@@ -180,6 +_,20 @@
         return this.raid;
     }
 
+    public boolean isCaptain() {
+        ItemStack itemstack = this.getItemBySlot(EquipmentSlot.HEAD);
+        boolean flag = !itemstack.isEmpty()
+            && ItemStack.matches(itemstack, Raid.getLeaderBannerInstance(this.registryAccess().lookupOrThrow(Registries.BANNER_PATTERN)));
+        boolean flag1 = this.isPatrolLeader();
+        return flag && flag1;
+    }
+
+    public boolean hasRaid() {
+        return !(this.level() instanceof ServerLevel serverlevel)
+            ? false
+            : this.getCurrentRaid() != null || serverlevel.getRaidAt(this.blockPosition()) != null;
+    }
+
     public boolean hasActiveRaid() {
         return this.getCurrentRaid() != null && this.getCurrentRaid().isActive();
     }
@@ -316,7 +_,7 @@
             super.start();
             this.mob.getNavigation().stop();
 
-            for(Raider raider : this.mob
+            for (Raider raider : this.mob
                 .level()
                 .getNearbyEntities(Raider.class, this.shoutTargeting, this.mob, this.mob.getBoundingBox().inflate(8.0, 8.0, 8.0))) {
                 raider.setTarget(this.mob.getTarget());
@@ -328,7 +_,7 @@
             super.stop();
             LivingEntity livingentity = this.mob.getTarget();
             if (livingentity != null) {
-                for(Raider raider : this.mob
+                for (Raider raider : this.mob
                     .level()
                     .getNearbyEntities(Raider.class, this.shoutTargeting, this.mob, this.mob.getBoundingBox().inflate(8.0, 8.0, 8.0))) {
                     raider.setTarget(livingentity);
@@ -489,13 +_,11 @@
 
         @Override
         public boolean canContinueToUse() {
-            if (this.raider.getNavigation().isDone()) {
-                return false;
-            } else {
-                return this.raider.getTarget() == null
+            return this.raider.getNavigation().isDone()
+                ? false
+                : this.raider.getTarget() == null
                     && !this.poiPos.closerToCenterThan(this.raider.position(), (double)(this.raider.getBbWidth() + (float)this.distanceToPoi))
                     && !this.stuck;
-            }
         }
 
         @Override
@@ -532,7 +_,7 @@
         }
 
         private boolean hasNotVisited(BlockPos p_37943_) {
-            for(BlockPos blockpos : this.visited) {
+            for (BlockPos blockpos : this.visited) {
                 if (Objects.equals(p_37943_, blockpos)) {
                     return false;
                 }
