--- a/net/minecraft/data/loot/BlockLootSubProvider.java
+++ b/net/minecraft/data/loot/BlockLootSubProvider.java
@@ -82,7 +_,6 @@
     protected static final LootItemCondition.Builder HAS_NO_SILK_TOUCH = HAS_SILK_TOUCH.invert();
     protected static final LootItemCondition.Builder HAS_SHEARS = MatchTool.toolMatches(ItemPredicate.Builder.item().of(Items.SHEARS));
     private static final LootItemCondition.Builder HAS_SHEARS_OR_SILK_TOUCH = HAS_SHEARS.or(HAS_SILK_TOUCH);
-    protected static final LootItemCondition.Builder HAS_POTATO_PEELER = MatchTool.toolMatches(ItemPredicate.Builder.item().of(Items.POTATO_PEELER));
     private static final LootItemCondition.Builder HAS_NO_SHEARS_OR_SILK_TOUCH = HAS_SHEARS_OR_SILK_TOUCH.invert();
     protected final Set<Item> explosionResistant;
     protected final FeatureFlagSet enabledFeatures;
@@ -101,11 +_,11 @@
     }
 
     protected <T extends FunctionUserBuilder<T>> T applyExplosionDecay(ItemLike p_248695_, FunctionUserBuilder<T> p_248548_) {
-        return (T)(!this.explosionResistant.contains(p_248695_.asItem()) ? p_248548_.apply(ApplyExplosionDecay.explosionDecay()) : p_248548_.unwrap());
+        return !this.explosionResistant.contains(p_248695_.asItem()) ? p_248548_.apply(ApplyExplosionDecay.explosionDecay()) : p_248548_.unwrap();
     }
 
     protected <T extends ConditionUserBuilder<T>> T applyExplosionCondition(ItemLike p_249717_, ConditionUserBuilder<T> p_248851_) {
-        return (T)(!this.explosionResistant.contains(p_249717_.asItem()) ? p_248851_.when(ExplosionCondition.survivesExplosion()) : p_248851_.unwrap());
+        return !this.explosionResistant.contains(p_249717_.asItem()) ? p_248851_.when(ExplosionCondition.survivesExplosion()) : p_248851_.unwrap();
     }
 
     public LootTable.Builder createSingleItemTable(ItemLike p_251912_) {
@@ -117,11 +_,7 @@
         Block p_252253_, LootItemCondition.Builder p_248764_, LootPoolEntryContainer.Builder<?> p_249146_
     ) {
         return LootTable.lootTable()
-            .withPool(
-                LootPool.lootPool()
-                    .setRolls(ConstantValue.exactly(1.0F))
-                    .add(((LootPoolSingletonContainer.Builder)LootItem.lootTableItem(p_252253_).when(p_248764_)).otherwise(p_249146_))
-            );
+            .withPool(LootPool.lootPool().setRolls(ConstantValue.exactly(1.0F)).add(LootItem.lootTableItem(p_252253_).when(p_248764_).otherwise(p_249146_)));
     }
 
     protected static LootTable.Builder createSilkTouchDispatchTable(Block p_250203_, LootPoolEntryContainer.Builder<?> p_252089_) {
@@ -137,7 +_,9 @@
     }
 
     protected LootTable.Builder createSingleItemTableWithSilkTouch(Block p_249305_, ItemLike p_251905_) {
-        return createSilkTouchDispatchTable(p_249305_, this.applyExplosionCondition(p_249305_, LootItem.lootTableItem(p_251905_)));
+        return createSilkTouchDispatchTable(
+            p_249305_, (LootPoolEntryContainer.Builder<?>)this.applyExplosionCondition(p_249305_, LootItem.lootTableItem(p_251905_))
+        );
     }
 
     protected LootTable.Builder createSingleItemTable(ItemLike p_251584_, NumberProvider p_249865_) {
@@ -145,13 +_,20 @@
             .withPool(
                 LootPool.lootPool()
                     .setRolls(ConstantValue.exactly(1.0F))
-                    .add(this.applyExplosionDecay(p_251584_, LootItem.lootTableItem(p_251584_).apply(SetItemCountFunction.setCount(p_249865_))))
+                    .add(
+                        (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
+                            p_251584_, LootItem.lootTableItem(p_251584_).apply(SetItemCountFunction.setCount(p_249865_))
+                        )
+                    )
             );
     }
 
     protected LootTable.Builder createSingleItemTableWithSilkTouch(Block p_251449_, ItemLike p_248558_, NumberProvider p_250047_) {
         return createSilkTouchDispatchTable(
-            p_251449_, this.applyExplosionDecay(p_251449_, LootItem.lootTableItem(p_248558_).apply(SetItemCountFunction.setCount(p_250047_)))
+            p_251449_,
+            (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
+                p_251449_, LootItem.lootTableItem(p_248558_).apply(SetItemCountFunction.setCount(p_250047_))
+            )
         );
     }
 
@@ -176,7 +_,7 @@
                 LootPool.lootPool()
                     .setRolls(ConstantValue.exactly(1.0F))
                     .add(
-                        this.applyExplosionDecay(
+                        (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                             p_251313_,
                             LootItem.lootTableItem(p_251313_)
                                 .apply(
@@ -220,22 +_,7 @@
                         .setRolls(ConstantValue.exactly(1.0F))
                         .add(
                             LootItem.lootTableItem(p_252291_)
-                                .apply(CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY).copy(DataComponents.CUSTOM_NAME))
-                        )
-                )
-            );
-    }
-
-    protected LootTable.Builder createFletchingTableTableTable(Block p_341522_) {
-        return LootTable.lootTable()
-            .withPool(
-                this.applyExplosionCondition(
-                    p_341522_,
-                    LootPool.lootPool()
-                        .setRolls(ConstantValue.exactly(1.0F))
-                        .add(
-                            LootItem.lootTableItem(p_341522_)
-                                .apply(CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY).copy(DataComponents.FLETCHING))
+                                .apply(CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY).include(DataComponents.CUSTOM_NAME))
                         )
                 )
             );
@@ -252,10 +_,10 @@
                             LootItem.lootTableItem(p_252164_)
                                 .apply(
                                     CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY)
-                                        .copy(DataComponents.CUSTOM_NAME)
-                                        .copy(DataComponents.CONTAINER)
-                                        .copy(DataComponents.LOCK)
-                                        .copy(DataComponents.CONTAINER_LOOT)
+                                        .include(DataComponents.CUSTOM_NAME)
+                                        .include(DataComponents.CONTAINER)
+                                        .include(DataComponents.LOCK)
+                                        .include(DataComponents.CONTAINER_LOOT)
                                 )
                         )
                 )
@@ -265,7 +_,7 @@
     protected LootTable.Builder createCopperOreDrops(Block p_251306_) {
         return createSilkTouchDispatchTable(
             p_251306_,
-            this.applyExplosionDecay(
+            (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                 p_251306_,
                 LootItem.lootTableItem(Items.RAW_COPPER)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 5.0F)))
@@ -277,7 +_,7 @@
     protected LootTable.Builder createLapisOreDrops(Block p_251511_) {
         return createSilkTouchDispatchTable(
             p_251511_,
-            this.applyExplosionDecay(
+            (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                 p_251511_,
                 LootItem.lootTableItem(Items.LAPIS_LAZULI)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(4.0F, 9.0F)))
@@ -289,7 +_,7 @@
     protected LootTable.Builder createRedstoneOreDrops(Block p_251906_) {
         return createSilkTouchDispatchTable(
             p_251906_,
-            this.applyExplosionDecay(
+            (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                 p_251906_,
                 LootItem.lootTableItem(Items.REDSTONE)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(4.0F, 5.0F)))
@@ -309,8 +_,10 @@
                             LootItem.lootTableItem(p_249810_)
                                 .apply(
                                     CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY)
-                                        .copy(DataComponents.CUSTOM_NAME)
-                                        .copy(DataComponents.BANNER_PATTERNS)
+                                        .include(DataComponents.CUSTOM_NAME)
+                                        .include(DataComponents.ITEM_NAME)
+                                        .include(DataComponents.HIDE_ADDITIONAL_TOOLTIP)
+                                        .include(DataComponents.BANNER_PATTERNS)
                                 )
                         )
                 )
@@ -325,7 +_,7 @@
                     .setRolls(ConstantValue.exactly(1.0F))
                     .add(
                         LootItem.lootTableItem(p_250988_)
-                            .apply(CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY).copy(DataComponents.BEES))
+                            .apply(CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY).include(DataComponents.BEES))
                             .apply(CopyBlockState.copyState(p_250988_).copy(BeehiveBlock.HONEY_LEVEL))
                     )
             );
@@ -337,8 +_,9 @@
                 LootPool.lootPool()
                     .setRolls(ConstantValue.exactly(1.0F))
                     .add(
-                        ((LootPoolSingletonContainer.Builder)LootItem.lootTableItem(p_248770_).when(HAS_SILK_TOUCH))
-                            .apply(CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY).copy(DataComponents.BEES))
+                        LootItem.lootTableItem(p_248770_)
+                            .when(HAS_SILK_TOUCH)
+                            .apply(CopyComponentsFunction.copyComponents(CopyComponentsFunction.Source.BLOCK_ENTITY).include(DataComponents.BEES))
                             .apply(CopyBlockState.copyState(p_248770_).copy(BeehiveBlock.HONEY_LEVEL))
                             .otherwise(LootItem.lootTableItem(p_248770_))
                     )
@@ -359,14 +_,17 @@
 
     protected LootTable.Builder createOreDrop(Block p_250450_, Item p_249745_) {
         return createSilkTouchDispatchTable(
-            p_250450_, this.applyExplosionDecay(p_250450_, LootItem.lootTableItem(p_249745_).apply(ApplyBonusCount.addOreBonusCount(Enchantments.FORTUNE)))
+            p_250450_,
+            (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
+                p_250450_, LootItem.lootTableItem(p_249745_).apply(ApplyBonusCount.addOreBonusCount(Enchantments.FORTUNE))
+            )
         );
     }
 
     protected LootTable.Builder createMushroomBlockDrop(Block p_249959_, ItemLike p_249315_) {
         return createSilkTouchDispatchTable(
             p_249959_,
-            this.applyExplosionDecay(
+            (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                 p_249959_,
                 LootItem.lootTableItem(p_249315_)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(-6.0F, 2.0F)))
@@ -378,9 +_,10 @@
     protected LootTable.Builder createGrassDrops(Block p_252139_) {
         return createShearsDispatchTable(
             p_252139_,
-            this.applyExplosionDecay(
+            (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                 p_252139_,
-                LootItem.lootTableItem(Items.WHEAT_SEEDS).when(LootItemRandomChanceCondition.randomChance(0.125F))
+                LootItem.lootTableItem(Items.WHEAT_SEEDS)
+                    .when(LootItemRandomChanceCondition.randomChance(0.125F))
                     .apply(ApplyBonusCount.addUniformBonusCount(Enchantments.FORTUNE, 2))
             )
         );
@@ -394,7 +_,7 @@
                     LootPool.lootPool()
                         .setRolls(ConstantValue.exactly(1.0F))
                         .add(
-                            (LootPoolEntryContainer.Builder<?>)LootItem.lootTableItem(p_249098_)
+                            LootItem.lootTableItem(p_249098_)
                                 .apply(
                                     StemBlock.AGE.getPossibleValues(),
                                     p_249795_ -> SetItemCountFunction.setCount(BinomialDistributionGenerator.binomial(3, (float)(p_249795_ + 1) / 15.0F))
@@ -432,20 +_,21 @@
             .withPool(
                 LootPool.lootPool()
                     .add(
-                        this.applyExplosionDecay(
+                        (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                             p_249088_,
-                            LootItem.lootTableItem(p_249088_).when(p_251535_)
-                                    .apply(
-                                        Direction.values(),
-                                        p_251536_ -> SetItemCountFunction.setCount(ConstantValue.exactly(1.0F), true)
-                                                .when(
-                                                    LootItemBlockStatePropertyCondition.hasBlockStateProperties(p_249088_)
-                                                        .setProperties(
-                                                            StatePropertiesPredicate.Builder.properties()
-                                                                .hasProperty(MultifaceBlock.getFaceProperty(p_251536_), true)
-                                                        )
-                                                )
-                                    )
+                            LootItem.lootTableItem(p_249088_)
+                                .when(p_251535_)
+                                .apply(
+                                    Direction.values(),
+                                    p_251536_ -> SetItemCountFunction.setCount(ConstantValue.exactly(1.0F), true)
+                                            .when(
+                                                LootItemBlockStatePropertyCondition.hasBlockStateProperties(p_249088_)
+                                                    .setProperties(
+                                                        StatePropertiesPredicate.Builder.properties()
+                                                            .hasProperty(MultifaceBlock.getFaceProperty(p_251536_), true)
+                                                    )
+                                            )
+                                )
                                 .apply(SetItemCountFunction.setCount(ConstantValue.exactly(-1.0F), true))
                         )
                     )
@@ -455,7 +_,7 @@
     protected LootTable.Builder createLeavesDrops(Block p_250088_, Block p_250731_, float... p_248949_) {
         return createSilkTouchOrShearsDispatchTable(
                 p_250088_,
-                this.applyExplosionCondition(p_250088_, LootItem.lootTableItem(p_250731_))
+                ((LootPoolSingletonContainer.Builder)this.applyExplosionCondition(p_250088_, LootItem.lootTableItem(p_250731_)))
                     .when(BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.FORTUNE, p_248949_))
             )
             .withPool(
@@ -463,9 +_,9 @@
                     .setRolls(ConstantValue.exactly(1.0F))
                     .when(HAS_NO_SHEARS_OR_SILK_TOUCH)
                     .add(
-                        this.applyExplosionDecay(
+                        ((LootPoolSingletonContainer.Builder)this.applyExplosionDecay(
                                 p_250088_, LootItem.lootTableItem(Items.STICK).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F)))
-                            )
+                            ))
                             .when(BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.FORTUNE, NORMAL_LEAVES_STICK_CHANCES))
                     )
             );
@@ -478,7 +_,7 @@
                     .setRolls(ConstantValue.exactly(1.0F))
                     .when(HAS_NO_SHEARS_OR_SILK_TOUCH)
                     .add(
-                        this.applyExplosionCondition(p_249535_, LootItem.lootTableItem(Items.APPLE))
+                        ((LootPoolSingletonContainer.Builder)this.applyExplosionCondition(p_249535_, LootItem.lootTableItem(Items.APPLE)))
                             .when(BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.FORTUNE, 0.005F, 0.0055555557F, 0.00625F, 0.008333334F, 0.025F))
                     )
             );
@@ -487,9 +_,9 @@
     protected LootTable.Builder createMangroveLeavesDrops(Block p_251103_) {
         return createSilkTouchOrShearsDispatchTable(
             p_251103_,
-            this.applyExplosionDecay(
+            ((LootPoolSingletonContainer.Builder)this.applyExplosionDecay(
                     Blocks.MANGROVE_LEAVES, LootItem.lootTableItem(Items.STICK).apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F)))
-                )
+                ))
                 .when(BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.FORTUNE, NORMAL_LEAVES_STICK_CHANCES))
         );
     }
@@ -498,13 +_,7 @@
         return this.applyExplosionDecay(
             p_249457_,
             LootTable.lootTable()
-                .withPool(
-                    LootPool.lootPool()
-                        .add(
-                            ((LootPoolSingletonContainer.Builder)LootItem.lootTableItem(p_248599_).when(p_252202_))
-                                .otherwise(LootItem.lootTableItem(p_251915_))
-                        )
-                )
+                .withPool(LootPool.lootPool().add(LootItem.lootTableItem(p_248599_).when(p_252202_).otherwise(LootItem.lootTableItem(p_251915_))))
                 .withPool(
                     LootPool.lootPool()
                         .when(p_252202_)
@@ -521,11 +_,12 @@
     }
 
     protected LootTable.Builder createDoublePlantWithSeedDrops(Block p_248590_, Block p_248735_) {
-        LootPoolEntryContainer.Builder<?> builder = ((LootPoolSingletonContainer.Builder)LootItem.lootTableItem(p_248735_)
-                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(2.0F)))
-                .when(HAS_SHEARS))
+        LootPoolEntryContainer.Builder<?> builder = LootItem.lootTableItem(p_248735_)
+            .apply(SetItemCountFunction.setCount(ConstantValue.exactly(2.0F)))
+            .when(HAS_SHEARS)
             .otherwise(
-                this.applyExplosionCondition(p_248590_, LootItem.lootTableItem(Items.WHEAT_SEEDS)).when(LootItemRandomChanceCondition.randomChance(0.125F))
+                ((LootPoolSingletonContainer.Builder)this.applyExplosionCondition(p_248590_, LootItem.lootTableItem(Items.WHEAT_SEEDS)))
+                    .when(LootItemRandomChanceCondition.randomChance(0.125F))
             );
         return LootTable.lootTable()
             .withPool(
@@ -574,7 +_,7 @@
                 LootPool.lootPool()
                     .setRolls(ConstantValue.exactly(1.0F))
                     .add(
-                        this.applyExplosionDecay(
+                        (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                             p_250896_,
                             LootItem.lootTableItem(p_250896_)
                                 .apply(
@@ -598,7 +_,7 @@
                 LootPool.lootPool()
                     .setRolls(ConstantValue.exactly(1.0F))
                     .add(
-                        this.applyExplosionDecay(
+                        (LootPoolEntryContainer.Builder<?>)this.applyExplosionDecay(
                             p_273240_,
                             LootItem.lootTableItem(p_273240_)
                                 .apply(
@@ -626,12 +_,16 @@
 
     protected abstract void generate();
 
+    protected Iterable<Block> getKnownBlocks() {
+        return BuiltInRegistries.BLOCK;
+    }
+
     @Override
     public void generate(HolderLookup.Provider p_331472_, BiConsumer<ResourceKey<LootTable>, LootTable.Builder> p_249322_) {
         this.generate();
         Set<ResourceKey<LootTable>> set = new HashSet<>();
 
-        for(Block block : BuiltInRegistries.BLOCK) {
+        for(Block block : getKnownBlocks()) {
             if (block.isEnabled(this.enabledFeatures)) {
                 ResourceKey<LootTable> resourcekey = block.getLootTable();
                 if (resourcekey != BuiltInLootTables.EMPTY && set.add(resourcekey)) {
