--- a/net/minecraft/world/level/levelgen/Aquifer.java
+++ b/net/minecraft/world/level/levelgen/Aquifer.java
@@ -162,9 +_,9 @@
                     long k2 = 0L;
                     long l2 = 0L;
 
-                    for(int i3 = 0; i3 <= 1; ++i3) {
-                        for(int j3 = -1; j3 <= 1; ++j3) {
-                            for(int k3 = 0; k3 <= 1; ++k3) {
+                    for (int i3 = 0; i3 <= 1; i3++) {
+                        for (int j3 = -1; j3 <= 1; j3++) {
+                            for (int k3 = 0; k3 <= 1; k3++) {
                                 int l3 = l + i3;
                                 int i4 = i1 + j3;
                                 int j4 = j1 + k3;
@@ -356,7 +_,7 @@
             int k = p_188449_ - 12;
             boolean flag = false;
 
-            for(int[] aint : SURFACE_SAMPLING_OFFSETS_IN_CHUNKS) {
+            for (int[] aint : SURFACE_SAMPLING_OFFSETS_IN_CHUNKS) {
                 int l = p_188448_ + SectionPos.sectionToBlockCoord(aint[0]);
                 int i1 = p_188450_ + SectionPos.sectionToBlockCoord(aint[1]);
                 int j1 = this.noiseChunk.preliminarySurfaceLevel(l, i1);
