--- a/net/minecraft/world/level/levelgen/DensityFunction.java
+++ b/net/minecraft/world/level/levelgen/DensityFunction.java
@@ -79,7 +_,7 @@
 
     public static record NoiseHolder(Holder<NormalNoise.NoiseParameters> noiseData, @Nullable NormalNoise noise) {
         public static final Codec<DensityFunction.NoiseHolder> CODEC = NormalNoise.NoiseParameters.CODEC
-            .xmap(p_224011_ -> new DensityFunction.NoiseHolder(p_224011_, null), DensityFunction.NoiseHolder::noiseData);
+            .xmap(p_224011_ -> new DensityFunction.NoiseHolder((Holder<NormalNoise.NoiseParameters>)p_224011_, null), DensityFunction.NoiseHolder::noiseData);
 
         public NoiseHolder(Holder<NormalNoise.NoiseParameters> p_224001_) {
             this(p_224001_, null);
