--- a/net/minecraft/world/level/levelgen/FlatLevelSource.java
+++ b/net/minecraft/world/level/levelgen/FlatLevelSource.java
@@ -1,12 +_,11 @@
 package net.minecraft.world.level.levelgen;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
-import java.util.function.Function;
 import java.util.stream.Stream;
 import net.minecraft.Util;
 import net.minecraft.core.BlockPos;
@@ -29,7 +_,7 @@
 import net.minecraft.world.level.levelgen.structure.StructureSet;
 
 public class FlatLevelSource extends ChunkGenerator {
-    public static final Codec<FlatLevelSource> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<FlatLevelSource> CODEC = RecordCodecBuilder.mapCodec(
         p_255577_ -> p_255577_.group(FlatLevelGeneratorSettings.CODEC.fieldOf("settings").forGetter(FlatLevelSource::settings))
                 .apply(p_255577_, p_255577_.stable(FlatLevelSource::new))
     );
@@ -45,15 +_,12 @@
         Stream<Holder<StructureSet>> stream = this.settings
             .structureOverrides()
             .map(HolderSet::stream)
-            .orElseGet(
-                () -> p_256602_.listElements()
-                        .map((Function<? super Holder.Reference<StructureSet>, ? extends Holder.Reference<StructureSet>>)(p_255579_ -> p_255579_))
-            );
+            .orElseGet(() -> p_256602_.listElements().map(p_255579_ -> (Holder<StructureSet>)p_255579_));
         return ChunkGeneratorStructureState.createForFlat(p_255830_, p_256355_, this.biomeSource, stream);
     }
 
     @Override
-    protected Codec<? extends ChunkGenerator> codec() {
+    protected MapCodec<? extends ChunkGenerator> codec() {
         return CODEC;
     }
 
@@ -79,13 +_,13 @@
         Heightmap heightmap = p_224187_.getOrCreateHeightmapUnprimed(Heightmap.Types.OCEAN_FLOOR_WG);
         Heightmap heightmap1 = p_224187_.getOrCreateHeightmapUnprimed(Heightmap.Types.WORLD_SURFACE_WG);
 
-        for(int i = 0; i < Math.min(p_224187_.getHeight(), list.size()); ++i) {
+        for (int i = 0; i < Math.min(p_224187_.getHeight(), list.size()); i++) {
             BlockState blockstate = list.get(i);
             if (blockstate != null) {
                 int j = p_224187_.getMinBuildHeight() + i;
 
-                for(int k = 0; k < 16; ++k) {
-                    for(int l = 0; l < 16; ++l) {
+                for (int k = 0; k < 16; k++) {
+                    for (int l = 0; l < 16; l++) {
                         p_224187_.setBlockState(blockpos$mutableblockpos.set(k, j, l), blockstate, false);
                         heightmap.update(k, j, l, blockstate);
                         heightmap1.update(k, j, l, blockstate);
@@ -101,7 +_,7 @@
     public int getBaseHeight(int p_224160_, int p_224161_, Heightmap.Types p_224162_, LevelHeightAccessor p_224163_, RandomState p_224164_) {
         List<BlockState> list = this.settings.getLayers();
 
-        for(int i = Math.min(list.size(), p_224163_.getMaxBuildHeight()) - 1; i >= 0; --i) {
+        for (int i = Math.min(list.size(), p_224163_.getMaxBuildHeight()) - 1; i >= 0; i--) {
             BlockState blockstate = list.get(i);
             if (blockstate != null && p_224162_.isOpaque().test(blockstate)) {
                 return p_224163_.getMinBuildHeight() + i + 1;
@@ -120,7 +_,7 @@
                 .stream()
                 .limit((long)p_224157_.getHeight())
                 .map(p_204549_ -> p_204549_ == null ? Blocks.AIR.defaultBlockState() : p_204549_)
-                .toArray(p_204543_ -> new BlockState[p_204543_])
+                .toArray(BlockState[]::new)
         );
     }
 
