--- a/net/minecraft/world/level/levelgen/NoiseSettings.java
+++ b/net/minecraft/world/level/levelgen/NoiseSettings.java
@@ -2,7 +_,7 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
-import com.mojang.serialization.DataResult.PartialResult;
+import com.mojang.serialization.DataResult.Error;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.function.Function;
@@ -26,7 +_,6 @@
     protected static final NoiseSettings END_NOISE_SETTINGS = create(0, 128, 2, 1);
     protected static final NoiseSettings CAVES_NOISE_SETTINGS = create(-64, 192, 1, 2);
     protected static final NoiseSettings FLOATING_ISLANDS_NOISE_SETTINGS = create(0, 256, 2, 1);
-    protected static final NoiseSettings POTATO_NOISE_SETTINGS = create(0, 256, 4, 1);
 
     private static DataResult<NoiseSettings> guardY(NoiseSettings p_158721_) {
         if (p_158721_.minY() + p_158721_.height() > DimensionType.MAX_Y + 1) {
@@ -40,8 +_,8 @@
 
     public static NoiseSettings create(int p_224526_, int p_224527_, int p_224528_, int p_224529_) {
         NoiseSettings noisesettings = new NoiseSettings(p_224526_, p_224527_, p_224528_, p_224529_);
-        guardY(noisesettings).error().ifPresent(p_158719_ -> {
-            throw new IllegalStateException(p_158719_.message());
+        guardY(noisesettings).error().ifPresent(p_338093_ -> {
+            throw new IllegalStateException(p_338093_.message());
         });
         return noisesettings;
     }
