--- a/net/minecraft/world/level/levelgen/NoiseRouterData.java
+++ b/net/minecraft/world/level/levelgen/NoiseRouterData.java
@@ -423,41 +_,10 @@
         );
     }
 
-    private static NoiseRouter noNewCavesWithExtraZing(
-        HolderGetter<DensityFunction> p_340666_, HolderGetter<NormalNoise.NoiseParameters> p_342475_, DensityFunction p_340444_
-    ) {
-        DensityFunction densityfunction = getFunction(p_340666_, SHIFT_X);
-        DensityFunction densityfunction1 = getFunction(p_340666_, SHIFT_Z);
-        DensityFunction densityfunction2 = DensityFunctions.shiftedNoise2d(densityfunction, densityfunction1, 0.25, p_342475_.getOrThrow(Noises.TEMPERATURE));
-        DensityFunction densityfunction3 = DensityFunctions.shiftedNoise2d(densityfunction, densityfunction1, 0.25, p_342475_.getOrThrow(Noises.VEGETATION));
-        DensityFunction densityfunction4 = postProcess(p_340444_);
-        return new NoiseRouter(
-            DensityFunctions.zero(),
-            DensityFunctions.zero(),
-            DensityFunctions.zero(),
-            DensityFunctions.zero(),
-            densityfunction2,
-            densityfunction3,
-            getFunction(p_340666_, CONTINENTS),
-            DensityFunctions.zero(),
-            DensityFunctions.zero(),
-            DensityFunctions.zero(),
-            DensityFunctions.zero(),
-            densityfunction4,
-            DensityFunctions.zero(),
-            DensityFunctions.zero(),
-            DensityFunctions.zero()
-        );
-    }
-
     private static DensityFunction slideOverworld(boolean p_224490_, DensityFunction p_224491_) {
         return slide(p_224491_, -64, 384, p_224490_ ? 16 : 80, p_224490_ ? 0 : 64, -0.078125, 0, 24, p_224490_ ? 0.4 : 0.1171875);
     }
 
-    private static DensityFunction slidePotato(HolderGetter<DensityFunction> p_342611_) {
-        return slide(getFunction(p_342611_, BASE_3D_NOISE_END), 0, 256, 192, 96, -0.12, 5, 32, -0.1);
-    }
-
     private static DensityFunction slideNetherLike(HolderGetter<DensityFunction> p_256084_, int p_255802_, int p_255834_) {
         return slide(getFunction(p_256084_, BASE_3D_NOISE_NETHER), p_255802_, p_255834_, 24, 0, 0.9375, -8, 24, 2.5);
     }
@@ -476,10 +_,6 @@
 
     protected static NoiseRouter floatingIslands(HolderGetter<DensityFunction> p_256633_, HolderGetter<NormalNoise.NoiseParameters> p_255902_) {
         return noNewCaves(p_256633_, p_255902_, slideEndLike(getFunction(p_256633_, BASE_3D_NOISE_END), 0, 256));
-    }
-
-    protected static NoiseRouter potato(HolderGetter<DensityFunction> p_342308_, HolderGetter<NormalNoise.NoiseParameters> p_341762_) {
-        return noNewCavesWithExtraZing(p_342308_, p_341762_, slidePotato(p_342308_));
     }
 
     private static DensityFunction slideEnd(DensityFunction p_224506_) {
