--- a/net/minecraft/world/level/levelgen/PatrolSpawner.java
+++ b/net/minecraft/world/level/levelgen/PatrolSpawner.java
@@ -26,11 +_,11 @@
             return 0;
         } else {
             RandomSource randomsource = p_64570_.random;
-            --this.nextTick;
+            this.nextTick--;
             if (this.nextTick > 0) {
                 return 0;
             } else {
-                this.nextTick += 12000 + randomsource.nextInt(1200);
+                this.nextTick = this.nextTick + 12000 + randomsource.nextInt(1200);
                 long i = p_64570_.getDayTime() / 24000L;
                 if (i < 5L || !p_64570_.isDay()) {
                     return 0;
@@ -66,8 +_,8 @@
                                     int j1 = 0;
                                     int k1 = (int)Math.ceil((double)p_64570_.getCurrentDifficultyAt(blockpos$mutableblockpos).getEffectiveDifficulty()) + 1;
 
-                                    for(int l1 = 0; l1 < k1; ++l1) {
-                                        ++j1;
+                                    for (int l1 = 0; l1 < k1; l1++) {
+                                        j1++;
                                         blockpos$mutableblockpos.setY(
                                             p_64570_.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, blockpos$mutableblockpos).getY()
                                         );
