--- a/net/minecraft/world/level/levelgen/SurfaceRules.java
+++ b/net/minecraft/world/level/levelgen/SurfaceRules.java
@@ -124,7 +_,7 @@
         return SurfaceRules.Bandlands.INSTANCE;
     }
 
-    static <A> Codec<? extends A> register(Registry<Codec<? extends A>> p_224604_, String p_224605_, KeyDispatchDataCodec<? extends A> p_224606_) {
+    static <A> MapCodec<? extends A> register(Registry<MapCodec<? extends A>> p_224604_, String p_224605_, KeyDispatchDataCodec<? extends A> p_224606_) {
         return Registry.register(p_224604_, p_224605_, p_224606_.codec());
     }
 
@@ -238,9 +_,9 @@
     public interface ConditionSource extends Function<SurfaceRules.Context, SurfaceRules.Condition> {
         Codec<SurfaceRules.ConditionSource> CODEC = BuiltInRegistries.MATERIAL_CONDITION
             .byNameCodec()
-            .dispatch(p_224613_ -> p_224613_.codec().codec(), Function.identity());
+            .dispatch(p_338094_ -> p_338094_.codec().codec(), Function.identity());
 
-        static Codec<? extends SurfaceRules.ConditionSource> bootstrap(Registry<Codec<? extends SurfaceRules.ConditionSource>> p_204625_) {
+        static MapCodec<? extends SurfaceRules.ConditionSource> bootstrap(Registry<MapCodec<? extends SurfaceRules.ConditionSource>> p_204625_) {
             SurfaceRules.register(p_204625_, "biome", SurfaceRules.BiomeConditionSource.CODEC);
             SurfaceRules.register(p_204625_, "noise_threshold", SurfaceRules.NoiseThresholdConditionSource.CODEC);
             SurfaceRules.register(p_204625_, "vertical_gradient", SurfaceRules.VerticalGradientConditionSource.CODEC);
@@ -308,15 +_,15 @@
         }
 
         protected void updateXZ(int p_189570_, int p_189571_) {
-            ++this.lastUpdateXZ;
-            ++this.lastUpdateY;
+            this.lastUpdateXZ++;
+            this.lastUpdateY++;
             this.blockX = p_189570_;
             this.blockZ = p_189571_;
             this.surfaceDepth = this.system.getSurfaceDepth(p_189570_, p_189571_);
         }
 
         protected void updateY(int p_189577_, int p_189578_, int p_189579_, int p_189580_, int p_189581_, int p_189582_) {
-            ++this.lastUpdateY;
+            this.lastUpdateY++;
             this.biome = Suppliers.memoize(() -> this.biomeGetter.apply(this.pos.set(p_189580_, p_189581_, p_189582_)));
             this.blockY = p_189581_;
             this.waterHeight = p_189579_;
@@ -558,9 +_,9 @@
     public interface RuleSource extends Function<SurfaceRules.Context, SurfaceRules.SurfaceRule> {
         Codec<SurfaceRules.RuleSource> CODEC = BuiltInRegistries.MATERIAL_RULE
             .byNameCodec()
-            .dispatch(p_224627_ -> p_224627_.codec().codec(), Function.identity());
+            .dispatch(p_338095_ -> p_338095_.codec().codec(), Function.identity());
 
-        static Codec<? extends SurfaceRules.RuleSource> bootstrap(Registry<Codec<? extends SurfaceRules.RuleSource>> p_204631_) {
+        static MapCodec<? extends SurfaceRules.RuleSource> bootstrap(Registry<MapCodec<? extends SurfaceRules.RuleSource>> p_204631_) {
             SurfaceRules.register(p_204631_, "bandlands", SurfaceRules.Bandlands.CODEC);
             SurfaceRules.register(p_204631_, "block", SurfaceRules.BlockRuleSource.CODEC);
             SurfaceRules.register(p_204631_, "sequence", SurfaceRules.SequenceRuleSource.CODEC);
@@ -574,7 +_,7 @@
         @Nullable
         @Override
         public BlockState tryApply(int p_189694_, int p_189695_, int p_189696_) {
-            for(SurfaceRules.SurfaceRule surfacerules$surfacerule : this.rules) {
+            for (SurfaceRules.SurfaceRule surfacerules$surfacerule : this.rules) {
                 BlockState blockstate = surfacerules$surfacerule.tryApply(p_189694_, p_189695_, p_189696_);
                 if (blockstate != null) {
                     return blockstate;
@@ -601,7 +_,7 @@
             } else {
                 Builder<SurfaceRules.SurfaceRule> builder = ImmutableList.builder();
 
-                for(SurfaceRules.RuleSource surfacerules$rulesource : this.sequence) {
+                for (SurfaceRules.RuleSource surfacerules$rulesource : this.sequence) {
                     builder.add(surfacerules$rulesource.apply(p_189704_));
                 }
 
@@ -747,7 +_,7 @@
             final PositionalRandomFactory positionalrandomfactory = p_189841_.randomState.getOrCreateRandomFactory(this.randomName());
 
             class VerticalGradientCondition extends SurfaceRules.LazyYCondition {
-                VerticalGradientCondition() {
+                VerticalGradientCondition(SurfaceRules.VerticalGradientConditionSource p_189857_) {
                     super(p_189841_);
                 }
 
@@ -766,7 +_,7 @@
                 }
             }
 
-            return new VerticalGradientCondition();
+            return new VerticalGradientCondition(this);
         }
     }
 
