--- a/net/minecraft/world/level/levelgen/VerticalAnchor.java
+++ b/net/minecraft/world/level/levelgen/VerticalAnchor.java
@@ -3,13 +_,10 @@
 import com.mojang.datafixers.util.Either;
 import com.mojang.serialization.Codec;
 import java.util.function.Function;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.dimension.DimensionType;
 
 public interface VerticalAnchor {
-    Codec<VerticalAnchor> CODEC = ExtraCodecs.xor(
-            VerticalAnchor.Absolute.CODEC, ExtraCodecs.xor(VerticalAnchor.AboveBottom.CODEC, VerticalAnchor.BelowTop.CODEC)
-        )
+    Codec<VerticalAnchor> CODEC = Codec.xor(VerticalAnchor.Absolute.CODEC, Codec.xor(VerticalAnchor.AboveBottom.CODEC, VerticalAnchor.BelowTop.CODEC))
         .xmap(VerticalAnchor::merge, VerticalAnchor::split);
     VerticalAnchor BOTTOM = aboveBottom(0);
     VerticalAnchor TOP = belowTop(0);
@@ -35,7 +_,7 @@
     }
 
     private static VerticalAnchor merge(Either<VerticalAnchor.Absolute, Either<VerticalAnchor.AboveBottom, VerticalAnchor.BelowTop>> p_158925_) {
-        return p_158925_.map(Function.identity(), p_209698_ -> p_209698_.map(Function.identity(), Function.identity()));
+        return p_158925_.map(Function.identity(), Either::unwrap);
     }
 
     private static Either<VerticalAnchor.Absolute, Either<VerticalAnchor.AboveBottom, VerticalAnchor.BelowTop>> split(VerticalAnchor p_158927_) {
