--- a/net/minecraft/world/level/levelgen/XoroshiroRandomSource.java
+++ b/net/minecraft/world/level/levelgen/XoroshiroRandomSource.java
@@ -59,7 +_,7 @@
             long j = i * (long)p_190118_;
             long k = j & 4294967295L;
             if (k < (long)p_190118_) {
-                for(int l = Integer.remainderUnsigned(~p_190118_ + 1, p_190118_); k < (long)l; k = j & 4294967295L) {
+                for (int l = Integer.remainderUnsigned(~p_190118_ + 1, p_190118_); k < (long)l; k = j & 4294967295L) {
                     i = Integer.toUnsignedLong(this.nextInt());
                     j = i * (long)p_190118_;
                 }
@@ -97,7 +_,7 @@
 
     @Override
     public void consumeCount(int p_190111_) {
-        for(int i = 0; i < p_190111_; ++i) {
+        for (int i = 0; i < p_190111_; i++) {
             this.randomNumberGenerator.nextLong();
         }
     }
