--- a/net/minecraft/world/level/levelgen/Xoroshiro128PlusPlus.java
+++ b/net/minecraft/world/level/levelgen/Xoroshiro128PlusPlus.java
@@ -9,7 +_,7 @@
     private long seedHi;
     public static final Codec<Xoroshiro128PlusPlus> CODEC = Codec.LONG_STREAM
         .comapFlatMap(
-            p_287733_ -> Util.fixedSize(p_287733_, 2).map(p_287742_ -> new Xoroshiro128PlusPlus(p_287742_[0], p_287742_[1])),
+            p_338097_ -> Util.fixedSize(p_338097_, 2).map(p_287742_ -> new Xoroshiro128PlusPlus(p_287742_[0], p_287742_[1])),
             p_287687_ -> LongStream.of(p_287687_.seedLo, p_287687_.seedHi)
         );
 
