--- a/net/minecraft/server/players/OldUsersConverter.java
+++ b/net/minecraft/server/players/OldUsersConverter.java
@@ -36,7 +_,7 @@
     static List<String> readOldListFormat(File p_11074_, Map<String, String[]> p_11075_) throws IOException {
         List<String> list = Files.readLines(p_11074_, StandardCharsets.UTF_8);
 
-        for(String s : list) {
+        for (String s : list) {
             s = s.trim();
             if (!s.startsWith("#") && s.length() >= 1) {
                 String[] astring = s.split("\\|");
@@ -48,11 +_,11 @@
     }
 
     private static void lookupPlayers(MinecraftServer p_11087_, Collection<String> p_11088_, ProfileLookupCallback p_11089_) {
-        String[] astring = p_11088_.stream().filter(p_11077_ -> !StringUtil.isNullOrEmpty(p_11077_)).toArray(p_11070_ -> new String[p_11070_]);
+        String[] astring = p_11088_.stream().filter(p_11077_ -> !StringUtil.isNullOrEmpty(p_11077_)).toArray(String[]::new);
         if (p_11087_.usesAuthentication()) {
             p_11087_.getProfileRepository().findProfilesByNames(astring, p_11089_);
         } else {
-            for(String s : astring) {
+            for (String s : astring) {
                 p_11089_.onProfileLookupSucceeded(UUIDUtil.createOfflineProfile(s));
             }
         }
@@ -76,7 +_,7 @@
                     @Override
                     public void onProfileLookupSucceeded(GameProfile p_11123_) {
                         p_11082_.getProfileCache().add(p_11123_);
-                        String[] astring = (String[])map.get(p_11123_.getName().toLowerCase(Locale.ROOT));
+                        String[] astring = map.get(p_11123_.getName().toLowerCase(Locale.ROOT));
                         if (astring == null) {
                             OldUsersConverter.LOGGER.warn("Could not convert user banlist entry for {}", p_11123_.getName());
                             throw new OldUsersConverter.ConversionError("Profile not in the conversionlist");
@@ -128,8 +_,8 @@
                 Map<String, String[]> map = Maps.newHashMap();
                 readOldListFormat(OLD_IPBANLIST, map);
 
-                for(String s : map.keySet()) {
-                    String[] astring = (String[])map.get(s);
+                for (String s : map.keySet()) {
+                    String[] astring = map.get(s);
                     Date date = astring.length > 1 ? parseDate(astring[1], null) : null;
                     String s1 = astring.length > 2 ? astring[2] : null;
                     Date date1 = astring.length > 3 ? parseDate(astring[3], null) : null;
@@ -279,7 +_,7 @@
             File[] afile = file1.listFiles();
             List<String> list = Lists.newArrayList();
 
-            for(File file4 : afile) {
+            for (File file4 : afile) {
                 String s = file4.getName();
                 if (s.toLowerCase(Locale.ROOT).endsWith(".dat")) {
                     String s1 = s.substring(0, s.length() - ".dat".length());
@@ -322,7 +_,7 @@
                     private String getFileNameForProfile(String p_296396_) {
                         String s2 = null;
 
-                        for(String s3 : astring) {
+                        for (String s3 : astring) {
                             if (s3 != null && s3.equalsIgnoreCase(p_296396_)) {
                                 s2 = s3;
                                 break;
