--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -127,6 +_,7 @@
     private int simulationDistance;
     private boolean allowCommandsForAllPlayers;
     private int sendAllPlayerInfoIn;
+    private final List<ServerPlayer> playersView = java.util.Collections.unmodifiableList(players);
 
     public PlayerList(MinecraftServer server, LayeredRegistryAccess<RegistryLayer> registries, PlayerDataStorage playerIo, NotificationService notificationService) {
         this.server = server;
@@ -158,7 +_,7 @@
         LevelData leveldata = serverlevel.getLevelData();
         ServerGamePacketListenerImpl servergamepacketlistenerimpl = new ServerGamePacketListenerImpl(this.server, connection, player, cookie);
         connection.setupInboundProtocol(
-            GameProtocols.SERVERBOUND_TEMPLATE.bind(RegistryFriendlyByteBuf.decorator(this.server.registryAccess()), servergamepacketlistenerimpl),
+            GameProtocols.SERVERBOUND_TEMPLATE.bind(RegistryFriendlyByteBuf.decorator(this.server.registryAccess(), servergamepacketlistenerimpl.getConnectionType()), servergamepacketlistenerimpl),
             servergamepacketlistenerimpl
         );
         servergamepacketlistenerimpl.suspendFlushing();
@@ -184,10 +_,12 @@
         servergamepacketlistenerimpl.send(new ClientboundChangeDifficultyPacket(leveldata.getDifficulty(), leveldata.isDifficultyLocked()));
         servergamepacketlistenerimpl.send(new ClientboundPlayerAbilitiesPacket(player.getAbilities()));
         servergamepacketlistenerimpl.send(new ClientboundSetHeldSlotPacket(player.getInventory().getSelectedSlot()));
+        var dataSyncEvent = net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.OnDatapackSyncEvent(this, player));
         RecipeManager recipemanager = this.server.getRecipeManager();
         servergamepacketlistenerimpl.send(
             new ClientboundUpdateRecipesPacket(recipemanager.getSynchronizedItemProperties(), recipemanager.getSynchronizedStonecutterRecipes())
         );
+        net.neoforged.neoforge.common.CommonHooks.sendRecipes(player, dataSyncEvent.getRecipeTypesToSend(), recipemanager.recipeMap());
         this.sendPlayerPermissionLevel(player);
         player.getStats().markAllDirty();
         player.getRecipeBook().sendInitialRecipeBook(player);
@@ -218,6 +_,9 @@
         player.initInventoryMenu();
         this.server.notificationManager().playerJoined(player);
         servergamepacketlistenerimpl.resumeFlushing();
+
+        net.neoforged.neoforge.attachment.AttachmentSync.syncInitialPlayerAttachments(player);
+        net.neoforged.neoforge.event.EventHooks.firePlayerLoggedIn(player);
     }
 
     protected void updateEntireScoreboard(ServerScoreboard scoreboard, ServerPlayer player) {
@@ -287,6 +_,7 @@
     }
 
     protected void save(ServerPlayer player) {
+        if (player.connection == null) return;
         this.playerIo.save(player);
         ServerStatsCounter serverstatscounter = this.stats.get(player.getUUID());
         if (serverstatscounter != null) {
@@ -300,6 +_,7 @@
     }
 
     public void remove(ServerPlayer player) {
+        net.neoforged.neoforge.event.EventHooks.firePlayerLoggedOut(player);
         ServerLevel serverlevel = player.level();
         player.awardStat(Stats.LEAVE_GAME);
         this.save(player);
@@ -389,13 +_,20 @@
         TeleportTransition teleporttransition = serverPlayer.findRespawnPositionAndUseSpawnBlock(!keepAllPlayerData, TeleportTransition.DO_NOTHING);
         this.players.remove(serverPlayer);
         serverPlayer.level().removePlayerImmediately(serverPlayer, removalReason);
+
+        // Neo: Allow changing the respawn position of players. The local dimension transition is updated with the new target.
+        var event = net.neoforged.neoforge.event.EventHooks.firePlayerRespawnPositionEvent(serverPlayer, teleporttransition, keepAllPlayerData);
+        teleporttransition = event.getTeleportTransition();
+
         ServerLevel serverlevel = teleporttransition.newLevel();
         ServerPlayer serverplayer = new ServerPlayer(this.server, serverlevel, serverPlayer.getGameProfile(), serverPlayer.clientInformation());
         serverplayer.connection = serverPlayer.connection;
         serverplayer.restoreFrom(serverPlayer, keepAllPlayerData);
         serverplayer.setId(serverPlayer.getId());
         serverplayer.setMainArm(serverPlayer.getMainArm());
-        if (!teleporttransition.missingRespawnBlock()) {
+
+        // Neo: Allow the event to control if the original spawn position is copied
+        if (event.copyOriginalSpawnPosition()) {
             serverplayer.copyRespawnPosition(serverPlayer);
         }
 
@@ -426,6 +_,8 @@
         this.playersByUUID.put(serverplayer.getUUID(), serverplayer);
         serverplayer.initInventoryMenu();
         serverplayer.setHealth(serverplayer.getHealth());
+        net.neoforged.neoforge.attachment.AttachmentSync.syncInitialPlayerAttachments(serverplayer);
+        net.neoforged.neoforge.event.EventHooks.firePlayerRespawnEvent(serverplayer, keepAllPlayerData);
         ServerPlayer.RespawnConfig serverplayer$respawnconfig = serverplayer.getRespawnConfig();
         if (!keepAllPlayerData && serverplayer$respawnconfig != null) {
             LevelData.RespawnData leveldata$respawndata = serverplayer$respawnconfig.respawnData();
@@ -539,6 +_,7 @@
     }
 
     public void op(NameAndId nameAndId, Optional<LevelBasedPermissionSet> permissions, Optional<Boolean> canBypassPlayerLimit) {
+        if (net.neoforged.neoforge.event.EventHooks.onPermissionChanged(nameAndId, this.server.operatorUserPermissions(), this)) return;
         this.ops
             .add(
                 new ServerOpListEntry(
@@ -552,6 +_,7 @@
     }
 
     public void deop(NameAndId nameAndId) {
+        if (net.neoforged.neoforge.event.EventHooks.onPermissionChanged(nameAndId, LevelBasedPermissionSet.ALL, this)) return;
         if (this.ops.remove(nameAndId)) {
             ServerPlayer serverplayer = this.getPlayer(nameAndId.id());
             if (serverplayer != null) {
@@ -642,8 +_,12 @@
     public void sendLevelInfo(ServerPlayer player, ServerLevel level) {
         WorldBorder worldborder = level.getWorldBorder();
         player.connection.send(new ClientboundInitializeBorderPacket(worldborder));
+        if (player.connection.hasChannel(net.neoforged.neoforge.network.payload.ClientboundCustomSetTimePayload.TYPE)) {
+            player.connection.send(new net.neoforged.neoforge.network.payload.ClientboundCustomSetTimePayload(level.getGameTime(), level.getDayTime(), level.getGameRules().get(GameRules.ADVANCE_TIME), level.getDayTimeFraction(), level.getDayTimePerTick()));
+        } else {
         player.connection
             .send(new ClientboundSetTimePacket(level.getGameTime(), level.getDayTime(), level.getGameRules().get(GameRules.ADVANCE_TIME)));
+        }
         player.connection.send(new ClientboundSetDefaultSpawnPositionPacket(level.getRespawnData()));
         if (level.isRaining()) {
             player.connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.START_RAINING, 0.0F));
@@ -653,6 +_,7 @@
 
         player.connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.LEVEL_CHUNKS_LOAD_START, 0.0F));
         this.server.tickRateManager().updateJoiningPlayer(player);
+        net.neoforged.neoforge.attachment.AttachmentSync.syncInitialLevelAttachments(level, player);
     }
 
     public void sendAllPlayerInfo(ServerPlayer player) {
@@ -797,6 +_,8 @@
             this.advancements.put(uuid, playeradvancements);
         }
 
+        // Forge: don't overwrite active player with a fake one.
+        if (!(player instanceof net.neoforged.neoforge.common.util.FakePlayer))
         playeradvancements.setPlayer(player);
         return playeradvancements;
     }
@@ -820,7 +_,7 @@
     }
 
     public List<ServerPlayer> getPlayers() {
-        return this.players;
+        return this.playersView; // Neo: Return an unmodifiable view, we don't want people removing things without us knowing.
     }
 
     public @Nullable ServerPlayer getPlayer(UUID uuid) {
@@ -846,6 +_,7 @@
             playeradvancements.reload(this.server.getAdvancements());
         }
 
+        var dataSyncEvent = net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.OnDatapackSyncEvent(this, null));
         this.broadcastAll(new ClientboundUpdateTagsPacket(TagNetworkSerialization.serializeTagsToNetwork(this.registries)));
         RecipeManager recipemanager = this.server.getRecipeManager();
         ClientboundUpdateRecipesPacket clientboundupdaterecipespacket = new ClientboundUpdateRecipesPacket(
@@ -855,10 +_,15 @@
         for (ServerPlayer serverplayer : this.players) {
             serverplayer.connection.send(clientboundupdaterecipespacket);
             serverplayer.getRecipeBook().sendInitialRecipeBook(serverplayer);
+            net.neoforged.neoforge.common.CommonHooks.sendRecipes(serverplayer, dataSyncEvent.getRecipeTypesToSend(), recipemanager.recipeMap());
         }
     }
 
     public boolean isAllowCommandsForAllPlayers() {
         return this.allowCommandsForAllPlayers;
+    }
+
+    public PlayerDataStorage getPlayerIo() {
+        return this.playerIo;
     }
 }
