--- a/net/minecraft/server/players/SleepStatus.java
+++ b/net/minecraft/server/players/SleepStatus.java
@@ -36,11 +_,11 @@
         this.activePlayers = 0;
         this.sleepingPlayers = 0;
 
-        for(ServerPlayer serverplayer : p_144008_) {
+        for (ServerPlayer serverplayer : p_144008_) {
             if (!serverplayer.isSpectator()) {
-                ++this.activePlayers;
+                this.activePlayers++;
                 if (serverplayer.isSleeping()) {
-                    ++this.sleepingPlayers;
+                    this.sleepingPlayers++;
                 }
             }
         }
