--- a/net/minecraft/server/players/StoredUserList.java
+++ b/net/minecraft/server/players/StoredUserList.java
@@ -85,13 +_,13 @@
     private void removeExpired() {
         List<K> list = Lists.newArrayList();
 
-        for(V v : this.map.values()) {
+        for (V v : this.map.values()) {
             if (v.hasExpired()) {
                 list.add(v.getUser());
             }
         }
 
-        for(K k : list) {
+        for (K k : list) {
             this.map.remove(this.getKeyForUser(k));
         }
     }
@@ -120,7 +_,7 @@
                     return;
                 }
 
-                for(JsonElement jsonelement : jsonarray) {
+                for (JsonElement jsonelement : jsonarray) {
                     JsonObject jsonobject = GsonHelper.convertToJsonObject(jsonelement, "entry");
                     StoredUserEntry<K> storeduserentry = this.createEntry(jsonobject);
                     if (storeduserentry.getUser() != null) {
