--- a/net/minecraft/server/players/UserBanList.java
+++ b/net/minecraft/server/players/UserBanList.java
@@ -21,12 +_,7 @@
 
     @Override
     public String[] getUserList() {
-        return this.getEntries()
-            .stream()
-            .map(StoredUserEntry::getUser)
-            .filter(Objects::nonNull)
-            .map(GameProfile::getName)
-            .toArray(p_144013_ -> new String[p_144013_]);
+        return this.getEntries().stream().map(StoredUserEntry::getUser).filter(Objects::nonNull).map(GameProfile::getName).toArray(String[]::new);
     }
 
     protected String getKeyForUser(GameProfile p_11411_) {
