--- a/net/minecraft/commands/functions/CommandFunction.java
+++ b/net/minecraft/commands/functions/CommandFunction.java
@@ -31,7 +_,7 @@
     ) {
         FunctionBuilder<T> functionbuilder = new FunctionBuilder<>();
 
-        for(int i = 0; i < p_306307_.size(); ++i) {
+        for (int i = 0; i < p_306307_.size(); i++) {
             int j = i + 1;
             String s = p_306307_.get(i).trim();
             String s1;
@@ -47,7 +_,7 @@
                     String s2 = p_306307_.get(i).trim();
                     stringbuilder.append(s2);
                     checkCommandLineLength(stringbuilder);
-                } while(shouldConcatenateNextLine(stringbuilder));
+                } while (shouldConcatenateNextLine(stringbuilder));
 
                 s1 = stringbuilder.toString();
             } else {
