--- a/net/minecraft/commands/functions/MacroFunction.java
+++ b/net/minecraft/commands/functions/MacroFunction.java
@@ -55,7 +_,7 @@
         } else {
             List<String> list = new ArrayList<>(this.parameters.size());
 
-            for(String s : this.parameters) {
+            for (String s : this.parameters) {
                 Tag tag = p_305810_.get(s);
                 if (tag == null) {
                     throw new FunctionInstantiationException(
@@ -104,7 +_,7 @@
         List<UnboundEntryAction<T>> list = new ArrayList<>(this.entries.size());
         List<String> list1 = new ArrayList<>(p_306024_.size());
 
-        for(MacroFunction.Entry<T> entry : this.entries) {
+        for (MacroFunction.Entry<T> entry : this.entries) {
             lookupValues(p_306024_, entry.parameters(), list1);
             list.add(entry.instantiate(list1, p_306281_, this.id));
         }
