--- a/net/minecraft/world/level/chunk/status/ChunkStatus.java
+++ b/net/minecraft/world/level/chunk/status/ChunkStatus.java
@@ -100,9 +_,9 @@
     private static final IntList RANGE_BY_STATUS = Util.make(new IntArrayList(getStatusList().size()), p_331236_ -> {
         int i = 0;
 
-        for(int j = getStatusList().size() - 1; j >= 0; --j) {
-            while(i + 1 < STATUS_BY_RANGE.size() && j <= STATUS_BY_RANGE.get(i + 1).getIndex()) {
-                ++i;
+        for (int j = getStatusList().size() - 1; j >= 0; j--) {
+            while (i + 1 < STATUS_BY_RANGE.size() && j <= STATUS_BY_RANGE.get(i + 1).getIndex()) {
+                i++;
             }
 
             p_331236_.add(0, i);
@@ -136,7 +_,7 @@
         List<ChunkStatus> list = Lists.newArrayList();
 
         ChunkStatus chunkstatus;
-        for(chunkstatus = FULL; chunkstatus.getParent() != chunkstatus; chunkstatus = chunkstatus.getParent()) {
+        for (chunkstatus = FULL; chunkstatus.getParent() != chunkstatus; chunkstatus = chunkstatus.getParent()) {
             list.add(chunkstatus);
         }
 
@@ -200,7 +_,7 @@
                 profiledduration.finish();
             }
 
-            return p_331239_;
+            return (ChunkAccess)p_331239_;
         });
     }
 
