--- a/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
@@ -104,7 +_,7 @@
 
     protected void pageBack() {
         if (this.currentPage > 0) {
-            --this.currentPage;
+            this.currentPage--;
         }
 
         this.updateButtonVisibility();
@@ -112,7 +_,7 @@
 
     protected void pageForward() {
         if (this.currentPage < this.getNumPages() - 1) {
-            ++this.currentPage;
+            this.currentPage++;
         }
 
         this.updateButtonVisibility();
@@ -128,7 +_,7 @@
         if (super.keyPressed(p_98278_, p_98279_, p_98280_)) {
             return true;
         } else {
-            switch(p_98278_) {
+            switch (p_98278_) {
                 case 266:
                     this.backButton.onPress();
                     return true;
@@ -157,7 +_,7 @@
         p_281997_.drawString(this.font, this.pageMsg, i - i1 + 192 - 44, 18, 0, false);
         int k = Math.min(128 / 9, this.cachedPageComponents.size());
 
-        for(int l = 0; l < k; ++l) {
+        for (int l = 0; l < k; l++) {
             FormattedCharSequence formattedcharsequence = this.cachedPageComponents.get(l);
             p_281997_.drawString(this.font, formattedcharsequence, i + 36, 32 + l * 9, 0, false);
         }
