--- a/net/minecraft/client/gui/screens/inventory/BookEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/BookEditScreen.java
@@ -114,7 +_,7 @@
     @Override
     public void tick() {
         super.tick();
-        ++this.frameTick;
+        this.frameTick++;
     }
 
     @Override
@@ -150,7 +_,7 @@
 
     private void pageBack() {
         if (this.currentPage > 0) {
-            --this.currentPage;
+            this.currentPage--;
         }
 
         this.updateButtonVisibility();
@@ -159,11 +_,11 @@
 
     private void pageForward() {
         if (this.currentPage < this.getNumPages() - 1) {
-            ++this.currentPage;
+            this.currentPage++;
         } else {
             this.appendPageToBook();
             if (this.currentPage < this.getNumPages() - 1) {
-                ++this.currentPage;
+                this.currentPage++;
             }
         }
 
@@ -184,7 +_,7 @@
     private void eraseEmptyTrailingPages() {
         ListIterator<String> listiterator = this.pages.listIterator(this.pages.size());
 
-        while(listiterator.hasPrevious() && listiterator.previous().isEmpty()) {
+        while (listiterator.hasPrevious() && listiterator.previous().isEmpty()) {
             listiterator.remove();
         }
     }
@@ -265,7 +_,7 @@
             TextFieldHelper.CursorStep textfieldhelper$cursorstep = Screen.hasControlDown()
                 ? TextFieldHelper.CursorStep.WORD
                 : TextFieldHelper.CursorStep.CHARACTER;
-            switch(p_98153_) {
+            switch (p_98153_) {
                 case 257:
                 case 335:
                     this.pageEdit.insertText("\n");
@@ -342,7 +_,7 @@
     }
 
     private boolean titleKeyPressed(int p_98164_, int p_98165_, int p_98166_) {
-        switch(p_98164_) {
+        switch (p_98164_) {
             case 257:
             case 335:
                 if (!this.title.isEmpty()) {
@@ -396,7 +_,7 @@
             p_281724_.drawString(this.font, this.pageMsg, i - j1 + 192 - 44, 18, 0, false);
             BookEditScreen.DisplayCache bookeditscreen$displaycache = this.getDisplayCache();
 
-            for(BookEditScreen.LineInfo bookeditscreen$lineinfo : bookeditscreen$displaycache.lines) {
+            for (BookEditScreen.LineInfo bookeditscreen$lineinfo : bookeditscreen$displaycache.lines) {
                 p_281724_.drawString(this.font, bookeditscreen$lineinfo.asComponent, bookeditscreen$lineinfo.x, bookeditscreen$lineinfo.y, -16777216, false);
             }
 
@@ -423,7 +_,7 @@
     }
 
     private void renderHighlight(GuiGraphics p_282188_, Rect2i[] p_265482_) {
-        for(Rect2i rect2i : p_265482_) {
+        for (Rect2i rect2i : p_265482_) {
             int i = rect2i.getX();
             int j = rect2i.getY();
             int k = i + rect2i.getWidth();
@@ -559,7 +_,7 @@
                     int i3 = j1 + 1 > aint.length ? s.length() : aint[j1 + 1];
                     list1.add(this.createPartialLineSelection(s, stringsplitter, l2, i3, j1 * 9, aint[j1]));
 
-                    for(int j3 = j1 + 1; j3 < k1; ++j3) {
+                    for (int j3 = j1 + 1; j3 < k1; j3++) {
                         int j2 = j3 * 9;
                         String s1 = s.substring(aint[j3], aint[j3 + 1]);
                         int k2 = (int)stringsplitter.stringWidth(s1);
