--- a/net/minecraft/client/gui/screens/inventory/CrafterScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CrafterScreen.java
@@ -37,7 +_,7 @@
     @Override
     protected void slotClicked(Slot p_307465_, int p_307203_, int p_307325_, ClickType p_307680_) {
         if (p_307465_ instanceof CrafterSlot && !p_307465_.hasItem() && !this.player.isSpectator()) {
-            switch(p_307680_) {
+            switch (p_307680_) {
                 case PICKUP:
                     if (this.menu.isSlotDisabled(p_307203_)) {
                         this.enableSlot(p_307203_);
