--- a/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java
@@ -115,6 +_,8 @@
     private final Set<TagKey<Item>> visibleTags = new HashSet<>();
     private final boolean displayOperatorCreativeTab;
     private final EffectsInInventory effects;
+    private final List<net.neoforged.neoforge.client.gui.CreativeTabsScreenPage> pages = new java.util.ArrayList<>();
+    private net.neoforged.neoforge.client.gui.CreativeTabsScreenPage currentPage = new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(new java.util.ArrayList<>());
 
     public CreativeModeInventoryScreen(LocalPlayer player, FeatureFlagSet enabledFeatures, boolean displayOperatorCreativeTab) {
         super(new CreativeModeInventoryScreen.ItemPickerMenu(player), player.getInventory(), CommonComponents.EMPTY);
@@ -151,9 +_,11 @@
             return false;
         } else {
             if (searchTrees != null) {
-                List<ItemStack> list = List.copyOf(CreativeModeTabs.searchTab().getDisplayItems());
-                searchTrees.updateCreativeTooltips(holders, list);
-                searchTrees.updateCreativeTags(list);
+                CreativeModeTabs.allTabs().stream().filter(net.minecraft.world.item.CreativeModeTab::hasSearchBar).forEach(tab -> {
+                    List<ItemStack> list = List.copyOf(tab.getDisplayItems());
+                    searchTrees.updateCreativeTooltips(holders, list, net.neoforged.neoforge.client.CreativeModeTabSearchRegistry.getNameSearchKey(tab));
+                    searchTrees.updateCreativeTags(list, net.neoforged.neoforge.client.CreativeModeTabSearchRegistry.getTagSearchKey(tab));
+                });
             }
 
             return true;
@@ -163,7 +_,7 @@
     private void refreshCurrentTabContents(Collection<ItemStack> displayList) {
         int i = this.menu.getRowIndexForScroll(this.scrollOffs);
         this.menu.items.clear();
-        if (selectedTab.getType() == CreativeModeTab.Type.SEARCH) {
+        if (selectedTab.hasSearchBar()) {
             this.refreshSearchResults();
         } else {
             this.menu.items.addAll(displayList);
@@ -331,6 +_,34 @@
     protected void init() {
         if (this.minecraft.player.hasInfiniteMaterials()) {
             super.init();
+            this.pages.clear();
+            int tabIndex = 0;
+            List<CreativeModeTab> currentPage = new java.util.ArrayList<>();
+            for (CreativeModeTab sortedCreativeModeTab : net.neoforged.neoforge.common.CreativeModeTabRegistry.getSortedCreativeModeTabs().stream().filter(CreativeModeTab::hasAnyItems).toList()) {
+                currentPage.add(sortedCreativeModeTab);
+                tabIndex++;
+                if (tabIndex == 10) {
+                    this.pages.add(new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(currentPage));
+                    currentPage = new java.util.ArrayList<>();
+                    tabIndex = 0;
+                }
+            }
+            if (tabIndex != 0) {
+                this.pages.add(new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(currentPage));
+            }
+            if (this.pages.isEmpty()) {
+                this.currentPage = new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(new java.util.ArrayList<>());
+            } else {
+                this.currentPage = this.pages.get(0);
+            }
+            if (this.pages.size() > 1) {
+                addRenderableWidget(net.minecraft.client.gui.components.Button.builder(Component.literal("<"), b -> setCurrentPage(this.pages.get(Math.max(this.pages.indexOf(this.currentPage) - 1, 0)))).pos(leftPos,  topPos - 50).size(20, 20).build());
+                addRenderableWidget(net.minecraft.client.gui.components.Button.builder(Component.literal(">"), b -> setCurrentPage(this.pages.get(Math.min(this.pages.indexOf(this.currentPage) + 1, this.pages.size() - 1)))).pos(leftPos + imageWidth - 20, topPos - 50).size(20, 20).build());
+            }
+            this.currentPage = this.pages.stream().filter(page -> page.getVisibleTabs().contains(selectedTab)).findFirst().orElse(this.currentPage);
+            if (!this.currentPage.getVisibleTabs().contains(selectedTab)) {
+                selectedTab = this.currentPage.getVisibleTabs().get(0);
+            }
             this.searchBox = new EditBox(this.font, this.leftPos + 82, this.topPos + 6, 80, 9, Component.translatable("itemGroup.search"));
             this.searchBox.setMaxLength(50);
             this.searchBox.setBordered(false);
@@ -378,7 +_,7 @@
     public boolean charTyped(CharacterEvent event) {
         if (this.ignoreTextInput) {
             return false;
-        } else if (selectedTab.getType() != CreativeModeTab.Type.SEARCH) {
+        } else if (!selectedTab.hasSearchBar()) {
             return false;
         } else {
             String s = this.searchBox.getValue();
@@ -397,7 +_,7 @@
     @Override
     public boolean keyPressed(KeyEvent event) {
         this.ignoreTextInput = false;
-        if (selectedTab.getType() != CreativeModeTab.Type.SEARCH) {
+        if (!selectedTab.hasSearchBar()) {
             if (this.minecraft.options.keyChat.matches(event)) {
                 this.ignoreTextInput = true;
                 this.selectTab(CreativeModeTabs.searchTab());
@@ -433,6 +_,7 @@
     }
 
     private void refreshSearchResults() {
+        if (!selectedTab.hasSearchBar()) return;
         this.menu.items.clear();
         this.visibleTags.clear();
         String s = this.searchBox.getValue();
@@ -445,10 +_,10 @@
                 SearchTree<ItemStack> searchtree;
                 if (s.startsWith("#")) {
                     s = s.substring(1);
-                    searchtree = sessionsearchtrees.creativeTagSearch();
+                    searchtree = sessionsearchtrees.creativeTagSearch(net.neoforged.neoforge.client.CreativeModeTabSearchRegistry.getTagSearchKey(selectedTab));
                     this.updateVisibleTags(s);
                 } else {
-                    searchtree = sessionsearchtrees.creativeNameSearch();
+                    searchtree = sessionsearchtrees.creativeNameSearch(net.neoforged.neoforge.client.CreativeModeTabSearchRegistry.getNameSearchKey(selectedTab));
                 }
 
                 this.menu.items.addAll(searchtree.search(s.toLowerCase(Locale.ROOT)));
@@ -476,7 +_,7 @@
     @Override
     protected void renderLabels(GuiGraphics graphics, int xm, int ym) {
         if (selectedTab.showTitle()) {
-            graphics.drawString(this.font, selectedTab.getDisplayName(), 8, 6, -12566464, false);
+            graphics.drawString(this.font, selectedTab.getDisplayName(), 8, 6, selectedTab.getLabelColor(), false);
         }
     }
 
@@ -486,7 +_,7 @@
             double d0 = event.x() - this.leftPos;
             double d1 = event.y() - this.topPos;
 
-            for (CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+            for (CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
                 if (this.checkTabClicked(creativemodetab, d0, d1)) {
                     return true;
                 }
@@ -508,7 +_,7 @@
             double d1 = event.y() - this.topPos;
             this.scrolling = false;
 
-            for (CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+            for(CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
                 if (this.checkTabClicked(creativemodetab, d0, d1)) {
                     this.selectTab(creativemodetab);
                     return true;
@@ -526,6 +_,7 @@
     private void selectTab(CreativeModeTab tab) {
         CreativeModeTab creativemodetab = selectedTab;
         selectedTab = tab;
+        slotColor = tab.getSlotColor();
         this.quickCraftSlots.clear();
         this.menu.items.clear();
         this.clearDraggingState();
@@ -602,13 +_,15 @@
             this.originalSlots = null;
         }
 
-        if (selectedTab.getType() == CreativeModeTab.Type.SEARCH) {
+        if (selectedTab.hasSearchBar()) {
             this.searchBox.setVisible(true);
             this.searchBox.setCanLoseFocus(false);
             this.searchBox.setFocused(true);
             if (creativemodetab != tab) {
                 this.searchBox.setValue("");
             }
+            this.searchBox.setWidth(selectedTab.getSearchBarWidth());
+            this.searchBox.setX(this.leftPos + (82 /*default left*/ + 89 /*default width*/) - this.searchBox.getWidth());
 
             this.refreshSearchResults();
         } else {
@@ -671,18 +_,24 @@
         this.effects.render(graphics, mouseX, mouseY);
         super.render(graphics, mouseX, mouseY, a);
 
-        for (CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
-            if (this.checkTabHovering(graphics, creativemodetab, mouseX, mouseY)) {
-                break;
-            }
-        }
-
         if (this.destroyItemSlot != null
             && selectedTab.getType() == CreativeModeTab.Type.INVENTORY
             && this.isHovering(this.destroyItemSlot.x, this.destroyItemSlot.y, 16, 16, mouseX, mouseY)) {
             graphics.setTooltipForNextFrame(this.font, TRASH_SLOT_TOOLTIP, mouseX, mouseY);
         }
 
+        if (this.pages.size() != 1) {
+            Component page = Component.literal(String.format("%d / %d", this.pages.indexOf(this.currentPage) + 1, this.pages.size()));
+            // TODO 1.21.6: This previously bumped Z by 300, investigate if this needs nextStratum
+            graphics.drawString(font, page.getVisualOrderText(), leftPos + (imageWidth / 2) - (font.width(page) / 2), topPos - 44, -1);
+        }
+
+        for (CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
+            if (this.checkTabHovering(graphics, creativemodetab, mouseX, mouseY)) {
+                break;
+            }
+        }
+
         this.renderTooltip(graphics, mouseX, mouseY);
     }
 
@@ -695,10 +_,10 @@
     public List<Component> getTooltipFromContainerItem(ItemStack itemStack) {
         boolean flag = this.hoveredSlot != null && this.hoveredSlot instanceof CreativeModeInventoryScreen.CustomCreativeSlot;
         boolean flag1 = selectedTab.getType() == CreativeModeTab.Type.CATEGORY;
-        boolean flag2 = selectedTab.getType() == CreativeModeTab.Type.SEARCH;
+        boolean flag2 = selectedTab.hasSearchBar();
         TooltipFlag.Default tooltipflag$default = this.minecraft.options.advancedItemTooltips ? TooltipFlag.Default.ADVANCED : TooltipFlag.Default.NORMAL;
         TooltipFlag tooltipflag = flag ? tooltipflag$default.asCreative() : tooltipflag$default;
-        List<Component> list = itemStack.getTooltipLines(Item.TooltipContext.of(this.minecraft.level), this.minecraft.player, tooltipflag);
+        List<Component> list = itemStack.getTooltipLines(Item.TooltipContext.of(this.minecraft.level), this.minecraft.player, net.neoforged.neoforge.client.ClientTooltipFlag.of(tooltipflag));
         if (list.isEmpty()) {
             return list;
         } else if (flag1 && flag) {
@@ -716,7 +_,7 @@
             int i = 1;
 
             for (CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
-                if (creativemodetab.getType() != CreativeModeTab.Type.SEARCH && creativemodetab.contains(itemStack)) {
+                if (!creativemodetab.hasSearchBar() && creativemodetab.contains(itemStack)) {
                     list1.add(i++, creativemodetab.getDisplayName().copy().withStyle(ChatFormatting.BLUE));
                 }
             }
@@ -727,7 +_,7 @@
 
     @Override
     protected void renderBg(GuiGraphics graphics, float a, int xm, int ym) {
-        for (CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+        for (CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
             if (creativemodetab != selectedTab) {
                 this.renderTabButton(graphics, xm, ym, creativemodetab);
             }
@@ -754,10 +_,11 @@
         int k = this.topPos + 18;
         int i = k + 112;
         if (selectedTab.canScroll()) {
-            Identifier identifier = this.canScroll() ? SCROLLER_SPRITE : SCROLLER_DISABLED_SPRITE;
+            Identifier identifier = selectedTab.getScrollerSprite(); // this.canScroll() ? SCROLLER_SPRITE : SCROLLER_DISABLED_SPRITE;
             graphics.blitSprite(RenderPipelines.GUI_TEXTURED, identifier, j, k + (int)((i - k - 17) * this.scrollOffs), 12, 15);
         }
 
+        if (currentPage.getVisibleTabs().contains(selectedTab)) //Forge: only display tab selection when the selected tab is on the current page
         this.renderTabButton(graphics, xm, ym, selectedTab);
         if (selectedTab.getType() == CreativeModeTab.Type.INVENTORY) {
             InventoryScreen.renderEntityInInventoryFollowsMouse(
@@ -767,7 +_,7 @@
     }
 
     private int getTabX(CreativeModeTab tab) {
-        int i = tab.column();
+        int i = currentPage.getColumn(tab);
         int j = 27;
         int k = 27 * i;
         if (tab.isAlignedRight()) {
@@ -779,7 +_,7 @@
 
     private int getTabY(CreativeModeTab tab) {
         int i = 0;
-        if (tab.row() == CreativeModeTab.Row.TOP) {
+        if (currentPage.isTop(tab)) {
             i -= 32;
         } else {
             i += this.imageHeight;
@@ -807,8 +_,8 @@
 
     protected void renderTabButton(GuiGraphics graphics, int mouseX, int mouseY, CreativeModeTab tab) {
         boolean flag = tab == selectedTab;
-        boolean flag1 = tab.row() == CreativeModeTab.Row.TOP;
-        int i = tab.column();
+        boolean flag1 = currentPage.isTop(tab);
+        int i = currentPage.getColumn(tab);
         int j = this.leftPos + this.getTabX(tab);
         int k = this.topPos - (flag1 ? 28 : -(this.imageHeight - 4));
         Identifier[] aidentifier;
@@ -822,6 +_,7 @@
             graphics.requestCursor(CursorTypes.POINTING_HAND);
         }
 
+        // TODO 1.20.2: Deal with custom tab backgrounds, and deal with transparency.
         graphics.blitSprite(RenderPipelines.GUI_TEXTURED, aidentifier[Mth.clamp(i, 0, aidentifier.length)], j, k, 26, 32);
         int l = j + 13 - 8;
         int i1 = k + 16 - 8 + (flag1 ? 1 : -1);
@@ -858,6 +_,14 @@
         }
     }
 
+    public net.neoforged.neoforge.client.gui.CreativeTabsScreenPage getCurrentPage() {
+        return currentPage;
+    }
+
+    public void setCurrentPage(net.neoforged.neoforge.client.gui.CreativeTabsScreenPage currentPage) {
+        this.currentPage = currentPage;
+    }
+
     @OnlyIn(Dist.CLIENT)
     static class CustomCreativeSlot extends Slot {
         public CustomCreativeSlot(Container container, int slot, int x, int y) {
@@ -1038,6 +_,22 @@
         @Override
         public boolean mayPickup(Player player) {
             return this.target.mayPickup(player);
+        }
+
+        @Override
+        public int getSlotIndex() {
+            return this.target.getSlotIndex();
+        }
+
+        @Override
+        public boolean isSameInventory(Slot other) {
+            return this.target.isSameInventory(other);
+        }
+
+        @Override
+        public Slot setBackground(Identifier sprite) {
+            this.target.setBackground(sprite);
+            return this;
         }
     }
 }
