--- a/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java
@@ -60,8 +_,7 @@
         new ResourceLocation("container/creative_inventory/tab_top_unselected_4"),
         new ResourceLocation("container/creative_inventory/tab_top_unselected_5"),
         new ResourceLocation("container/creative_inventory/tab_top_unselected_6"),
-        new ResourceLocation("container/creative_inventory/tab_top_unselected_7"),
-        new ResourceLocation("container/creative_inventory/tab_top_unselected_8")
+        new ResourceLocation("container/creative_inventory/tab_top_unselected_7")
     };
     private static final ResourceLocation[] SELECTED_TOP_TABS = new ResourceLocation[]{
         new ResourceLocation("container/creative_inventory/tab_top_selected_1"),
@@ -70,8 +_,7 @@
         new ResourceLocation("container/creative_inventory/tab_top_selected_4"),
         new ResourceLocation("container/creative_inventory/tab_top_selected_5"),
         new ResourceLocation("container/creative_inventory/tab_top_selected_6"),
-        new ResourceLocation("container/creative_inventory/tab_top_selected_7"),
-        new ResourceLocation("container/creative_inventory/tab_top_selected_8")
+        new ResourceLocation("container/creative_inventory/tab_top_selected_7")
     };
     private static final ResourceLocation[] UNSELECTED_BOTTOM_TABS = new ResourceLocation[]{
         new ResourceLocation("container/creative_inventory/tab_bottom_unselected_1"),
@@ -114,6 +_,8 @@
     private boolean hasClickedOutside;
     private final Set<TagKey<Item>> visibleTags = new HashSet<>();
     private final boolean displayOperatorCreativeTab;
+    private final List<net.neoforged.neoforge.client.gui.CreativeTabsScreenPage> pages = new java.util.ArrayList<>();
+    private net.neoforged.neoforge.client.gui.CreativeTabsScreenPage currentPage = new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(new java.util.ArrayList<>());
 
     public CreativeModeInventoryScreen(Player p_259788_, FeatureFlagSet p_260074_, boolean p_259569_) {
         super(new CreativeModeInventoryScreen.ItemPickerMenu(p_259788_), p_259788_.getInventory(), CommonComponents.EMPTY);
@@ -130,7 +_,7 @@
 
     private void tryRefreshInvalidatedTabs(FeatureFlagSet p_259501_, boolean p_259713_, HolderLookup.Provider p_270898_) {
         if (CreativeModeTabs.tryRebuildTabContents(p_259501_, p_259713_, p_270898_)) {
-            for(CreativeModeTab creativemodetab : CreativeModeTabs.allTabs()) {
+            for (CreativeModeTab creativemodetab : CreativeModeTabs.allTabs()) {
                 Collection<ItemStack> collection = creativemodetab.getDisplayItems();
                 if (creativemodetab == selectedTab) {
                     if (creativemodetab.getType() == CreativeModeTab.Type.CATEGORY && collection.isEmpty()) {
@@ -146,7 +_,7 @@
     private void refreshCurrentTabContents(Collection<ItemStack> p_261591_) {
         int i = this.menu.getRowIndexForScroll(this.scrollOffs);
         this.menu.items.clear();
-        if (selectedTab.getType() == CreativeModeTab.Type.SEARCH) {
+        if (selectedTab.hasSearchBar()) {
             this.refreshSearchResults();
         } else {
             this.menu.items.addAll(p_261591_);
@@ -203,7 +_,7 @@
             }
 
             if (p_98556_ == this.destroyItemSlot && flag) {
-                for(int j = 0; j < this.minecraft.player.inventoryMenu.getItems().size(); ++j) {
+                for (int j = 0; j < this.minecraft.player.inventoryMenu.getItems().size(); j++) {
                     this.minecraft.gameMode.handleCreativeModeItemAdd(ItemStack.EMPTY, j);
                 }
             } else if (selectedTab.getType() == CreativeModeTab.Type.INVENTORY) {
@@ -284,7 +_,7 @@
                 ItemStack itemstack3 = p_98556_ == null ? ItemStack.EMPTY : this.menu.getSlot(p_98556_.index).getItem();
                 this.menu.clicked(p_98556_ == null ? p_98557_ : p_98556_.index, p_98558_, p_98559_, this.minecraft.player);
                 if (AbstractContainerMenu.getQuickcraftHeader(p_98558_) == 2) {
-                    for(int k = 0; k < 9; ++k) {
+                    for (int k = 0; k < 9; k++) {
                         this.minecraft.gameMode.handleCreativeModeItemAdd(this.menu.getSlot(45 + k).getItem(), 36 + k);
                     }
                 } else if (p_98556_ != null) {
@@ -313,6 +_,34 @@
     protected void init() {
         if (this.minecraft.gameMode.hasInfiniteItems()) {
             super.init();
+            this.pages.clear();
+            int tabIndex = 0;
+            List<CreativeModeTab> currentPage = new java.util.ArrayList<>();
+            for (CreativeModeTab sortedCreativeModeTab : net.neoforged.neoforge.common.CreativeModeTabRegistry.getSortedCreativeModeTabs()) {
+                currentPage.add(sortedCreativeModeTab);
+                tabIndex++;
+                if (tabIndex == 10) {
+                    this.pages.add(new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(currentPage));
+                    currentPage = new java.util.ArrayList<>();
+                    tabIndex = 0;
+                }
+            }
+            if (tabIndex != 0) {
+                this.pages.add(new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(currentPage));
+            }
+            if (this.pages.isEmpty()) {
+                this.currentPage = new net.neoforged.neoforge.client.gui.CreativeTabsScreenPage(new java.util.ArrayList<>());
+            } else {
+                this.currentPage = this.pages.get(0);
+            }
+            if (this.pages.size() > 1) {
+                addRenderableWidget(net.minecraft.client.gui.components.Button.builder(Component.literal("<"), b -> setCurrentPage(this.pages.get(Math.max(this.pages.indexOf(this.currentPage) - 1, 0)))).pos(leftPos,  topPos - 50).size(20, 20).build());
+                addRenderableWidget(net.minecraft.client.gui.components.Button.builder(Component.literal(">"), b -> setCurrentPage(this.pages.get(Math.min(this.pages.indexOf(this.currentPage) + 1, this.pages.size() - 1)))).pos(leftPos + imageWidth - 20, topPos - 50).size(20, 20).build());
+            }
+            this.currentPage = this.pages.stream().filter(page -> page.getVisibleTabs().contains(selectedTab)).findFirst().orElse(this.currentPage);
+            if (!this.currentPage.getVisibleTabs().contains(selectedTab)) {
+                selectedTab = this.currentPage.getVisibleTabs().get(0);
+            }
             this.searchBox = new EditBox(this.font, this.leftPos + 82, this.topPos + 6, 80, 9, Component.translatable("itemGroup.search"));
             this.searchBox.setMaxLength(50);
             this.searchBox.setBordered(false);
@@ -359,7 +_,7 @@
     public boolean charTyped(char p_98521_, int p_98522_) {
         if (this.ignoreTextInput) {
             return false;
-        } else if (selectedTab.getType() != CreativeModeTab.Type.SEARCH) {
+        } else if (!selectedTab.hasSearchBar()) {
             return false;
         } else {
             String s = this.searchBox.getValue();
@@ -378,7 +_,7 @@
     @Override
     public boolean keyPressed(int p_98547_, int p_98548_, int p_98549_) {
         this.ignoreTextInput = false;
-        if (selectedTab.getType() != CreativeModeTab.Type.SEARCH) {
+        if (!selectedTab.hasSearchBar()) {
             if (this.minecraft.options.keyChat.matches(p_98547_, p_98548_)) {
                 this.ignoreTextInput = true;
                 this.selectTab(CreativeModeTabs.searchTab());
@@ -414,6 +_,7 @@
     }
 
     private void refreshSearchResults() {
+        if (!selectedTab.hasSearchBar()) return;
         this.menu.items.clear();
         this.visibleTags.clear();
         String s = this.searchBox.getValue();
@@ -423,10 +_,10 @@
             SearchTree<ItemStack> searchtree;
             if (s.startsWith("#")) {
                 s = s.substring(1);
-                searchtree = this.minecraft.getSearchTree(SearchRegistry.CREATIVE_TAGS);
+                searchtree = this.minecraft.getSearchTree(net.neoforged.neoforge.client.CreativeModeTabSearchRegistry.getTagSearchKey(selectedTab));
                 this.updateVisibleTags(s);
             } else {
-                searchtree = this.minecraft.getSearchTree(SearchRegistry.CREATIVE_NAMES);
+                searchtree = this.minecraft.getSearchTree(net.neoforged.neoforge.client.CreativeModeTabSearchRegistry.getNameSearchKey(selectedTab));
             }
 
             this.menu.items.addAll(searchtree.search(s.toLowerCase(Locale.ROOT)));
@@ -453,7 +_,8 @@
     @Override
     protected void renderLabels(GuiGraphics p_283168_, int p_281774_, int p_281466_) {
         if (selectedTab.showTitle()) {
-            p_283168_.drawString(this.font, selectedTab.getDisplayName(), 8, 6, 4210752, false);
+            com.mojang.blaze3d.systems.RenderSystem.disableBlend();
+            p_283168_.drawString(this.font, selectedTab.getDisplayName(), 8, 6, selectedTab.getLabelColor(), false);
         }
     }
 
@@ -463,7 +_,7 @@
             double d0 = p_98531_ - (double)this.leftPos;
             double d1 = p_98532_ - (double)this.topPos;
 
-            for(CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+            for(CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
                 if (this.checkTabClicked(creativemodetab, d0, d1)) {
                     return true;
                 }
@@ -485,7 +_,7 @@
             double d1 = p_98623_ - (double)this.topPos;
             this.scrolling = false;
 
-            for(CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+            for(CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
                 if (this.checkTabClicked(creativemodetab, d0, d1)) {
                     this.selectTab(creativemodetab);
                     return true;
@@ -503,22 +_,23 @@
     private void selectTab(CreativeModeTab p_98561_) {
         CreativeModeTab creativemodetab = selectedTab;
         selectedTab = p_98561_;
+        slotColor = p_98561_.getSlotColor();
         this.quickCraftSlots.clear();
         this.menu.items.clear();
         this.clearDraggingState();
         if (selectedTab.getType() == CreativeModeTab.Type.HOTBAR) {
             HotbarManager hotbarmanager = this.minecraft.getHotbarManager();
 
-            for(int i = 0; i < 9; ++i) {
+            for (int i = 0; i < 9; i++) {
                 Hotbar hotbar = hotbarmanager.get(i);
                 if (hotbar.isEmpty()) {
-                    for(int j = 0; j < 9; ++j) {
+                    for (int j = 0; j < 9; j++) {
                         if (j == i) {
                             ItemStack itemstack = new ItemStack(Items.PAPER);
                             itemstack.set(DataComponents.CREATIVE_SLOT_LOCK, Unit.INSTANCE);
                             Component component = this.minecraft.options.keyHotbarSlots[i].getTranslatedKeyMessage();
                             Component component1 = this.minecraft.options.keySaveHotbarActivator.getTranslatedKeyMessage();
-                            itemstack.set(DataComponents.CUSTOM_NAME, Component.translatable("inventory.hotbarInfo", component1, component));
+                            itemstack.set(DataComponents.ITEM_NAME, Component.translatable("inventory.hotbarInfo", component1, component));
                             this.menu.items.add(itemstack);
                         } else {
                             this.menu.items.add(ItemStack.EMPTY);
@@ -540,7 +_,7 @@
 
             this.menu.slots.clear();
 
-            for(int k = 0; k < abstractcontainermenu.slots.size(); ++k) {
+            for (int k = 0; k < abstractcontainermenu.slots.size(); k++) {
                 int l;
                 int i1;
                 if (k >= 5 && k < 9) {
@@ -579,13 +_,15 @@
             this.originalSlots = null;
         }
 
-        if (selectedTab.getType() == CreativeModeTab.Type.SEARCH) {
+        if (selectedTab.hasSearchBar()) {
             this.searchBox.setVisible(true);
             this.searchBox.setCanLoseFocus(false);
             this.searchBox.setFocused(true);
             if (creativemodetab != p_98561_) {
                 this.searchBox.setValue("");
             }
+            this.searchBox.setWidth(selectedTab.getSearchBarWidth());
+            this.searchBox.setX(this.leftPos + (82 /*default left*/ + 89 /*default width*/) - this.searchBox.getWidth());
 
             this.refreshSearchResults();
         } else {
@@ -648,7 +_,7 @@
     public void render(GuiGraphics p_283000_, int p_281317_, int p_282770_, float p_281295_) {
         super.render(p_283000_, p_281317_, p_282770_, p_281295_);
 
-        for(CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+        for(CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
             if (this.checkTabHovering(p_283000_, creativemodetab, p_281317_, p_282770_)) {
                 break;
             }
@@ -660,6 +_,15 @@
             p_283000_.renderTooltip(this.font, TRASH_SLOT_TOOLTIP, p_281317_, p_282770_);
         }
 
+        if (this.pages.size() != 1) {
+            Component page = Component.literal(String.format("%d / %d", this.pages.indexOf(this.currentPage) + 1, this.pages.size()));
+            p_283000_.pose().pushPose();
+            p_283000_.pose().translate(0F, 0F, 300F);
+            p_283000_.drawString(font, page.getVisualOrderText(), leftPos + (imageWidth / 2) - (font.width(page) / 2), topPos - 44, -1);
+            p_283000_.pose().popPose();
+        }
+
+        com.mojang.blaze3d.systems.RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
         this.renderTooltip(p_283000_, p_281317_, p_282770_);
     }
 
@@ -667,26 +_,26 @@
     public List<Component> getTooltipFromContainerItem(ItemStack p_281769_) {
         boolean flag = this.hoveredSlot != null && this.hoveredSlot instanceof CreativeModeInventoryScreen.CustomCreativeSlot;
         boolean flag1 = selectedTab.getType() == CreativeModeTab.Type.CATEGORY;
-        boolean flag2 = selectedTab.getType() == CreativeModeTab.Type.SEARCH;
+        boolean flag2 = selectedTab.hasSearchBar();
         TooltipFlag.Default tooltipflag$default = this.minecraft.options.advancedItemTooltips ? TooltipFlag.Default.ADVANCED : TooltipFlag.Default.NORMAL;
         TooltipFlag tooltipflag = flag ? tooltipflag$default.asCreative() : tooltipflag$default;
-        List<Component> list = p_281769_.getTooltipLines(this.minecraft.player, tooltipflag);
+        List<Component> list = p_281769_.getTooltipLines(Item.TooltipContext.of(this.minecraft.level), this.minecraft.player, tooltipflag);
         if (flag1 && flag) {
             return list;
         } else {
             List<Component> list1 = Lists.newArrayList(list);
             if (flag2 && flag) {
-                this.visibleTags.forEach(p_205407_ -> {
-                    if (p_281769_.is(p_205407_)) {
-                        list1.add(1, Component.literal("#" + p_205407_.location()).withStyle(ChatFormatting.DARK_PURPLE));
+                this.visibleTags.forEach(p_339283_ -> {
+                    if (p_281769_.is((TagKey<Item>)p_339283_)) {
+                        list1.add(1, Component.literal("#" + p_339283_.location()).withStyle(ChatFormatting.DARK_PURPLE));
                     }
                 });
             }
 
             int i = 1;
 
-            for(CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
-                if (creativemodetab.getType() != CreativeModeTab.Type.SEARCH && creativemodetab.contains(p_281769_)) {
+            for (CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+                if (!creativemodetab.hasSearchBar() && creativemodetab.contains(p_281769_)) {
                     list1.add(i++, creativemodetab.getDisplayName().copy().withStyle(ChatFormatting.BLUE));
                 }
             }
@@ -697,14 +_,14 @@
 
     @Override
     protected void renderBg(GuiGraphics p_282663_, float p_282504_, int p_282089_, int p_282249_) {
-        for(CreativeModeTab creativemodetab : CreativeModeTabs.tabs()) {
+        for(CreativeModeTab creativemodetab : currentPage.getVisibleTabs()) {
             if (creativemodetab != selectedTab) {
                 this.renderTabButton(p_282663_, creativemodetab);
             }
         }
 
         p_282663_.blit(
-            new ResourceLocation("textures/gui/container/creative_inventory/tab_" + selectedTab.getBackgroundSuffix()),
+            selectedTab.getBackgroundLocation(),
             this.leftPos,
             this.topPos,
             0,
@@ -717,10 +_,11 @@
         int k = this.topPos + 18;
         int i = k + 112;
         if (selectedTab.canScroll()) {
-            ResourceLocation resourcelocation = this.canScroll() ? SCROLLER_SPRITE : SCROLLER_DISABLED_SPRITE;
+            ResourceLocation resourcelocation = selectedTab.getScrollerSprite(); // this.canScroll() ? SCROLLER_SPRITE : SCROLLER_DISABLED_SPRITE;
             p_282663_.blitSprite(resourcelocation, j, k + (int)((float)(i - k - 17) * this.scrollOffs), 12, 15);
         }
 
+        if (currentPage.getVisibleTabs().contains(selectedTab)) //Forge: only display tab selection when the selected tab is on the current page
         this.renderTabButton(p_282663_, selectedTab);
         if (selectedTab.getType() == CreativeModeTab.Type.INVENTORY) {
             InventoryScreen.renderEntityInInventoryFollowsMouse(
@@ -739,7 +_,7 @@
     }
 
     private int getTabX(CreativeModeTab p_260136_) {
-        int i = p_260136_.column();
+        int i = currentPage.getColumn(p_260136_);
         int j = 27;
         int k = 27 * i;
         if (p_260136_.isAlignedRight()) {
@@ -751,7 +_,7 @@
 
     private int getTabY(CreativeModeTab p_260181_) {
         int i = 0;
-        if (p_260181_.row() == CreativeModeTab.Row.TOP) {
+        if (currentPage.isTop(p_260181_)) {
             i -= 32;
         } else {
             i += this.imageHeight;
@@ -779,8 +_,8 @@
 
     protected void renderTabButton(GuiGraphics p_283590_, CreativeModeTab p_283489_) {
         boolean flag = p_283489_ == selectedTab;
-        boolean flag1 = p_283489_.row() == CreativeModeTab.Row.TOP;
-        int i = p_283489_.column();
+        boolean flag1 = currentPage.isTop(p_283489_);
+        int i = currentPage.getColumn(p_283489_);
         int j = this.leftPos + this.getTabX(p_283489_);
         int k = this.topPos - (flag1 ? 28 : -(this.imageHeight - 4));
         ResourceLocation[] aresourcelocation;
@@ -790,6 +_,7 @@
             aresourcelocation = flag ? SELECTED_BOTTOM_TABS : UNSELECTED_BOTTOM_TABS;
         }
 
+        //PATCH 1.20.2: Deal with custom tab backgrounds, and deal with transparency.
         p_283590_.blitSprite(aresourcelocation[Mth.clamp(i, 0, aresourcelocation.length)], j, k, 26, 32);
         p_283590_.pose().pushPose();
         p_283590_.pose().translate(0.0F, 0.0F, 100.0F);
@@ -813,7 +_,7 @@
         if (p_98601_) {
             List<ItemStack> list = hotbar.load(registryaccess);
 
-            for(int i = 0; i < Inventory.getSelectionSize(); ++i) {
+            for (int i = 0; i < Inventory.getSelectionSize(); i++) {
                 ItemStack itemstack = list.get(i);
                 localplayer.getInventory().setItem(i, itemstack);
                 p_98599_.gameMode.handleCreativeModeItemAdd(itemstack, 36 + i);
@@ -831,6 +_,14 @@
         }
     }
 
+    public net.neoforged.neoforge.client.gui.CreativeTabsScreenPage getCurrentPage() {
+        return currentPage;
+    }
+
+    public void setCurrentPage(net.neoforged.neoforge.client.gui.CreativeTabsScreenPage currentPage) {
+        this.currentPage = currentPage;
+    }
+
     @OnlyIn(Dist.CLIENT)
     static class CustomCreativeSlot extends Slot {
         public CustomCreativeSlot(Container p_98633_, int p_98634_, int p_98635_, int p_98636_) {
@@ -840,11 +_,9 @@
         @Override
         public boolean mayPickup(Player p_98638_) {
             ItemStack itemstack = this.getItem();
-            if (super.mayPickup(p_98638_) && !itemstack.isEmpty()) {
-                return itemstack.isItemEnabled(p_98638_.level().enabledFeatures()) && !itemstack.has(DataComponents.CREATIVE_SLOT_LOCK);
-            } else {
-                return itemstack.isEmpty();
-            }
+            return super.mayPickup(p_98638_) && !itemstack.isEmpty()
+                ? itemstack.isItemEnabled(p_98638_.level().enabledFeatures()) && !itemstack.has(DataComponents.CREATIVE_SLOT_LOCK)
+                : itemstack.isEmpty();
         }
     }
 
@@ -858,13 +_,13 @@
             this.inventoryMenu = p_98641_.inventoryMenu;
             Inventory inventory = p_98641_.getInventory();
 
-            for(int i = 0; i < 5; ++i) {
-                for(int j = 0; j < 9; ++j) {
+            for (int i = 0; i < 5; i++) {
+                for (int j = 0; j < 9; j++) {
                     this.addSlot(new CreativeModeInventoryScreen.CustomCreativeSlot(CreativeModeInventoryScreen.CONTAINER, i * 9 + j, 9 + j * 18, 18 + i * 18));
                 }
             }
 
-            for(int k = 0; k < 9; ++k) {
+            for (int k = 0; k < 9; k++) {
                 this.addSlot(new Slot(inventory, k, 9 + k * 18, 112));
             }
 
@@ -895,8 +_,8 @@
         public void scrollTo(float p_98643_) {
             int i = this.getRowIndexForScroll(p_98643_);
 
-            for(int j = 0; j < 5; ++j) {
-                for(int k = 0; k < 9; ++k) {
+            for (int j = 0; j < 5; j++) {
+                for (int k = 0; k < 9; k++) {
                     int l = k + (j + i) * 9;
                     if (l >= 0 && l < this.items.size()) {
                         CreativeModeInventoryScreen.CONTAINER.setItem(k + j * 9, this.items.get(l));
@@ -1017,6 +_,22 @@
         @Override
         public boolean mayPickup(Player p_98665_) {
             return this.target.mayPickup(p_98665_);
+        }
+
+        @Override
+        public int getSlotIndex() {
+            return this.target.getSlotIndex();
+        }
+
+        @Override
+        public boolean isSameInventory(Slot other) {
+            return this.target.isSameInventory(other);
+        }
+
+        @Override
+        public Slot setBackground(ResourceLocation atlas, ResourceLocation sprite) {
+            this.target.setBackground(atlas, sprite);
+            return this;
         }
     }
 }
