--- a/net/minecraft/client/gui/screens/inventory/EnchantmentNames.java
+++ b/net/minecraft/client/gui/screens/inventory/EnchantmentNames.java
@@ -92,7 +_,7 @@
         StringBuilder stringbuilder = new StringBuilder();
         int i = this.random.nextInt(2) + 3;
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             if (j != 0) {
                 stringbuilder.append(" ");
             }
