--- a/net/minecraft/client/gui/screens/inventory/EnchantmentScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/EnchantmentScreen.java
@@ -73,7 +_,7 @@
         int i = (this.width - this.imageWidth) / 2;
         int j = (this.height - this.imageHeight) / 2;
 
-        for(int k = 0; k < 3; ++k) {
+        for (int k = 0; k < 3; k++) {
             double d0 = p_98758_ - (double)(i + 60);
             double d1 = p_98759_ - (double)(j + 14 + 19 * k);
             if (d0 >= 0.0 && d1 >= 0.0 && d0 < 108.0 && d1 < 19.0 && this.menu.clickMenuButton(this.minecraft.player, k)) {
@@ -94,7 +_,7 @@
         EnchantmentNames.getInstance().initSeed((long)this.menu.getEnchantmentSeed());
         int k = this.menu.getGoldCount();
 
-        for(int l = 0; l < 3; ++l) {
+        for (int l = 0; l < 3; l++) {
             int i1 = i + 60;
             int j1 = i1 + 20;
             int k1 = this.menu.costs[l];
@@ -105,7 +_,7 @@
                 int l1 = 86 - this.font.width(s);
                 FormattedText formattedtext = EnchantmentNames.getInstance().getRandomName(this.font, l1);
                 int i2 = 6839882;
-                if ((k < l + 1 || this.minecraft.player.experienceLevel < k1) && !this.minecraft.player.getAbilities().instabuild) {
+                if (((k < l + 1 || this.minecraft.player.experienceLevel < k1) && !this.minecraft.player.getAbilities().instabuild) || this.menu.enchantClue[l] == -1) { // Forge: render buttons as disabled when enchantable but enchantability not met on lower levels
                     p_282430_.blitSprite(ENCHANTMENT_SLOT_DISABLED_SPRITE, i1, j + 14 + 19 * l, 108, 19);
                     p_282430_.blitSprite(DISABLED_LEVEL_SPRITES[l], i1 + 1, j + 15 + 19 * l, 16, 16);
                     p_282430_.drawWordWrap(this.font, formattedtext, j1, j + 16 + 19 * l, l1, (i2 & 16711422) >> 1);
@@ -161,15 +_,18 @@
         boolean flag = this.minecraft.player.getAbilities().instabuild;
         int i = this.menu.getGoldCount();
 
-        for(int j = 0; j < 3; ++j) {
+        for (int j = 0; j < 3; j++) {
             int k = this.menu.costs[j];
             Enchantment enchantment = Enchantment.byId(this.menu.enchantClue[j]);
             int l = this.menu.levelClue[j];
             int i1 = j + 1;
-            if (this.isHovering(60, 14 + 19 * j, 108, 17, (double)p_282491_, (double)p_281953_) && k > 0 && l >= 0 && enchantment != null) {
+            if (this.isHovering(60, 14 + 19 * j, 108, 17, (double)p_282491_, (double)p_281953_) && k > 0) {
                 List<Component> list = Lists.newArrayList();
-                list.add(Component.translatable("container.enchant.clue", enchantment.getFullname(l)).withStyle(ChatFormatting.WHITE));
-                if (!flag) {
+                list.add((Component.translatable("container.enchant.clue", enchantment == null ? "" : enchantment.getFullname(l))).withStyle(ChatFormatting.WHITE));
+                if (enchantment == null) {
+                    list.add(Component.literal(""));
+                    list.add(Component.translatable("neoforge.container.enchant.limitedEnchantability").withStyle(ChatFormatting.RED));
+                } else if (!flag) {
                     list.add(CommonComponents.EMPTY);
                     if (this.minecraft.player.experienceLevel < k) {
                         list.add(Component.translatable("container.enchant.level.requirement", this.menu.costs[j]).withStyle(ChatFormatting.RED));
@@ -205,16 +_,16 @@
             this.last = itemstack;
 
             do {
-                this.flipT += (float)(this.random.nextInt(4) - this.random.nextInt(4));
-            } while(this.flip <= this.flipT + 1.0F && this.flip >= this.flipT - 1.0F);
+                this.flipT = this.flipT + (float)(this.random.nextInt(4) - this.random.nextInt(4));
+            } while (this.flip <= this.flipT + 1.0F && this.flip >= this.flipT - 1.0F);
         }
 
-        ++this.time;
+        this.time++;
         this.oFlip = this.flip;
         this.oOpen = this.open;
         boolean flag = false;
 
-        for(int i = 0; i < 3; ++i) {
+        for (int i = 0; i < 3; i++) {
             if (this.menu.costs[i] != 0) {
                 flag = true;
             }
@@ -230,7 +_,7 @@
         float f1 = (this.flipT - this.flip) * 0.4F;
         float f = 0.2F;
         f1 = Mth.clamp(f1, -0.2F, 0.2F);
-        this.flipA += (f1 - this.flipA) * 0.9F;
-        this.flip += this.flipA;
+        this.flipA = this.flipA + (f1 - this.flipA) * 0.9F;
+        this.flip = this.flip + this.flipA;
     }
 }
