--- a/net/minecraft/client/gui/screens/inventory/SmithingScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/SmithingScreen.java
@@ -12,7 +_,6 @@
 import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.inventory.SmithingMenu;
 import net.minecraft.world.item.ArmorItem;
-import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.SmithingTemplateItem;
 import net.neoforged.api.distmarker.Dist;
@@ -81,14 +_,9 @@
 
     private Optional<SmithingTemplateItem> getTemplateItem() {
         ItemStack itemstack = this.menu.getSlot(0).getItem();
-        if (!itemstack.isEmpty()) {
-            Item item = itemstack.getItem();
-            if (item instanceof SmithingTemplateItem smithingtemplateitem) {
-                return Optional.of(smithingtemplateitem);
-            }
-        }
-
-        return Optional.empty();
+        return !itemstack.isEmpty() && itemstack.getItem() instanceof SmithingTemplateItem smithingtemplateitem
+            ? Optional.of(smithingtemplateitem)
+            : Optional.empty();
     }
 
     @Override
@@ -117,14 +_,13 @@
 
     private void updateArmorStandPreview(ItemStack p_268225_) {
         if (this.armorStandPreview != null) {
-            for(EquipmentSlot equipmentslot : EquipmentSlot.values()) {
+            for (EquipmentSlot equipmentslot : EquipmentSlot.values()) {
                 this.armorStandPreview.setItemSlot(equipmentslot, ItemStack.EMPTY);
             }
 
             if (!p_268225_.isEmpty()) {
                 ItemStack itemstack = p_268225_.copy();
-                Item item = p_268225_.getItem();
-                if (item instanceof ArmorItem armoritem) {
+                if (p_268225_.getItem() instanceof ArmorItem armoritem) {
                     this.armorStandPreview.setItemSlot(armoritem.getEquipmentSlot(), itemstack);
                 } else {
                     this.armorStandPreview.setItemSlot(EquipmentSlot.OFFHAND, itemstack);
@@ -153,14 +_,11 @@
                 if (this.hoveredSlot.index == 0) {
                     optional = Optional.of(MISSING_TEMPLATE_TOOLTIP);
                 }
-            } else {
-                Item item = itemstack.getItem();
-                if (item instanceof SmithingTemplateItem smithingtemplateitem && itemstack1.isEmpty()) {
-                    if (this.hoveredSlot.index == 1) {
-                        optional = Optional.of(smithingtemplateitem.getBaseSlotDescription());
-                    } else if (this.hoveredSlot.index == 2) {
-                        optional = Optional.of(smithingtemplateitem.getAdditionSlotDescription());
-                    }
+            } else if (itemstack.getItem() instanceof SmithingTemplateItem smithingtemplateitem && itemstack1.isEmpty()) {
+                if (this.hoveredSlot.index == 1) {
+                    optional = Optional.of(smithingtemplateitem.getBaseSlotDescription());
+                } else if (this.hoveredSlot.index == 2) {
+                    optional = Optional.of(smithingtemplateitem.getAdditionSlotDescription());
                 }
             }
         }
