--- a/net/minecraft/world/item/ArmorMaterials.java
+++ b/net/minecraft/world/item/ArmorMaterials.java
@@ -78,13 +_,6 @@
         p_323385_.put(ArmorItem.Type.HELMET, 3);
         p_323385_.put(ArmorItem.Type.BODY, 11);
     }), 10, SoundEvents.ARMOR_EQUIP_WOLF, 0.0F, 0.0F, () -> Ingredient.of(Items.ARMADILLO_SCUTE));
-    public static final Holder<ArmorMaterial> POTATO = register("potato", Util.make(new EnumMap<>(ArmorItem.Type.class), p_340176_ -> {
-        p_340176_.put(ArmorItem.Type.BOOTS, 0);
-        p_340176_.put(ArmorItem.Type.LEGGINGS, 0);
-        p_340176_.put(ArmorItem.Type.CHESTPLATE, 0);
-        p_340176_.put(ArmorItem.Type.HELMET, 0);
-        p_340176_.put(ArmorItem.Type.BODY, 0);
-    }), 9, SoundEvents.ARMOR_EQUIP_TURTLE, 0.0F, 0.0F, () -> Ingredient.of(Items.POISONOUS_POTATO_PLANT));
 
     public static Holder<ArmorMaterial> bootstrap(Registry<ArmorMaterial> p_323827_) {
         return LEATHER;
@@ -115,7 +_,7 @@
     ) {
         EnumMap<ArmorItem.Type, Integer> enummap = new EnumMap<>(ArmorItem.Type.class);
 
-        for(ArmorItem.Type armoritem$type : ArmorItem.Type.values()) {
+        for (ArmorItem.Type armoritem$type : ArmorItem.Type.values()) {
             enummap.put(armoritem$type, p_324599_.get(armoritem$type));
         }
 
