--- a/net/minecraft/world/item/BoatItem.java
+++ b/net/minecraft/world/item/BoatItem.java
@@ -43,7 +_,7 @@
             if (!list.isEmpty()) {
                 Vec3 vec31 = p_40623_.getEyePosition();
 
-                for(Entity entity : list) {
+                for (Entity entity : list) {
                     AABB aabb = entity.getBoundingBox().inflate((double)entity.getPickRadius());
                     if (aabb.contains(vec31)) {
                         return InteractionResultHolder.pass(itemstack);
