--- a/net/minecraft/world/item/BoneMealItem.java
+++ b/net/minecraft/world/item/BoneMealItem.java
@@ -38,7 +_,7 @@
         BlockPos blockpos = context.getClickedPos();
         BlockPos blockpos1 = blockpos.relative(context.getClickedFace());
         ItemStack itemstack = context.getItemInHand();
-        if (growCrop(itemstack, level, blockpos)) {
+        if (applyBonemeal(itemstack, level, blockpos, context.getPlayer())) {
             if (!level.isClientSide()) {
                 itemstack.causeUseVibration(context.getPlayer(), GameEvent.ITEM_INTERACT_FINISH);
                 level.levelEvent(1505, blockpos, 15);
@@ -61,8 +_,17 @@
         }
     }
 
+    @Deprecated //Forge: Use Player/Hand version
     public static boolean growCrop(ItemStack itemStack, Level level, BlockPos pos) {
+        if (level instanceof net.minecraft.server.level.ServerLevel)
+            return applyBonemeal(itemStack, level, pos, null);
+        return false;
+    }
+
+    public static boolean applyBonemeal(ItemStack itemStack, Level level, BlockPos pos, net.minecraft.world.entity.player.@Nullable Player player) {
         BlockState blockstate = level.getBlockState(pos);
+        var event = net.neoforged.neoforge.event.EventHooks.fireBonemealEvent(player, level, pos, blockstate, itemStack);
+        if (event.isCanceled()) return event.isSuccessful();
         if (blockstate.getBlock() instanceof BonemealableBlock bonemealableblock && bonemealableblock.isValidBonemealTarget(level, pos, blockstate)) {
             if (level instanceof ServerLevel) {
                 if (bonemealableblock.isBonemealSuccess(level, level.random, pos, blockstate)) {
