--- a/net/minecraft/world/item/BottleItem.java
+++ b/net/minecraft/world/item/BottleItem.java
@@ -31,7 +_,7 @@
         List<AreaEffectCloud> list = p_40656_.getEntitiesOfClass(
             AreaEffectCloud.class,
             p_40657_.getBoundingBox().inflate(2.0),
-            p_340177_ -> p_340177_ != null && p_340177_.isAlive() && p_340177_.getOwner() instanceof EnderDragon
+            p_337905_ -> p_337905_ != null && p_337905_.isAlive() && p_337905_.getOwner() instanceof EnderDragon
         );
         ItemStack itemstack = p_40657_.getItemInHand(p_40658_);
         if (!list.isEmpty()) {
