--- a/net/minecraft/world/item/BowItem.java
+++ b/net/minecraft/world/item/BowItem.java
@@ -32,6 +_,8 @@
                 return false;
             } else {
                 int i = this.getUseDuration(itemStack, entity) - remainingTime;
+                i = net.neoforged.neoforge.event.EventHooks.onArrowLoose(itemStack, level, player, i, !itemstack.isEmpty());
+                if (i < 0) return false;
                 float f = getPowerForTime(i);
                 if (f < 0.1) {
                     return false;
@@ -89,6 +_,10 @@
     public InteractionResult use(Level level, Player player, InteractionHand hand) {
         ItemStack itemstack = player.getItemInHand(hand);
         boolean flag = !player.getProjectile(itemstack).isEmpty();
+
+        InteractionResult ret = net.neoforged.neoforge.event.EventHooks.onArrowNock(itemstack, level, player, hand, flag);
+        if (ret != null) return ret;
+
         if (!player.hasInfiniteMaterials() && !flag) {
             return InteractionResult.FAIL;
         } else {
