--- a/net/minecraft/world/item/CrossbowItem.java
+++ b/net/minecraft/world/item/CrossbowItem.java
@@ -168,6 +_,7 @@
         Level p_40888_, LivingEntity p_40889_, InteractionHand p_40890_, ItemStack p_40891_, float p_40892_, float p_40893_, @Nullable LivingEntity p_331602_
     ) {
         if (!p_40888_.isClientSide()) {
+            if (p_40889_ instanceof Player player && net.neoforged.neoforge.event.EventHooks.onArrowLoose(p_40891_, p_40889_.level(), player, 1, true) < 0) return;
             ChargedProjectiles chargedprojectiles = p_40891_.set(DataComponents.CHARGED_PROJECTILES, ChargedProjectiles.EMPTY);
             if (chargedprojectiles != null && !chargedprojectiles.isEmpty()) {
                 this.shoot(p_40888_, p_40889_, p_40890_, p_40891_, chargedprojectiles.getItems(), p_40892_, p_40893_, p_40889_ instanceof Player, p_331602_);
@@ -228,7 +_,7 @@
     }
 
     private SoundEvent getStartSound(int p_40852_) {
-        switch(p_40852_) {
+        switch (p_40852_) {
             case 1:
                 return SoundEvents.CROSSBOW_QUICK_CHARGE_1;
             case 2:
@@ -250,16 +_,16 @@
     }
 
     @Override
-    public void appendHoverText(ItemStack p_40880_, @Nullable Level p_40881_, List<Component> p_40882_, TooltipFlag p_40883_) {
+    public void appendHoverText(ItemStack p_40880_, Item.TooltipContext p_339686_, List<Component> p_40882_, TooltipFlag p_40883_) {
         ChargedProjectiles chargedprojectiles = p_40880_.get(DataComponents.CHARGED_PROJECTILES);
         if (chargedprojectiles != null && !chargedprojectiles.isEmpty()) {
             ItemStack itemstack = chargedprojectiles.getItems().get(0);
             p_40882_.add(Component.translatable("item.minecraft.crossbow.projectile").append(CommonComponents.SPACE).append(itemstack.getDisplayName()));
             if (p_40883_.isAdvanced() && itemstack.is(Items.FIREWORK_ROCKET)) {
                 List<Component> list = Lists.newArrayList();
-                Items.FIREWORK_ROCKET.appendHoverText(itemstack, p_40881_, list, p_40883_);
+                Items.FIREWORK_ROCKET.appendHoverText(itemstack, p_339686_, list, p_40883_);
                 if (!list.isEmpty()) {
-                    for(int i = 0; i < list.size(); ++i) {
+                    for (int i = 0; i < list.size(); i++) {
                         list.set(i, Component.literal("  ").append(list.get(i)).withStyle(ChatFormatting.GRAY));
                     }
 
