--- a/net/minecraft/world/item/FireworkStarItem.java
+++ b/net/minecraft/world/item/FireworkStarItem.java
@@ -1,11 +_,9 @@
 package net.minecraft.world.item;
 
 import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.core.component.DataComponents;
 import net.minecraft.network.chat.Component;
 import net.minecraft.world.item.component.FireworkExplosion;
-import net.minecraft.world.level.Level;
 
 public class FireworkStarItem extends Item {
     public FireworkStarItem(Item.Properties p_41248_) {
@@ -13,7 +_,7 @@
     }
 
     @Override
-    public void appendHoverText(ItemStack p_41252_, @Nullable Level p_41253_, List<Component> p_41254_, TooltipFlag p_41255_) {
+    public void appendHoverText(ItemStack p_41252_, Item.TooltipContext p_339685_, List<Component> p_41254_, TooltipFlag p_41255_) {
         FireworkExplosion fireworkexplosion = p_41252_.get(DataComponents.FIREWORK_EXPLOSION);
         if (fireworkexplosion != null) {
             fireworkexplosion.addToTooltip(p_41254_::add, p_41255_);
