--- a/net/minecraft/world/item/HangingEntityItem.java
+++ b/net/minecraft/world/item/HangingEntityItem.java
@@ -2,7 +_,6 @@
 
 import java.util.List;
 import java.util.Optional;
-import javax.annotation.Nullable;
 import net.minecraft.ChatFormatting;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
@@ -86,8 +_,8 @@
     }
 
     @Override
-    public void appendHoverText(ItemStack p_270235_, @Nullable Level p_270688_, List<Component> p_270630_, TooltipFlag p_270170_) {
-        super.appendHoverText(p_270235_, p_270688_, p_270630_, p_270170_);
+    public void appendHoverText(ItemStack p_270235_, Item.TooltipContext p_339644_, List<Component> p_270630_, TooltipFlag p_270170_) {
+        super.appendHoverText(p_270235_, p_339644_, p_270630_, p_270170_);
         if (this.type == EntityType.PAINTING) {
             CustomData customdata = p_270235_.getOrDefault(DataComponents.ENTITY_DATA, CustomData.EMPTY);
             if (!customdata.isEmpty()) {
