--- a/net/minecraft/world/item/Items.java
+++ b/net/minecraft/world/item/Items.java
@@ -2433,11 +_,19 @@
     }
 
     public static Item registerBlock(Block block, Block... alternatives) {
-        Item item = registerBlock(block);
-
-        for (Block block : alternatives) {
-            Item.BY_BLOCK.put(block, item);
-        }
+        Item item = registerItem(
+                blockIdToItemId(block.builtInRegistryHolder().key()),
+                p -> new BlockItem(block, p) {
+                    @Override
+                    public void registerBlocks(java.util.Map<Block, Item> map, Item self) {
+                        super.registerBlocks(map, self);
+                        for (Block block : alternatives) {
+                            map.put(block, self);
+                        }
+                    }
+                },
+                new Item.Properties().useBlockDescriptionPrefix()
+        );
 
         return item;
     }
