--- a/net/minecraft/world/item/ItemCooldowns.java
+++ b/net/minecraft/world/item/ItemCooldowns.java
@@ -26,11 +_,11 @@
     }
 
     public void tick() {
-        ++this.tickCount;
+        this.tickCount++;
         if (!this.cooldowns.isEmpty()) {
             Iterator<Entry<Item, ItemCooldowns.CooldownInstance>> iterator = this.cooldowns.entrySet().iterator();
 
-            while(iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 Entry<Item, ItemCooldowns.CooldownInstance> entry = iterator.next();
                 if (entry.getValue().endTime <= this.tickCount) {
                     iterator.remove();
