--- a/net/minecraft/world/item/ItemUtils.java
+++ b/net/minecraft/world/item/ItemUtils.java
@@ -42,7 +_,7 @@
     public static void onContainerDestroyed(ItemEntity p_150953_, Stream<ItemStack> p_150954_) {
         Level level = p_150953_.level();
         if (!level.isClientSide) {
-            p_150954_.forEach(p_340200_ -> level.addFreshEntity(new ItemEntity(level, p_150953_.getX(), p_150953_.getY(), p_150953_.getZ(), p_340200_)));
+            p_150954_.forEach(p_337935_ -> level.addFreshEntity(new ItemEntity(level, p_150953_.getX(), p_150953_.getY(), p_150953_.getZ(), p_337935_)));
         }
     }
 }
