--- a/net/minecraft/world/item/KnowledgeBookItem.java
+++ b/net/minecraft/world/item/KnowledgeBookItem.java
@@ -37,7 +_,7 @@
                 RecipeManager recipemanager = p_42824_.getServer().getRecipeManager();
                 List<RecipeHolder<?>> list1 = new ArrayList<>(list.size());
 
-                for(ResourceLocation resourcelocation : list) {
+                for (ResourceLocation resourcelocation : list) {
                     Optional<RecipeHolder<?>> optional = recipemanager.byKey(resourcelocation);
                     if (!optional.isPresent()) {
                         LOGGER.error("Invalid recipe: {}", resourcelocation);
