--- a/net/minecraft/world/item/LeadItem.java
+++ b/net/minecraft/world/item/LeadItem.java
@@ -44,7 +_,7 @@
         AABB aabb = new AABB((double)i - 7.0, (double)j - 7.0, (double)k - 7.0, (double)i + 7.0, (double)j + 7.0, (double)k + 7.0);
         List<Mob> list = p_42831_.getEntitiesOfClass(Mob.class, aabb, p_316242_ -> p_316242_.getLeashHolder() == p_42830_);
 
-        for(Mob mob : list) {
+        for (Mob mob : list) {
             if (leashfenceknotentity == null) {
                 leashfenceknotentity = LeashFenceKnotEntity.getOrCreateKnot(p_42831_, p_42832_);
                 leashfenceknotentity.playPlacementSound();
