--- a/net/minecraft/world/item/MapItem.java
+++ b/net/minecraft/world/item/MapItem.java
@@ -57,6 +_,16 @@
 
     @Nullable
     public static MapItemSavedData getSavedData(ItemStack p_42854_, Level p_42855_) {
+        // Neo: Add instance method so that mods can override
+        Item map = p_42854_.getItem();
+        if(map instanceof MapItem) {
+            return ((MapItem)map).getCustomMapData(p_42854_, p_42855_);
+        }
+        return null;
+    }
+
+    @Nullable
+    protected MapItemSavedData getCustomMapData(ItemStack p_42854_, Level p_42855_) {
         MapId mapid = p_42854_.get(DataComponents.MAP_ID);
         return getSavedData(mapid, p_42855_);
     }
@@ -83,17 +_,17 @@
             }
 
             MapItemSavedData.HoldingPlayer mapitemsaveddata$holdingplayer = p_42896_.getHoldingPlayer((Player)p_42895_);
-            ++mapitemsaveddata$holdingplayer.step;
+            mapitemsaveddata$holdingplayer.step++;
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
             BlockPos.MutableBlockPos blockpos$mutableblockpos1 = new BlockPos.MutableBlockPos();
             boolean flag = false;
 
-            for(int k1 = l - j1 + 1; k1 < l + j1; ++k1) {
+            for (int k1 = l - j1 + 1; k1 < l + j1; k1++) {
                 if ((k1 & 15) == (mapitemsaveddata$holdingplayer.step & 15) || flag) {
                     flag = false;
                     double d0 = 0.0;
 
-                    for(int l1 = i1 - j1 - 1; l1 < i1 + j1; ++l1) {
+                    for (int l1 = i1 - j1 - 1; l1 < i1 + j1; l1++) {
                         if (k1 >= 0 && l1 >= -1 && k1 < 128 && l1 < 128) {
                             int i2 = Mth.square(k1 - l) + Mth.square(l1 - i1);
                             boolean flag1 = i2 > (j1 - 2) * (j1 - 2);
@@ -115,8 +_,8 @@
 
                                     d1 = 100.0;
                                 } else {
-                                    for(int i4 = 0; i4 < i; ++i4) {
-                                        for(int j3 = 0; j3 < i; ++j3) {
+                                    for (int i4 = 0; i4 < i; i4++) {
+                                        for (int j3 = 0; j3 < i; j3++) {
                                             blockpos$mutableblockpos.set(j2 + i4, 0, k2 + j3);
                                             int k3 = levelchunk.getHeight(
                                                     Heightmap.Types.WORLD_SURFACE, blockpos$mutableblockpos.getX(), blockpos$mutableblockpos.getZ()
@@ -124,16 +_,12 @@
                                                 + 1;
                                             BlockState blockstate;
                                             if (k3 <= p_42894_.getMinBuildHeight() + 1) {
-                                                if (p_42894_.isPotato()) {
-                                                    blockstate = Blocks.WARPED_WART_BLOCK.defaultBlockState();
-                                                } else {
-                                                    blockstate = Blocks.BEDROCK.defaultBlockState();
-                                                }
+                                                blockstate = Blocks.BEDROCK.defaultBlockState();
                                             } else {
                                                 do {
                                                     blockpos$mutableblockpos.setY(--k3);
                                                     blockstate = levelchunk.getBlockState(blockpos$mutableblockpos);
-                                                } while(
+                                                } while (
                                                     blockstate.getMapColor(p_42894_, blockpos$mutableblockpos) == MapColor.NONE
                                                         && k3 > p_42894_.getMinBuildHeight()
                                                 );
@@ -146,8 +_,8 @@
                                                     do {
                                                         blockpos$mutableblockpos1.setY(l3--);
                                                         blockstate1 = levelchunk.getBlockState(blockpos$mutableblockpos1);
-                                                        ++l2;
-                                                    } while(l3 > p_42894_.getMinBuildHeight() && !blockstate1.getFluidState().isEmpty());
+                                                        l2++;
+                                                    } while (l3 > p_42894_.getMinBuildHeight() && !blockstate1.getFluidState().isEmpty());
 
                                                     blockstate = this.getCorrectStateForFluidBlock(p_42894_, blockstate, blockpos$mutableblockpos);
                                                 }
@@ -216,21 +_,21 @@
                 int i1 = k / i - 64;
                 BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-                for(int j1 = 0; j1 < 128; ++j1) {
-                    for(int k1 = 0; k1 < 128; ++k1) {
+                for (int j1 = 0; j1 < 128; j1++) {
+                    for (int k1 = 0; k1 < 128; k1++) {
                         Holder<Biome> holder = p_42851_.getBiome(blockpos$mutableblockpos.set((l + k1) * i, 0, (i1 + j1) * i));
                         aboolean[j1 * 128 + k1] = holder.is(BiomeTags.WATER_ON_MAP_OUTLINES);
                     }
                 }
 
-                for(int j2 = 1; j2 < 127; ++j2) {
-                    for(int k2 = 1; k2 < 127; ++k2) {
+                for (int j2 = 1; j2 < 127; j2++) {
+                    for (int k2 = 1; k2 < 127; k2++) {
                         int l2 = 0;
 
-                        for(int l1 = -1; l1 < 2; ++l1) {
-                            for(int i2 = -1; i2 < 2; ++i2) {
+                        for (int l1 = -1; l1 < 2; l1++) {
+                            for (int i2 = -1; i2 < 2; i2++) {
                                 if ((l1 != 0 || i2 != 0) && isBiomeWatery(aboolean, j2 + l1, k2 + i2)) {
-                                    ++l2;
+                                    l2++;
                                 }
                             }
                         }
@@ -240,7 +_,7 @@
                         if (isBiomeWatery(aboolean, j2, k2)) {
                             mapcolor = MapColor.COLOR_ORANGE;
                             if (l2 > 7 && k2 % 2 == 0) {
-                                switch((j2 + (int)(Mth.sin((float)k2 + 0.0F) * 7.0F)) / 8 % 5) {
+                                switch ((j2 + (int)(Mth.sin((float)k2 + 0.0F) * 7.0F)) / 8 % 5) {
                                     case 0:
                                     case 4:
                                         mapcolor$brightness = MapColor.Brightness.LOW;
@@ -307,7 +_,7 @@
     public void onCraftedPostProcess(ItemStack p_42913_, Level p_42914_) {
         MapPostProcessing mappostprocessing = p_42913_.remove(DataComponents.MAP_POST_PROCESSING);
         if (mappostprocessing != null) {
-            switch(mappostprocessing) {
+            switch (mappostprocessing) {
                 case LOCK:
                     lockMap(p_42914_, p_42913_);
                     break;
@@ -337,9 +_,9 @@
     }
 
     @Override
-    public void appendHoverText(ItemStack p_42880_, @Nullable Level p_42881_, List<Component> p_42882_, TooltipFlag p_42883_) {
+    public void appendHoverText(ItemStack p_42880_, Item.TooltipContext p_339653_, List<Component> p_42882_, TooltipFlag p_42883_) {
         MapId mapid = p_42880_.get(DataComponents.MAP_ID);
-        MapItemSavedData mapitemsaveddata = p_42881_ == null ? null : getSavedData(mapid, p_42881_);
+        MapItemSavedData mapitemsaveddata = mapid != null ? p_339653_.mapData(mapid) : null;
         MapPostProcessing mappostprocessing = p_42880_.get(DataComponents.MAP_POST_PROCESSING);
         if (mapitemsaveddata != null && (mapitemsaveddata.locked || mappostprocessing == MapPostProcessing.LOCK)) {
             p_42882_.add(Component.translatable("filled_map.locked", mapid.id()).withStyle(ChatFormatting.GRAY));
