--- a/net/minecraft/world/item/StandingAndWallBlockItem.java
+++ b/net/minecraft/world/item/StandingAndWallBlockItem.java
@@ -32,7 +_,7 @@
         LevelReader levelreader = p_43255_.getLevel();
         BlockPos blockpos = p_43255_.getClickedPos();
 
-        for(Direction direction : p_43255_.getNearestLookingDirections()) {
+        for (Direction direction : p_43255_.getNearestLookingDirections()) {
             if (direction != this.attachmentDirection.getOpposite()) {
                 BlockState blockstate2 = direction == this.attachmentDirection ? this.getBlock().getStateForPlacement(p_43255_) : blockstate;
                 if (blockstate2 != null && this.canPlace(levelreader, blockstate2, blockpos)) {
@@ -49,5 +_,10 @@
     public void registerBlocks(Map<Block, Item> p_43252_, Item p_43253_) {
         super.registerBlocks(p_43252_, p_43253_);
         p_43252_.put(this.wallBlock, p_43253_);
+    }
+
+    public void removeFromBlockToItemMap(Map<Block, Item> blockToItemMap, Item itemIn) {
+        super.removeFromBlockToItemMap(blockToItemMap, itemIn);
+        blockToItemMap.remove(this.wallBlock);
     }
 }
