--- a/net/minecraft/world/item/SuspiciousStewItem.java
+++ b/net/minecraft/world/item/SuspiciousStewItem.java
@@ -2,7 +_,6 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.core.component.DataComponents;
 import net.minecraft.network.chat.Component;
 import net.minecraft.world.effect.MobEffectInstance;
@@ -19,17 +_,17 @@
     }
 
     @Override
-    public void appendHoverText(ItemStack p_260314_, @Nullable Level p_259224_, List<Component> p_259700_, TooltipFlag p_260021_) {
-        super.appendHoverText(p_260314_, p_259224_, p_259700_, p_260021_);
+    public void appendHoverText(ItemStack p_260314_, Item.TooltipContext p_339691_, List<Component> p_259700_, TooltipFlag p_260021_) {
+        super.appendHoverText(p_260314_, p_339691_, p_259700_, p_260021_);
         if (p_260021_.isCreative()) {
             List<MobEffectInstance> list = new ArrayList<>();
             SuspiciousStewEffects suspicioussteweffects = p_260314_.getOrDefault(DataComponents.SUSPICIOUS_STEW_EFFECTS, SuspiciousStewEffects.EMPTY);
 
-            for(SuspiciousStewEffects.Entry suspicioussteweffects$entry : suspicioussteweffects.effects()) {
+            for (SuspiciousStewEffects.Entry suspicioussteweffects$entry : suspicioussteweffects.effects()) {
                 list.add(suspicioussteweffects$entry.createEffectInstance());
             }
 
-            PotionContents.addPotionTooltip(list, p_259700_::add, 1.0F, p_259224_ == null ? 20.0F : p_259224_.tickRateManager().tickrate());
+            PotionContents.addPotionTooltip(list, p_259700_::add, 1.0F, p_339691_.tickRate());
         }
     }
 
@@ -37,7 +_,7 @@
     public ItemStack finishUsingItem(ItemStack p_43263_, Level p_43264_, LivingEntity p_43265_) {
         SuspiciousStewEffects suspicioussteweffects = p_43263_.getOrDefault(DataComponents.SUSPICIOUS_STEW_EFFECTS, SuspiciousStewEffects.EMPTY);
 
-        for(SuspiciousStewEffects.Entry suspicioussteweffects$entry : suspicioussteweffects.effects()) {
+        for (SuspiciousStewEffects.Entry suspicioussteweffects$entry : suspicioussteweffects.effects()) {
             p_43265_.addEffect(suspicioussteweffects$entry.createEffectInstance());
         }
 
