--- a/net/minecraft/world/item/WindChargeItem.java
+++ b/net/minecraft/world/item/WindChargeItem.java
@@ -1,17 +_,23 @@
 package net.minecraft.world.item;
 
 import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.core.Position;
+import net.minecraft.core.dispenser.BlockSource;
 import net.minecraft.sounds.SoundEvents;
 import net.minecraft.sounds.SoundSource;
 import net.minecraft.stats.Stats;
+import net.minecraft.util.RandomSource;
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.InteractionResultHolder;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.Projectile;
 import net.minecraft.world.entity.projectile.windcharge.WindCharge;
 import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.DispenserBlock;
 import net.minecraft.world.phys.Vec3;
 
-public class WindChargeItem extends Item {
+public class WindChargeItem extends Item implements ProjectileItem {
     private static final int COOLDOWN = 10;
 
     public WindChargeItem(Item.Properties p_326377_) {
@@ -46,5 +_,27 @@
         p_326042_.awardStat(Stats.ITEM_USED.get(this));
         itemstack.consume(1, p_326042_);
         return InteractionResultHolder.sidedSuccess(itemstack, p_326306_.isClientSide());
+    }
+
+    @Override
+    public Projectile asProjectile(Level p_338589_, Position p_338670_, ItemStack p_338308_, Direction p_338206_) {
+        RandomSource randomsource = p_338589_.getRandom();
+        double d0 = randomsource.triangle((double)p_338206_.getStepX(), 0.11485000000000001);
+        double d1 = randomsource.triangle((double)p_338206_.getStepY(), 0.11485000000000001);
+        double d2 = randomsource.triangle((double)p_338206_.getStepZ(), 0.11485000000000001);
+        return new WindCharge(p_338589_, p_338670_.x(), p_338670_.y(), p_338670_.z(), d0, d1, d2);
+    }
+
+    @Override
+    public void shoot(Projectile p_338260_, double p_338763_, double p_338177_, double p_338349_, float p_338273_, float p_338257_) {
+    }
+
+    @Override
+    public ProjectileItem.DispenseConfig createDispenseConfig() {
+        return ProjectileItem.DispenseConfig.builder()
+            .positionFunction((p_338288_, p_338801_) -> DispenserBlock.getDispensePosition(p_338288_, 1.0, Vec3.ZERO))
+            .uncertainty(6.6666665F)
+            .power(1.0F)
+            .build();
     }
 }
