--- a/net/minecraft/world/level/levelgen/material/MaterialRuleList.java
+++ b/net/minecraft/world/level/levelgen/material/MaterialRuleList.java
@@ -10,7 +_,7 @@
     @Nullable
     @Override
     public BlockState calculate(DensityFunction.FunctionContext p_209815_) {
-        for(NoiseChunk.BlockStateFiller noisechunk$blockstatefiller : this.materialRuleList) {
+        for (NoiseChunk.BlockStateFiller noisechunk$blockstatefiller : this.materialRuleList) {
             BlockState blockstate = noisechunk$blockstatefiller.calculate(p_209815_);
             if (blockstate != null) {
                 return blockstate;
