--- a/net/minecraft/client/resources/sounds/BiomeAmbientSoundsHandler.java
+++ b/net/minecraft/client/resources/sounds/BiomeAmbientSoundsHandler.java
@@ -62,7 +_,7 @@
                     }
 
                     p_174925_.fadeIn();
-                    return p_174925_;
+                    return (BiomeAmbientSoundsHandler.LoopSoundInstance)p_174925_;
                 }));
         }
 
@@ -83,9 +_,9 @@
                     );
                     int j = level.getBrightness(LightLayer.SKY, blockpos);
                     if (j > 0) {
-                        this.moodiness -= (float)j / (float)level.getMaxLightLevel() * 0.001F;
+                        this.moodiness = this.moodiness - (float)j / (float)level.getMaxLightLevel() * 0.001F;
                     } else {
-                        this.moodiness -= (float)(level.getBrightness(LightLayer.BLOCK, blockpos) - 1) / (float)p_274718_.getTickDelay();
+                        this.moodiness = this.moodiness - (float)(level.getBrightness(LightLayer.BLOCK, blockpos) - 1) / (float)p_274718_.getTickDelay();
                     }
         
                     if (this.moodiness >= 1.0F) {
@@ -132,7 +_,7 @@
                 this.stop();
             }
 
-            this.fade += this.fadeDirection;
+            this.fade = this.fade + this.fadeDirection;
             this.volume = Mth.clamp((float)this.fade / 40.0F, 0.0F, 1.0F);
         }
 
