--- a/net/minecraft/client/resources/sounds/GuardianAttackSoundInstance.java
+++ b/net/minecraft/client/resources/sounds/GuardianAttackSoundInstance.java
@@ -15,7 +_,7 @@
     private final Guardian guardian;
 
     public GuardianAttackSoundInstance(Guardian p_119690_) {
-        super(p_119690_.isToxic() ? SoundEvents.TOXIFIN_ATTACK : SoundEvents.GUARDIAN_ATTACK, SoundSource.HOSTILE, SoundInstance.createUnseededRandom());
+        super(SoundEvents.GUARDIAN_ATTACK, SoundSource.HOSTILE, SoundInstance.createUnseededRandom());
         this.guardian = p_119690_;
         this.attenuation = SoundInstance.Attenuation.NONE;
         this.looping = true;
