--- a/net/minecraft/client/resources/sounds/SoundEventRegistrationSerializer.java
+++ b/net/minecraft/client/resources/sounds/SoundEventRegistrationSerializer.java
@@ -33,7 +_,7 @@
         if (p_119831_.has("sounds")) {
             JsonArray jsonarray = GsonHelper.getAsJsonArray(p_119831_, "sounds");
 
-            for(int i = 0; i < jsonarray.size(); ++i) {
+            for (int i = 0; i < jsonarray.size(); i++) {
                 JsonElement jsonelement = jsonarray.get(i);
                 if (GsonHelper.isStringValue(jsonelement)) {
                     String s = GsonHelper.convertToString(jsonelement, "sound");
