--- a/net/minecraft/client/resources/sounds/UnderwaterAmbientSoundInstances.java
+++ b/net/minecraft/client/resources/sounds/UnderwaterAmbientSoundInstances.java
@@ -49,7 +_,7 @@
         public void tick() {
             if (!this.player.isRemoved() && this.fade >= 0) {
                 if (this.player.isUnderWater()) {
-                    ++this.fade;
+                    this.fade++;
                 } else {
                     this.fade -= 2;
                 }
