--- a/net/minecraft/client/resources/sounds/UnderwaterAmbientSoundHandler.java
+++ b/net/minecraft/client/resources/sounds/UnderwaterAmbientSoundHandler.java
@@ -23,7 +_,7 @@
 
     @Override
     public void tick() {
-        --this.tickDelay;
+        this.tickDelay--;
         if (this.tickDelay <= 0 && this.player.isUnderWater()) {
             float f = this.player.level().random.nextFloat();
             if (f < 1.0E-4F) {
