--- a/net/minecraft/client/renderer/texture/OverlayTexture.java
+++ b/net/minecraft/client/renderer/texture/OverlayTexture.java
@@ -17,8 +_,8 @@
     public OverlayTexture() {
         NativeImage nativeimage = this.texture.getPixels();
 
-        for(int i = 0; i < 16; ++i) {
-            for(int j = 0; j < 16; ++j) {
+        for (int i = 0; i < 16; i++) {
+            for (int j = 0; j < 16; j++) {
                 if (i < 8) {
                     nativeimage.setPixelRGBA(j, i, -1308622593);
                 } else {
