--- a/net/minecraft/client/renderer/texture/TextureManager.java
+++ b/net/minecraft/client/renderer/texture/TextureManager.java
@@ -150,7 +_,7 @@
 
     @Override
     public void tick() {
-        for(Tickable tickable : this.tickableTextures) {
+        for (Tickable tickable : this.tickableTextures) {
             tickable.tick();
         }
     }
@@ -185,7 +_,7 @@
             RealmsPopupScreen.updateCarouselImages(this.resourceManager);
             Iterator<Entry<ResourceLocation, AbstractTexture>> iterator = this.byPath.entrySet().iterator();
 
-            while(iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 Entry<ResourceLocation, AbstractTexture> entry = iterator.next();
                 ResourceLocation resourcelocation = entry.getKey();
                 AbstractTexture abstracttexture = entry.getValue();
