--- a/net/minecraft/commands/arguments/EntityAnchorArgument.java
+++ b/net/minecraft/commands/arguments/EntityAnchorArgument.java
@@ -63,7 +_,7 @@
         EYES("eyes", (p_90382_, p_90383_) -> new Vec3(p_90382_.x, p_90382_.y + (double)p_90383_.getEyeHeight(), p_90382_.z));
 
         static final Map<String, EntityAnchorArgument.Anchor> BY_NAME = Util.make(Maps.newHashMap(), p_90387_ -> {
-            for(EntityAnchorArgument.Anchor entityanchorargument$anchor : values()) {
+            for (EntityAnchorArgument.Anchor entityanchorargument$anchor : values()) {
                 p_90387_.put(entityanchorargument$anchor.name, entityanchorargument$anchor);
             }
         });
