--- a/net/minecraft/commands/arguments/GameProfileArgument.java
+++ b/net/minecraft/commands/arguments/GameProfileArgument.java
@@ -46,7 +_,7 @@
         } else {
             int i = p_94586_.getCursor();
 
-            while(p_94586_.canRead() && p_94586_.peek() != ' ') {
+            while (p_94586_.canRead() && p_94586_.peek() != ' ') {
                 p_94586_.skip();
             }
 
@@ -103,7 +_,7 @@
             } else {
                 List<GameProfile> list1 = Lists.newArrayList();
 
-                for(ServerPlayer serverplayer : list) {
+                for (ServerPlayer serverplayer : list) {
                     list1.add(serverplayer.getGameProfile());
                 }
 
