--- a/net/minecraft/commands/arguments/NbtPathArgument.java
+++ b/net/minecraft/commands/arguments/NbtPathArgument.java
@@ -64,7 +_,7 @@
         Object2IntMap<NbtPathArgument.Node> object2intmap = new Object2IntOpenHashMap<>();
         boolean flag = true;
 
-        while(p_99491_.canRead() && p_99491_.peek() != ' ') {
+        while (p_99491_.canRead() && p_99491_.peek() != ' ') {
             NbtPathArgument.Node nbtpathargument$node = parseNode(p_99491_, flag);
             list.add(nbtpathargument$node);
             object2intmap.put(nbtpathargument$node, p_99491_.getCursor() - i);
@@ -82,7 +_,7 @@
 
     private static NbtPathArgument.Node parseNode(StringReader p_99496_, boolean p_99497_) throws CommandSyntaxException {
         Object object;
-        switch(p_99496_.peek()) {
+        switch (p_99496_.peek()) {
             case '"':
             case '\'':
                 object = readObjectNode(p_99496_, p_99496_.readString());
@@ -130,7 +_,7 @@
     private static String readUnquotedName(StringReader p_99509_) throws CommandSyntaxException {
         int i = p_99509_.getCursor();
 
-        while(p_99509_.canRead() && isAllowedInUnquotedName(p_99509_.peek())) {
+        while (p_99509_.canRead() && isAllowedInUnquotedName(p_99509_.peek())) {
             p_99509_.skip();
         }
 
@@ -176,14 +_,14 @@
 
         @Override
         public void getOrCreateTag(Tag p_99528_, Supplier<Tag> p_99529_, List<Tag> p_99530_) {
-            if (p_99528_ instanceof CollectionTag collectiontag) {
+            if (p_99528_ instanceof CollectionTag<?> collectiontag) {
                 if (collectiontag.isEmpty()) {
                     Tag tag = p_99529_.get();
                     if (collectiontag.addTag(0, tag)) {
                         p_99530_.add(tag);
                     }
                 } else {
-                    p_99530_.addAll(collectiontag);
+                    p_99530_.addAll((Collection<? extends Tag>)collectiontag);
                 }
             }
         }
@@ -195,10 +_,9 @@
 
         @Override
         public int setTag(Tag p_99525_, Supplier<Tag> p_99526_) {
-            if (!(p_99525_ instanceof CollectionTag)) {
+            if (!(p_99525_ instanceof CollectionTag<?> collectiontag)) {
                 return 0;
             } else {
-                CollectionTag<?> collectiontag = (CollectionTag)p_99525_;
                 int i = collectiontag.size();
                 if (i == 0) {
                     collectiontag.addTag(0, p_99526_.get());
@@ -213,7 +_,7 @@
                         if (!collectiontag.addTag(0, tag)) {
                             return 0;
                         } else {
-                            for(int k = 1; k < i; ++k) {
+                            for (int k = 1; k < i; k++) {
                                 collectiontag.addTag(k, p_99526_.get());
                             }
 
@@ -226,7 +_,7 @@
 
         @Override
         public int removeTag(Tag p_99520_) {
-            if (p_99520_ instanceof CollectionTag collectiontag) {
+            if (p_99520_ instanceof CollectionTag<?> collectiontag) {
                 int i = collectiontag.size();
                 if (i > 0) {
                     collectiontag.clear();
@@ -308,11 +_,11 @@
 
         @Override
         public void getTag(Tag p_99554_, List<Tag> p_99555_) {
-            if (p_99554_ instanceof CollectionTag collectiontag) {
+            if (p_99554_ instanceof CollectionTag<?> collectiontag) {
                 int i = collectiontag.size();
                 int j = this.index < 0 ? i + this.index : this.index;
                 if (0 <= j && j < i) {
-                    p_99555_.add((Tag)collectiontag.get(j));
+                    p_99555_.add(collectiontag.get(j));
                 }
             }
         }
@@ -329,11 +_,11 @@
 
         @Override
         public int setTag(Tag p_99557_, Supplier<Tag> p_99558_) {
-            if (p_99557_ instanceof CollectionTag collectiontag) {
+            if (p_99557_ instanceof CollectionTag<?> collectiontag) {
                 int i = collectiontag.size();
                 int j = this.index < 0 ? i + this.index : this.index;
                 if (0 <= j && j < i) {
-                    Tag tag = (Tag)collectiontag.get(j);
+                    Tag tag = collectiontag.get(j);
                     Tag tag1 = p_99558_.get();
                     if (!tag1.equals(tag) && collectiontag.setTag(j, tag1)) {
                         return 1;
@@ -346,7 +_,7 @@
 
         @Override
         public int removeTag(Tag p_99552_) {
-            if (p_99552_ instanceof CollectionTag collectiontag) {
+            if (p_99552_ instanceof CollectionTag<?> collectiontag) {
                 int i = collectiontag.size();
                 int j = this.index < 0 ? i + this.index : this.index;
                 if (0 <= j && j < i) {
@@ -403,14 +_,14 @@
                 int j = listtag.size();
                 if (j == 0) {
                     listtag.add(p_99579_.get());
-                    ++i;
+                    i++;
                 } else {
-                    for(int k = 0; k < j; ++k) {
+                    for (int k = 0; k < j; k++) {
                         Tag tag = listtag.get(k);
                         if (this.predicate.test(tag)) {
                             Tag tag1 = p_99579_.get();
                             if (!tag1.equals(tag) && listtag.setTag(k, tag1)) {
-                                ++i;
+                                i++;
                             }
                         }
                     }
@@ -424,10 +_,10 @@
         public int removeTag(Tag p_99573_) {
             int i = 0;
             if (p_99573_ instanceof ListTag listtag) {
-                for(int j = listtag.size() - 1; j >= 0; --j) {
+                for (int j = listtag.size() - 1; j >= 0; j--) {
                     if (this.predicate.test(listtag.get(j))) {
                         listtag.remove(j);
-                        ++i;
+                        i++;
                     }
                 }
             }
@@ -567,7 +_,7 @@
         public List<Tag> get(Tag p_99639_) throws CommandSyntaxException {
             List<Tag> list = Collections.singletonList(p_99639_);
 
-            for(NbtPathArgument.Node nbtpathargument$node : this.nodes) {
+            for (NbtPathArgument.Node nbtpathargument$node : this.nodes) {
                 list = nbtpathargument$node.get(list);
                 if (list.isEmpty()) {
                     throw this.createNotFoundException(nbtpathargument$node);
@@ -580,7 +_,7 @@
         public int countMatching(Tag p_99644_) {
             List<Tag> list = Collections.singletonList(p_99644_);
 
-            for(NbtPathArgument.Node nbtpathargument$node : this.nodes) {
+            for (NbtPathArgument.Node nbtpathargument$node : this.nodes) {
                 list = nbtpathargument$node.get(list);
                 if (list.isEmpty()) {
                     return 0;
@@ -593,7 +_,7 @@
         private List<Tag> getOrCreateParents(Tag p_99651_) throws CommandSyntaxException {
             List<Tag> list = Collections.singletonList(p_99651_);
 
-            for(int i = 0; i < this.nodes.length - 1; ++i) {
+            for (int i = 0; i < this.nodes.length - 1; i++) {
                 NbtPathArgument.Node nbtpathargument$node = this.nodes[i];
                 int j = i + 1;
                 list = nbtpathargument$node.getOrCreate(list, this.nodes[j]::createPreferredParentTag);
@@ -620,14 +_,14 @@
                 return true;
             } else {
                 if (p_263392_ instanceof CompoundTag compoundtag) {
-                    for(String s : compoundtag.getAllKeys()) {
+                    for (String s : compoundtag.getAllKeys()) {
                         Tag tag = compoundtag.get(s);
                         if (tag != null && isTooDeep(tag, p_263386_ + 1)) {
                             return true;
                         }
                     }
                 } else if (p_263392_ instanceof ListTag) {
-                    for(Tag tag1 : (ListTag)p_263392_) {
+                    for (Tag tag1 : (ListTag)p_263392_) {
                         if (isTooDeep(tag1, p_263386_ + 1)) {
                             return true;
                         }
@@ -668,7 +_,7 @@
         public int insert(int p_263397_, CompoundTag p_263348_, List<Tag> p_263419_) throws CommandSyntaxException {
             List<Tag> list = new ArrayList<>(p_263419_.size());
 
-            for(Tag tag : p_263419_) {
+            for (Tag tag : p_263419_) {
                 Tag tag1 = tag.copy();
                 list.add(tag1);
                 if (isTooDeep(tag1, this.estimatePathDepth())) {
@@ -680,19 +_,18 @@
             int j = 0;
             boolean flag1 = false;
 
-            for(Tag tag2 : collection) {
-                if (!(tag2 instanceof CollectionTag)) {
+            for (Tag tag2 : collection) {
+                if (!(tag2 instanceof CollectionTag<?> collectiontag)) {
                     throw NbtPathArgument.ERROR_EXPECTED_LIST.create(tag2);
                 }
 
-                CollectionTag<?> collectiontag = (CollectionTag)tag2;
                 boolean flag = false;
                 int i = p_263397_ < 0 ? collectiontag.size() + p_263397_ + 1 : p_263397_;
 
-                for(Tag tag3 : list) {
+                for (Tag tag3 : list) {
                     try {
                         if (collectiontag.addTag(i, flag1 ? tag3.copy() : tag3)) {
-                            ++i;
+                            i++;
                             flag = true;
                         }
                     } catch (IndexOutOfBoundsException indexoutofboundsexception) {
@@ -710,7 +_,7 @@
         public int remove(Tag p_99649_) {
             List<Tag> list = Collections.singletonList(p_99649_);
 
-            for(int i = 0; i < this.nodes.length - 1; ++i) {
+            for (int i = 0; i < this.nodes.length - 1; i++) {
                 list = this.nodes[i].get(list);
             }
 
@@ -755,7 +_,7 @@
         default List<Tag> collect(List<Tag> p_99656_, BiConsumer<Tag, List<Tag>> p_99657_) {
             List<Tag> list = Lists.newArrayList();
 
-            for(Tag tag : p_99656_) {
+            for (Tag tag : p_99656_) {
                 p_99657_.accept(tag, list);
             }
 
