--- a/net/minecraft/commands/arguments/ResourceOrTagKeyArgument.java
+++ b/net/minecraft/commands/arguments/ResourceOrTagKeyArgument.java
@@ -65,8 +_,7 @@
 
     @Override
     public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> p_251659_, SuggestionsBuilder p_251141_) {
-        Object object = p_251659_.getSource();
-        return object instanceof SharedSuggestionProvider sharedsuggestionprovider
+        return p_251659_.getSource() instanceof SharedSuggestionProvider sharedsuggestionprovider
             ? sharedsuggestionprovider.suggestRegistryElements(this.registryKey, SharedSuggestionProvider.ElementSuggestionType.ALL, p_251141_, p_251659_)
             : p_251141_.buildFuture();
     }
