--- a/net/minecraft/commands/arguments/ResourceKeyArgument.java
+++ b/net/minecraft/commands/arguments/ResourceKeyArgument.java
@@ -86,8 +_,7 @@
 
     @Override
     public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> p_212399_, SuggestionsBuilder p_212400_) {
-        Object object = p_212399_.getSource();
-        return object instanceof SharedSuggestionProvider sharedsuggestionprovider
+        return p_212399_.getSource() instanceof SharedSuggestionProvider sharedsuggestionprovider
             ? sharedsuggestionprovider.suggestRegistryElements(this.registryKey, SharedSuggestionProvider.ElementSuggestionType.ELEMENTS, p_212400_, p_212399_)
             : p_212400_.buildFuture();
     }
