--- a/net/minecraft/commands/arguments/ResourceOrIdArgument.java
+++ b/net/minecraft/commands/arguments/ResourceOrIdArgument.java
@@ -11,7 +_,6 @@
 import java.util.Collection;
 import java.util.List;
 import javax.annotation.Nullable;
-import net.minecraft.Util;
 import net.minecraft.commands.CommandBuildContext;
 import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.core.Holder;
@@ -83,7 +_,7 @@
             return null;
         } else {
             RegistryOps<Tag> registryops = this.registryLookup.createSerializationContext(NbtOps.INSTANCE);
-            return Util.getOrThrow(this.codec.parse(registryops, tag), p_335883_ -> ERROR_FAILED_TO_PARSE.createWithContext(p_335906_, p_335883_));
+            return this.codec.parse(registryops, tag).getOrThrow(p_335883_ -> ERROR_FAILED_TO_PARSE.createWithContext(p_335906_, p_335883_));
         }
     }
 
