--- a/net/minecraft/world/phys/Vec3.java
+++ b/net/minecraft/world/phys/Vec3.java
@@ -15,7 +_,7 @@
     public static final Codec<Vec3> CODEC = Codec.DOUBLE
         .listOf()
         .comapFlatMap(
-            p_231079_ -> Util.fixedSize(p_231079_, 3).map(p_231081_ -> new Vec3(p_231081_.get(0), p_231081_.get(1), p_231081_.get(2))),
+            p_338175_ -> Util.fixedSize((List<Double>)p_338175_, 3).map(p_231081_ -> new Vec3(p_231081_.get(0), p_231081_.get(1), p_231081_.get(2))),
             p_231083_ -> List.of(p_231083_.x(), p_231083_.y(), p_231083_.z())
         );
     public static final Vec3 ZERO = new Vec3(0.0, 0.0, 0.0);
@@ -169,17 +_,12 @@
     public boolean equals(Object p_82552_) {
         if (this == p_82552_) {
             return true;
-        } else if (!(p_82552_ instanceof Vec3)) {
+        } else if (!(p_82552_ instanceof Vec3 vec3)) {
+            return false;
+        } else if (Double.compare(vec3.x, this.x) != 0) {
             return false;
         } else {
-            Vec3 vec3 = (Vec3)p_82552_;
-            if (Double.compare(vec3.x, this.x) != 0) {
-                return false;
-            } else if (Double.compare(vec3.y, this.y) != 0) {
-                return false;
-            } else {
-                return Double.compare(vec3.z, this.z) == 0;
-            }
+            return Double.compare(vec3.y, this.y) != 0 ? false : Double.compare(vec3.z, this.z) == 0;
         }
     }
 
