--- a/net/minecraft/client/renderer/block/MovingBlockRenderState.java
+++ b/net/minecraft/client/renderer/block/MovingBlockRenderState.java
@@ -61,5 +62,10 @@
     @Override
     public int getMinY() {
         return this.blockPos.getY();
+    }
+
+    @Override
+    public net.neoforged.neoforge.model.data.ModelData getModelData(BlockPos pos) {
+        return pos.equals(this.blockPos) ? this.modelData : net.neoforged.neoforge.model.data.ModelData.EMPTY;
     }
 }
