--- a/net/minecraft/client/renderer/entity/DisplayRenderer.java
+++ b/net/minecraft/client/renderer/entity/DisplayRenderer.java
@@ -62,7 +_,7 @@
     private Quaternionf calculateOrientation(Display.RenderState p_277846_, T p_271013_, float p_294432_, Quaternionf p_295809_) {
         Camera camera = this.entityRenderDispatcher.camera;
 
-        return switch(p_277846_.billboardConstraints()) {
+        return switch (p_277846_.billboardConstraints()) {
             case FIXED -> p_295809_.rotationYXZ(
             (float) (-Math.PI / 180.0) * entityYRot(p_271013_, p_294432_), (float) (Math.PI / 180.0) * entityXRot(p_271013_, p_294432_), 0.0F
         );
@@ -73,6 +_,7 @@
             (float) (-Math.PI / 180.0) * cameraYrot(camera), (float) (Math.PI / 180.0) * entityXRot(p_271013_, p_294432_), 0.0F
         );
             case CENTER -> p_295809_.rotationYXZ((float) (-Math.PI / 180.0) * cameraYrot(camera), (float) (Math.PI / 180.0) * cameraXRot(camera), 0.0F);
+            default -> throw new MatchException(null, null);
         };
     }
 
@@ -174,7 +_,7 @@
             List<Display.TextDisplay.CachedLine> list1 = new ArrayList<>(list.size());
             int i = 0;
 
-            for(FormattedCharSequence formattedcharsequence : list) {
+            for (FormattedCharSequence formattedcharsequence : list) {
                 int j = this.font.width(formattedcharsequence);
                 i = Math.max(i, j);
                 list1.add(new Display.TextDisplay.CachedLine(formattedcharsequence, j));
@@ -227,11 +_,12 @@
                 vertexconsumer.vertex(matrix4f, (float)k, -1.0F, 0.0F).color(i).uv2(p_278046_).endVertex();
             }
 
-            for(Display.TextDisplay.CachedLine display$textdisplay$cachedline : display$textdisplay$cachedinfo.lines()) {
-                float f1 = switch(display$textdisplay$align) {
+            for (Display.TextDisplay.CachedLine display$textdisplay$cachedline : display$textdisplay$cachedinfo.lines()) {
+                float f1 = switch (display$textdisplay$align) {
                     case LEFT -> 0.0F;
                     case RIGHT -> (float)(k - display$textdisplay$cachedline.width());
                     case CENTER -> (float)k / 2.0F - (float)display$textdisplay$cachedline.width() / 2.0F;
+                    default -> throw new MatchException(null, null);
                 };
                 this.font
                     .drawInBatch(
