--- a/net/minecraft/client/renderer/entity/ElderGuardianRenderer.java
+++ b/net/minecraft/client/renderer/entity/ElderGuardianRenderer.java
@@ -1,7 +_,7 @@
 package net.minecraft.client.renderer.entity;
 
 import com.mojang.blaze3d.vertex.PoseStack;
-import net.minecraft.client.model.geom.ModelLayerLocation;
+import net.minecraft.client.model.geom.ModelLayers;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.entity.monster.ElderGuardian;
 import net.minecraft.world.entity.monster.Guardian;
@@ -11,10 +_,9 @@
 @OnlyIn(Dist.CLIENT)
 public class ElderGuardianRenderer extends GuardianRenderer {
     public static final ResourceLocation GUARDIAN_ELDER_LOCATION = new ResourceLocation("textures/entity/guardian_elder.png");
-    public static final ResourceLocation PLAGUEWHALE_LOCATION = new ResourceLocation("textures/entity/plaguewhale.png");
 
-    public ElderGuardianRenderer(EntityRendererProvider.Context p_173966_, ModelLayerLocation p_342079_) {
-        super(p_173966_, 1.2F, p_342079_);
+    public ElderGuardianRenderer(EntityRendererProvider.Context p_173966_) {
+        super(p_173966_, 1.2F, ModelLayers.ELDER_GUARDIAN);
     }
 
     protected void scale(Guardian p_114129_, PoseStack p_114130_, float p_114131_) {
@@ -23,6 +_,6 @@
 
     @Override
     public ResourceLocation getTextureLocation(Guardian p_114127_) {
-        return p_114127_.isToxic() ? PLAGUEWHALE_LOCATION : GUARDIAN_ELDER_LOCATION;
+        return GUARDIAN_ELDER_LOCATION;
     }
 }
