--- a/net/minecraft/client/renderer/entity/EnderDragonRenderer.java
+++ b/net/minecraft/client/renderer/entity/EnderDragonRenderer.java
@@ -74,7 +_,7 @@
             p_114211_.pushPose();
             p_114211_.translate(0.0F, -1.0F, -2.0F);
 
-            for(int i = 0; (float)i < (f5 + f5 * f5) / 2.0F * 60.0F; ++i) {
+            for (int i = 0; (float)i < (f5 + f5 * f5) / 2.0F * 60.0F; i++) {
                 p_114211_.mulPose(Axis.XP.rotationDegrees(randomsource.nextFloat() * 360.0F));
                 p_114211_.mulPose(Axis.YP.rotationDegrees(randomsource.nextFloat() * 360.0F));
                 p_114211_.mulPose(Axis.ZP.rotationDegrees(randomsource.nextFloat() * 360.0F));
@@ -148,7 +_,7 @@
         float f6 = 0.0F;
         PoseStack.Pose posestack$pose = p_114193_.last();
 
-        for(int j = 1; j <= 8; ++j) {
+        for (int j = 1; j <= 8; j++) {
             float f7 = Mth.sin((float)j * (float) (Math.PI * 2) / 8.0F) * 0.75F;
             float f8 = Mth.cos((float)j * (float) (Math.PI * 2) / 8.0F) * 0.75F;
             float f9 = (float)j / 8.0F;
@@ -362,7 +_,7 @@
             float f7 = Mth.wrapDegrees((float)(this.entity.getLatencyPos(5, this.a)[0] + (double)(f6 / 2.0F)));
             float f8 = f * (float) (Math.PI * 2);
 
-            for(int i = 0; i < 5; ++i) {
+            for (int i = 0; i < 5; i++) {
                 double[] adouble1 = this.entity.getLatencyPos(5 - i, this.a);
                 float f9 = (float)Math.cos((double)((float)i * 0.45F + f8)) * 0.15F;
                 this.neck.yRot = Mth.wrapDegrees((float)(adouble1[0] - adouble[0])) * (float) (Math.PI / 180.0) * 1.5F;
@@ -438,7 +_,7 @@
             f2 = 0.0F;
             adouble = this.entity.getLatencyPos(11, this.a);
 
-            for(int j = 0; j < 12; ++j) {
+            for (int j = 0; j < 12; j++) {
                 adouble2 = this.entity.getLatencyPos(12 + j, this.a);
                 f11 += Mth.sin((float)j * 0.45F + f8) * 0.05F;
                 this.neck.yRot = (Mth.wrapDegrees((float)(adouble2[0] - adouble[0])) * 1.5F + 180.0F) * (float) (Math.PI / 180.0);
