--- a/net/minecraft/client/renderer/entity/EntityRenderers.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderers.java
@@ -32,11 +_,11 @@
 
     public static Map<EntityType<?>, EntityRenderer<?>> createEntityRenderers(EntityRendererProvider.Context p_174050_) {
         Builder<EntityType<?>, EntityRenderer<?>> builder = ImmutableMap.builder();
-        PROVIDERS.forEach((p_258154_, p_258155_) -> {
+        PROVIDERS.forEach((p_339310_, p_339311_) -> {
             try {
-                builder.put(p_258154_, p_258155_.create(p_174050_));
+                builder.put((EntityType<?>)p_339310_, p_339311_.create(p_174050_));
             } catch (Exception exception) {
-                throw new IllegalArgumentException("Failed to create model for " + BuiltInRegistries.ENTITY_TYPE.getKey(p_258154_), exception);
+                throw new IllegalArgumentException("Failed to create model for " + BuiltInRegistries.ENTITY_TYPE.getKey((EntityType<?>)p_339310_), exception);
             }
         });
         return builder.build();
@@ -44,11 +_,11 @@
 
     public static Map<PlayerSkin.Model, EntityRenderer<? extends Player>> createPlayerRenderers(EntityRendererProvider.Context p_174052_) {
         Builder<PlayerSkin.Model, EntityRenderer<? extends Player>> builder = ImmutableMap.builder();
-        PLAYER_PROVIDERS.forEach((p_295831_, p_234608_) -> {
+        PLAYER_PROVIDERS.forEach((p_339306_, p_339307_) -> {
             try {
-                builder.put(p_295831_, p_234608_.create(p_174052_));
+                builder.put(p_339306_, p_339307_.create(p_174052_));
             } catch (Exception exception) {
-                throw new IllegalArgumentException("Failed to create player model for " + p_295831_, exception);
+                throw new IllegalArgumentException("Failed to create player model for " + p_339306_, exception);
             }
         });
         return builder.build();
@@ -57,7 +_,7 @@
     public static boolean validateRegistrations() {
         boolean flag = true;
 
-        for(EntityType<?> entitytype : BuiltInRegistries.ENTITY_TYPE) {
+        for (EntityType<?> entitytype : BuiltInRegistries.ENTITY_TYPE) {
             if (entitytype != EntityType.PLAYER && !PROVIDERS.containsKey(entitytype)) {
                 LOGGER.warn("No renderer registered for {}", BuiltInRegistries.ENTITY_TYPE.getKey(entitytype));
                 flag = false;
@@ -73,10 +_,8 @@
         register(EntityType.ARMADILLO, ArmadilloRenderer::new);
         register(EntityType.ARMOR_STAND, ArmorStandRenderer::new);
         register(EntityType.ARROW, TippableArrowRenderer::new);
-        register(EntityType.VINE_PROJECTILE, NoopRenderer::new);
         register(EntityType.AXOLOTL, AxolotlRenderer::new);
         register(EntityType.BAT, BatRenderer::new);
-        register(EntityType.BATATO, BatatoRenderer::new);
         register(EntityType.BEE, BeeRenderer::new);
         register(EntityType.BLAZE, BlazeRenderer::new);
         register(EntityType.BLOCK_DISPLAY, DisplayRenderer.BlockDisplayRenderer::new);
@@ -99,7 +_,7 @@
         register(EntityType.DRAGON_FIREBALL, DragonFireballRenderer::new);
         register(EntityType.DROWNED, DrownedRenderer::new);
         register(EntityType.EGG, ThrownItemRenderer::new);
-        register(EntityType.ELDER_GUARDIAN, p_339522_ -> new ElderGuardianRenderer(p_339522_, ModelLayers.ELDER_GUARDIAN));
+        register(EntityType.ELDER_GUARDIAN, ElderGuardianRenderer::new);
         register(EntityType.ENDERMAN, EndermanRenderer::new);
         register(EntityType.ENDERMITE, EndermiteRenderer::new);
         register(EntityType.ENDER_DRAGON, EnderDragonRenderer::new);
@@ -109,12 +_,11 @@
         register(EntityType.EVOKER_FANGS, EvokerFangsRenderer::new);
         register(EntityType.EXPERIENCE_BOTTLE, ThrownItemRenderer::new);
         register(EntityType.EXPERIENCE_ORB, ExperienceOrbRenderer::new);
-        register(EntityType.EYE_OF_ENDER, p_339523_ -> new ThrownItemRenderer<>(p_339523_, 1.0F, true));
+        register(EntityType.EYE_OF_ENDER, p_174084_ -> new ThrownItemRenderer<>(p_174084_, 1.0F, true));
         register(EntityType.FALLING_BLOCK, FallingBlockRenderer::new);
-        register(EntityType.FIREBALL, p_174060_ -> new ThrownItemRenderer<>(p_174060_, 3.0F, true));
+        register(EntityType.FIREBALL, p_174082_ -> new ThrownItemRenderer<>(p_174082_, 3.0F, true));
         register(EntityType.FIREWORK_ROCKET, FireworkEntityRenderer::new);
         register(EntityType.FISHING_BOBBER, FishingHookRenderer::new);
-        register(EntityType.LASHING_POTATO_HOOK, LashingPotatoHookRenderer::new);
         register(EntityType.FOX, FoxRenderer::new);
         register(EntityType.FROG, FrogRenderer::new);
         register(EntityType.FURNACE_MINECART, p_174080_ -> new MinecartRenderer<>(p_174080_, ModelLayers.FURNACE_MINECART));
@@ -123,7 +_,7 @@
         register(EntityType.GLOW_ITEM_FRAME, ItemFrameRenderer::new);
         register(EntityType.GLOW_SQUID, p_174076_ -> new GlowSquidRenderer(p_174076_, new SquidModel<>(p_174076_.bakeLayer(ModelLayers.GLOW_SQUID))));
         register(EntityType.GOAT, GoatRenderer::new);
-        register(EntityType.GUARDIAN, p_339524_ -> new GuardianRenderer(p_339524_, ModelLayers.GUARDIAN));
+        register(EntityType.GUARDIAN, GuardianRenderer::new);
         register(EntityType.HOGLIN, HoglinRenderer::new);
         register(EntityType.HOPPER_MINECART, p_174074_ -> new MinecartRenderer<>(p_174074_, ModelLayers.HOPPER_MINECART));
         register(EntityType.HORSE, HorseRenderer::new);
@@ -134,6 +_,7 @@
         register(EntityType.ITEM, ItemEntityRenderer::new);
         register(EntityType.ITEM_DISPLAY, DisplayRenderer.ItemDisplayRenderer::new);
         register(EntityType.ITEM_FRAME, ItemFrameRenderer::new);
+        register(EntityType.OMINOUS_ITEM_SPAWNER, OminousItemSpawnerRenderer::new);
         register(EntityType.LEASH_KNOT, LeashKnotRenderer::new);
         register(EntityType.LIGHTNING_BOLT, LightningBoltRenderer::new);
         register(EntityType.LLAMA, p_174072_ -> new LlamaRenderer(p_174072_, ModelLayers.LLAMA));
@@ -160,7 +_,6 @@
                 )
         );
         register(EntityType.PILLAGER, PillagerRenderer::new);
-        register(EntityType.PLAGUEWHALE, p_339525_ -> new ElderGuardianRenderer(p_339525_, ModelLayers.ELDER_GUARDIAN_SLAB));
         register(EntityType.POLAR_BEAR, PolarBearRenderer::new);
         register(EntityType.POTION, ThrownItemRenderer::new);
         register(EntityType.PUFFERFISH, PufferfishRenderer::new);
@@ -174,8 +_,7 @@
         register(EntityType.SKELETON, SkeletonRenderer::new);
         register(EntityType.SKELETON_HORSE, p_174062_ -> new UndeadHorseRenderer(p_174062_, ModelLayers.SKELETON_HORSE));
         register(EntityType.SLIME, SlimeRenderer::new);
-        register(EntityType.MEGA_SPUD, MegaSpudRenderer::new);
-        register(EntityType.SMALL_FIREBALL, p_174082_ -> new ThrownItemRenderer<>(p_174082_, 0.75F, true));
+        register(EntityType.SMALL_FIREBALL, p_174060_ -> new ThrownItemRenderer<>(p_174060_, 0.75F, true));
         register(EntityType.SNIFFER, SnifferRenderer::new);
         register(EntityType.SNOWBALL, ThrownItemRenderer::new);
         register(EntityType.SNOW_GOLEM, SnowGolemRenderer::new);
@@ -189,7 +_,6 @@
         register(EntityType.TEXT_DISPLAY, DisplayRenderer.TextDisplayRenderer::new);
         register(EntityType.TNT, TntRenderer::new);
         register(EntityType.TNT_MINECART, TntMinecartRenderer::new);
-        register(EntityType.TOXIFIN, p_339526_ -> new GuardianRenderer(p_339526_, ModelLayers.GUARDIAN_SLAB));
         register(EntityType.TRADER_LLAMA, p_174054_ -> new LlamaRenderer(p_174054_, ModelLayers.TRADER_LLAMA));
         register(EntityType.TRIDENT, ThrownTridentRenderer::new);
         register(EntityType.TROPICAL_FISH, TropicalFishRenderer::new);
@@ -207,7 +_,6 @@
         register(EntityType.WOLF, WolfRenderer::new);
         register(EntityType.ZOGLIN, ZoglinRenderer::new);
         register(EntityType.ZOMBIE, ZombieRenderer::new);
-        register(EntityType.POISONOUS_POTATO_ZOMBIE, PoisonousPotatoZombieRenderer::new);
         register(EntityType.ZOMBIE_HORSE, p_234612_ -> new UndeadHorseRenderer(p_234612_, ModelLayers.ZOMBIE_HORSE));
         register(EntityType.ZOMBIE_VILLAGER, ZombieVillagerRenderer::new);
         register(
@@ -216,7 +_,5 @@
                     p_234610_, ModelLayers.ZOMBIFIED_PIGLIN, ModelLayers.ZOMBIFIED_PIGLIN_INNER_ARMOR, ModelLayers.ZOMBIFIED_PIGLIN_OUTER_ARMOR, true
                 )
         );
-        register(EntityType.GRID_CARRIER, GridCarrierEntityRenderer::new);
-        register(EntityType.EYE_OF_POTATO, p_174084_ -> new ThrownItemRenderer<>(p_174084_, 1.0F, true));
     }
 }
