--- a/net/minecraft/client/renderer/entity/FallingBlockRenderer.java
+++ b/net/minecraft/client/renderer/entity/FallingBlockRenderer.java
@@ -10,7 +_,6 @@
 import net.minecraft.world.entity.item.FallingBlockEntity;
 import net.minecraft.world.level.block.RenderShape;
 import net.minecraft.world.level.block.state.BlockState;
-import net.minecraft.world.phys.Vec3;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
@@ -44,7 +_,7 @@
         super.extractRenderState(entity, state, partialTicks);
         BlockPos pos = BlockPos.containing(entity.getX(), entity.getBoundingBox().maxY, entity.getZ());
         state.movingBlockRenderState.randomSeedPos = entity.getStartPos();
-        state.movingBlockRenderState.pos = new Vec3(pos);
+        state.movingBlockRenderState.blockPos = pos;
         state.movingBlockRenderState.blockState = entity.getBlockState();
         if (entity.level() instanceof ClientLevel clientLevel) {
             state.movingBlockRenderState.biome = clientLevel.getBiome(pos);
