--- a/net/minecraft/client/renderer/entity/FishingHookRenderer.java
+++ b/net/minecraft/client/renderer/entity/FishingHookRenderer.java
@@ -12,7 +_,6 @@
 import net.minecraft.world.entity.HumanoidArm;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.entity.projectile.FishingHook;
-import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.phys.Vec3;
@@ -44,58 +_,57 @@
             vertex(vertexconsumer, posestack$pose, p_114710_, 1.0F, 1, 1, 0);
             vertex(vertexconsumer, posestack$pose, p_114710_, 0.0F, 1, 0, 0);
             p_114708_.popPose();
-            Vec3 vec3 = getPlayerHandPos(player, p_114707_, Items.FISHING_ROD, this.entityRenderDispatcher);
-            double d0 = Mth.lerp((double)p_114707_, p_114705_.xo, p_114705_.getX());
-            double d1 = Mth.lerp((double)p_114707_, p_114705_.yo, p_114705_.getY()) + 0.25;
-            double d2 = Mth.lerp((double)p_114707_, p_114705_.zo, p_114705_.getZ());
-            float f = (float)(vec3.x - d0);
-            float f1 = (float)(vec3.y - d1);
-            float f2 = (float)(vec3.z - d2);
+            int i = player.getMainArm() == HumanoidArm.RIGHT ? 1 : -1;
+            ItemStack itemstack = player.getMainHandItem();
+            if (!itemstack.canPerformAction(net.neoforged.neoforge.common.ToolActions.FISHING_ROD_CAST)) {
+                i = -i;
+            }
+
+            float f = player.getAttackAnim(p_114707_);
+            float f1 = Mth.sin(Mth.sqrt(f) * (float) Math.PI);
+            float f2 = Mth.lerp(p_114707_, player.yBodyRotO, player.yBodyRot) * (float) (Math.PI / 180.0);
+            double d0 = (double)Mth.sin(f2);
+            double d1 = (double)Mth.cos(f2);
+            double d2 = (double)i * 0.35;
+            double d3 = 0.8;
+            double d4;
+            double d5;
+            double d6;
+            float f3;
+            if ((this.entityRenderDispatcher.options == null || this.entityRenderDispatcher.options.getCameraType().isFirstPerson())
+                && player == Minecraft.getInstance().player) {
+                double d7 = 960.0 / (double)this.entityRenderDispatcher.options.fov().get().intValue();
+                Vec3 vec3 = this.entityRenderDispatcher.camera.getNearPlane().getPointOnPlane((float)i * 0.525F, -0.1F);
+                vec3 = vec3.scale(d7);
+                vec3 = vec3.yRot(f1 * 0.5F);
+                vec3 = vec3.xRot(-f1 * 0.7F);
+                d4 = Mth.lerp((double)p_114707_, player.xo, player.getX()) + vec3.x;
+                d5 = Mth.lerp((double)p_114707_, player.yo, player.getY()) + vec3.y;
+                d6 = Mth.lerp((double)p_114707_, player.zo, player.getZ()) + vec3.z;
+                f3 = player.getEyeHeight();
+            } else {
+                d4 = Mth.lerp((double)p_114707_, player.xo, player.getX()) - d1 * d2 - d0 * 0.8;
+                d5 = player.yo + (double)player.getEyeHeight() + (player.getY() - player.yo) * (double)p_114707_ - 0.45;
+                d6 = Mth.lerp((double)p_114707_, player.zo, player.getZ()) - d0 * d2 + d1 * 0.8;
+                f3 = player.isCrouching() ? -0.1875F : 0.0F;
+            }
+
+            double d9 = Mth.lerp((double)p_114707_, p_114705_.xo, p_114705_.getX());
+            double d10 = Mth.lerp((double)p_114707_, p_114705_.yo, p_114705_.getY()) + 0.25;
+            double d8 = Mth.lerp((double)p_114707_, p_114705_.zo, p_114705_.getZ());
+            float f4 = (float)(d4 - d9);
+            float f5 = (float)(d5 - d10) + f3;
+            float f6 = (float)(d6 - d8);
             VertexConsumer vertexconsumer1 = p_114709_.getBuffer(RenderType.lineStrip());
             PoseStack.Pose posestack$pose1 = p_114708_.last();
-            int i = 16;
+            int j = 16;
 
-            for(int j = 0; j <= 16; ++j) {
-                stringVertex(f, f1, f2, vertexconsumer1, posestack$pose1, fraction(j, 16), fraction(j + 1, 16));
+            for (int k = 0; k <= 16; k++) {
+                stringVertex(f4, f5, f6, vertexconsumer1, posestack$pose1, fraction(k, 16), fraction(k + 1, 16));
             }
 
             p_114708_.popPose();
             super.render(p_114705_, p_114706_, p_114707_, p_114708_, p_114709_, p_114710_);
-        }
-    }
-
-    public static Vec3 getPlayerHandPos(Player p_341593_, float p_340319_, Item p_341350_, EntityRenderDispatcher p_341461_) {
-        int i = p_341593_.getMainArm() == HumanoidArm.RIGHT ? 1 : -1;
-        ItemStack itemstack = p_341593_.getMainHandItem();
-        if (!itemstack.is(p_341350_)) {
-            i = -i;
-        }
-
-        float f = p_341593_.getAttackAnim(p_340319_);
-        float f1 = Mth.sin(Mth.sqrt(f) * (float) Math.PI);
-        float f2 = Mth.lerp(p_340319_, p_341593_.yBodyRotO, p_341593_.yBodyRot) * (float) (Math.PI / 180.0);
-        double d0 = (double)Mth.sin(f2);
-        double d1 = (double)Mth.cos(f2);
-        double d2 = (double)i * 0.35;
-        double d3 = 0.8;
-        if ((p_341461_.options == null || p_341461_.options.getCameraType().isFirstPerson()) && p_341593_ == Minecraft.getInstance().player) {
-            double d4 = 960.0 / (double)p_341461_.options.fov().get().intValue();
-            Vec3 vec3 = p_341461_.camera.getNearPlane().getPointOnPlane((float)i * 0.525F, -0.1F);
-            vec3 = vec3.scale(d4);
-            vec3 = vec3.yRot(f1 * 0.5F);
-            vec3 = vec3.xRot(-f1 * 0.7F);
-            return new Vec3(
-                Mth.lerp((double)p_340319_, p_341593_.xo, p_341593_.getX()) + vec3.x,
-                Mth.lerp((double)p_340319_, p_341593_.yo, p_341593_.getY()) + vec3.y + (double)p_341593_.getEyeHeight(),
-                Mth.lerp((double)p_340319_, p_341593_.zo, p_341593_.getZ()) + vec3.z
-            );
-        } else {
-            float f3 = p_341593_.isCrouching() ? -0.1875F : 0.0F;
-            return new Vec3(
-                Mth.lerp((double)p_340319_, p_341593_.xo, p_341593_.getX()) - d1 * d2 - d0 * 0.8,
-                p_341593_.yo + (double)p_341593_.getEyeHeight() + (p_341593_.getY() - p_341593_.yo) * (double)p_340319_ - 0.45 + (double)f3,
-                Mth.lerp((double)p_340319_, p_341593_.zo, p_341593_.getZ()) - d0 * d2 + d1 * 0.8
-            );
         }
     }
 
