--- a/net/minecraft/client/renderer/entity/IllusionerRenderer.java
+++ b/net/minecraft/client/renderer/entity/IllusionerRenderer.java
@@ -50,7 +_,7 @@
             Vec3[] avec3 = p_114952_.getIllusionOffsets(p_114954_);
             float f = this.getBob(p_114952_, p_114954_);
 
-            for(int i = 0; i < avec3.length; ++i) {
+            for (int i = 0; i < avec3.length; i++) {
                 p_114955_.pushPose();
                 p_114955_.translate(
                     avec3[i].x + (double)Mth.cos((float)i + f * 0.5F) * 0.025,
