--- a/net/minecraft/client/renderer/entity/LivingEntityRenderer.java
+++ b/net/minecraft/client/renderer/entity/LivingEntityRenderer.java
@@ -18,7 +_,6 @@
 import net.minecraft.core.Direction;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.Mth;
-import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.Pose;
 import net.minecraft.world.entity.player.Player;
@@ -51,34 +_,33 @@
     }
 
     public void render(T p_115308_, float p_115309_, float p_115310_, PoseStack p_115311_, MultiBufferSource p_115312_, int p_115313_) {
+        if (net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLivingEvent.Pre<T, M>(p_115308_, this, p_115310_, p_115311_, p_115312_, p_115313_)).isCanceled()) return;
         p_115311_.pushPose();
         this.model.attackTime = this.getAttackAnim(p_115308_, p_115310_);
-        this.model.riding = p_115308_.isPassenger();
+        boolean shouldSit = p_115308_.isPassenger() && (p_115308_.getVehicle() != null && p_115308_.getVehicle().shouldRiderSit());
+        this.model.riding = shouldSit;
         this.model.young = p_115308_.isBaby();
         float f = Mth.rotLerp(p_115310_, p_115308_.yBodyRotO, p_115308_.yBodyRot);
         float f1 = Mth.rotLerp(p_115310_, p_115308_.yHeadRotO, p_115308_.yHeadRot);
         float f2 = f1 - f;
-        if (p_115308_.isPassenger()) {
-            Entity $$14 = p_115308_.getVehicle();
-            if ($$14 instanceof LivingEntity livingentity) {
-                f = Mth.rotLerp(p_115310_, livingentity.yBodyRotO, livingentity.yBodyRot);
-                f2 = f1 - f;
-                float f7 = Mth.wrapDegrees(f2);
-                if (f7 < -85.0F) {
-                    f7 = -85.0F;
-                }
-
-                if (f7 >= 85.0F) {
-                    f7 = 85.0F;
-                }
-
-                f = f1 - f7;
-                if (f7 * f7 > 2500.0F) {
-                    f += f7 * 0.2F;
-                }
-
-                f2 = f1 - f;
-            }
+        if (shouldSit && p_115308_.getVehicle() instanceof LivingEntity livingentity) {
+            f = Mth.rotLerp(p_115310_, livingentity.yBodyRotO, livingentity.yBodyRot);
+            f2 = f1 - f;
+            float f7 = Mth.wrapDegrees(f2);
+            if (f7 < -85.0F) {
+                f7 = -85.0F;
+            }
+
+            if (f7 >= 85.0F) {
+                f7 = 85.0F;
+            }
+
+            f = f1 - f7;
+            if (f7 * f7 > 2500.0F) {
+                f += f7 * 0.2F;
+            }
+
+            f2 = f1 - f;
         }
 
         float f6 = Mth.lerp(p_115310_, p_115308_.xRotO, p_115308_.getXRot());
@@ -105,7 +_,7 @@
         p_115311_.translate(0.0F, -1.501F, 0.0F);
         float f4 = 0.0F;
         float f5 = 0.0F;
-        if (!p_115308_.isPassenger() && p_115308_.isAlive()) {
+        if (!shouldSit && p_115308_.isAlive()) {
             f4 = p_115308_.walkAnimation.speed(p_115310_);
             f5 = p_115308_.walkAnimation.position(p_115310_);
             if (p_115308_.isBaby()) {
@@ -131,26 +_,19 @@
         }
 
         if (!p_115308_.isSpectator()) {
-            for(RenderLayer<T, M> renderlayer : this.layers) {
+            for (RenderLayer<T, M> renderlayer : this.layers) {
                 renderlayer.render(p_115311_, p_115312_, p_115313_, p_115308_, f5, f4, p_115310_, f9, f2, f6);
             }
         }
 
         p_115311_.popPose();
         super.render(p_115308_, p_115309_, p_115310_, p_115311_, p_115312_, p_115313_);
-    }
-
-    public static ResourceLocation potatoify(ResourceLocation p_341075_) {
-        return p_341075_.withPath(p_341138_ -> p_341138_.replaceFirst(".png$", "_potato.png"));
+        net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderLivingEvent.Post<T, M>(p_115308_, this, p_115310_, p_115311_, p_115312_, p_115313_));
     }
 
     @Nullable
     protected RenderType getRenderType(T p_115322_, boolean p_115323_, boolean p_115324_, boolean p_115325_) {
         ResourceLocation resourcelocation = this.getTextureLocation(p_115322_);
-        if (p_115322_.hasPotatoVariant() && p_115322_.isPotato()) {
-            resourcelocation = potatoify(resourcelocation);
-        }
-
         if (p_115324_) {
             return RenderType.itemEntityTranslucentCull(resourcelocation);
         } else if (p_115323_) {
@@ -169,7 +_,7 @@
     }
 
     private static float sleepDirectionToRotation(Direction p_115329_) {
-        switch(p_115329_) {
+        switch (p_115329_) {
             case SOUTH:
                 return 90.0F;
             case WEST:
@@ -252,7 +_,7 @@
                 Team team1 = localplayer.getTeam();
                 if (team != null) {
                     Team.Visibility team$visibility = team.getNameTagVisibility();
-                    switch(team$visibility) {
+                    switch (team$visibility) {
                         case ALWAYS:
                             return flag;
                         case NEVER:
