--- a/net/minecraft/client/renderer/entity/MobRenderer.java
+++ b/net/minecraft/client/renderer/entity/MobRenderer.java
@@ -74,11 +_,11 @@
         int k = p_115462_.level().getBrightness(LightLayer.SKY, blockpos);
         int l = p_115462_.level().getBrightness(LightLayer.SKY, blockpos1);
 
-        for(int i1 = 0; i1 <= 24; ++i1) {
+        for (int i1 = 0; i1 <= 24; i1++) {
             addVertexPair(vertexconsumer, matrix4f, f, f1, f2, i, j, k, l, 0.025F, 0.025F, f5, f6, i1, false);
         }
 
-        for(int j1 = 24; j1 >= 0; --j1) {
+        for (int j1 = 24; j1 >= 0; j1--) {
             addVertexPair(vertexconsumer, matrix4f, f, f1, f2, i, j, k, l, 0.025F, 0.0F, f5, f6, j1, true);
         }
 
