--- a/net/minecraft/client/renderer/entity/PaintingRenderer.java
+++ b/net/minecraft/client/renderer/entity/PaintingRenderer.java
@@ -80,8 +_,8 @@
         double d0 = 1.0 / (double)i;
         double d1 = 1.0 / (double)j;
 
-        for(int k = 0; k < i; ++k) {
-            for(int l = 0; l < j; ++l) {
+        for (int k = 0; k < i; k++) {
+            for (int l = 0; l < j; l++) {
                 float f15 = f + (float)((k + 1) * 16);
                 float f16 = f + (float)(k * 16);
                 float f17 = f1 + (float)((l + 1) * 16);
