--- a/net/minecraft/client/renderer/entity/ShulkerRenderer.java
+++ b/net/minecraft/client/renderer/entity/ShulkerRenderer.java
@@ -25,7 +_,7 @@
     private static final ResourceLocation[] TEXTURE_LOCATION = Sheets.SHULKER_TEXTURE_LOCATION
         .stream()
         .map(p_115919_ -> new ResourceLocation("textures/" + p_115919_.texture().getPath() + ".png"))
-        .toArray(p_115877_ -> new ResourceLocation[p_115877_]);
+        .toArray(ResourceLocation[]::new);
 
     public ShulkerRenderer(EntityRendererProvider.Context p_174370_) {
         super(p_174370_, new ShulkerModel<>(p_174370_.bakeLayer(ModelLayers.SHULKER)), 0.0F);
