--- a/net/minecraft/client/renderer/entity/SkeletonRenderer.java
+++ b/net/minecraft/client/renderer/entity/SkeletonRenderer.java
@@ -11,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SkeletonRenderer<T extends AbstractSkeleton> extends HumanoidMobRenderer<T, SkeletonModel<T>> {
-    public static final ResourceLocation SKELETON_LOCATION = new ResourceLocation("textures/entity/skeleton/skeleton.png");
+    private static final ResourceLocation SKELETON_LOCATION = new ResourceLocation("textures/entity/skeleton/skeleton.png");
 
     public SkeletonRenderer(EntityRendererProvider.Context p_174380_) {
         this(p_174380_, ModelLayers.SKELETON, ModelLayers.SKELETON_INNER_ARMOR, ModelLayers.SKELETON_OUTER_ARMOR);
