--- a/net/minecraft/client/renderer/entity/TropicalFishRenderer.java
+++ b/net/minecraft/client/renderer/entity/TropicalFishRenderer.java
@@ -28,16 +_,18 @@
     }
 
     public ResourceLocation getTextureLocation(TropicalFish p_116217_) {
-        return switch(p_116217_.getVariant().base()) {
+        return switch (p_116217_.getVariant().base()) {
             case SMALL -> MODEL_A_TEXTURE;
             case LARGE -> MODEL_B_TEXTURE;
+            default -> throw new MatchException(null, null);
         };
     }
 
     public void render(TropicalFish p_116219_, float p_116220_, float p_116221_, PoseStack p_116222_, MultiBufferSource p_116223_, int p_116224_) {
-        ColorableHierarchicalModel<TropicalFish> colorablehierarchicalmodel = switch(p_116219_.getVariant().base()) {
+        ColorableHierarchicalModel<TropicalFish> colorablehierarchicalmodel = switch (p_116219_.getVariant().base()) {
             case SMALL -> this.modelA;
             case LARGE -> this.modelB;
+            default -> throw new MatchException(null, null);
         };
         this.model = colorablehierarchicalmodel;
         float[] afloat = p_116219_.getBaseColor().getTextureDiffuseColors();
