--- a/net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen.java
@@ -2,8 +_,10 @@
 
 import it.unimi.dsi.fastutil.booleans.BooleanConsumer;
 import java.util.Collection;
+import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.MultiLineLabel;
 import net.minecraft.client.gui.components.MultiLineTextWidget;
@@ -63,7 +_,9 @@
         );
         gridlayout$rowhelper.addChild(Button.builder(CommonComponents.GUI_PROCEED, p_252248_ -> this.callback.accept(true)).build());
         gridlayout$rowhelper.addChild(Button.builder(CommonComponents.GUI_BACK, p_250397_ -> this.callback.accept(false)).build());
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_321368_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_321368_);
+        });
         this.layout.arrangeElements();
         this.repositionElements();
     }
@@ -82,6 +_,8 @@
     class DetailsScreen extends Screen {
         private static final Component TITLE = Component.translatable("selectWorld.experimental.details.title");
         final HeaderAndFooterLayout layout = new HeaderAndFooterLayout(this);
+        @Nullable
+        private ConfirmExperimentalFeaturesScreen.DetailsScreen.PackList list;
 
         DetailsScreen() {
             super(TITLE);
@@ -90,17 +_,23 @@
         @Override
         protected void init() {
             this.layout.addTitleHeader(TITLE, this.font);
-            this.layout
+            this.list = this.layout
                 .addToContents(
                     new ConfirmExperimentalFeaturesScreen.DetailsScreen.PackList(this.minecraft, ConfirmExperimentalFeaturesScreen.this.enabledPacks)
                 );
             this.layout.addToFooter(Button.builder(CommonComponents.GUI_BACK, p_251286_ -> this.onClose()).build());
-            this.layout.visitWidgets(this::addRenderableWidget);
+            this.layout.visitWidgets(p_329748_ -> {
+                AbstractWidget abstractwidget = this.addRenderableWidget(p_329748_);
+            });
             this.repositionElements();
         }
 
         @Override
         protected void repositionElements() {
+            if (this.list != null) {
+                this.list.updateSize(this.width, this.layout);
+            }
+
             this.layout.arrangeElements();
         }
 
@@ -112,20 +_,14 @@
         @OnlyIn(Dist.CLIENT)
         class PackList extends ObjectSelectionList<ConfirmExperimentalFeaturesScreen.DetailsScreen.PackListEntry> {
             public PackList(Minecraft p_249776_, Collection<Pack> p_251183_) {
-                super(
-                    p_249776_, DetailsScreen.this.width, DetailsScreen.this.layout.getContentHeight(), DetailsScreen.this.layout.getHeaderHeight(), (9 + 2) * 3
-                );
+                super(p_249776_, DetailsScreen.this.width, DetailsScreen.this.layout.getContentHeight(), DetailsScreen.this.layout.getHeaderHeight(), (9 + 2) * 3);
 
-                for(Pack pack : p_251183_) {
+                for (Pack pack : p_251183_) {
                     String s = FeatureFlags.printMissingFlags(FeatureFlags.VANILLA_SET, pack.getRequestedFeatures());
                     if (!s.isEmpty()) {
                         Component component = ComponentUtils.mergeStyles(pack.getTitle().copy(), Style.EMPTY.withBold(true));
                         Component component1 = Component.translatable("selectWorld.experimental.details.entry", s);
-                        this.addEntry(
-                            DetailsScreen.this.new PackListEntry(
-                                component, component1, MultiLineLabel.create(DetailsScreen.this.font, component1, this.getRowWidth())
-                            )
-                        );
+                        this.addEntry(DetailsScreen.this.new PackListEntry(component, component1, MultiLineLabel.create(DetailsScreen.this.font, component1, this.getRowWidth())));
                     }
                 }
             }
