--- a/net/minecraft/world/level/storage/loot/parameters/LootContextParamSet.java
+++ b/net/minecraft/world/level/storage/loot/parameters/LootContextParamSet.java
@@ -31,7 +_,7 @@
     @Override
     public String toString() {
         return "["
-            + Joiner.on(", ").join(this.all.stream().map(p_81400_ -> (this.required.contains(p_81400_) ? "!" : "") + p_81400_.getName()).iterator())
+            + Joiner.on(", ").join(this.all.stream().map(p_339580_ -> (this.required.contains(p_339580_) ? "!" : "") + p_339580_.getName()).iterator())
             + "]";
     }
 
