--- a/net/minecraft/world/entity/ai/behavior/warden/SonicBoom.java
+++ b/net/minecraft/world/entity/ai/behavior/warden/SonicBoom.java
@@ -56,7 +_,7 @@
     }
 
     protected void tick(ServerLevel p_217724_, Warden p_217725_, long p_217726_) {
-        p_217725_.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).ifPresent(p_340062_ -> p_217725_.getLookControl().setLookAt(p_340062_.position()));
+        p_217725_.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).ifPresent(p_337787_ -> p_217725_.getLookControl().setLookAt(p_337787_.position()));
         if (!p_217725_.getBrain().hasMemoryValue(MemoryModuleType.SONIC_BOOM_SOUND_DELAY)
             && !p_217725_.getBrain().hasMemoryValue(MemoryModuleType.SONIC_BOOM_SOUND_COOLDOWN)) {
             p_217725_.getBrain().setMemoryWithExpiry(MemoryModuleType.SONIC_BOOM_SOUND_COOLDOWN, Unit.INSTANCE, (long)(DURATION - TICKS_BEFORE_PLAYING_SOUND));
@@ -70,7 +_,7 @@
                     Vec3 vec32 = vec31.normalize();
                     int i = Mth.floor(vec31.length()) + 7;
     
-                    for(int j = 1; j < i; ++j) {
+                    for (int j = 1; j < i; j++) {
                         Vec3 vec33 = vec3.add(vec32.scale((double)j));
                         p_217724_.sendParticles(ParticleTypes.SONIC_BOOM, vec33.x, vec33.y, vec33.z, 1, 0.0, 0.0, 0.0, 0.0);
                     }
