--- a/net/minecraft/world/level/levelgen/feature/BambooFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/BambooFeature.java
@@ -42,8 +_,8 @@
                 if (randomsource.nextFloat() < probabilityfeatureconfiguration.probability) {
                     int k = randomsource.nextInt(4) + 1;
 
-                    for(int l = blockpos.getX() - k; l <= blockpos.getX() + k; ++l) {
-                        for(int i1 = blockpos.getZ() - k; i1 <= blockpos.getZ() + k; ++i1) {
+                    for (int l = blockpos.getX() - k; l <= blockpos.getX() + k; l++) {
+                        for (int i1 = blockpos.getZ() - k; i1 <= blockpos.getZ() + k; i1++) {
                             int j1 = l - blockpos.getX();
                             int k1 = i1 - blockpos.getZ();
                             if (j1 * j1 + k1 * k1 <= k * k) {
@@ -56,7 +_,7 @@
                     }
                 }
 
-                for(int l1 = 0; l1 < j && worldgenlevel.isEmptyBlock(blockpos$mutableblockpos); ++l1) {
+                for (int l1 = 0; l1 < j && worldgenlevel.isEmptyBlock(blockpos$mutableblockpos); l1++) {
                     worldgenlevel.setBlock(blockpos$mutableblockpos, BAMBOO_TRUNK, 2);
                     blockpos$mutableblockpos.move(Direction.UP, 1);
                 }
@@ -68,7 +_,7 @@
                 }
             }
 
-            ++i;
+            i++;
         }
 
         return i > 0;
