--- a/net/minecraft/world/level/levelgen/feature/BasaltColumnsFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/BasaltColumnsFeature.java
@@ -16,10 +_,8 @@
 public class BasaltColumnsFeature extends Feature<ColumnFeatureConfiguration> {
     private static final ImmutableList<Block> CANNOT_PLACE_ON = ImmutableList.of(
         Blocks.LAVA,
-        Blocks.WATER,
         Blocks.BEDROCK,
         Blocks.MAGMA_BLOCK,
-        Blocks.SLIME_BLOCK,
         Blocks.SOUL_SAND,
         Blocks.NETHER_BRICKS,
         Blocks.NETHER_BRICK_FENCE,
@@ -53,14 +_,12 @@
             int l = flag ? 50 : 15;
             boolean flag1 = false;
 
-            for(BlockPos blockpos1 : BlockPos.randomBetweenClosed(
+            for (BlockPos blockpos1 : BlockPos.randomBetweenClosed(
                 randomsource, l, blockpos.getX() - k, blockpos.getY(), blockpos.getZ() - k, blockpos.getX() + k, blockpos.getY(), blockpos.getZ() + k
             )) {
                 int i1 = j - blockpos1.distManhattan(blockpos);
                 if (i1 >= 0) {
-                    flag1 |= this.placeColumn(
-                        worldgenlevel, i, blockpos1, i1, columnfeatureconfiguration.reach().sample(randomsource), columnfeatureconfiguration.state()
-                    );
+                    flag1 |= this.placeColumn(worldgenlevel, i, blockpos1, i1, columnfeatureconfiguration.reach().sample(randomsource));
                 }
             }
 
@@ -68,10 +_,10 @@
         }
     }
 
-    private boolean placeColumn(LevelAccessor p_65168_, int p_65169_, BlockPos p_65170_, int p_65171_, int p_65172_, BlockState p_342202_) {
+    private boolean placeColumn(LevelAccessor p_65168_, int p_65169_, BlockPos p_65170_, int p_65171_, int p_65172_) {
         boolean flag = false;
 
-        for(BlockPos blockpos : BlockPos.betweenClosed(
+        for (BlockPos blockpos : BlockPos.betweenClosed(
             p_65170_.getX() - p_65172_, p_65170_.getY(), p_65170_.getZ() - p_65172_, p_65170_.getX() + p_65172_, p_65170_.getY(), p_65170_.getZ() + p_65172_
         )) {
             int i = blockpos.distManhattan(p_65170_);
@@ -81,14 +_,13 @@
             if (blockpos1 != null) {
                 int j = p_65171_ - i / 2;
 
-                for(BlockPos.MutableBlockPos blockpos$mutableblockpos = blockpos1.mutable(); j >= 0; --j) {
+                for (BlockPos.MutableBlockPos blockpos$mutableblockpos = blockpos1.mutable(); j >= 0; j--) {
                     if (isAirOrLavaOcean(p_65168_, p_65169_, blockpos$mutableblockpos)) {
-                        this.setBlock(p_65168_, blockpos$mutableblockpos, p_342202_);
+                        this.setBlock(p_65168_, blockpos$mutableblockpos, Blocks.BASALT.defaultBlockState());
                         blockpos$mutableblockpos.move(Direction.UP);
                         flag = true;
                     } else {
-                        if (!p_65168_.getBlockState(blockpos$mutableblockpos).is(Blocks.BASALT)
-                            && !p_65168_.getBlockState(blockpos$mutableblockpos).is(Blocks.ANCIENT_DEBRIS)) {
+                        if (!p_65168_.getBlockState(blockpos$mutableblockpos).is(Blocks.BASALT)) {
                             break;
                         }
 
@@ -103,8 +_,8 @@
 
     @Nullable
     private static BlockPos findSurface(LevelAccessor p_65159_, int p_65160_, BlockPos.MutableBlockPos p_65161_, int p_65162_) {
-        while(p_65161_.getY() > p_65159_.getMinBuildHeight() + 1 && p_65162_ > 0) {
-            --p_65162_;
+        while (p_65161_.getY() > p_65159_.getMinBuildHeight() + 1 && p_65162_ > 0) {
+            p_65162_--;
             if (canPlaceAt(p_65159_, p_65160_, p_65161_)) {
                 return p_65161_;
             }
@@ -127,8 +_,8 @@
 
     @Nullable
     private static BlockPos findAir(LevelAccessor p_65174_, BlockPos.MutableBlockPos p_65175_, int p_65176_) {
-        while(p_65175_.getY() < p_65174_.getMaxBuildHeight() && p_65176_ > 0) {
-            --p_65176_;
+        while (p_65175_.getY() < p_65174_.getMaxBuildHeight() && p_65176_ > 0) {
+            p_65176_--;
             BlockState blockstate = p_65174_.getBlockState(p_65175_);
             if (CANNOT_PLACE_ON.contains(blockstate.getBlock())) {
                 return null;
