--- a/net/minecraft/world/level/levelgen/feature/BlockBlobFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/BlockBlobFeature.java
@@ -19,7 +_,7 @@
         RandomSource randomsource = p_159471_.random();
 
         BlockStateConfiguration blockstateconfiguration;
-        for(blockstateconfiguration = p_159471_.config(); blockpos.getY() > worldgenlevel.getMinBuildHeight() + 3; blockpos = blockpos.below()) {
+        for (blockstateconfiguration = p_159471_.config(); blockpos.getY() > worldgenlevel.getMinBuildHeight() + 3; blockpos = blockpos.below()) {
             if (!worldgenlevel.isEmptyBlock(blockpos.below())) {
                 BlockState blockstate = worldgenlevel.getBlockState(blockpos.below());
                 if (isDirt(blockstate) || isStone(blockstate)) {
@@ -31,13 +_,13 @@
         if (blockpos.getY() <= worldgenlevel.getMinBuildHeight() + 3) {
             return false;
         } else {
-            for(int l = 0; l < 3; ++l) {
+            for (int l = 0; l < 3; l++) {
                 int i = randomsource.nextInt(2);
                 int j = randomsource.nextInt(2);
                 int k = randomsource.nextInt(2);
                 float f = (float)(i + j + k) * 0.333F + 0.5F;
 
-                for(BlockPos blockpos1 : BlockPos.betweenClosed(blockpos.offset(-i, -j, -k), blockpos.offset(i, j, k))) {
+                for (BlockPos blockpos1 : BlockPos.betweenClosed(blockpos.offset(-i, -j, -k), blockpos.offset(i, j, k))) {
                     if (blockpos1.distSqr(blockpos) <= (double)(f * f)) {
                         worldgenlevel.setBlock(blockpos1, blockstateconfiguration.state, 3);
                     }
