--- a/net/minecraft/world/level/levelgen/feature/BonusChestFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/BonusChestFeature.java
@@ -30,8 +_,8 @@
         IntArrayList intarraylist1 = Util.toShuffledList(IntStream.rangeClosed(chunkpos.getMinBlockZ(), chunkpos.getMaxBlockZ()), randomsource);
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(Integer integer : intarraylist) {
-            for(Integer integer1 : intarraylist1) {
+        for (Integer integer : intarraylist) {
+            for (Integer integer1 : intarraylist1) {
                 blockpos$mutableblockpos.set(integer, 0, integer1);
                 BlockPos blockpos = worldgenlevel.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, blockpos$mutableblockpos);
                 if (worldgenlevel.isEmptyBlock(blockpos) || worldgenlevel.getBlockState(blockpos).getCollisionShape(worldgenlevel, blockpos).isEmpty()) {
@@ -39,7 +_,7 @@
                     RandomizableContainer.setBlockEntityLootTable(worldgenlevel, randomsource, blockpos, BuiltInLootTables.SPAWN_BONUS_CHEST);
                     BlockState blockstate = Blocks.TORCH.defaultBlockState();
 
-                    for(Direction direction : Direction.Plane.HORIZONTAL) {
+                    for (Direction direction : Direction.Plane.HORIZONTAL) {
                         BlockPos blockpos1 = blockpos.relative(direction);
                         if (blockstate.canSurvive(worldgenlevel, blockpos1)) {
                             worldgenlevel.setBlock(blockpos1, blockstate, 2);
