--- a/net/minecraft/world/level/levelgen/feature/DeltaFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/DeltaFeature.java
@@ -38,7 +_,7 @@
         int l = deltafeatureconfiguration.size().sample(randomsource);
         int i1 = Math.max(k, l);
 
-        for(BlockPos blockpos1 : BlockPos.withinManhattan(blockpos, k, 0, l)) {
+        for (BlockPos blockpos1 : BlockPos.withinManhattan(blockpos, k, 0, l)) {
             if (blockpos1.distManhattan(blockpos) > i1) {
                 break;
             }
@@ -67,7 +_,7 @@
         } else if (CANNOT_REPLACE.contains(blockstate.getBlock())) {
             return false;
         } else {
-            for(Direction direction : DIRECTIONS) {
+            for (Direction direction : DIRECTIONS) {
                 boolean flag = p_65552_.getBlockState(p_65553_.relative(direction)).isAir();
                 if (flag && direction != Direction.UP || !flag && direction == Direction.UP) {
                     return false;
