--- a/net/minecraft/world/level/levelgen/feature/DiskFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/DiskFeature.java
@@ -25,7 +_,7 @@
         int l = diskconfiguration.radius().sample(randomsource);
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(BlockPos blockpos1 : BlockPos.betweenClosed(blockpos.offset(-l, 0, -l), blockpos.offset(l, 0, l))) {
+        for (BlockPos blockpos1 : BlockPos.betweenClosed(blockpos.offset(-l, 0, -l), blockpos.offset(l, 0, l))) {
             int i1 = blockpos1.getX() - blockpos.getX();
             int j1 = blockpos1.getZ() - blockpos.getZ();
             if (i1 * i1 + j1 * j1 <= l * l) {
@@ -41,7 +_,7 @@
     ) {
         boolean flag = false;
 
-        for(int i = p_224999_; i > p_225000_; --i) {
+        for (int i = p_224999_; i > p_225000_; i--) {
             p_225001_.setY(i);
             if (p_224996_.target().test(p_224997_, p_225001_)) {
                 BlockState blockstate = p_224996_.stateProvider().getState(p_224997_, p_224998_, p_225001_);
