--- a/net/minecraft/world/level/levelgen/feature/DripstoneClusterFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/DripstoneClusterFeature.java
@@ -38,8 +_,8 @@
             int j = dripstoneclusterconfiguration.radius.sample(randomsource);
             int k = dripstoneclusterconfiguration.radius.sample(randomsource);
 
-            for(int l = -j; l <= j; ++l) {
-                for(int i1 = -k; i1 <= k; ++i1) {
+            for (int l = -j; l <= j; l++) {
+                for (int i1 = -k; i1 <= k; i1++) {
                     double d0 = this.getChanceOfStalagmiteOrStalactite(j, k, l, i1, dripstoneclusterconfiguration);
                     BlockPos blockpos1 = blockpos.offset(l, 0, i1);
                     this.placeColumn(worldgenlevel, randomsource, blockpos1, l, i1, f, d0, i, f1, dripstoneclusterconfiguration);
@@ -167,7 +_,7 @@
             if (p_159620_.getBlockState(p_159621_.above()).getFluidState().is(FluidTags.WATER)) {
                 return false;
             } else {
-                for(Direction direction : Direction.Plane.HORIZONTAL) {
+                for (Direction direction : Direction.Plane.HORIZONTAL) {
                     if (!this.canBeAdjacentToWater(p_159620_, p_159621_.relative(direction))) {
                         return false;
                     }
@@ -188,7 +_,7 @@
     private void replaceBlocksWithDripstoneBlocks(WorldGenLevel p_159589_, BlockPos p_159590_, int p_159591_, Direction p_159592_) {
         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_159590_.mutable();
 
-        for(int i = 0; i < p_159591_; ++i) {
+        for (int i = 0; i < p_159591_; i++) {
             if (!DripstoneUtils.placeDripstoneBlockIfPossible(p_159589_, blockpos$mutableblockpos)) {
                 return;
             }
