--- a/net/minecraft/world/level/levelgen/feature/EndIslandFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/EndIslandFeature.java
@@ -20,9 +_,9 @@
         BlockPos blockpos = p_159717_.origin();
         float f = (float)randomsource.nextInt(3) + 4.0F;
 
-        for(int i = 0; f > 0.5F; --i) {
-            for(int j = Mth.floor(-f); j <= Mth.ceil(f); ++j) {
-                for(int k = Mth.floor(-f); k <= Mth.ceil(f); ++k) {
+        for (int i = 0; f > 0.5F; i--) {
+            for (int j = Mth.floor(-f); j <= Mth.ceil(f); j++) {
+                for (int k = Mth.floor(-f); k <= Mth.ceil(f); k++) {
                     if ((float)(j * j + k * k) <= (f + 1.0F) * (f + 1.0F)) {
                         this.setBlock(worldgenlevel, blockpos.offset(j, i, k), Blocks.END_STONE.defaultBlockState());
                     }
