--- a/net/minecraft/world/level/levelgen/feature/FillLayerFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/FillLayerFeature.java
@@ -17,8 +_,8 @@
         WorldGenLevel worldgenlevel = p_159780_.level();
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int i = 0; i < 16; ++i) {
-            for(int j = 0; j < 16; ++j) {
+        for (int i = 0; i < 16; i++) {
+            for (int j = 0; j < 16; j++) {
                 int k = blockpos.getX() + i;
                 int l = blockpos.getZ() + j;
                 int i1 = worldgenlevel.getMinBuildHeight() + layerconfiguration.height;
