--- a/net/minecraft/world/level/levelgen/feature/GeodeFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/GeodeFeature.java
@@ -57,7 +_,7 @@
         boolean flag = (double)randomsource.nextFloat() < geodecracksettings.generateCrackChance;
         int l = 0;
 
-        for(int i1 = 0; i1 < k; ++i1) {
+        for (int i1 = 0; i1 < k; i1++) {
             int j1 = geodeconfiguration.outerWallDistance.sample(randomsource);
             int k1 = geodeconfiguration.outerWallDistance.sample(randomsource);
             int l1 = geodeconfiguration.outerWallDistance.sample(randomsource);
@@ -97,16 +_,16 @@
         List<BlockPos> list2 = Lists.newArrayList();
         Predicate<BlockState> predicate = isReplaceable(geodeconfiguration.geodeBlockSettings.cannotReplace);
 
-        for(BlockPos blockpos3 : BlockPos.betweenClosed(blockpos.offset(i, i, i), blockpos.offset(j, j, j))) {
+        for (BlockPos blockpos3 : BlockPos.betweenClosed(blockpos.offset(i, i, i), blockpos.offset(j, j, j))) {
             double d8 = normalnoise.getValue((double)blockpos3.getX(), (double)blockpos3.getY(), (double)blockpos3.getZ()) * geodeconfiguration.noiseMultiplier;
             double d6 = 0.0;
             double d7 = 0.0;
 
-            for(Pair<BlockPos, Integer> pair : list) {
+            for (Pair<BlockPos, Integer> pair : list) {
                 d6 += Mth.invSqrt(blockpos3.distSqr(pair.getFirst()) + (double)pair.getSecond().intValue()) + d8;
             }
 
-            for(BlockPos blockpos6 : list1) {
+            for (BlockPos blockpos6 : list1) {
                 d7 += Mth.invSqrt(blockpos3.distSqr(blockpos6) + (double)geodecracksettings.crackPointOffset) + d8;
             }
 
@@ -114,7 +_,7 @@
                 if (flag && d7 >= d5 && d6 < d1) {
                     this.safeSetBlock(worldgenlevel, blockpos3, Blocks.AIR.defaultBlockState(), predicate);
 
-                    for(Direction direction1 : DIRECTIONS) {
+                    for (Direction direction1 : DIRECTIONS) {
                         BlockPos blockpos2 = blockpos3.relative(direction1);
                         FluidState fluidstate = worldgenlevel.getFluidState(blockpos2);
                         if (!fluidstate.isEmpty()) {
@@ -144,25 +_,24 @@
         }
 
         List<BlockState> list3 = geodeblocksettings.innerPlacements;
-        if (!list3.isEmpty()) {
-            for(BlockPos blockpos4 : list2) {
-                BlockState blockstate1 = Util.getRandom(list3, randomsource);
-
-                for(Direction direction : DIRECTIONS) {
-                    if (blockstate1.hasProperty(BlockStateProperties.FACING)) {
-                        blockstate1 = blockstate1.setValue(BlockStateProperties.FACING, direction);
-                    }
-
-                    BlockPos blockpos5 = blockpos4.relative(direction);
-                    BlockState blockstate2 = worldgenlevel.getBlockState(blockpos5);
-                    if (blockstate1.hasProperty(BlockStateProperties.WATERLOGGED)) {
-                        blockstate1 = blockstate1.setValue(BlockStateProperties.WATERLOGGED, Boolean.valueOf(blockstate2.getFluidState().isSource()));
-                    }
-
-                    if (BuddingAmethystBlock.canClusterGrowAtState(blockstate2)) {
-                        this.safeSetBlock(worldgenlevel, blockpos5, blockstate1, predicate);
-                        break;
-                    }
+
+        for (BlockPos blockpos4 : list2) {
+            BlockState blockstate1 = Util.getRandom(list3, randomsource);
+
+            for (Direction direction : DIRECTIONS) {
+                if (blockstate1.hasProperty(BlockStateProperties.FACING)) {
+                    blockstate1 = blockstate1.setValue(BlockStateProperties.FACING, direction);
+                }
+
+                BlockPos blockpos5 = blockpos4.relative(direction);
+                BlockState blockstate2 = worldgenlevel.getBlockState(blockpos5);
+                if (blockstate1.hasProperty(BlockStateProperties.WATERLOGGED)) {
+                    blockstate1 = blockstate1.setValue(BlockStateProperties.WATERLOGGED, Boolean.valueOf(blockstate2.getFluidState().isSource()));
+                }
+
+                if (BuddingAmethystBlock.canClusterGrowAtState(blockstate2)) {
+                    this.safeSetBlock(worldgenlevel, blockpos5, blockstate1, predicate);
+                    break;
                 }
             }
         }
