--- a/net/minecraft/world/level/levelgen/feature/GlowstoneFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/GlowstoneFeature.java
@@ -28,16 +_,16 @@
             } else {
                 worldgenlevel.setBlock(blockpos, Blocks.GLOWSTONE.defaultBlockState(), 2);
 
-                for(int i = 0; i < 1500; ++i) {
+                for (int i = 0; i < 1500; i++) {
                     BlockPos blockpos1 = blockpos.offset(
                         randomsource.nextInt(8) - randomsource.nextInt(8), -randomsource.nextInt(12), randomsource.nextInt(8) - randomsource.nextInt(8)
                     );
                     if (worldgenlevel.getBlockState(blockpos1).isAir()) {
                         int j = 0;
 
-                        for(Direction direction : Direction.values()) {
+                        for (Direction direction : Direction.values()) {
                             if (worldgenlevel.getBlockState(blockpos1.relative(direction)).is(Blocks.GLOWSTONE)) {
-                                ++j;
+                                j++;
                             }
 
                             if (j > 1) {
