--- a/net/minecraft/world/level/levelgen/feature/HugeRedMushroomFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/HugeRedMushroomFeature.java
@@ -22,12 +_,12 @@
         BlockPos.MutableBlockPos p_225086_,
         HugeMushroomFeatureConfiguration p_225087_
     ) {
-        for(int i = p_225085_ - 3; i <= p_225085_; ++i) {
+        for (int i = p_225085_ - 3; i <= p_225085_; i++) {
             int j = i < p_225085_ ? p_225087_.foliageRadius : p_225087_.foliageRadius - 1;
             int k = p_225087_.foliageRadius - 2;
 
-            for(int l = -j; l <= j; ++l) {
-                for(int i1 = -j; i1 <= j; ++i1) {
+            for (int l = -j; l <= j; l++) {
+                for (int i1 = -j; i1 <= j; i1++) {
                     boolean flag = l == -j;
                     boolean flag1 = l == j;
                     boolean flag2 = i1 == -j;
