--- a/net/minecraft/world/level/levelgen/feature/KelpFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/KelpFeature.java
@@ -28,13 +_,13 @@
             BlockState blockstate1 = Blocks.KELP_PLANT.defaultBlockState();
             int k = 1 + randomsource.nextInt(10);
 
-            for(int l = 0; l <= k; ++l) {
+            for (int l = 0; l <= k; l++) {
                 if (worldgenlevel.getBlockState(blockpos1).is(Blocks.WATER)
                     && worldgenlevel.getBlockState(blockpos1.above()).is(Blocks.WATER)
                     && blockstate1.canSurvive(worldgenlevel, blockpos1)) {
                     if (l == k) {
                         worldgenlevel.setBlock(blockpos1, blockstate.setValue(KelpBlock.AGE, Integer.valueOf(randomsource.nextInt(4) + 20)), 2);
-                        ++i;
+                        i++;
                     } else {
                         worldgenlevel.setBlock(blockpos1, blockstate1, 2);
                     }
@@ -42,7 +_,7 @@
                     BlockPos blockpos2 = blockpos1.below();
                     if (blockstate.canSurvive(worldgenlevel, blockpos2) && !worldgenlevel.getBlockState(blockpos2.below()).is(Blocks.KELP)) {
                         worldgenlevel.setBlock(blockpos2, blockstate.setValue(KelpBlock.AGE, Integer.valueOf(randomsource.nextInt(4) + 20)), 2);
-                        ++i;
+                        i++;
                     }
                     break;
                 }
