--- a/net/minecraft/world/level/levelgen/feature/LargeDripstoneFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/LargeDripstoneFeature.java
@@ -12,8 +_,6 @@
 import net.minecraft.world.level.WorldGenLevel;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
-import net.minecraft.world.level.block.PointedDripstoneBlock;
-import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.levelgen.Column;
 import net.minecraft.world.level.levelgen.Heightmap;
 import net.minecraft.world.level.levelgen.feature.configurations.LargeDripstoneConfiguration;
@@ -30,7 +_,6 @@
         BlockPos blockpos = p_159967_.origin();
         LargeDripstoneConfiguration largedripstoneconfiguration = p_159967_.config();
         RandomSource randomsource = p_159967_.random();
-        PointedDripstoneBlock pointeddripstoneblock = largedripstoneconfiguration.block;
         if (!DripstoneUtils.isEmptyOrWater(worldgenlevel, blockpos)) {
             return false;
         } else {
@@ -39,47 +_,35 @@
                 blockpos,
                 largedripstoneconfiguration.floorToCeilingSearchRange,
                 DripstoneUtils::isEmptyOrWater,
-                p_341968_ -> DripstoneUtils.isDripstoneBaseOrLava(p_341968_)
+                DripstoneUtils::isDripstoneBaseOrLava
             );
-            if (optional.isEmpty()) {
-                return false;
-            } else if (optional.get() instanceof Column.Range || pointeddripstoneblock == Blocks.POTATO_BUD && optional.get().getFloor().isPresent()) {
-                Column column = optional.get();
-                double d0 = column instanceof Column.Range column$range ? (double)column$range.height() : (double)Mth.randomBetween(randomsource, 10.0F, 20.0F);
-                if (d0 < 4.0) {
+            if (!optional.isEmpty() && optional.get() instanceof Column.Range) {
+                Column.Range column$range = (Column.Range)optional.get();
+                if (column$range.height() < 4) {
                     return false;
                 } else {
-                    int k = (int)(d0 * (double)largedripstoneconfiguration.maxColumnRadiusToCaveHeightRatio);
-                    int i = Mth.clamp(k, largedripstoneconfiguration.columnRadius.getMinValue(), largedripstoneconfiguration.columnRadius.getMaxValue());
-                    int j = Mth.randomBetweenInclusive(randomsource, largedripstoneconfiguration.columnRadius.getMinValue(), i);
-                    Optional<LargeDripstoneFeature.LargeDripstone> optional1;
-                    if (column.getCeiling().isPresent()) {
-                        optional1 = Optional.of(
-                            makeDripstone(
-                                blockpos.atY(column.getCeiling().getAsInt() - 1),
-                                false,
-                                randomsource,
-                                j,
-                                largedripstoneconfiguration.stalactiteBluntness,
-                                largedripstoneconfiguration.heightScale
-                            )
-                        );
-                    } else {
-                        optional1 = Optional.empty();
-                    }
-
+                    int i = (int)((float)column$range.height() * largedripstoneconfiguration.maxColumnRadiusToCaveHeightRatio);
+                    int j = Mth.clamp(i, largedripstoneconfiguration.columnRadius.getMinValue(), largedripstoneconfiguration.columnRadius.getMaxValue());
+                    int k = Mth.randomBetweenInclusive(randomsource, largedripstoneconfiguration.columnRadius.getMinValue(), j);
                     LargeDripstoneFeature.LargeDripstone largedripstonefeature$largedripstone = makeDripstone(
-                        blockpos.atY(column.getFloor().getAsInt() + 1),
+                        blockpos.atY(column$range.ceiling() - 1),
+                        false,
+                        randomsource,
+                        k,
+                        largedripstoneconfiguration.stalactiteBluntness,
+                        largedripstoneconfiguration.heightScale
+                    );
+                    LargeDripstoneFeature.LargeDripstone largedripstonefeature$largedripstone1 = makeDripstone(
+                        blockpos.atY(column$range.floor() + 1),
                         true,
                         randomsource,
-                        j,
+                        k,
                         largedripstoneconfiguration.stalagmiteBluntness,
                         largedripstoneconfiguration.heightScale
                     );
                     LargeDripstoneFeature.WindOffsetter largedripstonefeature$windoffsetter;
-                    if (optional1.isPresent()
-                        && optional1.get().isSuitableForWind(largedripstoneconfiguration)
-                        && largedripstonefeature$largedripstone.isSuitableForWind(largedripstoneconfiguration)) {
+                    if (largedripstonefeature$largedripstone.isSuitableForWind(largedripstoneconfiguration)
+                        && largedripstonefeature$largedripstone1.isSuitableForWind(largedripstoneconfiguration)) {
                         largedripstonefeature$windoffsetter = new LargeDripstoneFeature.WindOffsetter(
                             blockpos.getY(), randomsource, largedripstoneconfiguration.windSpeed
                         );
@@ -87,19 +_,18 @@
                         largedripstonefeature$windoffsetter = LargeDripstoneFeature.WindOffsetter.noWind();
                     }
 
-                    boolean flag = optional1.isPresent()
-                        && optional1.get().moveBackUntilBaseIsInsideStoneAndShrinkRadiusIfNecessary(worldgenlevel, largedripstonefeature$windoffsetter);
-                    boolean flag1 = largedripstonefeature$largedripstone.moveBackUntilBaseIsInsideStoneAndShrinkRadiusIfNecessary(
-                        worldgenlevel, largedripstonefeature$windoffsetter
-                    );
-                    if (optional1.isPresent() && flag) {
-                        optional1.get().placeBlocks(pointeddripstoneblock, worldgenlevel, randomsource, largedripstonefeature$windoffsetter);
+                    boolean flag = largedripstonefeature$largedripstone.moveBackUntilBaseIsInsideStoneAndShrinkRadiusIfNecessary(
+                        worldgenlevel, largedripstonefeature$windoffsetter
+                    );
+                    boolean flag1 = largedripstonefeature$largedripstone1.moveBackUntilBaseIsInsideStoneAndShrinkRadiusIfNecessary(
+                        worldgenlevel, largedripstonefeature$windoffsetter
+                    );
+                    if (flag) {
+                        largedripstonefeature$largedripstone.placeBlocks(worldgenlevel, randomsource, largedripstonefeature$windoffsetter);
                     }
 
                     if (flag1) {
-                        largedripstonefeature$largedripstone.placeBlocks(
-                            pointeddripstoneblock, worldgenlevel, randomsource, largedripstonefeature$windoffsetter
-                        );
+                        largedripstonefeature$largedripstone1.placeBlocks(worldgenlevel, randomsource, largedripstonefeature$windoffsetter);
                     }
 
                     return true;
@@ -122,7 +_,7 @@
         p_159962_.setBlock(p_159965_.offset(p_159963_.atY(p_159964_.ceiling() - 1)), Blocks.DIAMOND_BLOCK.defaultBlockState(), 2);
         p_159962_.setBlock(p_159965_.offset(p_159963_.atY(p_159964_.floor() + 1)), Blocks.GOLD_BLOCK.defaultBlockState(), 2);
 
-        for(BlockPos.MutableBlockPos blockpos$mutableblockpos = p_159963_.atY(p_159964_.floor() + 2).mutable();
+        for (BlockPos.MutableBlockPos blockpos$mutableblockpos = p_159963_.atY(p_159964_.floor() + 2).mutable();
             blockpos$mutableblockpos.getY() < p_159964_.ceiling() - 1;
             blockpos$mutableblockpos.move(Direction.UP)
         ) {
@@ -161,11 +_,11 @@
         }
 
         boolean moveBackUntilBaseIsInsideStoneAndShrinkRadiusIfNecessary(WorldGenLevel p_159990_, LargeDripstoneFeature.WindOffsetter p_159991_) {
-            while(this.radius > 1) {
+            while (this.radius > 1) {
                 BlockPos.MutableBlockPos blockpos$mutableblockpos = this.root.mutable();
                 int i = Math.min(10, this.getHeight());
 
-                for(int j = 0; j < i; ++j) {
+                for (int j = 0; j < i; j++) {
                     if (p_159990_.getBlockState(blockpos$mutableblockpos).is(Blocks.LAVA)) {
                         return false;
                     }
@@ -188,9 +_,9 @@
             return (int)DripstoneUtils.getDripstoneHeight((double)p_159988_, (double)this.radius, this.scale, this.bluntness);
         }
 
-        void placeBlocks(PointedDripstoneBlock p_342370_, WorldGenLevel p_225146_, RandomSource p_225147_, LargeDripstoneFeature.WindOffsetter p_225148_) {
-            for(int i = -this.radius; i <= this.radius; ++i) {
-                for(int j = -this.radius; j <= this.radius; ++j) {
+        void placeBlocks(WorldGenLevel p_225146_, RandomSource p_225147_, LargeDripstoneFeature.WindOffsetter p_225148_) {
+            for (int i = -this.radius; i <= this.radius; i++) {
+                for (int j = -this.radius; j <= this.radius; j++) {
                     float f = Mth.sqrt((float)(i * i + j * j));
                     if (!(f > (float)this.radius)) {
                         int k = this.getHeightAtRadius(f);
@@ -201,19 +_,15 @@
 
                             BlockPos.MutableBlockPos blockpos$mutableblockpos = this.root.offset(i, 0, j).mutable();
                             boolean flag = false;
-                            int l = this.pointingUp && p_342370_ == Blocks.POINTED_DRIPSTONE
+                            int l = this.pointingUp
                                 ? p_225146_.getHeight(Heightmap.Types.WORLD_SURFACE_WG, blockpos$mutableblockpos.getX(), blockpos$mutableblockpos.getZ())
                                 : Integer.MAX_VALUE;
 
-                            for(int i1 = 0; i1 < k && blockpos$mutableblockpos.getY() < l; ++i1) {
+                            for (int i1 = 0; i1 < k && blockpos$mutableblockpos.getY() < l; i1++) {
                                 BlockPos blockpos = p_225148_.offset(blockpos$mutableblockpos);
-                                if (Math.abs(blockpos.getZ() - this.root.getZ()) > 16 || Math.abs(blockpos.getX() - this.root.getX()) > 16) {
-                                    blockpos = blockpos$mutableblockpos;
-                                }
-
                                 if (DripstoneUtils.isEmptyOrWaterOrLava(p_225146_, blockpos)) {
                                     flag = true;
-                                    Block block = p_342370_.getLargeBlock();
+                                    Block block = Blocks.DRIPSTONE_BLOCK;
                                     p_225146_.setBlock(blockpos, block.defaultBlockState(), 2);
                                 } else if (flag && p_225146_.getBlockState(blockpos).is(BlockTags.BASE_STONE_OVERWORLD)) {
                                     break;
