--- a/net/minecraft/world/level/levelgen/feature/MultifaceGrowthFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/MultifaceGrowthFeature.java
@@ -30,11 +_,11 @@
             } else {
                 BlockPos.MutableBlockPos blockpos$mutableblockpos = blockpos.mutable();
 
-                for(Direction direction : list) {
+                for (Direction direction : list) {
                     blockpos$mutableblockpos.set(blockpos);
                     List<Direction> list1 = multifacegrowthconfiguration.getShuffledDirectionsExcept(randomsource, direction.getOpposite());
 
-                    for(int i = 0; i < multifacegrowthconfiguration.searchRange; ++i) {
+                    for (int i = 0; i < multifacegrowthconfiguration.searchRange; i++) {
                         blockpos$mutableblockpos.setWithOffset(blockpos, direction);
                         BlockState blockstate = worldgenlevel.getBlockState(blockpos$mutableblockpos);
                         if (!isAirOrWater(blockstate) && !blockstate.is(multifacegrowthconfiguration.placeBlock)) {
@@ -62,7 +_,7 @@
     ) {
         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_225159_.mutable();
 
-        for(Direction direction : p_225163_) {
+        for (Direction direction : p_225163_) {
             BlockState blockstate = p_225158_.getBlockState(blockpos$mutableblockpos.setWithOffset(p_225159_, direction));
             if (blockstate.is(p_225161_.canBePlacedOn)) {
                 BlockState blockstate1 = p_225161_.placeBlock.getStateForPlacement(p_225160_, p_225158_, p_225159_, direction);
