--- a/net/minecraft/world/level/levelgen/feature/PointedDripstoneFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/PointedDripstoneFeature.java
@@ -6,9 +_,6 @@
 import net.minecraft.core.Direction;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.LevelAccessor;
-import net.minecraft.world.level.block.Block;
-import net.minecraft.world.level.block.Blocks;
-import net.minecraft.world.level.block.PointedDripstoneBlock;
 import net.minecraft.world.level.levelgen.feature.configurations.PointedDripstoneConfiguration;
 
 public class PointedDripstoneFeature extends Feature<PointedDripstoneConfiguration> {
@@ -38,7 +_,6 @@
     }
 
     private static Optional<Direction> getTipDirection(LevelAccessor p_225199_, BlockPos p_225200_, RandomSource p_225201_) {
-        PointedDripstoneBlock pointeddripstoneblock = (PointedDripstoneBlock)Blocks.POINTED_DRIPSTONE;
         boolean flag = DripstoneUtils.isDripstoneBase(p_225199_.getBlockState(p_225200_.above()));
         boolean flag1 = DripstoneUtils.isDripstoneBase(p_225199_.getBlockState(p_225200_.below()));
         if (flag && flag1) {
@@ -53,10 +_,9 @@
     private static void createPatchOfDripstoneBlocks(
         LevelAccessor p_225194_, RandomSource p_225195_, BlockPos p_225196_, PointedDripstoneConfiguration p_225197_
     ) {
-        Block block = Blocks.POINTED_DRIPSTONE;
         DripstoneUtils.placeDripstoneBlockIfPossible(p_225194_, p_225196_);
 
-        for(Direction direction : Direction.Plane.HORIZONTAL) {
+        for (Direction direction : Direction.Plane.HORIZONTAL) {
             if (!(p_225195_.nextFloat() > p_225197_.chanceOfDirectionalSpread)) {
                 BlockPos blockpos = p_225196_.relative(direction);
                 DripstoneUtils.placeDripstoneBlockIfPossible(p_225194_, blockpos);
