--- a/net/minecraft/world/level/levelgen/feature/ReplaceBlockFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/ReplaceBlockFeature.java
@@ -17,7 +_,7 @@
         BlockPos blockpos = p_160216_.origin();
         ReplaceBlockConfiguration replaceblockconfiguration = p_160216_.config();
 
-        for(OreConfiguration.TargetBlockState oreconfiguration$targetblockstate : replaceblockconfiguration.targetStates) {
+        for (OreConfiguration.TargetBlockState oreconfiguration$targetblockstate : replaceblockconfiguration.targetStates) {
             if (oreconfiguration$targetblockstate.target.test(worldgenlevel.getBlockState(blockpos), p_160216_.random())) {
                 worldgenlevel.setBlock(blockpos, oreconfiguration$targetblockstate.state, 2);
                 break;
