--- a/net/minecraft/world/level/levelgen/feature/ReplaceBlobsFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/ReplaceBlobsFeature.java
@@ -36,7 +_,7 @@
             int l = Math.max(i, Math.max(j, k));
             boolean flag = false;
 
-            for(BlockPos blockpos1 : BlockPos.withinManhattan(blockpos, i, j, k)) {
+            for (BlockPos blockpos1 : BlockPos.withinManhattan(blockpos, i, j, k)) {
                 if (blockpos1.distManhattan(blockpos) > l) {
                     break;
                 }
@@ -54,7 +_,7 @@
 
     @Nullable
     private static BlockPos findTarget(LevelAccessor p_66635_, BlockPos.MutableBlockPos p_66636_, Block p_66637_) {
-        while(p_66636_.getY() > p_66635_.getMinBuildHeight() + 1) {
+        while (p_66636_.getY() > p_66635_.getMinBuildHeight() + 1) {
             BlockState blockstate = p_66635_.getBlockState(p_66636_);
             if (blockstate.is(p_66637_)) {
                 return p_66636_;
