--- a/net/minecraft/world/level/levelgen/feature/ScatteredOreFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/ScatteredOreFeature.java
@@ -23,11 +_,11 @@
         int i = randomsource.nextInt(oreconfiguration.size + 1);
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             this.offsetTargetPos(blockpos$mutableblockpos, randomsource, blockpos, Math.min(j, 7));
             BlockState blockstate = worldgenlevel.getBlockState(blockpos$mutableblockpos);
 
-            for(OreConfiguration.TargetBlockState oreconfiguration$targetblockstate : oreconfiguration.targetStates) {
+            for (OreConfiguration.TargetBlockState oreconfiguration$targetblockstate : oreconfiguration.targetStates) {
                 if (OreFeature.canPlaceOre(
                     blockstate, worldgenlevel::getBlockState, randomsource, oreconfiguration, oreconfiguration$targetblockstate, blockpos$mutableblockpos
                 )) {
