--- a/net/minecraft/world/level/levelgen/feature/SculkPatchFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SculkPatchFeature.java
@@ -30,14 +_,14 @@
             SculkSpreader sculkspreader = SculkSpreader.createWorldGenSpreader();
             int i = sculkpatchconfiguration.spreadRounds() + sculkpatchconfiguration.growthRounds();
 
-            for(int j = 0; j < i; ++j) {
-                for(int k = 0; k < sculkpatchconfiguration.chargeCount(); ++k) {
+            for (int j = 0; j < i; j++) {
+                for (int k = 0; k < sculkpatchconfiguration.chargeCount(); k++) {
                     sculkspreader.addCursors(blockpos, sculkpatchconfiguration.amountPerCharge());
                 }
 
                 boolean flag = j < sculkpatchconfiguration.spreadRounds();
 
-                for(int l = 0; l < sculkpatchconfiguration.spreadAttempts(); ++l) {
+                for (int l = 0; l < sculkpatchconfiguration.spreadAttempts(); l++) {
                     sculkspreader.updateCursors(worldgenlevel, blockpos, randomsource, flag);
                 }
 
@@ -52,7 +_,7 @@
 
             int i1 = sculkpatchconfiguration.extraRareGrowths().sample(randomsource);
 
-            for(int j1 = 0; j1 < i1; ++j1) {
+            for (int j1 = 0; j1 < i1; j1++) {
                 BlockPos blockpos1 = blockpos.offset(randomsource.nextInt(5) - 2, 0, randomsource.nextInt(5) - 2);
                 if (worldgenlevel.getBlockState(blockpos1).isAir()
                     && worldgenlevel.getBlockState(blockpos1.below()).isFaceSturdy(worldgenlevel, blockpos1.below(), Direction.UP)) {
