--- a/net/minecraft/world/level/levelgen/feature/SimpleBlockFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SimpleBlockFeature.java
@@ -3,9 +_,7 @@
 import com.mojang.serialization.Codec;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.WorldGenLevel;
-import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.DoublePlantBlock;
-import net.minecraft.world.level.block.PotatoBlock;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.levelgen.feature.configurations.SimpleBlockConfiguration;
 
@@ -20,24 +_,20 @@
         WorldGenLevel worldgenlevel = p_160341_.level();
         BlockPos blockpos = p_160341_.origin();
         BlockState blockstate = simpleblockconfiguration.toPlace().getState(p_160341_.random(), blockpos);
-        if (blockstate.is(Blocks.POTATOES)) {
-            blockstate = PotatoBlock.withCorrectTaterBoost(blockstate, worldgenlevel.getBlockState(blockpos.below()));
-        }
-
-        return blockstate.canSurvive(worldgenlevel, blockpos) ? place(blockstate, worldgenlevel, blockpos) : false;
-    }
-
-    public static boolean place(BlockState p_341562_, WorldGenLevel p_342044_, BlockPos p_341883_) {
-        if (p_341562_.getBlock() instanceof DoublePlantBlock) {
-            if (!p_342044_.isEmptyBlock(p_341883_.above())) {
-                return false;
+        if (blockstate.canSurvive(worldgenlevel, blockpos)) {
+            if (blockstate.getBlock() instanceof DoublePlantBlock) {
+                if (!worldgenlevel.isEmptyBlock(blockpos.above())) {
+                    return false;
+                }
+
+                DoublePlantBlock.placeAt(worldgenlevel, blockstate, blockpos, 2);
+            } else {
+                worldgenlevel.setBlock(blockpos, blockstate, 2);
             }
 
-            DoublePlantBlock.placeAt(p_342044_, p_341562_, p_341883_, 2);
+            return true;
         } else {
-            p_342044_.setBlock(p_341883_, p_341562_, 2);
+            return false;
         }
-
-        return true;
     }
 }
