--- a/net/minecraft/world/level/levelgen/feature/SnowAndFreezeFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SnowAndFreezeFeature.java
@@ -23,8 +_,8 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
         BlockPos.MutableBlockPos blockpos$mutableblockpos1 = new BlockPos.MutableBlockPos();
 
-        for(int i = 0; i < 16; ++i) {
-            for(int j = 0; j < 16; ++j) {
+        for (int i = 0; i < 16; i++) {
+            for (int j = 0; j < 16; j++) {
                 int k = blockpos.getX() + i;
                 int l = blockpos.getZ() + j;
                 int i1 = worldgenlevel.getHeight(Heightmap.Types.MOTION_BLOCKING, k, l);
