--- a/net/minecraft/world/level/levelgen/feature/UnderwaterMagmaFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/UnderwaterMagmaFeature.java
@@ -59,7 +_,7 @@
 
     private boolean isValidPlacement(WorldGenLevel p_160575_, BlockPos p_160576_) {
         if (!this.isWaterOrAir(p_160575_, p_160576_) && !this.isWaterOrAir(p_160575_, p_160576_.below())) {
-            for(Direction direction : Direction.Plane.HORIZONTAL) {
+            for (Direction direction : Direction.Plane.HORIZONTAL) {
                 if (this.isWaterOrAir(p_160575_, p_160576_.relative(direction))) {
                     return false;
                 }
