--- a/net/minecraft/world/level/levelgen/feature/VinesFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/VinesFeature.java
@@ -21,7 +_,7 @@
         if (!worldgenlevel.isEmptyBlock(blockpos)) {
             return false;
         } else {
-            for(Direction direction : Direction.values()) {
+            for (Direction direction : Direction.values()) {
                 if (direction != Direction.DOWN && VineBlock.isAcceptableNeighbour(worldgenlevel, blockpos.relative(direction), direction)) {
                     worldgenlevel.setBlock(
                         blockpos, Blocks.VINE.defaultBlockState().setValue(VineBlock.getPropertyForFace(direction), Boolean.valueOf(true)), 2
