--- a/net/minecraft/world/level/levelgen/feature/VoidStartPlatformFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/VoidStartPlatformFeature.java
@@ -31,8 +_,8 @@
             BlockPos blockpos = PLATFORM_OFFSET.atY(p_160633_.origin().getY() + PLATFORM_OFFSET.getY());
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(int i = chunkpos.getMinBlockZ(); i <= chunkpos.getMaxBlockZ(); ++i) {
-                for(int j = chunkpos.getMinBlockX(); j <= chunkpos.getMaxBlockX(); ++j) {
+            for (int i = chunkpos.getMinBlockZ(); i <= chunkpos.getMaxBlockZ(); i++) {
+                for (int j = chunkpos.getMinBlockX(); j <= chunkpos.getMaxBlockX(); j++) {
                     if (checkerboardDistance(blockpos.getX(), blockpos.getZ(), j, i) <= 16) {
                         blockpos$mutableblockpos.set(j, blockpos.getY(), i);
                         if (blockpos$mutableblockpos.equals(blockpos)) {
