--- a/net/minecraft/world/level/levelgen/feature/WaterloggedVegetationPatchFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/WaterloggedVegetationPatchFeature.java
@@ -33,13 +_,13 @@
         Set<BlockPos> set1 = new HashSet<>();
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(BlockPos blockpos : set) {
+        for (BlockPos blockpos : set) {
             if (!isExposed(p_225339_, set, blockpos, blockpos$mutableblockpos)) {
                 set1.add(blockpos);
             }
         }
 
-        for(BlockPos blockpos1 : set1) {
+        for (BlockPos blockpos1 : set1) {
             p_225339_.setBlock(blockpos1, Blocks.WATER.defaultBlockState(), 2);
         }
 
