--- a/net/minecraft/world/level/levelgen/feature/WeepingVinesFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/WeepingVinesFeature.java
@@ -43,7 +_,7 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
         BlockPos.MutableBlockPos blockpos$mutableblockpos1 = new BlockPos.MutableBlockPos();
 
-        for(int i = 0; i < 200; ++i) {
+        for (int i = 0; i < 200; i++) {
             blockpos$mutableblockpos.setWithOffset(
                 p_225362_,
                 p_225361_.nextInt(6) - p_225361_.nextInt(6),
@@ -53,10 +_,10 @@
             if (p_225360_.isEmptyBlock(blockpos$mutableblockpos)) {
                 int j = 0;
 
-                for(Direction direction : DIRECTIONS) {
+                for (Direction direction : DIRECTIONS) {
                     BlockState blockstate = p_225360_.getBlockState(blockpos$mutableblockpos1.setWithOffset(blockpos$mutableblockpos, direction));
                     if (blockstate.is(Blocks.NETHERRACK) || blockstate.is(Blocks.NETHER_WART_BLOCK)) {
-                        ++j;
+                        j++;
                     }
 
                     if (j > 1) {
@@ -74,7 +_,7 @@
     private void placeRoofWeepingVines(LevelAccessor p_225364_, RandomSource p_225365_, BlockPos p_225366_) {
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-        for(int i = 0; i < 100; ++i) {
+        for (int i = 0; i < 100; i++) {
             blockpos$mutableblockpos.setWithOffset(
                 p_225366_,
                 p_225365_.nextInt(8) - p_225365_.nextInt(8),
@@ -104,7 +_,7 @@
     public static void placeWeepingVinesColumn(
         LevelAccessor p_225353_, RandomSource p_225354_, BlockPos.MutableBlockPos p_225355_, int p_225356_, int p_225357_, int p_225358_
     ) {
-        for(int i = 0; i <= p_225356_; ++i) {
+        for (int i = 0; i <= p_225356_; i++) {
             if (p_225353_.isEmptyBlock(p_225355_)) {
                 if (i == p_225356_ || !p_225353_.isEmptyBlock(p_225355_.below())) {
                     p_225353_.setBlock(
