--- a/net/minecraft/client/gui/components/spectator/SpectatorGui.java
+++ b/net/minecraft/client/gui/components/spectator/SpectatorGui.java
@@ -71,7 +_,7 @@
 
         p_282945_.setColor(1.0F, 1.0F, 1.0F, 1.0F);
 
-        for(int i = 0; i < 9; ++i) {
+        for (int i = 0; i < 9; i++) {
             this.renderSlot(p_282945_, i, p_282945_.guiWidth() / 2 - 90 + i * 20 + 2, (float)(p_281730_ + 3), p_281688_, p_282361_.getItem(i));
         }
 
@@ -123,7 +_,7 @@
     public void onMouseScrolled(int p_205381_) {
         int i = this.menu.getSelectedSlot() + p_205381_;
 
-        while(i >= 0 && i <= 8 && (this.menu.getItem(i) == SpectatorMenu.EMPTY_SLOT || !this.menu.getItem(i).isEnabled())) {
+        while (i >= 0 && i <= 8 && (this.menu.getItem(i) == SpectatorMenu.EMPTY_SLOT || !this.menu.getItem(i).isEnabled())) {
             i += p_205381_;
         }
 
