--- a/net/minecraft/world/entity/ai/attributes/AttributeMap.java
+++ b/net/minecraft/world/entity/ai/attributes/AttributeMap.java
@@ -43,7 +_,7 @@
 
     @Nullable
     public AttributeInstance getInstance(Holder<Attribute> p_250010_) {
-        return this.attributes.computeIfAbsent(p_250010_, p_315936_ -> this.supplier.createInstance(this::onAttributeModified, p_315936_));
+        return this.attributes.computeIfAbsent(p_250010_, p_315936_ -> this.supplier.createInstance(this::onAttributeModified, (Holder<Attribute>)p_315936_));
     }
 
     public boolean hasAttribute(Holder<Attribute> p_248893_) {
@@ -82,7 +_,7 @@
     public ListTag save() {
         ListTag listtag = new ListTag();
 
-        for(AttributeInstance attributeinstance : this.attributes.values()) {
+        for (AttributeInstance attributeinstance : this.attributes.values()) {
             listtag.add(attributeinstance.save());
         }
 
@@ -90,7 +_,7 @@
     }
 
     public void load(ListTag p_22169_) {
-        for(int i = 0; i < p_22169_.size(); ++i) {
+        for (int i = 0; i < p_22169_.size(); i++) {
             CompoundTag compoundtag = p_22169_.getCompound(i);
             String s = compoundtag.getString("Name");
             ResourceLocation resourcelocation = ResourceLocation.tryParse(s);
