--- a/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
+++ b/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
@@ -49,7 +_,7 @@
 
     @VisibleForTesting
     Map<UUID, AttributeModifier> getModifiers(AttributeModifier.Operation p_22105_) {
-        return this.modifiersByOperation.computeIfAbsent(p_22105_, p_332604_ -> new Object2ObjectOpenHashMap());
+        return this.modifiersByOperation.computeIfAbsent(p_22105_, p_332604_ -> new Object2ObjectOpenHashMap<>());
     }
 
     public Set<AttributeModifier> getModifiers() {
@@ -123,7 +_,7 @@
     }
 
     public void removeModifiers() {
-        for(AttributeModifier attributemodifier : this.getModifiers()) {
+        for (AttributeModifier attributemodifier : this.getModifiers()) {
             this.removeModifier(attributemodifier);
         }
     }
@@ -140,17 +_,17 @@
     private double calculateValue() {
         double d0 = this.getBaseValue();
 
-        for(AttributeModifier attributemodifier : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_VALUE)) {
+        for (AttributeModifier attributemodifier : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_VALUE)) {
             d0 += attributemodifier.amount();
         }
 
         double d1 = d0;
 
-        for(AttributeModifier attributemodifier1 : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_MULTIPLIED_BASE)) {
+        for (AttributeModifier attributemodifier1 : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_MULTIPLIED_BASE)) {
             d1 += d0 * attributemodifier1.amount();
         }
 
-        for(AttributeModifier attributemodifier2 : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_MULTIPLIED_TOTAL)) {
+        for (AttributeModifier attributemodifier2 : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_MULTIPLIED_TOTAL)) {
             d1 *= 1.0 + attributemodifier2.amount();
         }
 
@@ -168,7 +_,8 @@
         this.permanentModifiers.clear();
         this.permanentModifiers.putAll(p_22103_.permanentModifiers);
         this.modifiersByOperation.clear();
-        p_22103_.modifiersByOperation.forEach((p_332605_, p_332606_) -> this.getModifiers(p_332605_).putAll(p_332606_));
+        p_22103_.modifiersByOperation
+            .forEach((p_332605_, p_332606_) -> this.getModifiers(p_332605_).putAll((Map<? extends UUID, ? extends AttributeModifier>)p_332606_));
         this.setDirty();
     }
 
@@ -182,7 +_,7 @@
         if (!this.permanentModifiers.isEmpty()) {
             ListTag listtag = new ListTag();
 
-            for(AttributeModifier attributemodifier : this.permanentModifiers.values()) {
+            for (AttributeModifier attributemodifier : this.permanentModifiers.values()) {
                 listtag.add(attributemodifier.save());
             }
 
@@ -197,7 +_,7 @@
         if (p_22114_.contains("Modifiers", 9)) {
             ListTag listtag = p_22114_.getList("Modifiers", 10);
 
-            for(int i = 0; i < listtag.size(); ++i) {
+            for (int i = 0; i < listtag.size(); i++) {
                 AttributeModifier attributemodifier = AttributeModifier.load(listtag.getCompound(i));
                 if (attributemodifier != null) {
                     this.modifierById.put(attributemodifier.id(), attributemodifier);
