--- a/net/minecraft/world/entity/animal/axolotl/Axolotl.java
+++ b/net/minecraft/world/entity/animal/axolotl/Axolotl.java
@@ -528,6 +_,11 @@
     }
 
     @Override
+    public boolean canAttack(LivingEntity target) {
+        return false;
+    }
+
+    @Override
     public Brain<Axolotl> getBrain() {
         return (Brain<Axolotl>) super.getBrain();
     }
@@ -554,8 +_,8 @@
     }
 
     @Override
-    public @Nullable Entity getTarget() {
-        return this.getTargetFromBrain();
+    public @Nullable LivingEntity getTarget() {
+        return (LivingEntity) this.getTargetFromBrain();
     }
 
     public static boolean checkAxolotlSpawnRules(
