--- a/net/minecraft/util/valueproviders/BiasedToBottomInt.java
+++ b/net/minecraft/util/valueproviders/BiasedToBottomInt.java
@@ -2,26 +_,25 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
-import java.util.function.Function;
 import net.minecraft.util.RandomSource;
 
 public class BiasedToBottomInt extends IntProvider {
-    public static final Codec<BiasedToBottomInt> CODEC = RecordCodecBuilder.<BiasedToBottomInt>create(
+    public static final MapCodec<BiasedToBottomInt> CODEC = RecordCodecBuilder.<BiasedToBottomInt>mapCodec(
             p_146373_ -> p_146373_.group(
                         Codec.INT.fieldOf("min_inclusive").forGetter(p_146381_ -> p_146381_.minInclusive),
                         Codec.INT.fieldOf("max_inclusive").forGetter(p_146378_ -> p_146378_.maxInclusive)
                     )
                     .apply(p_146373_, BiasedToBottomInt::new)
         )
-        .comapFlatMap(
+        .validate(
             p_274930_ -> p_274930_.maxInclusive < p_274930_.minInclusive
                     ? DataResult.error(
                         () -> "Max must be at least min, min_inclusive: " + p_274930_.minInclusive + ", max_inclusive: " + p_274930_.maxInclusive
                     )
-                    : DataResult.success(p_274930_),
-            Function.identity()
+                    : DataResult.success(p_274930_)
         );
     private final int minInclusive;
     private final int maxInclusive;
