--- a/net/minecraft/util/valueproviders/ClampedNormalFloat.java
+++ b/net/minecraft/util/valueproviders/ClampedNormalFloat.java
@@ -2,14 +_,14 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
-import java.util.function.Function;
 import net.minecraft.util.Mth;
 import net.minecraft.util.RandomSource;
 
 public class ClampedNormalFloat extends FloatProvider {
-    public static final Codec<ClampedNormalFloat> CODEC = RecordCodecBuilder.<ClampedNormalFloat>create(
+    public static final MapCodec<ClampedNormalFloat> CODEC = RecordCodecBuilder.<ClampedNormalFloat>mapCodec(
             p_146431_ -> p_146431_.group(
                         Codec.FLOAT.fieldOf("mean").forGetter(p_146449_ -> p_146449_.mean),
                         Codec.FLOAT.fieldOf("deviation").forGetter(p_146447_ -> p_146447_.deviation),
@@ -18,11 +_,10 @@
                     )
                     .apply(p_146431_, ClampedNormalFloat::new)
         )
-        .comapFlatMap(
+        .validate(
             p_274935_ -> p_274935_.max < p_274935_.min
                     ? DataResult.error(() -> "Max must be larger than min: [" + p_274935_.min + ", " + p_274935_.max + "]")
-                    : DataResult.success(p_274935_),
-            Function.identity()
+                    : DataResult.success(p_274935_)
         );
     private final float mean;
     private final float deviation;
