--- a/net/minecraft/util/valueproviders/ConstantInt.java
+++ b/net/minecraft/util/valueproviders/ConstantInt.java
@@ -1,13 +_,12 @@
 package net.minecraft.util.valueproviders;
 
 import com.mojang.serialization.Codec;
-import net.minecraft.util.ExtraCodecs;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.util.RandomSource;
 
 public class ConstantInt extends IntProvider {
     public static final ConstantInt ZERO = new ConstantInt(0);
-    public static final Codec<ConstantInt> CODEC = ExtraCodecs.withAlternative(Codec.INT, Codec.INT.fieldOf("value").codec())
-        .xmap(ConstantInt::new, ConstantInt::getValue);
+    public static final MapCodec<ConstantInt> CODEC = Codec.INT.fieldOf("value").xmap(ConstantInt::of, ConstantInt::getValue);
     private final int value;
 
     public static ConstantInt of(int p_146484_) {
