--- a/net/minecraft/util/valueproviders/TrapezoidFloat.java
+++ b/net/minecraft/util/valueproviders/TrapezoidFloat.java
@@ -2,13 +_,13 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
-import java.util.function.Function;
 import net.minecraft.util.RandomSource;
 
 public class TrapezoidFloat extends FloatProvider {
-    public static final Codec<TrapezoidFloat> CODEC = RecordCodecBuilder.<TrapezoidFloat>create(
+    public static final MapCodec<TrapezoidFloat> CODEC = RecordCodecBuilder.<TrapezoidFloat>mapCodec(
             p_146578_ -> p_146578_.group(
                         Codec.FLOAT.fieldOf("min").forGetter(p_146588_ -> p_146588_.min),
                         Codec.FLOAT.fieldOf("max").forGetter(p_146586_ -> p_146586_.max),
@@ -16,7 +_,7 @@
                     )
                     .apply(p_146578_, TrapezoidFloat::new)
         )
-        .comapFlatMap(
+        .validate(
             p_274953_ -> {
                 if (p_274953_.max < p_274953_.min) {
                     return DataResult.error(() -> "Max must be larger than min: [" + p_274953_.min + ", " + p_274953_.max + "]");
@@ -25,8 +_,7 @@
                         ? DataResult.error(() -> "Plateau can at most be the full span: [" + p_274953_.min + ", " + p_274953_.max + "]")
                         : DataResult.success(p_274953_);
                 }
-            },
-            Function.identity()
+            }
         );
     private final float min;
     private final float max;
