--- a/net/minecraft/util/valueproviders/WeightedListInt.java
+++ b/net/minecraft/util/valueproviders/WeightedListInt.java
@@ -1,6 +_,6 @@
 package net.minecraft.util.valueproviders;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -9,7 +_,7 @@
 import net.minecraft.util.random.WeightedEntry;
 
 public class WeightedListInt extends IntProvider {
-    public static final Codec<WeightedListInt> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<WeightedListInt> CODEC = RecordCodecBuilder.mapCodec(
         p_185920_ -> p_185920_.group(
                     SimpleWeightedRandomList.wrappedCodec(IntProvider.CODEC).fieldOf("distribution").forGetter(p_185918_ -> p_185918_.distribution)
                 )
@@ -25,9 +_,9 @@
         int i = Integer.MAX_VALUE;
         int j = Integer.MIN_VALUE;
 
-        for(WeightedEntry.Wrapper<IntProvider> wrapper : list) {
-            int k = wrapper.getData().getMinValue();
-            int l = wrapper.getData().getMaxValue();
+        for (WeightedEntry.Wrapper<IntProvider> wrapper : list) {
+            int k = wrapper.data().getMinValue();
+            int l = wrapper.data().getMaxValue();
             i = Math.min(i, k);
             j = Math.max(j, l);
         }
