--- a/net/minecraft/world/level/storage/loot/predicates/ConditionReference.java
+++ b/net/minecraft/world/level/storage/loot/predicates/ConditionReference.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.storage.loot.predicates;
 
 import com.mojang.logging.LogUtils;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.Holder;
@@ -13,7 +_,7 @@
 
 public record ConditionReference(ResourceKey<LootItemCondition> name) implements LootItemCondition {
     private static final Logger LOGGER = LogUtils.getLogger();
-    public static final Codec<ConditionReference> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<ConditionReference> CODEC = RecordCodecBuilder.mapCodec(
         p_335367_ -> p_335367_.group(ResourceKey.codec(Registries.PREDICATE).fieldOf("name").forGetter(ConditionReference::name))
                 .apply(p_335367_, ConditionReference::new)
     );
@@ -32,7 +_,7 @@
             p_81560_.resolver()
                 .get(Registries.PREDICATE, this.name)
                 .ifPresentOrElse(
-                    p_335370_ -> ((LootItemCondition)p_335370_.value()).validate(p_81560_.enterElement(".{" + this.name.location() + "}", this.name)),
+                    p_339583_ -> p_339583_.value().validate(p_81560_.enterElement(".{" + this.name.location() + "}", this.name)),
                     () -> p_81560_.reportProblem("Unknown condition table called " + this.name.location())
                 );
         }
