--- a/net/minecraft/world/level/storage/loot/predicates/InvertedLootItemCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/InvertedLootItemCondition.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.predicates;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Set;
@@ -9,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 
 public record InvertedLootItemCondition(LootItemCondition term) implements LootItemCondition {
-    public static final Codec<InvertedLootItemCondition> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<InvertedLootItemCondition> CODEC = RecordCodecBuilder.mapCodec(
         p_335371_ -> p_335371_.group(LootItemConditions.DIRECT_CODEC.fieldOf("term").forGetter(InvertedLootItemCondition::term))
                 .apply(p_335371_, InvertedLootItemCondition::new)
     );
