--- a/net/minecraft/world/level/storage/loot/predicates/LootItemConditionType.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LootItemConditionType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.predicates;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 
-public record LootItemConditionType(Codec<? extends LootItemCondition> codec) {
+public record LootItemConditionType(MapCodec<? extends LootItemCondition> codec) {
 }
