--- a/net/minecraft/world/level/storage/loot/predicates/LootItemRandomChanceWithEnchantedBonusCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LootItemRandomChanceWithEnchantedBonusCondition.java
@@ -49,6 +_,9 @@
             enchantmentLevel = EnchantmentHelper.getEnchantmentLevel(this.enchantment, livingKiller);
         }
 
+        // Neo: Allow mods to modify the enchantment level for loot calculations with relevant context.
+        enchantmentLevel = net.neoforged.neoforge.event.EventHooks.getEntityLootEnchantmentLevel(this.enchantment, enchantmentLevel, context);
+
         float chance = enchantmentLevel > 0 ? this.enchantedChance.calculate(enchantmentLevel) : this.unenchantedChance;
         return context.getRandom().nextFloat() < chance;
     }
