--- a/net/minecraft/world/level/storage/loot/predicates/LootItemEntityPropertyCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LootItemEntityPropertyCondition.java
@@ -1,13 +_,12 @@
 package net.minecraft.world.level.storage.loot.predicates;
 
 import com.google.common.collect.ImmutableSet;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Optional;
 import java.util.Set;
 import net.minecraft.advancements.critereon.EntityPredicate;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
@@ -15,12 +_,12 @@
 import net.minecraft.world.phys.Vec3;
 
 public record LootItemEntityPropertyCondition(Optional<EntityPredicate> predicate, LootContext.EntityTarget entityTarget) implements LootItemCondition {
-    public static final Codec<LootItemEntityPropertyCondition> CODEC = RecordCodecBuilder.create(
-        p_298191_ -> p_298191_.group(
-                    ExtraCodecs.strictOptionalField(EntityPredicate.CODEC, "predicate").forGetter(LootItemEntityPropertyCondition::predicate),
+    public static final MapCodec<LootItemEntityPropertyCondition> CODEC = RecordCodecBuilder.mapCodec(
+        p_338171_ -> p_338171_.group(
+                    EntityPredicate.CODEC.optionalFieldOf("predicate").forGetter(LootItemEntityPropertyCondition::predicate),
                     LootContext.EntityTarget.CODEC.fieldOf("entity").forGetter(LootItemEntityPropertyCondition::entityTarget)
                 )
-                .apply(p_298191_, LootItemEntityPropertyCondition::new)
+                .apply(p_338171_, LootItemEntityPropertyCondition::new)
     );
 
     @Override
