--- a/net/minecraft/world/level/storage/loot/predicates/ValueCheckCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/ValueCheckCondition.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.storage.loot.predicates;
 
 import com.google.common.collect.Sets;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Set;
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.providers.number.NumberProviders;
 
 public record ValueCheckCondition(NumberProvider provider, IntRange range) implements LootItemCondition {
-    public static final Codec<ValueCheckCondition> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<ValueCheckCondition> CODEC = RecordCodecBuilder.mapCodec(
         p_298196_ -> p_298196_.group(
                     NumberProviders.CODEC.fieldOf("value").forGetter(ValueCheckCondition::provider),
                     IntRange.CODEC.fieldOf("range").forGetter(ValueCheckCondition::range)
