--- a/net/minecraft/server/level/progress/LoggerChunkProgressListener.java
+++ b/net/minecraft/server/level/progress/LoggerChunkProgressListener.java
@@ -42,7 +_,7 @@
     @Override
     public void onStatusChange(ChunkPos p_9633_, @Nullable ChunkStatus p_332174_) {
         if (p_332174_ == ChunkStatus.FULL) {
-            ++this.count;
+            this.count++;
         }
 
         int i = this.getProgress();
