--- a/net/minecraft/world/level/block/state/properties/BlockSetType.java
+++ b/net/minecraft/world/level/block/state/properties/BlockSetType.java
@@ -6,7 +_,6 @@
 import java.util.stream.Stream;
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.sounds.SoundEvents;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.block.SoundType;
 
 public record BlockSetType(
@@ -26,7 +_,7 @@
     SoundEvent buttonClickOn
 ) {
     private static final Map<String, BlockSetType> TYPES = new Object2ObjectArrayMap<>();
-    public static final Codec<BlockSetType> CODEC = ExtraCodecs.stringResolverCodec(BlockSetType::name, TYPES::get);
+    public static final Codec<BlockSetType> CODEC = Codec.stringResolver(BlockSetType::name, TYPES::get);
     public static final BlockSetType IRON = register(
         new BlockSetType(
             "iron",
@@ -162,24 +_,6 @@
     public static final BlockSetType WARPED = register(
         new BlockSetType(
             "warped",
-            true,
-            true,
-            true,
-            BlockSetType.PressurePlateSensitivity.EVERYTHING,
-            SoundType.NETHER_WOOD,
-            SoundEvents.NETHER_WOOD_DOOR_CLOSE,
-            SoundEvents.NETHER_WOOD_DOOR_OPEN,
-            SoundEvents.NETHER_WOOD_TRAPDOOR_CLOSE,
-            SoundEvents.NETHER_WOOD_TRAPDOOR_OPEN,
-            SoundEvents.NETHER_WOOD_PRESSURE_PLATE_CLICK_OFF,
-            SoundEvents.NETHER_WOOD_PRESSURE_PLATE_CLICK_ON,
-            SoundEvents.NETHER_WOOD_BUTTON_CLICK_OFF,
-            SoundEvents.NETHER_WOOD_BUTTON_CLICK_ON
-        )
-    );
-    public static final BlockSetType POTATO = register(
-        new BlockSetType(
-            "potato",
             true,
             true,
             true,
