--- a/net/minecraft/world/level/block/state/properties/EnumProperty.java
+++ b/net/minecraft/world/level/block/state/properties/EnumProperty.java
@@ -19,7 +_,7 @@
         super(p_61579_, p_61580_);
         this.values = ImmutableSet.copyOf(p_61581_);
 
-        for(T t : p_61581_) {
+        for (T t : p_61581_) {
             String s = t.getSerializedName();
             if (this.names.containsKey(s)) {
                 throw new IllegalArgumentException("Multiple values have the same name '" + s + "'");
@@ -48,7 +_,7 @@
         if (this == p_61606_) {
             return true;
         } else {
-            if (p_61606_ instanceof EnumProperty enumproperty && super.equals(p_61606_)) {
+            if (p_61606_ instanceof EnumProperty<?> enumproperty && super.equals(p_61606_)) {
                 return this.values.equals(enumproperty.values) && this.names.equals(enumproperty.names);
             }
 
