--- a/net/minecraft/server/rcon/PktUtils.java
+++ b/net/minecraft/server/rcon/PktUtils.java
@@ -10,8 +_,8 @@
         int i = p_11491_ - 1;
         int j = p_11490_ > i ? i : p_11490_;
 
-        while(0 != p_11489_[j] && j < i) {
-            ++j;
+        while (0 != p_11489_[j] && j < i) {
+            j++;
         }
 
         return new String(p_11489_, p_11490_, j - p_11490_, StandardCharsets.UTF_8);
