--- a/net/minecraft/client/gui/layouts/EqualSpacingLayout.java
+++ b/net/minecraft/client/gui/layouts/EqualSpacingLayout.java
@@ -31,7 +_,7 @@
             int i = 0;
             int j = this.orientation.getSecondaryLength(this);
 
-            for(EqualSpacingLayout.ChildContainer equalspacinglayout$childcontainer : this.children) {
+            for (EqualSpacingLayout.ChildContainer equalspacinglayout$childcontainer : this.children) {
                 i += this.orientation.getPrimaryLength(equalspacinglayout$childcontainer);
                 j = Math.max(j, this.orientation.getSecondaryLength(equalspacinglayout$childcontainer));
             }
@@ -42,25 +_,24 @@
             EqualSpacingLayout.ChildContainer equalspacinglayout$childcontainer1 = iterator.next();
             this.orientation.setPrimaryPosition(equalspacinglayout$childcontainer1, l);
             l += this.orientation.getPrimaryLength(equalspacinglayout$childcontainer1);
-            EqualSpacingLayout.ChildContainer equalspacinglayout$childcontainer2;
             if (this.children.size() >= 2) {
-                for(Divisor divisor = new Divisor(k, this.children.size() - 1);
-                    divisor.hasNext();
-                    l += this.orientation.getPrimaryLength(equalspacinglayout$childcontainer2)
-                ) {
+                Divisor divisor = new Divisor(k, this.children.size() - 1);
+
+                while (divisor.hasNext()) {
                     l += divisor.nextInt();
-                    equalspacinglayout$childcontainer2 = iterator.next();
+                    EqualSpacingLayout.ChildContainer equalspacinglayout$childcontainer2 = iterator.next();
                     this.orientation.setPrimaryPosition(equalspacinglayout$childcontainer2, l);
+                    l += this.orientation.getPrimaryLength(equalspacinglayout$childcontainer2);
                 }
             }
 
             int i1 = this.orientation.getSecondaryPosition(this);
 
-            for(EqualSpacingLayout.ChildContainer equalspacinglayout$childcontainer3 : this.children) {
+            for (EqualSpacingLayout.ChildContainer equalspacinglayout$childcontainer3 : this.children) {
                 this.orientation.setSecondaryPosition(equalspacinglayout$childcontainer3, i1, j);
             }
 
-            switch(this.orientation) {
+            switch (this.orientation) {
                 case HORIZONTAL:
                     this.height = j;
                     break;
@@ -109,35 +_,35 @@
         VERTICAL;
 
         int getPrimaryLength(LayoutElement p_295781_) {
-            return switch(this) {
+            return switch (this) {
                 case HORIZONTAL -> p_295781_.getWidth();
                 case VERTICAL -> p_295781_.getHeight();
             };
         }
 
         int getPrimaryLength(EqualSpacingLayout.ChildContainer p_295365_) {
-            return switch(this) {
+            return switch (this) {
                 case HORIZONTAL -> p_295365_.getWidth();
                 case VERTICAL -> p_295365_.getHeight();
             };
         }
 
         int getSecondaryLength(LayoutElement p_294949_) {
-            return switch(this) {
+            return switch (this) {
                 case HORIZONTAL -> p_294949_.getHeight();
                 case VERTICAL -> p_294949_.getWidth();
             };
         }
 
         int getSecondaryLength(EqualSpacingLayout.ChildContainer p_295288_) {
-            return switch(this) {
+            return switch (this) {
                 case HORIZONTAL -> p_295288_.getHeight();
                 case VERTICAL -> p_295288_.getWidth();
             };
         }
 
         void setPrimaryPosition(EqualSpacingLayout.ChildContainer p_295434_, int p_294763_) {
-            switch(this) {
+            switch (this) {
                 case HORIZONTAL:
                     p_295434_.setX(p_294763_, p_295434_.getWidth());
                     break;
@@ -147,7 +_,7 @@
         }
 
         void setSecondaryPosition(EqualSpacingLayout.ChildContainer p_295472_, int p_295477_, int p_296148_) {
-            switch(this) {
+            switch (this) {
                 case HORIZONTAL:
                     p_295472_.setY(p_295477_, p_296148_);
                     break;
@@ -157,14 +_,14 @@
         }
 
         int getPrimaryPosition(LayoutElement p_294749_) {
-            return switch(this) {
+            return switch (this) {
                 case HORIZONTAL -> p_294749_.getX();
                 case VERTICAL -> p_294749_.getY();
             };
         }
 
         int getSecondaryPosition(LayoutElement p_295927_) {
-            return switch(this) {
+            return switch (this) {
                 case HORIZONTAL -> p_295927_.getY();
                 case VERTICAL -> p_295927_.getX();
             };
