--- a/net/minecraft/world/level/storage/loot/entries/CompositeEntryBase.java
+++ b/net/minecraft/world/level/storage/loot/entries/CompositeEntryBase.java
@@ -1,11 +_,10 @@
 package net.minecraft.world.level.storage.loot.entries;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
 import java.util.function.Consumer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.ValidationContext;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
@@ -27,7 +_,7 @@
             p_79434_.reportProblem("Empty children list");
         }
 
-        for(int i = 0; i < this.children.size(); ++i) {
+        for (int i = 0; i < this.children.size(); i++) {
             this.children.get(i).validate(p_79434_.forChild(".entry[" + i + "]"));
         }
     }
@@ -39,13 +_,11 @@
         return !this.canRun(p_79439_) ? false : this.composedChildren.expand(p_79439_, p_79440_);
     }
 
-    public static <T extends CompositeEntryBase> Codec<T> createCodec(CompositeEntryBase.CompositeEntryConstructor<T> p_299248_) {
-        return RecordCodecBuilder.create(
-            p_298970_ -> p_298970_.group(
-                        ExtraCodecs.strictOptionalField(LootPoolEntries.CODEC.listOf(), "children", List.of()).forGetter(p_299120_ -> p_299120_.children)
-                    )
-                    .and(commonFields(p_298970_).t1())
-                    .apply(p_298970_, p_299248_::create)
+    public static <T extends CompositeEntryBase> MapCodec<T> createCodec(CompositeEntryBase.CompositeEntryConstructor<T> p_299248_) {
+        return RecordCodecBuilder.mapCodec(
+            p_338125_ -> p_338125_.group(LootPoolEntries.CODEC.listOf().optionalFieldOf("children", List.of()).forGetter(p_299120_ -> p_299120_.children))
+                    .and(commonFields(p_338125_).t1())
+                    .apply(p_338125_, p_299248_::create)
         );
     }
 
