--- a/net/minecraft/world/level/storage/loot/entries/DynamicLoot.java
+++ b/net/minecraft/world/level/storage/loot/entries/DynamicLoot.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.entries;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class DynamicLoot extends LootPoolSingletonContainer {
-    public static final Codec<DynamicLoot> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<DynamicLoot> CODEC = RecordCodecBuilder.mapCodec(
         p_298006_ -> p_298006_.group(ResourceLocation.CODEC.fieldOf("name").forGetter(p_298012_ -> p_298012_.name))
                 .and(singletonFields(p_298006_))
                 .apply(p_298006_, DynamicLoot::new)
