--- a/net/minecraft/world/level/storage/loot/entries/EmptyLootItem.java
+++ b/net/minecraft/world/level/storage/loot/entries/EmptyLootItem.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.entries;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -11,7 +_,9 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class EmptyLootItem extends LootPoolSingletonContainer {
-    public static final Codec<EmptyLootItem> CODEC = RecordCodecBuilder.create(p_299288_ -> singletonFields(p_299288_).apply(p_299288_, EmptyLootItem::new));
+    public static final MapCodec<EmptyLootItem> CODEC = RecordCodecBuilder.mapCodec(
+        p_299288_ -> singletonFields(p_299288_).apply(p_299288_, EmptyLootItem::new)
+    );
 
     private EmptyLootItem(int p_79519_, int p_79520_, List<LootItemCondition> p_298942_, List<LootItemFunction> p_298310_) {
         super(p_79519_, p_79520_, p_298942_, p_298310_);
