--- a/net/minecraft/world/level/storage/loot/entries/LootPoolSingletonContainer.java
+++ b/net/minecraft/world/level/storage/loot/entries/LootPoolSingletonContainer.java
@@ -8,7 +_,6 @@
 import java.util.List;
 import java.util.function.BiFunction;
 import java.util.function.Consumer;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.util.Mth;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.storage.loot.LootContext;
@@ -46,18 +_,18 @@
         Instance<T> p_299239_
     ) {
         return p_299239_.group(
-                ExtraCodecs.strictOptionalField(Codec.INT, "weight", 1).forGetter(p_299262_ -> p_299262_.weight),
-                ExtraCodecs.strictOptionalField(Codec.INT, "quality", 0).forGetter(p_299272_ -> p_299272_.quality)
+                Codec.INT.optionalFieldOf("weight", Integer.valueOf(1)).forGetter(p_299262_ -> p_299262_.weight),
+                Codec.INT.optionalFieldOf("quality", Integer.valueOf(0)).forGetter(p_299272_ -> p_299272_.quality)
             )
             .and(commonFields(p_299239_).t1())
-            .and(ExtraCodecs.strictOptionalField(LootItemFunctions.ROOT_CODEC.listOf(), "functions", List.of()).forGetter(p_298225_ -> p_298225_.functions));
+            .and(LootItemFunctions.ROOT_CODEC.listOf().optionalFieldOf("functions", List.of()).forGetter(p_298225_ -> p_298225_.functions));
     }
 
     @Override
     public void validate(ValidationContext p_79686_) {
         super.validate(p_79686_);
 
-        for(int i = 0; i < this.functions.size(); ++i) {
+        for (int i = 0; i < this.functions.size(); i++) {
             this.functions.get(i).validate(p_79686_.forChild(".functions[" + i + "]"));
         }
     }
