--- a/net/minecraft/world/level/storage/loot/entries/LootPoolEntryType.java
+++ b/net/minecraft/world/level/storage/loot/entries/LootPoolEntryType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.storage.loot.entries;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 
-public record LootPoolEntryType(Codec<? extends LootPoolEntryContainer> codec) {
+public record LootPoolEntryType(MapCodec<? extends LootPoolEntryContainer> codec) {
 }
