--- a/net/minecraft/world/level/storage/loot/entries/TagEntry.java
+++ b/net/minecraft/world/level/storage/loot/entries/TagEntry.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.storage.loot.entries;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.List;
@@ -16,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class TagEntry extends LootPoolSingletonContainer {
-    public static final Codec<TagEntry> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<TagEntry> CODEC = RecordCodecBuilder.mapCodec(
         p_298033_ -> p_298033_.group(
                     TagKey.codec(Registries.ITEM).fieldOf("name").forGetter(p_298040_ -> p_298040_.tag),
                     Codec.BOOL.fieldOf("expand").forGetter(p_298039_ -> p_298039_.expand)
@@ -42,14 +_,14 @@
 
     @Override
     public void createItemStack(Consumer<ItemStack> p_79854_, LootContext p_79855_) {
-        BuiltInRegistries.ITEM.getTagOrEmpty(this.tag).forEach(p_205094_ -> p_79854_.accept(new ItemStack(p_205094_)));
+        BuiltInRegistries.ITEM.getTagOrEmpty(this.tag).forEach(p_205094_ -> p_79854_.accept(new ItemStack((Holder<Item>)p_205094_)));
     }
 
     private boolean expandTag(LootContext p_79846_, Consumer<LootPoolEntry> p_79847_) {
         if (!this.canRun(p_79846_)) {
             return false;
         } else {
-            for(final Holder<Item> holder : BuiltInRegistries.ITEM.getTagOrEmpty(this.tag)) {
+            for (final Holder<Item> holder : BuiltInRegistries.ITEM.getTagOrEmpty(this.tag)) {
                 p_79847_.accept(new LootPoolSingletonContainer.EntryBase() {
                     @Override
                     public void createItemStack(Consumer<ItemStack> p_79869_, LootContext p_79870_) {
