--- a/net/minecraft/server/bossevents/CustomBossEvent.java
+++ b/net/minecraft/server/bossevents/CustomBossEvent.java
@@ -88,10 +_,10 @@
         Set<UUID> set = Sets.newHashSet();
         Set<ServerPlayer> set1 = Sets.newHashSet();
 
-        for(UUID uuid : this.players) {
+        for (UUID uuid : this.players) {
             boolean flag = false;
 
-            for(ServerPlayer serverplayer : p_136269_) {
+            for (ServerPlayer serverplayer : p_136269_) {
                 if (serverplayer.getUUID().equals(uuid)) {
                     flag = true;
                     break;
@@ -103,10 +_,10 @@
             }
         }
 
-        for(ServerPlayer serverplayer1 : p_136269_) {
+        for (ServerPlayer serverplayer1 : p_136269_) {
             boolean flag1 = false;
 
-            for(UUID uuid2 : this.players) {
+            for (UUID uuid2 : this.players) {
                 if (serverplayer1.getUUID().equals(uuid2)) {
                     flag1 = true;
                     break;
@@ -118,8 +_,8 @@
             }
         }
 
-        for(UUID uuid1 : set) {
-            for(ServerPlayer serverplayer3 : this.getPlayers()) {
+        for (UUID uuid1 : set) {
+            for (ServerPlayer serverplayer3 : this.getPlayers()) {
                 if (serverplayer3.getUUID().equals(uuid1)) {
                     this.removePlayer(serverplayer3);
                     break;
@@ -129,7 +_,7 @@
             this.players.remove(uuid1);
         }
 
-        for(ServerPlayer serverplayer2 : set1) {
+        for (ServerPlayer serverplayer2 : set1) {
             this.addPlayer(serverplayer2);
         }
 
@@ -149,7 +_,7 @@
         compoundtag.putBoolean("CreateWorldFog", this.shouldCreateWorldFog());
         ListTag listtag = new ListTag();
 
-        for(UUID uuid : this.players) {
+        for (UUID uuid : this.players) {
             listtag.add(NbtUtils.createUUID(uuid));
         }
 
@@ -168,7 +_,7 @@
         custombossevent.setPlayBossMusic(p_136273_.getBoolean("PlayBossMusic"));
         custombossevent.setCreateWorldFog(p_136273_.getBoolean("CreateWorldFog"));
 
-        for(Tag tag : p_136273_.getList("Players", 11)) {
+        for (Tag tag : p_136273_.getList("Players", 11)) {
             custombossevent.addOfflinePlayer(NbtUtils.loadUUID(tag));
         }
 
