--- a/net/minecraft/server/bossevents/CustomBossEvents.java
+++ b/net/minecraft/server/bossevents/CustomBossEvents.java
@@ -39,7 +_,7 @@
     public CompoundTag save(HolderLookup.Provider p_330783_) {
         CompoundTag compoundtag = new CompoundTag();
 
-        for(CustomBossEvent custombossevent : this.events.values()) {
+        for (CustomBossEvent custombossevent : this.events.values()) {
             compoundtag.put(custombossevent.getTextId().toString(), custombossevent.save(p_330783_));
         }
 
@@ -47,20 +_,20 @@
     }
 
     public void load(CompoundTag p_136296_, HolderLookup.Provider p_331580_) {
-        for(String s : p_136296_.getAllKeys()) {
+        for (String s : p_136296_.getAllKeys()) {
             ResourceLocation resourcelocation = new ResourceLocation(s);
             this.events.put(resourcelocation, CustomBossEvent.load(p_136296_.getCompound(s), resourcelocation, p_331580_));
         }
     }
 
     public void onPlayerConnect(ServerPlayer p_136294_) {
-        for(CustomBossEvent custombossevent : this.events.values()) {
+        for (CustomBossEvent custombossevent : this.events.values()) {
             custombossevent.onPlayerConnect(p_136294_);
         }
     }
 
     public void onPlayerDisconnect(ServerPlayer p_136306_) {
-        for(CustomBossEvent custombossevent : this.events.values()) {
+        for (CustomBossEvent custombossevent : this.events.values()) {
             custombossevent.onPlayerDisconnect(p_136306_);
         }
     }
