--- a/net/minecraft/world/ticks/ProtoChunkTicks.java
+++ b/net/minecraft/world/ticks/ProtoChunkTicks.java
@@ -41,7 +_,7 @@
     public Tag save(long p_193308_, Function<T, String> p_193309_) {
         ListTag listtag = new ListTag();
 
-        for(SavedTick<T> savedtick : this.ticks) {
+        for (SavedTick<T> savedtick : this.ticks) {
             listtag.add(savedtick.save(p_193309_));
         }
 
