--- a/net/minecraft/world/ticks/SavedTick.java
+++ b/net/minecraft/world/ticks/SavedTick.java
@@ -25,10 +_,8 @@
         public boolean equals(@Nullable SavedTick<?> p_193366_, @Nullable SavedTick<?> p_193367_) {
             if (p_193366_ == p_193367_) {
                 return true;
-            } else if (p_193366_ != null && p_193367_ != null) {
-                return p_193366_.type() == p_193367_.type() && p_193366_.pos().equals(p_193367_.pos());
             } else {
-                return false;
+                return p_193366_ != null && p_193367_ != null ? p_193366_.type() == p_193367_.type() && p_193366_.pos().equals(p_193367_.pos()) : false;
             }
         }
     };
@@ -36,11 +_,11 @@
     public static <T> void loadTickList(ListTag p_193351_, Function<String, Optional<T>> p_193352_, ChunkPos p_193353_, Consumer<SavedTick<T>> p_193354_) {
         long i = p_193353_.toLong();
 
-        for(int j = 0; j < p_193351_.size(); ++j) {
+        for (int j = 0; j < p_193351_.size(); j++) {
             CompoundTag compoundtag = p_193351_.getCompound(j);
             loadTick(compoundtag, p_193352_).ifPresent(p_210665_ -> {
                 if (ChunkPos.asLong(p_210665_.pos()) == i) {
-                    p_193354_.accept(p_210665_);
+                    p_193354_.accept((SavedTick<T>)p_210665_);
                 }
             });
         }
@@ -49,7 +_,7 @@
     public static <T> Optional<SavedTick<T>> loadTick(CompoundTag p_210670_, Function<String, Optional<T>> p_210671_) {
         return p_210671_.apply(p_210670_.getString("i")).map(p_210668_ -> {
             BlockPos blockpos = new BlockPos(p_210670_.getInt("x"), p_210670_.getInt("y"), p_210670_.getInt("z"));
-            return new SavedTick<>(p_210668_, blockpos, p_210670_.getInt("t"), TickPriority.byValue(p_210670_.getInt("p")));
+            return new SavedTick<>((T)p_210668_, blockpos, p_210670_.getInt("t"), TickPriority.byValue(p_210670_.getInt("p")));
         });
     }
 
