--- a/net/minecraft/data/tags/DamageTypeTagsProvider.java
+++ b/net/minecraft/data/tags/DamageTypeTagsProvider.java
@@ -9,9 +_,14 @@
 import net.minecraft.world.damagesource.DamageTypes;
 
 public class DamageTypeTagsProvider extends TagsProvider<DamageType> {
+    /** @deprecated Forge: Use the {@linkplain #DamageTypeTagsProvider(PackOutput, CompletableFuture, String, net.neoforged.neoforge.common.data.ExistingFileHelper) mod id variant} */
+    @Deprecated
     public DamageTypeTagsProvider(PackOutput p_270719_, CompletableFuture<HolderLookup.Provider> p_270256_) {
         super(p_270719_, Registries.DAMAGE_TYPE, p_270256_);
     }
+    public DamageTypeTagsProvider(PackOutput p_270719_, CompletableFuture<HolderLookup.Provider> p_270256_, String modId, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.common.data.ExistingFileHelper existingFileHelper) {
+        super(p_270719_, Registries.DAMAGE_TYPE, p_270256_, modId, existingFileHelper);
+    }
 
     @Override
     protected void addTags(HolderLookup.Provider p_270108_) {
@@ -35,8 +_,7 @@
                 DamageTypes.FELL_OUT_OF_WORLD,
                 DamageTypes.GENERIC_KILL,
                 DamageTypes.SONIC_BOOM,
-                DamageTypes.OUTSIDE_BORDER,
-                DamageTypes.POTATO_MAGIC
+                DamageTypes.OUTSIDE_BORDER
             );
         this.tag(DamageTypeTags.BYPASSES_SHIELD).addTag(DamageTypeTags.BYPASSES_ARMOR).add(DamageTypes.FALLING_ANVIL, DamageTypes.FALLING_STALACTITE);
         this.tag(DamageTypeTags.BYPASSES_INVULNERABILITY).add(DamageTypes.FELL_OUT_OF_WORLD, DamageTypes.GENERIC_KILL);
@@ -53,8 +_,7 @@
                 DamageTypes.UNATTRIBUTED_FIREBALL,
                 DamageTypes.FIREBALL,
                 DamageTypes.WITHER_SKULL,
-                DamageTypes.THROWN,
-                DamageTypes.POTATO_MAGIC
+                DamageTypes.THROWN
             );
         this.tag(DamageTypeTags.WITCH_RESISTANT_TO).add(DamageTypes.MAGIC, DamageTypes.INDIRECT_MAGIC, DamageTypes.SONIC_BOOM, DamageTypes.THORNS);
         this.tag(DamageTypeTags.IS_EXPLOSION).add(DamageTypes.FIREWORKS, DamageTypes.EXPLOSION, DamageTypes.PLAYER_EXPLOSION, DamageTypes.BAD_RESPAWN_POINT);
@@ -89,5 +_,6 @@
                 DamageTypes.THORNS,
                 DamageTypes.WITHER
             );
+        this.tag(DamageTypeTags.IS_PLAYER_ATTACK).add(DamageTypes.PLAYER_ATTACK);
     }
 }
