--- a/net/minecraft/data/tags/EntityTypeTagsProvider.java
+++ b/net/minecraft/data/tags/EntityTypeTagsProvider.java
@@ -8,8 +_,13 @@
 import net.minecraft.world.entity.EntityType;
 
 public class EntityTypeTagsProvider extends IntrinsicHolderTagsProvider<EntityType<?>> {
+    /** @deprecated Forge: Use the {@linkplain #EntityTypeTagsProvider(PackOutput, CompletableFuture, String, net.neoforged.neoforge.common.data.ExistingFileHelper) mod id variant} */
+    @Deprecated
     public EntityTypeTagsProvider(PackOutput p_256095_, CompletableFuture<HolderLookup.Provider> p_256572_) {
-        super(p_256095_, Registries.ENTITY_TYPE, p_256572_, p_256665_ -> p_256665_.builtInRegistryHolder().key());
+        this(p_256095_, p_256572_, "vanilla", null);
+    }
+    public EntityTypeTagsProvider(PackOutput p_256095_, CompletableFuture<HolderLookup.Provider> p_256572_, String modId, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.common.data.ExistingFileHelper existingFileHelper) {
+        super(p_256095_, Registries.ENTITY_TYPE, p_256572_, p_256665_ -> p_256665_.builtInRegistryHolder().key(), modId, existingFileHelper);
     }
 
     @Override
@@ -23,8 +_,7 @@
                 EntityType.ZOMBIFIED_PIGLIN,
                 EntityType.ZOGLIN,
                 EntityType.DROWNED,
-                EntityType.HUSK,
-                EntityType.POISONOUS_POTATO_ZOMBIE
+                EntityType.HUSK
             );
         this.tag(EntityTypeTags.RAIDERS)
             .add(EntityType.EVOKER, EntityType.PILLAGER, EntityType.RAVAGER, EntityType.VINDICATOR, EntityType.ILLUSIONER, EntityType.WITCH);
@@ -48,15 +_,12 @@
             .add(
                 EntityType.TROPICAL_FISH, EntityType.PUFFERFISH, EntityType.SALMON, EntityType.COD, EntityType.SQUID, EntityType.GLOW_SQUID, EntityType.TADPOLE
             );
-        this.tag(EntityTypeTags.AXOLOTL_ALWAYS_HOSTILES)
-            .add(EntityType.DROWNED, EntityType.GUARDIAN, EntityType.ELDER_GUARDIAN, EntityType.TOXIFIN, EntityType.PLAGUEWHALE);
+        this.tag(EntityTypeTags.AXOLOTL_ALWAYS_HOSTILES).add(EntityType.DROWNED, EntityType.GUARDIAN, EntityType.ELDER_GUARDIAN);
         this.tag(EntityTypeTags.FREEZE_IMMUNE_ENTITY_TYPES).add(EntityType.STRAY, EntityType.POLAR_BEAR, EntityType.SNOW_GOLEM, EntityType.WITHER);
         this.tag(EntityTypeTags.FREEZE_HURTS_EXTRA_TYPES).add(EntityType.STRIDER, EntityType.BLAZE, EntityType.MAGMA_CUBE);
         this.tag(EntityTypeTags.CAN_BREATHE_UNDER_WATER)
             .addTag(EntityTypeTags.UNDEAD)
             .add(
-                EntityType.PLAGUEWHALE,
-                EntityType.TOXIFIN,
                 EntityType.AXOLOTL,
                 EntityType.FROG,
                 EntityType.GUARDIAN,
@@ -79,7 +_,6 @@
                 EntityType.SHULKER,
                 EntityType.ALLAY,
                 EntityType.BAT,
-                EntityType.BATATO,
                 EntityType.BEE,
                 EntityType.BLAZE,
                 EntityType.CAT,
@@ -112,7 +_,7 @@
             .add(EntityType.TURTLE)
             .add(EntityType.AXOLOTL)
             .add(EntityType.GUARDIAN)
-            .add(EntityType.ELDER_GUARDIAN, EntityType.TOXIFIN, EntityType.PLAGUEWHALE)
+            .add(EntityType.ELDER_GUARDIAN)
             .add(EntityType.COD)
             .add(EntityType.PUFFERFISH)
             .add(EntityType.SALMON)
@@ -134,7 +_,7 @@
         this.tag(EntityTypeTags.NOT_SCARY_FOR_PUFFERFISH)
             .add(EntityType.TURTLE)
             .add(EntityType.GUARDIAN)
-            .add(EntityType.ELDER_GUARDIAN, EntityType.TOXIFIN, EntityType.PLAGUEWHALE)
+            .add(EntityType.ELDER_GUARDIAN)
             .add(EntityType.COD)
             .add(EntityType.PUFFERFISH)
             .add(EntityType.SALMON)
