--- a/net/minecraft/data/tags/StructureTagsProvider.java
+++ b/net/minecraft/data/tags/StructureTagsProvider.java
@@ -9,15 +_,19 @@
 import net.minecraft.world.level.levelgen.structure.Structure;
 
 public class StructureTagsProvider extends TagsProvider<Structure> {
+    /** @deprecated Forge: Use the {@linkplain #StructureTagsProvider(PackOutput, CompletableFuture, String, net.neoforged.neoforge.common.data.ExistingFileHelper) mod id variant} */
+    @Deprecated
     public StructureTagsProvider(PackOutput p_256522_, CompletableFuture<HolderLookup.Provider> p_256661_) {
         super(p_256522_, Registries.STRUCTURE, p_256661_);
     }
+    public StructureTagsProvider(PackOutput p_256522_, CompletableFuture<HolderLookup.Provider> p_256661_, String modId, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.common.data.ExistingFileHelper existingFileHelper) {
+        super(p_256522_, Registries.STRUCTURE, p_256661_, modId, existingFileHelper);
+    }
 
     @Override
     protected void addTags(HolderLookup.Provider p_256087_) {
         this.tag(StructureTags.VILLAGE)
             .add(BuiltinStructures.VILLAGE_PLAINS)
-            .add(BuiltinStructures.VILLAGE_POTATO)
             .add(BuiltinStructures.VILLAGE_DESERT)
             .add(BuiltinStructures.VILLAGE_SAVANNA)
             .add(BuiltinStructures.VILLAGE_SNOWY)
@@ -33,8 +_,6 @@
             .add(BuiltinStructures.RUINED_PORTAL_OCEAN)
             .add(BuiltinStructures.RUINED_PORTAL_STANDARD)
             .add(BuiltinStructures.RUINED_PORTAL_SWAMP);
-        this.tag(StructureTags.RUINED_PORTATOL).add(BuiltinStructures.RUINED_PORTATOL);
-        this.tag(StructureTags.COLOSSEUM).add(BuiltinStructures.COLOSSEUM);
         this.tag(StructureTags.CATS_SPAWN_IN).add(BuiltinStructures.SWAMP_HUT);
         this.tag(StructureTags.CATS_SPAWN_AS_BLACK).add(BuiltinStructures.SWAMP_HUT);
         this.tag(StructureTags.EYE_OF_ENDER_LOCATED).add(BuiltinStructures.STRONGHOLD);
