--- a/net/minecraft/data/tags/TagsProvider.java
+++ b/net/minecraft/data/tags/TagsProvider.java
@@ -32,26 +_,49 @@
     private final CompletableFuture<TagsProvider.TagLookup<T>> parentProvider;
     protected final ResourceKey<? extends Registry<T>> registryKey;
     protected final Map<Identifier, TagBuilder> builders = Maps.newLinkedHashMap();
+    protected final String modId;
 
+    /**
+     * @deprecated Forge: Use the {@linkplain #TagsProvider(PackOutput, ResourceKey, CompletableFuture, String) mod id variant}
+     */
+    @Deprecated
     protected TagsProvider(PackOutput output, ResourceKey<? extends Registry<T>> registryKey, CompletableFuture<HolderLookup.Provider> lookupProvider) {
-        this(output, registryKey, lookupProvider, CompletableFuture.completedFuture(TagsProvider.TagLookup.empty()));
+        this(output, registryKey, lookupProvider, "vanilla");
+    }
+    protected TagsProvider(PackOutput output, ResourceKey<? extends Registry<T>> registryKey, CompletableFuture<HolderLookup.Provider> lookupProvider, String modId) {
+        this(output, registryKey, lookupProvider, CompletableFuture.completedFuture(TagsProvider.TagLookup.empty()), modId);
     }
 
+    /**
+     * @deprecated Forge: Use the {@linkplain #TagsProvider(PackOutput, ResourceKey, CompletableFuture, CompletableFuture, String) mod id variant}
+     */
+    @Deprecated
     protected TagsProvider(
         PackOutput output,
         ResourceKey<? extends Registry<T>> registryKey,
         CompletableFuture<HolderLookup.Provider> lookupProvider,
         CompletableFuture<TagsProvider.TagLookup<T>> parentProvider
     ) {
+        this(output, registryKey, lookupProvider, parentProvider, "vanilla");
+    }
+
+    protected TagsProvider(PackOutput output, ResourceKey<? extends Registry<T>> registryKey, CompletableFuture<HolderLookup.Provider> lookupProvider, CompletableFuture<TagsProvider.TagLookup<T>> parentProvider, String modId) {
         this.pathProvider = output.createRegistryTagsPathProvider(registryKey);
         this.registryKey = registryKey;
         this.parentProvider = parentProvider;
         this.lookupProvider = lookupProvider;
+        this.modId = modId;
+    }
+
+    // Forge: Allow customizing the path for a given tag or returning null
+    @org.jspecify.annotations.Nullable
+    protected Path getPath(Identifier id) {
+        return this.pathProvider.json(id);
     }
 
     @Override
     public String getName() {
-        return "Tags for " + this.registryKey.identifier();
+        return "Tags for " + this.registryKey.identifier() + " mod id " + this.modId;
     }
 
     protected abstract void addTags(HolderLookup.Provider registries);
@@ -84,7 +_,10 @@
                                     Identifier identifier = entry.getKey();
                                     TagBuilder tagbuilder = entry.getValue();
                                     List<TagEntry> list = tagbuilder.build();
-                                    List<TagEntry> list1 = list.stream().filter(e -> !e.verifyIfPresent(predicate, predicate1)).toList();
+                                    List<TagEntry> list1 = java.util.stream.Stream.concat(list.stream(), tagbuilder.getRemoveEntries())
+                                               // Neo: Assume tags from other namespaces always exists
+                                              .filter((e) -> e.getId().getNamespace().equals(modId) && !e.verifyIfPresent(predicate, predicate1))
+                                              .toList();
                                     if (!list1.isEmpty()) {
                                         throw new IllegalArgumentException(
                                             String.format(
@@ -95,8 +_,10 @@
                                             )
                                         );
                                     } else {
-                                        Path path = this.pathProvider.json(identifier);
-                                        return DataProvider.saveStable(cache, c.contents, TagFile.CODEC, new TagFile(list, false), path);
+                                        Path path = this.getPath(identifier);
+                                        if (path == null) return CompletableFuture.completedFuture(null); // Neo: Allow running this data provider without writing it. Recipe provider needs valid tags.
+                                        var removed = tagbuilder.getRemoveEntries().toList();
+                                        return DataProvider.saveStable(cache, c.contents, TagFile.CODEC, new TagFile(list, tagbuilder.isReplace(), removed), path);
                                     }
                                 }
                             )
