--- a/net/minecraft/data/tags/UpdateOneTwentyOneEntityTypeTagsProvider.java
+++ b/net/minecraft/data/tags/UpdateOneTwentyOneEntityTypeTagsProvider.java
@@ -31,5 +_,7 @@
                 EntityType.SLIME
             );
         this.tag(EntityTypeTags.SKELETONS).add(EntityType.BOGGED);
+        this.tag(EntityTypeTags.IMMUNE_TO_INFESTED).add(EntityType.SILVERFISH);
+        this.tag(EntityTypeTags.IMMUNE_TO_OOZING).add(EntityType.SLIME);
     }
 }
