--- a/net/minecraft/data/tags/VanillaBlockTagsProvider.java
+++ b/net/minecraft/data/tags/VanillaBlockTagsProvider.java
@@ -5,7 +_,6 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.data.PackOutput;
 import net.minecraft.tags.BlockTags;
-import net.minecraft.world.item.DyeColor;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 
@@ -37,7 +_,6 @@
             );
         this.tag(BlockTags.PLANKS)
             .add(
-                Blocks.POTATO_PLANKS,
                 Blocks.OAK_PLANKS,
                 Blocks.SPRUCE_PLANKS,
                 Blocks.BIRCH_PLANKS,
@@ -53,7 +_,6 @@
         this.tag(BlockTags.STONE_BRICKS).add(Blocks.STONE_BRICKS, Blocks.MOSSY_STONE_BRICKS, Blocks.CRACKED_STONE_BRICKS, Blocks.CHISELED_STONE_BRICKS);
         this.tag(BlockTags.WOODEN_BUTTONS)
             .add(
-                Blocks.POTATO_BUTTON,
                 Blocks.OAK_BUTTON,
                 Blocks.SPRUCE_BUTTON,
                 Blocks.BIRCH_BUTTON,
@@ -89,7 +_,6 @@
             );
         this.tag(BlockTags.WOODEN_DOORS)
             .add(
-                Blocks.POTATO_DOOR,
                 Blocks.OAK_DOOR,
                 Blocks.SPRUCE_DOOR,
                 Blocks.BIRCH_DOOR,
@@ -104,7 +_,6 @@
             );
         this.tag(BlockTags.WOODEN_STAIRS)
             .add(
-                Blocks.POTATO_STAIRS,
                 Blocks.OAK_STAIRS,
                 Blocks.SPRUCE_STAIRS,
                 Blocks.BIRCH_STAIRS,
@@ -126,7 +_,6 @@
                 Blocks.ACACIA_SLAB,
                 Blocks.DARK_OAK_SLAB,
                 Blocks.CRIMSON_SLAB,
-                Blocks.POTATO_SLAB,
                 Blocks.WARPED_SLAB,
                 Blocks.MANGROVE_SLAB,
                 Blocks.BAMBOO_SLAB,
@@ -141,7 +_,6 @@
                 Blocks.BIRCH_FENCE,
                 Blocks.JUNGLE_FENCE,
                 Blocks.CRIMSON_FENCE,
-                Blocks.POTATO_FENCE,
                 Blocks.WARPED_FENCE,
                 Blocks.MANGROVE_FENCE,
                 Blocks.BAMBOO_FENCE,
@@ -190,7 +_,7 @@
             .add(Blocks.DARK_OAK_LOG)
             .add(Blocks.MANGROVE_LOG)
             .add(Blocks.CHERRY_LOG);
-        this.tag(BlockTags.LOGS).addTag(BlockTags.LOGS_THAT_BURN).addTag(BlockTags.CRIMSON_STEMS).addTag(BlockTags.WARPED_STEMS).add(Blocks.POTATO_STEM);
+        this.tag(BlockTags.LOGS).addTag(BlockTags.LOGS_THAT_BURN).addTag(BlockTags.CRIMSON_STEMS).addTag(BlockTags.WARPED_STEMS);
         this.tag(BlockTags.ANVIL).add(Blocks.ANVIL, Blocks.CHIPPED_ANVIL, Blocks.DAMAGED_ANVIL);
         this.tag(BlockTags.SMALL_FLOWERS)
             .add(
@@ -206,17 +_,13 @@
                 Blocks.OXEYE_DAISY,
                 Blocks.CORNFLOWER,
                 Blocks.LILY_OF_THE_VALLEY,
-                Blocks.POTATO_FLOWER,
                 Blocks.WITHER_ROSE,
                 Blocks.TORCHFLOWER
             );
         this.tag(BlockTags.DIRT)
             .add(
                 Blocks.DIRT,
-                Blocks.TERREDEPOMME,
                 Blocks.GRASS_BLOCK,
-                Blocks.PEELGRASS_BLOCK,
-                Blocks.CORRUPTED_PEELGRASS_BLOCK,
                 Blocks.PODZOL,
                 Blocks.COARSE_DIRT,
                 Blocks.MYCELIUM,
@@ -230,7 +_,6 @@
             .addTag(BlockTags.DIRT)
             .add(
                 Blocks.SAND,
-                Blocks.GRAVTATER,
                 Blocks.RED_SAND,
                 Blocks.GRAVEL,
                 Blocks.BROWN_MUSHROOM,
@@ -274,7 +_,6 @@
                 Blocks.POTTED_CACTUS,
                 Blocks.POTTED_CORNFLOWER,
                 Blocks.POTTED_LILY_OF_THE_VALLEY,
-                Blocks.POTTED_POTATO_FLOWER,
                 Blocks.POTTED_WITHER_ROSE,
                 Blocks.POTTED_BAMBOO,
                 Blocks.POTTED_CRIMSON_FUNGUS,
@@ -332,7 +_,6 @@
                 Blocks.DARK_OAK_PRESSURE_PLATE,
                 Blocks.CRIMSON_PRESSURE_PLATE,
                 Blocks.WARPED_PRESSURE_PLATE,
-                Blocks.POTATO_PRESSURE_PLATE,
                 Blocks.MANGROVE_PRESSURE_PLATE,
                 Blocks.BAMBOO_PRESSURE_PLATE,
                 Blocks.CHERRY_PRESSURE_PLATE
@@ -346,11 +_,6 @@
             .addTag(BlockTags.WOODEN_STAIRS)
             .add(Blocks.BAMBOO_MOSAIC_STAIRS)
             .add(
-                Blocks.POTONE_STAIRS,
-                Blocks.TATERSTONE_STAIRS,
-                Blocks.EXPIRED_BAKED_POTATO_BRICK_STAIRS,
-                Blocks.CHARRED_BAKED_POTATO_BRICK_STAIRS,
-                Blocks.BAKED_POTATO_BRICK_STAIRS,
                 Blocks.COBBLESTONE_STAIRS,
                 Blocks.SANDSTONE_STAIRS,
                 Blocks.NETHER_BRICK_STAIRS,
@@ -397,11 +_,6 @@
             .addTag(BlockTags.WOODEN_SLABS)
             .add(Blocks.BAMBOO_MOSAIC_SLAB)
             .add(
-                Blocks.POTONE_SLAB,
-                Blocks.BAKED_POTATO_BRICK_SLAB,
-                Blocks.CHARRED_BAKED_POTATO_BRICK_SLAB,
-                Blocks.TATERSTONE_SLAB,
-                Blocks.EXPIRED_BAKED_POTATO_BRICK_SLAB,
                 Blocks.STONE_SLAB,
                 Blocks.SMOOTH_STONE_SLAB,
                 Blocks.STONE_BRICK_SLAB,
@@ -450,11 +_,6 @@
             );
         this.tag(BlockTags.WALLS)
             .add(
-                Blocks.POTONE_WALL,
-                Blocks.TATERSTONE_WALL,
-                Blocks.BAKED_POTATO_BRICK_WALL,
-                Blocks.CHARRED_BAKED_POTATO_BRICK_WALL,
-                Blocks.EXPIRED_BAKED_POTATO_BRICK_WALL,
                 Blocks.COBBLESTONE_WALL,
                 Blocks.MOSSY_COBBLESTONE_WALL,
                 Blocks.BRICK_WALL,
@@ -484,16 +_,15 @@
             .add(Blocks.TUBE_CORAL_FAN, Blocks.BRAIN_CORAL_FAN, Blocks.BUBBLE_CORAL_FAN, Blocks.FIRE_CORAL_FAN, Blocks.HORN_CORAL_FAN);
         this.tag(BlockTags.WALL_CORALS)
             .add(Blocks.TUBE_CORAL_WALL_FAN, Blocks.BRAIN_CORAL_WALL_FAN, Blocks.BUBBLE_CORAL_WALL_FAN, Blocks.FIRE_CORAL_WALL_FAN, Blocks.HORN_CORAL_WALL_FAN);
-        this.tag(BlockTags.SAND).add(Blocks.SAND, Blocks.RED_SAND, Blocks.SUSPICIOUS_SAND, Blocks.GRAVTATER);
-        this.tag(BlockTags.SMELTS_TO_GLASS).add(Blocks.SAND, Blocks.GRAVTATER, Blocks.RED_SAND);
+        this.tag(BlockTags.SAND).add(Blocks.SAND, Blocks.RED_SAND, Blocks.SUSPICIOUS_SAND);
+        this.tag(BlockTags.SMELTS_TO_GLASS).add(Blocks.SAND, Blocks.RED_SAND);
         this.tag(BlockTags.RAILS).add(Blocks.RAIL, Blocks.POWERED_RAIL, Blocks.DETECTOR_RAIL, Blocks.ACTIVATOR_RAIL);
         this.tag(BlockTags.CORAL_BLOCKS)
             .add(Blocks.TUBE_CORAL_BLOCK, Blocks.BRAIN_CORAL_BLOCK, Blocks.BUBBLE_CORAL_BLOCK, Blocks.FIRE_CORAL_BLOCK, Blocks.HORN_CORAL_BLOCK);
         this.tag(BlockTags.ICE).add(Blocks.ICE, Blocks.PACKED_ICE, Blocks.BLUE_ICE, Blocks.FROSTED_ICE);
-        this.tag(BlockTags.VALID_SPAWN).add(Blocks.GRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.PODZOL);
+        this.tag(BlockTags.VALID_SPAWN).add(Blocks.GRASS_BLOCK, Blocks.PODZOL);
         this.tag(BlockTags.LEAVES)
             .add(
-                Blocks.POTATO_LEAVES,
                 Blocks.JUNGLE_LEAVES,
                 Blocks.OAK_LEAVES,
                 Blocks.SPRUCE_LEAVES,
@@ -528,7 +_,6 @@
             );
         this.tag(BlockTags.WOODEN_TRAPDOORS)
             .add(
-                Blocks.POTATO_TRAPDOOR,
                 Blocks.ACACIA_TRAPDOOR,
                 Blocks.BIRCH_TRAPDOOR,
                 Blocks.DARK_OAK_TRAPDOOR,
@@ -549,7 +_,6 @@
             .add(Blocks.BAMBOO, Blocks.BAMBOO_SAPLING, Blocks.GRAVEL, Blocks.SUSPICIOUS_GRAVEL);
         this.tag(BlockTags.STANDING_SIGNS)
             .add(
-                Blocks.POTATO_SIGN,
                 Blocks.OAK_SIGN,
                 Blocks.SPRUCE_SIGN,
                 Blocks.BIRCH_SIGN,
@@ -564,7 +_,6 @@
             );
         this.tag(BlockTags.WALL_SIGNS)
             .add(
-                Blocks.POTATO_WALL_SIGN,
                 Blocks.OAK_WALL_SIGN,
                 Blocks.SPRUCE_WALL_SIGN,
                 Blocks.BIRCH_WALL_SIGN,
@@ -580,7 +_,6 @@
         this.tag(BlockTags.SIGNS).addTag(BlockTags.STANDING_SIGNS).addTag(BlockTags.WALL_SIGNS);
         this.tag(BlockTags.CEILING_HANGING_SIGNS)
             .add(
-                Blocks.POTATO_HANGING_SIGN,
                 Blocks.OAK_HANGING_SIGN,
                 Blocks.SPRUCE_HANGING_SIGN,
                 Blocks.BIRCH_HANGING_SIGN,
@@ -595,7 +_,6 @@
             );
         this.tag(BlockTags.WALL_HANGING_SIGNS)
             .add(
-                Blocks.POTATO_WALL_HANGING_SIGN,
                 Blocks.OAK_WALL_HANGING_SIGN,
                 Blocks.SPRUCE_WALL_HANGING_SIGN,
                 Blocks.BIRCH_WALL_HANGING_SIGN,
@@ -693,8 +_,6 @@
                 Blocks.TORCHFLOWER_CROP,
                 Blocks.PITCHER_CROP
             );
-        this.tag(BlockTags.GROWS_POTATOES)
-            .add(Blocks.POISON_FARMLAND, Blocks.FARMLAND, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.GRAVTATER);
         this.tag(BlockTags.BEE_GROWABLES).addTag(BlockTags.CROPS).add(Blocks.SWEET_BERRY_BUSH).add(Blocks.CAVE_VINES).add(Blocks.CAVE_VINES_PLANT);
         this.tag(BlockTags.SHULKER_BOXES)
             .add(
@@ -716,12 +_,11 @@
                 Blocks.WHITE_SHULKER_BOX,
                 Blocks.YELLOW_SHULKER_BOX
             );
-        this.tag(BlockTags.PORTALS).add(Blocks.NETHER_PORTAL, Blocks.POTATO_PORTAL, Blocks.END_PORTAL, Blocks.END_GATEWAY);
+        this.tag(BlockTags.PORTALS).add(Blocks.NETHER_PORTAL, Blocks.END_PORTAL, Blocks.END_GATEWAY);
         this.tag(BlockTags.FIRE).add(Blocks.FIRE, Blocks.SOUL_FIRE);
-        this.tag(BlockTags.NYLIUM).add(Blocks.CRIMSON_NYLIUM, Blocks.WARPED_NYLIUM, Blocks.CORRUPTED_PEELGRASS_BLOCK);
+        this.tag(BlockTags.NYLIUM).add(Blocks.CRIMSON_NYLIUM, Blocks.WARPED_NYLIUM);
         this.tag(BlockTags.WART_BLOCKS).add(Blocks.NETHER_WART_BLOCK, Blocks.WARPED_WART_BLOCK);
-        this.tag(BlockTags.BEACON_BASE_BLOCKS)
-            .add(Blocks.AMBER_BLOCK, Blocks.NETHERITE_BLOCK, Blocks.EMERALD_BLOCK, Blocks.DIAMOND_BLOCK, Blocks.GOLD_BLOCK, Blocks.IRON_BLOCK);
+        this.tag(BlockTags.BEACON_BASE_BLOCKS).add(Blocks.NETHERITE_BLOCK, Blocks.EMERALD_BLOCK, Blocks.DIAMOND_BLOCK, Blocks.GOLD_BLOCK, Blocks.IRON_BLOCK);
         this.tag(BlockTags.SOUL_SPEED_BLOCKS).add(Blocks.SOUL_SAND, Blocks.SOUL_SOIL);
         this.tag(BlockTags.WALL_POST_OVERRIDE)
             .add(Blocks.TORCH, Blocks.SOUL_TORCH, Blocks.REDSTONE_TORCH, Blocks.TRIPWIRE)
@@ -748,18 +_,17 @@
             .add(Blocks.SOUL_WALL_TORCH)
             .add(Blocks.SOUL_CAMPFIRE);
         this.tag(BlockTags.HOGLIN_REPELLENTS).add(Blocks.WARPED_FUNGUS).add(Blocks.POTTED_WARPED_FUNGUS).add(Blocks.NETHER_PORTAL).add(Blocks.RESPAWN_ANCHOR);
-        this.tag(BlockTags.GOLD_ORES).add(Blocks.GOLD_ORE, Blocks.NETHER_GOLD_ORE, Blocks.DEEPSLATE_GOLD_ORE, Blocks.POTONE_GOLD_ORE);
-        this.tag(BlockTags.IRON_ORES).add(Blocks.IRON_ORE, Blocks.DEEPSLATE_IRON_ORE, Blocks.POTONE_IRON_ORE);
-        this.tag(BlockTags.POISONOUS_POTATO_ORES).add(Blocks.POISONOUS_POTATO_ORE, Blocks.DEEPSLATE_POISONOUS_POTATO_ORE);
-        this.tag(BlockTags.DIAMOND_ORES).add(Blocks.DIAMOND_ORE, Blocks.DEEPSLATE_DIAMOND_ORE, Blocks.POTONE_DIAMOND_ORE);
-        this.tag(BlockTags.REDSTONE_ORES).add(Blocks.REDSTONE_ORE, Blocks.DEEPSLATE_REDSTONE_ORE, Blocks.POTONE_REDSTONE_ORE);
+        this.tag(BlockTags.GOLD_ORES).add(Blocks.GOLD_ORE, Blocks.NETHER_GOLD_ORE, Blocks.DEEPSLATE_GOLD_ORE);
+        this.tag(BlockTags.IRON_ORES).add(Blocks.IRON_ORE, Blocks.DEEPSLATE_IRON_ORE);
+        this.tag(BlockTags.DIAMOND_ORES).add(Blocks.DIAMOND_ORE, Blocks.DEEPSLATE_DIAMOND_ORE);
+        this.tag(BlockTags.REDSTONE_ORES).add(Blocks.REDSTONE_ORE, Blocks.DEEPSLATE_REDSTONE_ORE);
         this.tag(BlockTags.COAL_ORES).add(Blocks.COAL_ORE, Blocks.DEEPSLATE_COAL_ORE);
         this.tag(BlockTags.EMERALD_ORES).add(Blocks.EMERALD_ORE, Blocks.DEEPSLATE_EMERALD_ORE);
-        this.tag(BlockTags.COPPER_ORES).add(Blocks.COPPER_ORE, Blocks.DEEPSLATE_COPPER_ORE, Blocks.POTONE_COPPER_ORE);
-        this.tag(BlockTags.LAPIS_ORES).add(Blocks.LAPIS_ORE, Blocks.DEEPSLATE_LAPIS_ORE, Blocks.POTONE_LAPIS_ORE);
+        this.tag(BlockTags.COPPER_ORES).add(Blocks.COPPER_ORE, Blocks.DEEPSLATE_COPPER_ORE);
+        this.tag(BlockTags.LAPIS_ORES).add(Blocks.LAPIS_ORE, Blocks.DEEPSLATE_LAPIS_ORE);
         this.tag(BlockTags.SOUL_FIRE_BASE_BLOCKS).add(Blocks.SOUL_SAND, Blocks.SOUL_SOIL);
         this.tag(BlockTags.STRIDER_WARM_BLOCKS).add(Blocks.LAVA);
-        this.tag(BlockTags.CAMPFIRES).add(Blocks.CAMPFIRE, Blocks.SOUL_CAMPFIRE, Blocks.FRYING_TABLE);
+        this.tag(BlockTags.CAMPFIRES).add(Blocks.CAMPFIRE, Blocks.SOUL_CAMPFIRE);
         this.tag(BlockTags.GUARDED_BY_PIGLINS)
             .add(Blocks.GOLD_BLOCK, Blocks.BARREL, Blocks.CHEST, Blocks.ENDER_CHEST, Blocks.GILDED_BLACKSTONE, Blocks.TRAPPED_CHEST, Blocks.RAW_GOLD_BLOCK)
             .addTag(BlockTags.SHULKER_BOXES)
@@ -767,7 +_,6 @@
         this.tag(BlockTags.PREVENT_MOB_SPAWNING_INSIDE).addTag(BlockTags.RAILS);
         this.tag(BlockTags.FENCE_GATES)
             .add(
-                Blocks.POTATO_FENCE_GATE,
                 Blocks.ACACIA_FENCE_GATE,
                 Blocks.BIRCH_FENCE_GATE,
                 Blocks.DARK_OAK_FENCE_GATE,
@@ -788,7 +_,7 @@
         this.tag(BlockTags.STONE_ORE_REPLACEABLES).add(Blocks.STONE).add(Blocks.GRANITE).add(Blocks.DIORITE).add(Blocks.ANDESITE);
         this.tag(BlockTags.DEEPSLATE_ORE_REPLACEABLES).add(Blocks.DEEPSLATE).add(Blocks.TUFF);
         this.tag(BlockTags.BASE_STONE_OVERWORLD)
-            .add(Blocks.STONE, Blocks.POTONE)
+            .add(Blocks.STONE)
             .add(Blocks.GRANITE)
             .add(Blocks.DIORITE)
             .add(Blocks.ANDESITE)
@@ -813,8 +_,7 @@
                 Blocks.PACKED_ICE,
                 Blocks.RAW_IRON_BLOCK,
                 Blocks.RAW_COPPER_BLOCK
-            )
-            .addTag(BlockTags.POISONOUS_POTATO_ORES);
+            );
         this.tag(BlockTags.NETHER_CARVER_REPLACEABLES)
             .addTag(BlockTags.BASE_STONE_OVERWORLD)
             .addTag(BlockTags.BASE_STONE_NETHER)
@@ -868,11 +_,11 @@
             .add(Blocks.POWDER_SNOW, Blocks.SCULK_VEIN, Blocks.GLOW_LICHEN, Blocks.LILY_PAD, Blocks.SMALL_AMETHYST_BUD, Blocks.PINK_PETALS);
         this.tag(BlockTags.COMBINATION_STEP_SOUND_BLOCKS)
             .addTag(BlockTags.WOOL_CARPETS)
-            .add(Blocks.MOSS_CARPET, Blocks.SNOW, Blocks.POISONOUS_MASHED_POTATO, Blocks.NETHER_SPROUTS, Blocks.WARPED_ROOTS, Blocks.CRIMSON_ROOTS);
+            .add(Blocks.MOSS_CARPET, Blocks.SNOW, Blocks.NETHER_SPROUTS, Blocks.WARPED_ROOTS, Blocks.CRIMSON_ROOTS);
         this.tag(BlockTags.DRIPSTONE_REPLACEABLE).addTag(BlockTags.BASE_STONE_OVERWORLD);
         this.tag(BlockTags.CAVE_VINES).add(Blocks.CAVE_VINES_PLANT).add(Blocks.CAVE_VINES);
         this.tag(BlockTags.MOSS_REPLACEABLE).addTag(BlockTags.BASE_STONE_OVERWORLD).addTag(BlockTags.CAVE_VINES).addTag(BlockTags.DIRT);
-        this.tag(BlockTags.LUSH_GROUND_REPLACEABLE).addTag(BlockTags.MOSS_REPLACEABLE).add(Blocks.CLAY).add(Blocks.GRAVEL).add(Blocks.SAND, Blocks.GRAVTATER);
+        this.tag(BlockTags.LUSH_GROUND_REPLACEABLE).addTag(BlockTags.MOSS_REPLACEABLE).add(Blocks.CLAY).add(Blocks.GRAVEL).add(Blocks.SAND);
         this.tag(BlockTags.AZALEA_ROOT_REPLACEABLE)
             .addTag(BlockTags.BASE_STONE_OVERWORLD)
             .addTag(BlockTags.DIRT)
@@ -880,7 +_,7 @@
             .add(Blocks.RED_SAND)
             .add(Blocks.CLAY)
             .add(Blocks.GRAVEL)
-            .add(Blocks.SAND, Blocks.GRAVTATER)
+            .add(Blocks.SAND)
             .add(Blocks.SNOW_BLOCK)
             .add(Blocks.POWDER_SNOW);
         this.tag(BlockTags.SMALL_DRIPLEAF_PLACEABLE).add(Blocks.CLAY).add(Blocks.MOSS_BLOCK);
@@ -888,36 +_,6 @@
         this.tag(BlockTags.OCCLUDES_VIBRATION_SIGNALS).addTag(BlockTags.WOOL);
         this.tag(BlockTags.DAMPENS_VIBRATIONS).addTag(BlockTags.WOOL).addTag(BlockTags.WOOL_CARPETS);
         this.tag(BlockTags.SNOW).add(Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.POWDER_SNOW);
-        this.tag(BlockTags.POTONE_MINEABLE)
-            .add(
-                Blocks.RESIN_ORE,
-                Blocks.POTONE_DIAMOND_ORE,
-                Blocks.POTONE_REDSTONE_ORE,
-                Blocks.POTONE_COPPER_ORE,
-                Blocks.POTONE_IRON_ORE,
-                Blocks.POTONE_GOLD_ORE,
-                Blocks.POTONE_LAPIS_ORE,
-                Blocks.POTONE,
-                Blocks.POTONE_SLAB,
-                Blocks.POTONE_STAIRS,
-                Blocks.POTONE_WALL,
-                Blocks.TATERSTONE,
-                Blocks.TATERSTONE_SLAB,
-                Blocks.TATERSTONE_STAIRS,
-                Blocks.TATERSTONE_WALL,
-                Blocks.EXPIRED_BAKED_POTATO_BRICKS,
-                Blocks.EXPIRED_BAKED_POTATO_BRICK_SLAB,
-                Blocks.EXPIRED_BAKED_POTATO_BRICK_STAIRS,
-                Blocks.EXPIRED_BAKED_POTATO_BRICK_WALL,
-                Blocks.CHARRED_BAKED_POTATO_BRICKS,
-                Blocks.CHARRED_BAKED_POTATO_BRICK_SLAB,
-                Blocks.CHARRED_BAKED_POTATO_BRICK_STAIRS,
-                Blocks.CHARRED_BAKED_POTATO_BRICK_WALL,
-                Blocks.BAKED_POTATO_BRICKS,
-                Blocks.BAKED_POTATO_BRICK_SLAB,
-                Blocks.BAKED_POTATO_BRICK_STAIRS,
-                Blocks.BAKED_POTATO_BRICK_WALL
-            );
         this.tag(BlockTags.MINEABLE_WITH_AXE)
             .add(
                 Blocks.NOTE_BLOCK,
@@ -1004,10 +_,8 @@
             .addTag(BlockTags.ALL_HANGING_SIGNS)
             .add(Blocks.BAMBOO_MOSAIC, Blocks.BAMBOO_MOSAIC_SLAB, Blocks.BAMBOO_MOSAIC_STAIRS)
             .addTag(BlockTags.BAMBOO_BLOCKS)
-            .add(Blocks.CHISELED_BOOKSHELF)
-            .add(Blocks.BIG_BRAIN);
+            .add(Blocks.CHISELED_BOOKSHELF);
         this.tag(BlockTags.MINEABLE_WITH_HOE)
-            .addTag(BlockTags.POTONE_MINEABLE)
             .add(
                 Blocks.NETHER_WART_BLOCK,
                 Blocks.WARPED_WART_BLOCK,
@@ -1035,12 +_,10 @@
                 Blocks.SCULK_VEIN,
                 Blocks.SCULK_SHRIEKER,
                 Blocks.PINK_PETALS,
-                Blocks.CHERRY_LEAVES,
-                Blocks.FLOATATO
+                Blocks.CHERRY_LEAVES
             );
         this.tag(BlockTags.MINEABLE_WITH_PICKAXE)
             .add(
-                Blocks.AMBER_BLOCK,
                 Blocks.STONE,
                 Blocks.GRANITE,
                 Blocks.POLISHED_GRANITE,
@@ -1370,34 +_,18 @@
             .add(Blocks.MUD_BRICKS)
             .add(Blocks.MUD_BRICK_STAIRS)
             .add(Blocks.MUD_BRICK_SLAB)
-            .add(Blocks.PACKED_MUD)
-            .add(
-                Blocks.POISONOUS_POTATO_CUTTER,
-                Blocks.FLOATATER,
-                Blocks.POISONOUS_POTATO_BLOCK,
-                Blocks.COMPRESSED_POISONOUS_POTATO_BLOCK,
-                Blocks.DOUBLE_COMPRESSED_POISONOUS_POTATO_BLOCK,
-                Blocks.TRIPLE_COMPRESSED_POISONOUS_POTATO_BLOCK,
-                Blocks.QUADRUPLE_COMPRESSED_POISONOUS_POTATO_BLOCK
-            )
-            .addTag(BlockTags.POISONOUS_POTATO_ORES);
+            .add(Blocks.PACKED_MUD);
         this.tag(BlockTags.MINEABLE_WITH_SHOVEL)
             .add(
                 Blocks.CLAY,
                 Blocks.DIRT,
-                Blocks.POISON_FARMLAND,
-                Blocks.POISON_PATH,
-                Blocks.TERREDEPOMME,
                 Blocks.COARSE_DIRT,
                 Blocks.PODZOL,
                 Blocks.FARMLAND,
                 Blocks.GRASS_BLOCK,
-                Blocks.PEELGRASS_BLOCK,
-                Blocks.CORRUPTED_PEELGRASS_BLOCK,
                 Blocks.GRAVEL,
                 Blocks.MYCELIUM,
                 Blocks.SAND,
-                Blocks.GRAVTATER,
                 Blocks.RED_SAND,
                 Blocks.SNOW_BLOCK,
                 Blocks.SNOW,
@@ -1408,8 +_,7 @@
                 Blocks.MUDDY_MANGROVE_ROOTS,
                 Blocks.MUD,
                 Blocks.SUSPICIOUS_SAND,
-                Blocks.SUSPICIOUS_GRAVEL,
-                Blocks.POISONOUS_MASHED_POTATO
+                Blocks.SUSPICIOUS_GRAVEL
             )
             .addTag(BlockTags.CONCRETE_POWDER);
         this.tag(BlockTags.SWORD_EFFICIENT)
@@ -1418,11 +_,6 @@
             .addTag(BlockTags.SMALL_FLOWERS)
             .addTag(BlockTags.CROPS)
             .add(
-                Blocks.POTATO_BUD,
-                Blocks.POTATO_STEM,
-                Blocks.POTATO_LEAVES,
-                Blocks.POTATO_FRUIT,
-                Blocks.POTATO_SPROUTS,
                 Blocks.SHORT_GRASS,
                 Blocks.FERN,
                 Blocks.DEAD_BUSH,
@@ -1478,10 +_,8 @@
         this.tag(BlockTags.NEEDS_IRON_TOOL)
             .add(
                 Blocks.DIAMOND_BLOCK,
-                Blocks.AMBER_BLOCK,
                 Blocks.DIAMOND_ORE,
                 Blocks.DEEPSLATE_DIAMOND_ORE,
-                Blocks.POTONE_DIAMOND_ORE,
                 Blocks.EMERALD_ORE,
                 Blocks.DEEPSLATE_EMERALD_ORE,
                 Blocks.EMERALD_BLOCK,
@@ -1489,10 +_,8 @@
                 Blocks.RAW_GOLD_BLOCK,
                 Blocks.GOLD_ORE,
                 Blocks.DEEPSLATE_GOLD_ORE,
-                Blocks.POTONE_GOLD_ORE,
                 Blocks.REDSTONE_ORE,
-                Blocks.DEEPSLATE_REDSTONE_ORE,
-                Blocks.POTONE_REDSTONE_ORE
+                Blocks.DEEPSLATE_REDSTONE_ORE
             );
         this.tag(BlockTags.NEEDS_STONE_TOOL)
             .add(
@@ -1500,16 +_,13 @@
                 Blocks.RAW_IRON_BLOCK,
                 Blocks.IRON_ORE,
                 Blocks.DEEPSLATE_IRON_ORE,
-                Blocks.POTONE_IRON_ORE,
                 Blocks.LAPIS_BLOCK,
                 Blocks.LAPIS_ORE,
-                Blocks.POTONE_LAPIS_ORE,
                 Blocks.DEEPSLATE_LAPIS_ORE,
                 Blocks.COPPER_BLOCK,
                 Blocks.RAW_COPPER_BLOCK,
                 Blocks.COPPER_ORE,
                 Blocks.DEEPSLATE_COPPER_ORE,
-                Blocks.POTONE_COPPER_ORE,
                 Blocks.CUT_COPPER_SLAB,
                 Blocks.CUT_COPPER_STAIRS,
                 Blocks.CUT_COPPER,
@@ -1542,8 +_,7 @@
                 Blocks.WAXED_OXIDIZED_CUT_COPPER_STAIRS,
                 Blocks.WAXED_OXIDIZED_CUT_COPPER,
                 Blocks.LIGHTNING_ROD
-            )
-            .addTag(BlockTags.POISONOUS_POTATO_ORES);
+            );
         this.tag(BlockTags.INCORRECT_FOR_NETHERITE_TOOL);
         this.tag(BlockTags.INCORRECT_FOR_DIAMOND_TOOL);
         this.tag(BlockTags.INCORRECT_FOR_IRON_TOOL).addTag(BlockTags.NEEDS_DIAMOND_TOOL);
@@ -1553,33 +_,22 @@
         this.tag(BlockTags.FEATURES_CANNOT_REPLACE).add(Blocks.BEDROCK, Blocks.SPAWNER, Blocks.CHEST, Blocks.END_PORTAL_FRAME, Blocks.REINFORCED_DEEPSLATE);
         this.tag(BlockTags.LAVA_POOL_STONE_CANNOT_REPLACE).addTag(BlockTags.FEATURES_CANNOT_REPLACE).addTag(BlockTags.LEAVES).addTag(BlockTags.LOGS);
         this.tag(BlockTags.GEODE_INVALID_BLOCKS).add(Blocks.BEDROCK, Blocks.WATER, Blocks.LAVA, Blocks.ICE, Blocks.PACKED_ICE, Blocks.BLUE_ICE);
-        this.tag(BlockTags.ANIMALS_SPAWNABLE_ON).add(Blocks.GRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK);
+        this.tag(BlockTags.ANIMALS_SPAWNABLE_ON).add(Blocks.GRASS_BLOCK);
         this.tag(BlockTags.ARMADILLO_SPAWNABLE_ON)
             .addTag(BlockTags.ANIMALS_SPAWNABLE_ON)
             .addTag(BlockTags.BADLANDS_TERRACOTTA)
-            .add(Blocks.RED_SAND, Blocks.COARSE_DIRT, Blocks.GRAVTATER);
+            .add(Blocks.RED_SAND, Blocks.COARSE_DIRT);
         this.tag(BlockTags.AXOLOTLS_SPAWNABLE_ON).add(Blocks.CLAY);
         this.tag(BlockTags.GOATS_SPAWNABLE_ON)
             .addTag(BlockTags.ANIMALS_SPAWNABLE_ON)
-            .add(Blocks.STONE, Blocks.POTONE, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.PACKED_ICE, Blocks.GRAVEL);
+            .add(Blocks.STONE, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.PACKED_ICE, Blocks.GRAVEL);
         this.tag(BlockTags.MOOSHROOMS_SPAWNABLE_ON).add(Blocks.MYCELIUM);
-        this.tag(BlockTags.PARROTS_SPAWNABLE_ON)
-            .add(Blocks.GRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.AIR)
-            .addTag(BlockTags.LEAVES)
-            .addTag(BlockTags.LOGS);
+        this.tag(BlockTags.PARROTS_SPAWNABLE_ON).add(Blocks.GRASS_BLOCK, Blocks.AIR).addTag(BlockTags.LEAVES).addTag(BlockTags.LOGS);
         this.tag(BlockTags.POLAR_BEARS_SPAWNABLE_ON_ALTERNATE).add(Blocks.ICE);
-        this.tag(BlockTags.RABBITS_SPAWNABLE_ON)
-            .add(Blocks.GRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.SAND, Blocks.GRAVTATER);
-        this.tag(BlockTags.FOXES_SPAWNABLE_ON)
-            .add(
-                Blocks.GRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.PODZOL, Blocks.COARSE_DIRT
-            );
-        this.tag(BlockTags.WOLVES_SPAWNABLE_ON)
-            .add(
-                Blocks.GRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.COARSE_DIRT, Blocks.PODZOL
-            );
-        this.tag(BlockTags.FROGS_SPAWNABLE_ON)
-            .add(Blocks.GRASS_BLOCK, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.MUD, Blocks.MANGROVE_ROOTS, Blocks.MUDDY_MANGROVE_ROOTS);
+        this.tag(BlockTags.RABBITS_SPAWNABLE_ON).add(Blocks.GRASS_BLOCK, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.SAND);
+        this.tag(BlockTags.FOXES_SPAWNABLE_ON).add(Blocks.GRASS_BLOCK, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.PODZOL, Blocks.COARSE_DIRT);
+        this.tag(BlockTags.WOLVES_SPAWNABLE_ON).add(Blocks.GRASS_BLOCK, Blocks.SNOW, Blocks.SNOW_BLOCK, Blocks.COARSE_DIRT, Blocks.PODZOL);
+        this.tag(BlockTags.FROGS_SPAWNABLE_ON).add(Blocks.GRASS_BLOCK, Blocks.MUD, Blocks.MANGROVE_ROOTS, Blocks.MUDDY_MANGROVE_ROOTS);
         this.tag(BlockTags.TERRACOTTA)
             .add(
                 Blocks.TERRACOTTA,
@@ -1642,7 +_,7 @@
             .addTag(BlockTags.TERRACOTTA)
             .addTag(BlockTags.NYLIUM)
             .addTag(BlockTags.BASE_STONE_NETHER)
-            .add(Blocks.SAND, Blocks.GRAVTATER, Blocks.RED_SAND)
+            .add(Blocks.SAND, Blocks.RED_SAND)
             .add(Blocks.GRAVEL)
             .add(Blocks.SOUL_SAND)
             .add(Blocks.SOUL_SOIL)
@@ -1662,7 +_,7 @@
             .add(Blocks.CRACKED_DEEPSLATE_TILES)
             .add(Blocks.POLISHED_DEEPSLATE);
         this.tag(BlockTags.VIBRATION_RESONATORS).add(Blocks.AMETHYST_BLOCK);
-        this.tag(BlockTags.CONVERTABLE_TO_MUD).add(Blocks.DIRT, Blocks.TERREDEPOMME, Blocks.COARSE_DIRT, Blocks.ROOTED_DIRT);
+        this.tag(BlockTags.CONVERTABLE_TO_MUD).add(Blocks.DIRT, Blocks.COARSE_DIRT, Blocks.ROOTED_DIRT);
         this.tag(BlockTags.ANCIENT_CITY_REPLACEABLE)
             .add(Blocks.DEEPSLATE)
             .add(Blocks.DEEPSLATE_BRICKS)
@@ -1714,23 +_,19 @@
         this.tag(BlockTags.COMPLETES_FIND_TREE_TUTORIAL).addTag(BlockTags.LOGS).addTag(BlockTags.LEAVES).addTag(BlockTags.WART_BLOCKS);
         this.tag(BlockTags.SNAPS_GOAT_HORN)
             .addTag(BlockTags.OVERWORLD_NATURAL_LOGS)
-            .add(Blocks.STONE, Blocks.POTONE)
+            .add(Blocks.STONE)
             .add(Blocks.PACKED_ICE)
             .add(Blocks.IRON_ORE)
             .add(Blocks.COAL_ORE)
             .add(Blocks.COPPER_ORE)
-            .add(Blocks.EMERALD_ORE)
-            .add(Blocks.POISONOUS_POTATO_ORE);
+            .add(Blocks.EMERALD_ORE);
         this.tag(BlockTags.INVALID_SPAWN_INSIDE).add(Blocks.END_PORTAL, Blocks.END_GATEWAY);
         this.tag(BlockTags.SAND).add(Blocks.SUSPICIOUS_SAND);
         this.tag(BlockTags.TRAIL_RUINS_REPLACEABLE).add(Blocks.GRAVEL);
         this.tag(BlockTags.SNIFFER_DIGGABLE_BLOCK)
             .add(
                 Blocks.DIRT,
-                Blocks.TERREDEPOMME,
                 Blocks.GRASS_BLOCK,
-                Blocks.PEELGRASS_BLOCK,
-                Blocks.CORRUPTED_PEELGRASS_BLOCK,
                 Blocks.PODZOL,
                 Blocks.COARSE_DIRT,
                 Blocks.ROOTED_DIRT,
@@ -1762,8 +_,5 @@
             );
         this.tag(BlockTags.CAMEL_SAND_STEP_SOUND_BLOCKS).addTag(BlockTags.SAND).addTag(BlockTags.CONCRETE_POWDER);
         this.tag(BlockTags.DOES_NOT_BLOCK_HOPPERS).addTag(BlockTags.BEEHIVES);
-        IntrinsicHolderTagsProvider.IntrinsicTagAppender<Block> intrinsictagappender1 = this.tag(BlockTags.POTATOSTONE_BASE);
-        intrinsictagappender1.add(Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK, Blocks.GRAVTATER, Blocks.TERREDEPOMME);
-        Blocks.POTATO_PEELS_BLOCK_MAP.forEach((p_339744_, p_339745_) -> intrinsictagappender1.add(p_339745_));
     }
 }
