--- a/net/minecraft/util/datafix/schemas/V3818_3.java
+++ b/net/minecraft/util/datafix/schemas/V3818_3.java
@@ -6,7 +_,6 @@
 import com.mojang.datafixers.util.Pair;
 import java.util.Map;
 import java.util.function.Supplier;
-import net.minecraft.util.datafix.ExtraDataFixUtils;
 import net.minecraft.util.datafix.fixes.References;
 
 public class V3818_3 extends NamespacedSchema {
@@ -20,7 +_,7 @@
         p_331415_.registerType(
             true,
             References.DATA_COMPONENTS,
-            () -> ExtraDataFixUtils.optionalFields(
+            () -> DSL.optionalFields(
                     Pair.of("minecraft:bees", DSL.list(DSL.optionalFields("entity_data", References.ENTITY_TREE.in(p_331415_)))),
                     Pair.of("minecraft:block_entity_data", References.BLOCK_ENTITY.in(p_331415_)),
                     Pair.of("minecraft:bundle_contents", DSL.list(References.ITEM_STACK.in(p_331415_))),
