--- a/net/minecraft/util/datafix/schemas/V704.java
+++ b/net/minecraft/util/datafix/schemas/V704.java
@@ -15,7 +_,6 @@
 import java.util.Map;
 import java.util.Objects;
 import java.util.function.Supplier;
-import net.minecraft.util.datafix.ExtraDataFixUtils;
 import net.minecraft.util.datafix.fixes.References;
 
 public class V704 extends Schema {
@@ -196,7 +_,13 @@
     @Override
     public void registerTypes(Schema p_18065_, Map<String, Supplier<TypeTemplate>> p_18066_, Map<String, Supplier<TypeTemplate>> p_18067_) {
         super.registerTypes(p_18065_, p_18066_, p_18067_);
-        p_18065_.registerType(false, References.BLOCK_ENTITY, () -> DSL.taggedChoiceLazy("id", NamespacedSchema.namespacedString(), p_18067_));
+        p_18065_.registerType(
+            true,
+            References.BLOCK_ENTITY,
+            () -> DSL.optionalFields(
+                    "components", References.DATA_COMPONENTS.in(p_18065_), DSL.taggedChoiceLazy("id", NamespacedSchema.namespacedString(), p_18067_)
+                )
+        );
         p_18065_.registerType(
             true,
             References.ITEM_STACK,
@@ -205,7 +_,7 @@
                         "id",
                         References.ITEM_NAME.in(p_18065_),
                         "tag",
-                        ExtraDataFixUtils.optionalFields(
+                        DSL.optionalFields(
                             Pair.of("EntityTag", References.ENTITY_TREE.in(p_18065_)),
                             Pair.of("BlockEntityTag", References.BLOCK_ENTITY.in(p_18065_)),
                             Pair.of("CanDestroy", DSL.list(References.BLOCK_NAME.in(p_18065_))),
