--- a/net/minecraft/util/datafix/schemas/V705.java
+++ b/net/minecraft/util/datafix/schemas/V705.java
@@ -11,7 +_,6 @@
 import com.mojang.serialization.DynamicOps;
 import java.util.Map;
 import java.util.function.Supplier;
-import net.minecraft.util.datafix.ExtraDataFixUtils;
 import net.minecraft.util.datafix.fixes.References;
 
 public class V705 extends NamespacedSchema {
@@ -98,6 +_,31 @@
         .put("minecraft:zombie_horse_spawn_egg", "minecraft:zombie_horse")
         .put("minecraft:zombie_villager_spawn_egg", "minecraft:zombie_villager")
         .put("minecraft:zombified_piglin_spawn_egg", "minecraft:zombified_piglin")
+        .put("minecraft:item_frame", "minecraft:item_frame")
+        .put("minecraft:boat", "minecraft:boat")
+        .put("minecraft:oak_boat", "minecraft:boat")
+        .put("minecraft:oak_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:spruce_boat", "minecraft:boat")
+        .put("minecraft:spruce_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:birch_boat", "minecraft:boat")
+        .put("minecraft:birch_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:jungle_boat", "minecraft:boat")
+        .put("minecraft:jungle_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:acacia_boat", "minecraft:boat")
+        .put("minecraft:acacia_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:cherry_boat", "minecraft:boat")
+        .put("minecraft:cherry_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:dark_oak_boat", "minecraft:boat")
+        .put("minecraft:dark_oak_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:mangrove_boat", "minecraft:boat")
+        .put("minecraft:mangrove_chest_boat", "minecraft:chest_boat")
+        .put("minecraft:bamboo_raft", "minecraft:boat")
+        .put("minecraft:bamboo_chest_raft", "minecraft:chest_boat")
+        .put("minecraft:minecart", "minecraft:minecart")
+        .put("minecraft:chest_minecart", "minecraft:chest_minecart")
+        .put("minecraft:furnace_minecart", "minecraft:furnace_minecart")
+        .put("minecraft:tnt_minecart", "minecraft:tnt_minecart")
+        .put("minecraft:hopper_minecart", "minecraft:hopper_minecart")
         .build();
     protected static final HookFunction ADD_NAMES = new HookFunction() {
         @Override
@@ -293,7 +_,7 @@
                         "id",
                         References.ITEM_NAME.in(p_18150_),
                         "tag",
-                        ExtraDataFixUtils.optionalFields(
+                        DSL.optionalFields(
                             Pair.of("EntityTag", References.ENTITY_TREE.in(p_18150_)),
                             Pair.of("BlockEntityTag", References.BLOCK_ENTITY.in(p_18150_)),
                             Pair.of("CanDestroy", DSL.list(References.BLOCK_NAME.in(p_18150_))),
