--- a/net/minecraft/server/packs/resources/FallbackResourceManager.java
+++ b/net/minecraft/server/packs/resources/FallbackResourceManager.java
@@ -62,7 +_,7 @@
 
     @Override
     public Optional<Resource> getResource(ResourceLocation p_215419_) {
-        for(int i = this.fallbacks.size() - 1; i >= 0; --i) {
+        for (int i = this.fallbacks.size() - 1; i >= 0; i--) {
             FallbackResourceManager.PackEntry fallbackresourcemanager$packentry = this.fallbacks.get(i);
             PackResources packresources = fallbackresourcemanager$packentry.resources;
             if (packresources != null) {
@@ -101,7 +_,7 @@
         boolean flag = false;
         String s = null;
 
-        for(int i = this.fallbacks.size() - 1; i >= 0; --i) {
+        for (int i = this.fallbacks.size() - 1; i >= 0; i--) {
             FallbackResourceManager.PackEntry fallbackresourcemanager$packentry = this.fallbacks.get(i);
             PackResources packresources = fallbackresourcemanager$packentry.resources;
             if (packresources != null) {
@@ -160,7 +_,7 @@
         Map<ResourceLocation, ResourceWithSourceAndIndex> map1 = new HashMap<>();
         int i = this.fallbacks.size();
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             FallbackResourceManager.PackEntry fallbackresourcemanager$packentry = this.fallbacks.get(j);
             fallbackresourcemanager$packentry.filterAll(map.keySet());
             fallbackresourcemanager$packentry.filterAll(map1.keySet());
@@ -202,7 +_,7 @@
         return () -> {
             ResourceLocation resourcelocation = getMetadataLocation(p_215369_);
 
-            for(int i = this.fallbacks.size() - 1; i >= p_215370_; --i) {
+            for (int i = this.fallbacks.size() - 1; i >= p_215370_; i--) {
                 FallbackResourceManager.PackEntry fallbackresourcemanager$packentry = this.fallbacks.get(i);
                 PackResources packresources = fallbackresourcemanager$packentry.resources;
                 if (packresources != null) {
@@ -237,7 +_,7 @@
     private static void applyPackFiltersToExistingResources(
         FallbackResourceManager.PackEntry p_215393_, Map<ResourceLocation, FallbackResourceManager.EntryStack> p_215394_
     ) {
-        for(FallbackResourceManager.EntryStack fallbackresourcemanager$entrystack : p_215394_.values()) {
+        for (FallbackResourceManager.EntryStack fallbackresourcemanager$entrystack : p_215394_.values()) {
             if (p_215393_.isFiltered(fallbackresourcemanager$entrystack.fileLocation)) {
                 fallbackresourcemanager$entrystack.fileSources.clear();
             } else if (p_215393_.isFiltered(fallbackresourcemanager$entrystack.metadataLocation())) {
@@ -284,18 +_,18 @@
     public Map<ResourceLocation, List<Resource>> listResourceStacks(String p_215416_, Predicate<ResourceLocation> p_215417_) {
         Map<ResourceLocation, FallbackResourceManager.EntryStack> map = Maps.newHashMap();
 
-        for(FallbackResourceManager.PackEntry fallbackresourcemanager$packentry : this.fallbacks) {
+        for (FallbackResourceManager.PackEntry fallbackresourcemanager$packentry : this.fallbacks) {
             applyPackFiltersToExistingResources(fallbackresourcemanager$packentry, map);
             this.listPackResources(fallbackresourcemanager$packentry, p_215416_, p_215417_, map);
         }
 
         TreeMap<ResourceLocation, List<Resource>> treemap = Maps.newTreeMap();
 
-        for(FallbackResourceManager.EntryStack fallbackresourcemanager$entrystack : map.values()) {
+        for (FallbackResourceManager.EntryStack fallbackresourcemanager$entrystack : map.values()) {
             if (!fallbackresourcemanager$entrystack.fileSources.isEmpty()) {
                 List<Resource> list = new ArrayList<>();
 
-                for(FallbackResourceManager.ResourceWithSource fallbackresourcemanager$resourcewithsource : fallbackresourcemanager$entrystack.fileSources) {
+                for (FallbackResourceManager.ResourceWithSource fallbackresourcemanager$resourcewithsource : fallbackresourcemanager$entrystack.fileSources) {
                     PackResources packresources = fallbackresourcemanager$resourcewithsource.source;
                     IoSupplier<InputStream> iosupplier = fallbackresourcemanager$entrystack.metaSources.get(packresources);
                     IoSupplier<ResourceMetadata> iosupplier1 = iosupplier != null ? convertToMetadata(iosupplier) : ResourceMetadata.EMPTY_SUPPLIER;
