--- a/net/minecraft/server/packs/resources/ProfiledReloadInstance.java
+++ b/net/minecraft/server/packs/resources/ProfiledReloadInstance.java
@@ -62,7 +_,7 @@
         long i = 0L;
         LOGGER.info("Resource reload finished after {} ms", this.total.elapsed(TimeUnit.MILLISECONDS));
 
-        for(ProfiledReloadInstance.State profiledreloadinstance$state : p_215484_) {
+        for (ProfiledReloadInstance.State profiledreloadinstance$state : p_215484_) {
             ProfileResults profileresults = profiledreloadinstance$state.preparationResult;
             ProfileResults profileresults1 = profiledreloadinstance$state.reloadResult;
             long j = TimeUnit.NANOSECONDS.toMillis(profiledreloadinstance$state.preparationNanos.get());
