--- a/net/minecraft/server/packs/resources/SimpleReloadInstance.java
+++ b/net/minecraft/server/packs/resources/SimpleReloadInstance.java
@@ -5,6 +_,7 @@
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
 import java.util.concurrent.Executor;
 import java.util.concurrent.atomic.AtomicInteger;
 import net.minecraft.Util;
@@ -54,32 +_,39 @@
         CompletableFuture<?> completablefuture = p_10813_;
         this.preparingListeners = Sets.newHashSet(p_10811_);
 
-        for(final PreparableReloadListener preparablereloadlistener : p_10811_) {
+        for (final PreparableReloadListener preparablereloadlistener : p_10811_) {
             final CompletableFuture<?> completablefuture1 = completablefuture;
-            CompletableFuture<S> completablefuture2 = p_10812_.create(new PreparableReloadListener.PreparationBarrier() {
-                @Override
-                public <T> CompletableFuture<T> wait(T p_10858_) {
+            CompletableFuture<S> completablefuture2 = p_10812_.create(
+                new PreparableReloadListener.PreparationBarrier() {
+                    @Override
+                    public <T> CompletableFuture<T> wait(T p_10858_) {
+                        p_10809_.execute(() -> {
+                            SimpleReloadInstance.this.preparingListeners.remove(preparablereloadlistener);
+                            if (SimpleReloadInstance.this.preparingListeners.isEmpty()) {
+                                SimpleReloadInstance.this.allPreparations.complete(Unit.INSTANCE);
+                            }
+                        });
+                        return SimpleReloadInstance.this.allPreparations
+                            .thenCombine((CompletionStage<? extends T>)completablefuture1, (p_10861_, p_10862_) -> p_10858_);
+                    }
+                },
+                p_10810_,
+                preparablereloadlistener,
+                p_10842_ -> {
+                    this.startedTaskCounter.incrementAndGet();
+                    p_10808_.execute(() -> {
+                        p_10842_.run();
+                        this.doneTaskCounter.incrementAndGet();
+                    });
+                },
+                p_10836_ -> {
+                    this.startedReloads++;
                     p_10809_.execute(() -> {
-                        SimpleReloadInstance.this.preparingListeners.remove(preparablereloadlistener);
-                        if (SimpleReloadInstance.this.preparingListeners.isEmpty()) {
-                            SimpleReloadInstance.this.allPreparations.complete(Unit.INSTANCE);
-                        }
+                        p_10836_.run();
+                        this.finishedReloads++;
                     });
-                    return SimpleReloadInstance.this.allPreparations.thenCombine(completablefuture1, (p_10861_, p_10862_) -> p_10858_);
                 }
-            }, p_10810_, preparablereloadlistener, p_10842_ -> {
-                this.startedTaskCounter.incrementAndGet();
-                p_10808_.execute(() -> {
-                    p_10842_.run();
-                    this.doneTaskCounter.incrementAndGet();
-                });
-            }, p_10836_ -> {
-                ++this.startedReloads;
-                p_10809_.execute(() -> {
-                    p_10836_.run();
-                    ++this.finishedReloads;
-                });
-            });
+            );
             list.add(completablefuture2);
             completablefuture = completablefuture2;
         }
