--- a/net/minecraft/util/ArrayListDeque.java
+++ b/net/minecraft/util/ArrayListDeque.java
@@ -1,19 +_,17 @@
 package net.minecraft.util;
 
 import com.google.common.annotations.VisibleForTesting;
-import java.io.Serializable;
 import java.util.AbstractList;
-import java.util.Deque;
 import java.util.Iterator;
+import java.util.List;
 import java.util.NoSuchElementException;
 import java.util.Objects;
-import java.util.RandomAccess;
 import java.util.function.Consumer;
 import java.util.function.Predicate;
 import java.util.function.UnaryOperator;
 import javax.annotation.Nullable;
 
-public class ArrayListDeque<T> extends AbstractList<T> implements Serializable, Cloneable, Deque<T>, RandomAccess {
+public class ArrayListDeque<T> extends AbstractList<T> implements ListAndDeque<T> {
     private static final int MIN_GROWTH = 1;
     private Object[] contents;
     private int head;
@@ -85,22 +_,22 @@
         if (p_294490_ == this.size) {
             this.contents[i] = p_294693_;
         } else if (p_294490_ == 0) {
-            --this.head;
+            this.head--;
             if (this.head < 0) {
-                this.head += this.contents.length;
+                this.head = this.head + this.contents.length;
             }
 
             this.contents[this.getIndex(0)] = p_294693_;
         } else {
-            for(int j = this.size - 1; j >= p_294490_; --j) {
+            for (int j = this.size - 1; j >= p_294490_; j--) {
                 this.contents[this.getIndex(j + 1)] = this.contents[this.getIndex(j)];
             }
 
             this.contents[i] = p_294693_;
         }
 
-        ++this.modCount;
-        ++this.size;
+        this.modCount++;
+        this.size++;
     }
 
     private void grow() {
@@ -118,19 +_,19 @@
         T t = this.getInner(i);
         if (p_295380_ == 0) {
             this.contents[i] = null;
-            ++this.head;
+            this.head++;
         } else if (p_295380_ == this.size - 1) {
             this.contents[i] = null;
         } else {
-            for(int j = p_295380_ + 1; j < this.size; ++j) {
+            for (int j = p_295380_ + 1; j < this.size; j++) {
                 this.contents[this.getIndex(j - 1)] = this.get(j);
             }
 
             this.contents[this.getIndex(this.size - 1)] = null;
         }
 
-        ++this.modCount;
-        --this.size;
+        this.modCount++;
+        this.size--;
         return t;
     }
 
@@ -138,10 +_,10 @@
     public boolean removeIf(Predicate<? super T> p_296232_) {
         int i = 0;
 
-        for(int j = 0; j < this.size; ++j) {
+        for (int j = 0; j < this.size; j++) {
             T t = this.get(j);
             if (p_296232_.test(t)) {
-                ++i;
+                i++;
             } else if (i != 0) {
                 this.contents[this.getIndex(j - i)] = t;
                 this.contents[this.getIndex(j)] = null;
@@ -154,14 +_,14 @@
     }
 
     private void copyCount(Object[] p_294388_, int p_294959_) {
-        for(int i = 0; i < p_294959_; ++i) {
+        for (int i = 0; i < p_294959_; i++) {
             p_294388_[i] = this.get(i);
         }
     }
 
     @Override
     public void replaceAll(UnaryOperator<T> p_295123_) {
-        for(int i = 0; i < this.size; ++i) {
+        for (int i = 0; i < this.size; i++) {
             int j = this.getIndex(i);
             this.contents[j] = Objects.requireNonNull(p_295123_.apply(this.getInner(i)));
         }
@@ -169,7 +_,7 @@
 
     @Override
     public void forEach(Consumer<? super T> p_296263_) {
-        for(int i = 0; i < this.size; ++i) {
+        for (int i = 0; i < this.size; i++) {
             p_296263_.accept(this.get(i));
         }
     }
@@ -214,6 +_,11 @@
         }
     }
 
+    @Override
+    public ListAndDeque<T> reversed() {
+        return new ArrayListDeque.ReversedView(this);
+    }
+
     @Nullable
     @Override
     public T pollFirst() {
@@ -258,7 +_,7 @@
 
     @Override
     public boolean removeFirstOccurrence(Object p_294109_) {
-        for(int i = 0; i < this.size; ++i) {
+        for (int i = 0; i < this.size; i++) {
             T t = this.get(i);
             if (Objects.equals(p_294109_, t)) {
                 this.remove(i);
@@ -271,7 +_,7 @@
 
     @Override
     public boolean removeLastOccurrence(Object p_295642_) {
-        for(int i = this.size - 1; i >= 0; --i) {
+        for (int i = this.size - 1; i >= 0; i--) {
             T t = this.get(i);
             if (Objects.equals(p_295642_, t)) {
                 this.remove(i);
@@ -283,51 +_,10 @@
     }
 
     @Override
-    public boolean offer(T p_294969_) {
-        return this.offerLast(p_294969_);
-    }
-
-    @Override
-    public T remove() {
-        return this.removeFirst();
-    }
-
-    @Nullable
-    @Override
-    public T poll() {
-        return this.pollFirst();
-    }
-
-    @Override
-    public T element() {
-        return this.getFirst();
-    }
-
-    @Nullable
-    @Override
-    public T peek() {
-        return this.peekFirst();
-    }
-
-    @Override
-    public void push(T p_294939_) {
-        this.addFirst(p_294939_);
-    }
-
-    @Override
-    public T pop() {
-        return this.removeFirst();
-    }
-
-    @Override
     public Iterator<T> descendingIterator() {
         return new ArrayListDeque.DescendingIterator();
     }
 
-    public ArrayListDeque<T> reversed() {
-        throw new UnsupportedOperationException("Method was added in java 21 and is required to be overridden for classes implementing both List and Deque");
-    }
-
     class DescendingIterator implements Iterator<T> {
         private int index = ArrayListDeque.this.size() - 1;
 
@@ -347,6 +_,158 @@
         @Override
         public void remove() {
             ArrayListDeque.this.remove(this.index + 1);
+        }
+    }
+
+    class ReversedView extends AbstractList<T> implements ListAndDeque<T> {
+        private final ArrayListDeque<T> source;
+
+        public ReversedView(ArrayListDeque<T> p_339677_) {
+            this.source = p_339677_;
+        }
+
+        @Override
+        public ListAndDeque<T> reversed() {
+            return this.source;
+        }
+
+        @Override
+        public T getFirst() {
+            return this.source.getLast();
+        }
+
+        @Override
+        public T getLast() {
+            return this.source.getFirst();
+        }
+
+        @Override
+        public void addFirst(T p_339593_) {
+            this.source.addLast(p_339593_);
+        }
+
+        @Override
+        public void addLast(T p_339649_) {
+            this.source.addFirst(p_339649_);
+        }
+
+        @Override
+        public boolean offerFirst(T p_339654_) {
+            return this.source.offerLast(p_339654_);
+        }
+
+        @Override
+        public boolean offerLast(T p_339690_) {
+            return this.source.offerFirst(p_339690_);
+        }
+
+        @Override
+        public T pollFirst() {
+            return this.source.pollLast();
+        }
+
+        @Override
+        public T pollLast() {
+            return this.source.pollFirst();
+        }
+
+        @Override
+        public T peekFirst() {
+            return this.source.peekLast();
+        }
+
+        @Override
+        public T peekLast() {
+            return this.source.peekFirst();
+        }
+
+        @Override
+        public T removeFirst() {
+            return this.source.removeLast();
+        }
+
+        @Override
+        public T removeLast() {
+            return this.source.removeFirst();
+        }
+
+        @Override
+        public boolean removeFirstOccurrence(Object p_339684_) {
+            return this.source.removeLastOccurrence(p_339684_);
+        }
+
+        @Override
+        public boolean removeLastOccurrence(Object p_339678_) {
+            return this.source.removeFirstOccurrence(p_339678_);
+        }
+
+        @Override
+        public Iterator<T> descendingIterator() {
+            return this.source.iterator();
+        }
+
+        @Override
+        public int size() {
+            return this.source.size();
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return this.source.isEmpty();
+        }
+
+        @Override
+        public boolean contains(Object p_339624_) {
+            return this.source.contains(p_339624_);
+        }
+
+        @Override
+        public T get(int p_339634_) {
+            return this.source.get(this.reverseIndex(p_339634_));
+        }
+
+        @Override
+        public T set(int p_339669_, T p_339609_) {
+            return this.source.set(this.reverseIndex(p_339669_), p_339609_);
+        }
+
+        @Override
+        public void add(int p_339646_, T p_339625_) {
+            this.source.add(this.reverseIndex(p_339646_) + 1, p_339625_);
+        }
+
+        @Override
+        public T remove(int p_339650_) {
+            return this.source.remove(this.reverseIndex(p_339650_));
+        }
+
+        @Override
+        public int indexOf(Object p_339682_) {
+            return this.reverseIndex(this.source.lastIndexOf(p_339682_));
+        }
+
+        @Override
+        public int lastIndexOf(Object p_339602_) {
+            return this.reverseIndex(this.source.indexOf(p_339602_));
+        }
+
+        @Override
+        public List<T> subList(int p_339640_, int p_339642_) {
+            return this.source.subList(this.reverseIndex(p_339642_) + 1, this.reverseIndex(p_339640_) + 1).reversed();
+        }
+
+        @Override
+        public Iterator<T> iterator() {
+            return this.source.descendingIterator();
+        }
+
+        @Override
+        public void clear() {
+            this.source.clear();
+        }
+
+        private int reverseIndex(int p_339612_) {
+            return p_339612_ == -1 ? -1 : this.source.size() - 1 - p_339612_;
         }
     }
 }
