--- a/net/minecraft/util/ByIdMap.java
+++ b/net/minecraft/util/ByIdMap.java
@@ -14,7 +_,7 @@
         } else {
             Int2ObjectMap<T> int2objectmap = new Int2ObjectOpenHashMap<>();
 
-            for(T t : p_263043_) {
+            for (T t : p_263043_) {
                 int i = p_263047_.applyAsInt(t);
                 T t1 = int2objectmap.put(i, t);
                 if (t1 != null) {
@@ -39,7 +_,7 @@
             T[] at = (T[])p_263053_.clone();
             Arrays.fill(at, null);
 
-            for(T t : p_263053_) {
+            for (T t : p_263053_) {
                 int j = p_262976_.applyAsInt(t);
                 if (j < 0 || j >= i) {
                     throw new IllegalArgumentException("Values are not continous, found index " + j + " for value " + t);
@@ -53,7 +_,7 @@
                 at[j] = t;
             }
 
-            for(int k = 0; k < i; ++k) {
+            for (int k = 0; k < i; k++) {
                 if (at[k] == null) {
                     throw new IllegalArgumentException("Missing value at index: " + k);
                 }
@@ -67,7 +_,7 @@
         T[] at = createSortedArray(p_263112_, p_262975_);
         int i = at.length;
 
-        return switch(p_263075_) {
+        return switch (p_263075_) {
             case ZERO -> {
                 T t = at[0];
                 yield p_262927_ -> p_262927_ >= 0 && p_262927_ < i ? at[p_262927_] : t;
