--- a/net/minecraft/util/ClassTreeIdRegistry.java
+++ b/net/minecraft/util/ClassTreeIdRegistry.java
@@ -15,7 +_,7 @@
         } else {
             Class<?> oclass = p_326222_;
 
-            while((oclass = oclass.getSuperclass()) != Object.class) {
+            while ((oclass = oclass.getSuperclass()) != Object.class) {
                 int j = this.classToLastIdCache.getInt(oclass);
                 if (j != -1) {
                     return j;
