--- a/net/minecraft/util/CsvOutput.java
+++ b/net/minecraft/util/CsvOutput.java
@@ -34,7 +_,7 @@
     }
 
     private void writeLine(Stream<?> p_13623_) throws IOException {
-        this.output.write((String)p_13623_.map(CsvOutput::getStringValue).collect(Collectors.joining(",")) + "\r\n");
+        this.output.write(p_13623_.map(CsvOutput::getStringValue).collect(Collectors.joining(",")) + "\r\n");
     }
 
     private static String getStringValue(@Nullable Object p_13621_) {
