--- a/net/minecraft/util/DebugBuffer.java
+++ b/net/minecraft/util/DebugBuffer.java
@@ -23,7 +_,7 @@
         do {
             j = this.index.get();
             k = (j + 1) % i;
-        } while(!this.index.compareAndSet(j, k));
+        } while (!this.index.compareAndSet(j, k));
 
         this.data.set(k, p_144626_);
     }
@@ -32,7 +_,7 @@
         int i = this.index.get();
         Builder<T> builder = ImmutableList.builder();
 
-        for(int j = 0; j < this.data.length(); ++j) {
+        for (int j = 0; j < this.data.length(); j++) {
             int k = Math.floorMod(i - j, this.data.length());
             T t = this.data.get(k);
             if (t != null) {
