--- a/net/minecraft/util/InclusiveRange.java
+++ b/net/minecraft/util/InclusiveRange.java
@@ -26,28 +_,34 @@
     }
 
     public static <T extends Comparable<T>> Codec<InclusiveRange<T>> codec(Codec<T> p_184575_, T p_184576_, T p_184577_) {
-        return ExtraCodecs.validate(
-            codec(p_184575_),
-            p_274898_ -> {
-                if (p_274898_.minInclusive().compareTo(p_184576_) < 0) {
-                    return DataResult.error(
-                        () -> "Range limit too low, expected at least " + p_184576_ + " [" + p_274898_.minInclusive() + "-" + p_274898_.maxInclusive() + "]"
-                    );
-                } else {
-                    return p_274898_.maxInclusive().compareTo(p_184577_) > 0
-                        ? DataResult.error(
-                            () -> "Range limit too high, expected at most "
-                                    + p_184577_
+        return codec(p_184575_)
+            .validate(
+                p_274898_ -> {
+                    if (p_274898_.minInclusive().compareTo(p_184576_) < 0) {
+                        return DataResult.error(
+                            () -> "Range limit too low, expected at least "
+                                    + p_184576_
                                     + " ["
                                     + p_274898_.minInclusive()
                                     + "-"
                                     + p_274898_.maxInclusive()
                                     + "]"
-                        )
-                        : DataResult.success(p_274898_);
+                        );
+                    } else {
+                        return p_274898_.maxInclusive().compareTo(p_184577_) > 0
+                            ? DataResult.error(
+                                () -> "Range limit too high, expected at most "
+                                        + p_184577_
+                                        + " ["
+                                        + p_274898_.minInclusive()
+                                        + "-"
+                                        + p_274898_.maxInclusive()
+                                        + "]"
+                            )
+                            : DataResult.success(p_274898_);
+                    }
                 }
-            }
-        );
+            );
     }
 
     public static <T extends Comparable<T>> DataResult<InclusiveRange<T>> create(T p_184581_, T p_184582_) {
