--- a/net/minecraft/util/SimpleBitStorage.java
+++ b/net/minecraft/util/SimpleBitStorage.java
@@ -213,10 +_,10 @@
         int j = 0;
 
         int i;
-        for(i = 0; i <= p_198165_ - this.valuesPerLong; i += this.valuesPerLong) {
+        for (i = 0; i <= p_198165_ - this.valuesPerLong; i += this.valuesPerLong) {
             long k = 0L;
 
-            for(int i1 = this.valuesPerLong - 1; i1 >= 0; --i1) {
+            for (int i1 = this.valuesPerLong - 1; i1 >= 0; i1--) {
                 k <<= p_198164_;
                 k |= (long)p_198166_[i + i1] & this.mask;
             }
@@ -228,7 +_,7 @@
         if (k1 > 0) {
             long l = 0L;
 
-            for(int j1 = k1 - 1; j1 >= 0; --j1) {
+            for (int j1 = k1 - 1; j1 >= 0; j1--) {
                 l <<= p_198164_;
                 l |= (long)p_198166_[i + j1] & this.mask;
             }
@@ -319,8 +_,8 @@
     public void getAll(IntConsumer p_184734_) {
         int i = 0;
 
-        for(long j : this.data) {
-            for(int k = 0; k < this.valuesPerLong; ++k) {
+        for (long j : this.data) {
+            for (int k = 0; k < this.valuesPerLong; k++) {
                 p_184734_.accept((int)(j & this.mask));
                 j >>= this.bits;
                 if (++i >= this.size) {
@@ -335,10 +_,10 @@
         int i = this.data.length;
         int j = 0;
 
-        for(int k = 0; k < i - 1; ++k) {
+        for (int k = 0; k < i - 1; k++) {
             long l = this.data[k];
 
-            for(int i1 = 0; i1 < this.valuesPerLong; ++i1) {
+            for (int i1 = 0; i1 < this.valuesPerLong; i1++) {
                 p_198168_[j + i1] = (int)(l & this.mask);
                 l >>= this.bits;
             }
@@ -350,7 +_,7 @@
         if (j1 > 0) {
             long k1 = this.data[i - 1];
 
-            for(int l1 = 0; l1 < j1; ++l1) {
+            for (int l1 = 0; l1 < j1; l1++) {
                 p_198168_[j + l1] = (int)(k1 & this.mask);
                 k1 >>= this.bits;
             }
