--- a/net/minecraft/util/SingleKeyCache.java
+++ b/net/minecraft/util/SingleKeyCache.java
@@ -3,24 +_,24 @@
 import java.util.Objects;
 import java.util.function.Function;
 import javax.annotation.Nullable;
-import net.minecraft.Maybe;
 
 public class SingleKeyCache<K, V> {
     private final Function<K, V> computeValue;
     @Nullable
     private K cacheKey = (K)null;
-    private Maybe<V> cachedValue = Maybe.no();
+    @Nullable
+    private V cachedValue;
 
     public SingleKeyCache(Function<K, V> p_270132_) {
         this.computeValue = p_270132_;
     }
 
     public V getValue(K p_270953_) {
-        if (this.cachedValue.isEmpty() || !Objects.equals(this.cacheKey, p_270953_)) {
-            this.cachedValue = Maybe.yes(this.computeValue.apply(p_270953_));
+        if (this.cachedValue == null || !Objects.equals(this.cacheKey, p_270953_)) {
+            this.cachedValue = this.computeValue.apply(p_270953_);
             this.cacheKey = p_270953_;
         }
 
-        return this.cachedValue.getValue();
+        return this.cachedValue;
     }
 }
