--- a/net/minecraft/util/SortedArraySet.java
+++ b/net/minecraft/util/SortedArraySet.java
@@ -84,11 +_,11 @@
         }
 
         this.contents[p_14257_] = p_14256_;
-        ++this.size;
+        this.size++;
     }
 
     void removeInternal(int p_14275_) {
-        --this.size;
+        this.size--;
         if (p_14275_ != this.size) {
             System.arraycopy(this.contents, p_14275_ + 1, this.contents, p_14275_, this.size - p_14275_);
         }
@@ -159,7 +_,7 @@
     @Override
     public <U> U[] toArray(U[] p_14286_) {
         if (p_14286_.length < this.size) {
-            return (U[])Arrays.copyOf(this.contents, this.size, p_14286_.getClass());
+            return (U[])Arrays.copyOf(this.contents, this.size, (Class<? extends T[]>)p_14286_.getClass());
         } else {
             System.arraycopy(this.contents, 0, p_14286_, 0, this.size);
             if (p_14286_.length > this.size) {
@@ -181,7 +_,7 @@
         if (this == p_14279_) {
             return true;
         } else {
-            if (p_14279_ instanceof SortedArraySet sortedarrayset && this.comparator.equals(sortedarrayset.comparator)) {
+            if (p_14279_ instanceof SortedArraySet<?> sortedarrayset && this.comparator.equals(sortedarrayset.comparator)) {
                 return this.size == sortedarrayset.size && Arrays.equals(this.contents, sortedarrayset.contents);
             }
 
@@ -214,7 +_,7 @@
                 throw new IllegalStateException();
             } else {
                 SortedArraySet.this.removeInternal(this.last);
-                --this.index;
+                this.index--;
                 this.last = -1;
             }
         }
