--- a/net/minecraft/util/SpawnUtil.java
+++ b/net/minecraft/util/SpawnUtil.java
@@ -27,15 +_,15 @@
     ) {
         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_216407_.mutable();
 
-        for(int i = 0; i < p_216408_; ++i) {
+        for (int i = 0; i < p_216408_; i++) {
             int j = Mth.randomBetweenInclusive(p_216406_.random, -p_216409_, p_216409_);
             int k = Mth.randomBetweenInclusive(p_216406_.random, -p_216409_, p_216409_);
             blockpos$mutableblockpos.setWithOffset(p_216407_, j, p_216410_, k);
             if (p_216406_.getWorldBorder().isWithinBounds(blockpos$mutableblockpos)
                 && moveToPossibleSpawnPosition(p_216406_, p_216410_, blockpos$mutableblockpos, p_216411_)) {
-                T t = p_216404_.create(p_216406_, null, blockpos$mutableblockpos, p_216405_, false, false);
+                T t = (T)p_216404_.create(p_216406_, null, blockpos$mutableblockpos, p_216405_, false, false);
                 if (t != null) {
-                    if (t.checkSpawnRules(p_216406_, p_216405_) && t.checkSpawnObstruction(p_216406_)) {
+                    if (net.neoforged.neoforge.event.EventHooks.checkSpawnPosition(t, p_216406_, p_216405_)) {
                         p_216406_.addFreshEntityWithPassengers(t);
                         return Optional.of(t);
                     }
@@ -52,7 +_,7 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos().set(p_216401_);
         BlockState blockstate = p_216399_.getBlockState(blockpos$mutableblockpos);
 
-        for(int i = p_216400_; i >= -p_216400_; --i) {
+        for (int i = p_216400_; i >= -p_216400_; i--) {
             p_216401_.move(Direction.DOWN);
             blockpos$mutableblockpos.setWithOffset(p_216401_, Direction.UP);
             BlockState blockstate1 = p_216399_.getBlockState(p_216401_);
@@ -69,27 +_,23 @@
 
     public interface Strategy {
         @Deprecated
-        SpawnUtil.Strategy LEGACY_IRON_GOLEM = (p_289751_, p_289752_, p_289753_, p_289754_, p_289755_) -> {
-            if (!p_289753_.is(Blocks.COBWEB)
-                && !p_289753_.is(Blocks.CACTUS)
-                && !p_289753_.is(Blocks.GLASS_PANE)
-                && !(p_289753_.getBlock() instanceof StainedGlassPaneBlock)
-                && !(p_289753_.getBlock() instanceof StainedGlassBlock)
-                && !(p_289753_.getBlock() instanceof LeavesBlock)
-                && !p_289753_.is(Blocks.CONDUIT)
-                && !p_289753_.is(Blocks.ICE)
-                && !p_289753_.is(Blocks.TNT)
-                && !p_289753_.is(Blocks.GLOWSTONE)
-                && !p_289753_.is(Blocks.BEACON)
-                && !p_289753_.is(Blocks.SEA_LANTERN)
-                && !p_289753_.is(Blocks.FROSTED_ICE)
-                && !p_289753_.is(Blocks.TINTED_GLASS)
-                && !p_289753_.is(Blocks.GLASS)) {
-                return (p_289755_.isAir() || p_289755_.liquid()) && (p_289753_.isSolid() || p_289753_.is(Blocks.POWDER_SNOW));
-            } else {
-                return false;
-            }
-        };
+        SpawnUtil.Strategy LEGACY_IRON_GOLEM = (p_289751_, p_289752_, p_289753_, p_289754_, p_289755_) -> !p_289753_.is(Blocks.COBWEB)
+                    && !p_289753_.is(Blocks.CACTUS)
+                    && !p_289753_.is(Blocks.GLASS_PANE)
+                    && !(p_289753_.getBlock() instanceof StainedGlassPaneBlock)
+                    && !(p_289753_.getBlock() instanceof StainedGlassBlock)
+                    && !(p_289753_.getBlock() instanceof LeavesBlock)
+                    && !p_289753_.is(Blocks.CONDUIT)
+                    && !p_289753_.is(Blocks.ICE)
+                    && !p_289753_.is(Blocks.TNT)
+                    && !p_289753_.is(Blocks.GLOWSTONE)
+                    && !p_289753_.is(Blocks.BEACON)
+                    && !p_289753_.is(Blocks.SEA_LANTERN)
+                    && !p_289753_.is(Blocks.FROSTED_ICE)
+                    && !p_289753_.is(Blocks.TINTED_GLASS)
+                    && !p_289753_.is(Blocks.GLASS)
+                ? (p_289755_.isAir() || p_289755_.liquid()) && (p_289753_.isSolid() || p_289753_.is(Blocks.POWDER_SNOW))
+                : false;
         SpawnUtil.Strategy ON_TOP_OF_COLLIDER = (p_216416_, p_216417_, p_216418_, p_216419_, p_216420_) -> p_216420_.getCollisionShape(p_216416_, p_216419_)
                     .isEmpty()
                 && Block.isFaceFull(p_216418_.getCollisionShape(p_216416_, p_216417_), Direction.UP);
