--- a/net/minecraft/util/StringDecomposer.java
+++ b/net/minecraft/util/StringDecomposer.java
@@ -16,7 +_,7 @@
     public static boolean iterate(String p_14318_, Style p_14319_, FormattedCharSink p_14320_) {
         int i = p_14318_.length();
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             char c0 = p_14318_.charAt(j);
             if (Character.isHighSurrogate(c0)) {
                 if (j + 1 >= i) {
@@ -32,7 +_,7 @@
                         return false;
                     }
 
-                    ++j;
+                    j++;
                 } else if (!p_14320_.accept(j, p_14319_, 65533)) {
                     return false;
                 }
@@ -47,7 +_,7 @@
     public static boolean iterateBackwards(String p_14338_, Style p_14339_, FormattedCharSink p_14340_) {
         int i = p_14338_.length();
 
-        for(int j = i - 1; j >= 0; --j) {
+        for (int j = i - 1; j >= 0; j--) {
             char c0 = p_14338_.charAt(j);
             if (Character.isLowSurrogate(c0)) {
                 if (j - 1 < 0) {
@@ -85,7 +_,7 @@
         int i = p_14312_.length();
         Style style = p_14314_;
 
-        for(int j = p_14313_; j < i; ++j) {
+        for (int j = p_14313_; j < i; j++) {
             char c0 = p_14312_.charAt(j);
             if (c0 == 167) {
                 if (j + 1 >= i) {
@@ -98,7 +_,7 @@
                     style = chatformatting == ChatFormatting.RESET ? p_14315_ : style.applyLegacyFormat(chatformatting);
                 }
 
-                ++j;
+                j++;
             } else if (Character.isHighSurrogate(c0)) {
                 if (j + 1 >= i) {
                     if (!p_14316_.accept(j, style, 65533)) {
@@ -113,7 +_,7 @@
                         return false;
                     }
 
-                    ++j;
+                    j++;
                 } else if (!p_14316_.accept(j, style, 65533)) {
                     return false;
                 }
