--- a/net/minecraft/util/StringRepresentable.java
+++ b/net/minecraft/util/StringRepresentable.java
@@ -42,12 +_,12 @@
 
     static <T extends StringRepresentable> Function<String, T> createNameLookup(T[] p_304419_, Function<String, String> p_304658_) {
         if (p_304419_.length > 16) {
-            Map<String, T> map = Arrays.stream(p_304419_)
-                .collect(Collectors.toMap(p_304335_ -> p_304658_.apply(p_304335_.getSerializedName()), p_304719_ -> p_304719_));
+            Map<String, T> map = Arrays.<StringRepresentable>stream(p_304419_)
+                .collect(Collectors.toMap(p_304335_ -> p_304658_.apply(p_304335_.getSerializedName()), p_304719_ -> (T)p_304719_));
             return p_304332_ -> p_304332_ == null ? null : map.get(p_304332_);
         } else {
             return p_304338_ -> {
-                for(T t : p_304419_) {
+                for (T t : p_304419_) {
                     if (p_304658_.apply(t.getSerializedName()).equals(p_304338_)) {
                         return t;
                     }
@@ -91,7 +_,7 @@
 
         public StringRepresentableCodec(S[] p_304774_, Function<String, S> p_304834_, ToIntFunction<S> p_304925_) {
             this.codec = ExtraCodecs.orCompressed(
-                ExtraCodecs.stringResolverCodec(StringRepresentable::getSerializedName, p_304834_),
+                Codec.stringResolver(StringRepresentable::getSerializedName, p_304834_),
                 ExtraCodecs.idResolverCodec(p_304925_, p_304986_ -> p_304986_ >= 0 && p_304986_ < p_304774_.length ? p_304774_[p_304986_] : null, -1)
             );
         }
