--- a/net/minecraft/util/StringUtil.java
+++ b/net/minecraft/util/StringUtil.java
@@ -43,8 +_,8 @@
             Matcher matcher = LINE_PATTERN.matcher(p_145003_);
             int i = 1;
 
-            while(matcher.find()) {
-                ++i;
+            while (matcher.find()) {
+                i++;
             }
 
             return i;
@@ -74,7 +_,7 @@
     public static String filterText(String p_332058_, boolean p_330572_) {
         StringBuilder stringbuilder = new StringBuilder();
 
-        for(char c0 : p_332058_.toCharArray()) {
+        for (char c0 : p_332058_.toCharArray()) {
             if (isAllowedChatCharacter(c0)) {
                 stringbuilder.append(c0);
             } else if (p_330572_ && c0 == '\n') {
