--- a/net/minecraft/util/ThreadingDetector.java
+++ b/net/minecraft/util/ThreadingDetector.java
@@ -81,8 +_,6 @@
     }
 
     private static String stackTrace(Thread p_199421_) {
-        return p_199421_.getName()
-            + ": \n\tat "
-            + (String)Arrays.stream(p_199421_.getStackTrace()).map(Object::toString).collect(Collectors.joining("\n\tat "));
+        return p_199421_.getName() + ": \n\tat " + Arrays.stream(p_199421_.getStackTrace()).map(Object::toString).collect(Collectors.joining("\n\tat "));
     }
 }
