--- a/net/minecraft/world/entity/ai/village/poi/PoiManager.java
+++ b/net/minecraft/world/entity/ai/village/poi/PoiManager.java
@@ -244,12 +_,12 @@
                         SectionPos.sectionRelative(p_217902_.getY()),
                         SectionPos.sectionRelative(p_217902_.getZ())
                     );
-                    PoiTypes.forState(blockstate).ifPresent(p_217931_ -> p_27072_.accept(p_217902_, p_217931_));
+                    PoiTypes.forState(blockstate).ifPresent(p_217931_ -> p_27072_.accept(p_217902_, (Holder<PoiType>)p_217931_));
                 }
             );
     }
 
-    public void ensureLoadedAndValid(LevelReader p_27057_, BlockPos p_27058_, int p_27059_, ChunkStatus p_341699_) {
+    public void ensureLoadedAndValid(LevelReader p_27057_, BlockPos p_27058_, int p_27059_) {
         SectionPos.aroundChunk(
                 new ChunkPos(p_27058_), Math.floorDiv(p_27059_, 16), this.levelHeightAccessor.getMinSection(), this.levelHeightAccessor.getMaxSection()
             )
@@ -257,7 +_,7 @@
             .filter(p_217963_ -> !p_217963_.getSecond().map(PoiSection::isValid).orElse(false))
             .map(p_217891_ -> p_217891_.getFirst().chunk())
             .filter(p_217961_ -> this.loadedChunks.add(p_217961_.toLong()))
-            .forEach(p_340105_ -> p_27057_.getChunk(p_340105_.x, p_340105_.z, p_341699_));
+            .forEach(p_330057_ -> p_27057_.getChunk(p_330057_.x, p_330057_.z, ChunkStatus.EMPTY));
     }
 
     final class DistanceTracker extends SectionTracker {
