--- a/net/minecraft/world/entity/ai/village/poi/PoiRecord.java
+++ b/net/minecraft/world/entity/ai/village/poi/PoiRecord.java
@@ -49,7 +_,7 @@
         if (this.freeTickets <= 0) {
             return false;
         } else {
-            --this.freeTickets;
+            this.freeTickets--;
             this.setDirty.run();
             return true;
         }
@@ -59,7 +_,7 @@
         if (this.freeTickets >= this.poiType.value().maxTickets()) {
             return false;
         } else {
-            ++this.freeTickets;
+            this.freeTickets++;
             this.setDirty.run();
             return true;
         }
