--- a/net/minecraft/network/VarInt.java
+++ b/net/minecraft/network/VarInt.java
@@ -9,7 +_,7 @@
     private static final int DATA_BITS_PER_BYTE = 7;
 
     public static int getByteSize(int p_295994_) {
-        for(int i = 1; i < 5; ++i) {
+        for (int i = 1; i < 5; i++) {
             if ((p_295994_ & -1 << i * 7) == 0) {
                 return i;
             }
@@ -33,13 +_,13 @@
             if (j > 5) {
                 throw new RuntimeException("VarInt too big");
             }
-        } while(hasContinuationBit(b0));
+        } while (hasContinuationBit(b0));
 
         return i;
     }
 
     public static ByteBuf write(ByteBuf p_295539_, int p_294256_) {
-        while((p_294256_ & -128) != 0) {
+        while ((p_294256_ & -128) != 0) {
             p_295539_.writeByte(p_294256_ & 127 | 128);
             p_294256_ >>>= 7;
         }
