--- a/net/minecraft/network/VarLong.java
+++ b/net/minecraft/network/VarLong.java
@@ -9,7 +_,7 @@
     private static final int DATA_BITS_PER_BYTE = 7;
 
     public static int getByteSize(long p_295931_) {
-        for(int i = 1; i < 10; ++i) {
+        for (int i = 1; i < 10; i++) {
             if ((p_295931_ & -1L << i * 7) == 0L) {
                 return i;
             }
@@ -33,13 +_,13 @@
             if (j > 10) {
                 throw new RuntimeException("VarLong too big");
             }
-        } while(hasContinuationBit(b0));
+        } while (hasContinuationBit(b0));
 
         return i;
     }
 
     public static ByteBuf write(ByteBuf p_296144_, long p_295377_) {
-        while((p_295377_ & -128L) != 0L) {
+        while ((p_295377_ & -128L) != 0L) {
             p_296144_.writeByte((int)(p_295377_ & 127L) | 128);
             p_295377_ >>>= 7;
         }
