--- a/net/minecraft/world/level/block/entity/BaseContainerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BaseContainerBlockEntity.java
@@ -32,11 +_,11 @@
     }
 
     @Override
-    public void load(CompoundTag p_155080_, HolderLookup.Provider p_324522_) {
-        super.load(p_155080_, p_324522_);
-        this.lockKey = LockCode.fromTag(p_155080_);
-        if (p_155080_.contains("CustomName", 8)) {
-            this.name = Component.Serializer.fromJson(p_155080_.getString("CustomName"), p_324522_);
+    protected void loadAdditional(CompoundTag p_338606_, HolderLookup.Provider p_338309_) {
+        super.loadAdditional(p_338606_, p_338309_);
+        this.lockKey = LockCode.fromTag(p_338606_);
+        if (p_338606_.contains("CustomName", 8)) {
+            this.name = Component.Serializer.fromJson(p_338606_.getString("CustomName"), p_338309_);
         }
     }
 
@@ -87,7 +_,7 @@
 
     @Override
     public boolean isEmpty() {
-        for(ItemStack itemstack : this.getItems()) {
+        for (ItemStack itemstack : this.getItems()) {
             if (!itemstack.isEmpty()) {
                 return false;
             }
@@ -142,20 +_,22 @@
     protected abstract AbstractContainerMenu createMenu(int p_58627_, Inventory p_58628_);
 
     @Override
-    public void applyComponents(DataComponentMap p_331820_) {
-        this.name = p_331820_.get(DataComponents.CUSTOM_NAME);
-        this.lockKey = p_331820_.getOrDefault(DataComponents.LOCK, LockCode.NO_LOCK);
-        p_331820_.getOrDefault(DataComponents.CONTAINER, ItemContainerContents.EMPTY).copyInto(this.getItems());
+    protected void applyImplicitComponents(BlockEntity.DataComponentInput p_338855_) {
+        super.applyImplicitComponents(p_338855_);
+        this.name = p_338855_.get(DataComponents.CUSTOM_NAME);
+        this.lockKey = p_338855_.getOrDefault(DataComponents.LOCK, LockCode.NO_LOCK);
+        p_338855_.getOrDefault(DataComponents.CONTAINER, ItemContainerContents.EMPTY).copyInto(this.getItems());
     }
 
     @Override
-    public void collectComponents(DataComponentMap.Builder p_330778_) {
-        p_330778_.set(DataComponents.CUSTOM_NAME, this.name);
+    protected void collectImplicitComponents(DataComponentMap.Builder p_338252_) {
+        super.collectImplicitComponents(p_338252_);
+        p_338252_.set(DataComponents.CUSTOM_NAME, this.name);
         if (!this.lockKey.equals(LockCode.NO_LOCK)) {
-            p_330778_.set(DataComponents.LOCK, this.lockKey);
+            p_338252_.set(DataComponents.LOCK, this.lockKey);
         }
 
-        p_330778_.set(DataComponents.CONTAINER, ItemContainerContents.copyOf(this.getItems()));
+        p_338252_.set(DataComponents.CONTAINER, ItemContainerContents.copyOf(this.getItems()));
     }
 
     @Override
