--- a/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
@@ -77,7 +_,7 @@
     private final ContainerData dataAccess = new ContainerData() {
         @Override
         public int get(int p_58711_) {
-            return switch(p_58711_) {
+            return switch (p_58711_) {
                 case 0 -> BeaconBlockEntity.this.levels;
                 case 1 -> BeaconMenu.encodeEffect(BeaconBlockEntity.this.primaryPower);
                 case 2 -> BeaconMenu.encodeEffect(BeaconBlockEntity.this.secondaryPower);
@@ -87,7 +_,7 @@
 
         @Override
         public void set(int p_58713_, int p_58714_) {
-            switch(p_58713_) {
+            switch (p_58713_) {
                 case 0:
                     BeaconBlockEntity.this.levels = p_58714_;
                     break;
@@ -136,11 +_,11 @@
             : p_155111_.checkingBeamSections.get(p_155111_.checkingBeamSections.size() - 1);
         int l = p_155108_.getHeight(Heightmap.Types.WORLD_SURFACE, i, k);
 
-        for(int i1 = 0; i1 < 10 && blockpos.getY() <= l; ++i1) {
+        for (int i1 = 0; i1 < 10 && blockpos.getY() <= l; i1++) {
             BlockState blockstate = p_155108_.getBlockState(blockpos);
             Block block = blockstate.getBlock();
-            if (block instanceof BeaconBeamBlock) {
-                float[] afloat = ((BeaconBeamBlock)block).getColor().getTextureDiffuseColors();
+            float[] afloat = blockstate.getBeaconColorMultiplier(p_155108_, blockpos, p_155109_);
+            if (afloat != null) {
                 if (p_155111_.checkingBeamSections.size() <= 1) {
                     beaconblockentity$beaconbeamsection = new BeaconBlockEntity.BeaconBeamSection(afloat);
                     p_155111_.checkingBeamSections.add(beaconblockentity$beaconbeamsection);
@@ -169,7 +_,7 @@
             }
 
             blockpos = blockpos.above();
-            ++p_155111_.lastCheckY;
+            p_155111_.lastCheckY++;
         }
 
         int j1 = p_155111_.levels;
@@ -193,7 +_,7 @@
                 if (!flag && flag1) {
                     playSound(p_155108_, p_155109_, SoundEvents.BEACON_ACTIVATE);
 
-                    for(ServerPlayer serverplayer : p_155108_.getEntitiesOfClass(
+                    for (ServerPlayer serverplayer : p_155108_.getEntitiesOfClass(
                         ServerPlayer.class, new AABB((double)i, (double)j, (double)k, (double)i, (double)(j - 4), (double)k).inflate(10.0, 5.0, 10.0)
                     )) {
                         CriteriaTriggers.CONSTRUCT_BEACON.trigger(serverplayer, p_155111_.levels);
@@ -208,7 +_,7 @@
     private static int updateBase(Level p_155093_, int p_155094_, int p_155095_, int p_155096_) {
         int i = 0;
 
-        for(int j = 1; j <= 4; i = j++) {
+        for (int j = 1; j <= 4; i = j++) {
             int k = p_155095_ - j;
             if (k < p_155093_.getMinBuildHeight()) {
                 break;
@@ -216,8 +_,8 @@
 
             boolean flag = true;
 
-            for(int l = p_155094_ - j; l <= p_155094_ + j && flag; ++l) {
-                for(int i1 = p_155096_ - j; i1 <= p_155096_ + j; ++i1) {
+            for (int l = p_155094_ - j; l <= p_155094_ + j && flag; l++) {
+                for (int i1 = p_155096_ - j; i1 <= p_155096_ + j; i1++) {
                     if (!p_155093_.getBlockState(new BlockPos(l, k, i1)).is(BlockTags.BEACON_BASE_BLOCKS)) {
                         flag = false;
                         break;
@@ -253,12 +_,12 @@
             AABB aabb = new AABB(p_155099_).inflate(d0).expandTowards(0.0, (double)p_155098_.getHeight(), 0.0);
             List<Player> list = p_155098_.getEntitiesOfClass(Player.class, aabb);
 
-            for(Player player : list) {
+            for (Player player : list) {
                 player.addEffect(new MobEffectInstance(p_316599_, j, i, true, true));
             }
 
             if (p_155100_ >= 4 && !Objects.equals(p_316599_, p_316343_) && p_316343_ != null) {
-                for(Player player1 : list) {
+                for (Player player1 : list) {
                     player1.addEffect(new MobEffectInstance(p_316343_, j, 0, true, true));
                 }
             }
@@ -279,7 +_,7 @@
 
     @Override
     public CompoundTag getUpdateTag(HolderLookup.Provider p_324570_) {
-        return this.saveWithoutMetadata(p_324570_);
+        return this.saveCustomOnly(p_324570_);
     }
 
     private static void storeEffect(CompoundTag p_298214_, String p_298983_, @Nullable Holder<MobEffect> p_316689_) {
@@ -299,15 +_,15 @@
     }
 
     @Override
-    public void load(CompoundTag p_155113_, HolderLookup.Provider p_324371_) {
-        super.load(p_155113_, p_324371_);
-        this.primaryPower = loadEffect(p_155113_, "primary_effect");
-        this.secondaryPower = loadEffect(p_155113_, "secondary_effect");
-        if (p_155113_.contains("CustomName", 8)) {
-            this.name = Component.Serializer.fromJson(p_155113_.getString("CustomName"), p_324371_);
+    protected void loadAdditional(CompoundTag p_338669_, HolderLookup.Provider p_338291_) {
+        super.loadAdditional(p_338669_, p_338291_);
+        this.primaryPower = loadEffect(p_338669_, "primary_effect");
+        this.secondaryPower = loadEffect(p_338669_, "secondary_effect");
+        if (p_338669_.contains("CustomName", 8)) {
+            this.name = Component.Serializer.fromJson(p_338669_.getString("CustomName"), p_338291_);
         }
 
-        this.lockKey = LockCode.fromTag(p_155113_);
+        this.lockKey = LockCode.fromTag(p_338669_);
     }
 
     @Override
@@ -352,16 +_,18 @@
     }
 
     @Override
-    public void applyComponents(DataComponentMap p_330941_) {
-        this.name = p_330941_.get(DataComponents.CUSTOM_NAME);
-        this.lockKey = p_330941_.getOrDefault(DataComponents.LOCK, LockCode.NO_LOCK);
+    protected void applyImplicitComponents(BlockEntity.DataComponentInput p_338364_) {
+        super.applyImplicitComponents(p_338364_);
+        this.name = p_338364_.get(DataComponents.CUSTOM_NAME);
+        this.lockKey = p_338364_.getOrDefault(DataComponents.LOCK, LockCode.NO_LOCK);
     }
 
     @Override
-    public void collectComponents(DataComponentMap.Builder p_331091_) {
-        p_331091_.set(DataComponents.CUSTOM_NAME, this.name);
+    protected void collectImplicitComponents(DataComponentMap.Builder p_338239_) {
+        super.collectImplicitComponents(p_338239_);
+        p_338239_.set(DataComponents.CUSTOM_NAME, this.name);
         if (!this.lockKey.equals(LockCode.NO_LOCK)) {
-            p_331091_.set(DataComponents.LOCK, this.lockKey);
+            p_338239_.set(DataComponents.LOCK, this.lockKey);
         }
     }
 
@@ -387,7 +_,7 @@
         }
 
         protected void increaseHeight() {
-            ++this.height;
+            this.height++;
         }
 
         public float[] getColor() {
