--- a/net/minecraft/world/level/block/entity/BellBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BellBlockEntity.java
@@ -57,7 +_,7 @@
         Level p_155181_, BlockPos p_155182_, BlockState p_155183_, BellBlockEntity p_155184_, BellBlockEntity.ResonationEndAction p_155185_
     ) {
         if (p_155184_.shaking) {
-            ++p_155184_.ticks;
+            p_155184_.ticks++;
         }
 
         if (p_155184_.ticks >= 50) {
@@ -72,7 +_,7 @@
 
         if (p_155184_.resonating) {
             if (p_155184_.resonationTicks < 40) {
-                ++p_155184_.resonationTicks;
+                p_155184_.resonationTicks++;
             } else {
                 p_155185_.run(p_155181_, p_155182_, p_155184_.nearbyEntities);
                 p_155184_.resonating = false;
@@ -109,7 +_,7 @@
         }
 
         if (!this.level.isClientSide) {
-            for(LivingEntity livingentity : this.nearbyEntities) {
+            for (LivingEntity livingentity : this.nearbyEntities) {
                 if (livingentity.isAlive() && !livingentity.isRemoved() && blockpos.closerToCenterThan(livingentity.position(), 32.0)) {
                     livingentity.getBrain().setMemory(MemoryModuleType.HEARD_BELL_TIME, this.level.getGameTime());
                 }
@@ -118,7 +_,7 @@
     }
 
     private static boolean areRaidersNearby(BlockPos p_155200_, List<LivingEntity> p_155201_) {
-        for(LivingEntity livingentity : p_155201_) {
+        for (LivingEntity livingentity : p_155201_) {
             if (livingentity.isAlive()
                 && !livingentity.isRemoved()
                 && p_155200_.closerToCenterThan(livingentity.position(), 32.0)
@@ -136,7 +_,7 @@
 
     private static void showBellParticles(Level p_155208_, BlockPos p_155209_, List<LivingEntity> p_155210_) {
         MutableInt mutableint = new MutableInt(16700985);
-        int i = (int)p_155210_.stream().filter(p_340255_ -> p_155209_.closerToCenterThan(p_340255_.position(), 48.0)).count();
+        int i = (int)p_155210_.stream().filter(p_337986_ -> p_155209_.closerToCenterThan(p_337986_.position(), 48.0)).count();
         p_155210_.stream()
             .filter(p_155213_ -> isRaiderWithinRange(p_155209_, p_155213_))
             .forEach(
@@ -150,7 +_,7 @@
                     double d2 = (double)((float)p_155209_.getZ() + 0.5F) + 1.0 / d0 * (p_333683_.getZ() - (double)p_155209_.getZ());
                     int j = Mth.clamp((i - 21) / -2, 3, 15);
         
-                    for(int k = 0; k < j; ++k) {
+                    for (int k = 0; k < j; k++) {
                         int l = mutableint.addAndGet(5);
                         p_155208_.addParticle(
                             ColorParticleOption.create(ParticleTypes.ENTITY_EFFECT, l), d1, (double)((float)p_155209_.getY() + 0.5F), d2, 0.0, 0.0, 0.0
