--- a/net/minecraft/world/level/block/entity/BlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BlockEntity.java
@@ -1,13 +_,21 @@
 package net.minecraft.world.level.block.entity;
 
 import com.mojang.logging.LogUtils;
+import com.mojang.serialization.Codec;
+import java.util.HashSet;
+import java.util.Set;
 import javax.annotation.Nullable;
 import net.minecraft.CrashReportCategory;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.HolderLookup;
 import net.minecraft.core.component.DataComponentMap;
+import net.minecraft.core.component.DataComponentPatch;
+import net.minecraft.core.component.DataComponentType;
+import net.minecraft.core.component.PatchedDataComponentMap;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.NbtOps;
+import net.minecraft.nbt.Tag;
 import net.minecraft.network.protocol.Packet;
 import net.minecraft.network.protocol.game.ClientGamePacketListener;
 import net.minecraft.resources.ResourceLocation;
@@ -17,7 +_,7 @@
 import net.minecraft.world.level.block.state.BlockState;
 import org.slf4j.Logger;
 
-public abstract class BlockEntity {
+public abstract class BlockEntity extends net.neoforged.neoforge.attachment.AttachmentHolder implements net.neoforged.neoforge.common.extensions.IBlockEntityExtension {
     private static final Logger LOGGER = LogUtils.getLogger();
     private final BlockEntityType<?> type;
     @Nullable
@@ -25,6 +_,9 @@
     protected final BlockPos worldPosition;
     protected boolean remove;
     private BlockState blockState;
+    private DataComponentMap components = DataComponentMap.EMPTY;
+    @Nullable
+    private CompoundTag customPersistentData;
 
     public BlockEntity(BlockEntityType<?> p_155228_, BlockPos p_155229_, BlockState p_155230_) {
         this.type = p_155228_;
@@ -49,10 +_,27 @@
         return this.level != null;
     }
 
-    public void load(CompoundTag p_155245_, HolderLookup.Provider p_323967_) {
+    protected void loadAdditional(CompoundTag p_338466_, HolderLookup.Provider p_338445_) {
+        if (p_338466_.contains("NeoForgeData", net.minecraft.nbt.Tag.TAG_COMPOUND)) this.customPersistentData = p_338466_.getCompound("NeoForgeData");
+        if (p_338466_.contains(ATTACHMENTS_NBT_KEY, net.minecraft.nbt.Tag.TAG_COMPOUND)) deserializeAttachments(p_338445_, p_338466_.getCompound(ATTACHMENTS_NBT_KEY));
+    }
+
+    public final void loadWithComponents(CompoundTag p_338356_, HolderLookup.Provider p_338558_) {
+        this.loadAdditional(p_338356_, p_338558_);
+        BlockEntity.ComponentHelper.COMPONENTS_CODEC
+            .parse(p_338558_.createSerializationContext(NbtOps.INSTANCE), p_338356_)
+            .resultOrPartial(p_337987_ -> LOGGER.warn("Failed to load components: {}", p_337987_))
+            .ifPresent(p_337995_ -> this.components = p_337995_);
+    }
+
+    public final void loadCustomOnly(CompoundTag p_338387_, HolderLookup.Provider p_338713_) {
+        this.loadAdditional(p_338387_, p_338713_);
     }
 
     protected void saveAdditional(CompoundTag p_187471_, HolderLookup.Provider p_323635_) {
+        if (this.customPersistentData != null) p_187471_.put("NeoForgeData", this.customPersistentData.copy());
+        var attachmentsTag = serializeAttachments(p_323635_);
+        if (attachmentsTag != null) p_187471_.put(ATTACHMENTS_NBT_KEY, attachmentsTag);
     }
 
     public final CompoundTag saveWithFullMetadata(HolderLookup.Provider p_323767_) {
@@ -70,6 +_,22 @@
     public final CompoundTag saveWithoutMetadata(HolderLookup.Provider p_324030_) {
         CompoundTag compoundtag = new CompoundTag();
         this.saveAdditional(compoundtag, p_324030_);
+        BlockEntity.ComponentHelper.COMPONENTS_CODEC
+            .encodeStart(p_324030_.createSerializationContext(NbtOps.INSTANCE), this.components)
+            .resultOrPartial(p_337988_ -> LOGGER.warn("Failed to save components: {}", p_337988_))
+            .ifPresent(p_337994_ -> compoundtag.merge((CompoundTag)p_337994_));
+        return compoundtag;
+    }
+
+    public final CompoundTag saveCustomOnly(HolderLookup.Provider p_338656_) {
+        CompoundTag compoundtag = new CompoundTag();
+        this.saveAdditional(compoundtag, p_338656_);
+        return compoundtag;
+    }
+
+    public final CompoundTag saveCustomAndMetadata(HolderLookup.Provider p_339688_) {
+        CompoundTag compoundtag = this.saveCustomOnly(p_339688_);
+        this.saveMetadata(compoundtag);
         return compoundtag;
     }
 
@@ -87,7 +_,7 @@
     }
 
     public void saveToItem(ItemStack p_187477_, HolderLookup.Provider p_323484_) {
-        CompoundTag compoundtag = this.saveWithoutMetadata(p_323484_);
+        CompoundTag compoundtag = this.saveCustomOnly(p_323484_);
         this.removeComponentsFromTag(compoundtag);
         BlockItem.setBlockEntityData(p_187477_, this.getType(), compoundtag);
         p_187477_.applyComponents(this.collectComponents());
@@ -115,10 +_,10 @@
                     LOGGER.error("Failed to create block entity {}", s, throwable);
                     return null;
                 }
-            }).map(p_323417_ -> {
+            }).map(p_337992_ -> {
                 try {
-                    p_323417_.load(p_155244_, p_323542_);
-                    return p_323417_;
+                    p_337992_.loadWithComponents(p_155244_, p_323542_);
+                    return (BlockEntity)p_337992_;
                 } catch (Throwable throwable) {
                     LOGGER.error("Failed to load data for block entity {}", s, throwable);
                     return null;
@@ -166,10 +_,14 @@
 
     public void setRemoved() {
         this.remove = true;
+        this.invalidateCapabilities();
+        requestModelDataUpdate();
     }
 
     public void clearRemoved() {
         this.remove = false;
+        // Neo: invalidate capabilities on block entity placement
+        invalidateCapabilities();
     }
 
     public boolean triggerEvent(int p_58889_, int p_58890_) {
@@ -192,15 +_,61 @@
         return this.type;
     }
 
+    @Override
+    public CompoundTag getPersistentData() {
+        if (this.customPersistentData == null)
+            this.customPersistentData = new CompoundTag();
+        return this.customPersistentData;
+    }
+
+    @Override
+    @Nullable
+    public final <T> T setData(net.neoforged.neoforge.attachment.AttachmentType<T> type, T data) {
+        setChanged();
+        return super.setData(type, data);
+    }
+
+    @Override
+    @Nullable
+    public final <T> T removeData(net.neoforged.neoforge.attachment.AttachmentType<T> type) {
+        setChanged();
+        return super.removeData(type);
+    }
+
     @Deprecated
     public void setBlockState(BlockState p_155251_) {
         this.blockState = p_155251_;
     }
 
-    public void applyComponents(DataComponentMap p_330364_) {
-    }
-
-    public void collectComponents(DataComponentMap.Builder p_330786_) {
+    protected void applyImplicitComponents(BlockEntity.DataComponentInput p_338718_) {
+    }
+
+    public final void applyComponentsFromItemStack(ItemStack p_338428_) {
+        this.applyComponents(p_338428_.getPrototype(), p_338428_.getComponentsPatch());
+    }
+
+    public final void applyComponents(DataComponentMap p_330364_, DataComponentPatch p_338381_) {
+        final Set<DataComponentType<?>> set = new HashSet<>();
+        final DataComponentMap datacomponentmap = PatchedDataComponentMap.fromPatch(p_330364_, p_338381_);
+        this.applyImplicitComponents(new BlockEntity.DataComponentInput() {
+            @Nullable
+            @Override
+            public <T> T get(DataComponentType<T> p_338266_) {
+                set.add(p_338266_);
+                return datacomponentmap.get(p_338266_);
+            }
+
+            @Override
+            public <T> T getOrDefault(DataComponentType<? extends T> p_338358_, T p_338352_) {
+                set.add(p_338358_);
+                return datacomponentmap.getOrDefault(p_338358_, p_338352_);
+            }
+        });
+        DataComponentPatch datacomponentpatch = p_338381_.forget(set::contains);
+        this.components = datacomponentpatch.split().added();
+    }
+
+    protected void collectImplicitComponents(DataComponentMap.Builder p_338210_) {
     }
 
     @Deprecated
@@ -209,7 +_,30 @@
 
     public final DataComponentMap collectComponents() {
         DataComponentMap.Builder datacomponentmap$builder = DataComponentMap.builder();
-        this.collectComponents(datacomponentmap$builder);
+        datacomponentmap$builder.addAll(this.components);
+        this.collectImplicitComponents(datacomponentmap$builder);
         return datacomponentmap$builder.build();
+    }
+
+    public DataComponentMap components() {
+        return this.components;
+    }
+
+    public void setComponents(DataComponentMap p_338529_) {
+        this.components = p_338529_;
+    }
+
+    static class ComponentHelper {
+        public static final Codec<DataComponentMap> COMPONENTS_CODEC = DataComponentMap.CODEC.optionalFieldOf("components", DataComponentMap.EMPTY).codec();
+
+        private ComponentHelper() {
+        }
+    }
+
+    protected interface DataComponentInput {
+        @Nullable
+        <T> T get(DataComponentType<T> p_338658_);
+
+        <T> T getOrDefault(DataComponentType<? extends T> p_338573_, T p_338734_);
     }
 }
