--- a/net/minecraft/world/level/block/entity/CommandBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/CommandBlockEntity.java
@@ -79,12 +_,12 @@
     }
 
     @Override
-    public void load(CompoundTag p_155383_, HolderLookup.Provider p_323771_) {
-        super.load(p_155383_, p_323771_);
-        this.commandBlock.load(p_155383_, p_323771_);
-        this.powered = p_155383_.getBoolean("powered");
-        this.conditionMet = p_155383_.getBoolean("conditionMet");
-        this.setAutomatic(p_155383_.getBoolean("auto"));
+    protected void loadAdditional(CompoundTag p_338502_, HolderLookup.Provider p_338638_) {
+        super.loadAdditional(p_338502_, p_338638_);
+        this.commandBlock.load(p_338502_, p_338638_);
+        this.powered = p_338502_.getBoolean("powered");
+        this.conditionMet = p_338502_.getBoolean("conditionMet");
+        this.setAutomatic(p_338502_.getBoolean("auto"));
     }
 
     @Override
@@ -167,14 +_,15 @@
     }
 
     @Override
-    public void applyComponents(DataComponentMap p_331932_) {
-        this.commandBlock.setCustomName(p_331932_.get(DataComponents.CUSTOM_NAME));
+    protected void applyImplicitComponents(BlockEntity.DataComponentInput p_338181_) {
+        super.applyImplicitComponents(p_338181_);
+        this.commandBlock.setCustomName(p_338181_.get(DataComponents.CUSTOM_NAME));
     }
 
     @Override
-    public void collectComponents(DataComponentMap.Builder p_331910_) {
-        super.collectComponents(p_331910_);
-        p_331910_.set(DataComponents.CUSTOM_NAME, this.commandBlock.getCustomName());
+    protected void collectImplicitComponents(DataComponentMap.Builder p_338580_) {
+        super.collectImplicitComponents(p_338580_);
+        p_338580_.set(DataComponents.CUSTOM_NAME, this.commandBlock.getCustomName());
     }
 
     @Override
