--- a/net/minecraft/world/level/block/entity/CrafterBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/CrafterBlockEntity.java
@@ -78,11 +_,7 @@
     }
 
     public boolean isSlotDisabled(int p_307461_) {
-        if (p_307461_ >= 0 && p_307461_ < 9) {
-            return this.containerData.get(p_307461_) == 1;
-        } else {
-            return false;
-        }
+        return p_307461_ >= 0 && p_307461_ < 9 ? this.containerData.get(p_307461_) == 1 : false;
     }
 
     @Override
@@ -94,16 +_,14 @@
             int i = itemstack.getCount();
             if (i >= itemstack.getMaxStackSize()) {
                 return false;
-            } else if (itemstack.isEmpty()) {
-                return true;
             } else {
-                return !this.smallerStackExist(i, itemstack, p_307543_);
+                return itemstack.isEmpty() ? true : !this.smallerStackExist(i, itemstack, p_307543_);
             }
         }
     }
 
     private boolean smallerStackExist(int p_307396_, ItemStack p_307520_, int p_307348_) {
-        for(int i = p_307348_ + 1; i < 9; ++i) {
+        for (int i = p_307348_ + 1; i < 9; i++) {
             if (!this.isSlotDisabled(i)) {
                 ItemStack itemstack = this.getItem(i);
                 if (itemstack.isEmpty() || itemstack.getCount() < p_307396_ && ItemStack.isSameItemSameComponents(itemstack, p_307520_)) {
@@ -116,8 +_,8 @@
     }
 
     @Override
-    public void load(CompoundTag p_307457_, HolderLookup.Provider p_324538_) {
-        super.load(p_307457_, p_324538_);
+    protected void loadAdditional(CompoundTag p_307457_, HolderLookup.Provider p_324538_) {
+        super.loadAdditional(p_307457_, p_324538_);
         this.craftingTicksRemaining = p_307457_.getInt("crafting_ticks_remaining");
         this.items = NonNullList.withSize(this.getContainerSize(), ItemStack.EMPTY);
         if (!this.tryLoadLootTable(p_307457_)) {
@@ -126,11 +_,11 @@
 
         int[] aint = p_307457_.getIntArray("disabled_slots");
 
-        for(int i = 0; i < 9; ++i) {
+        for (int i = 0; i < 9; i++) {
             this.containerData.set(i, 0);
         }
 
-        for(int j : aint) {
+        for (int j : aint) {
             if (this.slotCanBeDisabled(j)) {
                 this.containerData.set(j, 1);
             }
@@ -158,7 +_,7 @@
 
     @Override
     public boolean isEmpty() {
-        for(ItemStack itemstack : this.items) {
+        for (ItemStack itemstack : this.items) {
             if (!itemstack.isEmpty()) {
                 return false;
             }
@@ -208,7 +_,7 @@
 
     @Override
     public void fillStackedContents(StackedContents p_307380_) {
-        for(ItemStack itemstack : this.items) {
+        for (ItemStack itemstack : this.items) {
             p_307380_.accountSimpleStack(itemstack);
         }
     }
@@ -216,7 +_,7 @@
     private void addDisabledSlots(CompoundTag p_307523_) {
         IntList intlist = new IntArrayList();
 
-        for(int i = 0; i < 9; ++i) {
+        for (int i = 0; i < 9; i++) {
             if (this.isSlotDisabled(i)) {
                 intlist.add(i);
             }
@@ -255,10 +_,10 @@
     public int getRedstoneSignal() {
         int i = 0;
 
-        for(int j = 0; j < this.getContainerSize(); ++j) {
+        for (int j = 0; j < this.getContainerSize(); j++) {
             ItemStack itemstack = this.getItem(j);
             if (!itemstack.isEmpty() || this.isSlotDisabled(j)) {
-                ++i;
+                i++;
             }
         }
 
