--- a/net/minecraft/world/level/block/entity/DecoratedPotBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/DecoratedPotBlockEntity.java
@@ -47,12 +_,12 @@
     }
 
     @Override
-    public void load(CompoundTag p_272924_, HolderLookup.Provider p_324508_) {
-        super.load(p_272924_, p_324508_);
-        this.decorations = PotDecorations.load(p_272924_);
-        if (!this.tryLoadLootTable(p_272924_)) {
-            if (p_272924_.contains("item", 10)) {
-                this.item = ItemStack.parse(p_324508_, p_272924_.getCompound("item")).orElse(ItemStack.EMPTY);
+    protected void loadAdditional(CompoundTag p_338486_, HolderLookup.Provider p_338310_) {
+        super.loadAdditional(p_338486_, p_338310_);
+        this.decorations = PotDecorations.load(p_338486_);
+        if (!this.tryLoadLootTable(p_338486_)) {
+            if (p_338486_.contains("item", 10)) {
+                this.item = ItemStack.parse(p_338310_, p_338486_.getCompound("item")).orElse(ItemStack.EMPTY);
             } else {
                 this.item = ItemStack.EMPTY;
             }
@@ -65,7 +_,7 @@
 
     @Override
     public CompoundTag getUpdateTag(HolderLookup.Provider p_324359_) {
-        return this.saveWithoutMetadata(p_324359_);
+        return this.saveCustomOnly(p_324359_);
     }
 
     public Direction getDirection() {
@@ -77,7 +_,7 @@
     }
 
     public void setFromItem(ItemStack p_273109_) {
-        this.applyComponents(p_273109_.getComponents());
+        this.applyComponentsFromItemStack(p_273109_);
     }
 
     public ItemStack getPotAsItem() {
@@ -114,15 +_,17 @@
     }
 
     @Override
-    public void collectComponents(DataComponentMap.Builder p_331987_) {
-        p_331987_.set(DataComponents.POT_DECORATIONS, this.decorations);
-        p_331987_.set(DataComponents.CONTAINER, ItemContainerContents.copyOf(List.of(this.item)));
+    protected void collectImplicitComponents(DataComponentMap.Builder p_338608_) {
+        super.collectImplicitComponents(p_338608_);
+        p_338608_.set(DataComponents.POT_DECORATIONS, this.decorations);
+        p_338608_.set(DataComponents.CONTAINER, ItemContainerContents.copyOf(List.of(this.item)));
     }
 
     @Override
-    public void applyComponents(DataComponentMap p_331563_) {
-        this.decorations = p_331563_.getOrDefault(DataComponents.POT_DECORATIONS, PotDecorations.EMPTY);
-        this.item = p_331563_.getOrDefault(DataComponents.CONTAINER, ItemContainerContents.EMPTY).copyOne();
+    protected void applyImplicitComponents(BlockEntity.DataComponentInput p_338521_) {
+        super.applyImplicitComponents(p_338521_);
+        this.decorations = p_338521_.getOrDefault(DataComponents.POT_DECORATIONS, PotDecorations.EMPTY);
+        this.item = p_338521_.getOrDefault(DataComponents.CONTAINER, ItemContainerContents.EMPTY).copyOne();
     }
 
     @Override
