--- a/net/minecraft/world/level/block/entity/DispenserBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/DispenserBlockEntity.java
@@ -35,7 +_,7 @@
         int i = -1;
         int j = 1;
 
-        for(int k = 0; k < this.items.size(); ++k) {
+        for (int k = 0; k < this.items.size(); k++) {
             if (!this.items.get(k).isEmpty() && p_222762_.nextInt(j++) == 0) {
                 i = k;
             }
@@ -45,7 +_,7 @@
     }
 
     public int addItem(ItemStack p_59238_) {
-        for(int i = 0; i < this.items.size(); ++i) {
+        for (int i = 0; i < this.items.size(); i++) {
             if (this.items.get(i).isEmpty()) {
                 this.setItem(i, p_59238_);
                 return i;
@@ -61,8 +_,8 @@
     }
 
     @Override
-    public void load(CompoundTag p_155496_, HolderLookup.Provider p_323868_) {
-        super.load(p_155496_, p_323868_);
+    protected void loadAdditional(CompoundTag p_155496_, HolderLookup.Provider p_323868_) {
+        super.loadAdditional(p_155496_, p_323868_);
         this.items = NonNullList.withSize(this.getContainerSize(), ItemStack.EMPTY);
         if (!this.tryLoadLootTable(p_155496_)) {
             ContainerHelper.loadAllItems(p_155496_, this.items, p_323868_);
