--- a/net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
@@ -92,9 +_,9 @@
     }
 
     @Override
-    public void applyComponents(DataComponentMap p_331227_) {
-        super.applyComponents(p_331227_);
-        SeededContainerLoot seededcontainerloot = p_331227_.get(DataComponents.CONTAINER_LOOT);
+    protected void applyImplicitComponents(BlockEntity.DataComponentInput p_338832_) {
+        super.applyImplicitComponents(p_338832_);
+        SeededContainerLoot seededcontainerloot = p_338832_.get(DataComponents.CONTAINER_LOOT);
         if (seededcontainerloot != null) {
             this.lootTable = seededcontainerloot.lootTable();
             this.lootTableSeed = seededcontainerloot.seed();
@@ -102,10 +_,10 @@
     }
 
     @Override
-    public void collectComponents(DataComponentMap.Builder p_330445_) {
-        super.collectComponents(p_330445_);
+    protected void collectImplicitComponents(DataComponentMap.Builder p_338607_) {
+        super.collectImplicitComponents(p_338607_);
         if (this.lootTable != null) {
-            p_330445_.set(DataComponents.CONTAINER_LOOT, new SeededContainerLoot(this.lootTable, this.lootTableSeed));
+            p_338607_.set(DataComponents.CONTAINER_LOOT, new SeededContainerLoot(this.lootTable, this.lootTableSeed));
         }
     }
 
