--- a/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java
@@ -77,8 +_,8 @@
     }
 
     @Override
-    public void load(CompoundTag p_222868_, HolderLookup.Provider p_324354_) {
-        super.load(p_222868_, p_324354_);
+    protected void loadAdditional(CompoundTag p_222868_, HolderLookup.Provider p_324354_) {
+        super.loadAdditional(p_222868_, p_324354_);
         if (p_222868_.contains("warning_level", 99)) {
             this.warningLevel = p_222868_.getInt("warning_level");
         }
@@ -103,27 +_,27 @@
 
     @Nullable
     public static ServerPlayer tryGetPlayer(@Nullable Entity p_222862_) {
-        if (p_222862_ instanceof ServerPlayer serverplayer1) {
-            return serverplayer1;
+        if (p_222862_ instanceof ServerPlayer) {
+            return (ServerPlayer)p_222862_;
         } else {
             if (p_222862_ != null) {
                 LivingEntity $$6 = p_222862_.getControllingPassenger();
-                if ($$6 instanceof ServerPlayer serverplayer) {
-                    return serverplayer;
+                if ($$6 instanceof ServerPlayer) {
+                    return (ServerPlayer)$$6;
                 }
             }
 
             if (p_222862_ instanceof Projectile projectile) {
                 Entity entity = projectile.getOwner();
-                if (entity instanceof ServerPlayer serverplayer3) {
-                    return serverplayer3;
+                if (entity instanceof ServerPlayer) {
+                    return (ServerPlayer)entity;
                 }
             }
 
             if (p_222862_ instanceof ItemEntity itementity) {
                 Entity entity1 = itementity.getOwner();
-                if (entity1 instanceof ServerPlayer serverplayer2) {
-                    return serverplayer2;
+                if (entity1 instanceof ServerPlayer) {
+                    return (ServerPlayer)entity1;
                 }
             }
 
