--- a/net/minecraft/world/level/block/entity/SignText.java
+++ b/net/minecraft/world/level/block/entity/SignText.java
@@ -22,17 +_,18 @@
     private static final Codec<Component[]> LINES_CODEC = ComponentSerialization.FLAT_CODEC
         .listOf()
         .comapFlatMap(
-            p_277790_ -> Util.fixedSize(p_277790_, 4).map(p_277881_ -> new Component[]{p_277881_.get(0), p_277881_.get(1), p_277881_.get(2), p_277881_.get(3)}),
+            p_337999_ -> Util.fixedSize((List<Component>)p_337999_, 4)
+                    .map(p_277881_ -> new Component[]{p_277881_.get(0), p_277881_.get(1), p_277881_.get(2), p_277881_.get(3)}),
             p_277460_ -> List.of(p_277460_[0], p_277460_[1], p_277460_[2], p_277460_[3])
         );
     public static final Codec<SignText> DIRECT_CODEC = RecordCodecBuilder.create(
-        p_277675_ -> p_277675_.group(
+        p_338000_ -> p_338000_.group(
                     LINES_CODEC.fieldOf("messages").forGetter(p_277822_ -> p_277822_.messages),
-                    LINES_CODEC.optionalFieldOf("filtered_messages").forGetter(SignText::filteredMessages),
+                    LINES_CODEC.lenientOptionalFieldOf("filtered_messages").forGetter(SignText::filteredMessages),
                     DyeColor.CODEC.fieldOf("color").orElse(DyeColor.BLACK).forGetter(p_277343_ -> p_277343_.color),
                     Codec.BOOL.fieldOf("has_glowing_text").orElse(false).forGetter(p_277555_ -> p_277555_.hasGlowingText)
                 )
-                .apply(p_277675_, SignText::load)
+                .apply(p_338000_, SignText::load)
     );
     public static final int LINES = 4;
     private final Component[] messages;
@@ -107,7 +_,7 @@
             this.renderMessagedFiltered = p_277336_;
             this.renderMessages = new FormattedCharSequence[4];
 
-            for(int i = 0; i < 4; ++i) {
+            for (int i = 0; i < 4; i++) {
                 this.renderMessages[i] = p_277538_.apply(this.getMessage(i, p_277336_));
             }
         }
@@ -116,7 +_,7 @@
     }
 
     private Optional<Component[]> filteredMessages() {
-        for(int i = 0; i < 4; ++i) {
+        for (int i = 0; i < 4; i++) {
             if (!this.filteredMessages[i].equals(this.messages[i])) {
                 return Optional.of(this.filteredMessages);
             }
@@ -126,7 +_,7 @@
     }
 
     public boolean hasAnyClickCommands(Player p_277865_) {
-        for(Component component : this.getMessages(p_277865_.isTextFilteringEnabled())) {
+        for (Component component : this.getMessages(p_277865_.isTextFilteringEnabled())) {
             Style style = component.getStyle();
             ClickEvent clickevent = style.getClickEvent();
             if (clickevent != null && clickevent.getAction() == ClickEvent.Action.RUN_COMMAND) {
