--- a/net/minecraft/world/level/block/entity/SignBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SignBlockEntity.java
@@ -26,7 +_,6 @@
 import net.minecraft.util.Mth;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.Level;
-import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.SignBlock;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.Vec2;
@@ -56,8 +_,7 @@
     }
 
     public boolean isFacingFrontText(Player p_277382_) {
-        Block $$2 = this.getBlockState().getBlock();
-        if ($$2 instanceof SignBlock signblock) {
+        if (this.getBlockState().getBlock() instanceof SignBlock signblock) {
             Vec3 vec3 = signblock.getSignHitboxCenterPosition(this.getBlockState());
             double d0 = p_277382_.getX() - ((double)this.getBlockPos().getX() + vec3.x);
             double d1 = p_277382_.getZ() - ((double)this.getBlockPos().getZ() + vec3.z);
@@ -105,8 +_,8 @@
     }
 
     @Override
-    public void load(CompoundTag p_155716_, HolderLookup.Provider p_324351_) {
-        super.load(p_155716_, p_324351_);
+    protected void loadAdditional(CompoundTag p_155716_, HolderLookup.Provider p_324351_) {
+        super.loadAdditional(p_155716_, p_324351_);
         DynamicOps<Tag> dynamicops = p_324351_.createSerializationContext(NbtOps.INSTANCE);
         if (p_155716_.contains("front_text")) {
             SignText.DIRECT_CODEC
@@ -126,7 +_,7 @@
     }
 
     private SignText loadLines(SignText p_278305_) {
-        for(int i = 0; i < 4; ++i) {
+        for (int i = 0; i < 4; i++) {
             Component component = this.loadLine(p_278305_.getMessage(i, false));
             Component component1 = this.loadLine(p_278305_.getMessage(i, true));
             p_278305_ = p_278305_.setMessage(i, component, component1);
@@ -136,8 +_,7 @@
     }
 
     private Component loadLine(Component p_278307_) {
-        Level level = this.level;
-        if (level instanceof ServerLevel serverlevel) {
+        if (this.level instanceof ServerLevel serverlevel) {
             try {
                 return ComponentUtils.updateForEntity(createCommandSourceStack(null, serverlevel, this.worldPosition), p_278307_, null, 0);
             } catch (CommandSyntaxException commandsyntaxexception) {
@@ -163,7 +_,7 @@
     }
 
     private SignText setMessages(Player p_277396_, List<FilteredText> p_277744_, SignText p_277359_) {
-        for(int i = 0; i < p_277744_.size(); ++i) {
+        for (int i = 0; i < p_277744_.size(); i++) {
             FilteredText filteredtext = p_277744_.get(i);
             Style style = p_277359_.getMessage(i, p_277396_.isTextFilteringEnabled()).getStyle();
             if (p_277396_.isTextFilteringEnabled()) {
@@ -209,7 +_,7 @@
     public boolean executeClickCommandsIfPresent(Player p_279304_, Level p_279201_, BlockPos p_278282_, boolean p_278254_) {
         boolean flag = false;
 
-        for(Component component : this.getText(p_278254_).getMessages(p_279304_.isTextFilteringEnabled())) {
+        for (Component component : this.getText(p_278254_).getMessages(p_279304_.isTextFilteringEnabled())) {
             Style style = component.getStyle();
             ClickEvent clickevent = style.getClickEvent();
             if (clickevent != null && clickevent.getAction() == ClickEvent.Action.RUN_COMMAND) {
@@ -235,7 +_,7 @@
 
     @Override
     public CompoundTag getUpdateTag(HolderLookup.Provider p_324439_) {
-        return this.saveWithoutMetadata(p_324439_);
+        return this.saveCustomOnly(p_324439_);
     }
 
     @Override
@@ -273,8 +_,7 @@
 
     public boolean playerIsTooFarAwayToEdit(UUID p_277978_) {
         Player player = this.level.getPlayerByUUID(p_277978_);
-        return player == null
-            || player.distanceToSqr((double)this.getBlockPos().getX(), (double)this.getBlockPos().getY(), (double)this.getBlockPos().getZ()) > 64.0;
+        return player == null || !player.canInteractWithBlock(this.getBlockPos(), 4.0);
     }
 
     public static void tick(Level p_277662_, BlockPos p_278050_, BlockState p_277927_, SignBlockEntity p_277928_) {
