--- a/net/minecraft/world/level/block/entity/SkullBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SkullBlockEntity.java
@@ -39,7 +_,9 @@
     @Nullable
     private static Executor mainThreadExecutor;
     @Nullable
-    private static LoadingCache<String, CompletableFuture<Optional<GameProfile>>> profileCache;
+    private static LoadingCache<String, CompletableFuture<Optional<GameProfile>>> profileCacheByName;
+    @Nullable
+    private static LoadingCache<UUID, CompletableFuture<Optional<GameProfile>>> profileCacheById;
     public static final Executor CHECKED_MAIN_THREAD_EXECUTOR = p_294078_ -> {
         Executor executor = mainThreadExecutor;
         if (executor != null) {
@@ -61,43 +_,60 @@
 
     public static void setup(final Services p_222886_, Executor p_222887_) {
         mainThreadExecutor = p_222887_;
-        final BooleanSupplier booleansupplier = () -> profileCache == null;
-        profileCache = CacheBuilder.newBuilder()
-            .expireAfterAccess(Duration.ofMinutes(10L))
-            .maximumSize(256L)
-            .build(
-                new CacheLoader<String, CompletableFuture<Optional<GameProfile>>>() {
-                    public CompletableFuture<Optional<GameProfile>> load(String p_304652_) {
-                        return booleansupplier.getAsBoolean()
-                            ? CompletableFuture.completedFuture(Optional.empty())
-                            : SkullBlockEntity.loadProfile(p_304652_, p_222886_, booleansupplier);
-                    }
+        final BooleanSupplier booleansupplier = () -> profileCacheById == null;
+        profileCacheByName = CacheBuilder.newBuilder()
+            .expireAfterAccess(Duration.ofMinutes(10L))
+            .maximumSize(256L)
+            .build(new CacheLoader<String, CompletableFuture<Optional<GameProfile>>>() {
+                public CompletableFuture<Optional<GameProfile>> load(String p_304652_) {
+                    return SkullBlockEntity.fetchProfileByName(p_304652_, p_222886_);
+                }
+            });
+        profileCacheById = CacheBuilder.newBuilder()
+            .expireAfterAccess(Duration.ofMinutes(10L))
+            .maximumSize(256L)
+            .build(new CacheLoader<UUID, CompletableFuture<Optional<GameProfile>>>() {
+                public CompletableFuture<Optional<GameProfile>> load(UUID p_339657_) {
+                    return SkullBlockEntity.fetchProfileById(p_339657_, p_222886_, booleansupplier);
+                }
+            });
+    }
+
+    static CompletableFuture<Optional<GameProfile>> fetchProfileByName(String p_339683_, Services p_339592_) {
+        return p_339592_.profileCache()
+            .getAsync(p_339683_)
+            .thenCompose(
+                p_339545_ -> {
+                    LoadingCache<UUID, CompletableFuture<Optional<GameProfile>>> loadingcache = profileCacheById;
+                    return loadingcache != null && !p_339545_.isEmpty()
+                        ? loadingcache.getUnchecked(p_339545_.get().getId()).thenApply(p_339543_ -> p_339543_.or(() -> p_339545_))
+                        : CompletableFuture.completedFuture(Optional.empty());
                 }
             );
     }
 
-    public static void clear() {
-        mainThreadExecutor = null;
-        profileCache = null;
-    }
-
-    static CompletableFuture<Optional<GameProfile>> loadProfile(String p_304754_, Services p_304699_, BooleanSupplier p_304484_) {
-        return p_304699_.profileCache().getAsync(p_304754_).thenApplyAsync(p_304381_ -> {
-            if (p_304381_.isPresent() && !p_304484_.getAsBoolean()) {
-                UUID uuid = p_304381_.get().getId();
-                ProfileResult profileresult = p_304699_.sessionService().fetchProfile(uuid, true);
-                return profileresult != null ? Optional.ofNullable(profileresult.profile()) : p_304381_;
-            } else {
+    static CompletableFuture<Optional<GameProfile>> fetchProfileById(UUID p_339663_, Services p_339676_, BooleanSupplier p_339631_) {
+        return CompletableFuture.supplyAsync(() -> {
+            if (p_339631_.getAsBoolean()) {
                 return Optional.empty();
+            } else {
+                ProfileResult profileresult = p_339676_.sessionService().fetchProfile(p_339663_, true);
+                return Optional.ofNullable(profileresult).map(ProfileResult::profile);
             }
         }, Util.backgroundExecutor());
     }
 
+    public static void clear() {
+        mainThreadExecutor = null;
+        profileCacheByName = null;
+        profileCacheById = null;
+    }
+
     @Override
     protected void saveAdditional(CompoundTag p_187518_, HolderLookup.Provider p_324418_) {
         super.saveAdditional(p_187518_, p_324418_);
         if (this.owner != null) {
-            p_187518_.put("profile", Util.getOrThrow(ResolvableProfile.CODEC.encodeStart(NbtOps.INSTANCE, this.owner), IllegalStateException::new));
+            p_187518_.put("profile", ResolvableProfile.CODEC.encodeStart(NbtOps.INSTANCE, this.owner).getOrThrow());
         }
 
         if (this.noteBlockSound != null) {
@@ -110,8 +_,8 @@
     }
 
     @Override
-    public void load(CompoundTag p_155745_, HolderLookup.Provider p_323876_) {
-        super.load(p_155745_, p_323876_);
+    protected void loadAdditional(CompoundTag p_155745_, HolderLookup.Provider p_323876_) {
+        super.loadAdditional(p_155745_, p_323876_);
         if (p_155745_.contains("profile")) {
             ResolvableProfile.CODEC
                 .parse(NbtOps.INSTANCE, p_155745_.get("profile"))
@@ -133,7 +_,7 @@
     public static void animation(Level p_261710_, BlockPos p_262153_, BlockState p_262021_, SkullBlockEntity p_261594_) {
         if (p_262021_.hasProperty(SkullBlock.POWERED) && p_262021_.getValue(SkullBlock.POWERED)) {
             p_261594_.isAnimating = true;
-            ++p_261594_.animationTickCount;
+            p_261594_.animationTickCount++;
         } else {
             p_261594_.isAnimating = false;
         }
@@ -159,11 +_,11 @@
 
     @Override
     public CompoundTag getUpdateTag(HolderLookup.Provider p_323711_) {
-        return this.saveWithoutMetadata(p_323711_);
+        return this.saveCustomOnly(p_323711_);
     }
 
     public void setOwner(@Nullable ResolvableProfile p_332738_) {
-        synchronized(this) {
+        synchronized (this) {
             this.owner = p_332738_;
         }
 
@@ -182,24 +_,31 @@
     }
 
     public static CompletableFuture<Optional<GameProfile>> fetchGameProfile(String p_295932_) {
-        LoadingCache<String, CompletableFuture<Optional<GameProfile>>> loadingcache = profileCache;
+        LoadingCache<String, CompletableFuture<Optional<GameProfile>>> loadingcache = profileCacheByName;
         return loadingcache != null && StringUtil.isValidPlayerName(p_295932_)
             ? loadingcache.getUnchecked(p_295932_)
             : CompletableFuture.completedFuture(Optional.empty());
     }
 
-    @Override
-    public void applyComponents(DataComponentMap p_330564_) {
-        this.setOwner(p_330564_.get(DataComponents.PROFILE));
-        this.noteBlockSound = p_330564_.get(DataComponents.NOTE_BLOCK_SOUND);
-        this.customName = p_330564_.get(DataComponents.CUSTOM_NAME);
-    }
-
-    @Override
-    public void collectComponents(DataComponentMap.Builder p_330651_) {
-        p_330651_.set(DataComponents.PROFILE, this.owner);
-        p_330651_.set(DataComponents.NOTE_BLOCK_SOUND, this.noteBlockSound);
-        p_330651_.set(DataComponents.CUSTOM_NAME, this.customName);
+    public static CompletableFuture<Optional<GameProfile>> fetchGameProfile(UUID p_339604_) {
+        LoadingCache<UUID, CompletableFuture<Optional<GameProfile>>> loadingcache = profileCacheById;
+        return loadingcache != null ? loadingcache.getUnchecked(p_339604_) : CompletableFuture.completedFuture(Optional.empty());
+    }
+
+    @Override
+    protected void applyImplicitComponents(BlockEntity.DataComponentInput p_338654_) {
+        super.applyImplicitComponents(p_338654_);
+        this.setOwner(p_338654_.get(DataComponents.PROFILE));
+        this.noteBlockSound = p_338654_.get(DataComponents.NOTE_BLOCK_SOUND);
+        this.customName = p_338654_.get(DataComponents.CUSTOM_NAME);
+    }
+
+    @Override
+    protected void collectImplicitComponents(DataComponentMap.Builder p_338880_) {
+        super.collectImplicitComponents(p_338880_);
+        p_338880_.set(DataComponents.PROFILE, this.owner);
+        p_338880_.set(DataComponents.NOTE_BLOCK_SOUND, this.noteBlockSound);
+        p_338880_.set(DataComponents.CUSTOM_NAME, this.customName);
     }
 
     @Override
