--- a/net/minecraft/world/level/block/entity/TrialSpawnerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/TrialSpawnerBlockEntity.java
@@ -1,7 +_,7 @@
 package net.minecraft.world.level.block.entity;
 
 import com.mojang.logging.LogUtils;
-import com.mojang.serialization.DataResult.PartialResult;
+import com.mojang.serialization.DataResult.Error;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.HolderLookup;
 import net.minecraft.nbt.CompoundTag;
@@ -32,9 +_,14 @@
     }
 
     @Override
-    public void load(CompoundTag p_311954_, HolderLookup.Provider p_324032_) {
-        super.load(p_311954_, p_324032_);
-        this.trialSpawner.codec().parse(NbtOps.INSTANCE, p_311954_).resultOrPartial(LOGGER::error).ifPresent(p_311911_ -> this.trialSpawner = p_311911_);
+    protected void loadAdditional(CompoundTag p_338752_, HolderLookup.Provider p_338872_) {
+        super.loadAdditional(p_338752_, p_338872_);
+        if (p_338752_.contains("normal_config")) {
+            CompoundTag compoundtag = p_338752_.getCompound("normal_config").copy();
+            p_338752_.put("ominous_config", compoundtag.merge(p_338752_.getCompound("ominous_config")));
+        }
+
+        this.trialSpawner.codec().parse(NbtOps.INSTANCE, p_338752_).resultOrPartial(LOGGER::error).ifPresent(p_311911_ -> this.trialSpawner = p_311911_);
         if (this.level != null) {
             this.markUpdated();
         }
@@ -46,9 +_,8 @@
         this.trialSpawner
             .codec()
             .encodeStart(NbtOps.INSTANCE, this.trialSpawner)
-            .get()
-            .ifLeft(p_312175_ -> p_311806_.merge((CompoundTag)p_312175_))
-            .ifRight(p_312606_ -> LOGGER.warn("Failed to encode TrialSpawner {}", p_312606_.message()));
+            .ifSuccess(p_312175_ -> p_311806_.merge((CompoundTag)p_312175_))
+            .ifError(p_338001_ -> LOGGER.warn("Failed to encode TrialSpawner {}", p_338001_.message()));
     }
 
     public ClientboundBlockEntityDataPacket getUpdatePacket() {
