--- a/net/minecraft/client/data/models/ItemModelOutput.java
+++ b/net/minecraft/client/data/models/ItemModelOutput.java
@@ -13,4 +_,10 @@
     void accept(ResourceLocation p_422487_, ItemModel.Unbaked p_421069_);
 
     void copy(Item p_387316_, Item p_387995_);
+
+    /**
+     * Neo: Pulled up from {@link ModelProvider.ItemInfoCollector} to give modders full control over the {@link net.minecraft.client.renderer.item.ClientItem} instead of just the {@link ItemModel.Unbaked} in {@link #accept(Item, ItemModel.Unbaked)
+     */
+    default void register(Item item, net.minecraft.client.renderer.item.ClientItem clientItem) {
+    }
 }
