--- a/net/minecraft/world/entity/decoration/PaintingVariants.java
+++ b/net/minecraft/world/entity/decoration/PaintingVariants.java
@@ -36,13 +_,6 @@
     public static final ResourceKey<PaintingVariant> WIND = create("wind");
     public static final ResourceKey<PaintingVariant> WATER = create("water");
     public static final ResourceKey<PaintingVariant> FIRE = create("fire");
-    public static final ResourceKey<PaintingVariant> POISONOUS_POTATO = create("poisonous_potato");
-    public static final ResourceKey<PaintingVariant> MR_POTATO = create("mr_potato");
-    public static final ResourceKey<PaintingVariant> ABSTRACTATO = create("abstractato");
-    public static final ResourceKey<PaintingVariant> BURNING_POTATO = create("burning_potato");
-    public static final ResourceKey<PaintingVariant> UBIQUITATO = create("ubiquitato");
-    public static final ResourceKey<PaintingVariant> CECI = create("ceci");
-    public static final ResourceKey<PaintingVariant> POTATOE = create("potatoe");
 
     public static PaintingVariant bootstrap(Registry<PaintingVariant> p_218943_) {
         Registry.register(p_218943_, KEBAB, new PaintingVariant(16, 16));
@@ -74,13 +_,6 @@
         Registry.register(p_218943_, WIND, new PaintingVariant(32, 32));
         Registry.register(p_218943_, WATER, new PaintingVariant(32, 32));
         Registry.register(p_218943_, FIRE, new PaintingVariant(32, 32));
-        Registry.register(p_218943_, POISONOUS_POTATO, new PaintingVariant(128, 128));
-        Registry.register(p_218943_, MR_POTATO, new PaintingVariant(32, 48));
-        Registry.register(p_218943_, ABSTRACTATO, new PaintingVariant(32, 32));
-        Registry.register(p_218943_, BURNING_POTATO, new PaintingVariant(64, 64));
-        Registry.register(p_218943_, UBIQUITATO, new PaintingVariant(48, 48));
-        Registry.register(p_218943_, CECI, new PaintingVariant(128, 64));
-        Registry.register(p_218943_, POTATOE, new PaintingVariant(64, 32));
         return Registry.register(p_218943_, DONKEY_KONG, new PaintingVariant(64, 48));
     }
 
