--- a/net/minecraft/util/debugchart/DebugSampleSubscriptionTracker.java
+++ b/net/minecraft/util/debugchart/DebugSampleSubscriptionTracker.java
@@ -22,7 +_,7 @@
         this.playerList = p_323956_;
         this.subscriptions = new EnumMap<>(RemoteDebugSampleType.class);
 
-        for(RemoteDebugSampleType remotedebugsampletype : RemoteDebugSampleType.values()) {
+        for (RemoteDebugSampleType remotedebugsampletype : RemoteDebugSampleType.values()) {
             this.subscriptions.put(remotedebugsampletype, Maps.newHashMap());
         }
     }
@@ -32,7 +_,7 @@
     }
 
     public void broadcast(ClientboundDebugSamplePacket p_324398_) {
-        for(ServerPlayer serverplayer : this.subscriptions.get(p_324398_.debugSampleType()).keySet()) {
+        for (ServerPlayer serverplayer : this.subscriptions.get(p_324398_.debugSampleType()).keySet()) {
             serverplayer.connection.send(p_324398_);
         }
     }
@@ -50,7 +_,7 @@
     }
 
     private void handleSubscriptions(long p_324350_, int p_323574_) {
-        for(DebugSampleSubscriptionTracker.SubscriptionRequest debugsamplesubscriptiontracker$subscriptionrequest : this.subscriptionRequestQueue) {
+        for (DebugSampleSubscriptionTracker.SubscriptionRequest debugsamplesubscriptiontracker$subscriptionrequest : this.subscriptionRequestQueue) {
             this.subscriptions
                 .get(debugsamplesubscriptiontracker$subscriptionrequest.sampleType())
                 .put(
@@ -60,7 +_,7 @@
     }
 
     private void handleUnsubscriptions(long p_323971_, int p_324253_) {
-        for(Map<ServerPlayer, DebugSampleSubscriptionTracker.SubscriptionStartedAt> map : this.subscriptions.values()) {
+        for (Map<ServerPlayer, DebugSampleSubscriptionTracker.SubscriptionStartedAt> map : this.subscriptions.values()) {
             map.entrySet()
                 .removeIf(
                     p_323887_ -> {
