--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -87,7 +_,7 @@
     public void close() throws IOException {
         ExceptionCollector<IOException> exceptioncollector = new ExceptionCollector<>();
 
-        for(RegionFile regionfile : this.regionCache.values()) {
+        for (RegionFile regionfile : this.regionCache.values()) {
             try {
                 regionfile.close();
             } catch (IOException ioexception) {
@@ -99,7 +_,7 @@
     }
 
     public void flush() throws IOException {
-        for(RegionFile regionfile : this.regionCache.values()) {
+        for (RegionFile regionfile : this.regionCache.values()) {
             regionfile.flush();
         }
     }
