--- a/net/minecraft/world/level/chunk/storage/RegionFile.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFile.java
@@ -77,7 +_,7 @@
 
                 long j = Files.size(p_63633_);
 
-                for(int k = 0; k < 1024; ++k) {
+                for (int k = 0; k < 1024; k++) {
                     int l = this.offsets.get(k);
                     if (l != 0) {
                         int i1 = getSectorNumber(l);
