--- a/net/minecraft/world/level/chunk/storage/SerializableChunkData.java
+++ b/net/minecraft/world/level/chunk/storage/SerializableChunkData.java
@@ -92,7 +_,9 @@
     List<SerializableChunkData.SectionData> sectionData,
     List<CompoundTag> entities,
     List<CompoundTag> blockEntities,
-    CompoundTag structureData
+    CompoundTag structureData,
+    @Nullable CompoundTag attachmentData,
+    @Nullable ListTag auxLightData
 ) {
     private static final Codec<List<SavedTick<Block>>> BLOCK_TICKS_CODEC = SavedTick.codec(BuiltInRegistries.BLOCK.byNameCodec()).listOf();
     private static final Codec<List<SavedTick<Fluid>>> FLUID_TICKS_CODEC = SavedTick.codec(BuiltInRegistries.FLUID.byNameCodec()).listOf();
@@ -108,6 +_,33 @@
     public static final String BLOCK_LIGHT_TAG = "BlockLight";
     public static final String SKY_LIGHT_TAG = "SkyLight";
 
+    /**
+     * @deprecated Neo: use constructor with additional data instead
+     */
+    @Deprecated
+    SerializableChunkData(
+            PalettedContainerFactory containerFactory,
+            ChunkPos chunkPos,
+            int minSectionY,
+            long lastUpdateTime,
+            long inhabitedTime,
+            ChunkStatus chunkStatus,
+            BlendingData.@Nullable Packed blendingData,
+            @Nullable BelowZeroRetrogen belowZeroRetrogen,
+            UpgradeData upgradeData,
+            long @Nullable [] carvingMask,
+            Map<Heightmap.Types, long[]> heightmaps,
+            ChunkAccess.PackedTicks packedTicks,
+            ShortList[] postProcessingSections,
+            boolean lightCorrect,
+            List<SerializableChunkData.SectionData> sectionData,
+            List<CompoundTag> entities,
+            List<CompoundTag> blockEntities,
+            CompoundTag structureData
+    ) {
+        this(containerFactory, chunkPos, minSectionY, lastUpdateTime, inhabitedTime, chunkStatus, blendingData, belowZeroRetrogen, upgradeData, carvingMask, heightmaps, packedTicks, postProcessingSections, lightCorrect, sectionData, entities, blockEntities, structureData, null, null);
+    }
+
     public static SerializableChunkData parse(LevelHeightAccessor levelHeight, PalettedContainerFactory containerFactory, CompoundTag chunkData) {
         if (chunkData.getString("Status").isEmpty()) {
             return null;
@@ -186,6 +_,9 @@
                 }
             }
 
+            CompoundTag attachmentData = chunkData.getCompound(net.neoforged.neoforge.attachment.AttachmentHolder.ATTACHMENTS_NBT_KEY).orElse(null);
+            ListTag auxLightData = chunkData.getList(net.neoforged.neoforge.common.world.LevelChunkAuxiliaryLightManager.LIGHT_NBT_KEY).orElse(null);
+
             return new SerializableChunkData(
                 containerFactory,
                 chunkpos,
@@ -204,7 +_,9 @@
                 list2,
                 list3,
                 list4,
-                compoundtag1
+                compoundtag1,
+                attachmentData,
+                auxLightData
             );
         }
     }
@@ -264,6 +_,9 @@
                 postLoadChunk(level, this.entities, this.blockEntities),
                 BlendingData.unpack(this.blendingData)
             );
+            if (this.auxLightData != null) {
+                ((LevelChunk) chunkaccess).getAuxLightManager(chunkPos).deserializeNBT(this.auxLightData);
+            }
         } else {
             ProtoChunkTicks<Block> protochunkticks = ProtoChunkTicks.load(this.packedTicks.blocks());
             ProtoChunkTicks<Fluid> protochunkticks1 = ProtoChunkTicks.load(this.packedTicks.fluids());
@@ -312,6 +_,10 @@
             }
         }
 
+        if (this.attachmentData != null) {
+            chunkaccess.readAttachmentsFromNBT(level.registryAccess(), this.attachmentData);
+        }
+
         if (chunktype == ChunkType.LEVELCHUNK) {
             return new ImposterProtoChunk((LevelChunk)chunkaccess, false);
         } else {
@@ -395,6 +_,18 @@
             CompoundTag compoundtag1 = packStructureData(
                 StructurePieceSerializationContext.fromLevel(level), chunkpos, chunk.getAllStarts(), chunk.getAllReferences()
             );
+
+            CompoundTag attachmentData = null;
+            try {
+                attachmentData = chunk.writeAttachmentsToNBT(level.registryAccess());
+            } catch (Exception exception) {
+                LOGGER.error("Failed to write chunk attachments. An attachment has likely thrown an exception trying to write state. It will not persist. Report this to the mod author", exception);
+            }
+            ListTag auxLightData = null;
+            if (chunk instanceof LevelChunk levelChunk) {
+                auxLightData = levelChunk.getAuxLightManager(chunkpos).serializeNBT();
+            }
+
             return new SerializableChunkData(
                 level.palettedContainerFactory(),
                 chunkpos,
@@ -413,7 +_,9 @@
                 list,
                 list2,
                 list1,
-                compoundtag1
+                compoundtag1,
+                attachmentData,
+                auxLightData
             );
         }
     }
@@ -481,6 +_,14 @@
         this.heightmaps.forEach((type, data) -> compoundtag2.put(type.getSerializationKey(), new LongArrayTag(data)));
         compoundtag.put("Heightmaps", compoundtag2);
         compoundtag.put("structures", this.structureData);
+
+        if (attachmentData != null) {
+            compoundtag.put(net.neoforged.neoforge.attachment.AttachmentHolder.ATTACHMENTS_NBT_KEY, attachmentData);
+        }
+        if (auxLightData != null) {
+            compoundtag.put(net.neoforged.neoforge.common.world.LevelChunkAuxiliaryLightManager.LIGHT_NBT_KEY, auxLightData);
+        }
+
         return compoundtag;
     }
 
