--- a/net/minecraft/recipebook/PlaceRecipe.java
+++ b/net/minecraft/recipebook/PlaceRecipe.java
@@ -2,7 +_,6 @@
 
 import java.util.Iterator;
 import net.minecraft.util.Mth;
-import net.minecraft.world.item.crafting.Recipe;
 import net.minecraft.world.item.crafting.RecipeHolder;
 import net.minecraft.world.item.crafting.ShapedRecipe;
 
@@ -10,27 +_,26 @@
     default void placeRecipe(int p_135409_, int p_135410_, int p_135411_, RecipeHolder<?> p_301225_, Iterator<T> p_135413_, int p_135414_) {
         int i = p_135409_;
         int j = p_135410_;
-        Recipe<?> recipe = p_301225_.value();
-        if (recipe instanceof ShapedRecipe shapedrecipe) {
+        if (p_301225_.value() instanceof net.neoforged.neoforge.common.crafting.IShapedRecipe<?> shapedrecipe) {
             i = shapedrecipe.getWidth();
             j = shapedrecipe.getHeight();
         }
 
         int k1 = 0;
 
-        for(int k = 0; k < p_135410_; ++k) {
+        for (int k = 0; k < p_135410_; k++) {
             if (k1 == p_135411_) {
-                ++k1;
+                k1++;
             }
 
             boolean flag = (float)j < (float)p_135410_ / 2.0F;
             int l = Mth.floor((float)p_135410_ / 2.0F - (float)j / 2.0F);
             if (flag && l > k) {
                 k1 += p_135409_;
-                ++k;
+                k++;
             }
 
-            for(int i1 = 0; i1 < p_135409_; ++i1) {
+            for (int i1 = 0; i1 < p_135409_; i1++) {
                 if (!p_135413_.hasNext()) {
                     return;
                 }
@@ -51,7 +_,7 @@
                     break;
                 }
 
-                ++k1;
+                k1++;
             }
         }
     }
