--- a/com/mojang/blaze3d/platform/GlDebug.java
+++ b/com/mojang/blaze3d/platform/GlDebug.java
@@ -35,7 +_,7 @@
     }
 
     public static String sourceToString(int p_84056_) {
-        switch(p_84056_) {
+        switch (p_84056_) {
             case 33350:
                 return "API";
             case 33351:
@@ -54,7 +_,7 @@
     }
 
     public static String typeToString(int p_84058_) {
-        switch(p_84058_) {
+        switch (p_84058_) {
             case 33356:
                 return "ERROR";
             case 33357:
@@ -75,7 +_,7 @@
     }
 
     public static String severityToString(int p_84060_) {
-        switch(p_84060_) {
+        switch (p_84060_) {
             case 33387:
                 return "NOTIFICATION";
             case 37190:
@@ -92,10 +_,10 @@
     private static void printDebugLog(int p_84039_, int p_84040_, int p_84041_, int p_84042_, int p_84043_, long p_84044_, long p_84045_) {
         String s = GLDebugMessageCallback.getMessage(p_84043_, p_84044_);
         GlDebug.LogEntry gldebug$logentry;
-        synchronized(MESSAGE_BUFFER) {
+        synchronized (MESSAGE_BUFFER) {
             gldebug$logentry = lastEntry;
             if (gldebug$logentry != null && gldebug$logentry.isSame(p_84039_, p_84040_, p_84041_, p_84042_, s)) {
-                ++gldebug$logentry.count;
+                gldebug$logentry.count++;
             } else {
                 gldebug$logentry = new GlDebug.LogEntry(p_84039_, p_84040_, p_84041_, p_84042_, s);
                 MESSAGE_BUFFER.add(gldebug$logentry);
@@ -107,10 +_,10 @@
     }
 
     public static List<String> getLastOpenGlDebugMessages() {
-        synchronized(MESSAGE_BUFFER) {
+        synchronized (MESSAGE_BUFFER) {
             List<String> list = Lists.newArrayListWithCapacity(MESSAGE_BUFFER.size());
 
-            for(GlDebug.LogEntry gldebug$logentry : MESSAGE_BUFFER) {
+            for (GlDebug.LogEntry gldebug$logentry : MESSAGE_BUFFER) {
                 list.add(gldebug$logentry + " x " + gldebug$logentry.count);
             }
 
@@ -133,7 +_,7 @@
                     GL11.glEnable(33346);
                 }
 
-                for(int i = 0; i < DEBUG_LEVELS.size(); ++i) {
+                for (int i = 0; i < DEBUG_LEVELS.size(); i++) {
                     boolean flag = i < p_84050_;
                     KHRDebug.glDebugMessageControl(4352, 4352, DEBUG_LEVELS.get(i), (int[])null, flag);
                 }
@@ -145,7 +_,7 @@
                     GL11.glEnable(33346);
                 }
 
-                for(int j = 0; j < DEBUG_LEVELS_ARB.size(); ++j) {
+                for (int j = 0; j < DEBUG_LEVELS_ARB.size(); j++) {
                     boolean flag1 = j < p_84050_;
                     ARBDebugOutput.glDebugMessageControlARB(4352, 4352, DEBUG_LEVELS_ARB.get(j), (int[])null, flag1);
                 }
