--- a/com/mojang/blaze3d/platform/GLX.java
+++ b/com/mojang/blaze3d/platform/GLX.java
@@ -21,9 +_,7 @@
 import org.lwjgl.glfw.GLFW;
 import org.lwjgl.glfw.GLFWErrorCallback;
 import org.lwjgl.glfw.GLFWErrorCallbackI;
-import org.lwjgl.glfw.GLFWNativeGLX;
 import org.lwjgl.glfw.GLFWVidMode;
-import org.lwjgl.opengl.GL;
 import org.lwjgl.system.MemoryUtil;
 import org.slf4j.Logger;
 import oshi.SystemInfo;
@@ -60,7 +_,6 @@
 
     public static LongSupplier _initGlfw() {
         RenderSystem.assertInInitPhase();
-        GLFWNativeGLX.setPath(GL.getFunctionProvider());
         Window.checkGlfwError((p_242032_, p_242033_) -> {
             throw new IllegalStateException(String.format(Locale.ROOT, "GLFW error before init: [0x%X]%s", p_242032_, p_242033_));
         });
@@ -74,7 +_,7 @@
         } else {
             LongSupplier longsupplier = () -> (long)(GLFW.glfwGetTime() * 1.0E9);
 
-            for(String s : list) {
+            for (String s : list) {
                 LOGGER.error("GLFW error collected during initialization: {}", s);
             }
 
