--- a/com/mojang/blaze3d/platform/InputConstants.java
+++ b/com/mojang/blaze3d/platform/InputConstants.java
@@ -166,12 +_,12 @@
         if (InputConstants.Key.NAME_MAP.containsKey(p_84852_)) {
             return InputConstants.Key.NAME_MAP.get(p_84852_);
         } else {
-            for(InputConstants.Type inputconstants$type : InputConstants.Type.values()) {
+            for (InputConstants.Type inputconstants$type : InputConstants.Type.values()) {
                 if (p_84852_.startsWith(inputconstants$type.defaultPrefix)) {
                     String s = p_84852_.substring(inputconstants$type.defaultPrefix.length() + 1);
                     int i = Integer.parseInt(s);
                     if (inputconstants$type == InputConstants.Type.MOUSE) {
-                        --i;
+                        i--;
                     }
 
                     return inputconstants$type.getOrCreate(i);
@@ -221,13 +_,13 @@
 
     static {
         Lookup lookup = MethodHandles.lookup();
-        MethodType methodtype = MethodType.methodType(Boolean.TYPE);
+        MethodType methodtype = MethodType.methodType(boolean.class);
         MethodHandle methodhandle = null;
         int i = 0;
 
         try {
             methodhandle = lookup.findStatic(GLFW.class, "glfwRawMouseMotionSupported", methodtype);
-            MethodHandle methodhandle1 = lookup.findStaticGetter(GLFW.class, "GLFW_RAW_MOUSE_MOTION", Integer.TYPE);
+            MethodHandle methodhandle1 = lookup.findStaticGetter(GLFW.class, "GLFW_RAW_MOUSE_MOTION", int.class);
             i = (int)methodhandle1.invokeExact();
         } catch (NoSuchFieldException | NoSuchMethodException nosuchmethodexception) {
         } catch (Throwable throwable) {
