--- a/com/mojang/blaze3d/platform/NativeImage.java
+++ b/com/mojang/blaze3d/platform/NativeImage.java
@@ -244,7 +_,7 @@
             IntBuffer intbuffer = MemoryUtil.memIntBuffer(this.pixels, i);
             IntBuffer intbuffer1 = MemoryUtil.memIntBuffer(nativeimage.pixels, i);
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 intbuffer1.put(j, p_267084_.applyAsInt(intbuffer.get(j)));
             }
 
@@ -260,7 +_,7 @@
             int i = this.width * this.height;
             IntBuffer intbuffer = MemoryUtil.memIntBuffer(this.pixels, i);
 
-            for(int j = 0; j < i; ++j) {
+            for (int j = 0; j < i; j++) {
                 intbuffer.put(j, p_285490_.applyAsInt(intbuffer.get(j)));
             }
         }
@@ -397,8 +_,8 @@
             this.checkAllocated();
             int[] aint = new int[this.getWidth() * this.getHeight()];
 
-            for(int i = 0; i < this.getHeight(); ++i) {
-                for(int j = 0; j < this.getWidth(); ++j) {
+            for (int i = 0; i < this.getHeight(); i++) {
+                for (int j = 0; j < this.getWidth(); j++) {
                     int k = this.getPixelRGBA(j, i);
                     aint[j + i * this.getWidth()] = FastColor.ARGB32.color(
                         FastColor.ABGR32.alpha(k), FastColor.ABGR32.red(k), FastColor.ABGR32.green(k), FastColor.ABGR32.blue(k)
@@ -484,8 +_,8 @@
         this.format.setPackPixelStoreState();
         GlStateManager._getTexImage(3553, p_85046_, this.format.glFormat(), 5121, this.pixels);
         if (p_85047_ && this.format.hasAlpha()) {
-            for(int i = 0; i < this.getHeight(); ++i) {
-                for(int j = 0; j < this.getWidth(); ++j) {
+            for (int i = 0; i < this.getHeight(); i++) {
+                for (int j = 0; j < this.getWidth(); j++) {
                     this.setPixelRGBA(j, i, this.getPixelRGBA(j, i) | 255 << this.format.alphaOffset());
                 }
             }
@@ -608,7 +_,7 @@
                 int j = Math.min(this.getWidth(), p_85055_.getWidth());
                 int k = Math.min(this.getHeight(), p_85055_.getHeight());
 
-                for(int l = 0; l < k; ++l) {
+                for (int l = 0; l < k; l++) {
                     int i1 = l * p_85055_.getWidth() * i;
                     int j1 = l * this.getWidth() * i;
                     MemoryUtil.memCopy(p_85055_.pixels + (long)i1, this.pixels + (long)j1, (long)j);
@@ -618,8 +_,8 @@
     }
 
     public void fillRect(int p_84998_, int p_84999_, int p_85000_, int p_85001_, int p_85002_) {
-        for(int i = p_84999_; i < p_84999_ + p_85001_; ++i) {
-            for(int j = p_84998_; j < p_84998_ + p_85000_; ++j) {
+        for (int i = p_84999_; i < p_84999_ + p_85001_; i++) {
+            for (int j = p_84998_; j < p_84998_ + p_85000_; j++) {
                 this.setPixelRGBA(j, i, p_85002_);
             }
         }
@@ -632,8 +_,8 @@
     public void copyRect(
         NativeImage p_261644_, int p_262056_, int p_261490_, int p_261959_, int p_262110_, int p_261522_, int p_261505_, boolean p_261480_, boolean p_261622_
     ) {
-        for(int i = 0; i < p_261505_; ++i) {
-            for(int j = 0; j < p_261522_; ++j) {
+        for (int i = 0; i < p_261505_; i++) {
+            for (int j = 0; j < p_261522_; j++) {
                 int k = p_261480_ ? p_261522_ - 1 - j : j;
                 int l = p_261622_ ? p_261505_ - 1 - i : i;
                 int i1 = this.getPixelRGBA(p_262056_ + j, p_261490_ + i);
@@ -649,7 +_,7 @@
         long k = MemoryUtil.nmemAlloc((long)j);
 
         try {
-            for(int l = 0; l < this.getHeight() / 2; ++l) {
+            for (int l = 0; l < this.getHeight() / 2; l++) {
                 int i1 = l * this.getWidth() * i;
                 int j1 = (this.getHeight() - 1 - l) * this.getWidth() * i;
                 MemoryUtil.memCopy(this.pixels + (long)i1, k, (long)j);
@@ -831,7 +_,7 @@
         }
 
         static NativeImage.Format getStbFormat(int p_85168_) {
-            switch(p_85168_) {
+            switch (p_85168_) {
                 case 1:
                     return LUMINANCE;
                 case 2:
