--- a/com/mojang/blaze3d/platform/ScreenManager.java
+++ b/com/mojang/blaze3d/platform/ScreenManager.java
@@ -24,7 +_,7 @@
         GLFW.glfwSetMonitorCallback(this::onMonitorChange);
         PointerBuffer pointerbuffer = GLFW.glfwGetMonitors();
         if (pointerbuffer != null) {
-            for(int i = 0; i < pointerbuffer.limit(); ++i) {
+            for (int i = 0; i < pointerbuffer.limit(); i++) {
                 long j = pointerbuffer.get(i);
                 this.monitors.put(j, p_85265_.createMonitor(j));
             }
@@ -63,7 +_,7 @@
             long k1 = GLFW.glfwGetPrimaryMonitor();
             LOGGER.debug("Selecting monitor - primary: {}, current monitors: {}", k1, this.monitors);
 
-            for(Monitor monitor1 : this.monitors.values()) {
+            for (Monitor monitor1 : this.monitors.values()) {
                 int l1 = monitor1.getX();
                 int i2 = l1 + monitor1.getCurrentMode().getWidth();
                 int j2 = monitor1.getY();
