--- a/com/mojang/blaze3d/platform/TextureUtil.java
+++ b/com/mojang/blaze3d/platform/TextureUtil.java
@@ -66,7 +_,7 @@
             GlStateManager._texParameter(3553, 34049, 0.0F);
         }
 
-        for(int i = 0; i <= p_85300_; ++i) {
+        for (int i = 0; i <= p_85300_; i++) {
             GlStateManager._texImage2D(3553, i, p_85298_.glFormat(), p_85301_ >> i, p_85302_ >> i, 0, 6408, 5121, null);
         }
     }
@@ -87,7 +_,7 @@
         ByteBuffer bytebuffer = MemoryUtil.memAlloc(p_273297_);
 
         try {
-            while(p_273208_.read(bytebuffer) != -1) {
+            while (p_273208_.read(bytebuffer) != -1) {
                 if (!bytebuffer.hasRemaining()) {
                     bytebuffer = MemoryUtil.memRealloc(bytebuffer, bytebuffer.capacity() * 2);
                 }
@@ -110,7 +_,7 @@
         RenderSystem.assertOnRenderThread();
         bind(p_285400_);
 
-        for(int i = 0; i <= p_285244_; ++i) {
+        for (int i = 0; i <= p_285244_; i++) {
             int j = p_285373_ >> i;
             int k = p_285206_ >> i;
 
