--- a/com/mojang/blaze3d/platform/Window.java
+++ b/com/mojang/blaze3d/platform/Window.java
@@ -95,6 +_,10 @@
         GLFW.glfwWindowHint(139267, 3);
         GLFW.glfwWindowHint(139272, 204801);
         GLFW.glfwWindowHint(139270, 1);
+        var earlyLoadingScreen = net.neoforged.fml.loading.EarlyLoadingScreenController.current();
+        if (earlyLoadingScreen != null) {
+            this.handle = takeOverWindow(earlyLoadingScreen, title);
+        } else {
         this.handle = GLFW.glfwCreateWindow(this.width, this.height, title, this.fullscreen && initialMonitor != null ? initialMonitor.getMonitor() : 0L, 0L);
         if (initialMonitor != null) {
             VideoMode mode = initialMonitor.getPreferredVidMode(this.fullscreen ? this.preferredFullscreenVideoMode : Optional.empty());
@@ -107,6 +_,7 @@
             this.windowedX = this.x = actualX[0];
             this.windowedY = this.y = actualY[0];
         }
+        }
 
         this.setMode();
         this.refreshFramebufferSize();
@@ -208,8 +_,10 @@
     }
 
     public void defaultErrorCallback(int errorCode, long description) {
-        RenderSystem.assertOnRenderThread();
         String errorString = MemoryUtil.memUTF8(description);
+        if (!RenderSystem.isOnRenderThread()) {
+            throw new IllegalStateException("Encountered GL error off-thread @ " + errorSection + ": " + errorCode + ": " + errorString);
+        }
         LOGGER.error("########## GL ERROR ##########");
         LOGGER.error("@ {}", this.errorSection);
         LOGGER.error("{}: {}", errorCode, errorString);
@@ -515,5 +_,27 @@
         private WindowInitFailed(String message) {
             super(message);
         }
+    }
+
+    // Neo take over window and its properties from early display
+    private long takeOverWindow(net.neoforged.fml.loading.EarlyLoadingScreenController earlyLoadingScreen, String title) {
+        long window = earlyLoadingScreen.takeOverGlfwWindow();
+
+        GLFW.glfwSetWindowTitle(window, title);
+
+        var x = new int[1];
+        var y = new int[1];
+        GLFW.glfwGetWindowPos(window, x, y);
+        this.x = this.windowedX = x[0];
+        this.y = this.windowedY = y[0];
+
+        var width = new int[1];
+        var height = new int[1];
+        GLFW.glfwGetWindowSize(window, width, height);
+        // The window height and width can be 0 if minimized
+        this.width = this.windowedWidth = Math.max(width[0], 1);
+        this.height = this.windowedHeight = Math.max(height[0], 1);
+
+        return window;
     }
 }
