--- a/net/minecraft/client/renderer/texture/atlas/SpriteSourceType.java
+++ b/net/minecraft/client/renderer/texture/atlas/SpriteSourceType.java
@@ -1,9 +_,9 @@
 package net.minecraft.client.renderer.texture.atlas;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
-public record SpriteSourceType(Codec<? extends SpriteSource> codec) {
+public record SpriteSourceType(MapCodec<? extends SpriteSource> codec) {
 }
