--- a/net/minecraft/util/profiling/FilledProfileResults.java
+++ b/net/minecraft/util/profiling/FilledProfileResults.java
@@ -87,7 +_,7 @@
 
         long l = 0L;
 
-        for(String s1 : this.entries.keySet()) {
+        for (String s1 : this.entries.keySet()) {
             if (isDirectChild(p_18493_, s1)) {
                 l += this.getEntry(s1).getDuration();
             }
@@ -102,7 +_,7 @@
             i = l;
         }
 
-        for(String s2 : this.entries.keySet()) {
+        for (String s2 : this.entries.keySet()) {
             if (isDirectChild(p_18493_, s2)) {
                 ProfilerPathEntry profilerpathentry2 = this.getEntry(s2);
                 long i1 = profilerpathentry2.getDuration();
@@ -221,7 +_,7 @@
     private static StringBuilder indentLine(StringBuilder p_18498_, int p_18499_) {
         p_18498_.append(String.format(Locale.ROOT, "[%02d] ", p_18499_));
 
-        for(int i = 0; i < p_18499_; ++i) {
+        for (int i = 0; i < p_18499_; i++) {
             p_18498_.append("|   ");
         }
 
@@ -242,7 +_,7 @@
                     .append('\n')
         );
         if (list.size() >= 3) {
-            for(int i = 1; i < list.size(); ++i) {
+            for (int i = 1; i < list.size(); i++) {
                 ResultField resultfield = list.get(i);
                 indentLine(p_18484_, p_18482_)
                     .append(resultfield.name)
