--- a/net/minecraft/client/model/AnimationUtils.java
+++ b/net/minecraft/client/model/AnimationUtils.java
@@ -56,8 +_,8 @@
     }
 
     public static void bobModelPart(ModelPart p_170342_, float p_170343_, float p_170344_) {
-        p_170342_.zRot += p_170344_ * (Mth.cos(p_170343_ * 0.09F) * 0.05F + 0.05F);
-        p_170342_.xRot += p_170344_ * Mth.sin(p_170343_ * 0.067F) * 0.05F;
+        p_170342_.zRot = p_170342_.zRot + p_170344_ * (Mth.cos(p_170343_ * 0.09F) * 0.05F + 0.05F);
+        p_170342_.xRot = p_170342_.xRot + p_170344_ * Mth.sin(p_170343_ * 0.067F) * 0.05F;
     }
 
     public static void bobArms(ModelPart p_102083_, ModelPart p_102084_, float p_102085_) {
