--- a/net/minecraft/client/model/DrownedModel.java
+++ b/net/minecraft/client/model/DrownedModel.java
@@ -68,8 +_,8 @@
                 - this.swimAmount * 0.35F * Mth.sin(0.1F * p_102529_);
             this.rightArm.zRot = this.rotlerpRad(this.swimAmount, this.rightArm.zRot, -0.15F);
             this.leftArm.zRot = this.rotlerpRad(this.swimAmount, this.leftArm.zRot, 0.15F);
-            this.leftLeg.xRot -= this.swimAmount * 0.55F * Mth.sin(0.1F * p_102529_);
-            this.rightLeg.xRot += this.swimAmount * 0.55F * Mth.sin(0.1F * p_102529_);
+            this.leftLeg.xRot = this.leftLeg.xRot - this.swimAmount * 0.55F * Mth.sin(0.1F * p_102529_);
+            this.rightLeg.xRot = this.rightLeg.xRot + this.swimAmount * 0.55F * Mth.sin(0.1F * p_102529_);
             this.head.xRot = 0.0F;
         }
     }
