--- a/net/minecraft/client/model/ElytraModel.java
+++ b/net/minecraft/client/model/ElytraModel.java
@@ -75,9 +_,9 @@
 
         this.leftWing.y = f2;
         if (p_102544_ instanceof AbstractClientPlayer abstractclientplayer) {
-            abstractclientplayer.elytraRotX += (f - abstractclientplayer.elytraRotX) * 0.1F;
-            abstractclientplayer.elytraRotY += (f3 - abstractclientplayer.elytraRotY) * 0.1F;
-            abstractclientplayer.elytraRotZ += (f1 - abstractclientplayer.elytraRotZ) * 0.1F;
+            abstractclientplayer.elytraRotX = abstractclientplayer.elytraRotX + (f - abstractclientplayer.elytraRotX) * 0.1F;
+            abstractclientplayer.elytraRotY = abstractclientplayer.elytraRotY + (f3 - abstractclientplayer.elytraRotY) * 0.1F;
+            abstractclientplayer.elytraRotZ = abstractclientplayer.elytraRotZ + (f1 - abstractclientplayer.elytraRotZ) * 0.1F;
             this.leftWing.xRot = abstractclientplayer.elytraRotX;
             this.leftWing.yRot = abstractclientplayer.elytraRotY;
             this.leftWing.zRot = abstractclientplayer.elytraRotZ;
