--- a/net/minecraft/client/model/HumanoidModel.java
+++ b/net/minecraft/client/model/HumanoidModel.java
@@ -268,20 +_,16 @@
     }
 
     private void poseRightArm(T p_102876_) {
-        switch(this.rightArmPose) {
+        switch (this.rightArmPose) {
             case EMPTY:
                 this.rightArm.yRot = 0.0F;
-                break;
-            case BLOCK:
-                this.poseBlockingArm(this.rightArm, true);
                 break;
             case ITEM:
                 this.rightArm.xRot = this.rightArm.xRot * 0.5F - (float) (Math.PI / 10);
                 this.rightArm.yRot = 0.0F;
                 break;
-            case THROW_SPEAR:
-                this.rightArm.xRot = this.rightArm.xRot * 0.5F - (float) Math.PI;
-                this.rightArm.yRot = 0.0F;
+            case BLOCK:
+                this.poseBlockingArm(this.rightArm, true);
                 break;
             case BOW_AND_ARROW:
                 this.rightArm.yRot = -0.1F + this.head.yRot;
@@ -289,16 +_,16 @@
                 this.rightArm.xRot = (float) (-Math.PI / 2) + this.head.xRot;
                 this.leftArm.xRot = (float) (-Math.PI / 2) + this.head.xRot;
                 break;
+            case THROW_SPEAR:
+                this.rightArm.xRot = this.rightArm.xRot * 0.5F - (float) Math.PI;
+                this.rightArm.yRot = 0.0F;
+                break;
             case CROSSBOW_CHARGE:
                 AnimationUtils.animateCrossbowCharge(this.rightArm, this.leftArm, p_102876_, true);
                 break;
             case CROSSBOW_HOLD:
                 AnimationUtils.animateCrossbowHold(this.rightArm, this.leftArm, this.head, true);
                 break;
-            case BRUSH:
-                this.rightArm.xRot = this.rightArm.xRot * 0.5F - (float) (Math.PI / 5);
-                this.rightArm.yRot = 0.0F;
-                break;
             case SPYGLASS:
                 this.rightArm.xRot = Mth.clamp(this.head.xRot - 1.9198622F - (p_102876_.isCrouching() ? (float) (Math.PI / 12) : 0.0F), -2.4F, 3.3F);
                 this.rightArm.yRot = this.head.yRot - (float) (Math.PI / 12);
@@ -306,24 +_,26 @@
             case TOOT_HORN:
                 this.rightArm.xRot = Mth.clamp(this.head.xRot, -1.2F, 1.2F) - 1.4835298F;
                 this.rightArm.yRot = this.head.yRot - (float) (Math.PI / 6);
+                break;
+            case BRUSH:
+                this.rightArm.xRot = this.rightArm.xRot * 0.5F - (float) (Math.PI / 5);
+                this.rightArm.yRot = 0.0F;
+            default:
+                this.rightArmPose.applyTransform(this, p_102876_, net.minecraft.world.entity.HumanoidArm.RIGHT);
         }
     }
 
     private void poseLeftArm(T p_102879_) {
-        switch(this.leftArmPose) {
+        switch (this.leftArmPose) {
             case EMPTY:
                 this.leftArm.yRot = 0.0F;
-                break;
-            case BLOCK:
-                this.poseBlockingArm(this.leftArm, false);
                 break;
             case ITEM:
                 this.leftArm.xRot = this.leftArm.xRot * 0.5F - (float) (Math.PI / 10);
                 this.leftArm.yRot = 0.0F;
                 break;
-            case THROW_SPEAR:
-                this.leftArm.xRot = this.leftArm.xRot * 0.5F - (float) Math.PI;
-                this.leftArm.yRot = 0.0F;
+            case BLOCK:
+                this.poseBlockingArm(this.leftArm, false);
                 break;
             case BOW_AND_ARROW:
                 this.rightArm.yRot = -0.1F + this.head.yRot - 0.4F;
@@ -331,16 +_,16 @@
                 this.rightArm.xRot = (float) (-Math.PI / 2) + this.head.xRot;
                 this.leftArm.xRot = (float) (-Math.PI / 2) + this.head.xRot;
                 break;
+            case THROW_SPEAR:
+                this.leftArm.xRot = this.leftArm.xRot * 0.5F - (float) Math.PI;
+                this.leftArm.yRot = 0.0F;
+                break;
             case CROSSBOW_CHARGE:
                 AnimationUtils.animateCrossbowCharge(this.rightArm, this.leftArm, p_102879_, false);
                 break;
             case CROSSBOW_HOLD:
                 AnimationUtils.animateCrossbowHold(this.rightArm, this.leftArm, this.head, false);
                 break;
-            case BRUSH:
-                this.leftArm.xRot = this.leftArm.xRot * 0.5F - (float) (Math.PI / 5);
-                this.leftArm.yRot = 0.0F;
-                break;
             case SPYGLASS:
                 this.leftArm.xRot = Mth.clamp(this.head.xRot - 1.9198622F - (p_102879_.isCrouching() ? (float) (Math.PI / 12) : 0.0F), -2.4F, 3.3F);
                 this.leftArm.yRot = this.head.yRot + (float) (Math.PI / 12);
@@ -348,6 +_,12 @@
             case TOOT_HORN:
                 this.leftArm.xRot = Mth.clamp(this.head.xRot, -1.2F, 1.2F) - 1.4835298F;
                 this.leftArm.yRot = this.head.yRot + (float) (Math.PI / 6);
+                break;
+            case BRUSH:
+                this.leftArm.xRot = this.leftArm.xRot * 0.5F - (float) (Math.PI / 5);
+                this.leftArm.yRot = 0.0F;
+            default:
+                this.leftArmPose.applyTransform(this, p_102879_, net.minecraft.world.entity.HumanoidArm.LEFT);
         }
     }
 
@@ -370,9 +_,9 @@
             this.rightArm.x = -Mth.cos(this.body.yRot) * 5.0F;
             this.leftArm.z = -Mth.sin(this.body.yRot) * 5.0F;
             this.leftArm.x = Mth.cos(this.body.yRot) * 5.0F;
-            this.rightArm.yRot += this.body.yRot;
-            this.leftArm.yRot += this.body.yRot;
-            this.leftArm.xRot += this.body.yRot;
+            this.rightArm.yRot = this.rightArm.yRot + this.body.yRot;
+            this.leftArm.yRot = this.leftArm.yRot + this.body.yRot;
+            this.leftArm.xRot = this.leftArm.xRot + this.body.yRot;
             f = 1.0F - this.attackTime;
             f *= f;
             f *= f;
@@ -380,8 +_,8 @@
             float f1 = Mth.sin(f * (float) Math.PI);
             float f2 = Mth.sin(this.attackTime * (float) Math.PI) * -(this.head.xRot - 0.7F) * 0.75F;
             modelpart.xRot -= f1 * 1.2F + f2;
-            modelpart.yRot += this.body.yRot * 2.0F;
-            modelpart.zRot += Mth.sin(this.attackTime * (float) Math.PI) * -0.4F;
+            modelpart.yRot = modelpart.yRot + this.body.yRot * 2.0F;
+            modelpart.zRot = modelpart.zRot + Mth.sin(this.attackTime * (float) Math.PI) * -0.4F;
         }
     }
 
@@ -446,7 +_,7 @@
     }
 
     @OnlyIn(Dist.CLIENT)
-    public static enum ArmPose {
+    public static enum ArmPose implements net.neoforged.neoforge.common.IExtensibleEnum {
         EMPTY(false),
         ITEM(false),
         BLOCK(false),
@@ -462,10 +_,29 @@
 
         private ArmPose(boolean p_102896_) {
             this.twoHanded = p_102896_;
+            this.forgeArmPose = null;
         }
 
         public boolean isTwoHanded() {
             return this.twoHanded;
         }
+        // FORGE START
+        @javax.annotation.Nullable
+        private final net.neoforged.neoforge.client.IArmPoseTransformer forgeArmPose;
+
+        private ArmPose(boolean twoHanded, @javax.annotation.Nonnull net.neoforged.neoforge.client.IArmPoseTransformer forgeArmPose) {
+            this.twoHanded = twoHanded;
+            com.google.common.base.Preconditions.checkNotNull(forgeArmPose, "Cannot create new ArmPose with null transformer!");
+            this.forgeArmPose = forgeArmPose;
+        }
+
+        public static ArmPose create(String name, boolean twoHanded, @javax.annotation.Nonnull net.neoforged.neoforge.client.IArmPoseTransformer forgeArmPose) {
+            throw new IllegalStateException("Enum not extended");
+        }
+
+        public <T extends LivingEntity> void applyTransform(HumanoidModel<T> model, T entity, net.minecraft.world.entity.HumanoidArm arm) {
+            if (this.forgeArmPose != null) this.forgeArmPose.applyTransform(model, entity, arm);
+        }
+        // FORGE END
     }
 }
