--- a/net/minecraft/client/model/ModelUtils.java
+++ b/net/minecraft/client/model/ModelUtils.java
@@ -8,11 +_,11 @@
     public static float rotlerpRad(float p_103126_, float p_103127_, float p_103128_) {
         float f = p_103127_ - p_103126_;
 
-        while(f < (float) -Math.PI) {
+        while (f < (float) -Math.PI) {
             f += (float) (Math.PI * 2);
         }
 
-        while(f >= (float) Math.PI) {
+        while (f >= (float) Math.PI) {
             f -= (float) (Math.PI * 2);
         }
 
