--- a/net/minecraft/client/model/ParrotModel.java
+++ b/net/minecraft/client/model/ParrotModel.java
@@ -112,7 +_,24 @@
         this.tail.x = 0.0F;
         this.rightWing.x = -1.5F;
         this.leftWing.x = 1.5F;
-        switch(p_103242_) {
+        switch (p_103242_) {
+            case STANDING:
+                this.leftLeg.xRot = this.leftLeg.xRot + Mth.cos(p_103244_ * 0.6662F) * 1.4F * p_103245_;
+                this.rightLeg.xRot = this.rightLeg.xRot + Mth.cos(p_103244_ * 0.6662F + (float) Math.PI) * 1.4F * p_103245_;
+            case FLYING:
+            case ON_SHOULDER:
+            default:
+                float f2 = p_103246_ * 0.3F;
+                this.head.y = 15.69F + f2;
+                this.tail.xRot = 1.015F + Mth.cos(p_103244_ * 0.6662F) * 0.3F * p_103245_;
+                this.tail.y = 21.07F + f2;
+                this.body.y = 16.5F + f2;
+                this.leftWing.zRot = -0.0873F - p_103246_;
+                this.leftWing.y = 16.94F + f2;
+                this.rightWing.zRot = 0.0873F + p_103246_;
+                this.rightWing.y = 16.94F + f2;
+                this.leftLeg.y = 22.0F + f2;
+                this.rightLeg.y = 22.0F + f2;
             case SITTING:
                 break;
             case PARTY:
@@ -133,24 +_,6 @@
                 this.rightWing.y = 16.94F + f1;
                 this.tail.x = f;
                 this.tail.y = 21.07F + f1;
-                break;
-            case STANDING:
-                this.leftLeg.xRot += Mth.cos(p_103244_ * 0.6662F) * 1.4F * p_103245_;
-                this.rightLeg.xRot += Mth.cos(p_103244_ * 0.6662F + (float) Math.PI) * 1.4F * p_103245_;
-            case FLYING:
-            case ON_SHOULDER:
-            default:
-                float f2 = p_103246_ * 0.3F;
-                this.head.y = 15.69F + f2;
-                this.tail.xRot = 1.015F + Mth.cos(p_103244_ * 0.6662F) * 0.3F * p_103245_;
-                this.tail.y = 21.07F + f2;
-                this.body.y = 16.5F + f2;
-                this.leftWing.zRot = -0.0873F - p_103246_;
-                this.leftWing.y = 16.94F + f2;
-                this.rightWing.zRot = 0.0873F + p_103246_;
-                this.rightWing.y = 16.94F + f2;
-                this.leftLeg.y = 22.0F + f2;
-                this.rightLeg.y = 22.0F + f2;
         }
     }
 
@@ -167,7 +_,14 @@
         this.rightLeg.y = 22.0F;
         this.leftLeg.zRot = 0.0F;
         this.rightLeg.zRot = 0.0F;
-        switch(p_103240_) {
+        switch (p_103240_) {
+            case FLYING:
+                this.leftLeg.xRot += (float) (Math.PI * 2.0 / 9.0);
+                this.rightLeg.xRot += (float) (Math.PI * 2.0 / 9.0);
+            case STANDING:
+            case ON_SHOULDER:
+            default:
+                break;
             case SITTING:
                 float f = 1.9F;
                 this.head.y = 17.59F;
@@ -178,21 +_,14 @@
                 this.leftWing.y = 18.84F;
                 this.rightWing.zRot = 0.0873F;
                 this.rightWing.y = 18.84F;
-                ++this.leftLeg.y;
-                ++this.rightLeg.y;
-                ++this.leftLeg.xRot;
-                ++this.rightLeg.xRot;
+                this.leftLeg.y++;
+                this.rightLeg.y++;
+                this.leftLeg.xRot++;
+                this.rightLeg.xRot++;
                 break;
             case PARTY:
                 this.leftLeg.zRot = (float) (-Math.PI / 9);
                 this.rightLeg.zRot = (float) (Math.PI / 9);
-            case STANDING:
-            case ON_SHOULDER:
-            default:
-                break;
-            case FLYING:
-                this.leftLeg.xRot += (float) (Math.PI * 2.0 / 9.0);
-                this.rightLeg.xRot += (float) (Math.PI * 2.0 / 9.0);
         }
     }
 
