--- a/net/minecraft/client/model/SquidModel.java
+++ b/net/minecraft/client/model/SquidModel.java
@@ -37,7 +_,7 @@
         int j = 8;
         CubeListBuilder cubelistbuilder = CubeListBuilder.create().texOffs(48, 0).addBox(-1.0F, 0.0F, -1.0F, 2.0F, 18.0F, 2.0F);
 
-        for(int k = 0; k < 8; ++k) {
+        for (int k = 0; k < 8; k++) {
             double d0 = (double)k * Math.PI * 2.0 / 8.0;
             float f = (float)Math.cos(d0) * 5.0F;
             float f1 = 15.0F;
@@ -52,7 +_,7 @@
 
     @Override
     public void setupAnim(T p_103878_, float p_103879_, float p_103880_, float p_103881_, float p_103882_, float p_103883_) {
-        for(ModelPart modelpart : this.tentacles) {
+        for (ModelPart modelpart : this.tentacles) {
             modelpart.xRot = p_103881_;
         }
     }
