--- a/net/minecraft/world/entity/monster/AbstractIllager.java
+++ b/net/minecraft/world/entity/monster/AbstractIllager.java
@@ -32,10 +_,8 @@
     public boolean isAlliedTo(Entity p_320563_) {
         if (super.isAlliedTo(p_320563_)) {
             return true;
-        } else if (!p_320563_.getType().is(EntityTypeTags.ILLAGER_FRIENDS)) {
-            return false;
         } else {
-            return this.getTeam() == null && p_320563_.getTeam() == null;
+            return !p_320563_.getType().is(EntityTypeTags.ILLAGER_FRIENDS) ? false : this.getTeam() == null && p_320563_.getTeam() == null;
         }
     }
 
