--- a/net/minecraft/world/entity/monster/CrossbowAttackMob.java
+++ b/net/minecraft/world/entity/monster/CrossbowAttackMob.java
@@ -16,7 +_,7 @@
     void onCrossbowAttackPerformed();
 
     default void performCrossbowAttack(LivingEntity body, float crossbowPower) {
-        InteractionHand interactionhand = ProjectileUtil.getWeaponHoldingHand(body, Items.CROSSBOW);
+        InteractionHand interactionhand = ProjectileUtil.getWeaponHoldingHand(body, item -> item instanceof CrossbowItem);
         ItemStack itemstack = body.getItemInHand(interactionhand);
         if (itemstack.getItem() instanceof CrossbowItem crossbowitem) {
             crossbowitem.performShooting(
