--- a/net/minecraft/world/entity/monster/Endermite.java
+++ b/net/minecraft/world/entity/monster/Endermite.java
@@ -103,7 +_,7 @@
     public void aiStep() {
         super.aiStep();
         if (this.level().isClientSide) {
-            for(int i = 0; i < 2; ++i) {
+            for (int i = 0; i < 2; i++) {
                 this.level()
                     .addParticle(
                         ParticleTypes.PORTAL,
@@ -117,7 +_,7 @@
             }
         } else {
             if (!this.isPersistenceRequired()) {
-                ++this.life;
+                this.life++;
             }
 
             if (this.life >= 2400) {
