--- a/net/minecraft/world/entity/monster/EnderMan.java
+++ b/net/minecraft/world/entity/monster/EnderMan.java
@@ -82,7 +_,7 @@
     private int targetChangeTime;
     private static final UniformInt PERSISTENT_ANGER_TIME = TimeUtil.rangeOfSeconds(20, 39);
     private long persistentAngerEndTime;
-    private @Nullable EntityReference<Entity> persistentAngerTarget;
+    private @Nullable EntityReference<LivingEntity> persistentAngerTarget;
 
     public EnderMan(EntityType<? extends EnderMan> type, Level level) {
         super(type, level);
@@ -99,7 +_,7 @@
         this.goalSelector.addGoal(8, new RandomLookAroundGoal(this));
         this.goalSelector.addGoal(10, new EnderMan.EndermanLeaveBlockGoal(this));
         this.goalSelector.addGoal(11, new EnderMan.EndermanTakeBlockGoal(this));
-        this.targetSelector.addGoal(1, new EnderMan.EndermanLookForPlayerGoal(this, this::isAngryAt));
+        this.targetSelector.addGoal(1, new EnderMan.EndermanLookForPlayerGoal(this, (target, level) -> target instanceof LivingEntity livingTarget && this.isAngryAt(livingTarget, level)));
         this.targetSelector.addGoal(2, new HurtByTargetGoal(this));
         this.targetSelector.addGoal(3, new NearestAttackableTargetGoal<>(this, Endermite.class, true, false));
         this.targetSelector.addGoal(4, new ResetUniversalAngerTargetGoal<>(this, false));
@@ -120,8 +_,7 @@
     }
 
     @Override
-    public void setTarget(@Nullable Entity target) {
-        super.setTarget(target);
+    public void setTarget(@Nullable LivingEntity target) {
         AttributeInstance movementSpeed = this.getAttribute(Attributes.MOVEMENT_SPEED);
         if (target == null) {
             this.targetChangeTime = 0;
@@ -135,6 +_,7 @@
                 movementSpeed.addTransientModifier(SPEED_MODIFIER_ATTACKING);
             }
         }
+        super.setTarget(target); //Forge: Moved down to allow event handlers to write data manager values.
     }
 
     @Override
@@ -161,12 +_,12 @@
     }
 
     @Override
-    public void setPersistentAngerTarget(@Nullable EntityReference<Entity> persistentAngerTarget) {
+    public void setPersistentAngerTarget(@Nullable EntityReference<LivingEntity> persistentAngerTarget) {
         this.persistentAngerTarget = persistentAngerTarget;
     }
 
     @Override
-    public @Nullable EntityReference<Entity> getPersistentAngerTarget() {
+    public @Nullable EntityReference<LivingEntity> getPersistentAngerTarget() {
         return this.persistentAngerTarget;
     }
 
@@ -207,7 +_,7 @@
     }
 
     private boolean isBeingStaredBy(Player player) {
-        return !LivingEntity.PLAYER_NOT_WEARING_DISGUISE_ITEM.test(player) ? false : this.isLookingAtMe(player, 0.025, true, false, this.getEyeY());
+        return !LivingEntity.PLAYER_NOT_WEARING_DISGUISE_ITEM_FOR_TARGET.test(player, this) ? false : this.isLookingAtMe(player, 0.025, true, false, this.getEyeY()) && !net.neoforged.neoforge.common.CommonHooks.shouldSuppressEnderManAnger(this, player);
     }
 
     @Override
@@ -400,7 +_,7 @@
 
     private static class EndermanFreezeWhenLookedAt extends Goal {
         private final EnderMan enderman;
-        private @Nullable Entity target;
+        private @Nullable LivingEntity target;
 
         public EndermanFreezeWhenLookedAt(EnderMan enderman) {
             this.enderman = enderman;
@@ -441,7 +_,7 @@
             if (this.enderman.getCarriedBlock() == null) {
                 return false;
             } else {
-                return !getServerLevel(this.enderman).getGameRules().get(GameRules.MOB_GRIEFING)
+                return !net.neoforged.neoforge.event.EventHooks.canEntityGrief(getServerLevel(this.enderman.level()), this.enderman)
                     ? false
                     : this.enderman.getRandom().nextInt(reducedTickDelay(2000)) == 0;
             }
@@ -461,7 +_,7 @@
             BlockState carried = this.enderman.getCarriedBlock();
             if (carried != null) {
                 carried = Block.updateFromNeighbourShapes(carried, this.enderman.level(), pos);
-                if (this.canPlaceBlock(level, pos, carried, targetState, belowState, below)) {
+                if (this.canPlaceBlock(level, pos, carried, targetState, belowState, below) && !net.neoforged.neoforge.event.EventHooks.onBlockPlace(enderman, net.neoforged.neoforge.common.util.BlockSnapshot.create(level.dimension(), level, below), net.minecraft.core.Direction.UP)) {
                     level.setBlock(pos, carried, 3);
                     level.gameEvent(GameEvent.BLOCK_PLACE, pos, GameEvent.Context.of(this.enderman, carried));
                     this.enderman.setCarriedBlock(null);
@@ -473,6 +_,7 @@
             return targetState.isAir()
                 && !belowState.isAir()
                 && !belowState.is(Blocks.BEDROCK)
+                && !belowState.is(net.neoforged.neoforge.common.Tags.Blocks.ENDERMAN_PLACE_ON_BLACKLIST)
                 && belowState.isCollisionShapeFullBlock(level, below)
                 && carried.canSurvive(level, pos)
                 && level.getEntities(this.enderman, AABB.unitCubeFromLowerCorner(Vec3.atLowerCornerOf(pos))).isEmpty();
@@ -491,7 +_,7 @@
         public EndermanLookForPlayerGoal(EnderMan enderman, TargetingConditions.@Nullable Selector isAngryAt) {
             super(enderman, Player.class, 10, false, false, isAngryAt);
             this.enderman = enderman;
-            this.isAngerInducing = (target, level) -> (enderman.isBeingStaredBy((Player)target) || enderman.isAngryAt(target, level))
+            this.isAngerInducing = (target, level) -> (enderman.isBeingStaredBy((Player)target) || (target instanceof LivingEntity livingTarget && enderman.isAngryAt(livingTarget, level)))
                 && !enderman.hasIndirectPassenger(target);
             this.startAggroTargetConditions = TargetingConditions.forCombat().range(this.getFollowDistance()).selector(this.isAngerInducing);
         }
@@ -583,7 +_,7 @@
             if (this.enderman.getCarriedBlock() != null) {
                 return false;
             } else {
-                return !getServerLevel(this.enderman).getGameRules().get(GameRules.MOB_GRIEFING)
+                return !net.neoforged.neoforge.event.EventHooks.canEntityGrief(getServerLevel(this.enderman.level()), this.enderman)
                     ? false
                     : this.enderman.getRandom().nextInt(reducedTickDelay(20)) == 0;
             }
