--- a/net/minecraft/world/entity/monster/Guardian.java
+++ b/net/minecraft/world/entity/monster/Guardian.java
@@ -118,7 +_,7 @@
         return this.entityData.get(DATA_ID_ATTACK_TARGET) != 0;
     }
 
-    public @Nullable Entity getActiveAttackTarget() {
+    public @Nullable LivingEntity getActiveAttackTarget() {
         if (!this.hasActiveAttackTarget()) {
             return null;
         } else if (this.level().isClientSide()) {
@@ -431,7 +_,7 @@
         }
 
         @Override
-        public boolean test(@Nullable Entity target, ServerLevel level) {
+        public boolean test(@Nullable LivingEntity target, ServerLevel level) {
             return (target instanceof Player || target instanceof Squid || target instanceof Axolotl) && target.distanceToSqr(this.guardian) > 9.0;
         }
     }
