--- a/net/minecraft/world/entity/monster/Illusioner.java
+++ b/net/minecraft/world/entity/monster/Illusioner.java
@@ -48,7 +_,7 @@
         this.xpReward = 5;
         this.clientSideIllusionOffsets = new Vec3[2][4];
 
-        for(int i = 0; i < 4; ++i) {
+        for (int i = 0; i < 4; i++) {
             this.clientSideIllusionOffsets[0][i] = Vec3.ZERO;
             this.clientSideIllusionOffsets[1][i] = Vec3.ZERO;
         }
@@ -90,7 +_,7 @@
     public void aiStep() {
         super.aiStep();
         if (this.level().isClientSide && this.isInvisible()) {
-            --this.clientSideIllusionTicks;
+            this.clientSideIllusionTicks--;
             if (this.clientSideIllusionTicks < 0) {
                 this.clientSideIllusionTicks = 0;
             }
@@ -100,7 +_,7 @@
                 float f = -6.0F;
                 int j = 13;
 
-                for(int k = 0; k < 4; ++k) {
+                for (int k = 0; k < 4; k++) {
                     this.clientSideIllusionOffsets[0][k] = this.clientSideIllusionOffsets[1][k];
                     this.clientSideIllusionOffsets[1][k] = new Vec3(
                         (double)(-6.0F + (float)this.random.nextInt(13)) * 0.5,
@@ -109,7 +_,7 @@
                     );
                 }
 
-                for(int l = 0; l < 16; ++l) {
+                for (int l = 0; l < 16; l++) {
                     this.level().addParticle(ParticleTypes.CLOUD, this.getRandomX(0.5), this.getRandomY(), this.getZ(0.5), 0.0, 0.0, 0.0);
                 }
 
@@ -118,7 +_,7 @@
             } else if (this.hurtTime == this.hurtDuration - 1) {
                 this.clientSideIllusionTicks = 3;
 
-                for(int i = 0; i < 4; ++i) {
+                for (int i = 0; i < 4; i++) {
                     this.clientSideIllusionOffsets[0][i] = this.clientSideIllusionOffsets[1][i];
                     this.clientSideIllusionOffsets[1][i] = new Vec3(0.0, 0.0, 0.0);
                 }
@@ -139,7 +_,7 @@
             d0 = Math.pow(d0, 0.25);
             Vec3[] avec3 = new Vec3[4];
 
-            for(int i = 0; i < 4; ++i) {
+            for (int i = 0; i < 4; i++) {
                 avec3[i] = this.clientSideIllusionOffsets[1][i].scale(1.0 - d0).add(this.clientSideIllusionOffsets[0][i].scale(d0));
             }
 
@@ -173,8 +_,10 @@
 
     @Override
     public void performRangedAttack(LivingEntity p_32918_, float p_32919_) {
-        ItemStack itemstack = this.getProjectile(this.getItemInHand(ProjectileUtil.getWeaponHoldingHand(this, Items.BOW)));
+        ItemStack itemstack = this.getProjectile(this.getItemInHand(ProjectileUtil.getWeaponHoldingHand(this, item -> item instanceof net.minecraft.world.item.BowItem)));
         AbstractArrow abstractarrow = ProjectileUtil.getMobArrow(this, itemstack, p_32919_);
+        if (this.getMainHandItem().getItem() instanceof net.minecraft.world.item.BowItem)
+            abstractarrow = ((net.minecraft.world.item.BowItem)this.getMainHandItem().getItem()).customArrow(abstractarrow, itemstack);
         double d0 = p_32918_.getX() - this.getX();
         double d1 = p_32918_.getY(0.3333333333333333) - abstractarrow.getY();
         double d2 = p_32918_.getZ() - this.getZ();
@@ -202,10 +_,10 @@
                 return false;
             } else if (Illusioner.this.getTarget() == null) {
                 return false;
-            } else if (Illusioner.this.getTarget().getId() == this.lastTargetId) {
-                return false;
             } else {
-                return Illusioner.this.level().getCurrentDifficultyAt(Illusioner.this.blockPosition()).isHarderThan((float)Difficulty.NORMAL.ordinal());
+                return Illusioner.this.getTarget().getId() == this.lastTargetId
+                    ? false
+                    : Illusioner.this.level().getCurrentDifficultyAt(Illusioner.this.blockPosition()).isHarderThan((float)Difficulty.NORMAL.ordinal());
             }
         }
 
@@ -247,11 +_,7 @@
     class IllusionerMirrorSpellGoal extends SpellcasterIllager.SpellcasterUseSpellGoal {
         @Override
         public boolean canUse() {
-            if (!super.canUse()) {
-                return false;
-            } else {
-                return !Illusioner.this.hasEffect(MobEffects.INVISIBILITY);
-            }
+            return !super.canUse() ? false : !Illusioner.this.hasEffect(MobEffects.INVISIBILITY);
         }
 
         @Override
