--- a/net/minecraft/world/entity/monster/SpellcasterIllager.java
+++ b/net/minecraft/world/entity/monster/SpellcasterIllager.java
@@ -54,11 +_,7 @@
     }
 
     public boolean isCastingSpell() {
-        if (this.level().isClientSide) {
-            return this.entityData.get(DATA_SPELL_CASTING_ID) > 0;
-        } else {
-            return this.spellCastingTickCount > 0;
-        }
+        return this.level().isClientSide ? this.entityData.get(DATA_SPELL_CASTING_ID) > 0 : this.spellCastingTickCount > 0;
     }
 
     public void setIsCastingSpell(SpellcasterIllager.IllagerSpell p_33728_) {
@@ -74,7 +_,7 @@
     protected void customServerAiStep() {
         super.customServerAiStep();
         if (this.spellCastingTickCount > 0) {
-            --this.spellCastingTickCount;
+            this.spellCastingTickCount--;
         }
     }
 
@@ -186,10 +_,8 @@
             LivingEntity livingentity = SpellcasterIllager.this.getTarget();
             if (livingentity == null || !livingentity.isAlive()) {
                 return false;
-            } else if (SpellcasterIllager.this.isCastingSpell()) {
-                return false;
             } else {
-                return SpellcasterIllager.this.tickCount >= this.nextAttackTickCount;
+                return SpellcasterIllager.this.isCastingSpell() ? false : SpellcasterIllager.this.tickCount >= this.nextAttackTickCount;
             }
         }
 
@@ -214,7 +_,7 @@
 
         @Override
         public void tick() {
-            --this.attackWarmupDelay;
+            this.attackWarmupDelay--;
             if (this.attackWarmupDelay == 0) {
                 this.performSpellCasting();
                 SpellcasterIllager.this.playSound(SpellcasterIllager.this.getCastingSoundEvent(), 1.0F, 1.0F);
