--- a/net/minecraft/world/entity/monster/Vex.java
+++ b/net/minecraft/world/entity/monster/Vex.java
@@ -229,11 +_,12 @@
         @Override
         public boolean canUse() {
             LivingEntity livingentity = Vex.this.getTarget();
-            if (livingentity != null && livingentity.isAlive() && !Vex.this.getMoveControl().hasWanted() && Vex.this.random.nextInt(reducedTickDelay(7)) == 0) {
-                return Vex.this.distanceToSqr(livingentity) > 4.0;
-            } else {
-                return false;
-            }
+            return livingentity != null
+                    && livingentity.isAlive()
+                    && !Vex.this.getMoveControl().hasWanted()
+                    && Vex.this.random.nextInt(reducedTickDelay(7)) == 0
+                ? Vex.this.distanceToSqr(livingentity) > 4.0
+                : false;
         }
 
         @Override
@@ -352,7 +_,7 @@
                 blockpos = Vex.this.blockPosition();
             }
 
-            for(int i = 0; i < 3; ++i) {
+            for (int i = 0; i < 3; i++) {
                 BlockPos blockpos1 = blockpos.offset(Vex.this.random.nextInt(15) - 7, Vex.this.random.nextInt(11) - 5, Vex.this.random.nextInt(15) - 7);
                 if (Vex.this.level().isEmptyBlock(blockpos1)) {
                     Vex.this.moveControl
