--- a/net/minecraft/world/entity/monster/Zoglin.java
+++ b/net/minecraft/world/entity/monster/Zoglin.java
@@ -43,7 +_,6 @@
 import net.minecraft.world.entity.ai.memory.NearestVisibleLivingEntities;
 import net.minecraft.world.entity.ai.sensing.Sensor;
 import net.minecraft.world.entity.ai.sensing.SensorType;
-import net.minecraft.world.entity.livingblock.LivingBlock;
 import net.minecraft.world.entity.monster.hoglin.HoglinBase;
 import net.minecraft.world.entity.schedule.Activity;
 import net.minecraft.world.level.Level;
@@ -166,13 +_,13 @@
 
     @Override
     public boolean doHurtTarget(ServerLevel level, Entity target) {
-        if (!(target instanceof LivingEntity) && !(target instanceof LivingBlock)) {
-            return false;
-        } else {
+        if (target instanceof LivingEntity livingEntity) {
             this.attackAnimationRemainingTicks = 10;
             level.broadcastEntityEvent(this, (byte)4);
             this.makeSound(SoundEvents.ZOGLIN_ATTACK);
-            return HoglinBase.hurtAndThrowTarget(level, this, target);
+            return HoglinBase.hurtAndThrowTarget(level, this, livingEntity);
+        } else {
+            return false;
         }
     }
 
@@ -205,9 +_,9 @@
         return wasHurt;
     }
 
-    private void setAttackTarget(Entity target) {
+    public void setAttackTarget(Entity target) {
         this.brain.eraseMemory(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE);
-        this.brain.setMemoryWithExpiry(MemoryModuleType.ATTACK_TARGET, target, 200L);
+        this.brain.setMemoryWithExpiry(MemoryModuleType.ATTACK_TARGET, (LivingEntity)target, 200L);
     }
 
     @Override
@@ -301,7 +_,7 @@
     }
 
     @Override
-    public @Nullable Entity getTarget() {
+    public @Nullable LivingEntity getTarget() {
         return this.getTargetFromBrain();
     }
 
