--- a/net/minecraft/world/entity/monster/ZombieVillager.java
+++ b/net/minecraft/world/entity/monster/ZombieVillager.java
@@ -90,11 +_,7 @@
             .ifPresent(p_204072_ -> p_34397_.put("VillagerData", p_204072_));
         if (this.tradeOffers != null) {
             p_34397_.put(
-                "Offers",
-                Util.getOrThrow(
-                    MerchantOffers.CODEC.encodeStart(this.registryAccess().createSerializationContext(NbtOps.INSTANCE), this.tradeOffers),
-                    IllegalStateException::new
-                )
+                "Offers", MerchantOffers.CODEC.encodeStart(this.registryAccess().createSerializationContext(NbtOps.INSTANCE), this.tradeOffers).getOrThrow()
             );
         }
 
@@ -143,7 +_,7 @@
         if (!this.level().isClientSide && this.isAlive() && this.isConverting()) {
             int i = this.getConversionProgress();
             this.villagerConversionTime -= i;
-            if (this.villagerConversionTime <= 0) {
+            if (this.villagerConversionTime <= 0 && net.neoforged.neoforge.event.EventHooks.canLivingConvert(this, EntityType.VILLAGER, (timer) -> this.villagerConversionTime = timer)) {
                 this.finishConversion((ServerLevel)this.level());
             }
         }
@@ -217,7 +_,7 @@
     private void finishConversion(ServerLevel p_34399_) {
         Villager villager = this.convertTo(EntityType.VILLAGER, false);
 
-        for(EquipmentSlot equipmentslot : EquipmentSlot.values()) {
+        for (EquipmentSlot equipmentslot : EquipmentSlot.values()) {
             ItemStack itemstack = this.getItemBySlot(equipmentslot);
             if (!itemstack.isEmpty()) {
                 if (EnchantmentHelper.hasBindingCurse(itemstack)) {
@@ -255,6 +_,7 @@
         if (!this.isSilent()) {
             p_34399_.levelEvent(null, 1027, this.blockPosition(), 0);
         }
+        net.neoforged.neoforge.event.EventHooks.onLivingConvert(this, villager);
     }
 
     private int getConversionProgress() {
@@ -263,16 +_,16 @@
             int j = 0;
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for(int k = (int)this.getX() - 4; k < (int)this.getX() + 4 && j < 14; ++k) {
-                for(int l = (int)this.getY() - 4; l < (int)this.getY() + 4 && j < 14; ++l) {
-                    for(int i1 = (int)this.getZ() - 4; i1 < (int)this.getZ() + 4 && j < 14; ++i1) {
+            for (int k = (int)this.getX() - 4; k < (int)this.getX() + 4 && j < 14; k++) {
+                for (int l = (int)this.getY() - 4; l < (int)this.getY() + 4 && j < 14; l++) {
+                    for (int i1 = (int)this.getZ() - 4; i1 < (int)this.getZ() + 4 && j < 14; i1++) {
                         BlockState blockstate = this.level().getBlockState(blockpos$mutableblockpos.set(k, l, i1));
                         if (blockstate.is(Blocks.IRON_BARS) || blockstate.getBlock() instanceof BedBlock) {
                             if (this.random.nextFloat() < 0.3F) {
-                                ++i;
+                                i++;
                             }
 
-                            ++j;
+                            j++;
                         }
                     }
                 }
