--- a/net/minecraft/client/renderer/block/BlockModelShaper.java
+++ b/net/minecraft/client/renderer/block/BlockModelShaper.java
@@ -22,8 +_,15 @@
         this.modelManager = p_110880_;
     }
 
+    @Deprecated
     public TextureAtlasSprite getParticleIcon(BlockState p_110883_) {
-        return this.getBlockModel(p_110883_).getParticleIcon();
+        return this.getBlockModel(p_110883_).getParticleIcon(net.neoforged.neoforge.client.model.data.ModelData.EMPTY);
+    }
+
+    public TextureAtlasSprite getTexture(BlockState p_110883_, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
+        var data = level.getModelDataManager().getAtOrEmpty(pos);
+        BakedModel model = this.getBlockModel(p_110883_);
+        return model.getParticleIcon(model.getModelData(level, pos, p_110883_, data));
     }
 
     public BakedModel getBlockModel(BlockState p_110894_) {
@@ -54,7 +_,7 @@
     public static String statePropertiesToString(Map<Property<?>, Comparable<?>> p_110888_) {
         StringBuilder stringbuilder = new StringBuilder();
 
-        for(Entry<Property<?>, Comparable<?>> entry : p_110888_.entrySet()) {
+        for (Entry<Property<?>, Comparable<?>> entry : p_110888_.entrySet()) {
             if (stringbuilder.length() != 0) {
                 stringbuilder.append(',');
             }
