--- a/net/minecraft/client/renderer/block/LiquidBlockRenderer.java
+++ b/net/minecraft/client/renderer/block/LiquidBlockRenderer.java
@@ -38,12 +_,17 @@
         this.waterStill = materials.get(ModelBakery.WATER_STILL);
         this.waterFlowing = materials.get(ModelBakery.WATER_FLOW);
         this.waterOverlay = materials.get(ModelBakery.WATER_OVERLAY);
+        net.neoforged.neoforge.client.textures.FluidSpriteCache.reload();
     }
 
     private static boolean isNeighborSameFluid(FluidState fluidState, FluidState neighborFluidState) {
         return neighborFluidState.getType().isSame(fluidState.getType());
     }
 
+    private static boolean isNeighborStateHidingOverlay(FluidState selfState, BlockState otherState, Direction neighborFace) {
+        return otherState.shouldHideAdjacentFluidFace(neighborFace, selfState);
+    }
+
     private static boolean isFaceOccludedByState(Direction direction, float height, BlockState state) {
         VoxelShape voxelshape = state.getFaceOcclusionShape(direction.getOpposite());
         if (voxelshape == Shapes.empty()) {
@@ -65,15 +_,22 @@
         return isFaceOccludedByState(direction.getOpposite(), 1.0F, state);
     }
 
+    /** @deprecated Neo: use overload that accepts BlockState */
+    @Deprecated
     public static boolean shouldRenderFace(FluidState fluidState, BlockState blockState, Direction direction, FluidState neighborFluidState) {
         return !isFaceOccludedBySelf(blockState, direction) && !isNeighborSameFluid(fluidState, neighborFluidState);
     }
 
+    public static boolean shouldRenderFace(FluidState fluidState, BlockState selfState, Direction direction, BlockState otherState) {
+        return !isFaceOccludedBySelf(selfState, direction) && !isNeighborStateHidingOverlay(fluidState, otherState, direction.getOpposite());
+    }
+
     public void tesselate(BlockAndTintGetter level, BlockPos pos, VertexConsumer builder, BlockState blockState, FluidState fluidState) {
-        boolean flag = fluidState.is(FluidTags.LAVA);
-        TextureAtlasSprite textureatlassprite = flag ? this.lavaStill : this.waterStill;
-        TextureAtlasSprite textureatlassprite1 = flag ? this.lavaFlowing : this.waterFlowing;
-        int i = flag ? 16777215 : BiomeColors.getAverageWaterColor(level, pos);
+        TextureAtlasSprite[] atextureatlassprite = net.neoforged.neoforge.client.textures.FluidSpriteCache.getFluidSprites(level, pos, fluidState);
+        TextureAtlasSprite textureatlassprite = atextureatlassprite[0]; // Neo: Custom still sprite
+        TextureAtlasSprite textureatlassprite1 = atextureatlassprite[1]; // Neo: Custom flowing sprite
+        int i = net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions.of(fluidState).getTintColor(fluidState, level, pos);
+        float alpha = (i >> 24 & 255) / 255.0F;
         float f = (i >> 16 & 0xFF) / 255.0F;
         float f1 = (i >> 8 & 0xFF) / 255.0F;
         float f2 = (i & 0xFF) / 255.0F;
@@ -89,12 +_,12 @@
         FluidState fluidstate4 = blockstate4.getFluidState();
         BlockState blockstate5 = level.getBlockState(pos.relative(Direction.EAST));
         FluidState fluidstate5 = blockstate5.getFluidState();
-        boolean flag1 = !isNeighborSameFluid(fluidState, fluidstate1);
-        boolean flag2 = shouldRenderFace(fluidState, blockState, Direction.DOWN, fluidstate) && !isFaceOccludedByNeighbor(Direction.DOWN, 0.8888889F, blockstate);
-        boolean flag3 = shouldRenderFace(fluidState, blockState, Direction.NORTH, fluidstate2);
-        boolean flag4 = shouldRenderFace(fluidState, blockState, Direction.SOUTH, fluidstate3);
-        boolean flag5 = shouldRenderFace(fluidState, blockState, Direction.WEST, fluidstate4);
-        boolean flag6 = shouldRenderFace(fluidState, blockState, Direction.EAST, fluidstate5);
+        boolean flag1 = !isNeighborStateHidingOverlay(fluidState, blockstate1, Direction.DOWN);
+        boolean flag2 = shouldRenderFace(fluidState, blockState, Direction.DOWN, blockstate) && !isFaceOccludedByNeighbor(Direction.DOWN, 0.8888889F, blockstate);
+        boolean flag3 = shouldRenderFace(fluidState, blockState, Direction.NORTH, blockstate2);
+        boolean flag4 = shouldRenderFace(fluidState, blockState, Direction.SOUTH, blockstate3);
+        boolean flag5 = shouldRenderFace(fluidState, blockState, Direction.WEST, blockstate4);
+        boolean flag6 = shouldRenderFace(fluidState, blockState, Direction.EAST, blockstate5);
         if (flag1 || flag2 || flag6 || flag5 || flag3 || flag4) {
             float f3 = level.getShade(Direction.DOWN, true);
             float f4 = level.getShade(Direction.UP, true);
@@ -169,15 +_,15 @@
                 float f54 = f4 * f;
                 float f55 = f4 * f1;
                 float f56 = f4 * f2;
-                this.vertex(builder, f37 + 0.0F, f38 + f8, f39 + 0.0F, f54, f55, f56, f17, f21, l);
-                this.vertex(builder, f37 + 0.0F, f38 + f10, f39 + 1.0F, f54, f55, f56, f18, f22, l);
-                this.vertex(builder, f37 + 1.0F, f38 + f9, f39 + 1.0F, f54, f55, f56, f19, f23, l);
-                this.vertex(builder, f37 + 1.0F, f38 + f7, f39 + 0.0F, f54, f55, f56, f20, f24, l);
+                this.vertex(builder, f37 + 0.0F, f38 + f8, f39 + 0.0F, f54, f55, f56, alpha, f17, f21, l);
+                this.vertex(builder, f37 + 0.0F, f38 + f10, f39 + 1.0F, f54, f55, f56, alpha, f18, f22, l);
+                this.vertex(builder, f37 + 1.0F, f38 + f9, f39 + 1.0F, f54, f55, f56, alpha, f19, f23, l);
+                this.vertex(builder, f37 + 1.0F, f38 + f7, f39 + 0.0F, f54, f55, f56, alpha, f20, f24, l);
                 if (fluidState.shouldRenderBackwardUpFace(level, pos.above())) {
-                    this.vertex(builder, f37 + 0.0F, f38 + f8, f39 + 0.0F, f54, f55, f56, f17, f21, l);
-                    this.vertex(builder, f37 + 1.0F, f38 + f7, f39 + 0.0F, f54, f55, f56, f20, f24, l);
-                    this.vertex(builder, f37 + 1.0F, f38 + f9, f39 + 1.0F, f54, f55, f56, f19, f23, l);
-                    this.vertex(builder, f37 + 0.0F, f38 + f10, f39 + 1.0F, f54, f55, f56, f18, f22, l);
+                    this.vertex(builder, f37 + 0.0F, f38 + f8, f39 + 0.0F, f54, f55, f56, alpha, f17, f21, l);
+                    this.vertex(builder, f37 + 1.0F, f38 + f7, f39 + 0.0F, f54, f55, f56, alpha, f20, f24, l);
+                    this.vertex(builder, f37 + 1.0F, f38 + f9, f39 + 1.0F, f54, f55, f56, alpha, f19, f23, l);
+                    this.vertex(builder, f37 + 0.0F, f38 + f10, f39 + 1.0F, f54, f55, f56, alpha, f18, f22, l);
                 }
             }
 
@@ -190,10 +_,10 @@
                 float f47 = f3 * f;
                 float f49 = f3 * f1;
                 float f51 = f3 * f2;
-                this.vertex(builder, f37, f38 + f16, f39 + 1.0F, f47, f49, f51, f41, f44, k);
-                this.vertex(builder, f37, f38 + f16, f39, f47, f49, f51, f41, f43, k);
-                this.vertex(builder, f37 + 1.0F, f38 + f16, f39, f47, f49, f51, f42, f43, k);
-                this.vertex(builder, f37 + 1.0F, f38 + f16, f39 + 1.0F, f47, f49, f51, f42, f44, k);
+                this.vertex(builder, f37, f38 + f16, f39 + 1.0F, f47, f49, f51, alpha, f41, f44, k);
+                this.vertex(builder, f37, f38 + f16, f39, f47, f49, f51, alpha, f41, f43, k);
+                this.vertex(builder, f37 + 1.0F, f38 + f16, f39, f47, f49, f51, alpha, f42, f43, k);
+                this.vertex(builder, f37 + 1.0F, f38 + f16, f39 + 1.0F, f47, f49, f51, alpha, f42, f44, k);
             }
 
             int j = this.getLightColor(level, pos);
@@ -247,10 +_,9 @@
                 if (flag7 && !isFaceOccludedByNeighbor(direction, Math.max(f45, f46), level.getBlockState(pos.relative(direction)))) {
                     BlockPos blockpos = pos.relative(direction);
                     TextureAtlasSprite textureatlassprite2 = textureatlassprite1;
-                    if (!flag) {
-                        Block block = level.getBlockState(blockpos).getBlock();
-                        if (block instanceof HalfTransparentBlock || block instanceof LeavesBlock) {
-                            textureatlassprite2 = this.waterOverlay;
+                    if (atextureatlassprite[2] != null) {
+                        if (level.getBlockState(blockpos).shouldDisplayFluidOverlay(level, blockpos, fluidState)) {
+                            textureatlassprite2 = atextureatlassprite[2];
                         }
                     }
 
@@ -263,15 +_,15 @@
                     float f34 = f4 * f33 * f;
                     float f35 = f4 * f33 * f1;
                     float f36 = f4 * f33 * f2;
-                    this.vertex(builder, f48, f38 + f45, f50, f34, f35, f36, f57, f30, j);
-                    this.vertex(builder, f52, f38 + f46, f53, f34, f35, f36, f29, f31, j);
-                    this.vertex(builder, f52, f38 + f16, f53, f34, f35, f36, f29, f32, j);
-                    this.vertex(builder, f48, f38 + f16, f50, f34, f35, f36, f57, f32, j);
-                    if (textureatlassprite2 != this.waterOverlay) {
-                        this.vertex(builder, f48, f38 + f16, f50, f34, f35, f36, f57, f32, j);
-                        this.vertex(builder, f52, f38 + f16, f53, f34, f35, f36, f29, f32, j);
-                        this.vertex(builder, f52, f38 + f46, f53, f34, f35, f36, f29, f31, j);
-                        this.vertex(builder, f48, f38 + f45, f50, f34, f35, f36, f57, f30, j);
+                    this.vertex(builder, f48, f38 + f45, f50, f34, f35, f36, alpha, f57, f30, j);
+                    this.vertex(builder, f52, f38 + f46, f53, f34, f35, f36, alpha, f29, f31, j);
+                    this.vertex(builder, f52, f38 + f16, f53, f34, f35, f36, alpha, f29, f32, j);
+                    this.vertex(builder, f48, f38 + f16, f50, f34, f35, f36, alpha, f57, f32, j);
+                    if (textureatlassprite2 != atextureatlassprite[2]) { // Neo: use custom fluid's overlay texture
+                        this.vertex(builder, f48, f38 + f16, f50, f34, f35, f36, alpha, f57, f32, j);
+                        this.vertex(builder, f52, f38 + f16, f53, f34, f35, f36, alpha, f29, f32, j);
+                        this.vertex(builder, f52, f38 + f46, f53, f34, f35, f36, alpha, f29, f31, j);
+                        this.vertex(builder, f48, f38 + f45, f50, f34, f35, f36, alpha, f57, f30, j);
                     }
                 }
             }
@@ -312,6 +_,26 @@
     private float getHeight(BlockAndTintGetter level, Fluid fluidType, BlockPos pos) {
         BlockState blockstate = level.getBlockState(pos);
         return this.getHeight(level, fluidType, pos, blockstate, blockstate.getFluidState());
+    }
+
+    private void vertex(
+            VertexConsumer builder,
+            float x,
+            float y,
+            float z,
+            float red,
+            float green,
+            float blue,
+            float alpha,
+            float u,
+            float v,
+            int light
+    ) {
+        builder.addVertex(x, y, z)
+                .setColor(red, green, blue, alpha)
+                .setUv(u, v)
+                .setLight(light)
+                .setNormal(0.0F, 1.0F, 0.0F);
     }
 
     private float getHeight(BlockAndTintGetter level, Fluid fluidType, BlockPos pos, BlockState state, FluidState fluidState) {
