--- a/net/minecraft/client/renderer/block/MovingBlockRenderState.java
+++ b/net/minecraft/client/renderer/block/MovingBlockRenderState.java
@@ -23,6 +_,7 @@
     public @Nullable Holder<Biome> biome;
     public CardinalLighting cardinalLighting = CardinalLighting.DEFAULT;
     public LevelLightEngine lightEngine = LevelLightEngine.EMPTY;
+    public net.neoforged.neoforge.model.data.ModelData modelData = net.neoforged.neoforge.model.data.ModelData.EMPTY;
 
     @Override
     public CardinalLighting cardinalLighting() {
