--- a/net/minecraft/world/level/levelgen/feature/featuresize/TwoLayersFeatureSize.java
+++ b/net/minecraft/world/level/levelgen/feature/featuresize/TwoLayersFeatureSize.java
@@ -1,12 +_,13 @@
 package net.minecraft.world.level.levelgen.feature.featuresize;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.OptionalInt;
 
 public class TwoLayersFeatureSize extends FeatureSize {
-    public static final Codec<TwoLayersFeatureSize> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<TwoLayersFeatureSize> CODEC = RecordCodecBuilder.mapCodec(
         p_68356_ -> p_68356_.group(
                     Codec.intRange(0, 81).fieldOf("limit").orElse(1).forGetter(p_161341_ -> p_161341_.limit),
                     Codec.intRange(0, 16).fieldOf("lower_size").orElse(0).forGetter(p_161339_ -> p_161339_.lowerSize),
