--- a/net/minecraft/network/protocol/ProtocolInfoBuilder.java
+++ b/net/minecraft/network/protocol/ProtocolInfoBuilder.java
@@ -44,7 +_,7 @@
     private StreamCodec<ByteBuf, Packet<? super T>> buildPacketCodec(Function<ByteBuf, B> p_320922_, List<ProtocolInfoBuilder.CodecEntry<T, ?, B>> p_320733_) {
         ProtocolCodecBuilder<ByteBuf, T> protocolcodecbuilder = new ProtocolCodecBuilder<>(this.flow);
 
-        for(ProtocolInfoBuilder.CodecEntry<T, ?, B> codecentry : p_320733_) {
+        for (ProtocolInfoBuilder.CodecEntry<T, ?, B> codecentry : p_320733_) {
             codecentry.addToBuilder(protocolcodecbuilder, p_320922_);
         }
 
