--- a/net/minecraft/util/monitoring/jmx/MinecraftServerStatistics.java
+++ b/net/minecraft/util/monitoring/jmx/MinecraftServerStatistics.java
@@ -31,7 +_,7 @@
     private final MBeanInfo mBeanInfo;
     private final Map<String, MinecraftServerStatistics.AttributeDescription> attributeDescriptionByName = Stream.of(
             new MinecraftServerStatistics.AttributeDescription("tickTimes", this::getTickTimes, "Historical tick times (ms)", long[].class),
-            new MinecraftServerStatistics.AttributeDescription("averageTickTime", this::getAverageTickTime, "Current average tick time (ms)", Long.TYPE)
+            new MinecraftServerStatistics.AttributeDescription("averageTickTime", this::getAverageTickTime, "Current average tick time (ms)", long.class)
         )
         .collect(Collectors.toMap(p_18332_ -> p_18332_.name, Function.identity()));
 
@@ -41,7 +_,7 @@
             .values()
             .stream()
             .map(MinecraftServerStatistics.AttributeDescription::asMBeanAttributeInfo)
-            .toArray(p_145923_ -> new MBeanAttributeInfo[p_145923_]);
+            .toArray(MBeanAttributeInfo[]::new);
         this.mBeanInfo = new MBeanInfo(
             MinecraftServerStatistics.class.getSimpleName(), "metrics for dedicated server", ambeanattributeinfo, null, null, new MBeanNotificationInfo[0]
         );
