--- a/net/minecraft/client/renderer/block/model/multipart/KeyValueCondition.java
+++ b/net/minecraft/client/renderer/block/model/multipart/KeyValueCondition.java
@@ -63,7 +_,7 @@
                 String.format(Locale.ROOT, "Unknown value '%s' for property '%s' on '%s' in '%s'", p_111947_, this.key, p_111945_.getOwner(), this.value)
             );
         } else {
-            return p_111951_ -> p_111951_.getValue(p_111946_).equals(optional.get());
+            return p_339295_ -> p_339295_.getValue(p_111946_).equals(optional.get());
         }
     }
 
