--- a/net/minecraft/client/renderer/block/model/multipart/MultiPartModel.java
+++ b/net/minecraft/client/renderer/block/model/multipart/MultiPartModel.java
@@ -37,8 +_,9 @@
         return this.shared.particleIcon;
     }
 
+    // Neo: Implement our overloads (here and below) so child models can have custom logic
     @Override
-    public void collectParts(RandomSource random, List<BlockModelPart> output) {
+    public void collectParts(net.minecraft.world.level.BlockAndTintGetter level, net.minecraft.core.BlockPos pos, BlockState state, RandomSource random, List<BlockModelPart> output) {
         if (this.models == null) {
             this.models = this.shared.selectModels(this.blockState);
         }
@@ -84,7 +_,6 @@
                     selectedModels.set(i);
                 }
             }
-
             return this.subsets.computeIfAbsent(selectedModels, selected -> {
                 Builder<BlockStateModel> result = ImmutableList.builder();
 
