--- a/net/minecraft/client/renderer/block/model/multipart/MultiPart.java
+++ b/net/minecraft/client/renderer/block/model/multipart/MultiPart.java
@@ -48,7 +_,7 @@
     public Set<MultiVariant> getMultiVariants() {
         Set<MultiVariant> set = Sets.newHashSet();
 
-        for(Selector selector : this.selectors) {
+        for (Selector selector : this.selectors) {
             set.add(selector.getVariant());
         }
 
@@ -59,11 +_,10 @@
     public boolean equals(Object p_111984_) {
         if (this == p_111984_) {
             return true;
-        } else if (!(p_111984_ instanceof MultiPart)) {
-            return false;
         } else {
-            MultiPart multipart = (MultiPart)p_111984_;
-            return Objects.equals(this.definition, multipart.definition) && Objects.equals(this.selectors, multipart.selectors);
+            return !(p_111984_ instanceof MultiPart multipart)
+                ? false
+                : Objects.equals(this.definition, multipart.definition) && Objects.equals(this.selectors, multipart.selectors);
         }
     }
 
@@ -87,7 +_,7 @@
     public BakedModel bake(ModelBaker p_249988_, Function<Material, TextureAtlasSprite> p_111972_, ModelState p_111973_, ResourceLocation p_111974_) {
         MultiPartBakedModel.Builder multipartbakedmodel$builder = new MultiPartBakedModel.Builder();
 
-        for(Selector selector : this.getSelectors()) {
+        for (Selector selector : this.getSelectors()) {
             BakedModel bakedmodel = selector.getVariant().bake(p_249988_, p_111972_, p_111973_, p_111974_);
             if (bakedmodel != null) {
                 multipartbakedmodel$builder.add(selector.getPredicate(this.definition), bakedmodel);
@@ -112,7 +_,7 @@
         private List<Selector> getSelectors(JsonDeserializationContext p_111991_, JsonArray p_111992_) {
             List<Selector> list = Lists.newArrayList();
 
-            for(JsonElement jsonelement : p_111992_) {
+            for (JsonElement jsonelement : p_111992_) {
                 list.add(p_111991_.deserialize(jsonelement, Selector.class));
             }
 
