--- a/net/minecraft/server/packs/linkfs/LinkFSPath.java
+++ b/net/minecraft/server/packs/linkfs/LinkFSPath.java
@@ -80,8 +_,7 @@
 
     @Override
     public File toFile() {
-        PathContents pathcontents = this.pathContents;
-        if (pathcontents instanceof PathContents.FileContents pathcontents$filecontents) {
+        if (this.pathContents instanceof PathContents.FileContents pathcontents$filecontents) {
             return pathcontents$filecontents.contents().toFile();
         } else {
             throw new UnsupportedOperationException("Path " + this.pathToString() + " does not represent file");
@@ -139,7 +_,7 @@
         if (p_251923_ >= 0 && p_248807_ <= list.size() && p_251923_ < p_248807_) {
             LinkFSPath linkfspath = null;
 
-            for(int i = p_251923_; i < p_248807_; ++i) {
+            for (int i = p_251923_; i < p_248807_; i++) {
                 linkfspath = this.createRelativePath(linkfspath, list.get(i));
             }
 
@@ -163,7 +_,7 @@
                 if (i > list.size()) {
                     return false;
                 } else {
-                    for(int j = 0; j < i; ++j) {
+                    for (int j = 0; j < i; j++) {
                         if (!list1.get(j).equals(list.get(j))) {
                             return false;
                         }
@@ -192,7 +_,7 @@
                 if (j < 0) {
                     return false;
                 } else {
-                    for(int k = i - 1; k >= 0; --k) {
+                    for (int k = i - 1; k >= 0; k--) {
                         if (!list1.get(k).equals(list.get(j + k))) {
                             return false;
                         }
@@ -218,7 +_,7 @@
     private LinkFSPath resolve(List<String> p_252101_) {
         LinkFSPath linkfspath = this;
 
-        for(String s : p_252101_) {
+        for (String s : p_252101_) {
             linkfspath = linkfspath.resolveName(s);
         }
 
@@ -228,16 +_,13 @@
     LinkFSPath resolveName(String p_249718_) {
         if (isRelativeOrMissing(this.pathContents)) {
             return new LinkFSPath(this.fileSystem, p_249718_, this, this.pathContents);
+        } else if (this.pathContents instanceof PathContents.DirectoryContents pathcontents$directorycontents) {
+            LinkFSPath linkfspath = pathcontents$directorycontents.children().get(p_249718_);
+            return linkfspath != null ? linkfspath : new LinkFSPath(this.fileSystem, p_249718_, this, PathContents.MISSING);
+        } else if (this.pathContents instanceof PathContents.FileContents) {
+            return new LinkFSPath(this.fileSystem, p_249718_, this, PathContents.MISSING);
         } else {
-            PathContents $$2 = this.pathContents;
-            if ($$2 instanceof PathContents.DirectoryContents pathcontents$directorycontents) {
-                LinkFSPath linkfspath = pathcontents$directorycontents.children().get(p_249718_);
-                return linkfspath != null ? linkfspath : new LinkFSPath(this.fileSystem, p_249718_, this, PathContents.MISSING);
-            } else if (this.pathContents instanceof PathContents.FileContents) {
-                return new LinkFSPath(this.fileSystem, p_249718_, this, PathContents.MISSING);
-            } else {
-                throw new AssertionError("All content types should be already handled");
-            }
+            throw new AssertionError("All content types should be already handled");
         }
     }
 
@@ -255,7 +_,7 @@
             if (list.size() >= list1.size()) {
                 throw new IllegalArgumentException();
             } else {
-                for(int i = 0; i < list.size(); ++i) {
+                for (int i = 0; i < list.size(); i++) {
                     if (!list.get(i).equals(list1.get(i))) {
                         throw new IllegalArgumentException();
                     }
@@ -305,10 +_,10 @@
                 boolean flag = this.hasRealContents();
                 if (flag != linkfspath.hasRealContents()) {
                     return false;
-                } else if (flag) {
-                    return this.pathContents == linkfspath.pathContents;
                 } else {
-                    return Objects.equals(this.parent, linkfspath.parent) && Objects.equals(this.name, linkfspath.name);
+                    return flag
+                        ? this.pathContents == linkfspath.pathContents
+                        : Objects.equals(this.parent, linkfspath.parent) && Objects.equals(this.name, linkfspath.name);
                 }
             }
         } else {
@@ -362,14 +_,12 @@
 
     @Nullable
     public Path getTargetPath() {
-        PathContents pathcontents = this.pathContents;
-        return pathcontents instanceof PathContents.FileContents pathcontents$filecontents ? pathcontents$filecontents.contents() : null;
+        return this.pathContents instanceof PathContents.FileContents pathcontents$filecontents ? pathcontents$filecontents.contents() : null;
     }
 
     @Nullable
     public PathContents.DirectoryContents getDirectoryContents() {
-        PathContents pathcontents = this.pathContents;
-        return pathcontents instanceof PathContents.DirectoryContents pathcontents$directorycontents ? pathcontents$directorycontents : null;
+        return this.pathContents instanceof PathContents.DirectoryContents pathcontents$directorycontents ? pathcontents$directorycontents : null;
     }
 
     public BasicFileAttributeView getBasicAttributeView() {
