--- a/net/minecraft/world/level/levelgen/feature/treedecorators/AttachedToLeavesDecorator.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/AttachedToLeavesDecorator.java
@@ -1,6 +_,7 @@
 package net.minecraft.world.level.levelgen.feature.treedecorators;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.HashSet;
@@ -14,36 +_,31 @@
 import net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider;
 
 public class AttachedToLeavesDecorator extends TreeDecorator {
-    public static final Codec<AttachedToLeavesDecorator> CODEC = RecordCodecBuilder.create(
-        p_340289_ -> p_340289_.group(
+    public static final MapCodec<AttachedToLeavesDecorator> CODEC = RecordCodecBuilder.mapCodec(
+        p_225996_ -> p_225996_.group(
                     Codec.floatRange(0.0F, 1.0F).fieldOf("probability").forGetter(p_226014_ -> p_226014_.probability),
-                    Codec.BOOL.optionalFieldOf("use_logs", Boolean.valueOf(false)).forGetter(p_340290_ -> p_340290_.useLogs),
                     Codec.intRange(0, 16).fieldOf("exclusion_radius_xz").forGetter(p_226012_ -> p_226012_.exclusionRadiusXZ),
                     Codec.intRange(0, 16).fieldOf("exclusion_radius_y").forGetter(p_226010_ -> p_226010_.exclusionRadiusY),
-                    Codec.list(BlockStateProvider.CODEC).fieldOf("block_provider").forGetter(p_340291_ -> p_340291_.blockProvider),
+                    BlockStateProvider.CODEC.fieldOf("block_provider").forGetter(p_226008_ -> p_226008_.blockProvider),
                     Codec.intRange(1, 16).fieldOf("required_empty_blocks").forGetter(p_226006_ -> p_226006_.requiredEmptyBlocks),
                     ExtraCodecs.nonEmptyList(Direction.CODEC.listOf()).fieldOf("directions").forGetter(p_225998_ -> p_225998_.directions)
                 )
-                .apply(p_340289_, AttachedToLeavesDecorator::new)
+                .apply(p_225996_, AttachedToLeavesDecorator::new)
     );
     protected final float probability;
     protected final int exclusionRadiusXZ;
     protected final int exclusionRadiusY;
-    protected final List<BlockStateProvider> blockProvider;
+    protected final BlockStateProvider blockProvider;
     protected final int requiredEmptyBlocks;
     protected final List<Direction> directions;
-    protected boolean useLogs;
 
-    public AttachedToLeavesDecorator(
-        float p_225988_, boolean p_341867_, int p_225989_, int p_225990_, List<BlockStateProvider> p_225993_, int p_225992_, List<Direction> p_340505_
-    ) {
+    public AttachedToLeavesDecorator(float p_225988_, int p_225989_, int p_225990_, BlockStateProvider p_225991_, int p_225992_, List<Direction> p_225993_) {
         this.probability = p_225988_;
-        this.useLogs = p_341867_;
         this.exclusionRadiusXZ = p_225989_;
         this.exclusionRadiusY = p_225990_;
-        this.blockProvider = p_225993_;
+        this.blockProvider = p_225991_;
         this.requiredEmptyBlocks = p_225992_;
-        this.directions = p_340505_;
+        this.directions = p_225993_;
     }
 
     @Override
@@ -51,27 +_,24 @@
         Set<BlockPos> set = new HashSet<>();
         RandomSource randomsource = p_226000_.random();
 
-        for(BlockPos blockpos : this.useLogs ? Util.shuffledCopy(p_226000_.logs(), randomsource) : Util.shuffledCopy(p_226000_.leaves(), randomsource)) {
+        for (BlockPos blockpos : Util.shuffledCopy(p_226000_.leaves(), randomsource)) {
             Direction direction = Util.getRandom(this.directions, randomsource);
             BlockPos blockpos1 = blockpos.relative(direction);
             if (!set.contains(blockpos1) && randomsource.nextFloat() < this.probability && this.hasRequiredEmptyBlocks(p_226000_, blockpos, direction)) {
                 BlockPos blockpos2 = blockpos1.offset(-this.exclusionRadiusXZ, -this.exclusionRadiusY, -this.exclusionRadiusXZ);
                 BlockPos blockpos3 = blockpos1.offset(this.exclusionRadiusXZ, this.exclusionRadiusY, this.exclusionRadiusXZ);
 
-                for(BlockPos blockpos4 : BlockPos.betweenClosed(blockpos2, blockpos3)) {
+                for (BlockPos blockpos4 : BlockPos.betweenClosed(blockpos2, blockpos3)) {
                     set.add(blockpos4.immutable());
                 }
 
-                for(BlockStateProvider blockstateprovider : this.blockProvider) {
-                    p_226000_.setBlock(blockpos1, blockstateprovider.getState(randomsource, blockpos1));
-                    blockpos1 = blockpos1.relative(direction);
-                }
+                p_226000_.setBlock(blockpos1, this.blockProvider.getState(randomsource, blockpos1));
             }
         }
     }
 
     private boolean hasRequiredEmptyBlocks(TreeDecorator.Context p_226002_, BlockPos p_226003_, Direction p_226004_) {
-        for(int i = 1; i <= this.requiredEmptyBlocks; ++i) {
+        for (int i = 1; i <= this.requiredEmptyBlocks; i++) {
             BlockPos blockpos = p_226003_.relative(p_226004_, i);
             if (!p_226002_.isAir(blockpos)) {
                 return false;
