--- a/net/minecraft/world/level/levelgen/feature/treedecorators/LeaveVineDecorator.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/LeaveVineDecorator.java
@@ -1,16 +_,16 @@
 package net.minecraft.world.level.levelgen.feature.treedecorators;
 
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.BlockPos;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.block.VineBlock;
 import net.minecraft.world.level.block.state.properties.BooleanProperty;
 
 public class LeaveVineDecorator extends TreeDecorator {
-    public static final Codec<LeaveVineDecorator> CODEC = Codec.floatRange(0.0F, 1.0F)
+    public static final MapCodec<LeaveVineDecorator> CODEC = Codec.floatRange(0.0F, 1.0F)
         .fieldOf("probability")
-        .xmap(LeaveVineDecorator::new, p_226037_ -> p_226037_.probability)
-        .codec();
+        .xmap(LeaveVineDecorator::new, p_226037_ -> p_226037_.probability);
     private final float probability;
 
     @Override
@@ -60,7 +_,7 @@
         p_226043_.placeVine(p_226041_, p_226042_);
         int i = 4;
 
-        for(BlockPos blockpos = p_226041_.below(); p_226043_.isAir(blockpos) && i > 0; --i) {
+        for (BlockPos blockpos = p_226041_.below(); p_226043_.isAir(blockpos) && i > 0; i--) {
             p_226043_.placeVine(blockpos, p_226042_);
             blockpos = blockpos.below();
         }
