--- a/net/minecraft/world/level/levelgen/feature/treedecorators/TreeDecoratorType.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/TreeDecoratorType.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.levelgen.feature.treedecorators;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 
@@ -11,17 +_,17 @@
     public static final TreeDecoratorType<BeehiveDecorator> BEEHIVE = register("beehive", BeehiveDecorator.CODEC);
     public static final TreeDecoratorType<AlterGroundDecorator> ALTER_GROUND = register("alter_ground", AlterGroundDecorator.CODEC);
     public static final TreeDecoratorType<AttachedToLeavesDecorator> ATTACHED_TO_LEAVES = register("attached_to_leaves", AttachedToLeavesDecorator.CODEC);
-    private final Codec<P> codec;
-
-    private static <P extends TreeDecorator> TreeDecoratorType<P> register(String p_70053_, Codec<P> p_70054_) {
-        return Registry.register(BuiltInRegistries.TREE_DECORATOR_TYPE, p_70053_, new TreeDecoratorType<>(p_70054_));
-    }
-
-    public TreeDecoratorType(Codec<P> p_70050_) {
-        this.codec = p_70050_;
-    }
-
-    public Codec<P> codec() {
+    private final MapCodec<P> codec;
+
+    private static <P extends TreeDecorator> TreeDecoratorType<P> register(String p_70053_, MapCodec<P> p_338285_) {
+        return Registry.register(BuiltInRegistries.TREE_DECORATOR_TYPE, p_70053_, new TreeDecoratorType<>(p_338285_));
+    }
+
+    private TreeDecoratorType(MapCodec<P> p_338862_) {
+        this.codec = p_338862_;
+    }
+
+    public MapCodec<P> codec() {
         return this.codec;
     }
 }
