--- a/net/minecraft/DetectedVersion.java
+++ b/net/minecraft/DetectedVersion.java
@@ -28,12 +_,12 @@
 
     private DetectedVersion() {
         this.id = UUID.randomUUID().toString().replaceAll("-", "");
-        this.name = "24w14potato";
+        this.name = "24w14a";
         this.stable = false;
-        this.worldVersion = new DataVersion(3824, "april2024");
+        this.worldVersion = new DataVersion(3827, "main");
         this.protocolVersion = SharedConstants.getProtocolVersion();
-        this.resourcePackVersion = 30;
-        this.dataPackVersion = 36;
+        this.resourcePackVersion = 31;
+        this.dataPackVersion = 38;
         this.buildTime = new Date();
     }
 
