--- a/net/minecraft/FileUtil.java
+++ b/net/minecraft/FileUtil.java
@@ -22,7 +_,7 @@
     private static final Pattern STRICT_PATH_SEGMENT_CHECK = Pattern.compile("[-._a-z0-9]+");
 
     public static String findAvailableName(Path p_133731_, String p_133732_, String p_133733_) throws IOException {
-        for(char c0 : SharedConstants.ILLEGAL_FILE_CHARACTERS) {
+        for (char c0 : SharedConstants.ILLEGAL_FILE_CHARACTERS) {
             p_133732_ = p_133732_.replace(c0, '_');
         }
 
@@ -42,7 +_,7 @@
             p_133732_ = p_133732_.substring(0, 255 - p_133733_.length());
         }
 
-        while(true) {
+        while (true) {
             String s = p_133732_;
             if (j != 0) {
                 String s1 = " (" + j + ")";
@@ -62,7 +_,7 @@
                 Files.deleteIfExists(path1);
                 return p_133731_.relativize(path1).toString();
             } catch (FileAlreadyExistsException filealreadyexistsexception) {
-                ++j;
+                j++;
             }
         }
     }
@@ -73,7 +_,7 @@
     }
 
     public static boolean isPathPortable(Path p_133735_) {
-        for(Path path : p_133735_) {
+        for (Path path : p_133735_) {
             if (RESERVED_WINDOWS_FILENAMES.matcher(path.toString()).matches()) {
                 return false;
             }
@@ -103,7 +_,7 @@
     public static DataResult<List<String>> decomposePath(String p_248866_) {
         int i = p_248866_.indexOf(47);
         if (i == -1) {
-            return switch(p_248866_) {
+            return switch (p_248866_) {
                 case "", ".", ".." -> DataResult.error(() -> "Invalid path '" + p_248866_ + "'");
                 default -> !isValidStrictPathSegment(p_248866_)
                 ? DataResult.error(() -> "Invalid path '" + p_248866_ + "'")
@@ -114,9 +_,9 @@
             int j = 0;
             boolean flag = false;
 
-            while(true) {
+            while (true) {
                 String s = p_248866_.substring(j, i);
-                switch(s) {
+                switch (s) {
                     case "":
                     case ".":
                     case "..":
@@ -145,13 +_,13 @@
     public static Path resolvePath(Path p_251522_, List<String> p_251495_) {
         int i = p_251495_.size();
 
-        return switch(i) {
+        return switch (i) {
             case 0 -> p_251522_;
             case 1 -> p_251522_.resolve(p_251495_.get(0));
             default -> {
                 String[] astring = new String[i - 1];
 
-                for(int j = 1; j < i; ++j) {
+                for (int j = 1; j < i; j++) {
                     astring[j - 1] = p_251495_.get(j);
                 }
 
@@ -168,7 +_,7 @@
         if (p_249502_.length == 0) {
             throw new IllegalArgumentException("Path must have at least one element");
         } else {
-            for(String s : p_249502_) {
+            for (String s : p_249502_) {
                 if (s.equals("..") || s.equals(".") || !isValidStrictPathSegment(s)) {
                     throw new IllegalArgumentException("Illegal segment " + s + " in path " + Arrays.toString((Object[])p_249502_));
                 }
