--- a/net/minecraft/Optionull.java
+++ b/net/minecraft/Optionull.java
@@ -13,11 +_,11 @@
     }
 
     public static <T, R> R mapOrDefault(@Nullable T p_270215_, Function<T, R> p_270557_, R p_270839_) {
-        return (R)(p_270215_ == null ? p_270839_ : p_270557_.apply(p_270215_));
+        return p_270215_ == null ? p_270839_ : p_270557_.apply(p_270215_);
     }
 
     public static <T, R> R mapOrElse(@Nullable T p_270820_, Function<T, R> p_270536_, Supplier<R> p_270756_) {
-        return (R)(p_270820_ == null ? p_270756_.get() : p_270536_.apply(p_270820_));
+        return p_270820_ == null ? p_270756_.get() : p_270536_.apply(p_270820_);
     }
 
     @Nullable
@@ -28,12 +_,12 @@
 
     public static <T> T firstOrDefault(Collection<T> p_270625_, T p_270960_) {
         Iterator<T> iterator = p_270625_.iterator();
-        return (T)(iterator.hasNext() ? iterator.next() : p_270960_);
+        return iterator.hasNext() ? iterator.next() : p_270960_;
     }
 
     public static <T> T firstOrElse(Collection<T> p_270529_, Supplier<T> p_270239_) {
         Iterator<T> iterator = p_270529_.iterator();
-        return (T)(iterator.hasNext() ? iterator.next() : p_270239_.get());
+        return iterator.hasNext() ? iterator.next() : p_270239_.get();
     }
 
     public static <T> boolean isNullOrEmpty(@Nullable T[] p_270794_) {
