--- a/net/minecraft/SharedConstants.java
+++ b/net/minecraft/SharedConstants.java
@@ -15,22 +_,22 @@
     @Deprecated
     public static final boolean SNAPSHOT = true;
     @Deprecated
-    public static final int WORLD_VERSION = 3824;
-    @Deprecated
-    public static final String SERIES = "april2024";
-    @Deprecated
-    public static final String VERSION_STRING = "24w14potato";
+    public static final int WORLD_VERSION = 3827;
+    @Deprecated
+    public static final String SERIES = "main";
+    @Deprecated
+    public static final String VERSION_STRING = "24w14a";
     @Deprecated
     public static final int RELEASE_NETWORK_PROTOCOL_VERSION = 766;
     @Deprecated
-    public static final int SNAPSHOT_NETWORK_PROTOCOL_VERSION = 183;
+    public static final int SNAPSHOT_NETWORK_PROTOCOL_VERSION = 184;
     public static final int SNBT_NAG_VERSION = 3798;
     private static final int SNAPSHOT_PROTOCOL_BIT = 30;
-    public static final boolean THROW_ON_TASK_FAILURE = false;
-    @Deprecated
-    public static final int RESOURCE_PACK_FORMAT = 30;
-    @Deprecated
-    public static final int DATA_PACK_FORMAT = 36;
+    public static final boolean CRASH_EAGERLY = true;
+    @Deprecated
+    public static final int RESOURCE_PACK_FORMAT = 31;
+    @Deprecated
+    public static final int DATA_PACK_FORMAT = 38;
     @Deprecated
     public static final int LANGUAGE_FORMAT = 1;
     public static final int REPORT_FORMAT_VERSION = 1;
@@ -123,7 +_,8 @@
     public static final boolean USE_WORKFLOWS_HOOKS = false;
     public static final boolean USE_DEVONLY = false;
     public static boolean CHECK_DATA_FIXER_SCHEMA = true;
-    public static boolean IS_RUNNING_IN_IDE;
+    public static boolean IS_RUNNING_IN_IDE = !net.neoforged.fml.loading.FMLLoader.isProduction();
+    public static final boolean IS_RUNNING_WITH_JDWP = java.lang.management.ManagementFactory.getRuntimeMXBean().getInputArguments().stream().anyMatch(str -> str.startsWith("-agentlib:jdwp"));
     public static Set<TypeReference> DATA_FIX_TYPES_TO_OPTIMIZE = Set.of();
     public static final int WORLD_RESOLUTION = 16;
     public static final int MAX_CHAT_LENGTH = 256;
@@ -167,17 +_,13 @@
     }
 
     public static int getProtocolVersion() {
-        return 1073742007;
+        return 1073742008;
     }
 
     public static boolean debugVoidTerrain(ChunkPos p_183708_) {
         int i = p_183708_.getMinBlockX();
         int j = p_183708_.getMinBlockZ();
-        if (!debugGenerateSquareTerrainWithoutNoise) {
-            return false;
-        } else {
-            return i > 8192 || i < 0 || j > 1024 || j < 0;
-        }
+        return !debugGenerateSquareTerrainWithoutNoise ? false : i > 8192 || i < 0 || j > 1024 || j < 0;
     }
 
     public static void enableDataFixerOptimizations() {
@@ -185,6 +_,7 @@
     }
 
     static {
+        if (System.getProperty("io.netty.leakDetection.level") == null) // Forge: allow level to be manually specified
         ResourceLeakDetector.setLevel(NETTY_LEAK_DETECTION);
         CommandSyntaxException.ENABLE_COMMAND_STACK_TRACES = false;
         CommandSyntaxException.BUILT_IN_EXCEPTIONS = new BrigadierExceptions();
