--- a/net/minecraft/SystemReport.java
+++ b/net/minecraft/SystemReport.java
@@ -90,7 +_,7 @@
     private void putPhysicalMemory(List<PhysicalMemory> p_143532_) {
         int i = 0;
 
-        for(PhysicalMemory physicalmemory : p_143532_) {
+        for (PhysicalMemory physicalmemory : p_143532_) {
             String s = String.format(Locale.ROOT, "Memory slot #%d ", i++);
             this.setDetail(s + "capacity (MB)", () -> String.format(Locale.ROOT, "%.2f", (float)physicalmemory.getCapacity() / 1048576.0F));
             this.setDetail(s + "clockSpeed (GHz)", () -> String.format(Locale.ROOT, "%.2f", (float)physicalmemory.getClockSpeed() / 1.0E9F));
@@ -113,7 +_,7 @@
     private void putGraphics(List<GraphicsCard> p_143553_) {
         int i = 0;
 
-        for(GraphicsCard graphicscard : p_143553_) {
+        for (GraphicsCard graphicscard : p_143553_) {
             String s = String.format(Locale.ROOT, "Graphics card #%d ", i++);
             this.setDetail(s + "name", graphicscard::getName);
             this.setDetail(s + "vendor", graphicscard::getVendor);
@@ -150,7 +_,7 @@
         return this.entries
             .entrySet()
             .stream()
-            .map(p_143534_ -> (String)p_143534_.getKey() + ": " + (String)p_143534_.getValue())
+            .map(p_143534_ -> p_143534_.getKey() + ": " + p_143534_.getValue())
             .collect(Collectors.joining(System.lineSeparator()));
     }
 }
