--- a/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -343,7 +371,10 @@
         BlockState state = level.getBlockState(pos);
         if (!state.getBlock().isEnabled(level.enabledFeatures())) {
             return InteractionResult.FAIL;
-        } else if (this.gameModeForPlayer == GameType.SPECTATOR) {
+        }
+        net.neoforged.neoforge.event.entity.player.PlayerInteractEvent.RightClickBlock event = net.neoforged.neoforge.common.CommonHooks.onRightClickBlock(player, hand, pos, hitResult);
+        if (event.isCanceled()) return event.getCancellationResult();
+        if (this.gameModeForPlayer == GameType.SPECTATOR) {
             MenuProvider menuProvider = state.getMenuProvider(level, pos);
             if (menuProvider != null) {
                 player.openMenu(menuProvider);
@@ -352,10 +383,15 @@
                 return InteractionResult.PASS;
             }
         } else {
+            UseOnContext context = new UseOnContext(player, hand, hitResult);
+            if (event.getUseItem() != net.minecraft.util.TriState.FALSE) {
+                InteractionResult result = itemStack.onItemUseFirst(context);
+                if (result != InteractionResult.PASS) return result;
+            }
             boolean haveSomethingInOurHands = !player.getMainHandItem().isEmpty() || !player.getOffhandItem().isEmpty();
-            boolean suppressUsingBlock = player.isSecondaryUseActive() && haveSomethingInOurHands;
+            boolean suppressUsingBlock = (player.isSecondaryUseActive() && haveSomethingInOurHands) && !(player.getMainHandItem().doesSneakBypassUse(level, pos, player) && player.getOffhandItem().doesSneakBypassUse(level, pos, player));
             ItemStack usedItemStack = itemStack.copy();
-            if (!suppressUsingBlock) {
+            if (event.getUseBlock().isTrue() || (event.getUseBlock().isDefault() && !suppressUsingBlock)) {
                 InteractionResult itemUse = state.useItemOn(player.getItemInHand(hand), level, player, hand, hitResult);
                 if (itemUse.consumesAction()) {
                     CriteriaTriggers.ITEM_USED_ON_BLOCK.trigger(player, pos, usedItemStack);
