--- a/net/minecraft/world/grid/FlyingTickable.java
+++ /dev/null
@@ -1,11 +_,0 @@
-package net.minecraft.world.grid;
-
-import net.minecraft.core.BlockPos;
-import net.minecraft.core.Direction;
-import net.minecraft.world.level.Level;
-import net.minecraft.world.level.block.state.BlockState;
-import net.minecraft.world.phys.Vec3;
-
-public interface FlyingTickable {
-    void flyingTick(Level p_341544_, SubGridBlocks p_341971_, BlockState p_341328_, BlockPos p_340828_, Vec3 p_342033_, Direction p_341170_);
-}
