--- a/net/minecraft/server/level/ChunkTracker.java
+++ b/net/minecraft/server/level/ChunkTracker.java
@@ -20,8 +_,8 @@
             int i = chunkpos.x;
             int j = chunkpos.z;
 
-            for(int k = -1; k <= 1; ++k) {
-                for(int l = -1; l <= 1; ++l) {
+            for (int k = -1; k <= 1; k++) {
+                for (int l = -1; l <= 1; l++) {
                     long i1 = ChunkPos.asLong(i + k, j + l);
                     if (i1 != p_140707_) {
                         this.checkNeighbor(p_140707_, i1, p_140708_, p_140709_);
@@ -38,8 +_,8 @@
         int j = chunkpos.x;
         int k = chunkpos.z;
 
-        for(int l = -1; l <= 1; ++l) {
-            for(int i1 = -1; i1 <= 1; ++i1) {
+        for (int l = -1; l <= 1; l++) {
+            for (int i1 = -1; i1 <= 1; i1++) {
                 long j1 = ChunkPos.asLong(j + l, k + i1);
                 if (j1 == p_140711_) {
                     j1 = ChunkPos.INVALID_CHUNK_POS;
