--- a/net/minecraft/server/level/ChunkResult.java
+++ b/net/minecraft/server/level/ChunkResult.java
@@ -25,8 +_,8 @@
 
     @Nullable
     static <R> R orElse(ChunkResult<? extends R> p_331283_, @Nullable R p_331474_) {
-        R r = p_331283_.orElse(null);
-        return (R)(r != null ? r : p_331474_);
+        R r = (R)p_331283_.orElse(null);
+        return r != null ? r : p_331474_;
     }
 
     @Nullable
@@ -62,7 +_,7 @@
 
         @Override
         public <R> ChunkResult<R> map(Function<T, R> p_331060_) {
-            return new ChunkResult.Fail<>(this.error);
+            return new ChunkResult.Fail(this.error);
         }
 
         @Override
