--- a/net/minecraft/server/level/PlayerRespawnLogic.java
+++ b/net/minecraft/server/level/PlayerRespawnLogic.java
@@ -28,7 +_,7 @@
             } else {
                 BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-                for(int k = i + 1; k >= p_183929_.getMinBuildHeight(); --k) {
+                for (int k = i + 1; k >= p_183929_.getMinBuildHeight(); k--) {
                     blockpos$mutableblockpos.set(p_183930_, k, p_183931_);
                     BlockState blockstate = p_183929_.getBlockState(blockpos$mutableblockpos);
                     if (!blockstate.getFluidState().isEmpty()) {
@@ -50,8 +_,8 @@
         if (SharedConstants.debugVoidTerrain(p_183934_)) {
             return null;
         } else {
-            for(int i = p_183934_.getMinBlockX(); i <= p_183934_.getMaxBlockX(); ++i) {
-                for(int j = p_183934_.getMinBlockZ(); j <= p_183934_.getMaxBlockZ(); ++j) {
+            for (int i = p_183934_.getMinBlockX(); i <= p_183934_.getMaxBlockX(); i++) {
+                for (int j = p_183934_.getMinBlockZ(); j <= p_183934_.getMaxBlockZ(); j++) {
                     BlockPos blockpos = getOverworldRespawnPos(p_183933_, i, j);
                     if (blockpos != null) {
                         return blockpos;
