--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -98,9 +_,9 @@
 
         if (this.entity instanceof ItemFrame itemframe && this.tickCount % 10 == 0) {
             ItemStack itemstack = itemframe.getItem();
-            if (itemstack.getItem() instanceof MapItem) {
+            if (true) {
                 MapId mapid = itemstack.get(DataComponents.MAP_ID);
-                MapItemSavedData mapitemsaveddata = MapItem.getSavedData(mapid, this.level);
+                MapItemSavedData mapitemsaveddata = MapItem.getSavedData(itemstack, this.level);
                 if (mapitemsaveddata != null) {
                     for (ServerPlayer serverplayer : this.level.players()) {
                         mapitemsaveddata.tickCarriedBy(serverplayer, itemstack);
@@ -256,6 +_,7 @@
     public void removePairing(ServerPlayer player) {
         this.entity.stopSeenByPlayer(player);
         player.connection.send(new ClientboundRemoveEntitiesPacket(this.entity.getId()));
+        net.neoforged.neoforge.event.EventHooks.onStopEntityTracking(this.entity, player);
     }
 
     public void addPairing(ServerPlayer player) {
@@ -263,9 +_,10 @@
         this.sendPairingData(player, list::add);
         player.connection.send(new ClientboundBundlePacket(list));
         this.entity.startSeenByPlayer(player);
+        net.neoforged.neoforge.event.EventHooks.onStartEntityTracking(this.entity, player);
     }
 
-    public void sendPairingData(ServerPlayer player, Consumer<Packet<ClientGamePacketListener>> broadcast) {
+    public void sendPairingData(ServerPlayer player, Consumer<Packet<? super ClientGamePacketListener>> broadcast) {
         this.entity.updateDataBeforeSync();
         if (this.entity.isRemoved()) {
             LOGGER.warn("Fetching packet for removed entity {}", this.entity);
@@ -273,6 +_,7 @@
 
         Packet<ClientGamePacketListener> packet = this.entity.getAddEntityPacket(this);
         broadcast.accept(packet);
+        this.entity.sendPairingData(player, payload -> broadcast.accept(payload.toVanillaClientbound()));
         if (this.trackedDataValues != null) {
             broadcast.accept(new ClientboundSetEntityDataPacket(this.entity.getId(), this.trackedDataValues));
         }
@@ -310,6 +_,8 @@
         if (this.entity instanceof Leashable leashable && leashable.isLeashed()) {
             broadcast.accept(new ClientboundSetEntityLinkPacket(this.entity, leashable.getLeashHolder()));
         }
+
+        net.neoforged.neoforge.attachment.AttachmentSync.syncInitialEntityAttachments(this.entity, player, broadcast);
     }
 
     public Vec3 getPositionBase() {
