--- a/net/minecraft/server/level/ServerChunkCache.java
+++ b/net/minecraft/server/level/ServerChunkCache.java
@@ -8,6 +_,7 @@
 import java.util.Arrays;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
 import java.util.concurrent.Executor;
 import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
@@ -43,7 +_,7 @@
 import net.minecraft.world.level.storage.DimensionDataStorage;
 import net.minecraft.world.level.storage.LevelStorageSource;
 
-public class ServerChunkCache extends ChunkSource {
+public class ServerChunkCache extends ChunkSource implements net.neoforged.neoforge.common.extensions.IServerChunkCacheExtension {
     private static final List<ChunkStatus> CHUNK_STATUSES = ChunkStatus.getStatusList();
     private final DistanceManager distanceManager;
     public final ServerLevel level;
@@ -118,7 +_,7 @@
     }
 
     private void storeInCache(long p_8367_, @Nullable ChunkAccess p_8368_, ChunkStatus p_331839_) {
-        for(int i = 3; i > 0; --i) {
+        for (int i = 3; i > 0; i--) {
             this.lastChunkPos[i] = this.lastChunkPos[i - 1];
             this.lastChunkStatus[i] = this.lastChunkStatus[i - 1];
             this.lastChunk[i] = this.lastChunk[i - 1];
@@ -139,7 +_,7 @@
             profilerfiller.incrementCounter("getChunk");
             long i = ChunkPos.asLong(p_8360_, p_8361_);
 
-            for(int j = 0; j < 4; ++j) {
+            for (int j = 0; j < 4; j++) {
                 if (i == this.lastChunkPos[j] && p_330876_ == this.lastChunkStatus[j]) {
                     ChunkAccess chunkaccess = this.lastChunk[j];
                     if (chunkaccess != null || !p_8363_) {
@@ -148,6 +_,10 @@
                 }
             }
 
+            ChunkHolder chunkholder = this.getVisibleChunkIfPresent(i);
+            if (chunkholder != null && chunkholder.currentlyLoading != null)
+                return chunkholder.currentlyLoading; // Neo: If the requested chunk is loading, bypass the future chain to prevent a deadlock.
+
             profilerfiller.incrementCounter("getChunkCacheMiss");
             CompletableFuture<ChunkResult<ChunkAccess>> completablefuture = this.getChunkFutureMainThread(p_8360_, p_8361_, p_330876_, p_8363_);
             this.mainThreadProcessor.managedBlock(completablefuture::isDone);
@@ -171,7 +_,7 @@
             this.level.getProfiler().incrementCounter("getChunkNow");
             long i = ChunkPos.asLong(p_8357_, p_8358_);
 
-            for(int j = 0; j < 4; ++j) {
+            for (int j = 0; j < 4; j++) {
                 if (i == this.lastChunkPos[j] && this.lastChunkStatus[j] == ChunkStatus.FULL) {
                     ChunkAccess chunkaccess = this.lastChunk[j];
                     return chunkaccess instanceof LevelChunk ? (LevelChunk)chunkaccess : null;
@@ -182,6 +_,7 @@
             if (chunkholder == null) {
                 return null;
             } else {
+                if (chunkholder.currentlyLoading != null) return chunkholder.currentlyLoading; // Forge: If the requested chunk is loading, bypass the future chain to prevent a deadlock.
                 ChunkResult<ChunkAccess> chunkresult = chunkholder.getFutureIfPresent(ChunkStatus.FULL).getNow(null);
                 if (chunkresult == null) {
                     return null;
@@ -216,7 +_,7 @@
             completablefuture = CompletableFuture.<CompletableFuture<ChunkResult<ChunkAccess>>>supplyAsync(
                     () -> this.getChunkFutureMainThread(p_8432_, p_8433_, p_330326_, p_8435_), this.mainThreadProcessor
                 )
-                .thenCompose(p_331360_ -> p_331360_);
+                .thenCompose(p_331360_ -> (CompletionStage<ChunkResult<ChunkAccess>>)p_331360_);
         }
 
         return completablefuture;
@@ -265,7 +_,7 @@
         } else {
             int j = CHUNK_STATUSES.size() - 1;
 
-            while(true) {
+            while (true) {
                 ChunkStatus chunkstatus = CHUNK_STATUSES.get(j);
                 ChunkAccess chunkaccess = chunkholder.getFutureIfPresentUnchecked(chunkstatus).getNow(ChunkHolder.UNLOADED_CHUNK).orElse(null);
                 if (chunkaccess != null) {
@@ -276,7 +_,7 @@
                     return null;
                 }
 
-                --j;
+                j--;
             }
         }
     }
@@ -351,7 +_,7 @@
             profilerfiller.push("filteringLoadedChunks");
             List<ServerChunkCache.ChunkAndHolder> list = Lists.newArrayListWithCapacity(this.chunkMap.size());
 
-            for(ChunkHolder chunkholder : this.chunkMap.getChunks()) {
+            for (ChunkHolder chunkholder : this.chunkMap.getChunks()) {
                 LevelChunk levelchunk = chunkholder.getTickingChunk();
                 if (levelchunk != null) {
                     list.add(new ServerChunkCache.ChunkAndHolder(levelchunk, chunkholder));
@@ -371,10 +_,10 @@
                 int k = this.level.getGameRules().getInt(GameRules.RULE_RANDOMTICKING);
                 boolean flag = this.level.getLevelData().getGameTime() % 400L == 0L;
 
-                for(ServerChunkCache.ChunkAndHolder serverchunkcache$chunkandholder : list) {
+                for (ServerChunkCache.ChunkAndHolder serverchunkcache$chunkandholder : list) {
                     LevelChunk levelchunk1 = serverchunkcache$chunkandholder.chunk;
                     ChunkPos chunkpos = levelchunk1.getPos();
-                    if (this.level.isNaturalSpawningAllowed(chunkpos) && this.chunkMap.anyPlayerCloseEnoughForSpawning(chunkpos)) {
+                    if ((this.level.isNaturalSpawningAllowed(chunkpos) && this.chunkMap.anyPlayerCloseEnoughForSpawning(chunkpos)) || this.distanceManager.shouldForceTicks(chunkpos.toLong())) {
                         levelchunk1.incrementInhabitedTime(j);
                         if (flag1 && (this.spawnEnemies || this.spawnFriendlies) && this.level.getWorldBorder().isWithinBounds(chunkpos)) {
                             NaturalSpawner.spawnForChunk(this.level, levelchunk1, naturalspawner$spawnstate, this.spawnFriendlies, this.spawnEnemies, flag);
@@ -453,11 +_,17 @@
     }
 
     public <T> void addRegionTicket(TicketType<T> p_8388_, ChunkPos p_8389_, int p_8390_, T p_8391_) {
-        this.distanceManager.addRegionTicket(p_8388_, p_8389_, p_8390_, p_8391_);
+        addRegionTicket(p_8388_, p_8389_, p_8390_, p_8391_, false);
+    }
+    public <T> void addRegionTicket(TicketType<T> p_8388_, ChunkPos p_8389_, int p_8390_, T p_8391_, boolean forceTicks) {
+        this.distanceManager.addRegionTicket(p_8388_, p_8389_, p_8390_, p_8391_, forceTicks);
     }
 
     public <T> void removeRegionTicket(TicketType<T> p_8439_, ChunkPos p_8440_, int p_8441_, T p_8442_) {
-        this.distanceManager.removeRegionTicket(p_8439_, p_8440_, p_8441_, p_8442_);
+        removeRegionTicket(p_8439_, p_8440_, p_8441_, p_8442_, false);
+    }
+    public <T> void removeRegionTicket(TicketType<T> p_8439_, ChunkPos p_8440_, int p_8441_, T p_8442_, boolean forceTicks) {
+        this.distanceManager.removeRegionTicket(p_8439_, p_8440_, p_8441_, p_8442_, forceTicks);
     }
 
     @Override
