--- a/net/minecraft/world/level/storage/DimensionDataStorage.java
+++ b/net/minecraft/world/level/storage/DimensionDataStorage.java
@@ -38,13 +_,21 @@
 
 public class DimensionDataStorage implements AutoCloseable {
     private static final Logger LOGGER = LogUtils.getLogger();
+    private final net.minecraft.server.level.@org.jspecify.annotations.Nullable ServerLevel level;
     private final Map<SavedDataType<?>, Optional<SavedData>> cache = new HashMap<>();
     private final DataFixer fixerUpper;
     private final HolderLookup.Provider registries;
     private final Path dataFolder;
     private CompletableFuture<?> pendingWriteFuture = CompletableFuture.completedFuture(null);
 
+    // Neo: Use the version that has a level
+    @Deprecated
     public DimensionDataStorage(Path dataFolder, DataFixer fixerUpper, HolderLookup.Provider registries) {
+        this(null, dataFolder, fixerUpper, registries);
+    }
+
+    public DimensionDataStorage(net.minecraft.server.level.@org.jspecify.annotations.Nullable ServerLevel level, Path dataFolder, DataFixer fixerUpper, HolderLookup.Provider registries) {
+        this.level = level;
         this.fixerUpper = fixerUpper;
         this.dataFolder = dataFolder;
         this.registries = registries;
@@ -59,7 +_,7 @@
         if (t != null) {
             return t;
         } else {
-            T t1 = (T)type.constructor().get();
+            T t1 = (T)type.factory().create(this.level);
             this.set(type, t1);
             return t1;
         }
@@ -83,7 +_,7 @@
                     type.id(), type.dataFixType(), SharedConstants.getCurrentVersion().dataVersion().version()
                 );
                 RegistryOps<Tag> registryops = this.registries.createSerializationContext(NbtOps.INSTANCE);
-                return type.codec()
+                return type.codecFactory().create(this.level)
                     .parse(registryops, compoundtag.get("data"))
                     .resultOrPartial(error -> LOGGER.error("Failed to parse saved data for '{}': {}", type, error))
                     .orElse(null);
@@ -100,7 +_,7 @@
         data.setDirty();
     }
 
-    public CompoundTag readTagFromDisk(String id, DataFixTypes type, int newVersion) throws IOException {
+    public CompoundTag readTagFromDisk(String id, @Nullable DataFixTypes type, int newVersion) throws IOException {
         CompoundTag compoundtag1;
         try (
             InputStream inputstream = Files.newInputStream(this.getDataFile(id));
@@ -115,9 +_,19 @@
                 }
             }
 
-            int i = NbtUtils.getDataVersion(compoundtag, 1343);
-            compoundtag1 = type.update(this.fixerUpper, compoundtag, i, newVersion);
+            if (type != null) {
+                int i = NbtUtils.getDataVersion(compoundtag, 1343);
+                compoundtag1 = type.update(this.fixerUpper, compoundtag, i, newVersion);
+            } else {
+                compoundtag1 = compoundtag;
+            }
         }
+
+        // Neo: delete any temporary files so that we don't inflate disk space unnecessarily.
+        this.pendingWriteFuture = this.pendingWriteFuture.thenRunAsync(
+                () -> net.neoforged.neoforge.common.IOUtilities.tryCleanupTempFiles(this.dataFolder, id),
+                Util.ioPool()
+        );
 
         return compoundtag1;
     }
@@ -189,7 +_,7 @@
     }
 
     private <T extends SavedData> CompoundTag encodeUnchecked(SavedDataType<T> type, SavedData data, RegistryOps<Tag> ops) {
-        Codec<T> codec = type.codec();
+        Codec<T> codec = type.codecFactory().create(this.level);
         CompoundTag compoundtag = new CompoundTag();
         compoundtag.put("data", codec.encodeStart(ops, (T)data).getOrThrow());
         NbtUtils.addCurrentDataVersion(compoundtag);
@@ -200,7 +_,11 @@
         Path path = this.getDataFile(type.id());
 
         try {
-            NbtIo.writeCompressed(tag, path);
+            // Neo: ensure parent directories exist if the SavedData's path contains slashes
+            if (!Files.exists(path)) {
+                Files.createDirectories(path.getParent());
+            }
+            net.neoforged.neoforge.common.IOUtilities.writeNbtCompressed(tag, path);
         } catch (IOException ioexception) {
             LOGGER.error("Could not save data to {}", path.getFileName(), ioexception);
         }
