--- a/net/minecraft/client/renderer/chunk/RenderRegionCache.java
+++ b/net/minecraft/client/renderer/chunk/RenderRegionCache.java
@@ -33,7 +_,8 @@
             }
         }
 
-        return new RenderSectionRegion(level, l, i1, j1, asectioncopy);
+        var modelDataManager = level.getModelDataManager().snapshotSectionRegion(l, i1, j1, k1, l1, i2);
+        return new RenderSectionRegion(level, l, i1, j1, asectioncopy, modelDataManager);
     }
 
     private SectionCopy getSectionDataCopy(Level level, int sectionX, int sectionY, int sectionZ) {
