--- a/net/minecraft/client/renderer/chunk/RenderRegionCache.java
+++ b/net/minecraft/client/renderer/chunk/RenderRegionCache.java
@@ -17,14 +_,19 @@
 
     @Nullable
     public RenderChunkRegion createRegion(Level p_200466_, BlockPos p_200467_, BlockPos p_200468_, int p_200469_) {
+        return createRegion(p_200466_, p_200467_, p_200468_, p_200469_, true);
+    }
+
+    @Nullable
+    public RenderChunkRegion createRegion(Level p_200466_, BlockPos p_200467_, BlockPos p_200468_, int p_200469_, boolean nullForEmpty) {
         int i = SectionPos.blockToSectionCoord(p_200467_.getX() - p_200469_);
         int j = SectionPos.blockToSectionCoord(p_200467_.getZ() - p_200469_);
         int k = SectionPos.blockToSectionCoord(p_200468_.getX() + p_200469_);
         int l = SectionPos.blockToSectionCoord(p_200468_.getZ() + p_200469_);
         RenderRegionCache.ChunkInfo[][] arenderregioncache$chunkinfo = new RenderRegionCache.ChunkInfo[k - i + 1][l - j + 1];
 
-        for(int i1 = i; i1 <= k; ++i1) {
-            for(int j1 = j; j1 <= l; ++j1) {
+        for (int i1 = i; i1 <= k; i1++) {
+            for (int j1 = j; j1 <= l; j1++) {
                 arenderregioncache$chunkinfo[i1 - i][j1 - j] = this.chunkInfoCache
                     .computeIfAbsent(
                         ChunkPos.asLong(i1, j1),
@@ -33,18 +_,21 @@
             }
         }
 
-        if (isAllEmpty(p_200467_, p_200468_, i, j, arenderregioncache$chunkinfo)) {
+        if (nullForEmpty && isAllEmpty(p_200467_, p_200468_, i, j, arenderregioncache$chunkinfo)) {
             return null;
         } else {
             RenderChunk[][] arenderchunk = new RenderChunk[k - i + 1][l - j + 1];
 
-            for(int l1 = i; l1 <= k; ++l1) {
-                for(int k1 = j; k1 <= l; ++k1) {
+            for (int l1 = i; l1 <= k; l1++) {
+                for (int k1 = j; k1 <= l; k1++) {
                     arenderchunk[l1 - i][k1 - j] = arenderregioncache$chunkinfo[l1 - i][k1 - j].renderChunk();
                 }
             }
 
-            return new RenderChunkRegion(p_200466_, i, j, arenderchunk);
+            int sectionMinY = SectionPos.blockToSectionCoord(p_200467_.getY() - p_200469_);
+            int sectionMaxY = SectionPos.blockToSectionCoord(p_200468_.getY() + p_200469_);
+            var modelDataManager = p_200466_.getModelDataManager().snapshotSectionRegion(i, sectionMinY, j, k, sectionMaxY, l);
+            return new RenderChunkRegion(p_200466_, i, j, arenderchunk, modelDataManager);
         }
     }
 
@@ -54,8 +_,8 @@
         int k = SectionPos.blockToSectionCoord(p_200472_.getX());
         int l = SectionPos.blockToSectionCoord(p_200472_.getZ());
 
-        for(int i1 = i; i1 <= k; ++i1) {
-            for(int j1 = j; j1 <= l; ++j1) {
+        for (int i1 = i; i1 <= k; i1++) {
+            for (int j1 = j; j1 <= l; j1++) {
                 LevelChunk levelchunk = p_200475_[i1 - p_200473_][j1 - p_200474_].chunk();
                 if (!levelchunk.isYSpaceEmpty(p_200471_.getY(), p_200472_.getY())) {
                     return false;
