--- a/net/minecraft/client/renderer/chunk/SectionRenderDispatcher.java
+++ b/net/minecraft/client/renderer/chunk/SectionRenderDispatcher.java
@@ -346,9 +_,10 @@
 
         public SectionRenderDispatcher.RenderSection.CompileTask createCompileTask(RenderRegionCache cache) {
             this.cancelTasks();
+            var additionalRenderers = net.neoforged.neoforge.client.ClientHooks.gatherAdditionalRenderers(this.renderOrigin, SectionRenderDispatcher.this.level);
             RenderSectionRegion rendersectionregion = cache.createRegion(SectionRenderDispatcher.this.level, this.sectionNode);
             boolean flag = this.sectionMesh.get() != CompiledSectionMesh.UNCOMPILED;
-            this.lastRebuildTask = new SectionRenderDispatcher.RenderSection.RebuildTask(rendersectionregion, flag);
+            this.lastRebuildTask = new SectionRenderDispatcher.RenderSection.RebuildTask(rendersectionregion, flag, additionalRenderers);
             return this.lastRebuildTask;
         }
 
@@ -375,6 +_,15 @@
             );
         }
 
+        // Neo: start
+
+        @Override
+        public boolean isEmpty() {
+            return !getSectionMesh().hasRenderableLayers();
+        }
+
+        // Neo: end
+
         @OnlyIn(Dist.CLIENT)
         public abstract class CompileTask {
             protected final AtomicBoolean isCancelled = new AtomicBoolean(false);
@@ -403,10 +_,18 @@
         @OnlyIn(Dist.CLIENT)
         class RebuildTask extends SectionRenderDispatcher.RenderSection.CompileTask {
             protected final RenderSectionRegion region;
+            private final java.util.List<net.neoforged.neoforge.client.event.AddSectionGeometryEvent.AdditionalSectionRenderer> additionalRenderers;
 
+            /** @deprecated Neo: use {@link #RebuildTask(RenderSectionRegion, boolean, java.util.List)} instead */
+            @Deprecated
             public RebuildTask(RenderSectionRegion region, boolean isRecompile) {
+                this(region, isRecompile, java.util.List.of());
+            }
+
+            public RebuildTask(RenderSectionRegion region, boolean isRecompile, java.util.List<net.neoforged.neoforge.client.event.AddSectionGeometryEvent.AdditionalSectionRenderer> additionalRenderers) {
                 super(isRecompile);
                 this.region = region;
+                this.additionalRenderers = additionalRenderers;
             }
 
             @Override
@@ -427,7 +_,7 @@
                         SectionCompiler.Results sectioncompiler$results;
                         try (Zone zone = Profiler.get().zone("Compile Section")) {
                             sectioncompiler$results = SectionRenderDispatcher.this.sectionCompiler
-                                .compile(sectionpos, this.region, RenderSection.this.createVertexSorting(sectionpos), buffers);
+                                .compile(sectionpos, this.region, RenderSection.this.createVertexSorting(sectionpos), buffers, this.additionalRenderers);
                         }
 
                         TranslucencyPointOfView translucencypointofview = TranslucencyPointOfView.of(SectionRenderDispatcher.this.cameraPosition, i);
