--- a/net/minecraft/client/renderer/chunk/SectionRenderDispatcher.java
+++ b/net/minecraft/client/renderer/chunk/SectionRenderDispatcher.java
@@ -352,9 +_,10 @@
 
         public SectionRenderDispatcher.RenderSection.CompileTask createCompileTask(RenderRegionCache cache) {
             this.cancelTasks();
+            var additionalRenderers = net.neoforged.neoforge.client.ClientHooks.gatherAdditionalRenderers(this.renderOrigin, SectionRenderDispatcher.this.level);
             RenderSectionRegion region = cache.createRegion(SectionRenderDispatcher.this.level, this.sectionNode);
             boolean isRecompile = this.sectionMesh.get() != CompiledSectionMesh.UNCOMPILED;
-            this.lastRebuildTask = new SectionRenderDispatcher.RenderSection.RebuildTask(region, isRecompile);
+            this.lastRebuildTask = new SectionRenderDispatcher.RenderSection.RebuildTask(region, isRecompile, additionalRenderers);
             return this.lastRebuildTask;
         }
 
@@ -436,7 +_,7 @@
                         SectionCompiler.Results results;
                         try (Zone ignored = Profiler.get().zone("Compile Section")) {
                             results = SectionRenderDispatcher.this.sectionCompiler
-                                .compile(sectionPos, this.region, RenderSection.this.createVertexSorting(sectionPos), buffers);
+                                .compile(sectionPos, this.region, RenderSection.this.createVertexSorting(sectionPos), buffers, this.additionalRenderers);
                         }
 
                         TranslucencyPointOfView translucencyPointOfView = TranslucencyPointOfView.of(SectionRenderDispatcher.this.cameraPosition, sectionNode);
