--- a/net/minecraft/client/renderer/chunk/VisibilitySet.java
+++ b/net/minecraft/client/renderer/chunk/VisibilitySet.java
@@ -12,8 +_,8 @@
     private final BitSet data = new BitSet(FACINGS * FACINGS);
 
     public void add(Set<Direction> p_112991_) {
-        for(Direction direction : p_112991_) {
-            for(Direction direction1 : p_112991_) {
+        for (Direction direction : p_112991_) {
+            for (Direction direction1 : p_112991_) {
                 this.set(direction, direction1, true);
             }
         }
@@ -37,16 +_,16 @@
         StringBuilder stringbuilder = new StringBuilder();
         stringbuilder.append(' ');
 
-        for(Direction direction : Direction.values()) {
+        for (Direction direction : Direction.values()) {
             stringbuilder.append(' ').append(direction.toString().toUpperCase().charAt(0));
         }
 
         stringbuilder.append('\n');
 
-        for(Direction direction2 : Direction.values()) {
+        for (Direction direction2 : Direction.values()) {
             stringbuilder.append(direction2.toString().toUpperCase().charAt(0));
 
-            for(Direction direction1 : Direction.values()) {
+            for (Direction direction1 : Direction.values()) {
                 if (direction2 == direction1) {
                     stringbuilder.append("  ");
                 } else {
