--- a/net/minecraft/world/level/biome/FixedBiomeSource.java
+++ b/net/minecraft/world/level/biome/FixedBiomeSource.java
@@ -2,7 +_,7 @@
 
 import com.google.common.collect.Sets;
 import com.mojang.datafixers.util.Pair;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import java.util.Set;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
@@ -13,7 +_,7 @@
 import net.minecraft.world.level.LevelReader;
 
 public class FixedBiomeSource extends BiomeSource implements BiomeManager.NoiseBiomeSource {
-    public static final Codec<FixedBiomeSource> CODEC = Biome.CODEC.fieldOf("biome").xmap(FixedBiomeSource::new, p_204259_ -> p_204259_.biome).stable().codec();
+    public static final MapCodec<FixedBiomeSource> CODEC = Biome.CODEC.fieldOf("biome").xmap(FixedBiomeSource::new, p_204259_ -> p_204259_.biome).stable();
     private final Holder<Biome> biome;
 
     public FixedBiomeSource(Holder<Biome> p_204257_) {
@@ -26,7 +_,7 @@
     }
 
     @Override
-    protected Codec<? extends BiomeSource> codec() {
+    protected MapCodec<? extends BiomeSource> codec() {
         return CODEC;
     }
 
