--- a/net/minecraft/world/level/biome/MultiNoiseBiomeSource.java
+++ b/net/minecraft/world/level/biome/MultiNoiseBiomeSource.java
@@ -7,7 +_,6 @@
 import com.mojang.serialization.MapCodec;
 import java.util.List;
 import java.util.Optional;
-import java.util.function.Function;
 import java.util.stream.Stream;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Holder;
@@ -18,13 +_,12 @@
 
 public class MultiNoiseBiomeSource extends BiomeSource {
     private static final MapCodec<Holder<Biome>> ENTRY_CODEC = Biome.CODEC.fieldOf("biome");
-    public static final MapCodec<Climate.ParameterList<Holder<Biome>>> DIRECT_CODEC = Climate.ParameterList.<Holder<Biome>>codec(ENTRY_CODEC).fieldOf("biomes");
+    public static final MapCodec<Climate.ParameterList<Holder<Biome>>> DIRECT_CODEC = Climate.ParameterList.codec(ENTRY_CODEC).fieldOf("biomes");
     private static final MapCodec<Holder<MultiNoiseBiomeSourceParameterList>> PRESET_CODEC = MultiNoiseBiomeSourceParameterList.CODEC
         .fieldOf("preset")
         .withLifecycle(Lifecycle.stable());
-    public static final Codec<MultiNoiseBiomeSource> CODEC = Codec.mapEither(DIRECT_CODEC, PRESET_CODEC)
-        .xmap(MultiNoiseBiomeSource::new, p_275170_ -> p_275170_.parameters)
-        .codec();
+    public static final MapCodec<MultiNoiseBiomeSource> CODEC = Codec.mapEither(DIRECT_CODEC, PRESET_CODEC)
+        .xmap(MultiNoiseBiomeSource::new, p_275170_ -> p_275170_.parameters);
     private final Either<Climate.ParameterList<Holder<Biome>>, Holder<MultiNoiseBiomeSourceParameterList>> parameters;
 
     private MultiNoiseBiomeSource(Either<Climate.ParameterList<Holder<Biome>>, Holder<MultiNoiseBiomeSourceParameterList>> p_275370_) {
@@ -40,11 +_,7 @@
     }
 
     private Climate.ParameterList<Holder<Biome>> parameters() {
-        return this.parameters
-            .map(
-                (Function<? super Climate.ParameterList<Holder<Biome>>, ? extends Climate.ParameterList<Holder<Biome>>>)(p_275171_ -> p_275171_),
-                p_275172_ -> p_275172_.value().parameters()
-            );
+        return this.parameters.map(p_275171_ -> p_275171_, p_275172_ -> p_275172_.value().parameters());
     }
 
     @Override
@@ -53,7 +_,7 @@
     }
 
     @Override
-    protected Codec<? extends BiomeSource> codec() {
+    protected MapCodec<? extends BiomeSource> codec() {
         return CODEC;
     }
 
