--- a/net/minecraft/world/level/biome/MultiNoiseBiomeSourceParameterLists.java
+++ b/net/minecraft/world/level/biome/MultiNoiseBiomeSourceParameterLists.java
@@ -9,13 +_,11 @@
 public class MultiNoiseBiomeSourceParameterLists {
     public static final ResourceKey<MultiNoiseBiomeSourceParameterList> NETHER = register("nether");
     public static final ResourceKey<MultiNoiseBiomeSourceParameterList> OVERWORLD = register("overworld");
-    public static final ResourceKey<MultiNoiseBiomeSourceParameterList> POTATO = register("potato");
 
     public static void bootstrap(BootstrapContext<MultiNoiseBiomeSourceParameterList> p_321503_) {
         HolderGetter<Biome> holdergetter = p_321503_.lookup(Registries.BIOME);
         p_321503_.register(NETHER, new MultiNoiseBiomeSourceParameterList(MultiNoiseBiomeSourceParameterList.Preset.NETHER, holdergetter));
         p_321503_.register(OVERWORLD, new MultiNoiseBiomeSourceParameterList(MultiNoiseBiomeSourceParameterList.Preset.OVERWORLD, holdergetter));
-        p_321503_.register(POTATO, new MultiNoiseBiomeSourceParameterList(MultiNoiseBiomeSourceParameterList.Preset.POTATO, holdergetter));
     }
 
     private static ResourceKey<MultiNoiseBiomeSourceParameterList> register(String p_275281_) {
