--- a/net/minecraft/world/level/biome/TheEndBiomeSource.java
+++ b/net/minecraft/world/level/biome/TheEndBiomeSource.java
@@ -1,6 +_,6 @@
 package net.minecraft.world.level.biome;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.stream.Stream;
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.levelgen.DensityFunction;
 
 public class TheEndBiomeSource extends BiomeSource {
-    public static final Codec<TheEndBiomeSource> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<TheEndBiomeSource> CODEC = RecordCodecBuilder.mapCodec(
         p_255555_ -> p_255555_.group(
                     RegistryOps.retrieveElement(Biomes.THE_END),
                     RegistryOps.retrieveElement(Biomes.END_HIGHLANDS),
@@ -52,7 +_,7 @@
     }
 
     @Override
-    protected Codec<? extends BiomeSource> codec() {
+    protected MapCodec<? extends BiomeSource> codec() {
         return CODEC;
     }
 
