--- a/net/minecraft/client/telemetry/events/AggregatedTelemetryEvent.java
+++ b/net/minecraft/client/telemetry/events/AggregatedTelemetryEvent.java
@@ -25,7 +_,7 @@
     public void tick(TelemetryEventSender p_263410_) {
         if (this.shouldTakeSample()) {
             this.takeSample();
-            ++this.sampleCount;
+            this.sampleCount++;
             this.lastSampleTime = Instant.now();
         }
 
