--- a/net/minecraft/client/telemetry/events/GameLoadTimesEvent.java
+++ b/net/minecraft/client/telemetry/events/GameLoadTimesEvent.java
@@ -58,13 +_,13 @@
         p_286524_.send(
             TelemetryEventType.GAME_LOAD_TIMES,
             p_286285_ -> {
-                synchronized(this) {
+                synchronized (this) {
                     this.measurements
                         .forEach(
                             (p_286804_, p_286275_) -> {
                                 if (!p_286275_.isRunning()) {
                                     long i = p_286275_.elapsed(TimeUnit.MILLISECONDS);
-                                    p_286285_.put(p_286804_, new GameLoadTimesEvent.Measurement((int)i));
+                                    p_286285_.put((TelemetryProperty<GameLoadTimesEvent.Measurement>)p_286804_, new GameLoadTimesEvent.Measurement((int)i));
                                 } else {
                                     LOGGER.warn(
                                         "Measurement {} was discarded since it was still ongoing when the event {} was sent.",
