--- a/net/minecraft/world/level/levelgen/structure/placement/RandomSpreadStructurePlacement.java
+++ b/net/minecraft/world/level/levelgen/structure/placement/RandomSpreadStructurePlacement.java
@@ -2,34 +_,31 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.DataResult;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Optional;
 import net.minecraft.core.Vec3i;
-import net.minecraft.util.ExtraCodecs;
 import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.chunk.ChunkGeneratorStructureState;
 import net.minecraft.world.level.levelgen.LegacyRandomSource;
 import net.minecraft.world.level.levelgen.WorldgenRandom;
 
 public class RandomSpreadStructurePlacement extends StructurePlacement {
-    public static final Codec<RandomSpreadStructurePlacement> CODEC = ExtraCodecs.validate(
-            RecordCodecBuilder.mapCodec(
-                p_204996_ -> placementCodec(p_204996_)
-                        .and(
-                            p_204996_.group(
-                                Codec.intRange(0, 4096).fieldOf("spacing").forGetter(RandomSpreadStructurePlacement::spacing),
-                                Codec.intRange(0, 4096).fieldOf("separation").forGetter(RandomSpreadStructurePlacement::separation),
-                                RandomSpreadType.CODEC
-                                    .optionalFieldOf("spread_type", RandomSpreadType.LINEAR)
-                                    .forGetter(RandomSpreadStructurePlacement::spreadType)
-                            )
+    public static final MapCodec<RandomSpreadStructurePlacement> CODEC = RecordCodecBuilder.<RandomSpreadStructurePlacement>mapCodec(
+            p_204996_ -> placementCodec(p_204996_)
+                    .and(
+                        p_204996_.group(
+                            Codec.intRange(0, 4096).fieldOf("spacing").forGetter(RandomSpreadStructurePlacement::spacing),
+                            Codec.intRange(0, 4096).fieldOf("separation").forGetter(RandomSpreadStructurePlacement::separation),
+                            RandomSpreadType.CODEC
+                                .optionalFieldOf("spread_type", RandomSpreadType.LINEAR)
+                                .forGetter(RandomSpreadStructurePlacement::spreadType)
                         )
-                        .apply(p_204996_, RandomSpreadStructurePlacement::new)
-            ),
-            RandomSpreadStructurePlacement::validate
+                    )
+                    .apply(p_204996_, RandomSpreadStructurePlacement::new)
         )
-        .codec();
+        .validate(RandomSpreadStructurePlacement::validate);
     private final int spacing;
     private final int separation;
     private final RandomSpreadType spreadType;
