--- a/net/minecraft/world/level/levelgen/structure/placement/RandomSpreadType.java
+++ b/net/minecraft/world/level/levelgen/structure/placement/RandomSpreadType.java
@@ -21,7 +_,7 @@
     }
 
     public int evaluate(RandomSource p_227019_, int p_227020_) {
-        return switch(this) {
+        return switch (this) {
             case LINEAR -> p_227019_.nextInt(p_227020_);
             case TRIANGULAR -> (p_227019_.nextInt(p_227020_) + p_227019_.nextInt(p_227020_)) / 2;
         };
