--- a/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
+++ b/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
@@ -50,7 +_,7 @@
 
     @Override
     public void tick() {
-        ++this.time;
+        this.time++;
         if (this.level() instanceof ServerLevel) {
             BlockPos blockpos = this.blockPosition();
             if (((ServerLevel)this.level()).getDragonFight() != null && this.level().getBlockState(blockpos).isAir()) {
