--- a/net/minecraft/util/worldupdate/WorldUpgrader.java
+++ b/net/minecraft/util/worldupdate/WorldUpgrader.java
@@ -179,11 +_,11 @@
                 float f = (float)WorldUpgrader.this.totalFiles;
                 WorldUpgrader.this.status = this.upgradingStatus;
 
-                while(WorldUpgrader.this.running) {
+                while (WorldUpgrader.this.running) {
                     boolean flag = false;
                     float f1 = 0.0F;
 
-                    for(WorldUpgrader.DimensionToUpgrade<T> dimensiontoupgrade : list) {
+                    for (WorldUpgrader.DimensionToUpgrade<T> dimensiontoupgrade : list) {
                         ResourceKey<Level> resourcekey = dimensiontoupgrade.dimensionKey;
                         ListIterator<WorldUpgrader.FileToUpgrade> listiterator = dimensiontoupgrade.files;
                         T t = dimensiontoupgrade.storage;
@@ -191,7 +_,7 @@
                             WorldUpgrader.FileToUpgrade worldupgrader$filetoupgrade = listiterator.next();
                             boolean flag1 = true;
 
-                            for(ChunkPos chunkpos : worldupgrader$filetoupgrade.chunksToUpgrade) {
+                            for (ChunkPos chunkpos : worldupgrader$filetoupgrade.chunksToUpgrade) {
                                 flag1 = flag1 && this.processOnePosition(resourcekey, t, chunkpos);
                                 flag = true;
                             }
@@ -218,7 +_,7 @@
 
                 WorldUpgrader.this.status = this.finishedStatus;
 
-                for(WorldUpgrader.DimensionToUpgrade<T> dimensiontoupgrade1 : list) {
+                for (WorldUpgrader.DimensionToUpgrade<T> dimensiontoupgrade1 : list) {
                     try {
                         dimensiontoupgrade1.storage.close();
                     } catch (Exception exception) {
@@ -231,7 +_,7 @@
         private List<WorldUpgrader.DimensionToUpgrade<T>> getDimensionsToUpgrade() {
             List<WorldUpgrader.DimensionToUpgrade<T>> list = Lists.newArrayList();
 
-            for(ResourceKey<Level> resourcekey : WorldUpgrader.this.levels) {
+            for (ResourceKey<Level> resourcekey : WorldUpgrader.this.levels) {
                 RegionStorageInfo regionstorageinfo = new RegionStorageInfo(WorldUpgrader.this.levelStorage.getLevelId(), resourcekey, this.type);
                 Path path = WorldUpgrader.this.levelStorage.getDimensionPath(resourcekey).resolve(this.folderName);
                 T t = this.createStorage(regionstorageinfo, path);
@@ -246,8 +_,8 @@
 
         private ListIterator<WorldUpgrader.FileToUpgrade> getFilesToProcess(RegionStorageInfo p_325920_, Path p_326489_) {
             List<WorldUpgrader.FileToUpgrade> list = getAllChunkPositions(p_325920_, p_326489_);
-            WorldUpgrader.this.totalFiles += list.size();
-            WorldUpgrader.this.totalChunks += list.stream().mapToInt(p_321550_ -> p_321550_.chunksToUpgrade.size()).sum();
+            WorldUpgrader.this.totalFiles = WorldUpgrader.this.totalFiles + list.size();
+            WorldUpgrader.this.totalChunks = WorldUpgrader.this.totalChunks + list.stream().mapToInt(p_321550_ -> p_321550_.chunksToUpgrade.size()).sum();
             return list.listIterator();
         }
 
@@ -258,7 +_,7 @@
             } else {
                 List<WorldUpgrader.FileToUpgrade> list = Lists.newArrayList();
 
-                for(File file1 : afile) {
+                for (File file1 : afile) {
                     Matcher matcher = WorldUpgrader.REGEX.matcher(file1.getName());
                     if (matcher.matches()) {
                         int i = Integer.parseInt(matcher.group(1)) << 5;
@@ -266,8 +_,8 @@
                         List<ChunkPos> list1 = Lists.newArrayList();
 
                         try (RegionFile regionfile = new RegionFile(p_326177_, file1.toPath(), p_326228_, true)) {
-                            for(int k = 0; k < 32; ++k) {
-                                for(int l = 0; l < 32; ++l) {
+                            for (int k = 0; k < 32; k++) {
+                                for (int l = 0; l < 32; l++) {
                                     ChunkPos chunkpos = new ChunkPos(k + i, l + j);
                                     if (regionfile.doesChunkExist(chunkpos)) {
                                         list1.add(chunkpos);
@@ -303,9 +_,9 @@
             }
 
             if (flag) {
-                ++WorldUpgrader.this.converted;
+                WorldUpgrader.this.converted++;
             } else {
-                ++WorldUpgrader.this.skipped;
+                WorldUpgrader.this.skipped++;
             }
 
             return flag;
@@ -363,7 +_,7 @@
                     compoundtag1.remove("isLightOn");
                     ListTag listtag = compoundtag1.getList("sections", 10);
 
-                    for(int j = 0; j < listtag.size(); ++j) {
+                    for (int j = 0; j < listtag.size(); j++) {
                         CompoundTag compoundtag2 = listtag.getCompound(j);
                         flag = flag || compoundtag2.contains("BlockLight");
                         compoundtag2.remove("BlockLight");
