--- a/net/minecraft/client/resources/model/ModelBakery.java
+++ b/net/minecraft/client/resources/model/ModelBakery.java
@@ -65,7 +_,13 @@
     private final Map<Identifier, ClientItem> clientInfos;
     private final Map<Identifier, ResolvedModel> resolvedModels;
     private final ResolvedModel missingModel;
+    private final net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels standaloneModels;
+    private final net.neoforged.neoforge.client.entity.animation.json.AnimationLoader.PendingAnimations pendingAnimations;
 
+    /**
+     * @deprecated Neo: use {@link #ModelBakery(EntityModelSet, MaterialSet, PlayerSkinRenderCache, Map, Map, Map, ResolvedModel, net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels, net.neoforged.neoforge.client.entity.animation.json.AnimationLoader.PendingAnimations)} instead
+     */
+    @Deprecated
     public ModelBakery(
         EntityModelSet entityModelSet,
         MaterialSet materials,
@@ -75,6 +_,20 @@
         Map<Identifier, ResolvedModel> resolvedModels,
         ResolvedModel missingModel
     ) {
+        this(entityModelSet, materials, playerSkinRenderCache, unbakedBlockStateModels, clientInfos, resolvedModels, missingModel, net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels.EMPTY, net.neoforged.neoforge.client.entity.animation.json.AnimationLoader.PendingAnimations.EMPTY);
+    }
+
+    public ModelBakery(
+        EntityModelSet entityModelSet,
+        MaterialSet materials,
+        PlayerSkinRenderCache playerSkinRenderCache,
+        Map<BlockState, BlockStateModel.UnbakedRoot> unbakedBlockStateModels,
+        Map<Identifier, ClientItem> clientInfos,
+        Map<Identifier, ResolvedModel> resolvedModels,
+        ResolvedModel missingModel,
+        net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels standaloneModels,
+        net.neoforged.neoforge.client.entity.animation.json.AnimationLoader.PendingAnimations pendingAnimations
+    ) {
         this.entityModelSet = entityModelSet;
         this.materials = materials;
         this.playerSkinRenderCache = playerSkinRenderCache;
@@ -82,6 +_,8 @@
         this.clientInfos = clientInfos;
         this.resolvedModels = resolvedModels;
         this.missingModel = missingModel;
+        this.standaloneModels = standaloneModels;
+        this.pendingAnimations = pendingAnimations;
     }
 
     public CompletableFuture<ModelBakery.BakingResult> bakeModels(SpriteGetter sprites, Executor taskExecutor) {
@@ -105,7 +_,7 @@
                     return clientInfo.model()
                         .bake(
                             new ItemModel.BakingContext(
-                                baker, this.entityModelSet, this.materials, this.playerSkinRenderCache, missingModels.item, clientInfo.registrySwapper()
+                                baker, this.entityModelSet, this.materials, this.playerSkinRenderCache, missingModels.item, clientInfo.registrySwapper(), this.pendingAnimations
                             )
                         );
                 } catch (Exception var6x) {
@@ -115,6 +_,8 @@
             },
             taskExecutor
         );
+        CompletableFuture<net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.BakedModels> standaloneModelsFuture =
+                net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.bake(this.standaloneModels, baker, taskExecutor);
         Map<Identifier, ClientItem.Properties> itemStackModelProperties = new HashMap<>(this.clientInfos.size());
         this.clientInfos.forEach((id, clientInfo) -> {
             ClientItem.Properties properties = clientInfo.properties();
@@ -122,13 +_,15 @@
                 itemStackModelProperties.put(id, properties);
             }
         });
+        bakedBlockStateModelFuture = bakedBlockStateModelFuture.thenCombine(standaloneModelsFuture, (stateModels, standaloneModels) -> stateModels);
         return bakedBlockStateModelFuture.thenCombine(
             bakedItemStackModelFuture,
             (bakedBlockStateModels, bakedItemStateModels) -> new ModelBakery.BakingResult(
                 missingModels,
                 (Map<BlockState, BlockStateModel>)bakedBlockStateModels,
                 (Map<Identifier, ItemModel>)bakedItemStateModels,
-                itemStackModelProperties
+                itemStackModelProperties,
+                standaloneModelsFuture.join()
             )
         );
     }
