--- a/net/minecraft/world/level/timers/FunctionCallback.java
+++ b/net/minecraft/world/level/timers/FunctionCallback.java
@@ -1,5 +_,6 @@
 package net.minecraft.world.level.timers;
 
+import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.commands.functions.CommandFunction;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.resources.ResourceLocation;
@@ -15,7 +_,8 @@
 
     public void handle(MinecraftServer p_82172_, TimerQueue<MinecraftServer> p_82173_, long p_82174_) {
         ServerFunctionManager serverfunctionmanager = p_82172_.getFunctions();
-        serverfunctionmanager.get(this.functionId).ifPresent(p_305770_ -> serverfunctionmanager.execute(p_305770_, serverfunctionmanager.getGameLoopSender()));
+        serverfunctionmanager.get(this.functionId)
+            .ifPresent(p_305770_ -> serverfunctionmanager.execute((CommandFunction<CommandSourceStack>)p_305770_, serverfunctionmanager.getGameLoopSender()));
     }
 
     public static class Serializer extends TimerCallback.Serializer<MinecraftServer, FunctionCallback> {
