--- a/net/minecraft/client/player/RemotePlayer.java
+++ b/net/minecraft/client/player/RemotePlayer.java
@@ -33,6 +_,7 @@
 
     @Override
     public boolean hurt(DamageSource p_108772_, float p_108773_) {
+        net.neoforged.neoforge.common.CommonHooks.onPlayerAttack(this, p_108772_, p_108773_);
         return true;
     }
 
@@ -46,12 +_,12 @@
     public void aiStep() {
         if (this.lerpSteps > 0) {
             this.lerpPositionAndRotationStep(this.lerpSteps, this.lerpX, this.lerpY, this.lerpZ, this.lerpYRot, this.lerpXRot);
-            --this.lerpSteps;
+            this.lerpSteps--;
         }
 
         if (this.lerpHeadSteps > 0) {
             this.lerpHeadRotationStep(this.lerpHeadSteps, this.lerpYHeadRot);
-            --this.lerpHeadSteps;
+            this.lerpHeadSteps--;
         }
 
         if (this.lerpDeltaMovementSteps > 0) {
@@ -62,7 +_,7 @@
                     (this.lerpDeltaMovement.z - this.getDeltaMovement().z) / (double)this.lerpDeltaMovementSteps
                 )
             );
-            --this.lerpDeltaMovementSteps;
+            this.lerpDeltaMovementSteps--;
         }
 
         this.oBob = this.bob;
@@ -74,7 +_,7 @@
             f = 0.0F;
         }
 
-        this.bob += (f - this.bob) * 0.4F;
+        this.bob = this.bob + (f - this.bob) * 0.4F;
         this.level().getProfiler().push("push");
         this.pushEntities();
         this.level().getProfiler().pop();
