--- a/net/minecraft/client/gui/screens/advancements/AdvancementWidgetType.java
+++ b/net/minecraft/client/gui/screens/advancements/AdvancementWidgetType.java
@@ -37,10 +_,11 @@
     }
 
     public ResourceLocation frameSprite(AdvancementType p_312712_) {
-        return switch(p_312712_) {
+        return switch (p_312712_) {
             case TASK -> this.taskFrameSprite;
             case CHALLENGE -> this.challengeFrameSprite;
             case GOAL -> this.goalFrameSprite;
+            default -> throw new MatchException(null, null);
         };
     }
 }
