--- a/net/minecraft/client/renderer/blockentity/HangingSignRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/HangingSignRenderer.java
@@ -47,7 +_,7 @@
         this.hangingSignModels = WoodType.values()
             .collect(
                 ImmutableMap.toImmutableMap(
-                    p_249901_ -> p_249901_,
+                    p_249901_ -> (WoodType)p_249901_,
                     p_251956_ -> new HangingSignRenderer.HangingSignModel(p_248772_.bakeLayer(ModelLayers.createHangingSignModelName(p_251956_)))
                 )
             );
