--- a/net/minecraft/data/worldgen/features/MiscOverworldFeatures.java
+++ b/net/minecraft/data/worldgen/features/MiscOverworldFeatures.java
@@ -23,7 +_,6 @@
     public static final ResourceKey<ConfiguredFeature<?, ?>> ICE_SPIKE = FeatureUtils.createKey("ice_spike");
     public static final ResourceKey<ConfiguredFeature<?, ?>> ICE_PATCH = FeatureUtils.createKey("ice_patch");
     public static final ResourceKey<ConfiguredFeature<?, ?>> FOREST_ROCK = FeatureUtils.createKey("forest_rock");
-    public static final ResourceKey<ConfiguredFeature<?, ?>> POTATO_CLOUD = FeatureUtils.createKey("potato_cloud");
     public static final ResourceKey<ConfiguredFeature<?, ?>> ICEBERG_PACKED = FeatureUtils.createKey("iceberg_packed");
     public static final ResourceKey<ConfiguredFeature<?, ?>> ICEBERG_BLUE = FeatureUtils.createKey("iceberg_blue");
     public static final ResourceKey<ConfiguredFeature<?, ?>> BLUE_ICE = FeatureUtils.createKey("blue_ice");
@@ -36,7 +_,6 @@
     public static final ResourceKey<ConfiguredFeature<?, ?>> BONUS_CHEST = FeatureUtils.createKey("bonus_chest");
     public static final ResourceKey<ConfiguredFeature<?, ?>> VOID_START_PLATFORM = FeatureUtils.createKey("void_start_platform");
     public static final ResourceKey<ConfiguredFeature<?, ?>> DESERT_WELL = FeatureUtils.createKey("desert_well");
-    public static final ResourceKey<ConfiguredFeature<?, ?>> HASH_WELL = FeatureUtils.createKey("hash_well");
     public static final ResourceKey<ConfiguredFeature<?, ?>> SPRING_LAVA_OVERWORLD = FeatureUtils.createKey("spring_lava_overworld");
     public static final ResourceKey<ConfiguredFeature<?, ?>> SPRING_LAVA_FROZEN = FeatureUtils.createKey("spring_lava_frozen");
     public static final ResourceKey<ConfiguredFeature<?, ?>> SPRING_WATER = FeatureUtils.createKey("spring_water");
@@ -50,25 +_,13 @@
             new DiskConfiguration(
                 RuleBasedBlockStateProvider.simple(Blocks.PACKED_ICE),
                 BlockPredicate.matchesBlocks(
-                    List.of(
-                        Blocks.DIRT,
-                        Blocks.TERREDEPOMME,
-                        Blocks.GRASS_BLOCK,
-                        Blocks.PEELGRASS_BLOCK,
-                        Blocks.CORRUPTED_PEELGRASS_BLOCK,
-                        Blocks.PODZOL,
-                        Blocks.COARSE_DIRT,
-                        Blocks.MYCELIUM,
-                        Blocks.SNOW_BLOCK,
-                        Blocks.ICE
-                    )
+                    List.of(Blocks.DIRT, Blocks.GRASS_BLOCK, Blocks.PODZOL, Blocks.COARSE_DIRT, Blocks.MYCELIUM, Blocks.SNOW_BLOCK, Blocks.ICE)
                 ),
                 UniformInt.of(2, 3),
                 1
             )
         );
         FeatureUtils.register(p_321835_, FOREST_ROCK, Feature.FOREST_ROCK, new BlockStateConfiguration(Blocks.MOSSY_COBBLESTONE.defaultBlockState()));
-        FeatureUtils.register(p_321835_, POTATO_CLOUD, Feature.CLOUD, new BlockStateConfiguration(Blocks.FLOATATO.defaultBlockState()));
         FeatureUtils.register(p_321835_, ICEBERG_PACKED, Feature.ICEBERG, new BlockStateConfiguration(Blocks.PACKED_ICE.defaultBlockState()));
         FeatureUtils.register(p_321835_, ICEBERG_BLUE, Feature.ICEBERG, new BlockStateConfiguration(Blocks.BLUE_ICE.defaultBlockState()));
         FeatureUtils.register(p_321835_, BLUE_ICE, Feature.BLUE_ICE);
@@ -85,10 +_,7 @@
             DISK_CLAY,
             Feature.DISK,
             new DiskConfiguration(
-                RuleBasedBlockStateProvider.simple(Blocks.CLAY),
-                BlockPredicate.matchesBlocks(List.of(Blocks.DIRT, Blocks.CLAY, Blocks.TERREDEPOMME)),
-                UniformInt.of(2, 3),
-                1
+                RuleBasedBlockStateProvider.simple(Blocks.CLAY), BlockPredicate.matchesBlocks(List.of(Blocks.DIRT, Blocks.CLAY)), UniformInt.of(2, 3), 1
             )
         );
         FeatureUtils.register(
@@ -97,9 +_,7 @@
             Feature.DISK,
             new DiskConfiguration(
                 RuleBasedBlockStateProvider.simple(Blocks.GRAVEL),
-                BlockPredicate.matchesBlocks(
-                    List.of(Blocks.DIRT, Blocks.GRASS_BLOCK, Blocks.TERREDEPOMME, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK)
-                ),
+                BlockPredicate.matchesBlocks(List.of(Blocks.DIRT, Blocks.GRASS_BLOCK)),
                 UniformInt.of(2, 5),
                 2
             )
@@ -117,9 +_,7 @@
                         )
                     )
                 ),
-                BlockPredicate.matchesBlocks(
-                    List.of(Blocks.DIRT, Blocks.GRASS_BLOCK, Blocks.TERREDEPOMME, Blocks.PEELGRASS_BLOCK, Blocks.CORRUPTED_PEELGRASS_BLOCK)
-                ),
+                BlockPredicate.matchesBlocks(List.of(Blocks.DIRT, Blocks.GRASS_BLOCK)),
                 UniformInt.of(2, 6),
                 2
             )
@@ -143,7 +_,7 @@
                         )
                     )
                 ),
-                BlockPredicate.matchesBlocks(List.of(Blocks.DIRT, Blocks.TERREDEPOMME, Blocks.MUD)),
+                BlockPredicate.matchesBlocks(List.of(Blocks.DIRT, Blocks.MUD)),
                 UniformInt.of(2, 6),
                 2
             )
@@ -151,7 +_,6 @@
         FeatureUtils.register(p_321835_, BONUS_CHEST, Feature.BONUS_CHEST);
         FeatureUtils.register(p_321835_, VOID_START_PLATFORM, Feature.VOID_START_PLATFORM);
         FeatureUtils.register(p_321835_, DESERT_WELL, Feature.DESERT_WELL);
-        FeatureUtils.register(p_321835_, HASH_WELL, Feature.HASH_WELL);
         FeatureUtils.register(
             p_321835_,
             SPRING_LAVA_OVERWORLD,
@@ -164,15 +_,13 @@
                 HolderSet.direct(
                     Block::builtInRegistryHolder,
                     Blocks.STONE,
-                    Blocks.POTONE,
                     Blocks.GRANITE,
                     Blocks.DIORITE,
                     Blocks.ANDESITE,
                     Blocks.DEEPSLATE,
                     Blocks.TUFF,
                     Blocks.CALCITE,
-                    Blocks.DIRT,
-                    Blocks.TERREDEPOMME
+                    Blocks.DIRT
                 )
             )
         );
@@ -200,7 +_,6 @@
                 HolderSet.direct(
                     Block::builtInRegistryHolder,
                     Blocks.STONE,
-                    Blocks.POTONE,
                     Blocks.GRANITE,
                     Blocks.DIORITE,
                     Blocks.ANDESITE,
@@ -208,7 +_,6 @@
                     Blocks.TUFF,
                     Blocks.CALCITE,
                     Blocks.DIRT,
-                    Blocks.TERREDEPOMME,
                     Blocks.SNOW_BLOCK,
                     Blocks.POWDER_SNOW,
                     Blocks.PACKED_ICE
