--- a/net/minecraft/data/worldgen/features/TreeFeatures.java
+++ b/net/minecraft/data/worldgen/features/TreeFeatures.java
@@ -66,9 +_,6 @@
     public static final ResourceKey<ConfiguredFeature<?, ?>> CRIMSON_FUNGUS = FeatureUtils.createKey("crimson_fungus");
     public static final ResourceKey<ConfiguredFeature<?, ?>> CRIMSON_FUNGUS_PLANTED = FeatureUtils.createKey("crimson_fungus_planted");
     public static final ResourceKey<ConfiguredFeature<?, ?>> WARPED_FUNGUS = FeatureUtils.createKey("warped_fungus");
-    public static final ResourceKey<ConfiguredFeature<?, ?>> POTATO_TREE_TALL = FeatureUtils.createKey("potato_tree_tall");
-    public static final ResourceKey<ConfiguredFeature<?, ?>> POTATO_TREE = FeatureUtils.createKey("potato_tree");
-    public static final ResourceKey<ConfiguredFeature<?, ?>> MOTHER_POTATO_TREE = FeatureUtils.createKey("mother_potato_tree");
     public static final ResourceKey<ConfiguredFeature<?, ?>> WARPED_FUNGUS_PLANTED = FeatureUtils.createKey("warped_fungus_planted");
     public static final ResourceKey<ConfiguredFeature<?, ?>> HUGE_BROWN_MUSHROOM = FeatureUtils.createKey("huge_brown_mushroom");
     public static final ResourceKey<ConfiguredFeature<?, ?>> HUGE_RED_MUSHROOM = FeatureUtils.createKey("huge_red_mushroom");
@@ -189,7 +_,6 @@
             Blocks.CORNFLOWER,
             Blocks.WITHER_ROSE,
             Blocks.LILY_OF_THE_VALLEY,
-            Blocks.POTATO_FLOWER,
             Blocks.BROWN_MUSHROOM,
             Blocks.RED_MUSHROOM,
             Blocks.WHEAT,
@@ -313,150 +_,11 @@
                 2
             )
         );
-        BeehiveDecorator beehivedecorator = new BeehiveDecorator(0.002F, true);
-        BeehiveDecorator beehivedecorator1 = new BeehiveDecorator(0.01F, true);
-        BeehiveDecorator beehivedecorator2 = new BeehiveDecorator(0.02F, true);
-        BeehiveDecorator beehivedecorator3 = new BeehiveDecorator(0.05F, true);
-        BeehiveDecorator beehivedecorator4 = new BeehiveDecorator(0.5F, false);
-        BeehiveDecorator beehivedecorator5 = new BeehiveDecorator(1.0F, true);
-        FeatureUtils.register(
-            p_321605_,
-            MOTHER_POTATO_TREE,
-            Feature.TREE,
-            new TreeConfiguration.TreeConfigurationBuilder(
-                    BlockStateProvider.simple(Blocks.POTATO_STEM),
-                    new UpwardsBranchingTrunkPlacer(32, 1, 20, UniformInt.of(1, 10), 0.4F, UniformInt.of(0, 1), holdergetter.getOrThrow(BlockTags.LOGS), false),
-                    BlockStateProvider.simple(Blocks.POTATO_LEAVES),
-                    new AcaciaFoliagePlacer(UniformInt.of(3, 4), ConstantInt.of(0)),
-                    Optional.empty(),
-                    new TwoLayersFeatureSize(3, 0, 2)
-                )
-                .decorators(
-                    List.of(
-                        new AttachedToLeavesDecorator(
-                            0.1F,
-                            false,
-                            1,
-                            0,
-                            List.of(
-                                BlockStateProvider.simple(Blocks.POTATO_PEDICULE.defaultBlockState()),
-                                BlockStateProvider.simple(Blocks.POTATO_FRUIT.defaultBlockState())
-                            ),
-                            3,
-                            List.of(Direction.DOWN)
-                        ),
-                        new AttachedToLeavesDecorator(
-                            0.5F,
-                            true,
-                            1,
-                            0,
-                            List.of(
-                                BlockStateProvider.simple(Blocks.POTATO_PEDICULE.defaultBlockState()),
-                                BlockStateProvider.simple(Blocks.POTATO_FRUIT.defaultBlockState())
-                            ),
-                            3,
-                            List.of(Direction.DOWN)
-                        ),
-                        beehivedecorator5,
-                        beehivedecorator4,
-                        beehivedecorator4,
-                        beehivedecorator4,
-                        beehivedecorator4,
-                        beehivedecorator4,
-                        beehivedecorator4
-                    )
-                )
-                .ignoreVines()
-                .build()
-        );
-        FeatureUtils.register(
-            p_321605_,
-            POTATO_TREE_TALL,
-            Feature.TREE,
-            new TreeConfiguration.TreeConfigurationBuilder(
-                    BlockStateProvider.simple(Blocks.POTATO_STEM),
-                    new UpwardsBranchingTrunkPlacer(4, 20, 20, UniformInt.of(1, 8), 0.4F, UniformInt.of(0, 1), holdergetter.getOrThrow(BlockTags.LOGS), false),
-                    BlockStateProvider.simple(Blocks.POTATO_LEAVES),
-                    new AcaciaFoliagePlacer(UniformInt.of(2, 4), ConstantInt.of(0)),
-                    Optional.empty(),
-                    new TwoLayersFeatureSize(3, 0, 2)
-                )
-                .decorators(
-                    List.of(
-                        new AttachedToLeavesDecorator(
-                            0.02F,
-                            false,
-                            1,
-                            0,
-                            List.of(
-                                BlockStateProvider.simple(Blocks.POTATO_PEDICULE.defaultBlockState()),
-                                BlockStateProvider.simple(Blocks.POTATO_FRUIT.defaultBlockState())
-                            ),
-                            3,
-                            List.of(Direction.DOWN)
-                        ),
-                        new AttachedToLeavesDecorator(
-                            0.3F,
-                            true,
-                            1,
-                            0,
-                            List.of(
-                                BlockStateProvider.simple(Blocks.POTATO_PEDICULE.defaultBlockState()),
-                                BlockStateProvider.simple(Blocks.POTATO_FRUIT.defaultBlockState())
-                            ),
-                            3,
-                            List.of(Direction.DOWN)
-                        ),
-                        beehivedecorator4
-                    )
-                )
-                .ignoreVines()
-                .build()
-        );
-        FeatureUtils.register(
-            p_321605_,
-            POTATO_TREE,
-            Feature.TREE,
-            new TreeConfiguration.TreeConfigurationBuilder(
-                    BlockStateProvider.simple(Blocks.POTATO_STEM),
-                    new UpwardsBranchingTrunkPlacer(2, 1, 12, UniformInt.of(1, 6), 0.5F, UniformInt.of(0, 1), holdergetter.getOrThrow(BlockTags.LOGS), false),
-                    BlockStateProvider.simple(Blocks.POTATO_LEAVES),
-                    new AcaciaFoliagePlacer(UniformInt.of(2, 3), ConstantInt.of(0)),
-                    Optional.empty(),
-                    new TwoLayersFeatureSize(3, 0, 2)
-                )
-                .decorators(
-                    List.of(
-                        new AttachedToLeavesDecorator(
-                            0.005F,
-                            false,
-                            1,
-                            0,
-                            List.of(
-                                BlockStateProvider.simple(Blocks.POTATO_PEDICULE.defaultBlockState()),
-                                BlockStateProvider.simple(Blocks.POTATO_FRUIT.defaultBlockState())
-                            ),
-                            3,
-                            List.of(Direction.DOWN)
-                        ),
-                        new AttachedToLeavesDecorator(
-                            0.05F,
-                            true,
-                            1,
-                            0,
-                            List.of(
-                                BlockStateProvider.simple(Blocks.POTATO_PEDICULE.defaultBlockState()),
-                                BlockStateProvider.simple(Blocks.POTATO_FRUIT.defaultBlockState())
-                            ),
-                            3,
-                            List.of(Direction.DOWN)
-                        ),
-                        beehivedecorator1
-                    )
-                )
-                .ignoreVines()
-                .build()
-        );
+        BeehiveDecorator beehivedecorator = new BeehiveDecorator(0.002F);
+        BeehiveDecorator beehivedecorator1 = new BeehiveDecorator(0.01F);
+        BeehiveDecorator beehivedecorator2 = new BeehiveDecorator(0.02F);
+        BeehiveDecorator beehivedecorator3 = new BeehiveDecorator(0.05F);
+        BeehiveDecorator beehivedecorator4 = new BeehiveDecorator(1.0F);
         FeatureUtils.register(p_321605_, OAK, Feature.TREE, createOak().build());
         FeatureUtils.register(
             p_321605_,
@@ -571,7 +_,7 @@
                 .build()
         );
         FeatureUtils.register(p_321605_, SUPER_BIRCH_BEES_0002, Feature.TREE, createSuperBirch().decorators(ImmutableList.of(beehivedecorator)).build());
-        FeatureUtils.register(p_321605_, SUPER_BIRCH_BEES, Feature.TREE, createSuperBirch().decorators(ImmutableList.of(beehivedecorator5)).build());
+        FeatureUtils.register(p_321605_, SUPER_BIRCH_BEES, Feature.TREE, createSuperBirch().decorators(ImmutableList.of(beehivedecorator4)).build());
         FeatureUtils.register(
             p_321605_,
             SWAMP_OAK,
@@ -617,7 +_,7 @@
             new TreeConfiguration.TreeConfigurationBuilder(
                     BlockStateProvider.simple(Blocks.MANGROVE_LOG),
                     new UpwardsBranchingTrunkPlacer(
-                        2, 1, 4, UniformInt.of(1, 4), 0.5F, UniformInt.of(0, 1), holdergetter.getOrThrow(BlockTags.MANGROVE_LOGS_CAN_GROW_THROUGH), true
+                        2, 1, 4, UniformInt.of(1, 4), 0.5F, UniformInt.of(0, 1), holdergetter.getOrThrow(BlockTags.MANGROVE_LOGS_CAN_GROW_THROUGH)
                     ),
                     BlockStateProvider.simple(Blocks.MANGROVE_LEAVES),
                     new RandomSpreadFoliagePlacer(ConstantInt.of(3), ConstantInt.of(0), ConstantInt.of(2), 70),
@@ -643,17 +_,14 @@
                         new LeaveVineDecorator(0.125F),
                         new AttachedToLeavesDecorator(
                             0.14F,
-                            false,
                             1,
                             0,
-                            List.of(
-                                new RandomizedIntStateProvider(
-                                    BlockStateProvider.simple(
-                                        Blocks.MANGROVE_PROPAGULE.defaultBlockState().setValue(MangrovePropaguleBlock.HANGING, Boolean.valueOf(true))
-                                    ),
-                                    MangrovePropaguleBlock.AGE,
-                                    UniformInt.of(0, 4)
-                                )
+                            new RandomizedIntStateProvider(
+                                BlockStateProvider.simple(
+                                    Blocks.MANGROVE_PROPAGULE.defaultBlockState().setValue(MangrovePropaguleBlock.HANGING, Boolean.valueOf(true))
+                                ),
+                                MangrovePropaguleBlock.AGE,
+                                UniformInt.of(0, 4)
                             ),
                             2,
                             List.of(Direction.DOWN)
@@ -671,7 +_,7 @@
             new TreeConfiguration.TreeConfigurationBuilder(
                     BlockStateProvider.simple(Blocks.MANGROVE_LOG),
                     new UpwardsBranchingTrunkPlacer(
-                        4, 1, 9, UniformInt.of(1, 6), 0.5F, UniformInt.of(0, 1), holdergetter.getOrThrow(BlockTags.MANGROVE_LOGS_CAN_GROW_THROUGH), true
+                        4, 1, 9, UniformInt.of(1, 6), 0.5F, UniformInt.of(0, 1), holdergetter.getOrThrow(BlockTags.MANGROVE_LOGS_CAN_GROW_THROUGH)
                     ),
                     BlockStateProvider.simple(Blocks.MANGROVE_LEAVES),
                     new RandomSpreadFoliagePlacer(ConstantInt.of(3), ConstantInt.of(0), ConstantInt.of(2), 70),
@@ -697,17 +_,14 @@
                         new LeaveVineDecorator(0.125F),
                         new AttachedToLeavesDecorator(
                             0.14F,
-                            false,
                             1,
                             0,
-                            List.of(
-                                new RandomizedIntStateProvider(
-                                    BlockStateProvider.simple(
-                                        Blocks.MANGROVE_PROPAGULE.defaultBlockState().setValue(MangrovePropaguleBlock.HANGING, Boolean.valueOf(true))
-                                    ),
-                                    MangrovePropaguleBlock.AGE,
-                                    UniformInt.of(0, 4)
-                                )
+                            new RandomizedIntStateProvider(
+                                BlockStateProvider.simple(
+                                    Blocks.MANGROVE_PROPAGULE.defaultBlockState().setValue(MangrovePropaguleBlock.HANGING, Boolean.valueOf(true))
+                                ),
+                                MangrovePropaguleBlock.AGE,
+                                UniformInt.of(0, 4)
                             ),
                             2,
                             List.of(Direction.DOWN)
@@ -727,6 +_,6 @@
         FeatureUtils.register(p_321605_, FANCY_OAK_BEES_0002, Feature.TREE, createFancyOak().decorators(List.of(beehivedecorator)).build());
         FeatureUtils.register(p_321605_, FANCY_OAK_BEES_002, Feature.TREE, createFancyOak().decorators(List.of(beehivedecorator2)).build());
         FeatureUtils.register(p_321605_, FANCY_OAK_BEES_005, Feature.TREE, createFancyOak().decorators(List.of(beehivedecorator3)).build());
-        FeatureUtils.register(p_321605_, FANCY_OAK_BEES, Feature.TREE, createFancyOak().decorators(List.of(beehivedecorator5)).build());
+        FeatureUtils.register(p_321605_, FANCY_OAK_BEES, Feature.TREE, createFancyOak().decorators(List.of(beehivedecorator4)).build());
     }
 }
