--- a/net/minecraft/client/gui/ComponentPath.java
+++ b/net/minecraft/client/gui/ComponentPath.java
@@ -20,7 +_,7 @@
     static ComponentPath path(GuiEventListener p_265555_, ContainerEventHandler... p_265487_) {
         ComponentPath componentpath = leaf(p_265555_);
 
-        for(ContainerEventHandler containereventhandler : p_265487_) {
+        for (ContainerEventHandler containereventhandler : p_265487_) {
             componentpath = path(containereventhandler, componentpath);
         }
 
