--- a/net/minecraft/client/renderer/block/model/BlockElement.java
+++ b/net/minecraft/client/renderer/block/model/BlockElement.java
@@ -29,27 +_,34 @@
     public final Map<Direction, BlockElementFace> faces;
     public final BlockElementRotation rotation;
     public final boolean shade;
+    private net.neoforged.neoforge.client.model.ExtraFaceData faceData;
 
     public BlockElement(
         Vector3f p_253626_, Vector3f p_254426_, Map<Direction, BlockElementFace> p_254454_, @Nullable BlockElementRotation p_254229_, boolean p_253661_
     ) {
+      this(p_253626_, p_254426_, p_254454_, p_254229_, p_253661_, net.neoforged.neoforge.client.model.ExtraFaceData.DEFAULT);
+    }
+
+    public BlockElement(Vector3f p_253626_, Vector3f p_254426_, Map<Direction, BlockElementFace> p_254454_, @Nullable BlockElementRotation p_254229_, boolean p_253661_, net.neoforged.neoforge.client.model.ExtraFaceData faceData) {
         this.from = p_253626_;
         this.to = p_254426_;
         this.faces = p_254454_;
         this.rotation = p_254229_;
         this.shade = p_253661_;
         this.fillUvs();
+        this.setFaceData(faceData);
+        this.faces.values().forEach(face -> face.parent = this);
     }
 
     private void fillUvs() {
-        for(Entry<Direction, BlockElementFace> entry : this.faces.entrySet()) {
+        for (Entry<Direction, BlockElementFace> entry : this.faces.entrySet()) {
             float[] afloat = this.uvsByFace(entry.getKey());
             entry.getValue().uv.setMissingUv(afloat);
         }
     }
 
     public float[] uvsByFace(Direction p_111321_) {
-        switch(p_111321_) {
+        switch (p_111321_) {
             case DOWN:
                 return new float[]{this.from.x(), 16.0F - this.to.z(), this.to.x(), 16.0F - this.from.z()};
             case UP:
@@ -80,7 +_,9 @@
                 throw new JsonParseException("Expected shade to be a Boolean");
             } else {
                 boolean flag = GsonHelper.getAsBoolean(jsonobject, "shade", true);
-                return new BlockElement(vector3f, vector3f1, map, blockelementrotation, flag);
+                if (jsonobject.has("forge_data")) throw new JsonParseException("forge_data should be replaced by neoforge_data"); // TODO 1.22: Remove
+                var faceData = net.neoforged.neoforge.client.model.ExtraFaceData.read(jsonobject.get("neoforge_data"), net.neoforged.neoforge.client.model.ExtraFaceData.DEFAULT);
+                return new BlockElement(vector3f, vector3f1, map, blockelementrotation, flag, faceData);
             }
         }
 
@@ -132,7 +_,7 @@
             Map<Direction, BlockElementFace> map = Maps.newEnumMap(Direction.class);
             JsonObject jsonobject = GsonHelper.getAsJsonObject(p_111341_, "faces");
 
-            for(Entry<String, JsonElement> entry : jsonobject.entrySet()) {
+            for (Entry<String, JsonElement> entry : jsonobject.entrySet()) {
                 Direction direction = this.getFacing(entry.getKey());
                 map.put(direction, p_111340_.deserialize(entry.getValue(), BlockElementFace.class));
             }
@@ -184,12 +_,20 @@
             } else {
                 float[] afloat = new float[3];
 
-                for(int i = 0; i < afloat.length; ++i) {
+                for (int i = 0; i < afloat.length; i++) {
                     afloat[i] = GsonHelper.convertToFloat(jsonarray.get(i), p_111336_ + "[" + i + "]");
                 }
 
                 return new Vector3f(afloat[0], afloat[1], afloat[2]);
             }
         }
+    }
+
+    public net.neoforged.neoforge.client.model.ExtraFaceData getFaceData() {
+        return this.faceData;
+    }
+
+    public void setFaceData(net.neoforged.neoforge.client.model.ExtraFaceData faceData) {
+        this.faceData = java.util.Objects.requireNonNull(faceData);
     }
 }
