--- a/net/minecraft/client/renderer/block/model/BlockFaceUV.java
+++ b/net/minecraft/client/renderer/block/model/BlockFaceUV.java
@@ -85,7 +_,7 @@
                 } else {
                     float[] afloat = new float[4];
 
-                    for(int i = 0; i < afloat.length; ++i) {
+                    for (int i = 0; i < afloat.length; i++) {
                         afloat[i] = GsonHelper.convertToFloat(jsonarray.get(i), "uv[" + i + "]");
                     }
 
