--- a/net/minecraft/client/renderer/block/model/ItemModelGenerator.java
+++ b/net/minecraft/client/renderer/block/model/ItemModelGenerator.java
@@ -58,8 +_,8 @@
                 break;
             }
 
-            SpriteContents spritecontents = modelBaker.sprites().get(material, name).contents();
-            list.addAll(processFrames(i, s, spritecontents));
+            net.minecraft.client.renderer.texture.TextureAtlasSprite sprite = modelBaker.sprites().get(material, name);
+            list.addAll(processFrames(i, s, sprite.contents()));
         }
 
         return SimpleUnbakedGeometry.bake(list, textureSlots, modelBaker, modelState, name);
