--- a/net/minecraft/world/phys/shapes/BitSetDiscreteVoxelShape.java
+++ b/net/minecraft/world/phys/shapes/BitSetDiscreteVoxelShape.java
@@ -31,9 +_,9 @@
         bitsetdiscretevoxelshape.yMax = p_165940_;
         bitsetdiscretevoxelshape.zMax = p_165941_;
 
-        for(int i = p_165936_; i < p_165939_; ++i) {
-            for(int j = p_165937_; j < p_165940_; ++j) {
-                for(int k = p_165938_; k < p_165941_; ++k) {
+        for (int i = p_165936_; i < p_165939_; i++) {
+            for (int j = p_165937_; j < p_165940_; j++) {
+                for (int k = p_165938_; k < p_165941_; k++) {
                     bitsetdiscretevoxelshape.fillUpdateBounds(i, j, k, false);
                 }
             }
@@ -49,9 +_,9 @@
         } else {
             this.storage = new BitSet(this.xSize * this.ySize * this.zSize);
 
-            for(int i = 0; i < this.xSize; ++i) {
-                for(int j = 0; j < this.ySize; ++j) {
-                    for(int k = 0; k < this.zSize; ++k) {
+            for (int i = 0; i < this.xSize; i++) {
+                for (int j = 0; j < this.ySize; j++) {
+                    for (int k = 0; k < this.zSize; k++) {
                         if (p_82602_.isFull(i, j, k)) {
                             this.storage.set(this.getIndex(i, j, k));
                         }
@@ -155,11 +_,11 @@
     protected static void forAllBoxes(DiscreteVoxelShape p_165964_, DiscreteVoxelShape.IntLineConsumer p_165965_, boolean p_165966_) {
         BitSetDiscreteVoxelShape bitsetdiscretevoxelshape = new BitSetDiscreteVoxelShape(p_165964_);
 
-        for(int i = 0; i < bitsetdiscretevoxelshape.ySize; ++i) {
-            for(int j = 0; j < bitsetdiscretevoxelshape.xSize; ++j) {
+        for (int i = 0; i < bitsetdiscretevoxelshape.ySize; i++) {
+            for (int j = 0; j < bitsetdiscretevoxelshape.xSize; j++) {
                 int k = -1;
 
-                for(int l = 0; l <= bitsetdiscretevoxelshape.zSize; ++l) {
+                for (int l = 0; l <= bitsetdiscretevoxelshape.zSize; l++) {
                     if (bitsetdiscretevoxelshape.isFullWide(j, i, l)) {
                         if (p_165966_) {
                             if (k == -1) {
@@ -173,17 +_,17 @@
                         int j1 = i;
                         bitsetdiscretevoxelshape.clearZStrip(k, l, j, i);
 
-                        while(bitsetdiscretevoxelshape.isZStripFull(k, l, i1 + 1, i)) {
+                        while (bitsetdiscretevoxelshape.isZStripFull(k, l, i1 + 1, i)) {
                             bitsetdiscretevoxelshape.clearZStrip(k, l, i1 + 1, i);
-                            ++i1;
+                            i1++;
                         }
 
-                        while(bitsetdiscretevoxelshape.isXZRectangleFull(j, i1 + 1, k, l, j1 + 1)) {
-                            for(int k1 = j; k1 <= i1; ++k1) {
+                        while (bitsetdiscretevoxelshape.isXZRectangleFull(j, i1 + 1, k, l, j1 + 1)) {
+                            for (int k1 = j; k1 <= i1; k1++) {
                                 bitsetdiscretevoxelshape.clearZStrip(k, l, k1, j1 + 1);
                             }
 
-                            ++j1;
+                            j1++;
                         }
 
                         p_165965_.consume(j, i, k, i1 + 1, j1 + 1, l);
@@ -195,15 +_,13 @@
     }
 
     private boolean isZStripFull(int p_82609_, int p_82610_, int p_82611_, int p_82612_) {
-        if (p_82611_ < this.xSize && p_82612_ < this.ySize) {
-            return this.storage.nextClearBit(this.getIndex(p_82611_, p_82612_, p_82609_)) >= this.getIndex(p_82611_, p_82612_, p_82610_);
-        } else {
-            return false;
-        }
+        return p_82611_ < this.xSize && p_82612_ < this.ySize
+            ? this.storage.nextClearBit(this.getIndex(p_82611_, p_82612_, p_82609_)) >= this.getIndex(p_82611_, p_82612_, p_82610_)
+            : false;
     }
 
     private boolean isXZRectangleFull(int p_165927_, int p_165928_, int p_165929_, int p_165930_, int p_165931_) {
-        for(int i = p_165927_; i < p_165928_; ++i) {
+        for (int i = p_165927_; i < p_165928_; i++) {
             if (!this.isZStripFull(p_165929_, p_165930_, i, p_165931_)) {
                 return false;
             }
