--- a/net/minecraft/world/phys/shapes/DiscreteVoxelShape.java
+++ b/net/minecraft/world/phys/shapes/DiscreteVoxelShape.java
@@ -48,7 +_,7 @@
     public abstract void fill(int p_165998_, int p_165999_, int p_166000_);
 
     public boolean isEmpty() {
-        for(Direction.Axis direction$axis : AXIS_VALUES) {
+        for (Direction.Axis direction$axis : AXIS_VALUES) {
             if (this.firstFull(direction$axis) >= this.lastFull(direction$axis)) {
                 return true;
             }
@@ -69,7 +_,7 @@
             if (p_165996_ < this.getSize(direction$axis) && p_165997_ < this.getSize(direction$axis1)) {
                 AxisCycle axiscycle = AxisCycle.between(Direction.Axis.X, p_165995_);
 
-                for(int j = 0; j < i; ++j) {
+                for (int j = 0; j < i; j++) {
                     if (this.isFull(axiscycle, j, p_165996_, p_165997_)) {
                         return j;
                     }
@@ -92,7 +_,7 @@
                 int i = this.getSize(p_82842_);
                 AxisCycle axiscycle = AxisCycle.between(Direction.Axis.X, p_82842_);
 
-                for(int j = i - 1; j >= 0; --j) {
+                for (int j = i - 1; j >= 0; j--) {
                     if (this.isFull(axiscycle, j, p_82843_, p_82844_)) {
                         return j + 1;
                     }
@@ -135,18 +_,18 @@
         int k = this.getSize(axiscycle.cycle(Direction.Axis.Y));
         int l = this.getSize(axiscycle.cycle(Direction.Axis.Z));
 
-        for(int i1 = 0; i1 <= j; ++i1) {
-            for(int j1 = 0; j1 <= k; ++j1) {
+        for (int i1 = 0; i1 <= j; i1++) {
+            for (int j1 = 0; j1 <= k; j1++) {
                 int i = -1;
 
-                for(int k1 = 0; k1 <= l; ++k1) {
+                for (int k1 = 0; k1 <= l; k1++) {
                     int l1 = 0;
                     int i2 = 0;
 
-                    for(int j2 = 0; j2 <= 1; ++j2) {
-                        for(int k2 = 0; k2 <= 1; ++k2) {
+                    for (int j2 = 0; j2 <= 1; j2++) {
+                        for (int k2 = 0; k2 <= 1; k2++) {
                             if (this.isFullWide(axiscycle, i1 + j2 - 1, j1 + k2 - 1, k1)) {
-                                ++l1;
+                                l1++;
                                 i2 ^= j2 ^ k2;
                             }
                         }
@@ -202,11 +_,11 @@
         Direction direction = Direction.fromAxisAndDirection(direction$axis, Direction.AxisDirection.NEGATIVE);
         Direction direction1 = Direction.fromAxisAndDirection(direction$axis, Direction.AxisDirection.POSITIVE);
 
-        for(int l = 0; l < i; ++l) {
-            for(int i1 = 0; i1 < j; ++i1) {
+        for (int l = 0; l < i; l++) {
+            for (int i1 = 0; i1 < j; i1++) {
                 boolean flag = false;
 
-                for(int j1 = 0; j1 <= k; ++j1) {
+                for (int j1 = 0; j1 <= k; j1++) {
                     boolean flag1 = j1 != k && this.isFull(axiscycle, l, i1, j1);
                     if (!flag && flag1) {
                         p_82813_.consume(
