--- a/net/minecraft/world/phys/shapes/IdenticalMerger.java
+++ b/net/minecraft/world/phys/shapes/IdenticalMerger.java
@@ -13,7 +_,7 @@
     public boolean forMergedIndexes(IndexMerger.IndexConsumer p_82906_) {
         int i = this.coords.size() - 1;
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             if (!p_82906_.merge(j, j, j)) {
                 return false;
             }
