--- a/net/minecraft/world/phys/shapes/IndirectMerger.java
+++ b/net/minecraft/world/phys/shapes/IndirectMerger.java
@@ -25,7 +_,7 @@
         int i1 = 0;
         int j1 = 0;
 
-        while(true) {
+        while (true) {
             boolean flag2 = i1 >= i;
             boolean flag3 = j1 >= j;
             if (flag2 && flag3) {
@@ -35,12 +_,12 @@
 
             boolean flag4 = !flag2 && (flag3 || p_83001_.getDouble(i1) < p_83002_.getDouble(j1) + 1.0E-7);
             if (flag4) {
-                ++i1;
+                i1++;
                 if (flag && (j1 == 0 || flag3)) {
                     continue;
                 }
             } else {
-                ++j1;
+                j1++;
                 if (flag1 && (i1 == 0 || flag2)) {
                     continue;
                 }
@@ -53,7 +_,7 @@
                 this.firstIndices[l] = k1;
                 this.secondIndices[l] = l1;
                 this.result[l] = d1;
-                ++l;
+                l++;
                 d0 = d1;
             } else {
                 this.firstIndices[l - 1] = k1;
@@ -66,7 +_,7 @@
     public boolean forMergedIndexes(IndexMerger.IndexConsumer p_83007_) {
         int i = this.resultLength - 1;
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             if (!p_83007_.merge(this.firstIndices[j], this.secondIndices[j], j)) {
                 return false;
             }
