--- a/net/minecraft/world/phys/shapes/NonOverlappingMerger.java
+++ b/net/minecraft/world/phys/shapes/NonOverlappingMerger.java
@@ -29,7 +_,7 @@
     private boolean forNonSwappedIndexes(IndexMerger.IndexConsumer p_83024_) {
         int i = this.lower.size();
 
-        for(int j = 0; j < i; ++j) {
+        for (int j = 0; j < i; j++) {
             if (!p_83024_.merge(j, -1, j)) {
                 return false;
             }
@@ -37,7 +_,7 @@
 
         int l = this.upper.size() - 1;
 
-        for(int k = 0; k < l; ++k) {
+        for (int k = 0; k < l; k++) {
             if (!p_83024_.merge(i - 1, k, i + k)) {
                 return false;
             }
