--- a/net/minecraft/world/phys/shapes/Shapes.java
+++ b/net/minecraft/world/phys/shapes/Shapes.java
@@ -94,7 +_,7 @@
     @VisibleForTesting
     protected static int findBits(double p_83042_, double p_83043_) {
         if (!(p_83042_ < -1.0E-7) && !(p_83043_ > 1.0000001)) {
-            for(int i = 0; i <= 3; ++i) {
+            for (int i = 0; i <= 3; i++) {
                 int j = 1 << i;
                 double d0 = p_83042_ * (double)j;
                 double d1 = p_83043_ * (double)j;
@@ -176,7 +_,7 @@
                     boolean flag2 = p_83160_.apply(true, false);
                     boolean flag3 = p_83160_.apply(false, true);
 
-                    for(Direction.Axis direction$axis : AxisCycle.AXIS_VALUES) {
+                    for (Direction.Axis direction$axis : AxisCycle.AXIS_VALUES) {
                         if (p_83158_.max(direction$axis) < p_83159_.min(direction$axis) - 1.0E-7) {
                             return flag2 || flag3;
                         }
@@ -220,7 +_,7 @@
     }
 
     public static double collide(Direction.Axis p_193136_, AABB p_193137_, Iterable<VoxelShape> p_193138_, double p_193139_) {
-        for(VoxelShape voxelshape : p_193138_) {
+        for (VoxelShape voxelshape : p_193138_) {
             if (Math.abs(p_193139_) < 1.0E-7) {
                 return 0.0;
             }
@@ -302,10 +_,10 @@
     public static boolean faceShapeOccludes(VoxelShape p_83146_, VoxelShape p_83147_) {
         if (p_83146_ == block() || p_83147_ == block()) {
             return true;
-        } else if (p_83146_.isEmpty() && p_83147_.isEmpty()) {
-            return false;
         } else {
-            return !joinIsNotEmpty(block(), joinUnoptimized(p_83146_, p_83147_, BooleanOp.OR), BooleanOp.ONLY_FIRST);
+            return p_83146_.isEmpty() && p_83147_.isEmpty()
+                ? false
+                : !joinIsNotEmpty(block(), joinUnoptimized(p_83146_, p_83147_, BooleanOp.OR), BooleanOp.ONLY_FIRST);
         }
     }
 
