--- a/net/minecraft/client/gui/screens/social/PlayerSocialManager.java
+++ b/net/minecraft/client/gui/screens/social/PlayerSocialManager.java
@@ -10,7 +_,6 @@
 import java.util.concurrent.CompletableFuture;
 import net.minecraft.Util;
 import net.minecraft.client.Minecraft;
-import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.client.multiplayer.PlayerInfo;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
@@ -74,15 +_,13 @@
     public void addPlayer(PlayerInfo p_100677_) {
         GameProfile gameprofile = p_100677_.getProfile();
         this.discoveredNamesToUUID.put(gameprofile.getName(), gameprofile.getId());
-        Screen screen = this.minecraft.screen;
-        if (screen instanceof SocialInteractionsScreen socialinteractionsscreen) {
+        if (this.minecraft.screen instanceof SocialInteractionsScreen socialinteractionsscreen) {
             socialinteractionsscreen.onAddPlayer(p_100677_);
         }
     }
 
     public void removePlayer(UUID p_100691_) {
-        Screen screen = this.minecraft.screen;
-        if (screen instanceof SocialInteractionsScreen socialinteractionsscreen) {
+        if (this.minecraft.screen instanceof SocialInteractionsScreen socialinteractionsscreen) {
             socialinteractionsscreen.onRemovePlayer(p_100691_);
         }
     }
