--- a/net/minecraft/client/gui/screens/social/SocialInteractionsScreen.java
+++ b/net/minecraft/client/gui/screens/social/SocialInteractionsScreen.java
@@ -10,6 +_,7 @@
 import net.minecraft.client.GameNarrator;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.EditBox;
 import net.minecraft.client.gui.components.events.GuiEventListener;
@@ -143,7 +_,9 @@
         this.initialized = true;
         this.showPage(this.page);
         this.layout.addToFooter(Button.builder(CommonComponents.GUI_DONE, p_329743_ -> this.onClose()).width(200).build());
-        this.layout.visitWidgets(this::addRenderableWidget);
+        this.layout.visitWidgets(p_329745_ -> {
+            AbstractWidget abstractwidget = this.addRenderableWidget(p_329745_);
+        });
         this.repositionElements();
     }
 
@@ -177,7 +_,7 @@
         this.hiddenButton.setMessage(TAB_HIDDEN);
         this.blockedButton.setMessage(TAB_BLOCKED);
         boolean flag = false;
-        switch(p_100772_) {
+        switch (p_100772_) {
             case ALL:
                 this.allButton.setMessage(TAB_ALL_SELECTED);
                 Collection<UUID> collection = this.minecraft.player.connection.getOnlinePlayerIds();
