--- a/net/minecraft/client/gui/screens/social/SocialInteractionsPlayerList.java
+++ b/net/minecraft/client/gui/screens/social/SocialInteractionsPlayerList.java
@@ -59,7 +_,7 @@
     private void addOnlinePlayers(Collection<UUID> p_240813_, Map<UUID, PlayerEntry> p_240796_) {
         ClientPacketListener clientpacketlistener = this.minecraft.player.connection;
 
-        for(UUID uuid : p_240813_) {
+        for (UUID uuid : p_240813_) {
             PlayerInfo playerinfo = clientpacketlistener.getPlayerInfo(uuid);
             if (playerinfo != null) {
                 boolean flag = playerinfo.hasVerifiableChat();
@@ -71,7 +_,7 @@
     }
 
     private void updatePlayersFromChatLog(Map<UUID, PlayerEntry> p_240780_, boolean p_240827_) {
-        for(GameProfile gameprofile : collectProfilesFromChatLog(this.minecraft.getReportingContext().chatLog())) {
+        for (GameProfile gameprofile : collectProfilesFromChatLog(this.minecraft.getReportingContext().chatLog())) {
             PlayerEntry playerentry;
             if (p_240827_) {
                 playerentry = p_240780_.computeIfAbsent(
@@ -103,10 +_,13 @@
     private static Collection<GameProfile> collectProfilesFromChatLog(ChatLog p_250748_) {
         Set<GameProfile> set = new ObjectLinkedOpenHashSet<>();
 
-        for(int i = p_250748_.end(); i >= p_250748_.start(); --i) {
+        for (int i = p_250748_.end(); i >= p_250748_.start(); i--) {
             LoggedChatEvent loggedchatevent = p_250748_.lookup(i);
-            if (loggedchatevent instanceof LoggedChatMessage.Player loggedchatmessage$player && loggedchatmessage$player.message().hasSignature()) {
-                set.add(loggedchatmessage$player.profile());
+            if (loggedchatevent instanceof LoggedChatMessage.Player) {
+                LoggedChatMessage.Player loggedchatmessage$player = (LoggedChatMessage.Player)loggedchatevent;
+                if (loggedchatmessage$player.message().hasSignature()) {
+                    set.add(loggedchatmessage$player.profile());
+                }
             }
         }
 
@@ -163,7 +_,7 @@
     public void addPlayer(PlayerInfo p_100715_, SocialInteractionsScreen.Page p_100716_) {
         UUID uuid = p_100715_.getProfile().getId();
 
-        for(PlayerEntry playerentry : this.players) {
+        for (PlayerEntry playerentry : this.players) {
             if (playerentry.getPlayerId().equals(uuid)) {
                 playerentry.setRemoved(false);
                 return;
@@ -182,7 +_,7 @@
     }
 
     public void removePlayer(UUID p_100723_) {
-        for(PlayerEntry playerentry : this.players) {
+        for (PlayerEntry playerentry : this.players) {
             if (playerentry.getPlayerId().equals(p_100723_)) {
                 playerentry.setRemoved(true);
                 return;
