--- a/net/minecraft/data/worldgen/DimensionTypes.java
+++ b/net/minecraft/data/worldgen/DimensionTypes.java
@@ -89,25 +_,5 @@
                 new DimensionType.MonsterSettings(false, true, UniformInt.of(0, 7), 0)
             )
         );
-        p_321848_.register(
-            BuiltinDimensionTypes.POTATO,
-            new DimensionType(
-                OptionalLong.empty(),
-                true,
-                false,
-                false,
-                true,
-                1.0,
-                true,
-                false,
-                0,
-                256,
-                256,
-                BlockTags.INFINIBURN_OVERWORLD,
-                BuiltinDimensionTypes.POTATO_EFFECTS,
-                0.04F,
-                new DimensionType.MonsterSettings(true, false, UniformInt.of(0, 7), 0)
-            )
-        );
     }
 }
