--- a/net/minecraft/data/worldgen/Pools.java
+++ b/net/minecraft/data/worldgen/Pools.java
@@ -24,7 +_,6 @@
         Holder<StructureTemplatePool> holder = holdergetter.getOrThrow(EMPTY);
         p_321867_.register(EMPTY, new StructureTemplatePool(holder, ImmutableList.of(), StructureTemplatePool.Projection.RIGID));
         BastionPieces.bootstrap(p_321867_);
-        ColosseumPieces.bootstrap(p_321867_);
         PillagerOutpostPools.bootstrap(p_321867_);
         VillagePools.bootstrap(p_321867_);
         AncientCityStructurePieces.bootstrap(p_321867_);
