--- a/net/minecraft/data/worldgen/StructureSets.java
+++ b/net/minecraft/data/worldgen/StructureSets.java
@@ -26,7 +_,6 @@
             new StructureSet(
                 List.of(
                     StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.VILLAGE_PLAINS)),
-                    StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.VILLAGE_POTATO)),
                     StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.VILLAGE_DESERT)),
                     StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.VILLAGE_SAVANNA)),
                     StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.VILLAGE_SNOWY)),
@@ -103,8 +_,7 @@
             new StructureSet(
                 List.of(
                     StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.MINESHAFT)),
-                    StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.MINESHAFT_MESA)),
-                    StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.MINESHAFT_POTATO))
+                    StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.MINESHAFT_MESA))
                 ),
                 new RandomSpreadStructurePlacement(
                     Vec3i.ZERO, StructurePlacement.FrequencyReductionMethod.LEGACY_TYPE_3, 0.004F, 0, Optional.empty(), 1, 0, RandomSpreadType.LINEAR
@@ -127,13 +_,6 @@
             )
         );
         p_321852_.register(
-            BuiltinStructureSets.RUINED_PORTATOLS,
-            new StructureSet(
-                List.of(StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.RUINED_PORTATOL))),
-                new RandomSpreadStructurePlacement(28, 8, RandomSpreadType.LINEAR, 31415926)
-            )
-        );
-        p_321852_.register(
             BuiltinStructureSets.SHIPWRECKS,
             new StructureSet(
                 List.of(
@@ -161,12 +_,6 @@
                     StructureSet.entry(holdergetter.getOrThrow(BuiltinStructures.BASTION_REMNANT), 3)
                 ),
                 new RandomSpreadStructurePlacement(27, 4, RandomSpreadType.LINEAR, 30084232)
-            )
-        );
-        p_321852_.register(
-            BuiltinStructureSets.COLOSSEA,
-            new StructureSet(
-                holdergetter.getOrThrow(BuiltinStructures.COLOSSEUM), new RandomSpreadStructurePlacement(24, 5, RandomSpreadType.LINEAR, 123456789)
             )
         );
         p_321852_.register(
