--- a/net/minecraft/data/worldgen/TrialChambersStructurePools.java
+++ b/net/minecraft/data/worldgen/TrialChambersStructurePools.java
@@ -100,8 +_,8 @@
                 holder1,
                 List.of(
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_1", holder3), 1),
-                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_5", holder3), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_6", holder3), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption", holder3), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted", holder3), 1)
                 ),
                 StructureTemplatePool.Projection.RIGID
@@ -109,6 +_,22 @@
         );
         Pools.register(
             p_321754_,
+            "trial_chambers/corridor",
+            new StructureTemplatePool(
+                holder,
+                List.of(
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/second_plate"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/intersection/intersection_1", holder3), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/intersection/intersection_2", holder3), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/first_plate"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium_1", holder3), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/entrance_1", holder3), 1)
+                ),
+                StructureTemplatePool.Projection.RIGID
+            )
+        );
+        Pools.register(
+            p_321754_,
             "trial_chambers/chamber/addon",
             new StructureTemplatePool(
                 holder,
@@ -172,6 +_,26 @@
         );
         Pools.register(
             p_321754_,
+            "trial_chambers/chamber/eruption",
+            new StructureTemplatePool(
+                holder,
+                List.of(
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/center_1"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/breeze_slice_1"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/slice_1"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/slice_2"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/slice_3"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/quadrant_1"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/quadrant_2"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/quadrant_3"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/quadrant_4"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption/quadrant_5"), 1)
+                ),
+                StructureTemplatePool.Projection.RIGID
+            )
+        );
+        Pools.register(
+            p_321754_,
             "trial_chambers/chamber/slanted",
             new StructureTemplatePool(
                 holder,
@@ -189,7 +_,8 @@
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted/ramp_1"), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted/ramp_2"), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted/ramp_3"), 1),
-                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted/ramp_4"), 1)
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted/ramp_4"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted/ominous_upper_arm_1"), 1)
                 ),
                 StructureTemplatePool.Projection.RIGID
             )
@@ -206,6 +_,7 @@
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal/slice_3"), 3),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal/slice_4"), 3),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal/slice_5"), 3),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal/ominous_slice_1"), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal/quadrant_1"), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal/quadrant_2"), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal/quadrant_3"), 1),
@@ -219,39 +_,7 @@
         );
         Pools.register(
             p_321754_,
-            "trial_chambers/corridor/second",
-            new StructureTemplatePool(
-                holder,
-                List.of(
-                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/second_plate"), 1),
-                    Pair.of(StructurePoolElement.single("trial_chambers/intersection/intersection_1", holder3), 1),
-                    Pair.of(StructurePoolElement.single("trial_chambers/intersection/intersection_2", holder3), 1)
-                ),
-                StructureTemplatePool.Projection.RIGID
-            )
-        );
-        Pools.register(
-            p_321754_,
-            "trial_chambers/corridor/first",
-            new StructureTemplatePool(
-                holder,
-                List.of(
-                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/first_plate"), 1),
-                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/entrance_1", holder3), 2)
-                ),
-                StructureTemplatePool.Projection.RIGID
-            )
-        );
-        Pools.register(
-            p_321754_,
-            "trial_chambers/corridor/first/straight",
-            new StructureTemplatePool(
-                holder, List.of(Pair.of(StructurePoolElement.single("trial_chambers/corridor/first_plate"), 1)), StructureTemplatePool.Projection.RIGID
-            )
-        );
-        Pools.register(
-            p_321754_,
-            "trial_chambers/corridor",
+            "trial_chambers/corridor/slices",
             new StructureTemplatePool(
                 holder,
                 List.of(
@@ -261,7 +_,7 @@
                     Pair.of(StructurePoolElement.single("trial_chambers/corridor/straight_4", holder3), 2),
                     Pair.of(StructurePoolElement.single("trial_chambers/corridor/straight_5", holder3), 2),
                     Pair.of(StructurePoolElement.single("trial_chambers/corridor/straight_6", holder3), 2),
-                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/straight_7", holder3), 2),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/straight_7", holder3), 1),
                     Pair.of(StructurePoolElement.single("trial_chambers/corridor/straight_8", holder3), 2)
                 ),
                 StructureTemplatePool.Projection.RIGID
@@ -293,9 +_,9 @@
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_1", holder3), 150),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_2", holder3), 150),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_4", holder3), 150),
-                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_5", holder3), 150),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_6", holder3), 150),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/chamber_8", holder3), 150),
+                    Pair.of(StructurePoolElement.single("trial_chambers/chamber/eruption", holder3), 150),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/slanted", holder3), 150),
                     Pair.of(StructurePoolElement.single("trial_chambers/chamber/pedestal", holder3), 150),
                     Pair.of(StructurePoolElement.single("trial_chambers/hallway/rubble_chamber", holder3), 10),
@@ -361,6 +_,23 @@
         );
         Pools.register(
             p_321754_,
+            "trial_chambers/atrium",
+            new StructureTemplatePool(
+                holder,
+                List.of(
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium/bogged_relief"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium/breeze_relief"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium/spiral_relief"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium/spider_relief"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium/grand_staircase_1"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium/grand_staircase_2"), 1),
+                    Pair.of(StructurePoolElement.single("trial_chambers/corridor/atrium/grand_staircase_3"), 1)
+                ),
+                StructureTemplatePool.Projection.RIGID
+            )
+        );
+        Pools.register(
+            p_321754_,
             "trial_chambers/decor",
             new StructureTemplatePool(
                 holder,
@@ -394,14 +_,21 @@
             p_321754_,
             "trial_chambers/reward/all",
             new StructureTemplatePool(
-                holder, List.of(Pair.of(StructurePoolElement.single("trial_chambers/reward/connectors/default"), 1)), StructureTemplatePool.Projection.RIGID
+                holder, List.of(Pair.of(StructurePoolElement.single("trial_chambers/reward/vault"), 1)), StructureTemplatePool.Projection.RIGID
+            )
+        );
+        Pools.register(
+            p_321754_,
+            "trial_chambers/reward/ominous_vault",
+            new StructureTemplatePool(
+                holder, List.of(Pair.of(StructurePoolElement.single("trial_chambers/reward/ominous_vault"), 1)), StructureTemplatePool.Projection.RIGID
             )
         );
         Pools.register(
             p_321754_,
             "trial_chambers/reward/contents/default",
             new StructureTemplatePool(
-                holder, List.of(Pair.of(StructurePoolElement.single("trial_chambers/reward/default"), 1)), StructureTemplatePool.Projection.RIGID
+                holder, List.of(Pair.of(StructurePoolElement.single("trial_chambers/reward/vault"), 1)), StructureTemplatePool.Projection.RIGID
             )
         );
         Pools.register(
