--- a/net/minecraft/world/level/lighting/ChunkSkyLightSources.java
+++ b/net/minecraft/world/level/lighting/ChunkSkyLightSources.java
@@ -35,8 +_,8 @@
         if (i == -1) {
             this.fill(this.minY);
         } else {
-            for(int j = 0; j < 16; ++j) {
-                for(int k = 0; k < 16; ++k) {
+            for (int j = 0; j < 16; j++) {
+                for (int k = 0; k < 16; k++) {
                     int l = Math.max(this.findLowestSourceY(p_285152_, i, k, j), this.minY);
                     this.set(index(k, j), l);
                 }
@@ -50,7 +_,7 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos1 = this.mutablePos2.setWithOffset(blockpos$mutableblockpos, Direction.DOWN);
         BlockState blockstate = Blocks.AIR.defaultBlockState();
 
-        for(int j = p_285171_; j >= 0; --j) {
+        for (int j = p_285171_; j >= 0; j--) {
             LevelChunkSection levelchunksection = p_285214_.getSection(j);
             if (levelchunksection.hasOnlyAir()) {
                 blockstate = Blocks.AIR.defaultBlockState();
@@ -58,7 +_,7 @@
                 blockpos$mutableblockpos.setY(SectionPos.sectionToBlockCoord(l));
                 blockpos$mutableblockpos1.setY(blockpos$mutableblockpos.getY() - 1);
             } else {
-                for(int k = 15; k >= 0; --k) {
+                for (int k = 15; k >= 0; k--) {
                     BlockState blockstate1 = levelchunksection.getBlockState(p_285021_, k, p_285226_);
                     if (isEdgeOccluded(p_285214_, blockpos$mutableblockpos, blockstate, blockpos$mutableblockpos1, blockstate1)) {
                         return blockpos$mutableblockpos.getY();
@@ -117,7 +_,7 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos1 = this.mutablePos2.setWithOffset(p_285119_, Direction.DOWN);
         BlockState blockstate = p_285096_;
 
-        while(blockpos$mutableblockpos1.getY() >= this.minY) {
+        while (blockpos$mutableblockpos1.getY() >= this.minY) {
             BlockState blockstate1 = p_285279_.getBlockState(blockpos$mutableblockpos1);
             if (isEdgeOccluded(p_285279_, blockpos$mutableblockpos, blockstate, blockpos$mutableblockpos1, blockstate1)) {
                 return blockpos$mutableblockpos.getY();
@@ -149,7 +_,7 @@
     public int getHighestLowestSourceY() {
         int i = Integer.MIN_VALUE;
 
-        for(int j = 0; j < this.heightmap.getSize(); ++j) {
+        for (int j = 0; j < this.heightmap.getSize(); j++) {
             int k = this.heightmap.get(j);
             if (k > i) {
                 i = k;
@@ -162,7 +_,7 @@
     private void fill(int p_285311_) {
         int i = p_285311_ - this.minY;
 
-        for(int j = 0; j < this.heightmap.getSize(); ++j) {
+        for (int j = 0; j < this.heightmap.getSize(); j++) {
             this.heightmap.set(j, i);
         }
     }
