--- a/net/minecraft/world/level/lighting/LayerLightEventListener.java
+++ b/net/minecraft/world/level/lighting/LayerLightEventListener.java
@@ -12,23 +_,22 @@
 
     int getLightValue(BlockPos p_75710_);
 
-    public static record ConstantLayer(int lightLevel) implements LayerLightEventListener {
-        public static final LayerLightEventListener.ConstantLayer ZERO = new LayerLightEventListener.ConstantLayer(0);
-        public static final LayerLightEventListener.ConstantLayer FULL_BRIGHT = new LayerLightEventListener.ConstantLayer(15);
+    public static enum DummyLightLayerEventListener implements LayerLightEventListener {
+        INSTANCE;
 
         @Nullable
         @Override
-        public DataLayer getDataLayerData(SectionPos p_340936_) {
+        public DataLayer getDataLayerData(SectionPos p_75718_) {
             return null;
         }
 
         @Override
-        public int getLightValue(BlockPos p_341861_) {
-            return this.lightLevel;
+        public int getLightValue(BlockPos p_75723_) {
+            return 0;
         }
 
         @Override
-        public void checkBlock(BlockPos p_341947_) {
+        public void checkBlock(BlockPos p_164434_) {
         }
 
         @Override
@@ -42,15 +_,15 @@
         }
 
         @Override
-        public void updateSectionStatus(SectionPos p_340458_, boolean p_341024_) {
-        }
-
-        @Override
-        public void setLightEnabled(ChunkPos p_342284_, boolean p_341071_) {
-        }
-
-        @Override
-        public void propagateLightSources(ChunkPos p_342315_) {
+        public void updateSectionStatus(SectionPos p_75720_, boolean p_75721_) {
+        }
+
+        @Override
+        public void setLightEnabled(ChunkPos p_164431_, boolean p_164432_) {
+        }
+
+        @Override
+        public void propagateLightSources(ChunkPos p_285209_) {
         }
     }
 }
