--- a/net/minecraft/world/level/lighting/LayerLightSectionStorage.java
+++ b/net/minecraft/world/level/lighting/LayerLightSectionStorage.java
@@ -46,7 +_,7 @@
 
     @Nullable
     protected DataLayer getDataLayer(long p_75759_, boolean p_75760_) {
-        return this.getDataLayer((M)(p_75760_ ? this.updatingSectionData : this.visibleSectionData), p_75759_);
+        return this.getDataLayer(p_75760_ ? this.updatingSectionData : this.visibleSectionData, p_75759_);
     }
 
     @Nullable
@@ -111,9 +_,9 @@
         int j = SectionPos.y(p_281610_);
         int k = SectionPos.z(p_281610_);
 
-        for(int l = -1; l <= 1; ++l) {
-            for(int i1 = -1; i1 <= 1; ++i1) {
-                for(int j1 = -1; j1 <= 1; ++j1) {
+        for (int l = -1; l <= 1; l++) {
+            for (int i1 = -1; i1 <= 1; i1++) {
+                for (int j1 = -1; j1 <= 1; j1++) {
                     this.sectionsAffectedByLightUpdates.add(SectionPos.asLong(i + i1, j + j1, k + l));
                 }
             }
@@ -133,7 +_,7 @@
         if (this.hasInconsistencies) {
             this.hasInconsistencies = false;
 
-            for(long i : this.toRemove) {
+            for (long i : this.toRemove) {
                 DataLayer datalayer = this.queuedSections.remove(i);
                 DataLayer datalayer1 = this.updatingSectionData.removeLayer(i);
                 if (this.columnsToRetainQueuedDataFor.contains(SectionPos.getZeroNode(i))) {
@@ -147,7 +_,7 @@
 
             this.updatingSectionData.clearCache();
 
-            for(long k : this.toRemove) {
+            for (long k : this.toRemove) {
                 this.onNodeRemoved(k);
                 this.changedSections.add(k);
             }
@@ -155,7 +_,7 @@
             this.toRemove.clear();
             ObjectIterator<Entry<DataLayer>> objectiterator = Long2ObjectMaps.fastIterator(this.queuedSections);
 
-            while(objectiterator.hasNext()) {
+            while (objectiterator.hasNext()) {
                 Entry<DataLayer> entry = objectiterator.next();
                 long j = entry.getLongKey();
                 if (this.storingLightForSection(j)) {
@@ -216,9 +_,9 @@
             this.putSectionState(p_75788_, b1);
             int i = p_75789_ ? -1 : 1;
 
-            for(int j = -1; j <= 1; ++j) {
-                for(int k = -1; k <= 1; ++k) {
-                    for(int l = -1; l <= 1; ++l) {
+            for (int j = -1; j <= 1; j++) {
+                for (int k = -1; k <= 1; k++) {
+                    for (int l = -1; l <= 1; l++) {
                         if (j != 0 || k != 0 || l != 0) {
                             long i1 = SectionPos.offset(p_75788_, j, k, l);
                             byte b2 = this.sectionStates.get(i1);
@@ -268,7 +_,7 @@
         if (!this.sectionsAffectedByLightUpdates.isEmpty()) {
             LongIterator longiterator = this.sectionsAffectedByLightUpdates.iterator();
 
-            while(longiterator.hasNext()) {
+            while (longiterator.hasNext()) {
                 long i = longiterator.nextLong();
                 this.chunkSource.onLightUpdate(this.layer, SectionPos.of(i));
             }
