--- a/net/minecraft/world/level/lighting/LeveledPriorityQueue.java
+++ b/net/minecraft/world/level/lighting/LeveledPriorityQueue.java
@@ -11,7 +_,7 @@
         this.levelCount = p_278289_;
         this.queues = new LongLinkedOpenHashSet[p_278289_];
 
-        for(int i = 0; i < p_278289_; ++i) {
+        for (int i = 0; i < p_278289_; i++) {
             this.queues[i] = new LongLinkedOpenHashSet(p_278259_, 0.5F) {
                 @Override
                 protected void rehash(int p_278313_) {
@@ -58,7 +_,7 @@
         int i = this.firstQueuedLevel;
         this.firstQueuedLevel = p_278303_;
 
-        for(int j = i + 1; j < p_278303_; ++j) {
+        for (int j = i + 1; j < p_278303_; j++) {
             if (!this.queues[j].isEmpty()) {
                 this.firstQueuedLevel = j;
                 break;
