--- a/net/minecraft/world/level/lighting/LevelLightEngine.java
+++ b/net/minecraft/world/level/lighting/LevelLightEngine.java
@@ -36,11 +_,7 @@
 
     @Override
     public boolean hasLightWork() {
-        if (this.skyEngine != null && this.skyEngine.hasLightWork()) {
-            return true;
-        } else {
-            return this.blockEngine != null && this.blockEngine.hasLightWork();
-        }
+        return this.skyEngine != null && this.skyEngine.hasLightWork() ? true : this.blockEngine != null && this.blockEngine.hasLightWork();
     }
 
     @Override
@@ -92,9 +_,9 @@
 
     public LayerLightEventListener getLayerListener(LightLayer p_75815_) {
         if (p_75815_ == LightLayer.BLOCK) {
-            return (LayerLightEventListener)(this.blockEngine == null ? LayerLightEventListener.ConstantLayer.ZERO : this.blockEngine);
+            return (LayerLightEventListener)(this.blockEngine == null ? LayerLightEventListener.DummyLightLayerEventListener.INSTANCE : this.blockEngine);
         } else {
-            return (LayerLightEventListener)(this.skyEngine == null ? LayerLightEventListener.ConstantLayer.ZERO : this.skyEngine);
+            return (LayerLightEventListener)(this.skyEngine == null ? LayerLightEventListener.DummyLightLayerEventListener.INSTANCE : this.skyEngine);
         }
     }
 
