--- a/net/minecraft/world/level/lighting/SkyLightSectionStorage.java
+++ b/net/minecraft/world/level/lighting/SkyLightSectionStorage.java
@@ -33,7 +_,7 @@
         if (k != skylightsectionstorage$skydatalayerstoragemap.currentLowestY && j < k) {
             DataLayer datalayer = this.getDataLayer(skylightsectionstorage$skydatalayerstoragemap, i);
             if (datalayer == null) {
-                for(p_164458_ = BlockPos.getFlatIndex(p_164458_);
+                for (p_164458_ = BlockPos.getFlatIndex(p_164458_);
                     datalayer == null;
                     datalayer = this.getDataLayer(skylightsectionstorage$skydatalayerstoragemap, i)
                 ) {
@@ -76,8 +_,8 @@
         int j = SectionPos.y(p_75887_);
         if (this.updatingSectionData.topSections.get(i) == j + 1) {
             long k;
-            for(k = p_75887_; !this.storingLightForSection(k) && this.hasLightDataAtOrBelow(j); k = SectionPos.offset(k, Direction.DOWN)) {
-                --j;
+            for (k = p_75887_; !this.storingLightForSection(k) && this.hasLightDataAtOrBelow(j); k = SectionPos.offset(k, Direction.DOWN)) {
+                j--;
             }
 
             if (this.storingLightForSection(k)) {
@@ -99,7 +_,7 @@
                 long j = SectionPos.offset(p_75883_, Direction.UP);
 
                 DataLayer datalayer1;
-                while((datalayer1 = this.getDataLayer(j, true)) == null) {
+                while ((datalayer1 = this.getDataLayer(j, true)) == null) {
                     j = SectionPos.offset(j, Direction.UP);
                 }
 
@@ -117,7 +_,7 @@
             byte[] abyte = p_182513_.getData();
             byte[] abyte1 = new byte[2048];
 
-            for(int i = 0; i < 16; ++i) {
+            for (int i = 0; i < 16; i++) {
                 System.arraycopy(abyte, 0, abyte1, i * 128, 128);
             }
 
