--- a/net/minecraft/world/level/lighting/SpatialLongSet.java
+++ b/net/minecraft/world/level/lighting/SpatialLongSet.java
@@ -93,7 +_,7 @@
                 long[] along = this.key;
                 l = (int)HashCommon.mix(i) & this.mask;
 
-                for(long i1 = along[l]; i1 != 0L; i1 = along[l]) {
+                for (long i1 = along[l]; i1 != 0L; i1 = along[l]) {
                     if (i1 == i) {
                         this.lastPos = l;
                         this.lastOuterKey = i;
@@ -110,7 +_,7 @@
                 this.first = this.last = l;
                 this.link[l] = -1L;
             } else {
-                this.link[this.last] ^= (this.link[this.last] ^ (long)l & 4294967295L) & 4294967295L;
+                this.link[this.last] = this.link[this.last] ^ (this.link[this.last] ^ (long)l & 4294967295L) & 4294967295L;
                 this.link[l] = ((long)this.last & 4294967295L) << 32 | 4294967295L;
                 this.last = l;
             }
@@ -124,7 +_,7 @@
 
         private boolean replaceBit(int p_164487_, long p_164488_) {
             boolean flag = (this.value[p_164487_] & p_164488_) != 0L;
-            this.value[p_164487_] |= p_164488_;
+            this.value[p_164487_] = this.value[p_164487_] | p_164488_;
             return flag;
         }
 
@@ -140,7 +_,7 @@
                 long[] along = this.key;
                 int l = (int)HashCommon.mix(i) & this.mask;
 
-                for(long i1 = along[l]; i1 != 0L; i1 = along[l]) {
+                for (long i1 = along[l]; i1 != 0L; i1 = along[l]) {
                     if (i == i1) {
                         this.lastPos = l;
                         this.lastOuterKey = i;
@@ -158,12 +_,12 @@
             if ((this.value[this.n] & p_164504_) == 0L) {
                 return false;
             } else {
-                this.value[this.n] &= ~p_164504_;
+                this.value[this.n] = this.value[this.n] & ~p_164504_;
                 if (this.value[this.n] != 0L) {
                     return true;
                 } else {
                     this.containsNullKey = false;
-                    --this.size;
+                    this.size--;
                     this.fixPointers(this.n);
                     if (this.size < this.maxFill / 4 && this.n > 16) {
                         this.rehash(this.n / 2);
@@ -178,12 +_,12 @@
             if ((this.value[p_164495_] & p_164496_) == 0L) {
                 return false;
             } else {
-                this.value[p_164495_] &= ~p_164496_;
+                this.value[p_164495_] = this.value[p_164495_] & ~p_164496_;
                 if (this.value[p_164495_] != 0L) {
                     return true;
                 } else {
                     this.lastPos = -1;
-                    --this.size;
+                    this.size--;
                     this.fixPointers(p_164495_);
                     this.shiftKeys(p_164495_);
                     if (this.size < this.maxFill / 4 && this.n > 16) {
@@ -202,7 +_,7 @@
                 int i = this.first;
                 long j = this.key[i];
                 int k = Long.numberOfTrailingZeros(this.value[i]);
-                this.value[i] &= ~(1L << k);
+                this.value[i] = this.value[i] & ~(1L << k);
                 if (this.value[i] == 0L) {
                     this.removeFirstLong();
                     this.lastPos = -1;
