--- a/net/minecraft/util/datafix/fixes/AbstractUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/AbstractUUIDFix.java
@@ -27,17 +_,20 @@
     }
 
     protected static Optional<Dynamic<?>> replaceUUIDString(Dynamic<?> p_14591_, String p_14592_, String p_14593_) {
-        return createUUIDFromString(p_14591_, p_14592_).map(p_14616_ -> p_14591_.remove(p_14592_).set(p_14593_, p_14616_));
+        return createUUIDFromString(p_14591_, p_14592_).map(p_14616_ -> p_14591_.remove(p_14592_).set(p_14593_, (Dynamic<?>)p_14616_));
     }
 
     protected static Optional<Dynamic<?>> replaceUUIDMLTag(Dynamic<?> p_14609_, String p_14610_, String p_14611_) {
-        return p_14609_.get(p_14610_).result().flatMap(AbstractUUIDFix::createUUIDFromML).map(p_14598_ -> p_14609_.remove(p_14610_).set(p_14611_, p_14598_));
+        return p_14609_.get(p_14610_)
+            .result()
+            .flatMap(AbstractUUIDFix::createUUIDFromML)
+            .map(p_14598_ -> p_14609_.remove(p_14610_).set(p_14611_, (Dynamic<?>)p_14598_));
     }
 
     protected static Optional<Dynamic<?>> replaceUUIDLeastMost(Dynamic<?> p_14618_, String p_14619_, String p_14620_) {
         String s = p_14619_ + "Most";
         String s1 = p_14619_ + "Least";
-        return createUUIDFromLongs(p_14618_, s, s1).map(p_14604_ -> p_14618_.remove(s).remove(s1).set(p_14620_, p_14604_));
+        return createUUIDFromLongs(p_14618_, s, s1).map(p_14604_ -> p_14618_.remove(s).remove(s1).set(p_14620_, (Dynamic<?>)p_14604_));
     }
 
     protected static Optional<Dynamic<?>> createUUIDFromString(Dynamic<?> p_14588_, String p_14589_) {
