--- a/net/minecraft/util/datafix/fixes/AttributesRename.java
+++ b/net/minecraft/util/datafix/fixes/AttributesRename.java
@@ -42,12 +_,12 @@
             DSL.remainderFinder(),
             p_325650_ -> p_325650_.update(
                     "AttributeModifiers",
-                    p_325647_ -> DataFixUtils.orElse(
-                            p_325647_.asStreamOpt()
+                    p_337594_ -> DataFixUtils.orElse(
+                            p_337594_.asStreamOpt()
                                 .result()
                                 .map(p_325642_ -> p_325642_.map(p_325645_ -> p_325645_.update("AttributeName", this::fixName)))
-                                .map(p_325647_::createList),
-                            p_325647_
+                                .map(p_337594_::createList),
+                            p_337594_
                         )
                 )
         );
@@ -58,12 +_,12 @@
             DSL.remainderFinder(),
             p_325651_ -> p_325651_.update(
                     "Attributes",
-                    p_325644_ -> DataFixUtils.orElse(
-                            p_325644_.asStreamOpt()
+                    p_337595_ -> DataFixUtils.orElse(
+                            p_337595_.asStreamOpt()
                                 .result()
                                 .map(p_325646_ -> p_325646_.map(p_325643_ -> p_325643_.update("Name", this::fixName)))
-                                .map(p_325644_::createList),
-                            p_325644_
+                                .map(p_337595_::createList),
+                            p_337595_
                         )
                 )
         );
