--- a/net/minecraft/util/datafix/fixes/BannerPatternFormatFix.java
+++ b/net/minecraft/util/datafix/fixes/BannerPatternFormatFix.java
@@ -6,7 +_,6 @@
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.serialization.Dynamic;
 import java.util.Map;
-import net.minecraft.util.datafix.ExtraDataFixUtils;
 
 public class BannerPatternFormatFix extends NamedEntityFix {
     private static final Map<String, String> PATTERN_ID_MAP = Map.ofEntries(
@@ -63,25 +_,24 @@
     }
 
     private static Dynamic<?> fixTag(Dynamic<?> p_330421_) {
-        return ExtraDataFixUtils.renameAndFixField(
-            p_330421_, "Patterns", "patterns", p_330685_ -> p_330685_.createList(p_330685_.asStream().map(BannerPatternFormatFix::fixLayer))
+        return p_330421_.renameAndFixField(
+            "Patterns", "patterns", p_330685_ -> p_330685_.createList(p_330685_.asStream().map(BannerPatternFormatFix::fixLayer))
         );
     }
 
     private static Dynamic<?> fixLayer(Dynamic<?> p_331951_) {
-        p_331951_ = ExtraDataFixUtils.renameAndFixField(
-            p_331951_,
+        p_331951_ = p_331951_.renameAndFixField(
             "Pattern",
             "pattern",
-            p_331355_ -> DataFixUtils.orElse(
-                    p_331355_.asString().map(p_331145_ -> PATTERN_ID_MAP.getOrDefault(p_331145_, p_331145_)).map(p_331355_::createString).result(), p_331355_
+            p_337596_ -> DataFixUtils.orElse(
+                    p_337596_.asString().map(p_331145_ -> PATTERN_ID_MAP.getOrDefault(p_331145_, p_331145_)).map(p_337596_::createString).result(), p_337596_
                 )
         );
-        return ExtraDataFixUtils.renameAndFixField(p_331951_, "Color", "color", p_330677_ -> p_330677_.createString(fixColor(p_330677_.asInt(0))));
+        return p_331951_.renameAndFixField("Color", "color", p_330677_ -> p_330677_.createString(fixColor(p_330677_.asInt(0))));
     }
 
     public static String fixColor(int p_331992_) {
-        return switch(p_331992_) {
+        return switch (p_331992_) {
             case 1 -> "orange";
             case 2 -> "magenta";
             case 3 -> "light_blue";
