--- a/net/minecraft/util/datafix/fixes/BeehiveFieldRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/BeehiveFieldRenameFix.java
@@ -22,15 +_,15 @@
 
     private Dynamic<?> fixBee(Dynamic<?> p_331694_) {
         p_331694_ = p_331694_.remove("EntityData");
-        p_331694_ = ExtraDataFixUtils.renameField(p_331694_, "TicksInHive", "ticks_in_hive");
-        return ExtraDataFixUtils.renameField(p_331694_, "MinOccupationTicks", "min_ticks_in_hive");
+        p_331694_ = p_331694_.renameField("TicksInHive", "ticks_in_hive");
+        return p_331694_.renameField("MinOccupationTicks", "min_ticks_in_hive");
     }
 
     @Override
     public TypeRewriteRule makeRule() {
         Type<?> type = this.getInputSchema().getChoiceType(References.BLOCK_ENTITY, "minecraft:beehive");
         OpticFinder<?> opticfinder = DSL.namedChoice("minecraft:beehive", type);
-        ListType<?> listtype = (ListType)type.findFieldType("Bees");
+        ListType<?> listtype = (ListType<?>)type.findFieldType("Bees");
         Type<?> type1 = listtype.getElement();
         OpticFinder<?> opticfinder1 = DSL.fieldFinder("Bees", listtype);
         OpticFinder<?> opticfinder2 = DSL.typeFinder(type1);
