--- a/net/minecraft/util/datafix/fixes/BlendingDataFix.java
+++ b/net/minecraft/util/datafix/fixes/BlendingDataFix.java
@@ -43,7 +_,7 @@
             if (!STATUSES_TO_SKIP_BLENDING.contains(s)) {
                 p_240279_ = updateBlendingData(p_240279_, 384, -64);
             } else if (optional1.isPresent()) {
-                Dynamic<?> dynamic = optional1.get();
+                Dynamic<?> dynamic = (Dynamic<?>)optional1.get();
                 String s1 = NamespacedSchema.ensureNamespaced(dynamic.get("target_status").asString("empty"));
                 if (!STATUSES_TO_SKIP_BLENDING.contains(s1)) {
                     p_240279_ = updateBlendingData(p_240279_, 256, 0);
