--- a/net/minecraft/util/datafix/fixes/BlockEntityJukeboxFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityJukeboxFix.java
@@ -5,7 +_,6 @@
 import com.mojang.datafixers.Typed;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.datafixers.types.Type;
-import com.mojang.datafixers.util.Pair;
 import com.mojang.serialization.Dynamic;
 
 public class BlockEntityJukeboxFix extends NamedEntityFix {
@@ -29,8 +_,7 @@
                 dynamic1 = dynamic1.set("Count", dynamic1.createByte((byte)1));
                 return p_14846_.set(
                         opticfinder,
-                        (Typed)((Pair)type1.readTyped(dynamic1).result().orElseThrow(() -> new IllegalStateException("Could not create record item stack.")))
-                            .getFirst()
+                        type1.readTyped(dynamic1).result().orElseThrow(() -> new IllegalStateException("Could not create record item stack.")).getFirst()
                     )
                     .set(DSL.remainderFinder(), dynamic);
             }
