--- a/net/minecraft/util/datafix/fixes/BlockPosFormatAndRenamesFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockPosFormatAndRenamesFix.java
@@ -25,24 +_,24 @@
     }
 
     private Typed<?> fixFields(Typed<?> p_326273_, Map<String, String> p_326439_) {
-        return p_326273_.update(DSL.remainderFinder(), p_326349_ -> {
-            for(Entry<String, String> entry : p_326439_.entrySet()) {
-                p_326349_ = ExtraDataFixUtils.renameAndFixField(p_326349_, entry.getKey(), entry.getValue(), ExtraDataFixUtils::fixBlockPos);
+        return p_326273_.update(DSL.remainderFinder(), p_337600_ -> {
+            for (Entry<String, String> entry : p_326439_.entrySet()) {
+                p_337600_ = p_337600_.renameAndFixField(entry.getKey(), entry.getValue(), ExtraDataFixUtils::fixBlockPos);
             }
 
-            return p_326349_;
+            return p_337600_;
         });
     }
 
     private <T> Dynamic<T> fixMapSavedData(Dynamic<T> p_326466_) {
-        return p_326466_.update("frames", p_326158_ -> p_326158_.createList(p_326158_.asStream().map(p_326479_ -> {
-                p_326479_ = ExtraDataFixUtils.renameAndFixField(p_326479_, "Pos", "pos", ExtraDataFixUtils::fixBlockPos);
-                p_326479_ = ExtraDataFixUtils.renameField(p_326479_, "Rotation", "rotation");
-                return ExtraDataFixUtils.renameField(p_326479_, "EntityId", "entity_id");
-            }))).update("banners", p_326387_ -> p_326387_.createList(p_326387_.asStream().map(p_326028_ -> {
-                p_326028_ = ExtraDataFixUtils.renameField(p_326028_, "Pos", "pos");
-                p_326028_ = ExtraDataFixUtils.renameField(p_326028_, "Color", "color");
-                return ExtraDataFixUtils.renameField(p_326028_, "Name", "name");
+        return p_326466_.update("frames", p_326158_ -> p_326158_.createList(p_326158_.asStream().map(p_337598_ -> {
+                p_337598_ = p_337598_.renameAndFixField("Pos", "pos", ExtraDataFixUtils::fixBlockPos);
+                p_337598_ = p_337598_.renameField("Rotation", "rotation");
+                return p_337598_.renameField("EntityId", "entity_id");
+            }))).update("banners", p_326387_ -> p_326387_.createList(p_326387_.asStream().map(p_337601_ -> {
+                p_337601_ = p_337601_.renameField("Pos", "pos");
+                p_337601_ = p_337601_.renameField("Color", "color");
+                return p_337601_.renameField("Name", "name");
             })));
     }
 
@@ -74,7 +_,7 @@
         p_326170_.add(this.createEntityFixer(References.ENTITY, "minecraft:end_crystal", Map.of("BeamTarget", "beam_target")));
         p_326170_.add(this.createEntityFixer(References.ENTITY, "minecraft:wandering_trader", Map.of("WanderTarget", "wander_target")));
 
-        for(String s : PATROLLING_MOBS) {
+        for (String s : PATROLLING_MOBS) {
             p_326170_.add(this.createEntityFixer(References.ENTITY, s, Map.of("PatrolTarget", "patrol_target")));
         }
 
@@ -82,9 +_,7 @@
             this.fixTypeEverywhereTyped(
                 "BlockPos format in Leash for mobs",
                 this.getInputSchema().getType(References.ENTITY),
-                p_326408_ -> p_326408_.update(
-                        DSL.remainderFinder(), p_326323_ -> ExtraDataFixUtils.renameAndFixField(p_326323_, "Leash", "leash", ExtraDataFixUtils::fixBlockPos)
-                    )
+                p_326408_ -> p_326408_.update(DSL.remainderFinder(), p_337602_ -> p_337602_.renameAndFixField("Leash", "leash", ExtraDataFixUtils::fixBlockPos))
             )
         );
     }
