--- a/net/minecraft/util/datafix/fixes/ChunkToProtochunkFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkToProtochunkFix.java
@@ -49,17 +_,17 @@
     }
 
     private static <T> Dynamic<T> repackBiomes(Dynamic<T> p_199880_) {
-        return p_199880_.update("Biomes", p_199862_ -> DataFixUtils.orElse(p_199862_.asByteBufferOpt().result().map(p_199868_ -> {
+        return p_199880_.update("Biomes", p_337614_ -> DataFixUtils.orElse(p_337614_.asByteBufferOpt().result().map(p_199868_ -> {
                 int[] aint = new int[256];
 
-                for(int i = 0; i < aint.length; ++i) {
+                for (int i = 0; i < aint.length; i++) {
                     if (i < p_199868_.capacity()) {
                         aint[i] = p_199868_.get(i) & 255;
                     }
                 }
 
                 return p_199880_.createIntList(Arrays.stream(aint));
-            }), p_199862_));
+            }), p_337614_));
     }
 
     private static <T> Dynamic<T> repackTicks(Dynamic<T> p_199882_) {
