--- a/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
@@ -27,11 +_,9 @@
     public TypeRewriteRule makeRule() {
         Type<?> type = this.getOutputSchema().getType(References.CHUNK);
         Type<?> type1 = type.findFieldType("Level");
-        Type<?> type2 = type1.findFieldType("TileEntities");
-        if (!(type2 instanceof ListType)) {
+        if (!(type1.findFieldType("TileEntities") instanceof ListType<?> listtype)) {
             throw new IllegalStateException("Tile entity type is not a list type.");
         } else {
-            ListType<?> listtype = (ListType)type2;
             return this.cap(type1, listtype);
         }
     }
@@ -58,7 +_,7 @@
                     int k = dynamic.get("zPos").asInt(0);
                     List<TE> list = Lists.newArrayList(typed.getOrCreate(opticfinder1));
         
-                    for(Dynamic<?> dynamic1 : dynamic.get("Sections").asList(Function.identity())) {
+                    for (Dynamic<?> dynamic1 : dynamic.get("Sections").asList(Function.identity())) {
                         int l = dynamic1.get("Y").asInt(0);
                         Streams.mapWithIndex(dynamic1.get("Blocks").asIntStream(), (p_274917_, p_274918_) -> {
                                 if (416 == (p_274917_ & 0xFF) << 4) {
@@ -78,12 +_,12 @@
                                 }
                             })
                             .forEachOrdered(
-                                p_274922_ -> {
-                                    if (p_274922_ != null) {
+                                p_337607_ -> {
+                                    if (p_337607_ != null) {
                                         list.add(
-                                            (TE)((Pair)type.read(dynamic1.createMap(p_274922_))
-                                                    .result()
-                                                    .orElseThrow(() -> new IllegalStateException("Could not parse newly created bed block entity.")))
+                                            type.read(dynamic1.createMap((Map<? extends Dynamic<?>, ? extends Dynamic<?>>)p_337607_))
+                                                .result()
+                                                .orElseThrow(() -> new IllegalStateException("Could not parse newly created bed block entity."))
                                                 .getFirst()
                                         );
                                     }
