--- a/net/minecraft/util/datafix/fixes/ChunkStructuresTemplateRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkStructuresTemplateRenameFix.java
@@ -133,7 +_,9 @@
     }
 
     private Dynamic<?> fixChildren(Dynamic<?> p_275363_) {
-        return p_275363_.update("Children", p_274926_ -> p_275363_.createList(p_274926_.asStream().map(p_274924_ -> this.fixTag(p_275363_, p_274924_))));
+        return p_275363_.update(
+            "Children", p_274926_ -> p_275363_.createList(p_274926_.asStream().map(p_274924_ -> this.fixTag(p_275363_, (Dynamic<?>)p_274924_)))
+        );
     }
 
     private Dynamic<?> fixTag(Dynamic<?> p_15281_, Dynamic<?> p_15282_) {
