--- a/net/minecraft/util/datafix/fixes/ChunkPalettedStorageFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkPalettedStorageFix.java
@@ -95,7 +_,7 @@
         mapDoor(p_15105_, "dark_oak_door", 3152);
     });
     static final Map<String, Dynamic<?>> NOTE_BLOCK_MAP = DataFixUtils.make(Maps.newHashMap(), p_15102_ -> {
-        for(int i = 0; i < 26; ++i) {
+        for (int i = 0; i < 26; i++) {
             p_15102_.put("true" + i, BlockStateData.parse("{Name:'minecraft:note_block',Properties:{powered:'true',note:'" + i + "'}}"));
             p_15102_.put("false" + i, BlockStateData.parse("{Name:'minecraft:note_block',Properties:{powered:'false',note:'" + i + "'}}"));
         }
@@ -119,14 +_,14 @@
         p_15070_.put(15, "black");
     });
     static final Map<String, Dynamic<?>> BED_BLOCK_MAP = DataFixUtils.make(Maps.newHashMap(), p_15095_ -> {
-        for(Entry<String> entry : DYE_COLOR_MAP.int2ObjectEntrySet()) {
+        for (Entry<String> entry : DYE_COLOR_MAP.int2ObjectEntrySet()) {
             if (!Objects.equals(entry.getValue(), "red")) {
                 addBeds(p_15095_, entry.getIntKey(), entry.getValue());
             }
         }
     });
     static final Map<String, Dynamic<?>> BANNER_BLOCK_MAP = DataFixUtils.make(Maps.newHashMap(), p_15072_ -> {
-        for(Entry<String> entry : DYE_COLOR_MAP.int2ObjectEntrySet()) {
+        for (Entry<String> entry : DYE_COLOR_MAP.int2ObjectEntrySet()) {
             if (!Objects.equals(entry.getValue(), "white")) {
                 addBanners(p_15072_, 15 - entry.getIntKey(), entry.getValue());
             }
@@ -145,7 +_,7 @@
         p_15078_.put(p_15079_ + "south", BlockStateData.parse("{Name:'minecraft:" + p_15080_ + "_wall_" + p_15081_ + "',Properties:{facing:'south'}}"));
         p_15078_.put(p_15079_ + "west", BlockStateData.parse("{Name:'minecraft:" + p_15080_ + "_wall_" + p_15081_ + "',Properties:{facing:'west'}}"));
 
-        for(int i = 0; i < 16; ++i) {
+        for (int i = 0; i < 16; i++) {
             p_15078_.put("" + p_15079_ + i, BlockStateData.parse("{Name:'minecraft:" + p_15080_ + "_" + p_15081_ + "',Properties:{rotation:'" + i + "'}}"));
         }
     }
@@ -417,7 +_,7 @@
     }
 
     private static void addBanners(Map<String, Dynamic<?>> p_15097_, int p_15098_, String p_15099_) {
-        for(int i = 0; i < 16; ++i) {
+        for (int i = 0; i < 16; i++) {
             p_15097_.put(i + "_" + p_15098_, BlockStateData.parse("{Name:'minecraft:" + p_15099_ + "_banner',Properties:{rotation:'" + i + "'}}"));
         }
 
@@ -447,7 +_,7 @@
     private Dynamic<?> fix(Dynamic<?> p_15093_) {
         Optional<? extends Dynamic<?>> optional = p_15093_.get("Level").result();
         return optional.isPresent() && optional.get().get("Sections").asStreamOpt().result().isPresent()
-            ? p_15093_.set("Level", new ChunkPalettedStorageFix.UpgradeChunk(optional.get()).write())
+            ? p_15093_.set("Level", new ChunkPalettedStorageFix.UpgradeChunk((Dynamic<?>)optional.get()).write())
             : p_15093_;
     }
 
@@ -680,7 +_,7 @@
                 ChunkPalettedStorageFix.idFor(this.palette, ChunkPalettedStorageFix.AIR);
                 this.listTag.add(ChunkPalettedStorageFix.AIR);
 
-                for(int i = 0; i < 4096; ++i) {
+                for (int i = 0; i < 4096; i++) {
                     int j = i & 15;
                     int k = i >> 8 & 15;
                     int l = i >> 4 & 15;
@@ -726,7 +_,7 @@
                 int i = Math.max(4, DataFixUtils.ceillog2(this.seen.size()));
                 PackedBitStorage packedbitstorage = new PackedBitStorage(i, 4096);
 
-                for(int j = 0; j < this.buffer.length; ++j) {
+                for (int j = 0; j < this.buffer.length; j++) {
                     packedbitstorage.set(j, this.buffer[j]);
                 }
 
@@ -752,15 +_,14 @@
             this.z = p_15222_.get("zPos").asInt(0) << 4;
             p_15222_.get("TileEntities")
                 .asStreamOpt()
-                .result()
-                .ifPresent(
+                .ifSuccess(
                     p_15241_ -> p_15241_.forEach(
                             p_145228_ -> {
                                 int l3 = p_145228_.get("x").asInt(0) - this.x & 15;
                                 int i4 = p_145228_.get("y").asInt(0);
                                 int j4 = p_145228_.get("z").asInt(0) - this.z & 15;
                                 int k4 = i4 << 8 | j4 << 4 | l3;
-                                if (this.blockEntities.put(k4, p_145228_) != null) {
+                                if (this.blockEntities.put(k4, (Dynamic<?>)p_145228_) != null) {
                                     ChunkPalettedStorageFix.LOGGER
                                         .warn("In chunk: {}x{} found a duplicate block entity at position: [{}, {}, {}]", this.x, this.z, l3, i4, j4);
                                 }
@@ -768,19 +_,19 @@
                         )
                 );
             boolean flag = p_15222_.get("convertedFromAlphaFormat").asBoolean(false);
-            p_15222_.get("Sections").asStreamOpt().result().ifPresent(p_15235_ -> p_15235_.forEach(p_145226_ -> {
-                    ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section1 = new ChunkPalettedStorageFix.Section(p_145226_);
+            p_15222_.get("Sections").asStreamOpt().ifSuccess(p_15235_ -> p_15235_.forEach(p_145226_ -> {
+                    ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section1 = new ChunkPalettedStorageFix.Section((Dynamic<?>)p_145226_);
                     this.sides = chunkpalettedstoragefix$section1.upgrade(this.sides);
                     this.sections[chunkpalettedstoragefix$section1.y] = chunkpalettedstoragefix$section1;
                 }));
 
-            for(ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section : this.sections) {
+            for (ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section : this.sections) {
                 if (chunkpalettedstoragefix$section != null) {
-                    for(java.util.Map.Entry<Integer, IntList> entry : chunkpalettedstoragefix$section.toFix.entrySet()) {
+                    for (java.util.Map.Entry<Integer, IntList> entry : chunkpalettedstoragefix$section.toFix.entrySet()) {
                         int i = chunkpalettedstoragefix$section.y << 12;
-                        switch(entry.getKey()) {
+                        switch (entry.getKey()) {
                             case 2:
-                                for(int i3 : entry.getValue()) {
+                                for (int i3 : entry.getValue()) {
                                     i3 |= i;
                                     Dynamic<?> dynamic11 = this.getBlock(i3);
                                     if ("minecraft:grass_block".equals(ChunkPalettedStorageFix.getName(dynamic11))) {
@@ -792,7 +_,7 @@
                                 }
                                 break;
                             case 3:
-                                for(int l2 : entry.getValue()) {
+                                for (int l2 : entry.getValue()) {
                                     l2 |= i;
                                     Dynamic<?> dynamic10 = this.getBlock(l2);
                                     if ("minecraft:podzol".equals(ChunkPalettedStorageFix.getName(dynamic10))) {
@@ -804,7 +_,7 @@
                                 }
                                 break;
                             case 25:
-                                for(int k2 : entry.getValue()) {
+                                for (int k2 : entry.getValue()) {
                                     k2 |= i;
                                     Dynamic<?> dynamic9 = this.removeBlockEntity(k2);
                                     if (dynamic9 != null) {
@@ -817,7 +_,7 @@
                                 }
                                 break;
                             case 26:
-                                for(int j2 : entry.getValue()) {
+                                for (int j2 : entry.getValue()) {
                                     j2 |= i;
                                     Dynamic<?> dynamic8 = this.getBlockEntity(j2);
                                     Dynamic<?> dynamic14 = this.getBlock(j2);
@@ -842,7 +_,7 @@
                             case 195:
                             case 196:
                             case 197:
-                                for(int i2 : entry.getValue()) {
+                                for (int i2 : entry.getValue()) {
                                     i2 |= i;
                                     Dynamic<?> dynamic7 = this.getBlock(i2);
                                     if (ChunkPalettedStorageFix.getName(dynamic7).endsWith("_door")) {
@@ -864,7 +_,7 @@
                                 }
                                 break;
                             case 86:
-                                for(int l1 : entry.getValue()) {
+                                for (int l1 : entry.getValue()) {
                                     l1 |= i;
                                     Dynamic<?> dynamic6 = this.getBlock(l1);
                                     if ("minecraft:carved_pumpkin".equals(ChunkPalettedStorageFix.getName(dynamic6))) {
@@ -876,7 +_,7 @@
                                 }
                                 break;
                             case 110:
-                                for(int k1 : entry.getValue()) {
+                                for (int k1 : entry.getValue()) {
                                     k1 |= i;
                                     Dynamic<?> dynamic5 = this.getBlock(k1);
                                     if ("minecraft:mycelium".equals(ChunkPalettedStorageFix.getName(dynamic5))) {
@@ -888,7 +_,7 @@
                                 }
                                 break;
                             case 140:
-                                for(int j1 : entry.getValue()) {
+                                for (int j1 : entry.getValue()) {
                                     j1 |= i;
                                     Dynamic<?> dynamic4 = this.removeBlockEntity(j1);
                                     if (dynamic4 != null) {
@@ -902,7 +_,7 @@
                                 }
                                 break;
                             case 144:
-                                for(int i1 : entry.getValue()) {
+                                for (int i1 : entry.getValue()) {
                                     i1 |= i;
                                     Dynamic<?> dynamic3 = this.getBlockEntity(i1);
                                     if (dynamic3 != null) {
@@ -923,7 +_,7 @@
                                 }
                                 break;
                             case 175:
-                                for(int l : entry.getValue()) {
+                                for (int l : entry.getValue()) {
                                     l |= i;
                                     Dynamic<?> dynamic2 = this.getBlock(l);
                                     if ("upper".equals(ChunkPalettedStorageFix.getProperty(dynamic2, "half"))) {
@@ -947,7 +_,7 @@
                                 break;
                             case 176:
                             case 177:
-                                for(int j : entry.getValue()) {
+                                for (int j : entry.getValue()) {
                                     j |= i;
                                     Dynamic<?> dynamic = this.getBlockEntity(j);
                                     Dynamic<?> dynamic1 = this.getBlock(j);
@@ -978,7 +_,7 @@
         }
 
         public static int relative(int p_15227_, ChunkPalettedStorageFix.Direction p_15228_) {
-            switch(p_15228_.getAxis()) {
+            switch (p_15228_.getAxis()) {
                 case X:
                     int i = (p_15227_ & 15) + p_15228_.getAxisDirection().getStep();
                     return i >= 0 && i <= 15 ? p_15227_ & -16 | i : -1;
@@ -1028,7 +_,7 @@
             Dynamic<?> dynamic1 = dynamic.emptyMap();
             List<Dynamic<?>> list = Lists.newArrayList();
 
-            for(ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section : this.sections) {
+            for (ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section : this.sections) {
                 if (chunkpalettedstoragefix$section != null) {
                     list.add(chunkpalettedstoragefix$section.write());
                     dynamic1 = dynamic1.set(
