--- a/net/minecraft/util/datafix/fixes/DropInvalidSignDataFix.java
+++ b/net/minecraft/util/datafix/fixes/DropInvalidSignDataFix.java
@@ -23,7 +_,7 @@
         p_295232_ = p_295232_.update("front_text", DropInvalidSignDataFix::fixText);
         p_295232_ = p_295232_.update("back_text", DropInvalidSignDataFix::fixText);
 
-        for(String s : FIELDS_TO_DROP) {
+        for (String s : FIELDS_TO_DROP) {
             p_295232_ = p_295232_.remove(s);
         }
 
