--- a/net/minecraft/util/datafix/fixes/EntityRidingToPassengersFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityRidingToPassengersFix.java
@@ -62,13 +_,15 @@
                             Optional<Pair<String, Pair<Either<List<NewEntityTree>, Unit>, Entity>>> optional = Optional.empty();
                             Pair<String, Pair<Either<OldEntityTree, Unit>, Entity>> pair = p_145320_;
         
-                            while(true) {
+                            while (true) {
                                 Either<List<NewEntityTree>, Unit> either = DataFixUtils.orElse(
                                     optional.map(
                                         p_145326_ -> {
                                             Typed<NewEntityTree> typed = p_15645_.pointTyped(p_15653_)
                                                 .orElseThrow(() -> new IllegalStateException("Could not create new entity tree"));
-                                            NewEntityTree newentitytree = typed.set(opticfinder1, p_145326_)
+                                            NewEntityTree newentitytree = typed.set(
+                                                    opticfinder1, (Pair<String, Pair<Either<List<NewEntityTree>, Unit>, Entity>>)p_145326_
+                                                )
                                                 .getOptional(opticfinder2)
                                                 .orElseThrow(() -> new IllegalStateException("Should always have an entity tree here"));
                                             return Either.left(ImmutableList.of(newentitytree));
