--- a/net/minecraft/util/datafix/fixes/EntityHorseSaddleFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityHorseSaddleFix.java
@@ -23,7 +_,7 @@
         Optional<? extends Typed<?>> optional = p_15445_.getOptionalTyped(opticfinder1);
         Dynamic<?> dynamic = p_15445_.get(DSL.remainderFinder());
         if (optional.isEmpty() && dynamic.get("Saddle").asBoolean(false)) {
-            Typed<?> typed = (Typed)type.pointTyped(p_15445_.getOps()).orElseThrow(IllegalStateException::new);
+            Typed<?> typed = type.pointTyped(p_15445_.getOps()).orElseThrow(IllegalStateException::new);
             typed = typed.set(opticfinder, Pair.of(References.ITEM_NAME.typeName(), "minecraft:saddle"));
             Dynamic<?> dynamic1 = dynamic.emptyMap();
             dynamic1 = dynamic1.set("Count", dynamic1.createByte((byte)1));
