--- a/net/minecraft/util/datafix/fixes/EntityUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityUUIDFix.java
@@ -10,7 +_,6 @@
 import com.mojang.serialization.Dynamic;
 import java.util.Optional;
 import java.util.Set;
-import java.util.function.Supplier;
 import org.slf4j.Logger;
 
 public class EntityUUIDFix extends AbstractUUIDFix {
@@ -31,27 +_,27 @@
         return this.fixTypeEverywhereTyped("EntityUUIDFixes", this.getInputSchema().getType(this.typeReference), p_15725_ -> {
             p_15725_ = p_15725_.update(DSL.remainderFinder(), EntityUUIDFix::updateEntityUUID);
 
-            for(String s : ABSTRACT_HORSES) {
+            for (String s : ABSTRACT_HORSES) {
                 p_15725_ = this.updateNamedChoice(p_15725_, s, EntityUUIDFix::updateAnimalOwner);
             }
 
-            for(String s1 : TAMEABLE_ANIMALS) {
+            for (String s1 : TAMEABLE_ANIMALS) {
                 p_15725_ = this.updateNamedChoice(p_15725_, s1, EntityUUIDFix::updateAnimalOwner);
             }
 
-            for(String s2 : ANIMALS) {
+            for (String s2 : ANIMALS) {
                 p_15725_ = this.updateNamedChoice(p_15725_, s2, EntityUUIDFix::updateAnimal);
             }
 
-            for(String s3 : MOBS) {
+            for (String s3 : MOBS) {
                 p_15725_ = this.updateNamedChoice(p_15725_, s3, EntityUUIDFix::updateMob);
             }
 
-            for(String s4 : LIVING_ENTITIES) {
+            for (String s4 : LIVING_ENTITIES) {
                 p_15725_ = this.updateNamedChoice(p_15725_, s4, EntityUUIDFix::updateLivingEntity);
             }
 
-            for(String s5 : PROJECTILES) {
+            for (String s5 : PROJECTILES) {
                 p_15725_ = this.updateNamedChoice(p_15725_, s5, EntityUUIDFix::updateProjectile);
             }
 
@@ -105,15 +_,11 @@
     private static Dynamic<?> updateFox(Dynamic<?> p_15758_) {
         Optional<Dynamic<?>> optional = p_15758_.get("TrustedUUIDs")
             .result()
-            .map(
-                p_15748_ -> p_15758_.createList(
-                        p_15748_.asStream().map(p_145341_ -> createUUIDFromML(p_145341_).orElseGet((Supplier<? extends Dynamic<?>>)(() -> {
-                                LOGGER.warn("Trusted contained invalid data.");
-                                return p_145341_;
-                            })))
-                    )
-            );
-        return DataFixUtils.orElse(optional.map(p_15743_ -> p_15758_.remove("TrustedUUIDs").set("Trusted", p_15743_)), p_15758_);
+            .map(p_15748_ -> p_15758_.createList(p_15748_.asStream().map(p_145341_ -> createUUIDFromML((Dynamic<?>)p_145341_).orElseGet(() -> {
+                        LOGGER.warn("Trusted contained invalid data.");
+                        return p_145341_;
+                    }))));
+        return DataFixUtils.orElse(optional.map(p_15743_ -> p_15758_.remove("TrustedUUIDs").set("Trusted", (Dynamic<?>)p_15743_)), p_15758_);
     }
 
     private static Dynamic<?> updateHurtBy(Dynamic<?> p_15760_) {
@@ -143,7 +_,8 @@
                             p_145337_ -> p_145337_.update(
                                     "Modifiers",
                                     p_145335_ -> p_145337_.createList(
-                                            p_145335_.asStream().map(p_145339_ -> replaceUUIDLeastMost(p_145339_, "UUID", "UUID").orElse(p_145339_))
+                                            p_145335_.asStream()
+                                                .map(p_145339_ -> replaceUUIDLeastMost((Dynamic<?>)p_145339_, "UUID", "UUID").orElse((Dynamic<?>)p_145339_))
                                         )
                                 )
                         )
@@ -152,7 +_,9 @@
     }
 
     private static Dynamic<?> updateProjectile(Dynamic<?> p_15769_) {
-        return DataFixUtils.orElse(p_15769_.get("OwnerUUID").result().map(p_15728_ -> p_15769_.remove("OwnerUUID").set("Owner", p_15728_)), p_15769_);
+        return DataFixUtils.orElse(
+            p_15769_.get("OwnerUUID").result().map(p_15728_ -> p_15769_.remove("OwnerUUID").set("Owner", (Dynamic<?>)p_15728_)), p_15769_
+        );
     }
 
     public static Dynamic<?> updateEntityUUID(Dynamic<?> p_15735_) {
