--- a/net/minecraft/util/datafix/fixes/EntityStringUuidFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityStringUuidFix.java
@@ -21,15 +_,15 @@
             this.getInputSchema().getType(References.ENTITY),
             p_15697_ -> p_15697_.update(
                     DSL.remainderFinder(),
-                    p_145331_ -> {
-                        Optional<String> optional = p_145331_.get("UUID").asString().result();
+                    p_337618_ -> {
+                        Optional<String> optional = p_337618_.get("UUID").asString().result();
                         if (optional.isPresent()) {
                             UUID uuid = UUID.fromString(optional.get());
-                            return p_145331_.remove("UUID")
-                                .set("UUIDMost", p_145331_.createLong(uuid.getMostSignificantBits()))
-                                .set("UUIDLeast", p_145331_.createLong(uuid.getLeastSignificantBits()));
+                            return p_337618_.remove("UUID")
+                                .set("UUIDMost", p_337618_.createLong(uuid.getMostSignificantBits()))
+                                .set("UUIDLeast", p_337618_.createLong(uuid.getLeastSignificantBits()));
                         } else {
-                            return p_145331_;
+                            return p_337618_;
                         }
                     }
                 )
