--- a/net/minecraft/util/datafix/fixes/EntityVariantFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityVariantFix.java
@@ -22,13 +_,13 @@
 
     private static <T> Dynamic<T> updateAndRename(Dynamic<T> p_216637_, String p_216638_, String p_216639_, Function<Dynamic<T>, Dynamic<T>> p_216640_) {
         return p_216637_.map(
-            p_216646_ -> {
+            p_337624_ -> {
                 DynamicOps<T> dynamicops = p_216637_.getOps();
                 Function<T, T> function = p_216656_ -> p_216640_.apply(new Dynamic<>(dynamicops, p_216656_)).getValue();
-                return dynamicops.get(p_216646_, p_216638_)
-                    .map(p_216652_ -> dynamicops.set((T)p_216646_, p_216639_, function.apply(p_216652_)))
+                return dynamicops.get((T)p_337624_, p_216638_)
+                    .map(p_216652_ -> dynamicops.set((T)p_337624_, p_216639_, function.apply((T)p_216652_)))
                     .result()
-                    .orElse(p_216646_);
+                    .orElse((T)p_337624_);
             }
         );
     }
@@ -41,8 +_,8 @@
                     p_216632_,
                     this.fieldName,
                     "variant",
-                    p_216658_ -> DataFixUtils.orElse(
-                            p_216658_.asNumber().map(p_216635_ -> p_216658_.createString(this.idConversions.apply(p_216635_.intValue()))).result(), p_216658_
+                    p_337619_ -> DataFixUtils.orElse(
+                            p_337619_.asNumber().map(p_216635_ -> p_337619_.createString(this.idConversions.apply(p_216635_.intValue()))).result(), p_337619_
                         )
                 )
         );
