--- a/net/minecraft/util/datafix/fixes/EntityPaintingFieldsRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityPaintingFieldsRenameFix.java
@@ -4,7 +_,6 @@
 import com.mojang.datafixers.Typed;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.serialization.Dynamic;
-import net.minecraft.util.datafix.ExtraDataFixUtils;
 
 public class EntityPaintingFieldsRenameFix extends NamedEntityFix {
     public EntityPaintingFieldsRenameFix(Schema p_216606_) {
@@ -12,7 +_,7 @@
     }
 
     public Dynamic<?> fixTag(Dynamic<?> p_216610_) {
-        return ExtraDataFixUtils.renameField(ExtraDataFixUtils.renameField(p_216610_, "Motive", "variant"), "Facing", "facing");
+        return p_216610_.renameField("Motive", "variant").renameField("Facing", "facing");
     }
 
     @Override
