--- a/net/minecraft/util/datafix/fixes/EntityZombieSplitFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityZombieSplitFix.java
@@ -15,7 +_,7 @@
         if (Objects.equals("Zombie", p_15801_)) {
             String s = "Zombie";
             int i = p_15802_.get("ZombieType").asInt(0);
-            switch(i) {
+            switch (i) {
                 case 0:
                 default:
                     break;
