--- a/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
+++ b/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
@@ -38,7 +_,7 @@
     @SafeVarargs
     private <T> Function<Typed<?>, Typed<?>> chainAllFilters(Function<Typed<?>, Typed<?>>... p_309142_) {
         return p_309020_ -> {
-            for(Function<Typed<?>, Typed<?>> function : p_309142_) {
+            for (Function<Typed<?>, Typed<?>> function : p_309142_) {
                 p_309020_ = function.apply(p_309020_);
             }
 
