--- a/net/minecraft/util/datafix/fixes/HeightmapRenamingFix.java
+++ b/net/minecraft/util/datafix/fixes/HeightmapRenamingFix.java
@@ -29,31 +_,31 @@
         if (optional.isEmpty()) {
             return p_15899_;
         } else {
-            Dynamic<?> dynamic = optional.get();
+            Dynamic<?> dynamic = (Dynamic<?>)optional.get();
             Optional<? extends Dynamic<?>> optional1 = dynamic.get("LIQUID").result();
             if (optional1.isPresent()) {
                 dynamic = dynamic.remove("LIQUID");
-                dynamic = dynamic.set("WORLD_SURFACE_WG", optional1.get());
+                dynamic = dynamic.set("WORLD_SURFACE_WG", (Dynamic<?>)optional1.get());
             }
 
             Optional<? extends Dynamic<?>> optional2 = dynamic.get("SOLID").result();
             if (optional2.isPresent()) {
                 dynamic = dynamic.remove("SOLID");
-                dynamic = dynamic.set("OCEAN_FLOOR_WG", optional2.get());
-                dynamic = dynamic.set("OCEAN_FLOOR", optional2.get());
+                dynamic = dynamic.set("OCEAN_FLOOR_WG", (Dynamic<?>)optional2.get());
+                dynamic = dynamic.set("OCEAN_FLOOR", (Dynamic<?>)optional2.get());
             }
 
             Optional<? extends Dynamic<?>> optional3 = dynamic.get("LIGHT").result();
             if (optional3.isPresent()) {
                 dynamic = dynamic.remove("LIGHT");
-                dynamic = dynamic.set("LIGHT_BLOCKING", optional3.get());
+                dynamic = dynamic.set("LIGHT_BLOCKING", (Dynamic<?>)optional3.get());
             }
 
             Optional<? extends Dynamic<?>> optional4 = dynamic.get("RAIN").result();
             if (optional4.isPresent()) {
                 dynamic = dynamic.remove("RAIN");
-                dynamic = dynamic.set("MOTION_BLOCKING", optional4.get());
-                dynamic = dynamic.set("MOTION_BLOCKING_NO_LEAVES", optional4.get());
+                dynamic = dynamic.set("MOTION_BLOCKING", (Dynamic<?>)optional4.get());
+                dynamic = dynamic.set("MOTION_BLOCKING_NO_LEAVES", (Dynamic<?>)optional4.get());
             }
 
             return p_15899_.set("Heightmaps", dynamic);
