--- a/net/minecraft/util/datafix/fixes/IglooMetadataRemovalFix.java
+++ b/net/minecraft/util/datafix/fixes/IglooMetadataRemovalFix.java
@@ -31,7 +_,7 @@
 
     private static <T> Dynamic<T> removeIglooPieces(Dynamic<T> p_15909_) {
         return p_15909_.asStreamOpt()
-            .map(p_15907_ -> p_15907_.filter(p_145382_ -> !isIglooPiece(p_145382_)))
+            .map(p_15907_ -> p_15907_.filter(p_145382_ -> !isIglooPiece((Dynamic<?>)p_145382_)))
             .map(p_15909_::createList)
             .result()
             .orElse(p_15909_);
