--- a/net/minecraft/util/datafix/fixes/ItemRemoveBlockEntityTagFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemRemoveBlockEntityTagFix.java
@@ -32,18 +_,18 @@
             if (optional.isPresent() && this.items.contains(optional.get().getSecond())) {
                 Optional<? extends Typed<?>> optional1 = p_242866_.getOptionalTyped(opticfinder1);
                 if (optional1.isPresent()) {
-                    Typed<?> typed = optional1.get();
+                    Typed<?> typed = (Typed<?>)optional1.get();
                     Optional<? extends Typed<?>> optional2 = typed.getOptionalTyped(opticfinder2);
                     if (optional2.isPresent()) {
                         Optional<? extends Dynamic<?>> optional3 = typed.write().result();
-                        Dynamic<?> dynamic = optional3.isPresent() ? optional3.get() : typed.get(DSL.remainderFinder());
+                        Dynamic<?> dynamic = (Dynamic<?>)(optional3.isPresent() ? optional3.get() : typed.get(DSL.remainderFinder()));
                         Dynamic<?> dynamic1 = dynamic.remove("BlockEntityTag");
                         Optional<? extends Pair<? extends Typed<?>, ?>> optional4 = opticfinder1.type().readTyped(dynamic1).result();
                         if (optional4.isEmpty()) {
                             return p_242866_;
                         }
 
-                        return p_242866_.<Object, Object>set((OpticFinder<Object>)opticfinder1, (Typed<Object>)((Pair)optional4.get()).getFirst());
+                        return p_242866_.set(opticfinder1, (Typed<?>)optional4.get().getFirst());
                     }
                 }
             }
