--- a/net/minecraft/util/datafix/fixes/ItemStackEnchantmentNamesFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemStackEnchantmentNamesFix.java
@@ -72,21 +_,21 @@
             .map(p_16073_::createList)
             .result();
         if (optional.isPresent()) {
-            p_16073_ = p_16073_.remove("ench").set("Enchantments", optional.get());
+            p_16073_ = p_16073_.remove("ench").set("Enchantments", (Dynamic<?>)optional.get());
         }
 
         return p_16073_.update(
             "StoredEnchantments",
-            p_16079_ -> DataFixUtils.orElse(
-                    p_16079_.asStreamOpt()
+            p_337641_ -> DataFixUtils.orElse(
+                    p_337641_.asStreamOpt()
                         .map(
                             p_145421_ -> p_145421_.map(
                                     p_145423_ -> p_145423_.set("id", p_145423_.createString(MAP.getOrDefault(p_145423_.get("id").asInt(0), "null")))
                                 )
                         )
-                        .map(p_16079_::createList)
+                        .map(p_337641_::createList)
                         .result(),
-                    p_16079_
+                    p_337641_
                 )
         );
     }
