--- a/net/minecraft/util/datafix/fixes/ItemWrittenBookPagesStrictJsonFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemWrittenBookPagesStrictJsonFix.java
@@ -20,8 +_,8 @@
     public Dynamic<?> fixTag(Dynamic<?> p_16172_) {
         return p_16172_.update(
             "pages",
-            p_16175_ -> DataFixUtils.orElse(
-                    p_16175_.asStreamOpt().map(p_145441_ -> p_145441_.map(ComponentDataFixUtils::rewriteFromLenient)).map(p_16172_::createList).result(),
+            p_337643_ -> DataFixUtils.orElse(
+                    p_337643_.asStreamOpt().map(p_145441_ -> p_145441_.map(ComponentDataFixUtils::rewriteFromLenient)).map(p_16172_::createList).result(),
                     p_16172_.emptyList()
                 )
         );
