--- a/net/minecraft/util/datafix/fixes/ItemCustomNameToComponentFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemCustomNameToComponentFix.java
@@ -19,16 +_,10 @@
     private Dynamic<?> fixTag(Dynamic<?> p_15935_) {
         Optional<? extends Dynamic<?>> optional = p_15935_.get("display").result();
         if (optional.isPresent()) {
-            Dynamic<?> dynamic = optional.get();
+            Dynamic<?> dynamic = (Dynamic<?>)optional.get();
             Optional<String> optional1 = dynamic.get("Name").asString().result();
             if (optional1.isPresent()) {
                 dynamic = dynamic.set("Name", ComponentDataFixUtils.createPlainTextComponent(dynamic.getOps(), optional1.get()));
-            } else {
-                Optional<String> optional2 = dynamic.get("LocName").asString().result();
-                if (optional2.isPresent()) {
-                    dynamic = dynamic.set("Name", ComponentDataFixUtils.createTranslatableComponent(dynamic.getOps(), optional2.get()));
-                    dynamic = dynamic.remove("LocName");
-                }
             }
 
             return p_15935_.set("display", dynamic);
