--- a/net/minecraft/util/datafix/fixes/ItemBannerColorFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemBannerColorFix.java
@@ -35,17 +_,17 @@
                     Dynamic<?> dynamic = p_15924_.get(DSL.remainderFinder());
                     Optional<? extends Typed<?>> optional1 = p_15924_.getOptionalTyped(opticfinder1);
                     if (optional1.isPresent()) {
-                        Typed<?> typed = optional1.get();
+                        Typed<?> typed = (Typed<?>)optional1.get();
                         Optional<? extends Typed<?>> optional2 = typed.getOptionalTyped(opticfinder2);
                         if (optional2.isPresent()) {
-                            Typed<?> typed1 = optional2.get();
+                            Typed<?> typed1 = (Typed<?>)optional2.get();
                             Dynamic<?> dynamic1 = typed.get(DSL.remainderFinder());
                             Dynamic<?> dynamic2 = typed1.getOrCreate(DSL.remainderFinder());
                             if (dynamic2.get("Base").asNumber().result().isPresent()) {
                                 dynamic = dynamic.set("Damage", dynamic.createShort((short)(dynamic2.get("Base").asInt(0) & 15)));
                                 Optional<? extends Dynamic<?>> optional3 = dynamic1.get("display").result();
                                 if (optional3.isPresent()) {
-                                    Dynamic<?> dynamic3 = optional3.get();
+                                    Dynamic<?> dynamic3 = (Dynamic<?>)optional3.get();
                                     Dynamic<?> dynamic4 = dynamic3.createMap(
                                         ImmutableMap.of(dynamic3.createString("Lore"), dynamic3.createList(Stream.of(dynamic3.createString("(+NBT"))))
                                     );
