--- a/net/minecraft/util/datafix/fixes/LevelUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/LevelUUIDFix.java
@@ -7,7 +_,6 @@
 import com.mojang.datafixers.util.Pair;
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Dynamic;
-import java.util.function.Supplier;
 import org.slf4j.Logger;
 
 public class LevelUUIDFix extends AbstractUUIDFix {
@@ -53,11 +_,10 @@
                             p_145500_ -> p_145500_.update(
                                     "Players",
                                     p_145494_ -> p_145500_.createList(
-                                            p_145494_.asStream()
-                                                .map(p_145502_ -> createUUIDFromML(p_145502_).orElseGet((Supplier<? extends Dynamic<?>>)(() -> {
-                                                        LOGGER.warn("CustomBossEvents contains invalid UUIDs.");
-                                                        return p_145502_;
-                                                    })))
+                                            p_145494_.asStream().map(p_145502_ -> createUUIDFromML((Dynamic<?>)p_145502_).orElseGet(() -> {
+                                                    LOGGER.warn("CustomBossEvents contains invalid UUIDs.");
+                                                    return p_145502_;
+                                                }))
                                         )
                                 )
                         )
