--- a/net/minecraft/util/datafix/fixes/LevelLegacyWorldGenSettingsFix.java
+++ b/net/minecraft/util/datafix/fixes/LevelLegacyWorldGenSettingsFix.java
@@ -27,11 +_,11 @@
             p_307516_ -> p_307516_.update(DSL.remainderFinder(), p_307487_ -> {
                     Dynamic<?> dynamic = p_307487_.get("WorldGenSettings").orElseEmptyMap();
     
-                    for(String s : OLD_SETTINGS_KEYS) {
+                    for (String s : OLD_SETTINGS_KEYS) {
                         Optional<? extends Dynamic<?>> optional = p_307487_.get(s).result();
                         if (optional.isPresent()) {
                             p_307487_ = p_307487_.remove(s);
-                            dynamic = dynamic.set(s, optional.get());
+                            dynamic = dynamic.set(s, (Dynamic<?>)optional.get());
                         }
                     }
     
