--- a/net/minecraft/util/datafix/fixes/OminousBannerRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/OminousBannerRenameFix.java
@@ -13,7 +_,7 @@
     protected <T> Dynamic<T> fixItemStackTag(Dynamic<T> p_216696_) {
         Optional<? extends Dynamic<?>> optional = p_216696_.get("display").result();
         if (optional.isPresent()) {
-            Dynamic<?> dynamic = optional.get();
+            Dynamic<?> dynamic = (Dynamic<?>)optional.get();
             Optional<String> optional1 = dynamic.get("Name").asString().result();
             if (optional1.isPresent()) {
                 String s = optional1.get();
