--- a/net/minecraft/util/datafix/fixes/OptionsRenameFieldFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsRenameFieldFix.java
@@ -28,7 +_,8 @@
             p_16676_ -> p_16676_.update(
                     DSL.remainderFinder(),
                     p_145592_ -> DataFixUtils.orElse(
-                            p_145592_.get(this.fieldFrom).result().map(p_145595_ -> p_145592_.set(this.fieldTo, p_145595_).remove(this.fieldFrom)), p_145592_
+                            p_145592_.get(this.fieldFrom).result().map(p_145595_ -> p_145592_.set(this.fieldTo, (Dynamic<?>)p_145595_).remove(this.fieldFrom)),
+                            p_145592_
                         )
                 )
         );
