--- a/net/minecraft/util/datafix/fixes/PoiTypeRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/PoiTypeRenameFix.java
@@ -18,7 +_,7 @@
     protected <T> Stream<Dynamic<T>> processRecords(Stream<Dynamic<T>> p_216716_) {
         return p_216716_.map(
             p_216714_ -> p_216714_.update(
-                    "type", p_216718_ -> DataFixUtils.orElse(p_216718_.asString().map(this.renamer).map(p_216718_::createString).result(), p_216718_)
+                    "type", p_337661_ -> DataFixUtils.orElse(p_337661_.asString().map(this.renamer).map(p_337661_::createString).result(), p_337661_)
                 )
         );
     }
