--- a/net/minecraft/util/datafix/fixes/SavedDataUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/SavedDataUUIDFix.java
@@ -6,7 +_,6 @@
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Dynamic;
-import java.util.function.Supplier;
 import org.slf4j.Logger;
 
 public class SavedDataUUIDFix extends AbstractUUIDFix {
@@ -35,11 +_,11 @@
                                                             p_145680_ -> p_145680_.createList(
                                                                     p_145680_.asStream()
                                                                         .map(
-                                                                            p_145682_ -> createUUIDFromLongs(p_145682_, "UUIDMost", "UUIDLeast")
-                                                                                    .orElseGet((Supplier<? extends Dynamic<?>>)(() -> {
+                                                                            p_145682_ -> createUUIDFromLongs((Dynamic<?>)p_145682_, "UUIDMost", "UUIDLeast")
+                                                                                    .orElseGet(() -> {
                                                                                         LOGGER.warn("HeroesOfTheVillage contained invalid UUIDs.");
                                                                                         return p_145682_;
-                                                                                    }))
+                                                                                    })
                                                                         )
                                                                 )
                                                         )
