--- a/net/minecraft/util/datafix/fixes/SavedDataFeaturePoolElementFix.java
+++ b/net/minecraft/util/datafix/fixes/SavedDataFeaturePoolElementFix.java
@@ -47,7 +_,7 @@
 
     private static Stream<? extends Dynamic<?>> updateChildren(Stream<? extends Dynamic<?>> p_145661_) {
         return p_145661_.map(
-            (Function<? super Dynamic<?>, ? extends Dynamic<?>>)(p_145667_ -> {
+            p_145667_ -> {
                 String s = p_145667_.get("id").asString("");
                 if (!PIECE_TYPE.contains(s)) {
                     return p_145667_;
@@ -57,7 +_,7 @@
                         ? p_145667_
                         : p_145667_.update("pool_element", p_145669_ -> p_145669_.update("feature", SavedDataFeaturePoolElementFix::fixFeature));
                 }
-            })
+            }
         );
     }
 
@@ -67,14 +_,14 @@
         } else {
             OptionalDynamic<T> optionaldynamic = p_145650_.get(p_145651_[0]);
 
-            for(int i = 1; i < p_145651_.length; ++i) {
+            for (int i = 1; i < p_145651_.length; i++) {
                 String s = p_145651_[i];
                 Matcher matcher = INDEX_PATTERN.matcher(s);
                 if (matcher.matches()) {
                     int j = Integer.parseInt(matcher.group(1));
                     List<? extends Dynamic<T>> list = optionaldynamic.asList(Function.identity());
                     if (j >= 0 && j < list.size()) {
-                        optionaldynamic = new OptionalDynamic<>(p_145650_.getOps(), DataResult.success(list.get(j)));
+                        optionaldynamic = new OptionalDynamic<>(p_145650_.getOps(), DataResult.success((Dynamic<T>)list.get(j)));
                     } else {
                         optionaldynamic = new OptionalDynamic<>(p_145650_.getOps(), DataResult.error(() -> "Missing id:" + j));
                     }
