--- a/net/minecraft/util/datafix/fixes/ScoreboardDisplaySlotFix.java
+++ b/net/minecraft/util/datafix/fixes/ScoreboardDisplaySlotFix.java
@@ -61,9 +_,9 @@
                                     "DisplaySlots",
                                     p_295769_ -> p_295769_.updateMapValues(
                                             p_295633_ -> p_295633_.mapFirst(
-                                                    p_294211_ -> DataFixUtils.orElse(
-                                                            p_294211_.asString().result().map(ScoreboardDisplaySlotFix::rename).map(p_294211_::createString),
-                                                            p_294211_
+                                                    p_337665_ -> DataFixUtils.orElse(
+                                                            p_337665_.asString().result().map(ScoreboardDisplaySlotFix::rename).map(p_337665_::createString),
+                                                            p_337665_
                                                         )
                                                 )
                                         )
