--- a/net/minecraft/util/datafix/fixes/StatsCounterFix.java
+++ b/net/minecraft/util/datafix/fixes/StatsCounterFix.java
@@ -242,14 +_,14 @@
             Map<Dynamic<?>, Dynamic<?>> map = Maps.newHashMap();
             Optional<? extends Map<? extends Dynamic<?>, ? extends Dynamic<?>>> optional = dynamic.getMapValues().result();
             if (optional.isPresent()) {
-                for(Entry<? extends Dynamic<?>, ? extends Dynamic<?>> entry : optional.get().entrySet()) {
+                for (Entry<? extends Dynamic<?>, ? extends Dynamic<?>> entry : optional.get().entrySet()) {
                     if (entry.getValue().asNumber().result().isPresent()) {
                         String s = entry.getKey().asString("");
                         StatsCounterFix.StatType statscounterfix$stattype = unpackLegacyKey(s);
                         if (statscounterfix$stattype != null) {
                             Dynamic<?> dynamic1 = dynamic.createString(statscounterfix$stattype.type());
                             Dynamic<?> dynamic2 = map.computeIfAbsent(dynamic1, p_145701_ -> dynamic.emptyMap());
-                            map.put(dynamic1, dynamic2.set(statscounterfix$stattype.typeKey(), entry.getValue()));
+                            map.put(dynamic1, dynamic2.set(statscounterfix$stattype.typeKey(), (Dynamic<?>)entry.getValue()));
                         }
                     }
                 }
@@ -270,13 +_,13 @@
                 Dynamic<?> dynamic = p_311567_.get(DSL.remainderFinder());
                 Dynamic<?> dynamic1 = dynamic.update(
                     "CriteriaName",
-                    p_293832_ -> DataFixUtils.orElse(
-                            p_293832_.asString()
+                    p_337666_ -> DataFixUtils.orElse(
+                            p_337666_.asString()
                                 .result()
                                 .map(
                                     p_293833_ -> {
                                         if (SPECIAL_OBJECTIVE_CRITERIA.contains(p_293833_)) {
-                                            return p_293833_;
+                                            return (String)p_293833_;
                                         } else {
                                             StatsCounterFix.StatType statscounterfix$stattype = unpackLegacyKey(p_293833_);
                                             return statscounterfix$stattype == null
@@ -287,8 +_,8 @@
                                         }
                                     }
                                 )
-                                .map(p_293832_::createString),
-                            p_293832_
+                                .map(p_337666_::createString),
+                            p_337666_
                         )
                 );
                 return Util.readTypedOrThrow(type1, dynamic1);
