--- a/net/minecraft/util/datafix/fixes/StructuresBecomeConfiguredFix.java
+++ b/net/minecraft/util/datafix/fixes/StructuresBecomeConfiguredFix.java
@@ -1,24 +_,29 @@
 package net.minecraft.util.datafix.fixes;
 
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 import com.google.common.collect.ImmutableMap.Builder;
 import com.mojang.datafixers.DataFix;
 import com.mojang.datafixers.TypeRewriteRule;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.datafixers.types.Type;
-import com.mojang.datafixers.util.Pair;
+import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Dynamic;
 import it.unimi.dsi.fastutil.objects.Object2IntArrayMap;
-import java.util.ArrayList;
 import java.util.Comparator;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Map.Entry;
 import java.util.function.Function;
+import java.util.stream.LongStream;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
 
 public class StructuresBecomeConfiguredFix extends DataFix {
+    private static final Logger LOGGER = LogUtils.getLogger();
     private static final Map<String, StructuresBecomeConfiguredFix.Conversion> CONVERSION_MAP = ImmutableMap.<String, StructuresBecomeConfiguredFix.Conversion>builder(
             
         )
@@ -134,63 +_,64 @@
     }
 
     private Dynamic<?> updateStarts(Dynamic<?> p_207700_, Dynamic<?> p_207701_) {
-        Map<? extends Dynamic<?>, ? extends Dynamic<?>> map = (Map)p_207700_.getMapValues().result().get();
-        List<Dynamic<?>> list = new ArrayList<>();
-        map.forEach((p_207721_, p_207722_) -> {
-            if (p_207722_.get("id").asString("INVALID").equals("INVALID")) {
-                list.add(p_207721_);
+        Map<? extends Dynamic<?>, ? extends Dynamic<?>> map = p_207700_.getMapValues().result().orElse(Map.of());
+        HashMap<Dynamic<?>, Dynamic<?>> hashmap = Maps.newHashMap();
+        map.forEach((p_339494_, p_339495_) -> {
+            if (!p_339495_.get("id").asString("INVALID").equals("INVALID")) {
+                Dynamic<?> dynamic = this.findUpdatedStructureType((Dynamic<?>)p_339494_, p_207701_);
+                if (dynamic == null) {
+                    LOGGER.warn("Encountered unknown structure in datafixer: " + p_339494_.asString("<missing key>"));
+                } else {
+                    hashmap.computeIfAbsent(dynamic, p_339502_ -> p_339495_.set("id", dynamic));
+                }
             }
         });
-
-        for(Dynamic<?> dynamic : list) {
-            p_207700_ = p_207700_.remove(dynamic.asString(""));
-        }
-
-        return p_207700_.updateMapValues(p_207715_ -> this.updateStart(p_207715_, p_207701_));
-    }
-
-    private Pair<Dynamic<?>, Dynamic<?>> updateStart(Pair<Dynamic<?>, Dynamic<?>> p_207685_, Dynamic<?> p_207686_) {
-        Dynamic<?> dynamic = this.findUpdatedStructureType(p_207685_, p_207686_);
-        return new Pair<>(dynamic, p_207685_.getSecond().set("id", dynamic));
+        return p_207701_.createMap(hashmap);
     }
 
     private Dynamic<?> updateReferences(Dynamic<?> p_207717_, Dynamic<?> p_207718_) {
-        Map<? extends Dynamic<?>, ? extends Dynamic<?>> map = (Map)p_207717_.getMapValues().result().get();
-        List<Dynamic<?>> list = new ArrayList<>();
-        map.forEach((p_207704_, p_207705_) -> {
-            if (p_207705_.asLongStream().count() == 0L) {
-                list.add(p_207704_);
+        Map<? extends Dynamic<?>, ? extends Dynamic<?>> map = p_207717_.getMapValues().result().orElse(Map.of());
+        HashMap<Dynamic<?>, Dynamic<?>> hashmap = Maps.newHashMap();
+        map.forEach(
+            (p_339498_, p_339499_) -> {
+                if (p_339499_.asLongStream().count() != 0L) {
+                    Dynamic<?> dynamic = this.findUpdatedStructureType((Dynamic<?>)p_339498_, p_207718_);
+                    if (dynamic == null) {
+                        LOGGER.warn("Encountered unknown structure in datafixer: " + p_339498_.asString("<missing key>"));
+                    } else {
+                        hashmap.compute(
+                            dynamic,
+                            (p_339504_, p_339505_) -> p_339505_ == null
+                                    ? p_339499_
+                                    : p_339499_.createLongList(LongStream.concat(p_339505_.asLongStream(), p_339499_.asLongStream()))
+                        );
+                    }
+                }
             }
-        });
-
-        for(Dynamic<?> dynamic : list) {
-            p_207717_ = p_207717_.remove(dynamic.asString(""));
-        }
-
-        return p_207717_.updateMapValues(p_207698_ -> this.updateReference(p_207698_, p_207718_));
-    }
-
-    private Pair<Dynamic<?>, Dynamic<?>> updateReference(Pair<Dynamic<?>, Dynamic<?>> p_207711_, Dynamic<?> p_207712_) {
-        return p_207711_.mapFirst(p_207690_ -> this.findUpdatedStructureType(p_207711_, p_207712_));
-    }
-
-    private Dynamic<?> findUpdatedStructureType(Pair<Dynamic<?>, Dynamic<?>> p_207724_, Dynamic<?> p_207725_) {
-        String s = p_207724_.getFirst().asString("UNKNOWN").toLowerCase(Locale.ROOT);
+        );
+        return p_207718_.createMap(hashmap);
+    }
+
+    @Nullable
+    private Dynamic<?> findUpdatedStructureType(Dynamic<?> p_207725_, Dynamic<?> p_339616_) {
+        String s = p_207725_.asString("UNKNOWN").toLowerCase(Locale.ROOT);
         StructuresBecomeConfiguredFix.Conversion structuresbecomeconfiguredfix$conversion = CONVERSION_MAP.get(s);
+        if (structuresbecomeconfiguredfix$conversion == null) structuresbecomeconfiguredfix$conversion = net.neoforged.neoforge.common.CommonHooks.getStructureConversion(s); // Neo: hook for mods to register conversions through RegisterStructureConversionsEvent
         if (structuresbecomeconfiguredfix$conversion == null) {
-            throw new IllegalStateException("Found unknown structure: " + s);
+            // Porting 1.20.5 check if this is correct
+            if (net.neoforged.neoforge.common.CommonHooks.checkStructureNamespace(s)) return p_339616_.createString(s); // Neo: pass-through structure IDs which have a non-"minecraft" namespace
+            if (true) return p_339616_.createString("unknown." + s); // Neo: Pass-through with "unknown." prefix, so deserializer logs and ignores rather than fixer throwing an exception and dropping chunk data
+            return null;
         } else {
-            Dynamic<?> dynamic = p_207724_.getSecond();
             String s1 = structuresbecomeconfiguredfix$conversion.fallback;
             if (!structuresbecomeconfiguredfix$conversion.biomeMapping().isEmpty()) {
-                Optional<String> optional = this.guessConfiguration(p_207725_, structuresbecomeconfiguredfix$conversion);
+                Optional<String> optional = this.guessConfiguration(p_339616_, structuresbecomeconfiguredfix$conversion);
                 if (optional.isPresent()) {
                     s1 = optional.get();
                 }
             }
 
-            Dynamic<?> dynamic1 = dynamic.createString(s1);
-            return dynamic1;
+            return p_339616_.createString(s1);
         }
     }
 
@@ -222,7 +_,7 @@
         private static Map<String, String> unpack(Map<List<String>, String> p_207749_) {
             Builder<String, String> builder = ImmutableMap.builder();
 
-            for(Entry<List<String>, String> entry : p_207749_.entrySet()) {
+            for (Entry<List<String>, String> entry : p_207749_.entrySet()) {
                 entry.getKey().forEach(p_207745_ -> builder.put(p_207745_, entry.getValue()));
             }
 
