--- a/net/minecraft/util/datafix/fixes/VariantRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/VariantRenameFix.java
@@ -22,8 +_,8 @@
             DSL.remainderFinder(),
             p_216750_ -> p_216750_.update(
                     "variant",
-                    p_216755_ -> DataFixUtils.orElse(
-                            p_216755_.asString().map(p_216753_ -> p_216755_.createString(this.renames.getOrDefault(p_216753_, p_216753_))).result(), p_216755_
+                    p_337668_ -> DataFixUtils.orElse(
+                            p_337668_.asString().map(p_216753_ -> p_337668_.createString(this.renames.getOrDefault(p_216753_, p_216753_))).result(), p_337668_
                         )
                 )
         );
