--- a/net/minecraft/server/commands/BanPlayerCommands.java
+++ b/net/minecraft/server/commands/BanPlayerCommands.java
@@ -44,13 +_,13 @@
         UserBanList userbanlist = p_136565_.getServer().getPlayerList().getBans();
         int i = 0;
 
-        for(GameProfile gameprofile : p_136566_) {
+        for (GameProfile gameprofile : p_136566_) {
             if (!userbanlist.isBanned(gameprofile)) {
                 UserBanListEntry userbanlistentry = new UserBanListEntry(
                     gameprofile, null, p_136565_.getTextName(), null, p_136567_ == null ? null : p_136567_.getString()
                 );
                 userbanlist.add(userbanlistentry);
-                ++i;
+                i++;
                 p_136565_.sendSuccess(
                     () -> Component.translatable("commands.ban.success", Component.literal(gameprofile.getName()), userbanlistentry.getReason()), true
                 );
