--- a/net/minecraft/server/commands/DebugMobSpawningCommand.java
+++ b/net/minecraft/server/commands/DebugMobSpawningCommand.java
@@ -15,7 +_,7 @@
         LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.literal("debugmobspawning")
             .requires(p_180113_ -> p_180113_.hasPermission(2));
 
-        for(MobCategory mobcategory : MobCategory.values()) {
+        for (MobCategory mobcategory : MobCategory.values()) {
             literalargumentbuilder.then(
                 Commands.literal(mobcategory.getName())
                     .then(
