--- a/net/minecraft/server/commands/DebugConfigCommand.java
+++ b/net/minecraft/server/commands/DebugConfigCommand.java
@@ -47,9 +_,8 @@
     private static Iterable<String> getUuidsInConfig(MinecraftServer p_294210_) {
         Set<String> set = new HashSet<>();
 
-        for(Connection connection : p_294210_.getConnection().getConnections()) {
-            PacketListener packetlistener = connection.getPacketListener();
-            if (packetlistener instanceof ServerConfigurationPacketListenerImpl serverconfigurationpacketlistenerimpl) {
+        for (Connection connection : p_294210_.getConnection().getConnections()) {
+            if (connection.getPacketListener() instanceof ServerConfigurationPacketListenerImpl serverconfigurationpacketlistenerimpl) {
                 set.add(serverconfigurationpacketlistenerimpl.getOwner().getId().toString());
             }
         }
@@ -65,11 +_,13 @@
     }
 
     private static int unconfig(CommandSourceStack p_294216_, UUID p_296468_) {
-        for(Connection connection : p_294216_.getServer().getConnection().getConnections()) {
+        for (Connection connection : p_294216_.getServer().getConnection().getConnections()) {
             PacketListener packetlistener = connection.getPacketListener();
-            if (packetlistener instanceof ServerConfigurationPacketListenerImpl serverconfigurationpacketlistenerimpl
-                && serverconfigurationpacketlistenerimpl.getOwner().getId().equals(p_296468_)) {
-                serverconfigurationpacketlistenerimpl.returnToWorld();
+            if (packetlistener instanceof ServerConfigurationPacketListenerImpl) {
+                ServerConfigurationPacketListenerImpl serverconfigurationpacketlistenerimpl = (ServerConfigurationPacketListenerImpl)packetlistener;
+                if (serverconfigurationpacketlistenerimpl.getOwner().getId().equals(p_296468_)) {
+                    serverconfigurationpacketlistenerimpl.returnToWorld();
+                }
             }
         }
 
