--- a/net/minecraft/server/commands/DifficultyCommand.java
+++ b/net/minecraft/server/commands/DifficultyCommand.java
@@ -19,13 +_,13 @@
     public static void register(CommandDispatcher<CommandSourceStack> p_136939_) {
         LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.literal("difficulty");
 
-        for(Difficulty difficulty : Difficulty.values()) {
+        for (Difficulty difficulty : Difficulty.values()) {
             literalargumentbuilder.then(Commands.literal(difficulty.getKey()).executes(p_136937_ -> setDifficulty(p_136937_.getSource(), difficulty)));
         }
 
-        p_136939_.register(literalargumentbuilder.requires(p_136943_ -> p_136943_.hasPermission(2)).executes(p_339779_ -> {
-            Difficulty difficulty1 = p_339779_.getSource().getLevel().getDifficulty();
-            p_339779_.getSource().sendSuccess(() -> Component.translatable("commands.difficulty.query", difficulty1.getDisplayName()), false);
+        p_136939_.register(literalargumentbuilder.requires(p_136943_ -> p_136943_.hasPermission(2)).executes(p_329887_ -> {
+            Difficulty difficulty1 = p_329887_.getSource().getLevel().getDifficulty();
+            p_329887_.getSource().sendSuccess(() -> Component.translatable("commands.difficulty.query", difficulty1.getDisplayName()), false);
             return difficulty1.getId();
         }));
     }
