--- a/net/minecraft/server/commands/EnchantCommand.java
+++ b/net/minecraft/server/commands/EnchantCommand.java
@@ -76,14 +_,15 @@
         } else {
             int i = 0;
 
-            for(Entity entity : p_248848_) {
-                if (entity instanceof LivingEntity livingentity) {
+            for (Entity entity : p_248848_) {
+                if (entity instanceof LivingEntity) {
+                    LivingEntity livingentity = (LivingEntity)entity;
                     ItemStack itemstack = livingentity.getMainHandItem();
                     if (!itemstack.isEmpty()) {
                         if (enchantment.canEnchant(itemstack)
                             && EnchantmentHelper.isEnchantmentCompatible(EnchantmentHelper.getEnchantmentsForCrafting(itemstack).keySet(), enchantment)) {
                             itemstack.enchant(enchantment, p_249941_);
-                            ++i;
+                            i++;
                         } else if (p_248848_.size() == 1) {
                             throw ERROR_INCOMPATIBLE.create(itemstack.getItem().getName(itemstack).getString());
                         }
