--- a/net/minecraft/server/commands/EnchantCommand.java
+++ b/net/minecraft/server/commands/EnchantCommand.java
@@ -80,7 +_,7 @@
                 if (entity instanceof LivingEntity livingentity) {
                     ItemStack itemstack = livingentity.getMainHandItem();
                     if (!itemstack.isEmpty()) {
-                        if (enchantment.canEnchant(itemstack)
+                        if (itemstack.supportsEnchantment(enchantmentHolder) // Neo: Respect IItemExtension#supportsEnchantment
                             && EnchantmentHelper.isEnchantmentCompatible(EnchantmentHelper.getEnchantmentsForCrafting(itemstack).keySet(), enchantmentHolder)) {
                             itemstack.enchant(enchantmentHolder, level);
                             i++;
