--- a/net/minecraft/server/commands/ExperienceCommand.java
+++ b/net/minecraft/server/commands/ExperienceCommand.java
@@ -140,7 +_,7 @@
     private static int addExperience(
         CommandSourceStack p_137317_, Collection<? extends ServerPlayer> p_137318_, int p_137319_, ExperienceCommand.Type p_137320_
     ) {
-        for(ServerPlayer serverplayer : p_137318_) {
+        for (ServerPlayer serverplayer : p_137318_) {
             p_137320_.add.accept(serverplayer, p_137319_);
         }
 
@@ -165,9 +_,9 @@
     ) throws CommandSyntaxException {
         int i = 0;
 
-        for(ServerPlayer serverplayer : p_137327_) {
+        for (ServerPlayer serverplayer : p_137327_) {
             if (p_137329_.set.test(serverplayer, p_137328_)) {
-                ++i;
+                i++;
             }
         }
 
@@ -192,14 +_,14 @@
     }
 
     static enum Type {
-        POINTS("points", Player::giveExperiencePoints, (p_339790_, p_339791_) -> {
-            if (p_339791_ >= p_339790_.getXpNeededForNextLevel()) {
+        POINTS("points", Player::giveExperiencePoints, (p_337532_, p_337533_) -> {
+            if (p_337533_ >= p_337532_.getXpNeededForNextLevel()) {
                 return false;
             } else {
-                p_339790_.setExperiencePoints(p_339791_);
+                p_337532_.setExperiencePoints(p_337533_);
                 return true;
             }
-        }, p_339789_ -> Mth.floor(p_339789_.experienceProgress * (float)p_339789_.getXpNeededForNextLevel())),
+        }, p_337531_ -> Mth.floor(p_337531_.experienceProgress * (float)p_337531_.getXpNeededForNextLevel())),
         LEVELS("levels", ServerPlayer::giveExperienceLevels, (p_137360_, p_137361_) -> {
             p_137360_.setExperienceLevels(p_137361_);
             return true;
