--- a/net/minecraft/server/commands/FunctionCommand.java
+++ b/net/minecraft/server/commands/FunctionCommand.java
@@ -66,7 +_,7 @@
     public static void register(CommandDispatcher<CommandSourceStack> p_137715_) {
         LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.literal("with");
 
-        for(DataCommands.DataProvider datacommands$dataprovider : DataCommands.SOURCE_PROVIDERS) {
+        for (DataCommands.DataProvider datacommands$dataprovider : DataCommands.SOURCE_PROVIDERS) {
             datacommands$dataprovider.wrap(literalargumentbuilder, p_305702_ -> p_305702_.executes(new FunctionCommand.FunctionCustomExecutor() {
                     @Override
                     protected CompoundTag arguments(CommandContext<CommandSourceStack> p_306295_) throws CommandSyntaxException {
@@ -168,7 +_,7 @@
         T t = p_309547_.clearCallbacks();
         CommandResultCallback commandresultcallback = CommandResultCallback.chain(p_309655_.callback(), p_309663_.currentFrame().returnValueConsumer());
 
-        for(CommandFunction<T> commandfunction : p_309534_) {
+        for (CommandFunction<T> commandfunction : p_309534_) {
             ResourceLocation resourcelocation = commandfunction.id();
             CommandResultCallback commandresultcallback1 = decorateOutputIfNeeded(p_309655_, p_309585_, resourcelocation, commandresultcallback);
             instantiateAndQueueFunctions(p_309696_, p_309663_, commanddispatcher, t, commandfunction, resourcelocation, commandresultcallback1, true);
@@ -195,7 +_,7 @@
                 CommandResultCallback commandresultcallback1 = decorateOutputIfNeeded(p_309693_, p_309570_, resourcelocation, commandresultcallback);
                 instantiateAndQueueFunctions(p_309637_, p_309574_, commanddispatcher, t, commandfunction, resourcelocation, commandresultcallback1, false);
             } else if (commandresultcallback == CommandResultCallback.EMPTY) {
-                for(CommandFunction<T> commandfunction1 : p_309573_) {
+                for (CommandFunction<T> commandfunction1 : p_309573_) {
                     ResourceLocation resourcelocation2 = commandfunction1.id();
                     CommandResultCallback commandresultcallback2 = decorateOutputIfNeeded(p_309693_, p_309570_, resourcelocation2, commandresultcallback);
                     instantiateAndQueueFunctions(p_309637_, p_309574_, commanddispatcher, t, commandfunction1, resourcelocation2, commandresultcallback2, false);
@@ -214,7 +_,7 @@
                 Accumulator functioncommand$1accumulator = new Accumulator();
                 CommandResultCallback commandresultcallback4 = (p_309467_, p_309468_) -> functioncommand$1accumulator.add(p_309468_);
 
-                for(CommandFunction<T> commandfunction2 : p_309573_) {
+                for (CommandFunction<T> commandfunction2 : p_309573_) {
                     ResourceLocation resourcelocation1 = commandfunction2.id();
                     CommandResultCallback commandresultcallback3 = decorateOutputIfNeeded(p_309693_, p_309570_, resourcelocation1, commandresultcallback4);
                     instantiateAndQueueFunctions(p_309637_, p_309574_, commanddispatcher, t, commandfunction2, resourcelocation1, commandresultcallback3, false);
