--- a/net/minecraft/server/commands/GameModeCommand.java
+++ b/net/minecraft/server/commands/GameModeCommand.java
@@ -57,10 +_,10 @@
     private static int setMode(CommandContext<CommandSourceStack> p_137732_, Collection<ServerPlayer> p_137733_, GameType p_137734_) {
         int i = 0;
 
-        for(ServerPlayer serverplayer : p_137733_) {
+        for (ServerPlayer serverplayer : p_137733_) {
             if (serverplayer.setGameMode(p_137734_)) {
                 logGamemodeChange(p_137732_.getSource(), serverplayer, p_137734_);
-                ++i;
+                i++;
             }
         }
 
