--- a/net/minecraft/server/commands/HelpCommand.java
+++ b/net/minecraft/server/commands/HelpCommand.java
@@ -21,7 +_,7 @@
                 .executes(p_288460_ -> {
                     Map<CommandNode<CommandSourceStack>, String> map = p_137788_.getSmartUsage(p_137788_.getRoot(), p_288460_.getSource());
         
-                    for(String s : map.values()) {
+                    for (String s : map.values()) {
                         p_288460_.getSource().sendSuccess(() -> Component.literal("/" + s), false);
                     }
         
@@ -41,7 +_,7 @@
                                         Iterables.getLast(parseresults.getContext().getNodes()).getNode(), p_288458_.getSource()
                                     );
                     
-                                    for(String s : map.values()) {
+                                    for (String s : map.values()) {
                                         p_288458_.getSource().sendSuccess(() -> Component.literal("/" + parseresults.getReader().getString() + " " + s), false);
                                     }
                     
