--- a/net/minecraft/server/commands/LootCommand.java
+++ b/net/minecraft/server/commands/LootCommand.java
@@ -291,7 +_,7 @@
         Container container = getContainer(p_137961_, p_137962_);
         List<ItemStack> list = Lists.newArrayListWithCapacity(p_137963_.size());
 
-        for(ItemStack itemstack : p_137963_) {
+        for (ItemStack itemstack : p_137963_) {
             if (distributeToContainer(container, itemstack.copy())) {
                 container.setChanged();
                 list.add(itemstack);
@@ -305,7 +_,7 @@
     private static boolean distributeToContainer(Container p_137886_, ItemStack p_137887_) {
         boolean flag = false;
 
-        for(int i = 0; i < p_137886_.getContainerSize() && !p_137887_.isEmpty(); ++i) {
+        for (int i = 0; i < p_137886_.getContainerSize() && !p_137887_.isEmpty(); i++) {
             ItemStack itemstack = p_137886_.getItem(i);
             if (p_137886_.canPlaceItem(i, p_137887_)) {
                 if (itemstack.isEmpty()) {
@@ -335,7 +_,7 @@
         if (p_137956_ >= 0 && p_137956_ < i) {
             List<ItemStack> list = Lists.newArrayListWithCapacity(p_137958_.size());
 
-            for(int j = 0; j < p_137957_; ++j) {
+            for (int j = 0; j < p_137957_; j++) {
                 int k = p_137956_ + j;
                 ItemStack itemstack = j < p_137958_.size() ? p_137958_.get(j) : ItemStack.EMPTY;
                 if (container.canPlaceItem(k, itemstack)) {
@@ -358,8 +_,8 @@
     private static int playerGive(Collection<ServerPlayer> p_137985_, List<ItemStack> p_137986_, LootCommand.Callback p_137987_) throws CommandSyntaxException {
         List<ItemStack> list = Lists.newArrayListWithCapacity(p_137986_.size());
 
-        for(ItemStack itemstack : p_137986_) {
-            for(ServerPlayer serverplayer : p_137985_) {
+        for (ItemStack itemstack : p_137986_) {
+            for (ServerPlayer serverplayer : p_137985_) {
                 if (serverplayer.getInventory().add(itemstack.copy())) {
                     list.add(itemstack);
                 }
@@ -371,7 +_,7 @@
     }
 
     private static void setSlots(Entity p_137889_, List<ItemStack> p_137890_, int p_137891_, int p_137892_, List<ItemStack> p_137893_) {
-        for(int i = 0; i < p_137892_; ++i) {
+        for (int i = 0; i < p_137892_; i++) {
             ItemStack itemstack = i < p_137890_.size() ? p_137890_.get(i) : ItemStack.EMPTY;
             SlotAccess slotaccess = p_137889_.getSlot(p_137891_ + i);
             if (slotaccess != SlotAccess.NULL && slotaccess.set(itemstack.copy())) {
@@ -385,7 +_,7 @@
     ) throws CommandSyntaxException {
         List<ItemStack> list = Lists.newArrayListWithCapacity(p_137982_.size());
 
-        for(Entity entity : p_137979_) {
+        for (Entity entity : p_137979_) {
             if (entity instanceof ServerPlayer serverplayer) {
                 setSlots(entity, p_137982_, p_137980_, p_137981_, list);
                 serverplayer.containerMenu.broadcastChanges();
