--- a/net/minecraft/server/commands/PardonCommand.java
+++ b/net/minecraft/server/commands/PardonCommand.java
@@ -37,10 +_,10 @@
         UserBanList userbanlist = p_138103_.getServer().getPlayerList().getBans();
         int i = 0;
 
-        for(GameProfile gameprofile : p_138104_) {
+        for (GameProfile gameprofile : p_138104_) {
             if (userbanlist.isBanned(gameprofile)) {
                 userbanlist.remove(gameprofile);
-                ++i;
+                i++;
                 p_138103_.sendSuccess(() -> Component.translatable("commands.pardon.success", Component.literal(gameprofile.getName())), true);
             }
         }
