--- a/net/minecraft/server/commands/ParticleCommand.java
+++ b/net/minecraft/server/commands/ParticleCommand.java
@@ -152,7 +_,7 @@
     ) throws CommandSyntaxException {
         int i = 0;
 
-        for(ServerPlayer serverplayer : p_138136_) {
+        for (ServerPlayer serverplayer : p_138136_) {
             if (p_138129_.getLevel()
                 .sendParticles(
                     serverplayer,
@@ -167,7 +_,7 @@
                     p_138132_.z,
                     (double)p_138133_
                 )) {
-                ++i;
+                i++;
             }
         }
 
