--- a/net/minecraft/server/commands/PlaySoundCommand.java
+++ b/net/minecraft/server/commands/PlaySoundCommand.java
@@ -45,7 +_,7 @@
                     )
             );
 
-        for(SoundSource soundsource : SoundSource.values()) {
+        for (SoundSource soundsource : SoundSource.values()) {
             requiredargumentbuilder.then(source(soundsource));
         }
 
@@ -162,7 +_,7 @@
         int i = 0;
         long j = p_138161_.getLevel().getRandom().nextLong();
 
-        for(ServerPlayer serverplayer : p_138162_) {
+        for (ServerPlayer serverplayer : p_138162_) {
             double d1 = p_138165_.x - serverplayer.getX();
             double d2 = p_138165_.y - serverplayer.getY();
             double d3 = p_138165_.z - serverplayer.getZ();
@@ -180,7 +_,7 @@
             }
 
             serverplayer.connection.send(new ClientboundSoundPacket(holder, p_138164_, vec3.x(), vec3.y(), vec3.z(), f, p_138167_, j));
-            ++i;
+            i++;
         }
 
         if (i == 0) {
