--- a/net/minecraft/server/commands/RaidCommand.java
+++ b/net/minecraft/server/commands/RaidCommand.java
@@ -52,7 +_,7 @@
                     Commands.literal("setomen")
                         .then(
                             Commands.argument("level", IntegerArgumentType.integer(0))
-                                .executes(p_180481_ -> setBadOmenLevel(p_180481_.getSource(), IntegerArgumentType.getInteger(p_180481_, "level")))
+                                .executes(p_337538_ -> setRaidOmenLevel(p_337538_.getSource(), IntegerArgumentType.getInteger(p_337538_, "level")))
                         )
                 )
                 .then(Commands.literal("glow").executes(p_180471_ -> glow(p_180471_.getSource())))
@@ -62,7 +_,7 @@
     private static int glow(CommandSourceStack p_180473_) throws CommandSyntaxException {
         Raid raid = getRaid(p_180473_.getPlayerOrException());
         if (raid != null) {
-            for(Raider raider : raid.getAllRaiders()) {
+            for (Raider raider : raid.getAllRaiders()) {
                 raider.addEffect(new MobEffectInstance(MobEffects.GLOWING, 1000, 1));
             }
         }
@@ -70,16 +_,16 @@
         return 1;
     }
 
-    private static int setBadOmenLevel(CommandSourceStack p_180475_, int p_180476_) throws CommandSyntaxException {
+    private static int setRaidOmenLevel(CommandSourceStack p_180475_, int p_180476_) throws CommandSyntaxException {
         Raid raid = getRaid(p_180475_.getPlayerOrException());
         if (raid != null) {
-            int i = raid.getMaxBadOmenLevel();
+            int i = raid.getMaxRaidOmenLevel();
             if (p_180476_ > i) {
-                p_180475_.sendFailure(Component.literal("Sorry, the max bad omen level you can set is " + i));
+                p_180475_.sendFailure(Component.literal("Sorry, the max raid omen level you can set is " + i));
             } else {
-                int j = raid.getBadOmenLevel();
-                raid.setBadOmenLevel(p_180476_);
-                p_180475_.sendSuccess(() -> Component.literal("Changed village's bad omen level from " + j + " to " + p_180476_), false);
+                int j = raid.getRaidOmenLevel();
+                raid.setRaidOmenLevel(p_180476_);
+                p_180475_.sendSuccess(() -> Component.literal("Changed village's raid omen level from " + j + " to " + p_180476_), false);
             }
         } else {
             p_180475_.sendFailure(Component.literal("No raid found here"));
@@ -124,9 +_,9 @@
             return -1;
         } else {
             Raids raids = serverplayer.serverLevel().getRaids();
-            Raid raid = raids.createOrExtendRaid(serverplayer);
+            Raid raid = raids.createOrExtendRaid(serverplayer, serverplayer.blockPosition());
             if (raid != null) {
-                raid.setBadOmenLevel(p_180486_);
+                raid.setRaidOmenLevel(p_180486_);
                 raids.setDirty();
                 p_180485_.sendSuccess(() -> Component.literal("Created a raid in your local village"), false);
             } else {
@@ -160,8 +_,8 @@
             StringBuilder stringbuilder1 = new StringBuilder();
             stringbuilder1.append("Num groups spawned: ");
             stringbuilder1.append(raid.getGroupsSpawned());
-            stringbuilder1.append(" Bad omen level: ");
-            stringbuilder1.append(raid.getBadOmenLevel());
+            stringbuilder1.append(" Raid omen level: ");
+            stringbuilder1.append(raid.getRaidOmenLevel());
             stringbuilder1.append(" Num mobs: ");
             stringbuilder1.append(raid.getTotalRaidersAlive());
             stringbuilder1.append(" Raid health: ");
