--- a/net/minecraft/server/commands/ReloadCommand.java
+++ b/net/minecraft/server/commands/ReloadCommand.java
@@ -29,7 +_,7 @@
         Collection<String> collection = Lists.newArrayList(p_138225_);
         Collection<String> collection1 = p_138224_.getDataConfiguration().dataPacks().getDisabled();
 
-        for(String s : p_138223_.getAvailableIds()) {
+        for (String s : p_138223_.getAvailableIds()) {
             if (!collection1.contains(s) && !collection.contains(s)) {
                 collection.add(s);
             }
