--- a/net/minecraft/server/commands/ResetChunksCommand.java
+++ b/net/minecraft/server/commands/ResetChunksCommand.java
@@ -66,12 +_,12 @@
         int k = chunkpos.x - p_183686_;
         int l = chunkpos.x + p_183686_;
 
-        for(int i1 = i; i1 <= j; ++i1) {
-            for(int j1 = k; j1 <= l; ++j1) {
+        for (int i1 = i; i1 <= j; i1++) {
+            for (int j1 = k; j1 <= l; j1++) {
                 ChunkPos chunkpos1 = new ChunkPos(j1, i1);
                 LevelChunk levelchunk = serverchunkcache.getChunk(j1, i1, false);
                 if (levelchunk != null && (!p_183687_ || !levelchunk.isOldNoiseGeneration())) {
-                    for(BlockPos blockpos : BlockPos.betweenClosed(
+                    for (BlockPos blockpos : BlockPos.betweenClosed(
                         chunkpos1.getMinBlockX(),
                         serverlevel.getMinBuildHeight(),
                         chunkpos1.getMinBlockZ(),
@@ -89,7 +_,7 @@
         long j3 = System.currentTimeMillis();
         int k3 = (p_183686_ * 2 + 1) * (p_183686_ * 2 + 1);
 
-        for(ChunkStatus chunkstatus : ImmutableList.of(
+        for (ChunkStatus chunkstatus : ImmutableList.of(
             ChunkStatus.BIOMES, ChunkStatus.NOISE, ChunkStatus.SURFACE, ChunkStatus.CARVERS, ChunkStatus.FEATURES, ChunkStatus.INITIALIZE_LIGHT
         )) {
             long k1 = System.currentTimeMillis();
@@ -98,16 +_,16 @@
                 serverlevel, serverchunkcache.getGenerator(), serverlevel.getStructureManager(), serverchunkcache.getLightEngine()
             );
 
-            for(int i2 = chunkpos.z - p_183686_; i2 <= chunkpos.z + p_183686_; ++i2) {
-                for(int j2 = chunkpos.x - p_183686_; j2 <= chunkpos.x + p_183686_; ++j2) {
+            for (int i2 = chunkpos.z - p_183686_; i2 <= chunkpos.z + p_183686_; i2++) {
+                for (int j2 = chunkpos.x - p_183686_; j2 <= chunkpos.x + p_183686_; j2++) {
                     ChunkPos chunkpos2 = new ChunkPos(j2, i2);
                     LevelChunk levelchunk1 = serverchunkcache.getChunk(j2, i2, false);
                     if (levelchunk1 != null && (!p_183687_ || !levelchunk1.isOldNoiseGeneration())) {
                         List<ChunkAccess> list = Lists.newArrayList();
                         int k2 = Math.max(1, chunkstatus.getRange());
 
-                        for(int l2 = chunkpos2.z - k2; l2 <= chunkpos2.z + k2; ++l2) {
-                            for(int i3 = chunkpos2.x - k2; i3 <= chunkpos2.x + k2; ++i3) {
+                        for (int l2 = chunkpos2.z - k2; l2 <= chunkpos2.z + k2; l2++) {
+                            for (int i3 = chunkpos2.x - k2; i3 <= chunkpos2.x + k2; i3++) {
                                 ChunkAccess chunkaccess = serverchunkcache.getChunk(i3, l2, chunkstatus.getParent(), true);
                                 ChunkAccess chunkaccess1;
                                 if (chunkaccess instanceof ImposterProtoChunk) {
@@ -143,12 +_,12 @@
 
         long l3 = System.currentTimeMillis();
 
-        for(int i4 = chunkpos.z - p_183686_; i4 <= chunkpos.z + p_183686_; ++i4) {
-            for(int l1 = chunkpos.x - p_183686_; l1 <= chunkpos.x + p_183686_; ++l1) {
+        for (int i4 = chunkpos.z - p_183686_; i4 <= chunkpos.z + p_183686_; i4++) {
+            for (int l1 = chunkpos.x - p_183686_; l1 <= chunkpos.x + p_183686_; l1++) {
                 ChunkPos chunkpos3 = new ChunkPos(l1, i4);
                 LevelChunk levelchunk2 = serverchunkcache.getChunk(l1, i4, false);
                 if (levelchunk2 != null && (!p_183687_ || !levelchunk2.isOldNoiseGeneration())) {
-                    for(BlockPos blockpos1 : BlockPos.betweenClosed(
+                    for (BlockPos blockpos1 : BlockPos.betweenClosed(
                         chunkpos3.getMinBlockX(),
                         serverlevel.getMinBuildHeight(),
                         chunkpos3.getMinBlockZ(),
