--- a/net/minecraft/server/commands/SaveOffCommand.java
+++ b/net/minecraft/server/commands/SaveOffCommand.java
@@ -16,7 +_,7 @@
             CommandSourceStack commandsourcestack = p_138287_.getSource();
             boolean flag = false;
 
-            for(ServerLevel serverlevel : commandsourcestack.getServer().getAllLevels()) {
+            for (ServerLevel serverlevel : commandsourcestack.getServer().getAllLevels()) {
                 if (serverlevel != null && !serverlevel.noSave) {
                     serverlevel.noSave = true;
                     flag = true;
