--- a/net/minecraft/server/commands/SetSpawnCommand.java
+++ b/net/minecraft/server/commands/SetSpawnCommand.java
@@ -89,7 +_,7 @@
                     pos.getZ(),
                     f,
                     f1,
-                    s,
+                    source.getLevel().getDescription(), // Neo: Use dimension translation, if one exists
                     targets.iterator().next().getDisplayName()
                 ),
                 true
