--- a/net/minecraft/server/commands/SpawnArmorTrimsCommand.java
+++ b/net/minecraft/server/commands/SpawnArmorTrimsCommand.java
@@ -122,8 +_,8 @@
         int j = 0;
         int k = 0;
 
-        for(ArmorTrim armortrim : nonnulllist) {
-            for(ArmorMaterial armormaterial : registry2) {
+        for (ArmorTrim armortrim : nonnulllist) {
+            for (ArmorMaterial armormaterial : registry2) {
                 if (armormaterial != ArmorMaterials.LEATHER.value()) {
                     double d1 = (double)blockpos.getX() + 0.5 - (double)(j % registry1.size()) * 3.0;
                     double d2 = (double)blockpos.getY() + 0.5 + (double)(k % i) * 3.0;
@@ -132,23 +_,26 @@
                     armorstand.setYRot(180.0F);
                     armorstand.setNoGravity(true);
 
-                    for(EquipmentSlot equipmentslot : EquipmentSlot.values()) {
+                    for (EquipmentSlot equipmentslot : EquipmentSlot.values()) {
                         Item item = MATERIAL_AND_SLOT_TO_ITEM.get(Pair.of(armormaterial, equipmentslot));
                         if (item != null) {
                             ItemStack itemstack = new ItemStack(item);
                             itemstack.set(DataComponents.TRIM, armortrim);
                             armorstand.setItemSlot(equipmentslot, itemstack);
-                            if (item instanceof ArmorItem armoritem && armoritem.getMaterial().is(ArmorMaterials.TURTLE)) {
-                                armorstand.setCustomName(
-                                    armortrim.pattern()
-                                        .value()
-                                        .copyWithStyle(armortrim.material())
-                                        .copy()
-                                        .append(" ")
-                                        .append(armortrim.material().value().description())
-                                );
-                                armorstand.setCustomNameVisible(true);
-                                continue;
+                            if (item instanceof ArmorItem) {
+                                ArmorItem armoritem = (ArmorItem)item;
+                                if (armoritem.getMaterial().is(ArmorMaterials.TURTLE)) {
+                                    armorstand.setCustomName(
+                                        armortrim.pattern()
+                                            .value()
+                                            .copyWithStyle(armortrim.material())
+                                            .copy()
+                                            .append(" ")
+                                            .append(armortrim.material().value().description())
+                                    );
+                                    armorstand.setCustomNameVisible(true);
+                                    continue;
+                                }
                             }
 
                             armorstand.setInvisible(true);
@@ -156,11 +_,11 @@
                     }
 
                     level.addFreshEntity(armorstand);
-                    ++k;
+                    k++;
                 }
             }
 
-            ++j;
+            j++;
         }
 
         p_266993_.sendSuccess(() -> Component.literal("Armorstands with trimmed armor spawned around you"), true);
