--- a/net/minecraft/server/commands/SpreadPlayersCommand.java
+++ b/net/minecraft/server/commands/SpreadPlayersCommand.java
@@ -63,14 +_,14 @@
                                                 .then(
                                                     Commands.argument("targets", EntityArgument.entities())
                                                         .executes(
-                                                            p_339813_ -> spreadPlayers(
-                                                                    p_339813_.getSource(),
-                                                                    Vec2Argument.getVec2(p_339813_, "center"),
-                                                                    FloatArgumentType.getFloat(p_339813_, "spreadDistance"),
-                                                                    FloatArgumentType.getFloat(p_339813_, "maxRange"),
-                                                                    p_339813_.getSource().getLevel().getMaxBuildHeight(),
-                                                                    BoolArgumentType.getBool(p_339813_, "respectTeams"),
-                                                                    EntityArgument.getEntities(p_339813_, "targets")
+                                                            p_329902_ -> spreadPlayers(
+                                                                    p_329902_.getSource(),
+                                                                    Vec2Argument.getVec2(p_329902_, "center"),
+                                                                    FloatArgumentType.getFloat(p_329902_, "spreadDistance"),
+                                                                    FloatArgumentType.getFloat(p_329902_, "maxRange"),
+                                                                    p_329902_.getSource().getLevel().getMaxBuildHeight(),
+                                                                    BoolArgumentType.getBool(p_329902_, "respectTeams"),
+                                                                    EntityArgument.getEntities(p_329902_, "targets")
                                                                 )
                                                         )
                                                 )
@@ -145,7 +_,7 @@
     private static int getNumberOfTeams(Collection<? extends Entity> p_138728_) {
         Set<Team> set = Sets.newHashSet();
 
-        for(Entity entity : p_138728_) {
+        for (Entity entity : p_138728_) {
             if (entity instanceof Player) {
                 set.add(entity.getTeam());
             } else {
@@ -173,24 +_,26 @@
         double d0 = Float.MAX_VALUE;
 
         int i;
-        for(i = 0; i < 10000 && flag; ++i) {
+        for (i = 0; i < 10000 && flag; i++) {
             flag = false;
             d0 = Float.MAX_VALUE;
 
-            for(int j = 0; j < p_214750_.length; ++j) {
+            for (int j = 0; j < p_214750_.length; j++) {
                 SpreadPlayersCommand.Position spreadplayerscommand$position = p_214750_[j];
                 int k = 0;
                 SpreadPlayersCommand.Position spreadplayerscommand$position1 = new SpreadPlayersCommand.Position();
 
-                for(int l = 0; l < p_214750_.length; ++l) {
+                for (int l = 0; l < p_214750_.length; l++) {
                     if (j != l) {
                         SpreadPlayersCommand.Position spreadplayerscommand$position2 = p_214750_[l];
                         double d1 = spreadplayerscommand$position.dist(spreadplayerscommand$position2);
                         d0 = Math.min(d1, d0);
                         if (d1 < p_214742_) {
-                            ++k;
-                            spreadplayerscommand$position1.x += spreadplayerscommand$position2.x - spreadplayerscommand$position.x;
-                            spreadplayerscommand$position1.z += spreadplayerscommand$position2.z - spreadplayerscommand$position.z;
+                            k++;
+                            spreadplayerscommand$position1.x = spreadplayerscommand$position1.x
+                                + (spreadplayerscommand$position2.x - spreadplayerscommand$position.x);
+                            spreadplayerscommand$position1.z = spreadplayerscommand$position1.z
+                                + (spreadplayerscommand$position2.z - spreadplayerscommand$position.z);
                         }
                     }
                 }
@@ -215,7 +_,7 @@
             }
 
             if (!flag) {
-                for(SpreadPlayersCommand.Position spreadplayerscommand$position3 : p_214750_) {
+                for (SpreadPlayersCommand.Position spreadplayerscommand$position3 : p_214750_) {
                     if (!spreadplayerscommand$position3.isSafe(p_214743_, p_214749_)) {
                         spreadplayerscommand$position3.randomize(p_214744_, p_214745_, p_214746_, p_214747_, p_214748_);
                         flag = true;
@@ -244,7 +_,7 @@
         int i = 0;
         Map<Team, SpreadPlayersCommand.Position> map = Maps.newHashMap();
 
-        for(Entity entity : p_138730_) {
+        for (Entity entity : p_138730_) {
             SpreadPlayersCommand.Position spreadplayerscommand$position;
             if (p_138734_) {
                 Team team = entity instanceof Player ? entity.getTeam() : null;
@@ -257,18 +_,25 @@
                 spreadplayerscommand$position = p_138732_[i++];
             }
 
-            entity.teleportTo(
-                p_138731_,
-                (double)Mth.floor(spreadplayerscommand$position.x) + 0.5,
-                (double)spreadplayerscommand$position.getSpawnY(p_138731_, p_138733_),
-                (double)Mth.floor(spreadplayerscommand$position.z) + 0.5,
-                Set.of(),
-                entity.getYRot(),
-                entity.getXRot()
+            net.neoforged.neoforge.event.entity.EntityTeleportEvent.SpreadPlayersCommand event = net.neoforged.neoforge.event.EventHooks.onEntityTeleportSpreadPlayersCommand(entity,
+                      (double)Mth.floor(spreadplayerscommand$position.x) + 0.5,
+                      spreadplayerscommand$position.getSpawnY(p_138731_, p_138733_),
+                      (double)Mth.floor(spreadplayerscommand$position.z) + 0.5
             );
+            if (!event.isCanceled()) {
+                entity.teleportTo(
+                          p_138731_,
+                          event.getTargetX(),
+                          event.getTargetY(),
+                          event.getTargetZ(),
+                          Set.of(),
+                          entity.getYRot(),
+                          entity.getXRot()
+                );
+            }
             double d2 = Double.MAX_VALUE;
 
-            for(SpreadPlayersCommand.Position spreadplayerscommand$position1 : p_138732_) {
+            for (SpreadPlayersCommand.Position spreadplayerscommand$position1 : p_138732_) {
                 if (spreadplayerscommand$position != spreadplayerscommand$position1) {
                     double d1 = spreadplayerscommand$position.dist(spreadplayerscommand$position1);
                     d2 = Math.min(d1, d2);
@@ -286,7 +_,7 @@
     ) {
         SpreadPlayersCommand.Position[] aspreadplayerscommand$position = new SpreadPlayersCommand.Position[p_214735_];
 
-        for(int i = 0; i < aspreadplayerscommand$position.length; ++i) {
+        for (int i = 0; i < aspreadplayerscommand$position.length; i++) {
             SpreadPlayersCommand.Position spreadplayerscommand$position = new SpreadPlayersCommand.Position();
             spreadplayerscommand$position.randomize(p_214734_, p_214736_, p_214737_, p_214738_, p_214739_);
             aspreadplayerscommand$position[i] = spreadplayerscommand$position;
@@ -316,8 +_,8 @@
         }
 
         public void moveAway(SpreadPlayersCommand.Position p_138777_) {
-            this.x -= p_138777_.x;
-            this.z -= p_138777_.z;
+            this.x = this.x - p_138777_.x;
+            this.z = this.z - p_138777_.z;
         }
 
         public boolean clamp(double p_138754_, double p_138755_, double p_138756_, double p_138757_) {
@@ -345,19 +_,17 @@
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(this.x, (double)(p_138760_ + 1), this.z);
             boolean flag = p_138759_.getBlockState(blockpos$mutableblockpos).isAir();
             blockpos$mutableblockpos.move(Direction.DOWN);
+            boolean flag1 = p_138759_.getBlockState(blockpos$mutableblockpos).isAir();
 
-            boolean flag2;
-            for(boolean flag1 = p_138759_.getBlockState(blockpos$mutableblockpos).isAir();
-                blockpos$mutableblockpos.getY() > p_138759_.getMinBuildHeight();
-                flag1 = flag2
-            ) {
+            while (blockpos$mutableblockpos.getY() > p_138759_.getMinBuildHeight()) {
                 blockpos$mutableblockpos.move(Direction.DOWN);
-                flag2 = p_138759_.getBlockState(blockpos$mutableblockpos).isAir();
+                boolean flag2 = p_138759_.getBlockState(blockpos$mutableblockpos).isAir();
                 if (!flag2 && flag1 && flag) {
                     return blockpos$mutableblockpos.getY() + 1;
                 }
 
                 flag = flag1;
+                flag1 = flag2;
             }
 
             return p_138760_ + 1;
