--- a/net/minecraft/server/commands/StopSoundCommand.java
+++ b/net/minecraft/server/commands/StopSoundCommand.java
@@ -37,7 +_,7 @@
                     )
             );
 
-        for(SoundSource soundsource : SoundSource.values()) {
+        for (SoundSource soundsource : SoundSource.values()) {
             requiredargumentbuilder.then(
                 Commands.literal(soundsource.getName())
                     .executes(p_138807_ -> stopSound(p_138807_.getSource(), EntityArgument.getPlayers(p_138807_, "targets"), soundsource, null))
@@ -64,7 +_,7 @@
     ) {
         ClientboundStopSoundPacket clientboundstopsoundpacket = new ClientboundStopSoundPacket(p_138804_, p_138803_);
 
-        for(ServerPlayer serverplayer : p_138802_) {
+        for (ServerPlayer serverplayer : p_138802_) {
             serverplayer.connection.send(clientboundstopsoundpacket);
         }
 
