--- a/net/minecraft/server/commands/TagCommand.java
+++ b/net/minecraft/server/commands/TagCommand.java
@@ -66,7 +_,7 @@
     private static Collection<String> getTags(Collection<? extends Entity> p_138853_) {
         Set<String> set = Sets.newHashSet();
 
-        for(Entity entity : p_138853_) {
+        for (Entity entity : p_138853_) {
             set.addAll(entity.getTags());
         }
 
@@ -76,9 +_,9 @@
     private static int addTag(CommandSourceStack p_138849_, Collection<? extends Entity> p_138850_, String p_138851_) throws CommandSyntaxException {
         int i = 0;
 
-        for(Entity entity : p_138850_) {
+        for (Entity entity : p_138850_) {
             if (entity.addTag(p_138851_)) {
-                ++i;
+                i++;
             }
         }
 
@@ -100,9 +_,9 @@
     private static int removeTag(CommandSourceStack p_138857_, Collection<? extends Entity> p_138858_, String p_138859_) throws CommandSyntaxException {
         int i = 0;
 
-        for(Entity entity : p_138858_) {
+        for (Entity entity : p_138858_) {
             if (entity.removeTag(p_138859_)) {
-                ++i;
+                i++;
             }
         }
 
@@ -124,7 +_,7 @@
     private static int listTags(CommandSourceStack p_138846_, Collection<? extends Entity> p_138847_) {
         Set<String> set = Sets.newHashSet();
 
-        for(Entity entity : p_138847_) {
+        for (Entity entity : p_138847_) {
             set.addAll(entity.getTags());
         }
 
