--- a/net/minecraft/server/commands/TeamCommand.java
+++ b/net/minecraft/server/commands/TeamCommand.java
@@ -337,7 +_,7 @@
     private static int leaveTeam(CommandSourceStack p_138918_, Collection<ScoreHolder> p_138919_) {
         Scoreboard scoreboard = p_138918_.getServer().getScoreboard();
 
-        for(ScoreHolder scoreholder : p_138919_) {
+        for (ScoreHolder scoreholder : p_138919_) {
             scoreboard.removePlayerFromTeam(scoreholder.getScoreboardName());
         }
 
@@ -353,7 +_,7 @@
     private static int joinTeam(CommandSourceStack p_138895_, PlayerTeam p_138896_, Collection<ScoreHolder> p_138897_) {
         Scoreboard scoreboard = p_138895_.getServer().getScoreboard();
 
-        for(ScoreHolder scoreholder : p_138897_) {
+        for (ScoreHolder scoreholder : p_138897_) {
             scoreboard.addPlayerToTeam(scoreholder.getScoreboardName(), p_138896_);
         }
 
@@ -475,7 +_,7 @@
         if (collection.isEmpty()) {
             throw ERROR_TEAM_ALREADY_EMPTY.create();
         } else {
-            for(String s : collection) {
+            for (String s : collection) {
                 scoreboard.removePlayerFromTeam(s, p_138885_);
             }
 
