--- a/net/minecraft/server/commands/TitleCommand.java
+++ b/net/minecraft/server/commands/TitleCommand.java
@@ -108,7 +_,7 @@
     private static int clearTitle(CommandSourceStack p_139109_, Collection<ServerPlayer> p_139110_) {
         ClientboundClearTitlesPacket clientboundcleartitlespacket = new ClientboundClearTitlesPacket(false);
 
-        for(ServerPlayer serverplayer : p_139110_) {
+        for (ServerPlayer serverplayer : p_139110_) {
             serverplayer.connection.send(clientboundcleartitlespacket);
         }
 
@@ -124,7 +_,7 @@
     private static int resetTitle(CommandSourceStack p_139125_, Collection<ServerPlayer> p_139126_) {
         ClientboundClearTitlesPacket clientboundcleartitlespacket = new ClientboundClearTitlesPacket(true);
 
-        for(ServerPlayer serverplayer : p_139126_) {
+        for (ServerPlayer serverplayer : p_139126_) {
             serverplayer.connection.send(clientboundcleartitlespacket);
         }
 
@@ -140,7 +_,7 @@
     private static int showTitle(
         CommandSourceStack p_142781_, Collection<ServerPlayer> p_142782_, Component p_142783_, String p_142784_, Function<Component, Packet<?>> p_142785_
     ) throws CommandSyntaxException {
-        for(ServerPlayer serverplayer : p_142782_) {
+        for (ServerPlayer serverplayer : p_142782_) {
             serverplayer.connection.send(p_142785_.apply(ComponentUtils.updateForEntity(p_142781_, p_142783_, serverplayer, 0)));
         }
 
@@ -158,7 +_,7 @@
     private static int setTimes(CommandSourceStack p_139112_, Collection<ServerPlayer> p_139113_, int p_139114_, int p_139115_, int p_139116_) {
         ClientboundSetTitlesAnimationPacket clientboundsettitlesanimationpacket = new ClientboundSetTitlesAnimationPacket(p_139114_, p_139115_, p_139116_);
 
-        for(ServerPlayer serverplayer : p_139113_) {
+        for (ServerPlayer serverplayer : p_139113_) {
             serverplayer.connection.send(clientboundsettitlesanimationpacket);
         }
 
