--- a/net/minecraft/server/commands/TransferCommand.java
+++ b/net/minecraft/server/commands/TransferCommand.java
@@ -64,7 +_,7 @@
         if (p_321481_.isEmpty()) {
             throw ERROR_NO_PLAYERS.create();
         } else {
-            for(ServerPlayer serverplayer : p_321481_) {
+            for (ServerPlayer serverplayer : p_321481_) {
                 serverplayer.connection.send(new ClientboundTransferPacket(p_321828_, p_321760_));
             }
 
