--- a/net/minecraft/server/commands/TriggerCommand.java
+++ b/net/minecraft/server/commands/TriggerCommand.java
@@ -80,7 +_,7 @@
         if (scoreholder != null) {
             Scoreboard scoreboard = p_139149_.getServer().getScoreboard();
 
-            for(Objective objective : scoreboard.getObjectives()) {
+            for (Objective objective : scoreboard.getObjectives()) {
                 if (objective.getCriteria() == ObjectiveCriteria.TRIGGER) {
                     ReadOnlyScoreInfo readonlyscoreinfo = scoreboard.getPlayerScoreInfo(scoreholder, objective);
                     if (readonlyscoreinfo != null && !readonlyscoreinfo.isLocked()) {
