--- a/net/minecraft/server/commands/WardenSpawnTrackerCommand.java
+++ b/net/minecraft/server/commands/WardenSpawnTrackerCommand.java
@@ -37,7 +_,7 @@
     }
 
     private static int setWarningLevel(CommandSourceStack p_214783_, Collection<? extends Player> p_214784_, int p_214785_) {
-        for(Player player : p_214784_) {
+        for (Player player : p_214784_) {
             player.getWardenSpawnTracker().ifPresent(p_248188_ -> p_248188_.setWarningLevel(p_214785_));
         }
 
@@ -53,7 +_,7 @@
     }
 
     private static int resetTracker(CommandSourceStack p_214780_, Collection<? extends Player> p_214781_) {
-        for(Player player : p_214781_) {
+        for (Player player : p_214781_) {
             player.getWardenSpawnTracker().ifPresent(WardenSpawnTracker::reset);
         }
 
