--- a/net/minecraft/world/level/block/entity/trialspawner/PlayerDetector.java
+++ b/net/minecraft/world/level/block/entity/trialspawner/PlayerDetector.java
@@ -12,28 +_,47 @@
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.animal.Sheep;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.ClipContext;
+import net.minecraft.world.level.Level;
 import net.minecraft.world.level.entity.EntityTypeTest;
 import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.HitResult;
+import net.minecraft.world.phys.Vec3;
+import net.minecraft.world.phys.shapes.CollisionContext;
 
 public interface PlayerDetector {
-    PlayerDetector NO_CREATIVE_PLAYERS = (p_323427_, p_323428_, p_323429_, p_323430_) -> p_323428_.getPlayers(
-                p_323427_, p_340263_ -> p_340263_.blockPosition().closerThan(p_323429_, p_323430_) && !p_340263_.isCreative() && !p_340263_.isSpectator()
-            )
-            .stream()
-            .map(Entity::getUUID)
-            .toList();
-    PlayerDetector INCLUDING_CREATIVE_PLAYERS = (p_323431_, p_323432_, p_323433_, p_323434_) -> p_323432_.getPlayers(
-                p_323431_, p_340266_ -> p_340266_.blockPosition().closerThan(p_323433_, p_323434_) && !p_340266_.isSpectator()
-            )
-            .stream()
-            .map(Entity::getUUID)
-            .toList();
-    PlayerDetector SHEEP = (p_323420_, p_323421_, p_323422_, p_323423_) -> {
-        AABB aabb = new AABB(p_323422_).inflate(p_323423_);
-        return p_323421_.<Sheep>getEntities(p_323420_, EntityType.SHEEP, aabb, LivingEntity::isAlive).stream().map(Entity::getUUID).toList();
+    PlayerDetector NO_CREATIVE_PLAYERS = (p_338030_, p_338031_, p_338032_, p_338033_, p_338034_) -> p_338031_.getPlayers(
+                p_338030_, p_338022_ -> p_338022_.blockPosition().closerThan(p_338032_, p_338033_) && !p_338022_.isCreative() && !p_338022_.isSpectator()
+            )
+            .stream()
+            .filter(p_338010_ -> !p_338034_ || inLineOfSight(p_338030_, p_338032_.getCenter(), p_338010_.getEyePosition()))
+            .map(Entity::getUUID)
+            .toList();
+    PlayerDetector INCLUDING_CREATIVE_PLAYERS = (p_338015_, p_338016_, p_338017_, p_338018_, p_338019_) -> p_338016_.getPlayers(
+                p_338015_, p_338025_ -> p_338025_.blockPosition().closerThan(p_338017_, p_338018_) && !p_338025_.isSpectator()
+            )
+            .stream()
+            .filter(p_338029_ -> !p_338019_ || inLineOfSight(p_338015_, p_338017_.getCenter(), p_338029_.getEyePosition()))
+            .map(Entity::getUUID)
+            .toList();
+    PlayerDetector SHEEP = (p_338002_, p_338003_, p_338004_, p_338005_, p_338006_) -> {
+        AABB aabb = new AABB(p_338004_).inflate(p_338005_);
+        return p_338003_.getEntities(p_338002_, EntityType.SHEEP, aabb, LivingEntity::isAlive)
+            .stream()
+            .filter(p_338014_ -> !p_338006_ || inLineOfSight(p_338002_, p_338004_.getCenter(), p_338014_.getEyePosition()))
+            .map(Entity::getUUID)
+            .toList();
     };
 
-    List<UUID> detect(ServerLevel p_312124_, PlayerDetector.EntitySelector p_323893_, BlockPos p_312149_, double p_324310_);
+    List<UUID> detect(ServerLevel p_312124_, PlayerDetector.EntitySelector p_323893_, BlockPos p_312149_, double p_324310_, boolean p_338443_);
+
+    private static boolean inLineOfSight(Level p_338760_, Vec3 p_338259_, Vec3 p_338810_) {
+        BlockHitResult blockhitresult = p_338760_.clip(
+            new ClipContext(p_338810_, p_338259_, ClipContext.Block.VISUAL, ClipContext.Fluid.NONE, CollisionContext.empty())
+        );
+        return blockhitresult.getBlockPos().equals(BlockPos.containing(p_338259_)) || blockhitresult.getType() == HitResult.Type.MISS;
+    }
 
     public interface EntitySelector {
         PlayerDetector.EntitySelector SELECT_FROM_LEVEL = new PlayerDetector.EntitySelector() {
