--- a/net/minecraft/core/particles/ColorParticleOption.java
+++ b/net/minecraft/core/particles/ColorParticleOption.java
@@ -3,6 +_,7 @@
 import com.mojang.brigadier.StringReader;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import io.netty.buffer.ByteBuf;
 import java.util.Locale;
 import net.minecraft.core.HolderLookup;
@@ -31,8 +_,8 @@
     private final ParticleType<? extends ColorParticleOption> type;
     private final int color;
 
-    public static Codec<ColorParticleOption> codec(ParticleType<ColorParticleOption> p_333777_) {
-        return Codec.INT.xmap(p_333828_ -> new ColorParticleOption(p_333777_, p_333828_), p_333908_ -> p_333908_.color);
+    public static MapCodec<ColorParticleOption> codec(ParticleType<ColorParticleOption> p_333777_) {
+        return Codec.INT.xmap(p_333828_ -> new ColorParticleOption(p_333777_, p_333828_), p_333908_ -> p_333908_.color).fieldOf("value");
     }
 
     public static StreamCodec<? super ByteBuf, ColorParticleOption> streamCodec(ParticleType<ColorParticleOption> p_333948_) {
@@ -45,7 +_,7 @@
     }
 
     @Override
-    public ParticleType<?> getType() {
+    public ParticleType<? extends ColorParticleOption> getType() {
         return this.type;
     }
 
