--- a/net/minecraft/core/particles/DustParticleOptions.java
+++ b/net/minecraft/core/particles/DustParticleOptions.java
@@ -3,6 +_,7 @@
 import com.mojang.brigadier.StringReader;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.core.HolderLookup;
@@ -16,7 +_,7 @@
 public class DustParticleOptions extends DustParticleOptionsBase {
     public static final Vector3f REDSTONE_PARTICLE_COLOR = Vec3.fromRGB24(16711680).toVector3f();
     public static final DustParticleOptions REDSTONE = new DustParticleOptions(REDSTONE_PARTICLE_COLOR, 1.0F);
-    public static final Codec<DustParticleOptions> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<DustParticleOptions> CODEC = RecordCodecBuilder.mapCodec(
         p_253370_ -> p_253370_.group(
                     ExtraCodecs.VECTOR3F.fieldOf("color").forGetter(p_253371_ -> p_253371_.color),
                     Codec.FLOAT.fieldOf("scale").forGetter(p_175795_ -> p_175795_.scale)
