--- a/net/minecraft/core/particles/ItemParticleOption.java
+++ b/net/minecraft/core/particles/ItemParticleOption.java
@@ -2,7 +_,7 @@
 
 import com.mojang.brigadier.StringReader;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.commands.arguments.item.ItemInput;
 import net.minecraft.commands.arguments.item.ItemParser;
 import net.minecraft.core.HolderLookup;
@@ -23,8 +_,8 @@
     private final ParticleType<ItemParticleOption> type;
     private final ItemStack itemStack;
 
-    public static Codec<ItemParticleOption> codec(ParticleType<ItemParticleOption> p_123711_) {
-        return ItemStack.CODEC.xmap(p_123714_ -> new ItemParticleOption(p_123711_, p_123714_), p_123709_ -> p_123709_.itemStack);
+    public static MapCodec<ItemParticleOption> codec(ParticleType<ItemParticleOption> p_123711_) {
+        return ItemStack.CODEC.xmap(p_123714_ -> new ItemParticleOption(p_123711_, p_123714_), p_123709_ -> p_123709_.itemStack).fieldOf("value");
     }
 
     public static StreamCodec<? super RegistryFriendlyByteBuf, ItemParticleOption> streamCodec(ParticleType<ItemParticleOption> p_320881_) {
@@ -33,7 +_,7 @@
 
     public ItemParticleOption(ParticleType<ItemParticleOption> p_123705_, ItemStack p_123706_) {
         this.type = p_123705_;
-        this.itemStack = p_123706_;
+        this.itemStack = p_123706_.copy(); //Forge: Fix stack updating after the fact causing particle changes.
     }
 
     @Override
