--- a/net/minecraft/core/particles/ParticleType.java
+++ b/net/minecraft/core/particles/ParticleType.java
@@ -1,6 +_,6 @@
 package net.minecraft.core.particles;
 
-import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.codec.StreamCodec;
 
@@ -21,7 +_,7 @@
         return this.deserializer;
     }
 
-    public abstract Codec<T> codec();
+    public abstract MapCodec<T> codec();
 
     public abstract StreamCodec<? super RegistryFriendlyByteBuf, T> streamCodec();
 }
