--- a/net/minecraft/core/particles/SculkChargeParticleOptions.java
+++ b/net/minecraft/core/particles/SculkChargeParticleOptions.java
@@ -3,6 +_,7 @@
 import com.mojang.brigadier.StringReader;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
 import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Locale;
@@ -13,7 +_,7 @@
 import net.minecraft.network.codec.StreamCodec;
 
 public record SculkChargeParticleOptions(float roll) implements ParticleOptions {
-    public static final Codec<SculkChargeParticleOptions> CODEC = RecordCodecBuilder.create(
+    public static final MapCodec<SculkChargeParticleOptions> CODEC = RecordCodecBuilder.mapCodec(
         p_235920_ -> p_235920_.group(Codec.FLOAT.fieldOf("roll").forGetter(p_235922_ -> p_235922_.roll)).apply(p_235920_, SculkChargeParticleOptions::new)
     );
     public static final StreamCodec<RegistryFriendlyByteBuf, SculkChargeParticleOptions> STREAM_CODEC = StreamCodec.composite(
